--- udisks-1.0.4-old/src/daemon.c	2011-08-25 20:27:33.000000000 +0200
+++ udisks-1.0.4-new/src/daemon.c	2020-04-01 18:25:46.245662698 +0200
@@ -45,6 +45,7 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <sys/utsname.h>
+#include <sys/sysmacros.h>
 #include <net/if_arp.h>
 #include <fcntl.h>
 #include <signal.h>
diff -Naur udisks-1.0.4-old/src/device.c udisks-1.0.4-new/src/device.c
--- udisks-1.0.4-old/src/device.c	2011-08-25 23:31:20.000000000 +0200
+++ udisks-1.0.4-new/src/device.c	2020-04-01 18:24:58.418998908 +0200
@@ -33,6 +33,7 @@
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <sys/resource.h>
+#include <sys/sysmacros.h>
 #include <fcntl.h>
 #include <pwd.h>
 #include <grp.h>
--- udisks-1.0.4-old/src/mount-monitor.c	2011-08-25 20:27:33.000000000 +0200
+++ udisks-1.0.4-new/src/mount-monitor.c	2020-04-01 18:23:25.142337872 +0200
@@ -29,6 +29,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <mntent.h>
 
 #include <glib.h>
--- udisks-1.0.4-old/src/probers/part-id.c	2011-08-25 20:27:33.000000000 +0200
+++ udisks-1.0.4-new/src/probers/part-id.c	2020-04-01 18:43:43.742264590 +0200
@@ -26,6 +26,7 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <fcntl.h>
 
 #include <libudev.h>
--- udisks-1.0.4-old/tools/udisks.c	2011-08-25 20:27:33.000000000 +0200
+++ udisks-1.0.4-new/tools/udisks.c	2020-04-01 18:46:12.445588968 +0200
@@ -31,6 +31,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
+#include <sys/sysmacros.h>
 #include <fcntl.h>
 #include <pwd.h>
 #include <grp.h>
diff -Naur udisks-1.0.4-old/tools/umount-udisks.c udisks-1.0.4-new/tools/umount-udisks.c
--- udisks-1.0.4-old/tools/umount-udisks.c	2011-08-25 20:27:33.000000000 +0200
+++ udisks-1.0.4-new/tools/umount-udisks.c	2020-04-01 18:49:29.055577129 +0200
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <fcntl.h>
 #include <pwd.h>
 #include <grp.h>
