diff -Naru xtrans-1.2.7-old/Xtranssock.c xtrans-1.2.7-new/Xtranssock.c
--- xtrans-1.2.7-old/Xtranssock.c	2012-03-22 20:04:35.000000000 -0700
+++ xtrans-1.2.7-new/Xtranssock.c	2012-03-22 22:30:10.000000000 -0700
@@ -2436,7 +2436,7 @@
 Xtransport	TRANS(SocketUNIXFuncs) = {
 	/* Socket Interface */
 	"unix",
-#if !defined(LOCALCONN) && !defined(HAVE_ABSTRACT_SOCKETS)
+#if !defined(LOCALCONN)
         TRANS_ALIAS,
 #else
 	0,
