diff --speed-large-files --no-dereference --minimal -Naur fontconfig-2.17.1/Makefile.am fontconfig-2.17.1/Makefile.am
--- fontconfig-2.17.1/Makefile.am	2025-07-02 06:46:34.000000000 +0200
+++ fontconfig-2.17.1/Makefile.am	2026-04-12 13:03:19.933961547 +0200
@@ -24,7 +24,7 @@
 SUBDIRS=fontconfig fc-case fc-lang src \
 	fc-cache fc-cat fc-conflist fc-list fc-match \
 	fc-pattern fc-query fc-scan fc-validate conf.d \
-	its po po-conf test
+	its test
 if ENABLE_DOCS
 SUBDIRS += doc
 endif
