diff -Naur VisualBoyAdvance-1.7.2-old/src/gb/GB.cpp VisualBoyAdvance-1.7.2-new/src/gb/GB.cpp
--- VisualBoyAdvance-1.7.2-old/src/gb/GB.cpp	2004-05-13 12:35:26.000000000 -0700
+++ VisualBoyAdvance-1.7.2-new/src/gb/GB.cpp	2009-06-23 10:06:41.000000000 -0700
@@ -567,7 +567,7 @@
 bool gbIsGameboyRom(char * file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gb") == 0)
diff -Naur VisualBoyAdvance-1.7.2-old/src/GBA.cpp VisualBoyAdvance-1.7.2-new/src/GBA.cpp
--- VisualBoyAdvance-1.7.2-old/src/GBA.cpp	2004-05-13 07:37:59.000000000 -0700
+++ VisualBoyAdvance-1.7.2-new/src/GBA.cpp	2009-06-23 10:04:09.000000000 -0700
@@ -1130,7 +1130,7 @@
 bool CPUIsZipFile(const char * file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".zip") == 0)
@@ -1145,7 +1145,7 @@
 {
   cpuIsMultiBoot = false;
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gba") == 0)
@@ -1169,7 +1169,7 @@
 bool CPUIsGBABios(const char * file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gba") == 0)
@@ -1189,7 +1189,7 @@
 bool CPUIsELF(const char *file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
     
     if(p != NULL) {
       if(_stricmp(p, ".elf") == 0)
diff -Naur VisualBoyAdvance-1.7.2-old/src/Util.cpp VisualBoyAdvance-1.7.2-new/src/Util.cpp
--- VisualBoyAdvance-1.7.2-old/src/Util.cpp	2004-05-20 10:42:37.000000000 -0700
+++ VisualBoyAdvance-1.7.2-new/src/Util.cpp	2009-06-23 10:05:44.000000000 -0700
@@ -478,7 +478,7 @@
 {
   cpuIsMultiBoot = false;
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gba") == 0)
@@ -502,7 +502,7 @@
 bool utilIsGBImage(const char * file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gb") == 0)
@@ -522,7 +522,7 @@
 bool utilIsZipFile(const char *file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".zip") == 0)
@@ -537,7 +537,7 @@
 bool utilIsRarFile(const char *file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".rar") == 0)
@@ -552,7 +552,7 @@
 bool utilIsGzipFile(const char *file)
 {
   if(strlen(file) > 3) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gz") == 0)
