diff --git a/platform/sunxi/sunxi.c b/platform/sunxi/sunxi.c
index c149859..9565a52 100644
--- a/platform/sunxi/sunxi.c
+++ b/platform/sunxi/sunxi.c
@@ -89,6 +89,38 @@ struct resource *mali_create_mp2_resources(unsigned long address,
 	return res;
 }
 
+struct resource *mali_create_mali450_mp4_resources_nopmu(unsigned long address,
+							int irq_gp, int irq_gpmmu,
+							int irq_pp,
+							int irq_pp0, int irq_ppmmu0,
+							int irq_pp1, int irq_ppmmu1,
+							int irq_pp2, int irq_ppmmu2,
+							int irq_pp3, int irq_ppmmu3,
+							int *len)
+{
+	struct resource target[] = {
+		MALI_GPU_RESOURCES_MALI450_MP4(address,
+					irq_gp, irq_gpmmu,
+					irq_pp0, irq_ppmmu0,
+					irq_pp1, irq_ppmmu1,
+					irq_pp2, irq_ppmmu2,
+					irq_pp3, irq_ppmmu3,
+					irq_pp
+			)
+	};
+	struct resource *res;
+
+	res = kzalloc(sizeof(target), GFP_KERNEL);
+	if (!res)
+		return NULL;
+
+	memcpy(res, target, sizeof(target));
+
+	*len = ARRAY_SIZE(target);
+
+	return res;
+}
+
 struct resource *mali_create_mali450_mp4_resources(unsigned long address,
 						   int irq_gp, int irq_gpmmu,
 						   int irq_pp,
@@ -280,7 +312,19 @@ int mali_platform_device_register(void)
 		goto err_free_mem_region;
 	}
 
-	if (of_device_is_compatible(np, "arm,mali-450") &&
+	if (of_device_is_compatible(np, "allwinner,sun50i-h5-mali") &&
+		(irq_pp >= 0) &&
+		(irq_pp2 >= 0) && (irq_ppmmu2 >= 0) &&
+		(irq_pp3 >= 0) && (irq_ppmmu3 >= 0))
+		mali_res = mali_create_mali450_mp4_resources_nopmu(res.start,
+								irq_gp, irq_gpmmu,
+								irq_pp,
+								irq_pp0, irq_ppmmu0,
+								irq_pp1, irq_ppmmu1,
+								irq_pp2, irq_ppmmu2,
+								irq_pp3, irq_ppmmu3,
+								&len);
+	else if (of_device_is_compatible(np, "arm,mali-450") &&
 	    (irq_pp >= 0) &&
 	    (irq_pp2 >= 0) && (irq_ppmmu2 >= 0) &&
 	    (irq_pp3 >= 0) && (irq_ppmmu3 >= 0))
