diff --git a/platform/sunxi/sunxi.c b/platform/sunxi/sunxi.c
index 9565a52..42925a4 100644
--- a/platform/sunxi/sunxi.c
+++ b/platform/sunxi/sunxi.c
@@ -151,11 +151,45 @@ struct resource *mali_create_mali450_mp4_resources(unsigned long address,
 	return res;
 }
 
+#ifdef CONFIG_ARCH_MESON
+struct resource *mali_create_mali450_mp3_resources(unsigned long address,
+						   int irq_gp, int irq_gpmmu,
+						   int irq_pp,
+						   int irq_pp0, int irq_ppmmu0,
+						   int irq_pp1, int irq_ppmmu1,
+						   int irq_pp2, int irq_ppmmu2,
+						   int *len)
+{
+	struct resource target[] = {
+		MALI_GPU_RESOURCES_MALI450_MP3_PMU(address,
+						   irq_gp, irq_gpmmu,
+						   irq_pp0, irq_ppmmu0,
+						   irq_pp1, irq_ppmmu1,
+						   irq_pp2, irq_ppmmu2, irq_pp)
+	};
+	struct resource *res;
+
+	res = kzalloc(sizeof(target), GFP_KERNEL);
+	if (!res)
+		return NULL;
+
+	memcpy(res, target, sizeof(target));
+
+	*len = ARRAY_SIZE(target);
+
+	return res;
+}
+#endif
+
 static const struct of_device_id mali_dt_ids[] = {
 	{ .compatible = "allwinner,sun4i-a10-mali" },
 	{ .compatible = "allwinner,sun7i-a20-mali" },
 	{ .compatible = "allwinner,sun8i-h3-mali" },
 	{ .compatible = "allwinner,sun50i-h5-mali" },
+#ifdef CONFIG_ARCH_MESON
+        { .compatible = "amlogic,meson-gxbb-mali" },
+        { .compatible = "amlogic,meson-gxl-mali" },
+#endif
 	{ /* sentinel */ },
 };
 MODULE_DEVICE_TABLE(of, mali_dt_ids);
@@ -324,6 +358,31 @@ int mali_platform_device_register(void)
 								irq_pp2, irq_ppmmu2,
 								irq_pp3, irq_ppmmu3,
 								&len);
+#ifdef CONFIG_ARCH_MESON
+	else if ((of_device_is_compatible(np, "amlogic,meson-gxbb-mali") ||
+			of_device_is_compatible(np, "amlogic,meson-gxl-mali")) &&
+		(irq_pp >= 0) &&
+		(irq_pp2 >= 0) && (irq_ppmmu2 >= 0)) {
+		static const struct mali_gpu_device_data mali_gpu_data_meson = {
+			.fb_start = 0x0,
+			.fb_size = 0xFFFFF000,
+			.shared_mem_size = 256 * 1024 * 1024,
+			.control_interval = 200, /* 1000ms */
+			.pmu_domain_config = {
+				0x1, 0x2, 0x4, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x2, 0x0
+			},
+			.pmu_switch_delay = 0xFFFF,
+		};
+		memcpy(&mali_gpu_data, &mali_gpu_data_meson, sizeof(mali_gpu_data));
+		mali_res = mali_create_mali450_mp3_resources(res.start,
+							irq_gp, irq_gpmmu,
+							irq_pp,
+							irq_pp0, irq_ppmmu0,
+							irq_pp1, irq_ppmmu1,
+							irq_pp2, irq_ppmmu2,
+							&len);
+	}
+#endif
 	else if (of_device_is_compatible(np, "arm,mali-450") &&
 	    (irq_pp >= 0) &&
 	    (irq_pp2 >= 0) && (irq_ppmmu2 >= 0) &&
