diff -Naur fstrcmp-0.7.D001-old/configure.ac fstrcmp-0.7.D001-new/configure.ac
--- fstrcmp-0.7.D001-old/configure.ac	2014-03-05 01:13:45.000000000 +0100
+++ fstrcmp-0.7.D001-new/configure.ac	2021-10-20 12:33:10.339779455 +0200
@@ -84,6 +84,12 @@
   exit 1
 fi
 
+dnl Initialize libtool
+LT_PREREQ([2.2])
+LT_INIT([disable-static win32-dll])
+dnl when using libtool 2.x create libtool early, because it's used in configure
+m4_ifdef([LT_OUTPUT], [LT_OUTPUT])
+
 AC_CHECK_PROGS(GROFF, groff roff)
 AC_CHECK_PROGS(SOELIM, gsoelim soelim)
 AC_CHECK_PROGS(REFER, refer grefer)
diff -Naur fstrcmp-0.7.D001-old/Makefile.in fstrcmp-0.7.D001-new/Makefile.in
--- fstrcmp-0.7.D001-old/Makefile.in	2014-03-05 01:13:45.000000000 +0100
+++ fstrcmp-0.7.D001-new/Makefile.in	2021-10-20 12:43:04.123102061 +0200
@@ -146,6 +146,8 @@
 #
 SOELIM = @SOELIM@
 
+top_builddir = @top_builddir@
+
 # ---------------------------------------------------------
 # You should not need to change anything below this line.
 
