diff -Naur glibc-2.12.2-old/sunrpc/Makefile glibc-2.12.2-new/sunrpc/Makefile
--- glibc-2.12.2-old/sunrpc/Makefile	2010-12-13 02:47:26.000000000 -0800
+++ glibc-2.12.2-new/sunrpc/Makefile	2011-07-15 12:45:12.000000000 -0700
@@ -134,7 +134,7 @@
 $(objpfx)rpcgen: $(addprefix $(objpfx),$(rpcgen-objs)) \
   $(sort $(filter $(common-objpfx)libc%,$(link-libc))) \
   $(addprefix $(csu-objpfx),start.o) $(+preinit) $(+postinit)
-	$(+link)
+	$(+link-static)
 
 # This makes sure -DNOT_IN_libc is passed for all these modules.
 cpp-srcs-left := $(rpcgen-objs:.o=.c)
@@ -142,7 +142,7 @@
 include $(patsubst %,$(..)cppflags-iterator.mk,$(cpp-srcs-left))
 
 # Tell rpcgen where to find the C preprocessor.
-rpcgen-cmd = CPP='$(CC) -E -x c-header' $(built-program-cmd) -Y ../scripts
+rpcgen-cmd = CPP='$(CC) -E -x c-header' $(built-program-file) -Y ../scripts
 
 # Install the rpc data base file.
 $(inst_sysconfdir)/rpc: etc.rpc $(+force)
