diff -Naur glibc-2.15-old/sunrpc/Makefile glibc-2.15-new/sunrpc/Makefile
--- glibc-2.15-old/sunrpc/Makefile	2012-03-22 08:46:32.000000000 -0700
+++ glibc-2.15-new/sunrpc/Makefile	2012-03-22 08:46:55.000000000 -0700
@@ -159,7 +159,7 @@
 $(objpfx)rpcgen: $(addprefix $(objpfx),$(rpcgen-objs)) \
   $(sort $(filter $(common-objpfx)libc%,$(link-libc))) \
   $(addprefix $(csu-objpfx),start.o) $(+preinit) $(+postinit)
-	$(+link)
+	$(+link-static)
 
 # This makes sure -DNOT_IN_libc is passed for all these modules.
 cpp-srcs-left := $(rpcgen-objs:.o=.c)
@@ -167,7 +167,7 @@
 include $(patsubst %,$(..)cppflags-iterator.mk,$(cpp-srcs-left))
 
 # Tell rpcgen where to find the C preprocessor.
-rpcgen-cmd = CPP='$(CC) -E -x c-header' $(built-program-cmd) -Y ../scripts
+rpcgen-cmd = CPP='$(CC) -E -x c-header' $(built-program-file) -Y ../scripts
 
 # Install the rpc data base file.
 $(inst_sysconfdir)/rpc: etc.rpc $(+force)
