diff -Naur rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-old/os_dep/linux/recv_linux.c rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-new/os_dep/linux/recv_linux.c
--- rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-old/os_dep/linux/recv_linux.c	2021-01-13 20:39:20.000000000 +0100
+++ rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-new/os_dep/linux/recv_linux.c	2021-03-20 17:19:17.883329438 +0100
@@ -355,8 +355,12 @@
 
 #ifdef CONFIG_RTW_GRO
 		if (pregistrypriv->en_gro) {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 12, 0)
 			if (rtw_napi_gro_receive(&padapter->napi, pskb) != GRO_DROP)
 				rx_ok = _TRUE;
+#else
+			rx_ok = _TRUE;
+#endif
 			goto next;
 		}
 #endif /* CONFIG_RTW_GRO */
