diff --speed-large-files --no-dereference --minimal -Naur pvr.mythtv-c525168fb2980512f831e41b9a3774e7fb4eeae9/src/pvrclient-mythtv.h pvr.mythtv-c525168fb2980512f831e41b9a3774e7fb4eeae9/src/pvrclient-mythtv.h
--- pvr.mythtv-c525168fb2980512f831e41b9a3774e7fb4eeae9/src/pvrclient-mythtv.h	2024-08-04 09:19:23.000000000 +0200
+++ pvr.mythtv-c525168fb2980512f831e41b9a3774e7fb4eeae9/src/pvrclient-mythtv.h	2024-11-09 10:08:32.138345496 +0100
@@ -122,11 +122,11 @@
   bool SeekTime(double time, bool backwards, double& startpts) override;
 
   // Recording playback
-  bool OpenRecordedStream(const kodi::addon::PVRRecording& recinfo) override;
-  void CloseRecordedStream() override;
-  int ReadRecordedStream(unsigned char *pBuffer, unsigned int iBufferSize) override;
-  int64_t SeekRecordedStream(int64_t iPosition, int iWhence) override;
-  int64_t LengthRecordedStream() override;
+  bool OpenRecordedStream(const kodi::addon::PVRRecording& recinfo);
+  void CloseRecordedStream();
+  int ReadRecordedStream(unsigned char *pBuffer, unsigned int iBufferSize);
+  int64_t SeekRecordedStream(int64_t iPosition, int iWhence);
+  int64_t LengthRecordedStream();
 
   // Menu hook
   PVR_ERROR CallChannelMenuHook(const kodi::addon::PVRMenuhook& menuhook, const kodi::addon::PVRChannel& item) override;
