diff --speed-large-files --no-dereference --minimal -Naur u-boot-6d910b7f12318e5a5bb8d1b2093fe5a9ba17dfce/arch/arm/mach-rockchip/resource_img.c u-boot-6d910b7f12318e5a5bb8d1b2093fe5a9ba17dfce/arch/arm/mach-rockchip/resource_img.c
--- u-boot-6d910b7f12318e5a5bb8d1b2093fe5a9ba17dfce/arch/arm/mach-rockchip/resource_img.c	2019-11-11 20:23:15.000000000 +0100
+++ u-boot-6d910b7f12318e5a5bb8d1b2093fe5a9ba17dfce/arch/arm/mach-rockchip/resource_img.c	2022-10-31 16:42:10.153230445 +0100
@@ -113,8 +113,8 @@
 
 	ret = memcmp(RESOURCE_MAGIC, hdr->magic, RESOURCE_MAGIC_SIZE);
 	if (ret) {
-		printf("bad resource image magic: %s\n",
-		       hdr->magic ? hdr->magic : "none");
+//		printf("bad resource image magic: %s\n",
+//		       hdr->magic ? hdr->magic : "none");
 		ret = -EINVAL;
 	}
 	debug("resource image header:\n");
diff --speed-large-files --no-dereference --minimal -Naur u-boot-6d910b7f12318e5a5bb8d1b2093fe5a9ba17dfce/Makefile u-boot-6d910b7f12318e5a5bb8d1b2093fe5a9ba17dfce/Makefile
--- u-boot-6d910b7f12318e5a5bb8d1b2093fe5a9ba17dfce/Makefile	2019-11-11 20:23:15.000000000 +0100
+++ u-boot-6d910b7f12318e5a5bb8d1b2093fe5a9ba17dfce/Makefile	2022-10-31 16:58:32.866546095 +0100
@@ -605,6 +605,7 @@
 endif
 
 KBUILD_CFLAGS += $(call cc-option,-Wno-format-nonliteral)
+KBUILD_CFLAGS += $(call cc-disable-warning, address-of-packed-member)
 
 # turn jbsr into jsr for m68k
 ifeq ($(ARCH),m68k)
