diff --speed-large-files --no-dereference --minimal -Naur u-boot-f18abd08ab7b374d3c9e3865cfa193d6113463b5/arch/arm/dts/Makefile u-boot-f18abd08ab7b374d3c9e3865cfa193d6113463b5/arch/arm/dts/Makefile
--- u-boot-f18abd08ab7b374d3c9e3865cfa193d6113463b5/arch/arm/dts/Makefile	2021-05-21 11:11:58.000000000 +0200
+++ u-boot-f18abd08ab7b374d3c9e3865cfa193d6113463b5/arch/arm/dts/Makefile	2022-10-31 20:35:38.379644486 +0100
@@ -28,29 +28,6 @@
 	exynos5422-odroidxu3.dtb
 dtb-$(CONFIG_EXYNOS7420) += exynos7420-espresso7420.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += \
-	rk3036-sdk.dtb \
-	rk3066a-mk808.dtb \
-	rk3126-evb.dtb \
-	rk3128-evb.dtb \
-	rk3188-radxarock.dtb \
-	rk3288-evb.dtb \
-	rk3288-fennec.dtb \
-	rk3288-firefly.dtb \
-	rk3288-miqi.dtb \
-	rk3288-phycore-rdk.dtb \
-	rk3288-popmetal.dtb \
-	rk3288-rock2-square.dtb \
-	rk3288-tinker.dtb \
-	rk3288-veyron-jerry.dtb \
-	rk3288-veyron-mickey.dtb \
-	rk3288-veyron-minnie.dtb \
-	rk3288-vyasa.dtb \
-	rk3308-evb.dtb \
-	rk3328-evb.dtb \
-	rk3368-lion.dtb \
-	rk3368-sheep.dtb \
-	rk3368-geekbox.dtb \
-	rk3368-px5-evb.dtb \
 	rk3399-evb.dtb \
 	rk3399-firefly.dtb \
 	rk3399-puma-ddr1333.dtb \
@@ -62,8 +39,7 @@
 	rk3399-rock960-model-c.dtb \
 	rk3399pro-ficus2-linux.dtb \
 	rk3399pro-rockpi-n10-linux.dtb \
-	rk3399-fuhai.dtb \
-	rv1108-evb.dtb
+	rk3399-fuhai.dtb
 dtb-$(CONFIG_ARCH_MESON) += \
 	meson-gxbb-odroidc2.dtb
 dtb-$(CONFIG_TEGRA) += tegra20-harmony.dtb \
