diff --speed-large-files --no-dereference --minimal -Naur u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/command.c u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/command.c
--- u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/command.c	2023-04-13 05:19:49.000000000 +0200
+++ u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/command.c	2023-05-03 11:18:21.653229318 +0200
@@ -501,10 +501,10 @@
 	return result;
 }
 
-enum command_ret_t cmd_process(int flag, int argc, char * const argv[],
+int cmd_process(int flag, int argc, char * const argv[],
 			       int *repeatable, ulong *ticks)
 {
-	enum command_ret_t rc = CMD_RET_SUCCESS;
+	int rc = CMD_RET_SUCCESS;
 	cmd_tbl_t *cmdtp;
 
 	/* Look up command in command table */
diff --speed-large-files --no-dereference --minimal -Naur u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/edid.c u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/edid.c
--- u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/edid.c	2023-04-13 05:19:49.000000000 +0200
+++ u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/edid.c	2023-05-03 11:21:13.103224246 +0200
@@ -3579,7 +3579,7 @@
 {
 	const u8 *cea = drm_find_cea_extension(edid);
 	const u8 *db, *hdmi = NULL, *video = NULL;
-	u8 dbl, hdmi_len, video_len = 0;
+	u8 dbl, hdmi_len = 0, video_len = 0;
 	int modes = 0;
 
 	if (cea && cea_revision(cea) >= 3) {
diff --speed-large-files --no-dereference --minimal -Naur u-boot-f18abd08ab7b374d3c9e3865cfa193d6113463b5/lib/avb/libavb/avb_slot_verify.c u-boot-f18abd08ab7b374d3c9e3865cfa193d6113463b5/lib/avb/libavb/avb_slot_verify.c
--- u-boot-f18abd08ab7b374d3c9e3865cfa193d6113463b5/lib/avb/libavb/avb_slot_verify.c	2021-05-21 11:11:58.000000000 +0200
+++ u-boot-f18abd08ab7b374d3c9e3865cfa193d6113463b5/lib/avb/libavb/avb_slot_verify.c	2023-05-03 12:50:43.286398516 +0200
@@ -192,7 +192,7 @@
   bool image_preloaded = false;
   uint8_t* digest;
   size_t digest_len;
-  const char* found;
+  const char* found = 0;
   uint64_t image_size = 0;
   size_t expected_digest_len = 0;
   uint8_t expected_digest_buf[AVB_SHA512_DIGEST_SIZE];
