diff --speed-large-files --no-dereference --minimal -Naur mariadb-12.3.2/plugin/auth_ed25519/CMakeLists.txt mariadb-12.3.2/plugin/auth_ed25519/CMakeLists.txt
--- mariadb-12.3.2/plugin/auth_ed25519/CMakeLists.txt	2026-05-26 21:21:07.000000000 +0200
+++ mariadb-12.3.2/plugin/auth_ed25519/CMakeLists.txt	2026-06-14 17:06:01.055328220 +0200
@@ -14,6 +14,7 @@
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
 
 ADD_CONVENIENCE_LIBRARY(ref10 ${REF10_SOURCES})
+TARGET_INCLUDE_DIRECTORIES(ref10 PRIVATE ${CMAKE_SOURCE_DIR}/include)
 IF(MSVC)
    # Silence conversion (integer truncantion) warnings from reference code
   SET_SOURCE_FILES_PROPERTIES(${REF10_SOURCES} PROPERTY COMPILE_FLAGS "/wd4244 /wd4146")
