diff --speed-large-files --no-dereference --minimal -Naur mariadb-10.8.2/CMakeLists.txt mariadb-10.8.2/CMakeLists.txt
--- mariadb-10.8.2/CMakeLists.txt	2022-02-10 21:43:37.000000000 +0100
+++ mariadb-10.8.2/CMakeLists.txt	2022-03-18 08:33:29.470222174 +0100
@@ -376,6 +376,7 @@
   ADD_DEFINITIONS(-D_FILE_OFFSET_BITS=${_FILE_OFFSET_BITS})
 ENDIF()
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR}/include
+                    ${CMAKE_CURRENT_SOURCE_DIR}/include
                     ${CMAKE_SOURCE_DIR}/include/providers)
 
 # Add bundled or system zlib.
