diff --speed-large-files --no-dereference --minimal -Naur mariadb-11.7.2/storage/mroonga/CMakeLists.txt mariadb-11.7.2/storage/mroonga/CMakeLists.txt
--- mariadb-11.7.2/storage/mroonga/CMakeLists.txt	2025-02-11 20:05:58.000000000 +0100
+++ mariadb-11.7.2/storage/mroonga/CMakeLists.txt	2025-05-08 14:31:50.714371672 +0200
@@ -17,7 +17,7 @@
 # License along with this library; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1335  USA
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 2.8.12...4.0)
 project(mroonga)
 
 if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
diff --speed-large-files --no-dereference --minimal -Naur mariadb-11.7.2/storage/mroonga/vendor/groonga/CMakeLists.txt mariadb-11.7.2/storage/mroonga/vendor/groonga/CMakeLists.txt
--- mariadb-11.7.2/storage/mroonga/vendor/groonga/CMakeLists.txt	2025-02-11 20:05:58.000000000 +0100
+++ mariadb-11.7.2/storage/mroonga/vendor/groonga/CMakeLists.txt	2025-05-08 14:31:27.291038829 +0200
@@ -15,7 +15,7 @@
 
 # https://buildbot.askmonty.org/buildbot/builders/work-amd64-valgrind/builds/5263/steps/compile/logs/stdio
 # says CMake 2.6.2... We want to drop old software support...
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 2.8.12...4.0)
 # cmake_minimum_required(VERSION 2.6.4) # CentOS 5
 set(GRN_PROJECT_NAME "groonga")
 set(GRN_PROJECT_LABEL "Groonga")
diff --speed-large-files --no-dereference --minimal -Naur mariadb-11.7.2/storage/mroonga/vendor/groonga/vendor/plugins/groonga-normalizer-mysql/CMakeLists.txt mariadb-11.7.2/storage/mroonga/vendor/groonga/vendor/plugins/groonga-normalizer-mysql/CMakeLists.txt
--- mariadb-11.7.2/storage/mroonga/vendor/groonga/vendor/plugins/groonga-normalizer-mysql/CMakeLists.txt	2025-02-11 20:05:58.000000000 +0100
+++ mariadb-11.7.2/storage/mroonga/vendor/groonga/vendor/plugins/groonga-normalizer-mysql/CMakeLists.txt	2025-05-08 14:31:27.291038829 +0200
@@ -15,7 +15,7 @@
 # Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston,
 # MA 02110-1335  USA
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 2.8.12...4.0)
 if(NOT DEFINED GROONGA_NORMALIZER_MYSQL_PROJECT_NAME)
     set(GROONGA_NORMALIZER_MYSQL_PROJECT_NAME "groonga-normalizer-mysql")
 endif()
diff --speed-large-files --no-dereference --minimal -Naur mariadb-11.7.2/wsrep-lib/CMakeLists.txt mariadb-11.7.2/wsrep-lib/CMakeLists.txt
--- mariadb-11.7.2/wsrep-lib/CMakeLists.txt	2025-02-11 20:06:02.000000000 +0100
+++ mariadb-11.7.2/wsrep-lib/CMakeLists.txt	2025-05-08 14:31:27.291038829 +0200
@@ -2,7 +2,7 @@
 # Copyright (C) 2021 Codership Oy <info@codership.com>
 #
 
-cmake_minimum_required (VERSION 2.8)
+cmake_minimum_required (VERSION 2.8...4.0)
 
 # Parse version from version header file and store it into
 # WSREP_LIB_VERSION.
