diff -Naur mysql-8.0.26-old/cmake/libevent.cmake mysql-8.0.26-new/cmake/libevent.cmake
--- mysql-8.0.26-old/cmake/libevent.cmake	2021-07-01 09:53:49.000000000 +0200
+++ mysql-8.0.26-new/cmake/libevent.cmake	2021-09-19 22:07:15.922732623 +0200
@@ -128,12 +128,13 @@
   ELSE()
     MESSAGE(FATAL_ERROR "WITH_LIBEVENT must be bundled or system")
   ENDIF()
-  FIND_LIBEVENT_VERSION()
-  IF(LIBEVENT_VERSION VERSION_LESS MIN_LIBEVENT_VERSION_REQUIRED)
-    MESSAGE(FATAL_ERROR
-      "LIBEVENT version must be at least ${MIN_LIBEVENT_VERSION_REQUIRED}, "
-      "found ${LIBEVENT_VERSION}.\nPlease use -DWITH_LIBEVENT=bundled")
-  ENDIF()
+#  FIND_LIBEVENT_VERSION()
+#  IF(LIBEVENT_VERSION VERSION_LESS MIN_LIBEVENT_VERSION_REQUIRED)
+#    MESSAGE(FATAL_ERROR
+#      "LIBEVENT version must be at least ${MIN_LIBEVENT_VERSION_REQUIRED}, "
+#      "found ${LIBEVENT_VERSION}.\nPlease use -DWITH_LIBEVENT=bundled")
+#  ENDIF()
   SET(HAVE_LIBEVENT2 1)
   SET(HAVE_LIBEVENT2 1 CACHE BOOL "")
+  SET(LIBEVENT_VERSION "2.1")
 ENDMACRO()
