diff -Naur libX11-1.6.3-old/configure.ac libX11-1.6.3-new/configure.ac
--- libX11-1.6.3-old/configure.ac	2015-03-09 23:28:45.000000000 +0100
+++ libX11-1.6.3-new/configure.ac	2015-11-12 19:13:24.382209091 +0100
@@ -34,6 +34,7 @@
 
 # Required when PKG_CHECK_MODULES called within an if statement
 PKG_PROG_PKG_CONFIG
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
 
 if test x"$CC_FOR_BUILD" = x; then
 	if test x"$cross_compiling" = xyes; then
@@ -308,7 +309,7 @@
 AC_ARG_WITH([keysymdefdir],
         [AC_HELP_STRING([--with-keysymdefdir=DIR], [The location of keysymdef.h (defaults to xproto include dir)])],
         [KEYSYMDEFDIR=$withval],
-        [KEYSYMDEFDIR=`$PKG_CONFIG --variable=includedir xproto`/X11])
+        [KEYSYMDEFDIR=$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=includedir xproto`/X11])
 
 if test ! -d "$KEYSYMDEFDIR"; then
 	AC_MSG_ERROR([$KEYSYMDEFDIR doesn't exist or isn't a directory])
