diff -Naur jzintv-20181225-src-old/src/Makefile jzintv-20181225-src-new/src/Makefile
--- jzintv-20181225-src-old/src/Makefile	2018-12-25 09:00:00.000000000 +0100
+++ jzintv-20181225-src-new/src/Makefile	2020-04-05 11:16:08.361611207 +0200
@@ -56,8 +56,8 @@
  	   -Dlinux
 
 
- CC  = $(P) gcc-8.1.0 -std=c99
- CXX = $(P) g++-8.1.0 -std=c++14
+# CC  = $(P) gcc-8.1.0 -std=c99
+# CXX = $(P) g++-8.1.0 -std=c++14
 #CC  = $(P) gcc-3.4
 #CC  = $(P) /usr/local/bin/gcc -V4.1.1
 #CXX = $(P) /usr/local/bin/g++ 
@@ -69,15 +69,15 @@
 #DEF_FLAGS += -DNEED_INOUT
  DEF_FLAGS += -Dlinux
 
- LTO = -flto=4 -ffat-lto-objects
+# LTO = -flto=4 -ffat-lto-objects
 
 #OPT_FLAGS = -ggdb3
 #OPT_FLAGS = -O
 #OPT_FLAGS = -O2 -ggdb3 -pg
 #OPT_FLAGS =  -ggdb3 -O6 -fomit-frame-pointer -fprefetch-loop-arrays -msse #-DBENCHMARK_STIC
- OPT_FLAGS =  $(LTO) -O6 -fomit-frame-pointer -fprefetch-loop-arrays -msse2
- OPT_FLAGS += -fsplit-loops -funswitch-loops -fvisibility=hidden
- OPT_FLAGS += -fstrict-aliasing
+# OPT_FLAGS =  $(LTO) -O6 -fomit-frame-pointer -fprefetch-loop-arrays -msse2
+# OPT_FLAGS += -fsplit-loops -funswitch-loops -fvisibility=hidden
+# OPT_FLAGS += -fstrict-aliasing
 #OPT_FLAGS = -O6 -msse2 -pg -ggdb -DLOCAL=
 #OPT_FLAGS = -tpp6 -axMiKW -ip -vec_report3 -opt_report -ansi_alias -restrict -DHAVE_RESTRICT -align -O3 -Ob1 # -ipo # intel icc flags
 
