diff -Naur v4l-utils-20200507-316a273fa4-old/Makefile.am v4l-utils-20200507-316a273fa4-new/Makefile.am
--- v4l-utils-20200507-316a273fa4-old/Makefile.am	2020-05-07 20:48:53.000000000 +0200
+++ v4l-utils-20200507-316a273fa4-new/Makefile.am	2020-05-07 21:09:41.363333333 +0200
@@ -1,7 +1,7 @@
 AUTOMAKE_OPTIONS = foreign
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = v4l-utils-po libdvbv5-po lib
+SUBDIRS = lib
 
 if WITH_V4LUTILS
 SUBDIRS += utils contrib
