diff -Naur gtk+-2.24.28-old/gtk/Makefile.am gtk+-2.24.28-new/gtk/Makefile.am
--- gtk+-2.24.28-old/gtk/Makefile.am	2014-09-29 22:02:16.000000000 +0200
+++ gtk+-2.24.28-new/gtk/Makefile.am	2020-04-12 08:44:59.826666658 +0200
@@ -15,8 +15,8 @@
 GTK_PRINT_PREVIEW_COMMAND="evince --unlink-tempfile --preview --print-settings %s %f"
 endif
 
-SUBDIRS = theme-bits . tests
-DIST_SUBDIRS=theme-bits tests
+SUBDIRS = theme-bits . 
+DIST_SUBDIRS=theme-bits 
 
 if HAVE_PAPI_CUPS
 GTK_PRINT_BACKENDS=file,papi,cups
diff -Naur gtk+-2.24.28-old/Makefile.am gtk+-2.24.28-new/Makefile.am
--- gtk+-2.24.28-old/Makefile.am	2014-09-29 22:02:16.000000000 +0200
+++ gtk+-2.24.28-new/Makefile.am	2020-04-12 09:51:57.730000001 +0200
@@ -1,7 +1,7 @@
 ## Makefile.am for GTK+
 include $(top_srcdir)/Makefile.decl
 
-SRC_SUBDIRS = gdk gtk modules demos tests perf
+SRC_SUBDIRS = gdk gtk modules perf
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
 
 # require automake 1.4
diff -Naur gtk+-2.24.28-old/Makefile.in gtk+-2.24.28-new/Makefile.in
--- gtk+-2.24.28-old/Makefile.in	2015-05-13 13:26:30.000000000 +0200
+++ gtk+-2.24.28-new/Makefile.in	2020-04-12 09:40:39.719999996 +0200
@@ -563,7 +563,7 @@
 	|| { echo "Gtk+Tests:ERROR: Failed to start Xvfb environment for X11 target tests."; exit 1; } \
 	&& DISPLAY=:$$XID && export DISPLAY
 
-SRC_SUBDIRS = gdk gtk modules demos tests perf
+SRC_SUBDIRS = gdk gtk modules perf
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
 
 # require automake 1.4
