diff --git a/src/panfrost/midgard/midgard_ra.c b/src/panfrost/midgard/midgard_ra.c
index c2c1874aa82..f5162150193 100644
--- a/src/panfrost/midgard/midgard_ra.c
+++ b/src/panfrost/midgard/midgard_ra.c
@@ -446,7 +446,7 @@ allocate_registers(compiler_context *ctx, bool *spilled)
          * tex outputs use work r0/r1. Note we still use TEXR/TEXW classes,
          * noting that this handles interferences and sizes correctly. */
 
-        if (ctx->gpu_id == 0x0720) {
+        if (1) {
                 l->class_start[REG_CLASS_TEXR] = l->class_start[REG_CLASS_LDST];
                 l->class_start[REG_CLASS_TEXW] = l->class_start[REG_CLASS_WORK];
         }
