diff -Naur mesa-mesa-20.3.0-rc1-old/src/gallium/drivers/v3d/meson.build mesa-mesa-20.3.0-rc1-new/src/gallium/drivers/v3d/meson.build
--- mesa-mesa-20.3.0-rc1-old/src/gallium/drivers/v3d/meson.build	2020-11-10 01:30:12.000000000 +0100
+++ mesa-mesa-20.3.0-rc1-new/src/gallium/drivers/v3d/meson.build	2020-11-19 14:45:42.063333332 +0100
@@ -74,7 +74,7 @@
 
 v3d_neon_c_args = []
 if host_machine.cpu_family() == 'arm'
-    v3d_neon_c_args = '-mfpu=neon'
+    v3d_neon_c_args = ['-mfpu=neon', '-mfloat-abi=softfp']
 endif
 
 libv3d_neon = static_library(

diff -Naur mesa-mesa-20.3.0-rc1-old/src/gallium/drivers/vc4/meson.build mesa-mesa-20.3.0-rc1-new/src/gallium/drivers/vc4/meson.build
--- mesa-mesa-20.3.0-rc1-old/src/gallium/drivers/vc4/meson.build	2020-11-10 01:30:12.000000000 +0100
+++ mesa-mesa-20.3.0-rc1-new/src/gallium/drivers/vc4/meson.build	2020-11-19 14:45:42.063333332 +0100
@@ -91,7 +91,7 @@
     include_directories : [
       inc_src, inc_include, inc_gallium, inc_gallium_aux, inc_broadcom
     ],
-    c_args : '-mfpu=neon',
+    c_args : ['-mfpu=neon', '-mfloat-abi=softfp'],
   )
   vc4_c_args += '-DUSE_ARM_ASM'
 endif
diff -Naur mesa-mesa-20.3.0-rc1-old/src/gallium/winsys/lima/drm/meson.build mesa-mesa-20.3.0-rc1-new/src/gallium/winsys/lima/drm/meson.build
--- mesa-mesa-20.3.0-rc1-old/src/gallium/winsys/lima/drm/meson.build	2020-11-10 01:30:12.000000000 +0100
+++ mesa-mesa-20.3.0-rc1-new/src/gallium/winsys/lima/drm/meson.build	2020-11-19 14:38:59.909999995 +0100
@@ -19,12 +19,25 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 # IN THE SOFTWARE.
 
-liblimawinsys = static_library(
-  'limawinsys',
-  files('lima_drm_winsys.c'),
-  include_directories : [
-    inc_src, inc_include,
-    inc_gallium, inc_gallium_aux, inc_gallium_drivers,
-  ],
-  dependencies : idep_mesautil,
-)
+if host_machine.cpu_family() == 'arm'
+  liblimawinsys = static_library(
+    'limawinsys',
+    files('lima_drm_winsys.c'),
+    include_directories : [
+      inc_src, inc_include,
+      inc_gallium, inc_gallium_aux, inc_gallium_drivers,
+    ],
+    c_args : ['-mfpu=neon', '-mfloat-abi=softfp'],
+    dependencies : idep_mesautil,
+  )
+else
+  liblimawinsys = static_library(
+    'limawinsys',
+    files('lima_drm_winsys.c'),
+    include_directories : [
+      inc_src, inc_include,
+      inc_gallium, inc_gallium_aux, inc_gallium_drivers,
+    ],
+  dependencies : idep_mesautil,
+  )
+endif
