diff -Naur mesa-fcc7a1e7b6d6bc149938c1203a55e1ede3517af5-old/src/gallium/drivers/panfrost/pan_screen.c mesa-fcc7a1e7b6d6bc149938c1203a55e1ede3517af5-new/src/gallium/drivers/panfrost/pan_screen.c
--- mesa-fcc7a1e7b6d6bc149938c1203a55e1ede3517af5-old/src/gallium/drivers/panfrost/pan_screen.c	2020-08-18 14:12:27.000000000 +0200
+++ mesa-fcc7a1e7b6d6bc149938c1203a55e1ede3517af5-new/src/gallium/drivers/panfrost/pan_screen.c	2020-08-18 15:59:11.240000001 +0200
@@ -137,7 +137,7 @@
         case PIPE_CAP_SURFACE_SAMPLE_COUNT:
         case PIPE_CAP_PRIMITIVE_RESTART:
         case PIPE_CAP_PRIMITIVE_RESTART_FIXED_INDEX:
-                return !is_bifrost;
+                return 1;
 
         case PIPE_CAP_SAMPLER_VIEW_TARGET:
         case PIPE_CAP_TEXTURE_SWIZZLE:
@@ -157,22 +157,22 @@
                 return 1;
 
         case PIPE_CAP_MAX_STREAM_OUTPUT_BUFFERS:
-                return is_bifrost ? 0 : 4;
+                return is_bifrost ? 4 : 4;
         case PIPE_CAP_MAX_STREAM_OUTPUT_SEPARATE_COMPONENTS:
         case PIPE_CAP_MAX_STREAM_OUTPUT_INTERLEAVED_COMPONENTS:
-                return is_bifrost ? 0 : 64;
+                return is_bifrost ? 64 : 64;
         case PIPE_CAP_STREAM_OUTPUT_PAUSE_RESUME:
         case PIPE_CAP_STREAM_OUTPUT_INTERLEAVE_BUFFERS:
-                return is_bifrost ? 0 : 1;
+                return is_bifrost ? 1 : 1;
 
         case PIPE_CAP_MAX_TEXTURE_ARRAY_LAYERS:
-                return is_bifrost ? 0 : 256;
+                return is_bifrost ? 256 : 256;
 
         case PIPE_CAP_GLSL_FEATURE_LEVEL:
         case PIPE_CAP_GLSL_FEATURE_LEVEL_COMPATIBILITY:
-                return is_gl3 ? 330 : (is_bifrost ? 120 : 140);
+                return is_gl3 ? 330 : (is_bifrost ? 140 : 140);
         case PIPE_CAP_ESSL_FEATURE_LEVEL:
-                return is_bifrost ? 120 : 300;
+                return is_bifrost ? 300 : 300;
 
         case PIPE_CAP_CONSTANT_BUFFER_OFFSET_ALIGNMENT:
                 return 16;
@@ -198,7 +198,7 @@
         case PIPE_CAP_MAX_TEXTURE_2D_SIZE:
                 return 4096;
         case PIPE_CAP_MAX_TEXTURE_3D_LEVELS:
-                return is_bifrost ? 0 : 13;
+                return is_bifrost ? 13 : 13;
         case PIPE_CAP_MAX_TEXTURE_CUBE_LEVELS:
                 return 13;
 
@@ -220,7 +220,7 @@
 
         case PIPE_CAP_SEAMLESS_CUBE_MAP:
         case PIPE_CAP_SEAMLESS_CUBE_MAP_PER_TEXTURE:
-                return !is_bifrost;
+                return 1;
 
         case PIPE_CAP_MAX_VERTEX_ELEMENT_SRC_OFFSET:
                 return 0xffff;
@@ -249,7 +249,7 @@
         }
 
         case PIPE_CAP_SHADER_STENCIL_EXPORT:
-                return !is_bifrost;
+                return 1;
 
         case PIPE_CAP_SHADER_BUFFER_OFFSET_ALIGNMENT:
                 return 4;
@@ -320,7 +320,7 @@
                 return 0;
 
         case PIPE_SHADER_CAP_INDIRECT_INPUT_ADDR:
-                return is_bifrost ? 0 : 1;
+                return is_bifrost ? 1 : 1;
         case PIPE_SHADER_CAP_INDIRECT_OUTPUT_ADDR:
                 return 0;
 
@@ -328,7 +328,7 @@
                 return 0;
 
         case PIPE_SHADER_CAP_INDIRECT_CONST_ADDR:
-                return is_bifrost ? 0 : 1;
+                return is_bifrost ? 1 : 1;
 
         case PIPE_SHADER_CAP_SUBROUTINES:
                 return 0;
