diff --speed-large-files --no-dereference --minimal -Naur mesa-mesa-24.1.0/src/intel/compiler/brw_kernel.c mesa-mesa-24.1.0/src/intel/compiler/brw_kernel.c
--- mesa-mesa-24.1.0/src/intel/compiler/brw_kernel.c	2024-05-22 20:07:58.000000000 +0200
+++ mesa-mesa-24.1.0/src/intel/compiler/brw_kernel.c	2024-05-26 16:47:20.373183360 +0200
@@ -293,7 +293,7 @@
    if (spirv_options.clc_shader == NULL) {
       fprintf(stderr, "ERROR: libclc shader missing."
               " Consider installing the libclc package\n");
-      abort();
+      //abort();
    }
 
    assert(spirv_size % 4 == 0);
