diff -Naur mesa-87fa8d9ebc0dcd8c0ba1405d590ee6337fce69ba-old/src/gallium/drivers/panfrost/pan_screen.c mesa-87fa8d9ebc0dcd8c0ba1405d590ee6337fce69ba-new/src/gallium/drivers/panfrost/pan_screen.c
--- mesa-87fa8d9ebc0dcd8c0ba1405d590ee6337fce69ba-old/src/gallium/drivers/panfrost/pan_screen.c	2019-09-01 05:17:55.000000000 +0200
+++ mesa-87fa8d9ebc0dcd8c0ba1405d590ee6337fce69ba-new/src/gallium/drivers/panfrost/pan_screen.c	2019-09-03 11:48:55.913333301 +0200
@@ -630,6 +630,7 @@
         /* Check if we're loading against a supported GPU model. */
 
         switch (screen->gpu_id) {
+        case 0x720: /* T720 - unsupported yet*/
         case 0x750: /* T760 */
         case 0x820: /* T820 */
         case 0x860: /* T860 */
