diff -Naur mesa-6747a984f59ea9a2dd74b98d59cb8fdb028969ae-old/src/gallium/drivers/panfrost/pan_screen.c mesa-6747a984f59ea9a2dd74b98d59cb8fdb028969ae-new/src/gallium/drivers/panfrost/pan_screen.c
--- mesa-6747a984f59ea9a2dd74b98d59cb8fdb028969ae-old/src/gallium/drivers/panfrost/pan_screen.c	2020-04-28 10:06:33.000000000 +0200
+++ mesa-6747a984f59ea9a2dd74b98d59cb8fdb028969ae-new/src/gallium/drivers/panfrost/pan_screen.c	2020-04-28 17:34:33.726666643 +0200
@@ -750,6 +750,10 @@
         case 0x750: /* T760 */
         case 0x820: /* T820 */
         case 0x860: /* T860 */
+        case 0x7003: /* G31 */
+        case 0x7002: /* G52 */
+        case 0x7093: /* G31 */
+        case 0x7212: /* G52 */
                 break;
         default:
                 /* Fail to load against untested models */
