diff --speed-large-files --no-dereference --minimal -Naur u-boot-orangepi-6fe17fac388aad17490cf386578b7532975e567f/arch/arm/dts/sun50i-h616-orangepi-zero2w.dts u-boot-orangepi-6fe17fac388aad17490cf386578b7532975e567f/arch/arm/dts/sun50i-h616-orangepi-zero2w.dts
--- u-boot-orangepi-6fe17fac388aad17490cf386578b7532975e567f/arch/arm/dts/sun50i-h616-orangepi-zero2w.dts	2023-09-13 05:22:34.000000000 +0200
+++ u-boot-orangepi-6fe17fac388aad17490cf386578b7532975e567f/arch/arm/dts/sun50i-h616-orangepi-zero2w.dts	2023-10-28 21:58:25.786682578 +0200
@@ -6,37 +6,41 @@
 /dts-v1/;
 
 #include "sun50i-h616.dtsi"
+
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/leds/common.h>
 
 / {
-	model = "OrangePi Zero2 W";
+	model = "OrangePi Zero2W";
 	compatible = "xunlong,orangepi-zero2w", "allwinner,sun50i-h616";
 
 	aliases {
-		ethernet0 = &emac0;
+		ethernet0 = &emac1;
 		serial0 = &uart0;
+		mmc0 = &mmc0;  /* mmc0/mmcblk0 for SD */
+		mmc1 = &mmc1;  /* mmc1/mmcblk1 for SDIO */
 	};
 
 	chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
-	leds {
+	leds: leds {
 		compatible = "gpio-leds";
 
-		led-0 {
-			function = LED_FUNCTION_POWER;
-			color = <LED_COLOR_ID_RED>;
+		led-red {
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			label = "red_led";
 			gpios = <&pio 2 12 GPIO_ACTIVE_HIGH>; /* PC12 */
-			default-state = "on";
+			linux,default-trigger = "mmc0";
 		};
 
-		led-1 {
+		led-green {
 			function = LED_FUNCTION_STATUS;
-			color = <LED_COLOR_ID_GREEN>;
+			label = "green_led";
 			gpios = <&pio 2 13 GPIO_ACTIVE_HIGH>; /* PC13 */
+			linux,default-trigger = "heartbeat";
 		};
 	};
 
@@ -61,31 +65,34 @@
 	};
 };
 
-&ehci0 {
+&cpu0 {
+	cpu-supply = <&reg_dcdc2>;
 	status = "okay";
 };
 
-&ehci1 {
+&emac1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rmii_pins>;
+	phy-mode = "rmii";
+	phy-handle = <&rmii_phy>;
+	phy-supply = <&reg_dldo1>;
+	allwinner,rx-delay-ps = <3100>;
+	allwinner,tx-delay-ps = <700>;
 	status = "okay";
 };
 
-/* USB 2 & 3 are on headers only. */
-
-&emac0 {
-	status = "disabled";
+&mdio1 {
+	rmii_phy: ethernet-phy@16 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <16>;
+	};
 };
 
 &mmc0 {
+	vmmc-supply = <&reg_dldo1>;
 	cd-gpios = <&pio 5 6 GPIO_ACTIVE_LOW>;	/* PF6 */
 	bus-width = <4>;
-	status = "okay";
-};
-
-&ohci0 {
-	status = "okay";
-};
-
-&ohci1 {
+	max-frequency = <50000000>;
 	status = "okay";
 };
 
@@ -94,19 +101,14 @@
 
 	axp313a: pmic@36 {
 		compatible = "x-powers,axp313a";
-		status = "okay";
 		reg = <0x36>;
 		wakeup-source;
 
-		standby_param: standby_param {
-			vcc-dram = <0x4>;
-		};
-
 		regulators{
 			reg_dcdc1: dcdc1 {
 				regulator-name = "axp313a-dcdc1";
 				regulator-min-microvolt = <500000>;
-				regulator-max-microvolt = <3400000>;
+				regulator-max-microvolt = <1100000>;
 				regulator-step-delay-us = <25>;
 				regulator-final-delay-us = <50>;
 				regulator-always-on;
@@ -115,7 +117,7 @@
 			reg_dcdc2: dcdc2 {
 				regulator-name = "axp313a-dcdc2";
 				regulator-min-microvolt = <500000>;
-				regulator-max-microvolt = <1540000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-step-delay-us = <25>;
 				regulator-final-delay-us = <50>;
 				regulator-ramp-delay = <200>;
@@ -131,7 +133,7 @@
 				regulator-always-on;
 			};
 
-			reg_aldo1: aldo1 {
+			reg_aldo1: ldo1 {
 				regulator-name = "axp313a-aldo1";
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
@@ -140,7 +142,7 @@
 				regulator-always-on;
 			};
 
-			reg_dldo1: dldo1 {
+			reg_dldo1: ldo2 {
 				regulator-name = "axp313a-dldo1";
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
@@ -152,18 +154,6 @@
 	};
 };
 
-&spi0  {
-	status = "okay";
-
-	flash@0 {
-		#address-cells = <1>;
-		#size-cells = <1>;
-		compatible = "jedec,spi-nor";
-		reg = <0>;
-		spi-max-frequency = <40000000>;
-	};
-};
-
 &uart0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart0_ph_pins>;
@@ -171,6 +161,17 @@
 };
 
 &usbotg {
+	/*
+	 * PHY0 pins are connected to a USB-C socket, but a role switch
+	 * is not implemented: both CC pins are pulled to GND.
+	 * The VBUS pins power the device, so a fixed peripheral mode
+	 * is the best choice.
+	 * The board can be powered via GPIOs, in this case port0 *can*
+	 * act as a host (with a cable/adapter ignoring CC), as VBUS is
+	 * then provided by the GPIOs. Any user of this setup would
+	 * need to adjust the DT accordingly: dr_mode set to "host",
+	 * enabling OHCI0 and EHCI0.
+	 */
 	dr_mode = "peripheral";
 	status = "okay";
 };
@@ -179,3 +180,35 @@
 	usb1_vbus-supply = <&reg_usb1_vbus>;
 	status = "okay";
 };
+
+&ehci0 {
+	status = "disabled";
+};
+
+&ehci1 {
+	status = "okay";
+};
+
+&ehci2 {
+	status = "okay";
+};
+
+&ehci3 {
+	status = "okay";
+};
+
+&ohci0 {
+	status = "disabled";
+};
+
+&ohci1 {
+	status = "okay";
+};
+
+&ohci2 {
+	status = "okay";
+};
+
+&ohci3 {
+	status = "okay";
+};
diff --speed-large-files --no-dereference --minimal -Naur u-boot-orangepi-6fe17fac388aad17490cf386578b7532975e567f/configs/orangepi_zero2w_defconfig u-boot-orangepi-6fe17fac388aad17490cf386578b7532975e567f/configs/orangepi_zero2w_defconfig
--- u-boot-orangepi-6fe17fac388aad17490cf386578b7532975e567f/configs/orangepi_zero2w_defconfig	2023-09-13 05:22:34.000000000 +0200
+++ u-boot-orangepi-6fe17fac388aad17490cf386578b7532975e567f/configs/orangepi_zero2w_defconfig	2023-10-29 18:55:10.140003675 +0100
@@ -11,8 +11,19 @@
 CONFIG_MACH_SUN50I_H616=y
 CONFIG_MMC0_CD_PIN="PF6"
 CONFIG_R_I2C_ENABLE=y
-# CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
-CONFIG_AXP_DCDC3_VOLT=1100
-CONFIG_SPL_I2C_SUPPORT=y
-CONFIG_AXP313A_POWER=y
 CONFIG_SPL_SPI_SUNXI=y
+CONFIG_SYS_MONITOR_LEN=786432
+CONFIG_SPL_I2C=y
+CONFIG_SPL_I2C_SUPPORT=y
+CONFIG_SPL_SYS_I2C_LEGACY=y
+CONFIG_SYS_I2C_MVTWSI=y
+CONFIG_SYS_I2C_SLAVE=0x7f
+CONFIG_SYS_I2C_SPEED=400000
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_PHY_REALTEK=y
+CONFIG_SUN8I_EMAC=y
+CONFIG_SPI=y
+CONFIG_AXP313A_POWER=y
+CONFIG_I2C3_ENABLE=y
+CONFIG_AXP_DCDC3_VOLT=1150
+
