diff -Naur busybox-1.20.2-old/networking/tftp.c busybox-1.20.2-new/networking/tftp.c
--- busybox-1.20.2-old/networking/tftp.c	2012-07-02 21:51:24.000000000 -0700
+++ busybox-1.20.2-new/networking/tftp.c	2012-07-02 21:52:01.000000000 -0700
@@ -58,7 +58,7 @@
 #define TFTP_BLKSIZE_DEFAULT       512  /* according to RFC 1350, don't change */
 #define TFTP_BLKSIZE_DEFAULT_STR "512"
 /* Was 50 ms but users asked to bump it up a bit */
-#define TFTP_TIMEOUT_MS            100
+#define TFTP_TIMEOUT_MS            250
 #define TFTP_MAXTIMEOUT_MS        2000
 #define TFTP_NUM_RETRIES            12  /* number of backed-off retries */
 
