diff -Naur busybox-1.20.2-old/include/libbb.h busybox-1.20.2-new/include/libbb.h
--- busybox-1.20.2-old/include/libbb.h	2012-07-02 07:08:25.000000000 -0700
+++ busybox-1.20.2-new/include/libbb.h	2012-07-03 07:02:34.000000000 -0700
@@ -40,6 +40,7 @@
 #include <sys/poll.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
+#include <sys/resource.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <sys/time.h>
