diff -Naur busybox-1.20.2-old/scripts/trylink busybox-1.20.2-new/scripts/trylink
--- busybox-1.20.2-old/scripts/trylink	2012-06-26 06:35:45.000000000 -0700
+++ busybox-1.20.2-new/scripts/trylink	2012-07-02 21:50:25.000000000 -0700
@@ -49,7 +49,7 @@
     local tempname="/tmp/temp.$$.$RANDOM"
     # Can use "-o /dev/null", but older gcc tend to *unlink it* on failure! :(
     # "-xc": C language. "/dev/null" is an empty source file.
-    if $CC $1 -shared -xc /dev/null -o "$tempname".o >/dev/null 2>&1; then
+    if $CC $CFLAGS $LDFLAGS $1 -shared -xc /dev/null -o "$tempname".o >/dev/null 2>&1; then
 	echo "$1";
     else
 	echo "$2";
