diff -Naur mythtv-a2cdb9047a68d0d36c8be1493b68a88f6dfe6af0-old/mythplugins/configure mythtv-a2cdb9047a68d0d36c8be1493b68a88f6dfe6af0-new/mythplugins/configure
--- mythtv-a2cdb9047a68d0d36c8be1493b68a88f6dfe6af0-old/mythplugins/configure	2021-08-25 12:32:47.000000000 +0200
+++ mythtv-a2cdb9047a68d0d36c8be1493b68a88f6dfe6af0-new/mythplugins/configure	2021-09-24 10:58:38.306029567 +0200
@@ -296,6 +296,7 @@
 newexif
 dcraw
 cdio
+cross_compile
 "
 
 DEPEND_LIST="
@@ -337,6 +338,7 @@
   --enable-all             Enable all options
   --previous               use previous configure parameters if possible
   --extra-ldflags=ELDFLAGS add ELDFLAGS to LDFLAGS [$LDFLAGS]
+  --enable-cross-compile   assume a cross-compiler is used
 
 MythArchive related options:
   --enable-mytharchive     build the mytharchive plugin [$archive]
@@ -396,6 +398,7 @@
     python
     sysroot
     mythroot
+    cross_compile
 "
 
 CONFIGURATION_OPTS=""
@@ -465,7 +468,11 @@
 check_py_lib(){
     log check_py_lib "$@"
     lib=${1}
-    check_cmd $python <<EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Python:$lib Be sure You have it for cross target!"
+        return 0
+    else
+        check_cmd $python <<EOF
 from distutils.sysconfig import get_python_lib
 import sys
 for mythroot in '${mythroot}'.split(':'):
@@ -477,6 +484,7 @@
     sys.exit(1)
 sys.exit(0)
 EOF
+    fi
 }
 
 # set temporary file name
@@ -638,7 +646,12 @@
 if enabled weather; then
     check_pl_lib(){
         log check_pl_lib "$@"
-        check_cmd perl -e "use ${1};"
+        if enabled cross_compile ; then
+            echo "You are cross-compiling! Assuming YES for Perl:$@ Be sure You have it for cross target!"
+            return 0
+        else
+            check_cmd perl -e "use ${1};"
+        fi
     }
 
     disable_weather(){
diff -Naur mythtv-a2cdb9047a68d0d36c8be1493b68a88f6dfe6af0-old/mythtv/configure mythtv-a2cdb9047a68d0d36c8be1493b68a88f6dfe6af0-new/mythtv/configure
--- mythtv-a2cdb9047a68d0d36c8be1493b68a88f6dfe6af0-old/mythtv/configure	2021-09-24 11:06:25.009354480 +0200
+++ mythtv-a2cdb9047a68d0d36c8be1493b68a88f6dfe6af0-new/mythtv/configure	2021-09-24 11:07:38.436019819 +0200
@@ -1254,7 +1254,11 @@
 check_py_lib(){
     log check_py_lib "$@"
     lib=${1}
-    check_cmd $python <<EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Python:$lib Be sure You have it for cross target!"
+        return 0
+    else
+        check_cmd $python <<EOF
 import sys
 try:
     import $lib
@@ -1262,11 +1266,17 @@
     sys.exit(1)
 sys.exit(0)
 EOF
+    fi
 }
 
 check_pl_lib(){
     log check_pl_lib "$@"
-    check_cmd perl -e "use ${1};"
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Perl:$@ Be sure You have it for cross target!"
+        return 0
+    else
+        check_cmd perl -e "use ${1};"
+    fi
 }
 
 check_ld_cxx(){
diff --speed-large-files --no-dereference --minimal -Naur mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/configure mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/configure
--- mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/configure	2021-12-22 23:49:54.873302122 +0100
+++ mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/configure	2021-12-22 23:49:10.276636260 +0100
@@ -1230,23 +1230,33 @@
 }
 
 is_python3() {
-    log is_python3
-    check_cmd $python << EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for IsPython3 . Be sure You have it for cross target!"
+        return 0
+    else
+        log is_python3
+        check_cmd $python << EOF
 import sys
 if sys.version_info > (3,0):
     sys.exit(0)
 else:
     sys.exit(1)
 EOF
+    fi
 }
 
 check_python(){
-    log check_python
-    check_cmd $python <<EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Python ver > 3.6.0. Be sure You have it for cross target!"
+        return 0
+    else
+        log check_python
+        check_cmd $python <<EOF
 import sys
 if sys.version_info < (3,6,0):
     sys.exit(1)
 EOF
+    fi
 }
 
 check_py_lib(){
