diff --speed-large-files --no-dereference --minimal -Naur mythtv-5432d11672605878a7ab1964996edc7b0103087d/mythtv/libs/libmythbase/mythversion.h mythtv-5432d11672605878a7ab1964996edc7b0103087d/mythtv/libs/libmythbase/mythversion.h
--- mythtv-5432d11672605878a7ab1964996edc7b0103087d/mythtv/libs/libmythbase/mythversion.h	2022-03-19 18:21:45.000000000 +0100
+++ mythtv-5432d11672605878a7ab1964996edc7b0103087d/mythtv/libs/libmythbase/mythversion.h	2022-03-23 00:32:00.657130768 +0100
@@ -5,6 +5,8 @@
 #include "mythbaseexp.h"
 #include "mythconfig.h"
 
+#include "version.h"
+
 #if ( QT_VERSION < QT_MIN_VERSION )
 #error "You need a newer version of Qt to compile MythTV. Please re-run 'configure'."
 #endif
