diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2021-05-18 16:11:38.803333272 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2021-05-18 16:11:41.216666605 +0200
@@ -166,6 +166,12 @@
             if (m_metaAlbumPos > 0)
                 result["album"] = match.captured(m_metaAlbumPos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_metaFormat) );
+        }
     }
 
     return result;
