diff --speed-large-files --no-dereference --minimal -Naur mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/libs/libmythtv/tv_play.cpp mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/libs/libmythtv/tv_play.cpp	2022-07-30 02:43:05.000000000 +0200
+++ mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/libs/libmythtv/tv_play.cpp	2022-08-04 15:51:48.419994599 +0200
@@ -459,7 +459,7 @@
 void TV::InitKeys()
 {
     REG_KEY("TV Frontend", ACTION_PLAYBACK, QT_TRANSLATE_NOOP("MythControls",
-            "Play Program"), "P");
+            "Play Program"), "Ctrl+P");
     REG_KEY("TV Frontend", ACTION_STOP, QT_TRANSLATE_NOOP("MythControls",
             "Stop Program"), "");
     REG_KEY("TV Frontend", ACTION_TOGGLERECORD, QT_TRANSLATE_NOOP("MythControls",
@@ -781,7 +781,7 @@
     REG_KEY("TV Playback", ACTION_SWITCHANGLE, QT_TRANSLATE_NOOP("MythControls",
             "Switch angle"), "");
     REG_KEY("TV Playback", ACTION_OSDNAVIGATION, QT_TRANSLATE_NOOP("MythControls",
-            "OSD Navigation"), "");
+            "OSD Navigation"), "Select");
     REG_KEY("TV Playback", ACTION_ZOOMUP, QT_TRANSLATE_NOOP("MythControls",
             "Zoom mode - shift up"), "");
     REG_KEY("TV Playback", ACTION_ZOOMDOWN, QT_TRANSLATE_NOOP("MythControls",
@@ -885,7 +885,7 @@
 
     /* OSD playback information screen */
     REG_KEY("TV Playback", ACTION_TOGGLEOSDDEBUG,
-            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "");
+            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "Ctrl+I");
 
     /* 3D/Frame compatible/Stereoscopic TV */
     REG_KEY("TV Playback", ACTION_3DNONE,
diff --speed-large-files --no-dereference --minimal -Naur mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/libs/libmythui/mythmainwindow.cpp	2022-07-30 02:43:05.000000000 +0200
+++ mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/libs/libmythui/mythmainwindow.cpp	2022-08-04 15:47:01.936662970 +0200
@@ -797,7 +797,7 @@
     RegisterKey("Global", "PREVIOUS", QT_TRANSLATE_NOOP("MythControls",
         "Move to preview widget"), "Backtab");
     RegisterKey("Global", ACTION_SELECT, QT_TRANSLATE_NOOP("MythControls",
-        "Select"), "Return,Enter,Space");
+        "Select"), "Return,Enter,Space,Select");
     RegisterKey("Global", "BACKSPACE", QT_TRANSLATE_NOOP("MythControls",
         "Backspace"),       "Backspace");
     RegisterKey("Global", "ESCAPE", QT_TRANSLATE_NOOP("MythControls",
@@ -928,9 +928,9 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc");
+        "Display System Exit Prompt"),      "");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
-        "System Exit"),                     "");
+        "System Exit"),                     "Ctrl+Esc");
     RegisterKey("Main Menu",    "STANDBYMODE",QT_TRANSLATE_NOOP("MythControls",
         "Enter Standby Mode"),              "");
     RegisterKey("Long Press",    "LONGPRESS1",QT_TRANSLATE_NOOP("MythControls",
diff --speed-large-files --no-dereference --minimal -Naur mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/globalsettings.cpp	2022-08-04 15:52:21.766661065 +0200
+++ mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/globalsettings.cpp	2022-08-04 15:47:01.939996303 +0200
@@ -1730,7 +1730,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1784,7 +1784,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -1932,7 +1932,7 @@
     gs->setLabel(MainGeneralSettings::tr("Idle time before entering standby "
                                          "mode (minutes)"));
 
-    gs->setValue(90);
+    gs->setValue(0);
 
     gs->setHelpText(MainGeneralSettings::tr("Number of minutes to wait when "
                                             "the frontend is idle before "
@@ -2090,7 +2090,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -3213,7 +3213,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
diff --speed-large-files --no-dereference --minimal -Naur mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/mythfrontend.cpp mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/mythfrontend.cpp
--- mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/mythfrontend.cpp	2022-08-04 15:52:22.599994393 +0200
+++ mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/mythfrontend.cpp	2022-08-04 15:47:01.939996303 +0200
@@ -1634,7 +1634,7 @@
 
      // Gallery keybindings
      REG_KEY("Images", "PLAY", QT_TRANSLATE_NOOP("MythControls",
-         "Start/Stop Slideshow"), "P");
+         "Start/Stop Slideshow"), "Ctrl+P");
      REG_KEY("Images", "RECURSIVESHOW", QT_TRANSLATE_NOOP("MythControls",
          "Start Recursive Slideshow"), "R");
      REG_KEY("Images", "ROTRIGHT", QT_TRANSLATE_NOOP("MythControls",
diff --speed-large-files --no-dereference --minimal -Naur mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/videoplayersettings.cpp mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/videoplayersettings.cpp
--- mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/videoplayersettings.cpp	2022-07-30 02:43:05.000000000 +0200
+++ mythtv-5b989f3746f70af5aba05ecc89ff7be15a66acbd/mythtv/programs/mythfrontend/videoplayersettings.cpp	2022-08-04 15:47:01.939996303 +0200
@@ -57,7 +57,7 @@
     m_dvdDriveEdit->SetText(gCoreContext->GetSetting("DVDDeviceLocation",
                            "default"));
     m_blurayMountEdit->SetText(gCoreContext->GetSetting("BluRayMountpoint",
-                           "/media/cdrom"));
+                           "/media/disk"));
     m_altPlayerEdit->SetText(gCoreContext->GetSetting(
                            "mythvideo.VideoAlternatePlayer", "Internal"));
 
