diff --speed-large-files --no-dereference --minimal -Naur mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdisplaydrm.cpp mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdisplaydrm.cpp
--- mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdisplaydrm.cpp	2021-12-20 20:08:22.000000000 +0100
+++ mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdisplaydrm.cpp	2021-12-30 11:42:29.906521990 +0100
@@ -183,5 +183,5 @@
         return false;
     }
 
-    return m_device->SwitchMode(m_modeMap.value(mode));
+    return m_device->SwitchMode(m_modeMap.value(mode), m_screen);
 }
diff --speed-large-files --no-dereference --minimal -Naur mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdrmdevice.cpp mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdrmdevice.cpp
--- mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdrmdevice.cpp	2021-12-20 20:08:22.000000000 +0100
+++ mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdrmdevice.cpp	2021-12-30 15:23:34.706282777 +0100
@@ -655,7 +655,7 @@
  * using the atomic API. The old/non-atomic API *may* work but is untested and
  * realistically not much else is going to work without atomic.
 */
-bool MythDRMDevice::SwitchMode(int ModeIndex)
+bool MythDRMDevice::SwitchMode(int ModeIndex, QScreen *qScreen)
 {
     if (!(m_authenticated && m_atomic && m_connector.get() && m_crtc.get()))
         return false;
@@ -671,20 +671,14 @@
     auto modeid = MythDRMProperty::GetProperty("mode_id", m_crtc->m_properties);
     if (crtcid.get() && modeid.get())
     {
-        uint32_t blobid = 0;
-        // Presumably blobid does not need to be released? Can't find any documentation but
-        // there is the matching drmModeDestroyPropertyBlob...
-        if (drmModeCreatePropertyBlob(m_fd, &m_connector->m_modes[index], sizeof(drmModeModeInfo), &blobid) == 0)
-        {
-            QueueAtomics( {{ m_connector->m_id, crtcid->m_id, m_crtc->m_id },
-                           { m_crtc->m_id, modeid->m_id, blobid }} );
-            m_adjustedRefreshRate = m_connector->m_modes[index]->m_rate;
-            result = true;
-        }
-        else
-        {
-            LOG(VB_GENERAL, LOG_ERR, LOC + "Failed to create mode blob");
-        }
+        QByteArray property = "dri_set_mode_index";
+        //property is literal while ModeIndex is Int. so convert ModeIndex to literal then append
+        property.append(ModeIndex + 48);
+
+        if (auto * modeindex = qGuiApp->platformNativeInterface()->nativeResourceForScreen(property, qScreen); modeindex) {
+            LOG(VB_GENERAL, LOG_DEBUG, LOC + QString("nativeResourceForScreen mode_index %1 is set").arg(index));
+            result = true;
+        }
     }
 #endif
     return result;
diff --speed-large-files --no-dereference --minimal -Naur mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdrmdevice.h mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdrmdevice.h
--- mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdrmdevice.h	2021-12-20 20:08:22.000000000 +0100
+++ mythtv-0f4b8e886ec7d2674deac22066e1aa8ccf32832c/mythtv/libs/libmythui/platforms/mythdrmdevice.h	2021-12-30 11:43:58.713187042 +0100
@@ -41,7 +41,7 @@
     DRMConn  GetConnector   () const;
     const DRMModes& GetModes() const;
     bool     CanSwitchModes () const;
-    bool     SwitchMode     (int ModeIndex);
+    bool     SwitchMode     (int ModeIndex, QScreen *qScreen);
 
 #if defined (USING_QTPRIVATEHEADERS)
     static inline bool    s_mythDRMVideo     = qEnvironmentVariableIsSet("MYTHTV_DRM_VIDEO");
diff --speed-large-files --no-dereference --minimal -Naur mythtv-5a0d709f86becdfae8c04e509a45beceb75473de/mythtv/libs/libmythui/mythdisplay.cpp mythtv-5a0d709f86becdfae8c04e509a45beceb75473de/mythtv/libs/libmythui/mythdisplay.cpp
--- mythtv-5a0d709f86becdfae8c04e509a45beceb75473de/mythtv/libs/libmythui/mythdisplay.cpp	2021-12-30 16:52:20.000000000 +0100
+++ mythtv-5a0d709f86becdfae8c04e509a45beceb75473de/mythtv/libs/libmythui/mythdisplay.cpp	2021-12-31 11:05:48.296539723 +0100
@@ -740,7 +740,7 @@
         return true;
     }
 
-    LOG(VB_GENERAL, LOG_ERR, LOC + QString("Trying mode %1x%2@%3Hz")
+    LOG(VB_GENERAL, LOG_INFO, LOC + QString("Trying mode %1x%2@%3Hz")
         .arg(next.Width()).arg(next.Height()).arg(next.RefreshRate(), 0, 'f', 3));
 
     if (!SwitchToVideoMode(next.Resolution(), targetrate))
