From 3c3e62288de6d8f141434c6528ea35dee3005b39 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Tue, 25 Jun 2019 05:54:48 +0200
Subject: [PATCH 28/29] Even more BT delay

---
 drivers/bluetooth/hci_bcm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/bluetooth/hci_bcm.c b/drivers/bluetooth/hci_bcm.c
index 303550c..a0e7539 100644
--- a/drivers/bluetooth/hci_bcm.c
+++ b/drivers/bluetooth/hci_bcm.c
@@ -260,7 +260,7 @@ static int bcm_gpio_set_power(struct bcm_device *dev, bool powered)
 	}
 
 	/* wait for device to power on and come out of reset */
-	usleep_range(50000, 60000);
+	usleep_range(100000, 120000);
 
 	dev->res_enabled = powered;
 
-- 
2.7.1

