diff --speed-large-files --no-dereference --minimal -Naur old/ThirdParty/ANGLE/src/common/mathutil.h new/ThirdParty/ANGLE/src/common/mathutil.h
--- old/ThirdParty/ANGLE/src/common/mathutil.h	2021-12-23 01:02:54.000000000 +0100
+++ new/ThirdParty/ANGLE/src/common/mathutil.h	2023-04-29 19:02:46.609839201 +0200
@@ -16,6 +16,7 @@
 #include <algorithm>
 #include <string.h>
 #include <stdlib.h>
+#include <cstdint>
 
 namespace gl
 {
