diff --speed-large-files --no-dereference --minimal -Naur qtwebkit-84ad475fcaadc156568964eb1994f440a7ba9fdd/Source/WebCore/page/csp/ContentSecurityPolicy.cpp qtwebkit-84ad475fcaadc156568964eb1994f440a7ba9fdd/Source/WebCore/page/csp/ContentSecurityPolicy.cpp
--- qtwebkit-84ad475fcaadc156568964eb1994f440a7ba9fdd/Source/WebCore/page/csp/ContentSecurityPolicy.cpp	2021-12-23 01:02:54.000000000 +0100
+++ qtwebkit-84ad475fcaadc156568964eb1994f440a7ba9fdd/Source/WebCore/page/csp/ContentSecurityPolicy.cpp	2024-05-07 15:19:57.133716538 +0200
@@ -232,7 +232,7 @@
         cryptoDigest->addBytes(contentCString.data(), contentCString.length());
         Vector<uint8_t> digest = cryptoDigest->computeHash();
         for (auto& policy : policies) {
-            if ((policy.get()->*allowed)(std::make_pair(algorithm, digest)))
+//            if ((policy.get()->*allowed)(std::make_pair(algorithm, digest)))
                 return true;
         }
     }
