diff -Naur qtwebkit-5.212.0-alpha2-old/WebKit/qt/WebCoreSupport/QWebPageAdapter.cpp qtwebkit-5.212.0-alpha2-new/WebKit/qt/WebCoreSupport/QWebPageAdapter.cpp
--- qtwebkit-5.212.0-alpha2-old/WebKit/qt/WebCoreSupport/QWebPageAdapter.cpp	2017-06-09 16:11:36.000000000 +0200
+++ qtwebkit-5.212.0-alpha2-new//WebKit/qt/WebCoreSupport/QWebPageAdapter.cpp	2018-07-18 15:58:31.146666665 +0200
@@ -390,7 +390,10 @@
 
 QString QWebPageAdapter::selectedHtml() const
 {
-    return page->focusController().focusedOrMainFrame().editor().selectedRange()->toHTML();
+    RefPtr<Range> range = page->focusController().focusedOrMainFrame().editor().selectedRange();
+    if (!range)
+        return QString();
+    return range->toHTML();
 }
 
 bool QWebPageAdapter::isContentEditable() const
