diff --speed-large-files --no-dereference --minimal -Naur libxcrypt-4.5.2/lib/crypt-gost-yescrypt.c libxcrypt-4.5.2/lib/crypt-gost-yescrypt.c
--- libxcrypt-4.5.2/lib/crypt-gost-yescrypt.c	2025-11-10 17:31:08.000000000 +0000
+++ libxcrypt-4.5.2/lib/crypt-gost-yescrypt.c	2026-03-03 12:02:12.266579595 +0000
@@ -131,7 +131,7 @@
   intbuf->outbuf[1] = 'g';
 
   /* extract yescrypt output from "$y$param$salt$output" */
-  char *hptr = strchr ((const char *) intbuf->retval + 3, '$');
+  char *hptr = strchr ((char *) intbuf->retval + 3, '$');
   if (!hptr)
     {
       errno = EINVAL;
diff --speed-large-files --no-dereference --minimal -Naur libxcrypt-4.5.2/lib/crypt-sm3-yescrypt.c libxcrypt-4.5.2/lib/crypt-sm3-yescrypt.c
--- libxcrypt-4.5.2/lib/crypt-sm3-yescrypt.c	2025-11-10 17:31:08.000000000 +0000
+++ libxcrypt-4.5.2/lib/crypt-sm3-yescrypt.c	2026-03-03 12:02:42.931994196 +0000
@@ -136,7 +136,7 @@
   intbuf->outbuf[3] = '3';
 
   /* extract yescrypt output from "$y$param$salt$output" */
-  char *hptr = strchr ((const char *) intbuf->retval + 3, '$');
+  char *hptr = strchr ((char *) intbuf->retval + 3, '$');
   if (!hptr)
     {
       errno = EINVAL;
