diff -Naur xvattr-1.3-old/Makefile.in xvattr-1.3-new/Makefile.in
--- xvattr-1.3-old/Makefile.in	2002-06-13 12:43:51.000000000 -0700
+++ xvattr-1.3-new/Makefile.in	2007-01-31 20:41:12.000000000 -0800
@@ -78,7 +78,7 @@
 am__quote = @am__quote@
 install_sh = @install_sh@
 
-bin_PROGRAMS = xvattr gxvattr
+bin_PROGRAMS = xvattr
 xvattr_SOURCES = xvattr.c
 
 xvattr_LDADD = $(X_LIBS) $(X_PRE_LIBS) $(X_EXTRA_LIBS) -lX11 -lXext @LIBOBJS@
@@ -97,7 +97,7 @@
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_CLEAN_FILES = xvattr.pod
-bin_PROGRAMS = xvattr$(EXEEXT) gxvattr$(EXEEXT)
+bin_PROGRAMS = xvattr$(EXEEXT)
 PROGRAMS = $(bin_PROGRAMS)
 
 am_gxvattr_OBJECTS = gxvattr.$(OBJEXT)
