diff -Naur libmad-0.15.1b-old/configure.ac libmad-0.15.1b-new/configure.ac
--- libmad-0.15.1b-old/configure.ac	2010-04-27 15:35:49.000000000 -0700
+++ libmad-0.15.1b-new/configure.ac	2010-04-27 15:37:46.000000000 -0700
@@ -427,6 +427,10 @@
 dnl LTLIBOBJS=`echo "$LIBOBJS" | sed -e 's/\.o/.lo/g'`
 dnl AC_SUBST(LTLIBOBJS)
 
-AC_CONFIG_FILES([Makefile msvc++/Makefile  \
-	libmad.list])
+AC_CONFIG_FILES([
+	Makefile
+	msvc++/Makefile
+	libmad.list
+	mad.pc
+])
 AC_OUTPUT
diff -Naur libmad-0.15.1b-old/mad.pc.in libmad-0.15.1b-new/mad.pc.in
--- libmad-0.15.1b-old/mad.pc.in	1969-12-31 16:00:00.000000000 -0800
+++ libmad-0.15.1b-new/mad.pc.in	2010-04-27 15:37:46.000000000 -0700
@@ -0,0 +1,9 @@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mad
+Description: MPEG Audio Decoder
+Requires:
+Version: 0.14.2b
+Libs: -L${libdir} -lmad -lm
+Cflags: -I${includedir}
diff -Naur libmad-0.15.1b-old/Makefile.am libmad-0.15.1b-new/Makefile.am
--- libmad-0.15.1b-old/Makefile.am	2010-04-27 15:35:32.000000000 -0700
+++ libmad-0.15.1b-new/Makefile.am	2010-04-27 15:37:46.000000000 -0700
@@ -49,7 +49,10 @@
 			synth.c decoder.c layer12.c layer3.c huffman.c  \
 			$(headers) $(data_includes)
 
-EXTRA_libmad_la_SOURCES =	imdct_l_arm.S #synth_mmx.S
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = mad.pc
+
+EXTRA_libmad_la_SOURCES =	imdct_l_arm.S #synth_mmx.S mad.pc.in
 
 libmad_la_DEPENDENCIES =	@ASO_OBJS@
 libmad_la_LIBADD =		@ASO_OBJS@
