diff -Naur libmad-0.15.1b-old/configure.ac libmad-0.15.1b-new/configure.ac
--- libmad-0.15.1b-old/configure.ac	2004-01-23 01:41:32.000000000 -0800
+++ libmad-0.15.1b-new/configure.ac	2010-04-27 15:35:49.000000000 -0700
@@ -140,7 +140,6 @@
     case "$optimize" in
 	-O|"-O "*)
 	    optimize="-O"
-	    optimize="$optimize -fforce-mem"
 	    optimize="$optimize -fforce-addr"
 	    : #x optimize="$optimize -finline-functions"
 	    : #- optimize="$optimize -fstrength-reduce"
