diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.8/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux-6.6.8/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux-6.6.8/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2023-12-22 15:41:31.870013092 +0100
+++ linux-6.6.8/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2023-12-22 15:40:13.003346163 +0100
@@ -1268,7 +1268,7 @@
 				};
 
 				cpu-crit {
-					temperature = <100000>;
+					temperature = <1000000>;
 					hysteresis = <0>;
 					type = "critical";
 				};
@@ -1298,7 +1298,7 @@
 				};
 
 				gpu-crit {
-					temperature = <100000>;
+					temperature = <1000000>;
 					hysteresis = <0>;
 					type = "critical";
 				};
