diff -Naur fftw-3.3.3-old/configure.ac fftw-3.3.3-new/configure.ac
--- fftw-3.3.3-old/configure.ac	2012-11-25 04:33:44.000000000 -0800
+++ fftw-3.3.3-new/configure.ac	2013-01-03 17:45:30.000000000 -0800
@@ -7,7 +7,7 @@
 SHARED_VERSION_INFO="6:2:3" # CURRENT:REVISION:AGE
 
 AM_INIT_AUTOMAKE(1.7)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIR([m4])
 AM_MAINTAINER_MODE
 AC_SUBST(SHARED_VERSION_INFO)
