diff -Naur unifdef-1.0-old/Makefile unifdef-1.0-new/Makefile
--- unifdef-1.0-old/Makefile	2000-02-11 19:54:36.000000000 -0800
+++ unifdef-1.0-new/Makefile	2006-12-09 22:20:55.000000000 -0800
@@ -1,6 +1,9 @@
-CC     = gcc
-CFLAGS = -O2 
-DEST   = /usr
+CC      ?= gcc
+CFLAGS  ?= -O2 
+DESTDIR ?=
+prefix  ?= /usr
+sbindir ?= $(prefix)/bin
+mandir  ?= $(prefix)/usr/share/man
 
 unifdef: unifdef.o
 	$(CC) unifdef.o -o unifdef
@@ -9,5 +12,5 @@
 	rm -f unifdef *.o
 
 install: unifdef
-	install -m 0755 unifdef $(DEST)/bin
-	install -m 0644 unifdef.1 $(DEST)/man/man1 
+	install -m 0755 unifdef $(DESTDIR)$(sbindir)
+	install -m 0644 unifdef.1 $(DESTDIR)$(mandir)/man1 
