
From: devmfc <devmfc@mvand.net>
Subject: [PATCH 29/32] [DEVMFC] irqchip/meson-gpio: Add support for meson SC2
 SoCs

---
 drivers/irqchip/irq-meson-gpio.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/irqchip/irq-meson-gpio.c b/drivers/irqchip/irq-meson-gpio.c
index cd789fa515..0db666faaf 100644
--- a/drivers/irqchip/irq-meson-gpio.c
+++ b/drivers/irqchip/irq-meson-gpio.c
@@ -158,6 +158,10 @@ static const struct meson_gpio_irq_params t7_params = {
 	INIT_MESON_S4_COMMON_DATA(157)
 };
 
+static const struct meson_gpio_irq_params sc2_params = {
+	INIT_MESON_S4_COMMON_DATA(87)
+};
+
 static const struct of_device_id meson_irq_gpio_matches[] __maybe_unused = {
 	{ .compatible = "amlogic,meson8-gpio-intc", .data = &meson8_params },
 	{ .compatible = "amlogic,meson8b-gpio-intc", .data = &meson8b_params },
@@ -170,6 +174,7 @@ static const struct of_device_id meson_irq_gpio_matches[] __maybe_unused = {
 	{ .compatible = "amlogic,meson-s4-gpio-intc", .data = &s4_params },
 	{ .compatible = "amlogic,c3-gpio-intc", .data = &c3_params },
 	{ .compatible = "amlogic,t7-gpio-intc", .data = &t7_params },
+	{ .compatible = "amlogic,meson-sc2-gpio-intc", .data = &sc2_params },
 	{ }
 };
 
-- 
2.46.0

