
From: devmfc <devmfc@mvand.net>
Subject: net: wireless: broadcom add sdio ID 43342

diff --speed-large-files --no-dereference --minimal -Naur linux-6.18-rc5/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c linux-6.18-rc5/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
--- linux-6.18-rc5/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c	2025-11-15 09:23:00.418797846 +0100
+++ linux-6.18-rc5/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c	2025-11-15 09:20:02.642130576 +0100
@@ -609,6 +609,7 @@
 BRCMF_FW_DEF(4330, "brcmfmac4330-sdio");
 BRCMF_FW_DEF(4334, "brcmfmac4334-sdio");
 BRCMF_FW_DEF(43340, "brcmfmac43340-sdio");
+BRCMF_FW_DEF(43342, "brcmfmac43342-sdio");
 BRCMF_FW_DEF(4335, "brcmfmac4335-sdio");
 BRCMF_FW_DEF(43362, "brcmfmac43362-sdio");
 BRCMF_FW_DEF(4339, "brcmfmac4339-sdio");
@@ -646,6 +647,7 @@
 	BRCMF_FW_ENTRY(BRCM_CC_4334_CHIP_ID, 0xFFFFFFFF, 4334),
 	BRCMF_FW_ENTRY(BRCM_CC_43340_CHIP_ID, 0xFFFFFFFF, 43340),
 	BRCMF_FW_ENTRY(BRCM_CC_43341_CHIP_ID, 0xFFFFFFFF, 43340),
+	BRCMF_FW_ENTRY(BRCM_CC_43342_CHIP_ID, 0xFFFFFFFF, 43342),
 	BRCMF_FW_ENTRY(BRCM_CC_4335_CHIP_ID, 0xFFFFFFFF, 4335),
 	BRCMF_FW_ENTRY(BRCM_CC_43362_CHIP_ID, 0xFFFFFFFE, 43362),
 	BRCMF_FW_ENTRY(BRCM_CC_4339_CHIP_ID, 0xFFFFFFFF, 4339),
diff --speed-large-files --no-dereference --minimal -Naur linux-6.18-rc5/drivers/net/wireless/broadcom/brcm80211/include/brcm_hw_ids.h linux-6.18-rc5/drivers/net/wireless/broadcom/brcm80211/include/brcm_hw_ids.h
--- linux-6.18-rc5/drivers/net/wireless/broadcom/brcm80211/include/brcm_hw_ids.h	2025-11-10 00:10:19.000000000 +0100
+++ linux-6.18-rc5/drivers/net/wireless/broadcom/brcm80211/include/brcm_hw_ids.h	2025-11-15 09:22:34.738797755 +0100
@@ -27,6 +27,7 @@
 #define BRCM_CC_4334_CHIP_ID		0x4334
 #define BRCM_CC_43340_CHIP_ID		43340
 #define BRCM_CC_43341_CHIP_ID		43341
+#define BRCM_CC_43342_CHIP_ID		43342
 #define BRCM_CC_43362_CHIP_ID		43362
 #define BRCM_CC_4335_CHIP_ID		0x4335
 #define BRCM_CC_4339_CHIP_ID		0x4339
@@ -60,6 +61,7 @@
 #define CY_CC_4373_CHIP_ID		0x4373
 #define CY_CC_43012_CHIP_ID		43012
 #define CY_CC_43439_CHIP_ID		43439
+#define CY_CC_89459_CHIP_ID		0x4355
 
 /* USB Device IDs */
 #define BRCM_USB_43143_DEVICE_ID	0xbd1e
