diff --speed-large-files --no-dereference --minimal -Naur linux-6.18.4/drivers/thermal/amlogic_thermal.c linux-6.18.4/drivers/thermal/amlogic_thermal.c
--- linux-6.18.4/drivers/thermal/amlogic_thermal.c	2026-01-08 10:17:22.000000000 +0100
+++ linux-6.18.4/drivers/thermal/amlogic_thermal.c	2026-01-12 12:18:25.136689876 +0100
@@ -208,6 +208,13 @@
 	.n = 324,
 };
 
+static const struct amlogic_thermal_soc_calib_data amlogic_thermal_s4 = {
+	.A = 9411,
+	.B = 1159,
+	.m = 424,
+	.n = 324,
+};
+
 static const struct amlogic_thermal_data amlogic_thermal_g12a_cpu_param = {
 	.u_efuse_off = 0x128,
 	.calibration_parameters = &amlogic_thermal_g12a,
@@ -226,6 +233,12 @@
 	.regmap_config = &amlogic_thermal_regmap_config_g12a,
 };
 
+static const struct amlogic_thermal_data amlogic_thermal_s4_cpu_param = {
+	.u_efuse_off = 0x128,
+	.calibration_parameters = &amlogic_thermal_s4,
+	.regmap_config = &amlogic_thermal_regmap_config_g12a,
+};
+
 static const struct of_device_id of_amlogic_thermal_match[] = {
 	{
 		.compatible = "amlogic,g12a-ddr-thermal",
@@ -239,6 +252,10 @@
 		.compatible = "amlogic,a1-cpu-thermal",
 		.data = &amlogic_thermal_a1_cpu_param,
 	},
+	{
+		.compatible = "amlogic,s4-cpu-thermal",
+		.data = &amlogic_thermal_s4_cpu_param,
+	},
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, of_amlogic_thermal_match);
