diff --speed-large-files --no-dereference --minimal -Naur linux-6.18.7/drivers/media/platform/amlogic/vdec/aml_vdec.c linux-6.18.7/drivers/media/platform/amlogic/vdec/aml_vdec.c
--- linux-6.18.7/drivers/media/platform/amlogic/vdec/aml_vdec.c	2026-01-27 09:32:31.653342544 +0100
+++ linux-6.18.7/drivers/media/platform/amlogic/vdec/aml_vdec.c	2026-01-26 21:39:51.010032978 +0100
@@ -136,7 +136,7 @@
 
 	ret = ctx->codec_ops->run(ctx);
 
-	v4l2_m2m_buf_copy_metadata(src_buf, dst_buf);
+	v4l2_m2m_buf_copy_metadata(src_buf, dst_buf, true);
 	if (src_req)
 		v4l2_ctrl_request_complete(src_req, &ctx->ctrl_handler);
 	if (ret < 0)
diff --speed-large-files --no-dereference --minimal -Naur linux-6.18.7/drivers/media/platform/amlogic/vdec/aml_vdec_drv.c linux-6.18.7/drivers/media/platform/amlogic/vdec/aml_vdec_drv.c
--- linux-6.18.7/drivers/media/platform/amlogic/vdec/aml_vdec_drv.c	2026-01-27 09:32:31.653342544 +0100
+++ linux-6.18.7/drivers/media/platform/amlogic/vdec/aml_vdec_drv.c	2026-01-26 21:51:24.220051549 +0100
@@ -20,7 +20,7 @@
 	struct aml_vdec_ctx *ctx = NULL;
 	int ret = 0;
 
-	ctx = kzalloc_obj(*ctx, GFP_KERNEL);
+	ctx = kzalloc(sizeof(*ctx), GFP_KERNEL);
 	if (!ctx)
 		return -ENOMEM;
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.18.7/drivers/media/platform/amlogic/vdec/h264.c linux-6.18.7/drivers/media/platform/amlogic/vdec/h264.c
--- linux-6.18.7/drivers/media/platform/amlogic/vdec/h264.c	2026-01-27 09:32:31.653342544 +0100
+++ linux-6.18.7/drivers/media/platform/amlogic/vdec/h264.c	2026-01-26 21:52:36.283386801 +0100
@@ -1952,7 +1952,7 @@
 	struct aml_h264_ctx *h264_ctx;
 	int ret = 0;
 
-	h264_ctx = kzalloc_obj(*h264_ctx, GFP_KERNEL);
+	h264_ctx = kzalloc(sizeof(*h264_ctx), GFP_KERNEL);
 	if (!h264_ctx)
 		return -ENOMEM;
 
