diff --speed-large-files --no-dereference --minimal -Naur linux-6.18.4/arch/arm64/boot/dts/amlogic/meson-s4.dtsi linux-6.18.4/arch/arm64/boot/dts/amlogic/meson-s4.dtsi
--- linux-6.18.4/arch/arm64/boot/dts/amlogic/meson-s4.dtsi	2026-01-11 21:26:55.643389821 +0100
+++ linux-6.18.4/arch/arm64/boot/dts/amlogic/meson-s4.dtsi	2026-01-11 21:21:08.803380648 +0100
@@ -17,6 +17,8 @@
 #include <dt-bindings/reset/amlogic,meson-g12a-audio-reset.h>
 #include <dt-bindings/sound/meson-g12a-toacodec.h>
 #include <dt-bindings/sound/meson-g12a-tohdmitx.h>
+#include <dt-bindings/thermal/thermal.h>
+
 / {
 	cpus {
 		#address-cells = <2>;
@@ -920,6 +922,16 @@
 				status = "disabled";
 			};
 
+			cpu_temp: temperature-sensor@20000 {
+				compatible = "amlogic,g12a-cpu-thermal",
+					     "amlogic,g12a-thermal";
+				reg = <0x0 0x20000 0x0 0x50>;
+				interrupts = <GIC_SPI 30 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&clkc_periphs CLKID_TS>;
+				#thermal-sensor-cells = <0>;
+				amlogic,ao-secure = <&sec_ao>;
+			};
+
 			reset: reset-controller@2000 {
 				compatible = "amlogic,meson-s4-reset";
 				reg = <0x0 0x2000 0x0 0x98>;
@@ -1459,4 +1472,31 @@
 		resets = <&reset RESET_ACODEC>;
 	};
 
+	thermal-zones {
+		cpu_thermal: cpu-thermal {
+			polling-delay = <1000>;
+			polling-delay-passive = <100>;
+			thermal-sensors = <&cpu_temp>;
+
+			trips {
+				cpu_passive: cpu-passive {
+					temperature = <85000>; /* millicelsius */
+					hysteresis = <2000>; /* millicelsius */
+					type = "passive";
+				};
+
+				cpu_hot: cpu-hot {
+					temperature = <95000>; /* millicelsius */
+					hysteresis = <2000>; /* millicelsius */
+					type = "hot";
+				};
+
+				cpu_critical: cpu-critical {
+					temperature = <110000>; /* millicelsius */
+					hysteresis = <2000>; /* millicelsius */
+				    type = "critical";
+				};
+			};
+		};
+	};
 };
