diff --speed-large-files --no-dereference --minimal -Naur linux-6.18-rc5/arch/arm64/boot/dts/amlogic/meson-gx-p23x-q20x.dtsi linux-6.18-rc5/arch/arm64/boot/dts/amlogic/meson-gx-p23x-q20x.dtsi
--- linux-6.18-rc5/arch/arm64/boot/dts/amlogic/meson-gx-p23x-q20x.dtsi	2025-11-10 00:10:19.000000000 +0100
+++ linux-6.18-rc5/arch/arm64/boot/dts/amlogic/meson-gx-p23x-q20x.dtsi	2025-11-13 18:24:30.052123868 +0100
@@ -99,6 +99,7 @@
 
 	cvbs_connector: cvbs-connector {
 		compatible = "composite-video-connector";
+		status = "disabled";
 
 		port {
 			cvbs_connector_in: endpoint {
