diff --speed-large-files --no-dereference --minimal -Naur linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-ac2xx.dtsi linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-ac2xx.dtsi
--- linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-ac2xx.dtsi	2025-10-29 14:10:32.000000000 +0100
+++ linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-ac2xx.dtsi	2025-10-30 20:14:47.845459892 +0100
@@ -28,7 +28,7 @@
 		reset-gpios = <&gpio BOOT_12 GPIO_ACTIVE_LOW>;
 	};
 
-	cvbs-connector {
+	cvbs_connector: cvbs-connector {
 		compatible = "composite-video-connector";
 
 		port {
diff --speed-large-files --no-dereference --minimal -Naur linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-tx5-plus.dts linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-tx5-plus.dts
--- linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-tx5-plus.dts	2025-10-31 13:23:57.755495140 +0100
+++ linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-tx5-plus.dts	2025-10-30 20:14:07.045459750 +0100
@@ -113,3 +113,9 @@
 		vddio-supply = <&vddio_ao1v8>;
 	};
 };
+
+&cvbs_connector {
+	status = "disabled";
+};
+
+
diff --speed-large-files --no-dereference --minimal -Naur linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air.dts linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air.dts
--- linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air.dts	2025-10-29 14:10:32.000000000 +0100
+++ linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air.dts	2025-10-30 20:21:01.175461139 +0100
@@ -113,3 +113,7 @@
 &tohdmitx {
 	status = "okay";
 };
+
+&cvbs_connector {
+	status = "disabled";
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air-gbit.dts linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air-gbit.dts
--- linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air-gbit.dts	2025-10-29 14:10:32.000000000 +0100
+++ linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air-gbit.dts	2025-10-30 20:20:47.712127764 +0100
@@ -134,3 +134,7 @@
 &tohdmitx {
 	status = "okay";
 };
+
+&cvbs_connector {
+	status = "disabled";
+};
