diff --speed-large-files --no-dereference --minimal -Naur linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-ac2xx.dtsi linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-ac2xx.dtsi
--- linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-ac2xx.dtsi	2025-10-31 15:10:42.872183385 +0100
+++ linux-6.17.6/arch/arm64/boot/dts/amlogic/meson-sm1-ac2xx.dtsi	2025-10-31 14:45:25.395511541 +0100
@@ -248,7 +248,6 @@
 	pinctrl-names = "default", "clk-gate";
 
 	bus-width = <4>;
-	cap-sd-highspeed;
 	/* CRC errors are observed at 50MHz */
 	max-frequency = <35000000>;
 	disable-wp;
