diff --speed-large-files --no-dereference --minimal -Naur linux-6.18.4/arch/arm64/boot/dts/amlogic/meson-s4-s905w2-ap201.dts linux-6.18.4/arch/arm64/boot/dts/amlogic/meson-s4-s905w2-ap201.dts
--- linux-6.18.4/arch/arm64/boot/dts/amlogic/meson-s4-s905w2-ap201.dts	2026-01-12 13:09:17.070104785 +0100
+++ linux-6.18.4/arch/arm64/boot/dts/amlogic/meson-s4-s905w2-ap201.dts	2026-01-12 12:53:03.510170376 +0100
@@ -168,6 +168,57 @@
 				<689000 100>;
 	};
 
+	cpu_opp_table: cpu-opp-table {
+		compatible = "operating-points-v2";
+		status = "okay";
+		opp-shared;
+
+		opp00 {
+			opp-hz = /bits/ 64 <100000000>;
+			opp-microvolt = <769000>;
+		};
+		opp01 {
+			opp-hz = /bits/ 64 <250000000>;
+			opp-microvolt = <769000>;
+		};
+		opp02 {
+			opp-hz = /bits/ 64 <500000000>;
+			opp-microvolt = <769000>;
+		};
+		opp03 {
+			opp-hz = /bits/ 64 <666000000>;
+			opp-microvolt = <769000>;
+		};
+		opp04 {
+			opp-hz = /bits/ 64 <1000000000>;
+			opp-microvolt = <769000>;
+		};
+		opp05 {
+			opp-hz = /bits/ 64 <1200000000>;
+			opp-microvolt = <769000>;
+		};
+		opp06 {
+			opp-hz = /bits/ 64 <1404000000>;
+			opp-microvolt = <799000>;
+		};
+		opp07 {
+			opp-hz = /bits/ 64 <1500000000>;
+			opp-microvolt = <829000>;
+		};
+		opp08 {
+			opp-hz = /bits/ 64 <1608000000>;
+			opp-microvolt = <869000>;
+		};
+		opp09 {
+			opp-hz = /bits/ 64 <1704000000>;
+			opp-microvolt = <909000>;
+		};
+		opp10 {
+			opp-hz = /bits/ 64 <1800000000>;
+			opp-microvolt = <1009000>;
+		};
+	};
+
 	dmics: audio-codec-1 {
 		compatible = "dmic-codec";
 		#sound-dai-cells = <0>;
@@ -572,3 +623,26 @@
 	status = "okay";
 };
 
+&cpu0 {
+	cpu-supply = <&vdd_cpu>;
+	operating-points-v2 = <&cpu_opp_table>;
+	//clocks = <&clkc CLKID_CPU_CLK>;
+};
+
+&cpu1 {
+	cpu-supply = <&vdd_cpu>;
+	operating-points-v2 = <&cpu_opp_table>;
+	//clocks = <&clkc CLKID_CPU_CLK>;
+};
+
+&cpu2 {
+	cpu-supply = <&vdd_cpu>;
+	operating-points-v2 = <&cpu_opp_table>;
+	//clocks = <&clkc CLKID_CPU_CLK>;
+};
+
+&cpu3 {
+	cpu-supply = <&vdd_cpu>;
+	operating-points-v2 = <&cpu_opp_table>;
+	//clocks = <&clkc CLKID_CPU_CLK>;
+};
