diff --speed-large-files --no-dereference --minimal -Naur linux-6.18-rc4/drivers/net/phy/phy_device.c linux-6.18-rc4/drivers/net/phy/phy_device.c
--- linux-6.18-rc4/drivers/net/phy/phy_device.c	2025-11-02 20:28:02.000000000 +0100
+++ linux-6.18-rc4/drivers/net/phy/phy_device.c	2025-11-04 15:43:22.332181273 +0100
@@ -3544,8 +3544,7 @@
  * @new_driver: new phy_driver to register
  * @owner: module owning this PHY
  */
-static int phy_driver_register(struct phy_driver *new_driver,
-			       struct module *owner)
+int phy_driver_register(struct phy_driver *new_driver, struct module *owner)
 {
 	int retval;
 
@@ -3588,11 +3587,7 @@
 
 	return 0;
 }
-
-static void phy_driver_unregister(struct phy_driver *drv)
-{
-	driver_unregister(&drv->mdiodrv.driver);
-}
+EXPORT_SYMBOL(phy_driver_register);
 
 int phy_drivers_register(struct phy_driver *new_driver, int n,
 			 struct module *owner)
@@ -3611,6 +3606,12 @@
 }
 EXPORT_SYMBOL(phy_drivers_register);
 
+void phy_driver_unregister(struct phy_driver *drv)
+{
+	driver_unregister(&drv->mdiodrv.driver);
+}
+EXPORT_SYMBOL(phy_driver_unregister);
+
 void phy_drivers_unregister(struct phy_driver *drv, int n)
 {
 	int i;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.18-rc4/include/linux/phy.h linux-6.18-rc4/include/linux/phy.h
--- linux-6.18-rc4/include/linux/phy.h	2025-11-02 20:28:02.000000000 +0100
+++ linux-6.18-rc4/include/linux/phy.h	2025-11-04 15:43:22.332181273 +0100
@@ -2030,7 +2030,9 @@
 		return genphy_read_status(phydev);
 }
 
+void phy_driver_unregister(struct phy_driver *drv);
 void phy_drivers_unregister(struct phy_driver *drv, int n);
+int phy_driver_register(struct phy_driver *new_driver, struct module *owner);
 int phy_drivers_register(struct phy_driver *new_driver, int n,
 			 struct module *owner);
 void phy_error(struct phy_device *phydev);
