diff --speed-large-files --no-dereference --minimal -Naur linux-6.18.2/drivers/irqchip/irq-brcmstb-l2.c linux-6.18.2/drivers/irqchip/irq-brcmstb-l2.c
--- linux-6.18.2/drivers/irqchip/irq-brcmstb-l2.c	2025-12-18 14:03:43.000000000 +0100
+++ linux-6.18.2/drivers/irqchip/irq-brcmstb-l2.c	2025-12-18 13:06:57.237050477 +0100
@@ -51,6 +51,16 @@
 	.cpu_mask_clear		= 0x0C
 };
 
+/* Register offsets in the 2711 L2 level interrupt controller */
+static const struct brcmstb_intc_init_params l2_2711_lvl_intc_init = {
+	.handler		= handle_level_irq,
+	.cpu_status		= 0x00,
+	.cpu_clear		= 0x08,
+	.cpu_mask_status	= 0x0c,
+	.cpu_mask_set		= 0x10,
+	.cpu_mask_clear		= 0x14
+};
+
 /* L2 intc private data structure */
 struct brcmstb_l2_intc_data {
 	struct irq_domain *domain;
@@ -138,12 +148,13 @@
 	irq_reg_writel(gc, ~b->saved_mask, ct->regs.enable);
 }
 
-static int brcmstb_l2_intc_probe(struct platform_device *pdev, struct device_node *parent,
-				 const struct brcmstb_intc_init_params *init_params)
+static int __init brcmstb_l2_intc_of_init(struct device_node *np,
+					  struct device_node *parent,
+					  const struct brcmstb_intc_init_params
+					  *init_params)
 {
 	unsigned int clr = IRQ_NOREQUEST | IRQ_NOPROBE | IRQ_NOAUTOEN;
 	unsigned int set = 0;
-	struct device_node *np = pdev->dev.of_node;
 	struct brcmstb_l2_intc_data *data;
 	struct irq_chip_type *ct;
 	int ret;
@@ -256,21 +267,30 @@
 	return ret;
 }
 
-static int brcmstb_l2_edge_intc_probe(struct platform_device *pdev, struct device_node *parent)
+static int __init brcmstb_l2_edge_intc_of_init(struct device_node *np,
+	struct device_node *parent)
 {
-	return brcmstb_l2_intc_probe(pdev, parent, &l2_edge_intc_init);
+	return brcmstb_l2_intc_of_init(np, parent, &l2_edge_intc_init);
 }
 
-static int brcmstb_l2_lvl_intc_probe(struct platform_device *pdev, struct device_node *parent)
+static int __init brcmstb_l2_lvl_intc_of_init(struct device_node *np,
+	struct device_node *parent)
 {
-	return brcmstb_l2_intc_probe(pdev, parent, &l2_lvl_intc_init);
+	return brcmstb_l2_intc_of_init(np, parent, &l2_lvl_intc_init);
+}
+
+static int __init brcmstb_l2_2711_lvl_intc_of_init(struct device_node *np,
+	struct device_node *parent)
+{
+	return brcmstb_l2_intc_of_init(np, parent, &l2_2711_lvl_intc_init);
 }
 
 IRQCHIP_PLATFORM_DRIVER_BEGIN(brcmstb_l2)
-IRQCHIP_MATCH("brcm,l2-intc", brcmstb_l2_edge_intc_probe)
-IRQCHIP_MATCH("brcm,hif-spi-l2-intc", brcmstb_l2_edge_intc_probe)
-IRQCHIP_MATCH("brcm,upg-aux-aon-l2-intc", brcmstb_l2_edge_intc_probe)
-IRQCHIP_MATCH("brcm,bcm7271-l2-intc", brcmstb_l2_lvl_intc_probe)
+IRQCHIP_MATCH("brcm,l2-intc", brcmstb_l2_edge_intc_of_init)
+IRQCHIP_MATCH("brcm,hif-spi-l2-intc", brcmstb_l2_edge_intc_of_init)
+IRQCHIP_MATCH("brcm,upg-aux-aon-l2-intc", brcmstb_l2_edge_intc_of_init)
+IRQCHIP_MATCH("brcm,bcm7271-l2-intc", brcmstb_l2_lvl_intc_of_init)
+IRQCHIP_MATCH("brcm,bcm2711-l2-intc", brcmstb_l2_2711_lvl_intc_of_init)
 IRQCHIP_PLATFORM_DRIVER_END(brcmstb_l2)
 MODULE_DESCRIPTION("Broadcom STB generic L2 interrupt controller");
 MODULE_LICENSE("GPL v2");
