diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc2/drivers/irqchip/irq-brcmstb-l2.c linux-6.16-rc2/drivers/irqchip/irq-brcmstb-l2.c
--- linux-6.16-rc2/drivers/irqchip/irq-brcmstb-l2.c	2025-06-15 22:49:41.000000000 +0200
+++ linux-6.16-rc2/drivers/irqchip/irq-brcmstb-l2.c	2025-06-20 14:59:56.271360415 +0200
@@ -51,6 +51,16 @@
 	.cpu_mask_clear		= 0x0C
 };
 
+/* Register offsets in the 2711 L2 level interrupt controller */
+static const struct brcmstb_intc_init_params l2_2711_lvl_intc_init = {
+	.handler		= handle_level_irq,
+	.cpu_status		= 0x00,
+	.cpu_clear		= 0x08,
+	.cpu_mask_status	= 0x0c,
+	.cpu_mask_set		= 0x10,
+	.cpu_mask_clear		= 0x14
+};
+
 /* L2 intc private data structure */
 struct brcmstb_l2_intc_data {
 	struct irq_domain *domain;
@@ -269,11 +279,18 @@
 	return brcmstb_l2_intc_of_init(np, parent, &l2_lvl_intc_init);
 }
 
+static int __init brcmstb_l2_2711_lvl_intc_of_init(struct device_node *np,
+	struct device_node *parent)
+{
+	return brcmstb_l2_intc_of_init(np, parent, &l2_2711_lvl_intc_init);
+}
+
 IRQCHIP_PLATFORM_DRIVER_BEGIN(brcmstb_l2)
 IRQCHIP_MATCH("brcm,l2-intc", brcmstb_l2_edge_intc_of_init)
 IRQCHIP_MATCH("brcm,hif-spi-l2-intc", brcmstb_l2_edge_intc_of_init)
 IRQCHIP_MATCH("brcm,upg-aux-aon-l2-intc", brcmstb_l2_edge_intc_of_init)
 IRQCHIP_MATCH("brcm,bcm7271-l2-intc", brcmstb_l2_lvl_intc_of_init)
+IRQCHIP_MATCH("brcm,bcm2711-l2-intc", brcmstb_l2_2711_lvl_intc_of_init)
 IRQCHIP_PLATFORM_DRIVER_END(brcmstb_l2)
 MODULE_DESCRIPTION("Broadcom STB generic L2 interrupt controller");
 MODULE_LICENSE("GPL v2");
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc2/drivers/irqchip/Kconfig linux-6.16-rc2/drivers/irqchip/Kconfig
--- linux-6.16-rc2/drivers/irqchip/Kconfig	2025-06-15 22:49:41.000000000 +0200
+++ linux-6.16-rc2/drivers/irqchip/Kconfig	2025-06-20 14:59:56.271360415 +0200
@@ -128,6 +128,14 @@
 
 	  If unsure say n.
 
+config BCM2712_MIP
+	bool "Broadcom 2712 MSI-X Interrupt Peripheral support"
+	depends on ARM_GIC
+	select GENERIC_IRQ_CHIP
+	select IRQ_DOMAIN
+	help
+	  Enable support for the Broadcom BCM2712 MSI-X target peripheral.
+
 config BCM6345_L1_IRQ
 	bool
 	select GENERIC_IRQ_CHIP
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc2/drivers/irqchip/Makefile linux-6.16-rc2/drivers/irqchip/Makefile
--- linux-6.16-rc2/drivers/irqchip/Makefile	2025-06-15 22:49:41.000000000 +0200
+++ linux-6.16-rc2/drivers/irqchip/Makefile	2025-06-20 14:59:56.271360415 +0200
@@ -132,3 +132,5 @@
 obj-$(CONFIG_MCHP_EIC)			+= irq-mchp-eic.o
 obj-$(CONFIG_SOPHGO_SG2042_MSI)		+= irq-sg2042-msi.o
 obj-$(CONFIG_SUNPLUS_SP7021_INTC)	+= irq-sp7021-intc.o
+
+KBUILD_CFLAGS += "-Wno-incompatible-pointer-types"
