diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc4/arch/arm/boot/dts/broadcom/bcm2711-rpi-4-b.dts linux-6.16-rc4/arch/arm/boot/dts/broadcom/bcm2711-rpi-4-b.dts
--- linux-6.16-rc4/arch/arm/boot/dts/broadcom/bcm2711-rpi-4-b.dts	2025-06-29 22:09:04.000000000 +0200
+++ linux-6.16-rc4/arch/arm/boot/dts/broadcom/bcm2711-rpi-4-b.dts	2025-07-01 22:07:36.668345071 +0200
@@ -11,6 +11,14 @@
 	compatible = "raspberrypi,4-model-b", "brcm,bcm2711";
 	model = "Raspberry Pi 4 Model B";
 
+	aliases {
+		mmc0 = &emmc2;
+		emmc2bus = &emmc2bus;
+		ethernet0 = &genet;
+		pcie0 = &pcie0;
+		blconfig = &blconfig;
+	};
+
 	chosen {
 		/* 8250 auxiliary UART instead of pl011 */
 		stdout-path = "serial1:115200n8";
@@ -158,10 +166,11 @@
 
 &leds {
 	led_pwr: led-pwr {
-		label = "PWR";
+		label = "DISK";
 		gpios = <&expgpio 2 GPIO_ACTIVE_LOW>;
-		default-state = "keep";
-		linux,default-trigger = "default-on";
+		default-state = "off";
+		function = LED_FUNCTION_DISK_ACTIVITY;
+		linux,default-trigger = "disk-activity";
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc4/arch/arm/boot/dts/broadcom/bcm283x-rpi-led-deprecated.dtsi linux-6.16-rc4/arch/arm/boot/dts/broadcom/bcm283x-rpi-led-deprecated.dtsi
--- linux-6.16-rc4/arch/arm/boot/dts/broadcom/bcm283x-rpi-led-deprecated.dtsi	2025-06-29 22:09:04.000000000 +0200
+++ linux-6.16-rc4/arch/arm/boot/dts/broadcom/bcm283x-rpi-led-deprecated.dtsi	2025-07-01 21:53:57.421696204 +0200
@@ -1,5 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0
 
+#include <dt-bindings/leds/common.h>
+
 / {
 	/*
 	 * This file provides the now deprecated ACT LED to the
@@ -12,6 +14,7 @@
 		led_act: led-act {
 			label = "ACT";
 			default-state = "keep";
+			function = LED_FUNCTION_STATUS;
 			linux,default-trigger = "heartbeat";
 		};
 	};
