diff --speed-large-files --no-dereference --minimal -Naur linux-6.17.3/drivers/media/platform/rockchip/rkvdec/rkvdec-hevc-common.c linux-6.17.3/drivers/media/platform/rockchip/rkvdec/rkvdec-hevc-common.c
--- linux-6.17.3/drivers/media/platform/rockchip/rkvdec/rkvdec-hevc-common.c	2025-10-20 09:39:15.706713074 +0200
+++ linux-6.17.3/drivers/media/platform/rockchip/rkvdec/rkvdec-hevc-common.c	2025-10-20 09:38:54.832576179 +0200
@@ -42,10 +42,17 @@
 	if (ctrl->id != V4L2_CID_STATELESS_HEVC_SPS)
 		return RKVDEC_IMG_FMT_ANY;
 
-	if (sps->bit_depth_luma_minus8 == 0)
-		return RKVDEC_IMG_FMT_420_8BIT;
-	else if (sps->bit_depth_luma_minus8 == 2)
-		return RKVDEC_IMG_FMT_420_10BIT;
+	if (sps->bit_depth_luma_minus8 == 0) {
+		if (sps->chroma_format_idc == 2)
+			return RKVDEC_IMG_FMT_422_8BIT;
+		else
+			return RKVDEC_IMG_FMT_420_8BIT;
+	} else if (sps->bit_depth_luma_minus8 == 2) {
+		if (sps->chroma_format_idc == 2)
+			return RKVDEC_IMG_FMT_422_10BIT;
+		else
+			return RKVDEC_IMG_FMT_420_10BIT;
+	}
 
 	return RKVDEC_IMG_FMT_ANY;
 }
