diff --speed-large-files --no-dereference --minimal -Naur linux-6.17.5/drivers/media/platform/rockchip/rkvdec/rkvdec-hevc-common.c linux-6.17.5/drivers/media/platform/rockchip/rkvdec/rkvdec-hevc-common.c
--- linux-6.17.5/drivers/media/platform/rockchip/rkvdec/rkvdec-hevc-common.c	2025-10-29 12:52:15.495469192 +0100
+++ linux-6.17.5/drivers/media/platform/rockchip/rkvdec/rkvdec-hevc-common.c	2025-10-29 12:48:48.612135188 +0100
@@ -515,17 +515,12 @@
 			      V4L2_CID_STATELESS_HEVC_SCALING_MATRIX);
 	run->scaling_matrix = ctrl ? ctrl->p_cur.p : NULL;
 
-	if (ctx->has_sps_st_rps) {
-		ctrl = v4l2_ctrl_find(&ctx->ctrl_hdl,
-				      V4L2_CID_STATELESS_HEVC_EXT_SPS_ST_RPS);
-		run->ext_sps_st_rps = ctrl ? ctrl->p_cur.p : NULL;
-	}
-	if (ctx->has_sps_lt_rps) {
-		ctrl = v4l2_ctrl_find(&ctx->ctrl_hdl,
-				      V4L2_CID_STATELESS_HEVC_EXT_SPS_LT_RPS);
-		run->ext_sps_lt_rps = ctrl ? ctrl->p_cur.p : NULL;
-	}
-
+	ctrl = v4l2_ctrl_find(&ctx->ctrl_hdl,
+			      V4L2_CID_STATELESS_HEVC_EXT_SPS_ST_RPS);
+	run->ext_sps_st_rps = ctrl ? ctrl->p_cur.p : NULL;
+	ctrl = v4l2_ctrl_find(&ctx->ctrl_hdl,
+			      V4L2_CID_STATELESS_HEVC_EXT_SPS_LT_RPS);
+	run->ext_sps_lt_rps = ctrl ? ctrl->p_cur.p : NULL;
 	rkvdec_run_preamble(ctx, &run->base);
 }
 
