diff --speed-large-files --no-dereference --minimal -Naur linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec-vp9.c linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec-vp9.c
--- linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec-vp9.c	2025-09-22 10:33:23.446718510 +0200
+++ linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec-vp9.c	2025-09-22 10:34:22.316718725 +0200
@@ -659,7 +659,7 @@
 
 	regs->vp9.reg44.strmd_error_e = 0xe;
 
-	rkvdec_memcpy_toio(rkvdec->regs, regs, MIN(sizeof(*regs), rkvdec->variant->num_regs));
+	rkvdec_memcpy_toio(rkvdec->regs, regs, sizeof(*regs));
 }
 
 static int validate_dec_params(struct rkvdec_ctx *ctx,
