diff --speed-large-files --no-dereference --minimal -Naur linux-6.18-rc5/arch/arm64/boot/dts/qcom/sc7280.dtsi linux-6.18-rc5/arch/arm64/boot/dts/qcom/sc7280.dtsi
--- linux-6.18-rc5/arch/arm64/boot/dts/qcom/sc7280.dtsi	2025-11-11 14:55:54.458787769 +0100
+++ linux-6.18-rc5/arch/arm64/boot/dts/qcom/sc7280.dtsi	2025-11-11 14:54:58.382120898 +0100
@@ -862,7 +862,7 @@
 		interrupts = <GIC_PPI 7 IRQ_TYPE_LEVEL_LOW>;
 	};
 
-	psci {
+	psci: psci {
 		compatible = "arm,psci-1.0";
 		method = "smc";
 
