diff --speed-large-files --no-dereference --minimal -Naur linux-6.7/arch/arm64/boot/dts/rockchip/rk356x.dtsi linux-6.7/arch/arm64/boot/dts/rockchip/rk356x.dtsi
--- linux-6.7/arch/arm64/boot/dts/rockchip/rk356x.dtsi	2024-01-17 10:03:54.683675652 +0100
+++ linux-6.7/arch/arm64/boot/dts/rockchip/rk356x.dtsi	2024-01-17 09:12:14.830409071 +0100
@@ -669,6 +669,7 @@
 		resets = <&cru SRST_RGA_CORE>, <&cru SRST_A_RGA>, <&cru SRST_H_RGA>;
 		reset-names = "core", "axi", "ahb";
 		power-domains = <&power RK3568_PD_RGA>;
+		status = "disabled";
 	};
 
 	vepu: video-codec@fdee0000 {
@@ -679,6 +680,7 @@
 		clock-names = "aclk", "hclk";
 		iommus = <&vepu_mmu>;
 		power-domains = <&power RK3568_PD_RGA>;
+		status = "disabled";
 	};
 
 	vepu_mmu: iommu@fdee0800 {
@@ -689,6 +691,7 @@
 		clock-names = "aclk", "iface";
 		power-domains = <&power RK3568_PD_RGA>;
 		#iommu-cells = <0>;
+		status = "disabled";
 	};
 
 	sdmmc2: mmc@fe000000 {
