diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.1/drivers/gpu/drm/rockchip/dw_hdmi-rockchip.c linux-6.6.1/drivers/gpu/drm/rockchip/dw_hdmi-rockchip.c
--- linux-6.6.1/drivers/gpu/drm/rockchip/dw_hdmi-rockchip.c	2023-11-10 11:10:53.553353533 +0100
+++ linux-6.6.1/drivers/gpu/drm/rockchip/dw_hdmi-rockchip.c	2023-11-10 11:09:36.523353279 +0100
@@ -1292,6 +1292,9 @@
 			    const struct drm_display_info *info,
 			    const struct drm_display_mode *mode)
 {
+	// hack hack to stop kernel oops on other rk devices....
+	return MODE_OK;
+
 	struct rockchip_hdmi *hdmi = data;
 	const struct dw_hdmi_mpll_config *mpll_cfg = rockchip_mpll_cfg;
 	int pclk = mode->clock * 1000;
