diff --speed-large-files --no-dereference --minimal -Naur linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
--- linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2023-12-28 13:42:37.123385122 +0100
+++ linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2023-12-28 13:40:56.310051457 +0100
@@ -18,8 +18,8 @@
 	compatible = "xunlong,orangepi-5-plus", "rockchip,rk3588";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		serial2 = &uart2;
 	};
 
@@ -82,8 +82,8 @@
 		led {
 			color = <LED_COLOR_ID_BLUE>;
 			function = LED_FUNCTION_INDICATOR;
-			function-enumerator = <1>;
 			gpios = <&gpio3 RK_PA6 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "mmc0";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
--- linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2023-12-28 13:42:37.193385123 +0100
+++ linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2023-12-28 13:39:33.820051175 +0100
@@ -13,8 +13,8 @@
 	compatible = "radxa,rock-5b", "rockchip,rk3588";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio;
 		serial2 = &uart2;
 	};
@@ -46,10 +46,10 @@
 		pinctrl-0 = <&led_rgb_b>;
 
 		led_rgb_b {
-			function = LED_FUNCTION_STATUS;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			color = <LED_COLOR_ID_BLUE>;
 			gpios = <&gpio0 RK_PB7 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "mmc0";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts
--- linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2023-12-28 13:42:37.086718456 +0100
+++ linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2023-12-28 13:39:23.913384479 +0100
@@ -16,6 +16,8 @@
 
 	aliases {
 		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
+		ethernet0 = &gmac1;
 		serial2 = &uart2;
 	};
 
@@ -43,9 +45,10 @@
 		pinctrl-0 = <&leds_gpio>;
 
 		led-1 {
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			gpios = <&gpio1 RK_PA2 GPIO_ACTIVE_HIGH>;
 			label = "status_led";
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "mmc0";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
--- linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2023-12-28 13:42:37.193385123 +0100
+++ linux-6.7-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2023-12-28 13:39:13.163384441 +0100
@@ -13,9 +13,10 @@
 	compatible = "radxa,rock-5a", "rockchip,rk3588s";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		serial2 = &uart2;
+		ethernet0 = &gmac1;
 	};
 
 	analog-sound {
@@ -43,9 +44,9 @@
 
 		io-led {
 			color = <LED_COLOR_ID_BLUE>;
-			function = LED_FUNCTION_STATUS;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "mmc0";
 		};
 	};
 
