diff -Naur iputils-s20121221-old/Makefile iputils-s20121221-new/Makefile
--- iputils-s20121221-old/Makefile	2012-12-21 06:01:07.000000000 -0800
+++ iputils-s20121221-new/Makefile	2012-12-22 06:52:59.000000000 -0800
@@ -110,7 +110,7 @@
 IPV6_TARGETS=tracepath6 traceroute6 ping6
 TARGETS=$(IPV4_TARGETS) $(IPV6_TARGETS)
 
-CFLAGS=$(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
+CFLAGS+=$(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
 LDLIBS=$(LDLIB) $(ADDLIB)
 
 UNAME_N:=$(shell uname -n)
