diff --speed-large-files --no-dereference --minimal -Naur linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h313-cpu-opp.dtsi linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h313-cpu-opp.dtsi
--- linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h313-cpu-opp.dtsi	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h313-cpu-opp.dtsi	2024-11-16 12:44:54.641683178 +0100
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+// Copyright (C) 2021 Piotr Oniszczuk <piotr.oniszczuk@gmail.com>
+
+/*
+	X96-q DDR3 vendor Android DT:
+	480000000	900mV
+	600000000	900mV
+	792000000	900mV
+	1008000000	920mV
+	1200000000	980mV
+	1344000000	1120mV
+	1416000000	1140mV
+	1512000000	1160mV
+*/
+
+/ {
+	cpu_opp_table: opp-table-cpu {
+		compatible = "allwinner,sun50i-h616-operating-points";
+		nvmem-cells = <&cpu_speed_grade>;
+		opp-shared;
+
+		opp-480000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <480000000>;
+			opp-microvolt = <900000>;
+			opp-supported-hw = <0x3f>;
+		};
+
+		opp-600000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <900000>;
+			opp-supported-hw = <0x3f>;
+		};
+
+		opp-792000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <792000000>;
+			opp-microvolt = <900000>;
+			opp-supported-hw = <0x3f>;
+		};
+
+		opp-1008000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1008000000>;
+			opp-microvolt = <920000>;
+			opp-supported-hw = <0x3f>;
+		};
+
+		opp-1200000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1200000000>;
+			opp-microvolt = <980000>;
+			opp-supported-hw = <0x3f>;
+		};
+
+		opp-1344000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1344000000>;
+			opp-microvolt = <1120000>;
+			opp-supported-hw = <0x3f>;
+		};
+
+		opp-1416000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1416000000>;
+			opp-microvolt = <1140000>;
+			opp-supported-hw = <0x3f>;
+		};
+
+		opp-1512000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1512000000>;
+			opp-microvolt = <1160000>;
+			opp-supported-hw = <0x3f>;
+		};
+
+		opp-1608000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1608000000>;
+			opp-microvolt = <1160000>;
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1704000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1704000000>;
+			opp-microvolt = <1200000>;
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1800000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1800000000>;
+			opp-microvolt = <1200000>;
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1896000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1896000000>;
+			opp-microvolt = <1200000>;
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1992000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1992000000>;
+			opp-microvolt = <1200000>;
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+	};
+};
+
+&cpu0 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu1 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu2 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu3 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-11-16 13:04:49.558353822 +0100
+++ linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-11-16 13:02:52.181686761 +0100
@@ -29,7 +29,9 @@
 			reg = <0>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
 			#cooling-cells = <2>;
+			operating-points-v2 = <&cpu_opp_table>;
 			i-cache-size = <0x8000>;
 			i-cache-line-size = <64>;
 			i-cache-sets = <256>;
@@ -45,7 +47,9 @@
 			reg = <1>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
 			#cooling-cells = <2>;
+			operating-points-v2 = <&cpu_opp_table>;
 			i-cache-size = <0x8000>;
 			i-cache-line-size = <64>;
 			i-cache-sets = <256>;
@@ -61,7 +65,9 @@
 			reg = <2>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
 			#cooling-cells = <2>;
+			operating-points-v2 = <&cpu_opp_table>;
 			i-cache-size = <0x8000>;
 			i-cache-line-size = <64>;
 			i-cache-sets = <256>;
@@ -77,7 +83,9 @@
 			reg = <3>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
 			#cooling-cells = <2>;
+			operating-points-v2 = <&cpu_opp_table>;
 			i-cache-size = <0x8000>;
 			i-cache-line-size = <64>;
 			i-cache-sets = <256>;
@@ -1273,6 +1281,25 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
+
+		sunxi-info {
+			compatible = "allwinner,sun50i-h6-sys-info";
+			status = "okay";
+		};
+
+		addr_mgt: addr-mgt {
+			compatible = "allwinner,sunxi-addr_mgt";
+			type_addr_wifi = <0x2>;
+			type_addr_bt = <0x2>;
+			type_addr_eth = <0x2>;
+			status = "okay";
+		};
+
+		dump_reg: dump_reg@20000 {
+			compatible = "allwinner,sunxi-dump-reg";
+			reg = <0x0 0x03001000 0x0 0x0f20>;
+			status = "okay";
+		};
 	};
 
 	thermal-zones {
@@ -1344,4 +1371,28 @@
 			};
 		};
 	};
+
+	gpu_opp_table: gpu-opp-table {
+		compatible = "operating-points-v2";
+		opp-125000000 {
+			opp-hz = /bits/ 64 <125000000>;
+			opp-microvolt = <860000>;
+		};
+		opp-250000000 {
+			opp-hz = /bits/ 64 <250000000>;
+			opp-microvolt = <860000>;
+		};
+		opp-432000000 {
+			opp-hz = /bits/ 64 <432000000>;
+			opp-microvolt = <910000>;
+		};
+		opp-600000000 {
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <960000>;
+		};
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			opp-microvolt = <1080000>;
+		};
+	};
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero.dtsi linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero.dtsi
--- linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero.dtsi	2024-11-10 23:19:35.000000000 +0100
+++ linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero.dtsi	2024-11-16 13:02:52.185020094 +0100
@@ -7,6 +7,7 @@
  */
 
 #include "sun50i-h616.dtsi"
+#include "sun50i-h616-cpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
