diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/Makefile linux-6.14.4/arch/arm64/boot/dts/allwinner/Makefile
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/Makefile	2025-04-28 12:27:48.970918271 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/Makefile	2025-04-28 12:25:47.084254230 +0200
@@ -1,4 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
+
+subdir-y	:= $(dts-dirs) overlay
+
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-amarula-relic.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-bananapi-m64.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-nanopi-a64.dtb
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/Makefile linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/Makefile
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/Makefile	2025-04-28 12:25:11.257588360 +0200
@@ -0,0 +1,13 @@
+# SPDX-License-Identifier: GPL-2.0
+dtb-$(CONFIG_ARCH_SUNXI) += \
+	sun50i-h616-enable-1.512ghz-cpu-opp.dtbo \
+	sun50i-h616-enable-1.608ghz-cpu-opp.dtbo \
+	sun50i-h616-enable-1.704ghz-cpu-opp.dtbo \
+	sun50i-h616-enable-1.800ghz-cpu-opp.dtbo \
+	sun50i-h616-enable-1.896ghz-cpu-opp.dtbo \
+	sun50i-h616-enable-1.992ghz-cpu-opp.dtbo \
+
+targets += $(dtb-y)
+
+always		:= $(dtb-y)
+clean-files	:= *.dtbo
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.512ghz-cpu-opp.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.512ghz-cpu-opp.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.512ghz-cpu-opp.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.512ghz-cpu-opp.dtso	2025-04-28 12:25:11.257588360 +0200
@@ -0,0 +1,10 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.608ghz-cpu-opp.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.608ghz-cpu-opp.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.608ghz-cpu-opp.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.608ghz-cpu-opp.dtso	2025-04-28 12:25:11.257588360 +0200
@@ -0,0 +1,15 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1120000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.704ghz-cpu-opp.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.704ghz-cpu-opp.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.704ghz-cpu-opp.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.704ghz-cpu-opp.dtso	2025-04-28 12:25:11.257588360 +0200
@@ -0,0 +1,20 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1120000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.800ghz-cpu-opp.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.800ghz-cpu-opp.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.800ghz-cpu-opp.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.800ghz-cpu-opp.dtso	2025-04-28 12:25:11.257588360 +0200
@@ -0,0 +1,25 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1120000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+
+	opp-1800000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.896ghz-cpu-opp.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.896ghz-cpu-opp.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.896ghz-cpu-opp.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.896ghz-cpu-opp.dtso	2025-04-28 12:25:11.257588360 +0200
@@ -0,0 +1,30 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1120000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+
+	opp-1800000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1896000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.992ghz-cpu-opp.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.992ghz-cpu-opp.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.992ghz-cpu-opp.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-enable-1.992ghz-cpu-opp.dtso	2025-04-28 12:25:11.257588360 +0200
@@ -0,0 +1,35 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1120000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+
+	opp-1800000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1896000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1992000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi	2025-04-25 10:51:21.000000000 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi	2025-04-28 12:25:11.257588360 +0200
@@ -106,6 +106,46 @@
 			clock-latency-ns = <244144>; /* 8 32k periods */
 			opp-supported-hw = <0x2a>;
 		};
+
+		opp-1608000000 {
+			opp-hz = /bits/ 64 <1608000000>;
+			opp-microvolt = <1120000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1704000000 {
+			opp-hz = /bits/ 64 <1704000000>;
+			opp-microvolt = <1160000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1800000000 {
+			opp-hz = /bits/ 64 <1800000000>;
+			opp-microvolt = <1200000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1896000000 {
+			opp-hz = /bits/ 64 <1896000000>;
+			opp-microvolt = <1200000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1992000000 {
+			opp-hz = /bits/ 64 <1992000000>;
+			opp-microvolt = <1200000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
 	};
 };
 
