diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/Makefile linux-6.14.4/arch/arm64/boot/dts/allwinner/Makefile
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/Makefile	2025-04-30 14:13:19.477620629 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/Makefile	2025-04-30 14:05:26.260964337 +0200
@@ -1,4 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
+
+subdir-y	:= $(dts-dirs) overlay
+
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-amarula-relic.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-bananapi-m64.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-nanopi-a64.dtb
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/Makefile linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/Makefile
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/Makefile	2025-04-30 14:12:34.737621599 +0200
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: GPL-2.0
+dtb-$(CONFIG_ARCH_SUNXI) += \
+	overclock-h313-cpu-to-1.512ghz.dtbo \
+	overclock-h313-cpu-to-1.608ghz.dtbo \
+	overclock-h313-cpu-to-1.704ghz.dtbo \
+	overclock-h313-cpu-to-1.800ghz.dtbo \
+	overclock-h313-cpu-to-1.896ghz.dtbo \
+	overclock-h313-cpu-to-1.992ghz.dtbo \
+	overclock-h616-cpu-to-1.512ghz.dtbo \
+	overclock-h616-cpu-to-1.608ghz.dtbo \
+	overclock-h616-cpu-to-1.704ghz.dtbo \
+	overclock-h616-cpu-to-1.800ghz.dtbo \
+	overclock-h616-cpu-to-1.896ghz.dtbo \
+	overclock-h616-cpu-to-1.992ghz.dtbo \
+
+targets += $(dtb-y)
+
+always		:= $(dtb-y)
+clean-files	:= *.dtbo
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.512ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.512ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.512ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.512ghz.dtso	2025-04-30 14:09:28.107625704 +0200
@@ -0,0 +1,10 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1160000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.608ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.608ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.608ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.608ghz.dtso	2025-04-30 14:09:52.847625170 +0200
@@ -0,0 +1,15 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1160000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.704ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.704ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.704ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.704ghz.dtso	2025-04-30 14:10:18.827624581 +0200
@@ -0,0 +1,20 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1160000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.800ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.800ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.800ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.800ghz.dtso	2025-04-30 14:10:45.827623963 +0200
@@ -0,0 +1,25 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1160000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1800000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.896ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.896ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.896ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.896ghz.dtso	2025-04-30 14:11:13.680956688 +0200
@@ -0,0 +1,30 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1160000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1800000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1896000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.992ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.992ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.992ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-cpu-to-1.992ghz.dtso	2025-04-30 14:11:37.217622847 +0200
@@ -0,0 +1,35 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1160000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1800000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1896000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1992000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.512ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.512ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.512ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.512ghz.dtso	2025-04-30 14:05:26.260964337 +0200
@@ -0,0 +1,10 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.608ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.608ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.608ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.608ghz.dtso	2025-04-30 14:05:26.260964337 +0200
@@ -0,0 +1,15 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1120000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.704ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.704ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.704ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.704ghz.dtso	2025-04-30 14:05:26.260964337 +0200
@@ -0,0 +1,20 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1120000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.800ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.800ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.800ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.800ghz.dtso	2025-04-30 14:05:26.260964337 +0200
@@ -0,0 +1,25 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1120000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+
+	opp-1800000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.896ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.896ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.896ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.896ghz.dtso	2025-04-30 14:05:26.260964337 +0200
@@ -0,0 +1,30 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1120000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+
+	opp-1800000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1896000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.992ghz.dtso linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.992ghz.dtso
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.992ghz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-cpu-to-1.992ghz.dtso	2025-04-30 14:05:26.260964337 +0200
@@ -0,0 +1,35 @@
+/dts-v1/;
+/plugin/;
+
+&{/opp-table-cpu} {
+	opp-1512000000 {
+		opp-microvolt = <1120000>;
+		opp-supported-hw = <0x3f>;
+		status = "okay";
+	};
+
+	opp-1608000000 {
+		opp-microvolt = <1120000>;
+		status = "okay";
+	};
+
+	opp-1704000000 {
+		opp-microvolt = <1160000>;
+		status = "okay";
+	};
+
+	opp-1800000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1896000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+
+	opp-1992000000 {
+		opp-microvolt = <1200000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi	2025-04-25 10:51:21.000000000 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi	2025-04-30 14:05:26.260964337 +0200
@@ -106,6 +106,46 @@
 			clock-latency-ns = <244144>; /* 8 32k periods */
 			opp-supported-hw = <0x2a>;
 		};
+
+		opp-1608000000 {
+			opp-hz = /bits/ 64 <1608000000>;
+			opp-microvolt = <1120000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1704000000 {
+			opp-hz = /bits/ 64 <1704000000>;
+			opp-microvolt = <1160000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1800000000 {
+			opp-hz = /bits/ 64 <1800000000>;
+			opp-microvolt = <1200000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1896000000 {
+			opp-hz = /bits/ 64 <1896000000>;
+			opp-microvolt = <1200000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1992000000 {
+			opp-hz = /bits/ 64 <1992000000>;
+			opp-microvolt = <1200000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts	2025-04-30 14:13:17.007620681 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts	2025-04-30 14:51:44.007570803 +0200
@@ -228,7 +228,7 @@
 			reg_dcdc2: dcdc2 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
