diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-tanix-tx1.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-tanix-tx1.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-tanix-tx1.dts	2025-04-27 18:43:59.587682534 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-tanix-tx1.dts	2025-04-27 18:31:13.907698995 +0200
@@ -100,6 +100,28 @@
 	status = "okay";
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &de {
 	status = "okay";
 };
@@ -178,7 +200,7 @@
 			reg_dcdc2: dcdc2 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q.dts	2025-04-27 18:43:59.527682535 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q.dts	2025-04-27 18:32:00.461031319 +0200
@@ -112,6 +112,28 @@
 	status = "okay";
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &de {
 	status = "okay";
 };
@@ -290,7 +312,7 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q-lpddr3.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q-lpddr3.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q-lpddr3.dts	2025-04-27 18:43:59.544349201 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q-lpddr3.dts	2025-04-27 18:31:29.184365333 +0200
@@ -112,6 +112,28 @@
 	status = "okay";
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &de {
 	status = "okay";
 };
@@ -236,7 +258,7 @@
 			reg_dcdc2: dcdc2 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q-v5.1.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q-v5.1.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q-v5.1.dts	2025-04-27 18:43:59.604349201 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h313-x96-q-v5.1.dts	2025-04-27 18:31:41.871031728 +0200
@@ -112,6 +112,28 @@
 	status = "okay";
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &de {
 	status = "okay";
 };
@@ -236,7 +258,7 @@
 			reg_dcdc2: dcdc2 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi	2025-04-25 10:51:21.000000000 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi	2025-04-27 18:33:05.491029930 +0200
@@ -104,7 +104,47 @@
 			opp-microvolt-speed3 = <1100000>;
 			opp-microvolt-speed5 = <1160000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
-			opp-supported-hw = <0x2a>;
+			opp-supported-hw = <0x2d>;
+		};
+
+		opp-1608000000 {
+			opp-hz = /bits/ 64 <1608000000>;
+			opp-microvolt = <1120000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1704000000 {
+			opp-hz = /bits/ 64 <1704000000>;
+			opp-microvolt = <1160000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1800000000 {
+			opp-hz = /bits/ 64 <1800000000>;
+			opp-microvolt = <1200000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1896000000 {
+			opp-hz = /bits/ 64 <1896000000>;
+			opp-microvolt = <1200000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
+		};
+
+		opp-1992000000 {
+			opp-hz = /bits/ 64 <1992000000>;
+			opp-microvolt = <1200000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-supported-hw = <0x3f>;
+			status = "disabled";
 		};
 	};
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts	2025-04-27 18:43:59.481015869 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts	2025-04-27 18:09:45.114393309 +0200
@@ -17,6 +17,28 @@
 	cpu-supply = <&reg_dcdca>;
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &emac0 {
 	allwinner,rx-delay-ps = <3100>;
 	allwinner,tx-delay-ps = <700>;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-pendoo-x12pro.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-pendoo-x12pro.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-pendoo-x12pro.dts	2025-04-27 18:43:59.621015867 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-pendoo-x12pro.dts	2025-04-27 18:18:27.101048741 +0200
@@ -113,6 +113,28 @@
 	status = "okay";
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &de {
 	status = "okay";
 };
@@ -295,7 +317,7 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s-axp313.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s-axp313.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s-axp313.dts	2025-04-27 18:43:59.511015869 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s-axp313.dts	2025-04-27 18:18:42.451048414 +0200
@@ -113,6 +113,28 @@
 	status = "okay";
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &de {
 	status = "okay";
 };
@@ -241,7 +263,7 @@
 			reg_dcdc2: dcdc2 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts	2025-04-27 18:43:59.497682535 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts	2025-04-27 18:19:03.501047970 +0200
@@ -113,6 +113,28 @@
 	status = "okay";
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &de {
 	status = "okay";
 };
@@ -294,7 +316,7 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts	2025-04-27 18:43:59.574349201 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts	2025-04-27 18:19:20.467714284 +0200
@@ -110,6 +110,28 @@
 	status = "okay";
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &de {
 	status = "okay";
 };
@@ -228,7 +250,7 @@
 			reg_dcdc2: dcdc2 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts	2025-04-27 18:43:59.481015869 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts	2025-04-27 18:08:43.587727953 +0200
@@ -17,6 +17,28 @@
 	cpu-supply = <&reg_dcdc2>;
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &emac0 {
 	allwinner,tx-delay-ps = <700>;
 	phy-mode = "rgmii-rxid";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-transpeed-8k618-t.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-transpeed-8k618-t.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-transpeed-8k618-t.dts	2025-04-27 18:43:59.634349199 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-transpeed-8k618-t.dts	2025-04-27 18:17:21.977716811 +0200
@@ -113,6 +113,50 @@
 	status = "okay";
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &de {
 	status = "okay";
 };
@@ -241,7 +285,7 @@
 			reg_dcdc2: dcdc2 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-vontar-h618.dts linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-vontar-h618.dts
--- linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-vontar-h618.dts	2025-04-27 18:43:59.557682535 +0200
+++ linux-6.14.4/arch/arm64/boot/dts/allwinner/sun50i-h618-vontar-h618.dts	2025-04-27 18:16:37.627717774 +0200
@@ -113,6 +113,28 @@
 	status = "okay";
 };
 
+&cpu_opp_table {
+	opp-1608000000 {
+		status = "disabled";
+	};
+
+	opp-1704000000 {
+		status = "disabled";
+	};
+
+	opp-1800000000 {
+		status = "disabled";
+	};
+
+	opp-1896000000 {
+		status = "disabled";
+	};
+
+	opp-1992000000 {
+		status = "disabled";
+	};
+};
+
 &de {
 	status = "okay";
 };
@@ -241,7 +263,7 @@
 			reg_dcdc2: dcdc2 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
