diff --speed-large-files --no-dereference --minimal -Naur linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts
--- linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2022-03-06 21:27:44.806398088 +0100
+++ linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2022-03-06 21:14:46.289745476 +0100
@@ -126,6 +126,14 @@
 		regulator-max-microvolt = <5000000>;
 	};
 
+	vdd_gpu: vdd-gpu {
+		compatible = "regulator-fixed";
+		regulator-name = "vdd_gpu";
+		regulator-min-microvolt = <1150000>;
+		regulator-max-microvolt = <1150000>;
+		regulator-always-on;
+	};
+
 	ir-receiver {
 		compatible = "gpio-ir-receiver";
 		gpios = <&gpio2 RK_PA2 GPIO_ACTIVE_LOW>;
@@ -233,8 +241,77 @@
 	};
 };
 
+&cpu0_opp_table {
+	compatible = "operating-points-v2";
+	opp-shared;
+
+	opp-408000000 {
+		opp-hz = /bits/ 64 <408000000>;
+		opp-microvolt = <950000>;
+		clock-latency-ns = <40000>;
+		opp-suspend;
+	};
+	opp-600000000 {
+		opp-hz = /bits/ 64 <600000000>;
+		opp-microvolt = <950000>;
+		clock-latency-ns = <40000>;
+	};
+	opp-816000000 {
+		opp-hz = /bits/ 64 <816000000>;
+		opp-microvolt = <1000000>;
+		clock-latency-ns = <40000>;
+	};
+	opp-1008000000 {
+		opp-hz = /bits/ 64 <1008000000>;
+		opp-microvolt = <1100000>;
+		clock-latency-ns = <40000>;
+	};
+	opp-1200000000 {
+		opp-hz = /bits/ 64 <1200000000>;
+		opp-microvolt = <1225000>;
+		clock-latency-ns = <40000>;
+	};
+	opp-1296000000 {
+		opp-hz = /bits/ 64 <1296000000>;
+		opp-microvolt = <1300000>;
+		clock-latency-ns = <40000>;
+	};
+	opp-1392000000 {
+		opp-hz = /bits/ 64 <1392000000>;
+		opp-microvolt = <1300000>;
+		clock-latency-ns = <40000>;
+	};
+	opp-1416000000 {
+		opp-hz = /bits/ 64 <1416000000>;
+		opp-microvolt = <1350000>;
+		clock-latency-ns = <40000>;
+	};
+};
+
 &gpu {
-	mali-supply = <&vdd_logic>;
+	mali-supply = <&vdd_gpu>;
+	status = "okay";
+};
+
+&gpu_opp_table {
+	compatible = "operating-points-v2";
+
+	opp-200000000 {
+		opp-hz = /bits/ 64 <200000000>;
+		opp-microvolt = <1150000>;
+	};
+	opp-300000000 {
+		opp-hz = /bits/ 64 <300000000>;
+		opp-microvolt = <1150000>;
+	};
+	opp-400000000 {
+		opp-hz = /bits/ 64 <400000000>;
+		opp-microvolt = <1150000>;
+	};
+	opp-500000000 {
+		opp-hz = /bits/ 64 <500000000>;
+		opp-microvolt = <1150000>;
+	};
 };
 
 &hdmi {
diff --speed-large-files --no-dereference --minimal -Naur linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328.dtsi linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328.dtsi
--- linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328.dtsi	2022-03-06 21:27:44.623064758 +0100
+++ linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328.dtsi	2022-03-06 20:49:35.493106048 +0100
@@ -641,20 +641,19 @@
 
 		opp-200000000 {
 			opp-hz = /bits/ 64 <200000000>;
-			opp-microvolt = <1075000>;
+			opp-microvolt = <1150000>;
 		};
 		opp-300000000 {
 			opp-hz = /bits/ 64 <300000000>;
-			opp-microvolt = <1075000>;
+			opp-microvolt = <1150000>;
 		};
 		opp-400000000 {
 			opp-hz = /bits/ 64 <400000000>;
-			opp-microvolt = <1075000>;
+			opp-microvolt = <1150000>;
 		};
 		opp-500000000 {
 			opp-hz = /bits/ 64 <500000000>;
 			opp-microvolt = <1150000>;
-			status = "disabled";
 		};
 	};
 
