diff --speed-large-files --no-dereference --minimal -Naur linux-6.14-rc6/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c linux-6.14-rc6/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
--- linux-6.14-rc6/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c	2025-03-12 09:30:32.486698554 +0100
+++ linux-6.14-rc6/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c	2025-03-12 09:27:14.170031223 +0100
@@ -2141,9 +2141,9 @@
 		dip |= FIELD_PREP(RK3588_DSP_IF_POL__DP0_PIN_POL, polflags);
 		break;
 	case ROCKCHIP_VOP2_EP_DP1:
-		die &= ~RK3588_SYS_DSP_INFACE_EN_MIPI1_MUX;
-		die |= RK3588_SYS_DSP_INFACE_EN_MIPI1 |
-			   FIELD_PREP(RK3588_SYS_DSP_INFACE_EN_MIPI1_MUX, vp->id);
+		die &= ~RK3588_SYS_DSP_INFACE_EN_DP1_MUX;
+		die |= RK3588_SYS_DSP_INFACE_EN_DP1 |
+			   FIELD_PREP(RK3588_SYS_DSP_INFACE_EN_DP1_MUX, vp->id);
 		dip &= ~RK3588_DSP_IF_POL__DP1_PIN_POL;
 		dip |= FIELD_PREP(RK3588_DSP_IF_POL__DP1_PIN_POL, polflags);
 		break;
