diff --speed-large-files --no-dereference --minimal -Naur linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
--- linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2025-02-08 14:06:45.200064828 +0100
+++ linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2025-02-08 13:46:19.180060597 +0100
@@ -1443,6 +1443,22 @@
 		status = "disabled";
 	};
 
+	hdmi0_sound: hdmi0-sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,mclk-fs = <128>;
+		simple-audio-card,name = "rockchip-hdmi0";
+		status = "disabled";
+
+		simple-audio-card,codec {
+			sound-dai = <&hdmi0>;
+		};
+
+		simple-audio-card,cpu {
+			sound-dai = <&i2s5_8ch>;
+		};
+	};
+
 	hdmi0: hdmi@fde80000 {
 		compatible = "rockchip,rk3588-dw-hdmi-qp";
 		reg = <0x0 0xfde80000 0x0 0x20000>;
@@ -1451,8 +1467,21 @@
 			 <&cru CLK_HDMITX0_REF>,
 			 <&cru MCLK_I2S5_8CH_TX>,
 			 <&cru CLK_HDMIHDP0>,
+			 <&cru DCLK_VOP0>,
+			 <&cru DCLK_VOP1>,
+			 <&cru DCLK_VOP2>,
+			 <&cru DCLK_VOP3>,
 			 <&cru HCLK_VO1>;
-		clock-names = "pclk", "earc", "ref", "aud", "hdp", "hclk_vo1";
+		clock-names = "pclk",
+			      "earc",
+			      "ref",
+			      "aud",
+			      "hdp",
+			      "dclk_vp0",
+			      "dclk_vp1",
+			      "dclk_vp2",
+			      "dclk_vp3",
+			      "hclk_vo1";
 		interrupts = <GIC_SPI 169 IRQ_TYPE_LEVEL_HIGH 0>,
 			     <GIC_SPI 170 IRQ_TYPE_LEVEL_HIGH 0>,
 			     <GIC_SPI 171 IRQ_TYPE_LEVEL_HIGH 0>,
@@ -1468,6 +1497,7 @@
 		reset-names = "ref", "hdp";
 		rockchip,grf = <&sys_grf>;
 		rockchip,vo-grf = <&vo1_grf>;
+		#sound-dai-cells = <0>;
 		status = "disabled";
 
 		ports {
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi
--- linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi	2025-02-08 14:06:45.170064828 +0100
+++ linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi	2025-02-08 13:52:21.413395236 +0100
@@ -140,6 +140,22 @@
 		status = "disabled";
 	};
 
+	hdmi1_sound: hdmi1-sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,mclk-fs = <128>;
+		simple-audio-card,name = "rockchip-hdmi1";
+		status = "disabled";
+
+		simple-audio-card,codec {
+			sound-dai = <&hdmi1>;
+		};
+
+		simple-audio-card,cpu {
+			sound-dai = <&i2s6_8ch>;
+		};
+	};
+
 	hdmi1: hdmi@fdea0000 {
 		compatible = "rockchip,rk3588-dw-hdmi-qp";
 		reg = <0x0 0xfdea0000 0x0 0x20000>;
@@ -148,8 +164,21 @@
 			 <&cru CLK_HDMITX1_REF>,
 			 <&cru MCLK_I2S6_8CH_TX>,
 			 <&cru CLK_HDMIHDP1>,
+			 <&cru DCLK_VOP0>,
+			 <&cru DCLK_VOP1>,
+			 <&cru DCLK_VOP2>,
+			 <&cru DCLK_VOP3>,
 			 <&cru HCLK_VO1>;
-		clock-names = "pclk", "earc", "ref", "aud", "hdp", "hclk_vo1";
+		clock-names = "pclk",
+			      "earc",
+			      "ref",
+			      "aud",
+			      "hdp",
+			      "dclk_vp0",
+			      "dclk_vp1",
+			      "dclk_vp2",
+			      "dclk_vp3",
+			      "hclk_vo1";
 		interrupts = <GIC_SPI 173 IRQ_TYPE_LEVEL_HIGH 0>,
 			     <GIC_SPI 174 IRQ_TYPE_LEVEL_HIGH 0>,
 			     <GIC_SPI 175 IRQ_TYPE_LEVEL_HIGH 0>,
@@ -165,6 +194,7 @@
 		reset-names = "ref", "hdp";
 		rockchip,grf = <&sys_grf>;
 		rockchip,vo-grf = <&vo1_grf>;
+		#sound-dai-cells = <0>;
 		status = "disabled";
 
 		ports {
