diff --speed-large-files --no-dereference --minimal -Naur linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts
--- linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts	2025-02-03 00:39:26.000000000 +0100
+++ linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts	2025-02-09 20:59:30.960008184 +0100
@@ -9,6 +9,7 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/leds/common.h>
+#include <dt-bindings/soc/rockchip,vop2.h>
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/pwm/pwm.h>
 #include "dt-bindings/usb/pd.h"
@@ -19,8 +20,8 @@
 	compatible = "radxa,rock-5-itx", "rockchip,rk3588";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio;
 	};
 
@@ -28,6 +29,17 @@
 		stdout-path = "serial2:1500000n8";
 	};
 
+	hdmi1-con {
+		compatible = "hdmi-connector";
+		type = "d";
+
+		port {
+			hdmi1_con_in: endpoint {
+				remote-endpoint = <&hdmi1_out_con>;
+			};
+		};
+	};
+
 	adc_keys: adc-keys {
 		compatible = "adc-keys";
 		io-channels = <&saradc 0>;
@@ -56,6 +68,7 @@
 			  "Headphone", "Headphones";
 	};
 
+
 	gpio-leds {
 		compatible = "gpio-leds";
 		pinctrl-names = "default";
@@ -63,7 +76,7 @@
 
 		power-led1 {
 			gpios = <&gpio0 RK_PB7 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "default-on";
+			linux,default-trigger = "heartbeat";
 		};
 
 		hdd-led2 {
@@ -84,7 +97,7 @@
 	fan0: pwm-fan {
 		compatible = "pwm-fan";
 		#cooling-cells = <2>;
-		cooling-levels = <0 64 128 192 255>;
+		cooling-levels = <0 96 128 164 196 224 255>;
 		fan-supply = <&vcc12v_dcin>;
 		pwms = <&pwm14 0 10000 0>;
 	};
@@ -458,6 +471,10 @@
 	};
 };
 
+&i2s6_8ch {
+	status = "okay";
+};
+
 &package_thermal {
 	polling-delay = <1000>;
 
@@ -721,6 +738,7 @@
 	broken-cd;
 	bus-width = <4>;
 	cap-sdio-irq;
+	disable-wp;
 	keep-power-in-suspend;
 	max-frequency = <150000000>;
 	mmc-pwrseq = <&sdio_pwrseq>;
@@ -1209,3 +1227,46 @@
 	rockchip,dp-lane-mux = <2 3>;
 	status = "okay";
 };
+
+&hdmi1_sound {
+	status = "okay";
+};
+
+&hdmi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&hdmim0_tx1_cec &hdmim0_tx1_hpd &hdmim1_tx1_scl &hdmim1_tx1_sda>;
+	status = "okay";
+};
+
+&hdmi1_in {
+	hdmi1_in_vp1: endpoint {
+		remote-endpoint = <&vp1_out_hdmi1>;
+	};
+};
+
+&hdmi1_out {
+	hdmi1_out_con: endpoint {
+		remote-endpoint = <&hdmi1_con_in>;
+	};
+};
+
+&hdptxphy1 {
+	status = "okay";
+};
+
+
+&vop_mmu {
+	status = "okay";
+};
+
+&vop {
+	status = "okay";
+};
+
+&vp1 {
+	vp1_out_hdmi1: endpoint@ROCKCHIP_VOP2_EP_HDMI1 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI1>;
+		remote-endpoint = <&hdmi1_in_vp1>;
+	};
+};
+
