diff --speed-large-files --no-dereference --minimal -Naur linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi
--- linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi	2025-02-03 00:39:26.000000000 +0100
+++ linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi	2025-02-10 15:53:05.726701907 +0100
@@ -14,8 +14,8 @@
 
 / {
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
@@ -86,7 +86,7 @@
 		led_blue_pwm: led-1 {
 			color = <LED_COLOR_ID_BLUE>;
 			function = LED_FUNCTION_STATUS;
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "disk-activity";
 			max-brightness = <255>;
 			status = "disabled";
 		};
@@ -94,6 +94,7 @@
 		led_green_pwm: led-2 {
 			color = <LED_COLOR_ID_GREEN>;
 			function = LED_FUNCTION_INDICATOR;
+			linux,default-trigger = "heartbeat";
 			function-enumerator = <2>;
 			max-brightness = <255>;
 		};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
--- linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2025-02-03 00:39:26.000000000 +0100
+++ linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2025-02-10 15:55:13.860035673 +0100
@@ -26,11 +26,23 @@
 		};
 	};
 
+	hdmi1-con {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi1_con_in: endpoint {
+				remote-endpoint = <&hdmi1_out_con>;
+			};
+		};
+	};
+
 	ir-receiver {
 		compatible = "gpio-ir-receiver";
 		gpios = <&gpio4 RK_PB3 GPIO_ACTIVE_LOW>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&ir_receiver_pin>;
+		linux,rc-map-name = "rc-beelink-gs1";
 	};
 
 	rfkill {
@@ -109,6 +121,10 @@
 	pwms = <&pwm3 0 50000 0>;
 };
 
+&hdmi0_sound {
+	status = "okay";
+};
+
 &hdmi0 {
 	status = "okay";
 };
@@ -129,6 +145,30 @@
 	status = "okay";
 };
 
+&hdmi1_sound {
+	status = "okay";
+};
+
+&hdmi1 {
+	status = "okay";
+};
+
+&hdmi1_in {
+	hdmi1_in_vp1: endpoint {
+		remote-endpoint = <&vp1_out_hdmi1>;
+	};
+};
+
+&hdmi1_out {
+	hdmi1_out_con: endpoint {
+		remote-endpoint = <&hdmi1_con_in>;
+	};
+};
+
+&hdptxphy1 {
+	status = "okay";
+};
+
 &hym8563 {
 	interrupt-parent = <&gpio0>;
 	interrupts = <RK_PB0 IRQ_TYPE_LEVEL_LOW>;
@@ -190,12 +232,22 @@
 };
 
 &led_blue_gpio {
+	linux,default-trigger = "disk-activity";
 	gpios = <&gpio3 RK_PA6 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 };
 
 &led_green_pwm {
 	pwms = <&pwm2 0 25000 0>;
+	status = "okay";
+};
+
+&i2s5_8ch {
+	status = "okay";
+};
+
+&i2s6_8ch {
+	status = "okay";
 };
 
 /* phy1 - M.KEY socket */
@@ -342,3 +394,10 @@
 		remote-endpoint = <&hdmi0_in_vp0>;
 	};
 };
+
+&vp1 {
+	vp1_out_hdmi1: endpoint@ROCKCHIP_VOP2_EP_HDMI1 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI1>;
+		remote-endpoint = <&hdmi1_in_vp1>;
+	};
+};
