diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.4/drivers/clk/sunxi-ng/ccu-sun55i-a523.c linux-6.12.4/drivers/clk/sunxi-ng/ccu-sun55i-a523.c
--- linux-6.12.4/drivers/clk/sunxi-ng/ccu-sun55i-a523.c	2024-12-17 16:00:38.893717838 +0100
+++ linux-6.12.4/drivers/clk/sunxi-ng/ccu-sun55i-a523.c	2024-12-17 15:58:40.160381326 +0100
@@ -729,6 +729,27 @@
 				      0x970, BIT(31) | BIT(30), 6, 0);
 static SUNXI_CCU_GATE_HWS(bus_emac0_clk, "bus-emac0", ahb_hws, 0x97c,
 			  BIT(0), 0);
+/* GMAC1 BSP code
+static SUNXI_CCU_GATE(gmac1_25m_clk, "gmac1-25m",
+		      "dcxo24M",
+		      0x0974, BIT(31) | BIT(30), 0);
+
+static SUNXI_CCU_GATE_ASSOC(gmac1_clk, "gmac1",
+			    "dcxo24M",
+			    0x098C, BIT(17) | BIT(16) | BIT(0),
+			    0x0E04, BIT(21) | BIT(9),
+			    0x0);
+static SUNXI_CCU_GATE(gmac1_mbus_gate_clk, "gmac1-mbus-gate",
+			    "dcxo24M",
+			    0x0804, BIT(12), 0);
+*/
+static SUNXI_CCU_GATE_HWS_WITH_PREDIV(emac1_25M_clk, "emac1-25M",
+				      pll_periph0_150M_hws,
+				      0x974, BIT(31) | BIT(30), 6, 0); //is '6' correct one?
+static SUNXI_CCU_GATE_HWS(bus_emac1_clk, "bus-emac1", ahb_hws, 0x98c,
+			  BIT(17) | BIT(16) | BIT(0), 0);
+static SUNXI_CCU_GATE_HWS(mbus_emac1_clk, "mbus-emac1", ahb_hws, 0x804,
+			  BIT(12), 0);
 
 static const struct clk_parent_data ir_rx_parents[] = {
 	{ .fw_name = "losc" },
@@ -1226,6 +1247,9 @@
 	&bus_spifc_clk.common,
 	&emac0_25M_clk.common,
 	&bus_emac0_clk.common,
+	&emac1_25M_clk.common,
+	&bus_emac1_clk.common,
+	&mbus_emac1_clk.common,
 	&ir_rx_clk.common,
 	&bus_ir_rx_clk.common,
 	&ir_tx_clk.common,
@@ -1400,6 +1424,9 @@
 		[CLK_BUS_SPIFC]		= &bus_spifc_clk.common.hw,
 		[CLK_EMAC0_25M]		= &emac0_25M_clk.common.hw,
 		[CLK_BUS_EMAC0]		= &bus_emac0_clk.common.hw,
+		[CLK_EMAC1_25M]		= &emac1_25M_clk.common.hw,
+		[CLK_BUS_EMAC1]		= &bus_emac1_clk.common.hw,
+		[CLK_MBUS_EMAC1]	= &mbus_emac1_clk.common.hw,
 		[CLK_IR_RX]		= &ir_rx_clk.common.hw,
 		[CLK_BUS_IR_RX]		= &bus_ir_rx_clk.common.hw,
 		[CLK_IR_TX]		= &ir_tx_clk.common.hw,
diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.4/drivers/clk/sunxi-ng/ccu-sun55i-a523.h linux-6.12.4/drivers/clk/sunxi-ng/ccu-sun55i-a523.h
--- linux-6.12.4/drivers/clk/sunxi-ng/ccu-sun55i-a523.h	2024-12-17 16:00:38.893717838 +0100
+++ linux-6.12.4/drivers/clk/sunxi-ng/ccu-sun55i-a523.h	2024-12-17 16:33:11.167103493 +0100
@@ -9,6 +9,6 @@
 #include <dt-bindings/clock/sun55i-a523-ccu.h>
 #include <dt-bindings/reset/sun55i-a523-ccu.h>
 
-#define CLK_NUMBER	(CLK_FANOUT2 + 1)
+#define CLK_NUMBER	(CLK_MBUS_EMAC1 + 1)
 
 #endif /* _CCU_SUN55I_A523_H */
diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.4/include/dt-bindings/clock/sun55i-a523-ccu.h linux-6.12.4/include/dt-bindings/clock/sun55i-a523-ccu.h
--- linux-6.12.4/include/dt-bindings/clock/sun55i-a523-ccu.h	2024-12-17 16:00:38.860384503 +0100
+++ linux-6.12.4/include/dt-bindings/clock/sun55i-a523-ccu.h	2024-12-17 16:06:28.963727216 +0100
@@ -183,5 +183,6 @@
 #define CLK_FANOUT0		174
 #define CLK_FANOUT1		175
 #define CLK_FANOUT2		176
+#define CLK_MBUS_EMAC1		177
 
 #endif /* _DT_BINDINGS_CLK_SUN55I_A523_CCU_H_ */
