diff -Naur mythstream_mythtv-r21640-old/libs/harvester.h mythstream_mythtv-r21640-new/libs/harvester.h
--- mythstream_mythtv-r21640-old/libs/harvester.h	2009-08-30 14:46:35.000000000 -0700
+++ mythstream_mythtv-r21640-new/libs/harvester.h	2011-06-24 15:47:49.000000000 -0700
@@ -23,7 +23,8 @@
 #include <q3process.h>
 #include <qthread.h>
 #include <q3process.h>
-#include <qhttp.h>
+#include <QtNetwork/qhttp.h>
+#include <QTextStream>
 #include <q3dict.h>
 #include <qmap.h>
 //Added by qt3to4:
