diff -Naur mythstream_mythtv-r21640-old/libs/storagehandlers.cpp mythstream_mythtv-r21640-new/libs/storagehandlers.cpp
--- mythstream_mythtv-r21640-old/libs/storagehandlers.cpp	2009-09-02 07:06:24.000000000 -0700
+++ mythstream_mythtv-r21640-new/libs/storagehandlers.cpp	2011-08-17 13:01:49.000000000 -0700
@@ -34,7 +34,6 @@
 #include <q3ptrlist.h>
 //Added by qt3to4:
 #include <Q3CString>
-#include <QSqlQuery>
 #include <QSqlError>
 
 #include "storagehandlers.h"
@@ -726,6 +725,11 @@
 DatabaseStorage::DatabaseStorage(QString table, QString name,
                                  int accessType, int key_pri, int key_mul1, int key_mul2)
            : GenStorage(name, accessType, key_pri, key_mul1, key_mul2)
+#ifdef MYTHTV
+           , myQuery(MSqlQuery::InitCon())
+#else
+           , myQuery()
+#endif
 {
     defaultTable = table;
     defaultName  = name;
@@ -791,7 +795,11 @@
     accessNeedsCompact = false;
     accessReadOnly     = false;
 
+#ifdef MYTHTV
+    myQuery = MSqlQuery(MSqlQuery::InitCon());
+#else
     myQuery = QSqlQuery(myDb);
+#endif
     
     bool result = loadDbFields();
     emit storageEvent( ident, selected, !result );
@@ -799,7 +807,6 @@
 }
 
 #ifdef MYTHTV
-
 bool DatabaseStorage::openDefaultDb(int ident)
 {
     closeStorage(); // myDb = 0    
diff -Naur mythstream_mythtv-r21640-old/libs/storagehandlers.h mythstream_mythtv-r21640-new/libs/storagehandlers.h
--- mythstream_mythtv-r21640-old/libs/storagehandlers.h	2008-04-07 08:02:53.000000000 -0700
+++ mythstream_mythtv-r21640-new/libs/storagehandlers.h	2011-08-17 12:58:46.000000000 -0700
@@ -40,10 +40,8 @@
 #include <mythtv/mythdbcon.h>
 #else
 #include <qsqlquery.h>
-    typedef QSqlQuery MSqlQuery;
 #endif
 
-
 typedef Q3ValueVector<QString> ValueList;
 
 
@@ -253,7 +251,11 @@
     QString defaultName;     //
     
     QSqlDatabase myDb;  // databases other than mythconverg
+#ifdef MYTHTV
+    MSqlQuery myQuery;
+#else
     QSqlQuery myQuery;
+#endif
        
     bool loadDbFields();
     bool loadListFromDb();
