diff -Naur mythstream_mythtv-r21640-old/libs/streams.res mythstream_mythtv-r21640-new/libs/streams.res
--- mythstream_mythtv-r21640-old/libs/streams.res	1969-12-31 16:00:00.000000000 -0800
+++ mythstream_mythtv-r21640-new/libs/streams.res	2009-11-10 16:08:43.000000000 -0800
@@ -0,0 +1,281 @@
+
+
+[item]
+Interaction
+Apple movie trailers
+whatever
+trailers
+*apple
+
+[item]
+Interaction
+Browse Shoutcast genres
+http://www.shoutcast.com
+[emptystring]
+*shoutcast/menu
+
+[item]
+Interaction
+Uitzending gemist - laatste uitzendingen (Dutch)
+http://www.uitzendinggemist.nl
+Uitzending gemist selectie
+omroep/menu
+
+[item]
+Interaction
+World Wide Media Project
+[emptystring]
+[emptystring]
+*wwmp/menu
+
+[item]
+Interaction
+tv-links - Monty Python's Flying Circus
+http://tv-links.co.uk/listings/1/2526
+[emptystring]
+tvlinks/listing
+
+[item]
+Interaction
+wwiTV list
+http://wwitv.com/television/tvbar.htm
+List of television stations
+wwitv/countries
+
+[item]
+Podcast feeds
+Beercasting
+http://beercasting.com/rss.xml
+(podcast)
+podcast
+
+[item]
+Podcast feeds
+Feed me weird things
+http://del.icio.us/rss/audio/feedmeweirdthings
+(podcast) http://del.icio.us/rss/audio/feedmeweirdthings
+rdfcast
+
+[item]
+Podcast feeds
+SETI Institute
+http://podcast.seti.org/index.xml
+podcast
+podcast
+
+[item]
+Radio
+BBC World Service
+http://www.bbc.co.uk/worldservice/meta/tx/nb/live_news_au_nb.ram
+International
+[emptystring]
+
+[item]
+Radio
+DR R&B, Denmark
+http://www.dr.dk/netradio/Metafiler/ASX/DR_RogB_128.asx
+Copenhagen, Denmark
+[emptystring]
+
+[item]
+Radio
+Energy Black, Switzerland
+http://broadcast.infomaniak.ch/energyblack-high.mp3.m3u
+Zurich, Switzerland
+[emptystring]
+
+[item]
+Radio
+Fun Radio, belgium
+http://www.funradio.be/funradiobe-low.high.pls
+[emptystring]
+[emptystring]
+
+[item]
+Radio
+Juventus Radio
+http://live.juventus.hu:8000/live.m3u
+Budapest, Hungary
+[emptystring]
+
+[item]
+Radio
+Shoutcast Country
+http://www.shoutcast.com/directory/index.phtml?sgenre=Country&numresult=50
+Country music stations
+shoutcast
+
+[item]
+Radio
+Shoutcast Rock
+http://www.shoutcast.com/directory/index.phtml?sgenre=Rock&numresult=50
+Shoutcast Rock stations
+shoutcast
+
+[item]
+Radio
+The Voice, Norway
+http://213.158.233.199:6006/listen.pls
+Oslo, Norway
+[emptystring]
+
+[item]
+Radio
+WGBH, Boston
+http://streams.wgbh.org/classical.pls
+Boston, Massachusetts
+[emptystring]
+
+[item]
+Radio
+WholeWheatRadio stream
+http://64.62.252.134:2680/listen.pls
+[emptystring]
+[emptystring]
+
+[item]
+TV
+Holland Doc
+mms://livemedia.omroep.nl/vprohollanddoc-bb
+Nederland 4 themakanaal
+[emptystring]
+
+[item]
+TV
+NASA TV
+http://www.nasa.gov/ram/35037main_portal.ram
+[emptystring]
+[emptystring]
+
+[item]
+TV
+NRK News TV
+mms://media00.nrk.no/mms2/nydi/AutoEnc/mn1840.wmv
+Norwegen
+[emptystring]
+
+[item]
+TV
+National Nine News TV
+mms://broadband.netshow.ninemsn.com.au/broadband/news/story1.wmv
+Australia
+[emptystring]
+
+[item]
+TV
+RBC TV
+http://tv.gldn.net:80/rbc
+Russia
+[emptystring]
+
+[item]
+TV
+Tweede Kamer
+mms://213.75.12.78/TKStream3
+Dutch Parliament
+[emptystring]
+
+[item]
+TV
+Uitzending gemist RSS feed
+http://www.uitzendinggemist.nl/index.php/rss
+Laatste uitzendingen
+omroep/rss
+
+[item]
+TV
+United Nations TV
+rtsp://a4.l878435134.c8784.g.lr.akamaistream.net/live/D/4/8784/v0001/reflector:35134
+[emptystring]
+[emptystring]
+
+[item]
+TV
+VPro 3Voor12 Central TV
+mms://livemedia.omroep.nl/vpro3voor12tvcentral-bb
+http://3voor12.vpro.nl
+[emptystring]
+
+[item]
+TV
+VPro 3Voor12 on Stage TV
+mms://livemedia.omroep.nl/vpro3voor12tvonstage-bb
+[emptystring]
+[emptystring]
+
+[item]
+TV
+YLE 24 News TV
+http://ra.yle.fi/ramgen/yle24/video/uusin_rk.rm
+Finland
+[emptystring]
+
+[item]
+TV
+Yomiuri News TV
+http://www.yomiuri.co.jp/stream/vnews/vnews-w.asx
+Japan
+[emptystring]
+
+[item]
+Video
+DailyMotion custom search
+http://www.dailymotion.com/rss/relevance/search/<:search term::funny:>/<:page::1:>
+Edit search term with virtual keyboard
+dailymotion/playall
+
+[item]
+Video
+DailyMotion playlist demo: Alicia Keys Live
+http://www.dailymotion.com/rss/relevance/search/Alicia+Keys+live/1
+Mark streams (M) and play
+dailymotion/playall
+
+[item]
+Video
+Dailymotion download and play: VH1/1
+http://www.dailymotion.com/rss/relevance/search/vh1/1
+[emptystring]
+dailymotion/feed
+
+[item]
+Video
+Dailymotion playlist demo: Amy Winehouse
+http://www.dailymotion.com/rss/relevance/search/Amy+Winehouse+live/1
+Mark streams (M) and play
+dailymotion/playall
+
+[item]
+Video
+YouTube top viewed today
+http://youtube.com/rss/global/top_viewed_today.rss
+Voor andere feeds zie youtube
+youtube/feed
+
+[item]
+stream lists
+Icecast list
+http://dir.xiph.org/yp.xml
+Harvested with icecast.pl script
+icecast
+
+[item]
+stream lists
+OperaCast list
+http://www.operacast.com/opstations.htm
+Stationlist from www.operacast.com
+operacast
+
+[item]
+stream lists
+Operanut list (default parser)
+http://operanut.com/radio.htm
+Some dead streams
+[emptystring]
+
+[item]
+stream lists
+Robins BBC parser
+ask the parser :)
+BBC parsing by Robin Gilks
+*bbc/robins_links
