diff -Naur mythstream_mythtv-r21640-old/mythstream/main.cpp mythstream_mythtv-r21640-new/mythstream/main.cpp
--- mythstream_mythtv-r21640-old/mythstream/main.cpp	2009-09-02 18:40:49.000000000 +0200
+++ mythstream_mythtv-r21640-new/mythstream/main.cpp	2011-09-10 20:56:28.000000000 +0200
@@ -36,7 +36,7 @@
 #include <mythtv/mythmedia.h>
 #include <mythtv/mythpluginapi.h>
 #include <mythtv/libmythui/myththemedmenu.h>
-#include <mythtv/libmythdb/mythversion.h>
+#include <mythtv/libmythbase/mythversion.h>
 #include <mythtv/libmythui/mythuihelper.h>
 
 using namespace std;
