diff -Naur mythstream_mythtv-r21640-old/parsers/youtube/vid.pl mythstream_mythtv-r21640-new/parsers/youtube/vid.pl
--- mythstream_mythtv-r21640-old/parsers/youtube/vid.pl	2009-09-02 15:18:49.000000000 -0700
+++ mythstream_mythtv-r21640-new/parsers/youtube/vid.pl	2009-11-11 06:09:55.000000000 -0800
@@ -61,9 +61,11 @@
 
 my $a, $b;
 
-if ( ($a, $b) = $data =~ /swfArgs\s\=\s\{.*?"t":\s*"([^"]+)".+?"video_id":\s*"([^"]+)"/ )
+if ( ($a, $b) = $data =~ /'SWF_ARGS'\s*\:\s*\{.*?"t":\s*"([^"]+)".+?"video_id":\s*"([^"]+)"/ )
 {
-      $url = "http://www.youtube.com/get_video?video_id=$b&t=$a";
+      my $url = "http://www.youtube.com/get_video?video_id=$b&t=$a";
+
+      my $fmt_h264 = ($data =~ /'SWF_URL'\s*:\s*.*\.flash\.canPlayH264Videos()/);
 
       $item = $doc->createElement('item');
       $root->appendChild($item);
@@ -73,6 +75,17 @@
       $item->appendChild( newNode('descr', "") );
       $item->appendChild( newNode('handler', "") );
 
+      if ($fmt_h264)
+      {
+         $item = $doc->createElement('item');
+         $root->appendChild($item);
+         
+         $item->appendChild( newNode('name', "play: $sname [H.264]") );
+         $item->appendChild( newNode('url',  $url . '&fmt=18') );
+         $item->appendChild( newNode('descr', "") );
+         $item->appendChild( newNode('handler', "") );
+      }
+
       if (! ($playlist_enable == 1) )
       {
          $item = $doc->createElement('item');
@@ -82,6 +95,17 @@
          $item->appendChild( newNode('url',  $url) );
          $item->appendChild( newNode('descr', "") );
          $item->appendChild( newNode('handler', "STREAM_DL") );
+
+         if ($fmt_h264)
+         {
+            $item = $doc->createElement('item');
+            $root->appendChild($item);
+            
+            $item->appendChild( newNode('name', "download: $sname [H.264]") );
+            $item->appendChild( newNode('url',  $url . '&fmt=18') );
+            $item->appendChild( newNode('descr', "") );
+            $item->appendChild( newNode('handler', "STREAM_DL") );
+         }
       }
 }
 
diff -Naur mythstream_mythtv-r21640-old/parsers/youtube/youvid.pl mythstream_mythtv-r21640-new/parsers/youtube/youvid.pl
--- mythstream_mythtv-r21640-old/parsers/youtube/youvid.pl	2009-09-02 15:18:49.000000000 -0700
+++ mythstream_mythtv-r21640-new/parsers/youtube/youvid.pl	2009-11-11 06:10:22.000000000 -0800
@@ -65,10 +65,12 @@
 
 my $a, $b;
 
-if ( ($a, $b) = $data =~ /swfArgs\s\=\s\{.*?"t":\s*"([^"]+)".+?"video_id":\s*"([^"]+)"/ )
+if ( ($a, $b) = $data =~ /'SWF_ARGS'\s*\:\s*\{.*?"t":\s*"([^"]+)".+?"video_id":\s*"([^"]+)"/ )
 {
       $url = "http://www.youtube.com/get_video?video_id=$b&t=$a";
 
+      my $fmt_h264 = ($data =~ /'SWF_URL'\s*:\s*.*\.flash\.canPlayH264Videos()/);
+
       $item = $doc->createElement('item');
       $root->appendChild($item);
       
@@ -77,6 +79,17 @@
       $item->appendChild( newNode('descr', "") );
       $item->appendChild( newNode('handler', "") );
 
+      if ($fmt_h264)
+      {
+         $item = $doc->createElement('item');
+         $root->appendChild($item);
+         
+         $item->appendChild( newNode('name', "play: $sname [H.264]") );
+         $item->appendChild( newNode('url',  $url . '&fmt=18') );
+         $item->appendChild( newNode('descr', "") );
+         $item->appendChild( newNode('handler', "") );
+      }
+
       $item = $doc->createElement('item');
       $root->appendChild($item);
       
@@ -84,6 +97,17 @@
       $item->appendChild( newNode('url',  $url) );
       $item->appendChild( newNode('descr', "") );
       $item->appendChild( newNode('handler', "STREAM_DL") );
+
+      if ($fmt_h264)
+      {
+         $item = $doc->createElement('item');
+         $root->appendChild($item);
+         
+         $item->appendChild( newNode('name', "download: $sname [H.264]") );
+         $item->appendChild( newNode('url',  $url . '&fmt=18') );
+         $item->appendChild( newNode('descr', "") );
+         $item->appendChild( newNode('handler', "STREAM_DL") );
+      }
 }
 
 print $head->toString;
