diff -Naur mythstream_mythtv-r21640-old/libs/streamconfig.cpp mythstream_mythtv-r21640-new/libs/streamconfig.cpp
--- mythstream_mythtv-r21640-old/libs/streamconfig.cpp	2009-09-02 07:40:18.000000000 -0700
+++ mythstream_mythtv-r21640-new/libs/streamconfig.cpp	2010-05-17 11:59:31.000000000 -0700
@@ -46,7 +46,7 @@
   
 Window hierarchy
   StreamTunedConfig: main config window with "stream" and "storage" tabs  //StreamTuned only
-  gContext->GetMainWindow() // MythTV only
+  GetMythMainWindow() // MythTV only
     StreamConfig: stream configuration handler (stream tab)
       StorageGroup: StreamConfig goupBox with storage controls
       StreamEditGroup: StreamConfig groupBox with stream property edit boxes
diff -Naur mythstream_mythtv-r21640-old/mythstream/main.cpp mythstream_mythtv-r21640-new/mythstream/main.cpp
--- mythstream_mythtv-r21640-old/mythstream/main.cpp	2009-09-02 09:40:49.000000000 -0700
+++ mythstream_mythtv-r21640-new/mythstream/main.cpp	2010-05-17 11:59:08.000000000 -0700
@@ -57,12 +57,12 @@
 
     if (sel == "stream_config")
     {
-       StreamConfig config(gContext->GetMainWindow(), "stream", streamStorage, repos);
+       StreamConfig config(GetMythMainWindow(), "stream", streamStorage, repos);
        config.exec();
     }
     else if (sel == "storage_config")
     {
-       StorageConfig config(gContext->GetMainWindow(), "storage", streamStorage, repos);
+       StorageConfig config(GetMythMainWindow(), "storage", streamStorage, repos);
        config.exec();
     }
 
@@ -149,7 +149,7 @@
 void runStream(void)
 {
     GetMythUI()->AddCurrentLocation("mythstream");
-    MythStream stream(gContext->GetMainWindow(), "stream");
+    MythStream stream(GetMythMainWindow(), "stream");
     stream.exec();
     GetMythUI()->RemoveCurrentLocation( );
 }
diff -Naur mythstream_mythtv-r21640-old/mythstream/mythstream.cpp mythstream_mythtv-r21640-new/mythstream/mythstream.cpp
--- mythstream_mythtv-r21640-old/mythstream/mythstream.cpp	2009-09-02 14:22:33.000000000 -0700
+++ mythstream_mythtv-r21640-new/mythstream/mythstream.cpp	2010-05-17 11:58:37.000000000 -0700
@@ -499,7 +499,7 @@
       QString parameter;
       int res;
 
-      popBox = new MythPopupBox(gContext->GetMainWindow(), "");
+      popBox = new MythPopupBox(GetMythMainWindow(), "");
       popBox->addLabel(label);
 
       editBox = new MythLineEdit(popBox, "");
@@ -545,7 +545,7 @@
       QString parameter;
       int res;
       
-      popBox = new MythPopupBox(gContext->GetMainWindow(), "");
+      popBox = new MythPopupBox(GetMythMainWindow(), "");
       popBox->addLabel("                  Edit Stream Item                 ");
 
       MythLineEdit *folderBox = new MythLineEdit(popBox, "");
@@ -1295,7 +1295,7 @@
   bool handled;
   QStringList actions;
 
-  handled = gContext->GetMainWindow()->TranslateKeyPress("Stream", e, actions);
+  handled = GetMythMainWindow()->TranslateKeyPress("Stream", e, actions);
   handled = false;
   
   for (int i = 0; i < actions.size() && !handled; i++)
