diff -Naur mythstream_mythtv-r21640-old/mythstream/mythstream.pro mythstream_mythtv-r21640-new/mythstream/mythstream.pro
--- mythstream_mythtv-r21640-old/mythstream/mythstream.pro	2009-08-26 00:31:07.000000000 -0700
+++ mythstream_mythtv-r21640-new/mythstream/mythstream.pro	2011-07-24 07:28:08.000000000 -0700
@@ -41,3 +41,5 @@
 #The following line was inserted by qt3to4
 QT += xml  sql qt3support 
 
+QT += network
+
