diff -Naur which-2.20-old/tilde/tilde.c which-2.20-new/tilde/tilde.c
--- which-2.20-old/tilde/tilde.c	2008-01-16 09:51:57.000000000 -0800
+++ which-2.20-new/tilde/tilde.c	2008-08-29 21:29:17.000000000 -0700
@@ -196,7 +196,10 @@
   int result_size, result_index;
 
   result_index = result_size = 0;
-  if (result = strchr (string, '~'))
+
+  result = strchr (string, '~');
+
+  if (!result)
     result = (char *)xmalloc (result_size = (strlen (string) + 16));
   else
     result = (char *)xmalloc (result_size = (strlen (string) + 1));
