diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/dts/sun50i-h616.dtsi u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/dts/sun50i-h616.dtsi
--- u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/dts/sun50i-h616.dtsi	2023-03-06 00:33:14.000000000 +0100
+++ u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/dts/sun50i-h616.dtsi	2023-03-16 10:09:59.978279120 +0100
@@ -168,6 +168,13 @@
 				drive-strength = <40>;
 			};
 
+			rmii_pins: rmii-pins {
+				pins = "PA0", "PA1", "PA2", "PA3", "PA4",
+				       "PA5", "PA6", "PA7", "PA8", "PA9";
+				function = "emac2";
+				drive-strength = <40>;
+			};
+
 			i2c0_pins: i2c0-pins {
 				pins = "PI6", "PI7";
 				function = "i2c0";
@@ -501,6 +508,26 @@
 				compatible = "snps,dwmac-mdio";
 				#address-cells = <1>;
 				#size-cells = <0>;
+			};
+		};
+
+		emac1: ethernet@5030000 {
+			compatible = "allwinner,sun50i-h616-emac0",
+				     "allwinner,sun50i-a64-emac";
+			reg = <0x05030000 0x10000>;
+			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "macirq";
+			clocks = <&ccu CLK_BUS_EMAC1>;
+			clock-names = "stmmaceth";
+			resets = <&ccu RST_BUS_EMAC1>;
+			reset-names = "stmmaceth";
+			syscon = <&syscon 1>;
+			status = "disabled";
+
+			mdio1: mdio {
+				compatible = "snps,dwmac-mdio";
+				#address-cells = <1>;
+				#size-cells = <0>;
 			};
 		};
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/board.c u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/board.c
--- u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/board.c	2023-03-06 00:33:14.000000000 +0100
+++ u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/board.c	2023-03-16 10:48:02.601528960 +0100
@@ -460,6 +460,7 @@
 	/* Needed early by sunxi_board_init if PMU is enabled */
 	i2c_init_board();
 	i2c_init(CONFIG_SYS_I2C_SPEED, CONFIG_SYS_I2C_SLAVE);
+	i2c_set_bus_num(1);
 #endif
 	sunxi_board_init();
 }
diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/Kconfig u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/Kconfig
--- u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/Kconfig	2023-03-06 00:33:14.000000000 +0100
+++ u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/Kconfig	2023-03-16 10:48:02.601528960 +0100
@@ -726,6 +726,12 @@
 	---help---
 	See I2C0_ENABLE help text.
 
+config I2C3_ENABLE
+	bool "Enable I2C/TWI controller 3"
+	select CMD_I2C
+	---help---
+	See I2C0_ENABLE help text.
+
 if SUNXI_GEN_SUN6I || SUN50I_GEN_H6
 config R_I2C_ENABLE
 	bool "Enable the PRCM I2C/TWI controller"
diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/board/sunxi/board.c u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/board/sunxi/board.c
--- u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/board/sunxi/board.c	2023-03-06 00:33:14.000000000 +0100
+++ u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/board/sunxi/board.c	2023-03-16 10:48:02.601528960 +0100
@@ -15,6 +15,7 @@
 #include <dm.h>
 #include <env.h>
 #include <hang.h>
+#include <i2c.h>
 #include <image.h>
 #include <init.h>
 #include <log.h>
@@ -107,6 +108,17 @@
 #endif
 #endif
 
+#ifdef CONFIG_I2C3_ENABLE
+#if defined(CONFIG_MACH_SUN50I_H616)
+	sunxi_gpio_set_cfgpin(SUNXI_GPA(10), 2);
+	sunxi_gpio_set_cfgpin(SUNXI_GPA(11), 2);
+	sunxi_gpio_set_cfgpin(SUNXI_GPA(12), 2);
+	sunxi_gpio_set_pull(SUNXI_GPA(10), SUNXI_GPIO_PULL_UP);
+	sunxi_gpio_set_pull(SUNXI_GPA(11), SUNXI_GPIO_PULL_UP);
+	clock_twi_onoff(3, 1);
+#endif
+#endif
+
 #ifdef CONFIG_R_I2C_ENABLE
 #ifdef CONFIG_MACH_SUN50I
 	clock_twi_onoff(5, 1);
@@ -575,6 +587,7 @@
 void sunxi_board_init(void)
 {
 	int power_failed = 0;
+	u8 data[2];
 
 #ifdef CONFIG_LED_STATUS
 	if (IS_ENABLED(CONFIG_SPL_DRIVERS_MISC))
@@ -669,6 +682,23 @@
 		clock_set_pll1(get_board_sys_clk());
 	else
 		printf("Failed to set core voltage! Can't set CPU frequency\n");
+
+	i2c_set_bus_num(0);
+	data[0] = 0;
+	data[1] = 0;
+	i2c_write(0x10, 0xfe, 1, data, 2);
+	i2c_write(0x10, 2, 1, data, 2);
+	data[1] = 1;
+	i2c_write(0x10, 2, 1, data, 2);
+	data[1] = 0xf;
+	i2c_write(0x10, 0x16, 1, data, 2);
+	data[1] = 3;
+	i2c_write(0x10, 0x14, 1, data, 2);
+	data[1] = 0x60;
+	i2c_write(0x10, 0xfe, 1, data, 2);
+	data[0] = 0x08;
+	data[1] = 0x14;
+	i2c_write(0x10, 0, 1, data, 2);
 }
 #endif /* CONFIG_SPL_BUILD */
 
