diff --speed-large-files --no-dereference --minimal -Naur u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/include/asm/arch-sunxi/boot0.h u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/include/asm/arch-sunxi/boot0.h
--- u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/include/asm/arch-sunxi/boot0.h	2022-12-12 18:51:28.000000000 +0100
+++ u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/include/asm/arch-sunxi/boot0.h	2023-03-24 18:37:57.461748842 +0100
@@ -42,7 +42,7 @@
 #ifndef CONFIG_SUN50I_GEN_H6
 	.word	0x017000a0	// writeable RVBAR mapping address
 #else
-	.word	0x09010040	// writeable RVBAR mapping address
+	.word	0x08100040	// writeable RVBAR mapping address
 #endif
 #ifdef CONFIG_SPL_BUILD
 	.word	CONFIG_SPL_TEXT_BASE
