diff --speed-large-files --no-dereference --minimal -Naur u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c
--- u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c	2022-12-12 18:51:28.000000000 +0100
+++ u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c	2023-03-28 13:29:21.341227856 +0200
@@ -12,6 +12,7 @@
  * (C) Copyright 2020 Jernej Skrabec <jernej.skrabec@siol.net>
  *
  */
+
 #include <common.h>
 #include <init.h>
 #include <log.h>
@@ -827,20 +828,20 @@
 		val = 3;
 	clrsetbits_le32(SUNXI_DRAM_PHY0_BASE + 0x3c, 0xf, val);
 
-	writel(0xd, SUNXI_DRAM_PHY0_BASE + 0x14);
-	writel(0xd, SUNXI_DRAM_PHY0_BASE + 0x35c);
-	writel(0xd, SUNXI_DRAM_PHY0_BASE + 0x368);
-	writel(0xd, SUNXI_DRAM_PHY0_BASE + 0x374);
+	writel(0x9, SUNXI_DRAM_PHY0_BASE + 0x14);
+	writel(0x9, SUNXI_DRAM_PHY0_BASE + 0x35c);
+	writel(0x9, SUNXI_DRAM_PHY0_BASE + 0x368);
+	writel(0x9, SUNXI_DRAM_PHY0_BASE + 0x374);
 
 	writel(0, SUNXI_DRAM_PHY0_BASE + 0x18);
 	writel(0, SUNXI_DRAM_PHY0_BASE + 0x360);
 	writel(0, SUNXI_DRAM_PHY0_BASE + 0x36c);
 	writel(0, SUNXI_DRAM_PHY0_BASE + 0x378);
 
-	writel(9, SUNXI_DRAM_PHY0_BASE + 0x1c);
-	writel(9, SUNXI_DRAM_PHY0_BASE + 0x364);
-	writel(9, SUNXI_DRAM_PHY0_BASE + 0x370);
-	writel(9, SUNXI_DRAM_PHY0_BASE + 0x37c);
+	writel(7, SUNXI_DRAM_PHY0_BASE + 0x1c);
+	writel(7, SUNXI_DRAM_PHY0_BASE + 0x364);
+	writel(7, SUNXI_DRAM_PHY0_BASE + 0x370);
+	writel(7, SUNXI_DRAM_PHY0_BASE + 0x37c);
 
 	ptr = (u32 *)(SUNXI_DRAM_PHY0_BASE + 0xc0);
 	for (i = 0; i < ARRAY_SIZE(phy_init); i++)
@@ -896,7 +897,7 @@
 	writel(1, &mctl_ctl->swctl);
 	mctl_await_completion(&mctl_ctl->swstat, 1, 1);
 
-	writel(0x1f14, &mctl_ctl->mrctrl1);
+	writel(0x1b50, &mctl_ctl->mrctrl1);
 	writel(0x80000030, &mctl_ctl->mrctrl0);
 	mctl_await_completion(&mctl_ctl->mrctrl0, BIT(31), 0);
 
@@ -904,7 +905,7 @@
 	writel(0x80001030, &mctl_ctl->mrctrl0);
 	mctl_await_completion(&mctl_ctl->mrctrl0, BIT(31), 0);
 
-	writel(0x20, &mctl_ctl->mrctrl1);
+	writel(0x10, &mctl_ctl->mrctrl1);
 	writel(0x80002030, &mctl_ctl->mrctrl0);
 	mctl_await_completion(&mctl_ctl->mrctrl0, BIT(31), 0);
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_timings/h616_ddr3_1333.c u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_timings/h616_ddr3_1333.c
--- u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_timings/h616_ddr3_1333.c	2022-12-12 18:51:28.000000000 +0100
+++ u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_timings/h616_ddr3_1333.c	2023-03-28 13:25:42.537902526 +0200
@@ -85,8 +85,9 @@
 	clrsetbits_le32(&mctl_ctl->rankctl, 0xff0, 0x660);
 
 	/* Configure DFI timing */
-	writel((tcl - 2) | 0x2000000 | (t_rdata_en << 16) | 0x808000,
-	       &mctl_ctl->dfitmg0);
+	//writel((tcl - 2) | 0x2000000 | (t_rdata_en << 16) | 0x808000,
+	//       &mctl_ctl->dfitmg0);
+	writel(0x2858003, &mctl_ctl->dfitmg0);
 	writel(0x100202, &mctl_ctl->dfitmg1);
 
 	/* set refresh timing */
