diff --speed-large-files --no-dereference --minimal -Naur u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c
--- u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c	2023-03-27 18:30:58.168394379 +0200
+++ u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c	2023-03-27 16:15:22.558075108 +0200
@@ -1095,6 +1095,28 @@
 	if (mctl_core_init(para))
 		return;
 
+	uint32_t* base=0x047FA000u;
+	int a;
+	printf("=> md.l 0x047FA000 \n");
+	for( a = 0; a < 82; a = a + 1 ){
+		printf("%08x: %08x %08x %08x %08x\n",base,*base,*(base+0x1),*(base+0x2),*(base+0x3));
+		base=base+0x4u;
+	}
+
+	base=0x047FB000u;
+	printf("=> md.l 0x047FB000 \n");
+	for( a = 0; a < 51; a = a + 1 ){
+		printf("%08x: %08x %08x %08x %08x\n",base,*base,*(base+0x1),*(base+0x2),*(base+0x3));
+		base=base+0x4u;
+	}
+
+	base=0x04800000u;
+	printf("=> md.l 0x04800000 \n");
+	for( a = 0; a < 184; a = a + 1 ){
+		printf("%08x: %08x %08x %08x %08x\n",base,*base,*(base+0x1),*(base+0x2),*(base+0x3));
+		base=base+0x4u;
+	}
+
 	panic("This DRAM setup is currently not supported.\n");
 }
 
