diff -Naur libcdaudio-0.99.12p2-old/configure.ac libcdaudio-0.99.12p2-new/configure.ac
--- libcdaudio-0.99.12p2-old/configure.ac	2013-01-03 20:03:42.000000000 -0800
+++ libcdaudio-0.99.12p2-new/configure.ac	2013-01-03 20:04:16.000000000 -0800
@@ -12,12 +12,11 @@
 AM_INIT_AUTOMAKE($PACKAGE, ${LIBCDAUDIO_VERSION_STRING})
 AM_MAINTAINER_MODE
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_PROG_AWK
 AC_PROG_CC
 AC_PROG_CPP
-AM_PROG_CC_STDC
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
