diff -Naur libcdaudio-0.99.12p2-old/configure.ac libcdaudio-0.99.12p2-new/configure.ac
--- libcdaudio-0.99.12p2-old/configure.ac	2005-08-15 23:02:52.000000000 -0700
+++ libcdaudio-0.99.12p2-new/configure.ac	2012-07-17 06:49:51.000000000 -0700
@@ -17,7 +17,6 @@
 AC_PROG_AWK
 AC_PROG_CC
 AC_PROG_CPP
-AM_C_PROTOTYPES
 AM_PROG_CC_STDC
 AC_PROG_INSTALL
 AC_PROG_LN_S
diff -Naur libcdaudio-0.99.12p2-old/Makefile.am libcdaudio-0.99.12p2-new/Makefile.am
--- libcdaudio-0.99.12p2-old/Makefile.am	2004-10-02 10:50:59.000000000 -0700
+++ libcdaudio-0.99.12p2-new/Makefile.am	2012-07-17 06:51:49.000000000 -0700
@@ -1,5 +1,4 @@
 ## Process this file with automake to produce Makefile.in
-AUTOMAKE_OPTIONS = ansi2knr
 SUBDIRS = src
 
 bin_SCRIPTS = libcdaudio-config
