
This add AXP313A PMIC support

diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/arch/arm/mach-sunxi/pmic_bus.c u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/arch/arm/mach-sunxi/pmic_bus.c
--- u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/arch/arm/mach-sunxi/pmic_bus.c	2021-04-26 11:12:35.000000000 +0200
+++ u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/arch/arm/mach-sunxi/pmic_bus.c	2023-04-02 15:22:33.341314955 +0200
@@ -66,7 +66,7 @@
 	return i2c_read(AXP152_I2C_ADDR, reg, 1, data, 1);
 #elif defined CONFIG_AXP209_POWER
 	return i2c_read(AXP209_I2C_ADDR, reg, 1, data, 1);
-#elif defined CONFIG_AXP305_POWER
+#elif (IS_ENABLED(CONFIG_AXP305_POWER) || IS_ENABLED(CONFIG_AXP313_POWER))
 	return i2c_read(AXP305_I2C_ADDR, reg, 1, data, 1);
 #elif defined CONFIG_AXP221_POWER || defined CONFIG_AXP809_POWER || defined CONFIG_AXP818_POWER
 # ifdef CONFIG_MACH_SUN6I
@@ -85,7 +85,7 @@
 	return i2c_write(AXP152_I2C_ADDR, reg, 1, &data, 1);
 #elif defined CONFIG_AXP209_POWER
 	return i2c_write(AXP209_I2C_ADDR, reg, 1, &data, 1);
-#elif defined CONFIG_AXP305_POWER
+#elif (IS_ENABLED(CONFIG_AXP305_POWER) || IS_ENABLED(CONFIG_AXP313_POWER))
 	return i2c_write(AXP305_I2C_ADDR, reg, 1, &data, 1);
 #elif defined CONFIG_AXP221_POWER || defined CONFIG_AXP809_POWER || defined CONFIG_AXP818_POWER
 # ifdef CONFIG_MACH_SUN6I
diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/board/sunxi/board.c u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/board/sunxi/board.c
--- u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/board/sunxi/board.c	2023-04-02 15:24:35.877977156 +0200
+++ u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/board/sunxi/board.c	2023-04-02 15:08:34.598012262 +0200
@@ -663,7 +663,8 @@
 
 #if defined CONFIG_AXP152_POWER || defined CONFIG_AXP209_POWER || \
 	defined CONFIG_AXP221_POWER || defined CONFIG_AXP305_POWER || \
-	defined CONFIG_AXP809_POWER || defined CONFIG_AXP818_POWER
+	defined CONFIG_AXP313_POWER || defined CONFIG_AXP809_POWER || \
+	defined CONFIG_AXP818_POWER
 	power_failed = axp_init();
 
 #if defined CONFIG_AXP221_POWER || defined CONFIG_AXP809_POWER || \
@@ -671,10 +672,12 @@
 	power_failed |= axp_set_dcdc1(CONFIG_AXP_DCDC1_VOLT);
 #endif
 #if !defined(CONFIG_AXP305_POWER)
+#if !defined CONFIG_AXP313_POWER
 	power_failed |= axp_set_dcdc2(CONFIG_AXP_DCDC2_VOLT);
+#endif
 	power_failed |= axp_set_dcdc3(CONFIG_AXP_DCDC3_VOLT);
 #endif
-#if !defined(CONFIG_AXP209_POWER) && !defined(CONFIG_AXP818_POWER)
+#if !defined(CONFIG_AXP209_POWER) && !defined(CONFIG_AXP818_POWER) && !defined(CONFIG_AXP313_POWER)
 	power_failed |= axp_set_dcdc4(CONFIG_AXP_DCDC4_VOLT);
 #endif
 #if defined CONFIG_AXP221_POWER || defined CONFIG_AXP809_POWER || \
@@ -686,10 +689,10 @@
 	defined CONFIG_AXP818_POWER
 	power_failed |= axp_set_aldo1(CONFIG_AXP_ALDO1_VOLT);
 #endif
-#if !defined(CONFIG_AXP305_POWER)
+#if !defined(CONFIG_AXP305_POWER) && !defined(CONFIG_AXP313_POWER)
 	power_failed |= axp_set_aldo2(CONFIG_AXP_ALDO2_VOLT);
 #endif
-#if !defined(CONFIG_AXP152_POWER) && !defined(CONFIG_AXP305_POWER)
+#if !defined(CONFIG_AXP152_POWER) && !defined(CONFIG_AXP305_POWER) && !defined(CONFIG_AXP313_POWER)
 	power_failed |= axp_set_aldo3(CONFIG_AXP_ALDO3_VOLT);
 #endif
 #ifdef CONFIG_AXP209_POWER
diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/axp313.c u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/axp313.c
--- u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/axp313.c	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/axp313.c	2023-04-02 15:08:34.598012262 +0200
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * AXP305 driver
+ *
+ * (C) Copyright 2020 Jernej Skrabec <jernej.skrabec@siol.net>
+ *
+ * Based on axp221.c
+ * (C) Copyright 2014 Hans de Goede <hdegoede@redhat.com>
+ * (C) Copyright 2013 Oliver Schinagl <oliver@schinagl.nl>
+ */
+
+#include <common.h>
+#include <command.h>
+#include <errno.h>
+#include <asm/arch/pmic_bus.h>
+#include <axp_pmic.h>
+
+#define AXP313_DCDC3_1220MV_OFFSET 71
+
+static u8 axp313_mvolt_to_cfg(int mvolt, int min, int max, int div)
+{
+	if (mvolt < min)
+		mvolt = min;
+	else if (mvolt > max)
+		mvolt = max;
+
+	return  (mvolt - min) / div;
+}
+
+int axp_set_dcdc3(unsigned int mvolt)
+{
+	int ret;
+	u8 cfg;
+
+	ret = pmic_bus_read(AXP313_DCDC3_VOLTAGE, &cfg);
+
+	if (mvolt >= 1220)
+		cfg = AXP313_DCDC3_1220MV_OFFSET +
+			axp313_mvolt_to_cfg(mvolt, 1220, 1860, 20);
+	else
+		cfg = axp313_mvolt_to_cfg(mvolt, 500, 1200, 10);
+	if (mvolt == 0)
+		return pmic_bus_clrbits(AXP313_OUTPUT_CTRL1,
+					AXP313_OUTPUT_CTRL1_DCDC3_EN);
+
+	ret = pmic_bus_write(AXP313_DCDC3_VOLTAGE, cfg);
+	if (ret){
+		return ret;
+	}		
+
+	return pmic_bus_setbits(AXP313_OUTPUT_CTRL1,
+				AXP313_OUTPUT_CTRL1_DCDC3_EN);
+}
+
+int axp_init(void)
+{
+	u8 axp_chip_id;
+	int ret;
+	ret = pmic_bus_init();
+	if (ret)
+		return ret;
+
+	ret = pmic_bus_read(AXP313_CHIP_VERSION, &axp_chip_id);
+	if (ret)
+		return ret;
+	if (axp_chip_id != 0x4B)
+		return -ENODEV;
+	return ret;
+}
+
+
diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/Kconfig u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/Kconfig
--- u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/Kconfig	2021-04-26 11:12:35.000000000 +0200
+++ u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/Kconfig	2023-04-02 15:08:34.598012262 +0200
@@ -13,7 +13,7 @@
 	depends on ARCH_SUNXI
 	default AXP209_POWER if MACH_SUN4I || MACH_SUN5I || MACH_SUN7I
 	default AXP221_POWER if MACH_SUN6I || MACH_SUN8I_A23 || MACH_SUN8I_A33 || MACH_SUN8I_R40
-	default AXP305_POWER if MACH_SUN50I_H616
+	default AXP313_POWER if MACH_SUN50I_H616
 	default AXP818_POWER if MACH_SUN8I_A83T
 	default SUNXI_NO_PMIC if MACH_SUNXI_H3_H5 || MACH_SUN50I || MACH_SUN8I_V3S
 
@@ -58,6 +58,15 @@
 	Select this to enable support for the axp305 pmic found on most
 	H616 boards.
 
+config AXP313_POWER
+	bool "axp313 pmic support"
+	depends on MACH_SUN50I_H616
+	select AXP_PMIC_BUS
+	select CMD_POWEROFF
+	---help---
+	Select this to enable support for the axp313 pmic found on most
+	H616 boards.
+
 config AXP809_POWER
 	bool "axp809 pmic support"
 	depends on MACH_SUN9I
@@ -118,12 +127,13 @@
 
 config AXP_DCDC3_VOLT
 	int "axp pmic dcdc3 voltage"
-	depends on AXP152_POWER || AXP209_POWER || AXP221_POWER || AXP809_POWER || AXP818_POWER
+	depends on AXP152_POWER || AXP209_POWER || AXP221_POWER || AXP313_POWER || AXP809_POWER || AXP818_POWER
 	default 900 if AXP809_POWER || AXP818_POWER
 	default 1500 if AXP152_POWER
 	default 1250 if AXP209_POWER
 	default 1100 if MACH_SUN8I_R40
 	default 1200 if MACH_SUN6I || MACH_SUN8I
+	default 1200 if AXP313_POWER
 	---help---
 	Set the voltage (mV) to program the axp pmic dcdc3 at, set to 0 to
 	disable dcdc3.
@@ -134,6 +144,7 @@
 	On A80 boards dcdc3 is used for VDD-CPUA(cluster 0) and should be 0.9V.
 	On A83T boards dcdc3 is used for VDD-CPUB(cluster 1) and should be 0.9V.
 	On R40 boards dcdc3 is VDD-SYS and VDD-GPU and should be 1.1V.
+	On H616 boards dcdc3 is VDD-DRAM and for LPDDR3 should be 1.2V.
 
 config AXP_DCDC4_VOLT
 	int "axp pmic dcdc4 voltage"
diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/Makefile u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/Makefile
--- u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/Makefile	2021-04-26 11:12:35.000000000 +0200
+++ u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/drivers/power/Makefile	2023-04-02 15:08:34.598012262 +0200
@@ -7,6 +7,7 @@
 obj-$(CONFIG_AXP209_POWER)	+= axp209.o
 obj-$(CONFIG_AXP221_POWER)	+= axp221.o
 obj-$(CONFIG_AXP305_POWER)	+= axp305.o
+obj-$(CONFIG_AXP313_POWER)	+= axp313.o
 obj-$(CONFIG_AXP809_POWER)	+= axp809.o
 obj-$(CONFIG_AXP818_POWER)	+= axp818.o
 obj-$(CONFIG_EXYNOS_TMU)	+= exynos-tmu.o
diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/include/axp313.h u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/include/axp313.h
--- u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/include/axp313.h	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/include/axp313.h	2023-04-02 15:08:34.598012262 +0200
@@ -0,0 +1,20 @@
+/* SPDX-License-Identifier: GPL-2.0+ */
+/*
+ * (C) Copyright 2020 Jernej Skrabec <jernej.skrabec@siol.net>
+ */
+
+enum axp313_reg {
+	AXP313_CHIP_VERSION = 0x3,
+	AXP313_OUTPUT_CTRL1 = 0x10,
+	AXP313_DCDC3_VOLTAGE = 0x15,
+	// AXP313_SHUTDOWN = 0x32,
+};
+
+#define AXP313_CHIP_VERSION_MASK	0xcf
+
+#define AXP313_OUTPUT_CTRL1_DCDC3_EN	(1 << 2)
+
+// #define AXP313_POWEROFF			(1 << 7)
+
+// #define AXP_POWER_STATUS		0x00
+// #define AXP_POWER_STATUS_ALDO_IN		BIT(0)
diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/include/axp_pmic.h u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/include/axp_pmic.h
--- u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/include/axp_pmic.h	2021-04-26 11:12:35.000000000 +0200
+++ u-boot-sunxi-647b392bf20614006917d5fcd3390347a668eee5/include/axp_pmic.h	2023-04-02 15:24:10.671311412 +0200
@@ -24,6 +24,9 @@
 #ifdef CONFIG_AXP818_POWER
 #include <axp818.h>
 #endif
+#ifdef CONFIG_AXP313_POWER
+#include <axp313.h>
+#endif
 
 int axp_set_dcdc1(unsigned int mvolt);
 int axp_set_dcdc2(unsigned int mvolt);
