diff --speed-large-files --no-dereference --minimal -Naur u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/clock_sun50i_h6.c u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/clock_sun50i_h6.c
--- u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/clock_sun50i_h6.c	2023-03-14 11:14:39.374856217 +0100
+++ u-boot-sunxi-8e2c0ee3bafbc283b58b66d9007847a5a3ed07be/arch/arm/mach-sunxi/clock_sun50i_h6.c	2023-03-14 11:14:24.974856786 +0100
@@ -16,6 +16,8 @@
 		/* this seems to enable PLLs on H616 */
 		setbits_le32(&prcm->sys_pwroff_gating, 0x10);
 		setbits_le32(&prcm->res_cal_ctrl, 2);
+		/* enable GPU */
+		writel(0, 0x7010254);
 	}
 
 	clrbits_le32(&prcm->res_cal_ctrl, 1);
