diff --speed-large-files --no-dereference --minimal -Naur u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/arch/arm/mach-sunxi/dram_sun50i_h616.c u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/arch/arm/mach-sunxi/dram_sun50i_h616.c
--- u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/arch/arm/mach-sunxi/dram_sun50i_h616.c	2023-03-29 09:57:28.000000000 +0200
+++ u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/arch/arm/mach-sunxi/dram_sun50i_h616.c	2023-04-01 12:52:21.614857948 +0200
@@ -1360,5 +1360,27 @@
 
 	mctl_set_master_priority();
 
+	uint32_t* base=0x047FA000u;
+	int a;
+	printf("=> md.l 0x047FA000 \n");
+	for( a = 0; a < 82; a = a + 1 ){
+		printf("%08x: %08x %08x %08x %08x\n",base,*base,*(base+0x1),*(base+0x2),*(base+0x3));
+		base=base+0x4u;
+	}
+
+	base=0x047FB000u;
+	printf("=> md.l 0x047FB000 \n");
+	for( a = 0; a < 51; a = a + 1 ){
+		printf("%08x: %08x %08x %08x %08x\n",base,*base,*(base+0x1),*(base+0x2),*(base+0x3));
+		base=base+0x4u;
+	}
+
+	base=0x04800000u;
+	printf("=> md.l 0x04800000 \n");
+	for( a = 0; a < 184; a = a + 1 ){
+		printf("%08x: %08x %08x %08x %08x\n",base,*base,*(base+0x1),*(base+0x2),*(base+0x3));
+		base=base+0x4u;
+	}
+
 	return size;
 };
