diff --speed-large-files --no-dereference --minimal -Naur u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/arch/arm/mach-sunxi/board.c u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/arch/arm/mach-sunxi/board.c
--- u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/arch/arm/mach-sunxi/board.c	2023-04-01 12:22:36.794923070 +0200
+++ u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/arch/arm/mach-sunxi/board.c	2023-04-01 12:21:28.931592188 +0200
@@ -9,6 +9,8 @@
  * Some init for sunxi platform.
  */
 
+#define DEBUG
+
 #include <common.h>
 #include <cpu_func.h>
 #include <init.h>
diff --speed-large-files --no-dereference --minimal -Naur u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/board/sunxi/board.c u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/board/sunxi/board.c
--- u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/board/sunxi/board.c	2023-04-01 12:22:36.794923070 +0200
+++ u-boot-b5be74120d4aee645c63aea40c91f315e9513d68/board/sunxi/board.c	2023-04-01 12:21:28.931592188 +0200
@@ -10,6 +10,8 @@
  * Some board init for the Allwinner A10-evb board.
  */
 
+#define DEBUG
+
 #include <common.h>
 #include <clock_legacy.h>
 #include <dm.h>
