diff --speed-large-files --no-dereference --minimal -Naur linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3328.dtsi linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3328.dtsi
--- linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3328.dtsi	2026-03-13 11:27:02.040700020 +0100
+++ linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3328.dtsi	2026-03-13 11:23:23.534615374 +0100
@@ -1156,7 +1156,7 @@
 	};
 
 	usbdrd3: usb@ff600000 {
-		compatible = "rockchip,rk3328-dwc3", "snps,dwc3";
+		compatible = "rockchip,rk3328-dwc3", "snps,dwc3", "snps,dwc3-simple";
 		reg = <0x0 0xff600000 0x0 0x100000>;
 		interrupts = <GIC_SPI 67 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru SCLK_USB3OTG_REF>, <&cru SCLK_USB3OTG_SUSPEND>,
