diff --speed-large-files --no-dereference --minimal -Naur linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk356x-base.dtsi linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk356x-base.dtsi
--- linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk356x-base.dtsi	2026-03-13 11:27:03.974241461 +0100
+++ linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk356x-base.dtsi	2026-03-13 11:21:55.372002871 +0100
@@ -260,7 +260,7 @@
 	};
 
 	usb_host0_xhci: usb@fcc00000 {
-		compatible = "rockchip,rk3568-dwc3", "snps,dwc3";
+		compatible = "rockchip,rk3568-dwc3", "snps,dwc3", "snps,dwc3-simple";
 		reg = <0x0 0xfcc00000 0x0 0x400000>;
 		interrupts = <GIC_SPI 169 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru CLK_USB3OTG0_REF>, <&cru CLK_USB3OTG0_SUSPEND>,
@@ -276,7 +276,7 @@
 	};
 
 	usb_host1_xhci: usb@fd000000 {
-		compatible = "rockchip,rk3568-dwc3", "snps,dwc3";
+		compatible = "rockchip,rk3568-dwc3", "snps,dwc3", "snps,dwc3-simple";
 		reg = <0x0 0xfd000000 0x0 0x400000>;
 		interrupts = <GIC_SPI 170 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru CLK_USB3OTG1_REF>, <&cru CLK_USB3OTG1_SUSPEND>,
@@ -294,7 +294,7 @@
 	};
 
 	usbdrd30: usbdrd {
-		compatible = "rockchip,rk3568-dwc3", "rockchip,rk3399-dwc3", "snps,dwc3";
+		compatible = "rockchip,rk3568-dwc3", "rockchip,rk3399-dwc3", "snps,dwc3", "snps,dwc3-simple";
 		reg = <0x0 0xfcc00000 0x0 0x400000>;
 		interrupts = <GIC_SPI 169 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru CLK_USB3OTG0_REF>, <&cru CLK_USB3OTG0_SUSPEND>,
@@ -315,7 +315,7 @@
 	};
 
 	usbhost30: usbhost {
-		compatible = "rockchip,rk3568-dwc3", "rockchip,rk3399-dwc3", "snps,dwc3";
+		compatible = "rockchip,rk3568-dwc3", "rockchip,rk3399-dwc3", "snps,dwc3", "snps,dwc3-simple";
 		reg = <0x0 0xfd000000 0x0 0x400000>;
 		interrupts = <GIC_SPI 170 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru CLK_USB3OTG1_REF>, <&cru CLK_USB3OTG1_SUSPEND>,
