diff --speed-large-files --no-dereference --minimal -Naur linux-7.0-rc1/arch/arm64/boot/dts/broadcom/bcm2712.dtsi linux-7.0-rc1/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
--- linux-7.0-rc1/arch/arm64/boot/dts/broadcom/bcm2712.dtsi	2026-02-22 22:18:59.000000000 +0100
+++ linux-7.0-rc1/arch/arm64/boot/dts/broadcom/bcm2712.dtsi	2026-03-04 14:48:18.109292819 +0100
@@ -1,5 +1,6 @@
 // SPDX-License-Identifier: (GPL-2.0 OR MIT)
 #include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/soc/bcm2835-pm.h>
 
 / {
 	compatible = "brcm,bcm2712";
@@ -45,6 +46,27 @@
 			clock-frequency = <96000000>;
 			clock-output-names = "sw-baud";
 		};
+
+	};
+
+	thermal-zones {
+		cpu_thermal: cpu-thermal {
+			polling-delay-passive = <1000>;
+			polling-delay = <1000>;
+			coefficients = <(-550) 450000>;
+			thermal-sensors = <&thermal>;
+
+			thermal_trips: trips {
+				cpu_crit: cpu-crit {
+					temperature	= <110000>;
+					hysteresis	= <0>;
+					type		= "critical";
+				};
+			};
+
+			cooling_maps: cooling-maps {
+			};
+		};
 	};
 
 	cpus: cpus {
