diff --speed-large-files --no-dereference --minimal -Naur linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
--- linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2026-03-13 11:27:03.990732489 +0100
+++ linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2026-03-13 11:20:01.784936057 +0100
@@ -465,7 +465,7 @@
 	};
 
 	usb_host0_xhci: usb@fc000000 {
-		compatible = "rockchip,rk3588-dwc3", "snps,dwc3";
+		compatible = "rockchip,rk3588-dwc3", "snps,dwc3", "snps,dwc3-simple";
 		reg = <0x0 0xfc000000 0x0 0x400000>;
 		interrupts = <GIC_SPI 220 IRQ_TYPE_LEVEL_HIGH 0>;
 		clocks = <&cru REF_CLK_USB3OTG0>, <&cru SUSPEND_CLK_USB3OTG0>,
@@ -531,7 +531,7 @@
 	};
 
 	usb_host2_xhci: usb@fcd00000 {
-		compatible = "rockchip,rk3588-dwc3", "snps,dwc3";
+		compatible = "rockchip,rk3588-dwc3", "snps,dwc3", "snps,dwc3-simple";
 		reg = <0x0 0xfcd00000 0x0 0x400000>;
 		interrupts = <GIC_SPI 222 IRQ_TYPE_LEVEL_HIGH 0>;
 		clocks = <&cru REF_CLK_USB3OTG2>, <&cru SUSPEND_CLK_USB3OTG2>,
diff --speed-large-files --no-dereference --minimal -Naur linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi
--- linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi	2026-03-09 00:56:54.000000000 +0100
+++ linux-7.0-rc3/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi	2026-03-13 11:20:40.099848327 +0100
@@ -48,7 +48,7 @@
 	};
 
 	usb_host1_xhci: usb@fc400000 {
-		compatible = "rockchip,rk3588-dwc3", "snps,dwc3";
+		compatible = "rockchip,rk3588-dwc3", "snps,dwc3", "snps,dwc3-simple";
 		reg = <0x0 0xfc400000 0x0 0x400000>;
 		interrupts = <GIC_SPI 221 IRQ_TYPE_LEVEL_HIGH 0>;
 		clocks = <&cru REF_CLK_USB3OTG1>, <&cru SUSPEND_CLK_USB3OTG1>,
