diff --speed-large-files --no-dereference --minimal -Naur linux-7.0-rc7/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi linux-7.0-rc7/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
--- linux-7.0-rc7/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2026-04-08 16:36:10.011333607 +0200
+++ linux-7.0-rc7/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2026-04-08 16:21:38.440305047 +0200
@@ -388,6 +388,21 @@
 		};
 	};
 
+	dp0_sound: dp0-sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,mclk-fs = <512>;
+		simple-audio-card,name = "DP0";
+		status = "disabled";
+
+		simple-audio-card,codec {
+			sound-dai = <&dp0 1>;
+		};
+
+		simple-audio-card,cpu {
+			sound-dai = <&spdif_tx2>;
+		};
+	};
+
 	pmu-a55 {
 		compatible = "arm,cortex-a55-pmu";
 		interrupts = <GIC_PPI 7 IRQ_TYPE_LEVEL_HIGH &ppi_partition0>;
@@ -1664,7 +1679,7 @@
 		phys = <&usbdp_phy0 PHY_TYPE_DP>;
 		power-domains = <&power RK3588_PD_VO0>;
 		resets = <&cru SRST_DP0>;
-		#sound-dai-cells = <0>;
+		#sound-dai-cells = <1>;
 		status = "disabled";
 
 		ports {
diff --speed-large-files --no-dereference --minimal -Naur linux-7.0-rc7/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi linux-7.0-rc7/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi
--- linux-7.0-rc7/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi	2026-04-08 16:36:09.979661876 +0200
+++ linux-7.0-rc7/arch/arm64/boot/dts/rockchip/rk3588-extra.dtsi	2026-04-08 16:06:11.339585984 +0200
@@ -23,6 +23,21 @@
 		};
 	};
 
+	dp1_sound: dp1-sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,mclk-fs = <512>;
+		simple-audio-card,name = "DP1";
+		status = "disabled";
+
+		simple-audio-card,codec {
+			sound-dai = <&dp1 1>;
+		};
+
+		simple-audio-card,cpu {
+			sound-dai = <&spdif_tx5>;
+		};
+	};
+
 	reserved-memory {
 		#address-cells = <2>;
 		#size-cells = <2>;
@@ -223,7 +238,7 @@
 		phys = <&usbdp_phy1 PHY_TYPE_DP>;
 		power-domains = <&power RK3588_PD_VO0>;
 		resets = <&cru SRST_DP1>;
-		#sound-dai-cells = <0>;
+		#sound-dai-cells = <1>;
 		status = "disabled";
 
 		ports {
