diff --speed-large-files --no-dereference --minimal -Naur linux-7.0.6/arch/arm64/boot/dts/qcom/qcs6490-radxa-dragon-q6a.dts linux-7.0.6/arch/arm64/boot/dts/qcom/qcs6490-radxa-dragon-q6a.dts
--- linux-7.0.6/arch/arm64/boot/dts/qcom/qcs6490-radxa-dragon-q6a.dts	2026-05-11 08:21:59.000000000 +0200
+++ linux-7.0.6/arch/arm64/boot/dts/qcom/qcs6490-radxa-dragon-q6a.dts	2026-05-14 17:09:43.799178690 +0200
@@ -78,6 +78,57 @@
 		stdout-path = "serial0:115200n8";
 	};
 
+	bridge {
+		compatible = "radxa,ra620";
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+
+				hdmi_bridge_in: endpoint {
+					remote-endpoint = <&usb_dp_qmpphy_out>;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+
+				hdmi_bridge_out: endpoint {
+					remote-endpoint = <&hdmi_connector_in>;
+				};
+			};
+		};
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		pinctrl-0 = <&kypd_vol_up_n>;
+		pinctrl-names = "default";
+
+		key-volume-up {
+			label = "Volume Up";
+			gpios = <&pm7325_gpios 6 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_VOLUMEUP>;
+			linux,can-disable;
+		};
+	};
+
+	hdmi-connector {
+		compatible = "hdmi-connector";
+		label = "hdmi";
+		type = "a";
+
+		port {
+			hdmi_connector_in: endpoint {
+				remote-endpoint = <&hdmi_bridge_out>;
+			};
+		};
+	};
+
 	usb2_1_con: connector-0 {
 		compatible = "usb-a-connector";
 		vbus-supply = <&vcc_5v_peri>;
@@ -119,11 +170,9 @@
 
 		user-led {
 			color = <LED_COLOR_ID_BLUE>;
-			function = LED_FUNCTION_INDICATOR;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			gpios = <&tlmm 42 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "none";
-			default-state = "off";
-			panic-indicator;
+			linux,default-trigger = "disk-activity";
 		};
 	};
 
@@ -275,8 +324,13 @@
 		vdd-s8-supply = <&vph_pwr>;
 		vdd-l1-l4-l12-l15-supply = <&vreg_s7b_0p536>;
 		vdd-l2-l7-supply = <&vreg_bob_3p296>;
+		vdd-l3-supply = <&vreg_s2b_0p572>;
+		vdd-l5-supply = <&vreg_s2b_0p572>;
 		vdd-l6-l9-l10-supply = <&vreg_s8b_1p2>;
+		vdd-l8-supply = <&vreg_s7b_0p536>;
 		vdd-l11-l17-l18-l19-supply = <&vreg_s1b_1p84>;
+		vdd-l13-supply = <&vreg_s7b_0p536>;
+		vdd-l14-l16-supply = <&vreg_s8b_1p2>;
 
 		vreg_s1b_1p84: smps1 {
 			regulator-name = "vreg_s1b_1p84";
@@ -284,6 +338,12 @@
 			regulator-max-microvolt = <2040000>;
 		};
 
+		vreg_s2b_0p572: smps2 {
+			regulator-name = "vreg_s2b_0p572";
+			regulator-min-microvolt = <572000>;
+			regulator-max-microvolt = <1048000>;
+		};
+
 		vreg_s7b_0p536: smps7 {
 			regulator-name = "vreg_s7b_0p536";
 			regulator-min-microvolt = <536000>;
@@ -317,6 +377,27 @@
 							RPMH_REGULATOR_MODE_HPM>;
 		};
 
+		vreg_l3b_0p32: ldo3 {
+			regulator-name = "vreg_l3b_0p32";
+			regulator-min-microvolt = <320000>;
+			regulator-max-microvolt = <904000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
+		};
+
+		vreg_l4b_0p752: ldo4 {
+			regulator-name = "vreg_l4b_0p752";
+			regulator-min-microvolt = <752000>;
+			regulator-max-microvolt = <816000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
+		};
+
+		vreg_l5b_0p44: ldo5 {
+			regulator-name = "vreg_l5b_0p44";
+			regulator-min-microvolt = <440000>;
+			regulator-max-microvolt = <832000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
+		};
+
 		vreg_l6b_1p2: ldo6 {
 			regulator-name = "vreg_l6b_1p2";
 			regulator-min-microvolt = <1200000>;
@@ -337,6 +418,13 @@
 							RPMH_REGULATOR_MODE_HPM>;
 		};
 
+		vreg_l8b_0p872: ldo8 {
+			regulator-name = "vreg_l8b_0p872";
+			regulator-min-microvolt = <872000>;
+			regulator-max-microvolt = <968000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
+		};
+
 		vreg_l9b_1p2: ldo9 {
 			regulator-name = "vreg_l9b_1p2";
 			regulator-min-microvolt = <1200000>;
@@ -347,6 +435,48 @@
 						   RPMH_REGULATOR_MODE_HPM>;
 		};
 
+		vreg_l11b_1p504: ldo11 {
+			regulator-name = "vreg_l11b_1p504";
+			regulator-min-microvolt = <1504000>;
+			regulator-max-microvolt = <2000000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_l12b_0p752: ldo12 {
+			regulator-name = "vreg_l12b_0p752";
+			regulator-min-microvolt = <752000>;
+			regulator-max-microvolt = <824000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_l13b_0p53: ldo13 {
+			regulator-name = "vreg_l13b_0p53";
+			regulator-min-microvolt = <536000>;
+			regulator-max-microvolt = <824000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_l14b_1p08: ldo14 {
+			regulator-name = "vreg_l14b_1p08";
+			regulator-min-microvolt = <1080000>;
+			regulator-max-microvolt = <1304000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_l15b_0p768: ldo15 {
+			regulator-name = "vreg_l15b_0p768";
+			regulator-min-microvolt = <768000>;
+			regulator-max-microvolt = <1016000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_l16b_1p1: ldo16 {
+			regulator-name = "vreg_l16b_1p1";
+			regulator-min-microvolt = <1104000>;
+			regulator-max-microvolt = <1296000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
 		vreg_l17b_1p8: ldo17 {
 			regulator-name = "vreg_l17b_1p8";
 			regulator-min-microvolt = <1800000>;
@@ -388,10 +518,44 @@
 		vdd-s9-supply = <&vph_pwr>;
 		vdd-s10-supply = <&vph_pwr>;
 		vdd-l1-l12-supply = <&vreg_s1b_1p84>;
+		vdd-l2-l8-supply = <&vreg_s1b_1p84>;
+		vdd-l3-l4-l5-l7-l13-supply = <&vreg_bob_3p296>;
 		vdd-l6-l9-l11-supply = <&vreg_bob_3p296>;
 		vdd-l10-supply = <&vreg_s7b_0p536>;
 		vdd-bob-supply = <&vph_pwr>;
 
+		vreg_s1c_2p19: smps1 {
+			regulator-name = "vreg_s1c_2p19";
+			regulator-min-microvolt = <2192000>;
+			regulator-max-microvolt = <2208000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_s2c_0p752: smps2 {
+			regulator-name = "vreg_s2c_0p752";
+			regulator-min-microvolt = <752000>;
+			regulator-max-microvolt = <800000>;
+		};
+
+		vreg_s5c_0p468: smps5 {
+			regulator-name = "vreg_s5c_0p468";
+			regulator-min-microvolt = <468000>;
+			regulator-max-microvolt = <1048000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_s7c_0p468: smps7 {
+			regulator-name = "vreg_s7c_0p468";
+			regulator-min-microvolt = <468000>;
+			regulator-max-microvolt = <800000>;
+		};
+
+		vreg_s9c_1p012: smps9 {
+			regulator-name = "vreg_s9c_1p012";
+			regulator-min-microvolt = <1012000>;
+			regulator-max-microvolt = <1168000>;
+		};
+
 		vreg_l1c_1p8: ldo1 {
 			regulator-name = "vreg_l1c_1p8";
 			regulator-min-microvolt = <1800000>;
@@ -402,6 +566,34 @@
 						   RPMH_REGULATOR_MODE_HPM>;
 		};
 
+		vreg_l2c_1p8: ldo2 {
+			regulator-name = "vreg_l2c_1p8";
+			regulator-min-microvolt = <1624000>;
+			regulator-max-microvolt = <1976000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_l3c_3p0: ldo3 {
+			regulator-name = "vreg_l3c_3p0";
+			regulator-min-microvolt = <2800000>;
+			regulator-max-microvolt = <3536000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_l4c_1p62: ldo4 {
+			regulator-name = "vreg_l4c_1p62";
+			regulator-min-microvolt = <1624000>;
+			regulator-max-microvolt = <3296000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_l5c_1p62: ldo5 {
+			regulator-name = "vreg_l5c_1p62";
+			regulator-min-microvolt = <1624000>;
+			regulator-max-microvolt = <3296000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
 		vreg_l6c_2p96: ldo6 {
 			regulator-name = "vreg_l6c_2p96";
 			regulator-min-microvolt = <1650000>;
@@ -412,6 +604,20 @@
 						   RPMH_REGULATOR_MODE_HPM>;
 		};
 
+		vreg_l7c_3p0: ldo7 {
+			regulator-name = "vreg_l7c_3p0";
+			regulator-min-microvolt = <3000000>;
+			regulator-max-microvolt = <3544000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_l8c_1p8: ldo8 {
+			regulator-name = "vreg_l8c_1p8";
+			regulator-min-microvolt = <1624000>;
+			regulator-max-microvolt = <2000000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
 		vreg_l9c_2p96: ldo9 {
 			regulator-name = "vreg_l9c_2p96";
 			regulator-min-microvolt = <2704000>;
@@ -432,6 +638,27 @@
 						   RPMH_REGULATOR_MODE_HPM>;
 		};
 
+		vreg_l11c_2p8: ldo11 {
+			regulator-name = "vreg_l11c_2p8";
+			regulator-min-microvolt = <2800000>;
+			regulator-max-microvolt = <3544000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_l12c_1p8: ldo12 {
+			regulator-name = "vreg_l12c_1p8";
+			regulator-min-microvolt = <1656000>;
+			regulator-max-microvolt = <2000000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
+		vreg_l13c_3p0: ldo13 {
+			regulator-name = "vreg_l13c_2p7";
+			regulator-min-microvolt = <2704000>;
+			regulator-max-microvolt = <3544000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_RET>;
+		};
+
 		vreg_bob_3p296: bob {
 			regulator-name = "vreg_bob_3p296";
 			regulator-min-microvolt = <3032000>;
@@ -504,6 +731,11 @@
 	status = "okay";
 };
 
+// Otherwise it takes a long time in ufshcd_init_crypto
+&ice {
+	status = "disabled";
+};
+
 &lpass_audiocc {
 	compatible = "qcom,qcm6490-lpassaudiocc";
 	/delete-property/ power-domains;
@@ -521,6 +753,22 @@
 	status = "okay";
 };
 
+&mdss {
+	status = "okay";
+};
+
+&mdss_dp {
+	status = "okay";
+	sound-name-prefix = "Display Port0";
+	pinctrl-names = "default";
+	pinctrl-0 = <&dp_hot_plug_det>;
+};
+
+&mdss_dp_out {
+	data-lanes = <0 1>;
+	remote-endpoint = <&usb_dp_qmpphy_dp_in>;
+};
+
 &pcie0 {
 	perst-gpios = <&tlmm 87 GPIO_ACTIVE_LOW>;
 	wake-gpios = <&tlmm 89 GPIO_ACTIVE_HIGH>;
@@ -572,6 +820,14 @@
 		function = PMIC_GPIO_FUNC_NORMAL;
 		bias-high-impedance;
 	};
+
+	kypd_vol_up_n: kypd-vol-up-n-state {
+		pins = "gpio6";
+		function = PMIC_GPIO_FUNC_NORMAL;
+		power-source = <1>;
+		bias-pull-up;
+		input-enable;
+	};
 };
 
 &pm7325_temp_alarm {
@@ -665,6 +921,12 @@
 	status = "okay";
 };
 
+&psci {
+	reset-types {
+		mode-edl = <0 0x1>;
+	};
+};
+
 &qspi {
 	/* It's not possible to use QSPI with iommu */
 	/* due to an error in qcom_smmu_write_s2cr */
@@ -769,6 +1031,22 @@
 			sound-dai = <&q6apm>;
 		};
 	};
+
+	dp0-dai-link {
+		link-name = "DP0 Playback";
+
+		cpu {
+			sound-dai = <&q6apmbedai DISPLAY_PORT_RX_0>;
+		};
+
+		codec {
+			sound-dai = <&mdss_dp>;
+		};
+
+		platform {
+			sound-dai = <&q6apm>;
+		};
+	};
 };
 
 /* Pin 11, 29, 31, 32 in GPIO header */
@@ -811,94 +1089,198 @@
 
 &tlmm {
 	gpio-line-names =
-		/* GPIO_0 ~ GPIO_3 */
-		"PIN_13", "PIN_15", "", "",
-		/* GPIO_4 ~ GPIO_7 */
-		"", "", "", "",
-		/* GPIO_8 ~ GPIO_11 */
-		"PIN_27", "PIN_28", "", "",
-		/* GPIO_12 ~ GPIO_15 */
-		"", "", "", "",
-		/* GPIO_16 ~ GPIO_19 */
-		"", "", "", "",
-		/* GPIO_20 ~ GPIO_23 */
-		"", "", "PIN_8", "PIN_10",
-		/* GPIO_24 ~ GPIO_27 */
-		"PIN_3", "PIN_5", "PIN_16", "PIN_27",
-		/* GPIO_28 ~ GPIO_31 */
-		"PIN_31", "PIN_11", "PIN_32", "PIN_29",
-		/* GPIO_32 ~ GPIO_35 */
-		"", "", "", "",
-		/* GPIO_36 ~ GPIO_39 */
-		"", "", "", "",
-		/* GPIO_40 ~ GPIO_43 */
-		"", "", "", "",
-		/* GPIO_44 ~ GPIO_47 */
-		"", "", "", "",
-		/* GPIO_48 ~ GPIO_51 */
-		"PIN_21", "PIN_19", "PIN_23", "PIN_24",
-		/* GPIO_52 ~ GPIO_55 */
-		"", "", "", "PIN_26",
-		/* GPIO_56 ~ GPIO_59 */
-		"PIN_33", "PIN_22", "PIN_37", "PIN_36",
-		/* GPIO_60 ~ GPIO_63 */
-		"", "", "", "",
-		/* GPIO_64 ~ GPIO_67 */
-		"", "", "", "",
-		/* GPIO_68 ~ GPIO_71 */
-		"", "", "", "",
-		/* GPIO_72 ~ GPIO_75 */
-		"", "", "", "",
-		/* GPIO_76 ~ GPIO_79 */
-		"", "", "", "",
-		/* GPIO_80 ~ GPIO_83 */
-		"", "", "", "",
-		/* GPIO_84 ~ GPIO_87 */
-		"", "", "", "",
-		/* GPIO_88 ~ GPIO_91 */
-		"", "", "", "",
-		/* GPIO_92 ~ GPIO_95 */
-		"", "", "", "",
-		/* GPIO_96 ~ GPIO_99 */
-		"PIN_7", "PIN_12", "PIN_38", "PIN_40",
-		/* GPIO_100 ~ GPIO_103 */
-		"PIN_35", "", "", "",
-		/* GPIO_104 ~ GPIO_107 */
-		"", "", "", "",
-		/* GPIO_108 ~ GPIO_111 */
-		"", "", "", "",
-		/* GPIO_112 ~ GPIO_115 */
-		"", "", "", "",
-		/* GPIO_116 ~ GPIO_119 */
-		"", "", "", "",
-		/* GPIO_120 ~ GPIO_123 */
-		"", "", "", "",
-		/* GPIO_124 ~ GPIO_127 */
-		"", "", "", "",
-		/* GPIO_128 ~ GPIO_131 */
-		"", "", "", "",
-		/* GPIO_132 ~ GPIO_135 */
-		"", "", "", "",
-		/* GPIO_136 ~ GPIO_139 */
-		"", "", "", "",
-		/* GPIO_140 ~ GPIO_143 */
-		"", "", "", "",
-		/* GPIO_144 ~ GPIO_147 */
-		"", "", "", "",
-		/* GPIO_148 ~ GPIO_151 */
-		"", "", "", "",
-		/* GPIO_152 ~ GPIO_155 */
-		"", "", "", "",
-		/* GPIO_156 ~ GPIO_159 */
-		"", "", "", "",
-		/* GPIO_160 ~ GPIO_163 */
-		"", "", "", "",
-		/* GPIO_164 ~ GPIO_167 */
-		"", "", "", "",
-		/* GPIO_168 ~ GPIO_171 */
-		"", "", "", "",
-		/* GPIO_172 ~ GPIO_174 */
-		"", "", "";
+			  "QUPSE0_I2C_SDA", /* GPIO_0 */
+			  "QUPSE0_I2C_SCL",
+			  "PCIE1_RESET_N",
+			  "PCIE1_WAKE_N",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "QUPSE2_I2C_SDA",
+			  "QUPSE2_I2C_SCL",
+
+			  "NC", /* GPIO_10 */
+			  "NC",
+			  "QSPI_DATA0",
+			  "QSPI_DATA1",
+			  "QSPI_CLK",
+			  "QSPI_CS",
+			  "QSPI_DATA2",
+			  "QSPI_DATA3",
+			  "NC",
+			  "NC",
+
+			  "CAM0_RST_N", /* GPIO_20 */
+			  "NC",
+			  "DBG_UART_TX",
+			  "DBG_UART_RX",
+			  "QUPSE6_I2C_SDA",
+			  "QUPSE6_I2C_SCL",
+			  "QUPSE6_UART_TX",
+			  "QUPSE6_UART_RX",
+			  "QUPSE7_SPI_MISO",
+			  "QUPSE7_SPI_MOSI",
+
+			  "QUPSE7_SPI_CLK", /* GPIO_30 */
+			  "QUPSE7_SPI_CS_N",
+			  "NC",
+			  "WIFI_PWR_EN",
+			  "NC",
+			  "NC",
+			  "CAM0_POWER_EN",
+			  "NC",
+			  "NC",
+			  "NC",
+
+			  "I2C_RTC_SDA", /* GPIO_40 */
+			  "I2C_RTC_SCL",
+			  "USER_DEF_LEDG",
+			  "NC",
+			  "DISP0_RESET_N",
+			  "NC",
+			  "NC",
+			  "DP_HPD",
+			  "QUP1SE4_SPI_MISO",
+			  "QUP1SE4_SPI_MOSI",
+
+			  "QUP1SE4_SPI_CLK", /* GPIO_50 */
+			  "QUP1SE4_SPI_CS0_N",
+			  "TS_SPI_MISO_I2C_SDA",
+			  "TS_SPI_MOSI_I2C_SCL",
+			  "NC",
+			  "QUP1SE4_SPI_CS1_N",
+			  "FP_SPI_MISO",
+			  "FP_SPI_MOSI",
+			  "FP_SPI_SCLK",
+			  "FP_SPI_CS_N",
+
+			  "NC", /* GPIO_60 */
+			  "NC",
+			  "NC",
+			  "NC",
+			  "CAM_MCLK0",
+			  "NC",
+			  "CAM_MCLK2",
+			  "CAM_MCLK3",
+			  "NC",
+			  "CCI_I2C0_SDA",
+
+			  "CCI_I2C0_SCL", /* GPIO_70 */
+			  "NC",
+			  "NC",
+			  "CCI_I2C2_SDA",
+			  "CCI_I2C2_SCL",
+			  "CCI_I2C3_SDA",
+			  "CCI_I2C3_SCL",
+			  "CAM2_RESET_N",
+			  "CAM3_RESET_N",
+			  "PCIE1_CLK_REQ_N",
+
+			  "LCD_VCC_EN", /* GPIO_80 */
+			  "TS_INT_N",
+			  "FORCED_USB_BOOT",
+			  "WCD_RESET_N",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "SM_PCIE0_RESET_N_WCN",
+			  "SM_PCIE0_CLK_REQ_N_WCN",
+			  "SM_PCIE0_WAKE_N_WCN",
+
+			  "NC", /* GPIO_90 */
+			  "SD_CARD_DET_N",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "PRI_MI2S_MCLK",
+			  "MI2S0_SCK",
+			  "MI2S0_DATA0",
+			  "MI2S0_DATA1",
+
+			  "MI2S0_WS", /* GPIO_100 */
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "TS_RESET_N",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+
+			  "NC", /* GPIO_110 */
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+
+			  "BOOT_CONFIG_1", /* GPIO_120 */
+			  "NC",
+			  "BOOT_CONFIG_2",
+			  "NC",
+			  "BOOT_CONFIG_3",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+
+			  "NC", /* GPIO_130 */
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "FORCED_USB_BOOT_POL_SEL",
+
+			  "USB_PHY_PS", /* GPIO_140 */
+			  "NC",
+			  "NC",
+			  "NC",
+			  "WCD_SWR_TX_CLK",
+			  "WCD_SWR_TX_DATA0",
+			  "WCD_SWR_TX_DATA1",
+			  "WCD_SWR_RX_CLK",
+			  "WCD_SWR_RX_DATA0",
+			  "WCD_SWR_RX_DATA1",
+
+			  "NC", /* GPIO_150 */
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "WCD_SWR_TX_DATA2",
+			  "NC",
+
+			  "NC", /* GPIO_160 */
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC",
+
+			  "NC", /* GPIO_170 */
+			  "NC",
+			  "NC",
+			  "NC",
+			  "NC";
 
 	pcie0_reset_n: pcie0-reset-n-state {
 		pins = "gpio87";
@@ -940,6 +1322,13 @@
 		bias-pull-up;
 	};
 
+	ufs_cd: ufs-cd-state {
+		pins = "gpio7";
+		function = "gpio";
+		drive-strength = <2>;
+		bias-pull-up;
+	};
+
 	user_led: user-led-state {
 		pins = "gpio42";
 		function = "gpio";
@@ -959,6 +1348,55 @@
 	status = "okay";
 };
 
+&ufs_mem_hc {
+	cd-gpios = <&tlmm 7 GPIO_ACTIVE_LOW>;
+	reset-gpios = <&tlmm 175 GPIO_ACTIVE_LOW>;
+	vcc-supply = <&vreg_l7b_2p96>;
+	vcc-max-microamp = <800000>;
+	vccq-supply = <&vreg_l9b_1p2>;
+	vccq-max-microamp = <900000>;
+	vccq2-supply = <&vreg_l9b_1p2>;
+	vccq2-max-microamp = <1300000>;
+
+	pinctrl-0 = <&ufs_cd>;
+	pinctrl-names = "default";
+
+	/delete-property/ qcom,ice;
+
+	status = "okay";
+};
+
+&ufs_mem_phy {
+	vdda-phy-supply = <&vreg_l10c_0p88>;
+	vdda-pll-supply = <&vreg_l6b_1p2>;
+
+	status = "okay";
+};
+
+&usb_1 {
+	status = "okay";
+	dr_mode = "host";
+};
+
+&usb_1_hsphy {
+	vdda-pll-supply = <&vreg_l10c_0p88>;
+	vdda33-supply = <&vreg_l2b_3p072>;
+	vdda18-supply = <&vreg_l1c_1p8>;
+
+	status = "okay";
+};
+
+&usb_1_qmpphy {
+	vdda-phy-supply = <&vreg_l6b_1p2>;
+	vdda-pll-supply = <&vreg_l1b_0p912>;
+
+	status = "okay";
+};
+
+&usb_dp_qmpphy_out {
+	remote-endpoint = <&hdmi_bridge_in>;
+};
+
 &usb_2 {
 	dr_mode = "host";
 
