diff --speed-large-files --no-dereference --minimal -Naur crossguid-ca1bf4b810e2d188d04cb6286f957008ee1b7681/include/crossguid/guid.hpp crossguid-ca1bf4b810e2d188d04cb6286f957008ee1b7681/include/crossguid/guid.hpp
--- crossguid-ca1bf4b810e2d188d04cb6286f957008ee1b7681/include/crossguid/guid.hpp	2019-05-29 17:36:34.000000000 +0200
+++ crossguid-ca1bf4b810e2d188d04cb6286f957008ee1b7681/include/crossguid/guid.hpp	2026-03-30 12:04:06.789266673 +0200
@@ -29,6 +29,7 @@
 #include <jni.h>
 #endif
 
+#include <cstdint>
 #include <functional>
 #include <iostream>
 #include <array>
