diff --speed-large-files --no-dereference --minimal -Naur libcoverart-1.0.0/src/CMakeLists.txt libcoverart-1.0.0/src/CMakeLists.txt
--- libcoverart-1.0.0/src/CMakeLists.txt	2012-10-09 14:51:32.000000000 +0200
+++ libcoverart-1.0.0/src/CMakeLists.txt	2024-01-04 18:33:51.904567192 +0100
@@ -31,7 +31,8 @@
 	OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/caa_c.cc ${CMAKE_CURRENT_BINARY_DIR}/caa_c.h ${CMAKE_CURRENT_BINARY_DIR}/../include/coverart/caa_c.h
 	COMMAND make-c-interface ${CMAKE_CURRENT_SOURCE_DIR} cinterface.xml ${CMAKE_CURRENT_BINARY_DIR} caa_c.cc caa_c.h
 	COMMAND ${CMAKE_COMMAND} -E copy_if_different ${CMAKE_CURRENT_BINARY_DIR}/caa_c.h ${CMAKE_CURRENT_BINARY_DIR}/../include/coverart/caa_c.h
-	DEPENDS make-c-interface cinterface.xml *.inc
+	DEPENDS make-c-interface cinterface.xml c-int-coverart-defines.inc c-int-coverart-source.inc c-int-imagedata-defines.inc 
+	c-int-imagedata-source.inc c-int-source-funcs.inc
 )
 
 ADD_CUSTOM_TARGET(src_gen DEPENDS caa_c.h)
