diff -Naur mythtv-0.26-20130707-gcb02cec-old/mythplugins/configure mythtv-0.26-20130707-gcb02cec-new/mythplugins/configure
--- mythtv-0.26-20130707-gcb02cec-old/mythplugins/configure	2013-07-11 18:10:33.853416466 +0200
+++ mythtv-0.26-20130707-gcb02cec-new/mythplugins/configure	2013-07-11 18:10:48.593411260 +0200
@@ -458,10 +458,15 @@
 check_py_lib(){
     log check_py_lib "$@"
     lib=${1}
+if test "x${PYTHONXCPREFIX}" = "x" ; then
+   py_prefix=${prefix}
+else
+   py_prefix=${PYTHONXCPREFIX}
+fi
     check_cmd $python <<EOF
 from distutils.sysconfig import get_python_lib
 import sys
-sys.path.append(get_python_lib(prefix='${prefix}'))
+sys.path.append(get_python_lib(prefix='${py_prefix}'))
 try:
     import $lib
 except:
diff -Naur mythtv-0.26-20130707-gcb02cec-old/mythtv/bindings/python/Makefile mythtv-0.26-20130707-gcb02cec-new/mythtv/bindings/python/Makefile
--- mythtv-0.26-20130707-gcb02cec-old/mythtv/bindings/python/Makefile	2013-07-11 18:10:33.746749837 +0200
+++ mythtv-0.26-20130707-gcb02cec-new/mythtv/bindings/python/Makefile	2013-07-11 18:10:49.980077360 +0200
@@ -1,15 +1,15 @@
 include ../../config.mak
 
-ifdef INSTALL_ROOT
+#ifdef INSTALL_ROOT
 	ROOT_FLAGS = --root="$(INSTALL_ROOT)"
-endif
+#endif
 
 REAL_PREFIX = $(PREFIX)
-ifneq ($(PREFIX:/=), /usr)
-	ifneq ($(PREFIX:/=), /usr/local)
+#ifneq ($(PREFIX:/=), /usr)
+#	ifneq ($(PREFIX:/=), /usr/local)
 		PREFIX_FLAGS=--prefix="$(PREFIX)"
-	endif
-endif
+#	endif
+#endif
 
 all: python_build
 
@@ -19,7 +19,7 @@
 distclean: clean
 
 python_build: setup.py
-	$(PYTHON) setup.py build --mythtv-prefix="$(REAL_PREFIX)"
+	$(PYTHON) setup.py build --cross-compile --mythtv-prefix="$(REAL_PREFIX)"
 
 install: setup.py
 	$(PYTHON) setup.py install --skip-build $(ROOT_FLAGS) $(PREFIX_FLAGS)
diff -Naur mythtv-0.26-20130707-gcb02cec-old/mythtv/bindings/python/setup.py mythtv-0.26-20130707-gcb02cec-new/mythtv/bindings/python/setup.py
--- mythtv-0.26-20130707-gcb02cec-old/mythtv/bindings/python/setup.py	2013-07-11 18:10:33.746749837 +0200
+++ mythtv-0.26-20130707-gcb02cec-new/mythtv/bindings/python/setup.py	2013-07-11 18:10:50.680077036 +0200
@@ -1,10 +1,10 @@
 #!/usr/bin/env python
 
-from distutils.core import setup
-from distutils.cmd import Command
+from setuptools import setup
+from setuptools import Command
 from distutils.sysconfig import get_python_lib, project_base
 from distutils.command.install import INSTALL_SCHEMES
-from distutils.command.build import build as pybuild
+from distutilscross.crosscompile import build as pybuild
 
 import os
 import glob
