diff -Naur mythtv-0.26-20130707-gcb02cec-old/mythplugins/mytharchive/mytharchive/mytharchive.pro mythtv-0.26-20130707-gcb02cec-new/mythplugins/mytharchive/mytharchive/mytharchive.pro
--- mythtv-0.26-20130707-gcb02cec-old/mythplugins/mytharchive/mytharchive/mytharchive.pro	2013-07-11 18:10:47.840078194 +0200
+++ mythtv-0.26-20130707-gcb02cec-new/mythplugins/mytharchive/mytharchive/mytharchive.pro	2013-07-11 18:11:02.440078051 +0200
@@ -5,8 +5,8 @@
 QMAKE_COPY_DIR = sh ../../cpsvndir
 win32:QMAKE_COPY_DIR = sh ../../cpsimple
 
-INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libavformat
-INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libswscale
+INCLUDEPATH += $${PREFIX}/include/mythtv/libavformat
+INCLUDEPATH += $${PREFIX}/include/mythtv/libswscale
 DEPENDPATH *= $${INCLUDEPATH}
 
 LIBS += -lmythswscale
@@ -33,28 +33,28 @@
 SOURCES += mythburn.cpp themeselector.cpp editmetadata.cpp thumbfinder.cpp
 SOURCES += exportnative.cpp importnative.cpp
 
-burnuifiles.path = $${SYSROOT}$${PREFIX}/share/mythtv/themes/default
+burnuifiles.path = $${PREFIX}/share/mythtv/themes/default
 burnuifiles.files = mythburn-ui.xml
 
-scriptfiles.path = $${SYSROOT}$${PREFIX}/share/mythtv/mytharchive/scripts
+scriptfiles.path = $${PREFIX}/share/mythtv/mytharchive/scripts
 scriptfiles.files = ../mythburn/scripts/*
 
-introfiles.path = $${SYSROOT}$${PREFIX}/share/mythtv/mytharchive/intro
+introfiles.path = $${PREFIX}/share/mythtv/mytharchive/intro
 introfiles.files = ../mythburn/intro/*.mpg
 
-musicfiles.path = $${SYSROOT}$${PREFIX}/share/mythtv/mytharchive/music
+musicfiles.path = $${PREFIX}/share/mythtv/mytharchive/music
 musicfiles.files = ../mythburn/music/*.ac3
 
-imagefiles.path = $${SYSROOT}$${PREFIX}/share/mythtv/mytharchive/images
+imagefiles.path = $${PREFIX}/share/mythtv/mytharchive/images
 imagefiles.files = ../mythburn/images/*.png
 
-themefiles.path = $${SYSROOT}$${PREFIX}/share/mythtv/mytharchive/themes
+themefiles.path = $${PREFIX}/share/mythtv/mytharchive/themes
 themefiles.files = ../mythburn/themes/*
 
-profilefiles.path = $${SYSROOT}$${PREFIX}/share/mythtv/mytharchive/encoder_profiles
+profilefiles.path = $${PREFIX}/share/mythtv/mytharchive/encoder_profiles
 profilefiles.files = ../mythburn/encoder_profiles/*.xml
 
-nativeuifiles.path = $${SYSROOT}$${PREFIX}/share/mythtv/themes/default
+nativeuifiles.path = $${PREFIX}/share/mythtv/themes/default
 nativeuifiles.files = mythnative-ui.xml
 
 INSTALLS +=  scriptfiles introfiles themefiles imagefiles musicfiles
diff -Naur mythtv-0.26-20130707-gcb02cec-old/mythplugins/mytharchive/mytharchivehelper/mytharchivehelper.pro mythtv-0.26-20130707-gcb02cec-new/mythplugins/mytharchive/mytharchivehelper/mytharchivehelper.pro
--- mythtv-0.26-20130707-gcb02cec-old/mythplugins/mytharchive/mytharchivehelper/mytharchivehelper.pro	2013-07-11 18:10:47.820078201 +0200
+++ mythtv-0.26-20130707-gcb02cec-new/mythplugins/mytharchive/mytharchivehelper/mytharchivehelper.pro	2013-07-11 18:11:02.440078051 +0200
@@ -3,10 +3,10 @@
 include ( ../../../mythtv/version.pro )
 include ( ../../programs-libs.pro )
 
-INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libavutil
-INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libavcodec
-INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libavformat
-INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libswscale
+INCLUDEPATH += $${PREFIX}/include/mythtv/libavutil
+INCLUDEPATH += $${PREFIX}/include/mythtv/libavcodec
+INCLUDEPATH += $${PREFIX}/include/mythtv/libavformat
+INCLUDEPATH += $${PREFIX}/include/mythtv/libswscale
 DEPENDPATH *= $${INCLUDEPATH}
 
 TEMPLATE = app
diff -Naur mythtv-0.26-20130707-gcb02cec-old/mythplugins/programs-libs.pro mythtv-0.26-20130707-gcb02cec-new/mythplugins/programs-libs.pro
--- mythtv-0.26-20130707-gcb02cec-old/mythplugins/programs-libs.pro	2013-07-11 18:10:48.593411260 +0200
+++ mythtv-0.26-20130707-gcb02cec-new/mythplugins/programs-libs.pro	2013-07-11 18:11:02.440078051 +0200
@@ -1,7 +1,7 @@
-INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv
-INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmythbase
-INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmythui
-INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmyth
+INCLUDEPATH += $${PREFIX}/include/mythtv
+INCLUDEPATH += $${PREFIX}/include/mythtv/libmythbase
+INCLUDEPATH += $${PREFIX}/include/mythtv/libmythui
+INCLUDEPATH += $${PREFIX}/include/mythtv/libmyth
 DEPENDPATH *= $${INCLUDEPATH}
 
 LIBS += -L$${LIBDIR} $$EXTRA_LIBS -lmythbase-$$LIBVERSION
@@ -12,7 +12,7 @@
 LIBS += -lmythavformat
 
 # On Windows, libs are stored with the binaries:
-mingw:LIBS += -L$${SYSROOT}$${PREFIX}/bin
+mingw:LIBS += -L$${PREFIX}/bin
 
 # Trigger rebuilds if installed MythTV libs have changed
 include (../../targetdep.pro)
diff -Naur mythtv-0.26-20130707-gcb02cec-old/mythplugins/settings.pro mythtv-0.26-20130707-gcb02cec-new/mythplugins/settings.pro
--- mythtv-0.26-20130707-gcb02cec-old/mythplugins/settings.pro	2013-07-11 18:10:43.563412618 +0200
+++ mythtv-0.26-20130707-gcb02cec-new/mythplugins/settings.pro	2013-07-11 18:11:02.440078051 +0200
@@ -2,9 +2,9 @@
 
 LIBVERSION = 0.26
 
-INCLUDEPATH += $${SYSROOT}$${PREFIX}/include
+INCLUDEPATH += $${PREFIX}/include
 
-LIBS *= -L$${SYSROOT}$${PREFIX}/$${LIBDIRNAME}
+LIBS *= -L$${PREFIX}/$${LIBDIRNAME}
 
 isEmpty(TARGET_OS) : win32 {
     CONFIG += mingw
