diff -Naur mythtv-master-20130801-g5b93775-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20130801-g5b93775-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20130801-g5b93775-old/mythtv/programs/mythfrontend/main.cpp	2013-08-01 15:14:55.324199073 +0200
+++ mythtv-master-20130801-g5b93775-new/mythtv/programs/mythfrontend/main.cpp	2013-08-01 15:15:25.357529151 +0200
@@ -1680,6 +1680,18 @@
         mon->StartMonitoring();
         mainWindow->installEventFilter(mon);
     }
+    else
+    {
+        int port = gCoreContext->GetNumSetting("NetworkControlPort", 6546);
+        networkControl = new NetworkControl();
+        QList<QHostAddress> LocalhostList;
+        LocalhostList.clear();
+        LocalhostList.append(QHostAddress("127.0.0.1"));
+        if (!networkControl->listen(LocalhostList, port))
+            LOG(VB_GENERAL, LOG_ERR,
+                QString("NetworkControl failed to bind to port %1.")
+                   .arg(port));
+    }
 
     NetworkControl *networkControl = NULL;
     if (gCoreContext->GetNumSetting("NetworkControlEnabled", 0))
