diff -Naur mythtv-0.26-20130314-g29bce5c-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp mythtv-0.26-20130314-g29bce5c-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp
--- mythtv-0.26-20130314-g29bce5c-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2013-03-14 11:29:40.839866622 +0100
+++ mythtv-0.26-20130314-g29bce5c-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2013-03-14 11:29:59.399860931 +0100
@@ -42,7 +42,7 @@
 #ifdef Q_WS_MACX
     gc->setValue(QDir::homePath() + "/Pictures");
 #else
-    gc->setValue("/var/lib/pictures");
+    gc->setValue("/mnt/gallery");
 #endif
     gc->setHelpText(QObject::tr("This directory must exist and "
                        "MythGallery needs to have read permission."));
@@ -97,7 +97,7 @@
 {
     HostLineEdit *gc = new HostLineEdit("GalleryImportDirs");
     gc->setLabel(QObject::tr("Paths to import images from"));
-    gc->setValue("/mnt/cdrom:/mnt/camera");
+    gc->setValue("");
     gc->setHelpText(QObject::tr("This is a colon separated list of paths. "
                     "If the path in the list is a directory, its contents will "
                     "be copied. If it is an executable, it will be run."));
diff -Naur mythtv-0.26-20130314-g29bce5c-old/mythtv/libs/libmythmetadata/globals.cpp mythtv-0.26-20130314-g29bce5c-new/mythtv/libs/libmythmetadata/globals.cpp
--- mythtv-0.26-20130314-g29bce5c-old/mythtv/libs/libmythmetadata/globals.cpp	2013-03-14 11:29:35.823201499 +0100
+++ mythtv-0.26-20130314-g29bce5c-new/mythtv/libs/libmythmetadata/globals.cpp	2013-03-14 11:29:59.399860931 +0100
@@ -32,5 +32,5 @@
 #ifdef Q_WS_MACX
 const QString DEFAULT_VIDEOSTARTUP_DIR = QDir::homePath() + "/Movies";
 #else
-const QString DEFAULT_VIDEOSTARTUP_DIR = "/share/Movies/dvd";
+const QString DEFAULT_VIDEOSTARTUP_DIR = "/mnt/video";
 #endif
diff -Naur mythtv-0.26-20130314-g29bce5c-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-0.26-20130314-g29bce5c-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-0.26-20130314-g29bce5c-old/mythtv/programs/mythfrontend/globalsettings.cpp	2013-03-14 11:29:37.326534371 +0100
+++ mythtv-0.26-20130314-g29bce5c-new/mythtv/programs/mythfrontend/globalsettings.cpp	2013-03-14 11:29:59.413194259 +0100
@@ -1237,7 +1237,7 @@
     gc->addSelection(QObject::tr("Record date"), "Date");
     gc->addSelection(QObject::tr("Season/Episode"), "Season");
     gc->addSelection(QObject::tr("Original air date"), "OrigAirDate");
-    gc->addSelection(QObject::tr("Program ID"), "Id");
+    gc->addSelection(QObject::tr("Program ID"), "Id", true);
     gc->setHelpText(QObject::tr("Selects how to sort a shows episodes"));
     return gc;
 }
@@ -1447,7 +1447,7 @@
     HostComboBox *gc = new HostComboBox("PlaybackExitPrompt");
     gc->setLabel(QObject::tr("Action on playback exit"));
     gc->addSelection(QObject::tr("Just exit"), "0");
-    gc->addSelection(QObject::tr("Save position and exit"), "2");
+    gc->addSelection(QObject::tr("Save position and exit"), "2", true);
     gc->addSelection(QObject::tr("Always prompt (excluding Live TV)"), "1");
     gc->addSelection(QObject::tr("Always prompt (including Live TV)"), "4");
     gc->addSelection(QObject::tr("Prompt for Live TV only"), "8");
@@ -1462,7 +1462,7 @@
 {
     HostCheckBox *gc = new HostCheckBox("EndOfRecordingExitPrompt");
     gc->setLabel(QObject::tr("Prompt at end of recording"));
-    gc->setValue(false);
+    gc->setValue(true);
     gc->setHelpText(QObject::tr("If enabled, a menu will be displayed allowing "
                     "you to delete the recording when it has finished "
                     "playing."));
@@ -1525,7 +1525,7 @@
 // {
 //     HostCheckBox *gc = new HostCheckBox("PlaybackPreview");
 //     gc->setLabel(QObject::tr("Display live preview of recordings"));
-//     gc->setValue(true);
+//     gc->setValue(false);
 //     gc->setHelpText(QObject::tr("If enabled, a preview of the recording "
 //                     "will play in a small window on the \"Watch a "
 //                     "Recording\" menu."));
