diff -Naur mythtv-0.27-20131108-g45459f9-old/mythtv/libs/libmythtv/mpeg/H264Parser.cpp mythtv-0.27-20131108-g45459f9-new/mythtv/libs/libmythtv/mpeg/H264Parser.cpp
--- mythtv-0.27-20131108-g45459f9-old/mythtv/libs/libmythtv/mpeg/H264Parser.cpp	2013-11-08 15:27:55.948450626 +0100
+++ mythtv-0.27-20131108-g45459f9-new/mythtv/libs/libmythtv/mpeg/H264Parser.cpp	2013-11-08 15:28:18.155078236 +0100
@@ -596,7 +596,7 @@
          * determine if it is a keyframe or just a frame
          */
         AU_pending = false;
-        state_changed = true;
+        state_changed = seen_sps;
 
         on_frame = true;
         frame_start_offset = AU_offset;
