diff -Naur mythtv-0.27-20131019-g7323c32-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.27-20131019-g7323c32-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.27-20131019-g7323c32-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2013-10-19 11:34:42.000000000 +0200
+++ mythtv-0.27-20131019-g7323c32-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2013-10-26 11:23:10.491778174 +0200
@@ -3145,7 +3145,7 @@
     {
         buf += m_h264_parser->addBytes(buf, buf_end - buf, 0);
 
-        if (m_h264_parser->stateChanged())
+        if ( m_h264_parser->seen_SPS())
         {
             if (m_h264_parser->FieldType() != H264Parser::FIELD_BOTTOM)
             {
