diff -Naur mythtv-0.27-20131019-g7323c32-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.27-20131019-g7323c32-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.27-20131019-g7323c32-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2013-10-19 11:34:42.000000000 +0200
+++ mythtv-0.27-20131019-g7323c32-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2013-10-27 10:24:29.310875867 +0100
@@ -3152,7 +3152,8 @@
                 if (m_h264_parser->onFrameStart())
                     ++num_frames;
 
-                if (!m_h264_parser->onKeyFrameStart())
+                if (!m_h264_parser->onKeyFrameStart() ||
+                    !m_h264_parser->seen_SPS())
                     continue;
             }
             else
