diff -Naur mythtv-0.27-20131019-g7323c32-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.27-20131019-g7323c32-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.27-20131019-g7323c32-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2013-10-21 10:30:09.841262523 +0200
+++ mythtv-0.27-20131019-g7323c32-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2013-10-21 10:30:30.704490212 +0200
@@ -3281,7 +3281,7 @@
         }
     }
 
-    if (framesRead == 0 && !justAfterChange &&
+    if (false && framesRead == 0 && !justAfterChange &&
         !(pkt->flags & AV_PKT_FLAG_KEY))
     {
         av_free_packet(pkt);
