diff -Naur mythtv-0.27-20140225-g28b768a-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-0.27-20140225-g28b768a-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-0.27-20140225-g28b768a-old/mythtv/libs/libmythtv/tv_rec.cpp	2014-02-27 10:10:39.151188637 +0100
+++ mythtv-0.27-20140225-g28b768a-new/mythtv/libs/libmythtv/tv_rec.cpp	2014-02-27 10:10:54.041154464 +0100
@@ -3989,7 +3989,10 @@
             {
                 SetFlags(kFlagWaitingForSignal);
                 if (curRecording)
-                    signalMonitorDeadline = curRecording->GetScheduledEndTime();
+                {
+                    signalMonitorDeadline = curRecording->GetScheduledEndTime()
+                                            .addSecs(-50);
+                }
                 else
                 {
                     QDateTime expire = MythDate::current();
