diff -Naur mythtv-0.27-20140402-gb10d885-old/mythtv/libs/libmythtv/ringbuffer.cpp mythtv-0.27-20140402-gb10d885-new/mythtv/libs/libmythtv/ringbuffer.cpp
--- mythtv-0.27-20140402-gb10d885-old/mythtv/libs/libmythtv/ringbuffer.cpp	2014-04-03 10:33:06.697128781 +0200
+++ mythtv-0.27-20140402-gb10d885-new/mythtv/libs/libmythtv/ringbuffer.cpp	2014-04-03 10:33:23.583988148 +0200
@@ -552,6 +552,9 @@
 
     if (!mode)
     {
+        poslock.lockForWrite();
+        readpos = 0;
+        poslock.unlock();
         // reset the read offset as we are exiting the internal read mode
         readOffset = 0;
     }
