From 919724888e467596a9bc23525ec82c572158c683 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Sun, 13 Apr 2014 09:37:58 +0700
Subject: [PATCH] Add some logging for liveTV

---
 mythtv/libs/libmythtv/avformatdecoder.cpp |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/avformatdecoder.cpp b/mythtv/libs/libmythtv/avformatdecoder.cpp
index 324b322..552d3fe 100644
--- a/mythtv/libs/libmythtv/avformatdecoder.cpp
+++ b/mythtv/libs/libmythtv/avformatdecoder.cpp
@@ -1101,7 +1101,13 @@ int AvFormatDecoder::OpenFile(RingBuffer *rbuffer, bool novideo,
 
         avfRingBuffer->SetInInit(false);
 
-        if (!found)
+        if (found)
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, LOC +
+                QString("File successfully opened after %1ms")
+                .arg(timer.elapsed()));
+        }
+        else
         {
             LOG(VB_PLAYBACK, LOG_WARNING, LOC +
                 QString("No streams found in ram data after %1ms, defaulting to in-file")
-- 
1.7.10.2

