diff -Naur rpcbind-0.2.0-old/configure.in rpcbind-0.2.0-new/configure.in
--- rpcbind-0.2.0-old/configure.in	2009-05-29 06:38:22.000000000 -0700
+++ rpcbind-0.2.0-new/configure.in	2013-01-03 19:15:05.000000000 -0800
@@ -4,7 +4,7 @@
 # AM_MAINTAINER_MODE
 AC_CONFIG_SRCDIR([src/rpcbind.c])
  AC_PROG_CC
- AM_CONFIG_HEADER(config.h)
+ AC_CONFIG_HEADERS(config.h)
  AC_HEADER_DIRENT
  AC_PREFIX_DEFAULT(/usr)
 
