diff -Naur libdca-0.0.5-old/configure.ac libdca-0.0.5-new/configure.ac
--- libdca-0.0.5-old/configure.ac	2007-04-10 05:28:07.000000000 -0700
+++ libdca-0.0.5-new/configure.ac	2007-04-27 14:19:38.000000000 -0700
@@ -29,7 +29,7 @@
 
     dnl -O3
     changequote(<<,>>)
-    TRY_CFLAGS=`echo "$OPT_CFLAGS $CFLAGS"|sed "s/-O[0-9]*//g"`
+    TRY_CFLAGS=`echo "$OPT_CFLAGS $CFLAGS"|sed "s/-O[s0-9]*//g"`
     changequote([,])
     TRY_CFLAGS="$TRY_CFLAGS -O3"
     AC_TRY_CFLAGS([$TRY_CFLAGS],[OPT_CFLAGS=$TRY_CFLAGS; CFLAGS=""])
