diff -Naur mame-0.163-old/makefile mame-0.163-new/makefile
--- mame-0.163-old/makefile	2015-06-24 09:53:30.000000000 -0700
+++ mame-0.163-new/makefile	2015-07-10 13:43:54.396000023 -0700
@@ -73,6 +73,7 @@
 # OVERRIDE_CC = cc
 # OVERRIDE_CXX = c++
 # OVERRIDE_LD = ld
+# OVERRIDE_AR = ar
 
 # DEPRECATED = 1
 # LTO = 1
@@ -369,6 +370,12 @@
 LD := $(OVERRIDE_LD)
 endif
 endif
+ifdef OVERRIDE_AR
+PARAMS += --AR='$(OVERRIDE_AR)'
+ifndef CROSS_BUILD
+AR := $(OVERRIDE_AR)
+endif
+endif
 
 #-------------------------------------------------
 # sanity check the configuration
diff -Naur mame-0.163-old/scripts/genie.lua mame-0.163-new/scripts/genie.lua
--- mame-0.163-old/scripts/genie.lua	2015-06-24 09:53:30.000000000 -0700
+++ mame-0.163-new/scripts/genie.lua	2015-07-10 13:43:54.396000023 -0700
@@ -169,6 +169,11 @@
 }
 
 newoption {
+	trigger = "AR",
+	description = "AR replacement",
+}
+
+newoption {
 	trigger = "PROFILE",
 	description = "Enable profiling.",
 }
diff -Naur mame-0.163-old/scripts/toolchain.lua mame-0.163-new/scripts/toolchain.lua
--- mame-0.163-old/scripts/toolchain.lua	2015-06-24 09:53:32.000000000 -0700
+++ mame-0.163-new/scripts/toolchain.lua	2015-07-10 13:43:54.397000023 -0700
@@ -373,6 +373,9 @@
 	if (_OPTIONS["LD"] ~= nil) then
 		premake.gcc.ld  = _OPTIONS["LD"]
 	end	
+	if (_OPTIONS["AR"] ~= nil) then
+		premake.gcc.ar  = _OPTIONS["AR"]
+	end	
 	
 	configuration {} -- reset configuration
 
