diff -Naur mame-0.148-old/makefile mame-0.148-new/makefile
--- mame-0.148-old/makefile	2013-01-12 13:59:45.000000000 -0800
+++ mame-0.148-new/makefile	2013-01-12 14:30:41.000000000 -0800
@@ -826,12 +826,12 @@
 
 $(OBJ)/%.lh: $(SRC)/%.lay $(FILE2STR_TARGET)
 	@echo Converting $<...
-	@$(FILE2STR) $< $@ layout_$(basename $(notdir $<))
+	@mame-file2str  $< $@ layout_$(basename $(notdir $<))
 
 $(OBJ)/%.fh: $(SRC)/%.png $(PNG2BDC_TARGET) $(FILE2STR_TARGET)
 	@echo Converting $<...
-	@$(PNG2BDC) $< $(OBJ)/temp.bdc
-	@$(FILE2STR) $(OBJ)/temp.bdc $@ font_$(basename $(notdir $<)) UINT8
+	@mame-png2bdc $< $(OBJ)/temp.bdc
+	@mame-file2str $(OBJ)/temp.bdc $@ font_$(basename $(notdir $<)) UINT8
 
 $(DRIVLISTOBJ): $(DRIVLISTSRC)
 	@echo Compiling $<...
@@ -842,7 +842,7 @@
 
 $(DRIVLISTSRC): $(SRC)/$(TARGET)/$(SUBTARGET).lst $(MAKELIST_TARGET)
 	@echo Building driver list $<...
-	@$(MAKELIST) $< >$@
+	@mame-makelist $< >$@
 
 $(OBJ)/%.a:
 	@echo Archiving $@...
diff -Naur mame-0.148-old/src/emu/cpu/cpu.mak mame-0.148-new/src/emu/cpu/cpu.mak
--- mame-0.148-old/src/emu/cpu/cpu.mak	2013-01-12 13:59:45.000000000 -0800
+++ mame-0.148-new/src/emu/cpu/cpu.mak	2013-01-12 14:30:41.000000000 -0800
@@ -1277,7 +1277,7 @@
 # rule to generate the C files
 $(CPUOBJ)/m68000/m68kops.c: $(M68KMAKE) $(CPUSRC)/m68000/m68k_in.c
 	@echo Generating M68K source files...
-	$(M68KMAKE) $(CPUOBJ)/m68000 $(CPUSRC)/m68000/m68k_in.c
+	mame-m68kmake$(EXE) $(CPUOBJ)/m68000 $(CPUSRC)/m68000/m68k_in.c
 
 # rule to build the generator
 ifneq ($(CROSS_BUILD),1)
