diff -Naur mame-0.148-old/src/osd/sdl/sdl.mak mame-0.148-new/src/osd/sdl/sdl.mak
--- mame-0.148-old/src/osd/sdl/sdl.mak	2013-01-12 13:55:45.000000000 -0800
+++ mame-0.148-new/src/osd/sdl/sdl.mak	2013-01-12 14:30:11.000000000 -0800
@@ -685,9 +685,9 @@
 
 # the new debugger relies on GTK+ in addition to the base SDLMAME needs
 # Non-X11 builds can not use the debugger
-INCPATH += `pkg-config --cflags-only-I gtk+-2.0` `pkg-config --cflags-only-I gconf-2.0`
-CCOMFLAGS += `pkg-config --cflags-only-other gtk+-2.0` `pkg-config --cflags-only-other gconf-2.0`
-LIBS += `pkg-config --libs gtk+-2.0` `pkg-config --libs gconf-2.0`
+#INCPATH += `pkg-config --cflags-only-I gtk+-2.0` `pkg-config --cflags-only-I gconf-2.0`
+#CCOMFLAGS += `pkg-config --cflags-only-other gtk+-2.0` `pkg-config --cflags-only-other gconf-2.0`
+#LIBS += `pkg-config --libs gtk+-2.0` `pkg-config --libs gconf-2.0`
 #CCOMFLAGS += -DGTK_DISABLE_DEPRECATED
 
 # The newer debugger uses QT
