diff --speed-large-files --no-dereference --minimal -Naur wakelan-1.1/configure wakelan-1.1/configure
--- wakelan-1.1/configure	1998-08-30 09:56:56.000000000 +0200
+++ wakelan-1.1/configure	2024-05-06 19:17:00.040321643 +0200
@@ -633,7 +633,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 635 "configure"
 #include "confdefs.h"
-main(){return(0);}
+int main(){return(0);}
 EOF
 if { (eval echo configure:639: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest; then
   ac_cv_prog_cc_works=yes
