diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/hif.c smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/hif.c
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/hif.c	2024-10-01 11:03:26.080463171 +0200
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/hif.c	2024-10-01 11:03:09.040462731 +0200
@@ -48,7 +48,7 @@
 #endif
 #include "rx_linux.h"
 
-static char *wf_cfg = "/usr/lib/firmware/smartchip/wifi.cfg";
+static char *wf_cfg = "/lib/firmware/s9083s/wifi.cfg";
 
 #define RX_CMD_PARAM_LENGTH       8
 #define TX_CMD_PARAM_LENGTH       12
@@ -73,7 +73,7 @@
 #ifdef CONFIG_RICHV200
 static char *fw = "/usr/lib/firmware/ram-fw-9188-new-r1751-enc.bin";
 #else
-static char *fw = "/usr/lib/firmware/ram-fw-908x-old-r1549.bin";
+static char *fw = "/lib/firmware/s9083s/ram-fw-908x-old-r1549.bin";
 #endif
 #endif
 module_param(fw, charp, 0644);
