diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/os_porting/wf_os_api_mix.c smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/os_porting/wf_os_api_mix.c
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/os_porting/wf_os_api_mix.c	2022-03-21 09:42:51.000000000 +0100
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/os_porting/wf_os_api_mix.c	2024-04-20 14:53:57.286964818 +0200
@@ -211,7 +211,7 @@
 wf_u32 wf_os_api_rand32 (void)
 {
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0))
-    return prandom_u32();
+    return get_random_u32();
 #elif (LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,18))
     u32 random_int;
     get_random_bytes(&random_int, 4);
diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/proc.c smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/proc.c
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/proc.c	2022-03-21 09:42:51.000000000 +0100
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/proc.c	2024-04-20 14:48:43.733630441 +0200
@@ -692,7 +692,7 @@
     return 0;
 #else
 
-    ssize_t index = (ssize_t)PDE_DATA(file_inode(file));
+    ssize_t index = (ssize_t)pde_data(file_inode(file));
     const struct wf_proc_handle *hdl = proc_hdls + index;
     ssize_t (*write)(struct file *, const char __user *, size_t, loff_t *, void *) = hdl->write;
 
@@ -707,7 +707,7 @@
 
 static int wf_proc_open(struct inode *inode, struct file *file)
 {
-    ssize_t index = (ssize_t)PDE_DATA(inode);
+    ssize_t index = (ssize_t)pde_data(inode);
     const struct wf_proc_handle *hdl = proc_hdls + index;
     void *private = proc_get_parent_data(inode);
 
diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/rx_linux.c smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/rx_linux.c
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/rx_linux.c	2022-03-21 09:42:51.000000000 +0100
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/rx_linux.c	2024-04-20 14:50:59.560297595 +0200
@@ -148,7 +148,7 @@
         else
         {
             //LOG_I("[%s, %d]", __func__, __LINE__);
-            ret = netif_rx_ni(pskb);
+            ret = netif_rx(pskb);
         }
 
         if(ret == NET_RX_SUCCESS)
@@ -180,7 +180,7 @@
             else
             {
                 //LOG_I("[%s, %d]", __func__, __LINE__);
-                ret = netif_rx_ni(pskb);
+                ret = netif_rx(pskb);
             }
 
             if(ret == NET_RX_SUCCESS)
@@ -212,7 +212,7 @@
             else
             {
                 //LOG_I("[%s, %d]", __func__, __LINE__);
-                ret = netif_rx_ni(pskb);
+                ret = netif_rx(pskb);
             }
 
             if(ret == NET_RX_SUCCESS)
@@ -253,7 +253,7 @@
             else
             {
                 //LOG_I("[%s, %d]", __func__, __LINE__);
-                ret = netif_rx_ni(pskb);
+                ret = netif_rx(pskb);
             }
             if(ret == NET_RX_SUCCESS)
             {
diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/nic/adhoc.c smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/nic/adhoc.c
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/nic/adhoc.c	2022-03-21 09:42:51.000000000 +0100
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/nic/adhoc.c	2024-04-20 16:50:48.856988030 +0200
@@ -44,7 +44,7 @@
 inline static wf_u32 wf_adhoc_andom32(void)
 {
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0))
-    return prandom_u32();
+    return get_random_u32();
 #elif (LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,18))
     u32 random_int;
     get_random_bytes(&random_int, 4);
diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.c smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.c
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.c	2022-03-21 09:42:51.000000000 +0100
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.c	2024-04-20 16:48:43.310320949 +0200
@@ -438,7 +438,7 @@
     }
 }
 
-int isspace(int x)
+int isspace(x)
 {
     if(x==' '||x=='\t'||x=='\n'||x=='\f'||x=='\b'||x=='\r')
         return 1;
diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/mp/mp_trx.c smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/mp/mp_trx.c
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/mp/mp_trx.c	2022-03-21 09:42:51.000000000 +0100
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/mp/mp_trx.c	2024-04-20 16:50:29.433654626 +0200
@@ -112,7 +112,7 @@
 static wf_u32 mp_random32(void)
 {
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0))
-    return prandom_u32();
+    return get_random_u32();
 #elif (LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,18))
     wf_u32 random_int;
     get_random_bytes(&random_int, 4);
diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.c smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.c
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.c	2024-04-20 17:07:32.460324700 +0200
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.c	2024-04-20 17:04:03.170324005 +0200
@@ -438,7 +438,7 @@
     }
 }
 
-int isspace(x)
+int _isspace(int x)
 {
     if(x==' '||x=='\t'||x=='\n'||x=='\f'||x=='\b'||x=='\r')
         return 1;
@@ -460,7 +460,7 @@
     int sign;           /* if '-', then negative, otherwise positive */
 
     /* skip whitespace */
-    while ( isspace((int)(unsigned char)*nptr) )
+    while ( _isspace((int)(unsigned char)*nptr) )
         ++nptr;
 
     c = (int)(unsigned char)*nptr++;
diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.h smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.h
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.h	2022-03-21 09:42:51.000000000 +0100
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/iw/iw_priv_func.h	2024-04-20 17:06:07.833657769 +0200
@@ -56,6 +56,7 @@
 int wf_iw_ars(struct net_device *dev, struct iw_request_info *info, union iwreq_data *wdata, char *extra);
 int wf_iw_cmddl(struct net_device *dev, struct iw_request_info *info, union iwreq_data *wdata, char *extra);
 int wf_iw_txagg_timestart(struct net_device *dev, struct iw_request_info *info, union iwreq_data *wdata, char *extra);
+int _isspace(int x);
 
 #endif
 
diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/usb.c smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/usb.c
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/usb.c	2022-03-21 09:42:51.000000000 +0100
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/usb.c	2024-04-20 17:19:44.343660480 +0200
@@ -1098,7 +1098,7 @@
     .suspend        =   wf_usb_suspend,
     .resume         =   wf_usb_resume,
     .reset_resume   =   wf_usb_resume,
-    .drvwrap.driver.shutdown    = wf_usb_shutdown,
+    .driver.shutdown    = wf_usb_shutdown,
     .supports_autosuspend       = 1,
 };
 
diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/hif.c smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/hif.c
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/hif.c	2022-03-21 09:42:51.000000000 +0100
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/os/linux/hif/hif.c	2024-04-20 17:23:17.723661183 +0200
@@ -26,6 +26,7 @@
 #include <linux/usb.h>
 #include <linux/time.h>
 #include <linux/platform_device.h>
+#include <linux/of.h>
 
 #include "wf_debug.h"
 #include "wf_list.h"
diff --speed-large-files --no-dereference --minimal -Naur smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/Makefile smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/Makefile
--- smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/Makefile	2022-03-21 09:42:51.000000000 +0100
+++ smartchip-5d41ac040ecd2072498da88100305c1b444c12c7/trunk_driver/Makefile	2024-04-20 17:27:03.393661930 +0200
@@ -45,8 +45,8 @@
   CONFIG_CMD_METHOD         = mailbox
   CONFIG_WIFI_MODE          = all
   CONFIG_WIFI_FRAMEWORK     = wext
-  CONFIG_HIF_PORT           = both
-  CONFIG_CHIP               = s9188
+  CONFIG_HIF_PORT           = both
+  CONFIG_CHIP               = s9083
   CONFIG_POWER_SAVING       = n
   CONFIG_ARS_SUPPORT        = firmware
 
