diff --speed-large-files --no-dereference --minimal -Naur nvtop-33cf46840f58975d593c743b091eb1e37e323378/src/extract_gpuinfo_panfrost.c nvtop-33cf46840f58975d593c743b091eb1e37e323378/src/extract_gpuinfo_panfrost.c
--- nvtop-33cf46840f58975d593c743b091eb1e37e323378/src/extract_gpuinfo_panfrost.c	2025-02-20 10:54:38.000000000 +0100
+++ nvtop-33cf46840f58975d593c743b091eb1e37e323378/src/extract_gpuinfo_panfrost.c	2025-03-19 16:21:27.730109123 +0100
@@ -203,6 +203,7 @@
   if (prof_info->sysfs_filename == NULL)
     return false;
 
+  fprintf(stderr, "Reading profiling state from: %s\n", prof_info->sysfs_filename);
   fprofiling = fopen(prof_info->sysfs_filename, "r");
   if (fprofiling == NULL) {
     fprintf(stderr, "Profiling state not available in Panfrost\n");
