diff --speed-large-files --no-dereference --minimal -Naur nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/CMakeLists.txt nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/CMakeLists.txt
--- nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/CMakeLists.txt	2023-11-28 16:54:43.000000000 +0100
+++ nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/CMakeLists.txt	2023-12-07 18:18:13.233327425 +0100
@@ -1,3 +1,5 @@
+message(STATUS "Libdrm_INCLUDE_DIRS: ${Libdrm_INCLUDE_DIRS}")
+include_directories(${Libdrm_INCLUDE_DIRS})
 include(CheckCSourceCompiles)
 
 configure_file(
