diff --speed-large-files --no-dereference --minimal -Naur nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_amdgpu.c nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_amdgpu.c
--- nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_amdgpu.c	2023-11-28 16:54:43.000000000 +0100
+++ nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_amdgpu.c	2023-12-06 12:29:56.016435921 +0100
@@ -50,6 +50,8 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <uthash.h>
+#include "panfrost_drm.h"
+#include "panfrost_utils.h"
 #include <xf86drm.h>
 
 // extern
diff --speed-large-files --no-dereference --minimal -Naur nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_mali_common.c nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_mali_common.c
--- nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_mali_common.c	2023-11-28 16:54:43.000000000 +0100
+++ nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_mali_common.c	2023-12-06 11:53:05.766483470 +0100
@@ -24,6 +24,8 @@
 #include <dlfcn.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <drm/drm.h>
+#include <drm/drm_mode.h>
 #include <xf86drm.h>
 
 #include "mali_common.h"
diff --speed-large-files --no-dereference --minimal -Naur nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_panfrost.c nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_panfrost.c
--- nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_panfrost.c	2023-11-28 16:54:43.000000000 +0100
+++ nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_panfrost.c	2023-12-06 11:52:09.866484679 +0100
@@ -21,6 +21,8 @@
 
 #include <sys/stat.h>
 #include <sys/sysmacros.h>
+#include <drm/drm.h>
+#include <drm/drm_mode.h>
 #include <xf86drm.h>
 
 #include "nvtop/device_discovery.h"
diff --speed-large-files --no-dereference --minimal -Naur nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_panthor.c nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_panthor.c
--- nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_panthor.c	2023-11-28 16:54:43.000000000 +0100
+++ nvtop-eb56ee86ae0836872ee182fa2a15b1f1434d8efe/src/extract_gpuinfo_panthor.c	2023-12-06 11:52:49.546483815 +0100
@@ -18,6 +18,8 @@
  * along with nvtop.  If not, see <http://www.gnu.org/licenses/>.
  *
  */
+#include <drm/drm.h>
+#include <drm/drm_mode.h>
 #include <xf86drm.h>
 
 #include "nvtop/device_discovery.h"
