diff --speed-large-files --no-dereference --minimal -Naur json-c-json-c-0.17/doc/CMakeLists.txt json-c-json-c-0.17/doc/CMakeLists.txt
--- json-c-json-c-0.17/doc/CMakeLists.txt	2023-08-12 21:00:16.000000000 +0200
+++ json-c-json-c-0.17/doc/CMakeLists.txt	2023-12-11 11:20:16.950026366 +0100
@@ -1,6 +1,6 @@
 # generate doxygen documentation for json-c API
 
-find_package(Doxygen)
+#find_package(Doxygen)
 
 if (DOXYGEN_FOUND)
 
