diff -Naur xf86-video-intel-2.21.0-old/configure.ac xf86-video-intel-2.21.0-new/configure.ac
--- xf86-video-intel-2.21.0-old/configure.ac	2013-01-30 09:14:38.000000000 -0800
+++ xf86-video-intel-2.21.0-new/configure.ac	2013-02-02 05:57:49.000000000 -0800
@@ -53,6 +53,8 @@
 AC_PROG_LIBTOOL
 AC_SYS_LARGEFILE
 
+PKG_PROG_PKG_CONFIG
+
 # Are we in a git checkout?
 dot_git=no
 if test -e .git; then
@@ -352,7 +354,8 @@
 PKG_CHECK_MODULES(DRI2, [dri2proto >= 2.6],, DRI2=no)
 PKG_CHECK_MODULES(PCIACCESS, [pciaccess >= 0.10])
 
-sdkdir=`$PKG_CONFIG --variable=sdkdir xorg-server`
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 if test "x$enable_dri" != "xno"; then
         save_CFLAGS="$CFLAGS"
