diff -Naur libtirpc-0.2.2-old/configure.ac libtirpc-0.2.2-new/configure.ac
--- libtirpc-0.2.2-old/configure.ac	2011-05-02 05:10:40.000000000 -0700
+++ libtirpc-0.2.2-new/configure.ac	2011-07-03 22:38:39.000000000 -0700
@@ -1,7 +1,7 @@
 AC_INIT(libtirpc, 0.2.2)
 AM_INIT_AUTOMAKE(libtirpc, 0.2.2)
 AM_MAINTAINER_MODE
-AC_CONFIG_SRCDIR([src/auth_des.c])
+AC_CONFIG_SRCDIR([src/auth_none.c])
 AC_CONFIG_MACRO_DIR([m4])
 
 AC_ARG_ENABLE(gss,[  --enable-gss            Turn on gss api], [case "${enableval}" in
diff -Naur libtirpc-0.2.2-old/Makefile.am libtirpc-0.2.2-new/Makefile.am
--- libtirpc-0.2.2-old/Makefile.am	2011-05-02 05:10:40.000000000 -0700
+++ libtirpc-0.2.2-new/Makefile.am	2011-07-03 22:35:46.000000000 -0700
@@ -22,16 +22,13 @@
 			 tirpc/rpc/pmap_prot.h \
 			 tirpc/rpc/pmap_clnt.h \
 			 tirpc/rpc/nettype.h \
-			 tirpc/rpc/des.h \
-			 tirpc/rpc/des_crypt.h \
 			 tirpc/rpc/clnt_stat.h \
 			 tirpc/rpc/clnt_soc.h \
 			 tirpc/rpc/clnt.h \
 			 tirpc/rpc/auth_unix.h \
 			 tirpc/rpc/auth_kerb.h \
 			 tirpc/rpc/auth.h \
-			 tirpc/rpc/auth_gss.h \
-			 tirpc/rpc/auth_des.h
+			 tirpc/rpc/auth_gss.h
     
 pkgconfigdir=$(libdir)/pkgconfig
 pkgconfig_DATA = libtirpc.pc
diff -Naur libtirpc-0.2.2-old/man/Makefile.am libtirpc-0.2.2-new/man/Makefile.am
--- libtirpc-0.2.2-old/man/Makefile.am	2011-05-02 05:10:40.000000000 -0700
+++ libtirpc-0.2.2-new/man/Makefile.am	2011-07-03 22:32:03.000000000 -0700
@@ -1,6 +1,6 @@
 
 man5_MANS = netconfig.5
-man3_MANS = bindresvport.3t des_crypt.3t getnetconfig.3t getnetpath.3t \
+man3_MANS = bindresvport.3t getnetconfig.3t getnetpath.3t \
 		getrpcent.3t getrpcport.3t rpc.3t rpc_clnt_auth.3t rpc_clnt_calls.3t \
 		rpc_clnt_create.3t rpc_secure.3t rpc_soc.3t rpc_svc_calls.3t \
 		rpc_svc_create.3t rpc_svc_err.3t rpc_svc_reg.3t rpc_xdr.3t rtime.3t
diff -Naur libtirpc-0.2.2-old/src/Makefile.am libtirpc-0.2.2-new/src/Makefile.am
--- libtirpc-0.2.2-old/src/Makefile.am	2011-05-02 05:10:40.000000000 -0700
+++ libtirpc-0.2.2-new/src/Makefile.am	2011-07-03 22:03:10.000000000 -0700
@@ -50,7 +50,7 @@
         rpc_callmsg.c rpc_generic.c rpc_soc.c rpcb_clnt.c rpcb_prot.c \
         rpcb_st_xdr.c svc.c svc_auth.c svc_dg.c svc_auth_unix.c svc_generic.c \
         svc_raw.c svc_run.c svc_simple.c svc_vc.c getpeereid.c \
-        auth_time.c auth_des.c authdes_prot.c des_crypt.c
+        auth_time.c
 
 ## XDR
 libtirpc_la_SOURCES += xdr.c xdr_rec.c xdr_array.c xdr_float.c xdr_mem.c xdr_reference.c xdr_stdio.c
diff -Naur libtirpc-0.2.2-old/src/rpc_soc.c libtirpc-0.2.2-new/src/rpc_soc.c
--- libtirpc-0.2.2-old/src/rpc_soc.c	2011-05-02 05:10:40.000000000 -0700
+++ libtirpc-0.2.2-new/src/rpc_soc.c	2011-07-03 22:10:41.000000000 -0700
@@ -516,38 +516,6 @@
 }
 
 /*
- * Create the client des authentication object. Obsoleted by
- * authdes_seccreate().
- */
-AUTH *
-authdes_create(servername, window, syncaddr, ckey)
-	char *servername;		/* network name of server */
-	u_int window;			/* time to live */
-	struct sockaddr *syncaddr;	/* optional hostaddr to sync with */
-	des_block *ckey;		/* optional conversation key to use */
-{
-	AUTH *dummy;
-	AUTH *nauth;
-	char hostname[NI_MAXHOST];
-
-	if (syncaddr) {
-		/*
-		 * Change addr to hostname, because that is the way
-		 * new interface takes it.
-		 */
-		if (getnameinfo(syncaddr, sizeof(syncaddr), hostname,
-		    sizeof hostname, NULL, 0, 0) != 0)
-			goto fallback;
-
-		nauth = authdes_seccreate(servername, window, hostname, ckey);
-		return (nauth);
-	}
-fallback:
-	dummy = authdes_seccreate(servername, window, NULL, ckey);
-	return (dummy);
-}
-
-/*
  * Create a client handle for a unix connection. Obsoleted by clnt_vc_create()
  */
 CLIENT *
diff -Naur libtirpc-0.2.2-old/tirpc/rpc/auth.h libtirpc-0.2.2-new/tirpc/rpc/auth.h
--- libtirpc-0.2.2-old/tirpc/rpc/auth.h	2011-05-02 05:10:40.000000000 -0700
+++ libtirpc-0.2.2-new/tirpc/rpc/auth.h	2011-07-03 22:11:31.000000000 -0700
@@ -268,38 +268,6 @@
  * These are the various implementations of client side authenticators.
  */
 
-/*
- * System style authentication
- * AUTH *authunix_create(machname, uid, gid, len, aup_gids)
- *	char *machname;
- *	int uid;
- *	int gid;
- *	int len;
- *	int *aup_gids;
- */
-__BEGIN_DECLS
-extern AUTH *authunix_create(char *, uid_t, uid_t, int, uid_t *);
-extern AUTH *authunix_create_default(void);	/* takes no parameters */
-extern AUTH *authnone_create(void);		/* takes no parameters */
-__END_DECLS
-/*
- * DES style authentication
- * AUTH *authsecdes_create(servername, window, timehost, ckey)
- * 	char *servername;		- network name of server
- *	u_int window;			- time to live
- * 	const char *timehost;			- optional hostname to sync with
- * 	des_block *ckey;		- optional conversation key to use
- */
-__BEGIN_DECLS
-extern AUTH *authdes_create (char *, u_int, struct sockaddr *, des_block *);
-extern AUTH *authdes_seccreate (const char *, const u_int, const  char *,
-    const  des_block *);
-__END_DECLS
-
-__BEGIN_DECLS
-extern bool_t xdr_opaque_auth		(XDR *, struct opaque_auth *);
-__END_DECLS
-
 #define authsys_create(c,i1,i2,i3,ip) authunix_create((c),(i1),(i2),(i3),(ip))
 #define authsys_create_default() authunix_create_default()
 
diff -Naur libtirpc-0.2.2-old/tirpc/rpc/rpc.h libtirpc-0.2.2-new/tirpc/rpc/rpc.h
--- libtirpc-0.2.2-old/tirpc/rpc/rpc.h	2011-05-02 05:10:40.000000000 -0700
+++ libtirpc-0.2.2-new/tirpc/rpc/rpc.h	2011-07-03 22:34:26.000000000 -0700
@@ -52,12 +52,6 @@
 #include <rpc/rpc_msg.h>	/* protocol for rpc messages */
 #include <rpc/auth_unix.h>	/* protocol for unix style cred */
 
-/*
- *  Uncomment-out the next line if you are building the rpc library with
- *  DES Authentication (see the README file in the secure_rpc/ directory).
- */
-#include <rpc/auth_des.h>	/* protocol for des style cred */
-
 #ifdef HAVE_RPCSEC_GSS
 #include <rpc/auth_gss.h>   /* RPCSEC_GSS */
 #endif
