diff -Naur texinfo-4.13-old/configure.ac texinfo-4.13-new/configure.ac
--- texinfo-4.13-old/configure.ac	2008-09-18 11:45:56.000000000 -0700
+++ texinfo-4.13-new/configure.ac	2012-07-16 11:54:36.000000000 -0700
@@ -13,7 +13,7 @@
 
 dnl Must come before AM_INIT_AUTOMAKE.
 AC_CONFIG_AUX_DIR([build-aux])
-AM_INIT_AUTOMAKE([1.10.1 readme-alpha dist-lzma])
+AM_INIT_AUTOMAKE([1.10.1 readme-alpha])
 
 # Minimum Autoconf version required.
 AC_PREREQ(2.59)
