diff --speed-large-files --no-dereference --minimal -Naur libclc-18.1.6.src/cmake/CMakeDetermineCLCCompiler.cmake libclc-18.1.6.src/cmake/CMakeDetermineCLCCompiler.cmake
--- libclc-18.1.6.src/cmake/CMakeDetermineCLCCompiler.cmake	2024-05-18 16:14:07.000000000 +0200
+++ libclc-18.1.6.src/cmake/CMakeDetermineCLCCompiler.cmake	2024-05-26 14:20:31.249879279 +0200
@@ -12,7 +12,8 @@
 
 set(CMAKE_CLC_COMPILER_ENV_VAR "CLC_COMPILER")
 set(CMAKE_CLC_ARCHIVE_ENV_VAR "CLC_LINKER")
-find_file(clc_comp_in CMakeCLCCompiler.cmake.in PATHS ${CMAKE_ROOT}/Modules ${CMAKE_MODULE_PATH})
+# find_file(clc_comp_in CMakeCLCCompiler.cmake.in PATHS ${CMAKE_ROOT}/Modules ${CMAKE_MODULE_PATH})
+set(clc_comp_in "${CMAKE_MODULE_PATH}/CMakeCLCCompiler.cmake.in" )
 # configure all variables set in this file
 configure_file(${clc_comp_in} ${CMAKE_PLATFORM_INFO_DIR}/CMakeCLCCompiler.cmake @ONLY)
 mark_as_advanced(clc_comp_in)
diff --speed-large-files --no-dereference --minimal -Naur libclc-18.1.6.src/cmake/CMakeDetermineLLAsmCompiler.cmake libclc-18.1.6.src/cmake/CMakeDetermineLLAsmCompiler.cmake
--- libclc-18.1.6.src/cmake/CMakeDetermineLLAsmCompiler.cmake	2024-05-18 16:14:07.000000000 +0200
+++ libclc-18.1.6.src/cmake/CMakeDetermineLLAsmCompiler.cmake	2024-05-26 14:21:02.266545287 +0200
@@ -18,7 +18,8 @@
 set(CMAKE_LLAsm_PREPROCESSOR_ENV_VAR "LL_PREPROCESSOR")
 set(CMAKE_LLAsm_COMPILER_ENV_VAR "LL_ASSEMBLER")
 set(CMAKE_LLAsm_ARCHIVE_ENV_VAR "LL_LINKER")
-find_file(ll_comp_in CMakeLLAsmCompiler.cmake.in PATHS ${CMAKE_ROOT}/Modules ${CMAKE_MODULE_PATH})
+# find_file(ll_comp_in CMakeLLAsmCompiler.cmake.in PATHS ${CMAKE_ROOT}/Modules ${CMAKE_MODULE_PATH})
+set(ll_comp_in "${CMAKE_MODULE_PATH}/CMakeLLAsmCompiler.cmake.in" )
 # configure all variables set in this file
 configure_file(${ll_comp_in} ${CMAKE_PLATFORM_INFO_DIR}/CMakeLLAsmCompiler.cmake @ONLY)
 mark_as_advanced(ll_comp_in)
diff --speed-large-files --no-dereference --minimal -Naur libclc-18.1.6.src/CMakeLists.txt libclc-18.1.6.src/CMakeLists.txt
--- libclc-18.1.6.src/CMakeLists.txt	2024-05-18 16:14:07.000000000 +0200
+++ libclc-18.1.6.src/CMakeLists.txt	2024-05-26 14:19:59.413213280 +0200
@@ -326,7 +326,7 @@
 
 			# Add prepare target
 			add_custom_command( OUTPUT "${obj_suffix}"
-				            COMMAND prepare_builtins -o
+				            COMMAND ./prepare_builtins -o
 					    "${obj_suffix}"
 					    "builtins.opt.${obj_suffix}"
 					    DEPENDS "opt.${obj_suffix}"
