diff -Naur freeglut-2.8.0-old/configure.ac freeglut-2.8.0-new/configure.ac
--- freeglut-2.8.0-old/configure.ac	2011-09-27 18:37:26.000000000 -0700
+++ freeglut-2.8.0-new/configure.ac	2012-09-07 02:44:02.000000000 -0700
@@ -115,7 +115,6 @@
 	progs/demos/Makefile
 	progs/demos/One/Makefile
 	progs/demos/shapes/Makefile
-	progs/demos/smooth_opengl3/Makefile
 	progs/demos/spaceball/Makefile
 	progs/demos/subwin/Makefile
 	src/Makefile
diff -Naur freeglut-2.8.0-old/Makefile.am freeglut-2.8.0-new/Makefile.am
--- freeglut-2.8.0-old/Makefile.am	2009-04-05 06:13:31.000000000 -0700
+++ freeglut-2.8.0-new/Makefile.am	2012-09-07 02:44:02.000000000 -0700
@@ -46,10 +46,7 @@
 	progs/demos/demos_vs2008.sln \
 	progs/demos/shapes/shapesStatic.dsp \
 	progs/demos/shapes/shapesStatic_vs2008.vcproj \
-	progs/demos/shapes/shapes_vs2008.vcproj \
-	progs/demos/smooth_opengl3/smooth_opengl3Static.dsp \
-	progs/demos/smooth_opengl3/smooth_opengl3Static_vs2008.vcproj \
-	progs/demos/smooth_opengl3/smooth_opengl3_vs2008.vcproj
+	progs/demos/shapes/shapes_vs2008.vcproj
 
 # Thanks to WinDoze, some files in CVS have the "execute" bit set when they
 # shoudn't. We could fix this if we had direct access to the CVS repository,
diff -Naur freeglut-2.8.0-old/Makefile.in freeglut-2.8.0-new/Makefile.in
--- freeglut-2.8.0-old/Makefile.in	2012-01-02 04:18:08.000000000 -0800
+++ freeglut-2.8.0-new/Makefile.in	2012-09-07 02:44:02.000000000 -0700
@@ -275,11 +275,7 @@
 	progs/demos/demos_vs2008.sln \
 	progs/demos/shapes/shapesStatic.dsp \
 	progs/demos/shapes/shapesStatic_vs2008.vcproj \
-	progs/demos/shapes/shapes_vs2008.vcproj \
-	progs/demos/smooth_opengl3/smooth_opengl3Static.dsp \
-	progs/demos/smooth_opengl3/smooth_opengl3Static_vs2008.vcproj \
-	progs/demos/smooth_opengl3/smooth_opengl3_vs2008.vcproj
-
+	progs/demos/shapes/shapes_vs2008.vcproj
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
diff -Naur freeglut-2.8.0-old/progs/demos/Makefile.am freeglut-2.8.0-new/progs/demos/Makefile.am
--- freeglut-2.8.0-old/progs/demos/Makefile.am	2011-03-17 20:21:56.000000000 -0700
+++ freeglut-2.8.0-new/progs/demos/Makefile.am	2012-09-07 02:44:02.000000000 -0700
@@ -1,2 +1,2 @@
 EXTRA_DIST = demos.dsw
-SUBDIRS = CallbackMaker Fractals Fractals_random Lorenz One shapes smooth_opengl3 spaceball subwin
+SUBDIRS = CallbackMaker Fractals Fractals_random Lorenz One shapes spaceball subwin
diff -Naur freeglut-2.8.0-old/progs/demos/Makefile.in freeglut-2.8.0-new/progs/demos/Makefile.in
--- freeglut-2.8.0-old/progs/demos/Makefile.in	2012-01-02 04:18:07.000000000 -0800
+++ freeglut-2.8.0-new/progs/demos/Makefile.in	2012-09-07 02:44:02.000000000 -0700
@@ -215,7 +215,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 EXTRA_DIST = demos.dsw
-SUBDIRS = CallbackMaker Fractals Fractals_random Lorenz One shapes smooth_opengl3 spaceball subwin
+SUBDIRS = CallbackMaker Fractals Fractals_random Lorenz One shapes spaceball subwin
 all: all-recursive
 
 .SUFFIXES:
