diff --speed-large-files --no-dereference --minimal -Naur linux-6.2-rc5/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts linux-6.2-rc5/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts
--- linux-6.2-rc5/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts	2023-01-24 13:37:34.406418835 +0100
+++ linux-6.2-rc5/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts	2023-01-24 10:42:30.493274945 +0100
@@ -256,17 +207,23 @@
 
 &gmac1 {
 	assigned-clocks = <&cru SCLK_GMAC1_RX_TX>, <&cru SCLK_GMAC1>;
-	assigned-clock-parents = <&cru SCLK_GMAC1_RGMII_SPEED>, <&gmac1_clkin>;
-	clock_in_out = "input";
+	assigned-clock-parents = <&cru SCLK_GMAC1_RGMII_SPEED>;
+	assigned-clock-rates = <0>, <125000000>;
+	clock_in_out = "output";
 	phy-handle = <&rgmii_phy1>;
-	phy-mode = "rgmii-id";
-	phy-supply = <&vcc_3v3>;
+	phy-mode = "rgmii";
+	tx_delay = <0x42>;
+	rx_delay = <0x28>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&eth_phy_rst>;
+	snps,reset-gpio = <&gpio3 RK_PB0 GPIO_ACTIVE_LOW>;
+	snps,reset-active-low;
+	snps,reset-delays-us = <0 20000 100000>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&gmac1m1_miim
 		     &gmac1m1_tx_bus2
 		     &gmac1m1_rx_bus2
 		     &gmac1m1_rgmii_clk
-		     &gmac1m1_clkinout
 		     &gmac1m1_rgmii_bus>;
 	status = "okay";
 };
