diff --speed-large-files --no-dereference --minimal -Naur linux-6.2-rc5/arch/arm/boot/dts/bcm2711-rpi.dtsi linux-6.2-rc5/arch/arm/boot/dts/bcm2711-rpi.dtsi
--- linux-6.2-rc5/arch/arm/boot/dts/bcm2711-rpi.dtsi	2023-01-22 01:27:01.000000000 +0100
+++ linux-6.2-rc5/arch/arm/boot/dts/bcm2711-rpi.dtsi	2023-01-24 14:27:54.859697681 +0100
@@ -15,6 +15,9 @@
 		ethernet0 = &genet;
 		pcie0 = &pcie0;
 		blconfig = &blconfig;
+		mmc2 = &sdhost;
+		mmc1 = &sdhci;
+		mmc0 = &emmc2;
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.2-rc5/arch/arm/boot/dts/bcm283x-rpi-led-deprecated.dtsi linux-6.2-rc5/arch/arm/boot/dts/bcm283x-rpi-led-deprecated.dtsi
--- linux-6.2-rc5/arch/arm/boot/dts/bcm283x-rpi-led-deprecated.dtsi	2023-01-22 01:27:01.000000000 +0100
+++ linux-6.2-rc5/arch/arm/boot/dts/bcm283x-rpi-led-deprecated.dtsi	2023-01-24 14:25:46.159699996 +0100
@@ -1,5 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0
 
+#include <dt-bindings/leds/common.h>
+
 / {
 	/*
 	 * This file provides the now deprecated ACT LED to the
@@ -12,7 +14,8 @@
 		led_act: led-act {
 			label = "ACT";
 			default-state = "keep";
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 		};
 	};
 };
