diff --speed-large-files --no-dereference --minimal -Naur linux-6.2/arch/arm64/boot/dts/rockchip/rk3588s.dtsi linux-6.2/arch/arm64/boot/dts/rockchip/rk3588s.dtsi
--- linux-6.2/arch/arm64/boot/dts/rockchip/rk3588s.dtsi	2023-02-21 14:20:39.356722097 +0100
+++ linux-6.2/arch/arm64/boot/dts/rockchip/rk3588s.dtsi	2023-02-21 14:18:38.910059824 +0100
@@ -522,9 +522,11 @@
 
 			scmi_clk: protocol@14 {
 				reg = <0x14>;
-				assigned-clocks = <&scmi_clk SCMI_CLK_CPUB01>,
+				assigned-clocks = <&scmi_clk SCMI_CLK_CPUL>,
+						  <&scmi_clk SCMI_CLK_CPUB01>,
 						  <&scmi_clk SCMI_CLK_CPUB23>;
 				assigned-clock-rates = <1200000000>,
+						       <1200000000>,
 						       <1200000000>;
 				#clock-cells = <1>;
 			};
