diff -Naur cmake-3.10.2-old/Tests/CMakeLists.txt cmake-3.10.2-new/Tests/CMakeLists.txt
--- cmake-3.10.2-old/Tests/CMakeLists.txt	2018-01-18 15:48:42.000000000 +0100
+++ cmake-3.10.2-new/Tests/CMakeLists.txt	2018-07-21 15:03:09.224442247 +0200
@@ -1257,7 +1257,7 @@
   endif()
 
   if(NOT DEFINED CMake_TEST_Qt5)
-    set(CMake_TEST_Qt5 1)
+    set(CMake_TEST_Qt5 0)
   endif()
   if(CMake_TEST_Qt5)
     find_package(Qt5Widgets QUIET NO_MODULE)
