diff --speed-large-files --no-dereference --minimal -Naur cmake-4.3.0/Modules/FindICU.cmake cmake-4.3.0/Modules/FindICU.cmake
--- cmake-4.3.0/Modules/FindICU.cmake	2026-03-17 14:29:16.000000000 +0100
+++ cmake-4.3.0/Modules/FindICU.cmake	2026-03-31 16:19:48.316380251 +0200
@@ -270,7 +270,7 @@
   endforeach()
 
   # Find all ICU libraries
-  list(APPEND icu_library_suffixes "${_lib64}" "lib")
+  # list(APPEND icu_library_suffixes "${_lib64}" "lib")
   set(static_prefix )
   # static icu libraries compiled with MSVC have the prefix 's'
   if(MSVC)
