diff -Naur linux-5.8-rc2-old/drivers/gpu/drm/panfrost/panfrost_devfreq.c linux-5.8-rc2-new/drivers/gpu/drm/panfrost/panfrost_devfreq.c
--- linux-5.8-rc2-old/drivers/gpu/drm/panfrost/panfrost_devfreq.c	2020-07-14 13:05:23.489999995 +0200
+++ linux-5.8-rc2-new/drivers/gpu/drm/panfrost/panfrost_devfreq.c	2020-07-15 15:21:05.723333335 +0200
@@ -97,7 +97,7 @@
 	pfdevfreq->gov_data.downdifferential = 5;
 
 	devfreq = devm_devfreq_add_device(dev, &panfrost_devfreq_profile,
-					  DEVFREQ_GOV_SIMPLE_ONDEMAND,
+					  DEVFREQ_GOV_PERFORMANCE,
 					  &pfdevfreq->gov_data);
 	if (IS_ERR(devfreq)) {
 		DRM_DEV_ERROR(dev, "Couldn't initialize GPU devfreq\n");
