diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2023-12-07 12:36:28.806406403 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2023-12-07 12:32:52.709193470 +0100
@@ -322,3 +322,36 @@
 &usb3phy {
 	status = "okay";
 };
+
+&cpu_opp_table {
+	opp-1608000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1608000000>;
+		opp-microvolt-speed0 = <1090000 1090000 1200000>;
+		opp-microvolt-speed1 = <1030000 1030000 1200000>;
+		opp-microvolt-speed2 = <1030000 1030000 1200000>;
+	};
+
+	opp-1704000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1704000000>;
+		opp-microvolt-speed0 = <1120000 1120000 1200000>;
+		opp-microvolt-speed1 = <1060000 1060000 1200000>;
+		opp-microvolt-speed2 = <1060000 1060000 1200000>;
+	};
+
+	opp-1800000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1800000000>;
+		opp-microvolt-speed0 = <1160000 1160000 1200000>;
+		opp-microvolt-speed1 = <1100000 1100000 1200000>;
+		opp-microvolt-speed2 = <1100000 1100000 1200000>;
+	};
+};
+
+&gpu_opp_table {
+	opp-756000000 {
+		opp-hz = /bits/ 64 <756000000>;
+		opp-microvolt = <1040000 1040000 1200000>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-cpu-opp.dtsi linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-cpu-opp.dtsi
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-cpu-opp.dtsi	2023-12-03 07:33:10.000000000 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-cpu-opp.dtsi	2023-12-07 12:14:21.086434984 +0100
@@ -70,33 +70,6 @@
 			opp-microvolt-speed1 = <1000000 1000000 1200000>;
 			opp-microvolt-speed2 = <1000000 1000000 1200000>;
 		};
-
-		opp-1608000000 {
-			clock-latency-ns = <244144>; /* 8 32k periods */
-			opp-hz = /bits/ 64 <1608000000>;
-
-			opp-microvolt-speed0 = <1090000 1090000 1200000>;
-			opp-microvolt-speed1 = <1030000 1030000 1200000>;
-			opp-microvolt-speed2 = <1030000 1030000 1200000>;
-		};
-
-		opp-1704000000 {
-			clock-latency-ns = <244144>; /* 8 32k periods */
-			opp-hz = /bits/ 64 <1704000000>;
-
-			opp-microvolt-speed0 = <1120000 1120000 1200000>;
-			opp-microvolt-speed1 = <1060000 1060000 1200000>;
-			opp-microvolt-speed2 = <1060000 1060000 1200000>;
-		};
-
-		opp-1800000000 {
-			clock-latency-ns = <244144>; /* 8 32k periods */
-			opp-hz = /bits/ 64 <1800000000>;
-
-			opp-microvolt-speed0 = <1160000 1160000 1200000>;
-			opp-microvolt-speed1 = <1100000 1100000 1200000>;
-			opp-microvolt-speed2 = <1100000 1100000 1200000>;
-		};
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts	2023-12-07 12:36:29.316406393 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts	2023-12-07 12:32:35.001450754 +0100
@@ -234,3 +234,36 @@
 	mali-supply = <&reg_dcdcc>;
 	status = "okay";
 };
+
+&cpu_opp_table {
+	opp-1608000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1608000000>;
+		opp-microvolt-speed0 = <1090000 1090000 1200000>;
+		opp-microvolt-speed1 = <1030000 1030000 1200000>;
+		opp-microvolt-speed2 = <1030000 1030000 1200000>;
+	};
+
+	opp-1704000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1704000000>;
+		opp-microvolt-speed0 = <1120000 1120000 1200000>;
+		opp-microvolt-speed1 = <1060000 1060000 1200000>;
+		opp-microvolt-speed2 = <1060000 1060000 1200000>;
+	};
+
+	opp-1800000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1800000000>;
+		opp-microvolt-speed0 = <1160000 1160000 1200000>;
+		opp-microvolt-speed1 = <1100000 1100000 1200000>;
+		opp-microvolt-speed2 = <1100000 1100000 1200000>;
+	};
+};
+
+&gpu_opp_table {
+	opp-756000000 {
+		opp-hz = /bits/ 64 <756000000>;
+		opp-microvolt = <1040000 1040000 1200000>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-gpu-opp.dtsi linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-gpu-opp.dtsi
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-gpu-opp.dtsi	2023-12-03 07:33:10.000000000 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-gpu-opp.dtsi	2023-12-07 12:30:11.296414528 +0100
@@ -10,75 +10,25 @@
 			opp-microvolt = <810000 810000 1200000>;
 		};
 
-		opp-264000000 {
-			opp-hz = /bits/ 64 <264000000>;
-			opp-microvolt = <810000 810000 1200000>;
-		};
-
 		opp-312000000 {
 			opp-hz = /bits/ 64 <312000000>;
 			opp-microvolt = <810000 810000 1200000>;
 		};
 
-		opp-336000000 {
-			opp-hz = /bits/ 64 <336000000>;
-			opp-microvolt = <810000 810000 1200000>;
-		};
-
-		opp-360000000 {
-			opp-hz = /bits/ 64 <360000000>;
-			opp-microvolt = <820000 820000 1200000>;
-		};
-
-		opp-384000000 {
-			opp-hz = /bits/ 64 <384000000>;
-			opp-microvolt = <830000 830000 1200000>;
-		};
-
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
 			opp-microvolt = <840000 840000 1200000>;
 		};
 
-		opp-420000000 {
-			opp-hz = /bits/ 64 <420000000>;
-			opp-microvolt = <850000 850000 1200000>;
-		};
-
-		opp-432000000 {
-			opp-hz = /bits/ 64 <432000000>;
-			opp-microvolt = <860000 860000 1200000>;
-		};
-
-		opp-456000000 {
-			opp-hz = /bits/ 64 <456000000>;
-			opp-microvolt = <870000 870000 1200000>;
-		};
-
 		opp-504000000 {
 			opp-hz = /bits/ 64 <504000000>;
 			opp-microvolt = <890000 890000 1200000>;
 		};
 
-		opp-540000000 {
-			opp-hz = /bits/ 64 <540000000>;
-			opp-microvolt = <910000 910000 1200000>;
-		};
-
-		opp-576000000 {
-			opp-hz = /bits/ 64 <576000000>;
-			opp-microvolt = <930000 930000 1200000>;
-		};
-
 		opp-624000000 {
 			opp-hz = /bits/ 64 <624000000>;
 			opp-microvolt = <950000 950000 1200000>;
 		};
-
-		opp-756000000 {
-			opp-hz = /bits/ 64 <756000000>;
-			opp-microvolt = <1040000 1040000 1200000>;
-		};
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2023-12-07 12:36:29.316406393 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2023-12-07 12:32:16.228121583 +0100
@@ -382,3 +382,36 @@
 &usb3phy {
 	status = "okay";
 };
+
+&cpu_opp_table {
+	opp-1608000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1608000000>;
+		opp-microvolt-speed0 = <1090000 1090000 1200000>;
+		opp-microvolt-speed1 = <1030000 1030000 1200000>;
+		opp-microvolt-speed2 = <1030000 1030000 1200000>;
+	};
+
+	opp-1704000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1704000000>;
+		opp-microvolt-speed0 = <1120000 1120000 1200000>;
+		opp-microvolt-speed1 = <1060000 1060000 1200000>;
+		opp-microvolt-speed2 = <1060000 1060000 1200000>;
+	};
+
+	opp-1800000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1800000000>;
+		opp-microvolt-speed0 = <1160000 1160000 1200000>;
+		opp-microvolt-speed1 = <1100000 1100000 1200000>;
+		opp-microvolt-speed2 = <1100000 1100000 1200000>;
+	};
+};
+
+&gpu_opp_table {
+	opp-756000000 {
+		opp-hz = /bits/ 64 <756000000>;
+		opp-microvolt = <1040000 1040000 1200000>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3-lts.dts linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3-lts.dts
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3-lts.dts	2023-12-07 12:36:29.356406392 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3-lts.dts	2023-12-07 12:32:23.795123089 +0100
@@ -410,3 +410,36 @@
 &usb3phy {
 	status = "okay";
 };
+
+&cpu_opp_table {
+	opp-1608000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1608000000>;
+		opp-microvolt-speed0 = <1090000 1090000 1200000>;
+		opp-microvolt-speed1 = <1030000 1030000 1200000>;
+		opp-microvolt-speed2 = <1030000 1030000 1200000>;
+	};
+
+	opp-1704000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1704000000>;
+		opp-microvolt-speed0 = <1120000 1120000 1200000>;
+		opp-microvolt-speed1 = <1060000 1060000 1200000>;
+		opp-microvolt-speed2 = <1060000 1060000 1200000>;
+	};
+
+	opp-1800000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1800000000>;
+		opp-microvolt-speed0 = <1160000 1160000 1200000>;
+		opp-microvolt-speed1 = <1100000 1100000 1200000>;
+		opp-microvolt-speed2 = <1100000 1100000 1200000>;
+	};
+};
+
+&gpu_opp_table {
+	opp-756000000 {
+		opp-hz = /bits/ 64 <756000000>;
+		opp-microvolt = <1040000 1040000 1200000>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi	2023-12-07 13:03:45.943037841 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi	2023-12-07 13:04:07.826370703 +0100
@@ -5,6 +5,7 @@
 /dts-v1/;
 
 #include "sun50i-h6.dtsi"
+#include "sun50i-h6-cpu-opp.dtsi"
 #include "sun50i-h6-gpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
@@ -267,3 +268,36 @@
 	usb3_vbus-supply = <&reg_vcc5v>;
 	status = "okay";
 };
+
+&cpu_opp_table {
+	opp-1608000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1608000000>;
+		opp-microvolt-speed0 = <1090000 1090000 1200000>;
+		opp-microvolt-speed1 = <1030000 1030000 1200000>;
+		opp-microvolt-speed2 = <1030000 1030000 1200000>;
+	};
+
+	opp-1704000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1704000000>;
+		opp-microvolt-speed0 = <1120000 1120000 1200000>;
+		opp-microvolt-speed1 = <1060000 1060000 1200000>;
+		opp-microvolt-speed2 = <1060000 1060000 1200000>;
+	};
+
+	opp-1800000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1800000000>;
+		opp-microvolt-speed0 = <1160000 1160000 1200000>;
+		opp-microvolt-speed1 = <1100000 1100000 1200000>;
+		opp-microvolt-speed2 = <1100000 1100000 1200000>;
+	};
+};
+
+&gpu_opp_table {
+	opp-756000000 {
+		opp-hz = /bits/ 64 <756000000>;
+		opp-microvolt = <1040000 1040000 1200000>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts	2023-12-07 12:36:29.319739725 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts	2023-12-07 12:31:55.886412632 +0100
@@ -333,3 +333,36 @@
 	usb3_vbus-supply = <&reg_usb_vbus>;
 	status = "okay";
 };
+
+&cpu_opp_table {
+	opp-1608000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1608000000>;
+		opp-microvolt-speed0 = <1090000 1090000 1200000>;
+		opp-microvolt-speed1 = <1030000 1030000 1200000>;
+		opp-microvolt-speed2 = <1030000 1030000 1200000>;
+	};
+
+	opp-1704000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1704000000>;
+		opp-microvolt-speed0 = <1120000 1120000 1200000>;
+		opp-microvolt-speed1 = <1060000 1060000 1200000>;
+		opp-microvolt-speed2 = <1060000 1060000 1200000>;
+	};
+
+	opp-1800000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1800000000>;
+		opp-microvolt-speed0 = <1160000 1160000 1200000>;
+		opp-microvolt-speed1 = <1100000 1100000 1200000>;
+		opp-microvolt-speed2 = <1100000 1100000 1200000>;
+	};
+};
+
+&gpu_opp_table {
+	opp-756000000 {
+		opp-hz = /bits/ 64 <756000000>;
+		opp-microvolt = <1040000 1040000 1200000>;
+	};
+};
