diff --speed-large-files --no-dereference --minimal -Naur linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h313-cpu-opp.dtsi linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h313-cpu-opp.dtsi
--- linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h313-cpu-opp.dtsi	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h313-cpu-opp.dtsi	2024-04-26 18:04:45.030314171 +0200
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+// Copyright (C) 2021 Piotr Oniszczuk <piotr.oniszczuk@gmail.com>
+
+/*
+	X96-q DDR3 vendor Android DT:
+	480000000	900mV
+	600000000	900mV
+	792000000	900mV
+	1008000000	920mV
+	1200000000	980mV
+	1344000000	1120mV
+	1416000000	1140mV
+	1512000000	1160mV
+*/
+
+/ {
+	cpu_opp_table: opp-table-cpu {
+		compatible = "allwinner,sun50i-h616-operating-points";
+		nvmem-cells = <&cpu_speed_grade>;
+		opp-shared;
+
+		opp-480000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <480000000>;
+
+			opp-microvolt-speed0 = <900000 900000 1100000>;
+			opp-microvolt-speed1 = <900000 900000 1100000>;
+			opp-microvolt-speed2 = <900000 900000 1100000>;
+		};
+
+		opp-600000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <600000000>;
+
+			opp-microvolt-speed0 = <900000 900000 1100000>;
+			opp-microvolt-speed1 = <900000 900000 1100000>;
+			opp-microvolt-speed2 = <900000 900000 1100000>;
+		};
+
+		opp-792000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <792000000>;
+		        opp-microvolt-speed0 = <900000 900000 1100000>;
+			opp-microvolt-speed1 = <900000 900000 1100000>;
+			opp-microvolt-speed2 = <900000 900000 1100000>;
+		};
+
+		opp-1008000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1008000000>;
+
+			opp-microvolt-speed0 = <920000 920000 1100000>;
+			opp-microvolt-speed1 = <920000 920000 1100000>;
+			opp-microvolt-speed2 = <920000 920000 1100000>;
+		};
+
+		opp-1200000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1200000000>;
+
+			opp-microvolt-speed0 = <980000 980000 1100000>;
+			opp-microvolt-speed1 = <980000 980000 1100000>;
+			opp-microvolt-speed2 = <980000 980000 1100000>;
+		};
+
+		opp-1512000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1512000000>;
+
+			opp-microvolt-speed0 = <1100000 1100000 1100000>;
+			opp-microvolt-speed1 = <1100000 1100000 1100000>;
+			opp-microvolt-speed2 = <1100000 1100000 1100000>;
+		};
+	};
+};
+
+&cpu0 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu1 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu2 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu3 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1.dtsi linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1.dtsi
--- linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1.dtsi	2024-04-21 21:35:54.000000000 +0200
+++ linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1.dtsi	2024-04-26 18:04:45.030314171 +0200
@@ -6,6 +6,7 @@
 /dts-v1/;
 
 #include "sun50i-h616.dtsi"
+#include "sun50i-h616-cpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi
--- linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi	2024-04-26 18:04:45.030314171 +0200
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+// Copyright (C) 2021 Piotr Oniszczuk <piotr.oniszczuk@gmail.com>
+/*
+  Opi Zero2:bin0  bin1 Tx6s:bin0  bin1
+ 480000000  820mV  880mV  / 820mV  880mV
+ 600000000  820mV  880mV  / 820mV  880mV
+ 792000000  860mV  940mV  / 860mV  940mV
+ 1008000000 900mV  1020mV / 900mV  1020mV
+ 1200000000 960mV  1100mV / 960mV  1100mV
+ 1296000000 1100mV 1100mV / 1100mV 1100mV
+ 1344000000 1120mV 1120mV / 1120mV 1120mV
+ 1512000000 1120mV 1120mV / 1120mV 1120mV
+*/
+
+/ {
+	cpu_opp_table: opp-table-cpu {
+		compatible = "allwinner,sun50i-h616-operating-points";
+		nvmem-cells = <&cpu_speed_grade>;
+		opp-shared;
+
+		opp-480000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <480000000>;
+
+			opp-microvolt-speed0 = <900000 900000 1100000>;
+			opp-microvolt-speed1 = <900000 900000 1100000>;
+			opp-microvolt-speed2 = <900000 900000 1100000>;
+		};
+
+		opp-600000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <600000000>;
+
+			opp-microvolt-speed0 = <900000 900000 1100000>;
+			opp-microvolt-speed1 = <920000 920000 1100000>;
+			opp-microvolt-speed2 = <920000 920000 1100000>;
+		};
+
+		opp-792000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <792000000>;
+		        opp-microvolt-speed0 = <900000 900000 1100000>;
+			opp-microvolt-speed1 = <940000 940000 1100000>;
+			opp-microvolt-speed2 = <940000 940000 1100000>;
+		};
+
+		opp-1008000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1008000000>;
+
+			opp-microvolt-speed0 = <900000 900000 1100000>;
+			opp-microvolt-speed1 = <1020000 1020000 1100000>;
+			opp-microvolt-speed2 = <1020000 1020000 1100000>;
+		};
+
+		opp-1200000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1200000000>;
+
+			opp-microvolt-speed0 = <960000 960000 1100000>;
+			opp-microvolt-speed1 = <1100000 1100000 1100000>;
+			opp-microvolt-speed2 = <1100000 1100000 1100000>;
+		};
+
+		opp-1344000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1344000000>;
+
+			opp-microvolt-speed0 = <1050000 1050000 1100000>;
+			opp-microvolt-speed1 = <1100000 1100000 1100000>;
+			opp-microvolt-speed2 = <1100000 1100000 1100000>;
+		};
+
+		opp-1416000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1416000000>;
+
+			opp-microvolt-speed0 = <1100000 1100000 1100000>;
+			opp-microvolt-speed1 = <1100000 1100000 1100000>;
+			opp-microvolt-speed2 = <1100000 1100000 1100000>;
+		};
+
+		opp-1512000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1512000000>;
+
+			opp-microvolt-speed0 = <1100000 1100000 1100000>;
+			opp-microvolt-speed1 = <1100000 1100000 1100000>;
+			opp-microvolt-speed2 = <1100000 1100000 1100000>;
+		};
+
+		opp-1608000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1608000000>;
+
+			opp-microvolt-speed0 = <1120000 1120000 1120000>;
+			opp-microvolt-speed1 = <1120000 1120000 1120000>;
+			opp-microvolt-speed2 = <1120000 1120000 1120000>;
+		};
+	};
+};
+
+&cpu0 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu1 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu2 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu3 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-04-26 18:06:54.256981268 +0200
+++ linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-04-26 18:04:45.033647505 +0200
@@ -29,6 +29,9 @@
 			reg = <0>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			#cooling-cells = <2>;
+			operating-points-v2 = <&cpu_opp_table>;
 		};
 
 		cpu1: cpu@1 {
@@ -37,6 +40,9 @@
 			reg = <1>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			#cooling-cells = <2>;
+			operating-points-v2 = <&cpu_opp_table>;
 		};
 
 		cpu2: cpu@2 {
@@ -45,6 +51,9 @@
 			reg = <2>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			#cooling-cells = <2>;
+			operating-points-v2 = <&cpu_opp_table>;
 		};
 
 		cpu3: cpu@3 {
@@ -53,6 +62,9 @@
 			reg = <3>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			#cooling-cells = <2>;
+			operating-points-v2 = <&cpu_opp_table>;
 		};
 	};
 
@@ -239,6 +251,10 @@
 			#address-cells = <1>;
 			#size-cells = <1>;
 
+			cpu_speed_grade: cpu-speed-grade@0 {
+				reg = <0x00 0x02>;
+			};
+
 			ths_calibration: thermal-sensor-calibration@14 {
 				reg = <0x14 0x8>;
 			};
@@ -1188,6 +1204,25 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
+
+		sunxi-info {
+			compatible = "allwinner,sun50i-h6-sys-info";
+			status = "okay";
+		};
+
+		addr_mgt: addr-mgt {
+			compatible = "allwinner,sunxi-addr_mgt";
+			type_addr_wifi = <0x2>;
+			type_addr_bt = <0x2>;
+			type_addr_eth = <0x2>;
+			status = "okay";
+		};
+
+		dump_reg: dump_reg@20000 {
+			compatible = "allwinner,sunxi-dump-reg";
+			reg = <0x0 0x03001000 0x0 0x0f20>;
+			status = "okay";
+		};
 	};
 
 	thermal-zones {
@@ -1259,4 +1294,28 @@
 			};
 		};
 	};
+
+	gpu_opp_table: gpu-opp-table {
+		compatible = "operating-points-v2";
+		opp-125000000 {
+			opp-hz = /bits/ 64 <125000000>;
+			opp-microvolt = <860000>;
+		};
+		opp-250000000 {
+			opp-hz = /bits/ 64 <250000000>;
+			opp-microvolt = <860000>;
+		};
+		opp-432000000 {
+			opp-hz = /bits/ 64 <432000000>;
+			opp-microvolt = <910000>;
+		};
+		opp-600000000 {
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <960000>;
+		};
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			opp-microvolt = <1080000>;
+		};
+	};
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts
--- linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts	2024-04-21 21:35:54.000000000 +0200
+++ linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts	2024-04-26 18:04:45.033647505 +0200
@@ -6,6 +6,7 @@
 /dts-v1/;
 
 #include "sun50i-h616.dtsi"
+#include "sun50i-h616-cpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h618-longan-module-3h.dtsi linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h618-longan-module-3h.dtsi
--- linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h618-longan-module-3h.dtsi	2024-04-21 21:35:54.000000000 +0200
+++ linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h618-longan-module-3h.dtsi	2024-04-26 18:05:23.936980967 +0200
@@ -4,6 +4,7 @@
  */
 
 #include "sun50i-h616.dtsi"
+#include "sun50i-h616-cpu-opp.dtsi"
 
 &mmc2 {
 	pinctrl-names = "default";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h618-transpeed-8k618-t.dts linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h618-transpeed-8k618-t.dts
--- linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h618-transpeed-8k618-t.dts	2024-04-21 21:35:54.000000000 +0200
+++ linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h618-transpeed-8k618-t.dts	2024-04-26 18:04:45.033647505 +0200
@@ -6,6 +6,7 @@
 /dts-v1/;
 
 #include "sun50i-h616.dtsi"
+#include "sun50i-h616-cpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
