diff --speed-large-files --no-dereference --minimal -Naur linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-04-25 18:17:51.216959462 +0200
+++ linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-04-25 18:16:35.660292544 +0200
@@ -863,7 +863,7 @@
 					hysteresis = <0>;
 				};
 				cpu_critical: cpu-trip-2 {
-					temperature = <110000>;
+					temperature = <1100000>;
 					type = "critical";
 					hysteresis = <0>;
 				};
@@ -878,7 +878,7 @@
 
 			trips {
 				gpu_temp_critical: gpu-trip-0 {
-					temperature = <110000>;
+					temperature = <1100000>;
 					type = "critical";
 					hysteresis = <0>;
 				};
