diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc2/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts linux-6.1-rc2/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts
--- linux-6.1-rc2/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2022-10-24 00:27:33.000000000 +0200
+++ linux-6.1-rc2/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2022-10-24 12:42:21.088834720 +0200
@@ -146,6 +146,14 @@
 		regulator-max-microvolt = <5000000>;
 		vin-supply = <&vcc5v0_sys>;
 	};
+
+	ir-receiver {
+		compatible = "gpio-ir-receiver";
+		gpios = <&gpio3 RK_PC5 GPIO_ACTIVE_LOW>;
+		pinctrl-0 = <&ir_int>;
+		linux,rc-map-name = "rc-beelink-gs1";
+		status = "okay";
+	};
 };
 
 &combphy1 {
@@ -290,7 +298,7 @@
 				regulator-name = "vdd_gpu";
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <900000>;
+				regulator-min-microvolt = <825000>;
 				regulator-max-microvolt = <1350000>;
 				regulator-init-microvolt = <900000>;
 				regulator-ramp-delay = <6001>;
@@ -577,6 +585,12 @@
 			rockchip,pins = <0 RK_PC6 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	ir {
+		ir_int: ir-int {
+			rockchip,pins = <3 RK_PC5 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
 };
 
 &pmu_io_domains {
