diff --git a/arch/arm64/boot/dts/broadcom/bcm2712.dtsi b/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
index a41520d79cac1..1faf44c14b49a 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
@@ -702,7 +702,7 @@
 			interrupt-names = "cec-tx", "cec-rx", "cec-low",
 					  "hpd-connected", "hpd-removed";
 			ddc = <&ddc0>;
-			dmas = <&dma32 10>;
+			dmas = <&dma40 10>;
 			dma-names = "audio-rx";
 			clocks = <&firmware_clocks 13>, <&firmware_clocks 14>, <&dvp 0>, <&clk_27MHz>;
 			clock-names = "hdmi", "bvb", "audio", "cec";
@@ -736,7 +736,7 @@
 				     <14>, <15>;
 			interrupt-names = "cec-tx", "cec-rx", "cec-low",
 					  "hpd-connected", "hpd-removed";
-			dmas = <&dma32 17>;
+			dmas = <&dma40 17>;
 			dma-names = "audio-rx";
 			clocks = <&firmware_clocks 13>, <&firmware_clocks 14>, <&dvp 1>, <&clk_27MHz>;
 			clock-names = "hdmi", "bvb", "audio", "cec";
From 09d77a1e239a065e9adb38077ca18e3de5879f75 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Tue, 27 Feb 2024 12:42:05 +0000
Subject: [PATCH] arm64: dt: disable 2712 DMA lite channels

---
 arch/arm64/boot/dts/broadcom/bcm2712.dtsi | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/broadcom/bcm2712.dtsi b/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
index 1faf44c14b49a..ce1f4664cbb68 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
@@ -845,7 +845,7 @@
 		};
 
 		dma32: dma@10000 {
-			compatible = "brcm,bcm2712-dma";
+			compatible = "brcm,bcm2712-dma-lite";
 			reg = <0x10 0x00010000 0 0x600>;
 			interrupts = <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 81 IRQ_TYPE_LEVEL_HIGH>,
@@ -861,6 +861,7 @@
 					  "dma5";
 			#dma-cells = <1>;
 			brcm,dma-channel-mask = <0x0035>;
+			status = "disabled";
 		};
 
 		dma40: dma@10600 {
