diff --speed-large-files --no-dereference --minimal -Naur linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
--- linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-05-18 19:09:04.836206416 +0200
+++ linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-05-18 19:05:57.959539137 +0200
@@ -17,8 +17,8 @@
 	compatible = "xunlong,orangepi-5-plus", "rockchip,rk3588";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
@@ -80,8 +80,8 @@
 		led {
 			color = <LED_COLOR_ID_BLUE>;
 			function = LED_FUNCTION_INDICATOR;
-			function-enumerator = <1>;
 			gpios = <&gpio3 RK_PA6 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "mmc0";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
--- linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2024-05-18 19:09:04.729539749 +0200
+++ linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2024-05-18 19:04:52.096205584 +0200
@@ -12,8 +12,8 @@
 	compatible = "radxa,rock-5b", "rockchip,rk3588";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio;
 	};
 
@@ -44,10 +44,10 @@
 		pinctrl-0 = <&led_rgb_b>;
 
 		led_rgb_b {
-			function = LED_FUNCTION_STATUS;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			color = <LED_COLOR_ID_BLUE>;
 			gpios = <&gpio0 RK_PB7 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "mmc0";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts
--- linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2024-05-18 19:09:04.836206416 +0200
+++ linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2024-05-18 19:06:50.149539310 +0200
@@ -15,8 +15,9 @@
 	compatible = "xunlong,orangepi-5", "rockchip,rk3588s";
 
 	aliases {
-		ethernet0 = &gmac1;
 		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
+		ethernet0 = &gmac1;
 	};
 
 	chosen {
@@ -43,9 +44,10 @@
 		pinctrl-0 = <&leds_gpio>;
 
 		led-1 {
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			gpios = <&gpio1 RK_PA2 GPIO_ACTIVE_HIGH>;
 			label = "status_led";
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "mmc0";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
--- linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2024-05-18 19:09:04.729539749 +0200
+++ linux-6.9.1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2024-05-18 19:07:20.612872728 +0200
@@ -13,8 +13,8 @@
 
 	aliases {
 		ethernet0 = &gmac1;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	analog-sound {
@@ -42,9 +42,9 @@
 
 		io-led {
 			color = <LED_COLOR_ID_BLUE>;
-			function = LED_FUNCTION_STATUS;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "mmc0";
 		};
 	};
 
