From: =?utf-8?b?IkjDpXZhcmQgRi4gQWFzZW4i?= <havard.f.aasen@pfft.no>
Date: Fri, 29 Jul 2022 22:33:41 +0200
Subject: include ctime header

Bug-Debian: https://bugs.debian.org/1016273
Forwarded: no
---
 src/Toolbar.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Toolbar.cc b/src/Toolbar.cc
index 1e87b99..08b1a98 100644
--- a/src/Toolbar.cc
+++ b/src/Toolbar.cc
@@ -38,6 +38,7 @@
 #include <X11/Xutil.h>
 #include <sys/time.h>
 #include <assert.h>
+#include <ctime>
 
 
 long nextTimeout(int resolution)
