diff -Naur linux-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts linux-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
--- linux-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2021-03-31 22:03:42.449999726 +0200
+++ linux-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2021-03-31 22:04:49.239999732 +0200
@@ -298,10 +298,6 @@
 	vcc-pm-supply = <&reg_aldo1>;
 };
 
-&rtc {
-	clocks = <&ext_osc32k>;
-};
-
 &spdif {
 	status = "okay";
 };
