diff -Naur linux-5.11.8-old/include/media/hevc-ctrls.h linux-5.11.8-new/include/media/hevc-ctrls.h
--- linux-5.11.8-old/include/media/hevc-ctrls.h	2021-03-23 21:20:28.189997785 +0100
+++ linux-5.11.8-new/include/media/hevc-ctrls.h	2021-03-23 19:38:43.756664494 +0100
@@ -77,8 +77,12 @@
 	__u8	num_short_term_ref_pic_sets;
 	__u8	num_long_term_ref_pics_sps;
 	__u8	chroma_format_idc;
+	/* fields for rkvdec */
+	__u8	video_parameter_set_id;
+	__u8	seq_parameter_set_id;
+	__u8	num_slices;
 
-	__u8	padding;
+	__u8	padding[6];
 
 	__u64	flags;
 };
@@ -117,8 +121,13 @@
 	__s8	pps_beta_offset_div2;
 	__s8	pps_tc_offset_div2;
 	__u8	log2_parallel_merge_level_minus2;
+	/* fields for rkvdec */
+	__u8	pic_parameter_set_id;
+	__u8	num_ref_idx_l0_default_active_minus1;
+	__u8	num_ref_idx_l1_default_active_minus1;
+
+	__u8	padding;
 
-	__u8	padding[4];
 	__u64	flags;
 };
 
@@ -205,7 +214,11 @@
 	__u8	num_rps_poc_st_curr_after;
 	__u8	num_rps_poc_lt_curr;
 
-	__u8	padding;
+	/* fields for rkvdec */
+	__u16	short_term_ref_pic_set_size;
+	__u16	long_term_ref_pic_set_size;
+
+	__u8	padding[6];
 
 	__u32	entry_point_offset_minus1[256];
 
