

diff --speed-large-files --no-dereference --minimal -Naur NVIDIA-Linux-x86_64-340.108/kernel/nv.c NVIDIA-Linux-x86_64-340.108/kernel/nv.c
--- NVIDIA-Linux-x86_64-340.108/kernel/nv.c	2022-05-12 13:50:23.933212014 +0200
+++ NVIDIA-Linux-x86_64-340.108/kernel/nv.c	2022-05-12 13:44:11.269880750 +0200
@@ -31,7 +31,7 @@
 
 #if defined(NV_VMWARE) || (NV_BUILD_MODULE_INSTANCES != 0)
 #if defined(MODULE_LICENSE)
-MODULE_LICENSE("NVIDIA");
+MODULE_LICENSE("Dual BSD/GPL");
 #endif
 #if defined(MODULE_INFO)
 MODULE_INFO(supported, "external");
diff --speed-large-files --no-dereference --minimal -Naur NVIDIA-Linux-x86_64-340.108/kernel/nv-frontend.c NVIDIA-Linux-x86_64-340.108/kernel/nv-frontend.c
--- NVIDIA-Linux-x86_64-340.108/kernel/nv-frontend.c	2019-12-11 23:04:24.000000000 +0100
+++ NVIDIA-Linux-x86_64-340.108/kernel/nv-frontend.c	2022-05-12 13:47:29.663212989 +0200
@@ -15,7 +15,7 @@
 #include "nv-frontend.h"
 
 #if defined(MODULE_LICENSE)
-MODULE_LICENSE("NVIDIA");
+MODULE_LICENSE("Dual BSD/GPL");
 #endif
 #if defined(MODULE_INFO)
 MODULE_INFO(supported, "external");
