diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythui/mythuibuttonlist.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythui/mythuibuttonlist.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythui/mythuibuttonlist.cpp	2014-05-05 11:51:26.601181096 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythui/mythuibuttonlist.cpp	2014-05-05 11:51:39.794437797 +0200
@@ -2449,6 +2449,12 @@
         {
             ShowSearchDialog();
         }
+        else if (action == "0" || action == "1" || action == "2" || action == "3" ||
+                 action == "4" || action == "5" || action == "6" || action == "7" ||
+                 action == "8" || action == "9")
+        {
+            SetItemCurrent((int) ((GetCount() * (action.toInt() / 10.0))));
+        }
         else
             handled = false;
     }
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/videodlg.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/videodlg.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/videodlg.cpp	2014-05-05 11:51:26.907845981 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/videodlg.cpp	2014-05-05 11:51:39.931103670 +0200
@@ -2013,8 +2013,12 @@
  */
 bool VideoDialog::keyPressEvent(QKeyEvent *levent)
 {
-    if (GetFocusWidget()->keyPressEvent(levent))
+    if (levent->key() != Qt::Key_1 && levent->key() != Qt::Key_2 &&
+        levent->key() != Qt::Key_3 && levent->key() != Qt::Key_4)
+    {
+        if (GetFocusWidget()->keyPressEvent(levent))
         return true;
+    }
 
     bool handled = false;
     QStringList actions;
