diff -Naur mythtv-master-20140506-gd0e25ac-old/mythtv/bindings/python/MythTV/utility/dt.py mythtv-master-20140506-gd0e25ac-new/mythtv/bindings/python/MythTV/utility/dt.py
--- mythtv-master-20140506-gd0e25ac-old/mythtv/bindings/python/MythTV/utility/dt.py	2014-05-06 14:55:47.784996603 +0200
+++ mythtv-master-20140506-gd0e25ac-new/mythtv/bindings/python/MythTV/utility/dt.py	2014-05-06 14:56:01.394947945 +0200
@@ -15,7 +15,6 @@
 import re
 import time
 import singleton
-time.tzset()
 
 class basetzinfo( _pytzinfo ):
     """
