diff -Naur SDL-1.2.15-old/configure.in SDL-1.2.15-new/configure.in
--- SDL-1.2.15-old/configure.in	2012-01-18 22:30:05.000000000 -0800
+++ SDL-1.2.15-new/configure.in	2012-01-24 12:08:23.000000000 -0800
@@ -3,6 +3,7 @@
 AC_CONFIG_HEADER(include/SDL_config.h)
 AC_GNU_SOURCE
 AC_CONFIG_AUX_DIRS($srcdir/build-scripts)
+AC_CONFIG_MACRO_DIR([acinclude])
 
 dnl Set various version strings - taken gratefully from the GTk sources
 #
