diff -Naur SDL-1.2.15-old/configure.in SDL-1.2.15-new/configure.in
--- SDL-1.2.15-old/configure.in	2012-01-18 22:30:05.000000000 -0800
+++ SDL-1.2.15-new/configure.in	2015-07-22 15:50:24.972000154 -0700
@@ -1127,6 +1127,17 @@
             if test x$definitely_enable_video_x11_xrandr = xyes; then
                 AC_DEFINE(SDL_VIDEO_DRIVER_X11_XRANDR)
             fi
+            AC_MSG_CHECKING(for const parameter to _XData32)
+            have_const_param_xdata32=no
+            AC_TRY_COMPILE([
+              #include <X11/Xlibint.h>
+              extern int _XData32(Display *dpy,register _Xconst long *data,unsigned len);
+            ],[
+            ],[
+            have_const_param_xdata32=yes
+            AC_DEFINE(SDL_VIDEO_DRIVER_X11_CONST_PARAM_XDATA32)
+            ])
+            AC_MSG_RESULT($have_const_param_xdata32)
         fi
     fi
 }
diff -Naur SDL-1.2.15-old/include/SDL_config.h.in SDL-1.2.15-new/include/SDL_config.h.in
--- SDL-1.2.15-old/include/SDL_config.h.in	2012-01-18 22:30:05.000000000 -0800
+++ SDL-1.2.15-new/include/SDL_config.h.in	2015-07-22 15:50:24.972000154 -0700
@@ -282,6 +282,7 @@
 #undef SDL_VIDEO_DRIVER_WINDIB
 #undef SDL_VIDEO_DRIVER_WSCONS
 #undef SDL_VIDEO_DRIVER_X11
+#undef SDL_VIDEO_DRIVER_X11_CONST_PARAM_XDATA32
 #undef SDL_VIDEO_DRIVER_X11_DGAMOUSE
 #undef SDL_VIDEO_DRIVER_X11_DYNAMIC
 #undef SDL_VIDEO_DRIVER_X11_DYNAMIC_XEXT
diff -Naur SDL-1.2.15-old/src/video/x11/SDL_x11sym.h SDL-1.2.15-new/src/video/x11/SDL_x11sym.h
--- SDL-1.2.15-old/src/video/x11/SDL_x11sym.h	2012-01-18 22:30:06.000000000 -0800
+++ SDL-1.2.15-new/src/video/x11/SDL_x11sym.h	2015-07-22 15:50:24.972000154 -0700
@@ -165,7 +165,11 @@
  */
 #ifdef LONG64
 SDL_X11_MODULE(IO_32BIT)
+#if SDL_VIDEO_DRIVER_X11_CONST_PARAM_XDATA32
+SDL_X11_SYM(int,_XData32,(Display *dpy,register _Xconst long *data,unsigned len),(dpy,data,len),return)
+#else
 SDL_X11_SYM(int,_XData32,(Display *dpy,register long *data,unsigned len),(dpy,data,len),return)
+#endif
 SDL_X11_SYM(void,_XRead32,(Display *dpy,register long *data,long len),(dpy,data,len),)
 #endif
 
