diff --speed-large-files --no-dereference --minimal -Naur mkinitcpio-v34/Makefile mkinitcpio-v34/Makefile
--- mkinitcpio-v34/Makefile	2022-12-07 21:58:22.000000000 +0100
+++ mkinitcpio-v34/Makefile	2023-01-01 22:01:37.166648699 +0100
@@ -35,11 +35,11 @@
 install: all
 	install -dm755 $(addprefix $(DESTDIR),$(DIRS))
 
-	sed -e 's|\(^_f_config\)=.*|\1=/etc/mkinitcpio.conf|' \
-	    -e 's|\(^_f_functions\)=.*|\1=/usr/lib/initcpio/functions|' \
-	    -e 's|\(^_d_hooks\)=.*|\1=/etc/initcpio/hooks:/usr/lib/initcpio/hooks|' \
-	    -e 's|\(^_d_install\)=.*|\1=/etc/initcpio/install:/usr/lib/initcpio/install|' \
-	    -e 's|\(^_d_presets\)=.*|\1=/etc/mkinitcpio.d|' \
+	sed -e 's|\(^_f_config\)=.*|\1=$(DESTDIR)/etc/mkinitcpio.conf|' \
+	    -e 's|\(^_f_functions\)=.*|\1=$(DESTDIR)/usr/lib/initcpio/functions|' \
+	    -e 's|\(^_d_hooks\)=.*|\1=$(DESTDIR)/etc/initcpio/hooks:/usr/lib/initcpio/hooks|' \
+	    -e 's|\(^_d_install\)=.*|\1=$(DESTDIR)/etc/initcpio/install:$(DESTDIR)/usr/lib/initcpio/install|' \
+	    -e 's|\(^_d_presets\)=.*|\1=$(DESTDIR)/etc/mkinitcpio.d|' \
 	    -e 's|%VERSION%|$(VERSION)|g' \
 	    < mkinitcpio > $(DESTDIR)/usr/bin/mkinitcpio
 
