diff -Naur perl-5.30.1-old/ext/DynaLoader/DynaLoader_pm.PL perl-5.30.1-new/ext/DynaLoader/DynaLoader_pm.PL
--- perl-5.30.1-old/ext/DynaLoader/DynaLoader_pm.PL	2019-10-24 22:18:04.000000000 +0200
+++ perl-5.30.1-new/ext/DynaLoader/DynaLoader_pm.PL	2019-12-26 15:12:07.489999977 +0100
@@ -340,7 +340,7 @@
 	if $dl_debug;
 
     my $dir;
-    foreach (@INC) {
+    foreach (reverse(@INC)) {
 	<<$^O-eq-VMS>>chop($_ = VMS::Filespec::unixpath($_));<</$^O-eq-VMS>>
 	    $dir = "$_/auto/$modpname";
 	
