diff -Naur perl-5.22.1-old/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm perl-5.22.1-new/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm
--- perl-5.22.1-old/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm	2015-10-17 14:38:37.000000000 +0200
+++ perl-5.22.1-new/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm	2016-03-06 22:15:57.000000000 +0100
@@ -379,7 +379,7 @@
 sub perl_inc {
   my $self = shift;
 
-  $self->perl_src() || File::Spec->catdir($self->{config}{archlibexp},"CORE");
+  $self->perl_src() || File::Spec->catdir($ENV{DESTDIR},$self->{config}{archlibexp},"CORE");
 }
 
 sub DESTROY {
