diff -Naur perl-5.12.3-old/Configure perl-5.12.3-new/Configure
--- perl-5.12.3-old/Configure	2011-01-09 12:20:52.000000000 -0800
+++ perl-5.12.3-new/Configure	2011-01-26 17:55:21.000000000 -0800
@@ -1924,7 +1924,7 @@
 	*) exec 1>&4;;
 	esac
 	case "$config_sh" in
-	'') config_sh='config.sh';;
+	'') config_sh='./config.sh';;
 	esac
 	echo " "
 	echo "Fetching answers from $config_sh..."
@@ -1932,7 +1932,7 @@
 	. $config_sh
 	test "$override" && . ./optdef.sh
 	echo " "
-	. UU/extract
+	. ./UU/extract
 	rm -rf UU
 	echo "Extraction done."
 	exit 0
@@ -3110,7 +3110,7 @@
 if test -f config.sh; then
 	echo " "
 	rp="I see a config.sh file.  Shall I use it to set the defaults?"
-	. UU/myread
+	. ./UU/myread
 	case "$ans" in
 	n*|N*) echo "OK, I'll ignore it."
 		mv config.sh config.sh.old
@@ -3499,11 +3499,11 @@
 EOM
 
 	rp="Which of these apply, if any?"
-	. UU/myread
+	. ./UU/myread
 	tans=$ans
 	for file in $tans; do
 		if $test X$file = XPolicy -a -f Policy.sh; then
-			. Policy.sh
+			. ./Policy.sh
 			$cat Policy.sh >> UU/config.sh
 		elif $test -f $src/hints/$file.sh; then
 			. $src/hints/$file.sh
@@ -3515,7 +3515,7 @@
 			echo "$file.sh does not exist"
 			dflt=$file
 			rp="hint to use instead?"
-			. UU/myread
+			. ./UU/myread
 			for file in $ans; do
 				if $test -f "$src/hints/$file.sh"; then
 					. $src/hints/$file.sh
@@ -3751,7 +3751,7 @@
 	case "$tilde" in
 	true) rp="$rp (~name ok)";;
 	esac
-	. UU/myread
+	. ./UU/myread
 	if test -f UU/getfile.ok && \
 		$contains "^$ans\$" UU/getfile.ok >/dev/null 2>&1
 	then
@@ -3885,7 +3885,7 @@
 					dflt=n
 				fi
 				rp="$what $value doesn't exist.  Use that name anyway?"
-				. UU/myread
+				. ./UU/myread
 				dflt=''
 				case "$ans" in
 				y*) type='';;
@@ -22223,7 +22223,7 @@
 	echo " "
 	dflt=y
 	rp='I see a config.over file.  Do you wish to load it?'
-	. UU/myread
+	. ./UU/myread
 	case "$ans" in
 	n*) echo "OK, I'll ignore it.";;
 	*)	. ./config.over
@@ -23388,7 +23388,7 @@
 
 EOM
 	rp="Press return or use a shell escape to edit config.sh:"
-	. UU/myread
+	. ./UU/myread
 	nostick=''
 	case "$ans" in
 	'') ;;
@@ -23422,7 +23422,7 @@
 		;;
 	esac
 	rp="Run $make depend now?"
-	. UU/myread
+	. ./UU/myread
 	case "$ans" in
 	y*)
 		$make depend && echo "Now you must run '$make'."
