diff -Naur perl-5.12.3-old/dist/XSLoader/XSLoader_pm.PL perl-5.12.3-new/dist/XSLoader/XSLoader_pm.PL
--- perl-5.12.3-old/dist/XSLoader/XSLoader_pm.PL	2011-01-26 17:58:28.000000000 -0800
+++ perl-5.12.3-new/dist/XSLoader/XSLoader_pm.PL	2011-01-26 17:59:22.000000000 -0800
@@ -47,7 +47,7 @@
     my $boots = "$module\::bootstrap";
     goto &$boots if defined &$boots;
 
-    goto retry unless $module and defined &dl_load_file;
+    goto retry;
 
     my @modparts = split(/::/,$module);
     my $modfname = $modparts[-1];
