diff -Naur perl-5.12.3-old/Configure perl-5.12.3-new/Configure
--- perl-5.12.3-old/Configure	2011-01-26 17:55:57.000000000 -0800
+++ perl-5.12.3-new/Configure	2011-01-26 17:56:50.000000000 -0800
@@ -108,25 +108,25 @@
 fi
 
 : Proper PATH setting
-paths='/bin /usr/bin /usr/local/bin /usr/ucb /usr/local /usr/lbin'
-paths="$paths /opt/bin /opt/local/bin /opt/local /opt/lbin"
-paths="$paths /usr/5bin /etc /usr/gnu/bin /usr/new /usr/new/bin /usr/nbin"
-paths="$paths /opt/gnu/bin /opt/new /opt/new/bin /opt/nbin"
-paths="$paths /sys5.3/bin /sys5.3/usr/bin /bsd4.3/bin /bsd4.3/usr/ucb"
-paths="$paths /bsd4.3/usr/bin /usr/bsd /bsd43/bin /opt/ansic/bin /usr/ccs/bin"
-paths="$paths /etc /usr/lib /usr/ucblib /lib /usr/ccs/lib"
-paths="$paths /sbin /usr/sbin /usr/libexec"
-paths="$paths /system/gnu_library/bin"
+#paths='/bin /usr/bin /usr/local/bin /usr/ucb /usr/local /usr/lbin'
+#paths="$paths /opt/bin /opt/local/bin /opt/local /opt/lbin"
+#paths="$paths /usr/5bin /etc /usr/gnu/bin /usr/new /usr/new/bin /usr/nbin"
+#paths="$paths /opt/gnu/bin /opt/new /opt/new/bin /opt/nbin"
+#paths="$paths /sys5.3/bin /sys5.3/usr/bin /bsd4.3/bin /bsd4.3/usr/ucb"
+#paths="$paths /bsd4.3/usr/bin /usr/bsd /bsd43/bin /opt/ansic/bin /usr/ccs/bin"
+#paths="$paths /etc /usr/lib /usr/ucblib /lib /usr/ccs/lib"
+#paths="$paths /sbin /usr/sbin /usr/libexec"
+#paths="$paths /system/gnu_library/bin"
+
+#for p in $paths
+#do
+#	case "$p_$PATH$p_" in
+#	*$p_$p$p_*) ;;
+#	*) test -d $p && PATH=$PATH$p_$p ;;
+#	esac
+#done
 
-for p in $paths
-do
-	case "$p_$PATH$p_" in
-	*$p_$p$p_*) ;;
-	*) test -d $p && PATH=$PATH$p_$p ;;
-	esac
-done
-
-PATH=.$p_$PATH
+#PATH=.$p_$PATH
 export PATH
 
 : shall we be using ksh?
