diff -Naur perl-5.30.1-old/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Any.pm perl-5.30.1-new/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Any.pm
--- perl-5.30.1-old/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Any.pm	2019-10-24 23:27:54.000000000 +0200
+++ perl-5.30.1-new/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Any.pm	2019-12-28 14:48:31.246666641 +0100
@@ -3043,7 +3043,7 @@
 sub _perl_header_files {
     my $self = shift;
 
-    my $header_dir = $self->{PERL_SRC} || $ENV{PERL_SRC} || $self->catdir($Config{archlibexp}, 'CORE');
+    my $header_dir = $self->{PERL_SRC} || $ENV{PERL_SRC} || $self->catdir($ENV{DESTDIR},$Config{archlibexp}, 'CORE');
     opendir my $dh, $header_dir
         or die "Failed to opendir '$header_dir' to find header files: $!";
 
diff -Naur perl-5.30.1-old/ext/Errno/Errno_pm.PL perl-5.30.1-new/ext/Errno/Errno_pm.PL
--- perl-5.30.1-old/ext/Errno/Errno_pm.PL	2019-10-24 22:18:04.000000000 +0200
+++ perl-5.30.1-new/ext/Errno/Errno_pm.PL	2019-12-28 14:43:33.109999997 +0100
@@ -300,9 +300,6 @@
     if(!$ENV{'PERL_BUILD_EXPAND_CONFIG_VARS'}) {
         print <<"CONFIG_CHECK_END";
 use Config;
-"\$Config{'archname'}-\$Config{'osvers'}" eq
-"$archname-$Config{'osvers'}" or
-	die "Errno architecture ($archname-$Config{'osvers'}) does not match executable architecture (\$Config{'archname'}-\$Config{'osvers'})";
 
 CONFIG_CHECK_END
 }
