diff -Naur perl-5.12.3-old/ext/DynaLoader/DynaLoader_pm.PL perl-5.12.3-new/ext/DynaLoader/DynaLoader_pm.PL
--- perl-5.12.3-old/ext/DynaLoader/DynaLoader_pm.PL	2011-01-26 17:59:16.000000000 -0800
+++ perl-5.12.3-new/ext/DynaLoader/DynaLoader_pm.PL	2011-01-26 17:59:46.000000000 -0800
@@ -319,7 +319,7 @@
 		       <<|$^O-eq-MacOS>>"(auto/$modpname/$modfname.$dl_dlext)\n"<</$^O-eq-MacOS>>
 	if $dl_debug;
 
-    foreach (@INC) {
+    foreach (reverse(@INC)) {
 	<<$^O-eq-VMS>>chop($_ = VMS::Filespec::unixpath($_));<</$^O-eq-VMS>>
 	<<$^O-eq-MacOS>>
 	    my $path = $_;
