diff -Naur perl-5.30.1-old/dist/XSLoader/XSLoader_pm.PL perl-5.30.1-new/dist/XSLoader/XSLoader_pm.PL
--- perl-5.30.1-old/dist/XSLoader/XSLoader_pm.PL	2019-10-24 22:18:03.000000000 +0200
+++ perl-5.30.1-new/dist/XSLoader/XSLoader_pm.PL	2019-12-26 15:06:02.099999984 +0100
@@ -62,7 +62,7 @@
     my $boots = "$module\::bootstrap";
     goto &$boots if defined &$boots;
 
-    goto \&XSLoader::bootstrap_inherit unless $module and defined &dl_load_file;
+    goto \&XSLoader::bootstrap_inherit;
 
     my @modparts = split(/::/,$module);
     my $modfname = $modparts[-1];
