diff -Naur perl-5.14.0-old/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm perl-5.14.0-new/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm
--- perl-5.14.0-old/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm	2011-05-18 17:51:48.000000000 -0700
+++ perl-5.14.0-new/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm	2011-05-18 18:02:24.000000000 -0700
@@ -370,6 +370,7 @@
   my $self = shift;
 
   $self->perl_src() || File::Spec->catdir($self->{config}{archlibexp},"CORE");
+  $self->perl_src() || File::Spec->catdir($ENV{DESTDIR},$self->{config}{archlibexp},"CORE");
 }
 
 sub DESTROY {
