diff -Naur perl-5.12.3-old/cpan/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm perl-5.12.3-new/cpan/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm
--- perl-5.12.3-old/cpan/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm	2011-01-26 17:57:52.000000000 -0800
+++ perl-5.12.3-new/cpan/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm	2011-01-26 17:58:59.000000000 -0800
@@ -295,7 +295,7 @@
 sub perl_inc {
   my $self = shift;
 
-  $self->perl_src() || File::Spec->catdir($self->{config}{archlibexp},"CORE");
+  $self->perl_src() || File::Spec->catdir($ENV{DESTDIR},$self->{config}{archlibexp},"CORE");
 }
 
 sub DESTROY {
