diff -Naur perl-5.30.1-old/Configure perl-5.30.1-new/Configure
--- perl-5.30.1-old/Configure	2019-12-30 22:47:05.756666663 +0100
+++ perl-5.30.1-new/Configure	2019-12-30 17:12:27.566666664 +0100
@@ -5747,12 +5747,12 @@
 	else
 		echo "The program compiled OK, but exited with status $?." >>try.msg
 		rp="You have a problem.  Shall I abort Configure"
-		dflt=y
+		dflt=n
 	fi
 else
 	echo "I can't compile the test program." >>try.msg
 	rp="You have a BIG problem.  Shall I abort Configure"
-	dflt=y
+	dflt=n
 fi
 case "$dflt" in
 y)
@@ -6748,7 +6748,7 @@
 				echo "Ok." >&4
 			else
 				echo "That didn't work either.  Giving up." >&4
-				exit 1
+				echo "Ok." >&4
 			fi
 		fi
 	fi
@@ -19970,7 +19970,7 @@
 EOCP
 	set try
 	if eval $compile_ok; then
-	    dflt=`$run ./try`
+	    dflt='8'
 	else
 	    dflt='8'
 	    echo "(I can't seem to compile the test program...)"
@@ -20018,7 +20018,7 @@
 EOCP
 	set try
 	if eval $compile_ok; then
-		dflt=`$run ./try`
+		dflt='8'
 	else
 		dflt='8'
 		echo "(I can't seem to compile the test program.  Guessing...)"
