diff -Naur perl-5.12.3-old/ext/Errno/Errno_pm.PL perl-5.12.3-new/ext/Errno/Errno_pm.PL
--- perl-5.12.3-old/ext/Errno/Errno_pm.PL	2019-12-23 09:46:03.123843579 +0100
+++ perl-5.12.3-new/ext/Errno/Errno_pm.PL	2019-12-23 12:08:51.179543781 +0100
@@ -151,8 +151,7 @@
 	     ) {
 	# Some Linuxes have weird errno.hs which generate
 	# no #file or #line directives
-	my $linux_errno_h = -e '/usr/include/errno.h' ?
-	    '/usr/include/errno.h' : '/usr/local/include/errno.h';
+	my $linux_errno_h = $Config{usrinc}.'/errno.h';
 	$file{$linux_errno_h} = 1;
     } elsif ($^O eq 'MacOS') {
 	# note that we are only getting the GUSI errno's here ...
