diff -Naur gdbm-1.18.1-old/src/parseopt.c gdbm-1.18.1-new/src/parseopt.c
--- gdbm-1.18.1-old/src/parseopt.c	2018-05-30 11:39:15.000000000 +0200
+++ gdbm-1.18.1-new/src/parseopt.c	2020-04-26 08:44:52.493333335 +0200
@@ -255,8 +255,8 @@
 }
 
 char *parseopt_program_name;
-char *parseopt_program_doc;
-char *parseopt_program_args;
+extern char *parseopt_program_doc;
+extern char *parseopt_program_args;
 const char *program_bug_address = "<" PACKAGE_BUGREPORT ">";
 void (*parseopt_help_hook) (FILE *stream);
 
