diff -Naur Imaging-1.1.7-old/setup.py Imaging-1.1.7-new/setup.py
--- Imaging-1.1.7-old/setup.py	2009-11-15 08:06:10.000000000 -0800
+++ Imaging-1.1.7-new/setup.py	2010-06-26 17:09:58.000000000 -0700
@@ -83,7 +83,7 @@
 # --------------------------------------------------------------------
 
 from distutils import sysconfig
-from distutils.core import Extension, setup
+from setuptools import setup, Extension
 from distutils.command.build_ext import build_ext
 
 try:
