diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtbase/src/corelib/global/qglobal.h qt-everywhere-src-5.15.16/qtbase/src/corelib/global/qglobal.h
--- qt-everywhere-src-5.15.16/qtbase/src/corelib/global/qglobal.h	2024-11-08 14:36:08.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtbase/src/corelib/global/qglobal.h	2025-04-06 16:49:33.763372365 +0200
@@ -1284,12 +1284,6 @@
 
 #define QT_MODULE(x)
 
-#if !defined(QT_BOOTSTRAPPED) && defined(QT_REDUCE_RELOCATIONS) && defined(__ELF__) && \
-    (!defined(__PIC__) || (defined(__PIE__) && defined(Q_CC_GNU) && Q_CC_GNU >= 500))
-#  error "You must build your code with position independent code if Qt was built with -reduce-relocations. "\
-         "Compile your code with -fPIC (and not with -fPIE)."
-#endif
-
 namespace QtPrivate {
 //like std::enable_if
 template <bool B, typename T = void> struct QEnableIf;
