diff --speed-large-files --no-dereference --minimal -Naur linux-5.18-rc1/drivers/staging/media/hantro/hantro_g2_hevc_dec.c linux-5.18-rc1/drivers/staging/media/hantro/hantro_g2_hevc_dec.c
--- linux-5.18-rc1/drivers/staging/media/hantro/hantro_g2_hevc_dec.c	2022-04-03 23:08:21.000000000 +0200
+++ linux-5.18-rc1/drivers/staging/media/hantro/hantro_g2_hevc_dec.c	2022-04-11 09:19:04.800008899 +0200
@@ -420,7 +420,7 @@
 		chroma_addr = luma_addr + cr_offset;
 		mv_addr = luma_addr + mv_offset;
 
-		if (dpb[i].flags & V4L2_HEVC_DPB_ENTRY_LONG_TERM_REFERENCE)
+		if (dpb[i].rps == V4L2_HEVC_DPB_ENTRY_RPS_LT_CURR)
 			dpb_longterm_e |= BIT(V4L2_HEVC_DPB_ENTRIES_NUM_MAX - 1 - i);
 
 		hantro_write_addr(vpu, G2_REF_LUMA_ADDR(i), luma_addr);
diff --speed-large-files --no-dereference --minimal -Naur linux-5.18-rc1/drivers/staging/media/sunxi/cedrus/cedrus_h265.c linux-5.18-rc1/drivers/staging/media/sunxi/cedrus/cedrus_h265.c
--- linux-5.18-rc1/drivers/staging/media/sunxi/cedrus/cedrus_h265.c	2022-04-11 09:20:11.840010034 +0200
+++ linux-5.18-rc1/drivers/staging/media/sunxi/cedrus/cedrus_h265.c	2022-04-11 09:19:01.923342196 +0200
@@ -209,7 +209,7 @@
 		unsigned int index = list[i];
 		u8 value = list[i];
 
-		if (dpb[index].flags & V4L2_HEVC_DPB_ENTRY_LONG_TERM_REFERENCE)
+		if (dpb[index].rps == V4L2_HEVC_DPB_ENTRY_RPS_LT_CURR)
 			value |= VE_DEC_H265_SRAM_REF_PIC_LIST_LT_REF;
 
 		/* Each SRAM word gathers up to 4 references. */
diff --speed-large-files --no-dereference --minimal -Naur linux-5.18-rc1/include/media/hevc-ctrls.h linux-5.18-rc1/include/media/hevc-ctrls.h
--- linux-5.18-rc1/include/media/hevc-ctrls.h	2022-04-11 09:20:10.353343341 +0200
+++ linux-5.18-rc1/include/media/hevc-ctrls.h	2022-04-11 09:19:09.583342317 +0200
@@ -133,13 +133,15 @@
 	__u64	flags;
 };
 
-#define V4L2_HEVC_DPB_ENTRY_LONG_TERM_REFERENCE	0x01
+#define V4L2_HEVC_DPB_ENTRY_RPS_ST_CURR_BEFORE	0x01
+#define V4L2_HEVC_DPB_ENTRY_RPS_ST_CURR_AFTER	0x02
+#define V4L2_HEVC_DPB_ENTRY_RPS_LT_CURR		0x03
 
 #define V4L2_HEVC_DPB_ENTRIES_NUM_MAX		16
 
 struct v4l2_hevc_dpb_entry {
 	__u64	timestamp;
-	__u8	flags;
+	__u8	rps;
 	__u8	field_pic;
 	__u16	pic_order_cnt[2];
 	__u8	padding[2];
