diff -Naur linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi
--- linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi	1970-01-01 01:00:00.000000000 +0100
+++ linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616-cpu-opp.dtsi	2021-06-09 21:16:24.353320148 +0200
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+// Copyright (C) 2021 Piotr Oniszczuk <piotr.oniszczuk@gmail.com>
+/*
+  Opi Zero2:bin0  bin1 Tx6s:bin0  bin1
+ 480000000  820mV  880mV  / 820mV  880mV
+ 600000000  820mV  880mV  / 820mV  880mV
+ 792000000  860mV  940mV  / 860mV  940mV
+ 1008000000 900mV  1020mV / 900mV  1020mV
+ 1200000000 960mV  1100mV / 960mV  1100mV
+ 1296000000 1100mV 1100mV / 1100mV 1100mV
+ 1344000000 1120mV 1120mV / 1120mV 1120mV
+ 1512000000 1120mV 1120mV / 1120mV 1120mV
+*/
+
+/ {
+	cpu_opp_table: cpu-opp-table {
+		compatible = "operating-points-v2";
+		opp-shared;
+
+		opp-480000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <480000000>;
+			opp-microvolt = <820000 820000 1120000>;
+		};
+
+		opp-600000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <820000 820000 1120000>;
+		};
+
+		opp-792000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <792000000>;
+			opp-microvolt = <860000 860000 1120000>;
+		};
+
+		opp-1008000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1008000000>;
+			opp-microvolt = <900000 900000 1120000>;
+		};
+
+		opp-1200000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1200000000>;
+			opp-microvolt = <960000 960000 11200000>;
+		};
+
+		opp-1344000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1344000000>;
+			opp-microvolt = <1120000 1120000 1120000>;
+		};
+
+		opp-1416000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1416000000>;
+			opp-microvolt = <1120000 1120000 1120000>;
+		};
+
+		opp-1512000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1512000000>;
+			opp-microvolt = <1120000 1120000 1120000>;
+		};
+
+		opp-1608000000 {
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-hz = /bits/ 64 <1608000000>;
+			opp-microvolt = <1120000 1120000 1120000>;
+		};
+
+//		opp-1704000000 {
+//			clock-latency-ns = <244144>; /* 8 32k periods */
+//			opp-hz = /bits/ 64 <1704000000>;
+//			opp-microvolt = <1120000 1120000 1120000>;
+//		};
+
+//		opp-1800000000 {
+//			clock-latency-ns = <244144>; /* 8 32k periods */
+//			opp-hz = /bits/ 64 <1800000000>;
+//			opp-microvolt = <1120000 1120000 1120000>;
+//		};
+	};
+};
+
+&cpu0 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu1 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu2 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
+
+&cpu3 {
+	operating-points-v2 = <&cpu_opp_table>;
+};
diff -Naur linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2021-06-12 19:34:09.113317195 +0200
+++ linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2021-06-11 15:56:49.956651729 +0200
@@ -11,6 +11,7 @@
 #include <dt-bindings/reset/sun50i-h616-ccu.h>
 #include <dt-bindings/reset/sun50i-h6-r-ccu.h>
 #include <dt-bindings/reset/sun8i-de2.h>
+#include <dt-bindings/thermal/thermal.h>
 
 / {
 	interrupt-parent = <&gic>;
@@ -27,6 +28,8 @@
 			reg = <0>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			#cooling-cells = <2>;
 		};
 
 		cpu1: cpu@1 {
@@ -35,6 +38,8 @@
 			reg = <1>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			#cooling-cells = <2>;
 		};
 
 		cpu2: cpu@2 {
@@ -43,6 +48,8 @@
 			reg = <2>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			#cooling-cells = <2>;
 		};
 
 		cpu3: cpu@3 {
@@ -51,6 +58,8 @@
 			reg = <3>;
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			#cooling-cells = <2>;
 		};
 	};
 
@@ -216,6 +225,21 @@
 			#reset-cells = <1>;
 		};
 
+		sid: efuse@3006000 {
+			compatible = "allwinner,sun50i-h616-sid";
+			reg = <0x03006000 0x1000>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			cpu_speed_grade: cpu-speed-grade@0 {
+				reg = <0x00 0x02>;
+			};
+
+			ths_calibration: thermal-sensor-calibration@14 {
+				reg = <0x14 0x8>;
+			};
+		};
+
 		watchdog: watchdog@30090a0 {
 			compatible = "allwinner,sun50i-h616-wdt",
 				     "allwinner,sun6i-a31-wdt";
@@ -1019,6 +1043,95 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
+
+		ths: thermal-sensor@5070400 {
+			/* The Thermal Sensor Controller(THS) embeds four thermal sensors, 
+			sensor0 is located in GPU
+			sensor1 is located in VE
+			sensor2 is located in CPU
+			sensor3 is located in DDR
+			*/
+			compatible = "allwinner,sun50i-h616-ths";
+			reg = <0x05070400 0x400>;
+			interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&ccu CLK_BUS_THS>;
+			clock-names = "bus";
+			resets = <&ccu RST_BUS_THS>;
+			nvmem-cells = <&ths_calibration>;
+			nvmem-cell-names = "calibration";
+			#thermal-sensor-cells = <1>;
+		};
+	};
+
+	thermal-zones {
+		cpu-thermal {
+			polling-delay-passive = <0>;
+			polling-delay = <0>;
+			thermal-sensors = <&ths 2>;
+
+			trips {
+				cpu_alert: cpu-alert {
+					temperature = <85000>;
+					hysteresis = <2000>;
+					type = "passive";
+				};
+
+				cpu-crit {
+					temperature = <100000>;
+					hysteresis = <0>;
+					type = "critical";
+				};
+			};
+
+			cooling-maps {
+				map0 {
+					trip = <&cpu_alert>;
+					cooling-device = <&cpu0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>,
+							 <&cpu1 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>,
+							 <&cpu2 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>,
+							 <&cpu3 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+		};
+
+		gpu-thermal {
+			polling-delay-passive = <0>;
+			polling-delay = <0>;
+			thermal-sensors = <&ths 0>;
+
+			trips {
+				gpu_alert: gpu-alert {
+					temperature = <85000>;
+					hysteresis = <2000>;
+					type = "passive";
+				};
+
+				gpu-crit {
+					temperature = <100000>;
+					hysteresis = <0>;
+					type = "critical";
+				};
+			};
+
+			cooling-maps {
+				map0 {
+					trip = <&gpu_alert>;
+					cooling-device = <&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+		};
+
+		ve-thermal {
+			polling-delay-passive = <0>;
+			polling-delay = <0>;
+			thermal-sensors = <&ths 1>;
+		};
+
+		ddr-thermal {
+			polling-delay-passive = <0>;
+			polling-delay = <0>;
+			thermal-sensors = <&ths 3>;
+		};
 	};
 
 	gpu_opp_table: gpu-opp-table {
diff -Naur linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts
--- linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts	2021-06-12 19:34:09.193317195 +0200
+++ linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts	2021-06-12 13:51:44.769984055 +0200
@@ -10,6 +10,7 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/leds/common.h>
+#include "sun50i-h616-cpu-opp.dtsi"
 
 / {
 	model = "OrangePi Zero2";
@@ -257,14 +258,14 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
 				regulator-name = "vdd-cpu";
 			};
 
 			reg_dcdcc: dcdcc {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
 				regulator-name = "vdd-gpu-sys";
 			};
 
diff -Naur linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts
--- linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts	2021-06-12 19:34:09.383317195 +0200
+++ linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts	2021-06-09 21:24:47.056653480 +0200
@@ -10,6 +10,7 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/leds/common.h>
+#include "sun50i-h616-cpu-opp.dtsi"
 
 / {
 	model = "Tanix TX6s";
@@ -257,14 +258,14 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
 				regulator-name = "vdd-cpu";
 			};
 
 			reg_dcdcc: dcdcc {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
 				regulator-name = "vdd-gpu-sys";
 			};
 
diff -Naur linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts
--- linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts	2021-06-12 19:34:09.289983862 +0200
+++ linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts	2021-06-07 13:46:41.569989133 +0200
@@ -10,6 +10,7 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/leds/common.h>
+#include "sun50i-h616-cpu-opp.dtsi"
 
 / {
 	model = "X96 Mate";
@@ -257,14 +258,14 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
 				regulator-name = "vdd-cpu";
 			};
 
 			reg_dcdcc: dcdcc {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
 				regulator-name = "vdd-gpu-sys";
 			};
 
diff -Naur linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616-t95.dts linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616-t95.dts
--- linux-5.12.9-old/arch/arm64/boot/dts/allwinner/sun50i-h616-t95.dts	2021-06-12 19:34:09.289983862 +0200
+++ linux-5.12.9.new/arch/arm64/boot/dts/allwinner/sun50i-h616-t95.dts	2021-06-07 13:46:41.569989133 +0200
@@ -10,6 +10,7 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/leds/common.h>
+#include "sun50i-h616-cpu-opp.dtsi"
 
 / {
 	model = "T95";
@@ -257,14 +258,14 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
 				regulator-name = "vdd-cpu";
 			};
 
 			reg_dcdcc: dcdcc {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
 				regulator-name = "vdd-gpu-sys";
 			};
 
