diff --speed-large-files --no-dereference --minimal -Naur linux-5.19-rc5/drivers/staging/media/hantro/hantro_v4l2.c linux-5.19-rc5/drivers/staging/media/hantro/hantro_v4l2.c
--- linux-5.19-rc5/drivers/staging/media/hantro/hantro_v4l2.c	2022-07-09 14:29:06.196615873 +0200
+++ linux-5.19-rc5/drivers/staging/media/hantro/hantro_v4l2.c	2022-07-09 14:27:49.226616330 +0200
@@ -123,14 +123,8 @@
 		return -EINVAL;
 	}
 
-	/* For non-coded formats check if postprocessing scaling is possible */
-	if (fmt->codec_mode == HANTRO_MODE_NONE && hantro_needs_postproc(ctx, fmt)) {
-		return hanto_postproc_enum_framesizes(ctx, fsize);
-	} else if (fsize->index != 0) {
-		vpu_debug(0, "invalid frame size index (expected 0, got %d)\n",
-			  fsize->index);
+	if (fmt->codec_mode == HANTRO_MODE_NONE)
 		return -EINVAL;
-	}
 
 	fsize->type = V4L2_FRMSIZE_TYPE_STEPWISE;
 	fsize->stepwise = fmt->frmsize;
