diff --speed-large-files --no-dereference --minimal -Naur linux-5.19-rc4/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts linux-5.19-rc4/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts
--- linux-5.19-rc4/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts	2022-07-07 14:37:02.323287442 +0200
+++ linux-5.19-rc4/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts	2022-07-07 14:35:05.449954792 +0200
@@ -203,13 +203,15 @@
 	assigned-clock-rates = <0>, <125000000>;
 	clock_in_out = "output";
 	phy-handle = <&rgmii_phy1>;
-	phy-mode = "rgmii-id";
+	phy-mode = "rgmii";
 	pinctrl-names = "default";
 	pinctrl-0 = <&gmac1m1_miim
 		     &gmac1m1_tx_bus2
 		     &gmac1m1_rx_bus2
 		     &gmac1m1_rgmii_clk
 		     &gmac1m1_rgmii_bus>;
+	tx_delay = <0x42>;
+	rx_delay = <0x28>;
 	status = "okay";
 };
 
@@ -219,8 +221,11 @@
 };
 
 &hdmi {
+	assigned-clocks = <&cru CLK_HDMI_CEC>;
+	assigned-clock-rates = <32768>;
 	avdd-0v9-supply = <&vdda0v9_image>;
 	avdd-1v8-supply = <&vcca1v8_image>;
+	pinctrl-0 = <&hdmitx_scl &hdmitx_sda &hdmitxm1_cec>;
 	status = "okay";
 };
 
@@ -236,6 +241,14 @@
 	};
 };
 
+&i2s0_8ch {
+	status = "okay";
+};
+
+&hdmi_sound {
+	status = "okay";
+};
+
 &i2c0 {
 	status = "okay";
 
@@ -477,6 +490,11 @@
 };
 
 &i2s1_8ch {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s1m0_sclktx
+		     &i2s1m0_lrcktx
+		     &i2s1m0_sdi0
+		     &i2s1m0_sdo0>;
 	rockchip,trcm-sync-tx-only;
 	status = "okay";
 };
@@ -682,3 +700,15 @@
 		remote-endpoint = <&hdmi_in_vp0>;
 	};
 };
+
+&vpu {
+	status = "okay";
+};
+
+&vdpu_mmu {
+	status = "okay";
+};
+
+&usb_host0_ehci {
+	status = "okay";
+};
