diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/amlogic/meson-khadas-vim3.dtsi u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/amlogic/meson-khadas-vim3.dtsi
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/amlogic/meson-khadas-vim3.dtsi	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/amlogic/meson-khadas-vim3.dtsi	2026-05-23 19:04:53.625029181 +0200
@@ -12,6 +12,9 @@
 
 / {
 	aliases {
+		mmc0 = &sd_emmc_b;
+		mmc1 = &sd_emmc_c;
+		mmc2 = &sd_emmc_a;
 		serial0 = &uart_AO;
 		ethernet0 = &ethmac;
 		rtc0 = &rtc;
@@ -52,8 +55,9 @@
 
 		led-red {
 			color = <LED_COLOR_ID_RED>;
-			function = LED_FUNCTION_STATUS;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			gpios = <&gpio_expander 5 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "disk-activity";
 		};
 	};
 
@@ -93,9 +97,10 @@
 		regulator-name = "VCC_5V";
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
+		regulator-always-on;
 		vin-supply = <&dc_in>;
 
-		gpio = <&gpio GPIOH_8 GPIO_OPEN_DRAIN>;
+		//gpio = <&gpio GPIOH_8 GPIO_OPEN_DRAIN>;
 		enable-active-high;
 	};
 
@@ -189,7 +194,6 @@
 		assigned-clocks = <&clkc CLKID_MPLL2>,
 				  <&clkc CLKID_MPLL0>,
 				  <&clkc CLKID_MPLL1>;
-		assigned-clock-parents = <0>, <0>, <0>;
 		assigned-clock-rates = <294912000>,
 				       <270950400>,
 				       <393216000>;
