diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.07/arch/arm/include/asm/arch-sunxi/clock_sun50i_h6.h u-boot-2026.07/arch/arm/include/asm/arch-sunxi/clock_sun50i_h6.h
--- u-boot-2026.07/arch/arm/include/asm/arch-sunxi/clock_sun50i_h6.h	2026-07-07 01:50:43.000000000 +0200
+++ u-boot-2026.07/arch/arm/include/asm/arch-sunxi/clock_sun50i_h6.h	2026-08-22 13:21:55.667914671 +0200
@@ -23,6 +23,7 @@
 #define CCU_H6_APB1_CFG			0x520
 #define CCU_H6_APB2_CFG			0x524
 #define CCU_H6_MBUS_CFG			0x540
+#define CCU_H6_PWM_GATE_RESET		0x7ac
 #define CCU_H6_DRAM_CLK_CFG		0x800
 #define CCU_H6_MBUS_GATE		0x804
 #define CCU_H6_DRAM_GATE_RESET		0x80c
@@ -86,6 +87,10 @@
 #define CCM_CPU_AXI_AXI_MASK		0x3
 #define CCM_CPU_AXI_DEFAULT_FACTORS	0x301
 
+/* pwm_gate_reset bit field */
+#define CCU_PWM_RST			BIT(16)
+#define CCU_PWM_GATING			BIT(0)
+
 #ifdef CONFIG_MACH_SUN50I_H6				/* H6 */
 
 #define CCM_PLL6_DEFAULT		0xa0006300
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.07/arch/arm/include/asm/arch-sunxi/cpu_sun50i_h6.h u-boot-2026.07/arch/arm/include/asm/arch-sunxi/cpu_sun50i_h6.h
--- u-boot-2026.07/arch/arm/include/asm/arch-sunxi/cpu_sun50i_h6.h	2026-07-07 01:50:43.000000000 +0200
+++ u-boot-2026.07/arch/arm/include/asm/arch-sunxi/cpu_sun50i_h6.h	2026-08-22 13:21:55.667968071 +0200
@@ -13,6 +13,7 @@
 #define SUNXI_SIDC_BASE			0x03006000
 #define SUNXI_SID_BASE			0x03006200
 #define SUNXI_TIMER_BASE		0x03009000
+#define SUNXI_PWM_BASE			0x0300A000
 
 #define SUNXI_GIC400_BASE		0x03020000
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.07/arch/arm/include/asm/arch-sunxi/pwm_sun50i_h616.h u-boot-2026.07/arch/arm/include/asm/arch-sunxi/pwm_sun50i_h616.h
--- u-boot-2026.07/arch/arm/include/asm/arch-sunxi/pwm_sun50i_h616.h	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-2026.07/arch/arm/include/asm/arch-sunxi/pwm_sun50i_h616.h	2026-08-22 13:21:55.668009171 +0200
@@ -0,0 +1,85 @@
+/* SPDX-License-Identifier: GPL-2.0+ */
+/*
+ * (C) Copyright 2025 Richard Genoud, Bootlin <richard.genoud@bootlin.com>
+ */
+
+#ifndef _SUNXI_PWM_SUN50I_H616_H
+#define _SUNXI_PWM_
+
+#ifndef __ASSEMBLY__
+#include <linux/bitops.h>
+#endif
+
+/* PWM IRQ Enable Register */
+#define SUNXI_PWM_IER			0x0
+
+/* PWM IRQ Status Register */
+#define SUNXI_PWM_ISR			0x4
+
+/* PWM Capture IRQ Enable Register */
+#define SUNXI_PWM_CIER			0x10
+
+/* PWM Capture IRQ Status Register */
+#define SUNXI_PWM_CISR			0x14
+
+/* PWM01 Clock Configuration Register */
+#define SUNXI_PWM_CCR01R		0x20
+
+/* PWM23 Clock Configuration Register */
+#define SUNXI_PWM_CCR23R		0x24
+
+/* PWM45 Clock Configuration Register */
+#define SUNXI_PWM_CCR45R		0x28
+
+/* PWM01 Dead Zone Control Register */
+#define SUNXI_PWM_DZCR01R		0x30
+
+/* PWM23 Dead Zone Control Register */
+#define SUNXI_PWM_DZCR23R		0x34
+
+/* PWM45 Dead Zone Control Register */
+#define SUNXI_PWM_DZCR45R		0x38
+
+/* PWM Enable Register */
+#define SUNXI_PWM_ENR			0x40
+
+/* PWM Capture Enable Register */
+#define SUNXI_PWM_CER			0x44
+
+/* PWM Control Register */
+#define SUNXI_PWM_CTRL_REG(x)		(0x60 + (x) * 0x20)
+
+/* PWM Period Register */
+#define SUNXI_PWM_PERIOD_REG(x)		(0x64 + (x) * 0x20)
+
+/* PWM Count Register */
+#define SUNXI_PWM_COUNT_REG(x)		(0x68 + (x) * 0x20)
+
+/* PWM Capture Control Register */
+#define SUNXI_PWM_CCR(x)		(0x6c + (x) * 0x20)
+
+/* PWM Capture Rise Lock Register */
+#define SUNXI_PWM_CRLR(x)		(0x70 + (x) * 0x20)
+
+/* PWM Capture Fall Lock Register */
+#define SUNXI_PWM_CFLR(x)		(0x74 + (x) * 0x20)
+
+
+/* PWM Period bit field */
+#define PWM_ENTIRE_CYCLE(x)		(((x) - 1) << 16)
+#define PWM_ACTIVE_CYCLE(x)		((x) - 1)
+
+/* PWM Clock Configuration bit field */
+#define PMW_CLK_DIV_M(x)		(x)
+#define PWM_CLK_GATING			BIT(4)
+#define PWM23_CLK_SRC_BYPASS_TO_PWM2	BIT(5)
+#define PWM45_CLK_SRC_BYPASS_TO_PWM4	BIT(5)
+#define PWM01_CLK_SRC_BYPASS_TO_PWM1	BIT(6)
+#define PWM23_CLK_SRC_BYPASS_TO_PWM3	BIT(6)
+#define PWM45_CLK_SRC_BYPASS_TO_PWM5	BIT(6)
+#define PWM_CLK_SRC_APB1		BIT(7)
+
+/* PWM Enable bit field */
+#define PWM_ENABLE(x)			BIT(x)
+
+#endif
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.07/arch/arm/mach-sunxi/board.c u-boot-2026.07/arch/arm/mach-sunxi/board.c
--- u-boot-2026.07/arch/arm/mach-sunxi/board.c	2026-07-07 01:50:43.000000000 +0200
+++ u-boot-2026.07/arch/arm/mach-sunxi/board.c	2026-08-22 13:21:55.668072881 +0200
@@ -488,6 +488,12 @@
 	i2c_init_board();
 	i2c_init(CONFIG_SYS_I2C_SPEED, CONFIG_SYS_I2C_SLAVE);
 #endif
+	/*
+	 * pinmux pwm5:
+	 * TODO:
+	 * This should be done in dts
+	 */
+	sunxi_gpio_set_cfgpin(SUNXI_GPA(12), 2);
 	sunxi_board_init();
 }
 #endif /* CONFIG_XPL_BUILD */
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.07/arch/arm/mach-sunxi/clock_sun50i_h6.c u-boot-2026.07/arch/arm/mach-sunxi/clock_sun50i_h6.c
--- u-boot-2026.07/arch/arm/mach-sunxi/clock_sun50i_h6.c	2026-08-22 13:22:49.407875302 +0200
+++ u-boot-2026.07/arch/arm/mach-sunxi/clock_sun50i_h6.c	2026-08-22 13:21:55.668129531 +0200
@@ -5,6 +5,7 @@
 #include <asm/arch/clock.h>
 #include <asm/arch/prcm.h>
 #include <linux/delay.h>
+#include <asm/arch/pwm_sun50i_h616.h>
 
 #ifndef SUNXI_CPU_PLL_CFG_BASE
 #define SUNXI_CPU_PLL_CFG_BASE 0
@@ -15,6 +16,7 @@
 {
 	void *const ccm = (void *)SUNXI_CCM_BASE;
 	void *const prcm = (void *)SUNXI_PRCM_BASE;
+	void *const pwm = (void *)SUNXI_PWM_BASE;
 
 	if (IS_ENABLED(CONFIG_MACH_SUN50I_H616))
 		setbits_le32(prcm + CCU_PRCM_SYS_PWROFF_GATING, 0x10);
@@ -72,6 +74,20 @@
 		writel(MBUS_CLK_SRC_PLL6X2 | MBUS_CLK_M(3),
 		       ccm + CCU_H6_MBUS_CFG);
 	}
+
+	if (IS_ENABLED(CONFIG_MACH_SUN50I_H616)) {
+		/* add for pwm ephy */
+		writel(CCU_PWM_RST | CCU_PWM_GATING,
+		       ccm + CCU_H6_PWM_GATE_RESET);
+
+		writel(PWM_ENTIRE_CYCLE(9) | PWM_ACTIVE_CYCLE(5),
+		       pwm + SUNXI_PWM_PERIOD_REG(5));
+
+		writel(PWM_CLK_GATING | PWM45_CLK_SRC_BYPASS_TO_PWM5,
+		       pwm + SUNXI_PWM_CCR45R);
+
+		writel(PWM_ENABLE(5), pwm + SUNXI_PWM_ENR);
+	}
 }
 
 void clock_init_uart(void)
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.07/drivers/gpio/sunxi_gpio.c u-boot-2026.07/drivers/gpio/sunxi_gpio.c
--- u-boot-2026.07/drivers/gpio/sunxi_gpio.c	2026-07-07 01:50:43.000000000 +0200
+++ u-boot-2026.07/drivers/gpio/sunxi_gpio.c	2026-08-22 13:22:26.191431669 +0200
@@ -295,13 +295,10 @@
 		sunxi_gpio_set_value_bank(plat->regs, offset, value);
 		sunxi_gpio_set_cfgbank(plat->regs, offset, SUNXI_GPIO_OUTPUT);
 	} else if (flags & GPIOD_IS_IN) {
-		u32 pull = 0;
-
 		if (flags & GPIOD_PULL_UP)
-			pull = 1;
+			sunxi_gpio_set_pull_bank(plat->regs, offset, 1);
 		else if (flags & GPIOD_PULL_DOWN)
-			pull = 2;
-		sunxi_gpio_set_pull_bank(plat->regs, offset, pull);
+			sunxi_gpio_set_pull_bank(plat->regs, offset, 2);
 		sunxi_gpio_set_cfgbank(plat->regs, offset, SUNXI_GPIO_INPUT);
 	}
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.07/drivers/net/phy/phy.c u-boot-2026.07/drivers/net/phy/phy.c
--- u-boot-2026.07/drivers/net/phy/phy.c	2026-07-07 01:50:43.000000000 +0200
+++ u-boot-2026.07/drivers/net/phy/phy.c	2026-08-22 13:22:10.550252046 +0200
@@ -16,6 +16,7 @@
 #include <command.h>
 #include <miiphy.h>
 #include <phy.h>
+#include <asm/io.h>
 #include <errno.h>
 #include <asm-generic/gpio.h>
 #include <dm/device_compat.h>
@@ -385,10 +386,101 @@
 	return 0;
 }
 
+#define AC300_DEV 0x10
+static void ac300_ephy_enable(struct phy_device *phydev)
+{
+	struct mii_dev *bus = phydev->bus;
+	bus->write(bus, AC300_DEV, MDIO_DEVAD_NONE, 0x00, 0x0000);	/* reset ephy */
+	bus->write(bus, AC300_DEV, MDIO_DEVAD_NONE, 0x00, 0x0003);	/* de-reset ephy */
+
+	bus->write(bus, AC300_DEV, MDIO_DEVAD_NONE, 0x00, 0x00b7);	/* open clk gate */
+	bus->write(bus, AC300_DEV, MDIO_DEVAD_NONE, 0x05, 0xa81f);	/* enable io */
+
+	mdelay(10);
+	bus->write(bus, AC300_DEV, MDIO_DEVAD_NONE, 0x06, 0x5811);	/* shutdown ephy */
+	mdelay(10);
+	bus->write(bus, AC300_DEV, MDIO_DEVAD_NONE, 0x06, 0x5810);	/* powerup ephy */
+}
+
+static void disable_intelligent_ieee(struct phy_device *phydev)
+{
+	unsigned int value;
+
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0100);	/* switch to page 1 */
+	value = phy_read(phydev, MDIO_DEVAD_NONE, 0x17);	/* read address 0 0x17 register */
+	value &= ~(1 << 3);					/* reg 0x17 bit 3, set 0 to disable IEEE */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x17, value);
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0000);	/* switch to page 0 */
+}
+
+static void disable_802_3az_ieee(struct phy_device *phydev)
+{
+	unsigned int value;
+
+	phy_write(phydev, MDIO_DEVAD_NONE, 0xd, 0x7);
+	phy_write(phydev, MDIO_DEVAD_NONE, 0xe, 0x3c);
+	phy_write(phydev, MDIO_DEVAD_NONE, 0xd, 0x1 << 14 | 0x7);
+	value = phy_read(phydev, MDIO_DEVAD_NONE, 0xe);
+	value &= ~(0x1 << 1);
+	phy_write(phydev, MDIO_DEVAD_NONE, 0xd, 0x7);
+	phy_write(phydev, MDIO_DEVAD_NONE, 0xe, 0x3c);
+	phy_write(phydev, MDIO_DEVAD_NONE, 0xd, 0x1 << 14 | 0x7);
+	phy_write(phydev, MDIO_DEVAD_NONE, 0xe, value);
+
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0200);	/* switch to page 2 */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x18, 0x0000);
+}
+
+static void ephy_config_default(struct phy_device *phydev)
+{
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0100);	/* Switch to Page 1 */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x12, 0x4824);	/* Disable APS */
+
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0200);	/* Switch to Page 2 */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x18, 0x0000);	/* PHYAFE TRX optimization */
+
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0600);	/* Switch to Page 6 */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x14, 0x708b);	/* PHYAFE TX optimization */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x13, 0xF000);	/* PHYAFE RX optimization */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x15, 0x1530);
+
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0800);	/* Switch to Page 6 */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x18, 0x00bc);	/* PHYAFE TRX optimization */
+}
+
+static void __maybe_unused ephy_config_fixed(struct phy_device *phydev)
+{
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0100);	/* switch to Page 1 */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x12, 0x4824);	/* Disable APS */
+
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0200);	/* switch to Page 2 */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x18, 0x0000);	/* PHYAFE TRX optimization */
+
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0600);	/* switch to Page 6 */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x14, 0x7809);	/* PHYAFE TX optimization */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x13, 0xf000);	/* PHYAFE RX optimization */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x10, 0x5523);
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x15, 0x3533);
+
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0800);	/* switch to Page 8 */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1d, 0x0844);	/* disable auto offset */
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x18, 0x00bc);	/* PHYAFE TRX optimization */
+}
+
+static void __maybe_unused ephy_config_cali(struct phy_device *phydev, u16 ephy_cali)
+{
+	int value;
+	value = phy_read(phydev, MDIO_DEVAD_NONE, 0x06);
+	value &= ~(0x0F << 12);
+	value |= (0x0F & (0x03 + ephy_cali)) << 12;
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x06, value);
+}
+
 int genphy_config(struct phy_device *phydev)
 {
 	int val;
 	u32 features;
+	u16 ephy_cali;
 
 	features = (SUPPORTED_TP | SUPPORTED_MII
 			| SUPPORTED_AUI | SUPPORTED_FIBRE |
@@ -433,6 +525,38 @@
 
 	genphy_config_aneg(phydev);
 
+#define sunxi_ac300_key (1<<8)
+
+	val = readl(0x300622c);
+	ephy_cali = val & 0xffff;
+	if (val & sunxi_ac300_key) {
+		ac300_ephy_enable(phydev);
+		ephy_config_cali(phydev, ephy_cali);
+
+		/*
+		 * BIT9: the flag of calibration value
+		 * 0: Normal
+		 * 1: Low level of calibration value
+		 */
+		if (ephy_cali & 0x200) {
+			printf("ac300:ephy cali efuse read: fixed!\n");
+			ephy_config_fixed(phydev);
+		} else {
+			printf("ac300:ephy cali efuse read: default!\n");
+			ephy_config_default(phydev);
+		}
+	}
+
+	disable_intelligent_ieee(phydev);
+	disable_802_3az_ieee(phydev);
+	phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0000); /* Switch to Page 0 */
+
+	if (val & sunxi_ac300_key) {
+		val = phy_read(phydev, MDIO_DEVAD_NONE, 6);
+		val |= (0x1 << 11);
+		phy_write(phydev, MDIO_DEVAD_NONE, 6, val | (1 << 1));
+	}
+
 	return 0;
 }
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.07/drivers/net/sun8i_emac.c u-boot-2026.07/drivers/net/sun8i_emac.c
--- u-boot-2026.07/drivers/net/sun8i_emac.c	2026-07-07 01:50:43.000000000 +0200
+++ u-boot-2026.07/drivers/net/sun8i_emac.c	2026-08-22 13:22:20.291802908 +0200
@@ -348,6 +348,14 @@
 {
 	struct phy_device *phydev;
 
+	/* Connect AC300 and configure AC300 first */
+	phydev = phy_connect(priv->bus, 0x0, dev, priv->interface);
+	if (!phydev)
+		return -ENODEV;
+
+	phy_config(phydev);
+
+	/* Connect EPHY */
 	phydev = phy_connect(priv->bus, priv->phyaddr, dev, priv->interface);
 	if (!phydev)
 		return -ENODEV;
@@ -709,6 +717,9 @@
 
 	sun8i_mdio_init(dev->name, dev);
 	priv->bus = miiphy_get_dev_by_name(dev->name);
+	if (!priv->bus) {
+		printf("miiphy_get_dev_by_name failed\n");
+	}
 
 	return sun8i_phy_init(priv, dev);
 }
@@ -756,6 +767,29 @@
 	return 0;
 }
 
+static int sun8i_handle_phy_clk(struct udevice *dev, struct emac_eth_dev *priv)
+{
+	struct ofnode_phandle_args phandle;
+	int ret;
+
+	ret = ofnode_parse_phandle_with_args(dev_ofnode(dev), "phy-handle",
+					     NULL, 0, 0, &phandle);
+	if (ret)
+		return ret;
+
+	if (!ofnode_device_is_compatible(phandle.node,
+					 "ethernet-phy-ieee802.3-c22"))
+		return 0;
+
+	// ret = clk_get_by_index_nodev(phandle.node, 0, &priv->ephy_clk);
+	// if (ret && ret != -ENOENT) {
+	// 	dev_err(dev, "failed to get PHY clock\n");
+	// 	return ret;
+	// }
+
+	return 0;
+}
+
 static int sun8i_emac_eth_of_to_plat(struct udevice *dev)
 {
 	struct sun8i_eth_pdata *sun8i_pdata = dev_get_plat(dev);
@@ -833,6 +867,10 @@
 		ret = sun8i_handle_internal_phy(dev, priv);
 		if (ret)
 			return ret;
+	} else {
+		ret = sun8i_handle_phy_clk(dev, priv);
+		if (ret)
+			return ret;
 	}
 
 	priv->interface = pdata->phy_interface;
@@ -900,6 +938,11 @@
 	.support_rmii		= true,
 };
 
+static const struct emac_variant emac_variant_h616_1 = {
+	.syscon_offset		= 0x34,
+	.support_rmii		= true,
+};
+
 static const struct udevice_id sun8i_emac_eth_ids[] = {
 	{ .compatible = "allwinner,sun8i-a83t-emac",
 	  .data = (ulong)&emac_variant_a83t },
@@ -913,6 +956,8 @@
 	  .data = (ulong)&emac_variant_a64 },
 	{ .compatible = "allwinner,sun50i-h6-emac",
 	  .data = (ulong)&emac_variant_h6 },
+	{ .compatible = "allwinner,sun50i-h616-emac1",
+	  .data = (ulong)&emac_variant_h616_1 },
 	{ }
 };
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.07/drivers/pinctrl/sunxi/pinctrl-sunxi.c u-boot-2026.07/drivers/pinctrl/sunxi/pinctrl-sunxi.c
--- u-boot-2026.07/drivers/pinctrl/sunxi/pinctrl-sunxi.c	2026-07-07 01:50:43.000000000 +0200
+++ u-boot-2026.07/drivers/pinctrl/sunxi/pinctrl-sunxi.c	2026-08-22 13:22:04.065012281 +0200
@@ -738,12 +738,14 @@
 
 static const struct sunxi_pinctrl_function sun50i_h616_pinctrl_functions[] = {
 	{ "emac0",	2 },	/* PI0-PI16 */
+	{ "emac1",	2 },	/* PA0-PA9 */
 	{ "gpio_in",	0 },
 	{ "gpio_out",	1 },
 	{ "mmc0",	2 },	/* PF0-PF5 */
 	{ "mmc1",	2 },	/* PG0-PG5 */
 	{ "mmc2",	3 },	/* PC0-PC16 */
 	{ "nand0",	2 },	/* PC0-PC16 */
+	{ "pwm5",	2 },	/* PA12 */
 	{ "spi0",	4 },	/* PC0-PC7, PC15-PC16 */
 #if IS_ENABLED(CONFIG_UART0_PORT_F)
 	{ "uart0",	3 },	/* PF2-PF4 */
