diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.04/configs/orangepi_zero2w_defconfig u-boot-2026.04/configs/orangepi_zero2w_defconfig
--- u-boot-2026.04/configs/orangepi_zero2w_defconfig	2026-04-06 19:59:18.000000000 +0200
+++ u-boot-2026.04/configs/orangepi_zero2w_defconfig	2026-04-26 14:56:23.756465764 +0200
@@ -28,3 +27,20 @@
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_OHCI_HCD=y
 CONFIG_USB_MUSB_GADGET=y
+CONFIG_PHY_REALTEK=y
+CONFIG_SUN8I_EMAC=y
+CONFIG_I2C3_ENABLE=y
+CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_CMD_NET=y
+CONFIG_CMD_DHCP=y
+CONFIG_CMD_WGET=y
+CONFIG_CMD_TFTPBOOT=y
+CONFIG_CMD_PING=y
+CONFIG_PROT_TCP=y
+CONFIG_LIB_RAND=y
+CONFIG_WGET=y
+CONFIG_NET_LWIP=y
+CONFIG_CMD_WGET_LWIP=y
+CONFIG_BOOTCOMMAND="if dhcp ${scriptaddr} /PXEclient/pxelinux.cfg/sun50i-h618-orangepi-zero2w.scr; then source ${scriptaddr}; fi; run distro_bootcmd"
+CONFIG_CMD_MII=y
+CONFIG_CMD_MDIO=y
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.04/dts/upstream/src/arm64/allwinner/sun50i-h618-orangepi-zero2w.dts u-boot-2026.04/dts/upstream/src/arm64/allwinner/sun50i-h618-orangepi-zero2w.dts
--- u-boot-2026.04/dts/upstream/src/arm64/allwinner/sun50i-h618-orangepi-zero2w.dts	2026-04-06 19:59:18.000000000 +0200
+++ u-boot-2026.04/dts/upstream/src/arm64/allwinner/sun50i-h618-orangepi-zero2w.dts	2026-04-26 14:50:51.138244494 +0200
@@ -1,22 +1,23 @@
-// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+// SPDX-License-Identifier: (GPL-2.0+ or MIT)
 /*
- * Copyright (C) 2023 Arm Ltd.
+ * Author: piotr.oniszczuk@gmail.com
  */
 
 /dts-v1/;
 
 #include "sun50i-h616.dtsi"
 #include "sun50i-h616-cpu-opp.dtsi"
-
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/leds/common.h>
 
 / {
-	model = "OrangePi Zero 2W";
-	compatible = "xunlong,orangepi-zero2w", "allwinner,sun50i-h618";
+	model = "OrangePi Zero2 W";
+	compatible = "xunlong,orangepi-zero2w", "allwinner,sun50i-h616";
 
 	aliases {
+		mmc0 = &mmc0;  /* mmc0/mmcblk0 for SD */
+		ethernet0 = &emac1;
 		serial0 = &uart0;
 	};
 
@@ -24,13 +25,14 @@
 		stdout-path = "serial0:115200n8";
 	};
 
-	leds {
+	leds: leds {
 		compatible = "gpio-leds";
 
-		led-0 {
-			function = LED_FUNCTION_STATUS;
-			color = <LED_COLOR_ID_GREEN>;
+		led-green {
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			label = "red_green";
 			gpios = <&pio 2 13 GPIO_ACTIVE_HIGH>; /* PC13 */
+			linux,default-trigger = "disk-activity";
 		};
 	};
 
@@ -43,41 +45,74 @@
 		regulator-always-on;
 	};
 
-	reg_vcc3v3: vcc3v3 {
-		/* SY8089 DC/DC converter */
+	reg_usb1_vbus: usb1-vbus {
 		compatible = "regulator-fixed";
-		regulator-name = "vcc-3v3";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
+		regulator-name = "usb1-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
 		vin-supply = <&reg_vcc5v>;
-		regulator-always-on;
+		enable-active-high;
+		gpio = <&pio 2 16 GPIO_ACTIVE_HIGH>; /* PC16 */
+		status = "okay";
 	};
 };
 
-&codec {
-	allwinner,audio-routing = "Line Out", "LINEOUT";
+&cpu0 {
+	cpu-supply = <&reg_dcdc2>;
 	status = "okay";
 };
 
-&cpu0 {
-	cpu-supply = <&reg_dcdc2>;
+&ehci0 {
+	status = "okay";
 };
 
 &ehci1 {
 	status = "okay";
 };
 
-/* USB 2 & 3 are on the FPC connector (or the exansion board) */
+&ehci2 {
+	status = "okay";
+};
+
+&ehci3 {
+	status = "okay";
+};
 
 &gpu {
 	mali-supply = <&reg_dcdc1>;
 	status = "okay";
 };
 
+&i2c3 {
+	status = "okay";
+};
+
+&emac1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rmii_pins>;
+	phy-mode = "rmii";
+	phy-handle = <&rmii_phy>;
+	phy-supply = <&reg_aldo1>;
+	allwinner,rx-delay-ps = <3100>;
+	allwinner,tx-delay-ps = <700>;
+	status = "okay";
+};
+
+&mdio1 {
+	rmii_phy: ethernet-phy@16 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <16>;
+	};
+};
+
 &mmc0 {
-	cd-gpios = <&pio 5 6 GPIO_ACTIVE_LOW>;	/* PF6 */
+	vmmc-supply = <&reg_dldo1>;
+	broken-cd;
 	bus-width = <4>;
-	vmmc-supply = <&reg_vcc3v3>;
+	status = "okay";
+};
+
+&ohci0 {
 	status = "okay";
 };
 
@@ -85,81 +120,65 @@
 	status = "okay";
 };
 
-&pio {
-	vcc-pc-supply = <&reg_dldo1>;
-	vcc-pf-supply = <&reg_dldo1>;	/* internally via VCC-IO */
-	vcc-pg-supply = <&reg_aldo1>;
-	vcc-ph-supply = <&reg_dldo1>;	/* internally via VCC-IO */
-	vcc-pi-supply = <&reg_dldo1>;
+&ohci2 {
+	status = "okay";
+};
+
+&ohci3 {
+	status = "okay";
 };
 
 &r_i2c {
 	status = "okay";
 
-	axp313: pmic@36 {
+	axp313a: pmic@36 {
 		compatible = "x-powers,axp313a";
 		reg = <0x36>;
-		#interrupt-cells = <1>;
-		interrupt-controller;
-		interrupt-parent = <&pio>;
-		interrupts = <2 9 IRQ_TYPE_LEVEL_LOW>;	/* PC9 */
+		wakeup-source;
 
 		vin1-supply = <&reg_vcc5v>;
 		vin2-supply = <&reg_vcc5v>;
 		vin3-supply = <&reg_vcc5v>;
 
 		regulators {
-			/* Supplies VCC-PLL and DRAM */
-			reg_aldo1: aldo1 {
-				regulator-always-on;
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <1800000>;
-				regulator-name = "vcc1v8";
-			};
-
-			/* Supplies VCC-IO, so needs to be always on. */
-			reg_dldo1: dldo1 {
-				regulator-always-on;
-				regulator-min-microvolt = <3300000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-name = "vcc3v3";
-			};
-
 			reg_dcdc1: dcdc1 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <990000>;
-				regulator-name = "vdd-gpu-sys";
+				regulator-max-microvolt = <1160000>;
+				regulator-name = "vdd-gpu";
 			};
 
 			reg_dcdc2: dcdc2 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1100000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-cpu";
 			};
 
 			reg_dcdc3: dcdc3 {
 				regulator-always-on;
-				regulator-min-microvolt = <1100000>;
-				regulator-max-microvolt = <1100000>;
+				/* vdd-ram increased from 1100mV to 1150mV
+				 * as 1100mV gives me ustable board operation
+				 */
+				regulator-min-microvolt = <1150000>;
+				regulator-max-microvolt = <1150000>;
 				regulator-name = "vdd-dram";
 			};
-		};
-	};
-};
 
-&spi0  {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&spi0_pins>, <&spi0_cs0_pin>;
+			reg_aldo1: aldo1 {
+				regulator-always-on;
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-name = "vcc1v8";
+			};
 
-	flash@0 {
-		#address-cells = <1>;
-		#size-cells = <1>;
-		compatible = "jedec,spi-nor";
-		reg = <0>;
-		spi-max-frequency = <40000000>;
+			reg_dldo1: dldo1 {
+				regulator-always-on;
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-name = "vcc3v3";
+			};
+		};
 	};
 };
 
@@ -170,22 +189,26 @@
 };
 
 &usbotg {
-	/*
-	 * PHY0 pins are connected to a USB-C socket, but a role switch
-	 * is not implemented: both CC pins are pulled to GND.
-	 * The VBUS pins power the device, so a fixed peripheral mode
-	 * is the best choice.
-	 * The board can be powered via GPIOs, in this case port0 *can*
-	 * act as a host (with a cable/adapter ignoring CC), as VBUS is
-	 * then provided by the GPIOs. Any user of this setup would
-	 * need to adjust the DT accordingly: dr_mode set to "host",
-	 * enabling OHCI0 and EHCI0.
-	 */
 	dr_mode = "peripheral";
 	status = "okay";
 };
 
 &usbphy {
-	usb1_vbus-supply = <&reg_vcc5v>;
+	usb1_vbus-supply = <&reg_usb1_vbus>;
+	status = "okay";
+};
+
+&spi0  {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi0_pins>, <&spi0_cs0_pin>;
+
+	flash@0 {
+		status = "okay";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
+	};
 };
