diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.04/configs/transpeed-8k618-t_defconfig u-boot-2026.04/configs/transpeed-8k618-t_defconfig
--- u-boot-2026.04/configs/transpeed-8k618-t_defconfig	2026-04-06 19:59:18.000000000 +0200
+++ u-boot-2026.04/configs/transpeed-8k618-t_defconfig	2026-04-26 21:52:08.240593415 +0200
@@ -25,3 +25,16 @@
 CONFIG_AXP_DCDC3_VOLT=1360
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_OHCI_HCD=y
+CONFIG_PHY_REALTEK=y
+CONFIG_SUN8I_EMAC=y
+CONFIG_I2C3_ENABLE=y
+CONFIG_USB_MUSB_GADGET=y
+CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR=0x40
+CONFIG_MMC_SUNXI_SLOT_EXTRA=2
+CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_CMD_WGET=y
+CONFIG_LIB_RAND=y
+CONFIG_PROT_TCP=y
+CONFIG_NET_LWIP=y
+CONFIG_CMD_WGET_LWIP=y
+CONFIG_BOOTCOMMAND="if dhcp ${scriptaddr} /PXEclient/pxelinux.cfg/sun50i-h618-transpeed-8k618-t.scr; then source ${scriptaddr}; fi; run distro_bootcmd"
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2026.04/dts/upstream/src/arm64/allwinner/sun50i-h618-transpeed-8k618-t.dts u-boot-2026.04/dts/upstream/src/arm64/allwinner/sun50i-h618-transpeed-8k618-t.dts
--- u-boot-2026.04/dts/upstream/src/arm64/allwinner/sun50i-h618-transpeed-8k618-t.dts	2026-04-06 19:59:18.000000000 +0200
+++ u-boot-2026.04/dts/upstream/src/arm64/allwinner/sun50i-h618-transpeed-8k618-t.dts	2026-04-26 21:49:33.410684064 +0200
@@ -13,9 +13,13 @@
 
 / {
 	model = "Transpeed 8K618-T";
-	compatible = "transpeed,8k618-t", "allwinner,sun50i-h618";
+	compatible = "transpeed,8k618-t", "allwinner,sun50i-h616";
 
 	aliases {
+		mmc0 = &mmc0;
+		mmc1 = &mmc1;
+		mmc2 = &mmc2;
+		ethernet0 = &emac1;
 		ethernet1 = &sdio_wifi;
 		serial0 = &uart0;
 	};
@@ -61,6 +65,24 @@
 	cpu-supply = <&reg_dcdc2>;
 };
 
+&emac1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rmii_pins>;
+	phy-mode = "rmii";
+	phy-handle = <&rmii_phy>;
+	phy-supply = <&reg_aldo1>;
+	allwinner,rx-delay-ps = <3100>;
+	allwinner,tx-delay-ps = <700>;
+	status = "okay";
+};
+
+&mdio1 {
+	rmii_phy: ethernet-phy@16 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <16>;
+	};
+};
+
 &ehci0 {
 	status = "okay";
 };
