diff --speed-large-files --no-dereference --minimal -Naur u-boot-182cb30084516c3085d2ca5dde3f105f7625d774/dts/upstream/src/arm64/rockchip/rk3576.dtsi u-boot-182cb30084516c3085d2ca5dde3f105f7625d774/dts/upstream/src/arm64/rockchip/rk3576.dtsi
--- u-boot-182cb30084516c3085d2ca5dde3f105f7625d774/dts/upstream/src/arm64/rockchip/rk3576.dtsi	2025-07-29 02:23:01.000000000 +0200
+++ u-boot-182cb30084516c3085d2ca5dde3f105f7625d774/dts/upstream/src/arm64/rockchip/rk3576.dtsi	2025-08-01 15:41:45.876747716 +0200
@@ -1354,7 +1354,7 @@
 			resets = <&cru SRST_A_UFS_BIU>, <&cru SRST_A_UFS_SYS>,
 				 <&cru SRST_A_UFS>, <&cru SRST_P_UFS_GRF>;
 			reset-names = "biu", "sys", "ufs", "grf";
-			reset-gpios = <&gpio4 RK_PD0 GPIO_ACTIVE_LOW>;
+			//reset-gpios = <&gpio4 RK_PD0 GPIO_ACTIVE_LOW>;
 			status = "disabled";
 		};
