diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3588s-u-boot.dtsi u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3588s-u-boot.dtsi
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3588s-u-boot.dtsi	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3588s-u-boot.dtsi	2026-01-23 11:44:16.480067761 +0100
@@ -11,7 +11,7 @@
 	};
 
 	chosen {
-		u-boot,spl-boot-order = "same-as-spl", &sdmmc, &sdhci;
+		u-boot,spl-boot-order = &sfc, &sdhci, &sdmmc, "same-as-spl";
 	};
 
 	dmc {
