diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3566-orangepi-3b.dtsi u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3566-orangepi-3b.dtsi
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3566-orangepi-3b.dtsi	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3566-orangepi-3b.dtsi	2026-01-21 10:17:46.253440221 +0100
@@ -12,8 +12,8 @@
 
 	aliases {
 		ethernet0 = &gmac1;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc0;
+		mmc0 = &sdmmc0;
+		mmc1 = &sdhci;
 		mmc2 = &sdmmc1;
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3566-rock-3c.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3566-rock-3c.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3566-rock-3c.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3566-rock-3c.dts	2026-01-21 10:18:43.196775060 +0100
@@ -13,8 +13,8 @@
 
 	aliases {
 		ethernet0 = &gmac1;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc0;
+		mmc0 = &sdmmc0;
+		mmc1 = &sdhci;
 		mmc2 = &sdmmc1;
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3568-rock-3a.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3568-rock-3a.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3568-rock-3a.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3568-rock-3a.dts	2026-01-21 10:19:09.793442440 +0100
@@ -13,8 +13,8 @@
 
 	aliases {
 		ethernet0 = &gmac1;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc0;
+		mmc0 = &sdmmc0;
+		mmc1 = &sdhci;
 		mmc2 = &sdmmc2;
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3568-rock-3b.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3568-rock-3b.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3568-rock-3b.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3568-rock-3b.dts	2026-01-21 10:19:24.440109487 +0100
@@ -15,8 +15,8 @@
 	aliases {
 		ethernet0 = &gmac0;
 		ethernet1 = &gmac1;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc0;
+		mmc0 = &sdmmc0;
+		mmc1 = &sdhci;
 		mmc2 = &sdmmc2;
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-nanopc-t6.dtsi u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-nanopc-t6.dtsi
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-nanopc-t6.dtsi	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-nanopc-t6.dtsi	2026-01-21 10:13:10.653432952 +0100
@@ -19,8 +19,8 @@
 	compatible = "friendlyarm,nanopc-t6", "rockchip,rk3588";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	adc-keys-0 {
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-orangepi-5.dtsi u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-orangepi-5.dtsi
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-orangepi-5.dtsi	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-orangepi-5.dtsi	2026-01-21 10:12:02.133431133 +0100
@@ -14,8 +14,8 @@
 
 / {
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-rock-5b-5bp-5t.dtsi u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-rock-5b-5bp-5t.dtsi
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-rock-5b-5bp-5t.dtsi	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-rock-5b-5bp-5t.dtsi	2026-01-21 10:14:03.670101015 +0100
@@ -9,8 +9,8 @@
 
 / {
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio;
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-rock-5-itx.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-rock-5-itx.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-rock-5-itx.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588-rock-5-itx.dts	2026-01-21 10:10:35.796762154 +0100
@@ -20,8 +20,8 @@
 	compatible = "radxa,rock-5-itx", "rockchip,rk3588";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio;
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-orangepi-5b.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-orangepi-5b.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-orangepi-5b.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-orangepi-5b.dts	2026-01-21 10:15:19.880103016 +0100
@@ -9,8 +9,8 @@
 	compatible = "xunlong,orangepi-5b", "rockchip,rk3588s";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-rock-5a.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-rock-5a.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-rock-5a.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-rock-5a.dts	2026-01-21 10:15:44.346770318 +0100
@@ -14,8 +14,8 @@
 
 	aliases {
 		ethernet0 = &gmac1;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	analog-sound {
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-rock-5c.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-rock-5c.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-rock-5c.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/dts/upstream/src/arm64/rockchip/rk3588s-rock-5c.dts	2026-01-21 10:15:59.426770712 +0100
@@ -17,8 +17,8 @@
 
 	aliases {
 		ethernet0 = &gmac1;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
