diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3328-generic.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3328-generic.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3328-generic.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3328-generic.dts	2026-01-21 10:56:28.470168919 +0100
@@ -11,8 +11,8 @@
 	compatible = "rockchip,rk3328";
 
 	aliases {
-		mmc0 = &emmc;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &emmc;
 	};
 
 	chosen {
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3399-generic.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3399-generic.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3399-generic.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3399-generic.dts	2026-01-21 10:56:05.903501638 +0100
@@ -11,8 +11,8 @@
 	compatible = "rockchip,rk3399";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3528-generic.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3528-generic.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3528-generic.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3528-generic.dts	2026-01-21 10:55:35.516834157 +0100
@@ -11,8 +11,8 @@
 	compatible = "rockchip,rk3528";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		serial0 = &uart0;
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3568-generic.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3568-generic.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3568-generic.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3568-generic.dts	2026-01-21 10:55:14.353500263 +0100
@@ -11,8 +11,8 @@
 	compatible = "rockchip,rk3568";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc0;
+		mmc0 = &sdmmc0;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3576-generic.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3576-generic.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3576-generic.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3576-generic.dts	2026-01-21 10:54:58.316833173 +0100
@@ -13,8 +13,8 @@
 	compatible = "rockchip,rk3576";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
diff --speed-large-files --no-dereference --minimal -Naur u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3588-generic.dts u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3588-generic.dts
--- u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3588-generic.dts	2026-01-05 21:49:22.000000000 +0100
+++ u-boot-127a42c7257a6ffbbd1575ed1cbaa8f5408a44b3/arch/arm/dts/rk3588-generic.dts	2026-01-21 10:54:33.226832498 +0100
@@ -11,8 +11,8 @@
 	compatible = "rockchip,rk3588";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
