diff --speed-large-files --no-dereference --minimal -Naur mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythtv/external/libmythdvdnav/config.h.in mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythtv/external/libmythdvdnav/config.h.in
--- mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythtv/external/libmythdvdnav/config.h.in	2025-03-28 21:06:21.000000000 +0100
+++ mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythtv/external/libmythdvdnav/config.h.in	2025-03-30 12:59:39.903377773 +0200
@@ -12,6 +12,6 @@
 #cmakedefine01 HAVE_GETTIMEOFDAY
 #cmakedefine01 HAVE_BIGENDIAN
 #cmakedefine01 HAVE_SYS_PARAM_H
-#cmakedefine   HAVE_DVDCSS_DVDCSS_H
+//#cmakedefine   HAVE_DVDCSS_DVDCSS_H
 
 #endif /* CONFIG_H */
