diff --speed-large-files --no-dereference --minimal -Naur mythtv-66e978d669dd1cadd2d5bc37914132c161a2e31c/mythtv/programs/mythfrontend/videodlg.cpp mythtv-66e978d669dd1cadd2d5bc37914132c161a2e31c/mythtv/programs/mythfrontend/videodlg.cpp
--- mythtv-66e978d669dd1cadd2d5bc37914132c161a2e31c/mythtv/programs/mythfrontend/videodlg.cpp	2025-01-05 06:35:06.000000000 +0100
+++ mythtv-66e978d669dd1cadd2d5bc37914132c161a2e31c/mythtv/programs/mythfrontend/videodlg.cpp	2025-01-05 13:25:00.693242977 +0100
@@ -1390,14 +1390,10 @@
     }
     else
     {
-        if (m_d->m_videoList)
-        {
-            m_d->m_videoList->refreshList(m_d->m_isFileBrowser,
+        m_d->m_videoList->refreshList(m_d->m_isFileBrowser,
                 m_d->m_parentalLevel.GetLevel(),
                 m_d->m_isFlatList, m_d->m_groupType);
-        }
-        if(m_d->m_videoList)
-            m_d->m_rootNode = m_d->m_videoList->GetTreeRoot();
+        m_d->m_rootNode = m_d->m_videoList->GetTreeRoot();
     }
 
     m_d->m_treeLoaded = true;
@@ -3093,10 +3089,7 @@
 {
     VideoMetadata *metadata = GetMetadata(GetItemCurrent());
     if (metadata)
-    {
-        auto cache = m_d->m_videoList->getListCache();
-        PlayVideo(metadata->GetFilename(), cache);
-    }
+        PlayVideo(metadata->GetFilename(), m_d->m_videoList->getListCache());
 }
 
 /** \fn VideoDialog::playVideoAlt()
@@ -3107,10 +3100,8 @@
 {
     VideoMetadata *metadata = GetMetadata(GetItemCurrent());
     if (metadata)
-    {
-        auto cache = m_d->m_videoList->getListCache();
-        PlayVideo(metadata->GetFilename(), cache, true);
-    }
+        PlayVideo(metadata->GetFilename(),
+                  m_d->m_videoList->getListCache(), true);
 }
 
 /** \fn VideoDialog::playFolder()
@@ -3148,8 +3139,8 @@
                 {
                     playing_time.start();
                     video_started = true;
-                    auto cache = m_d->m_videoList->getListCache();
-                    PlayVideo(metadata->GetFilename(), cache);
+                    PlayVideo(metadata->GetFilename(),
+                              m_d->m_videoList->getListCache());
                 }
             }
             i++;
@@ -3619,7 +3610,7 @@
     if (!metadata)
         return;
 
-    if (m_d->m_videoList && m_d->m_videoList->Delete(metadata->GetID()))
+    if (m_d->m_videoList->Delete(metadata->GetID()))
     {
         if (m_videoButtonTree)
             m_videoButtonTree->RemoveItem(item, false); // FIXME Segfault when true
