diff --speed-large-files --no-dereference --minimal -Naur mythtv-2d88074be2d8dd5d3653b2ca99e95fdcf498ec65/mythtv/libs/libmythtv/audioplayer.cpp mythtv-2d88074be2d8dd5d3653b2ca99e95fdcf498ec65/mythtv/libs/libmythtv/audioplayer.cpp
--- mythtv-2d88074be2d8dd5d3653b2ca99e95fdcf498ec65/mythtv/libs/libmythtv/audioplayer.cpp	2025-12-08 19:57:48.000000000 +0100
+++ mythtv-2d88074be2d8dd5d3653b2ca99e95fdcf498ec65/mythtv/libs/libmythtv/audioplayer.cpp	2025-12-08 21:14:25.026686927 +0100
@@ -127,10 +127,11 @@
             aos.m_init = false;
 
         m_audioOutput = AudioOutput::OpenAudio(aos);
-        if (m_audioOutput == nullptr || !m_audioOutput->isConfigured())
+/*        if (m_audioOutput == nullptr || !m_audioOutput->isConfigured())
         {
             errMsg = tr("Unable to create AudioOutput.");
         }
+*/
         AddVisuals();
     }
     else if (!m_noAudioIn && m_audioOutput)
