diff -Naur mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythbase/mythcorecontext.cpp mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythbase/mythcorecontext.cpp
--- mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythbase/mythcorecontext.cpp	2020-10-10 15:48:08.000000000 +0200
+++ mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythbase/mythcorecontext.cpp	2020-10-10 16:07:16.856666666 +0200
@@ -321,7 +321,8 @@
     proto_mismatch = false;
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!CheckProtoVersion(serverSock, timeout_in_ms, true))
+    if (!GetBoolSetting("IgnoreProtoVerMismatch", false)
+        && !CheckProtoVersion(serverSock, timeout_in_ms, true))
     {
         proto_mismatch = true;
         return false;
diff -Naur mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythbase/remotefile.cpp mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythbase/remotefile.cpp
--- mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythbase/remotefile.cpp	2020-10-10 15:48:08.000000000 +0200
+++ mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythbase/remotefile.cpp	2020-10-10 16:07:16.856666666 +0200
@@ -160,7 +160,8 @@
     QStringList strlist;
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!gCoreContext->CheckProtoVersion(lsock, 5000))
+    if (!gCoreContext->GetBoolSetting("IgnoreProtoVerMismatch", false)
+        && !gCoreContext->CheckProtoVersion(lsock, 5000))
     {
         LOG(VB_GENERAL, LOG_ERR, loc +
             QString("Failed validation to server %1:%2").arg(host).arg(port));
diff -Naur mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp
--- mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp	2020-10-10 15:48:08.000000000 +0200
+++ mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp	2020-10-10 16:07:16.859999999 +0200
@@ -45,7 +45,8 @@
     }
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!m_socket->Validate())
+    if (!gCoreContext->GetBoolSetting("IgnoreProtoVerMismatch", false)
+        && !m_socket->Validate())
     {
         LOG(VB_GENERAL, LOG_NOTICE, "Unable to confirm protocol version with backend.");
         m_socket->DecrRef();
diff -Naur mythtv-master-20201010-g68e85b9-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20201010-g68e85b9-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20201010-g68e85b9-old/mythtv/programs/mythfrontend/main.cpp	2020-10-10 15:48:07.000000000 +0200
+++ mythtv-master-20201010-g68e85b9-new/mythtv/programs/mythfrontend/main.cpp	2020-10-10 16:07:16.859999999 +0200
@@ -2037,7 +2037,8 @@
             return GENERIC_EXIT_NO_THEME;
     }
 
-    if (!UpgradeTVDatabaseSchema(false, false, true))
+    if (!gCoreContext->GetBoolSetting("IgnoreSchemaVerMismatch", false)
+        && !UpgradeTVDatabaseSchema(false, false, true))
     {
         LOG(VB_GENERAL, LOG_ERR,
             "Couldn't upgrade database to new schema, exiting.");
