diff --speed-large-files --no-dereference --minimal -Naur mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/cmake/FindLibX264.cmake mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/cmake/FindLibX264.cmake
--- mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/cmake/FindLibX264.cmake	2025-03-28 21:06:21.000000000 +0100
+++ mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/cmake/FindLibX264.cmake	2025-03-29 12:53:15.550071606 +0100
@@ -38,7 +38,7 @@
   the directory containing the X264 headers
 
 #]=======================================================================]
-
+return()
 find_package(PkgConfig)
 pkg_check_modules(PC_LibX264 QUIET x264)
 if(NOT PC_LibX264_FOUND)
