diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythbase/mythcorecontext.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythbase/mythcorecontext.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythbase/mythcorecontext.cpp	2021-05-18 09:42:32.000000000 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythbase/mythcorecontext.cpp	2021-05-18 16:11:25.086666604 +0200
@@ -329,7 +329,8 @@
     proto_mismatch = false;
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!CheckProtoVersion(serverSock, timeout, true))
+    if (!GetBoolSetting("IgnoreProtoVerMismatch", false)
+        && !CheckProtoVersion(serverSock, timeout, true))
     {
         proto_mismatch = true;
         return false;
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythbase/remotefile.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythbase/remotefile.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythbase/remotefile.cpp	2021-05-18 09:42:32.000000000 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythbase/remotefile.cpp	2021-05-18 16:11:25.086666604 +0200
@@ -160,7 +160,8 @@
     QStringList strlist;
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!gCoreContext->CheckProtoVersion(lsock, 5s))
+    if (!gCoreContext->GetBoolSetting("IgnoreProtoVerMismatch", false)
+        && !gCoreContext->CheckProtoVersion(lsock, 5s))
     {
         LOG(VB_GENERAL, LOG_ERR, loc +
             QString("Failed validation to server %1:%2").arg(host).arg(port));
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp	2021-05-18 09:42:31.000000000 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp	2021-05-18 16:11:25.086666604 +0200
@@ -47,7 +47,8 @@
     }
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!m_socket->Validate())
+    if (!gCoreContext->GetBoolSetting("IgnoreProtoVerMismatch", false)
+        && !m_socket->Validate())
     {
         LOG(VB_GENERAL, LOG_NOTICE, "Unable to confirm protocol version with backend.");
         m_socket->DecrRef();
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythfrontend/main.cpp	2021-05-18 09:42:26.000000000 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythfrontend/main.cpp	2021-05-18 16:11:25.086666604 +0200
@@ -2033,7 +2033,8 @@
             return GENERIC_EXIT_NO_THEME;
     }
 
-    if (!UpgradeTVDatabaseSchema(false, false, true))
+    if (!gCoreContext->GetBoolSetting("IgnoreSchemaVerMismatch", false)
+        && !UpgradeTVDatabaseSchema(false, false, true))
     {
         LOG(VB_GENERAL, LOG_ERR,
             "Couldn't upgrade database to new schema, exiting.");
