diff --speed-large-files --no-dereference --minimal -Naur mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythtv/cmake/MythFindPackages.cmake mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythtv/cmake/MythFindPackages.cmake
--- mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythtv/cmake/MythFindPackages.cmake	2025-03-28 21:06:21.000000000 +0100
+++ mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythtv/cmake/MythFindPackages.cmake	2025-03-30 14:18:57.216727260 +0200
@@ -220,7 +220,6 @@
 #
 if(ENABLE_X11
-   AND NOT APPLE
-   AND NOT CMAKE_CROSSCOMPILING)
+   AND NOT APPLE)
   find_package(X11)
   if(TARGET X11::X11)
     if(NOT TARGET X11::Xrandr)
