diff --speed-large-files --no-dereference --minimal -Naur mythtv-e1f9afea2d0f9f65193dc554e88b9c9ce5295273/cmake/MythFindQt.cmake mythtv-e1f9afea2d0f9f65193dc554e88b9c9ce5295273/cmake/MythFindQt.cmake
--- mythtv-e1f9afea2d0f9f65193dc554e88b9c9ce5295273/cmake/MythFindQt.cmake	2025-04-02 12:55:50.000000000 +0200
+++ mythtv-e1f9afea2d0f9f65193dc554e88b9c9ce5295273/cmake/MythFindQt.cmake	2025-04-05 11:01:24.266721710 +0200
@@ -21,19 +21,21 @@
     Test
     Widgets
     Xml)
-if(NOT CMAKE_CROSSCOMPILING)
-  list(APPEND _REQUIRED_COMPONENTS DBus)
-  if(${QT_VERSION_MAJOR} EQUAL 5)
-    set(_OPTIONAL_COMPONENTS Script ScriptTools)
-    if(ENABLE_QTWEBENGINE)
-      list(APPEND _OPTIONAL_COMPONENTS Quick WebEngine WebEngineWidgets)
-    endif()
-  else()
-    if(ENABLE_QTWEBENGINE)
-      set(_OPTIONAL_COMPONENTS Quick WebEngineQuick WebEngineWidgets)
-    endif()
+
+list(APPEND _REQUIRED_COMPONENTS DBus)
+
+if(${QT_VERSION_MAJOR} EQUAL 5)
+  set(_OPTIONAL_COMPONENTS Script ScriptTools)
+  if(ENABLE_QTWEBENGINE)
+    list(APPEND _OPTIONAL_COMPONENTS Quick WebEngine WebEngineWidgets)
   endif()
-elseif(ANDROID)
+else()
+  if(ENABLE_QTWEBENGINE)
+    set(_OPTIONAL_COMPONENTS Quick WebEngineQuick WebEngineWidgets)
+  endif()
+endif()
+
+if(ANDROID)
   if(${QT_VERSION_MAJOR} EQUAL 5)
     list(APPEND _REQUIRED_COMPONENTS AndroidExtras)
   endif()
