diff --speed-large-files --no-dereference --minimal -Naur mythtv-a5c6628f511d0b1d5dd33879179e900017499409/mythtv/libs/libmythbase/programinfo.cpp mythtv-a5c6628f511d0b1d5dd33879179e900017499409/mythtv/libs/libmythbase/programinfo.cpp
--- mythtv-a5c6628f511d0b1d5dd33879179e900017499409/mythtv/libs/libmythbase/programinfo.cpp	2025-01-03 01:15:23.000000000 +0100
+++ mythtv-a5c6628f511d0b1d5dd33879179e900017499409/mythtv/libs/libmythbase/programinfo.cpp	2025-01-03 13:39:01.909728991 +0100
@@ -3698,7 +3698,7 @@
 
     MSqlQuery query(MSqlQuery::InitCon());
 
-    query.prepare("SELECT mark, type, `offset` "
+    query.prepare("SELECT mark, type, offset "
                   "FROM filemarkup "
                   "WHERE filename = :PATH AND "
                   "      type     = :TYPE "
