diff --speed-large-files --no-dereference --minimal -Naur mythtv-88bdf676b0264a4bebab564b5a6c7ef3a8c06b7f/mythtv/external/libmythdvdnav/dvdread/dvd_input.c mythtv-88bdf676b0264a4bebab564b5a6c7ef3a8c06b7f/mythtv/external/libmythdvdnav/dvdread/dvd_input.c
--- mythtv-88bdf676b0264a4bebab564b5a6c7ef3a8c06b7f/mythtv/external/libmythdvdnav/dvdread/dvd_input.c	2026-08-06 00:50:14.000000000 +0200
+++ mythtv-88bdf676b0264a4bebab564b5a6c7ef3a8c06b7f/mythtv/external/libmythdvdnav/dvdread/dvd_input.c	2026-08-12 11:03:14.245780020 +0200
@@ -19,6 +19,7 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+//#include <fcntl.h>
 #include "config.h"                  /* Required for HAVE_DVDCSS_DVDCSS_H */
 #include <stdio.h>                               /* fprintf */
 #include <stdlib.h>                              /* free */
@@ -30,6 +31,7 @@
 #include "dvdread/mythdvdio.h"
 #include <string.h>
 
+
 /* The function pointers that is the exported interface of this file. */
 dvd_input_t (*dvdinput_open)  (const char *, void *, dvd_reader_stream_cb *);
 int         (*dvdinput_close) (dvd_input_t);
@@ -265,6 +267,9 @@
 /**
  * Setup read functions with either libdvdcss or minimal DVD access.
  */
+extern void *dlopen(const char *filename, int flags);
+extern void *dlsym(void *handle, const char *symbol);
+extern int dlclose(void *handle);
 int dvdinput_setup(const char *path)
 {
   void *dvdcss_library = NULL;
@@ -285,7 +290,8 @@
 #else
   #define CSS_LIB "libdvdcss.so.2"
 #endif
-  dvdcss_library = dlopen(CSS_LIB, RTLD_LAZY);
+  //dvdcss_library = dlopen(CSS_LIB, RTLD_LAZY);
+  dvdcss_library = dlopen(CSS_LIB, 1);
 
   if(dvdcss_library != NULL) {
 #if defined(__OpenBSD__) && !defined(__ELF__) || defined(__OS2__)
