diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/tv_rec.cpp	2018-10-12 10:49:30.000000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/tv_rec.cpp	2018-10-12 10:51:15.303333332 +0200
@@ -99,6 +99,7 @@
       audioSampleRateDB(0),
       overRecordSecNrml(0),         overRecordSecCat(0),
       overRecordCategory(""),
+      rec_delay_per_card(0),
       // Configuration variables from setup rutines
       inputid(_inputid), parentid(0), ispip(false),
       // State variables
@@ -195,6 +196,7 @@
     overRecordSecNrml = gCoreContext->GetNumSetting("RecordOverTime");
     overRecordSecCat  = gCoreContext->GetNumSetting("CategoryOverTime") * 60;
     overRecordCategory= gCoreContext->GetSetting("OverTimeCategory");
+    rec_delay_per_card= gCoreContext->GetNumSetting("ConcurrentRecDelay", 0);
 
     eventThread->start();
 
@@ -1014,6 +1016,32 @@
     autoRunJobs.erase(autoJob);
 }
 
+static int no_capturecards(uint inputid)
+{
+    MSqlQuery query(MSqlQuery::InitCon());
+
+    QString str =
+        "SELECT COUNT(cardid) "
+        "FROM capturecard ";
+
+    if (inputid)
+        str += "WHERE cardid < :INPUTID";
+
+    query.prepare(str);
+
+    if (inputid)
+        query.bindValue(":INPUTID", inputid);
+
+    if (!query.exec() || !query.isActive())
+    {
+        MythDB::DBError("no_capturecards", query);
+        return -1;
+    }
+    else if (query.next())
+        return query.value(0).toInt();
+    return -1;
+}
+
 #define TRANSITION(ASTATE,BSTATE) \
    ((internalState == ASTATE) && (desiredNextState == BSTATE))
 #define SET_NEXT() do { nextState = desiredNextState; changed = true; } while(0)
@@ -1073,6 +1101,18 @@
     }
     else if (TRANSITION(kState_None, kState_RecordingOnly))
     {
+        if (rec_delay_per_card > 0)
+        {
+            int card_pos = no_capturecards(inputid);
+            int rec_start_delay = rec_delay_per_card * card_pos;
+
+            LOG(VB_GENERAL, LOG_INFO, LOC +
+                QString("Delaying start of recording by %1msec").arg(rec_start_delay));
+            std::this_thread::sleep_for(std::chrono::milliseconds(rec_start_delay));
+            LOG(VB_GENERAL, LOG_INFO, LOC +
+                "Now start recording...");
+        }
+
         SetPseudoLiveTVRecording(nullptr);
         tuningRequests.enqueue(TuningRequest(kFlagRecording, curRecording));
         SET_NEXT();
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/tv_rec.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/tv_rec.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/tv_rec.h	2018-10-12 10:49:30.000000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/tv_rec.h	2018-10-12 10:51:15.303333332 +0200
@@ -370,6 +370,7 @@
     int     overRecordSecNrml;
     int     overRecordSecCat;
     QString overRecordCategory;
+    int     rec_delay_per_card;
 
     // Configuration variables from setup routines
     uint              inputid;
