diff --speed-large-files --no-dereference --minimal -Naur mythtv-4efd8a3c211d5208ddfb889918a9681ed40f1c1f/mythplugins/configure mythtv-4efd8a3c211d5208ddfb889918a9681ed40f1c1f/mythplugins/configure
--- mythtv-4efd8a3c211d5208ddfb889918a9681ed40f1c1f/mythplugins/configure	2023-05-16 02:24:38.000000000 +0200
+++ mythtv-4efd8a3c211d5208ddfb889918a9681ed40f1c1f/mythplugins/configure	2023-05-16 11:02:48.936767526 +0200
@@ -302,6 +302,7 @@
 newexif
 dcraw
 cdio
+cross_compile
 "
 
 DEPEND_LIST="
@@ -343,6 +344,7 @@
   --enable-all             Enable all options
   --previous               use previous configure parameters if possible
   --extra-ldflags=ELDFLAGS add ELDFLAGS to LDFLAGS [$LDFLAGS]
+  --enable-cross-compile   assume a cross-compiler is used
 
 MythArchive related options:
   --enable-mytharchive     build the mytharchive plugin [$archive]
@@ -401,6 +403,7 @@
     python
     sysroot
     mythroot
+    cross_compile
 "
 
 CONFIGURATION_OPTS=""
@@ -470,7 +473,11 @@
 check_py_lib(){
     log check_py_lib "$@"
     lib=${1}
-    check_cmd $python <<EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Python:$lib Be sure You have it for cross target!"
+        return 0
+    else
+        check_cmd $python <<EOF
 from distutils.sysconfig import get_python_lib
 import sys
 for mythroot in '${mythroot}'.split(':'):
@@ -484,6 +491,7 @@
     sys.exit(1)
 sys.exit(0)
 EOF
+    fi
 }
 
 # set temporary file name
@@ -644,7 +652,12 @@
 if enabled weather; then
     check_pl_lib(){
         log check_pl_lib "$@"
-        check_cmd perl -e "use ${1};"
+        if enabled cross_compile ; then
+            echo "You are cross-compiling! Assuming YES for Perl:$@ Be sure You have it for cross target!"
+            return 0
+        else
+            check_cmd perl -e "use ${1};"
+        fi
     }
 
     disable_weather(){
diff --speed-large-files --no-dereference --minimal -Naur mythtv-4efd8a3c211d5208ddfb889918a9681ed40f1c1f/mythtv/configure mythtv-4efd8a3c211d5208ddfb889918a9681ed40f1c1f/mythtv/configure
--- mythtv-4efd8a3c211d5208ddfb889918a9681ed40f1c1f/mythtv/configure	2023-05-16 02:24:38.000000000 +0200
+++ mythtv-4efd8a3c211d5208ddfb889918a9681ed40f1c1f/mythtv/configure	2023-05-16 11:03:29.743432970 +0200
@@ -1239,30 +1239,44 @@
 }
 
 is_python3() {
-    log is_python3
-    check_cmd $python << EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for IsPython3 . Be sure You have it for cross target!"
+        return 0
+    else
+        log is_python3
+        check_cmd $python << EOF
 import sys
 if sys.version_info > (3,0):
     sys.exit(0)
 else:
     sys.exit(1)
 EOF
+    fi
 }
 
 check_python(){
-    log check_python "$@"
-    version=${1}
-    check_cmd $python <<EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Python ver > 3.6.0. Be sure You have it for cross target!"
+        return 0
+    else
+        log check_python "$@"
+        version=${1}
+        check_cmd $python <<EOF
 import sys
 if sys.version_info < $version:
     sys.exit(1)
 EOF
+    fi
 }
 
 check_py_lib(){
     log check_py_lib "$@"
     lib=${1}
-    check_cmd $python <<EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Python:$lib Be sure You have it for cross target!"
+        return 0
+    else
+        check_cmd $python <<EOF
 import sys
 try:
     import $lib
@@ -1270,6 +1284,7 @@
     sys.exit(1)
 sys.exit(0)
 EOF
+    fi
 }
 
 check_py_lib_version(){
@@ -1292,7 +1307,12 @@
 
 check_pl_lib(){
     log check_pl_lib "$@"
-    check_cmd perl -e "use ${1};"
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Perl:$@ Be sure You have it for cross target!"
+        return 0
+    else
+        check_cmd perl -e "use ${1};"
+    fi
 }
 
 check_ld_cxx(){
