diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythtv/cardutil.cpp	2020-09-04 11:35:02.793333334 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythtv/cardutil.cpp	2020-09-04 11:35:04.043333333 +0200
@@ -2451,7 +2451,7 @@
 QString CardUtil::GetDeviceLabel(uint inputid)
 {
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
