diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/cardutil.cpp	2020-05-07 12:45:35.556666667 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/cardutil.cpp	2020-05-07 12:45:36.956666667 +0200
@@ -2437,7 +2437,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
