diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/cardutil.cpp	2019-02-17 16:57:47.023333336 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/cardutil.cpp	2019-02-17 16:57:49.003333336 +0100
@@ -1944,7 +1944,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
