diff -Naur mythtv-master-20190320-gb121bce-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20190320-gb121bce-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20190320-gb121bce-old/mythtv/libs/libmythtv/cardutil.cpp	2019-03-20 22:45:04.833333333 +0100
+++ mythtv-master-20190320-gb121bce-new/mythtv/libs/libmythtv/cardutil.cpp	2019-03-20 22:45:07.993333334 +0100
@@ -1943,7 +1943,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
