diff -Naur mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmythtv/cardutil.cpp	2019-12-12 19:15:31.996666669 +0100
+++ mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmythtv/cardutil.cpp	2019-12-12 19:15:33.820000002 +0100
@@ -2443,7 +2443,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
