diff -Naur mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythtv/cardutil.cpp	2018-10-20 14:56:12.273333335 +0200
+++ mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythtv/cardutil.cpp	2018-10-20 14:56:17.463333334 +0200
@@ -1942,7 +1942,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
