diff -Naur mythtv-master-20200530-ga180830-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20200530-ga180830-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20200530-ga180830-old/mythtv/libs/libmythtv/cardutil.cpp	2020-05-30 12:59:55.473333334 +0200
+++ mythtv-master-20200530-ga180830-new/mythtv/libs/libmythtv/cardutil.cpp	2020-05-30 12:59:58.663333335 +0200
@@ -2437,7 +2437,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
