diff -Naur mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmythtv/cardutil.cpp	2020-04-09 12:24:48.620923978 +0200
+++ mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmythtv/cardutil.cpp	2020-04-09 12:24:50.137590590 +0200
@@ -2435,7 +2435,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
