diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythtv/cardutil.cpp	2019-03-29 08:39:48.300000001 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythtv/cardutil.cpp	2019-03-29 08:39:52.066666667 +0100
@@ -2187,7 +2187,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
