diff -Naur mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythtv/cardutil.cpp	2020-10-10 16:07:18.819999999 +0200
+++ mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythtv/cardutil.cpp	2020-10-10 16:07:20.326666666 +0200
@@ -2440,7 +2440,7 @@
 QString CardUtil::GetDeviceLabel(uint inputid)
 {
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
