diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/cardutil.cpp	2019-03-16 10:52:32.273333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/cardutil.cpp	2019-03-16 10:52:36.716666667 +0100
@@ -1944,7 +1944,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
