diff -Naur mythtv-master-20200207-g2869463-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20200207-g2869463-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20200207-g2869463-old/mythtv/libs/libmythtv/cardutil.cpp	2020-02-08 14:18:07.373333336 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/libs/libmythtv/cardutil.cpp	2020-02-08 14:18:09.146666668 +0100
@@ -2435,7 +2435,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
