diff -Naur mythtv-master-20200109-gc116014-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20200109-gc116014-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20200109-gc116014-old/mythtv/libs/libmythtv/cardutil.cpp	2020-01-09 22:03:50.203333334 +0100
+++ mythtv-master-20200109-gc116014-new/mythtv/libs/libmythtv/cardutil.cpp	2020-01-09 22:03:52.706666668 +0100
@@ -2435,7 +2435,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
