diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/cardutil.cpp	2020-09-15 09:22:05.800000001 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/cardutil.cpp	2020-09-15 09:22:07.556666667 +0200
@@ -2453,7 +2453,7 @@
 QString CardUtil::GetDeviceLabel(uint inputid)
 {
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
