diff -Naur mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/cardutil.cpp	2020-09-21 09:17:30.993333335 +0200
+++ mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/cardutil.cpp	2020-09-21 09:17:33.746666668 +0200
@@ -2453,7 +2453,7 @@
 QString CardUtil::GetDeviceLabel(uint inputid)
 {
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
