diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/cardutil.cpp	2019-12-27 12:18:38.576666667 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/cardutil.cpp	2019-12-27 12:18:39.866666667 +0100
@@ -2445,7 +2445,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
