diff -Naur mythtv-master-20200311-g625fc71-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20200311-g625fc71-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20200311-g625fc71-old/mythtv/libs/libmythtv/cardutil.cpp	2020-03-14 13:29:07.653333334 +0100
+++ mythtv-master-20200311-g625fc71-new/mythtv/libs/libmythtv/cardutil.cpp	2020-03-14 13:29:10.213333334 +0100
@@ -2435,7 +2435,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
