diff -Naur mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/cardutil.cpp	2020-09-25 09:20:29.050000001 +0200
+++ mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/cardutil.cpp	2020-09-25 09:20:32.680000001 +0200
@@ -2456,7 +2456,7 @@
 QString CardUtil::GetDeviceLabel(uint inputid)
 {
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
