diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/cardutil.cpp	2018-10-12 10:51:13.536666664 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/cardutil.cpp	2018-10-12 10:51:16.650000000 +0200
@@ -1942,7 +1942,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
