diff -Naur mythtv-master-build-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-master-build-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-master-build-old/mythtv/libs/libmythtv/tv_rec.cpp	2016-06-12 10:56:41.000000000 +0200
+++ mythtv-master-build-new/mythtv/libs/libmythtv/tv_rec.cpp	2016-06-12 11:17:10.423333315 +0200
@@ -97,6 +97,7 @@
       overRecordSecNrml(0),         overRecordSecCat(0),
       overRecordCategory(""),
       recLimit(0),
+      rec_delay_per_card(0),
       // Configuration variables from setup rutines
       inputid(_inputid), ispip(false),
       // State variables
@@ -179,6 +180,7 @@
     overRecordSecCat  = gCoreContext->GetNumSetting("CategoryOverTime") * 60;
     overRecordCategory= gCoreContext->GetSetting("OverTimeCategory");
     recLimit = CardUtil::GetRecLimit(inputid);
+    rec_delay_per_card= gCoreContext->GetNumSetting("ConcurrentRecDelay", 0);
 
     eventThread->start();
 
@@ -1006,6 +1008,32 @@
     autoRunJobs.erase(autoJob);
 }
 
+static int no_capturecards(uint inputid)
+{
+    MSqlQuery query(MSqlQuery::InitCon());
+
+    QString str =
+        "SELECT COUNT(cardid) "
+        "FROM capturecard ";
+
+    if (inputid)
+        str += "WHERE cardid < :INPUTID";
+
+    query.prepare(str);
+
+    if (inputid)
+        query.bindValue(":INPUTID", inputid);
+
+    if (!query.exec() || !query.isActive())
+    {
+        MythDB::DBError("no_capturecards", query);
+        return -1;
+    }
+    else if (query.next())
+        return query.value(0).toInt();
+    return -1;
+}
+
 #define TRANSITION(ASTATE,BSTATE) \
    ((internalState == ASTATE) && (desiredNextState == BSTATE))
 #define SET_NEXT() do { nextState = desiredNextState; changed = true; } while(0)
@@ -1065,6 +1093,18 @@
     }
     else if (TRANSITION(kState_None, kState_RecordingOnly))
     {
+        if (rec_delay_per_card > 0)
+        {
+            int card_pos = no_capturecards(inputid);
+            int rec_start_delay = rec_delay_per_card * card_pos;
+
+            LOG(VB_GENERAL, LOG_INFO, LOC +
+                QString("Delaying start of recording by %1msec").arg(rec_start_delay));
+            usleep(rec_start_delay*1000);
+            LOG(VB_GENERAL, LOG_INFO, LOC +
+                "Now start recording...");
+        }
+
         SetPseudoLiveTVRecording(NULL);
         tuningRequests.enqueue(TuningRequest(kFlagRecording, curRecording));
         SET_NEXT();
diff -Naur mythtv-master-build-old/mythtv/libs/libmythtv/tv_rec.h mythtv-master-build-new/mythtv/libs/libmythtv/tv_rec.h
--- mythtv-master-build-old/mythtv/libs/libmythtv/tv_rec.h	2016-06-12 10:56:41.000000000 +0200
+++ mythtv-master-build-new/mythtv/libs/libmythtv/tv_rec.h	2016-06-12 11:18:01.170000001 +0200
@@ -370,6 +370,7 @@
     QString overRecordCategory;
     InputGroupMap igrp;
     uint    recLimit;
+    int     rec_delay_per_card;
 
     // Configuration variables from setup routines
     uint              inputid;
