diff -Naur mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/eitcache.cpp mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/eitcache.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/eitcache.cpp	2015-04-30 10:48:01.859461995 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/eitcache.cpp	2015-04-30 10:48:50.589264325 +0200
@@ -242,7 +242,7 @@
     }
 
     if (eventMap->size())
-        LOG(VB_EIT, LOG_INFO, LOC + QString("Loaded %1 entries for channel %2")
+        LOG(VB_GENERAL, LOG_INFO, LOC + QString("Loaded %1 entries for channel %2")
                 .arg(eventMap->size()).arg(chanid));
 
     entryCnt += eventMap->size();
diff -Naur mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/eitscanner.cpp mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/eitscanner.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/eitscanner.cpp	2015-04-30 10:48:01.856128675 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/eitscanner.cpp	2015-04-30 10:48:50.589264325 +0200
@@ -109,7 +109,7 @@
         // but not in the last 60 seconds
         if (!activeScan && eitCount && (t.elapsed() > 60 * 1000))
         {
-            LOG(VB_EIT, LOG_INFO,
+            LOG(VB_GENERAL, LOG_INFO,
                 LOC_ID + QString("Added %1 EIT Events").arg(eitCount));
             eitCount = 0;
             RescheduleRecordings();
@@ -121,7 +121,7 @@
             // if there have been any new events, tell scheduler to run.
             if (eitCount)
             {
-                LOG(VB_EIT, LOG_INFO,
+                LOG(VB_GENERAL, LOG_INFO,
                     LOC_ID + QString("Added %1 EIT Events").arg(eitCount));
                 eitCount = 0;
                 RescheduleRecordings();
@@ -137,7 +137,7 @@
                 {
                     eitHelper->SetChannelID(ChannelUtil::GetChanID(
                         rec->GetSourceID(), *activeScanNextChan));
-                    LOG(VB_EIT, LOG_INFO,
+                    LOG(VB_GENERAL, LOG_INFO,
                         LOC_ID + QString("Now looking for EIT data on "
                                          "multiplex of channel %1")
                         .arg(*activeScanNextChan));
@@ -202,7 +202,7 @@
     eitHelper->SetChannelID(channel->GetChanID());
     eitHelper->SetSourceID(ChannelUtil::GetSourceIDForChannel(channel->GetChanID()));
 
-    LOG(VB_EIT, LOG_INFO, LOC_ID + "Started passive scan.");
+    LOG(VB_GENERAL, LOG_INFO, LOC_ID + "Started passive scan.");
 }
 
 /** \fn EITScanner::StopPassiveScan(void)
@@ -259,7 +259,7 @@
         activeScanNextChan = activeScanChannels.begin();
     }
 
-    LOG(VB_EIT, LOG_INFO, LOC_ID +
+    LOG(VB_GENERAL, LOG_INFO, LOC_ID +
         QString("StartActiveScan called with %1 multiplexes")
             .arg(activeScanChannels.size()));
 
