diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2015-01-06 16:48:41.237927434 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2015-01-06 16:49:13.444420566 +0100
@@ -507,6 +507,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(networkID).arg(sdt->OriginalNetworkID()).arg(transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != networkID || sdt->TSID() != transportID)
     {
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
