diff -Naur mythtv-master-20201010-g68e85b9-old/mythtv/contrib/user_jobs/mythlink.pl mythtv-master-20201010-g68e85b9-new/mythtv/contrib/user_jobs/mythlink.pl
--- mythtv-master-20201010-g68e85b9-old/mythtv/contrib/user_jobs/mythlink.pl	2020-10-10 16:07:23.423333333 +0200
+++ mythtv-master-20201010-g68e85b9-new/mythtv/contrib/user_jobs/mythlink.pl	2020-10-10 16:07:24.886666666 +0200
@@ -29,7 +29,7 @@
     our ($hostname, $dbh, $sh, $q, $count, $base_dir);
 
 # Default filename format
-    $dformat = '%T %- %Y-%m-%d, %g-%i %A %- %S';
+    $dformat = '%U/%T %- %Y-%m-%d, %g-%i %A %- %S';
 # Default separator character
     $dseparator = '-';
 # Default replacement character
