diff -Naur mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/cardutil.cpp	2015-04-30 10:48:58.749231232 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/cardutil.cpp	2015-04-30 10:49:56.462330569 +0200
@@ -1936,7 +1936,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardid, displayname "
                   "FROM capturecard WHERE cardid = :CARDID ");
     query.bindValue(":CARDID", cardid);
 
diff -Naur mythtv-master-20150429-g6523186-old/mythtv/programs/mythfrontend/statusbox.cpp mythtv-master-20150429-g6523186-new/mythtv/programs/mythfrontend/statusbox.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/programs/mythfrontend/statusbox.cpp	2015-04-30 10:49:09.772519863 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/programs/mythfrontend/statusbox.cpp	2015-04-30 10:49:56.462330569 +0200
@@ -862,7 +862,7 @@
 
         QString tun = tr("Tuner %1 %2 %3");
         QString devlabel = CardUtil::GetDeviceLabel(
-            query.value(1).toString(), query.value(2).toString());
+            query.value(0).toInt());
 
         QString shorttuner = tun.arg(cardid).arg("").arg(status);
         QString longtuner = tun.arg(cardid).arg(devlabel).arg(status);
