diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/contrib/user_jobs/mythlink.pl mythtv-master-20200904-gca3ee6a-new/mythtv/contrib/user_jobs/mythlink.pl
--- mythtv-master-20200904-gca3ee6a-old/mythtv/contrib/user_jobs/mythlink.pl	2020-09-04 11:35:14.523333331 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/contrib/user_jobs/mythlink.pl	2020-09-04 11:35:18.400000000 +0200
@@ -29,7 +29,7 @@
     our ($hostname, $dbh, $sh, $q, $count, $base_dir);
 
 # Default filename format
-    $dformat = '%T %- %Y-%m-%d, %g-%i %A %- %S';
+    $dformat = '%U/%T %- %Y-%m-%d, %g-%i %A %- %S';
 # Default separator character
     $dseparator = '-';
 # Default replacement character
