diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/cardutil.cpp	2015-01-06 16:49:02.507812850 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/cardutil.cpp	2015-01-06 16:49:32.344318792 +0100
@@ -2101,8 +2101,8 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
-                  "FROM capturecard WHERE cardid = :CARDID ");
+    query.prepare("SELECT cardid, displayname "
+                  "FROM cardinput WHERE cardid = :CARDID ");
     query.bindValue(":CARDID", cardid);
 
     if (query.exec() && query.next())
diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/statusbox.cpp mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/statusbox.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/statusbox.cpp	2015-01-06 16:49:06.447791625 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/statusbox.cpp	2015-01-06 16:49:32.344318792 +0100
@@ -866,7 +866,7 @@
 
         QString tun = tr("Tuner %1 %2 %3");
         QString devlabel = CardUtil::GetDeviceLabel(
-            query.value(1).toString(), query.value(2).toString());
+            query.value(0).toInt());
 
         QString shorttuner = tun.arg(cardid).arg("").arg(status);
         QString longtuner = tun.arg(cardid).arg(devlabel).arg(status);
