diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/contrib/user_jobs/mythlink.pl mythtv-master-20200507-ge586e5a-new/mythtv/contrib/user_jobs/mythlink.pl
--- mythtv-master-20200507-ge586e5a-old/mythtv/contrib/user_jobs/mythlink.pl	2020-05-07 12:45:40.750000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/contrib/user_jobs/mythlink.pl	2020-05-07 12:45:42.126666667 +0200
@@ -29,7 +29,7 @@
     our ($hostname, $dbh, $sh, $q, $count, $base_dir);
 
 # Default filename format
-    $dformat = '%T %- %Y-%m-%d, %g-%i %A %- %S';
+    $dformat = '%U/%T %- %Y-%m-%d, %g-%i %A %- %S';
 # Default separator character
     $dseparator = '-';
 # Default replacement character
