diff -Naur mythtv-master-20140506-gd0e25ac-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20140506-gd0e25ac-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20140506-gd0e25ac-old/mythtv/libs/libmythtv/cardutil.cpp	2014-05-06 14:52:44.105646687 +0200
+++ mythtv-master-20140506-gd0e25ac-new/mythtv/libs/libmythtv/cardutil.cpp	2014-05-06 14:53:06.068902946 +0200
@@ -2101,8 +2101,8 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
-                  "FROM capturecard WHERE cardid = :CARDID ");
+    query.prepare("SELECT cardid, displayname "
+                  "FROM cardinput WHERE cardid = :CARDID ");
     query.bindValue(":CARDID", cardid);
 
     if (query.exec() && query.next())
diff -Naur mythtv-master-20140506-gd0e25ac-old/mythtv/programs/mythfrontend/statusbox.cpp mythtv-master-20140506-gd0e25ac-new/mythtv/programs/mythfrontend/statusbox.cpp
--- mythtv-master-20140506-gd0e25ac-old/mythtv/programs/mythfrontend/statusbox.cpp	2014-05-06 14:52:52.378951008 +0200
+++ mythtv-master-20140506-gd0e25ac-new/mythtv/programs/mythfrontend/statusbox.cpp	2014-05-06 14:53:06.072236267 +0200
@@ -866,7 +866,7 @@
 
         QString tun = tr("Tuner %1 %2 %3");
         QString devlabel = CardUtil::GetDeviceLabel(
-            query.value(1).toString(), query.value(2).toString());
+            query.value(0).toInt());
 
         QString shorttuner = tun.arg(cardid).arg("").arg(status);
         QString longtuner = tun.arg(cardid).arg(devlabel).arg(status);
