diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/programs/mythbackend/scheduler.cpp mythtv-master-20150105-g4216e00-new/mythtv/programs/mythbackend/scheduler.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/programs/mythbackend/scheduler.cpp	2015-01-06 16:49:26.934347934 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/programs/mythbackend/scheduler.cpp	2015-01-06 16:49:49.094228564 +0100
@@ -179,8 +179,7 @@
         LOG(VB_GENERAL, LOG_ERR, LOC +
                 "No capture cards are defined in the database.\n\t\t\t"
                 "Perhaps you should re-read the installation instructions?");
-        error = GENERIC_EXIT_SETUP_ERROR;
-        return false;
+        return true;
     }
 
     query.prepare("SELECT sourceid,name FROM videosource ORDER BY sourceid;");
