diff -Naur mythtv-master-20150429-g6523186-old/mythtv/programs/mythbackend/scheduler.cpp mythtv-master-20150429-g6523186-new/mythtv/programs/mythbackend/scheduler.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/programs/mythbackend/scheduler.cpp	2015-04-30 10:49:50.355688657 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/programs/mythbackend/scheduler.cpp	2015-04-30 10:50:09.145612489 +0200
@@ -179,8 +179,7 @@
         LOG(VB_GENERAL, LOG_ERR, LOC +
                 "No capture cards are defined in the database.\n\t\t\t"
                 "Perhaps you should re-read the installation instructions?");
-        error = GENERIC_EXIT_SETUP_ERROR;
-        return false;
+        return true;
     }
 
     query.prepare("SELECT sourceid,name FROM videosource ORDER BY sourceid;");
