diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Movie/tmdb3.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Movie/tmdb3.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Movie/tmdb3.py	2018-10-12 10:51:33.570000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Movie/tmdb3.py	2018-10-12 10:51:36.790000000 +0200
@@ -26,7 +26,8 @@
 #       resolved upstream.
 # 0.3.7 Add handling for TMDB site returning insufficient results from a
 #       query
-
+import warnings
+warnings.filterwarnings("ignore")
 from optparse import OptionParser
 import sys
 import re
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/discid/disc.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/discid/disc.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/discid/disc.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/discid/disc.py	2018-10-12 10:51:36.790000000 +0200
@@ -25,6 +25,8 @@
 from discid.util import _encode, _decode, _sectors_to_seconds
 from discid.track import Track
 
+import warnings
+warnings.filterwarnings("ignore")
 
 # our implemented of libdiscid's enum discid_feature
 _FEATURE_MAPPING = {"read": 1 << 0, "mcn": 1 << 1, "isrc": 1 << 2}
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/alsong.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/alsong.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/alsong.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/alsong.py	2018-10-12 10:51:36.790000000 +0200
@@ -12,6 +12,9 @@
 from optparse import OptionParser
 from common import utilities
 
+import warnings
+warnings.filterwarnings("ignore")
+
 __author__      = "Paul Harrison and 'driip'"
 __title__       = "Alsong"
 __description__ = "Search http://lyrics.alsong.co.kr"
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/baidu.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/baidu.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/baidu.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/baidu.py	2018-10-12 10:51:36.790000000 +0200
@@ -14,6 +14,9 @@
 from optparse import OptionParser
 from common import utilities
 
+import warnings
+warnings.filterwarnings("ignore")
+
 __author__      = "Paul Harrison and 'ronie'"
 __title__       = "Baidu"
 __description__ = "Search http://www.baidu.com for lyrics"
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/darklyrics.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/darklyrics.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/darklyrics.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/darklyrics.py	2018-10-12 10:51:36.790000000 +0200
@@ -13,6 +13,9 @@
 from optparse import OptionParser
 from common import utilities
 
+import warnings
+warnings.filterwarnings("ignore")
+
 __author__      = "Paul Harrison and smory'"
 __title__       = "DarkLyrics"
 __description__ = "Search http://www.darklyrics.com/ - the largest metal lyrics archive on the Web"
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/embedlrc.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/embedlrc.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/embedlrc.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/embedlrc.py	2018-10-12 10:51:36.790000000 +0200
@@ -10,6 +10,9 @@
 from optparse import OptionParser
 from common import utilities
 
+import warnings
+warnings.filterwarnings("ignore")
+
 __author__      = "Paul Harrison and 'ronin'"
 __title__       = "EmbeddedLyrics"
 __description__ = "Search tracks tag for embedded lyrics"
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/filelyrics.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/filelyrics.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/filelyrics.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/filelyrics.py	2018-10-12 10:51:36.790000000 +0200
@@ -10,6 +10,9 @@
 from optparse import OptionParser
 from common import *
 
+import warnings
+warnings.filterwarnings("ignore")
+
 __author__      = "Paul Harrison"
 __title__       = "FileLyrics"
 __description__ = "Search the same directory as the track for lyrics"
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/genius.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/genius.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/genius.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/genius.py	2018-10-12 10:51:36.793333333 +0200
@@ -17,6 +17,9 @@
 from optparse import OptionParser
 from common import utilities
 
+import warnings
+warnings.filterwarnings("ignore")
+
 if sys.version_info < (2, 7):
     import simplejson
 else:
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/gomaudio.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/gomaudio.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/gomaudio.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/gomaudio.py	2018-10-12 10:51:36.793333333 +0200
@@ -16,6 +16,9 @@
 from common import utilities
 from common import audiofile
 
+import warnings
+warnings.filterwarnings("ignore")
+
 __author__      = "Paul Harrison and edge'"
 __title__       = "GomAudio"
 __description__ = "Search http://newlyrics.gomtv.com for lyrics"
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/letssingit.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/letssingit.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/letssingit.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/letssingit.py	2018-10-12 10:51:36.793333333 +0200
@@ -15,6 +15,9 @@
 from optparse import OptionParser
 from common import utilities
 
+import warnings
+warnings.filterwarnings("ignore")
+
 __author__      = "Paul Harrison and 'ronie'"
 __title__       = "LetsSingIt"
 __description__ = "Search https://www.letssingit.com/ for lyrics"
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/lyricscom.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/lyricscom.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/lyricscom.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/lyricscom.py	2018-10-12 10:51:36.793333333 +0200
@@ -14,6 +14,9 @@
 from optparse import OptionParser
 from common import utilities
 
+import warnings
+warnings.filterwarnings("ignore")
+
 __author__      = "Paul Harrison and 'ronie'"
 __title__       = "Lyrics.Com"
 __description__ = "Search http://www.lyrics.com for lyrics"
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/lyricsmode.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/lyricsmode.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/lyricsmode.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/lyricsmode.py	2018-10-12 10:51:36.793333333 +0200
@@ -5,6 +5,9 @@
 from optparse import OptionParser
 from common import utilities
 
+import warnings
+warnings.filterwarnings("ignore")
+
 __author__      = "Paul Harrison and ronie'"
 __title__       = "LyricsMode"
 __description__ = "Search http://www.lyricsmode.com for lyrics"
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/lyricswiki.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/lyricswiki.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/lyricswiki.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/lyricswiki.py	2018-10-12 10:51:36.793333333 +0200
@@ -2,6 +2,9 @@
 import sys, re, urllib2, socket, HTMLParser
 from optparse import OptionParser
 
+import warnings
+warnings.filterwarnings("ignore")
+
 if sys.version_info < (2, 7):
     import simplejson
 else:
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/ttplayer.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/ttplayer.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/lyrics/ttplayer.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/lyrics/ttplayer.py	2018-10-12 10:51:36.793333333 +0200
@@ -16,6 +16,9 @@
 from optparse import OptionParser
 from common import utilities
 
+import warnings
+warnings.filterwarnings("ignore")
+
 __author__      = "Paul Harrison and 'taxigps'"
 __title__       = "TTPlayer"
 __description__ = "Search http://lrcct2.ttplayer.com for lyrics"
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/musicbrainzngs/musicbrainz.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/musicbrainzngs/musicbrainz.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Music/musicbrainzngs/musicbrainz.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Music/musicbrainzngs/musicbrainz.py	2018-10-12 10:51:36.793333333 +0200
@@ -15,7 +15,7 @@
 import xml.etree.ElementTree as etree
 from xml.parsers import expat
 from warnings import warn
-
+warnings.filterwarnings("ignore")
 from musicbrainzngs import mbxml
 from musicbrainzngs import util
 from musicbrainzngs import compat
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Television/ttvdb.py mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Television/ttvdb.py
--- mythtv-master-20181012-g4651002-old/mythtv/programs/scripts/metadata/Television/ttvdb.py	2018-10-12 10:51:33.566666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/scripts/metadata/Television/ttvdb.py	2018-10-12 10:51:36.793333333 +0200
@@ -1226,6 +1226,8 @@
 
 # System modules
 import sys, os, re
+import warnings
+warnings.filterwarnings("ignore")
 from optparse import OptionParser
 from copy import deepcopy
 # shlex for doctest
