diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythbase/threadedfilewriter.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythbase/threadedfilewriter.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythbase/threadedfilewriter.cpp	2018-10-12 10:51:36.540000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythbase/threadedfilewriter.cpp	2018-10-12 10:51:39.163333333 +0200
@@ -209,7 +209,7 @@
  *  \param data  pointer to data to write to disk
  *  \param count size of data in bytes
  */
-int ThreadedFileWriter::Write(const void *data, uint count)
+uint ThreadedFileWriter::Write(const void *data, uint count)
 {
     if (count == 0)
         return 0;
@@ -217,7 +217,7 @@
     QMutexLocker locker(&buflock);
 
     if (ignore_writes)
-        return -1;
+        return count;
 
     uint written    = 0;
     uint left       = count;
@@ -238,7 +238,7 @@
                     "\n\t\t\tis insufficient to deal with the number of on-going "
                     "\n\t\t\trecordings, or you have a disk failure.");
                 ignore_writes = true;
-                return -1;
+                return count;
             }
             if (!m_warned)
             {
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythbase/threadedfilewriter.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythbase/threadedfilewriter.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythbase/threadedfilewriter.h	2018-10-12 10:51:36.540000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythbase/threadedfilewriter.h	2018-10-12 10:51:39.163333333 +0200
@@ -50,14 +50,13 @@
     bool ReOpen(QString newFilename = "");
 
     long long Seek(long long pos, int whence);
-    int Write(const void *data, uint count);
+    uint Write(const void *data, uint count);
 
     void SetWriteBufferMinWriteSize(uint newMinSize = kMinWriteSize);
 
     void Sync(void);
     void Flush(void);
     bool SetBlocking(bool block = true);
-    bool WritesFailing(void) const { return ignore_writes; }
 
   protected:
     void DiskLoop(void);
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2018-10-12 10:51:36.553333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2018-10-12 10:51:39.163333333 +0200
@@ -338,15 +338,8 @@
         }
     }
 
-    if (ringBuffer && ringBuffer->Write(tspacket.data(), TSPacket::kSize) < 0 &&
-        curRecording && curRecording->GetRecordingStatus() != RecStatus::Failing)
-    {
-        LOG(VB_GENERAL, LOG_INFO, LOC +
-            QString("BufferedWrite: Writes are failing, "
-                    "setting status to %1")
-            .arg(RecStatus::toString(RecStatus::Failing, kSingleRecord)));
-        SetRecordingStatus(RecStatus::Failing, __FILE__, __LINE__);
-    }
+    if (ringBuffer)
+        ringBuffer->Write(tspacket.data(), TSPacket::kSize);
 }
 
 enum { kExtractPTS, kExtractDTS };
