diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythbase/threadedfilewriter.cpp mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythbase/threadedfilewriter.cpp
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythbase/threadedfilewriter.cpp	2019-03-29 08:40:03.533333334 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythbase/threadedfilewriter.cpp	2019-03-29 08:40:05.203333334 +0100
@@ -187,7 +187,7 @@
  *  \param data  pointer to data to write to disk
  *  \param count size of data in bytes
  */
-int ThreadedFileWriter::Write(const void *data, uint count)
+uint ThreadedFileWriter::Write(const void *data, uint count)
 {
     if (count == 0)
         return 0;
@@ -195,7 +195,7 @@
     QMutexLocker locker(&m_buflock);
 
     if (m_ignore_writes)
-        return -1;
+        return count;
 
     uint written    = 0;
     uint left       = count;
@@ -216,7 +216,7 @@
                     "\n\t\t\tis insufficient to deal with the number of on-going "
                     "\n\t\t\trecordings, or you have a disk failure.");
                 m_ignore_writes = true;
-                return -1;
+                return count;
             }
             if (!m_warned)
             {
diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythbase/threadedfilewriter.h mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythbase/threadedfilewriter.h
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythbase/threadedfilewriter.h	2019-03-29 08:40:03.533333334 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythbase/threadedfilewriter.h	2019-03-29 08:40:05.203333334 +0100
@@ -54,14 +54,13 @@
     bool ReOpen(QString newFilename = "");
 
     long long Seek(long long pos, int whence);
-    int Write(const void *data, uint count);
+    uint Write(const void *data, uint count);
 
     void SetWriteBufferMinWriteSize(uint newMinSize = kMinWriteSize);
 
     void Sync(void);
     void Flush(void);
     bool SetBlocking(bool block = true);
-    bool WritesFailing(void) const { return m_ignore_writes; }
 
   protected:
     void DiskLoop(void);
diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2019-03-29 08:40:03.513333334 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2019-03-29 08:40:05.203333334 +0100
@@ -292,15 +292,8 @@
         }
     }
 
-    if (m_ringBuffer && m_ringBuffer->Write(tspacket.data(), TSPacket::kSize) < 0 &&
-        m_curRecording && m_curRecording->GetRecordingStatus() != RecStatus::Failing)
-    {
-        LOG(VB_GENERAL, LOG_INFO, LOC +
-            QString("BufferedWrite: Writes are failing, "
-                    "setting status to %1")
-            .arg(RecStatus::toString(RecStatus::Failing, kSingleRecord)));
-        SetRecordingStatus(RecStatus::Failing, __FILE__, __LINE__);
-    }
+    if (m_ringBuffer)
+        m_ringBuffer->Write(tspacket.data(), TSPacket::kSize);
 }
 
 enum { kExtractPTS, kExtractDTS };
