diff -Naur mythtv-master-20160412-gea5fdd4-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-master-20160412-gea5fdd4-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-master-20160412-gea5fdd4-old/mythtv/programs/mythbackend/mainserver.cpp	2016-04-12 11:05:12.000000000 +0200
+++ mythtv-master-20160412-gea5fdd4-new/mythtv/programs/mythbackend/mainserver.cpp	2016-04-12 11:23:58.386666666 +0200
@@ -1941,6 +1941,7 @@
             if (!controlSocketList.remove(socket))
                 return; // socket was disconnected
             ft = new FileTransfer(filename, socket, writemode);
+            fileTransferList.push_back(ft);
         }
         else
         {
@@ -1948,6 +1949,7 @@
             if (!controlSocketList.remove(socket))
                 return; // socket was disconnected
             ft = new FileTransfer(filename, socket, usereadahead, timeout_ms);
+            fileTransferList.push_back(ft);
         }
 
         if (!ft->isOpen())
@@ -1961,7 +1963,6 @@
             return;
         }
         ft->IncrRef();
-        fileTransferList.push_back(ft);
 
         retlist << QString::number(socket->GetSocketDescriptor());
         retlist << QString::number(ft->GetFileSize());
