diff -Naur mythtv-master-20181019-g402f35a-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-master-20181019-g402f35a-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-master-20181019-g402f35a-old/mythtv/programs/mythbackend/mainserver.cpp	2018-10-20 14:56:38.866666667 +0200
+++ mythtv-master-20181019-g402f35a-new/mythtv/programs/mythbackend/mainserver.cpp	2018-10-20 14:56:41.020000001 +0200
@@ -2016,6 +2016,7 @@
             if (!controlSocketList.remove(socket))
                 return; // socket was disconnected
             ft = new FileTransfer(filename, socket, writemode);
+            fileTransferList.push_back(ft);
         }
         else
         {
@@ -2023,6 +2024,7 @@
             if (!controlSocketList.remove(socket))
                 return; // socket was disconnected
             ft = new FileTransfer(filename, socket, usereadahead, timeout_ms);
+            fileTransferList.push_back(ft);
         }
 
         if (!ft->isOpen())
@@ -2036,7 +2038,6 @@
             return;
         }
         ft->IncrRef();
-        fileTransferList.push_back(ft);
 
         retlist << QString::number(socket->GetSocketDescriptor());
         retlist << QString::number(ft->GetFileSize());
