diff -Naur mythtv-master-20190320-gb121bce-old/mythtv/libs/libmythmetadata/metadatagrabber.cpp mythtv-master-20190320-gb121bce-new/mythtv/libs/libmythmetadata/metadatagrabber.cpp
--- mythtv-master-20190320-gb121bce-old/mythtv/libs/libmythmetadata/metadatagrabber.cpp	2019-03-20 22:45:18.323333333 +0100
+++ mythtv-master-20190320-gb121bce-new/mythtv/libs/libmythmetadata/metadatagrabber.cpp	2019-03-20 22:45:19.526666667 +0100
@@ -425,14 +425,16 @@
 MetadataLookupList MetaGrabberScript::RunGrabber(const QStringList &args,
                         MetadataLookup *lookup, bool passseas)
 {
+    int GrabberRunTimeout  = gCoreContext->GetNumSetting("GrabberRunTimeout", 60);
+
     MythSystemLegacy grabber(m_fullcommand, args, kMSStdOut);
     MetadataLookupList list;
 
-    LOG(VB_GENERAL, LOG_INFO, QString("Running Grabber: %1 %2")
-        .arg(m_fullcommand).arg(args.join(" ")));
+    LOG(VB_GENERAL, LOG_INFO, QString("Running Grabber: %1 %2 with timeout %3 sec")
+        .arg(m_fullcommand).arg(args.join(" ")).arg(GrabberRunTimeout));
 
     grabber.Run();
-    if (grabber.Wait(60) != GENERIC_EXIT_OK)
+    if (grabber.Wait(GrabberRunTimeout) != GENERIC_EXIT_OK)
         return list;
 
     QByteArray result = grabber.ReadAll();
