diff -Naur mythtv-master-20190607-gee89397-old/mythtv/libs/libmythmetadata/metadatagrabber.cpp mythtv-master-20190607-gee89397-new/mythtv/libs/libmythmetadata/metadatagrabber.cpp
--- mythtv-master-20190607-gee89397-old/mythtv/libs/libmythmetadata/metadatagrabber.cpp	2019-06-07 16:33:28.000000000 +0200
+++ mythtv-master-20190607-gee89397-new/mythtv/libs/libmythmetadata/metadatagrabber.cpp	2019-06-13 21:05:47.559999962 +0200
@@ -423,14 +423,16 @@
 MetadataLookupList MetaGrabberScript::RunGrabber(const QStringList &args,
                         MetadataLookup *lookup, bool passseas)
 {
+    int GrabberRunTimeout  = gCoreContext->GetNumSetting("GrabberRunTimeout", 60);
+
     MythSystemLegacy grabber(m_fullcommand, args, kMSStdOut);
     MetadataLookupList list;
 
-    LOG(VB_GENERAL, LOG_INFO, QString("Running Grabber: %1 %2")
-        .arg(m_fullcommand).arg(args.join(" ")));
+    LOG(VB_GENERAL, LOG_INFO, QString("Running Grabber: %1 %2 with timeout %3 sec")
+        .arg(m_fullcommand).arg(args.join(" ")).arg(GrabberRunTimeout));
 
     grabber.Run();
-    if (grabber.Wait(60) != GENERIC_EXIT_OK)
+    if (grabber.Wait(GrabberRunTimeout) != GENERIC_EXIT_OK)
         return list;
 
     QByteArray result = grabber.ReadAll();
diff -Naur mythtv-master-20190607-gee89397-old/mythtv/programs/scripts/metadata/Movie/tmdb3.py mythtv-master-20190607-gee89397-new/mythtv/programs/scripts/metadata/Movie/tmdb3.py
--- mythtv-master-20190607-gee89397-old/mythtv/programs/scripts/metadata/Movie/tmdb3.py	2019-06-13 21:03:28.656666667 +0200
+++ mythtv-master-20190607-gee89397-new/mythtv/programs/scripts/metadata/Movie/tmdb3.py	2019-06-13 21:06:35.733333337 +0200
@@ -298,7 +298,7 @@
     opts, args = parser.parse_args()
 
     signal.signal(signal.SIGALRM, timeouthandler)
-    signal.alarm(30)
+    signal.alarm(300)
 
     if opts.version:
         buildVersion()
