diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2015-01-06 16:50:08.117459403 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2015-01-06 16:50:33.653988547 +0100
@@ -880,7 +880,7 @@
             {
                 LOG(VB_GENERAL, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
-                break;
+                //break;
             }
 
             // bounds check
