diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2015-01-06 16:50:20.994056737 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2015-01-06 16:50:45.770589951 +0100
@@ -297,7 +297,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
