diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-master-20181012-g4651002-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/programs/mythbackend/mainserver.cpp	2018-10-12 10:51:41.930000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/mythbackend/mainserver.cpp	2018-10-12 10:51:44.999999997 +0200
@@ -31,7 +31,6 @@
 #include <QFile>
 #include <QDir>
 #include <QWaitCondition>
-#include <QWriteLocker>
 #include <QRegExp>
 #include <QEvent>
 #include <QUrl>
@@ -469,8 +468,6 @@
     PlaybackSock *pbs = GetPlaybackBySock(sock);
     if (pbs)
         pbs->IncrRef();
-
-    bool bIsControl = (pbs) ? false : controlSocketList.contains(sock);
     sockListLock.unlock();
 
     QStringList listline;
@@ -484,11 +481,6 @@
         }
         pbs->DecrRef();
     }
-    else if (!bIsControl)
-    {
-        // The socket has been disconnected
-        return;
-    }
     else if (!sock->ReadStringList(listline) || listline.empty())
     {
         LOG(VB_GENERAL, LOG_INFO, LOC + "No data in ProcessRequestWork()");
@@ -1752,31 +1744,26 @@
             socket->WriteStringList(errlist);
             return;
         }
-
         // Monitor connections are same as Playback but they don't
         // block shutdowns. See the Scheduler event loop for more.
 
         PlaybackSockEventsMode eventsMode =
             (PlaybackSockEventsMode)commands[3].toInt();
-
-        QWriteLocker lock(&sockListLock);
-        if (!controlSocketList.remove(socket))
-            return; // socket was disconnected
+        LOG(VB_NETWORK, LOG_INFO, LOC + QString("MainServer::ANN %1")
+                                            .arg(commands[1]));
+        LOG(VB_NETWORK, LOG_INFO, LOC +
+            QString("adding: %1 as a client (events: %2)")
+                .arg(commands[2]).arg(eventsMode));
         PlaybackSock *pbs = new PlaybackSock(this, socket, commands[2],
                                              eventsMode);
-        playbackList.push_back(pbs);
-        lock.unlock();
-
-        LOG(VB_GENERAL, LOG_INFO, LOC + QString("MainServer::ANN %1")
-                                      .arg(commands[1]));
-        LOG(VB_GENERAL, LOG_INFO, LOC +
-            QString("adding: %1(%2) as a client (events: %3)")
-                                      .arg(commands[2])
-                                      .arg(quintptr(socket),0,16)
-                                      .arg(eventsMode));
         pbs->setBlockShutdown((commands[1] == "Playback") ||
                               (commands[1] == "Frontend"));
 
+        sockListLock.lockForWrite();
+        controlSocketList.remove(socket);
+        playbackList.push_back(pbs);
+        sockListLock.unlock();
+
         if (commands[1] == "Frontend")
         {
             pbs->SetAsFrontend();
@@ -1807,15 +1794,14 @@
             return;
         }
 
-        QWriteLocker lock(&sockListLock);
-        if (!controlSocketList.remove(socket))
-            return; // socket was disconnected
         PlaybackSock *pbs = new PlaybackSock(this, socket, commands[2],
                                               kPBSEvents_Normal);
         pbs->setAsMediaServer();
         pbs->setBlockShutdown(false);
+        sockListLock.lockForWrite();
+        controlSocketList.remove(socket);
         playbackList.push_back(pbs);
-        lock.unlock();
+        sockListLock.unlock();
 
         gCoreContext->SendSystemEvent(
             QString("CLIENT_CONNECTED HOSTNAME %1").arg(commands[2]));
@@ -1832,17 +1818,11 @@
             return;
         }
 
-        QWriteLocker lock(&sockListLock);
-        if (!controlSocketList.remove(socket))
-            return; // socket was disconnected
-        PlaybackSock *pbs = new PlaybackSock(this, socket, commands[2],
-                                             kPBSEvents_None);
-        playbackList.push_back(pbs);
-        lock.unlock();
-
         LOG(VB_GENERAL, LOG_INFO, LOC +
             QString("adding: %1 as a slave backend server")
                                .arg(commands[2]));
+        PlaybackSock *pbs = new PlaybackSock(this, socket, commands[2],
+                                             kPBSEvents_None);
         pbs->setAsSlaveBackend();
         pbs->setIP(commands[3]);
 
@@ -1888,6 +1868,11 @@
 
         pbs->setBlockShutdown(false);
 
+        sockListLock.lockForWrite();
+        controlSocketList.remove(socket);
+        playbackList.push_back(pbs);
+        sockListLock.unlock();
+
         autoexpireUpdateTimer->start(1000);
 
         gCoreContext->SendSystemEvent(
@@ -2012,19 +1997,11 @@
                     return;
                 }
             }
-            QWriteLocker lock(&sockListLock);
-            if (!controlSocketList.remove(socket))
-                return; // socket was disconnected
             ft = new FileTransfer(filename, socket, writemode);
-            fileTransferList.push_back(ft);
         }
         else
         {
-            QWriteLocker lock(&sockListLock);
-            if (!controlSocketList.remove(socket))
-                return; // socket was disconnected
             ft = new FileTransfer(filename, socket, usereadahead, timeout_ms);
-            fileTransferList.push_back(ft);
         }
 
         if (!ft->isOpen())
@@ -2038,6 +2015,10 @@
             return;
         }
         ft->IncrRef();
+        sockListLock.lockForWrite();
+        controlSocketList.remove(socket);
+        fileTransferList.push_back(ft);
+        sockListLock.unlock();
 
         retlist << QString::number(socket->GetSocketDescriptor());
         retlist << QString::number(ft->GetFileSize());
@@ -7776,7 +7757,7 @@
     DeferredDeleteStruct dds = deferredDeleteList.front();
     while (dds.ts.secsTo(MythDate::current()) > 30)
     {
-        dds.sock->DecrRef();
+        delete dds.sock;
         deferredDeleteList.pop_front();
         if (deferredDeleteList.empty())
             return;
@@ -7901,10 +7882,6 @@
                 }
             }
 
-            LOG(VB_GENERAL, LOG_INFO, QString("%1 sock(%2) '%3' disconnected")
-                .arg(pbs->getBlockShutdown() ? "Playback" : "Monitor")
-                .arg(quintptr(socket),0,16)
-                .arg(pbs->getHostname()) );
             pbs->SetDisconnected();
             playbackList.erase(it);
 
@@ -7941,8 +7918,6 @@
         MythSocket *sock = (*ft)->getSocket();
         if (sock == socket)
         {
-            LOG(VB_GENERAL, LOG_INFO, QString("FileTransfer sock(%1) disconnected")
-                .arg(quintptr(socket),0,16) );
             (*ft)->DecrRef();
             fileTransferList.erase(ft);
             sockListLock.unlock();
@@ -7953,8 +7928,6 @@
     QSet<MythSocket*>::iterator cs = controlSocketList.find(socket);
     if (cs != controlSocketList.end())
     {
-        LOG(VB_GENERAL, LOG_INFO, QString("Control sock(%1) disconnected")
-            .arg(quintptr(socket),0,16) );
         (*cs)->DecrRef();
         controlSocketList.erase(cs);
         sockListLock.unlock();
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/mythbackend/playbacksock.h mythtv-master-20181012-g4651002-new/mythtv/programs/mythbackend/playbacksock.h
--- mythtv-master-20181012-g4651002-old/mythtv/programs/mythbackend/playbacksock.h	2018-10-12 10:51:41.686666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/mythbackend/playbacksock.h	2018-10-12 10:51:45.003333330 +0200
@@ -30,6 +30,7 @@
   public:
     PlaybackSock(MainServer *parent, MythSocket *lsock,
                  QString lhostname, PlaybackSockEventsMode eventsMode);
+    virtual ~PlaybackSock();
 
     void SetDisconnected(void) { disconnected = true; }
     bool IsDisconnected(void) const { return disconnected; }
@@ -108,10 +109,6 @@
 
     bool AddChildInput(uint childid);
 
-    // Enforce reference counting
-  protected:
-    virtual ~PlaybackSock();
-
   private:
     bool SendReceiveStringList(QStringList &strlist, uint min_reply_length = 0);
 
