diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-05-07 12:45:45.820000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-05-07 12:45:47.170000000 +0200
@@ -2271,12 +2271,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
