diff -Naur mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-09-21 09:17:44.150000001 +0200
+++ mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-09-21 09:17:46.710000002 +0200
@@ -2235,12 +2235,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
