diff -Naur mythtv-master-20210424-g69d12b67b6-old/mythtv/libs/libmythui/mythuiscreenbounds.cpp mythtv-master-20210424-g69d12b67b6-new/mythtv/libs/libmythui/mythuiscreenbounds.cpp
--- mythtv-master-20210424-g69d12b67b6-old/mythtv/libs/libmythui/mythuiscreenbounds.cpp	2021-04-24 16:55:48.899999039 +0200
+++ mythtv-master-20210424-g69d12b67b6-new/mythtv/libs/libmythui/mythuiscreenbounds.cpp	2021-04-24 16:55:54.719999039 +0200
@@ -223,12 +223,12 @@
 
 int MythUIScreenBounds::NormX(int X) const
 {
-    return qRound(X * m_screenHorizScale);
+    return (int)(X * m_screenHorizScale);
 }
 
 int MythUIScreenBounds::NormY(int Y) const
 {
-    return qRound(Y * m_screenVertScale);
+    return (int)(Y * m_screenVertScale);
 }
 
 void MythUIScreenBounds::GetScalingFactors(float& Horizontal, float& Vertical) const
