diff -Naur mythtv-master-20190408-g038e171-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20190408-g038e171-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20190408-g038e171-old/mythtv/libs/libmythui/mythmainwindow.cpp	2019-04-08 14:44:58.550000000 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/libs/libmythui/mythmainwindow.cpp	2019-04-08 14:45:00.183333334 +0200
@@ -2898,12 +2898,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
