diff -Naur mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-04-09 12:25:00.847590199 +0200
+++ mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-04-09 12:25:02.504256805 +0200
@@ -2727,12 +2727,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
