diff -Naur mythtv-master-20200530-ga180830-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20200530-ga180830-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20200530-ga180830-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-05-30 13:00:16.646666667 +0200
+++ mythtv-master-20200530-ga180830-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-05-30 13:00:18.116666667 +0200
@@ -2271,12 +2271,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
