diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythui/mythuiscreenbounds.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythui/mythuiscreenbounds.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythui/mythuiscreenbounds.cpp	2021-03-17 21:06:40.469999834 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythui/mythuiscreenbounds.cpp	2021-03-17 21:06:43.239999834 +0100
@@ -223,12 +223,12 @@
 
 int MythUIScreenBounds::NormX(int X) const
 {
-    return qRound(X * m_screenHorizScale);
+    return (int)(X * m_screenHorizScale);
 }
 
 int MythUIScreenBounds::NormY(int Y) const
 {
-    return qRound(Y * m_screenVertScale);
+    return (int)(Y * m_screenVertScale);
 }
 
 void MythUIScreenBounds::GetScalingFactors(float& Horizontal, float& Vertical) const
