diff -Naur mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-09-25 09:20:42.923333334 +0200
+++ mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-09-25 09:20:44.853333334 +0200
@@ -2198,12 +2198,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
