diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythui/mythmainwindow.cpp	2019-03-29 08:40:06.443333334 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythui/mythmainwindow.cpp	2019-03-29 08:40:07.840000001 +0100
@@ -2897,12 +2897,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
