diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-09-04 11:35:24.070000000 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-09-04 11:35:25.413333333 +0200
@@ -2235,12 +2235,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
