diff -Naur mythtv-master-20200418-g8760389-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20200418-g8760389-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20200418-g8760389-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-04-18 10:53:36.973333332 +0200
+++ mythtv-master-20200418-g8760389-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-04-18 10:53:44.209999999 +0200
@@ -2265,12 +2265,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
