diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-09-15 09:22:25.756666668 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-09-15 09:22:27.936666668 +0200
@@ -2235,12 +2235,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
