diff -Naur mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythui/mythmainwindow.cpp	2018-10-20 14:56:43.430000001 +0200
+++ mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythui/mythmainwindow.cpp	2018-10-20 14:56:46.136666667 +0200
@@ -2935,12 +2935,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return (int)(qRound(x * d->wmult));
+    return (int)(x * d->wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return (int)(qRound(y * d->hmult));
+    return (int)(y * d->hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
