diff -Naur mythtv-master-20190320-gb121bce-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20190320-gb121bce-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20190320-gb121bce-old/mythtv/libs/libmythui/mythmainwindow.cpp	2019-03-20 22:45:19.356666667 +0100
+++ mythtv-master-20190320-gb121bce-new/mythtv/libs/libmythui/mythmainwindow.cpp	2019-03-20 22:45:20.866666667 +0100
@@ -2897,12 +2897,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
