diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/channelscan/channelscan_sm.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/channelscan/channelscan_sm.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/channelscan/channelscan_sm.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/channelscan/channelscan_sm.cpp	2018-10-12 10:51:47.896666666 +0200
@@ -144,7 +144,7 @@
     : // Set in constructor
       m_scanMonitor(_scan_monitor),
       m_channel(_channel),
-      m_signalMonitor(SignalMonitor::Init(_cardtype, -1, _channel, true)),
+      m_signalMonitor(SignalMonitor::Init(_cardtype, -1, _channel)),
       m_sourceID(_sourceID),
       m_signalTimeout(signal_timeout),
       m_channelTimeout(channel_timeout),
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/analogsignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/analogsignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/analogsignalmonitor.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/analogsignalmonitor.cpp	2018-10-12 10:51:47.896666666 +0200
@@ -17,13 +17,11 @@
 #define LOC QString("AnalogSigMon[%1](%2): ") \
             .arg(capturecardnum).arg(channel->GetDevice())
 
-AnalogSignalMonitor::AnalogSignalMonitor(int db_cardnum,
-                                         V4LChannel *_channel,
-                                         bool _release_stream,
-                                         uint64_t _flags)
-    : SignalMonitor(db_cardnum, _channel, _release_stream, _flags),
-      m_usingv4l2(false), m_version(0), m_width(0), m_stable_time(2000),
-      m_lock_cnt(0), m_log_idx(40)
+AnalogSignalMonitor::AnalogSignalMonitor(
+    int db_cardnum, V4LChannel *_channel, uint64_t _flags) :
+    SignalMonitor(db_cardnum, _channel, _flags),
+    m_usingv4l2(false), m_version(0), m_width(0), m_stable_time(2000),
+    m_lock_cnt(0), m_log_idx(40)
 {
     int videofd = channel->GetFd();
     if (videofd >= 0)
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/analogsignalmonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/analogsignalmonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/analogsignalmonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/analogsignalmonitor.h	2018-10-12 10:51:47.896666666 +0200
@@ -12,9 +12,9 @@
 class AnalogSignalMonitor : public SignalMonitor
 {
   public:
-    AnalogSignalMonitor(int db_cardnum, V4LChannel *_channel,
-                        bool _release_stream,
-                        uint64_t _flags = kSigMon_WaitForSig);
+    AnalogSignalMonitor(
+        int db_cardnum, V4LChannel *_channel,
+        uint64_t _flags = kSigMon_WaitForSig);
 
     virtual void UpdateValues(void);
 
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/asisignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/asisignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/asisignalmonitor.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/asisignalmonitor.cpp	2018-10-12 10:51:47.896666666 +0200
@@ -38,12 +38,10 @@
  *  \param _channel ASIChannel for card
  *  \param _flags   Flags to start with
  */
-ASISignalMonitor::ASISignalMonitor(int db_cardnum,
-                                   ASIChannel *_channel,
-                                   bool _release_stream,
-                                   uint64_t _flags)
-    : DTVSignalMonitor(db_cardnum, _channel, _release_stream, _flags),
-      streamHandlerStarted(false), streamHandler(nullptr)
+ASISignalMonitor::ASISignalMonitor(
+    int db_cardnum, ASIChannel *_channel, uint64_t _flags) :
+    DTVSignalMonitor(db_cardnum, _channel, _flags),
+    streamHandlerStarted(false), streamHandler(nullptr)
 {
     LOG(VB_CHANNEL, LOG_INFO, LOC + "ctor");
     streamHandler = ASIStreamHandler::Get(_channel->GetDevice());
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/asisignalmonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/asisignalmonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/asisignalmonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/asisignalmonitor.h	2018-10-12 10:51:47.896666666 +0200
@@ -14,7 +14,7 @@
 {
   public:
     ASISignalMonitor(int db_cardnum, ASIChannel *_channel,
-                     bool _release_stream = true, uint64_t _flags = 0);
+                     uint64_t _flags = 0);
     virtual ~ASISignalMonitor();
 
     void Stop(void);
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/cetonsignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/cetonsignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/cetonsignalmonitor.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/cetonsignalmonitor.cpp	2018-10-12 10:51:47.896666666 +0200
@@ -30,12 +30,10 @@
  *  \param _channel CetonChannel for card
  *  \param _flags   Flags to start with
  */
-CetonSignalMonitor::CetonSignalMonitor(int db_cardnum,
-                                       CetonChannel* _channel,
-                                       bool _release_stream,
-                                       uint64_t _flags)
-    : DTVSignalMonitor(db_cardnum, _channel, _release_stream, _flags),
-      streamHandlerStarted(false), streamHandler(nullptr)
+CetonSignalMonitor::CetonSignalMonitor(
+    int db_cardnum, CetonChannel* _channel, uint64_t _flags) :
+    DTVSignalMonitor(db_cardnum, _channel, _flags),
+    streamHandlerStarted(false), streamHandler(nullptr)
 {
     LOG(VB_CHANNEL, LOG_INFO, LOC + "ctor");
 
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/cetonsignalmonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/cetonsignalmonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/cetonsignalmonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/cetonsignalmonitor.h	2018-10-12 10:51:47.896666666 +0200
@@ -19,7 +19,7 @@
 {
   public:
     CetonSignalMonitor(int db_cardnum, CetonChannel* _channel,
-                       bool _release_stream, uint64_t _flags = 0);
+                      uint64_t _flags = 0);
     virtual ~CetonSignalMonitor();
 
     void Stop(void);
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2018-10-12 10:51:47.896666666 +0200
@@ -43,9 +43,8 @@
 
 DTVSignalMonitor::DTVSignalMonitor(int db_cardnum,
                                    DTVChannel *_channel,
-                                   bool _release_stream,
                                    uint64_t wait_for_mask)
-    : SignalMonitor(db_cardnum, _channel, _release_stream, wait_for_mask),
+    : SignalMonitor(db_cardnum, _channel, wait_for_mask),
       stream_data(nullptr),
       seenPAT(QObject::tr("Seen")+" PAT", "seen_pat", 1, true, 0, 1, 0),
       seenPMT(QObject::tr("Seen")+" PMT", "seen_pmt", 1, true, 0, 1, 0),
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.h	2018-10-12 10:51:47.896666666 +0200
@@ -21,7 +21,6 @@
   public:
     DTVSignalMonitor(int db_cardnum,
                      DTVChannel *_channel,
-                     bool _release_stream,
                      uint64_t wait_for_mask);
     virtual ~DTVSignalMonitor();
 
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2018-10-12 10:51:47.896666666 +0200
@@ -43,9 +43,8 @@
  *  \param _flags   Flags to start with
  */
 DVBSignalMonitor::DVBSignalMonitor(int db_cardnum, DVBChannel* _channel,
-                                   bool _release_stream,
                                    uint64_t _flags)
-    : DTVSignalMonitor(db_cardnum, _channel, _release_stream, _flags),
+    : DTVSignalMonitor(db_cardnum, _channel, _flags),
       // This snr setup is incorrect for API 3.x but works better
       // than int16_t range in practice, however this is correct
       // for the 4.0 DVB API which uses a uint16_t for the snr
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.h	2018-10-12 10:51:47.896666666 +0200
@@ -20,7 +20,6 @@
 
   public:
     DVBSignalMonitor(int db_cardnum, DVBChannel* _channel,
-                     bool _release_stream,
                      uint64_t _flags =
                      kSigMon_WaitForSig    | kDVBSigMon_WaitForSNR |
                      kDVBSigMon_WaitForBER | kDVBSigMon_WaitForUB);
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/ExternalRecorder.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/ExternalRecorder.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/ExternalRecorder.cpp	2018-10-12 10:51:44.893333328 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/ExternalRecorder.cpp	2018-10-12 10:51:47.896666666 +0200
@@ -81,19 +81,11 @@
 
     StartNewFile();
 
-    m_stream_handler->LockReplay();
-
-    m_h264_parser.Reset();
-    _wait_for_keyframe_option = true;
-    _seen_sps = false;
-
     _stream_data->AddAVListener(this);
     _stream_data->AddWritingListener(this);
     m_stream_handler->AddListener(_stream_data, false, true);
 
-    m_stream_handler->ReplayStream();
-    m_stream_handler->UnlockReplay();
-
+    StartStreaming();
 
     while (IsRecordingRequested() && !IsErrored())
     {
@@ -197,8 +189,7 @@
     else if (IsPaused(true))
     {
         LOG(VB_RECORD, LOG_INFO, LOC + "PauseAndWait unpause");
-
-        // The SignalMonitor will StartStreaming
+        StartStreaming();
 
         if (_stream_data)
             _stream_data->Reset(_stream_data->DesiredProgram());
@@ -214,8 +205,12 @@
 
 bool ExternalRecorder::StartStreaming(void)
 {
+    m_h264_parser.Reset();
+    _wait_for_keyframe_option = true;
+    _seen_sps = false;
+
     LOG(VB_RECORD, LOG_INFO, LOC + "StartStreaming");
-    if (m_stream_handler && m_stream_handler->StartStreaming())
+    if (m_stream_handler && m_stream_handler->StartStreaming(true))
         return true;
 
     return false;
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/ExternalSignalMonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/ExternalSignalMonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/ExternalSignalMonitor.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/ExternalSignalMonitor.cpp	2018-10-12 10:51:47.896666666 +0200
@@ -36,12 +36,10 @@
  *  \param _channel ExternalChannel for card
  *  \param _flags   Flags to start with
  */
-ExternalSignalMonitor::ExternalSignalMonitor(int db_cardnum,
-                                             ExternalChannel *_channel,
-                                             bool _release_stream,
-                                             uint64_t _flags)
-    : DTVSignalMonitor(db_cardnum, _channel, _flags, _release_stream),
-      m_stream_handler(nullptr), m_stream_handler_started(false), m_lock_timeout(0)
+ExternalSignalMonitor::ExternalSignalMonitor(
+    int db_cardnum, ExternalChannel *_channel, uint64_t _flags) :
+    DTVSignalMonitor(db_cardnum, _channel, _flags),
+    m_stream_handler(nullptr), m_stream_handler_started(false), m_lock_timeout(0)
 {
     QString result;
 
@@ -71,13 +69,11 @@
     QString result;
 
     LOG(VB_CHANNEL, LOG_INFO, LOC + "Stop() -- begin");
+    m_stream_handler->StopStreaming();
 
     SignalMonitor::Stop();
-    if (release_stream && GetStreamData())
-    {
-        m_stream_handler->StopStreaming();
+    if (GetStreamData())
         m_stream_handler->RemoveListener(GetStreamData());
-    }
     m_stream_handler_started = false;
 
     LOG(VB_CHANNEL, LOG_INFO, LOC + "Stop() -- end");
@@ -147,7 +143,7 @@
         if (!m_stream_handler_started)
         {
             m_stream_handler->AddListener(GetStreamData());
-            m_stream_handler->StartStreaming();
+            m_stream_handler->StartStreaming(false);
             m_stream_handler_started = true;
         }
     }
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/ExternalSignalMonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/ExternalSignalMonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/ExternalSignalMonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/ExternalSignalMonitor.h	2018-10-12 10:51:47.896666666 +0200
@@ -16,7 +16,7 @@
 {
   public:
     ExternalSignalMonitor(int db_cardnum, ExternalChannel *_channel,
-                          bool _release_stream, uint64_t _flags = 0);
+                          uint64_t _flags = 0);
     virtual ~ExternalSignalMonitor();
 
     void Stop(void);
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/ExternalStreamHandler.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/ExternalStreamHandler.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/ExternalStreamHandler.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/ExternalStreamHandler.cpp	2018-10-12 10:51:47.896666666 +0200
@@ -632,8 +632,8 @@
             m_notify = false;
         }
 
-        while (read_len = 0, buffer.size() == PACKET_SIZE ||
-               (read_len = m_IO->Read(buffer, PACKET_SIZE - remainder, 100)) > 0)
+        while (read_len = 0, buffer.size() > 188*50 ||
+               (read_len = m_IO->Read(buffer, PACKET_SIZE, 100)) > 0)
         {
             if (m_IO->Error())
             {
@@ -650,6 +650,12 @@
             if (read_len > 0)
                 empty_cnt = 0;
 
+            if (_stream_data_list.empty())
+            {
+                LOG(VB_GENERAL, LOG_ERR, LOC + "_stream_data_list is empty");
+                continue;
+            }
+
             if (xon)
             {
                 if (timer.elapsed() >= 2000)
@@ -686,11 +692,6 @@
                 }
             }
 
-            len = remainder = buffer.size();
-
-            if (!m_stream_lock.tryLock())
-                continue;
-
             if (!_listener_lock.tryLock())
                 continue;
 
@@ -702,6 +703,8 @@
 
             _listener_lock.unlock();
 
+            len = buffer.size();
+
             if (m_replay)
             {
                 m_replay_buffer += buffer.left(len - remainder);
@@ -714,14 +717,11 @@
                 }
             }
 
-            m_stream_lock.unlock();
-
-            if (remainder == 0)
-                buffer.clear();
-            else if (len > remainder) // leftover bytes
+            if (remainder > 0 && (len > remainder)) // leftover bytes
                 buffer.remove(0, len - remainder);
+            else
+                buffer.clear();
         }
-
         if (m_IO->Error())
         {
             LOG(VB_GENERAL, LOG_ERR, LOC +
@@ -907,22 +907,30 @@
 
     if (streaming)
     {
-        return StartStreaming();
+        return StartStreaming(false);
     }
 
     return true;
 }
 
-void ExternalStreamHandler::ReplayStream(void)
+bool ExternalStreamHandler::StartStreaming(bool flush_buffer)
 {
-    if (m_replay)
-    {
-        QString    result;
+    QString result;
 
-        // Let the external app know that we could be busy for a little while
-        if (!m_poll_mode)
-            ProcessCommand(QString("XOFF"), 50, result);
+    QMutexLocker locker(&m_stream_lock);
 
+    LOG(VB_RECORD, LOG_INFO, LOC +
+        QString("StartStreaming with %1 current listeners")
+        .arg(StreamingCount()));
+
+    if (!IsAppOpen())
+    {
+        LOG(VB_GENERAL, LOG_ERR, LOC + "External Recorder not started.");
+        return false;
+    }
+
+    if (flush_buffer && m_replay)
+    {
         /* If the input is not a 'broadcast' it may only have one
          * copy of the SPS right at the beginning of the stream,
          * so make sure we don't miss it!
@@ -941,27 +949,6 @@
             .arg(m_replay_buffer.size()));
         m_replay_buffer.clear();
         m_replay = false;
-
-        // Let the external app know that we are ready
-        if (!m_poll_mode)
-            ProcessCommand(QString("XON"), 50, result);
-    }
-}
-
-bool ExternalStreamHandler::StartStreaming(void)
-{
-    QString result;
-
-    QMutexLocker locker(&m_stream_lock);
-
-    LOG(VB_RECORD, LOG_INFO, LOC +
-        QString("StartStreaming with %1 current listeners")
-        .arg(StreamingCount()));
-
-    if (!IsAppOpen())
-    {
-        LOG(VB_GENERAL, LOG_ERR, LOC + "External Recorder not started.");
-        return false;
     }
 
     if (StreamingCount() == 0)
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/ExternalStreamHandler.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/ExternalStreamHandler.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/ExternalStreamHandler.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/ExternalStreamHandler.h	2018-10-12 10:51:47.899999999 +0200
@@ -80,11 +80,7 @@
 
     bool RestartStream(void);
 
-    void LockReplay(void) { m_replay_lock.lock(); }
-    void UnlockReplay(bool enable_replay = false)
-        { m_replay = enable_replay; m_replay_lock.unlock(); }
-    void ReplayStream(void);
-    bool StartStreaming(void);
+    bool StartStreaming(bool flush_buffer);
     bool StopStreaming(void);
 
     bool CheckForError(void);
@@ -121,7 +117,6 @@
 
     QAtomicInt    m_streaming_cnt;
     QMutex        m_stream_lock;
-    QMutex        m_replay_lock;
 };
 
 #endif // _ExternalSTREAMHANDLER_H_
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/firewiresignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/firewiresignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/firewiresignalmonitor.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/firewiresignalmonitor.cpp	2018-10-12 10:51:47.899999999 +0200
@@ -44,16 +44,16 @@
  *  \param _channel FirewireChannel for card
  *  \param _flags   Flags to start with
  */
-FirewireSignalMonitor::FirewireSignalMonitor(int db_cardnum,
-                                             FirewireChannel *_channel,
-                                             bool _release_stream,
-                                             uint64_t _flags)
-    : DTVSignalMonitor(db_cardnum, _channel, _release_stream, _flags),
-      dtvMonitorRunning(false),
-      tableMonitorThread(nullptr),
-      stb_needs_retune(true),
-      stb_needs_to_wait_for_pat(false),
-      stb_needs_to_wait_for_power(false)
+FirewireSignalMonitor::FirewireSignalMonitor(
+    int db_cardnum,
+    FirewireChannel *_channel,
+    uint64_t _flags) :
+    DTVSignalMonitor(db_cardnum, _channel, _flags),
+    dtvMonitorRunning(false),
+    tableMonitorThread(nullptr),
+    stb_needs_retune(true),
+    stb_needs_to_wait_for_pat(false),
+    stb_needs_to_wait_for_power(false)
 {
     LOG(VB_CHANNEL, LOG_INFO, LOC + "ctor");
 
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/firewiresignalmonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/firewiresignalmonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/firewiresignalmonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/firewiresignalmonitor.h	2018-10-12 10:51:47.899999999 +0200
@@ -36,7 +36,6 @@
     friend class FirewireTableMonitorThread;
   public:
     FirewireSignalMonitor(int db_cardnum, FirewireChannel *_channel,
-                          bool _release_stream,
                           uint64_t _flags = kFWSigMon_WaitForPower);
 
     virtual void HandlePAT(const ProgramAssociationTable*);
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/hdhrsignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/hdhrsignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/hdhrsignalmonitor.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/hdhrsignalmonitor.cpp	2018-10-12 10:51:47.899999999 +0200
@@ -38,12 +38,10 @@
  *  \param _channel HDHRChannel for card
  *  \param _flags   Flags to start with
  */
-HDHRSignalMonitor::HDHRSignalMonitor(int db_cardnum,
-                                     HDHRChannel* _channel,
-                                     bool _release_stream,
-                                     uint64_t _flags)
-    : DTVSignalMonitor(db_cardnum, _channel, _release_stream, _flags),
-      streamHandlerStarted(false), streamHandler(nullptr)
+HDHRSignalMonitor::HDHRSignalMonitor(
+    int db_cardnum, HDHRChannel* _channel, uint64_t _flags) :
+    DTVSignalMonitor(db_cardnum, _channel, _flags),
+    streamHandlerStarted(false), streamHandler(nullptr)
 {
     LOG(VB_CHANNEL, LOG_INFO, LOC + "ctor");
 
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/hdhrsignalmonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/hdhrsignalmonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/hdhrsignalmonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/hdhrsignalmonitor.h	2018-10-12 10:51:47.899999999 +0200
@@ -14,7 +14,7 @@
 {
   public:
     HDHRSignalMonitor(int db_cardnum, HDHRChannel* _channel,
-                      bool _release_stream, uint64_t _flags = 0);
+                      uint64_t _flags = 0);
     virtual ~HDHRSignalMonitor();
 
     void Stop(void);
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/iptvsignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/iptvsignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/iptvsignalmonitor.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/iptvsignalmonitor.cpp	2018-10-12 10:51:47.899999999 +0200
@@ -24,10 +24,9 @@
  */
 IPTVSignalMonitor::IPTVSignalMonitor(int db_cardnum,
                                      IPTVChannel *_channel,
-                                     bool _release_stream,
-                                     uint64_t _flags)
-    : DTVSignalMonitor(db_cardnum, _channel, _release_stream, _flags),
-      m_streamHandlerStarted(false), m_locked(false)
+                                     uint64_t _flags) :
+    DTVSignalMonitor(db_cardnum, _channel, _flags),
+    m_streamHandlerStarted(false), m_locked(false)
 {
     LOG(VB_CHANNEL, LOG_INFO, LOC + "ctor");
     signalLock.SetValue(0);
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/iptvsignalmonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/iptvsignalmonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/iptvsignalmonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/iptvsignalmonitor.h	2018-10-12 10:51:47.899999999 +0200
@@ -13,7 +13,7 @@
     friend class IPTVTableMonitorThread;
   public:
     IPTVSignalMonitor(int db_cardnum, IPTVChannel *_channel,
-                      bool _release_stream, uint64_t _flags = 0);
+                      uint64_t _flags = 0);
     virtual ~IPTVSignalMonitor();
 
     void Stop(void);
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/scriptsignalmonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/scriptsignalmonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/scriptsignalmonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/scriptsignalmonitor.h	2018-10-12 10:51:47.899999999 +0200
@@ -10,9 +10,8 @@
 {
   public:
     ScriptSignalMonitor(int db_cardnum, ChannelBase *_channel,
-                        bool _release_stream,
                         uint64_t _flags = 0) :
-        SignalMonitor(db_cardnum, _channel, _release_stream, _flags)
+        SignalMonitor(db_cardnum, _channel, _flags)
     {
         signalLock.SetValue(true);
         signalStrength.SetValue(100);
@@ -21,7 +20,7 @@
     virtual void UpdateValues(void)
     {
         SignalMonitor::UpdateValues();
-
+        
         EmitStatus();
         if (IsAllGood())
             SendMessageAllGood();
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/signalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/signalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/signalmonitor.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/signalmonitor.cpp	2018-10-12 10:51:47.899999999 +0200
@@ -83,8 +83,7 @@
  */
 
 SignalMonitor *SignalMonitor::Init(QString cardtype, int db_cardnum,
-                                   ChannelBase *channel,
-                                   bool release_stream)
+                                   ChannelBase *channel)
 {
     (void) cardtype;
     (void) db_cardnum;
@@ -97,8 +96,7 @@
     {
         DVBChannel *dvbc = dynamic_cast<DVBChannel*>(channel);
         if (dvbc)
-            signalMonitor = new DVBSignalMonitor(db_cardnum, dvbc,
-                                                 release_stream);
+            signalMonitor = new DVBSignalMonitor(db_cardnum, dvbc);
     }
 #endif
 
@@ -107,15 +105,13 @@
     {
         V4LChannel *chan = dynamic_cast<V4LChannel*>(channel);
         if (chan)
-            signalMonitor = new AnalogSignalMonitor(db_cardnum, chan,
-                                                    release_stream);
+            signalMonitor = new AnalogSignalMonitor(db_cardnum, chan);
     }
     else if (cardtype.toUpper() == "V4L2ENC")
     {
         V4LChannel *chan = dynamic_cast<V4LChannel*>(channel);
         if (chan)
-            signalMonitor = new V4L2encSignalMonitor(db_cardnum, chan,
-                                                     release_stream);
+            signalMonitor = new V4L2encSignalMonitor(db_cardnum, chan);
     }
 #endif
 
@@ -124,8 +120,7 @@
     {
         HDHRChannel *hdhrc = dynamic_cast<HDHRChannel*>(channel);
         if (hdhrc)
-            signalMonitor = new HDHRSignalMonitor(db_cardnum, hdhrc,
-                                                  release_stream);
+            signalMonitor = new HDHRSignalMonitor(db_cardnum, hdhrc);
     }
 #endif
 
@@ -134,8 +129,7 @@
     {
         CetonChannel *cetonchan = dynamic_cast<CetonChannel*>(channel);
         if (cetonchan)
-            signalMonitor = new CetonSignalMonitor(db_cardnum, cetonchan,
-                                                   release_stream);
+            signalMonitor = new CetonSignalMonitor(db_cardnum, cetonchan);
     }
 #endif
 
@@ -144,8 +138,7 @@
     {
         IPTVChannel *fbc = dynamic_cast<IPTVChannel*>(channel);
         if (fbc)
-            signalMonitor = new IPTVSignalMonitor(db_cardnum, fbc,
-                                                  release_stream);
+            signalMonitor = new IPTVSignalMonitor(db_cardnum, fbc);
     }
 #endif
 
@@ -154,8 +147,7 @@
     {
         IPTVChannel *fbc = dynamic_cast<IPTVChannel*>(channel);
         if (fbc)
-            signalMonitor = new IPTVSignalMonitor(db_cardnum, fbc,
-                                                  release_stream);
+            signalMonitor = new IPTVSignalMonitor(db_cardnum, fbc);
     }
 #endif
 
@@ -164,8 +156,7 @@
     {
         FirewireChannel *fc = dynamic_cast<FirewireChannel*>(channel);
         if (fc)
-            signalMonitor = new FirewireSignalMonitor(db_cardnum, fc,
-                                                      release_stream);
+            signalMonitor = new FirewireSignalMonitor(db_cardnum, fc);
     }
 #endif
 
@@ -174,8 +165,7 @@
     {
         ASIChannel *fc = dynamic_cast<ASIChannel*>(channel);
         if (fc)
-            signalMonitor = new ASISignalMonitor(db_cardnum, fc,
-                                                 release_stream);
+            signalMonitor = new ASISignalMonitor(db_cardnum, fc);
     }
 #endif
 
@@ -183,14 +173,12 @@
     {
         ExternalChannel *fc = dynamic_cast<ExternalChannel*>(channel);
         if (fc)
-            signalMonitor = new ExternalSignalMonitor(db_cardnum, fc,
-                                                      release_stream);
+            signalMonitor = new ExternalSignalMonitor(db_cardnum, fc);
     }
 
     if (!signalMonitor && channel)
     {
-        signalMonitor = new ScriptSignalMonitor(db_cardnum, channel,
-                                                release_stream);
+        signalMonitor = new ScriptSignalMonitor(db_cardnum, channel);
     }
 
     if (!signalMonitor)
@@ -216,11 +204,10 @@
  *  \param wait_for_mask   SignalMonitorFlags to start with.
  */
 SignalMonitor::SignalMonitor(int _capturecardnum, ChannelBase *_channel,
-                             uint64_t wait_for_mask, bool _release_stream)
+                             uint64_t wait_for_mask)
     : MThread("SignalMonitor"),
       channel(_channel),               pParent(nullptr),
       capturecardnum(_capturecardnum), flags(wait_for_mask),
-      release_stream(_release_stream),
       update_rate(25),                 minimum_update_rate(5),
       update_done(false),              notify_frontend(true),
       tablemon(false),                 eit_scan(false),
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/signalmonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/signalmonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/signalmonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/signalmonitor.h	2018-10-12 10:51:47.899999999 +0200
@@ -38,8 +38,7 @@
     static inline bool IsRequired(const QString &cardtype);
     static inline bool IsSupported(const QString &cardtype);
     static SignalMonitor *Init(QString cardtype, int db_cardnum,
-                               ChannelBase *channel,
-                               bool release_stream);
+                               ChannelBase *channel);
     virtual ~SignalMonitor();
 
     // // // // // // // // // // // // // // // // // // // // // // // //
@@ -118,7 +117,7 @@
 
   protected:
     SignalMonitor(int _capturecardnum, ChannelBase *_channel,
-                  uint64_t wait_for_mask, bool _release_stream);
+                  uint64_t wait_for_mask);
 
     virtual void run(void);
 
@@ -200,7 +199,6 @@
     TVRec       *pParent;
     int          capturecardnum;
     volatile uint64_t flags;
-    bool         release_stream;
     int          update_rate;
     uint         minimum_update_rate;
     bool         update_done;
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/streamhandler.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/streamhandler.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/streamhandler.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/streamhandler.cpp	2018-10-12 10:51:47.899999999 +0200
@@ -105,7 +105,16 @@
         _needs_buffering      |= needs_buffering;
     }
 
-    _stream_data_list[data] = output_file;
+    StreamDataList::iterator it = _stream_data_list.find(data);
+    if (it != _stream_data_list.end())
+    {
+        LOG(VB_GENERAL, LOG_ERR, LOC + "Programmer Error, attempted "
+                "to add a listener which is already being listened to.");
+    }
+    else
+    {
+        _stream_data_list[data] = output_file;
+    }
 
     _listener_lock.unlock();
 
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/v4l2encsignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/v4l2encsignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/v4l2encsignalmonitor.cpp	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/v4l2encsignalmonitor.cpp	2018-10-12 10:51:47.899999999 +0200
@@ -35,13 +35,11 @@
  *  \param _channel V4lChannel for card
  *  \param _flags   Flags to start with
  */
-V4L2encSignalMonitor::V4L2encSignalMonitor(int db_cardnum,
-                                           V4LChannel *_channel,
-                                           bool _release_stream,
-                                           uint64_t _flags)
-    : DTVSignalMonitor(db_cardnum, _channel, _release_stream, _flags),
-      m_stream_handler(nullptr), m_isTS(false),
-      m_strength(0), m_stable_time(1500), m_width(0), m_height(0), m_lock_cnt(0)
+V4L2encSignalMonitor::V4L2encSignalMonitor(int db_cardnum, V4LChannel *_channel,
+                                     uint64_t _flags) :
+    DTVSignalMonitor(db_cardnum, _channel, _flags),
+    m_stream_handler(nullptr), m_isTS(false),
+    m_strength(0), m_stable_time(1500), m_width(0), m_height(0), m_lock_cnt(0)
 {
     LOG(VB_CHANNEL, LOG_INFO, LOC + "ctor");
 
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/v4l2encsignalmonitor.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/v4l2encsignalmonitor.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/v4l2encsignalmonitor.h	2018-10-12 10:51:44.889999995 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/v4l2encsignalmonitor.h	2018-10-12 10:51:47.899999999 +0200
@@ -17,7 +17,7 @@
 {
   public:
     V4L2encSignalMonitor(int db_cardnum, V4LChannel *_channel,
-                         bool _release_stream, uint64_t _flags = 0);
+                      uint64_t _flags = 0);
     virtual ~V4L2encSignalMonitor();
 
     void Stop(void);
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/tv_rec.cpp	2018-10-12 10:51:44.886666661 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/tv_rec.cpp	2018-10-12 10:51:47.899999999 +0200
@@ -2223,8 +2223,7 @@
     SignalMonitorValue::Init();
 
     if (SignalMonitor::IsSupported(genOpt.inputtype) && channel->Open())
-        signalMonitor = SignalMonitor::Init(genOpt.inputtype, inputid,
-                                            channel, false);
+        signalMonitor = SignalMonitor::Init(genOpt.inputtype, inputid, channel);
 
     if (signalMonitor)
     {
