diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20190216-g938606f-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmythui/mythmainwindow.cpp	2019-02-17 16:57:59.976666669 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmythui/mythmainwindow.cpp	2019-02-17 16:58:01.363333336 +0100
@@ -2903,12 +2903,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->wmult);
+    return (int)(x * d->wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->hmult);
+    return (int)(y * d->hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
