diff -Naur mythtv-master-20201003-gd5ba9b4-old/mythtv/libs/libmythui/mythuiscreenbounds.cpp mythtv-master-20201003-gd5ba9b4-new/mythtv/libs/libmythui/mythuiscreenbounds.cpp
--- mythtv-master-20201003-gd5ba9b4-old/mythtv/libs/libmythui/mythuiscreenbounds.cpp	2020-10-03 22:27:07.000000000 +0200
+++ mythtv-master-20201003-gd5ba9b4-new/mythtv/libs/libmythui/mythuiscreenbounds.cpp	2020-10-04 13:29:04.526666656 +0200
@@ -217,12 +217,12 @@
 
 int MythUIScreenBounds::NormX(int X) const
 {
-    return qRound(X * m_screenHorizScale);
+    return (int)(X * m_screenHorizScale);
 }
 
 int MythUIScreenBounds::NormY(int Y) const
 {
-    return qRound(Y * m_screenVertScale);
+    return (int)(Y * m_screenVertScale);
 }
 
 void MythUIScreenBounds::GetScalingFactors(float& Horizontal, float& Vertical) const
