diff -Naur mythtv-master-20200324-gaa46cdd-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20200324-gaa46cdd-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20200324-gaa46cdd-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-03-24 09:41:39.923333334 +0100
+++ mythtv-master-20200324-gaa46cdd-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-03-24 09:41:41.330000000 +0100
@@ -2727,12 +2727,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
