diff -Naur mythtv-master-20200109-gc116014-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20200109-gc116014-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20200109-gc116014-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-01-09 22:04:11.206666668 +0100
+++ mythtv-master-20200109-gc116014-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-01-09 22:04:13.950000001 +0100
@@ -2730,12 +2730,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
