diff -Naur mythtv-master-20181017-gd8a3e6a-old/mythtv/libs/libmythtv/recorders/signalmonitor.cpp mythtv-master-20181017-gd8a3e6a-new/mythtv/libs/libmythtv/recorders/signalmonitor.cpp
--- mythtv-master-20181017-gd8a3e6a-old/mythtv/libs/libmythtv/recorders/signalmonitor.cpp	2018-10-17 15:28:51.000000000 +0200
+++ mythtv-master-20181017-gd8a3e6a-new/mythtv/libs/libmythtv/recorders/signalmonitor.cpp	2018-10-19 10:22:39.433333331 +0200
@@ -216,7 +216,7 @@
  *  \param wait_for_mask   SignalMonitorFlags to start with.
  */
 SignalMonitor::SignalMonitor(int _capturecardnum, ChannelBase *_channel,
-                             uint64_t wait_for_mask, bool _release_stream)
+                             bool _release_stream, uint64_t wait_for_mask)
     : MThread("SignalMonitor"),
       channel(_channel),               pParent(nullptr),
       capturecardnum(_capturecardnum), flags(wait_for_mask),
diff -Naur mythtv-master-20181017-gd8a3e6a-old/mythtv/libs/libmythtv/recorders/signalmonitor.h mythtv-master-20181017-gd8a3e6a-new/mythtv/libs/libmythtv/recorders/signalmonitor.h
--- mythtv-master-20181017-gd8a3e6a-old/mythtv/libs/libmythtv/recorders/signalmonitor.h	2018-10-17 15:28:51.000000000 +0200
+++ mythtv-master-20181017-gd8a3e6a-new/mythtv/libs/libmythtv/recorders/signalmonitor.h	2018-10-19 10:19:30.613333335 +0200
@@ -118,7 +118,7 @@
 
   protected:
     SignalMonitor(int _capturecardnum, ChannelBase *_channel,
-                  uint64_t wait_for_mask, bool _release_stream);
+                  bool _release_stream, uint64_t wait_for_mask);
 
     virtual void run(void);
 
