diff -Naur mythtv-master-20181110-g3aebace-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20181110-g3aebace-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20181110-g3aebace-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2018-11-10 12:16:36.000000000 +0100
+++ mythtv-master-20181110-g3aebace-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2018-11-10 12:38:28.989999961 +0100
@@ -3996,6 +3996,14 @@
         {
             AVFrame tmppicture;
 
+//            VideoFrame *xf = picframe;
+//            picframe = m_parent->GetNextVideoFrame();
+
+//            unsigned char *buf = picframe->buf;
+            av_image_fill_arrays(tmppicture.data, tmppicture.linesize,
+                                 buf, AV_PIX_FMT_YUV420P, use_frame->width,
+                                 use_frame->height, IMAGE_ALIGN);
+
             tmppicture.data[0] = buf + picframe->offsets[0];
             tmppicture.data[1] = buf + picframe->offsets[1];
             tmppicture.data[2] = buf + picframe->offsets[2];
