diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Catalan.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Catalan.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Catalan.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Catalan.lang	2016-05-05 10:55:36.286666666 +0200
@@ -933,6 +933,8 @@
 "Select the correct show"
 "Server returned invalid data when attempting to retrieve metadata."
 "Server Statistics"
+"serviceid"
+"mplexid"
 "Set Host"
     Ajustar Host
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Czech.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Czech.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Czech.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Czech.lang	2016-05-05 10:55:36.286666666 +0200
@@ -655,6 +655,8 @@
     NAČÍTÁVÁM
 "Local Server Statistics"
     Statistiky lokálního serveru
+"serviceid"
+"mplexid"
 "LOCKDOWN_NOTICE"
 "LOCKDOWN_NOTICE_2"
 "Logs"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Danish.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Danish.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Danish.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Danish.lang	2016-05-05 10:55:36.286666666 +0200
@@ -1261,6 +1261,8 @@
     Serveren svarede med ugyldige data ved metadata opslag
 "Server Statistics"
     Server statistik
+"serviceid"
+"mplexid"
 "Set Host"
     Sæt vært
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Dutch.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Dutch.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Dutch.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Dutch.lang	2016-05-05 10:55:36.286666666 +0200
@@ -906,6 +906,8 @@
 "Select the correct show"
 "Server returned invalid data when attempting to retrieve metadata."
 "Server Statistics"
+"serviceid"
+"mplexid"
 "Set Host"
     Host instellen
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/English_CA.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/English_CA.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/English_CA.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/English_CA.lang	2016-05-05 10:55:36.286666666 +0200
@@ -732,6 +732,8 @@
 "Select the correct show"
 "Server returned invalid data when attempting to retrieve metadata."
 "Server Statistics"
+"serviceid"
+"mplexid"
 "Set Host"
 "Settings"
 "settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/English_GB.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/English_GB.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/English_GB.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/English_GB.lang	2016-05-05 10:55:36.286666666 +0200
@@ -748,6 +748,8 @@
 "Select the correct show"
 "Server returned invalid data when attempting to retrieve metadata."
 "Server Statistics"
+"serviceid"
+"mplexid"
 "Set Host"
 "Settings"
 "settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/English.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/English.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/English.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/English.lang	2016-05-05 10:55:36.286666666 +0200
@@ -732,6 +732,8 @@
 "Select the correct show"
 "Server returned invalid data when attempting to retrieve metadata."
 "Server Statistics"
+"mplexid"
+"serviceid"
 "Set Host"
 "settings"
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Finnish.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Finnish.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Finnish.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Finnish.lang	2016-05-05 10:55:36.286666666 +0200
@@ -1281,6 +1281,8 @@
 "Server returned invalid data when attempting to retrieve metadata."
 "Server Statistics"
     Serverin statistiikkaa
+"serviceid"
+"mplexid"
 "Set Host"
     Aseta palvelin
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/French_CA.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/French_CA.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/French_CA.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/French_CA.lang	2016-05-05 10:55:36.289999999 +0200
@@ -1314,6 +1314,8 @@
     Le serveur a retourné des données invalides pendant la recherche de métadonnées.
 "Server Statistics"
     Statistiques du serveur
+"serviceid"
+"mplexid"
 "Set Host"
     Choisir l'ordinateur
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/French.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/French.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/French.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/French.lang	2016-05-05 10:55:36.289999999 +0200
@@ -1314,6 +1314,8 @@
     Le serveur a retourné des données invalides pendant la recherche de métadonnées.
 "Server Statistics"
     Statistiques du serveur
+"serviceid"
+"mplexid"
 "Set Host"
     Choisir l'ordinateur
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/German.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/German.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/German.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/German.lang	2016-05-05 10:55:36.289999999 +0200
@@ -1297,6 +1297,8 @@
     Beim Versuch Metadaten abzurufen gab der Server ungültige Daten zurück.
 "Server Statistics"
     Serverstatistik
+"serviceid"
+"mplexid"
 "Set Host"
     Setze Host
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Hungarian.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Hungarian.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Hungarian.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Hungarian.lang	2016-05-05 10:55:36.289999999 +0200
@@ -1178,6 +1178,8 @@
 "Server returned invalid data when attempting to retrieve metadata."
 "Server Statistics"
     Szerver statisztikák
+"serviceid"
+"mplexid"
 "Set Host"
     Gép beállítása
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Japanese.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Japanese.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Japanese.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Japanese.lang	2016-05-05 10:55:36.289999999 +0200
@@ -907,6 +907,8 @@
 "Select the correct show"
 "Server returned invalid data when attempting to retrieve metadata."
 "Server Statistics"
+"serviceid"
+"mplexid"
 "Set Host"
     ホスト設定
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Norwegian_NB.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Norwegian_NB.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Norwegian_NB.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Norwegian_NB.lang	2016-05-05 10:55:36.289999999 +0200
@@ -1273,6 +1273,8 @@
     Tjeneren returnerte ugyldige data under henting av metadata.
 "Server Statistics"
     Tjenerstatistikk
+"serviceid"
+"mplexid"
 "Set Host"
     Sett vert
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Polish.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Polish.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Polish.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Polish.lang	2016-05-05 10:55:36.289999999 +0200
@@ -819,6 +819,8 @@
 "Select the correct show"
 "Server returned invalid data when attempting to retrieve metadata."
 "Server Statistics"
+"serviceid"
+"mplexid"
 "Set Host"
 "Settings"
     Ustawienia
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Slovenian.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Slovenian.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Slovenian.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Slovenian.lang	2016-05-05 10:55:36.289999999 +0200
@@ -1136,6 +1136,8 @@
 "Server returned invalid data when attempting to retrieve metadata."
 "Server Statistics"
     Stratistike strežnika
+"serviceid"
+"mplexid"
 "Set Host"
     Nastavi gostitelja
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Spanish_ES.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Spanish_ES.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Spanish_ES.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Spanish_ES.lang	2016-05-05 10:55:36.289999999 +0200
@@ -684,6 +684,8 @@
     CARGANDO
 "Local Server Statistics"
     Estadísticas del Servidor Local
+"serviceid"
+"mplexid"
 "LOCKDOWN_NOTICE"
 "LOCKDOWN_NOTICE_2"
 "Logs"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Spanish.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Spanish.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Spanish.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Spanish.lang	2016-05-05 10:55:36.289999999 +0200
@@ -1001,6 +1001,8 @@
 "Select the correct show"
 "Server returned invalid data when attempting to retrieve metadata."
 "Server Statistics"
+"serviceid"
+"mplexid"
 "Set Host"
     Ajustar Host
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/_shared/lang/Swedish.lang mythtv-new/mythplugins/mythweb/modules/_shared/lang/Swedish.lang
--- mythtv-old/mythplugins/mythweb/modules/_shared/lang/Swedish.lang	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/_shared/lang/Swedish.lang	2016-05-05 10:55:36.289999999 +0200
@@ -1301,6 +1301,8 @@
     Servern returnerade ogiltiga data under hämtning av metadata.
 "Server Statistics"
     Serverstatistik
+"serviceid"
+"mplexid"
 "Set Host"
     Sätt värd
 "Settings"
diff -Naur mythtv-old/mythplugins/mythweb/modules/tv/set_channels.php mythtv-new/mythplugins/mythweb/modules/tv/set_channels.php
--- mythtv-old/mythplugins/mythweb/modules/tv/set_channels.php	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/tv/set_channels.php	2016-05-05 10:57:12.806666641 +0200
@@ -21,6 +21,8 @@
             // Not deleting so grab values that can be empty
                 $query = 'UPDATE channel SET xmltvid       = ?,
                                              freqid        = ?,
+                                             mplexid       = ?,
+                                             serviceid     = ?,
                                              finetune      = ?,
                                              videofilters  = ?,
                                              brightness    = ?,
@@ -33,6 +35,8 @@
                                              visible       = ?';
                 $query_params[] = $data['xmltvid'];
                 $query_params[] = $data['freqid'];
+                $query_params[] = $data['mplexid'];
+                $query_params[] = $data['serviceid'];
                 $query_params[] = empty($data['finetune'])      ? 0 : $data['finetune'];
                 $query_params[] = $data['videofilters'];
                 $query_params[] = $data['brightness'];
@@ -63,7 +67,7 @@
             }
         }
     // Do a reschedule to refresh scheduled recordings;
-        MythBackend::find()->rescheduleRecording();
+        // MythBackend::find()->rescheduleRecording();
     }
 
 // Sortby
diff -Naur mythtv-old/mythplugins/mythweb/modules/tv/tmpl/default/set_channels.php mythtv-new/mythplugins/mythweb/modules/tv/tmpl/default/set_channels.php
--- mythtv-old/mythplugins/mythweb/modules/tv/tmpl/default/set_channels.php	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/tv/tmpl/default/set_channels.php	2016-05-05 10:55:36.289999999 +0200
@@ -15,21 +15,24 @@
 <tr class="menu" align="center">
     <td width="4%"><?php  echo t('delete')        ?></td>
     <td width="4%"><a href="<?php echo form_action ?>?sortby=sourceid"><?php  echo t('sourceid') ?></a></td>
-    <td width="4%"><a href="<?php echo form_action ?>?sortby=xmltvid"><?php  echo t('xmltvid') ?></a></td>
-    <td width="5%"><a href="<?php echo form_action ?>?sortby=channum"><?php  echo t('channum') ?></a></td>
-    <td width="5%"><a href="<?php echo form_action ?>?sortby=callsign"><?php echo t('callsign') ?></a></td>
-    <td width="5%"><a href="<?php echo form_action ?>?sortby=name"><?php     echo t('name') ?></a></td>
-    <td width="5%"><a href="<?php echo form_action ?>?sortby=freqid"><?php   echo t('freqid') ?></a></td>
+    <td width="4%"><a href="<?php echo form_action ?>?sortby=xmltvid"><?php   echo t('xmltvid') ?></a></td>
+    <td width="5%"><a href="<?php echo form_action ?>?sortby=channum"><?php   echo t('channum') ?></a></td>
+    <td width="5%"><a href="<?php echo form_action ?>?sortby=callsign"><?php  echo t('callsign') ?></a></td>
+    <td width="5%"><a href="<?php echo form_action ?>?sortby=name"><?php      echo t('name') ?></a></td>
+    <td width="5%"><a href="<?php echo form_action ?>?sortby=mplexid"><?php   echo t('mplexid') ?></a></td>
+    <td width="5%"><a href="<?php echo form_action ?>?sortby=serviceid"><?php echo t('serviceid') ?></a></td>
+    <td width="5%"><?php  echo t('visible')       ?></td>
+    <td width="5%"><?php  echo t('useonairguide') ?></td>
+    <td width="5%"><?php  echo t('commfree')      ?></td>
+    <td width="5%"><?php  echo t('recpriority')   ?></td>
+    <td width="5%"><a href="<?php echo form_action ?>?sortby=freqid"><?php    echo t('freqid') ?></a></td>
     <td width="5%"><?php  echo t('finetune')      ?></td>
     <td width="5%"><?php  echo t('videofilters')  ?></td>
     <td width="7%"><?php  echo t('brightness')    ?></td>
     <td width="7%"><?php  echo t('contrast')      ?></td>
     <td width="7%"><?php  echo t('color')         ?></td>
     <td width="7%"><?php  echo t('hue')           ?></td>
-    <td width="5%"><?php  echo t('recpriority')   ?></td>
-    <td width="5%"><?php  echo t('commfree')      ?></td>
-    <td width="5%"><?php  echo t('visible')       ?></td>
-    <td width="5%"><?php  echo t('useonairguide') ?></td>
+
 </tr><?php
     foreach ($channels as $chanid) {
         $channel =& Channel::find($chanid);
@@ -39,7 +42,13 @@
     <td><input type="text" size="5"  name="channel[<?php echo $channel->chanid ?>][xmltvid]"       id="xmltvid_<?php      echo $channel->chanid ?>" value="<?php echo html_entities($channel->xmltvid)      ?>" style="text-align: center" /></td>
     <td><input type="text" size="5"  name="channel[<?php echo $channel->chanid ?>][channum]"       id="channum_<?php      echo $channel->chanid ?>" value="<?php echo html_entities($channel->channum)      ?>" style="text-align: center" /></td>
     <td><input type="text" size="10" name="channel[<?php echo $channel->chanid ?>][callsign]"      id="callsign_<?php     echo $channel->chanid ?>" value="<?php echo html_entities($channel->callsign)     ?>" /></td>
-    <td><input type="text" size="27" name="channel[<?php echo $channel->chanid ?>][name]"          id="name_<?php         echo $channel->chanid ?>" value="<?php echo html_entities($channel->name) ?>" /></td>
+    <td><input type="text" size="27" name="channel[<?php echo $channel->chanid ?>][name]"          id="name_<?php         echo $channel->chanid ?>" value="<?php echo html_entities($channel->name)         ?>" /></td>
+    <td><input type="text" size="3"  name="channel[<?php echo $channel->chanid ?>][mplexid]"       id="mplexid_<?php      echo $channel->chanid ?>" value="<?php echo html_entities($channel->mplexid)      ?>" style="text-align: center" /></td>
+    <td><input type="text" size="3"  name="channel[<?php echo $channel->chanid ?>][serviceid]"     id="serviceid_<?php    echo $channel->chanid ?>" value="<?php echo html_entities($channel->serviceid)    ?>" style="text-align: center" /></td>
+    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][visible]"       value="1"<?php if (!empty($channel->visible))       echo ' CHECKED' ?> /></td>
+    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][useonairguide]" value="1"<?php if (!empty($channel->useonairguide)) echo ' CHECKED' ?> /></td>
+    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][commfree]"      value="1"<?php if (!empty($channel->commmethod) && $channel->commmethod == -2)      echo ' CHECKED'      ?> /></td>
+    <td><input type="text" size="2"  name="channel[<?php echo $channel->chanid ?>][recpriority]"   id="recpriority_<?php  echo $channel->chanid ?>" value="<?php echo html_entities($channel->recpriority)  ?>" style="text-align: center" /></td>
     <td><input type="text" size="3"  name="channel[<?php echo $channel->chanid ?>][freqid]"        id="freqid_<?php       echo $channel->chanid ?>" value="<?php echo html_entities($channel->freqid)       ?>" style="text-align: center" /></td>
     <td><input type="text" size="3"  name="channel[<?php echo $channel->chanid ?>][finetune]"      id="finetune_<?php     echo $channel->chanid ?>" value="<?php echo html_entities($channel->finetune)     ?>" style="text-align: center" /></td>
     <td><input type="text" size="3"  name="channel[<?php echo $channel->chanid ?>][videofilters]"  id="videofilters_<?php echo $channel->chanid ?>" value="<?php echo html_entities($channel->videofilters) ?>" style="text-align: center" /></td>
@@ -47,10 +56,6 @@
     <td><input type="text" size="5"  name="channel[<?php echo $channel->chanid ?>][contrast]"      id="contrast_<?php     echo $channel->chanid ?>" value="<?php echo html_entities($channel->contrast)     ?>" style="text-align: center" /></td>
     <td><input type="text" size="5"  name="channel[<?php echo $channel->chanid ?>][colour]"        id="colour_<?php       echo $channel->chanid ?>" value="<?php echo html_entities($channel->colour)       ?>" style="text-align: center" /></td>
     <td><input type="text" size="5"  name="channel[<?php echo $channel->chanid ?>][hue]"           id="hue_<?php          echo $channel->chanid ?>" value="<?php echo html_entities($channel->hue)          ?>" style="text-align: center" /></td>
-    <td><input type="text" size="2"  name="channel[<?php echo $channel->chanid ?>][recpriority]"   id="recpriority_<?php  echo $channel->chanid ?>" value="<?php echo html_entities($channel->recpriority)  ?>" style="text-align: center" /></td>
-    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][commfree]"      value="1"<?php if (!empty($channel->commmethod) && $channel->commmethod == -2)      echo ' CHECKED' ?> /></td>
-    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][visible]"       value="1"<?php if (!empty($channel->visible))       echo ' CHECKED' ?> /></td>
-    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][useonairguide]" value="1"<?php if (!empty($channel->useonairguide)) echo ' CHECKED' ?> /></td>
 </tr><?php
     }
 ?>
diff -Naur mythtv-old/mythplugins/mythweb/modules/tv/tmpl/lite/set_channels.php mythtv-new/mythplugins/mythweb/modules/tv/tmpl/lite/set_channels.php
--- mythtv-old/mythplugins/mythweb/modules/tv/tmpl/lite/set_channels.php	2016-05-05 10:53:21.000000000 +0200
+++ mythtv-new/mythplugins/mythweb/modules/tv/tmpl/lite/set_channels.php	2016-05-05 10:55:36.293333333 +0200
@@ -19,6 +19,12 @@
     <td width="5%"><a href="<?php echo form_action ?>?sortby=channum"><?php  echo t('channum') ?></a></td>
     <td width="5%"><a href="<?php echo form_action ?>?sortby=callsign"><?php echo t('callsign') ?></a></td>
     <td width="5%"><a href="<?php echo form_action ?>?sortby=name"><?php     echo t('name') ?></a></td>
+    <td width="5%"><?php  echo t('mplexid')       ?></td>
+    <td width="5%"><?php  echo t('serviceid')     ?></td>
+    <td width="5%"><?php  echo t('visible')       ?></td>
+    <td width="5%"><?php  echo t('useonairguide') ?></td>
+    <td width="5%"><?php  echo t('commfree')      ?></td>
+    <td width="5%"><?php  echo t('recpriority')   ?></td>
     <td width="5%"><a href="<?php echo form_action ?>?sortby=freqid"><?php   echo t('freqid') ?></a></td>
     <td width="5%"><?php  echo t('finetune')      ?></td>
     <td width="5%"><?php  echo t('videofilters')  ?></td>
@@ -26,10 +32,6 @@
     <td width="7%"><?php  echo t('contrast')      ?></td>
     <td width="7%"><?php  echo t('color')         ?></td>
     <td width="7%"><?php  echo t('hue')           ?></td>
-    <td width="5%"><?php  echo t('recpriority')   ?></td>
-    <td width="5%"><?php  echo t('commfree')      ?></td>
-    <td width="5%"><?php  echo t('visible')       ?></td>
-    <td width="5%"><?php  echo t('useonairguide') ?></td>
 </tr><?php
     foreach ($channels as $chanid) {
         $channel =& Channel::find($chanid);
@@ -40,6 +42,12 @@
     <td><input type="text" size="3"  name="channel[<?php echo $channel->chanid ?>][channum]"       id="channum_<?php      echo $channel->chanid ?>" value="<?php echo html_entities($channel->channum)      ?>" style="text-align: center" /></td>
     <td><input type="text" size="10" name="channel[<?php echo $channel->chanid ?>][callsign]"      id="callsign_<?php     echo $channel->chanid ?>" value="<?php echo html_entities($channel->callsign)     ?>" /></td>
     <td><input type="text" size="27" name="channel[<?php echo $channel->chanid ?>][name]"          id="name_<?php         echo $channel->chanid ?>" value="<?php echo html_entities($channel->name) ?>" /></td>
+    <td><input type="text" size="3"  name="channel[<?php echo $channel->chanid ?>][mplexid]"       id="mplexid_<?php      echo $channel->chanid ?>" value="<?php echo html_entities($channel->mplexid)      ?>" style="text-align: center" /></td>
+    <td><input type="text" size="3"  name="channel[<?php echo $channel->chanid ?>][serviceid]"     id="serviceid_<?php    echo $channel->chanid ?>" value="<?php echo html_entities($channel->serviceid)    ?>" style="text-align: center" /></td>
+    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][visible]"       value="1"<?php if (!empty($channel->visible))       echo ' CHECKED' ?> /></td>
+    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][useonairguide]" value="1"<?php if (!empty($channel->useonairguide)) echo ' CHECKED' ?> /></td>
+    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][commfree]"      value="1"<?php if (!empty($channel->commmethod) && $channel->commmethod == -2)      echo ' CHECKED' ?> /></td>
+    <td><input type="text" size="2"  name="channel[<?php echo $channel->chanid ?>][recpriority]"   id="recpriority_<?php  echo $channel->chanid ?>" value="<?php echo html_entities($channel->recpriority)  ?>" style="text-align: center" /></td>
     <td><input type="text" size="3"  name="channel[<?php echo $channel->chanid ?>][freqid]"        id="freqid_<?php       echo $channel->chanid ?>" value="<?php echo html_entities($channel->freqid)       ?>" style="text-align: center" /></td>
     <td><input type="text" size="3"  name="channel[<?php echo $channel->chanid ?>][finetune]"      id="finetune_<?php     echo $channel->chanid ?>" value="<?php echo html_entities($channel->finetune)     ?>" style="text-align: center" /></td>
     <td><input type="text" size="3"  name="channel[<?php echo $channel->chanid ?>][videofilters]"  id="videofilters_<?php echo $channel->chanid ?>" value="<?php echo html_entities($channel->videofilters) ?>" style="text-align: center" /></td>
@@ -47,10 +55,6 @@
     <td><input type="text" size="5"  name="channel[<?php echo $channel->chanid ?>][contrast]"      id="contrast_<?php     echo $channel->chanid ?>" value="<?php echo html_entities($channel->contrast)     ?>" style="text-align: center" /></td>
     <td><input type="text" size="5"  name="channel[<?php echo $channel->chanid ?>][colour]"        id="colour_<?php       echo $channel->chanid ?>" value="<?php echo html_entities($channel->colour)       ?>" style="text-align: center" /></td>
     <td><input type="text" size="5"  name="channel[<?php echo $channel->chanid ?>][hue]"           id="hue_<?php          echo $channel->chanid ?>" value="<?php echo html_entities($channel->hue)          ?>" style="text-align: center" /></td>
-    <td><input type="text" size="2"  name="channel[<?php echo $channel->chanid ?>][recpriority]"   id="recpriority_<?php  echo $channel->chanid ?>" value="<?php echo html_entities($channel->recpriority)  ?>" style="text-align: center" /></td>
-    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][commfree]"      value="1"<?php if (!empty($channel->commmethod) && $channel->commmethod == -2)      echo ' CHECKED' ?> /></td>
-    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][visible]"       value="1"<?php if (!empty($channel->visible))       echo ' CHECKED' ?> /></td>
-    <td><input type="checkbox"       name="channel[<?php echo $channel->chanid ?>][useonairguide]" value="1"<?php if (!empty($channel->useonairguide)) echo ' CHECKED' ?> /></td>
 </tr><?php
     }
 ?>
