diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/diseqc.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/diseqc.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/diseqc.cpp	2014-05-05 11:48:49.938760864 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/diseqc.cpp	2014-05-05 11:49:01.002029522 +0200
@@ -494,8 +494,13 @@
         return false;
     }
 
-    // apply any voltage change
-    ApplyVoltage(settings, tuning);
+    // apply any voltage change for non-SCR systems
+    DiSEqCDevSCR *scr = FindSCR(settings);
+    if (!scr)
+        ApplyVoltage(settings, tuning);
+    else
+        LOG(VB_CHANNEL, LOG_INFO, LOC + "SCR bus->skiping setting LNB Voltage");
+
 
     // turn off tone burst first if commands need to be sent
     if (m_root->IsCommandNeeded(settings, tuning))
@@ -726,13 +731,15 @@
         diseqc_bus_already_reset = false;
     }
 
+    // make sure the bus is powered
+    SetVoltage(SEC_VOLTAGE_18);
+    usleep(DISEQC_POWER_ON_WAIT);
+    // some DiSEqC devices need more time. see #8465
+    usleep(DISEQC_POWER_ON_WAIT);
+
+
     if (!diseqc_bus_already_reset || !is_SCR)
     {
-        // make sure the bus is powered
-        SetVoltage(SEC_VOLTAGE_18);
-        usleep(DISEQC_POWER_ON_WAIT);
-        // some DiSEqC devices need more time. see #8465
-        usleep(DISEQC_POWER_ON_WAIT);
 
         // issue a global reset command
         LOG(VB_CHANNEL, LOG_INFO, LOC + "Resetting DiSEqC Bus");
@@ -741,7 +748,7 @@
             LOG(VB_GENERAL, LOG_ERR, LOC + "DiSEqC reset failed" + ENO);
             return false;
         }
-        
+
         if (is_SCR)
             diseqc_bus_already_reset = true;
     }
@@ -2231,16 +2238,16 @@
     (void) repeats;
 
     // power on bus
-    if (!m_tree.SetVoltage(SEC_VOLTAGE_18))
-        return false;
-    usleep(DISEQC_LONG_WAIT);
+//    if (!m_tree.SetVoltage(SEC_VOLTAGE_18))
+//        return false;
+//    usleep(DISEQC_LONG_WAIT);
 
     // send command
     bool ret = m_tree.SendCommand(DISEQC_ADR_SW_ALL, cmd, repeats, data_len, data);
 
     // power off bus
-    if (!m_tree.SetVoltage(SEC_VOLTAGE_13))
-        return false;
+//    if (!m_tree.SetVoltage(SEC_VOLTAGE_13))
+//        return false;
 
     return ret;
 }
