diff -Naur mythtv-0.24-26493-old/mythplugins/mythweb/modules/stream/stream_flv.pl mythtv-0.24-26493-new/mythplugins/mythweb/modules/stream/stream_flv.pl
--- mythtv-0.24-26493-old/mythplugins/mythweb/modules/stream/stream_flv.pl	2010-09-24 19:20:16.000000000 +0200
+++ mythtv-0.24-26493-new/mythplugins/mythweb/modules/stream/stream_flv.pl	2010-10-09 14:58:09.000000000 +0200
@@ -29,7 +29,7 @@
     }
     sub shutdown_handler {
         kill(1, $ffmpeg_pid) if ($ffmpeg_pid);
-        kill(-1, $ffmpeg_pgid) if ($ffmpeg_pgid);
+        kill(-1, $ffmpeg_pid) if ($ffmpeg_pid);
     }
 
 # Find ffmpeg
