diff -Naur mythtv-master-20150429-g6523186-old/mythtv/bindings/python/MythTV/utility/dt.py mythtv-master-20150429-g6523186-new/mythtv/bindings/python/MythTV/utility/dt.py
--- mythtv-master-20150429-g6523186-old/mythtv/bindings/python/MythTV/utility/dt.py	2015-04-30 10:51:21.588652290 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/bindings/python/MythTV/utility/dt.py	2015-04-30 10:51:37.658587197 +0200
@@ -15,7 +15,6 @@
 import re
 import time
 import singleton
-time.tzset()
 
 class basetzinfo( _pytzinfo ):
     """
