diff -Naur mythtv-master-build-old/mythplugins/mythweb/modules/stream/stream_flv.pl mythtv-master-build-new/mythplugins/mythweb/modules/stream/stream_flv.pl
--- mythtv-master-build-old/mythplugins/mythweb/modules/stream/stream_flv.pl	2016-01-08 22:10:47.000000000 +0100
+++ mythtv-master-build-new/mythplugins/mythweb/modules/stream/stream_flv.pl	2016-01-08 22:17:18.442451547 +0100
@@ -138,7 +138,7 @@
     }
     # Guess the filesize based on duration and bitrate. This allows for progressive download behavior
     my $lengthSec;
-    $dur = `$ffmpeg -i $filename 2>&1 | grep "Duration" | cut -d ' ' -f 4 | sed s/,//`;
+    $dur = `$ffmpeg -i $filename 2>&1 | /usr/bin/grep "Duration" | /usr/bin/cut -d ' ' -f 4 | /bin/sed s/,//`;
     if ($dur && $dur =~ /\d*:\d*:.*/) {
         @times = split(':',$dur);
         $lengthSec = $times[0]*3600+$times[1]*60+$times[2];
