diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/bindings/python/MythTV/utility/dt.py mythtv-master-20150105-g4216e00-new/mythtv/bindings/python/MythTV/utility/dt.py
--- mythtv-master-20150105-g4216e00-old/mythtv/bindings/python/MythTV/utility/dt.py	2015-01-06 16:50:41.497279634 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/bindings/python/MythTV/utility/dt.py	2015-01-06 16:50:59.420516431 +0100
@@ -15,7 +15,6 @@
 import re
 import time
 import singleton
-time.tzset()
 
 class basetzinfo( _pytzinfo ):
     """
