diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/contrib/user_jobs/mythlink.pl mythtv-master-20150105-g4216e00-new/mythtv/contrib/user_jobs/mythlink.pl
--- mythtv-master-20150105-g4216e00-old/mythtv/contrib/user_jobs/mythlink.pl	2015-01-06 16:51:10.440457077 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/contrib/user_jobs/mythlink.pl	2015-01-06 16:51:27.883696463 +0100
@@ -29,7 +29,7 @@
     our ($hostname, $dbh, $sh, $q, $count, $base_dir);
 
 # Default filename format
-    $dformat = '%T %- %Y-%m-%d, %g-%i %A %- %S';
+    $dformat = '%U/%T %- %Y-%m-%d, %g-%i %A %- %S';
 # Default separator character
     $dseparator = '-';
 # Default replacement character
