diff --git a/mythtv/libs/libmythtv/captions/subtitlescreen.cpp b/mythtv/libs/libmythtv/captions/subtitlescreen.cpp
index 6b955533f5..ec0f74d6f3 100644
--- a/mythtv/libs/libmythtv/captions/subtitlescreen.cpp
+++ b/mythtv/libs/libmythtv/captions/subtitlescreen.cpp
@@ -1762,10 +1762,8 @@ void SubtitleScreen::Pulse(void)
         currentFrame ? currentFrame->m_timecode : std::chrono::milliseconds::max();
     bool needRescale = (m_textFontZoom != m_textFontZoomPrev);
 
-    for (it = m_childrenList.begin(); it != m_childrenList.end(); it = itNext)
+    for (MythUIType *child : std::as_const(m_childrenList))
     {
-        itNext = it + 1;
-        MythUIType *child = *it;
         auto *wrapper = dynamic_cast<SubWrapper *>(child);
         if (!wrapper)
             continue;
