diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2015-01-06 16:51:40.700294103 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2015-01-06 16:51:54.156888297 +0100
@@ -1271,7 +1271,7 @@
         if ((m_playbackinfo) || livetv || watchingrecording)
         {
             recordingHasPositionMap |= SyncPositionMap();
-            if (recordingHasPositionMap && !livetv && !watchingrecording)
+            if (recordingHasPositionMap && !watchingrecording) // && !livetv)
             {
                 hasFullPositionMap = true;
                 gopset = true;
@@ -3118,12 +3118,12 @@
         int tempKeyFrameDist = framesRead - 1 - prevgoppos;
         bool reset_kfd = false;
 
-        if (!gopset || livetv) // gopset: we've seen 2 keyframes
+        if (!gopset) // || livetv) // gopset: we've seen 2 keyframes
         {
             LOG(VB_PLAYBACK, LOG_INFO, LOC +
                 "gopset not set, syncing positionMap");
             SyncPositionMap();
-            if (tempKeyFrameDist > 0 && !livetv)
+            if (tempKeyFrameDist > 0) // && !livetv)
             {
                 LOG(VB_PLAYBACK, LOG_INFO, LOC +
                     QString("Initial key frame distance: %1.")
@@ -3164,7 +3164,7 @@
     lastKey = prevgoppos = framesRead - 1;
 
     if (can_reliably_parse_keyframes &&
-        !hasFullPositionMap && !livetv && !watchingrecording)
+        !hasFullPositionMap && !watchingrecording) // && !livetv)
     {
         long long last_frame = 0;
         {
