diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/tv_rec.cpp	2015-01-06 16:20:36.000000000 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/tv_rec.cpp	2015-01-08 16:25:44.902746380 +0100
@@ -3875,7 +3875,7 @@
     // Start signal monitoring for devices capable of monitoring
     if (use_sm)
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "Starting Signal Monitor");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Starting Signal Monitor");
         bool error = false;
         if (!SetupSignalMonitor(
                 !antadj, request.flags & kFlagEITScan, livetv | antadj))
@@ -3923,7 +3923,7 @@
                     signalMonitorDeadline =
                         curRecording->GetRecordingEndTime();
 
-                    LOG(VB_RECORD, LOG_DEBUG, LOC +
+                    LOG(VB_GENERAL, LOG_DEBUG, LOC +
                         QString("Start recording deadline: %1 "
                                 "Good signal deadline: %2")
                         .arg(startRecordingDeadline.toLocalTime()
@@ -3979,7 +3979,7 @@
 
     if (signalMonitor->IsAllGood())
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
         if (curRecording && (MythDate::current() > startRecordingDeadline))
         {
             newRecStatus = rsFailing;
@@ -4065,7 +4065,7 @@
             --signalMonitorCheckCnt;
         else
         {
-            LOG(VB_RECORD, LOG_INFO, LOC +
+            LOG(VB_GENERAL, LOG_INFO, LOC +
                 QString("TuningSignalCheck: Still waiting.  Will timeout @ %1")
                 .arg(signalMonitorDeadline.toLocalTime()
                      .toString("hh:mm:ss.zzz")));
