diff -Naur mythtv-master-20140506-gd0e25ac-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20140506-gd0e25ac-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20140506-gd0e25ac-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2014-05-06 14:53:47.835422543 +0200
+++ mythtv-master-20140506-gd0e25ac-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2014-05-06 14:54:00.918709674 +0200
@@ -864,7 +864,7 @@
             {
                 LOG(VB_GENERAL, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
-                break;
+                //break;
             }
 
             // bounds check
