diff -Naur mythtv-master-20150429-g6523186-old/mythtv/bindings/python/MythTV/dataheap.py mythtv-master-20150429-g6523186-new/mythtv/bindings/python/MythTV/dataheap.py
--- mythtv-master-20150429-g6523186-old/mythtv/bindings/python/MythTV/dataheap.py	2015-04-30 10:53:53.498037286 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/bindings/python/MythTV/dataheap.py	2015-04-30 10:54:17.487940231 +0200
@@ -276,6 +276,7 @@
     Recorded(data=None, db=None) -> Recorded object
             'data' is a tuple containing (chanid, storagegroup)
     """
+    _key   = ['chanid','starttime']
     _defaults = {'title':u'Unknown', 'subtitle':'',          'description':'',
                  'category':'',      'hostname':'',          'bookmark':0,
                  'editing':0,        'cutlist':0,            'autoexpire':0,
