diff -Naur mythtv-master-20150129-gc7638ce-old/mythtv/bindings/python/MythTV/static.py mythtv-master-20150129-gc7638ce-new/mythtv/bindings/python/MythTV/static.py
--- mythtv-master-20150129-gc7638ce-old/mythtv/bindings/python/MythTV/static.py	2015-01-29 17:03:04.000000000 +0100
+++ mythtv-master-20150129-gc7638ce-new/mythtv/bindings/python/MythTV/static.py	2015-02-01 11:12:47.560816110 +0100
@@ -8,8 +8,8 @@
 SCHEMA_VERSION = 1336
 NVSCHEMA_VERSION = 1007
 MUSICSCHEMA_VERSION = 1018
-PROTO_VERSION = '83'
-PROTO_TOKEN = 'BreakingGlass'
+PROTO_VERSION = '85'
+PROTO_TOKEN = 'BluePool'
 BACKEND_SEP = '[]:[]'
 INSTALL_PREFIX = '/usr/local'
 
