diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythupnp/upnpcds.h mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythupnp/upnpcds.h
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythupnp/upnpcds.h	2019-03-29 08:40:09.640000001 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythupnp/upnpcds.h	2019-03-29 08:40:11.130000001 +0100
@@ -310,7 +310,7 @@
         // Implement UPnpServiceImpl methods that we can
 
         QString GetServiceType() override // UPnpServiceImpl
-            { return "urn:schemas-upnp-org:service:ContentDirectory:4"; }
+            { return "urn:schemas-upnp-org:service:ContentDirectory:1"; }
         QString GetServiceId() override // UPnpServiceImpl
             { return "urn:upnp-org:serviceId:ContentDirectory"; }
         QString GetServiceControlURL() override // UPnpServiceImpl
diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythupnp/upnpcmgr.h mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythupnp/upnpcmgr.h
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythupnp/upnpcmgr.h	2019-03-29 08:40:09.640000001 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythupnp/upnpcmgr.h	2019-03-29 08:40:11.130000001 +0100
@@ -68,7 +68,7 @@
         // Implement UPnpServiceImpl methods that we can
 
         QString GetServiceType() override // UPnpServiceImpl
-            { return "urn:schemas-upnp-org:service:ConnectionManager:3"; }
+            { return "urn:schemas-upnp-org:service:ConnectionManager:1"; }
         QString GetServiceId() override // UPnpServiceImpl
             { return "urn:upnp-org:serviceId:ConnectionManager"; }
         QString GetServiceControlURL() override // UPnpServiceImpl
diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/programs/mythbackend/devicemaster.xml mythtv-master-20190328-gdcb0847-new/mythtv/programs/mythbackend/devicemaster.xml
--- mythtv-master-20190328-gdcb0847-old/mythtv/programs/mythbackend/devicemaster.xml	2019-03-29 08:40:09.256666667 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/programs/mythbackend/devicemaster.xml	2019-03-29 08:40:11.130000001 +0100
@@ -7,7 +7,7 @@
     <device>
         <!-- UDN auto generated -->
         <friendlyName>MythTV AV Media Server</friendlyName>
-        <deviceType>urn:schemas-upnp-org:device:MediaServer:4</deviceType>
+        <deviceType>urn:schemas-upnp-org:device:MediaServer:1</deviceType>
         <manufacturer>MythTV</manufacturer>
         <manufacturerURL>http://www.mythtv.org/</manufacturerURL>
         <modelName>MythTV AV Media Server</modelName>
