diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythupnp/upnpcds.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythupnp/upnpcds.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythupnp/upnpcds.h	2018-10-12 10:51:49.996666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythupnp/upnpcds.h	2018-10-12 10:51:53.233333333 +0200
@@ -319,7 +319,7 @@
 
         // Implement UPnpServiceImpl methods that we can
 
-        virtual QString GetServiceType      () { return "urn:schemas-upnp-org:service:ContentDirectory:4"; }
+        virtual QString GetServiceType      () { return "urn:schemas-upnp-org:service:ContentDirectory:1"; }
         virtual QString GetServiceId        () { return "urn:upnp-org:serviceId:ContentDirectory"; }
         virtual QString GetServiceControlURL() { return m_sControlUrl.mid( 1 ); }
         virtual QString GetServiceDescURL   () { return m_sControlUrl.mid( 1 ) + "/GetServDesc"; }
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythupnp/upnpcmgr.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythupnp/upnpcmgr.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythupnp/upnpcmgr.h	2018-10-12 10:51:49.996666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythupnp/upnpcmgr.h	2018-10-12 10:51:53.233333333 +0200
@@ -67,7 +67,7 @@
 
         // Implement UPnpServiceImpl methods that we can
 
-        virtual QString GetServiceType      () { return "urn:schemas-upnp-org:service:ConnectionManager:3"; }
+        virtual QString GetServiceType      () { return "urn:schemas-upnp-org:service:ConnectionManager:1"; }
         virtual QString GetServiceId        () { return "urn:upnp-org:serviceId:ConnectionManager"; }
         virtual QString GetServiceControlURL() { return m_sControlUrl.mid( 1 ); }
         virtual QString GetServiceDescURL   () { return m_sControlUrl.mid( 1 ) + "/GetServDesc"; }
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/mythbackend/devicemaster.xml mythtv-master-20181012-g4651002-new/mythtv/programs/mythbackend/devicemaster.xml
--- mythtv-master-20181012-g4651002-old/mythtv/programs/mythbackend/devicemaster.xml	2018-10-12 10:51:49.946666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/mythbackend/devicemaster.xml	2018-10-12 10:51:53.233333333 +0200
@@ -7,7 +7,7 @@
     <device>
         <!-- UDN auto generated -->
         <friendlyName>MythTV AV Media Server</friendlyName>
-        <deviceType>urn:schemas-upnp-org:device:MediaServer:4</deviceType>
+        <deviceType>urn:schemas-upnp-org:device:MediaServer:1</deviceType>
         <manufacturer>MythTV</manufacturer>
         <manufacturerURL>http://www.mythtv.org/</manufacturerURL>
         <modelName>MythTV AV Media Server</modelName>
