diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythupnp/upnpcds.h mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythupnp/upnpcds.h
--- mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythupnp/upnpcds.h	2020-09-04 11:35:26.523333334 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythupnp/upnpcds.h	2020-09-04 11:35:27.786666667 +0200
@@ -312,7 +312,7 @@
         // Implement UPnpServiceImpl methods that we can
 
         QString GetServiceType() override // UPnpServiceImpl
-            { return "urn:schemas-upnp-org:service:ContentDirectory:4"; }
+            { return "urn:schemas-upnp-org:service:ContentDirectory:1"; }
         QString GetServiceId() override // UPnpServiceImpl
             { return "urn:upnp-org:serviceId:ContentDirectory"; }
         QString GetServiceControlURL() override // UPnpServiceImpl
diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythupnp/upnpcmgr.h mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythupnp/upnpcmgr.h
--- mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythupnp/upnpcmgr.h	2020-09-04 11:35:26.523333334 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythupnp/upnpcmgr.h	2020-09-04 11:35:27.786666667 +0200
@@ -66,7 +66,7 @@
         // Implement UPnpServiceImpl methods that we can
 
         QString GetServiceType() override // UPnpServiceImpl
-            { return "urn:schemas-upnp-org:service:ConnectionManager:3"; }
+            { return "urn:schemas-upnp-org:service:ConnectionManager:1"; }
         QString GetServiceId() override // UPnpServiceImpl
             { return "urn:upnp-org:serviceId:ConnectionManager"; }
         QString GetServiceControlURL() override // UPnpServiceImpl
diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/programs/mythbackend/devicemaster.xml mythtv-master-20200904-gca3ee6a-new/mythtv/programs/mythbackend/devicemaster.xml
--- mythtv-master-20200904-gca3ee6a-old/mythtv/programs/mythbackend/devicemaster.xml	2020-09-04 11:35:26.280000000 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/programs/mythbackend/devicemaster.xml	2020-09-04 11:35:27.786666667 +0200
@@ -7,7 +7,7 @@
     <device>
         <!-- UDN auto generated -->
         <friendlyName>MythTV AV Media Server</friendlyName>
-        <deviceType>urn:schemas-upnp-org:device:MediaServer:4</deviceType>
+        <deviceType>urn:schemas-upnp-org:device:MediaServer:1</deviceType>
         <manufacturer>MythTV</manufacturer>
         <manufacturerURL>http://www.mythtv.org/</manufacturerURL>
         <modelName>MythTV AV Media Server</modelName>
