diff --git a/mythtv/libs/libmythupnp/upnpcds.h b/mythtv/libs/libmythupnp/upnpcds.h
index adc0d8b..f31634d 100644
--- a/mythtv/libs/libmythupnp/upnpcds.h
+++ b/mythtv/libs/libmythupnp/upnpcds.h
@@ -319,7 +319,7 @@ class UPNP_PUBLIC UPnpCDS : public Eventing
 
         // Implement UPnpServiceImpl methods that we can
 
-        virtual QString GetServiceType      () { return "urn:schemas-upnp-org:service:ContentDirectory:4"; }
+        virtual QString GetServiceType      () { return "urn:schemas-upnp-org:service:ContentDirectory:1"; }
         virtual QString GetServiceId        () { return "urn:upnp-org:serviceId:ContentDirectory"; }
         virtual QString GetServiceControlURL() { return m_sControlUrl.mid( 1 ); }
         virtual QString GetServiceDescURL   () { return m_sControlUrl.mid( 1 ) + "/GetServDesc"; }
diff --git a/mythtv/libs/libmythupnp/upnpcmgr.h b/mythtv/libs/libmythupnp/upnpcmgr.h
index ccb6fa9..5d15903 100644
--- a/mythtv/libs/libmythupnp/upnpcmgr.h
+++ b/mythtv/libs/libmythupnp/upnpcmgr.h
@@ -67,7 +67,7 @@ class UPNP_PUBLIC UPnpCMGR : public Eventing
 
         // Implement UPnpServiceImpl methods that we can
 
-        virtual QString GetServiceType      () { return "urn:schemas-upnp-org:service:ConnectionManager:3"; }
+        virtual QString GetServiceType      () { return "urn:schemas-upnp-org:service:ConnectionManager:1"; }
         virtual QString GetServiceId        () { return "urn:upnp-org:serviceId:ConnectionManager"; }
         virtual QString GetServiceControlURL() { return m_sControlUrl.mid( 1 ); }
         virtual QString GetServiceDescURL   () { return m_sControlUrl.mid( 1 ) + "/GetServDesc"; }
diff --git a/mythtv/programs/mythbackend/devicemaster.xml b/mythtv/programs/mythbackend/devicemaster.xml
index 4af91b9..7b675b2 100644
--- a/mythtv/programs/mythbackend/devicemaster.xml
+++ b/mythtv/programs/mythbackend/devicemaster.xml
@@ -7,7 +7,7 @@
     <device>
         <!-- UDN auto generated -->
         <friendlyName>MythTV AV Media Server</friendlyName>
-        <deviceType>urn:schemas-upnp-org:device:MediaServer:4</deviceType>
+        <deviceType>urn:schemas-upnp-org:device:MediaServer:1</deviceType>
         <manufacturer>MythTV</manufacturer>
         <manufacturerURL>http://www.mythtv.org/</manufacturerURL>
         <modelName>MythTV AV Media Server</modelName>
