diff -Naur mythtv-master-build-old/mythplugins/mythweb/modules/mythweb/tmpl/default/set_flvplayer.php mythtv-master-build-new/mythplugins/mythweb/modules/mythweb/tmpl/default/set_flvplayer.php
--- mythtv-master-build-old/mythplugins/mythweb/modules/mythweb/tmpl/default/set_flvplayer.php	2016-05-20 17:22:39.000000000 +0200
+++ mythtv-master-build-new/mythplugins/mythweb/modules/mythweb/tmpl/default/set_flvplayer.php	2016-05-20 15:59:44.000000000 +0200
@@ -23,9 +23,9 @@
          title="<?php echo t('Enable Flash Video player for recordings.') ?>"
         <?php
             $ffmpeg = '';
-            foreach (split (':', getenv ('PATH').':/usr/local/bin:/usr/bin') as $path) {
-                if (file_exists ($path."/ffmpeg")) {
-                    $ffmpeg = $path."/ffmpeg";
+            foreach (explode (':', getenv ('PATH').':/usr/local/bin:/usr/bin') as $path) {
+                if (file_exists ($path."/mythffmpeg")) {
+                    $ffmpeg = $path."/mythffmpeg";
                     break;
                 }
                 elseif (php_uname ('s') == 'Darwin' && file_exists ($path."/ffmpeg.app")) {
