diff -Naur mythtv-old/mythplugins/mythweb/modules/stream/stream_flv.pl mythtv-new/mythplugins/mythweb/modules/stream/stream_flv.pl
--- mythtv-old/mythplugins/mythweb/modules/stream/stream_flv.pl	2016-11-22 14:33:18.251763453 +0100
+++ mythtv-new/mythplugins/mythweb/modules/stream/stream_flv.pl	2016-11-22 14:32:19.748704132 +0100
@@ -35,10 +35,10 @@
 # Find ffmpeg
     $ffmpeg = '';
     foreach my $path (split(/:/, $ENV{'PATH'}.':/usr/local/bin:/usr/bin'), '.') {
-        if (-e "$path/mythffmpeg") {
-            $ffmpeg = "$path/mythffmpeg";
-            last;
-        }
+        #if (-e "$path/mythffmpeg") {
+        #    $ffmpeg = "$path/mythffmpeg";
+        #    last;
+        #}
         if (-e "$path/ffmpeg") {
             $ffmpeg = "$path/ffmpeg";
             last;
