diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/eitcache.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/eitcache.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/eitcache.cpp	2020-05-07 12:45:52.370000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/eitcache.cpp	2020-05-07 12:45:53.910000000 +0200
@@ -252,7 +252,7 @@
     }
 
     if (!eventMap->empty())
-        LOG(VB_EIT, LOG_INFO, LOC + QString("Loaded %1 entries for channel %2")
+        LOG(VB_GENERAL, LOG_INFO, LOC + QString("Loaded %1 entries for channel %2")
                 .arg(eventMap->size()).arg(chanid));
 
     m_entryCnt += eventMap->size();
diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/eitscanner.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/eitscanner.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/eitscanner.cpp	2020-05-07 12:45:52.370000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/eitscanner.cpp	2020-05-07 12:45:53.910000000 +0200
@@ -105,7 +105,7 @@
         // but not in the last 60 seconds
         if (!m_activeScan && eitCount && (t.elapsed() > 60 * 1000))
         {
-            LOG(VB_EIT, LOG_INFO,
+            LOG(VB_GENERAL, LOG_INFO,
                 LOC_ID + QString("Added %1 EIT Events").arg(eitCount));
             eitCount = 0;
             RescheduleRecordings();
@@ -117,7 +117,7 @@
             // if there have been any new events, tell scheduler to run.
             if (eitCount)
             {
-                LOG(VB_EIT, LOG_INFO,
+                LOG(VB_GENERAL, LOG_INFO,
                     LOC_ID + QString("Added %1 EIT Events").arg(eitCount));
                 eitCount = 0;
                 RescheduleRecordings();
@@ -136,7 +136,7 @@
                 {
                     m_eitHelper->SetChannelID(ChannelUtil::GetChanID(
                         m_rec->GetSourceID(), *m_activeScanNextChan));
-                    LOG(VB_EIT, LOG_INFO,
+                    LOG(VB_GENERAL, LOG_INFO,
                         LOC_ID + QString("Now looking for EIT data on "
                                          "multiplex of channel %1")
                         .arg(*m_activeScanNextChan));
@@ -209,7 +209,7 @@
     m_eitHelper->SetChannelID(m_channel->GetChanID());
     m_eitHelper->SetSourceID(ChannelUtil::GetSourceIDForChannel(m_channel->GetChanID()));
 
-    LOG(VB_EIT, LOG_INFO, LOC_ID + "Started passive scan.");
+    LOG(VB_GENERAL, LOG_INFO, LOC_ID + "Started passive scan.");
 }
 
 /** \fn EITScanner::StopPassiveScan(void)
@@ -269,7 +269,7 @@
         m_activeScanNextChan = m_activeScanChannels.begin();
     }
 
-    LOG(VB_EIT, LOG_INFO, LOC_ID +
+    LOG(VB_GENERAL, LOG_INFO, LOC_ID +
         QString("StartActiveScan called with %1 multiplexes")
             .arg(m_activeScanChannels.size()));
 
