diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/eitcache.cpp mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/eitcache.cpp
--- mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/eitcache.cpp	2019-01-05 12:44:57.710000000 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/eitcache.cpp	2019-01-05 12:44:59.670000000 +0100
@@ -259,7 +259,7 @@
     }
 
     if (eventMap->size())
-        LOG(VB_EIT, LOG_INFO, LOC + QString("Loaded %1 entries for channel %2")
+        LOG(VB_GENERAL, LOG_INFO, LOC + QString("Loaded %1 entries for channel %2")
                 .arg(eventMap->size()).arg(chanid));
 
     entryCnt += eventMap->size();
diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/eitscanner.cpp mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/eitscanner.cpp
--- mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/eitscanner.cpp	2019-01-05 12:44:57.716666667 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/eitscanner.cpp	2019-01-05 12:44:59.670000000 +0100
@@ -110,7 +110,7 @@
         // but not in the last 60 seconds
         if (!activeScan && eitCount && (t.elapsed() > 60 * 1000))
         {
-            LOG(VB_EIT, LOG_INFO,
+            LOG(VB_GENERAL, LOG_INFO,
                 LOC_ID + QString("Added %1 EIT Events").arg(eitCount));
             eitCount = 0;
             RescheduleRecordings();
@@ -122,7 +122,7 @@
             // if there have been any new events, tell scheduler to run.
             if (eitCount)
             {
-                LOG(VB_EIT, LOG_INFO,
+                LOG(VB_GENERAL, LOG_INFO,
                     LOC_ID + QString("Added %1 EIT Events").arg(eitCount));
                 eitCount = 0;
                 RescheduleRecordings();
@@ -141,7 +141,7 @@
                 {
                     eitHelper->SetChannelID(ChannelUtil::GetChanID(
                         rec->GetSourceID(), *activeScanNextChan));
-                    LOG(VB_EIT, LOG_INFO,
+                    LOG(VB_GENERAL, LOG_INFO,
                         LOC_ID + QString("Now looking for EIT data on "
                                          "multiplex of channel %1")
                         .arg(*activeScanNextChan));
@@ -215,7 +215,7 @@
     eitHelper->SetChannelID(channel->GetChanID());
     eitHelper->SetSourceID(ChannelUtil::GetSourceIDForChannel(channel->GetChanID()));
 
-    LOG(VB_EIT, LOG_INFO, LOC_ID + "Started passive scan.");
+    LOG(VB_GENERAL, LOG_INFO, LOC_ID + "Started passive scan.");
 }
 
 /** \fn EITScanner::StopPassiveScan(void)
@@ -273,7 +273,7 @@
         activeScanNextChan = activeScanChannels.begin();
     }
 
-    LOG(VB_EIT, LOG_INFO, LOC_ID +
+    LOG(VB_GENERAL, LOG_INFO, LOC_ID +
         QString("StartActiveScan called with %1 multiplexes")
             .arg(activeScanChannels.size()));
 
