diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-05-07 12:45:53.743333333 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-05-07 12:45:55.470000000 +0200
@@ -505,6 +505,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(m_networkID).arg(sdt->OriginalNetworkID()).arg(m_transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
