diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-02-17 16:58:07.836666669 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-02-17 16:58:09.320000002 +0100
@@ -508,6 +508,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(networkID).arg(sdt->OriginalNetworkID()).arg(transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != networkID || sdt->TSID() != transportID)
     {
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
