diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-10-02 13:02:08.550000000 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-10-02 13:02:10.063333333 +0200
@@ -512,6 +512,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(m_networkID).arg(sdt->OriginalNetworkID()).arg(m_transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
         DVBStreamData *dvb = GetDVBStreamData();
