diff -Naur mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-09-21 09:17:55.116666668 +0200
+++ mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-09-21 09:17:56.333333335 +0200
@@ -512,6 +512,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(m_networkID).arg(sdt->OriginalNetworkID()).arg(m_transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
         DVBStreamData *dvb = GetDVBStreamData();
