diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-03-29 08:40:15.000000001 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-03-29 08:40:16.170000001 +0100
@@ -502,6 +502,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(m_networkID).arg(sdt->OriginalNetworkID()).arg(m_transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
