diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-09-15 09:22:38.666666668 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-09-15 09:22:43.400000001 +0200
@@ -520,6 +520,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(m_networkID).arg(sdt->OriginalNetworkID()).arg(m_transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
         DVBStreamData *dvb = GetDVBStreamData();
