diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2018-10-12 10:52:00.943333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2018-10-12 10:52:05.226666666 +0200
@@ -507,6 +507,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(networkID).arg(sdt->OriginalNetworkID()).arg(transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != networkID || sdt->TSID() != transportID)
     {
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
