diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-12-27 12:18:55.653333333 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-12-27 12:18:57.046666667 +0100
@@ -505,6 +505,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(m_networkID).arg(sdt->OriginalNetworkID()).arg(m_transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
