diff -Naur mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2018-10-20 14:57:00.196666667 +0200
+++ mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2018-10-20 14:57:03.510000001 +0200
@@ -508,6 +508,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(networkID).arg(sdt->OriginalNetworkID()).arg(transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != networkID || sdt->TSID() != transportID)
     {
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
