diff -Naur mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-10-10 16:07:35.766666666 +0200
+++ mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-10-10 16:07:37.566666666 +0200
@@ -511,6 +511,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(m_networkID).arg(sdt->OriginalNetworkID()).arg(m_transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
         DVBStreamData *dvb = GetDVBStreamData();
