diff -Naur mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2020-09-21 09:17:56.310000001 +0200
+++ mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2020-09-21 09:17:57.546666668 +0200
@@ -874,7 +874,7 @@
                 tspacket->data()[i++] != 0x00 ||
                 tspacket->data()[i++] != 0x01)
             {
-                LOG(VB_GENERAL, LOG_ERR, LOC +
+                LOG(VB_RECORD, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
                 break;
             }
diff -Naur mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-09-21 09:17:56.333333335 +0200
+++ mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-09-21 09:17:57.546666668 +0200
@@ -353,7 +353,7 @@
     {
         if (insert_crc(m_seenTableCrc, *pmt))
         {
-            LOG(VB_CHANNEL, LOG_DEBUG, LOC +
+            LOG(VB_CHANNEL, LOG_ERR, LOC +
                 QString("Wrong PMT; pmt->ProgramNumber(%1) desired(%2)")
                 .arg(pmt->ProgramNumber()).arg(m_programNumber));
         }
