diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2018-10-12 10:52:05.083333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2018-10-12 10:52:07.613333333 +0200
@@ -888,7 +888,7 @@
                 tspacket->data()[i++] != 0x00 ||
                 tspacket->data()[i++] != 0x01)
             {
-                LOG(VB_GENERAL, LOG_ERR, LOC +
+                LOG(VB_RECORD, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
                 break;
             }
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2018-10-12 10:52:05.226666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2018-10-12 10:52:07.613333333 +0200
@@ -355,7 +355,7 @@
     {
         if (insert_crc(seen_table_crc, *pmt))
         {
-            LOG(VB_GENERAL, LOG_ERR, LOC +
+            LOG(VB_CHANNEL, LOG_ERR, LOC +
                 QString("Wrong PMT; pmt->pn(%1) desired(%2)")
                 .arg(pmt->ProgramNumber()).arg(programNumber));
         }
