diff -Naur mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2019-12-12 19:15:51.860000002 +0100
+++ mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2019-12-12 19:15:53.800000002 +0100
@@ -832,7 +832,7 @@
                 tspacket->data()[i++] != 0x00 ||
                 tspacket->data()[i++] != 0x01)
             {
-                LOG(VB_GENERAL, LOG_ERR, LOC +
+                LOG(VB_RECORD, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
                 break;
             }
diff -Naur mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-12-12 19:15:52.406666669 +0100
+++ mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-12-12 19:15:53.800000002 +0100
@@ -351,7 +351,7 @@
     {
         if (insert_crc(m_seen_table_crc, *pmt))
         {
-            LOG(VB_GENERAL, LOG_ERR, LOC +
+            LOG(VB_CHANNEL, LOG_ERR, LOC +
                 QString("Wrong PMT; pmt->pn(%1) desired(%2)")
                 .arg(pmt->ProgramNumber()).arg(m_programNumber));
         }
