diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2020-05-07 12:45:55.166666667 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2020-05-07 12:45:56.726666667 +0200
@@ -857,7 +857,7 @@
                 tspacket->data()[i++] != 0x00 ||
                 tspacket->data()[i++] != 0x01)
             {
-                LOG(VB_GENERAL, LOG_ERR, LOC +
+                LOG(VB_RECORD, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
                 break;
             }
diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-05-07 12:45:55.470000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-05-07 12:45:56.726666667 +0200
@@ -351,7 +351,7 @@
     {
         if (insert_crc(m_seen_table_crc, *pmt))
         {
-            LOG(VB_CHANNEL, LOG_DEBUG, LOC +
+            LOG(VB_CHANNEL, LOG_ERR, LOC +
                 QString("Wrong PMT; pmt->ProgramNumber(%1) desired(%2)")
                 .arg(pmt->ProgramNumber()).arg(m_programNumber));
         }
