diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2020-01-14 09:19:27.229999999 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2020-01-14 09:19:28.653333332 +0100
@@ -844,7 +844,7 @@
                 tspacket->data()[i++] != 0x00 ||
                 tspacket->data()[i++] != 0x01)
             {
-                LOG(VB_GENERAL, LOG_ERR, LOC +
+                LOG(VB_RECORD, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
                 break;
             }
diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-01-14 09:19:27.466666666 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-01-14 09:19:28.656666666 +0100
@@ -351,7 +351,7 @@
     {
         if (insert_crc(m_seen_table_crc, *pmt))
         {
-            LOG(VB_CHANNEL, LOG_DEBUG, LOC +
+            LOG(VB_CHANNEL, LOG_ERR, LOC +
                 QString("Wrong PMT; pmt->ProgramNumber(%1) desired(%2)")
                 .arg(pmt->ProgramNumber()).arg(m_programNumber));
         }
