diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2019-01-05 12:45:00.630000000 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2019-01-05 12:45:02.423333333 +0100
@@ -885,7 +885,7 @@
                 tspacket->data()[i++] != 0x00 ||
                 tspacket->data()[i++] != 0x01)
             {
-                LOG(VB_GENERAL, LOG_ERR, LOC +
+                LOG(VB_RECORD, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
                 break;
             }
diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-01-05 12:45:00.820000000 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-01-05 12:45:02.423333333 +0100
@@ -356,7 +356,7 @@
     {
         if (insert_crc(seen_table_crc, *pmt))
         {
-            LOG(VB_GENERAL, LOG_ERR, LOC +
+            LOG(VB_CHANNEL, LOG_ERR, LOC +
                 QString("Wrong PMT; pmt->pn(%1) desired(%2)")
                 .arg(pmt->ProgramNumber()).arg(programNumber));
         }
