diff -Naur mythtv-master-20200116-gdd96d68-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20200116-gdd96d68-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20200116-gdd96d68-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-01-16 10:16:45.353333334 +0100
+++ mythtv-master-20200116-gdd96d68-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-01-16 10:16:46.500000001 +0100
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
