diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2018-10-12 10:52:07.533333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2018-10-12 10:52:10.223333333 +0200
@@ -297,7 +297,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
