diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-05-07 12:45:56.610000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-05-07 12:45:58.003333333 +0200
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
