diff -Naur mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-09-25 09:21:03.613333335 +0200
+++ mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-09-25 09:21:06.086666668 +0200
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
