diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-03-29 08:40:17.033333334 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-03-29 08:40:18.946666667 +0100
@@ -298,7 +298,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
