diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-10-02 13:02:10.880000000 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-10-02 13:02:12.386666666 +0200
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
