diff -Naur mythtv-master-20200109-gc116014-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20200109-gc116014-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20200109-gc116014-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-01-09 22:04:28.890000001 +0100
+++ mythtv-master-20200109-gc116014-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-01-09 22:04:31.346666668 +0100
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
