diff -Naur mythtv-master-20200324-gaa46cdd-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20200324-gaa46cdd-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20200324-gaa46cdd-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-03-24 09:41:50.610000001 +0100
+++ mythtv-master-20200324-gaa46cdd-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-03-24 09:41:52.146666667 +0100
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
