diff -Naur mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-12-12 19:15:53.186666669 +0100
+++ mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-12-12 19:15:55.096666669 +0100
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
