diff -Naur mythtv-master-20200311-g625fc71-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20200311-g625fc71-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20200311-g625fc71-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-03-14 13:29:44.416666668 +0100
+++ mythtv-master-20200311-g625fc71-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-03-14 13:29:46.766666667 +0100
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
