diff -Naur mythtv-master-20200530-ga180830-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20200530-ga180830-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20200530-ga180830-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-05-30 13:00:32.020000001 +0200
+++ mythtv-master-20200530-ga180830-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-05-30 13:00:37.666666668 +0200
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
