diff -Naur mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-04-09 12:25:12.237589782 +0200
+++ mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-04-09 12:25:13.874256390 +0200
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
