diff -Naur mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2018-10-20 14:57:06.286666668 +0200
+++ mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2018-10-20 14:57:09.026666668 +0200
@@ -298,7 +298,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
