diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-12-27 12:18:57.783333333 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-12-27 12:18:59.303333333 +0100
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
