diff -Naur mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-04-08 14:45:07.580000000 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-04-08 14:45:08.886666667 +0200
@@ -298,7 +298,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
