diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-01-14 09:19:28.466666665 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-01-14 09:19:30.563333332 +0100
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
