diff -Naur mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2015-04-30 10:50:38.565493242 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2015-04-30 10:51:24.875305624 +0200
@@ -297,7 +297,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
