diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/tv_rec.cpp	2019-12-27 12:18:59.020000000 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/tv_rec.cpp	2019-12-27 12:19:00.370000000 +0100
@@ -3945,7 +3945,7 @@
     // Start signal monitoring for devices capable of monitoring
     if (use_sm)
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "Starting Signal Monitor");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Starting Signal Monitor");
         bool error = false;
         if (!SetupSignalMonitor(
                 !antadj, (request.m_flags & kFlagEITScan) != 0U, livetv || antadj))
@@ -3995,7 +3995,7 @@
                     m_signalMonitorDeadline =
                         m_curRecording->GetRecordingEndTime().addSecs(-10);
 
-                    LOG(VB_CHANNEL, LOG_DEBUG, LOC +
+                    LOG(VB_GENERAL, LOG_DEBUG, LOC +
                         QString("Pre-fail start deadline: %1 "
                                 "Start recording deadline: %2 "
                                 "Good signal deadline: %3")
@@ -4067,7 +4067,7 @@
 
     if (m_signalMonitor->IsAllGood())
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
         if (m_curRecording && (current_time > m_startRecordingDeadline))
         {
             newRecStatus = RecStatus::Failing;
@@ -4156,7 +4156,7 @@
             --m_signalMonitorCheckCnt;
         else
         {
-            LOG(VB_RECORD, LOG_INFO, LOC +
+            LOG(VB_GENERAL, LOG_INFO, LOC +
                 QString("TuningSignalCheck: Still waiting.  Will timeout @ %1")
                 .arg(m_signalMonitorDeadline.toLocalTime()
                      .toString("hh:mm:ss.zzz")));
