diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/tv_rec.cpp	2020-01-14 09:19:30.026666666 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/tv_rec.cpp	2020-01-14 09:19:31.659999999 +0100
@@ -3936,7 +3936,7 @@
     // Start signal monitoring for devices capable of monitoring
     if (use_sm)
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "Starting Signal Monitor");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Starting Signal Monitor");
         bool error = false;
         if (!SetupSignalMonitor(
                 !antadj, (request.m_flags & kFlagEITScan) != 0U, livetv || antadj))
@@ -3986,7 +3986,7 @@
                     m_signalMonitorDeadline =
                         m_curRecording->GetRecordingEndTime().addSecs(-10);
 
-                    LOG(VB_CHANNEL, LOG_DEBUG, LOC +
+                    LOG(VB_GENERAL, LOG_DEBUG, LOC +
                         QString("Pre-fail start deadline: %1 "
                                 "Start recording deadline: %2 "
                                 "Good signal deadline: %3")
@@ -4058,7 +4058,7 @@
 
     if (m_signalMonitor->IsAllGood())
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
         if (m_curRecording && (current_time > m_startRecordingDeadline))
         {
             newRecStatus = RecStatus::Failing;
@@ -4147,7 +4147,7 @@
             --m_signalMonitorCheckCnt;
         else
         {
-            LOG(VB_RECORD, LOG_INFO, LOC +
+            LOG(VB_GENERAL, LOG_INFO, LOC +
                 QString("TuningSignalCheck: Still waiting.  Will timeout @ %1")
                 .arg(m_signalMonitorDeadline.toLocalTime()
                      .toString("hh:mm:ss.zzz")));
