diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythtv/tv_rec.cpp	2019-03-29 08:40:18.626666667 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythtv/tv_rec.cpp	2019-03-29 08:40:20.670000001 +0100
@@ -3917,7 +3917,7 @@
     // Start signal monitoring for devices capable of monitoring
     if (use_sm)
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "Starting Signal Monitor");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Starting Signal Monitor");
         bool error = false;
         if (!SetupSignalMonitor(
                 !antadj, request.flags & kFlagEITScan, livetv || antadj))
@@ -3967,7 +3967,7 @@
                     m_signalMonitorDeadline =
                         m_curRecording->GetRecordingEndTime().addSecs(-10);
 
-                    LOG(VB_CHANNEL, LOG_DEBUG, LOC +
+                    LOG(VB_GENERAL, LOG_DEBUG, LOC +
                         QString("Pre-fail start deadline: %1 "
                                 "Start recording deadline: %2 "
                                 "Good signal deadline: %3")
@@ -4039,7 +4039,7 @@
 
     if (m_signalMonitor->IsAllGood())
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
         if (m_curRecording && (current_time > m_startRecordingDeadline))
         {
             newRecStatus = RecStatus::Failing;
@@ -4128,7 +4128,7 @@
             --m_signalMonitorCheckCnt;
         else
         {
-            LOG(VB_RECORD, LOG_INFO, LOC +
+            LOG(VB_GENERAL, LOG_INFO, LOC +
                 QString("TuningSignalCheck: Still waiting.  Will timeout @ %1")
                 .arg(m_signalMonitorDeadline.toLocalTime()
                      .toString("hh:mm:ss.zzz")));
