diff -Naur mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythtv/tv_rec.cpp	2018-10-20 14:57:08.913333334 +0200
+++ mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythtv/tv_rec.cpp	2018-10-20 14:57:11.720000001 +0200
@@ -3955,7 +3955,7 @@
     // Start signal monitoring for devices capable of monitoring
     if (use_sm)
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "Starting Signal Monitor");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Starting Signal Monitor");
         bool error = false;
         if (!SetupSignalMonitor(
                 !antadj, request.flags & kFlagEITScan, livetv | antadj))
@@ -4005,7 +4005,7 @@
                     signalMonitorDeadline =
                         curRecording->GetRecordingEndTime();
 
-                    LOG(VB_CHANNEL, LOG_DEBUG, LOC +
+                    LOG(VB_GENERAL, LOG_DEBUG, LOC +
                         QString("Pre-fail start deadline: %1 "
                                 "Start recording deadline: %2 "
                                 "Good signal deadline: %3")
@@ -4077,7 +4077,7 @@
 
     if (signalMonitor->IsAllGood())
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
         if (curRecording && (current_time > startRecordingDeadline))
         {
             newRecStatus = RecStatus::Failing;
@@ -4166,7 +4166,7 @@
             --signalMonitorCheckCnt;
         else
         {
-            LOG(VB_RECORD, LOG_INFO, LOC +
+            LOG(VB_GENERAL, LOG_INFO, LOC +
                 QString("TuningSignalCheck: Still waiting.  Will timeout @ %1")
                 .arg(signalMonitorDeadline.toLocalTime()
                      .toString("hh:mm:ss.zzz")));
