diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythtv/decoders/avformatdecoder.h mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythtv/decoders/avformatdecoder.h
--- mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2020-09-04 11:35:36.326666667 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2020-09-04 11:35:37.646666667 +0200
@@ -76,10 +76,11 @@
     }
     QString toString() const
     {
-        return QString("id(%1) %2Hz %3ch %4bps %5 (profile %6)")
+        return QString("id(%1) %2Hz %3/%7ch %4bps %5 (profile %6)")
             .arg(ff_codec_id_string(m_codecId),4).arg(m_sampleRate,6)
             .arg(m_channels,2).arg(AudioOutputSettings::FormatToBits(format),2)
-            .arg((m_doPassthru) ? "pt":"",3).arg(m_codecProfile);
+            .arg((m_doPassthru) ? "pt":"",3).arg(m_codecProfile)
+            .arg(m_originalChannels);
     }
 };
 
