diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/avformatdecoder.h mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/avformatdecoder.h
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/avformatdecoder.h	2018-10-12 10:52:12.713333332 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/avformatdecoder.h	2018-10-12 10:52:15.599999998 +0200
@@ -75,10 +75,11 @@
     }
     QString toString() const
     {
-        return QString("id(%1) %2Hz %3ch %4bps %5 (profile %6)")
+        return QString("id(%1) %2Hz %3/%7ch %4bps %5 (profile %6)")
             .arg(ff_codec_id_string(codec_id),4).arg(sample_rate,6)
             .arg(channels,2).arg(AudioOutputSettings::FormatToBits(format),2)
-            .arg((do_passthru) ? "pt":"",3).arg(codec_profile);
+            .arg((do_passthru) ? "pt":"",3).arg(codec_profile)
+            .arg(original_channels);
     }
 };
 
