diff -Naur mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmythtv/decoders/avformatdecoder.h mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmythtv/decoders/avformatdecoder.h
--- mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2020-04-09 12:25:15.107589677 +0200
+++ mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2020-04-09 12:25:16.780922950 +0200
@@ -74,10 +74,11 @@
     }
     QString toString() const
     {
-        return QString("id(%1) %2Hz %3ch %4bps %5 (profile %6)")
+        return QString("id(%1) %2Hz %3/%7ch %4bps %5 (profile %6)")
             .arg(ff_codec_id_string(m_codecId),4).arg(m_sampleRate,6)
             .arg(m_channels,2).arg(AudioOutputSettings::FormatToBits(format),2)
-            .arg((m_doPassthru) ? "pt":"",3).arg(m_codecProfile);
+            .arg((m_doPassthru) ? "pt":"",3).arg(m_codecProfile)
+            .arg(m_originalChannels);
     }
 };
 
