diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/decoders/avformatdecoder.h mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/decoders/avformatdecoder.h
--- mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2020-01-14 09:19:31.516666666 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2020-01-14 09:19:33.483333332 +0100
@@ -74,10 +74,11 @@
     }
     QString toString() const
     {
-        return QString("id(%1) %2Hz %3ch %4bps %5 (profile %6)")
+        return QString("id(%1) %2Hz %3/%7ch %4bps %5 (profile %6)")
             .arg(ff_codec_id_string(m_codecId),4).arg(m_sampleRate,6)
             .arg(m_channels,2).arg(AudioOutputSettings::FormatToBits(format),2)
-            .arg((m_doPassthru) ? "pt":"",3).arg(m_codecProfile);
+            .arg((m_doPassthru) ? "pt":"",3).arg(m_codecProfile)
+            .arg(m_originalChannels);
     }
 };
 
