diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-05-07 12:46:00.973333333 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-05-07 12:46:02.590000000 +0200
@@ -510,6 +510,11 @@
 
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
+        LOG(VB_GENERAL, LOG_WARNING, LOC +
+            QString("Wrong SDT; onid %1 tsid %2, desired %3 %4\n"
+                    "\t\t\tRescan your transports.")
+                .arg(m_detectedNetworkID).arg(m_detectedTransportID)
+                .arg(m_networkID).arg(m_transportID) );
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
     }
     else
