diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-09-15 09:22:51.363333335 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-09-15 09:22:52.693333335 +0200
@@ -525,6 +525,11 @@
 
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
+        LOG(VB_GENERAL, LOG_WARNING, LOC +
+            QString("Wrong SDT; onid %1 tsid %2, desired %3 %4\n"
+                    "\t\t\tRescan your transports.")
+                .arg(m_detectedNetworkID).arg(m_detectedTransportID)
+                .arg(m_networkID).arg(m_transportID) );
         DVBStreamData *dvb = GetDVBStreamData();
         if (dvb)
             dvb->SetVersionSDT(sdt->TSID(), -1, 0);
