diff -Naur mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-09-21 09:18:01.166666668 +0200
+++ mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-09-21 09:18:02.793333335 +0200
@@ -517,6 +517,11 @@
 
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
+        LOG(VB_GENERAL, LOG_WARNING, LOC +
+            QString("Wrong SDT; onid %1 tsid %2, desired %3 %4\n"
+                    "\t\t\tRescan your transports.")
+                .arg(m_detectedNetworkID).arg(m_detectedTransportID)
+                .arg(m_networkID).arg(m_transportID) );
         DVBStreamData *dvb = GetDVBStreamData();
         if (dvb)
             dvb->SetVersionSDT(sdt->TSID(), -1, 0);
