diff -Naur mythtv-master-20200114-ga315cd6-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-master-20200114-ga315cd6-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-master-20200114-ga315cd6-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2020-01-14 09:19:34.706666666 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2020-01-14 09:19:35.876666666 +0100
@@ -164,6 +164,12 @@
             if (m_metaAlbumPos > 0)
                 result["album"] = rx.cap(m_metaAlbumPos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_metaFormat) );
+        }
     }
 
     return result;
