diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-master-20210225-gc0426e3fea-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2021-03-04 20:25:32.439999619 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2021-03-04 20:25:37.416666284 +0100
@@ -165,6 +165,12 @@
             if (m_metaAlbumPos > 0)
                 result["album"] = rx.cap(m_metaAlbumPos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_metaFormat) );
+        }
     }
 
     return result;
