diff -Naur mythtv-master-20191227-gf15b4f0-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-master-20191227-gf15b4f0-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2019-12-27 12:19:02.953333333 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2019-12-27 12:19:04.096666667 +0100
@@ -164,6 +164,12 @@
             if (m_metaAlbumPos > 0)
                 result["album"] = rx.cap(m_metaAlbumPos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_metaFormat) );
+        }
     }
 
     return result;
