diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2018-10-12 10:52:17.743333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2018-10-12 10:52:20.626666666 +0200
@@ -163,6 +163,12 @@
             if (m_meta_album_pos > 0)
                 result["album"] = rx.cap(m_meta_album_pos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_meta_format) );
+        }
     }
 
     return result;
