diff -Naur mythtv-master-20200924-gb00d7c3-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-master-20200924-gb00d7c3-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-master-20200924-gb00d7c3-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2020-09-25 09:21:12.003333335 +0200
+++ mythtv-master-20200924-gb00d7c3-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2020-09-25 09:21:13.973333334 +0200
@@ -164,6 +164,12 @@
             if (m_metaAlbumPos > 0)
                 result["album"] = rx.cap(m_metaAlbumPos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_metaFormat) );
+        }
     }
 
     return result;
