diff -Naur mythtv-master-20200207-g2869463-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-master-20200207-g2869463-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-master-20200207-g2869463-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2020-02-08 14:18:40.070000002 +0100
+++ mythtv-master-20200207-g2869463-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2020-02-08 14:18:41.743333335 +0100
@@ -164,6 +164,12 @@
             if (m_metaAlbumPos > 0)
                 result["album"] = rx.cap(m_metaAlbumPos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_metaFormat) );
+        }
     }
 
     return result;
