diff -Naur mythtv-master-20200418-g8760389-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-master-20200418-g8760389-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-master-20200418-g8760389-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2020-04-18 10:54:03.643333332 +0200
+++ mythtv-master-20200418-g8760389-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2020-04-18 10:54:05.026666666 +0200
@@ -164,6 +164,12 @@
             if (m_metaAlbumPos > 0)
                 result["album"] = rx.cap(m_metaAlbumPos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_metaFormat) );
+        }
     }
 
     return result;
