diff -Naur mythtv-master-20210424-g69d12b67b6-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-master-20210424-g69d12b67b6-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-master-20210424-g69d12b67b6-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2021-04-24 16:56:11.826665707 +0200
+++ mythtv-master-20210424-g69d12b67b6-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2021-04-24 16:56:16.449999040 +0200
@@ -166,6 +166,12 @@
             if (m_metaAlbumPos > 0)
                 result["album"] = match.captured(m_metaAlbumPos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_metaFormat) );
+        }
     }
 
     return result;
