diff -Naur mythtv-master-20210424-g69d12b67b6-old/mythtv/libs/libmythbase/mythdb.cpp mythtv-master-20210424-g69d12b67b6-new/mythtv/libs/libmythbase/mythdb.cpp
--- mythtv-master-20210424-g69d12b67b6-old/mythtv/libs/libmythbase/mythdb.cpp	2021-04-24 16:56:15.689999040 +0200
+++ mythtv-master-20210424-g69d12b67b6-new/mythtv/libs/libmythbase/mythdb.cpp	2021-04-24 16:56:20.809999040 +0200
@@ -424,6 +424,11 @@
     if (query.exec() && query.next())
     {
         value = query.value(0).toString();
+
+        LOG(VB_DATABASE, LOG_INFO,
+            QString("GetSetting: DB for:'%1' for host:'%2' returns:'%3'")
+            .arg(key).arg(d->m_localhostname).arg(value));
+
     }
     else
     {
@@ -436,6 +441,11 @@
         if (query.exec() && query.next())
         {
             value = query.value(0).toString();
+
+            LOG(VB_DATABASE, LOG_INFO,
+                QString("GetSetting: no:'%1' found for:'%2'. Asking with NULL hostname returns:'%3'")
+                .arg(key).arg(d->m_localhostname).arg(value));
+
         }
     }
 
