diff -Naur mythtv-master-20200116-gdd96d68-old/mythtv/libs/libmythbase/mythdb.cpp mythtv-master-20200116-gdd96d68-new/mythtv/libs/libmythbase/mythdb.cpp
--- mythtv-master-20200116-gdd96d68-old/mythtv/libs/libmythbase/mythdb.cpp	2020-01-16 10:18:16.746666668 +0100
+++ mythtv-master-20200116-gdd96d68-new/mythtv/libs/libmythbase/mythdb.cpp	2020-01-16 10:20:26.156666648 +0100
@@ -409,6 +409,11 @@
     if (query.exec() && query.next())
     {
         value = query.value(0).toString();
+
+        LOG(VB_DATABASE, LOG_INFO,
+            QString("GetSetting: DB for:'%1' for host:'%2' returns:'%3'")
+            .arg(key).arg(d->m_localhostname).arg(value));
+
     }
     else
     {
@@ -421,6 +426,11 @@
         if (query.exec() && query.next())
         {
             value = query.value(0).toString();
+
+            LOG(VB_DATABASE, LOG_INFO,
+                QString("GetSetting: no:'%1' found for:'%2'. Asking with NULL hostname returns:'%3'")
+                .arg(key).arg(d->m_localhostname).arg(value));
+
         }
     }
 
