diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythbase/portchecker.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythbase/portchecker.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythbase/portchecker.cpp	2021-05-18 16:11:42.736666605 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythbase/portchecker.cpp	2021-05-18 16:11:45.799999938 +0200
@@ -180,18 +180,18 @@
         }
         else
             retryCount++;
-        // This retry count of 6 means 3 seconds of waiting for
+        // This retry count of 30 means 3 seconds of waiting for
         // connection before aborting and starting a new connection attempt.
-        if (retryCount > 6)
+        if (retryCount > 30)
             socket.abort();
         processEvents();
         // Check if user got impatient and canceled
         if (m_cancelCheck)
             break;
-        std::this_thread::sleep_for(500ms);
+        std::this_thread::sleep_for(100ms);
         state = socket.state();
-        LOG(VB_GENERAL, LOG_DEBUG, LOC + QString("socket state %1")
-            .arg(state));
+        LOG(VB_GENERAL, LOG_DEBUG, LOC + QString("host %1 port %2 socket state %3")
+            .arg(host).arg(port).arg(state));
         if (linkLocalOnly
           && state == QAbstractSocket::UnconnectedState
           && testedAll)
