diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythbase/portchecker.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythbase/portchecker.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythbase/portchecker.cpp	2021-02-25 16:40:36.000000000 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythbase/portchecker.cpp	2021-03-04 17:24:20.876666640 +0100
@@ -182,18 +182,18 @@
         }
         else
             retryCount++;
-        // This retry count of 6 means 3 seconds of waiting for
+        // This retry count of 30 means 3 seconds of waiting for
         // connection before aborting and starting a new connection attempt.
-        if (retryCount > 6)
+        if (retryCount > 30)
             socket.abort();
         processEvents();
         // Check if user got impatient and canceled
         if (m_cancelCheck)
             break;
-        std::this_thread::sleep_for(500ms);
+        std::this_thread::sleep_for(100ms);
         state = socket.state();
-        LOG(VB_GENERAL, LOG_DEBUG, LOC + QString("socket state %1")
-            .arg(state));
+        LOG(VB_GENERAL, LOG_DEBUG, LOC + QString("host %1 port %2 socket state %3")
+            .arg(host).arg(port).arg(state));
         if (linkLocalOnly
           && state == QAbstractSocket::UnconnectedState
           && testedAll)
