diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20181012-g4651002-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20181012-g4651002-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2018-10-12 10:52:20.399999999 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2018-10-12 10:52:23.623333333 +0200
@@ -463,6 +463,11 @@
 
     // get the MMAP path
     m_mmapPath = getZMSetting("ZM_PATH_MAP");
+    if (m_mmapPath == "")
+    {
+        m_mmapPath = "/dev/shm";
+    }
+
     if (m_debug)
     {
         cout << "Memory path directory is: " << m_mmapPath << endl;
