diff -Naur mythtv-master-20190408-g038e171-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20190408-g038e171-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20190408-g038e171-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-04-08 14:45:12.576666667 +0200
+++ mythtv-master-20190408-g038e171-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-04-08 14:45:14.263333333 +0200
@@ -452,6 +452,11 @@
 
     // get the MMAP path
     m_mmapPath = getZMSetting("ZM_PATH_MAP");
+    if (m_mmapPath == "")
+    {
+        m_mmapPath = "/dev/shm";
+    }
+
     if (m_debug)
     {
         cout << "Memory path directory is: " << m_mmapPath << endl;
