diff -Naur mythtv-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2017-11-17 14:16:48.000000000 +0100
+++ mythtv-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2018-05-06 12:57:23.000000000 +0200
@@ -463,6 +463,11 @@
 
     // get the MMAP path
     m_mmapPath = getZMSetting("ZM_PATH_MAP");
+    if (m_mmapPath == "")
+    {
+        m_mmapPath = "/dev/shm";
+    }
+
     if (m_debug)
     {
         cout << "Memory path directory is: " << m_mmapPath << endl;
