diff -Naur mythtv-master-20190328-gdcb0847-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20190328-gdcb0847-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20190328-gdcb0847-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-03-29 08:40:24.623333334 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-03-29 08:40:26.146666667 +0100
@@ -452,6 +452,11 @@
 
     // get the MMAP path
     m_mmapPath = getZMSetting("ZM_PATH_MAP");
+    if (m_mmapPath == "")
+    {
+        m_mmapPath = "/dev/shm";
+    }
+
     if (m_debug)
     {
         cout << "Memory path directory is: " << m_mmapPath << endl;
