diff -Naur mythtv-master-20190216-g938606f-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20190216-g938606f-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20190216-g938606f-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-02-17 16:58:17.370000002 +0100
+++ mythtv-master-20190216-g938606f-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-02-17 16:58:18.576666669 +0100
@@ -454,6 +454,11 @@
 
     // get the MMAP path
     m_mmapPath = getZMSetting("ZM_PATH_MAP");
+    if (m_mmapPath == "")
+    {
+        m_mmapPath = "/dev/shm";
+    }
+
     if (m_debug)
     {
         cout << "Memory path directory is: " << m_mmapPath << endl;
