diff -Naur mythtv-master-20181019-g402f35a-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20181019-g402f35a-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20181019-g402f35a-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2018-10-20 14:57:21.323333334 +0200
+++ mythtv-master-20181019-g402f35a-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2018-10-20 14:57:24.653333334 +0200
@@ -463,6 +463,11 @@
 
     // get the MMAP path
     m_mmapPath = getZMSetting("ZM_PATH_MAP");
+    if (m_mmapPath == "")
+    {
+        m_mmapPath = "/dev/shm";
+    }
+
     if (m_debug)
     {
         cout << "Memory path directory is: " << m_mmapPath << endl;
