diff -Naur mythtv-master-20190320-gb121bce-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20190320-gb121bce-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20190320-gb121bce-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-03-20 22:45:36.703333333 +0100
+++ mythtv-master-20190320-gb121bce-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-03-20 22:45:38.713333334 +0100
@@ -453,6 +453,11 @@
 
     // get the MMAP path
     m_mmapPath = getZMSetting("ZM_PATH_MAP");
+    if (m_mmapPath == "")
+    {
+        m_mmapPath = "/dev/shm";
+    }
+
     if (m_debug)
     {
         cout << "Memory path directory is: " << m_mmapPath << endl;
