diff -Naur mythtv-master-20190316-g1a44f54-old/mythplugins/mythzoneminder/mythzmserver/main.cpp mythtv-master-20190316-g1a44f54-new/mythplugins/mythzoneminder/mythzmserver/main.cpp
--- mythtv-master-20190316-g1a44f54-old/mythplugins/mythzoneminder/mythzmserver/main.cpp	2019-03-16 10:53:08.976666667 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythplugins/mythzoneminder/mythzmserver/main.cpp	2019-03-16 10:53:10.693333334 +0100
@@ -213,11 +213,11 @@
     {
         cout << "ZM is version '" << g_zmversion << "'" << endl;
     }
-    else
-    {
-        cout << "This version of ZM is to old you need 1.24.0 or later '" << g_zmversion << "'" << endl;
-        return EXIT_VERSION_ERROR;
-    }
+    //else
+    //{
+    //    cout << "This version of ZM is to old you need 1.24.0 or later '" << g_zmversion << "'" << endl;
+    //    return EXIT_VERSION_ERROR;
+    //}
 
     // connect to the DB
     connectToDatabase();
