diff -Naur mythtv-master-20190408-g038e171-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20190408-g038e171-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20190408-g038e171-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-04-08 14:45:14.950000000 +0200
+++ mythtv-master-20190408-g038e171-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-04-08 14:45:16.870000000 +0200
@@ -25,6 +25,7 @@
 #include <sys/stat.h>
 #include <sys/shm.h>
 #include <sys/mman.h>
+#include <algorithm>
 
 #ifdef linux
 #  include <sys/vfs.h>
@@ -1085,9 +1086,13 @@
     int frameNo = atoi(tokens[3].c_str());
     string eventTime(tokens[4]);
 
+    std::replace( eventTime.begin(), eventTime.end(), '/', '-');
+    std::string eventDate = eventTime.substr(0, 8);
+
     if (m_debug)
         cout << "Getting frame " << frameNo << " for event " << eventID
-             << " on monitor " << monitorID  << " event time is " << eventTime << endl;
+             << " on monitor " << monitorID  << " event time is " << eventTime
+             << " event date is " << eventDate << endl;
 
     string outStr;
 
@@ -1099,7 +1104,7 @@
 
     if (m_useDeepStorage)
     {
-        filepath = g_webPath + "/events/" + monitorID + "/" + eventTime + "/";
+        filepath = g_webPath + "/events/" + monitorID + "/20" + eventDate + "/" + eventID + "/" ;
         sprintf(str, m_eventFileFormat.c_str(), frameNo);
         filepath += str;
     }
@@ -1152,9 +1157,13 @@
     int frameID;
     int frameCount = 0;
 
+    std::replace( eventTime.begin(), eventTime.end(), '/', '-');
+    std::string eventDate = eventTime.substr(0, 8);
+
     if (m_debug)
         cout << "Getting analysis frame " << frameNo << " for event " << eventID
-             << " on monitor " << monitorID << " event time is " << eventTime << endl;
+             << " on monitor " << monitorID << " event time is " << eventTime
+             << " event date is " << eventDate << endl;
 
     // get the 'alarm' frames from the Frames table for this event
     MYSQL_RES *res;
@@ -1232,7 +1241,7 @@
     string frameFile;
 
     if (m_useDeepStorage)
-        filepath = g_webPath + "/events/" + monitorID + "/" + eventTime + "/";
+        filepath = g_webPath + "/events/" + monitorID + "/20" + eventDate + "/" + eventID + "/" ;
     else
         filepath = g_webPath + "/events/" + monitorID + "/" + eventID + "/";
 
