diff -Naur mythtv-master-20181019-g402f35a-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20181019-g402f35a-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20181019-g402f35a-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2018-10-20 14:57:27.043333334 +0200
+++ mythtv-master-20181019-g402f35a-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2018-10-20 14:57:29.883333335 +0200
@@ -25,6 +25,7 @@
 #include <sys/stat.h>
 #include <sys/shm.h>
 #include <sys/mman.h>
+#include <algorithm>
 
 #ifdef linux
 #  include <sys/vfs.h>
@@ -1102,9 +1103,13 @@
     int frameNo = atoi(tokens[3].c_str());
     string eventTime(tokens[4]);
 
+    std::replace( eventTime.begin(), eventTime.end(), '/', '-');
+    std::string eventDate = eventTime.substr(0, 8);
+
     if (m_debug)
         cout << "Getting frame " << frameNo << " for event " << eventID
-             << " on monitor " << monitorID  << " event time is " << eventTime << endl;
+             << " on monitor " << monitorID  << " event time is " << eventTime
+             << " event date is " << eventDate << endl;
 
     string outStr("");
 
@@ -1116,7 +1121,7 @@
 
     if (m_useDeepStorage)
     {
-        filepath = g_webPath + "/events/" + monitorID + "/" + eventTime + "/";
+        filepath = g_webPath + "/events/" + monitorID + "/20" + eventDate + "/" + eventID + "/" ;
         sprintf(str, m_eventFileFormat.c_str(), frameNo);
         filepath += str;
     }
@@ -1169,9 +1174,13 @@
     int frameID;
     int frameCount = 0;
 
+    std::replace( eventTime.begin(), eventTime.end(), '/', '-');
+    std::string eventDate = eventTime.substr(0, 8);
+
     if (m_debug)
         cout << "Getting analysis frame " << frameNo << " for event " << eventID
-             << " on monitor " << monitorID << " event time is " << eventTime << endl;
+             << " on monitor " << monitorID << " event time is " << eventTime
+             << " event date is " << eventDate << endl;
 
     // get the 'alarm' frames from the Frames table for this event
     MYSQL_RES *res;
@@ -1249,7 +1258,7 @@
     string frameFile("");
 
     if (m_useDeepStorage)
-        filepath = g_webPath + "/events/" + monitorID + "/" + eventTime + "/";
+        filepath = g_webPath + "/events/" + monitorID + "/20" + eventDate + "/" + eventID + "/" ;
     else
         filepath = g_webPath + "/events/" + monitorID + "/" + eventID + "/";
 
