diff -Naur mythtv-0.25-old/mythplugins/mythweb/modules/tv/tmpl/default/schedules.php mythtv-0.25-new/mythplugins/mythweb/modules/tv/tmpl/default/schedules.php
--- mythtv-0.25-old/mythplugins/mythweb/modules/tv/tmpl/default/schedules.php	2012-04-10 13:57:10.000000000 +0200
+++ mythtv-0.25-new/mythplugins/mythweb/modules/tv/tmpl/default/schedules.php	2012-05-16 19:02:47.867184947 +0200
@@ -85,6 +85,12 @@
 </tr>
 </thead>
 <?php
+
+	// Hack:  Disable warnings about "Creating default
+	// object from empty value" from below line
+	// "$schedule->channel->name = '[ '.t('Any').' ]';"
+	ini_set('error_reporting', E_ERROR);
+
         $prev_group = '';
         $cur_group  = '';
         foreach ($the_schedules as $schedule) {
