diff -Naur mythtv-master-build-old/mythplugins/mythweb/modules/_shared/lang/English.lang mythtv-master-build-new/mythplugins/mythweb/modules/_shared/lang/English.lang
--- mythtv-master-build-old/mythplugins/mythweb/modules/_shared/lang/English.lang	2016-12-07 21:46:06.000000000 +0100
+++ mythtv-master-build-new/mythplugins/mythweb/modules/_shared/lang/English.lang	2016-12-07 21:39:00.000000000 +0100
@@ -93,6 +93,7 @@
 "Audio Bitrate"
 "Audio for Hearing Impaired"
 "Audio for Visually Impaired"
+"Auth"
 "Auto Expire"
 "Auto-expire recordings"
 "Auto-flag commercials"
@@ -117,6 +118,7 @@
 "Browse All Albums"
 "Browse Music"
 "Browse the Music Database"
+"BW"
 "By Artist Beginning With"
 "By Genre"
 "Cache_APC"
@@ -169,6 +171,7 @@
 "commfree"
 "Conflict"
 "Conflicts"
+"Constel."
 "Context"
 "contrast"
 "Count"
@@ -224,6 +227,7 @@
 "Done, Unknown Status"
 "Down"
 "Download URL override"
+"DTV Mplex Info"
 "Dup Method"
 "Duplicate"
 "Duplicate Check method"
@@ -256,6 +260,7 @@
 "Everything"
 "Exec. Producer"
 "Extended Check"
+"FEC"
 "ffmpeg with MP3 support not detected"
 "File Size"
 "film"
@@ -282,6 +287,7 @@
 "Forget Old"
 "format help"
 "Found %s results for '%s'"
+"FREQ"
 "freqid"
 "Friday"
 "Frontends"
@@ -296,6 +302,7 @@
 "Go"
 "Google"
 "Group timeslots every"
+"Guard"
 "Guest Starring"
 "Guide rating"
 "Guide Settings"
@@ -308,11 +315,13 @@
 "HDTV"
 "Height"
 "Hide"
+"Hierarchy"
 "High definition"
 "Home"
 "Hosted by"
 "Hour"
 "Hour Format"
+"HP_code"
 "HTML5 Streaming"
 "hue"
 "Identifiable episode"
@@ -367,6 +376,7 @@
 "Input Name"
 "Insert"
 "Internet Reference #"
+"INV"
 "Jump"
 "Jump Points"
 "Jump To"
@@ -409,6 +419,7 @@
 "Logs"
 "Look up Metadata"
 "Low Space"
+"LP_code"
 "Manual"
 "Match related callsigns"
 "matches"
@@ -418,6 +429,7 @@
 "Metadata Lookup Error"
 "minutes"
 "Missing Cover"
+"MOD."
 "Modify priority by star rating (0.0 to 1.0 for movies only)"
 "Modify priority for a station on an input"
 "Modify priority for an input (Input priority)"
@@ -467,6 +479,7 @@
 "New Group"
 "New Search"
 "New Titles, Premieres"
+"NID"
 "No"
 "NO DATA"
 "No episodes"
@@ -551,6 +564,7 @@
 "Please wait for the pending AJAX request"
 "Plot"
 "Plot:"
+"POL."
 "Pop-out player"
 "Possible conflicts"
 "Possible conflicts with this show"
@@ -704,6 +718,7 @@
 "Return to Statistics Page"
 "Right"
 "right"
+"RollOff"
 "Root Directory"
 "Rows to show between timeslot info"
 "Running"
@@ -776,6 +791,7 @@
 "Showing all programs."
 "Showings between"
 "Shows"
+"SI"
 "Skip Ahead"
 "Skip Back"
 "Songs"
@@ -788,6 +804,7 @@
 "sourceid"
 "Space"
 "Special Searches"
+"SR"
 "Star character"
 "Start Date"
 "Start Early"
@@ -828,6 +845,7 @@
 "This time"
 "This will overwrite your current, unsaved playlist. Are you sure you want to continue?"
 "Thursday"
+"TID"
 "Time"
 "Time Span"
 "Time Stretch"
@@ -851,6 +869,7 @@
 "Total Running Time"
 "Total Time"
 "Total Time:  %s"
+"TR_mode"
 "Track #%s from the album '%s'"
 "Trakt.tv Home Page"
 "Transcode"
@@ -876,6 +895,7 @@
 "Upcoming Recordings"
 "Update"
 "Update Recording Settings"
+"Updated"
 "Use callsign"
 "useonairguide"
 "User Rating"
@@ -921,6 +941,7 @@
 "View Details of %s"
 "View Saved Playlist"
 "View Server Statistics"
+"Visible"
 "visible"
 "Visit $1"
 "Watched"
diff -Naur mythtv-master-build-old/mythplugins/mythweb/modules/_shared/lang/Polish.lang mythtv-master-build-new/mythplugins/mythweb/modules/_shared/lang/Polish.lang
--- mythtv-master-build-old/mythplugins/mythweb/modules/_shared/lang/Polish.lang	2016-12-07 21:46:06.000000000 +0100
+++ mythtv-master-build-new/mythplugins/mythweb/modules/_shared/lang/Polish.lang	2016-12-07 21:39:00.000000000 +0100
@@ -116,6 +116,7 @@
 "Audio Bitrate"
 "Audio for Hearing Impaired"
 "Audio for Visually Impaired"
+"Auth"
 "Auto Expire"
 "Auto-expire recordings"
     nadpisz nagrania
@@ -148,6 +149,7 @@
 "Browse All Albums"
 "Browse Music"
 "Browse the Music Database"
+"BW"
 "By Artist Beginning With"
 "By Genre"
 "Cache_APC"
@@ -210,6 +212,7 @@
 "Conflict"
 "Conflicts"
     konflikty
+"Constel."
 "Context"
     Kontekst
 "contrast"
@@ -282,6 +285,7 @@
 "Done, Unknown Status"
 "Down"
 "Download URL override"
+"DTV Mplex Info"
 "Dup Method"
 "Duplicate"
 "Duplicate Check method"
@@ -318,6 +322,7 @@
 "Everything"
 "Exec. Producer"
 "Extended Check"
+"FEC"
 "ffmpeg with MP3 support not detected"
 "File Size"
 "film"
@@ -347,6 +352,7 @@
 "Forget Old"
 "format help"
 "Found %s results for '%s'"
+"FREQ"
 "freqid"
 "Friday"
     Piątek
@@ -363,6 +369,7 @@
     Idź
 "Google"
 "Group timeslots every"
+"Guard"
 "Guest Starring"
 "Guide rating"
 "Guide Settings"
@@ -375,6 +382,7 @@
 "HDTV"
 "Height"
 "Hide"
+"Hierarchy"
 "High definition"
 "Home"
 "Hosted by"
@@ -383,6 +391,7 @@
     Godzina
 "Hour Format"
     Format godziny
+"HP_code"
 "HTML5 Streaming"
 "hue"
     kolor
@@ -427,6 +436,7 @@
 "Input Name"
 "Insert"
 "Internet Reference #"
+"INV"
 "Jump"
     Skocz
 "Jump Points"
@@ -477,6 +487,7 @@
     Logi
 "Look up Metadata"
 "Low Space"
+"LP_code"
 "Manual"
     Ręczny
 "Match related callsigns"
@@ -488,6 +499,7 @@
 "minutes"
     minut
 "Missing Cover"
+"MOD."
 "Modify priority by star rating (0.0 to 1.0 for movies only)"
 "Modify priority for a station on an input"
 "Modify priority for an input (Input priority)"
@@ -545,6 +557,7 @@
 "New Search"
 "New Titles, Premieres"
     Nowe tytuły, premiery
+"NID"
 "No"
 "NO DATA"
 "No episodes"
@@ -630,6 +643,7 @@
 "Please wait for the pending AJAX request"
 "Plot"
 "Plot:"
+"POL."
 "Pop-out player"
 "Possible conflicts"
 "Possible conflicts with this show"
@@ -796,6 +810,7 @@
 "Return to Statistics Page"
 "Right"
 "right"
+"RollOff"
 "Root Directory"
 "Rows to show between timeslot info"
 "Running"
@@ -875,6 +890,7 @@
 "Showing all programs."
 "Showings between"
 "Shows"
+"SI"
 "Skip Ahead"
 "Skip Back"
 "Songs"
@@ -886,6 +902,7 @@
 "sourceid"
 "Space"
 "Special Searches"
+"SR"
 "Star character"
 "Start Date"
     Data rozpoczęcia
@@ -933,6 +950,7 @@
 "This will overwrite your current, unsaved playlist. Are you sure you want to continue?"
 "Thursday"
     Czwartek
+"TID"
 "Time"
     Czas
 "Time Span"
@@ -962,6 +980,7 @@
 "Total Time"
     Całkowity Czas
 "Total Time:  %s"
+"TR_mode"
 "Track #%s from the album '%s'"
 "Trakt.tv Home Page"
 "Transcode"
@@ -996,6 +1015,7 @@
     Uaktualnij
 "Update Recording Settings"
     Uaktualnij Ustawienia Nagrywania
+"Updated"
 "Use callsign"
     Użyj CALLSIGN?
 "useonairguide"
@@ -1037,6 +1057,7 @@
 "View Details of %s"
 "View Saved Playlist"
 "View Server Statistics"
+"Visible"
 "visible"
     Widoczny
 "Visit $1"
diff -Naur mythtv-master-build-old/mythplugins/mythweb/modules/tv/init.php mythtv-master-build-new/mythplugins/mythweb/modules/tv/init.php
--- mythtv-master-build-old/mythplugins/mythweb/modules/tv/init.php	2016-12-07 21:46:05.000000000 +0100
+++ mythtv-master-build-new/mythplugins/mythweb/modules/tv/init.php	2016-12-02 12:11:00.000000000 +0100
@@ -20,6 +20,7 @@
         $Settings['tv'] = array('name'    => t('TV'),
                                 'choices' => array('session'    => t('My Session'),
                                                    'channels'   => t('Channel Info'),
+                                                   'multiplexes' => t('DTV Mplex Info'),
                                                    'playgroup'  => t('Playback Groups'),
                                                    'screens'    => t('Customize Screens'),
                                                   ),
diff -Naur mythtv-master-build-old/mythplugins/mythweb/modules/tv/set_multiplexes.php mythtv-master-build-new/mythplugins/mythweb/modules/tv/set_multiplexes.php
--- mythtv-master-build-old/mythplugins/mythweb/modules/tv/set_multiplexes.php	1970-01-01 01:00:00.000000000 +0100
+++ mythtv-master-build-new/mythplugins/mythweb/modules/tv/set_multiplexes.php	2016-12-07 21:33:00.000000000 +0100
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Configure MythTV Channels
+ *
+ * @license     GPL
+ *
+ * @package     MythWeb
+ * @subpackage  Settings
+ *
+ **/
+
+// Save?
+    if ($_POST['save']) {
+    // Parse the post variables and save each group of mplex info
+        foreach ($_POST['mplex'] as $mplexid => $data) {
+
+            if ($data['delete'] == 'true')
+                $db->query('DELETE FROM dtv_multiplex WHERE mplexid=?', $mplexid);
+            else {
+                $query_params = array();
+            // Not deleting so grab values that can be empty
+                $query = 'UPDATE dtv_multiplex SET sourceid          = ?,
+                                                   transportid       = ?,
+                                                   networkid         = ?,
+                                                   frequency         = ?,
+                                                   inversion         = ?,
+                                                   symbolrate        = ?,
+                                                   fec               = ?,
+                                                   polarity          = ?,
+                                                   modulation        = ?,
+                                                   bandwidth         = ?,
+                                                   lp_code_rate      = ?,
+                                                   transmission_mode = ?,
+                                                   guard_interval    = ?,
+                                                   visible           = ?,
+                                                   constellation     = ?,
+                                                   hierarchy         = ?,
+                                                   hp_code_rate      = ?,
+                                                   mod_sys           = ?,
+                                                   rolloff           = ?,
+                                                   sistandard        = ?,
+                                                   serviceversion    = ?,
+                                                   updatetimestamp   = ?,
+                                                   default_authority = ?';
+                $query_params[] = $data['sourceid'];
+                $query_params[] = $data['transportid'];
+                $query_params[] = $data['networkid'];
+                $query_params[] = $data['frequency'];
+                $query_params[] = $data['inversion'];
+                $query_params[] = $data['symbolrate'];
+                $query_params[] = $data['fec'];
+                $query_params[] = $data['polarity'];
+                $query_params[] = $data['modulation'];
+                $query_params[] = $data['bandwidth'];
+                $query_params[] = $data['lp_code_rate'];
+                $query_params[] = $data['transmission_mode'];
+                $query_params[] = $data['guard_interval'];
+                $query_params[] = $data['visible'];
+                $query_params[] = $data['constellation'];
+                $query_params[] = $data['hierarchy'];
+                $query_params[] = $data['hp_code_rate'];
+                $query_params[] = $data['mod_sys'];
+                $query_params[] = $data['rolloff'];
+                $query_params[] = $data['sistandard'];
+                $query_params[] = $data['serviceversion'];
+                $query_params[] = $data['updatetimestamp'];
+                $query_params[] = $data['default_authority'];
+
+                $db->query($query.' WHERE mplexid=?',
+                           $query_params,
+                           $mplexid
+                           );
+            }
+        }
+    }
+
+// Sortby
+    if ($_GET['sortby'])
+        $_SESSION['tv']['set']['chan_sort'] = $_GET['sortby'];
+    switch ($_SESSION['tv']['set']['chan_sort']) {
+        case 'callsign':
+        case 'name':
+            $sortby = $_SESSION['tv']['set']['chan_sort'];
+            break;
+        case 'channum':
+        case 'xmltvid':
+        case 'freqid':
+            $sortby = $_SESSION['tv']['set']['chan_sort'].' + 0, '.$_SESSION['tv']['set']['chan_sort'];
+            break;
+        case 'sourceid':
+            $sortby = 'sourceid';
+            break;
+        default:
+            $sortby = 'frequency';
+    }
+
+// Load all of the mplexes data from the database
+    $mplexes = array();
+//    $sh = $db->query('SELECT mplexid FROM dtv_multiplex ORDER BY '.$sortby);
+    $sh = $db->query('SELECT mplexid FROM dtv_multiplex ORDER BY frequency');
+    while ($row = $sh->fetch_col())
+        $mplexes[] = $row;
+    $sh->finish();
+
+// These settings affect all of mythtv
+    $Settings_Hosts = t('All Hosts');
diff -Naur mythtv-master-build-old/mythplugins/mythweb/modules/tv/tmpl/default/set_multiplexes.php mythtv-master-build-new/mythplugins/mythweb/modules/tv/tmpl/default/set_multiplexes.php
--- mythtv-master-build-old/mythplugins/mythweb/modules/tv/tmpl/default/set_multiplexes.php	1970-01-01 01:00:00.000000000 +0100
+++ mythtv-master-build-new/mythplugins/mythweb/modules/tv/tmpl/default/set_multiplexes.php	2016-12-07 21:40:00.000000000 +0100
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Configure MythTV Channel info
+ *
+ * @license     GPL
+ *
+ * @package     MythWeb
+ * @subpackage  Settings
+ *
+ **/
+?>
+<form class="form" method="post" action="<?php echo form_action ?>">
+
+<table border="0" cellpadding="4" cellspacing="2" class="list small">
+<tr class="menu" align="center">
+    <td width="4%"><?php  echo t('mplexid')   ?></td>
+    <td width="4%"><?php  echo t('sourceid')  ?></td>
+    <td width="5%"><?php  echo t('TID')       ?></td>
+    <td width="5%"><?php  echo t('NID')       ?></td>
+    <td width="5%"><?php  echo t('FREQ')      ?></td>
+    <td width="5%"><?php  echo t('INV')       ?></td>
+    <td width="5%"><?php  echo t('SR')        ?></td>
+    <td width="5%"><?php  echo t('FEC')       ?></td>
+    <td width="5%"><?php  echo t('POL.')      ?></td>
+    <td width="5%"><?php  echo t('MOD.')      ?></td>
+    <td width="5%"><?php  echo t('BW')        ?></td>
+    <td width="5%"><?php  echo t('LP_code')   ?></td>
+    <td width="5%"><?php  echo t('TR_mode')   ?></td>
+    <td width="5%"><?php  echo t('Guard')     ?></td>
+    <td width="5%"><?php  echo t('Visible')   ?></td>
+    <td width="5%"><?php  echo t('Constel.')  ?></td>
+    <td width="5%"><?php  echo t('Hierarchy') ?></td>
+    <td width="5%"><?php  echo t('HP_code')   ?></td>
+    <td width="5%"><?php  echo t('Type')      ?></td>
+    <td width="5%"><?php  echo t('RollOff')   ?></td>
+    <td width="5%"><?php  echo t('SI')        ?></td>
+    <td width="5%"><?php  echo t('Updated')   ?></td>
+    <td width="5%"><?php  echo t('Auth')      ?></td>
+</tr><?php
+    foreach ($mplexes as $mplexid) {
+        // Load mplex data from the database
+        $row = array();
+        $sh = $db->query('SELECT sourceid,
+                                 transportid,
+                                 networkid,
+                                 frequency,
+                                 inversion,
+                                 symbolrate,
+                                 fec,
+                                 polarity,
+                                 modulation,
+                                 bandwidth,
+                                 lp_code_rate,
+                                 transmission_mode,
+                                 guard_interval,
+                                 visible,
+                                 constellation,
+                                 hierarchy,
+                                 hp_code_rate,
+                                 mod_sys,
+                                 rolloff,
+                                 sistandard,
+                                 updatetimestamp,
+                                 default_authority FROM dtv_multiplex WHERE mplexid='.$mplexid);
+        $row = $sh->fetch_row();
+        $sh->finish();
+
+?><tr class="settings" align="center">
+    <td><?php echo html_entities($mplexid) ?></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][sourceid]"          id="sourceid_<?php echo $mplexid ?>"          value="<?php echo html_entities($row[0])     ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][transportid]"       id="transportid_<?php echo $mplexid ?>"       value="<?php echo html_entities($row[1])     ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][networkid]"         id="networkid_<?php echo $mplexid ?>"         value="<?php echo html_entities($row[2])     ?>" style="text-align: center" /></td>
+    <td><input type="text" size="10" name="mplex[<?php echo $mplexid ?>][frequency]"         id="frequency_<?php echo $mplexid ?>"         value="<?php echo html_entities($row[3])     ?>" style="text-align: center" /></td>
+    <td><input type="text" size="2"  name="mplex[<?php echo $mplexid ?>][inversion]"         id="inversion_<?php echo $mplexid ?>"         value="<?php echo html_entities($row[4])     ?>" style="text-align: center" /></td>
+    <td><input type="text" size="10" name="mplex[<?php echo $mplexid ?>][symbolrate]"        id="symbolrate_<?php echo $mplexid ?>"        value="<?php echo html_entities($row[5])     ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][fec]"               id="fec_<?php echo $mplexid ?>"               value="<?php echo html_entities($row[6])     ?>" style="text-align: center" /></td>
+    <td><input type="text" size="2"  name="mplex[<?php echo $mplexid ?>][polarity]"          id="polarity_<?php echo $mplexid ?>"          value="<?php echo html_entities($row[7])     ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][modulation]"        id="modulation_<?php echo $mplexid ?>"        value="<?php echo html_entities($row[8])     ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][bandwidth]"         id="bandwidth_<?php echo $mplexid ?>"         value="<?php echo html_entities($row[9])     ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][lp_code_rate]"      id="lp_code_rate_<?php echo $mplexid ?>"      value="<?php echo html_entities($row[10])    ?>" style="text-align: center" /></td>
+    <td><input type="text" size="2"  name="mplex[<?php echo $mplexid ?>][transmission_mode]" id="transmission_<?php echo $mplexid ?>"      value="<?php echo html_entities($row[11])    ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][guard_interval]"    id="guard_interval_<?php echo $mplexid ?>"    value="<?php echo html_entities($row[12])    ?>" style="text-align: center" /></td>
+    <td><input type="text" size="2"  name="mplex[<?php echo $mplexid ?>][visible]"           id="visible_<?php echo $mplexid ?>"           value="<?php echo html_entities($row[13])    ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][constellation]"     id="constellation_<?php echo $mplexid ?>"     value="<?php echo html_entities($row[14])    ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][hierarchy]"         id="hierarchy_<?php echo $mplexid ?>"         value="<?php echo html_entities($row[15])    ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][hp_code_rate]"      id="hp_code_rate_<?php echo $mplexid ?>"      value="<?php echo html_entities($row[16])    ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][mod_sys]"           id="mod_sys_<?php echo $mplexid ?>"           value="<?php echo html_entities($row[17])    ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][rolloff]"           id="rolloff_<?php echo $mplexid ?>"           value="<?php echo html_entities($row[18])    ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][sistandard]"        id="sistandard_<?php echo $mplexid ?>"        value="<?php echo html_entities($row[19])    ?>" style="text-align: center" /></td>
+    <td><input type="text" size="20" name="mplex[<?php echo $mplexid ?>][updatetimestamp]"   id="updatetimestamp_<?php echo $mplexid ?>"   value="<?php echo html_entities($row[20])    ?>" style="text-align: center" /></td>
+    <td><input type="text" size="5"  name="mplex[<?php echo $mplexid ?>][default_authority]" id="default_authority_<?php echo $mplexid ?>" value="<?php echo html_entities($row[21])    ?>" style="text-align: center" /></td>
+
+</tr><?php
+    }
+?>
+</table>
+
+<p align="center">
+<input type="submit" name="save" value="<?php echo t('Save') ?>">
+</p>
+
+</form>
