diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2015-01-06 16:51:59.970191365 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2015-01-06 16:52:22.333403219 +0100
@@ -729,6 +729,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/videodisplayprofile.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/videodisplayprofile.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/videodisplayprofile.cpp	2015-01-06 16:51:59.960192022 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/videodisplayprofile.cpp	2015-01-06 16:52:22.333403219 +0100
@@ -1183,6 +1183,16 @@
                   "");
 }
 
+void VideoDisplayProfile::CreateMiniMythProfiles(const QString &hostname)
+{
+    (void) QObject::tr("MiniMyth", "MiniMyth");
+    DeleteProfileGroup("MiniMyth", hostname);
+    uint groupid = CreateProfileGroup("MiniMyth", hostname);
+    CreateProfile(groupid, 1, ">", 0, 0, "", 0, 0,
+                  "ffmpeg", 1, true, "xv-blit", "softblend", false,
+                  "bobdeint", "onefield", "");
+}
+
 void VideoDisplayProfile::CreateProfiles(const QString &hostname)
 {
     CreateNewProfiles(hostname);
diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/videodisplayprofile.h mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/videodisplayprofile.h
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/videodisplayprofile.h	2015-01-06 16:51:59.946860010 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/videodisplayprofile.h	2015-01-06 16:52:22.333403219 +0100
@@ -150,6 +150,7 @@
     static void        CreateProfiles(const QString &hostname);
     static void        CreateNewProfiles(const QString &hostname);
     static void        CreateVDPAUProfiles(const QString &hostname);
+    static void        CreateMiniMythProfiles(const QString &hostname);
     static void        CreateVDAProfiles(const QString &hostname);
     static void        CreateOpenGLProfiles(const QString &hostname);
     static void        CreateVAAPIProfiles(const QString &hostname);
diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/globalsettings.cpp	2015-01-06 16:52:02.503510012 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/globalsettings.cpp	2015-01-06 16:52:22.340069850 +0100
@@ -1195,6 +1195,11 @@
     if (profiles.empty())
         return;
 
+    if (!profiles.contains("MiniMyth"))
+    {
+        VideoDisplayProfile::CreateMiniMythProfiles(host);
+        profiles = VideoDisplayProfile::GetProfiles(host);
+    }
     if (!profiles.contains("Normal") &&
         !profiles.contains("High Quality") &&
         !profiles.contains("Slim"))
