diff -Naur mythtv-master-20140506-gd0e25ac-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp mythtv-master-20140506-gd0e25ac-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp
--- mythtv-master-20140506-gd0e25ac-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2014-05-06 14:54:31.495268117 +0200
+++ mythtv-master-20140506-gd0e25ac-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2014-05-06 14:54:46.175216035 +0200
@@ -45,7 +45,7 @@
 #ifdef Q_OS_MAC
     gc->setValue(QDir::homePath() + "/Pictures");
 #else
-    gc->setValue("/var/lib/pictures");
+    gc->setValue("/mnt/gallery");
 #endif
     gc->setHelpText(GallerySettings::tr("This directory must exist and "
                                         "MythGallery needs to have read "
diff -Naur mythtv-master-20140506-gd0e25ac-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20140506-gd0e25ac-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20140506-gd0e25ac-old/mythtv/libs/libmythui/mythmainwindow.cpp	2014-05-06 14:54:24.205293950 +0200
+++ mythtv-master-20140506-gd0e25ac-new/mythtv/libs/libmythui/mythmainwindow.cpp	2014-05-06 14:54:46.175216035 +0200
@@ -1266,9 +1266,9 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc");
+        "Display System Exit Prompt"),      "");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
-        "System Exit"),                     "");
+        "System Exit"),                     "Ctrl+Esc");
 }
 
 void MythMainWindow::ReloadKeys()
diff -Naur mythtv-master-20140506-gd0e25ac-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20140506-gd0e25ac-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20140506-gd0e25ac-old/mythtv/programs/mythfrontend/globalsettings.cpp	2014-05-06 14:54:30.755270739 +0200
+++ mythtv-master-20140506-gd0e25ac-new/mythtv/programs/mythfrontend/globalsettings.cpp	2014-05-06 14:54:46.178549357 +0200
@@ -1641,7 +1641,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1713,7 +1713,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -1946,7 +1946,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -3002,7 +3002,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
