diff -Naur mythtv-master-20150105-g4216e00-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp mythtv-master-20150105-g4216e00-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp
--- mythtv-master-20150105-g4216e00-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2015-01-06 16:52:45.983275857 +0100
+++ mythtv-master-20150105-g4216e00-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2015-01-06 16:53:04.023178709 +0100
@@ -45,7 +45,7 @@
 #ifdef Q_OS_MAC
     gc->setValue(QDir::homePath() + "/Pictures");
 #else
-    gc->setValue("/var/lib/pictures");
+    gc->setValue("/mnt/gallery");
 #endif
     gc->setHelpText(GallerySettings::tr("This directory must exist and "
                                         "MythGallery needs to have read "
diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythui/mythmainwindow.cpp	2015-01-06 16:52:42.333295513 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythui/mythmainwindow.cpp	2015-01-06 16:53:04.023178709 +0100
@@ -1310,9 +1310,9 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc");
+        "Display System Exit Prompt"),      "");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
-        "System Exit"),                     "");
+        "System Exit"),                     "Ctrl+Esc");
     RegisterKey("Main Menu",    "STANDBYMODE",QT_TRANSLATE_NOOP("MythControls",
         "Enter Standby Mode"),              "");
 }
diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/globalsettings.cpp	2015-01-06 16:52:44.899948358 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/globalsettings.cpp	2015-01-06 16:53:04.023178709 +0100
@@ -1641,7 +1641,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1713,7 +1713,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -1947,7 +1947,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -3014,7 +3014,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/videoplayersettings.cpp mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/videoplayersettings.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/videoplayersettings.cpp	2015-01-06 16:52:44.919948250 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/videoplayersettings.cpp	2015-01-06 16:53:04.023178709 +0100
@@ -69,7 +69,7 @@
     m_dvdDriveEdit->SetText(gCoreContext->GetSetting("DVDDeviceLocation",
                            "default"));
     m_blurayMountEdit->SetText(gCoreContext->GetSetting("BluRayMountpoint",
-                           "/media/cdrom"));
+                           "/media/sr0"));
     m_altPlayerEdit->SetText(gCoreContext->GetSetting(
                            "mythvideo.VideoAlternatePlayer", "Internal"));
 
