diff -Naur mythtv-master-20140629-gb22277e-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp mythtv-master-20140629-gb22277e-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp
--- mythtv-master-20140629-gb22277e-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2014-06-29 12:41:03.000000000 +0200
+++ mythtv-master-20140629-gb22277e-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2014-06-29 12:51:23.081933369 +0200
@@ -45,7 +45,7 @@
 #ifdef Q_OS_MAC
     gc->setValue(QDir::homePath() + "/Pictures");
 #else
-    gc->setValue("/var/lib/pictures");
+    gc->setValue("/mnt/gallery");
 #endif
     gc->setHelpText(GallerySettings::tr("This directory must exist and "
                                         "MythGallery needs to have read "
diff -Naur mythtv-master-20140629-gb22277e-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20140629-gb22277e-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20140629-gb22277e-old/mythtv/libs/libmythui/mythmainwindow.cpp	2014-06-29 12:40:07.000000000 +0200
+++ mythtv-master-20140629-gb22277e-new/mythtv/libs/libmythui/mythmainwindow.cpp	2014-06-29 12:53:07.228216854 +0200
@@ -1306,9 +1306,9 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc");
+        "Display System Exit Prompt"),      "");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
-        "System Exit"),                     "");
+        "System Exit"),                     "Ctrl+Esc");
     RegisterKey("Main Menu",    "STANDBYMODE",QT_TRANSLATE_NOOP("MythControls",
         "Enter Standby Mode"),              "");
 }
diff -Naur mythtv-master-20140629-gb22277e-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20140629-gb22277e-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20140629-gb22277e-old/mythtv/programs/mythfrontend/globalsettings.cpp	2014-06-29 12:49:55.000000000 +0200
+++ mythtv-master-20140629-gb22277e-new/mythtv/programs/mythfrontend/globalsettings.cpp	2014-06-29 12:51:23.085266686 +0200
@@ -1641,7 +1641,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1713,7 +1713,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -1946,7 +1946,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -3013,7 +3013,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
