diff -Naur mythtv-master-20150913-g47e929b-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp mythtv-master-20150913-g47e929b-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp
--- mythtv-master-20150913-g47e929b-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2015-09-13 21:26:13.172813582 +0200
+++ mythtv-master-20150913-g47e929b-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2015-09-13 21:20:55.000000000 +0200
@@ -45,7 +45,7 @@
 #ifdef Q_OS_MAC
     gc->setValue(QDir::homePath() + "/Pictures");
 #else
-    gc->setValue("/var/lib/pictures");
+    gc->setValue("/mnt/gallery");
 #endif
     gc->setHelpText(GallerySettings::tr("This directory must exist and "
                                         "MythGallery needs to have read "
diff -Naur mythtv-master-20150913-g47e929b-old/mythplugins/mythgame/mythgame/gamesettings.cpp mythtv-master-20150913-g47e929b-new/mythplugins/mythgame/mythgame/gamesettings.cpp
--- mythtv-master-20150913-g47e929b-old/mythplugins/mythgame/mythgame/gamesettings.cpp	2015-09-13 21:11:53.000000000 +0200
+++ mythtv-master-20150913-g47e929b-new/mythplugins/mythgame/mythgame/gamesettings.cpp	2015-09-13 21:42:32.461498917 +0200
@@ -117,7 +117,7 @@
     HostLineEdit *gc = new HostLineEdit("mythgame.screenshotdir");
     gc->setLabel(MythGameGeneralSettings::tr("Directory where Game Screenshots "
                                              "are stored"));
-    gc->setValue(GetConfDir() + "/MythGame/Screenshots");
+    gc->setValue(GetConfDir() + "/myth/games/.screenshots");
     gc->setHelpText(MythGameGeneralSettings::tr("This directory will be the "
                                                 "default browse location when "
                                                 "assigning screenshots."));
@@ -129,7 +129,7 @@
     HostLineEdit *gc = new HostLineEdit("mythgame.fanartdir");
     gc->setLabel(MythGameGeneralSettings::tr("Directory where Game Fanart is "
                                              "stored"));
-    gc->setValue(GetConfDir() + "/MythGame/Fanart");
+    gc->setValue(GetConfDir() + "/myth/games/.fanart");
     gc->setHelpText(MythGameGeneralSettings::tr("This directory will be the "
                                                 "default browse location when "
                                                 "assigning fanart."));
@@ -141,7 +141,7 @@
     HostLineEdit *gc = new HostLineEdit("mythgame.boxartdir");
     gc->setLabel(MythGameGeneralSettings::tr("Directory where Game Boxart is "
                                              "stored"));
-    gc->setValue(GetConfDir() + "/MythGame/Boxart");
+    gc->setValue(GetConfDir() + "/myth/games/.boxart");
     gc->setHelpText(MythGameGeneralSettings::tr("This directory will be the "
                                                 "default browse location when "
                                                 "assigning boxart."));
diff -Naur mythtv-master-20150913-g47e929b-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20150913-g47e929b-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20150913-g47e929b-old/mythtv/libs/libmythui/mythmainwindow.cpp	2015-09-13 21:26:13.172813582 +0200
+++ mythtv-master-20150913-g47e929b-new/mythtv/libs/libmythui/mythmainwindow.cpp	2015-09-13 21:20:57.000000000 +0200
@@ -1287,9 +1287,9 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc");
+        "Display System Exit Prompt"),      "");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
-        "System Exit"),                     "");
+        "System Exit"),                     "Ctrl+Esc");
     RegisterKey("Main Menu",    "STANDBYMODE",QT_TRANSLATE_NOOP("MythControls",
         "Enter Standby Mode"),              "");
 }
diff -Naur mythtv-master-20150913-g47e929b-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20150913-g47e929b-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20150913-g47e929b-old/mythtv/programs/mythfrontend/globalsettings.cpp	2015-09-13 21:26:13.176146899 +0200
+++ mythtv-master-20150913-g47e929b-new/mythtv/programs/mythfrontend/globalsettings.cpp	2015-09-13 21:20:55.000000000 +0200
@@ -1643,7 +1643,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1715,7 +1715,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -1949,7 +1949,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -3016,7 +3016,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
diff -Naur mythtv-master-20150913-g47e929b-old/mythtv/programs/mythfrontend/videoplayersettings.cpp mythtv-master-20150913-g47e929b-new/mythtv/programs/mythfrontend/videoplayersettings.cpp
--- mythtv-master-20150913-g47e929b-old/mythtv/programs/mythfrontend/videoplayersettings.cpp	2015-09-13 21:26:13.176146899 +0200
+++ mythtv-master-20150913-g47e929b-new/mythtv/programs/mythfrontend/videoplayersettings.cpp	2015-09-13 21:20:55.000000000 +0200
@@ -69,7 +69,7 @@
     m_dvdDriveEdit->SetText(gCoreContext->GetSetting("DVDDeviceLocation",
                            "default"));
     m_blurayMountEdit->SetText(gCoreContext->GetSetting("BluRayMountpoint",
-                           "/media/cdrom"));
+                           "/media/sr0"));
     m_altPlayerEdit->SetText(gCoreContext->GetSetting(
                            "mythvideo.VideoAlternatePlayer", "Internal"));
 
