diff -Naur mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-old/mythtv/libs/libmythtv/tv_play.cpp	2021-10-28 21:37:48.000000000 +0200
+++ mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-new/mythtv/libs/libmythtv/tv_play.cpp	2021-10-29 16:30:20.946142364 +0200
@@ -468,7 +468,7 @@
 void TV::InitKeys()
 {
     REG_KEY("TV Frontend", ACTION_PLAYBACK, QT_TRANSLATE_NOOP("MythControls",
-            "Play Program"), "P");
+            "Play Program"), "Ctrl+P");
     REG_KEY("TV Frontend", ACTION_STOP, QT_TRANSLATE_NOOP("MythControls",
             "Stop Program"), "");
     REG_KEY("TV Frontend", ACTION_TOGGLERECORD, QT_TRANSLATE_NOOP("MythControls",
@@ -894,7 +894,7 @@
 
     /* OSD playback information screen */
     REG_KEY("TV Playback", ACTION_TOGGLEOSDDEBUG,
-            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "");
+            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "Ctrl+I");
 
     /* 3D/Frame compatible/Stereoscopic TV */
     REG_KEY("TV Playback", ACTION_3DNONE,
diff -Naur mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-old/mythtv/libs/libmythui/mythmainwindow.cpp	2021-10-28 21:37:48.000000000 +0200
+++ mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-new/mythtv/libs/libmythui/mythmainwindow.cpp	2021-10-29 16:30:20.946142364 +0200
@@ -795,7 +795,7 @@
     RegisterKey("Global", "PREVIOUS", QT_TRANSLATE_NOOP("MythControls",
         "Move to preview widget"), "Backtab");
     RegisterKey("Global", ACTION_SELECT, QT_TRANSLATE_NOOP("MythControls",
-        "Select"), "Return,Enter,Space");
+        "Select"), "Return,Enter,Space,Select");
     RegisterKey("Global", "BACKSPACE", QT_TRANSLATE_NOOP("MythControls",
         "Backspace"),       "Backspace");
     RegisterKey("Global", "ESCAPE", QT_TRANSLATE_NOOP("MythControls",
@@ -926,9 +926,9 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc");
+        "Display System Exit Prompt"),      "");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
-        "System Exit"),                     "");
+        "System Exit"),                     "Ctrl+Esc");
     RegisterKey("Main Menu",    "STANDBYMODE",QT_TRANSLATE_NOOP("MythControls",
         "Enter Standby Mode"),              "");
     RegisterKey("Long Press",    "LONGPRESS1",QT_TRANSLATE_NOOP("MythControls",
diff -Naur mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-old/mythtv/programs/mythfrontend/globalsettings.cpp	2021-10-29 16:28:52.729477295 +0200
+++ mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-new/mythtv/programs/mythfrontend/globalsettings.cpp	2021-10-29 16:31:40.289474277 +0200
@@ -1729,7 +1729,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1801,7 +1801,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -2107,7 +2107,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -3230,7 +3230,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
@@ -3276,7 +3276,7 @@
         "on and off from a MythTV remote "
         "if you have compatible hardware. "
         "These settings only take effect after a restart."));
-    gc->setValue(true);
+    gc->setValue(false);
     return gc;
 }
 
diff -Naur mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-old/mythtv/programs/mythfrontend/main.cpp mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-old/mythtv/programs/mythfrontend/main.cpp	2021-10-29 16:28:54.372810598 +0200
+++ mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-new/mythtv/programs/mythfrontend/main.cpp	2021-10-29 16:30:20.949475698 +0200
@@ -1604,7 +1604,7 @@
 
      // Gallery keybindings
      REG_KEY("Images", "PLAY", QT_TRANSLATE_NOOP("MythControls",
-         "Start/Stop Slideshow"), "P");
+         "Start/Stop Slideshow"), "Ctrl+P");
      REG_KEY("Images", "RECURSIVESHOW", QT_TRANSLATE_NOOP("MythControls",
          "Start Recursive Slideshow"), "R");
      REG_KEY("Images", "ROTRIGHT", QT_TRANSLATE_NOOP("MythControls",
diff -Naur mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-old/mythtv/programs/mythfrontend/videoplayersettings.cpp mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-new/mythtv/programs/mythfrontend/videoplayersettings.cpp
--- mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-old/mythtv/programs/mythfrontend/videoplayersettings.cpp	2021-10-28 21:37:48.000000000 +0200
+++ mythtv-8ebfeb3bbf2ed192bc42213b2a3645df4bfa6685-new/mythtv/programs/mythfrontend/videoplayersettings.cpp	2021-10-29 16:30:20.949475698 +0200
@@ -55,7 +55,7 @@
     m_dvdDriveEdit->SetText(gCoreContext->GetSetting("DVDDeviceLocation",
                            "default"));
     m_blurayMountEdit->SetText(gCoreContext->GetSetting("BluRayMountpoint",
-                           "/media/cdrom"));
+                           "/media/disk"));
     m_altPlayerEdit->SetText(gCoreContext->GetSetting(
                            "mythvideo.VideoAlternatePlayer", "Internal"));
 
