diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp mythtv-master-20181012-g4651002-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp
--- mythtv-master-20181012-g4651002-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2018-10-12 10:52:39.996666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2018-10-12 10:52:43.709999996 +0200
@@ -45,7 +45,7 @@
 #ifdef Q_OS_MAC
     gc->setValue(QDir::homePath() + "/Pictures");
 #else
-    gc->setValue("/var/lib/pictures");
+    gc->setValue("/mnt/gallery");
 #endif
     gc->setHelpText(GallerySettings::tr("This directory must exist and "
                                         "MythGallery needs to have read "
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythgame/mythgame/gamesettings.cpp mythtv-master-20181012-g4651002-new/mythplugins/mythgame/mythgame/gamesettings.cpp
--- mythtv-master-20181012-g4651002-old/mythplugins/mythgame/mythgame/gamesettings.cpp	2018-10-12 10:52:39.999999999 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythgame/mythgame/gamesettings.cpp	2018-10-12 10:52:43.709999996 +0200
@@ -117,7 +117,7 @@
     HostTextEditSetting *gc = new HostTextEditSetting("mythgame.screenshotdir");
     gc->setLabel(MythGameGeneralSettings::tr("Directory where Game Screenshots "
                                              "are stored"));
-    gc->setValue(GetConfDir() + "/MythGame/Screenshots");
+    gc->setValue(GetConfDir() + "/myth/games/.screenshots");
     gc->setHelpText(MythGameGeneralSettings::tr("This directory will be the "
                                                 "default browse location when "
                                                 "assigning screenshots."));
@@ -129,7 +129,7 @@
     HostTextEditSetting *gc = new HostTextEditSetting("mythgame.fanartdir");
     gc->setLabel(MythGameGeneralSettings::tr("Directory where Game Fanart is "
                                              "stored"));
-    gc->setValue(GetConfDir() + "/MythGame/Fanart");
+    gc->setValue(GetConfDir() + "/myth/games/.fanart");
     gc->setHelpText(MythGameGeneralSettings::tr("This directory will be the "
                                                 "default browse location when "
                                                 "assigning fanart."));
@@ -141,7 +141,7 @@
     HostTextEditSetting *gc = new HostTextEditSetting("mythgame.boxartdir");
     gc->setLabel(MythGameGeneralSettings::tr("Directory where Game Boxart is "
                                              "stored"));
-    gc->setValue(GetConfDir() + "/MythGame/Boxart");
+    gc->setValue(GetConfDir() + "/myth/games/.boxart");
     gc->setHelpText(MythGameGeneralSettings::tr("This directory will be the "
                                                 "default browse location when "
                                                 "assigning boxart."));
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/tv_play.cpp	2018-10-12 10:52:39.976666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/tv_play.cpp	2018-10-12 10:52:43.713333330 +0200
@@ -542,7 +542,7 @@
 void TV::InitKeys(void)
 {
     REG_KEY("TV Frontend", ACTION_PLAYBACK, QT_TRANSLATE_NOOP("MythControls",
-            "Play Program"), "P");
+            "Play Program"), "Ctrl+P");
     REG_KEY("TV Frontend", ACTION_STOP, QT_TRANSLATE_NOOP("MythControls",
             "Stop Program"), "");
     REG_KEY("TV Frontend", ACTION_TOGGLERECORD, QT_TRANSLATE_NOOP("MythControls",
@@ -983,7 +983,7 @@
 
     /* OSD playback information screen */
     REG_KEY("TV Playback", ACTION_TOGGLEOSDDEBUG,
-            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "");
+            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "Ctrl+I");
 
     /* 3D/Frame compatible/Stereoscopic TV */
     REG_KEY("TV Playback", ACTION_3DNONE,
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythui/mythmainwindow.cpp	2018-10-12 10:52:39.959999999 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythui/mythmainwindow.cpp	2018-10-12 10:52:43.713333330 +0200
@@ -1388,9 +1388,9 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc");
+        "Display System Exit Prompt"),      "");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
-        "System Exit"),                     "");
+        "System Exit"),                     "Ctrl+Esc");
     RegisterKey("Main Menu",    "STANDBYMODE",QT_TRANSLATE_NOOP("MythControls",
         "Enter Standby Mode"),              "");
     RegisterKey("Long Press",    "LONGPRESS1",QT_TRANSLATE_NOOP("MythControls",
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20181012-g4651002-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/programs/mythfrontend/globalsettings.cpp	2018-10-12 10:52:39.909999999 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/mythfrontend/globalsettings.cpp	2018-10-12 10:52:43.713333330 +0200
@@ -1601,7 +1601,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1673,7 +1673,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -1907,7 +1907,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -3008,7 +3008,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20181012-g4651002-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/programs/mythfrontend/main.cpp	2018-10-12 10:52:39.906666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/mythfrontend/main.cpp	2018-10-12 10:52:43.713333330 +0200
@@ -1606,7 +1606,7 @@
 
      // Gallery keybindings
      REG_KEY("Images", "PLAY", QT_TRANSLATE_NOOP("MythControls",
-         "Start/Stop Slideshow"), "P");
+         "Start/Stop Slideshow"), "Ctrl+P");
      REG_KEY("Images", "RECURSIVESHOW", QT_TRANSLATE_NOOP("MythControls",
          "Start Recursive Slideshow"), "R");
      REG_KEY("Images", "ROTRIGHT", QT_TRANSLATE_NOOP("MythControls",
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/programs/mythfrontend/videoplayersettings.cpp mythtv-master-20181012-g4651002-new/mythtv/programs/mythfrontend/videoplayersettings.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/programs/mythfrontend/videoplayersettings.cpp	2018-10-12 10:52:39.906666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/programs/mythfrontend/videoplayersettings.cpp	2018-10-12 10:52:43.713333330 +0200
@@ -66,7 +66,7 @@
     m_dvdDriveEdit->SetText(gCoreContext->GetSetting("DVDDeviceLocation",
                            "default"));
     m_blurayMountEdit->SetText(gCoreContext->GetSetting("BluRayMountpoint",
-                           "/media/cdrom"));
+                           "/media/disk"));
     m_altPlayerEdit->SetText(gCoreContext->GetSetting(
                            "mythvideo.VideoAlternatePlayer", "Internal"));
 
