diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythtv/tv_play.cpp	2021-03-17 21:07:11.479999833 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythtv/tv_play.cpp	2021-03-17 21:07:14.156666500 +0100
@@ -468,7 +468,7 @@
 void TV::InitKeys()
 {
     REG_KEY("TV Frontend", ACTION_PLAYBACK, QT_TRANSLATE_NOOP("MythControls",
-            "Play Program"), "P");
+            "Play Program"), "Ctrl+P");
     REG_KEY("TV Frontend", ACTION_STOP, QT_TRANSLATE_NOOP("MythControls",
             "Stop Program"), "");
     REG_KEY("TV Frontend", ACTION_TOGGLERECORD, QT_TRANSLATE_NOOP("MythControls",
@@ -894,7 +894,7 @@
 
     /* OSD playback information screen */
     REG_KEY("TV Playback", ACTION_TOGGLEOSDDEBUG,
-            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "");
+            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "Ctrl+I");
 
     /* 3D/Frame compatible/Stereoscopic TV */
     REG_KEY("TV Playback", ACTION_3DNONE,
diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythui/mythmainwindow.cpp	2021-03-17 21:07:11.449999833 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythui/mythmainwindow.cpp	2021-03-17 21:07:14.189999834 +0100
@@ -933,9 +933,9 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc");
+        "Display System Exit Prompt"),      "");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
-        "System Exit"),                     "");
+        "System Exit"),                     "Ctrl+Esc");
     RegisterKey("Main Menu",    "STANDBYMODE",QT_TRANSLATE_NOOP("MythControls",
         "Enter Standby Mode"),              "");
     RegisterKey("Long Press",    "LONGPRESS1",QT_TRANSLATE_NOOP("MythControls",
diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/programs/mythfrontend/globalsettings.cpp	2021-03-17 21:07:10.673333166 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/programs/mythfrontend/globalsettings.cpp	2021-03-17 21:07:14.193333167 +0100
@@ -1686,7 +1686,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1758,7 +1758,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -2064,7 +2064,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -3187,7 +3187,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/programs/mythfrontend/main.cpp	2021-03-17 21:07:10.673333166 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/programs/mythfrontend/main.cpp	2021-03-17 21:07:14.196666500 +0100
@@ -1593,7 +1593,7 @@
 
      // Gallery keybindings
      REG_KEY("Images", "PLAY", QT_TRANSLATE_NOOP("MythControls",
-         "Start/Stop Slideshow"), "P");
+         "Start/Stop Slideshow"), "Ctrl+P");
      REG_KEY("Images", "RECURSIVESHOW", QT_TRANSLATE_NOOP("MythControls",
          "Start Recursive Slideshow"), "R");
      REG_KEY("Images", "ROTRIGHT", QT_TRANSLATE_NOOP("MythControls",
diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/programs/mythfrontend/videoplayersettings.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/programs/mythfrontend/videoplayersettings.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/programs/mythfrontend/videoplayersettings.cpp	2021-03-17 21:07:10.673333166 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/programs/mythfrontend/videoplayersettings.cpp	2021-03-17 21:07:14.199999834 +0100
@@ -55,7 +55,7 @@
     m_dvdDriveEdit->SetText(gCoreContext->GetSetting("DVDDeviceLocation",
                            "default"));
     m_blurayMountEdit->SetText(gCoreContext->GetSetting("BluRayMountpoint",
-                           "/media/cdrom"));
+                           "/media/disk"));
     m_altPlayerEdit->SetText(gCoreContext->GetSetting(
                            "mythvideo.VideoAlternatePlayer", "Internal"));
 
