diff -Naur mythtv-master-20200625-gc3e17e6-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20200625-gc3e17e6-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20200625-gc3e17e6-old/mythtv/libs/libmythtv/tv_play.cpp	2020-06-29 11:46:30.826666667 +0200
+++ mythtv-master-20200625-gc3e17e6-new/mythtv/libs/libmythtv/tv_play.cpp	2020-06-29 11:47:58.780000001 +0200
@@ -505,7 +505,7 @@
 void TV::InitKeys(void)
 {
     REG_KEY("TV Frontend", ACTION_PLAYBACK, QT_TRANSLATE_NOOP("MythControls",
-            "Play Program"), "P");
+            "Play Program"), "Ctrl+P");
     REG_KEY("TV Frontend", ACTION_STOP, QT_TRANSLATE_NOOP("MythControls",
             "Stop Program"), "");
     REG_KEY("TV Frontend", ACTION_TOGGLERECORD, QT_TRANSLATE_NOOP("MythControls",
@@ -953,7 +953,7 @@
 
     /* OSD playback information screen */
     REG_KEY("TV Playback", ACTION_TOGGLEOSDDEBUG,
-            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "");
+            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "Ctrl+I");
 
     /* 3D/Frame compatible/Stereoscopic TV */
     REG_KEY("TV Playback", ACTION_3DNONE,
diff -Naur mythtv-master-20200625-gc3e17e6-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20200625-gc3e17e6-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20200625-gc3e17e6-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-06-29 11:46:26.106666667 +0200
+++ mythtv-master-20200625-gc3e17e6-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-06-29 11:47:58.783333334 +0200
@@ -904,9 +904,9 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc");
+        "Display System Exit Prompt"),      "");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
-        "System Exit"),                     "");
+        "System Exit"),                     "Ctrl+Esc");
     RegisterKey("Main Menu",    "STANDBYMODE",QT_TRANSLATE_NOOP("MythControls",
         "Enter Standby Mode"),              "");
     RegisterKey("Long Press",    "LONGPRESS1",QT_TRANSLATE_NOOP("MythControls",
diff -Naur mythtv-master-20200625-gc3e17e6-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20200625-gc3e17e6-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20200625-gc3e17e6-old/mythtv/programs/mythfrontend/globalsettings.cpp	2020-06-29 11:46:30.896666668 +0200
+++ mythtv-master-20200625-gc3e17e6-new/mythtv/programs/mythfrontend/globalsettings.cpp	2020-06-29 11:50:11.856666659 +0200
@@ -1675,7 +1675,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1728,7 +1728,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -2023,7 +2023,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -3138,7 +3138,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
@@ -3374,7 +3374,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Ignore devices"));
 
-    ge->setValue("");
+    ge->setValue("/dev/mmcblk0,/dev/mmcblk1,/dev/mmcblk0p2,/dev/mmcblk1p2,/dev/zram0,/dev/zram1,/dev/zram2,/dev/zram3,/dev/zram4,/dev/zram5,/dev/zram6,/dev/zram7");
 
     ge->setHelpText(MainGeneralSettings::tr("If there are any devices that you "
                                             "do not want to be monitored, list "
diff -Naur mythtv-master-20200625-gc3e17e6-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20200625-gc3e17e6-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20200625-gc3e17e6-old/mythtv/programs/mythfrontend/main.cpp	2020-06-29 11:46:30.830000001 +0200
+++ mythtv-master-20200625-gc3e17e6-new/mythtv/programs/mythfrontend/main.cpp	2020-06-29 11:47:58.786666668 +0200
@@ -1660,7 +1660,7 @@
 
      // Gallery keybindings
      REG_KEY("Images", "PLAY", QT_TRANSLATE_NOOP("MythControls",
-         "Start/Stop Slideshow"), "P");
+         "Start/Stop Slideshow"), "Ctrl+P");
      REG_KEY("Images", "RECURSIVESHOW", QT_TRANSLATE_NOOP("MythControls",
          "Start Recursive Slideshow"), "R");
      REG_KEY("Images", "ROTRIGHT", QT_TRANSLATE_NOOP("MythControls",
diff -Naur mythtv-master-20200625-gc3e17e6-old/mythtv/programs/mythfrontend/videoplayersettings.cpp mythtv-master-20200625-gc3e17e6-new/mythtv/programs/mythfrontend/videoplayersettings.cpp
--- mythtv-master-20200625-gc3e17e6-old/mythtv/programs/mythfrontend/videoplayersettings.cpp	2020-06-25 14:43:48.000000000 +0200
+++ mythtv-master-20200625-gc3e17e6-new/mythtv/programs/mythfrontend/videoplayersettings.cpp	2020-06-29 11:47:58.786666668 +0200
@@ -56,7 +56,7 @@
     m_dvdDriveEdit->SetText(gCoreContext->GetSetting("DVDDeviceLocation",
                            "default"));
     m_blurayMountEdit->SetText(gCoreContext->GetSetting("BluRayMountpoint",
-                           "/media/cdrom"));
+                           "/media/disk"));
     m_altPlayerEdit->SetText(gCoreContext->GetSetting(
                            "mythvideo.VideoAlternatePlayer", "Internal"));
 
