diff -Naur mythtv-master-20200109-gc116014-old/mythtv/bindings/perl/Makefile mythtv-master-20200109-gc116014-new/mythtv/bindings/perl/Makefile
--- mythtv-master-20200109-gc116014-old/mythtv/bindings/perl/Makefile	2020-01-09 18:00:48.163333334 +0100
+++ mythtv-master-20200109-gc116014-new/mythtv/bindings/perl/Makefile	2020-01-09 18:00:49.836666667 +0100
@@ -6,7 +6,7 @@
 
 ifneq ($(PREFIX:/=), /usr)
 	ifneq ($(PREFIX:/=), /usr/local)
-		PREFIX_FLAGS=PREFIX="$(PREFIX)"
+		PREFIX_FLAGS=PREFIX="/usr"
 	endif
 endif
 
