diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/bindings/perl/Makefile mythtv-master-20191227-gf15b4f0-new/mythtv/bindings/perl/Makefile
--- mythtv-master-20191227-gf15b4f0-old/mythtv/bindings/perl/Makefile	2020-01-03 18:36:08.000000000 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/bindings/perl/Makefile	2020-01-04 12:48:22.423333320 +0100
@@ -6,7 +6,7 @@
 
 ifneq ($(PREFIX:/=), /usr)
 	ifneq ($(PREFIX:/=), /usr/local)
-		PREFIX_FLAGS=PREFIX="$(PREFIX)"
+		PREFIX_FLAGS=PREFIX="/usr"
 	endif
 endif
 
