diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/bindings/perl/Makefile mythtv-master-20200114-ga315cd6-new/mythtv/bindings/perl/Makefile
--- mythtv-master-20200114-ga315cd6-old/mythtv/bindings/perl/Makefile	2020-01-14 09:19:41.863333332 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/bindings/perl/Makefile	2020-01-14 09:19:43.233333332 +0100
@@ -6,7 +6,7 @@
 
 ifneq ($(PREFIX:/=), /usr)
 	ifneq ($(PREFIX:/=), /usr/local)
-		PREFIX_FLAGS=PREFIX="$(PREFIX)"
+		PREFIX_FLAGS=PREFIX="/usr"
 	endif
 endif
 
