diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/bindings/perl/Makefile mythtv-master-20200904-gca3ee6a-new/mythtv/bindings/perl/Makefile
--- mythtv-master-20200904-gca3ee6a-old/mythtv/bindings/perl/Makefile	2020-09-04 11:35:48.993333333 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/bindings/perl/Makefile	2020-09-04 11:35:50.500000000 +0200
@@ -6,7 +6,7 @@
 
 ifneq ($(PREFIX:/=), /usr)
 	ifneq ($(PREFIX:/=), /usr/local)
-		PREFIX_FLAGS=PREFIX="$(PREFIX)"
+		PREFIX_FLAGS=PREFIX="/usr"
 	endif
 endif
 
diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/bindings/python/Makefile mythtv-master-20200904-gca3ee6a-new/mythtv/bindings/python/Makefile
--- mythtv-master-20200904-gca3ee6a-old/mythtv/bindings/python/Makefile	2020-09-04 11:35:48.993333333 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/bindings/python/Makefile	2020-09-04 11:35:50.500000000 +0200
@@ -19,7 +19,7 @@
 distclean: clean
 
 python_build: setup.py
-	$(PYTHON) setup.py build --mythtv-prefix="$(REAL_PREFIX)"
+	$(PYTHON) setup.py build --mythtv-prefix="/usr"
 
 install: setup.py
 	$(PYTHON) setup.py install --skip-build $(ROOT_FLAGS) $(PREFIX_FLAGS)
diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/programs/scripts/python_pathfix.py mythtv-master-20200904-gca3ee6a-new/mythtv/programs/scripts/python_pathfix.py
--- mythtv-master-20200904-gca3ee6a-old/mythtv/programs/scripts/python_pathfix.py	2020-09-04 11:35:48.890000000 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/programs/scripts/python_pathfix.py	2020-09-04 11:35:50.589999999 +0200
@@ -12,7 +12,7 @@
 def main():
 
     # Pick up default interpreter (ours)
-    python_interpreter = os.path.normcase(sys.executable).encode()
+    python_interpreter = '/usr/bin/python'
 
     # Don't bother reporting
     verbose = False
