diff -Naur mythtv-master-20200116-gdd96d68-old/mythtv/bindings/perl/Makefile mythtv-master-20200116-gdd96d68-new/mythtv/bindings/perl/Makefile
--- mythtv-master-20200116-gdd96d68-old/mythtv/bindings/perl/Makefile	2020-01-16 10:16:56.866666668 +0100
+++ mythtv-master-20200116-gdd96d68-new/mythtv/bindings/perl/Makefile	2020-01-16 10:16:58.260000001 +0100
@@ -6,7 +6,7 @@
 
 ifneq ($(PREFIX:/=), /usr)
 	ifneq ($(PREFIX:/=), /usr/local)
-		PREFIX_FLAGS=PREFIX="$(PREFIX)"
+		PREFIX_FLAGS=PREFIX="/usr"
 	endif
 endif
 
