diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmyth/langsettings.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmyth/langsettings.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmyth/langsettings.cpp	2021-03-17 21:07:16.293333167 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmyth/langsettings.cpp	2021-03-17 21:07:18.943333168 +0100
@@ -199,6 +199,7 @@
 
 void LanguageSelection::Save(void)
 {
+    QString hostname           = gCoreContext->GetHostName();
     MythUIButtonListItem *item = m_languageList->GetItemCurrent();
 
     if (!item)
@@ -210,7 +211,7 @@
     }
 
     QString langCode = item->GetData().toString();
-    gCoreContext->SaveSettingOnHost("Language", langCode, nullptr);
+    gCoreContext->SaveSettingOnHost("Language", langCode, hostname);
 
     item = m_countryList->GetItemCurrent();
 
@@ -223,7 +224,7 @@
     }
 
     QString countryCode = item->GetData().toString();
-    gCoreContext->SaveSettingOnHost("Country", countryCode, nullptr);
+    gCoreContext->SaveSettingOnHost("Country", countryCode, hostname);
 
     if (m_language != langCode)
         m_languageChanged = true;
diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/programs/mythfrontend/globalsettings.cpp	2021-03-17 21:07:15.499999834 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/programs/mythfrontend/globalsettings.cpp	2021-03-17 21:07:18.943333168 +0100
@@ -3106,9 +3106,9 @@
     return gs;
 }
 
-static GlobalComboBoxSetting *MythLanguage()
+static HostComboBoxSetting *MythLanguage()
 {
-    auto *gc = new GlobalComboBoxSetting("Language");
+    auto *gc = new HostComboBoxSetting("Language");
 
     gc->setLabel(AppearanceSettings::tr("Language"));
 
