diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmyth/langsettings.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmyth/langsettings.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmyth/langsettings.cpp	2020-05-07 12:46:15.120000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmyth/langsettings.cpp	2020-05-07 12:46:16.586666667 +0200
@@ -201,6 +201,7 @@
 
 void LanguageSelection::Save(void)
 {
+    QString hostname           = gCoreContext->GetHostName();
     MythUIButtonListItem *item = m_languageList->GetItemCurrent();
 
     if (!item)
@@ -212,7 +213,7 @@
     }
 
     QString langCode = item->GetData().toString();
-    gCoreContext->SaveSettingOnHost("Language", langCode, nullptr);
+    gCoreContext->SaveSettingOnHost("Language", langCode, hostname);
 
     item = m_countryList->GetItemCurrent();
 
@@ -225,7 +226,7 @@
     }
 
     QString countryCode = item->GetData().toString();
-    gCoreContext->SaveSettingOnHost("Country", countryCode, nullptr);
+    gCoreContext->SaveSettingOnHost("Country", countryCode, hostname);
 
     if (m_language != langCode)
         m_languageChanged = true;
diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20200507-ge586e5a-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/programs/mythfrontend/globalsettings.cpp	2020-05-07 12:46:14.840000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/programs/mythfrontend/globalsettings.cpp	2020-05-07 12:46:16.586666667 +0200
@@ -3073,9 +3073,9 @@
     return gs;
 }
 
-static GlobalComboBoxSetting *MythLanguage()
+static HostComboBoxSetting *MythLanguage()
 {
-    auto *gc = new GlobalComboBoxSetting("Language");
+    auto *gc = new HostComboBoxSetting("Language");
 
     gc->setLabel(AppearanceSettings::tr("Language"));
 
