diff --git a/mythtv/libs/libmythbase/mythcorecontext.cpp b/mythtv/libs/libmythbase/mythcorecontext.cpp
index a31e22a..98f8cac 100644
--- a/mythtv/libs/libmythbase/mythcorecontext.cpp
+++ b/mythtv/libs/libmythbase/mythcorecontext.cpp
@@ -297,7 +297,8 @@ bool MythCoreContext::SetupCommandSocket(MythSocket *serverSock,
     proto_mismatch = false;
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!CheckProtoVersion(serverSock, timeout_in_ms, true))
+    if (!GetNumSetting("IgnoreProtoVerMismatch", false)
+        && !CheckProtoVersion(serverSock, timeout_in_ms, true))
     {
         proto_mismatch = true;
         return false;
diff --git a/mythtv/libs/libmythbase/remotefile.cpp b/mythtv/libs/libmythbase/remotefile.cpp
index 5d6d086..8003676 100644
--- a/mythtv/libs/libmythbase/remotefile.cpp
+++ b/mythtv/libs/libmythbase/remotefile.cpp
@@ -167,7 +167,8 @@ MythSocket *RemoteFile::openSocket(bool control)
     QStringList strlist;
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!gCoreContext->CheckProtoVersion(lsock, 5000))
+    if (!gCoreContext->GetNumSetting("IgnoreProtoVerMismatch", false)
+        && !gCoreContext->CheckProtoVersion(lsock, 5000))
     {
         LOG(VB_GENERAL, LOG_ERR, loc +
             QString("Failed validation to server %1:%2").arg(host).arg(port));
diff --git a/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp b/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp
index 392437b..c9eb966 100644
--- a/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp
+++ b/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp
@@ -46,7 +46,8 @@ bool OutboundRequestHandler::DoConnectToMaster(void)
     }
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!m_socket->Validate())
+    if (!gCoreContext->GetNumSetting("IgnoreProtoVerMismatch", false)
+        && !m_socket->Validate())
     {
         LOG(VB_GENERAL, LOG_NOTICE, "Unable to confirm protocol version with backend.");
         m_socket->DecrRef();
diff --git a/mythtv/programs/mythfrontend/main.cpp b/mythtv/programs/mythfrontend/main.cpp
index 29c17d5..2784b30 100644
--- a/mythtv/programs/mythfrontend/main.cpp
+++ b/mythtv/programs/mythfrontend/main.cpp
@@ -2008,7 +2008,8 @@ int main(int argc, char **argv)
             return GENERIC_EXIT_NO_THEME;
     }
 
-    if (!UpgradeTVDatabaseSchema(false))
+    if (!gCoreContext->GetNumSetting("IgnoreSchemaVerMismatch", false)
+        && !UpgradeTVDatabaseSchema(false))
     {
         LOG(VB_GENERAL, LOG_ERR,
             "Couldn't upgrade database to new schema, exiting.");
