diff -Naur mythtv-master-20200116-gdd96d68-old/mythtv/libs/libmyth/langsettings.cpp mythtv-master-20200116-gdd96d68-new/mythtv/libs/libmyth/langsettings.cpp
--- mythtv-master-20200116-gdd96d68-old/mythtv/libs/libmyth/langsettings.cpp	2020-01-16 10:15:18.000000000 +0100
+++ mythtv-master-20200116-gdd96d68-new/mythtv/libs/libmyth/langsettings.cpp	2020-01-15 22:23:44.623333319 +0100
@@ -201,6 +201,7 @@
 
 void LanguageSelection::Save(void)
 {
+    QString hostname           = gCoreContext->GetHostName();
     MythUIButtonListItem *item = m_languageList->GetItemCurrent();
 
     if (!item)
@@ -212,7 +213,7 @@
     }
 
     QString langCode = item->GetData().toString();
-    gCoreContext->SaveSettingOnHost("Language", langCode, nullptr);
+    gCoreContext->SaveSettingOnHost("Language", langCode, hostname);
 
     item = m_countryList->GetItemCurrent();
 
@@ -225,7 +226,7 @@
     }
 
     QString countryCode = item->GetData().toString();
-    gCoreContext->SaveSettingOnHost("Country", countryCode, nullptr);
+    gCoreContext->SaveSettingOnHost("Country", countryCode, hostname);
 
     if (m_language != langCode)
         m_languageChanged = true;
diff -Naur mythtv-master-20200116-gdd96d68-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20200116-gdd96d68-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20200116-gdd96d68-old/mythtv/programs/mythfrontend/globalsettings.cpp	2020-01-16 10:30:08.063333334 +0100
+++ mythtv-master-20200116-gdd96d68-new/mythtv/programs/mythfrontend/globalsettings.cpp	2020-01-15 22:32:49.779999973 +0100
@@ -3011,9 +3030,9 @@
     return gs;
 }
 
-static GlobalComboBoxSetting *MythLanguage()
+static HostComboBoxSetting *MythLanguage()
 {
-    auto *gc = new GlobalComboBoxSetting("Language");
+    auto *gc = new HostComboBoxSetting("Language");
 
     gc->setLabel(AppearanceSettings::tr("Language"));
 
