diff -Naur mythtv-master-20200409-g0fc7abb082-old/mythtv/external/nv-codec-headers/Makefile mythtv-master-20200409-g0fc7abb082-new/mythtv/external/nv-codec-headers/Makefile
--- mythtv-master-20200409-g0fc7abb082-old/mythtv/external/nv-codec-headers/Makefile	2020-04-09 12:25:34.054255652 +0200
+++ mythtv-master-20200409-g0fc7abb082-new/mythtv/external/nv-codec-headers/Makefile	2020-04-09 12:25:36.124255577 +0200
@@ -11,13 +11,13 @@
 endif
 
 install: all
-	$(INSTALL) -m 0755 -d '$(DESTDIR)$(PREFIX)/include/ffnvcodec'
-	$(INSTALL) -m 0644 include/ffnvcodec/*.h '$(DESTDIR)$(PREFIX)/include/ffnvcodec'
-	$(INSTALL) -m 0755 -d '$(DESTDIR)$(PREFIX)/$(LIBDIR)/pkgconfig'
-	$(INSTALL) -m 0644 ffnvcodec.pc '$(DESTDIR)$(PREFIX)/$(LIBDIR)/pkgconfig'
+	$(INSTALL) -m 0755 -d '$(DESTDIR)/usr/include/ffnvcodec'
+	$(INSTALL) -m 0644 include/ffnvcodec/*.h '$(DESTDIR)/usr/include/ffnvcodec'
+	$(INSTALL) -m 0755 -d '$(DESTDIR)/usr/$(LIBDIR)/pkgconfig'
+	$(INSTALL) -m 0644 ffnvcodec.pc '$(DESTDIR)/usr/$(LIBDIR)/pkgconfig'
 
 uninstall:
-	rm -rf '$(DESTDIR)$(PREFIX)/include/ffnvcodec' '$(DESTDIR)$(PREFIX)/$(LIBDIR)/pkgconfig/ffnvcodec.pc'
+	rm -rf '$(DESTDIR)/include/ffnvcodec' '$(DESTDIR)/usr/$(LIBDIR)/pkgconfig/ffnvcodec.pc'
 
 .PHONY: all install uninstall
 
