diff -Naur mythtv-8646ad33eafe345120f58005d6e7783f6448f1e6-old/mythtv/libs/libmythui/devices/mythcecadapter.cpp mythtv-8646ad33eafe345120f58005d6e7783f6448f1e6-new/mythtv/libs/libmythui/devices/mythcecadapter.cpp
--- mythtv-8646ad33eafe345120f58005d6e7783f6448f1e6-old/mythtv/libs/libmythui/devices/mythcecadapter.cpp	2021-11-01 19:22:15.000000000 +0100
+++ mythtv-8646ad33eafe345120f58005d6e7783f6448f1e6-new/mythtv/libs/libmythui/devices/mythcecadapter.cpp	2021-11-03 09:29:53.086661869 +0100
@@ -94,7 +94,7 @@
     Close();
 
     // don't try if disabled
-    if (!gCoreContext->GetBoolSetting(LIBCEC_ENABLED, true))
+    if (!gCoreContext->GetBoolSetting(LIBCEC_ENABLED, false))
     {
         LOG(VB_GENERAL, LOG_INFO, LOC + "libCEC support is disabled.");
         return;
diff -Naur mythtv-8646ad33eafe345120f58005d6e7783f6448f1e6-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-8646ad33eafe345120f58005d6e7783f6448f1e6-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-8646ad33eafe345120f58005d6e7783f6448f1e6-old/mythtv/programs/mythfrontend/globalsettings.cpp	2021-11-03 09:40:19.586650572 +0100
+++ mythtv-8646ad33eafe345120f58005d6e7783f6448f1e6-new/mythtv/programs/mythfrontend/globalsettings.cpp	2021-11-03 09:43:38.186646988 +0100
@@ -3276,7 +3276,7 @@
         "on and off from a MythTV remote "
         "if you have compatible hardware. "
         "These settings only take effect after a restart."));
-    gc->setValue(true);
+    gc->setValue(false);
     return gc;
 }
 
