diff --speed-large-files --no-dereference --minimal -Naur mythtv-9c59a87a905b9ef1eaf8bb67629cbe412c5fc38f/mythtv/libs/libmythui/mythuithemehelper.h mythtv-9c59a87a905b9ef1eaf8bb67629cbe412c5fc38f/mythtv/libs/libmythui/mythuithemehelper.h
--- mythtv-9c59a87a905b9ef1eaf8bb67629cbe412c5fc38f/mythtv/libs/libmythui/mythuithemehelper.h	2022-08-06 16:36:43.000000000 +0200
+++ mythtv-9c59a87a905b9ef1eaf8bb67629cbe412c5fc38f/mythtv/libs/libmythui/mythuithemehelper.h	2022-08-08 13:21:45.626664729 +0200
@@ -4,8 +4,8 @@
 // MythTV
 #include "themeinfo.h"
 
-static constexpr const char* DEFAULT_UI_THEME  { "MythCenter" };
-static constexpr const char* FALLBACK_UI_THEME { "Terra"      };
+static constexpr const char* DEFAULT_UI_THEME  { "Monochrome" };
+static constexpr const char* FALLBACK_UI_THEME { "MythCenter" };
 
 class MUI_PUBLIC MythUIThemeHelper
 {
