diff --speed-large-files --no-dereference --minimal -Naur mythtv-65c97ac770a95b4b818d7ce3f512bc4ef79d2d8e/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-65c97ac770a95b4b818d7ce3f512bc4ef79d2d8e/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-65c97ac770a95b4b818d7ce3f512bc4ef79d2d8e/mythtv/programs/mythfrontend/globalsettings.cpp	2024-05-22 16:24:06.659593302 +0200
+++ mythtv-65c97ac770a95b4b818d7ce3f512bc4ef79d2d8e/mythtv/programs/mythfrontend/globalsettings.cpp	2024-05-22 16:21:55.376259535 +0200
@@ -2412,8 +2412,8 @@
         gCoreContext->GetResolutionSetting("GuiVidMode", w, h);
         if ((w <= 0) || (h <= 0))
         {
-            w = 640;
-            h = 480;
+            w = 1920;
+            h = 1080;
         }
 
         MythDisplayMode dscr(w, h, -1, -1, -1.0, 0);
