diff --speed-large-files --no-dereference --minimal -Naur mythtv-634c871972801165623818836b3e662589e70b23/mythplugins/configure mythtv-634c871972801165623818836b3e662589e70b23/mythplugins/configure
--- mythtv-634c871972801165623818836b3e662589e70b23/mythplugins/configure	2024-01-13 15:15:45.000000000 +0100
+++ mythtv-634c871972801165623818836b3e662589e70b23/mythplugins/configure	2024-01-14 16:39:53.399427095 +0100
@@ -322,6 +322,7 @@
 dcraw
 cdio
 musicbrainz
+cross_compile
 "
 
 DEPEND_LIST="
@@ -363,6 +364,7 @@
   --enable-all             Enable all options
   --previous               use previous configure parameters if possible
   --extra-ldflags=ELDFLAGS add ELDFLAGS to LDFLAGS [$LDFLAGS]
+  --enable-cross-compile   assume a cross-compiler is used
 
 MythArchive related options:
   --enable-mytharchive     build the mytharchive plugin [$archive]
@@ -422,6 +424,7 @@
     python
     sysroot
     mythroot
+    cross_compile
 "
 
 CONFIGURATION_OPTS=""
@@ -491,7 +494,11 @@
 check_py_lib(){
     log check_py_lib "$@"
     lib=${1}
-    check_cmd $python <<EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Python:$lib Be sure You have it for cross target!"
+        return 0
+    else
+        check_cmd $python <<EOF
 import sys
 try:
     from distutils.sysconfig import get_python_lib
@@ -515,6 +522,7 @@
     sys.exit(1)
 sys.exit(0)
 EOF
+    fi
 }
 
 # set temporary file name
@@ -675,7 +683,12 @@
 if enabled weather; then
     check_pl_lib(){
         log check_pl_lib "$@"
-        check_cmd perl -e "use ${1};"
+        if enabled cross_compile ; then
+            echo "You are cross-compiling! Assuming YES for Perl:$@ Be sure You have it for cross target!"
+            return 0
+        else
+            check_cmd perl -e "use ${1};"
+        fi
     }
 
     disable_weather(){
diff --speed-large-files --no-dereference --minimal -Naur mythtv-634c871972801165623818836b3e662589e70b23/mythtv/configure mythtv-634c871972801165623818836b3e662589e70b23/mythtv/configure
--- mythtv-634c871972801165623818836b3e662589e70b23/mythtv/configure	2024-01-13 15:15:45.000000000 +0100
+++ mythtv-634c871972801165623818836b3e662589e70b23/mythtv/configure	2024-01-14 16:38:50.399428450 +0100
@@ -1239,30 +1239,44 @@
 }
 
 is_python3() {
-    log is_python3
-    check_cmd $python << EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for IsPython3 . Be sure You have it for cross target!"
+        return 0
+    else
+        log is_python3
+        check_cmd $python << EOF
 import sys
 if sys.version_info > (3,0):
     sys.exit(0)
 else:
     sys.exit(1)
 EOF
+    fi
 }
 
 check_python(){
-    log check_python "$@"
-    version=${1}
-    check_cmd $python <<EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Python ver > 3.6.0. Be sure You have it for cross target!"
+        return 0
+    else
+        log check_python "$@"
+        version=${1}
+        check_cmd $python <<EOF
 import sys
 if sys.version_info < $version:
     sys.exit(1)
 EOF
+    fi
 }
 
 check_py_lib(){
     log check_py_lib "$@"
     lib=${1}
-    check_cmd $python <<EOF
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Python:$lib Be sure You have it for cross target!"
+        return 0
+    else
+        check_cmd $python <<EOF
 import sys
 try:
     import $lib
@@ -1270,6 +1284,7 @@
     sys.exit(1)
 sys.exit(0)
 EOF
+    fi
 }
 
 check_py_lib_version(){
@@ -1292,7 +1307,12 @@
 
 check_pl_lib(){
     log check_pl_lib "$@"
-    check_cmd perl -e "use ${1};"
+    if enabled cross_compile ; then
+        echo "You are cross-compiling! Assuming YES for Perl:$@ Be sure You have it for cross target!"
+        return 0
+    else
+        check_cmd perl -e "use ${1};"
+    fi
 }
 
 check_ld_cxx(){
diff --speed-large-files --no-dereference --minimal -Naur mythtv-e3a18ec40b25fde8a17fabdedaf6b1eade8a2466/mythtv/configure mythtv-e3a18ec40b25fde8a17fabdedaf6b1eade8a2466/mythtv/configure
--- mythtv-e3a18ec40b25fde8a17fabdedaf6b1eade8a2466/mythtv/configure	2024-01-27 12:42:21.023189471 +0100
+++ mythtv-e3a18ec40b25fde8a17fabdedaf6b1eade8a2466/mythtv/configure	2024-01-27 12:40:30.553191849 +0100
@@ -6592,8 +6592,8 @@
     check_py_lib MySQLdb       || disable_bindings_python "MySQLdb"
     check_py_lib lxml          || disable_bindings_python "lxml"
     check_py_lib requests      || disable_bindings_python "requests"
-    check_python "(3,11,1)" && check_py_lib wheel && check_py_lib_version pip "(23,0,1)" && USE_PYTHON_PIP="yes"
-    test x"$USE_PYTHON_PIP" = x"" && check_python "(3,12,0)" && disable_bindings_python "pip, wheel"
+    #check_python "(3,11,1)" && check_py_lib wheel && check_py_lib_version pip "(23,0,1)" && USE_PYTHON_PIP="yes"
+    #test x"$USE_PYTHON_PIP" = x"" && check_python "(3,12,0)" && disable_bindings_python "pip, wheel"
 fi
 
 # Check for perl dependencies
