diff --git a/mythtv/libs/libmythui/opengl/mythpainterwindowopengl.cpp b/mythtv/libs/libmythui/opengl/mythpainterwindowopengl.cpp
index 2334cf1c65..b0f64c5960 100644
--- a/mythtv/libs/libmythui/opengl/mythpainterwindowopengl.cpp
+++ b/mythtv/libs/libmythui/opengl/mythpainterwindowopengl.cpp
@@ -1,7 +1,10 @@
-// MythTV
+#include "opengl/mythpainterwindowopengl.h"
+
+#include <QGuiApplication>
+#include <QWindow>
+
 #include "libmythbase/mythlogging.h"
 #include "mythmainwindow.h"
-#include "opengl/mythpainterwindowopengl.h"
 
 #define LOC QString("GLPaintWin: ")
 
@@ -12,6 +15,13 @@ MythPainterWindowOpenGL::MythPainterWindowOpenGL(MythMainWindow *MainWin)
     setAttribute(Qt::WA_NoSystemBackground);
     setAttribute(Qt::WA_NativeWindow);
     setAttribute(Qt::WA_DontCreateNativeAncestors);
+    // The eglfs QPA platform works without setting the surface type and
+    // can only have one OpenGLSurface, which must be the top level widget
+    // (which for us is currently the MythMainWindow).
+    if (QGuiApplication::platformName() != "eglfs" && windowHandle() != nullptr)
+    {
+        windowHandle()->setSurfaceType(QWindow::OpenGLSurface);
+    }
     winId();
 #ifdef Q_OS_MACOS
      // must be visible before OpenGL initialisation on OSX
diff --git a/mythtv/libs/libmythui/opengl/mythrenderopengl.cpp b/mythtv/libs/libmythui/opengl/mythrenderopengl.cpp
index f4f709bfe8..850b922eb6 100644
--- a/mythtv/libs/libmythui/opengl/mythrenderopengl.cpp
+++ b/mythtv/libs/libmythui/opengl/mythrenderopengl.cpp
@@ -24,11 +24,6 @@
 
 #define LOC QString("OpenGL: ")
 
-#ifdef Q_OS_ANDROID
-#include <android/log.h>
-#include <QWindow>
-#endif
-
 static constexpr GLuint VERTEX_INDEX  { 0 };
 static constexpr GLuint COLOR_INDEX   { 1 };
 static constexpr GLuint TEXTURE_INDEX { 2 };
@@ -96,7 +91,6 @@ MythRenderOpenGL* MythRenderOpenGL::Create(QWidget *Widget)
     bool opengles = !qEnvironmentVariableIsEmpty("MYTHTV_OPENGL_ES");
     bool core     = !qEnvironmentVariableIsEmpty("MYTHTV_OPENGL_CORE");
     QSurfaceFormat format = QSurfaceFormat::defaultFormat();
-    format.setRenderableType(QSurfaceFormat::OpenGL);
     if (core)
     {
         format.setProfile(QSurfaceFormat::CoreProfile);
@@ -118,7 +112,6 @@ MythRenderOpenGL* MythRenderOpenGL::Create(QWidget *Widget)
     if (VERBOSE_LEVEL_CHECK(VB_GPU, LOG_INFO))
         format.setOption(QSurfaceFormat::DebugContext);
 
-    QSurfaceFormat::setDefaultFormat(format);
     return new MythRenderOpenGL(format, Widget);
 }
 
@@ -544,17 +537,20 @@ void MythRenderOpenGL::SetWidget(QWidget *Widget)
         return;
     }
 
-#if defined(Q_OS_ANDROID) || (QT_VERSION > QT_VERSION_CHECK(6,3,0))
+#if 0 //defined(Q_OS_ANDROID) || (QT_VERSION > QT_VERSION_CHECK(6,3,0))
     // Ensure surface type is always OpenGL
+    if (QGuiApplication::platformName() != "eglfs")
+    {
     m_window->setSurfaceType(QWindow::OpenGLSurface);
     m_window->destroy();
     m_window->create();
-    if (native && native->windowHandle() != m_window)
+    if (false && native && native->windowHandle() && native->windowHandle() != m_window)
     {
         native->windowHandle()->setSurfaceType(QWindow::OpenGLSurface);
         native->windowHandle()->destroy();
         native->windowHandle()->create();
     }
+    }
 #endif
 
 #ifdef CONFIG_QTWEBENGINE
 