diff --speed-large-files --no-dereference --minimal -Naur mythtv-6705a26e463a07f866ff93bcac955aa7166906cb/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp mythtv-6705a26e463a07f866ff93bcac955aa7166906cb/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp
--- mythtv-6705a26e463a07f866ff93bcac955aa7166906cb/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2024-10-15 07:14:12.000000000 +0200
+++ mythtv-6705a26e463a07f866ff93bcac955aa7166906cb/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2024-10-16 18:25:05.026724671 +0200
@@ -508,7 +508,7 @@
 
     // N.B. Interop support should already have been checked
     // Create the device context
-    auto * hwdeviceref = MythCodecContext::CreateDevice(AV_HWDEVICE_TYPE_DRM, nullptr);
+    auto * hwdeviceref = MythCodecContext::CreateDevice(AV_HWDEVICE_TYPE_V4L2REQUEST, nullptr);
     if (!hwdeviceref)
         return -1;
 
