diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavutil/buffer.c mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavutil/buffer.c
--- mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavutil/buffer.c	2020-01-14 09:19:45.099999999 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavutil/buffer.c	2020-01-14 09:19:46.416666665 +0100
@@ -272,6 +272,19 @@
     av_freep(&pool);
 }
 
+void av_buffer_pool_flush(AVBufferPool *pool)
+{
+    ff_mutex_lock(&pool->mutex);
+    while (pool->pool) {
+        BufferPoolEntry *buf = pool->pool;
+        pool->pool = buf->next;
+
+        buf->free(buf->opaque, buf->data);
+        av_freep(&buf);
+    }
+    ff_mutex_unlock(&pool->mutex);
+}
+
 void av_buffer_pool_uninit(AVBufferPool **ppool)
 {
     AVBufferPool *pool;
diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavutil/buffer.h mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavutil/buffer.h
--- mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavutil/buffer.h	2020-01-14 09:19:45.103333332 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavutil/buffer.h	2020-01-14 09:19:46.416666665 +0100
@@ -267,6 +267,11 @@
                                    void (*pool_free)(void *opaque));
 
 /**
+ * Free all available buffers in a buffer pool.
+ */
+ void av_buffer_pool_flush(AVBufferPool *pool);
+
+/**
  * Mark the pool as being available for freeing. It will actually be freed only
  * once all the allocated buffers associated with the pool are released. Thus it
  * is safe to call this function while some of the allocated buffers are still
