diff --speed-large-files --no-dereference --minimal -Naur mythtv-630afc10487ffdd7c9c9f7572cf7a78f39cb1d8b/mythtv/external/FFmpeg/libavcodec/h264_slice.c mythtv-630afc10487ffdd7c9c9f7572cf7a78f39cb1d8b/mythtv/external/FFmpeg/libavcodec/h264_slice.c
--- mythtv-630afc10487ffdd7c9c9f7572cf7a78f39cb1d8b/mythtv/external/FFmpeg/libavcodec/h264_slice.c	2022-01-25 22:15:51.149173718 +0100
+++ mythtv-630afc10487ffdd7c9c9f7572cf7a78f39cb1d8b/mythtv/external/FFmpeg/libavcodec/h264_slice.c	2022-01-25 22:11:25.102511848 +0100
@@ -1845,8 +1845,6 @@
         sl->idr_pic_id = get_ue_golomb_long(&sl->gb);
 
     pos = sl->gb.index;
-    sl->poc_lsb = 0;
-    sl->delta_poc_bottom = 0;
     if (sps->poc_type == 0) {
         sl->poc_lsb = get_bits(&sl->gb, sps->log2_max_poc_lsb);
 
@@ -1854,13 +1852,13 @@
             sl->delta_poc_bottom = get_se_golomb(&sl->gb);
     }
 
-    sl->delta_poc[0] = sl->delta_poc[1] = 0;
     if (sps->poc_type == 1 && !sps->delta_pic_order_always_zero_flag) {
         sl->delta_poc[0] = get_se_golomb(&sl->gb);
 
         if (pps->pic_order_present == 1 && picture_structure == PICT_FRAME)
             sl->delta_poc[1] = get_se_golomb(&sl->gb);
     }
+    sl->pic_order_cnt_bit_size = sl->gb.index - pos;
 
     sl->redundant_pic_count = 0;
     if (pps->redundant_pic_cnt_present)
@@ -1900,9 +1898,11 @@
 
     sl->explicit_ref_marking = 0;
     if (nal->ref_idc) {
+        pos = sl->gb.index;
         ret = ff_h264_decode_ref_pic_marking(sl, &sl->gb, nal, h->avctx);
         if (ret < 0 && (h->avctx->err_recognition & AV_EF_EXPLODE))
             return AVERROR_INVALIDDATA;
+        sl->ref_pic_marking_bit_size = sl->gb.index - pos;
     }
 
     if (sl->slice_type_nos != AV_PICTURE_TYPE_I && pps->cabac) {
