diff -Naur mythtv-master-20200530-ga180830-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c mythtv-master-20200530-ga180830-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c
--- mythtv-master-20200530-ga180830-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-05-30 13:01:13.976666668 +0200
+++ mythtv-master-20200530-ga180830-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-05-30 13:01:15.433333334 +0200
@@ -43,6 +43,11 @@
     AVDRMDeviceContext *hwctx = hwdev->hwctx;
     drmVersionPtr version;
 
+    if (device == NULL) {
+      hwctx->fd = -1;
+      return 0;
+    }
+
     hwctx->fd = open(device, O_RDWR);
     if (hwctx->fd < 0)
         return AVERROR(errno);
