diff -Naur mythtv-master-20200324-gaa46cdd-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c mythtv-master-20200324-gaa46cdd-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c
--- mythtv-master-20200324-gaa46cdd-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-03-24 09:42:18.970000001 +0100
+++ mythtv-master-20200324-gaa46cdd-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-03-24 09:42:20.360000000 +0100
@@ -43,6 +43,11 @@
     AVDRMDeviceContext *hwctx = hwdev->hwctx;
     drmVersionPtr version;
 
+    if (device == NULL) {
+      hwctx->fd = -1;
+      return 0;
+    }
+
     hwctx->fd = open(device, O_RDWR);
     if (hwctx->fd < 0)
         return AVERROR(errno);
