diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c
--- mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-01-14 09:19:52.686666666 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-01-14 09:19:54.493333332 +0100
@@ -43,6 +43,11 @@
     AVDRMDeviceContext *hwctx = hwdev->hwctx;
     drmVersionPtr version;
 
+    if (device == NULL) {
+      hwctx->fd = -1;
+      return 0;
+    }
+
     hwctx->fd = open(device, O_RDWR);
     if (hwctx->fd < 0)
         return AVERROR(errno);
