diff -Naur mythtv-master-20200207-g2869463-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c mythtv-master-20200207-g2869463-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c
--- mythtv-master-20200207-g2869463-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-02-08 14:19:07.530000002 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-02-08 14:19:09.343333336 +0100
@@ -43,6 +43,11 @@
     AVDRMDeviceContext *hwctx = hwdev->hwctx;
     drmVersionPtr version;
 
+    if (device == NULL) {
+      hwctx->fd = -1;
+      return 0;
+    }
+
     hwctx->fd = open(device, O_RDWR);
     if (hwctx->fd < 0)
         return AVERROR(errno);
