diff -Naur mythtv-master-20200116-gdd96d68-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c mythtv-master-20200116-gdd96d68-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c
--- mythtv-master-20200116-gdd96d68-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-01-16 10:17:07.706666668 +0100
+++ mythtv-master-20200116-gdd96d68-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-01-16 10:17:09.210000001 +0100
@@ -43,6 +43,11 @@
     AVDRMDeviceContext *hwctx = hwdev->hwctx;
     drmVersionPtr version;
 
+    if (device == NULL) {
+      hwctx->fd = -1;
+      return 0;
+    }
+
     hwctx->fd = open(device, O_RDWR);
     if (hwctx->fd < 0)
         return AVERROR(errno);
