diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c mythtv-master-20200507-ge586e5a-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c
--- mythtv-master-20200507-ge586e5a-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-05-07 12:46:28.460000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-05-07 12:46:30.056666667 +0200
@@ -43,6 +43,11 @@
     AVDRMDeviceContext *hwctx = hwdev->hwctx;
     drmVersionPtr version;
 
+    if (device == NULL) {
+      hwctx->fd = -1;
+      return 0;
+    }
+
     hwctx->fd = open(device, O_RDWR);
     if (hwctx->fd < 0)
         return AVERROR(errno);
