diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c mythtv-master-20191227-gf15b4f0-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c
--- mythtv-master-20191227-gf15b4f0-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2019-12-27 12:19:18.950000000 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2019-12-27 12:19:20.240000001 +0100
@@ -43,6 +43,11 @@
     AVDRMDeviceContext *hwctx = hwdev->hwctx;
     drmVersionPtr version;
 
+    if (device == NULL) {
+      hwctx->fd = -1;
+      return 0;
+    }
+
     hwctx->fd = open(device, O_RDWR);
     if (hwctx->fd < 0)
         return AVERROR(errno);
