diff -Naur mythtv-master-20191212-g4fddad7-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c mythtv-master-20191212-g4fddad7-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c
--- mythtv-master-20191212-g4fddad7-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2019-12-12 19:16:15.593333335 +0100
+++ mythtv-master-20191212-g4fddad7-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2019-12-12 19:16:17.210000002 +0100
@@ -43,6 +43,11 @@
     AVDRMDeviceContext *hwctx = hwdev->hwctx;
     drmVersionPtr version;
 
+    if (device == NULL) {
+      hwctx->fd = -1;
+      return 0;
+    }
+
     hwctx->fd = open(device, O_RDWR);
     if (hwctx->fd < 0)
         return AVERROR(errno);
