diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavcodec/h264dec.h mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavcodec/h264dec.h
--- mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavcodec/h264dec.h	2020-01-14 09:19:54.043333332 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavcodec/h264dec.h	2020-01-14 09:19:56.053333332 +0100
@@ -184,6 +184,8 @@
     int slice_type_nos;         ///< S free slice type (SI/SP are remapped to I/P)
     int slice_type_fixed;
 
+    int idr_pic_id;
+
     int qscale;
     int chroma_qp[2];   // QPc
     int qp_thresh;      ///< QP threshold to skip loopfilter
diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavcodec/h264_slice.c mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavcodec/h264_slice.c
--- mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavcodec/h264_slice.c	2020-01-14 09:19:54.073333332 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavcodec/h264_slice.c	2020-01-14 09:19:56.053333332 +0100
@@ -1823,7 +1823,7 @@
     }
 
     if (nal->type == H264_NAL_IDR_SLICE)
-        get_ue_golomb_long(&sl->gb); /* idr_pic_id */
+        sl->idr_pic_id = get_ue_golomb_long(&sl->gb);
 
     if (sps->poc_type == 0) {
         sl->poc_lsb = get_bits(&sl->gb, sps->log2_max_poc_lsb);
diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavcodec/v4l2_request_h264.c mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavcodec/v4l2_request_h264.c
--- mythtv-master-20200114-ga315cd6-old/mythtv/external/FFmpeg/libavcodec/v4l2_request_h264.c	2020-01-14 09:19:54.089999999 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/external/FFmpeg/libavcodec/v4l2_request_h264.c	2020-01-14 09:19:56.053333332 +0100
@@ -301,7 +301,7 @@
         .pic_parameter_set_id = sl->pps_id,
         .colour_plane_id = 0, /* what is this? */
         .frame_num = h->poc.frame_num,
-        .idr_pic_id = 0, /* what is this? */
+        .idr_pic_id = sl->idr_pic_id,
         .pic_order_cnt_lsb = sl->poc_lsb,
         .delta_pic_order_cnt_bottom = sl->delta_poc_bottom,
         .delta_pic_order_cnt0 = sl->delta_poc[0],
