diff -Naur mythtv-master-20191212-g4fddad7-old/mythtv/external/FFmpeg/libavcodec/h264-ctrls.h mythtv-master-20191212-g4fddad7-new/mythtv/external/FFmpeg/libavcodec/h264-ctrls.h
--- mythtv-master-20191212-g4fddad7-old/mythtv/external/FFmpeg/libavcodec/h264-ctrls.h	2019-12-12 19:16:19.433333335 +0100
+++ mythtv-master-20191212-g4fddad7-new/mythtv/external/FFmpeg/libavcodec/h264-ctrls.h	2019-12-12 19:16:20.806666668 +0100
@@ -185,6 +185,10 @@
 #define V4L2_H264_DPB_ENTRY_FLAG_VALID		0x01
 #define V4L2_H264_DPB_ENTRY_FLAG_ACTIVE		0x02
 #define V4L2_H264_DPB_ENTRY_FLAG_LONG_TERM	0x04
+#define V4L2_H264_DPB_ENTRY_FLAG_FIELD_PICTURE	0x08
+#define V4L2_H264_DPB_ENTRY_FLAG_REF_TOP	0x10
+#define V4L2_H264_DPB_ENTRY_FLAG_REF_BOTTOM	0x20
+#define V4L2_H264_DPB_ENTRY_FLAG_REF_FRAME	0x30
 
 struct v4l2_h264_dpb_entry {
 	__u64 reference_ts;
diff -Naur mythtv-master-20191212-g4fddad7-old/mythtv/external/FFmpeg/libavcodec/v4l2_request_h264.c mythtv-master-20191212-g4fddad7-new/mythtv/external/FFmpeg/libavcodec/v4l2_request_h264.c
--- mythtv-master-20191212-g4fddad7-old/mythtv/external/FFmpeg/libavcodec/v4l2_request_h264.c	2019-12-12 19:16:19.623333336 +0100
+++ mythtv-master-20191212-g4fddad7-new/mythtv/external/FFmpeg/libavcodec/v4l2_request_h264.c	2019-12-12 19:16:20.806666668 +0100
@@ -66,10 +66,13 @@
     entry->frame_num = pic->frame_num;
     entry->pic_num = pic->pic_id;
     entry->flags = V4L2_H264_DPB_ENTRY_FLAG_VALID;
+    entry->flags |= (pic->reference & 3) << 4;
     if (pic->reference)
         entry->flags |= V4L2_H264_DPB_ENTRY_FLAG_ACTIVE;
     if (pic->long_ref)
         entry->flags |= V4L2_H264_DPB_ENTRY_FLAG_LONG_TERM;
+    if (pic->field_picture)
+        entry->flags |= V4L2_H264_DPB_ENTRY_FLAG_FIELD_PICTURE;
     if (pic->field_poc[0] != INT_MAX)
         entry->top_field_order_cnt = pic->field_poc[0];
     if (pic->field_poc[1] != INT_MAX)
@@ -109,7 +112,8 @@
         struct v4l2_h264_dpb_entry *entry = &decode->dpb[i];
         if ((entry->flags & V4L2_H264_DPB_ENTRY_FLAG_VALID) &&
             entry->reference_ts == timestamp)
-            return i;
+            // TODO: signal reference type, possible using top 2 bits
+            return i | ((ref->reference & 3) << 6);
     }
 
     return 0;
