diff -Naur mythtv-master-20200418-g8760389-old/mythtv/configure mythtv-master-20200418-g8760389-new/mythtv/configure
--- mythtv-master-20200418-g8760389-old/mythtv/configure	2020-04-18 10:54:48.459999999 +0200
+++ mythtv-master-20200418-g8760389-new/mythtv/configure	2020-04-18 10:54:50.193333332 +0200
@@ -125,6 +125,7 @@
   --disable-vbox           disable support for V@Box TV Gateway boxes
   --disable-ceton          disable support for Ceton cards
   --disable-v4l2           disable Video4Linux support
+  --enable-v4l2request     enable Video4Linux V4l2-request support
   --disable-ivtv           disable ivtv support (PVR-x50) req. v4l2 support
   --disable-hdpvr          disable HD-PVR support
   --disable-dvb            disable DVB support
@@ -1948,7 +1949,6 @@
     wma_freqs
     wmv2dsp
 "
-
 MYTHTV_CONFIG_LIST='
     backend
     bdjava
@@ -1986,6 +1986,7 @@
     taglib
     v4l2
     v4l2prime
+    v4l2request
     v4l1
     valgrind
     x11
@@ -2781,6 +2782,7 @@
 enable v4l1
 enable v4l2
 enable v4l2prime
+disable v4l2request
 enable x11
 enable xrandr
 disable indevs
@@ -4736,6 +4738,7 @@
         disable v4l1
         disable v4l2
         disable v4l2prime
+        disable v4l2request
         disable x11
         # Workaround compile errors from missing u_int/uint def
         CFLAGS=`echo $CFLAGS | sed 's/-D_POSIX_C_SOURCE=200112//'`
@@ -4786,6 +4789,7 @@
         disable v4l1
         disable v4l2
         disable v4l2prime
+        disable v4l2request
         enable  windows
         disable x11
         disable qtdbus
@@ -7236,6 +7240,10 @@
     ffopts="$ffopts --enable-gnutls"
 fi
 
+if enabled v4l2request; then
+    ffopts="$ffopts --enable-v4l2_request --enable-libdrm --enable-libudev"
+fi
+
 ffmpeg_extra_cflags="$extra_cflags -w"
 
 ## Call FFmpeg configure here
@@ -7437,7 +7445,7 @@
   echo "NVDEC support             ${nvdec-no}"
 fi
   echo "DRM support               ${drm-no}"
-  echo "Video4Linux codecs        ${v4l2-no} (DRM ${v4l2prime-no})"
+  echo "Video4Linux codecs        ${v4l2-no} (DRM:${v4l2prime-no} v4l2-request:${v4l2request-no})"
   echo "MMAL decoder support      ${mmal-no}"
   echo "OpenGL                    ${opengl-no} (OpenGLES ${opengles-no})"
   echo "EGL support               ${egl-no}"
