diff -Naur mythtv-master-20191218-g410b405-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20191218-g410b405-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20191218-g410b405-old/mythtv/libs/libmythtv/mythplayer.cpp	2019-12-19 14:46:30.896666666 +0100
+++ mythtv-master-20191218-g410b405-new/mythtv/libs/libmythtv/mythplayer.cpp	2019-12-19 14:48:10.206666643 +0100
@@ -1587,7 +1587,7 @@
     auto playspeed1000 = static_cast<int64_t>(1000.0F / m_playSpeed);
     bool reset = false;
     // controller gain
-    static float const s_av_control_gain = 0.4F;
+    static float const s_av_control_gain = 0.1F;
     // time weighted exponential filter coefficient
     static float const s_sync_fc = 0.9F;
 
