diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c mythtv-master-20200507-ge586e5a-new/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c
--- mythtv-master-20200507-ge586e5a-old/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c	2020-05-07 12:46:54.300000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c	2020-05-07 12:46:56.336666667 +0200
@@ -135,14 +135,10 @@
     V4L2m2mContext *s = ((V4L2m2mPriv*)avctx->priv_data)->context;
     V4L2Context *const capture = &s->capture;
     V4L2Context *const output = &s->output;
-    AVPacket avpkt = {0};
     int ret;
 
-    if (s->buf_pkt.size) {
-        avpkt = s->buf_pkt;
-        memset(&s->buf_pkt, 0, sizeof(AVPacket));
-    } else {
-        ret = ff_decode_get_packet(avctx, &avpkt);
+    if (!s->buf_pkt.size) {
+        ret = ff_decode_get_packet(avctx, &s->buf_pkt);
         if (ret < 0 && ret != AVERROR_EOF)
             return ret;
     }
@@ -150,20 +146,17 @@
     if (s->draining)
         goto dequeue;
 
-    ret = ff_v4l2_context_enqueue_packet(output, &avpkt);
-    if (ret < 0) {
-        if (ret != AVERROR(EAGAIN))
-           return ret;
-
-        s->buf_pkt = avpkt;
+    ret = ff_v4l2_context_enqueue_packet(output, &s->buf_pkt);
+    if (ret == AVERROR(EAGAIN)) {
         /* no input buffers available, continue dequeing */
-    }
+    } else {
+        av_packet_unref(&s->buf_pkt);
+
+        if (ret < 0)
+            return ret;
 
-    if (avpkt.size) {
         ret = v4l2_try_start(avctx);
         if (ret) {
-            av_packet_unref(&avpkt);
-
             /* cant recover */
             if (ret == AVERROR(ENOMEM))
                 return ret;
@@ -173,8 +166,6 @@
     }
 
 dequeue:
-    if (!s->buf_pkt.size)
-        av_packet_unref(&avpkt);
     return ff_v4l2_context_dequeue_frame(capture, frame, -1);
 }
 
