diff -Naur mythtv-master-20200530-ga180830-old/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c mythtv-master-20200530-ga180830-new/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c
--- mythtv-master-20200530-ga180830-old/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c	2020-05-30 13:01:36.296666668 +0200
+++ mythtv-master-20200530-ga180830-new/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c	2020-05-30 13:01:37.470000001 +0200
@@ -137,15 +137,19 @@
     V4L2Context *const output = &s->output;
     int ret;
 
+    if (s->draining)
+        goto dequeue;
+
     if (!s->buf_pkt.size) {
         ret = ff_decode_get_packet(avctx, &s->buf_pkt);
-        if (ret < 0 && ret != AVERROR_EOF)
+        if (ret == AVERROR(EAGAIN))
+            goto dequeue;
+        else if (ret == AVERROR_EOF)
+            return ff_v4l2_context_enqueue_packet(output, &s->buf_pkt); /* null pkt */
+        else if (ret < 0)
             return ret;
     }
 
-    if (s->draining)
-        goto dequeue;
-
     ret = ff_v4l2_context_enqueue_packet(output, &s->buf_pkt);
     if (ret == AVERROR(EAGAIN)) {
         /* no input buffers available, continue dequeing */
