diff -Naur mythtv-master-20200530-ga180830-old/mythtv/external/FFmpeg/libavdevice/v4l2.c mythtv-master-20200530-ga180830-new/mythtv/external/FFmpeg/libavdevice/v4l2.c
--- mythtv-master-20200530-ga180830-old/mythtv/external/FFmpeg/libavdevice/v4l2.c	2020-05-30 13:01:38.466666667 +0200
+++ mythtv-master-20200530-ga180830-new/mythtv/external/FFmpeg/libavdevice/v4l2.c	2020-05-30 13:01:40.320000000 +0200
@@ -538,11 +538,10 @@
             s->frame_size = buf.bytesused;
 
         if (s->frame_size > 0 && buf.bytesused != s->frame_size) {
-            av_log(ctx, AV_LOG_ERROR,
+            av_log(ctx, AV_LOG_WARNING,
                    "Dequeued v4l2 buffer contains %d bytes, but %d were expected. Flags: 0x%08X.\n",
                    buf.bytesused, s->frame_size, buf.flags);
-            enqueue_buffer(s, &buf);
-            return AVERROR_INVALIDDATA;
+            buf.bytesused = 0;
         }
     }
 
