diff -Naur mythtv-master-20200311-g625fc71-old/mythtv/external/FFmpeg/libavdevice/v4l2.c mythtv-master-20200311-g625fc71-new/mythtv/external/FFmpeg/libavdevice/v4l2.c
--- mythtv-master-20200311-g625fc71-old/mythtv/external/FFmpeg/libavdevice/v4l2.c	2020-03-14 13:30:42.676666667 +0100
+++ mythtv-master-20200311-g625fc71-new/mythtv/external/FFmpeg/libavdevice/v4l2.c	2020-03-14 13:30:45.420000001 +0100
@@ -538,11 +538,10 @@
             s->frame_size = buf.bytesused;
 
         if (s->frame_size > 0 && buf.bytesused != s->frame_size) {
-            av_log(ctx, AV_LOG_ERROR,
+            av_log(ctx, AV_LOG_WARNING,
                    "Dequeued v4l2 buffer contains %d bytes, but %d were expected. Flags: 0x%08X.\n",
                    buf.bytesused, s->frame_size, buf.flags);
-            enqueue_buffer(s, &buf);
-            return AVERROR_INVALIDDATA;
+            buf.bytesused = 0;
         }
     }
 
