diff -Naur mythtv-master-20200128-ga4d1c3f-old/mythtv/external/FFmpeg/libavcodec/v4l2_m2m.c mythtv-master-20200128-ga4d1c3f-new/mythtv/external/FFmpeg/libavcodec/v4l2_m2m.c
--- mythtv-master-20200128-ga4d1c3f-old/mythtv/external/FFmpeg/libavcodec/v4l2_m2m.c	2020-01-28 17:04:16.000000000 +0100
+++ mythtv-master-20200128-ga4d1c3f-new/mythtv/external/FFmpeg/libavcodec/v4l2_m2m.c	2020-01-29 21:49:49.923333332 +0100
@@ -101,6 +101,7 @@
 {
     void *log_ctx = s->priv;
     int ret;
+    struct v4l2_format ofmt, cfmt;
 
     s->fd = open(s->devname, O_RDWR | O_NONBLOCK, 0);
     if (s->fd < 0)
@@ -149,6 +150,16 @@
 
     ofmt = s->output.format;
     cfmt = s->capture.format;
+    av_log(log_ctx, AV_LOG_INFO, "requesting formats: output=%s capture=%s\n",
+                                 av_fourcc2str(V4L2_TYPE_IS_MULTIPLANAR(ofmt.type) ?
+                                               ofmt.fmt.pix_mp.pixelformat :
+                                               ofmt.fmt.pix.pixelformat),
+                                 av_fourcc2str(V4L2_TYPE_IS_MULTIPLANAR(cfmt.type) ?
+                                               cfmt.fmt.pix_mp.pixelformat :
+                                               cfmt.fmt.pix.pixelformat));
+
+    ofmt = s->output.format;
+    cfmt = s->capture.format;
     av_log(log_ctx, AV_LOG_INFO, "requesting formats: output=%s capture=%s\n",
                                  av_fourcc2str(V4L2_TYPE_IS_MULTIPLANAR(ofmt.type) ?
                                                ofmt.fmt.pix_mp.pixelformat :
