diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2021-03-04 20:26:42.483332939 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2021-03-04 20:26:49.419999605 +0100
@@ -199,8 +199,10 @@
     // best guess currently - this matches the number of capture buffers to the
     // number of output buffers - and hence to the number of video buffers for
     // direct rendering
-    LOG(VB_PLAYBACK, LOG_INFO, LOC + "Setting number of capture buffers to 6");
-    av_opt_set_int(Context->priv_data, "num_capture_buffers", 6, 0);
+    LOG(VB_PLAYBACK, LOG_INFO, LOC + "Setting number of capture buffers to 16");
+    av_opt_set_int(Context->priv_data, "num_capture_buffers", 16, 0);
+    LOG(VB_PLAYBACK, LOG_INFO, LOC + "Setting number of output buffers to 8");
+    av_opt_set_int(Context->priv_data, "num_output_buffers", 8, 0);
 }
 
 /*! \brief Retrieve a frame from CPU memory
