diff --speed-large-files --no-dereference --minimal -Naur mythtv-675d5944ce23c8bbe0629750b5aeac63dac09fca/mythtv/external/FFmpeg/libavcodec/v4l2_request.c mythtv-675d5944ce23c8bbe0629750b5aeac63dac09fca/mythtv/external/FFmpeg/libavcodec/v4l2_request.c
--- mythtv-675d5944ce23c8bbe0629750b5aeac63dac09fca/mythtv/external/FFmpeg/libavcodec/v4l2_request.c	2021-12-15 12:26:14.706658787 +0100
+++ mythtv-675d5944ce23c8bbe0629750b5aeac63dac09fca/mythtv/external/FFmpeg/libavcodec/v4l2_request.c	2021-12-13 17:14:02.806212082 +0100
@@ -1037,6 +1037,9 @@
     case AV_CODEC_ID_VP8:
         hwfc->initial_pool_size += 3;
         break;
+    case AV_CODEC_ID_H264:
+        hwfc->initial_pool_size += 23;
+        break;
     default:
         hwfc->initial_pool_size += 2;
     }
