diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/external/FFmpeg/libavcodec/golomb.h mythtv-master-20210225-gc0426e3fea-new/mythtv/external/FFmpeg/libavcodec/golomb.h
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/external/FFmpeg/libavcodec/golomb.h	2021-03-04 20:26:47.959999605 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/external/FFmpeg/libavcodec/golomb.h	2021-03-04 20:26:54.003332939 +0100
@@ -91,7 +91,7 @@
         LAST_SKIP_BITS(re, gb, 32 - log);
         CLOSE_READER(re, gb);
         if (log < 7) {
-            av_log(NULL, AV_LOG_ERROR, "Invalid UE golomb code\n");
+            av_log(NULL, AV_LOG_DEBUG, "Invalid UE golomb code\n");
             return AVERROR_INVALIDDATA;
         }
         buf >>= log;
