diff -Naur mythtv-master-20210424-g69d12b67b6-old/mythtv/external/FFmpeg/libavcodec/golomb.h mythtv-master-20210424-g69d12b67b6-new/mythtv/external/FFmpeg/libavcodec/golomb.h
--- mythtv-master-20210424-g69d12b67b6-old/mythtv/external/FFmpeg/libavcodec/golomb.h	2021-04-24 16:57:29.513332374 +0200
+++ mythtv-master-20210424-g69d12b67b6-new/mythtv/external/FFmpeg/libavcodec/golomb.h	2021-04-24 16:57:35.059999041 +0200
@@ -91,7 +91,7 @@
         LAST_SKIP_BITS(re, gb, 32 - log);
         CLOSE_READER(re, gb);
         if (log < 7) {
-            av_log(NULL, AV_LOG_ERROR, "Invalid UE golomb code\n");
+            av_log(NULL, AV_LOG_DEBUG, "Invalid UE golomb code\n");
             return AVERROR_INVALIDDATA;
         }
         buf >>= log;
