diff --speed-large-files --no-dereference --minimal -Naur mythtv-a0f6508475c73c3853619c8e974a84dd9507864e/mythtv/external/FFmpeg/libavcodec/v4l2_request.c mythtv-a0f6508475c73c3853619c8e974a84dd9507864e/mythtv/external/FFmpeg/libavcodec/v4l2_request.c
--- mythtv-a0f6508475c73c3853619c8e974a84dd9507864e/mythtv/external/FFmpeg/libavcodec/v4l2_request.c	2021-12-17 11:02:47.399939581 +0100
+++ mythtv-a0f6508475c73c3853619c8e974a84dd9507864e/mythtv/external/FFmpeg/libavcodec/v4l2_request.c	2021-12-17 10:37:57.396633133 +0100
@@ -817,10 +817,10 @@
             av_log(avctx, AV_LOG_ERROR, "%s: capture stream off failed, %s (%d)\n", __func__, strerror(errno), errno);
     }
 
-    if (avctx->hw_frames_ctx) {
-        AVHWFramesContext *hwfc = (AVHWFramesContext*)avctx->hw_frames_ctx->data;
-        av_buffer_pool_flush(hwfc->pool);
-    }
+    //if (avctx->hw_frames_ctx) {
+    //    AVHWFramesContext *hwfc = (AVHWFramesContext*)avctx->hw_frames_ctx->data;
+    //    av_buffer_pool_flush(hwfc->pool);
+    //}
 
     if (ctx->video_fd >= 0)
         close(ctx->video_fd);
@@ -1003,7 +1003,7 @@
 {
     av_log(NULL, AV_LOG_DEBUG, "%s: hwfc=%p pool=%p\n", __func__, hwfc, hwfc->pool);
 
-    av_buffer_pool_flush(hwfc->pool);
+    //av_buffer_pool_flush(hwfc->pool);
     av_buffer_pool_uninit(&hwfc->pool);
 }
 
