diff -Naur mythtv-master-20200530-ga180830-old/mythtv/external/FFmpeg/libavcodec/v4l2_context.c mythtv-master-20200530-ga180830-new/mythtv/external/FFmpeg/libavcodec/v4l2_context.c
--- mythtv-master-20200530-ga180830-old/mythtv/external/FFmpeg/libavcodec/v4l2_context.c	2020-05-30 13:01:45.733333334 +0200
+++ mythtv-master-20200530-ga180830-new/mythtv/external/FFmpeg/libavcodec/v4l2_context.c	2020-05-30 13:01:47.240000001 +0200
@@ -393,6 +393,15 @@
             return NULL;
         }
 
+        if (ctx_to_m2mctx(ctx)->draining && !V4L2_TYPE_IS_OUTPUT(ctx->type)) {
+            int bytesused = V4L2_TYPE_IS_MULTIPLANAR(buf.type) ?
+                            buf.m.planes[0].bytesused : buf.bytesused;
+            if (bytesused == 0) {
+                ctx->done = 1;
+                return NULL;
+            }
+        }
+
         avbuf = &ctx->buffers[buf.index];
         avbuf->status = V4L2BUF_AVAILABLE;
         avbuf->buf = buf;
