diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/external/FFmpeg/libavcodec/v4l2_context.c mythtv-master-20200507-ge586e5a-new/mythtv/external/FFmpeg/libavcodec/v4l2_context.c
--- mythtv-master-20200507-ge586e5a-old/mythtv/external/FFmpeg/libavcodec/v4l2_context.c	2020-05-07 12:43:56.000000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/external/FFmpeg/libavcodec/v4l2_context.c	2020-05-14 20:36:03.641283897 +0200
@@ -311,7 +311,9 @@
     }
 
     for (;;) {
+        av_log(logger(ctx), AV_LOG_WARNING, "%s Entering poll\n", ctx->name);
         ret = poll(&pfd, 1, timeout);
+        av_log(logger(ctx), AV_LOG_WARNING, "%s Leaving  poll - ret %d\n", ctx->name, ret);
         if (ret > 0)
             break;
         if (errno == EINTR)
