--- mythtv-master-20200904-gca3ee6a-old/mythtv/external/FFmpeg/libavcodec/v4l2_buffers.c
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/external/FFmpeg/libavcodec/v4l2_buffers.c
@@ -563,7 +563,6 @@ int ff_v4l2_buffer_avframe_to_buf(const AVFrame *frame, V4L2Buffer *out)
     if (frame->format == AV_PIX_FMT_DRM_PRIME) {
         AVDRMFrameDescriptor *drm_desc = (AVDRMFrameDescriptor *)frame->data[0];
         int i;
-        av_assert0(out->buf.memory == V4L2_MEMORY_DMABUF);
 
         if (V4L2_TYPE_IS_MULTIPLANAR(out->buf.type)) {
             for (i = 0; i < drm_desc->nb_objects; i++) {
@@ -671,8 +670,7 @@ int ff_v4l2_buffer_initialize(V4L2Buffer* avbuf, int index)
     V4L2Context *ctx = avbuf->context;
     int ret, i;
 
-    avbuf->buf.memory = ctx->av_pix_fmt == AV_PIX_FMT_DRM_PRIME && V4L2_TYPE_IS_OUTPUT(ctx->type) ?
-        V4L2_MEMORY_DMABUF : V4L2_MEMORY_MMAP;
+    avbuf->buf.memory = V4L2_MEMORY_MMAP;
     avbuf->buf.type = ctx->type;
     avbuf->buf.index = index;
 
--- mythtv-master-20200904-gca3ee6a-old/mythtv/external/FFmpeg/libavcodec/v4l2_context.c
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/external/FFmpeg/libavcodec/v4l2_context.c
@@ -794,8 +794,7 @@ int ff_v4l2_context_init(V4L2Context* ctx)
 
     memset(&req, 0, sizeof(req));
     req.count = ctx->num_buffers;
-    req.memory = ctx->av_pix_fmt == AV_PIX_FMT_DRM_PRIME && V4L2_TYPE_IS_OUTPUT(ctx->type) ?
-        V4L2_MEMORY_DMABUF : V4L2_MEMORY_MMAP;
+    req.memory = V4L2_MEMORY_MMAP;
     req.type = ctx->type;
     ret = ioctl(s->fd, VIDIOC_REQBUFS, &req);
     if (ret < 0) {
