diff -Naur mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/recordingprofile.cpp mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/recordingprofile.cpp
--- mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/recordingprofile.cpp	2020-09-25 09:21:48.046666668 +0200
+++ mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/recordingprofile.cpp	2020-09-25 09:21:49.986666668 +0200
@@ -1221,10 +1221,10 @@
     };
 };
 
-class RecordingType : public MythUIComboBoxSetting, public CodecParamStorage
+class RecordingTypeStream : public MythUIComboBoxSetting, public CodecParamStorage
 {
   public:
-    explicit RecordingType(const RecordingProfile &parent) :
+    explicit RecordingTypeStream(const RecordingProfile &parent) :
         MythUIComboBoxSetting(this), CodecParamStorage(this, parent, "recordingtype")
     {
         setLabel(QObject::tr("Recording Type"));
@@ -1633,7 +1633,7 @@
     }
     else if (type.toUpper() == "DVB")
     {
-        addChild(new RecordingType(*this));
+        addChild(new RecordingTypeStream(*this));
     }
 
     if (CardUtil::IsTunerSharingCapable(type))
