diff -Naur mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/videosource.cpp mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/videosource.cpp
--- mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/videosource.cpp	2020-09-25 09:21:49.876666668 +0200
+++ mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/videosource.cpp	2020-09-25 09:21:51.796666668 +0200
@@ -396,10 +396,10 @@
     }
 };
 
-class ScanFrequency : public MythUITextEditSetting
+class ScanFrequencyStart : public MythUITextEditSetting
 {
   public:
-    explicit ScanFrequency(const VideoSource &parent) :
+    explicit ScanFrequencyStart(const VideoSource &parent) :
         MythUITextEditSetting(new VideoSourceDBStorage(this, parent, "scanfrequency"))
     {
        setLabel(QObject::tr("Scan Frequency"));
@@ -673,7 +673,7 @@
     addChild(m_name = new Name(*this));
     addChild(new XMLTVGrabber(*this));
     addChild(new FreqTableSelector(*this));
-    addChild(new ScanFrequency(*this));
+    addChild(new ScanFrequencyStart(*this));
     addChild(new DVBNetID(*this, -1, -1));
     addChild(new BouquetID(*this, 0, 0));
     addChild(new RegionID(*this, 0, 0));
