diff -Naur mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/recorders/v4lchannel.h mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/recorders/v4lchannel.h
--- mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/recorders/v4lchannel.h	2020-09-25 09:21:51.700000001 +0200
+++ mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/recorders/v4lchannel.h	2020-09-25 09:21:53.683333334 +0200
@@ -10,7 +10,7 @@
 #include "frequencies.h"
 
 #ifdef USING_V4L2
-#include "videodev2.h" // needed for v4l2_std_id type
+#include <linux/videodev2.h>
 #endif
 
 using namespace std;
diff -Naur mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/v4l2util.h mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/v4l2util.h
--- mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmythtv/v4l2util.h	2020-09-25 09:21:51.693333334 +0200
+++ mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmythtv/v4l2util.h	2020-09-25 09:21:53.683333334 +0200
@@ -2,7 +2,7 @@
 #define V4L2_UTIL_H
 
 #ifdef USING_V4L2
-#include "videodev2.h" // our copy
+#include <linux/videodev2.h>
 #endif
 
 #include "tv.h"
