From 02e0ed592d804394bc5041a1594698760fd21ae3 Mon Sep 17 00:00:00 2001
From: Gary Buhrmaster <gary.buhrmaster@gmail.com>
Date: Tue, 14 Jul 2020 15:11:54 +0000
Subject: [PATCH 7/7] update config to pass enable-lto to ffmpeg and export
 appropriate QMAKE vars

---
 mythtv/configure | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/mythtv/configure b/mythtv/configure
index 9c231921e5..5709c9f1aa 100755
--- a/mythtv/configure
+++ b/mythtv/configure
@@ -6374,6 +6374,7 @@ check_optflags -fno-signed-zeros
 if enabled lto; then
     test "$cc_type" != "$ld_type" && die "LTO requires same compiler and linker"
     check_cflags  -flto
+    check_cxxflags -flto
     check_ldflags -flto $cpuflags
     disable inline_asm_direct_symbol_refs
 fi
@@ -7218,6 +7219,7 @@ ffmpeg_optset arch cross_prefix sysroot sysinclude cc cxx ld cpu
 ffmpeg_optset extra_cxxflags extra_ldflags target_os
 ffmpeg_optset pkg_config prefix libdir as objcc dep_cc host_cc
 ffmpeg_optset host_ld
+ffmpeg_optset ar nm ranlib
 ffmpeg_optenable cross_compile libmp3lame libx264 libx265 libvpx libxvid
 ffmpeg_optenable vdpau libxml2 libass dxva2
 ffmpeg_optenable libbluray libfontconfig libfreetype libiec61883
@@ -7256,6 +7258,10 @@ if enabled gnutls; then
     ffopts="$ffopts --enable-gnutls"
 fi
 
+if enabled lto; then
+    ffopts="$ffopts --enable-lto"
+fi
+
 ffmpeg_extra_cflags="$extra_cflags -w"
 
 ## Call FFmpeg configure here
@@ -7586,7 +7592,9 @@ DEPX86ASMFLAGS=\$(X86ASMFLAGS)
 AR=$ar
 ARFLAGS=$arflags
 AR_O=$ar_o
+QMAKE_AR=$ar $arflags
 RANLIB=$ranlib
+QMAKE_RANLIB=$ranlib
 STRIP=$strip
 STRIPTYPE=$striptype
 NVCC=$nvcc
-- 
2.26.2

