diff -Naur mythtv-master-20201109-g62af17b-old/mythtv/external/FFmpeg/libavcodec/hevcdec.c mythtv-master-20201109-g62af17b-new/mythtv/external/FFmpeg/libavcodec/hevcdec.c
--- mythtv-master-20201109-g62af17b-old/mythtv/external/FFmpeg/libavcodec/hevcdec.c	2020-11-11 16:25:26.626666667 +0100
+++ mythtv-master-20201109-g62af17b-new/mythtv/external/FFmpeg/libavcodec/hevcdec.c	2020-11-11 13:13:43.126666667 +0100
@@ -348,15 +348,6 @@
         avctx->colorspace      = AVCOL_SPC_UNSPECIFIED;
     }
 
-    avctx->chroma_sample_location = AVCHROMA_LOC_UNSPECIFIED;
-    if (sps->chroma_format_idc == 1) {
-        if (sps->vui.chroma_loc_info_present_flag) {
-            if (sps->vui.chroma_sample_loc_type_top_field <= 5)
-                avctx->chroma_sample_location = sps->vui.chroma_sample_loc_type_top_field + 1;
-        } else
-            avctx->chroma_sample_location = AVCHROMA_LOC_LEFT;
-    }
-
     if (vps->vps_timing_info_present_flag) {
         num = vps->vps_num_units_in_tick;
         den = vps->vps_time_scale;
