diff -Naur mythtv-master-20200324-gaa46cdd-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200324-gaa46cdd-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200324-gaa46cdd-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-03-24 09:42:39.040000000 +0100
+++ mythtv-master-20200324-gaa46cdd-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-03-24 09:42:40.593333333 +0100
@@ -780,6 +780,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
