diff -Naur mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-21 09:18:26.100000001 +0200
+++ mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-21 09:18:27.453333335 +0200
@@ -786,6 +786,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
