diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-04 11:36:05.586666667 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-04 11:36:06.893333333 +0200
@@ -786,6 +786,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
