diff -Naur mythtv-master-20190320-gb121bce-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20190320-gb121bce-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20190320-gb121bce-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-20 22:45:49.430000000 +0100
+++ mythtv-master-20190320-gb121bce-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-20 22:45:50.696666667 +0100
@@ -786,6 +786,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
