diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-12-27 12:19:25.603333334 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-12-27 12:19:27.376666667 +0100
@@ -782,6 +782,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
