diff -Naur mythtv-master-20200116-gdd96d68-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200116-gdd96d68-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200116-gdd96d68-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-01-16 10:17:15.616666668 +0100
+++ mythtv-master-20200116-gdd96d68-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-01-16 10:17:17.156666668 +0100
@@ -780,6 +780,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
