diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythtv/osd.cpp	2015-01-06 16:52:54.789895099 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythtv/osd.cpp	2015-01-06 16:53:17.913103912 +0100
@@ -528,7 +528,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
