diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-10-02 13:02:36.046666666 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-10-02 13:02:37.650000000 +0200
@@ -786,6 +786,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
