diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-05-07 12:47:05.176666667 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-05-07 12:47:06.730000000 +0200
@@ -780,6 +780,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
