diff -Naur mythtv-master-20210424-g69d12b67b6-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20210424-g69d12b67b6-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20210424-g69d12b67b6-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2021-04-24 16:57:38.879999041 +0200
+++ mythtv-master-20210424-g69d12b67b6-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2021-04-24 16:57:43.836665707 +0200
@@ -779,6 +779,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
