diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-29 08:40:37.153333334 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-29 08:40:38.993333334 +0100
@@ -786,6 +786,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
