diff -Naur mythtv-master-20140506-gd0e25ac-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20140506-gd0e25ac-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20140506-gd0e25ac-old/mythtv/libs/libmythtv/osd.cpp	2014-05-06 14:54:37.815245704 +0200
+++ mythtv-master-20140506-gd0e25ac-new/mythtv/libs/libmythtv/osd.cpp	2014-05-06 14:54:56.801844952 +0200
@@ -525,7 +525,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
