diff -Naur mythtv-master-20200109-gc116014-old/mythtv/libs/libmyth/mediamonitor-unix.cpp mythtv-master-20200109-gc116014-new/mythtv/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-master-20200109-gc116014-old/mythtv/libs/libmyth/mediamonitor-unix.cpp	2020-01-09 18:01:09.906666667 +0100
+++ mythtv-master-20200109-gc116014-new/mythtv/libs/libmyth/mediamonitor-unix.cpp	2020-01-09 18:01:11.163333334 +0100
@@ -193,7 +193,7 @@
 bool MediaMonitorUnix::CheckMountable(void)
 {
 #if CONFIG_QTDBUS
-    for (int i = 0; i < 10; ++i, usleep(500000))
+    for (int i = 0; i < 3; ++i, usleep(500000))
     {
         // Connect to UDisks.  This can sometimes fail if mythfrontend
         // is started during system init
