diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmyth/mediamonitor-unix.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmyth/mediamonitor-unix.cpp	2018-10-12 10:52:45.943333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmyth/mediamonitor-unix.cpp	2018-10-12 10:52:49.206666666 +0200
@@ -197,7 +197,7 @@
 bool MediaMonitorUnix::CheckMountable(void)
 {
 #if CONFIG_QTDBUS
-    for (int i = 0; i < 10; ++i, usleep(500000))
+    for (int i = 0; i < 3; ++i, usleep(500000))
     {
         // Connect to UDisks.  This can sometimes fail if mythfrontend
         // is started during system init
