diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmyth/mediamonitor-unix.cpp mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmyth/mediamonitor-unix.cpp	2019-12-27 12:19:26.676666667 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmyth/mediamonitor-unix.cpp	2019-12-27 12:19:28.566666667 +0100
@@ -193,7 +193,7 @@
 bool MediaMonitorUnix::CheckMountable(void)
 {
 #if CONFIG_QTDBUS
-    for (int i = 0; i < 10; ++i, usleep(500000))
+    for (int i = 0; i < 3; ++i, usleep(500000))
     {
         // Connect to UDisks.  This can sometimes fail if mythfrontend
         // is started during system init
