diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmyth/mediamonitor-unix.cpp mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmyth/mediamonitor-unix.cpp	2020-09-15 09:23:27.523333335 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmyth/mediamonitor-unix.cpp	2020-09-15 09:23:28.713333335 +0200
@@ -200,7 +200,7 @@
 bool MediaMonitorUnix::CheckMountable(void)
 {
 #if CONFIG_QTDBUS
-    for (int i = 0; i < 10; ++i, usleep(500000))
+    for (int i = 0; i < 3; ++i, usleep(500000))
     {
         // Connect to UDisks.  This can sometimes fail if mythfrontend
         // is started during system init
