diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythbase/mythmedia.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythbase/mythmedia.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythbase/mythmedia.cpp	2018-10-12 10:52:49.049999999 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythbase/mythmedia.cpp	2018-10-12 10:52:51.553333333 +0200
@@ -134,20 +134,29 @@
         // Build a command line for mount/unmount and execute it...
         // Is there a better way to do this?
         if (QFile(PATHTO_PMOUNT).exists() && QFile(PATHTO_PUMOUNT).exists())
-            MountCommand = QString("%1 %2")
-                .arg((DoMount) ? PATHTO_PMOUNT : PATHTO_PUMOUNT)
-                .arg(m_DevicePath);
+            if(DoMount)
+            {
+                MountCommand = QString("%1 %2 /media/disk")
+                    .arg(PATHTO_PMOUNT)
+                    .arg(m_DevicePath);
+            }
+            else
+            {
+                MountCommand = QString("%1 %2")
+                    .arg(PATHTO_PUMOUNT)
+                    .arg(m_DevicePath);
+            }
         else
             MountCommand = QString("%1 %2")
                 .arg((DoMount) ? PATHTO_MOUNT : PATHTO_UNMOUNT)
                 .arg(m_DevicePath);
 
-        LOG(VB_MEDIA, LOG_INFO, QString("Executing '%1'").arg(MountCommand));
+        LOG(VB_GENERAL, LOG_INFO, QString("Executing '%1'").arg(MountCommand));
         int ret = myth_system(MountCommand, kMSDontBlockInputDevs);
         if (ret !=  GENERIC_EXIT_OK)
         {
             usleep(300000);
-            LOG(VB_MEDIA, LOG_INFO, QString("Retrying '%1'").arg(MountCommand));
+            LOG(VB_GENERAL, LOG_INFO, QString("Retrying '%1'").arg(MountCommand));
             ret = myth_system(MountCommand, kMSDontBlockInputDevs);
         }
         if (ret == GENERIC_EXIT_OK)
@@ -162,14 +171,14 @@
                 usleep(1000000-1);
                 for (int tries = 2; !findMountPath() && tries > 0; --tries)
                 {
-                    LOG(VB_MEDIA, LOG_INFO,
+                    LOG(VB_GENERAL, LOG_INFO,
                         QString("Repeating '%1'").arg(MountCommand));
                     myth_system(MountCommand, kMSDontBlockInputDevs);
                     usleep(500000);
                 }
                 if (!findMountPath())
                 {
-                    LOG(VB_MEDIA, LOG_ERR, "performMountCmd() attempted to"
+                    LOG(VB_GENERAL, LOG_ERR, "performMountCmd() attempted to"
                                            " find mounted media, but failed?");
                     return false;
                 }
