diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythbase/mythmedia.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythbase/mythmedia.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythbase/mythmedia.cpp	2021-03-04 20:27:07.449999602 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythbase/mythmedia.cpp	2021-03-04 20:27:13.126666268 +0100
@@ -134,9 +134,18 @@
         // Is there a better way to do this?
         if (QFile(PATHTO_PMOUNT).exists() && QFile(PATHTO_PUMOUNT).exists())
         {
-            MountCommand = QString("%1 %2")
-                .arg((DoMount) ? PATHTO_PMOUNT : PATHTO_PUMOUNT)
-                .arg(m_devicePath);
+            if(DoMount)
+            {
+                MountCommand = QString("%1 %2 /media/disk")
+                    .arg(PATHTO_PMOUNT)
+                    .arg(m_devicePath);
+            }
+            else
+            {
+                MountCommand = QString("%1 %2")
+                    .arg(PATHTO_PUMOUNT)
+                    .arg(m_devicePath);
+            }
         }
         else
         {
@@ -145,12 +154,12 @@
                 .arg(m_devicePath);
         }
 
-        LOG(VB_MEDIA, LOG_INFO, QString("Executing '%1'").arg(MountCommand));
+        LOG(VB_GENERAL, LOG_INFO, QString("Executing '%1'").arg(MountCommand));
         int ret = myth_system(MountCommand, kMSDontBlockInputDevs);
         if (ret !=  GENERIC_EXIT_OK)
         {
             usleep(300000);
-            LOG(VB_MEDIA, LOG_INFO, QString("Retrying '%1'").arg(MountCommand));
+            LOG(VB_GENERAL, LOG_INFO, QString("Retrying '%1'").arg(MountCommand));
             ret = myth_system(MountCommand, kMSDontBlockInputDevs);
         }
         if (ret == GENERIC_EXIT_OK)
@@ -165,14 +174,14 @@
                 usleep(1000000-1);
                 for (int tries = 2; !findMountPath() && tries > 0; --tries)
                 {
-                    LOG(VB_MEDIA, LOG_INFO,
+                    LOG(VB_GENERAL, LOG_INFO,
                         QString("Repeating '%1'").arg(MountCommand));
                     myth_system(MountCommand, kMSDontBlockInputDevs);
                     usleep(500000);
                 }
                 if (!findMountPath())
                 {
-                    LOG(VB_MEDIA, LOG_ERR, "performMountCmd() attempted to"
+                    LOG(VB_GENERAL, LOG_ERR, "performMountCmd() attempted to"
                                            " find mounted media, but failed?");
                     return false;
                 }
