diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-10-02 13:02:39.706666667 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-10-02 13:02:41.406666667 +0200
@@ -465,10 +465,6 @@
     // Sanity check
     if (m_active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor disabled by user setting.");
-        return;
-    }
 
     if (!m_thread)
         m_thread = new MonitorThread(this, m_monitorPollingInterval);
