diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmyth/mediamonitor-unix.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmyth/mediamonitor-unix.cpp	2015-01-06 16:53:25.253064387 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmyth/mediamonitor-unix.cpp	2015-01-06 16:53:47.759609860 +0100
@@ -185,7 +185,7 @@
 bool MediaMonitorUnix::CheckMountable(void)
 {
 #if CONFIG_QTDBUS
-    for (int i = 0; i < 10; ++i, usleep(500000))
+    for (int i = 0; i < 3; ++i, usleep(500000))
     {
         // Connect to UDisks.  This can sometimes fail if mythfrontend
         // is started during system init
