diff -Naur mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-12-12 19:16:26.533333336 +0100
+++ mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-12-12 19:16:28.243333336 +0100
@@ -459,10 +459,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
