diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2018-10-12 10:52:51.386666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2018-10-12 10:52:54.383333333 +0200
@@ -447,10 +447,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetNumSetting("MonitorDrives", 0)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
