diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2021-03-04 20:27:12.256666268 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2021-03-04 20:27:17.903332934 +0100
@@ -459,10 +459,6 @@
     // Sanity check
     if (m_active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor disabled by user setting.");
-        return;
-    }
 
     if (!m_thread)
         m_thread = new MonitorThread(this, m_monitorPollingInterval);
