diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-12-27 12:19:29.270000000 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-12-27 12:19:31.006666667 +0100
@@ -459,10 +459,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
