diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-01-14 09:20:07.136666666 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-01-14 09:20:08.826666666 +0100
@@ -459,10 +459,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
