diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-15 09:23:29.920000001 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-15 09:23:35.760000001 +0200
@@ -465,10 +465,6 @@
     // Sanity check
     if (m_active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor disabled by user setting.");
-        return;
-    }
 
     if (!m_thread)
         m_thread = new MonitorThread(this, m_monitorPollingInterval);
