diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-04 11:36:09.173333333 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-04 11:36:10.530000000 +0200
@@ -465,10 +465,6 @@
     // Sanity check
     if (m_active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor disabled by user setting.");
-        return;
-    }
 
     if (!m_thread)
         m_thread = new MonitorThread(this, m_monitorPollingInterval);
