diff -Naur mythtv-master-20190320-gb121bce-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20190320-gb121bce-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20190320-gb121bce-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-20 22:45:52.936666667 +0100
+++ mythtv-master-20190320-gb121bce-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-20 22:45:54.730000000 +0100
@@ -463,10 +463,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
