diff -Naur mythtv-master-20210424-g69d12b67b6-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20210424-g69d12b67b6-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20210424-g69d12b67b6-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2021-04-24 16:57:47.516665707 +0200
+++ mythtv-master-20210424-g69d12b67b6-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2021-04-24 16:57:52.316665707 +0200
@@ -458,10 +458,6 @@
     // Sanity check
     if (m_active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor disabled by user setting.");
-        return;
-    }
 
     if (!m_thread)
         m_thread = new MonitorThread(this, m_monitorPollingInterval);
