diff -Naur mythtv-master-20200324-gaa46cdd-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200324-gaa46cdd-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200324-gaa46cdd-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-03-24 09:42:42.973333334 +0100
+++ mythtv-master-20200324-gaa46cdd-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-03-24 09:42:44.590000000 +0100
@@ -459,10 +459,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
