diff -Naur mythtv-master-20200530-ga180830-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200530-ga180830-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200530-ga180830-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-05-30 13:01:53.623333334 +0200
+++ mythtv-master-20200530-ga180830-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-05-30 13:01:54.893333334 +0200
@@ -459,10 +459,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor disabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
