diff -Naur mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-21 09:18:29.770000002 +0200
+++ mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-21 09:18:30.976666668 +0200
@@ -465,10 +465,6 @@
     // Sanity check
     if (m_active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor disabled by user setting.");
-        return;
-    }
 
     if (!m_thread)
         m_thread = new MonitorThread(this, m_monitorPollingInterval);
