diff -Naur mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200924-gb00d7c3-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-25 09:22:07.003333334 +0200
+++ mythtv-master-20200924-gb00d7c3-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-09-25 09:22:10.260000001 +0200
@@ -465,10 +465,6 @@
     // Sanity check
     if (m_active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor disabled by user setting.");
-        return;
-    }
 
     if (!m_thread)
         m_thread = new MonitorThread(this, m_monitorPollingInterval);
