diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-29 08:40:41.370000001 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-29 08:40:42.833333334 +0100
@@ -463,10 +463,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
