diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythui/mediamonitor.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythui/mediamonitor.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythui/mediamonitor.cpp	2021-05-18 16:12:44.253333271 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythui/mediamonitor.cpp	2021-05-18 16:12:47.373333270 +0200
@@ -458,10 +458,6 @@
     // Sanity check
     if (m_active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor disabled by user setting.");
-        return;
-    }
 
     if (!m_thread)
         m_thread = new MonitorThread(this, m_monitorPollingInterval);
