diff -Naur mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-04-09 12:26:13.787587534 +0200
+++ mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-04-09 12:26:15.290920813 +0200
@@ -459,10 +459,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
