diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-16 10:53:25.426666667 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-16 10:53:27.100000000 +0100
@@ -463,10 +463,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
