diff -Naur mythtv-master-20181019-g402f35a-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20181019-g402f35a-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20181019-g402f35a-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2018-10-20 14:57:53.540000001 +0200
+++ mythtv-master-20181019-g402f35a-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2018-10-20 14:57:57.276666668 +0200
@@ -447,10 +447,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetNumSetting("MonitorDrives", 0)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
