diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-05-07 12:47:09.236666666 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-05-07 12:47:10.606666667 +0200
@@ -459,10 +459,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
