diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-01-05 12:45:21.690000000 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-01-05 12:45:23.456666667 +0100
@@ -447,10 +447,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
