diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythui/mythuibuttonlist.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythui/mythuibuttonlist.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythui/mythuibuttonlist.cpp	2015-01-06 16:53:52.052920005 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythui/mythuibuttonlist.cpp	2015-01-06 16:54:17.446116676 +0100
@@ -2489,6 +2489,12 @@
         {
             ShowSearchDialog();
         }
+        else if (action == "0" || action == "1" || action == "2" || action == "3" ||
+                 action == "4" || action == "5" || action == "6" || action == "7" ||
+                 action == "8" || action == "9")
+        {
+            SetItemCurrent((int) ((GetCount() * (action.toInt() / 10.0))));
+        }
         else
             handled = false;
     }
diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/videodlg.cpp mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/videodlg.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/videodlg.cpp	2015-01-06 16:53:58.089554236 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/videodlg.cpp	2015-01-06 16:54:17.452783307 +0100
@@ -2028,8 +2028,12 @@
  */
 bool VideoDialog::keyPressEvent(QKeyEvent *levent)
 {
-    if (GetFocusWidget()->keyPressEvent(levent))
+    if (levent->key() != Qt::Key_1 && levent->key() != Qt::Key_2 &&
+        levent->key() != Qt::Key_3 && levent->key() != Qt::Key_4)
+    {
+        if (GetFocusWidget()->keyPressEvent(levent))
         return true;
+    }
 
     bool handled = false;
     QStringList actions;
