diff -Naur mythtv-master-20200720-g1c94272-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200720-g1c94272-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200720-g1c94272-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-07-23 18:00:19.399999999 +0200
+++ mythtv-master-20200720-g1c94272-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-07-23 17:56:58.049999982 +0200
@@ -348,6 +348,26 @@
 {
     // User can specify that some devices are not monitored
     QString ignore = gCoreContext->GetSetting("IgnoreDevices", "");
+    // Prepend ignored devices list with some standard devices present in
+    // TVBoxes
+    ignore.prepend("\
+/dev/mmcblk0,\
+/dev/mmcblk1,\
+/dev/mmcblk2,\
+/dev/mmcblk0p2,\
+/dev/mmcblk1p2,\
+/dev/mmcblk2p2,\
+/dev/mmcblk1boot0,\
+/dev/mmcblk1boot1,\
+/dev/zram0,\
+/dev/zram1,\
+/dev/zram2,\
+/dev/zram3,\
+/dev/zram4,\
+/dev/zram5,\
+/dev/zram6,\
+/dev/zram7,\
+");
 
     if (ignore.length())
     {
@@ -360,8 +380,8 @@
     else
         m_IgnoreList = QStringList();  // Force empty list
 
-    LOG(VB_MEDIA, LOG_NOTICE, "Creating MediaMonitor");
-    LOG(VB_MEDIA, LOG_INFO, "IgnoreDevices=" + ignore);
+    LOG(VB_GENERAL, LOG_NOTICE, "Creating MediaMonitor");
+    LOG(VB_GENERAL, LOG_INFO, "IgnoreDevices=" + ignore);
 
     // If any of IgnoreDevices are symlinks, also add the real device
     QStringList::Iterator dev;
