diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2021-05-18 16:12:49.586666603 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2021-05-18 16:12:51.773333270 +0200
@@ -347,6 +347,46 @@
 {
     // User can specify that some devices are not monitored
     QString ignore = gCoreContext->GetSetting("IgnoreDevices", "");
+    // Prepend ignored devices list with some standard devices present in
+    // TVBoxes
+    ignore.prepend("\
+/dev/mmcblk0,\
+/dev/mmcblk1,\
+/dev/mmcblk2,\
+/dev/mmcblk3,\
+/dev/mmcblk0p1,\
+/dev/mmcblk0p2,\
+/dev/mmcblk0p3,\
+/dev/mmcblk0p4,\
+/dev/mmcblk1p1,\
+/dev/mmcblk1p2,\
+/dev/mmcblk1p3,\
+/dev/mmcblk1p4,\
+/dev/mmcblk2p1,\
+/dev/mmcblk2p2,\
+/dev/mmcblk2p3,\
+/dev/mmcblk2p4,\
+/dev/mmcblk3p1,\
+/dev/mmcblk3p2,\
+/dev/mmcblk3p3,\
+/dev/mmcblk3p4,\
+/dev/mmcblk0boot0,\
+/dev/mmcblk0boot1,\
+/dev/mmcblk1boot0,\
+/dev/mmcblk1boot1,\
+/dev/mmcblk2boot0,\
+/dev/mmcblk2boot1,\
+/dev/mmcblk3boot0,\
+/dev/mmcblk3boot1,\
+/dev/zram0,\
+/dev/zram1,\
+/dev/zram2,\
+/dev/zram3,\
+/dev/zram4,\
+/dev/zram5,\
+/dev/zram6,\
+/dev/zram7,\
+");
 
     if (!ignore.isEmpty())
     {
@@ -357,8 +397,8 @@
 #endif
     }
 
-    LOG(VB_MEDIA, LOG_NOTICE, "Creating MediaMonitor");
-    LOG(VB_MEDIA, LOG_INFO, "IgnoreDevices=" + ignore);
+    LOG(VB_GENERAL, LOG_NOTICE, "Creating MediaMonitor");
+    LOG(VB_GENERAL, LOG_INFO, "IgnoreDevices=" + ignore);
 
     // If any of IgnoreDevices are symlinks, also add the real device
     QStringList symlinked;
