diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/mythplayer.cpp	2018-10-12 10:52:57.493333319 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/mythplayer.cpp	2018-10-12 10:53:00.660000000 +0200
@@ -1039,8 +1039,20 @@
     if (!gCoreContext->IsDatabaseIgnored() &&
         player_ctx->playingInfo->QueryAutoExpire() == kLiveTVAutoExpire)
     {
-        gCoreContext->SaveSetting(
-            "DefaultChanid", player_ctx->playingInfo->GetChanID());
+        int livetv_always_same_chan = gCoreContext->GetNumSetting("NoRememberLastTVChan", 0);
+        if (livetv_always_same_chan == 1)
+        {
+            LOG(VB_GENERAL, LOG_INFO,
+                "Saving DefaultChanid disabled by NoRememberLastTVChan=1");
+        }
+        else
+        {
+            gCoreContext->SaveSetting(
+                "DefaultChanid", player_ctx->playingInfo->GetChanID());
+            LOG(VB_GENERAL, LOG_INFO, LOC +
+                QString("DefaultChanid saved as '%1'").arg(player_ctx->playingInfo->GetChanID()));
+        }
+
         if (player_ctx->recorder && player_ctx->recorder->IsValidRecorder())
         {
             int cardid = player_ctx->recorder->GetRecorderNumber();
