diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/configure mythtv-master-20210518-gc8c226c9b4-new/mythtv/configure
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/configure	2021-05-18 16:12:50.753333270 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/configure	2021-05-18 16:12:53.983333270 +0200
@@ -5219,7 +5219,7 @@
 }
 
 # Minimum supported Qt version
-QT5_MIN_VERSION_STR="5.9"
+QT5_MIN_VERSION_STR="4.8"
 QT6_MIN_VERSION_STR="6.0"
 
 # qmake-qt5 /usr/lib64/qt5/bin/qmake /usr/lib/x86_64-linux-gnu/qt5/bin/qmake
