diff -Naur mythtv-master-20190320-gb121bce-old/mythtv/configure mythtv-master-20190320-gb121bce-new/mythtv/configure
--- mythtv-master-20190320-gb121bce-old/mythtv/configure	2019-03-20 22:45:55.516666667 +0100
+++ mythtv-master-20190320-gb121bce-new/mythtv/configure	2019-03-20 22:45:57.190000000 +0100
@@ -5186,7 +5186,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.3"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
