diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/configure mythtv-master-20190328-gdcb0847-new/mythtv/configure
--- mythtv-master-20190328-gdcb0847-old/mythtv/configure	2019-03-29 08:40:43.890000001 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/configure	2019-03-29 08:40:45.623333334 +0100
@@ -5186,7 +5186,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.3"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
