diff -Naur mythtv-master-20181019-g402f35a-old/mythtv/configure mythtv-master-20181019-g402f35a-new/mythtv/configure
--- mythtv-master-20181019-g402f35a-old/mythtv/configure	2018-10-20 14:58:02.539999987 +0200
+++ mythtv-master-20181019-g402f35a-new/mythtv/configure	2018-10-20 14:58:05.686666668 +0200
@@ -5186,7 +5186,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.3"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
