diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/configure mythtv-master-20200915-g3f33f57-new/mythtv/configure
--- mythtv-master-20200915-g3f33f57-old/mythtv/configure	2020-09-15 09:23:41.413333335 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/configure	2020-09-15 09:23:42.656666668 +0200
@@ -5199,7 +5199,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.9"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
