diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/configure mythtv-master-20190316-g1a44f54-new/mythtv/configure
--- mythtv-master-20190316-g1a44f54-old/mythtv/configure	2019-03-16 10:53:28.413333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/configure	2019-03-16 10:53:30.193333334 +0100
@@ -5183,7 +5183,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.3"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
