diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/configure mythtv-master-20210225-gc0426e3fea-new/mythtv/configure
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/configure	2021-03-04 20:27:25.503332933 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/configure	2021-03-04 20:27:32.419999599 +0100
@@ -5219,7 +5219,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.9"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
