diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/configure mythtv-master-20190105-g3382e1a-new/mythtv/configure
--- mythtv-master-20190105-g3382e1a-old/mythtv/configure	2019-01-05 12:45:24.133333333 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/configure	2019-01-05 12:45:25.663333333 +0100
@@ -5179,7 +5179,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.3"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
