diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/configure mythtv-master-20200904-gca3ee6a-new/mythtv/configure
--- mythtv-master-20200904-gca3ee6a-old/mythtv/configure	2020-09-04 11:36:12.680000000 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/configure	2020-09-04 11:36:14.186666667 +0200
@@ -5192,7 +5192,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.9"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
