diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/configure mythtv-master-20200114-ga315cd6-new/mythtv/configure
--- mythtv-master-20200114-ga315cd6-old/mythtv/configure	2020-01-14 09:20:10.106666665 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/configure	2020-01-14 09:20:11.859999999 +0100
@@ -5170,7 +5170,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.7"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
