diff -Naur mythtv-master-20210424-g69d12b67b6-old/mythtv/configure mythtv-master-20210424-g69d12b67b6-new/mythtv/configure
--- mythtv-master-20210424-g69d12b67b6-old/mythtv/configure	2021-04-24 16:58:00.249999043 +0200
+++ mythtv-master-20210424-g69d12b67b6-new/mythtv/configure	2021-04-24 16:58:05.909999042 +0200
@@ -5223,7 +5223,7 @@
 # Minimum supported Qt version
 case $BRANCH in
     *[Qq][Tt]6*) QT_MIN_VERSION_STR="6.0" ;;
-    *)           QT_MIN_VERSION_STR="5.9" ;;
+    *)           QT_MIN_VERSION_STR="4.8" ;;
 esac
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
