diff -Naur mythtv-master-20200530-ga180830-old/mythtv/configure mythtv-master-20200530-ga180830-new/mythtv/configure
--- mythtv-master-20200530-ga180830-old/mythtv/configure	2020-05-30 13:01:55.800000001 +0200
+++ mythtv-master-20200530-ga180830-new/mythtv/configure	2020-05-30 13:01:57.500000001 +0200
@@ -5191,7 +5191,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.7"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
