diff -Naur mythtv-master-20210426-g4f0e519416-old/mythtv/configure mythtv-master-20210426-g4f0e519416-new/mythtv/configure
--- mythtv-master-20210426-g4f0e519416-old/mythtv/configure	2021-04-26 19:45:17.106665281 +0200
+++ mythtv-master-20210426-g4f0e519416-new/mythtv/configure	2021-04-26 19:46:43.053331947 +0200
@@ -5219,7 +5219,7 @@
 }
 
 # Minimum supported Qt version
-QT5_MIN_VERSION_STR="5.9"
+QT5_MIN_VERSION_STR="4.8"
 QT6_MIN_VERSION_STR="6.0"
 
 # qmake-qt5 /usr/lib64/qt5/bin/qmake /usr/lib/x86_64-linux-gnu/qt5/bin/qmake
