diff -Naur mythtv-master-20200207-g2869463-old/mythtv/configure mythtv-master-20200207-g2869463-new/mythtv/configure
--- mythtv-master-20200207-g2869463-old/mythtv/configure	2020-02-08 14:19:33.420000002 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/configure	2020-02-08 14:19:35.043333336 +0100
@@ -5190,7 +5190,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.7"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
