diff -Naur mythtv-master-20200409-g0fc7abb082-old/mythtv/configure mythtv-master-20200409-g0fc7abb082-new/mythtv/configure
--- mythtv-master-20200409-g0fc7abb082-old/mythtv/configure	2020-04-09 12:26:16.294254109 +0200
+++ mythtv-master-20200409-g0fc7abb082-new/mythtv/configure	2020-04-09 12:26:18.207587373 +0200
@@ -5195,7 +5195,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.7"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
