diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/configure mythtv-master-20200507-ge586e5a-new/mythtv/configure
--- mythtv-master-20200507-ge586e5a-old/mythtv/configure	2020-05-07 12:47:11.670000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/configure	2020-05-07 12:47:13.043333333 +0200
@@ -5183,7 +5183,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.7"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
