diff -Naur mythtv-master-20200921-g4c36f47-old/mythtv/configure mythtv-master-20200921-g4c36f47-new/mythtv/configure
--- mythtv-master-20200921-g4c36f47-old/mythtv/configure	2020-09-21 09:18:33.373333335 +0200
+++ mythtv-master-20200921-g4c36f47-new/mythtv/configure	2020-09-21 09:18:34.630000001 +0200
@@ -5199,7 +5199,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.9"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
