diff -Naur mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/tv_play.cpp	2015-04-30 10:58:40.726876342 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/tv_play.cpp	2015-04-30 10:59:32.406667702 +0200
@@ -1268,6 +1268,16 @@
                 // big in fullscreen mode
                 if (fullscreen)
                 {
+                    // Hack for ticket12311
+                    if ((maxHeight >= 1072) && (maxHeight <= 1200))
+                    { 
+                        LOG(VB_GENERAL, LOG_DEBUG, LOC + "Hack for ticket12311: Re-Adjusted Screen size. Original: "
+                        + QString("%1x").arg(maxWidth)
+                        + QString("%1. New: ").arg(maxHeight)
+                        + QString("%1x1080").arg(maxWidth));
+                        maxHeight = 1080;
+                    }
+
                     player_bounds.setSize(QSize(maxWidth, maxHeight));
 
                     // resize possibly avoids a bug on some systems
diff -Naur mythtv-master-20150429-g6523186-old/mythtv/libs/libmythui/mythuihelper.cpp mythtv-master-20150429-g6523186-new/mythtv/libs/libmythui/mythuihelper.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/libs/libmythui/mythuihelper.cpp	2015-04-30 10:58:40.770209500 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/libs/libmythui/mythuihelper.cpp	2015-04-30 10:59:32.413334342 +0200
@@ -893,8 +893,8 @@
 void MythUIHelper::GetScreenSettings(int &xbase, int &width, float &wmult,
                                      int &ybase, int &height, float &hmult)
 {
-    xbase  = d->m_screenxbase;
-    ybase  = d->m_screenybase;
+    xbase  = d->m_screenxbase - 1;
+    ybase  = d->m_screenybase - 1;
 
     height = d->m_screenheight;
     width = d->m_screenwidth;
