diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/external/libudfread/udfread.c mythtv-master-20210225-gc0426e3fea-new/mythtv/external/libudfread/udfread.c
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/external/libudfread/udfread.c	2021-03-17 21:07:53.373333166 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/external/libudfread/udfread.c	2021-03-17 21:07:56.699999833 +0100
@@ -87,7 +87,7 @@
     return result;
 }
 
-#elif defined(HAVE_PTHREAD_H)
+#elif defined(HAVE_PTHREADS)
 
 #include <pthread.h>
 
