diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/external/libudfread/udfread.c mythtv-master-20201001-g2a40eb3-new/mythtv/external/libudfread/udfread.c
--- mythtv-master-20201001-g2a40eb3-old/mythtv/external/libudfread/udfread.c	2020-10-02 13:02:45.000000000 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/external/libudfread/udfread.c	2020-10-02 13:02:46.360000000 +0200
@@ -87,7 +87,7 @@
     return result;
 }
 
-#elif defined(HAVE_PTHREAD_H)
+#elif defined(HAVE_PTHREADS)
 
 #include <pthread.h>
 
