diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/external/libudfread/udfread.c mythtv-master-20200507-ge586e5a-new/mythtv/external/libudfread/udfread.c
--- mythtv-master-20200507-ge586e5a-old/mythtv/external/libudfread/udfread.c	2020-05-07 12:47:12.890000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/external/libudfread/udfread.c	2020-05-07 12:47:14.183333333 +0200
@@ -87,7 +87,7 @@
     return result;
 }
 
-#elif defined(HAVE_PTHREAD_H)
+#elif defined(HAVE_PTHREADS)
 
 #include <pthread.h>
 
