diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/external/libudfread/udfread.c mythtv-master-20200904-gca3ee6a-new/mythtv/external/libudfread/udfread.c
--- mythtv-master-20200904-gca3ee6a-old/mythtv/external/libudfread/udfread.c	2020-09-04 11:36:14.013333333 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/external/libudfread/udfread.c	2020-09-04 11:36:15.353333333 +0200
@@ -87,7 +87,7 @@
     return result;
 }
 
-#elif defined(HAVE_PTHREAD_H)
+#elif defined(HAVE_PTHREADS)
 
 #include <pthread.h>
 
