diff -Naur mythtv-master-20210424-g69d12b67b6-old/mythtv/external/libudfread/udfread.c mythtv-master-20210424-g69d12b67b6-new/mythtv/external/libudfread/udfread.c
--- mythtv-master-20210424-g69d12b67b6-old/mythtv/external/libudfread/udfread.c	2021-04-24 16:58:03.863332376 +0200
+++ mythtv-master-20210424-g69d12b67b6-new/mythtv/external/libudfread/udfread.c	2021-04-24 16:58:11.556665709 +0200
@@ -87,7 +87,7 @@
     return result;
 }
 
-#elif defined(HAVE_PTHREAD_H)
+#elif defined(HAVE_PTHREADS)
 
 #include <pthread.h>
 
