diff -Naur mythtv-master-20191231-gbafc80d-old/mythplugins/configure mythtv-master-20191231-gbafc80d-new/mythplugins/configure
--- mythtv-master-20191231-gbafc80d-old/mythplugins/configure	2019-12-31 08:35:51.000000000 +0100
+++ mythtv-master-20191231-gbafc80d-new/mythplugins/configure	2019-12-31 11:56:17.079999989 +0100
@@ -314,6 +314,7 @@
 cc="gcc"
 qmake="qmake"
 python="python3"
+perl="perl"
 
 targetos=`uname -s`
 
@@ -331,6 +332,7 @@
   --qmake=QMAKE            use specified qmake [$qmake]
   --qmakespecs=QMAKE       use specified qmakespecs [$qmakespecs]
   --python=PATH            Force a specific python executable to use [$python]
+  --perl=PATH              Force a specific perl executable to use [$perl]
   --libdir-name=LIBNAME    install/look for libs in PREFIX/LIBNAME [$libdir_name]
   --help                   print this message
   --enable-all             Enable all options
@@ -395,6 +397,7 @@
     qmake
     qmakespecs
     python
+    perl
     sysroot
     mythroot
 "
@@ -663,7 +666,7 @@
 if enabled weather; then
     check_pl_lib(){
         log check_pl_lib "$@"
-        check_cmd perl -e "use ${1};"
+        check_cmd $perl -e "use ${1};"
     }
 
     disable_weather(){
@@ -768,6 +771,8 @@
 echo "        Qt minimum     $QT_MIN_VERSION_STR"
 echo "        Qt installed   `$qmake -query QT_VERSION`"
 echo "        qmake          $(which $qmake)"
+echo "        python         $python"
+echo "        perl           $perl"
 echo " "
 
 echo "#" > ./config.pro
