diff -Naur mythtv-master-20190328-gdcb0847-old/mythplugins/configure mythtv-master-20190328-gdcb0847-new/mythplugins/configure
--- mythtv-master-20190328-gdcb0847-old/mythplugins/configure	2019-03-29 08:40:45.546666667 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythplugins/configure	2019-03-29 08:40:46.973333334 +0100
@@ -660,13 +660,13 @@
         disable netvision
     }
 
-    check_py_lib MythTV  || disable_netvision "MythTV Python bindings (MythTV)"
-    check_py_lib pycurl  || disable_netvision "Python pycurl library (pycurl)"
+#    check_py_lib MythTV  || disable_netvision "MythTV Python bindings (MythTV)"
+#    check_py_lib pycurl  || disable_netvision "Python pycurl library (pycurl)"
     check_py_lib lxml    || disable_netvision "Python lxml library (lxml)"
     check_py_lib xml     || disable_netvision "Python XML library (xml)"
     check_py_lib urllib  || disable_netvision "Python URL library (urllib)"
     check_py_lib urllib2 || disable_netvision "Python URL library 2 (urllib2)"
-    check_py_lib oauth   || disable_netvision "Python OAuth library (oauth)"
+#    check_py_lib oauth   || disable_netvision "Python OAuth library (oauth)"
 
     disabled netvision && echo "Disabling MythNetvision due to missing dependencies."
 fi
