diff -Naur mythtv-master-20200921-g4c36f47-old/mythtv/external/libudfread/udfread.c mythtv-master-20200921-g4c36f47-new/mythtv/external/libudfread/udfread.c
--- mythtv-master-20200921-g4c36f47-old/mythtv/external/libudfread/udfread.c	2020-09-21 09:18:34.436666668 +0200
+++ mythtv-master-20200921-g4c36f47-new/mythtv/external/libudfread/udfread.c	2020-09-21 09:18:35.846666668 +0200
@@ -87,7 +87,7 @@
     return result;
 }
 
-#elif defined(HAVE_PTHREAD_H)
+#elif defined(HAVE_PTHREADS)
 
 #include <pthread.h>
 
