diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/external/libudfread/udfread.c mythtv-master-20210225-gc0426e3fea-new/mythtv/external/libudfread/udfread.c
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/external/libudfread/udfread.c	2021-03-04 20:27:31.343332932 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/external/libudfread/udfread.c	2021-03-04 20:27:37.196666264 +0100
@@ -87,7 +87,7 @@
     return result;
 }
 
-#elif defined(HAVE_PTHREAD_H)
+#elif defined(HAVE_PTHREADS)
 
 #include <pthread.h>
 
