diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/external/libudfread/udfread.c mythtv-master-20200915-g3f33f57-new/mythtv/external/libudfread/udfread.c
--- mythtv-master-20200915-g3f33f57-old/mythtv/external/libudfread/udfread.c	2020-09-15 09:23:42.583333335 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/external/libudfread/udfread.c	2020-09-15 09:23:43.823333335 +0200
@@ -87,7 +87,7 @@
     return result;
 }
 
-#elif defined(HAVE_PTHREAD_H)
+#elif defined(HAVE_PTHREADS)
 
 #include <pthread.h>
 
