diff -Naur mythtv-master-20200324-gaa46cdd-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20200324-gaa46cdd-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20200324-gaa46cdd-old/mythtv/programs/mythfrontend/main.cpp	2020-03-24 09:42:46.780000000 +0100
+++ mythtv-master-20200324-gaa46cdd-new/mythtv/programs/mythfrontend/main.cpp	2020-03-24 09:42:48.590000000 +0100
@@ -836,8 +836,22 @@
     switch (gCoreContext->GetNumSetting("DVDOnInsertDVD", 1))
     {
         case 0 : // Do nothing
-        case 1 : // Display menu (mythdvd)*/
-            break;
+        case 1 : // Display menu (mythdvd)
+        {
+            MythThemedMenuState *menu = dynamic_cast<MythThemedMenuState *>
+                (GetMythMainWindow()->GetMainStack()->GetTopScreen());
+            if (menu)
+            {
+                menu->m_buttonList->MoveToNamedPosition("Videos");
+                QKeyEvent key(QEvent::KeyPress, Qt::Key_Return, Qt::NoModifier);
+                menu->m_buttonList->keyPressEvent(&key);
+                menu = dynamic_cast<MythThemedMenuState *>
+                    (GetMythMainWindow()->GetMainStack()->GetTopScreen());
+                if (menu)
+                    menu->m_buttonList->MoveToNamedPosition("Play a DVD or Blu-ray");
+            }
+             break;
+        }
         case 2 : // play DVD or Blu-ray
             playDisc();
             break;
