diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythfrontend/mythfrontend.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythfrontend/mythfrontend.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythfrontend/mythfrontend.cpp	2021-05-18 16:12:55.269999937 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythfrontend/mythfrontend.cpp	2021-05-18 16:12:58.469999936 +0200
@@ -827,8 +827,9 @@
     switch (gCoreContext->GetNumSetting("DVDOnInsertDVD", 1))
     {
         case 0 : // Do nothing
-        case 1 : // Display menu (mythdvd)*/
-            break;
+        case 1 : // Display menu (mythdvd)
+            playDisc();
+            break;
         case 2 : // play DVD or Blu-ray
             playDisc();
             break;
diff --speed-large-files --no-dereference --minimal -Naur mythtv-d3aab1567edabbf2933a16e260428c90538b8fde/mythtv/programs/mythfrontend/mythfrontend.cpp mythtv-d3aab1567edabbf2933a16e260428c90538b8fde/mythtv/programs/mythfrontend/mythfrontend.cpp
--- mythtv-d3aab1567edabbf2933a16e260428c90538b8fde/mythtv/programs/mythfrontend/mythfrontend.cpp	2023-07-14 18:47:49.176440321 +0200
+++ mythtv-d3aab1567edabbf2933a16e260428c90538b8fde/mythtv/programs/mythfrontend/mythfrontend.cpp	2023-07-14 18:46:45.916441682 +0200
@@ -890,7 +890,7 @@
         }
     }
 
-    if (gCoreContext->GetBoolSetting("GalleryAutoLoad", false))
+    if (gCoreContext->GetBoolSetting("GalleryAutoLoad", true))
     {
         LOG(VB_GUI, LOG_INFO, "Main: Autostarting Gallery for new media");
         GetMythMainWindow()->JumpTo(JUMP_GALLERY_DEFAULT);
