diff -Naur mythtv-master-20190320-gb121bce-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20190320-gb121bce-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20190320-gb121bce-old/mythtv/programs/mythfrontend/main.cpp	2019-03-20 22:45:58.450000000 +0100
+++ mythtv-master-20190320-gb121bce-new/mythtv/programs/mythfrontend/main.cpp	2019-03-20 22:46:00.030000000 +0100
@@ -831,8 +831,22 @@
     {
         case 0 : // Do nothing
             break;
-        case 1 : // Display menu (mythdvd)*/
-            break;
+        case 1 : // Display menu (mythdvd)
+        {
+            MythThemedMenuState *menu = dynamic_cast<MythThemedMenuState *>
+                (GetMythMainWindow()->GetMainStack()->GetTopScreen());
+            if (menu)
+            {
+                menu->m_buttonList->MoveToNamedPosition("Videos");
+                QKeyEvent key(QEvent::KeyPress, Qt::Key_Return, Qt::NoModifier);
+                menu->m_buttonList->keyPressEvent(&key);
+                menu = dynamic_cast<MythThemedMenuState *>
+                    (GetMythMainWindow()->GetMainStack()->GetTopScreen());
+                if (menu)
+                    menu->m_buttonList->MoveToNamedPosition("Play a DVD or Blu-ray");
+            }
+             break;
+        }
         case 2 : // play DVD or Blu-ray
             playDisc();
             break;
