diff -Naur mythtv-master-20151223-g7c82c55-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20151223-g7c82c55-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20151223-g7c82c55-old/mythtv/programs/mythfrontend/main.cpp	2015-12-23 09:55:45.000000000 +0100
+++ mythtv-master-20151223-g7c82c55-new/mythtv/programs/mythfrontend/main.cpp	2015-12-23 10:02:16.438436352 +0100
@@ -797,8 +797,22 @@
     {
         case 0 : // Do nothing
             break;
-        case 1 : // Display menu (mythdvd)*/
-            break;
+        case 1 : // Display menu (mythdvd)
+        {
+            MythThemedMenuState *menu = dynamic_cast<MythThemedMenuState *>
+                (GetMythMainWindow()->GetMainStack()->GetTopScreen());
+            if (menu)
+            {
+                menu->m_buttonList->MoveToNamedPosition("Videos");
+                QKeyEvent key(QEvent::KeyPress, Qt::Key_Return, Qt::NoModifier);
+                menu->m_buttonList->keyPressEvent(&key);
+                menu = dynamic_cast<MythThemedMenuState *>
+                    (GetMythMainWindow()->GetMainStack()->GetTopScreen());
+                if (menu)
+                    menu->m_buttonList->MoveToNamedPosition("Play a DVD or Blu-ray");
+            }
+             break;
+        }
         case 2 : // play DVD or Blu-ray
             playDisc();
             break;
