diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythtv/mythvideooutgpu.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythtv/mythvideooutgpu.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythtv/mythvideooutgpu.cpp	2021-03-04 20:27:44.969999596 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythtv/mythvideooutgpu.cpp	2021-03-04 20:27:51.406666262 +0100
@@ -363,7 +363,7 @@
     QVector<MythVideoFrame*> release;
 
     m_videoBuffers.BeginLock(kVideoBuffer_pause);
-    while (m_videoBuffers.Size(kVideoBuffer_pause))
+    while (m_videoBuffers.Size(kVideoBuffer_pause) > 1)
     {
         auto * frame = m_videoBuffers.Dequeue(kVideoBuffer_pause);
         if (!retain || (frame != Frame))
