diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythui/mythrender_opengl.cpp mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythui/mythrender_opengl.cpp
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythui/mythrender_opengl.cpp	2019-03-29 08:40:48.856666667 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythui/mythrender_opengl.cpp	2019-03-29 08:40:50.463333334 +0100
@@ -128,6 +128,7 @@
     LOG(VB_GENERAL, LOG_INFO, "OpenGL ES2 forced for Android");
 #elif defined USE_OPENGL_QT5 && defined USING_OPENGLES
     int openGLVersionFlags = QGLFormat::OpenGL_ES_Version_2_0;
+    LOG(VB_GENERAL, LOG_INFO, "Opening OpenGL ES2");
 #else
     // Check OpenGL version supported
     int openGLVersionFlags = QGLFormat::openGLVersionFlags();
@@ -277,7 +278,7 @@
             m_window = w->windowHandle();
     }
 
-#ifdef ANDROID
+#ifdef USING_OPENGLES
     // change all window surfacetypes to OpenGLSurface
     // otherwise the raster gets painted on top of the GL surface
     m_window->setSurfaceType(QWindow::OpenGLSurface);
@@ -301,7 +302,7 @@
 {
     setDevice(w);
 
-#ifdef ANDROID
+#ifdef USING_OPENGLES
     // change all window surfacetypes to OpenGLSurface
     // otherwise the raster gets painted on top of the GL surface
     m_window->setSurfaceType(QWindow::OpenGLSurface);
diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythui/mythrender_opengl.h mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythui/mythrender_opengl.h
--- mythtv-master-20190328-gdcb0847-old/mythtv/libs/libmythui/mythrender_opengl.h	2019-03-29 08:40:48.853333334 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/libs/libmythui/mythrender_opengl.h	2019-03-29 08:40:50.463333334 +0100
@@ -10,7 +10,7 @@
 // will exclude it for Raspberry Pi. With this commented, all
 // code that depends on USE_OPENGL_QT5 will be bypassed and maybe can
 // be removed later.
-#if defined USING_OPENGLES && QT_VERSION >= QT_VERSION_CHECK(5, 4, 0) && defined(ANDROID)
+#if defined USING_OPENGLES && QT_VERSION >= QT_VERSION_CHECK(5, 4, 0)
 #define USE_OPENGL_QT5
 #include <QOpenGLContext>
 #else
