diff -Naur mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp
--- mythtv-master-20191212-g4fddad7-old/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2019-12-12 19:16:32.653333336 +0100
+++ mythtv-master-20191212-g4fddad7-new/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2019-12-12 19:16:34.263333335 +0100
@@ -228,7 +228,10 @@
             bool splanarm2m = caps & V4L2_CAP_VIDEO_M2M;
 
             if (!(mplanar || mplanarm2m || splanar || splanarm2m))
+            {
+                LOG(VB_PLAYBACK, LOG_INFO, LOC + QString("Device: %1 No mplanar/mplanarm2m/splanar/splanarm2m found. Skiping this device!").arg(v4l2dev.GetDeviceName()));
                 continue;
+            }
 
             v4l2_buf_type capturetype = V4L2_BUF_TYPE_VIDEO_CAPTURE;
             v4l2_buf_type outputtype = V4L2_BUF_TYPE_VIDEO_OUTPUT;
@@ -290,6 +293,12 @@
                             foundfmt = true;
                             break;
                         }
+                        else
+                        {
+                            LOG(VB_GENERAL, LOG_INFO, LOC + QString("V4L2 disabled as %1 has no YUV420/YUV420M/NV12/NV12M support for %2.")
+                                .arg(v4l2dev.GetDeviceName())
+                                .arg(avcodec_get_name(codec)));
+                        }
                         fdesc.index++;
                     }
 
