diff --git a/mythtv/libs/libmythtv/mythvideooutgpu.cpp b/mythtv/libs/libmythtv/mythvideooutgpu.cpp
index 66f5124f61..93d81b70c6 100644
--- a/mythtv/libs/libmythtv/mythvideooutgpu.cpp
+++ b/mythtv/libs/libmythtv/mythvideooutgpu.cpp
@@ -355,13 +355,13 @@ void MythVideoOutputGPU::DoneDisplayingFrame(MythVideoFrame* Frame)
     if (!Frame)
         return;
 
-    auto retain = MythVideoFrame::HardwareFormat(Frame->m_type);
+    auto retain = MythVideoFrame::HardwareFormat(Frame->m_type);
     QVector<MythVideoFrame*> release;
 
     m_videoBuffers.BeginLock(kVideoBuffer_pause);
-    while (m_videoBuffers.Size(kVideoBuffer_pause))
+    while (m_videoBuffers.Size(kVideoBuffer_pause) > 1)
     {
-        auto * frame = m_videoBuffers.Dequeue(kVideoBuffer_pause);
+        auto * frame = m_videoBuffers.Dequeue(kVideoBuffer_pause);
         if (!retain || (frame != Frame))
             release.append(frame);
     }
