diff -Naur mythtv-master-20210424-g69d12b67b6-old/mythtv/libs/libmythtv/mythvideooutgpu.cpp mythtv-master-20210424-g69d12b67b6-new/mythtv/libs/libmythtv/mythvideooutgpu.cpp
--- mythtv-master-20210424-g69d12b67b6-old/mythtv/libs/libmythtv/mythvideooutgpu.cpp	2021-04-24 16:58:22.246665710 +0200
+++ mythtv-master-20210424-g69d12b67b6-new/mythtv/libs/libmythtv/mythvideooutgpu.cpp	2021-04-24 16:58:27.199999043 +0200
@@ -365,7 +365,7 @@
     QVector<MythVideoFrame*> release;
 
     m_videoBuffers.BeginLock(kVideoBuffer_pause);
-    while (m_videoBuffers.Size(kVideoBuffer_pause))
+    while (m_videoBuffers.Size(kVideoBuffer_pause) > 1)
     {
         auto * frame = m_videoBuffers.Dequeue(kVideoBuffer_pause);
         if (!retain || (frame != Frame))
