diff -Naur mythtv-master-20200109-gc116014-old/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp mythtv-master-20200109-gc116014-new/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp
--- mythtv-master-20200109-gc116014-old/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2020-01-09 18:01:19.216666667 +0100
+++ mythtv-master-20200109-gc116014-new/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2020-01-09 18:01:20.573333334 +0100
@@ -250,7 +250,10 @@
             bool splanarm2m = caps & V4L2_CAP_VIDEO_M2M;
 
             if (!(mplanar || mplanarm2m || splanar || splanarm2m))
+            {
+                LOG(VB_PLAYBACK, LOG_INFO, LOC + QString("Device: %1 No mplanar/mplanarm2m/splanar/splanarm2m found. Skiping this device!").arg(v4l2dev.GetDeviceName()));
                 continue;
+            }
 
             v4l2_buf_type capturetype = V4L2_BUF_TYPE_VIDEO_CAPTURE;
             v4l2_buf_type outputtype = V4L2_BUF_TYPE_VIDEO_OUTPUT;
@@ -312,6 +315,12 @@
                             foundfmt = true;
                             break;
                         }
+                        else
+                        {
+                            LOG(VB_GENERAL, LOG_INFO, LOC + QString("V4L2 disabled as %1 has no YUV420/YUV420M/NV12/NV12M support for %2.")
+                                .arg(v4l2dev.GetDeviceName())
+                                .arg(avcodec_get_name(codec)));
+                        }
                         fdesc.index++;
                     }
 
