diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythui/mythnotificationcenter.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythui/mythnotificationcenter.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythui/mythnotificationcenter.cpp	2015-01-06 16:57:07.391887582 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythui/mythnotificationcenter.cpp	2015-01-06 16:57:23.491804297 +0100
@@ -214,74 +214,6 @@
     connect(m_timer, SIGNAL(timeout()), this, SLOT(ProcessTimer()));
 }
 
-
-MythNotificationScreen &MythNotificationScreen::operator=(const MythNotificationScreen &s)
-{
-    if (this == &s)
-        return *this;
-
-    // check if anything have changed
-    m_refresh = !(
-                  m_id == s.m_id &&
-                  m_image == s.m_image &&
-                  m_imagePath == s.m_imagePath &&
-                  m_title == s.m_title &&
-                  m_origin == s.m_origin &&
-                  m_description == s.m_description &&
-                  m_extra == s.m_extra &&
-                  m_duration == s.m_duration &&
-                  m_progress == s.m_progress &&
-                  m_progresstext == s.m_progresstext &&
-                  m_content == s.m_content &&
-                  m_fullscreen == s.m_fullscreen &&
-                  m_expiry == s.m_expiry &&
-                  m_index == s.m_index &&
-                  m_style == s.m_style &&
-                  m_visibility == s.m_visibility &&
-                  m_priority == s.m_priority &&
-                  m_type == s.m_type
-                  );
-
-    m_id            = s.m_id;
-    m_image         = s.m_image;
-    m_imagePath     = s.m_imagePath;
-    m_title         = s.m_title;
-    m_origin        = s.m_origin;
-    m_description   = s.m_description;
-    m_extra         = s.m_extra;
-    m_duration      = s.m_duration;
-    m_progress      = s.m_progress;
-    m_progresstext  = s.m_progresstext;
-    m_content       = s.m_content;
-    m_fullscreen    = s.m_fullscreen;
-    m_expiry        = s.m_expiry;
-    m_index         = s.m_index;
-    m_style         = s.m_style;
-    m_visibility    = s.m_visibility;
-    m_priority      = s.m_priority;
-    m_type          = s.m_type;
-
-    m_update        = m_content; // so all fields are initialised regardless of notification type
-
-    m_added         = false;
-    m_created       = false;
-
-    m_artworkImage     = NULL;
-    m_titleText        = NULL;
-    m_originText       = NULL;
-    m_descriptionText  = NULL;
-    m_extraText        = NULL;
-    m_progresstextText = NULL;
-    m_progressBar      = NULL;
-    m_errorState       = NULL;
-    m_mediaState       = NULL;
-
-    m_timer            = new QTimer(this);
-    connect(m_timer, SIGNAL(timeout()), this, SLOT(ProcessTimer()));
-
-    return *this;
-}
-
 MythNotificationScreen::~MythNotificationScreen()
 {
     m_timer->stop();
@@ -745,6 +677,67 @@
     GetScreenStack()->PopScreen(this, true, true);
 }
 
+MythNotificationScreen &MythNotificationScreen::operator=(const MythNotificationScreen &s)
+{
+    // check if anything have changed
+    m_refresh = !(
+                  m_id == s.m_id &&
+                  m_image == s.m_image &&
+                  m_imagePath == s.m_imagePath &&
+                  m_title == s.m_title &&
+                  m_origin == s.m_origin &&
+                  m_description == s.m_description &&
+                  m_extra == s.m_extra &&
+                  m_duration == s.m_duration &&
+                  m_progress == s.m_progress &&
+                  m_progresstext == s.m_progresstext &&
+                  m_content == s.m_content &&
+                  m_fullscreen == s.m_fullscreen &&
+                  m_expiry == s.m_expiry &&
+                  m_index == s.m_index &&
+                  m_style == s.m_style &&
+                  m_visibility == s.m_visibility &&
+                  m_priority == s.m_priority &&
+                  m_type == s.m_type
+                  );
+
+    m_id            = s.m_id;
+    m_image         = s.m_image;
+    m_imagePath     = s.m_imagePath;
+    m_title         = s.m_title;
+    m_origin        = s.m_origin;
+    m_description   = s.m_description;
+    m_extra         = s.m_extra;
+    m_duration      = s.m_duration;
+    m_progress      = s.m_progress;
+    m_progresstext  = s.m_progresstext;
+    m_content       = s.m_content;
+    m_fullscreen    = s.m_fullscreen;
+    m_expiry        = s.m_expiry;
+    m_index         = s.m_index;
+    m_style         = s.m_style;
+    m_visibility    = s.m_visibility;
+    m_priority      = s.m_priority;
+    m_type          = s.m_type;
+
+    m_update        = m_content; // so all fields are initialised regardless of notification type
+
+    m_artworkImage     = NULL;
+    m_titleText        = NULL;
+    m_originText       = NULL;
+    m_descriptionText  = NULL;
+    m_extraText        = NULL;
+    m_progresstextText = NULL;
+    m_progressBar      = NULL;
+    m_errorState       = NULL;
+    m_mediaState       = NULL;
+
+    m_timer            = new QTimer(this);
+    connect(m_timer, SIGNAL(timeout()), this, SLOT(ProcessTimer()));
+
+    return *this;
+}
+
 void MythNotificationScreen::SetSingleShotTimer(int s, bool update)
 {
     // only registered application can display non-expiring notification
diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythui/mythnotificationcenter_private.h mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythui/mythnotificationcenter_private.h
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythui/mythnotificationcenter_private.h	2015-01-06 16:56:56.931941687 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythui/mythnotificationcenter_private.h	2015-01-06 16:57:23.515137508 +0100
@@ -154,8 +154,6 @@
                              MythNotification &notification);
     MythNotificationScreen(MythScreenStack *stack,
                              const MythNotificationScreen &screen);
-    
-    MythNotificationScreen &operator=(const MythNotificationScreen &s);
 
     virtual ~MythNotificationScreen();
 
@@ -192,6 +190,8 @@
         kAll        = ~kNone,
     };
 
+    MythNotificationScreen &operator=(const MythNotificationScreen &s);
+
 signals:
     void ScreenDeleted();
 
