diff -Naur mythtv-master-20181124-gb89d3b6-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20181124-gb89d3b6-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20181124-gb89d3b6-old/mythtv/libs/libmythtv/mythplayer.cpp	2018-11-27 21:47:36.756666667 +0100
+++ mythtv-master-20181124-gb89d3b6-new/mythtv/libs/libmythtv/mythplayer.cpp	2018-11-27 21:19:58.736666665 +0100
@@ -2216,8 +2216,8 @@
         if (rtcbase == 0)
         {
             // cater for DVB radio
-            if (videotimecode == 0)
-                videotimecode = audiotimecode;
+//            if (videotimecode == 0)
+//                videotimecode = audiotimecode;
             // On first frame we get nothing, so exit out.
             if (videotimecode == 0)
                 return;
