diff -Naur mythtv-master-20150429-g6523186-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20150429-g6523186-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/libs/libmythui/mythmainwindow.cpp	2015-04-30 10:59:36.896649590 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/libs/libmythui/mythmainwindow.cpp	2015-04-30 11:00:05.083202502 +0200
@@ -2656,12 +2656,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return (int)(qRound(x * d->wmult));
+    return (int)(x * d->wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return (int)(qRound(y * d->hmult));
+    return (int)(y * d->hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
