diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/libs/libmythui/mythmainwindow.cpp	2015-01-06 16:57:10.738536936 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/libs/libmythui/mythmainwindow.cpp	2015-01-06 16:57:37.291732902 +0100
@@ -2650,12 +2650,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return (int)(qRound(x * d->wmult));
+    return (int)(x * d->wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return (int)(qRound(y * d->hmult));
+    return (int)(y * d->hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
