diff -Naur mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20150105-g4216e00-old/mythtv/programs/mythfrontend/main.cpp	2015-01-06 16:57:32.151759494 +0100
+++ mythtv-master-20150105-g4216e00-new/mythtv/programs/mythfrontend/main.cpp	2015-01-06 16:57:52.711653121 +0100
@@ -1627,6 +1627,7 @@
     // accelerated Flash playback to work without causing mythfrontend to abort.
     QApplication::setAttribute(Qt::AA_X11InitThreads);
 #endif
+    QApplication::setAttribute(Qt::AA_X11InitThreads);
     new QApplication(argc, argv);
     QCoreApplication::setApplicationName(MYTH_APPNAME_MYTHFRONTEND);
 
