diff -Naur mythtv-master-20160128-ge5b211f-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20160128-ge5b211f-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20160128-ge5b211f-old/mythtv/libs/libmythui/mythmainwindow.cpp	2016-01-29 11:14:48.000000000 +0100
+++ mythtv-master-20160128-ge5b211f-new/mythtv/libs/libmythui/mythmainwindow.cpp	2016-01-29 11:20:52.590293030 +0100
@@ -201,7 +201,6 @@
         enteringStandby(false),
         disableIdle(false),
         NC(NULL),
-        firstinit(true),
         m_bSavedPOS(false)
     {
     }
@@ -299,8 +298,6 @@
     bool enteringStandby;
     bool disableIdle;
     MythNotificationCenter *NC;
-        // window aspect
-    bool firstinit;
     bool m_bSavedPOS;
 };
 
@@ -1032,14 +1029,7 @@
     // Set window border based on fullscreen attribute
     Qt::WindowFlags flags = Qt::Window;
 
-    bool inwindow = GetMythDB()->GetNumSetting("RunFrontendInWindow", 0);
-    bool fullscreen = d->does_fill_screen && !GetMythUI()->IsGeometryOverridden();
-
-    // On Compiz/Unit, when the window is fullscreen and frameless changing
-    // screen position ends up stuck. Adding a border temporarily prevents this
-    setWindowFlags(windowFlags() & ~Qt::FramelessWindowHint);
-
-    if (!inwindow)
+    if (!GetMythDB()->GetNumSetting("RunFrontendInWindow", 0))
     {
         LOG(VB_GENERAL, LOG_INFO, "Using Frameless Window");
         flags |= Qt::FramelessWindowHint;
@@ -1050,20 +1040,10 @@
     flags |= Qt::MSWindowsOwnDC;
 #endif
 
-    if (fullscreen && !inwindow)
+    if (d->does_fill_screen && !GetMythUI()->IsGeometryOverridden())
     {
         LOG(VB_GENERAL, LOG_INFO, "Using Full Screen Window");
-        if (d->firstinit)
-        {
-            // During initialization, we force being fullscreen using setWindowState
-            // otherwise, in ubuntu's unity, the side bar often stays visible
-            setWindowState(Qt::WindowFullScreen);
-        }
-    }
-    else
-    {
-            // reset type
-        setWindowState(Qt::WindowNoState);
+        setWindowState(Qt::WindowFullScreen);
     }
 
     if (gCoreContext->GetNumSetting("AlwaysOnTop", false))
@@ -1072,7 +1052,6 @@
     }
 
     setWindowFlags(flags);
-    QTimer::singleShot(1000, this, SLOT(DelayedAction()));
 
     d->screenRect = QRect(d->xbase, d->ybase, d->screenwidth, d->screenheight);
     d->uiScreenRect = QRect(0, 0, d->screenwidth, d->screenheight);
@@ -1082,9 +1061,7 @@
                                         .arg(QString::number(d->screenheight)));
 
     setGeometry(d->xbase, d->ybase, d->screenwidth, d->screenheight);
-    // remove size constraints
-    setFixedSize(QWIDGETSIZE_MAX, QWIDGETSIZE_MAX);
-    resize(d->screenwidth, d->screenheight);
+    setFixedSize(QSize(d->screenwidth, d->screenheight));
 
     GetMythUI()->ThemeWidget(this);
 #ifdef Q_OS_MAC
@@ -1230,12 +1207,6 @@
     }
 }
 
-void MythMainWindow::DelayedAction(void)
-{
-    setFixedSize(QSize(d->screenwidth, d->screenheight));
-    Show();
-}
-
 void MythMainWindow::InitKeys()
 {
     RegisterKey("Global", ACTION_UP, QT_TRANSLATE_NOOP("MythControls",
@@ -1419,19 +1390,7 @@
 
 void MythMainWindow::Show(void)
 {
-    bool inwindow = GetMythDB()->GetNumSetting("RunFrontendInWindow", 0);
-    bool fullscreen = d->does_fill_screen && !GetMythUI()->IsGeometryOverridden();
-
-    if (fullscreen && !inwindow && !d->firstinit)
-    {
-        showFullScreen();
-    }
-    else
-    {
-        show();
-    }
-    d->firstinit = false;
-
+    show();
 #ifdef Q_WS_MACX_OLDQT
     if (d->does_fill_screen)
         HideMenuBar();
diff -Naur mythtv-master-20160128-ge5b211f-old/mythtv/libs/libmythui/mythmainwindow.h mythtv-master-20160128-ge5b211f-new/mythtv/libs/libmythui/mythmainwindow.h
--- mythtv-master-20160128-ge5b211f-old/mythtv/libs/libmythui/mythmainwindow.h	2016-01-28 12:33:54.000000000 +0100
+++ mythtv-master-20160128-ge5b211f-new/mythtv/libs/libmythui/mythmainwindow.h	2016-01-29 11:17:18.388610334 +0100
@@ -177,9 +177,6 @@
     void ShowMouseCursor(bool show);
 
     MythMainWindowPrivate *d;
-
-  private slots:
-    void DelayedAction(void);
 };
 
 MUI_PUBLIC MythMainWindow *GetMythMainWindow();
