diff -Naur mythtv-master-20150429-g6523186-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20150429-g6523186-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/libs/libmythui/mythmainwindow.cpp	2015-04-30 11:00:05.083202502 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/libs/libmythui/mythmainwindow.cpp	2015-04-30 11:00:51.986346584 +0200
@@ -196,8 +196,7 @@
         idleTime(0),
         standby(false),
         enteringStandby(false),
-        NC(NULL),
-        firstinit(true)
+        NC(NULL)
     {
     }
 
@@ -293,8 +292,6 @@
     bool standby;
     bool enteringStandby;
     MythNotificationCenter *NC;
-        // window aspect
-    bool firstinit;
 };
 
 // Make keynum in QKeyEvent be equivalent to what's in QKeySequence
@@ -985,14 +982,7 @@
     // Set window border based on fullscreen attribute
     Qt::WindowFlags flags = Qt::Window;
 
-    bool inwindow = GetMythDB()->GetNumSetting("RunFrontendInWindow", 0);
-    bool fullscreen = d->does_fill_screen && !GetMythUI()->IsGeometryOverridden();
-
-    // On Compiz/Unit, when the window is fullscreen and frameless changing
-    // screen position ends up stuck. Adding a border temporarily prevents this
-    setWindowFlags(windowFlags() & ~Qt::FramelessWindowHint);
-
-    if (!inwindow)
+    if (!GetMythDB()->GetNumSetting("RunFrontendInWindow", 0))
     {
         LOG(VB_GENERAL, LOG_INFO, "Using Frameless Window");
         flags |= Qt::FramelessWindowHint;
@@ -1003,20 +993,10 @@
     flags |= Qt::MSWindowsOwnDC;
 #endif
 
-    if (fullscreen && !inwindow)
+    if (d->does_fill_screen && !GetMythUI()->IsGeometryOverridden())
     {
         LOG(VB_GENERAL, LOG_INFO, "Using Full Screen Window");
-        if (d->firstinit)
-        {
-            // During initialization, we force being fullscreen using setWindowState
-            // otherwise, in ubuntu's unity, the side bar often stays visible
-            setWindowState(Qt::WindowFullScreen);
-        }
-    }
-    else
-    {
-            // reset type
-        setWindowState(Qt::WindowNoState);
+        setWindowState(Qt::WindowFullScreen);
     }
 
     if (gCoreContext->GetNumSetting("AlwaysOnTop", false))
@@ -1025,7 +1005,6 @@
     }
 
     setWindowFlags(flags);
-    QTimer::singleShot(1000, this, SLOT(DelayedAction()));
 
     d->screenRect = QRect(d->xbase, d->ybase, d->screenwidth, d->screenheight);
     d->uiScreenRect = QRect(0, 0, d->screenwidth, d->screenheight);
@@ -1035,9 +1014,7 @@
                                         .arg(QString::number(d->screenheight)));
 
     setGeometry(d->xbase, d->ybase, d->screenwidth, d->screenheight);
-    // remove size constraints
-    setFixedSize(QWIDGETSIZE_MAX, QWIDGETSIZE_MAX);
-    resize(d->screenwidth, d->screenheight);
+    setFixedSize(QSize(d->screenwidth, d->screenheight));
 
     GetMythUI()->ThemeWidget(this);
 #ifdef Q_OS_MAC
@@ -1161,12 +1138,6 @@
     }
 }
 
-void MythMainWindow::DelayedAction(void)
-{
-    setFixedSize(QSize(d->screenwidth, d->screenheight));
-    Show();
-}
-
 void MythMainWindow::InitKeys()
 {
     RegisterKey("Global", ACTION_UP, QT_TRANSLATE_NOOP("MythControls",
@@ -1350,19 +1321,7 @@
 
 void MythMainWindow::Show(void)
 {
-    bool inwindow = GetMythDB()->GetNumSetting("RunFrontendInWindow", 0);
-    bool fullscreen = d->does_fill_screen && !GetMythUI()->IsGeometryOverridden();
-
-    if (fullscreen && !inwindow && !d->firstinit)
-    {
-        showFullScreen();
-    }
-    else
-    {
-        show();
-    }
-    d->firstinit = false;
-
+    show();
 #ifdef Q_WS_MACX_OLDQT
     if (d->does_fill_screen)
         HideMenuBar();
diff -Naur mythtv-master-20150429-g6523186-old/mythtv/libs/libmythui/mythmainwindow.h mythtv-master-20150429-g6523186-new/mythtv/libs/libmythui/mythmainwindow.h
--- mythtv-master-20150429-g6523186-old/mythtv/libs/libmythui/mythmainwindow.h	2015-04-30 10:59:51.726589728 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/libs/libmythui/mythmainwindow.h	2015-04-30 11:00:51.986346584 +0200
@@ -174,9 +174,6 @@
     void ShowMouseCursor(bool show);
 
     MythMainWindowPrivate *d;
-
-  private slots:
-    void DelayedAction(void);
 };
 
 MUI_PUBLIC MythMainWindow *GetMythMainWindow();
