diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythtv/tv_play.cpp	2020-09-04 11:36:16.380000000 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythtv/tv_play.cpp	2020-09-04 11:36:17.710000000 +0200
@@ -8203,7 +8203,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
