diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/tv_play.cpp	2019-12-27 12:19:36.786666667 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/tv_play.cpp	2019-12-27 12:19:38.226666667 +0100
@@ -8226,7 +8226,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
