diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/tv_play.cpp	2020-05-07 12:47:15.416666667 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/tv_play.cpp	2020-05-07 12:47:17.123333333 +0200
@@ -8214,7 +8214,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
