diff -Naur mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/tv_play.cpp	2019-04-08 14:45:34.203333333 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/tv_play.cpp	2019-04-08 14:45:35.863333333 +0200
@@ -8238,7 +8238,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
