diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/tv_play.cpp	2020-10-02 13:02:47.280000000 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/tv_play.cpp	2020-10-02 13:02:49.070000000 +0200
@@ -6827,7 +6827,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
