diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/tv_play.cpp	2019-03-16 10:53:37.653333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/tv_play.cpp	2019-03-16 10:53:39.250000001 +0100
@@ -8253,7 +8253,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
