diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythtv/tv_play.cpp	2021-03-17 21:08:03.319999832 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythtv/tv_play.cpp	2021-03-17 21:08:06.139999833 +0100
@@ -6437,7 +6437,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode, slock, pat, pmt, mgt, vct,
