diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/tv_play.cpp	2020-09-15 09:23:44.950000002 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/tv_play.cpp	2020-09-15 09:23:46.340000001 +0200
@@ -8206,7 +8206,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
