diff -Naur mythtv-master-20200109-gc116014-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20200109-gc116014-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20200109-gc116014-old/mythtv/libs/libmythtv/tv_play.cpp	2020-01-09 18:01:20.483333334 +0100
+++ mythtv-master-20200109-gc116014-new/mythtv/libs/libmythtv/tv_play.cpp	2020-01-09 18:01:21.980000001 +0100
@@ -8211,7 +8211,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
