diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/tv_play.cpp	2019-02-17 16:58:42.900000002 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/tv_play.cpp	2019-02-17 16:58:44.316666669 +0100
@@ -8393,7 +8393,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
