diff -Naur mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20181019-g402f35a-old/mythtv/libs/libmythtv/tv_play.cpp	2018-10-20 14:58:11.216666667 +0200
+++ mythtv-master-20181019-g402f35a-new/mythtv/libs/libmythtv/tv_play.cpp	2018-10-20 14:58:14.120000001 +0200
@@ -8410,7 +8410,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
