diff -Naur mythtv-master-20190320-gb121bce-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20190320-gb121bce-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20190320-gb121bce-old/mythtv/libs/libmythtv/tv_play.cpp	2019-03-20 22:46:04.063333334 +0100
+++ mythtv-master-20190320-gb121bce-new/mythtv/libs/libmythtv/tv_play.cpp	2019-03-20 22:46:05.446666667 +0100
@@ -8241,7 +8241,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
