diff -Naur mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20200507-ge586e5a-old/mythtv/libs/libmythtv/osd.cpp	2020-05-07 12:47:17.020000000 +0200
+++ mythtv-master-20200507-ge586e5a-new/mythtv/libs/libmythtv/osd.cpp	2020-05-07 12:47:18.306666667 +0200
@@ -528,7 +528,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
