diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/libs/libmythtv/osd.cpp	2021-03-04 20:27:54.383332927 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/libs/libmythtv/osd.cpp	2021-03-04 20:28:01.306666258 +0100
@@ -350,7 +350,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
