diff -Naur mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20200114-ga315cd6-old/mythtv/libs/libmythtv/osd.cpp	2020-01-14 09:20:17.929999999 +0100
+++ mythtv-master-20200114-ga315cd6-new/mythtv/libs/libmythtv/osd.cpp	2020-01-14 09:20:20.543333332 +0100
@@ -528,7 +528,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
