diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/osd.cpp	2020-09-15 09:23:45.990000001 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/osd.cpp	2020-09-15 09:23:47.546666668 +0200
@@ -518,7 +518,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
