diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/osd.cpp	2019-03-16 10:53:39.013333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/osd.cpp	2019-03-16 10:53:40.583333334 +0100
@@ -565,7 +565,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
