diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythtv/osd.cpp	2021-05-18 16:13:02.366666603 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythtv/osd.cpp	2021-05-18 16:13:05.303333270 +0200
@@ -350,7 +350,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
