diff -Naur mythtv-master-20200530-ga180830-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20200530-ga180830-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20200530-ga180830-old/mythtv/libs/libmythtv/osd.cpp	2020-05-30 13:02:01.066666667 +0200
+++ mythtv-master-20200530-ga180830-new/mythtv/libs/libmythtv/osd.cpp	2020-05-30 13:02:02.376666667 +0200
@@ -528,7 +528,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
