diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/osd.cpp	2018-10-12 10:53:12.003333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/osd.cpp	2018-10-12 10:53:14.766666666 +0200
@@ -565,7 +565,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
