diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/osd.cpp	2019-12-27 12:19:37.856666667 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/osd.cpp	2019-12-27 12:19:39.286666667 +0100
@@ -532,7 +532,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
