diff -Naur mythtv-master-20200418-g8760389-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20200418-g8760389-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20200418-g8760389-old/mythtv/libs/libmythtv/osd.cpp	2020-04-18 10:55:18.719999999 +0200
+++ mythtv-master-20200418-g8760389-new/mythtv/libs/libmythtv/osd.cpp	2020-04-18 10:55:20.156666666 +0200
@@ -528,7 +528,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
