diff -Naur mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20200904-gca3ee6a-old/mythtv/libs/libmythtv/osd.cpp	2020-09-04 11:36:17.553333334 +0200
+++ mythtv-master-20200904-gca3ee6a-new/mythtv/libs/libmythtv/osd.cpp	2020-09-04 11:36:18.996666667 +0200
@@ -518,7 +518,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
