diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/osd.cpp	2020-10-02 13:02:48.543333333 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/osd.cpp	2020-10-02 13:02:50.363333333 +0200
@@ -518,7 +518,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
