diff -Naur mythtv-master-20200109-gc116014-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20200109-gc116014-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20200109-gc116014-old/mythtv/libs/libmythtv/osd.cpp	2020-01-09 18:01:21.870000001 +0100
+++ mythtv-master-20200109-gc116014-new/mythtv/libs/libmythtv/osd.cpp	2020-01-09 18:01:23.170000001 +0100
@@ -528,7 +528,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
