diff -Naur mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20200409-g0fc7abb082-old/mythtv/libs/libmythtv/osd.cpp	2020-04-09 12:26:21.047587269 +0200
+++ mythtv-master-20200409-g0fc7abb082-new/mythtv/libs/libmythtv/osd.cpp	2020-04-09 12:26:22.534253881 +0200
@@ -528,7 +528,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
