diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythui/mythnotificationcenter.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythui/mythnotificationcenter.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythui/mythnotificationcenter.cpp	2021-05-18 16:13:04.523333270 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythui/mythnotificationcenter.cpp	2021-05-18 16:13:07.583333269 +0200
@@ -331,6 +331,9 @@
         // if not, default to the main one
         foundtheme = LoadWindowFromXML("notification-ui.xml", theme, this);
     }
+    if (!m_style.isEmpty())
+        LOG(VB_GENERAL, LOG_NOTICE, QString("Custom notification style request: '%1', will load: '%2'")
+            .arg(m_style).arg(theme_attempt));
 
     if (!foundtheme) // If we cannot load the theme for any reason ...
         return false;
@@ -577,6 +580,11 @@
     {
         m_extra = tmp;
     }
+    tmp = data["asst"];
+    if (!(tmp.isNull() && (m_update & kMetaData)))
+    {
+        m_style = tmp;
+    }
     // We need to re-run init
     m_refresh = true;
 }
@@ -1489,6 +1497,7 @@
                                                                  "MythTV") : origin;
     data["asal"] = detail;
     data["asfm"] = extra;
+    data["asst"] = style;
 
     if (type == MythNotification::Error   ||
         type == MythNotification::Warning ||
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythui/mythudplistener.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythui/mythudplistener.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythui/mythudplistener.cpp	2021-05-18 16:13:04.519999936 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythui/mythudplistener.cpp	2021-05-18 16:13:07.593333270 +0200
@@ -101,6 +101,8 @@
     bool error = false;
     int visibility = 0;
     QString type = "normal";
+    QString style = "";
+    MythNotification::Priority priority = MythNotification::kDefault;
 
     auto node = element.firstChild();
     while (!node.isNull())
@@ -131,6 +133,8 @@
                 visibility = dom.text().toInt();
             else if (tagname == "type")
                 type = dom.text();
+            else if (notification && tagname == "style")
+                style = dom.text();
             else if (notification && tagname == "progress")
             {
                 bool ok = false;
@@ -155,12 +159,12 @@
             timeout = notification ? 5s : 0s;
         if (notification)
         {
-            origin = origin.isEmpty() ? tr("UDP Listener") : origin;
+            //origin = origin.isEmpty() ? tr("UDP Listener") : origin;
             ShowNotification(error ? MythNotification::Error :
                              MythNotification::TypeFromString(type),
                              msg, origin, description, image, extra,
                              progress_text, progress, timeout,
-                             fullscreen, static_cast<VNMask>(visibility));
+                             fullscreen, static_cast<VNMask>(visibility), priority, style);
         }
         else
         {
