diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythui/mythnotificationcenter.cpp mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythui/mythnotificationcenter.cpp
--- mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythui/mythnotificationcenter.cpp	2020-09-15 09:23:47.223333335 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythui/mythnotificationcenter.cpp	2020-09-15 09:23:48.983333335 +0200
@@ -331,6 +331,9 @@
         // if not, default to the main one
         foundtheme = LoadWindowFromXML("notification-ui.xml", theme, this);
     }
+    if (!m_style.isEmpty())
+        LOG(VB_GENERAL, LOG_NOTICE, QString("Custom notification style request: '%1', will load: '%2'")
+            .arg(m_style).arg(theme_attempt));
 
     if (!foundtheme) // If we cannot load the theme for any reason ...
         return false;
@@ -577,6 +580,11 @@
     {
         m_extra = tmp;
     }
+    tmp = data["asst"];
+    if (!(tmp.isNull() && (m_update & kMetaData)))
+    {
+        m_style = tmp;
+    }
     // We need to re-run init
     m_refresh = true;
 }
@@ -1489,6 +1497,7 @@
                                                                  "MythTV") : origin;
     data["asal"] = detail;
     data["asfm"] = extra;
+    data["asst"] = style;
 
     if (type == MythNotification::Error   ||
         type == MythNotification::Warning ||
diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythui/mythudplistener.cpp mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythui/mythudplistener.cpp
--- mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythui/mythudplistener.cpp	2020-09-15 09:23:47.223333335 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythui/mythudplistener.cpp	2020-09-15 09:23:48.983333335 +0200
@@ -113,6 +113,8 @@
     bool error = false;
     int visibility = 0;
     QString type = "normal";
+    QString style = "";
+    MythNotification::Priority priority = MythNotification::kDefault;
 
     QDomNode n = docElem.firstChild();
     while (!n.isNull())
@@ -142,6 +144,8 @@
                 visibility = e.text().toUInt();
             else if (e.tagName() == "type")
                 type = e.text();
+            else if (notification && e.tagName() == "style")
+                style = e.text();
             else if (notification && e.tagName() == "progress")
             {
                 bool ok = false;
@@ -172,7 +176,7 @@
                              MythNotification::TypeFromString(type),
                              msg, origin, description, image, extra,
                              progress_text, progress, timeout, fullscreen,
-                             visibility);
+                             visibility, priority, style);
         }
         else
         {
