diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmythui/mythnotificationcenter.cpp mythtv-master-20190216-g938606f-new/mythtv/libs/libmythui/mythnotificationcenter.cpp
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmythui/mythnotificationcenter.cpp	2019-02-17 16:58:45.636666669 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmythui/mythnotificationcenter.cpp	2019-02-17 16:58:47.196666669 +0100
@@ -369,6 +369,9 @@
         // if not, default to the main one
         foundtheme = LoadWindowFromXML("notification-ui.xml", theme, this);
     }
+    if (!m_style.isEmpty())
+        LOG(VB_GENERAL, LOG_NOTICE, QString("Custom notification style request: '%1', will load: '%2'")
+            .arg(m_style).arg(theme_attempt));
 
     if (!foundtheme) // If we cannot load the theme for any reason ...
         return false;
@@ -615,6 +618,11 @@
     {
         m_extra = tmp;
     }
+    tmp = data["asst"];
+    if (!(tmp.isNull() && (m_update & kMetaData)))
+    {
+        m_style = tmp;
+    }
     // We need to re-run init
     m_refresh = true;
 }
@@ -1543,6 +1551,7 @@
                                                                  "MythTV") : origin;
     data["asal"] = detail;
     data["asfm"] = extra;
+    data["asst"] = style;
 
     if (type == MythNotification::Error   ||
         type == MythNotification::Warning ||
diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmythui/mythudplistener.cpp mythtv-master-20190216-g938606f-new/mythtv/libs/libmythui/mythudplistener.cpp
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmythui/mythudplistener.cpp	2019-02-17 16:58:45.636666669 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmythui/mythudplistener.cpp	2019-02-17 16:58:47.196666669 +0100
@@ -114,6 +114,8 @@
     bool error = false;
     int visibility = 0;
     QString type = "normal";
+    QString style = "";
+    MythNotification::Priority priority = MythNotification::kDefault;
 
     QDomNode n = docElem.firstChild();
     while (!n.isNull())
@@ -143,6 +145,8 @@
                 visibility = e.text().toUInt();
             else if (e.tagName() == "type")
                 type = e.text();
+            else if (notification && e.tagName() == "style")
+                style = e.text();
             else if (notification && e.tagName() == "progress")
             {
                 bool ok;
@@ -173,7 +177,7 @@
                              MythNotification::TypeFromString(type),
                              msg, origin, description, image, extra,
                              progress_text, progress, timeout, fullscreen,
-                             visibility);
+                             visibility, priority, style);
         }
         else
         {
