From 0c9f98bb8889878194a7340139a3fe837155217e Mon Sep 17 00:00:00 2001
From: Roger Siddons <dizygotheca@ntlworld.com>
Date: Wed, 29 Apr 2015 14:41:23 +0100
Subject: [PATCH] Cache FE gallery images in remotecache for Minimyth2


diff --git a/mythtv/libs/libmythui/mythuihelper.cpp b/mythtv/libs/libmythui/mythuihelper.cpp
index 19e73e7..4ec6e86 100644
--- a/mythtv/libs/libmythui/mythuihelper.cpp
+++ b/mythtv/libs/libmythui/mythuihelper.cpp
@@ -590,17 +590,16 @@ MythImage *MythUIHelper::CacheImage(const QString &url, MythImage *im,
 
     if (!nodisk)
     {
-        QString dstfile = GetMythUI()->GetThemeCacheDir() + '/' + url;
+        QString cachedir = GetMythUI()->GetThemeCacheDir(url);
+        QString dstfile = cachedir + '/' + url;
 
         LOG(VB_GUI | VB_FILE, LOG_INFO, LOC +
             QString("Saved to Cache (%1)").arg(dstfile));
 
         // This would probably be better off somewhere else before any
         // Load() calls at all.
-        QDir themedir(GetMythUI()->GetThemeCacheDir());
-
-        if (!themedir.exists())
-            themedir.mkdir(GetMythUI()->GetThemeCacheDir());
+        QDir themedir(cachedir);
+        themedir.mkdir(cachedir);
 
         // Save to disk cache
         im->save(dstfile, "PNG");
@@ -691,7 +690,7 @@ void MythUIHelper::RemoveFromCacheByURL(const QString &url)
 
     QString dstfile;
 
-    dstfile = GetThemeCacheDir() + '/' + url;
+    dstfile = GetThemeCacheDir(url) + '/' + url;
     LOG(VB_GUI | VB_FILE, LOG_INFO, LOC +
         QString("RemoveFromCacheByURL removed :%1: from cache").arg(dstfile));
     QFile::remove(dstfile);
@@ -716,7 +715,7 @@ void MythUIHelper::RemoveFromCacheByFile(const QString &fname)
 
     // Loop through files to cache any that were not caught by
     // RemoveFromCacheByURL
-    QDir dir(GetThemeCacheDir());
+    QDir dir(GetThemeCacheDir(partialKey));
     QFileInfoList list = dir.entryInfoList();
 
     for (int i = 0; i < list.size(); ++i)
@@ -750,8 +749,12 @@ bool MythUIHelper::IsImageInCache(const QString &url)
     return false;
 }
 
-QString MythUIHelper::GetThemeCacheDir(void)
+QString MythUIHelper::GetThemeCacheDir(const QString &mrl)
 {
+    // Gallery thumbnails ("-Photographs-") and images ("-Photographs@") are cached separately
+    if (mrl.contains("-Photographs"))
+        return GetConfDir() + "/cache/remotecache/";
+
     QString cachedirname = GetConfDir() + "/cache/themecache/";
 
     QString tmpcachedir = cachedirname +
@@ -1604,7 +1607,7 @@ MythImage *MythUIHelper::LoadCacheImage(QString srcfile, QString label,
     if (ret || !(cacheMode & kCacheIgnoreDisk))
     {
         // Create url to image in disk cache
-        QString cachefilepath = GetThemeCacheDir() + '/' + label;
+        QString cachefilepath = GetThemeCacheDir(label) + '/' + label;
         QFileInfo cacheFileInfo(cachefilepath);
 
         // If the file isn't in the disk cache, then we don't want to bother
diff --git a/mythtv/libs/libmythui/mythuihelper.h b/mythtv/libs/libmythui/mythuihelper.h
index 759dd7f..e64d007 100644
--- a/mythtv/libs/libmythui/mythuihelper.h
+++ b/mythtv/libs/libmythui/mythuihelper.h
@@ -57,7 +57,7 @@ class MUI_PUBLIC MythUIHelper
     void RemoveFromCacheByURL(const QString &url);
     void RemoveFromCacheByFile(const QString &fname);
     bool IsImageInCache(const QString &url);
-    QString GetThemeCacheDir(void);
+    QString GetThemeCacheDir(const QString &mrl = "");
 
     void IncludeInCacheSize(MythImage *im);
     void ExcludeFromCacheSize(MythImage *im);
-- 
1.9.1

