diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/mythplayer.cpp	2018-10-12 10:53:22.953333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/mythplayer.cpp	2018-10-12 10:53:26.610000000 +0200
@@ -5006,7 +5006,7 @@
     infoMap.insert("buffersize",
         QString::number(player_ctx->buffer->GetBufferSize() >> 20));
     infoMap.insert("avsync",
-            QString::number((float)avsync_avg / (float)frame_interval, 'f', 2));
+            QString::number((float)avsync_avg / (float)frame_interval, 'f', 3));
     if (videoOutput)
     {
         QString frames = QString("%1/%2").arg(videoOutput->ValidVideoFrames())
@@ -5019,8 +5019,8 @@
     {
         infoMap["framerate"] = QString("%1%2%3")
             .arg(output_jmeter->GetLastFPS(), 0, 'f', 2)
-            .arg(QChar(0xB1, 0))
-            .arg(output_jmeter->GetLastSD(), 0, 'f', 2);
+            .arg(QChar(0xB1, 0)) // +/- symbol
+            .arg(output_jmeter->GetLastSD(), 0, 'f', 3);
         infoMap["load"] = output_jmeter->GetLastCPUStats();
     }
     GetCodecDescription(infoMap);
diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/ringbuffer.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/ringbuffer.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/ringbuffer.cpp	2018-10-12 10:53:22.946666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/ringbuffer.cpp	2018-10-12 10:53:26.610000000 +0200
@@ -1677,8 +1677,11 @@
     if (type == kRingBuffer_DVD || type == kRingBuffer_BD)
         return "N/A";
 
-    int avail = (rbwpos >= rbrpos) ? rbwpos - rbrpos : bufferSize - rbrpos + rbwpos;
-    return QString("%1%").arg((int)(((float)avail / (float)bufferSize) * 100.0));
+    unsigned avail = (rbwpos >= rbrpos) ? rbwpos - rbrpos :
+                                          bufferSize - rbrpos + rbwpos;
+    double d = 100.0 * avail / bufferSize;
+    if (d >= 99.95) return QString("100.%");
+    return QString("%1%").arg(d, 4, 'f', 1);
 }
 
 uint64_t RingBuffer::UpdateDecoderRate(uint64_t latest)
