diff -Naur mythtv-test-20191117-gb8e27f1-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-test-20191117-gb8e27f1-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-test-20191117-gb8e27f1-old/mythtv/libs/libmythtv/mythplayer.cpp	2019-11-17 13:04:18.000000000 +0100
+++ mythtv-test-20191117-gb8e27f1-new/mythtv/libs/libmythtv/mythplayer.cpp	2019-11-17 15:39:45.463333334 +0100
@@ -4866,7 +4866,8 @@
     if (decoder)
         infoMap["videocodecdesc"] = decoder->GetRawEncodingType();
     infoMap["videowidth"]     = QString::number(width);
-    infoMap["videoheight"]    = QString::number(height);
+    bool interlaced = is_interlaced(m_scan);
+    infoMap["videoheight"]    = QString::number(height) + (interlaced ? "i" : "p");
     infoMap["videoframerate"] = QString::number(video_frame_rate, 'f', 2);
     infoMap["deinterlacer"]   = DeinterlacerName(m_lastDeinterlacer,
                                                  m_lastDeinterlacer2x, m_lastFrameCodec);
@@ -4874,7 +4875,6 @@
     if (width < 640)
         return;
 
-    bool interlaced = is_interlaced(m_scan);
     if (width == 1920 || height == 1080 || height == 1088)
         infoMap["videodescrip"] = interlaced ? "HD_1080_I" : "HD_1080_P";
     else if ((width == 1280 || height == 720) && !interlaced)
