diff -Naur mythtv-master-20200418-g8760389-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20200418-g8760389-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20200418-g8760389-old/mythtv/libs/libmythtv/mythplayer.cpp	2020-04-18 10:55:23.446666666 +0200
+++ mythtv-master-20200418-g8760389-new/mythtv/libs/libmythtv/mythplayer.cpp	2020-04-18 10:55:24.646666666 +0200
@@ -4661,7 +4661,8 @@
     if (m_decoder)
         infoMap["videocodecdesc"] = m_decoder->GetRawEncodingType();
     infoMap["videowidth"]     = QString::number(width);
-    infoMap["videoheight"]    = QString::number(height);
+    bool interlaced = is_interlaced(m_scan);
+    infoMap["videoheight"]    = QString::number(height) + (interlaced ? "i" : "p");
     infoMap["videoframerate"] = QString::number(m_videoFrameRate, 'f', 2);
     infoMap["deinterlacer"]   = DeinterlacerName(m_lastDeinterlacer,
                                                  m_lastDeinterlacer2x, m_lastFrameCodec);
@@ -4669,7 +4670,6 @@
     if (width < 640)
         return;
 
-    bool interlaced = is_interlaced(m_scan);
     if (width == 1920 || height == 1080 || height == 1088)
         infoMap["videodescrip"] = interlaced ? "HD_1080_I" : "HD_1080_P";
     else if ((width == 1280 || height == 720) && !interlaced)
