diff -Naur mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20200915-g3f33f57-old/mythtv/libs/libmythtv/mythplayer.cpp	2020-09-15 09:23:51.340000002 +0200
+++ mythtv-master-20200915-g3f33f57-new/mythtv/libs/libmythtv/mythplayer.cpp	2020-09-15 09:23:57.506666668 +0200
@@ -4636,7 +4636,8 @@
     if (m_decoder)
         infoMap["videocodecdesc"] = m_decoder->GetRawEncodingType();
     infoMap["videowidth"]     = QString::number(width);
-    infoMap["videoheight"]    = QString::number(height);
+    bool interlaced = is_interlaced(m_scan);
+    infoMap["videoheight"]    = QString::number(height) + (interlaced ? "i" : "p");
     infoMap["videoframerate"] = QString::number(m_videoFrameRate, 'f', 2);
     infoMap["deinterlacer"]   = DeinterlacerName(m_lastDeinterlacer,
                                                  m_lastDeinterlacer2x, m_lastFrameCodec);
@@ -4644,7 +4645,6 @@
     if (width < 640)
         return;
 
-    bool interlaced = is_interlaced(m_scan);
     if (height > 2100)
         infoMap["videodescrip"] = interlaced ? "UHD_4K_I" : "UHD_4K_P";
     else if (width == 1920 || height == 1080 || height == 1088)
