diff -Naur mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythtv/libs/libmythtv/mythplayer.cpp	2019-12-27 12:19:42.256666667 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythtv/libs/libmythtv/mythplayer.cpp	2019-12-27 12:19:44.026666667 +0100
@@ -4601,7 +4601,8 @@
     if (m_decoder)
         infoMap["videocodecdesc"] = m_decoder->GetRawEncodingType();
     infoMap["videowidth"]     = QString::number(width);
-    infoMap["videoheight"]    = QString::number(height);
+    bool interlaced = is_interlaced(m_scan);
+    infoMap["videoheight"]    = QString::number(height) + (interlaced ? "i" : "p");
     infoMap["videoframerate"] = QString::number(m_videoFrameRate, 'f', 2);
     infoMap["deinterlacer"]   = DeinterlacerName(m_lastDeinterlacer,
                                                  m_lastDeinterlacer2x, m_lastFrameCodec);
@@ -4609,7 +4610,6 @@
     if (width < 640)
         return;
 
-    bool interlaced = is_interlaced(m_scan);
     if (width == 1920 || height == 1080 || height == 1088)
         infoMap["videodescrip"] = interlaced ? "HD_1080_I" : "HD_1080_P";
     else if ((width == 1280 || height == 720) && !interlaced)
