diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythtv/mythplayerui.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythtv/mythplayerui.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythtv/mythplayerui.cpp	2021-05-18 16:13:11.066666603 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythtv/mythplayerui.cpp	2021-05-18 16:13:14.299999937 +0200
@@ -821,14 +821,14 @@
     if (m_decoder)
         Map["videocodecdesc"] = m_decoder->GetRawEncodingType();
     Map["videowidth"]     = QString::number(width);
-    Map["videoheight"]    = QString::number(height);
+    bool interlaced = is_interlaced(GetScanType());
+    Map["videoheight"]    = QString::number(height) + (interlaced ? "i" : "p");
     Map["videoframerate"] = QString::number(m_videoFrameRate, 'f', 2);
     Map["deinterlacer"]   = GetDeinterlacerName();
 
     if (width < 640)
         return;
 
-    bool interlaced = is_interlaced(GetScanType());
     if (height > 2100)
         Map["videodescrip"] = interlaced ? "UHD_4K_I" : "UHD_4K_P";
     else if (width == 1920 || height == 1080 || height == 1088)
