diff -Naur mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20200921-g4c36f47-old/mythtv/libs/libmythtv/mythplayer.cpp	2020-09-21 09:18:45.766666668 +0200
+++ mythtv-master-20200921-g4c36f47-new/mythtv/libs/libmythtv/mythplayer.cpp	2020-09-21 09:18:47.493333335 +0200
@@ -4626,7 +4626,8 @@
     if (m_decoder)
         infoMap["videocodecdesc"] = m_decoder->GetRawEncodingType();
     infoMap["videowidth"]     = QString::number(width);
-    infoMap["videoheight"]    = QString::number(height);
+    bool interlaced = is_interlaced(m_scan);
+    infoMap["videoheight"]    = QString::number(height) + (interlaced ? "i" : "p");
     infoMap["videoframerate"] = QString::number(m_videoFrameRate, 'f', 2);
     infoMap["deinterlacer"]   = DeinterlacerName(m_lastDeinterlacer,
                                                  m_lastDeinterlacer2x, m_lastFrameCodec);
@@ -4634,7 +4635,6 @@
     if (width < 640)
         return;
 
-    bool interlaced = is_interlaced(m_scan);
     if (height > 2100)
         infoMap["videodescrip"] = interlaced ? "UHD_4K_I" : "UHD_4K_P";
     else if (width == 1920 || height == 1080 || height == 1088)
