diff -Naur mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20181012-g4651002-old/mythtv/libs/libmythtv/mythplayer.cpp	2018-10-12 10:53:26.610000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythtv/libs/libmythtv/mythplayer.cpp	2018-10-12 10:53:29.616666666 +0200
@@ -4821,13 +4821,13 @@
     if (decoder)
         infoMap["videocodecdesc"] = decoder->GetRawEncodingType();
     infoMap["videowidth"]     = QString::number(width);
-    infoMap["videoheight"]    = QString::number(height);
+    bool interlaced = is_interlaced(m_scan);
+    infoMap["videoheight"]    = QString::number(height) + (interlaced ? "i" : "");
     infoMap["videoframerate"] = QString::number(video_frame_rate, 'f', 2);
 
     if (width < 640)
         return;
 
-    bool interlaced = is_interlaced(m_scan);
     if (width == 1920 || height == 1080 || height == 1088)
         infoMap["videodescrip"] = interlaced ? "HD_1080_I" : "HD_1080_P";
     else if ((width == 1280 || height == 720) && !interlaced)
