diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/mythplayer.cpp	2019-01-05 12:45:37.260000000 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/mythplayer.cpp	2019-01-05 12:45:38.720000000 +0100
@@ -5074,13 +5074,13 @@
     if (decoder)
         infoMap["videocodecdesc"] = decoder->GetRawEncodingType();
     infoMap["videowidth"]     = QString::number(width);
-    infoMap["videoheight"]    = QString::number(height);
+    bool interlaced = is_interlaced(m_scan);
+    infoMap["videoheight"]    = QString::number(height) + (interlaced ? "i" : "");
     infoMap["videoframerate"] = QString::number(video_frame_rate, 'f', 2);
 
     if (width < 640)
         return;
 
-    bool interlaced = is_interlaced(m_scan);
     if (width == 1920 || height == 1080 || height == 1088)
         infoMap["videodescrip"] = interlaced ? "HD_1080_I" : "HD_1080_P";
     else if ((width == 1280 || height == 720) && !interlaced)
