diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/mythplayer.cpp	2020-10-02 13:03:00.976666666 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/mythplayer.cpp	2020-10-02 13:03:07.280000000 +0200
@@ -4441,7 +4441,8 @@
     if (m_decoder)
         infoMap["videocodecdesc"] = m_decoder->GetRawEncodingType();
     infoMap["videowidth"]     = QString::number(width);
-    infoMap["videoheight"]    = QString::number(height);
+    bool interlaced = is_interlaced(m_scan);
+    infoMap["videoheight"]    = QString::number(height) + (interlaced ? "i" : "p");
     infoMap["videoframerate"] = QString::number(m_videoFrameRate, 'f', 2);
     infoMap["deinterlacer"]   = DeinterlacerName(m_lastDeinterlacer,
                                                  m_lastDeinterlacer2x, m_lastFrameCodec);
@@ -4449,7 +4450,6 @@
     if (width < 640)
         return;
 
-    bool interlaced = is_interlaced(m_scan);
     if (height > 2100)
         infoMap["videodescrip"] = interlaced ? "UHD_4K_I" : "UHD_4K_P";
     else if (width == 1920 || height == 1080 || height == 1088)
