From ed5338381c519cbf3b6304a3870fdb134f24830e Mon Sep 17 00:00:00 2001
From: Roger Siddons <dizygotheca@ntlworld.com>
Date: Thu, 14 May 2015 17:11:17 +0100
Subject: [PATCH 1/4] Add logging


diff --git a/mythtv/programs/mythfrontend/gallerythumbview.cpp b/mythtv/programs/mythfrontend/gallerythumbview.cpp
index 229a8b8..ef610bd 100644
--- a/mythtv/programs/mythfrontend/gallerythumbview.cpp
+++ b/mythtv/programs/mythfrontend/gallerythumbview.cpp
@@ -58,6 +58,8 @@ GalleryThumbView::~GalleryThumbView()
 {
     gCoreContext->removeListener(this);
 
+    LOG(VB_GUI, LOG_DEBUG, "Exiting Gallery");
+
     delete m_view;
     delete m_db;
     delete m_infoList;
@@ -1208,6 +1210,8 @@ bool GalleryThumbView::DirSelectUp()
     ImageItem *im = m_view->GetParent();
     if (im)
     {
+        LOG(VB_GUI, LOG_DEBUG, QString("Thumbview: Going up from %1").arg(im->m_fileName));
+
         // Select the upfolder in the higher dir
         m_view->Select(im->m_id);
 
@@ -1225,8 +1229,12 @@ void GalleryThumbView::DirSelectDown()
 {
     ImageItem *im = m_view->GetSelected();
     if (im)
+    {
+        LOG(VB_GUI, LOG_DEBUG, QString("Thumbview: Going down to %1").arg(im->m_fileName));
+
         // Create tree rooted at selected item
         LoadData(im->m_id);
+    }
 }
 
 
-- 
1.9.1

