diff -Naur mythtv-old/mythtv/programs/mythfrontend/networkcontrol.cpp mythtv-new/mythtv/programs/mythfrontend/networkcontrol.cpp
--- mythtv-old/mythtv/programs/mythfrontend/networkcontrol.cpp	2021-02-06 23:53:31.476666533 +0100
+++ mythtv-new/mythtv/programs/mythfrontend/networkcontrol.cpp	2021-02-07 13:32:16.266665716 +0100
@@ -333,7 +333,7 @@
 
 void NetworkControl::deleteClient(void)
 {
-    LOG(VB_GENERAL, LOG_INFO, LOC + "Client Socket disconnected");
+    LOG(VB_NETWORK, LOG_INFO, LOC + "Client Socket disconnected");
     QMutexLocker locker(&m_clientLock);
 
     gCoreContext->SendSystemEvent("NET_CTRL_DISCONNECTED");
@@ -366,7 +366,7 @@
 {
     QString welcomeStr;
 
-    LOG(VB_GENERAL, LOG_INFO, LOC + QString("New connection established."));
+    LOG(VB_NETWORK, LOG_INFO, LOC + QString("New connection established."));
 
     gCoreContext->SendSystemEvent("NET_CTRL_CONNECTED");
 
