diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythfrontend/networkcontrol.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythfrontend/networkcontrol.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythfrontend/networkcontrol.cpp	2021-05-18 16:13:16.539999937 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythfrontend/networkcontrol.cpp	2021-05-18 16:13:18.826666604 +0200
@@ -336,7 +336,7 @@
 
 void NetworkControl::deleteClient(void)
 {
-    LOG(VB_GENERAL, LOG_INFO, LOC + "Client Socket disconnected");
+    LOG(VB_NETWORK, LOG_INFO, LOC + "Client Socket disconnected");
     QMutexLocker locker(&m_clientLock);
 
     gCoreContext->SendSystemEvent("NET_CTRL_DISCONNECTED");
@@ -369,7 +369,7 @@
 {
     QString welcomeStr;
 
-    LOG(VB_GENERAL, LOG_INFO, LOC + QString("New connection established."));
+    LOG(VB_NETWORK, LOG_INFO, LOC + QString("New connection established."));
 
     gCoreContext->SendSystemEvent("NET_CTRL_CONNECTED");
 
