diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythtv/programs/mythfrontend/networkcontrol.cpp mythtv-master-20210225-gc0426e3fea-new/mythtv/programs/mythfrontend/networkcontrol.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythtv/programs/mythfrontend/networkcontrol.cpp	2021-03-17 21:08:20.513333165 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythtv/programs/mythfrontend/networkcontrol.cpp	2021-03-17 21:08:22.969999832 +0100
@@ -333,7 +333,7 @@
 
 void NetworkControl::deleteClient(void)
 {
-    LOG(VB_GENERAL, LOG_INFO, LOC + "Client Socket disconnected");
+    LOG(VB_NETWORK, LOG_INFO, LOC + "Client Socket disconnected");
     QMutexLocker locker(&m_clientLock);
 
     gCoreContext->SendSystemEvent("NET_CTRL_DISCONNECTED");
@@ -366,7 +366,7 @@
 {
     QString welcomeStr;
 
-    LOG(VB_GENERAL, LOG_INFO, LOC + QString("New connection established."));
+    LOG(VB_NETWORK, LOG_INFO, LOC + QString("New connection established."));
 
     gCoreContext->SendSystemEvent("NET_CTRL_CONNECTED");
 
