diff -Naur mythtv-master-20190408-g038e171-old/mythplugins/mythmusic/mythmusic/musicplayer.cpp mythtv-master-20190408-g038e171-new/mythplugins/mythmusic/mythmusic/musicplayer.cpp
--- mythtv-master-20190408-g038e171-old/mythplugins/mythmusic/mythmusic/musicplayer.cpp	2019-04-08 14:45:44.076666667 +0200
+++ mythtv-master-20190408-g038e171-new/mythplugins/mythmusic/mythmusic/musicplayer.cpp	2019-04-08 14:45:45.126666667 +0200
@@ -676,6 +676,31 @@
                              QString("MusicPlayer: got invalid MUSIC_COMMAND "
                                      "PLAY_TRACK - %1").arg(me->Message()));
                 }
+                else if (list[2] == "PLAY_LIST")
+                {
+                    if (list.size() < 4)
+                    {
+                        LOG(VB_GENERAL, LOG_ERR,
+                             QString("MusicPlayer: got invalid MUSIC_COMMAND "
+                                     "PLAY_LIST - %1").arg(me->Message()));
+                    }
+                    else if (gMusicData && gMusicData->m_all_playlists)
+                    {
+                        int start = me->Message().indexOf("'");
+                        int end = me->Message().lastIndexOf("'");
+                        if (start != -1 && end != -1 && start != end)
+                        {
+                            QString name = me->Message().mid(start + 1, end - start - 1);
+                            Playlist *plist = gMusicData->m_all_playlists->getPlaylist(name);
+                            if (plist)
+                            {
+                                gMusicData->m_all_playlists->copyToActive(plist->getID());
+                                qApp->processEvents();
+                                setCurrentTrackPos(0);
+                            }
+                        }
+                    }
+                }
                 else if (list[2] == "GET_METADATA")
                 {
                     QString mdataStr;
@@ -704,6 +729,20 @@
                     MythEvent me2(message);
                     gCoreContext->dispatch(me2);
                 }
+                else if (list[2] == "GET_PLAYLISTS")
+                {
+                    QString mdataStr;
+
+                    if (gMusicData && gMusicData->m_all_playlists)
+                        mdataStr = gMusicData->m_all_playlists->getPlaylistNames().join(", ");
+                    else
+                        mdataStr = "<Uninitialised>";
+
+                    QString message = QString("MUSIC_CONTROL ANSWER %1 %2")
+                            .arg(gCoreContext->GetHostName()).arg(mdataStr);
+                    MythEvent me(message);
+                    gCoreContext->dispatch(me);
+                }
             }
             else
                 LOG(VB_GENERAL, LOG_ERR,
@@ -1476,7 +1515,8 @@
     if (!getCurrentMetadata())
     {
         m_currentTrack = -1;
-        stop(true);
+        if (isPlaying())
+            stop(true);
     }
 }
 
diff -Naur mythtv-master-20190408-g038e171-old/mythtv/programs/mythfrontend/networkcontrol.cpp mythtv-master-20190408-g038e171-new/mythtv/programs/mythfrontend/networkcontrol.cpp
--- mythtv-master-20190408-g038e171-old/mythtv/programs/mythfrontend/networkcontrol.cpp	2019-04-08 14:45:44.106666667 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/programs/mythfrontend/networkcontrol.cpp	2019-04-08 14:45:45.126666667 +0200
@@ -772,6 +772,37 @@
                 message = QString("MUSIC_COMMAND %1 PLAY_FILE '%2'")
                                 .arg(hostname)
                                 .arg(nc->getFrom(3));
+            else if (is_abbrev("list", nc->getArg(2)))
+            {
+                gotAnswer = false;
+
+                MythEvent me(QString("MUSIC_COMMAND %1 GET_PLAYLISTS")
+                                .arg(gCoreContext->GetHostName()) );
+                gCoreContext->dispatch(me);
+
+                QTime timer;
+                timer.start();
+                while (timer.elapsed() < FE_SHORT_TO && !gotAnswer)
+                {
+                    qApp->processEvents();
+                    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+                }
+
+                if (gotAnswer && answer == "<Uninitialised>")
+                {
+                    GetMythMainWindow()->JumpTo(jumpMap["playlistview"]);
+
+                    QTime timer;
+                    timer.start();
+                    while ((timer.elapsed() < FE_LONG_TO) &&
+                           (GetMythUI()->GetCurrentLocation().toLower() != "playlistview"))
+                        std::this_thread::sleep_for(std::chrono::milliseconds(10));
+                }
+
+                message = QString("MUSIC_COMMAND %1 PLAY_LIST '%2'")
+                                .arg(hostname)
+                                .arg(nc->getFrom(3));
+            }
             else
                 return QString("ERROR: Invalid 'play music' command");
         }
@@ -1048,6 +1079,27 @@
         return QString("ERROR: See 'help %1' for usage information "
                        "(parameters mismatch)").arg(nc->getArg(0));
     }
+    else if (is_abbrev("playlists", nc->getArg(1)))
+    {
+        gotAnswer = false;
+
+        MythEvent me(QString("MUSIC_COMMAND %1 GET_PLAYLISTS")
+                        .arg(gCoreContext->GetHostName()) );
+        gCoreContext->dispatch(me);
+
+        QTime timer;
+        timer.start();
+        while (timer.elapsed() < FE_SHORT_TO && !gotAnswer)
+        {
+            qApp->processEvents();
+            std::this_thread::sleep_for(std::chrono::milliseconds(10));
+        }
+
+        if (gotAnswer)
+            return answer;
+
+        return "unknown";
+    }
     else if (is_abbrev("events", nc->getArg(1)))
     {
         result = nc->getClient()->WantsEvents() ? "On" : "Off";
@@ -1458,7 +1510,8 @@
             "play music getstatus   - Get music player status playing/paused/stopped (MythMusic)\r\n"
             "play music file NAME   - Play specified file (MythMusic)\r\n"
             "play music track N     - Switch to specified track (MythMusic)\r\n"
-            "play music url URL     - Play specified URL (MythMusic)\r\n";
+            "play music url URL     - Play specified URL (MythMusic)\r\n"
+            "play music list NAME   - Play specified playlist (MythMusic)\r\n";
     }
     else if (is_abbrev("query", command))
     {
@@ -1478,7 +1531,8 @@
             "query loglevel        - Get current VERBOSE loglevel\r\n"
             "query version         - Query Frontend version details\r\n"
             "query channels        - Query available channels\r\n"
-            "query channels START LIMIT - Query available channels from START and limit results to LIMIT lines\r\n";
+            "query channels START LIMIT - Query available channels from START and limit results to LIMIT lines\r\n"
+            "query playlists       - Query playlists (MythMusic)\r\n";
     }
     else if (is_abbrev("set", command))
     {

diff -Naur mythtv-test-20191117-gb8e27f1-old/mythtv/programs/mythfrontend/networkcontrol.h mythtv-test-20191117-gb8e27f1-new/mythtv/programs/mythfrontend/networkcontrol.h
--- mythtv-test-20191117-gb8e27f1-old/mythtv/programs/mythfrontend/networkcontrol.h	2019-11-17 13:04:21.000000000 +0100
+++ mythtv-test-20191117-gb8e27f1-new/mythtv/programs/mythfrontend/networkcontrol.h	2019-11-17 16:42:09.506666667 +0100
@@ -112,7 +112,7 @@
     QString processLiveTV(NetworkCommand *nc);
     QString processPlay(NetworkCommand *nc, int clientID);
     QString processQuery(NetworkCommand *nc);
-    static QString processSet(NetworkCommand *nc);
+    QString processSet(NetworkCommand *nc);
     static QString processMessage(NetworkCommand *nc);
     static QString processNotification(NetworkCommand *nc);
     QString processTheme(NetworkCommand *nc);