diff -Naur mythtv-master-20150630-g44396c3-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20150630-g44396c3-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20150630-g44396c3-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2015-06-30 11:01:54.000000000 +0200
+++ mythtv-master-20150630-g44396c3-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2015-06-30 21:46:44.372369535 +0200
@@ -355,7 +355,7 @@
     {
         if (insert_crc(seen_table_crc, *pmt))
         {
-            LOG(VB_GENERAL, LOG_ERR, LOC +
+            LOG(VB_CHANNEL, LOG_ERR, LOC +
                 QString("Wrong PMT; pmt->pn(%1) desired(%2)")
                 .arg(pmt->ProgramNumber()).arg(programNumber));
         }
