diff -Naur mythtv-master-20150607-gde66bcd-old/mythtv/libs/libmythbase/remotefile.cpp mythtv-master-20150607-gde66bcd-new/mythtv/libs/libmythbase/remotefile.cpp
--- mythtv-master-20150607-gde66bcd-old/mythtv/libs/libmythbase/remotefile.cpp	2015-06-07 14:55:46.000000000 +0200
+++ mythtv-master-20150607-gde66bcd-new/mythtv/libs/libmythbase/remotefile.cpp	2015-06-07 14:57:55.970687243 +0200
@@ -48,7 +48,7 @@
     controlSock(NULL),    sock(NULL),
     query("QUERY_FILETRANSFER %1"),
     writemode(write),     completed(false),
-    localFile(-1),        fileWriter(NULL)
+    localFile(NULL),      fileWriter(NULL)
 {
     if (writemode)
     {
@@ -82,7 +82,7 @@
 bool RemoteFile::isLocal(const QString &path)
 {
     bool is_local = !path.isEmpty() &&
-        !path.startsWith("myth:") &&
+        !path.startsWith("/dev") && !path.startsWith("myth:") &&
         (path.startsWith("/") || QFile::exists(path));
     return is_local;
 }
@@ -228,7 +228,7 @@
 {
     if (isLocal())
     {
-        return writemode ? (fileWriter != NULL) : (localFile != -1);
+        return writemode ? (fileWriter != NULL) : (localFile != NULL);
     }
     return sock && controlSock;
 }
@@ -282,11 +282,13 @@
             return true;
         }
         // local mode, read only
-        localFile = ::open(path.toLocal8Bit().constData(), O_RDONLY);
-        if (localFile == -1)
+        localFile = new QFile(path);
+        if (!localFile->open(QIODevice::ReadOnly))
         {
             LOG(VB_FILE, LOG_ERR, QString("RemoteFile::Open(%1) Error: %2")
-                .arg(path).arg(strerror(errno)));
+                .arg(path).arg(localFile->error()));
+            delete localFile;
+            localFile = NULL;
             return false;
         }
         return true;
@@ -347,8 +349,8 @@
 {
     if (isLocal())
     {
-        ::close(localFile);
-        localFile = -1;
+        delete localFile;
+        localFile = NULL;
         delete fileWriter;
         fileWriter = NULL;
         return;
@@ -459,7 +461,7 @@
         else
         {
             QFileInfo info(url);
-            fileExists = info.exists() /*&& info.isFile()*/;
+            fileExists = info.exists() && info.isFile();
             fullFilePath = url;
         }
 
@@ -657,30 +659,26 @@
         long long offset = 0LL;
         if (whence == SEEK_SET)
         {
-            QFileInfo info(path);
-            offset = min(pos, info.size());
+            offset = min(pos, localFile->size());
         }
         else if (whence == SEEK_END)
         {
-            QFileInfo info(path);
-            offset = info.size() + pos;
+            offset = localFile->size() + pos;
         }
         else if (whence == SEEK_CUR)
         {
-            offset = ((curpos > 0) ? curpos : ::lseek64(localFile, 0, SEEK_CUR)) + pos;
+            offset = ((curpos > 0) ? curpos : localFile->pos()) + pos;
         }
         else
             return -1;
-
-        off64_t localpos = ::lseek64(localFile, (off64_t)pos, whence);
-        if (localpos != (off64_t)pos)
+        if (!localFile->seek(offset))
         {
             LOG(VB_FILE, LOG_ERR,
                 QString("RemoteFile::Seek(): Couldn't seek to offset %1")
                 .arg(offset));
             return -1;
         }
-        return (long long)localpos;
+        return localFile->pos();
     }
 
     if (!CheckConnection(false))
@@ -834,7 +832,8 @@
         }
         if (isOpen())
         {
-            return ::read(localFile, data, size);
+            QDataStream stream(localFile);
+            return stream.readRawData(static_cast<char*>(data), size);
         }
         LOG(VB_FILE, LOG_ERR, "RemoteFile:Read() called when local file not opened");
         return -1;
@@ -964,9 +963,15 @@
 {
     if (isLocal())
     {
-        if (isOpen() && writemode)
+        if (isOpen())
         {
-            fileWriter->Flush();
+            if (writemode)
+            {
+                fileWriter->Flush();
+                QFileInfo info(path);
+                return info.size();
+            }
+            return localFile->size();
         }
         if (Exists(path))
         {
diff -Naur mythtv-master-20150607-gde66bcd-old/mythtv/libs/libmythbase/remotefile.h mythtv-master-20150607-gde66bcd-new/mythtv/libs/libmythbase/remotefile.h
--- mythtv-master-20150607-gde66bcd-old/mythtv/libs/libmythbase/remotefile.h	2015-06-07 14:51:02.000000000 +0200
+++ mythtv-master-20150607-gde66bcd-new/mythtv/libs/libmythbase/remotefile.h	2015-06-07 14:57:55.970687243 +0200
@@ -93,7 +93,7 @@
 
     QStringList     possibleauxfiles;
     QStringList     auxfiles;
-    int             localFile;
+    QFile          *localFile;
     ThreadedFileWriter *fileWriter;
 };
 
