diff -Naur mythtv-master-20150525-ga96c03a-old/mythtv/libs/libmythtv/mythframe.h mythtv-master-20150525-ga96c03a-new/mythtv/libs/libmythtv/mythframe.h
--- mythtv-master-20150525-ga96c03a-old/mythtv/libs/libmythtv/mythframe.h	2015-05-25 10:30:25.000000000 +0200
+++ mythtv-master-20150525-ga96c03a-new/mythtv/libs/libmythtv/mythframe.h	2015-07-30 16:20:10.127904521 +0200
@@ -95,7 +95,7 @@
                         const int *p = 0,
                         const int *o = 0,
                         float _aspect = -1.0f, double _rate = -1.0f,
-                        int _aligned = 64) MUNUSED;
+                        int _aligned = 0) MUNUSED;
 static inline void clear(VideoFrame *vf);
 static inline bool compatible(const VideoFrame *a,
                               const VideoFrame *b) MUNUSED;
@@ -284,7 +284,7 @@
 }
 
 static inline uint buffersize(VideoFrameType type, int width, int height,
-                              int _aligned = 64)
+                              int _aligned = 16)
 {
     int  type_bpp = bitsperpixel(type);
     uint bpp = type_bpp / 4; /* bits per pixel div common factor */
diff -Naur mythtv-master-20150525-ga96c03a-old/mythtv/libs/libmythtv/videobuffers.h mythtv-master-20150525-ga96c03a-new/mythtv/libs/libmythtv/videobuffers.h
--- mythtv-master-20150525-ga96c03a-old/mythtv/libs/libmythtv/videobuffers.h	2015-05-25 10:30:31.000000000 +0200
+++ mythtv-master-20150525-ga96c03a-new/mythtv/libs/libmythtv/videobuffers.h	2015-07-30 16:20:10.127904521 +0200
@@ -47,7 +47,7 @@
 {
   public:
     YUVInfo(uint w, uint h, uint size, const int *p, const int *o,
-            int aligned = 64);
+            int aligned = 0);
 
   public:
     uint width;
