diff -Naur mythtv-master-20160218-ge4fface-old/mythtv/configure mythtv-master-20160218-ge4fface-new/mythtv/configure
--- mythtv-master-20160218-ge4fface-old/mythtv/configure	2016-02-18 07:52:30.000000000 +0100
+++ mythtv-master-20160218-ge4fface-new/mythtv/configure	2016-02-18 08:07:48.398731342 +0100
@@ -5689,7 +5689,7 @@
     if test -e $i ; then
         if is_qmake5 $i; then
             found_qmake=$i
-            check_cxxflags -DQT_DISABLE_DEPRECATED_BEFORE=0x050000
+            check_cxxflags -DQT_DISABLE_DEPRECATED_BEFORE
             break;
         else
             echo "found qmake at $i but version failed"
