diff -Naur mythtv-master-20160125-ga11d8cb-old/mythtv/libs/libmythbase/mythsystemlegacy.cpp mythtv-master-20160125-ga11d8cb-new/mythtv/libs/libmythbase/mythsystemlegacy.cpp
--- mythtv-master-20160125-ga11d8cb-old/mythtv/libs/libmythbase/mythsystemlegacy.cpp	2016-01-26 16:01:05.000000000 +0100
+++ mythtv-master-20160125-ga11d8cb-new/mythtv/libs/libmythbase/mythsystemlegacy.cpp	2016-01-26 21:47:22.000000000 +0100
@@ -520,6 +520,9 @@
     if (!ms->GetSetting("RunInBackground"))
         delete ms;
 
+    if (result == 2)
+        result = 0;
+
     return result;
 }
 
