diff --speed-large-files --no-dereference --minimal -Naur mythtv-3f3ee474d803ea0e8f1e10bbc83eb45abf00635e/mythtv/libs/libmythtv/mythplayer.cpp mythtv-3f3ee474d803ea0e8f1e10bbc83eb45abf00635e/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-3f3ee474d803ea0e8f1e10bbc83eb45abf00635e/mythtv/libs/libmythtv/mythplayer.cpp	2022-02-17 04:30:25.000000000 +0100
+++ mythtv-3f3ee474d803ea0e8f1e10bbc83eb45abf00635e/mythtv/libs/libmythtv/mythplayer.cpp	2022-02-17 21:07:58.234349812 +0100
@@ -759,10 +759,12 @@
             {
                 if (m_avSync.ResetAVSyncForLiveTV(&m_audio))
                 {
-                    LOG(VB_PLAYBACK, LOG_NOTICE, LOC + "Pause to allow live tv catch up");
                     LOG(VB_PLAYBACK, LOG_NOTICE, LOC + QString("Played: %1 Avail: %2 Buffered: %3 Margin: %4")
                         .arg(m_framesPlayed).arg(frameCount)
                         .arg(m_videoOutput->ValidVideoFrames()).arg(margin));
+                    LOG(VB_GENERAL, LOG_NOTICE, LOC + QString("Pause %1ms to allow live tv catch up")
+                        .arg(m_frameInterval.count() * 10 / 1000));
+                    std::this_thread::sleep_for(m_frameInterval * 10);
                 }
             }
         }
