diff --speed-large-files --no-dereference --minimal -Naur mythtv-594a0170077123801d56bf735f6f6f3a4c990014/mythtv/libs/libmythtv/mythplayer.cpp mythtv-594a0170077123801d56bf735f6f6f3a4c990014/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-594a0170077123801d56bf735f6f6f3a4c990014/mythtv/libs/libmythtv/mythplayer.cpp	2022-02-14 17:56:23.000000000 +0100
+++ mythtv-594a0170077123801d56bf735f6f6f3a4c990014/mythtv/libs/libmythtv/mythplayer.cpp	2022-02-14 18:57:59.361581781 +0100
@@ -725,7 +725,7 @@
 
 bool MythPlayer::PrebufferEnoughFrames(int min_buffers)
 {
-    if (!m_videoOutput || !m_videoOutput->ValidVideoFrames())
+    if (!m_videoOutput)
         return false;
 
     if (!(min_buffers
