diff -Naur mythtv-master-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-old/mythtv/libs/libmythtv/tv_play.cpp	2017-10-12 10:53:19.000000000 +0200
+++ mythtv-master-new/mythtv/libs/libmythtv/tv_play.cpp	2017-10-12 10:55:29.363333323 +0200
@@ -598,6 +598,10 @@
             "Change Recording Group"), "");
     REG_KEY("TV Frontend", "CHANGEGROUPVIEW", QT_TRANSLATE_NOOP("MythControls",
             "Change Group View"), "");
+    REG_KEY("TV Frontend", ACTION_ZOOMIN, QT_TRANSLATE_NOOP("MythControls",
+            "Zoom In"), "");
+    REG_KEY("TV Frontend", ACTION_ZOOMOUT, QT_TRANSLATE_NOOP("MythControls",
+            "Zoom Out"), "");
     REG_KEY("TV Frontend", ACTION_LISTRECORDEDEPISODES, QT_TRANSLATE_NOOP("MythControls",
             "List recorded episodes"), "");
     REG_KEY("TV Frontend", "THMBUP",     QT_TRANSLATE_NOOP("MythControls",
diff -Naur mythtv-master-old/mythtv/libs/libmythui/mythuiguidegrid.cpp mythtv-master-new/mythtv/libs/libmythui/mythuiguidegrid.cpp
--- mythtv-master-old/mythtv/libs/libmythui/mythuiguidegrid.cpp	2017-10-12 10:53:18.000000000 +0200
+++ mythtv-master-new/mythtv/libs/libmythui/mythuiguidegrid.cpp	2017-10-12 10:55:29.363333323 +0200
@@ -57,6 +57,8 @@
     m_drawCategoryText = true;
     m_categoryAlpha = 255;
 
+    m_needsUpdate = false;
+
     QMap<QString, QString> catColors;
     parseDefaultCategoryColors(catColors);
     SetCategoryColors(catColors);
@@ -258,6 +260,7 @@
 
     m_channelCount = gg->m_channelCount;
     m_timeCount = gg->m_timeCount;
+    m_rowCount = gg->m_rowCount;
     m_verticalLayout = gg->m_verticalLayout;
     m_categoryAlpha = gg->m_categoryAlpha;
     m_textOffset = gg->m_textOffset;
@@ -278,6 +281,26 @@
     m_drawCategoryColors = gg->m_drawCategoryColors;
     m_drawCategoryText = gg->m_drawCategoryText;
 
+    for (int i = 0; i < RECSTATUSSIZE; ++i)
+    {
+        MythImage *image = gg->m_recImages[i];
+        if (image)
+        {
+            m_recImages[i] = image;
+            image->IncrRef();
+        }
+    }
+
+    for (int i = 0; i < ARROWIMAGESIZE; ++i)
+    {
+        MythImage *image = gg->m_arrowImages[i];
+        if (image)
+        {
+            m_arrowImages[i] = image;
+            image->IncrRef();
+        }
+    }
+
     MythUIType::CopyFrom(base);
 }
 
@@ -685,6 +708,9 @@
 
 QPoint MythUIGuideGrid::GetRowAndColumn(QPoint position)
 {
+    if (!m_needsUpdate)
+        return QPoint(-1,-1);
+
     for (int i = 0; i < m_rowCount; i++)
     {
         QList<UIGTCon *>::iterator it = m_allData[i].begin();
@@ -721,6 +747,8 @@
 
     if (selected)
         m_selectedItem = *data;
+
+    Update();
 }
 
 bool MythUIGuideGrid::parseDefaultCategoryColors(QMap<QString, QString> &catColors)
@@ -827,6 +855,7 @@
 {
     for (int i = 0; i < m_rowCount; i++)
         ResetRow(i);
+    Update();
 }
 
 void MythUIGuideGrid::ResetRow(int row)
@@ -845,7 +874,7 @@
     else
         m_progPastCol = m_Area.x() + (m_Area.width() * ppast / 100);
 
-    SetRedraw();
+    Update();
 }
 
 void MythUIGuideGrid::SetMultiLine(bool multiline)
@@ -857,3 +886,9 @@
     else
         m_justification &= ~Qt::TextWordWrap;
 }
+
+void MythUIGuideGrid::Update()
+{
+    m_needsUpdate = true;
+    SetRedraw();
+}
diff -Naur mythtv-master-old/mythtv/libs/libmythui/mythuiguidegrid.h mythtv-master-new/mythtv/libs/libmythui/mythuiguidegrid.h
--- mythtv-master-old/mythtv/libs/libmythui/mythuiguidegrid.h	2017-10-12 10:53:18.000000000 +0200
+++ mythtv-master-new/mythtv/libs/libmythui/mythuiguidegrid.h	2017-10-12 10:57:51.353333331 +0200
@@ -69,7 +69,7 @@
     class UIGTCon
     {
       public:
-        UIGTCon() { m_arrow = m_recType = m_recStat = 0; };
+        UIGTCon() { m_arrow = m_recType = m_recStat = 0; }
         UIGTCon(const QRect &drawArea, const QString &title,
                 const QString &category, int arrow, int recType, int recStat) :
                 m_drawArea(drawArea),               m_title(title),
@@ -104,6 +104,8 @@
     void drawCurrent(MythPainter *p, int xoffset, int yoffset, UIGTCon *data,
                      int alphaMod);
 
+    void Update();
+
     QColor calcColor(const QColor &color, int alpha);
 
     QList<UIGTCon*> *m_allData;
@@ -142,6 +144,7 @@
 
     QMap<QString, QColor> m_categoryColors;
 
+    bool m_needsUpdate;
 };
 
 #endif
diff -Naur mythtv-master-old/mythtv/programs/mythfrontend/guidegrid.cpp mythtv-master-new/mythtv/programs/mythfrontend/guidegrid.cpp
--- mythtv-master-old/mythtv/programs/mythfrontend/guidegrid.cpp	2017-10-12 10:53:18.000000000 +0200
+++ mythtv-master-new/mythtv/programs/mythfrontend/guidegrid.cpp	2017-10-12 10:55:29.363333323 +0200
@@ -55,6 +55,8 @@
 // mythfrontend
 #include "progfind.h"
 
+#define GUIDEZOOMSETTING "GuideZoomLevel"
+
 QWaitCondition epgIsVisibleCond;
 
 #define LOC      QString("GuideGrid: ")
@@ -514,6 +516,7 @@
            m_updateTimer(new QTimer(this)),
            m_threadPool("GuideGridHelperPool"),
            m_changrpid(changrpid),
+           m_changrpname(),
            m_changrplist(ChannelGroup::GetChannelGroups(false)),
            m_jumpToChannelLock(QMutex::Recursive),
            m_jumpToChannel(NULL),
@@ -524,7 +527,12 @@
            m_longdateText(NULL),
            m_jumpToText(NULL),
            m_changroupname(NULL),
-           m_channelImage(NULL)
+           m_channelImage(NULL),
+           m_essentialWidgets(),
+           m_otherWidgets(),
+           m_zoomWidgets(),
+           m_zoomCurrent(0),
+           m_zoomMax(0)
 {
     connect(m_previewVideoRefreshTimer, SIGNAL(timeout()),
             this,                     SLOT(refreshVideo()));
@@ -550,6 +558,93 @@
     m_threadPool.setMaxThreadCount(1);
 }
 
+bool GuideGrid::FindZoomWidgets()
+{
+    bool ok = true;
+    MythUIType *widget;
+
+    foreach (const QString &name, m_essentialWidgets + m_otherWidgets)
+    {
+        // "0" suffix is optional for backward compatibility
+        widget = this->GetChild(name);
+        if (!widget)
+            widget = this->GetChild(name + "0");
+        if (widget)
+        {
+            ZoomLevels &zoomWidgets = m_zoomWidgets[name];
+
+            // find all zoom levels for this widget
+            int zoom = 0;
+            do
+            {
+                LOG(VB_GUI, LOG_DEBUG, QString("Using '%1' for zoom level %2")
+                    .arg(widget->objectName()).arg(zoom));
+
+                zoomWidgets.append(widget);
+                widget->SetVisible(false);
+
+                widget = this->GetChild(name + QString::number(++zoom));
+            }
+            while (widget);
+
+            // Note max zoom defined by theme
+            m_zoomMax = max(m_zoomMax, --zoom);
+        }
+        else
+        {
+            bool error = m_essentialWidgets.contains(name);
+            ok &= !error;
+            LOG(VB_GENERAL, error ? LOG_ERR : LOG_WARNING,
+                QString("Container '%1' is missing child '%2'")
+                .arg(objectName()).arg(name));
+        }
+    }
+    return ok;
+}
+
+MythUIType* GuideGrid::AssignWidget(const QString &name)
+{
+    ZoomLevels zoomWidgets = m_zoomWidgets.value(name);
+    if (zoomWidgets.isEmpty())
+        return NULL;
+    int index = min(m_zoomCurrent, zoomWidgets.size() - 1);
+    return zoomWidgets.at(index);
+}
+
+#define REASSIGN_WIDGET(widget, type, name) \
+    if (widget) \
+        widget->SetVisible(false); \
+    widget = dynamic_cast<type*>(AssignWidget(name)); \
+    if (widget) \
+        widget->SetVisible(true)
+
+void GuideGrid::SelectZoomWidgets(int change)
+{
+    // Constrain to zoom levels supported by theme
+    m_zoomCurrent = min(max(m_zoomCurrent + change, 0), m_zoomMax);
+
+    // Store any requested change, but not constraining adjustments
+    // Thus, changing to a theme with fewer zoom levels will not overwrite the
+    // setting
+    if (change != 0)
+        gCoreContext->SaveSetting(GUIDEZOOMSETTING, m_zoomCurrent);
+
+    // Hide the current widgets, select and show new ones
+    REASSIGN_WIDGET(m_timeList, MythUIButtonList, "timelist");
+    REASSIGN_WIDGET(m_channelList, MythUIButtonList, "channellist");
+    REASSIGN_WIDGET(m_guideGrid, MythUIGuideGrid, "guidegrid");
+    REASSIGN_WIDGET(m_dateText, MythUIText, "datetext");
+    REASSIGN_WIDGET(m_longdateText, MythUIText, "longdatetext");
+    REASSIGN_WIDGET(m_changroupname, MythUIText, "channelgroup");
+
+    // Reset guide characteristics
+    m_channelCount = m_guideGrid->getChannelCount();
+    m_timeCount = m_guideGrid->getTimeCount() * 6;
+    m_verticalLayout = m_guideGrid->isVerticalLayout();
+
+    m_currentEndTime = m_currentStartTime.addSecs(m_timeCount * 60 * 5);
+}
+
 bool GuideGrid::Create()
 {
     QString windowName = "programguide";
@@ -560,23 +655,22 @@
     if (!LoadWindowFromXML("schedule-ui.xml", windowName, this))
         return false;
 
-    bool err = false;
-    UIUtilE::Assign(this, m_timeList, "timelist", &err);
-    UIUtilE::Assign(this, m_channelList, "channellist", &err);
-    UIUtilE::Assign(this, m_guideGrid, "guidegrid", &err);
-    UIUtilW::Assign(this, m_dateText, "datetext");
-    UIUtilW::Assign(this, m_longdateText, "longdatetext");
-    UIUtilW::Assign(this, m_changroupname, "channelgroup");
-    UIUtilW::Assign(this, m_channelImage, "channelicon");
-    UIUtilW::Assign(this, m_jumpToText, "jumptotext");
+    m_essentialWidgets << "timelist" << "channellist" << "guidegrid";
+    m_otherWidgets << "channelgroup" << "datetext" << "longdatetext";
 
-    if (err)
+    if (!FindZoomWidgets())
     {
-        LOG(VB_GENERAL, LOG_ERR,
-            QString("Cannot load screen '%1'").arg(windowName));
+        LOG(VB_GENERAL, LOG_ERR, QString("Cannot load screen '%1'").arg(windowName));
         return false;
     }
 
+    UIUtilW::Assign(this, m_channelImage, "channelicon");
+    UIUtilW::Assign(this, m_jumpToText, "jumptotext");
+
+    m_zoomCurrent = gCoreContext->GetNumSetting(GUIDEZOOMSETTING, 0);
+
+    SelectZoomWidgets(0);
+
     BuildFocusList();
 
     MythUIImage *videoImage = dynamic_cast<MythUIImage *>(GetChild("video"));
@@ -585,12 +679,6 @@
     else
         m_videoRect = QRect(0,0,0,0);
 
-    m_channelCount = m_guideGrid->getChannelCount();
-    m_timeCount = m_guideGrid->getTimeCount() * 6;
-    m_verticalLayout = m_guideGrid->isVerticalLayout();
-
-    m_currentEndTime = m_currentStartTime.addSecs(m_timeCount * 60 * 5);
-
     LoadInBackground();
     return true;
 }
@@ -635,10 +723,9 @@
 
     updateDateText();
 
-    QString changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
+    m_changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
 
-    if (m_changroupname)
-        m_changroupname->SetText(changrpname);
+    UpdateGroupText();
 
     gCoreContext->addListener(this);
 }
@@ -865,6 +952,10 @@
             generateListings();
             updateChannels();
         }
+        else if (action == ACTION_ZOOMIN)
+            Zoom(1);
+        else if (action == ACTION_ZOOMOUT)
+            Zoom(-1);
         else
             handled = false;
     }
@@ -1125,6 +1216,12 @@
                 menuPopup->AddButton(tr("Remove from Channel Group"), NULL, true);
         }
 
+        if (m_zoomCurrent > 0)
+            menuPopup->AddButton(tr("Zoom Out"));
+
+        if (m_zoomCurrent < m_zoomMax)
+            menuPopup->AddButton(tr("Zoom In"));
+
         popupStack->AddScreen(menuPopup);
     }
     else
@@ -1971,6 +2068,14 @@
             {
                 ShowJumpToTime();
             }
+            else if (resulttext == tr("Zoom In"))
+            {
+                Zoom(1);
+            }
+            else if (resulttext == tr("Zoom Out"))
+            {
+                Zoom(-1);
+            }
         }
         else if (resultid == "recmenu")
         {
@@ -2024,11 +2129,9 @@
                 updateChannels();
                 updateInfo();
 
-                QString changrpname;
-                changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
+                m_changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
 
-                if (m_changroupname)
-                    m_changroupname->SetText(changrpname);
+                UpdateGroupText();
             }
         }
         else if (resultid == "jumptotime")
@@ -2060,6 +2163,12 @@
                                                  (MythDate::kDateFull | MythDate::kSimplify)));
 }
 
+void GuideGrid::UpdateGroupText(void)
+{
+    if (m_changroupname)
+        m_changroupname->SetText(m_changrpname);
+}
+
 void GuideGrid::updateProgramsUI(unsigned int firstRow, unsigned int numRows,
                                  int progPast,
                                  const QVector<ProgramList*> &proglists,
@@ -2255,17 +2364,14 @@
     int oldchangrpid = m_changrpid;
 
     m_changrpid = ChannelGroup::GetNextChannelGroup(m_changrplist, oldchangrpid);
+    m_changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
 
     if (oldchangrpid != m_changrpid)
       generateListings();
 
     updateChannels();
     updateInfo();
-
-    QString changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
-
-    if (m_changroupname)
-        m_changroupname->SetText(changrpname);
+    UpdateGroupText();
 }
 
 void GuideGrid::generateListings()
@@ -2745,3 +2851,39 @@
     else
         delete timedlg;
 }
+
+void GuideGrid::Zoom(int direction)
+{
+    ProgramInfo *pginfo = m_programInfos[m_currentRow][m_currentCol];
+    if (!pginfo)
+        return;
+
+    // Change widgets
+    SelectZoomWidgets(direction);
+
+    m_channelCount = min(m_guideGrid->getChannelCount(),
+                         max((int)GetChannelCount(), 1));
+
+    // Keep selected channel on-screen
+    int rowsOffScreen = m_currentRow - m_channelCount + 1;
+    if (rowsOffScreen > 0)
+    {
+        m_currentStartChannel += rowsOffScreen;
+        m_currentRow          -= rowsOffScreen;
+    }
+
+    // Keep selected program on-screen by putting it in first column
+    QDateTime progStart = pginfo->GetScheduledStartTime();
+    if (progStart >= m_currentEndTime)
+    {
+        m_currentStartTime = progStart;
+        m_currentCol = 0;
+    }
+
+    // Redraw
+    fillTimeInfos();
+    fillProgramInfos();
+    updateChannels();
+    updateDateText();
+    UpdateGroupText();
+}
diff -Naur mythtv-master-old/mythtv/programs/mythfrontend/guidegrid.h mythtv-master-new/mythtv/programs/mythfrontend/guidegrid.h
--- mythtv-master-old/mythtv/programs/mythfrontend/guidegrid.h	2017-10-12 10:53:18.000000000 +0200
+++ mythtv-master-new/mythtv/programs/mythfrontend/guidegrid.h	2017-10-12 10:55:29.363333323 +0200
@@ -232,6 +232,11 @@
     ProgramList GetProgramList(uint chanid) const;
     uint GetAlternateChannelIndex(uint chan_idx, bool with_same_channum) const;
     void updateDateText(void);
+    void UpdateGroupText(void);
+    bool FindZoomWidgets();
+    void SelectZoomWidgets(int change);
+    MythUIType* AssignWidget(const QString &name);
+    void Zoom(int direction);
 
   private:
     int   m_selectRecThreshold;
@@ -278,6 +283,7 @@
     MThreadPool       m_threadPool;
 
     int               m_changrpid;
+    QString           m_changrpname;
     ChannelGroupList  m_changrplist;
 
     QMutex            m_jumpToChannelLock;
@@ -291,6 +297,14 @@
     MythUIText       *m_jumpToText;
     MythUIText       *m_changroupname;
     MythUIImage      *m_channelImage;
+
+    // Zoomable widgets enable different sized grids
+    QStringList m_essentialWidgets, m_otherWidgets;
+
+    typedef QList<MythUIType*> ZoomLevels;
+    QMap<QString, ZoomLevels> m_zoomWidgets;
+    int               m_zoomCurrent; // Current zoom level
+    int               m_zoomMax; // Max zoom level defined by theme
 };
 
 #endif
