diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythtv/tv_play.cpp	2020-10-02 13:03:08.113333333 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythtv/tv_play.cpp	2020-10-02 13:03:10.356666666 +0200
@@ -549,6 +549,10 @@
             "Change Recording Group"), "");
     REG_KEY("TV Frontend", "CHANGEGROUPVIEW", QT_TRANSLATE_NOOP("MythControls",
             "Change Group View"), "");
+    REG_KEY("TV Frontend", ACTION_ZOOMIN, QT_TRANSLATE_NOOP("MythControls",
+            "Zoom In"), "");
+    REG_KEY("TV Frontend", ACTION_ZOOMOUT, QT_TRANSLATE_NOOP("MythControls",
+            "Zoom Out"), "");
     REG_KEY("TV Frontend", ACTION_LISTRECORDEDEPISODES, QT_TRANSLATE_NOOP("MythControls",
             "List recorded episodes"), "");
     /*
diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythui/mythuiguidegrid.cpp mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythui/mythuiguidegrid.cpp
--- mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythui/mythuiguidegrid.cpp	2020-10-02 13:03:08.100000000 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythui/mythuiguidegrid.cpp	2020-10-02 13:03:10.356666666 +0200
@@ -30,6 +30,8 @@
 {
     m_font = new MythFontProperties();
 
+    m_needsUpdate = false;
+
     QMap<QString, QString> catColors;
     parseDefaultCategoryColors(catColors);
     SetCategoryColors(catColors);
@@ -222,6 +224,7 @@
 
     m_channelCount = gg->m_channelCount;
     m_timeCount = gg->m_timeCount;
+    m_rowCount = gg->m_rowCount;
     m_verticalLayout = gg->m_verticalLayout;
     m_categoryAlpha = gg->m_categoryAlpha;
     m_textOffset = gg->m_textOffset;
@@ -659,6 +662,9 @@
 
 QPoint MythUIGuideGrid::GetRowAndColumn(QPoint position)
 {
+    if (!m_needsUpdate)
+        return QPoint(-1,-1);
+
     for (int i = 0; i < m_rowCount; i++)
     {
         QList<UIGTCon *>::iterator it = m_allData[i].begin();
@@ -695,6 +701,8 @@
 
     if (selected)
         m_selectedItem = *data;
+
+    Update();
 }
 
 bool MythUIGuideGrid::parseDefaultCategoryColors(QMap<QString, QString> &catColors)
@@ -792,6 +800,7 @@
 {
     for (int i = 0; i < m_rowCount; i++)
         ResetRow(i);
+    Update();
 }
 
 void MythUIGuideGrid::ResetRow(int row)
@@ -810,7 +819,7 @@
     else
         m_progPastCol = m_area.x() + (m_area.width() * ppast / 100);
 
-    SetRedraw();
+    Update();
 }
 
 void MythUIGuideGrid::SetMultiLine(bool multiline)
@@ -822,3 +831,9 @@
     else
         m_justification &= ~Qt::TextWordWrap;
 }
+
+void MythUIGuideGrid::Update()
+{
+    m_needsUpdate = true;
+    SetRedraw();
+}
diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythui/mythuiguidegrid.h mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythui/mythuiguidegrid.h
--- mythtv-master-20201001-g2a40eb3-old/mythtv/libs/libmythui/mythuiguidegrid.h	2020-10-02 13:03:08.100000000 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/libs/libmythui/mythuiguidegrid.h	2020-10-02 13:03:10.356666666 +0200
@@ -83,7 +83,7 @@
     class UIGTCon
     {
       public:
-        UIGTCon() { m_arrow = GridTimeNormal; m_recType = m_recStat = 0; };
+        UIGTCon() { m_arrow = GridTimeNormal; m_recType = m_recStat = 0; }
         UIGTCon(const QRect &drawArea, QString title,
                 const QString &category, int arrow, int recType, int recStat) :
                 m_drawArea(drawArea),               m_title(std::move(title)),
@@ -114,6 +114,8 @@
     void drawCurrent(MythPainter *p, int xoffset, int yoffset, UIGTCon *data,
                      int alphaMod);
 
+    void Update();
+
     static QColor calcColor(const QColor &color, int alpha);
 
     QList<UIGTCon*> *m_allData  {nullptr};
@@ -153,6 +155,7 @@
 
     QMap<QString, QColor> m_categoryColors;
 
+    bool m_needsUpdate;
 };
 
 #endif
diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/programs/mythfrontend/guidegrid.cpp mythtv-master-20201001-g2a40eb3-new/mythtv/programs/mythfrontend/guidegrid.cpp
--- mythtv-master-20201001-g2a40eb3-old/mythtv/programs/mythfrontend/guidegrid.cpp	2020-10-02 13:03:08.190000000 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/programs/mythfrontend/guidegrid.cpp	2020-10-02 13:03:10.360000000 +0200
@@ -55,6 +55,8 @@
 // mythfrontend
 #include "progfind.h"
 
+#define GUIDEZOOMSETTING "GuideZoomLevel"
+
 QWaitCondition epgIsVisibleCond;
 
 #define LOC      QString("GuideGrid: ")
@@ -491,7 +493,13 @@
     m_updateTimer(new QTimer(this)),
     m_threadPool("GuideGridHelperPool"),
     m_changrpid(changrpid),
-    m_changrplist(ChannelGroup::GetChannelGroups(false))
+    m_changrplist(ChannelGroup::GetChannelGroups(false)),
+    m_changrpname(),
+    m_essentialWidgets(),
+    m_otherWidgets(),
+    m_zoomWidgets(),
+    m_zoomCurrent(0),
+    m_zoomMax(0)
 {
     connect(m_updateTimer, SIGNAL(timeout()), SLOT(updateTimeout()) );
 
@@ -526,6 +534,93 @@
     }
 }
 
+bool GuideGrid::FindZoomWidgets()
+{
+    bool ok = true;
+    MythUIType *widget;
+
+    foreach (const QString &name, m_essentialWidgets + m_otherWidgets)
+    {
+        // "0" suffix is optional for backward compatibility
+        widget = this->GetChild(name);
+        if (!widget)
+            widget = this->GetChild(name + "0");
+        if (widget)
+        {
+            ZoomLevels &zoomWidgets = m_zoomWidgets[name];
+
+            // find all zoom levels for this widget
+            int zoom = 0;
+            do
+            {
+                LOG(VB_GUI, LOG_DEBUG, QString("Using '%1' for zoom level %2")
+                    .arg(widget->objectName()).arg(zoom));
+
+                zoomWidgets.append(widget);
+                widget->SetVisible(false);
+
+                widget = this->GetChild(name + QString::number(++zoom));
+            }
+            while (widget);
+
+            // Note max zoom defined by theme
+            m_zoomMax = max(m_zoomMax, --zoom);
+        }
+        else
+        {
+            bool error = m_essentialWidgets.contains(name);
+            ok &= !error;
+            LOG(VB_GENERAL, error ? LOG_ERR : LOG_WARNING,
+                QString("Container '%1' is missing child '%2'")
+                .arg(objectName()).arg(name));
+        }
+    }
+    return ok;
+}
+
+MythUIType* GuideGrid::AssignWidget(const QString &name)
+{
+    ZoomLevels zoomWidgets = m_zoomWidgets.value(name);
+    if (zoomWidgets.isEmpty())
+        return nullptr;
+    int index = min(m_zoomCurrent, zoomWidgets.size() - 1);
+    return zoomWidgets.at(index);
+}
+
+#define REASSIGN_WIDGET(widget, type, name) \
+    if (widget) \
+        widget->SetVisible(false); \
+    widget = dynamic_cast<type*>(AssignWidget(name)); \
+    if (widget) \
+        widget->SetVisible(true)
+
+void GuideGrid::SelectZoomWidgets(int change)
+{
+    // Constrain to zoom levels supported by theme
+    m_zoomCurrent = min(max(m_zoomCurrent + change, 0), m_zoomMax);
+
+    // Store any requested change, but not constraining adjustments
+    // Thus, changing to a theme with fewer zoom levels will not overwrite the
+    // setting
+    if (change != 0)
+        gCoreContext->SaveSetting(GUIDEZOOMSETTING, m_zoomCurrent);
+
+    // Hide the current widgets, select and show new ones
+    REASSIGN_WIDGET(m_timeList, MythUIButtonList, "timelist");
+    REASSIGN_WIDGET(m_channelList, MythUIButtonList, "channellist");
+    REASSIGN_WIDGET(m_guideGrid, MythUIGuideGrid, "guidegrid");
+    REASSIGN_WIDGET(m_dateText, MythUIText, "datetext");
+    REASSIGN_WIDGET(m_longdateText, MythUIText, "longdatetext");
+    REASSIGN_WIDGET(m_changroupname, MythUIText, "channelgroup");
+
+    // Reset guide characteristics
+    m_channelCount = m_guideGrid->getChannelCount();
+    m_timeCount = m_guideGrid->getTimeCount() * 6;
+    m_verticalLayout = m_guideGrid->isVerticalLayout();
+
+    m_currentEndTime = m_currentStartTime.addSecs(m_timeCount * 60 * 5);
+}
+
 bool GuideGrid::Create()
 {
     QString windowName = "programguide";
@@ -536,23 +631,22 @@
     if (!LoadWindowFromXML("schedule-ui.xml", windowName, this))
         return false;
 
-    bool err = false;
-    UIUtilE::Assign(this, m_timeList, "timelist", &err);
-    UIUtilE::Assign(this, m_channelList, "channellist", &err);
-    UIUtilE::Assign(this, m_guideGrid, "guidegrid", &err);
-    UIUtilW::Assign(this, m_dateText, "datetext");
-    UIUtilW::Assign(this, m_longdateText, "longdatetext");
-    UIUtilW::Assign(this, m_changroupname, "channelgroup");
-    UIUtilW::Assign(this, m_channelImage, "channelicon");
-    UIUtilW::Assign(this, m_jumpToText, "jumptotext");
+    m_essentialWidgets << "timelist" << "channellist" << "guidegrid";
+    m_otherWidgets << "channelgroup" << "datetext" << "longdatetext";
 
-    if (err)
+    if (!FindZoomWidgets())
     {
-        LOG(VB_GENERAL, LOG_ERR,
-            QString("Cannot load screen '%1'").arg(windowName));
+        LOG(VB_GENERAL, LOG_ERR, QString("Cannot load screen '%1'").arg(windowName));
         return false;
     }
 
+    UIUtilW::Assign(this, m_channelImage, "channelicon");
+    UIUtilW::Assign(this, m_jumpToText, "jumptotext");
+
+    m_zoomCurrent = gCoreContext->GetNumSetting(GUIDEZOOMSETTING, 0);
+
+    SelectZoomWidgets(0);
+
     BuildFocusList();
 
     MythUIImage *videoImage = dynamic_cast<MythUIImage *>(GetChild("video"));
@@ -561,12 +655,6 @@
     else
         m_videoRect = QRect(0,0,0,0);
 
-    m_channelCount = m_guideGrid->getChannelCount();
-    m_timeCount = m_guideGrid->getTimeCount() * 6;
-    m_verticalLayout = m_guideGrid->isVerticalLayout();
-
-    m_currentEndTime = m_currentStartTime.addSecs(m_timeCount * 60 * 5);
-
     LoadInBackground();
     return true;
 }
@@ -611,10 +699,9 @@
 
     updateDateText();
 
-    QString changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
+    m_changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
 
-    if (m_changroupname)
-        m_changroupname->SetText(changrpname);
+    UpdateGroupText();
 
     gCoreContext->addListener(this);
 }
@@ -836,6 +923,10 @@
             generateListings();
             updateChannels();
         }
+        else if (action == ACTION_ZOOMIN)
+            Zoom(1);
+        else if (action == ACTION_ZOOMOUT)
+            Zoom(-1);
         else
             handled = false;
     }
@@ -1095,6 +1186,12 @@
                 menuPopup->AddButton(tr("Remove from Channel Group"), nullptr, true);
         }
 
+        if (m_zoomCurrent > 0)
+            menuPopup->AddButton(tr("Zoom Out"));
+
+        if (m_zoomCurrent < m_zoomMax)
+            menuPopup->AddButton(tr("Zoom In"));
+
         popupStack->AddScreen(menuPopup);
     }
     else
@@ -1930,6 +2027,14 @@
             {
                 ShowJumpToTime();
             }
+            else if (resulttext == tr("Zoom In"))
+            {
+                Zoom(1);
+            }
+            else if (resulttext == tr("Zoom Out"))
+            {
+                Zoom(-1);
+            }
         }
         else if (resultid == "recmenu")
         {
@@ -1982,11 +2087,9 @@
                 updateChannels();
                 updateInfo();
 
-                QString changrpname;
-                changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
+                m_changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
 
-                if (m_changroupname)
-                    m_changroupname->SetText(changrpname);
+                UpdateGroupText();
             }
         }
         else if (resultid == "jumptotime")
@@ -2018,6 +2121,12 @@
                                                  (MythDate::kDateFull | MythDate::kSimplify)));
 }
 
+void GuideGrid::UpdateGroupText(void)
+{
+    if (m_changroupname)
+        m_changroupname->SetText(m_changrpname);
+}
+
 void GuideGrid::updateProgramsUI(unsigned int firstRow, unsigned int numRows,
                                  int progPast,
                                  const QVector<ProgramList*> &proglists,
@@ -2210,17 +2319,14 @@
     int oldchangrpid = m_changrpid;
 
     m_changrpid = ChannelGroup::GetNextChannelGroup(m_changrplist, oldchangrpid);
+    m_changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
 
     if (oldchangrpid != m_changrpid)
       generateListings();
 
     updateChannels();
     updateInfo();
-
-    QString changrpname = ChannelGroup::GetChannelGroupName(m_changrpid);
-
-    if (m_changroupname)
-        m_changroupname->SetText(changrpname);
+    UpdateGroupText();
 }
 
 void GuideGrid::generateListings()
@@ -2680,3 +2786,39 @@
     else
         delete timedlg;
 }
+
+void GuideGrid::Zoom(int direction)
+{
+    ProgramInfo *pginfo = m_programInfos[m_currentRow][m_currentCol];
+    if (!pginfo)
+        return;
+
+    // Change widgets
+    SelectZoomWidgets(direction);
+
+    m_channelCount = min(m_guideGrid->getChannelCount(),
+                         max((int)GetChannelCount(), 1));
+
+    // Keep selected channel on-screen
+    int rowsOffScreen = m_currentRow - m_channelCount + 1;
+    if (rowsOffScreen > 0)
+    {
+        m_currentStartChannel += rowsOffScreen;
+        m_currentRow          -= rowsOffScreen;
+    }
+
+    // Keep selected program on-screen by putting it in first column
+    QDateTime progStart = pginfo->GetScheduledStartTime();
+    if (progStart >= m_currentEndTime)
+    {
+        m_currentStartTime = progStart;
+        m_currentCol = 0;
+    }
+
+    // Redraw
+    fillTimeInfos();
+    fillProgramInfos();
+    updateChannels();
+    updateDateText();
+    UpdateGroupText();
+}
diff -Naur mythtv-master-20201001-g2a40eb3-old/mythtv/programs/mythfrontend/guidegrid.h mythtv-master-20201001-g2a40eb3-new/mythtv/programs/mythfrontend/guidegrid.h
--- mythtv-master-20201001-g2a40eb3-old/mythtv/programs/mythfrontend/guidegrid.h	2020-10-02 13:03:08.190000000 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythtv/programs/mythfrontend/guidegrid.h	2020-10-02 13:03:10.360000000 +0200
@@ -236,6 +236,11 @@
     ProgramList GetProgramList(uint chanid) const;
     uint GetAlternateChannelIndex(uint chan_idx, bool with_same_channum) const;
     void updateDateText(void);
+    void UpdateGroupText(void);
+    bool FindZoomWidgets();
+    void SelectZoomWidgets(int change);
+    MythUIType* AssignWidget(const QString &name);
+    void Zoom(int direction);
 
   private:
     int   m_selectRecThreshold      {16};
@@ -280,6 +285,7 @@
     MThreadPool       m_threadPool;
 
     int               m_changrpid {-1};
+    QString           m_changrpname;
     ChannelGroupList  m_changrplist;
 
     QMutex            m_jumpToChannelLock {QMutex::Recursive};
@@ -293,6 +299,14 @@
     MythUIText       *m_jumpToText        {nullptr};
     MythUIText       *m_changroupname     {nullptr};
     MythUIImage      *m_channelImage      {nullptr};
+
+    // Zoomable widgets enable different sized grids
+    QStringList m_essentialWidgets, m_otherWidgets;
+
+    typedef QList<MythUIType*> ZoomLevels;
+    QMap<QString, ZoomLevels> m_zoomWidgets;
+    int               m_zoomCurrent; // Current zoom level
+    int               m_zoomMax; // Max zoom level defined by theme
 };
 
 #endif
