diff -Naur mythtv-master-20160618-g0d24636-old/mythtv/libs/libmythui/libmythui.pro mythtv-master-20160618-g0d24636-new/mythtv/libs/libmythui/libmythui.pro
--- mythtv-master-20160618-g0d24636-old/mythtv/libs/libmythui/libmythui.pro	2016-06-18 11:12:41.000000000 +0200
+++ mythtv-master-20160618-g0d24636-new/mythtv/libs/libmythui/libmythui.pro	2016-06-21 18:30:59.786666651 +0200
@@ -4,11 +4,15 @@
 contains(QT_VERSION, ^4\\.[0-9]\\..*) {
 QT += webkit
 }
-contains(QT_VERSION, ^5\\.[0-9]\\..*):using_qtwebkit {
+contains(QT_VERSION, ^5\\.[0-5]\\..*) {
 QT += widgets
 QT += webkitwidgets
 android: QT += androidextras
 }
+contains(QT_VERSION, ^5\\.[6-9]\\..*) {
+QT += widgets
+android: QT += androidextras
+}
 
 TEMPLATE = lib
 TARGET = mythui-$$LIBVERSION
@@ -66,7 +70,7 @@
 SOURCES += mythnotificationcenter.cpp mythnotification.cpp
 SOURCES += mythuicomposite.cpp
 
-using_qtwebkit {
+!contains(QT_VERSION, ^5\\.[6-9]\\..*) {
 HEADERS += mythuiwebbrowser.h
 SOURCES += mythuiwebbrowser.cpp
 }
diff -Naur mythtv-master-20160618-g0d24636-old/mythtv/libs/libmythui/mythuitype.h mythtv-master-20160618-g0d24636-new/mythtv/libs/libmythui/mythuitype.h
--- mythtv-master-20160618-g0d24636-old/mythtv/libs/libmythui/mythuitype.h	2016-06-18 11:12:42.000000000 +0200
+++ mythtv-master-20160618-g0d24636-new/mythtv/libs/libmythui/mythuitype.h	2016-06-21 18:30:59.786666651 +0200
@@ -35,7 +35,7 @@
 class MythUITextEdit;
 class MythUIProgressBar;
 
-#ifdef CONFIG_QTWEBKIT
+#if ( QT_VERSION <= QT_VERSION_CHECK( 5, 5, 0 ))
 class MythUIWebBrowser;
 #endif
 
diff -Naur mythtv-master-20160618-g0d24636-old/mythtv/libs/libmythui/xmlparsebase.cpp mythtv-master-20160618-g0d24636-new/mythtv/libs/libmythui/xmlparsebase.cpp
--- mythtv-master-20160618-g0d24636-old/mythtv/libs/libmythui/xmlparsebase.cpp	2016-06-18 11:12:42.000000000 +0200
+++ mythtv-master-20160618-g0d24636-new/mythtv/libs/libmythui/xmlparsebase.cpp	2016-06-21 18:30:59.786666651 +0200
@@ -34,7 +34,7 @@
 #include "mythuiscrollbar.h"
 #include "mythuigroup.h"
 
-#ifdef CONFIG_QTWEBKIT
+#if ( QT_VERSION < QT_VERSION_CHECK( 5, 6, 0 ))
 #include "mythuiwebbrowser.h"
 #endif
 
@@ -495,7 +495,7 @@
         uitype = new MythUIProgressBar(parent, name);
     else if (type == "scrollbar")
         uitype = new MythUIScrollBar(parent, name);
-#ifdef CONFIG_QTWEBKIT
+#if ( QT_VERSION < QT_VERSION_CHECK( 5, 6, 0 ))
     else if (type == "webbrowser")
         uitype = new MythUIWebBrowser(parent, name);
 #endif
diff -Naur mythtv-master-20160618-g0d24636-old/mythtv/programs/mythfrontend/mythfrontend.pro mythtv-master-20160618-g0d24636-new/mythtv/programs/mythfrontend/mythfrontend.pro
--- mythtv-master-20160618-g0d24636-old/mythtv/programs/mythfrontend/mythfrontend.pro	2016-06-18 11:12:43.000000000 +0200
+++ mythtv-master-20160618-g0d24636-new/mythtv/programs/mythfrontend/mythfrontend.pro	2016-06-21 18:30:59.786666651 +0200
@@ -8,14 +8,18 @@
    QT += scripttools
 }
 contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-QT += webkit
+# QT += webkit
 using_qtdbus: CONFIG += qdbus
 }
-contains(QT_VERSION, ^5\\.[0-9]\\..*):using_qtwebkit {
+contains(QT_VERSION, ^5\\.[0-5]\\..*) {
 QT += widgets
 QT += webkitwidgets
 using_qtdbus: QT += dbus
 }
+contains(QT_VERSION, ^5\\.[6-9]\\..*) {
+QT += widgets
+using_qtdbus: QT += dbus
+}
 
 TEMPLATE = app
 CONFIG += thread
@@ -86,7 +90,10 @@
 HEADERS += services/frontend.h
 SOURCES += services/frontend.cpp
 
-using_qtwebkit {
+
+!contains(QT_VERSION, ^5\\.[6-9]\\..*) {
+    # temp disable features for Qt 5.6 until QtWebEngine is implemented correctly
+
     HEADERS += progdetails.h
     SOURCES += progdetails.cpp
 }
diff -Naur mythtv-master-20160618-g0d24636-old/mythtv/programs/mythfrontend/networkcontrol.cpp mythtv-master-20160618-g0d24636-new/mythtv/programs/mythfrontend/networkcontrol.cpp
--- mythtv-master-20160618-g0d24636-old/mythtv/programs/mythfrontend/networkcontrol.cpp	2016-06-21 18:29:06.000000000 +0200
+++ mythtv-master-20160618-g0d24636-new/mythtv/programs/mythfrontend/networkcontrol.cpp	2016-06-21 18:30:59.789999985 +0200
@@ -37,7 +37,7 @@
 #include "mythuicheckbox.h"
 #include "mythuispinbox.h"
 
-#ifdef CONFIG_QTWEBKIT
+#if ( QT_VERSION < QT_VERSION_CHECK( 5, 6, 0 ))
 #include "mythuiwebbrowser.h"
 #endif
 
@@ -1181,7 +1181,7 @@
         return "MythUIImage";
     else if (dynamic_cast<MythUISpinBox *>(type))
         return "MythUISpinBox";
-#ifdef CONFIG_QTWEBKIT
+#if ( QT_VERSION < QT_VERSION_CHECK( 5, 6, 0 ))
     else if (dynamic_cast<MythUIWebBrowser *>(type))
         return "MythUIWebBrowser";
 #endif
diff -Naur mythtv-master-20160618-g0d24636-old/mythtv/programs/mythfrontend/schedulecommon.cpp mythtv-master-20160618-g0d24636-new/mythtv/programs/mythfrontend/schedulecommon.cpp
--- mythtv-master-20160618-g0d24636-old/mythtv/programs/mythfrontend/schedulecommon.cpp	2016-06-18 11:12:43.000000000 +0200
+++ mythtv-master-20160618-g0d24636-new/mythtv/programs/mythfrontend/schedulecommon.cpp	2016-06-21 18:30:59.793333318 +0200
@@ -21,14 +21,14 @@
 // mythfrontend
 #include "scheduleeditor.h"
 
+#if QT_VERSION < QT_VERSION_CHECK(5,6,0)
+#include "progdetails.h"
+#endif
+
 #include "proglist.h"
 #include "customedit.h"
 #include "guidegrid.h"
 
-#ifdef CONFIG_QTWEBKIT
-#include "progdetails.h"
-#endif
-
 /**
 *  \brief Show the Program Details screen
 */
@@ -38,7 +38,7 @@
     if (!pginfo)
         return;
 
-#ifdef CONFIG_QTWEBKIT
+#if QT_VERSION < QT_VERSION_CHECK(5,6,0)
     MythScreenStack *mainStack = GetMythMainWindow()->GetMainStack();
     ProgDetails *details_dialog  = new ProgDetails(mainStack, pginfo);
 
diff -Naur mythtv-master-20160618-g0d24636-old/platform/win32/msvc/include/mythconfig.h mythtv-master-20160618-g0d24636-new/platform/win32/msvc/include/mythconfig.h
--- mythtv-master-20160618-g0d24636-old/platform/win32/msvc/include/mythconfig.h	2016-06-18 11:13:04.000000000 +0200
+++ mythtv-master-20160618-g0d24636-new/platform/win32/msvc/include/mythconfig.h	2016-06-21 18:30:59.796666652 +0200
@@ -1571,6 +1571,7 @@
 #define CONFIG_MHEG 1
 #define CONFIG_OPENGL 1
 #define CONFIG_OPENGL_VIDEO 1
+#define CONFIG_QTWEBKIT 1
 #define CONFIG_QTSCRIPT 1
 #define CONFIG_QTDBUS 0
 #define CONFIG_SDL 1
@@ -1581,6 +1582,6 @@
 #define CONFIG_X11 1
 #define CONFIG_XRANDR 0
 #define CONFIG_XV 0
-#define MYTH_BUILD_CONFIG " debug use_hidesyms using_alsa using_oss using_pulse using_pulseoutput using_backend using_bindings_php using_dvb using_frontend using_hdhomerun using_ceton using_hdpvr using_ivtv using_joystick_menu using_libdns_sd using_libfftw3 using_libxml2 using_lirc using_mheg using_opengl using_opengl_video using_qtscript using_qtdbus using_sdl using_taglib using_bindings_php using_mythtranscode using_opengl using_ffmpeg_threads using_mheg using_libxml2"
+#define MYTH_BUILD_CONFIG " debug use_hidesyms using_alsa using_oss using_pulse using_pulseoutput using_backend using_bindings_php using_dvb using_frontend using_hdhomerun using_ceton using_hdpvr using_ivtv using_joystick_menu using_libdns_sd using_libfftw3 using_libxml2 using_lirc using_mheg using_opengl using_opengl_video using_qtwebkit using_qtscript using_qtdbus using_sdl using_taglib using_bindings_php using_mythtranscode using_opengl using_ffmpeg_threads using_mheg using_libxml2"
 
 #endif /* MYTHTV_CONFIG_H */
