diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/programs/mythfrontend/playbackbox.cpp mythtv-master-20190328-gdcb0847-new/mythtv/programs/mythfrontend/playbackbox.cpp
--- mythtv-master-20190328-gdcb0847-old/mythtv/programs/mythfrontend/playbackbox.cpp	2019-03-29 08:41:09.623333334 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/programs/mythfrontend/playbackbox.cpp	2019-03-29 08:41:10.946666668 +0100
@@ -375,8 +375,12 @@
       // Recording Group settings
       m_groupDisplayName(ProgramInfo::i18n("All Programs")),
       m_recGroup("All Programs"),
+      m_liveTvGroupName(tr("Live TV")),
+      m_liveTvGroupLabel(m_liveTvGroupName.toLower()),
       m_watchGroupName(tr("Watch List")),
       m_watchGroupLabel(m_watchGroupName.toLower()),
+      m_unfinishedGroupName(tr("Unfinished")),
+      m_unfinishedGroupLabel(m_unfinishedGroupName.toLower()),
 
       // Other state
       m_programInfoCache(this),
@@ -1607,6 +1611,7 @@
 
     QMap<QString, QString> sortedList;
     QMap<int, QString> searchRule;
+    QMap<uint, ProgramInfo*> unfinished;
 
     m_programInfoCache.Refresh();
 
@@ -1703,10 +1708,8 @@
 
             if (!isLiveTvGroup && isLiveTVProg && (m_viewMask & VIEW_LIVETVGRP))
             {
-                QString tmpTitle = tr("Live TV");
-                sortedList[tmpTitle.toLower()] = tmpTitle;
-                m_progLists[tmpTitle.toLower()].push_front(p);
-                m_progLists[tmpTitle.toLower()].setAutoDelete(false);
+                m_progLists[m_liveTvGroupLabel].push_front(p);
+                m_progLists[m_liveTvGroupLabel].setAutoDelete(false);
                 continue;
             }
 
@@ -1758,13 +1761,18 @@
             {
                 m_watchlist.Add(p);
             }
+
+            if ((m_viewMask & VIEW_UNFINISHED) && p->GetProgressPercent() > 0)
+            {
+                unfinished.insert(p->GetProgressPercent(), p);
+            }
         }
     }
 
     if (sortedList.empty())
     {
-        LOG(VB_GENERAL, LOG_WARNING, LOC + "SortedList is Empty");
-        m_progLists[""];
+        LOG(VB_GENERAL, LOG_WARNING, LOC +
+            "No recordings selected for display (formerly SortedList is Empty)");
         m_titleList << "";
         m_playList.clear();
 
@@ -1842,9 +1850,21 @@
     m_titleList = QStringList("");
     if (!m_progLists[m_watchGroupLabel].empty())
         m_titleList << m_watchGroupName;
-    if ((!m_progLists["livetv"].empty()) &&
-        (!sortedList.values().contains(tr("Live TV"))))
-        m_titleList << tr("Live TV");
+
+    if (!unfinished.isEmpty())
+    {
+        // Most progressed at top
+        foreach (ProgramInfo* p, unfinished)
+            m_progLists[m_unfinishedGroupLabel].push_front(p);
+
+        m_progLists[m_unfinishedGroupLabel].setAutoDelete(false);
+        m_titleList << m_unfinishedGroupName;
+    }
+
+    if (m_progLists[m_liveTvGroupLabel].size() > 0
+            && !sortedList.values().contains(m_liveTvGroupName))
+        m_titleList << m_liveTvGroupName;
+
     m_titleList << sortedList.values();
 
     // Populate list of recording groups
@@ -4842,18 +4862,6 @@
                 m_parentScreen, SLOT(toggleRecGroupView(bool)));
     }
 
-    // TODO Do we need two separate settings to determine whether the watchlist
-    //      is shown? The filter setting be enough?
-        checkBox = dynamic_cast<MythUICheckBox*>(GetChild("watchlist"));
-        if (checkBox)
-        {
-            if (m_viewMask & PlaybackBox::VIEW_WATCHLIST)
-                checkBox->SetCheckState(MythUIStateType::Full);
-            connect(checkBox, SIGNAL(toggled(bool)),
-                    m_parentScreen, SLOT(toggleWatchListView(bool)));
-        }
-    //
-
     checkBox = dynamic_cast<MythUICheckBox*>(GetChild("searches"));
     if (checkBox)
     {
@@ -4863,17 +4871,32 @@
                 m_parentScreen, SLOT(toggleSearchView(bool)));
     }
 
-    // TODO Do we need two separate settings to determine whether livetv
-    //      recordings are shown? Same issue as the watchlist above
-        checkBox = dynamic_cast<MythUICheckBox*>(GetChild("livetv"));
-        if (checkBox)
-        {
-            if (m_viewMask & PlaybackBox::VIEW_LIVETVGRP)
-                checkBox->SetCheckState(MythUIStateType::Full);
-            connect(checkBox, SIGNAL(toggled(bool)),
-                    m_parentScreen, SLOT(toggleLiveTVView(bool)));
-        }
-    //
+    checkBox = dynamic_cast<MythUICheckBox*>(GetChild("watchlist"));
+    if (checkBox)
+    {
+        if (m_viewMask & PlaybackBox::VIEW_WATCHLIST)
+            checkBox->SetCheckState(MythUIStateType::Full);
+        connect(checkBox, SIGNAL(toggled(bool)),
+                m_parentScreen, SLOT(toggleWatchListView(bool)));
+    }
+
+    checkBox = dynamic_cast<MythUICheckBox*>(GetChild("unfinished"));
+    if (checkBox)
+    {
+        if (m_viewMask & PlaybackBox::VIEW_UNFINISHED)
+            checkBox->SetCheckState(MythUIStateType::Full);
+        connect(checkBox, SIGNAL(toggled(bool)),
+                m_parentScreen, SLOT(toggleUnfinishedView(bool)));
+    }
+
+    checkBox = dynamic_cast<MythUICheckBox*>(GetChild("livetv"));
+    if (checkBox)
+    {
+        if (m_viewMask & PlaybackBox::VIEW_LIVETVGRP)
+            checkBox->SetCheckState(MythUIStateType::Full);
+        connect(checkBox, SIGNAL(toggled(bool)),
+                m_parentScreen, SLOT(toggleLiveTVView(bool)));
+    }
 
     checkBox = dynamic_cast<MythUICheckBox*>(GetChild("watched"));
     if (checkBox)
diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/programs/mythfrontend/playbackbox.h mythtv-master-20190328-gdcb0847-new/mythtv/programs/mythfrontend/playbackbox.h
--- mythtv-master-20190328-gdcb0847-old/mythtv/programs/mythfrontend/playbackbox.h	2019-03-29 08:41:09.623333334 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/programs/mythfrontend/playbackbox.h	2019-03-29 08:41:10.946666668 +0100
@@ -151,6 +151,7 @@
         VIEW_WATCHLIST  =  0x0008,
         VIEW_SEARCHES   =  0x0010,
         VIEW_LIVETVGRP  =  0x0020,
+        VIEW_UNFINISHED =  0x0040,
         // insert new entries above here
         VIEW_WATCHED    =  0x8000
     } ViewMask;
@@ -263,6 +264,7 @@
     void toggleCategoryView(bool setOn)  { toggleView(VIEW_CATEGORIES, setOn); }
     void toggleRecGroupView(bool setOn)  { toggleView(VIEW_RECGROUPS, setOn); }
     void toggleWatchListView(bool setOn) { toggleView(VIEW_WATCHLIST, setOn); }
+    void toggleUnfinishedView(bool setOn){ toggleView(VIEW_UNFINISHED, setOn); }
     void toggleSearchView(bool setOn)    { toggleView(VIEW_SEARCHES, setOn); }
     void toggleLiveTVView(bool setOn)    { toggleView(VIEW_LIVETVGRP, setOn); }
     void toggleWatchedView(bool setOn)   { toggleView(VIEW_WATCHED, setOn); }
@@ -435,8 +437,12 @@
     QString             m_recGroup;
     QString             m_curGroupPassword;
     QString             m_newRecGroup;
+    QString             m_liveTvGroupName;
+    QString             m_liveTvGroupLabel;
     QString             m_watchGroupName;
     QString             m_watchGroupLabel;
+    QString             m_unfinishedGroupName;
+    QString             m_unfinishedGroupLabel;
     ViewMask            m_viewMask            {VIEW_TITLES};
 
     // Watchlist support
diff -Naur mythtv-master-20190328-gdcb0847-old/mythtv/themes/MythCenter-wide/recordings-ui.xml mythtv-master-20190328-gdcb0847-new/mythtv/themes/MythCenter-wide/recordings-ui.xml
--- mythtv-master-20190328-gdcb0847-old/mythtv/themes/MythCenter-wide/recordings-ui.xml	2019-03-29 08:41:09.626666667 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythtv/themes/MythCenter-wide/recordings-ui.xml	2019-03-29 08:41:10.946666668 +0100
@@ -526,29 +526,38 @@
         </textarea>
 
         <checkbox name="recgroups" from="basecheckbox">
-            <position>90,190</position>
+            <position>90,180</position>
         </checkbox>
 
         <textarea name="grouplabel" from="titlelabel">
-            <position>140,190</position>
+            <position>140,180</position>
             <value>Show Recording Groups</value>
         </textarea>
 
         <checkbox name="watchlist" from="basecheckbox">
-            <position>90,240</position>
+            <position>90,220</position>
         </checkbox>
 
         <textarea name="watchlabel" from="titlelabel">
-            <position>140,240</position>
+            <position>140,220</position>
             <value>Show Watch List</value>
         </textarea>
 
+        <checkbox name="unfinished" from="basecheckbox">
+            <position>90,260</position>
+        </checkbox>
+
+        <textarea name="unfinishedlabel" from="titlelabel">
+            <position>140,260</position>
+            <value>Show Unfinished</value>
+        </textarea>
+
         <checkbox name="searches" from="basecheckbox">
-            <position>90,290</position>
+            <position>90,300</position>
         </checkbox>
 
         <textarea name="searchlabel" from="titlelabel">
-            <position>140,290</position>
+            <position>140,300</position>
             <value>Show Searches</value>
         </textarea>
 
@@ -562,11 +571,11 @@
         </textarea>
 
         <checkbox name="watched" from="basecheckbox">
-            <position>90,390</position>
+            <position>90,380</position>
         </checkbox>
 
         <textarea name="watchedlabel" from="titlelabel">
-            <position>140,390</position>
+            <position>140,380</position>
             <value>Show Watched Programs</value>
         </textarea>
 
