diff -Naur mythtv-master-20171005-gd630544-old/mythtv/libs/libmythui/mythuihelper.cpp mythtv-master-20171005-gd630544-new/mythtv/libs/libmythui/mythuihelper.cpp
--- mythtv-master-20171005-gd630544-old/mythtv/libs/libmythui/mythuihelper.cpp	2017-10-05 18:10:23.000000000 +0200
+++ mythtv-master-20171005-gd630544-new/mythtv/libs/libmythui/mythuihelper.cpp	2017-10-11 20:54:42.000000000 +0200
@@ -893,39 +893,48 @@
  */
 void MythUIHelper::PruneCacheDir(QString dirname)
 {
-    LOG(VB_GENERAL, LOG_INFO, LOC +
-        QString("Pruning cache directory: %1").arg(dirname));
 
     int days = GetMythDB()->GetNumSetting("UIDiskCacheDays", 7);
-    QDateTime cutoff = MythDate::current().addDays(-days);
-    LOG(VB_GUI | VB_FILE, LOG_INFO, LOC +
-        QString("Removing files not accessed since %1")
-        .arg(cutoff.toLocalTime().toString(Qt::ISODate)));
-
-    int kept = 0, deleted = 0;
-    QDir dir(dirname);
-    dir.setFilter(QDir::Files | QDir::NoSymLinks | QDir::NoDotAndDotDot);
-    foreach (const QFileInfo &fi, dir.entryInfoList())
+    if (days == -1) {
+        LOG(VB_GENERAL, LOG_INFO, LOC +
+            QString("Pruning cache directory: %1 is disabled").arg(dirname));
+
+    }
+    else
     {
-        if (fi.lastRead() >= cutoff) {
-            kept += 1;
+        LOG(VB_GENERAL, LOG_INFO, LOC +
+            QString("Pruning cache directory: %1 for %2 days").arg(dirname).arg(days));
+
+        QDateTime cutoff = MythDate::current().addDays(-days);
+        LOG(VB_GUI | VB_FILE, LOG_INFO, LOC +
+            QString("Removing files not accessed since %1")
+            .arg(cutoff.toLocalTime().toString(Qt::ISODate)));
+
+        int kept = 0, deleted = 0;
+        QDir dir(dirname);
+        dir.setFilter(QDir::Files | QDir::NoSymLinks | QDir::NoDotAndDotDot);
+        foreach (const QFileInfo &fi, dir.entryInfoList())
+        {
+            if (fi.lastRead() >= cutoff) {
+                kept += 1;
+                LOG(VB_GUI | VB_FILE, LOG_DEBUG, LOC +
+                    QString("%1 Keep   %2")
+                    .arg(fi.lastRead().toLocalTime().toString(Qt::ISODate))
+                    .arg(fi.fileName()));
+                continue;
+            }
+            deleted += 1;
             LOG(VB_GUI | VB_FILE, LOG_DEBUG, LOC +
-                QString("%1 Keep   %2")
+                QString("%1 Delete %2")
                 .arg(fi.lastRead().toLocalTime().toString(Qt::ISODate))
                 .arg(fi.fileName()));
-            continue;
+            QFile file(fi.absoluteFilePath());
+            file.remove();
         }
-        deleted += 1;
-        LOG(VB_GUI | VB_FILE, LOG_DEBUG, LOC +
-            QString("%1 Delete %2")
-            .arg(fi.lastRead().toLocalTime().toString(Qt::ISODate))
-            .arg(fi.fileName()));
-        QFile file(fi.absoluteFilePath());
-        file.remove();
-    }
 
-    LOG(VB_GENERAL, LOG_INFO, LOC +
-        QString("Kept %1 files, deleted %2 files").arg(kept).arg(deleted));
+        LOG(VB_GUI | VB_FILE, LOG_INFO, LOC +
+            QString("Kept %1 files, deleted %2 files").arg(kept).arg(deleted));
+    }
 }
 
 void MythUIHelper::GetScreenBounds(int &xbase, int &ybase,
