diff -Naur mythtv-master-20170221-g672a509-old/mythtv/libs/libmythtv/recorders/recorderbase.cpp mythtv-master-20170221-g672a509-new/mythtv/libs/libmythtv/recorders/recorderbase.cpp
--- mythtv-master-20170221-g672a509-old/mythtv/libs/libmythtv/recorders/recorderbase.cpp	2017-02-21 11:28:46.000000000 +0100
+++ mythtv-master-20170221-g672a509-new/mythtv/libs/libmythtv/recorders/recorderbase.cpp	2017-02-21 12:21:29.583333325 +0100
@@ -392,8 +392,6 @@
 
     if (recq && tvrec)
         tvrec->RingBufferChanged(ringBuffer, curRecording, recq);
-
-    ringBufferCheckTimer.restart();
 }
 
 void RecorderBase::SetRecordingStatus(RecStatus::Type status,
@@ -638,18 +636,6 @@
     {
         positionMapLock.unlock();
     }
-
-    // Make sure a ringbuffer switch is checked at least every 10
-    // seconds.  Otherwise, this check is only performed on keyframes,
-    // and if there is a problem with the input we may never see one
-    // again, resulting in a wedged recording.
-    if (ringBufferCheckTimer.isRunning() &&
-        ringBufferCheckTimer.elapsed() > 10000)
-    {
-        LOG(VB_RECORD, LOG_WARNING, LOC + "It is has been over 10 seconds "
-            "since we've checked for a ringbuffer switch.");
-        CheckForRingBufferSwitch();
-    }
 }
 
 void RecorderBase::TryWriteProgStartMark(const frm_pos_map_t &durationDeltaCopy)
diff -Naur mythtv-master-20170221-g672a509-old/mythtv/libs/libmythtv/recorders/recorderbase.h mythtv-master-20170221-g672a509-new/mythtv/libs/libmythtv/recorders/recorderbase.h
--- mythtv-master-20170221-g672a509-old/mythtv/libs/libmythtv/recorders/recorderbase.h	2017-02-21 11:28:46.000000000 +0100
+++ mythtv-master-20170221-g672a509-new/mythtv/libs/libmythtv/recorders/recorderbase.h	2017-02-21 12:21:29.583333325 +0100
@@ -333,7 +333,6 @@
     QMutex         nextRingBufferLock;
     RingBuffer    *nextRingBuffer;
     RecordingInfo *nextRecording;
-    MythTimer      ringBufferCheckTimer;
 
     // Seektable  support
     MarkTypes      positionMapType;
