diff -Naur mythtv-master-20171119-g3393a71-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20171119-g3393a71-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20171119-g3393a71-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2017-11-19 11:38:09.000000000 +0100
+++ mythtv-master-20171119-g3393a71-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2017-11-19 15:17:40.556666655 +0100
@@ -447,10 +447,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetNumSetting("MonitorDrives", 0)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
