diff -Naur mythtv-master-20190320-gb121bce-old/mythplugins/mythmusic/mythmusic/mainvisual.cpp mythtv-master-20190320-gb121bce-new/mythplugins/mythmusic/mythmusic/mainvisual.cpp
--- mythtv-master-20190320-gb121bce-old/mythplugins/mythmusic/mythmusic/mainvisual.cpp	2019-03-20 22:46:24.803333333 +0100
+++ mythtv-master-20190320-gb121bce-new/mythplugins/mythmusic/mythmusic/mainvisual.cpp	2019-03-20 22:46:26.296666667 +0100
@@ -18,6 +18,7 @@
 
 // mythtv
 #include <mythuivideo.h>
+#include <mythuihelper.h>
 
 // mythmusic
 #include "visualize.h"
@@ -59,7 +60,12 @@
     m_updateTimer->stop();
     delete m_updateTimer;
 
-    delete m_vis;
+    if (m_vis)
+    {
+        if (!m_vis->xscreensaverenable())
+            GetMythUI()->DoRestoreScreensaver();
+        delete m_vis;
+    }
 
     while (!m_nodes.empty())
         delete m_nodes.takeLast();
@@ -78,6 +84,11 @@
     }
 }
 
+static bool inline IsVisFullscreen(QSize size)
+{
+    return size.width() >= 512 && size.height() >= 512;
+}
+
 void MainVisual::setVisual(const QString &name)
 {
     m_updateTimer->stop();
@@ -119,9 +130,12 @@
         if (pVisFactory->name() == visName)
         {
             m_vis = pVisFactory->create(this, pluginName);
-            m_vis->resize(m_visualizerVideo->GetArea().size());
+            const QSize size = m_visualizerVideo->GetArea().size();
+            m_vis->resize(size);
             m_fps = m_vis->getDesiredFPS();
             m_samples = m_vis->getDesiredSamples();
+            if (!m_vis->xscreensaverenable() && IsVisFullscreen(size))
+                GetMythUI()->DoDisableScreensaver();
 
             QMutexLocker locker(mutex());
             prepare();
@@ -248,7 +262,13 @@
     m_pixmap.fill(m_visualizerVideo->GetBackgroundColor());
 
     if (m_vis)
+    {
         m_vis->resize(size);
+        if (!m_vis->xscreensaverenable() && IsVisFullscreen(size))
+            GetMythUI()->DoDisableScreensaver();
+        else
+            GetMythUI()->DoRestoreScreensaver();
+    }
 }
 
 void MainVisual::customEvent(QEvent *event)
diff -Naur mythtv-master-20190320-gb121bce-old/mythplugins/mythmusic/mythmusic/visualize.cpp mythtv-master-20190320-gb121bce-new/mythplugins/mythmusic/mythmusic/visualize.cpp
--- mythtv-master-20190320-gb121bce-old/mythplugins/mythmusic/mythmusic/visualize.cpp	2019-03-20 22:46:24.803333333 +0100
+++ mythtv-master-20190320-gb121bce-new/mythplugins/mythmusic/mythmusic/visualize.cpp	2019-03-20 22:46:26.296666667 +0100
@@ -44,8 +44,6 @@
 VisualBase::VisualBase(bool screensaverenable)
     : m_xscreensaverenable(screensaverenable)
 {
-    if (!m_xscreensaverenable)
-        GetMythUI()->DoDisableScreensaver();
 }
 
 VisualBase::~VisualBase()
@@ -55,8 +53,6 @@
     //    that derived classes
     //    can destruct properly
     //
-    if (!m_xscreensaverenable)
-        GetMythUI()->DoRestoreScreensaver();
 }
 
 
diff -Naur mythtv-master-20190320-gb121bce-old/mythplugins/mythmusic/mythmusic/visualize.h mythtv-master-20190320-gb121bce-new/mythplugins/mythmusic/mythmusic/visualize.h
--- mythtv-master-20190320-gb121bce-old/mythplugins/mythmusic/mythmusic/visualize.h	2019-03-20 22:46:24.803333333 +0100
+++ mythtv-master-20190320-gb121bce-new/mythplugins/mythmusic/mythmusic/visualize.h	2019-03-20 22:46:26.296666667 +0100
@@ -93,6 +93,8 @@
     virtual unsigned long getDesiredSamples(void) { return SAMPLES_DEFAULT_SIZE; }
     void drawWarning(QPainter *p, const QColor &back, const QSize &size, QString warning, int fontsize = 28);
 
+    bool xscreensaverenable() const { return m_xscreensaverenable; }
+
   protected:
     int  m_fps                {20};
     bool m_xscreensaverenable {true};
