diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp
--- mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp	2018-10-12 10:53:57.160000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp	2018-10-12 10:53:59.336666666 +0200
@@ -6,6 +6,7 @@
 
 // ANSI C
 #include <cstdlib>
+#include <cstring>
 
 // C++
 #include <iostream>
@@ -44,7 +45,7 @@
 
     vorbis_info_init(&vi);
 
-    ogg_packet_clear(&op);
+    std::memset(&op, 0, sizeof op);
 
     float quality = 1.0;
     if (qualitylevel == 0)
@@ -79,6 +80,7 @@
     ogg_stream_packetin(&os, &header_comments);
     ogg_stream_packetin(&os, &header_codebooks);
 
+    ogg_page og;
     int result;
     while ((result = ogg_stream_flush(&os, &og)))
     {
@@ -97,8 +99,8 @@
 {
     addSamples(nullptr, 0); //flush
     ogg_stream_clear(&os);
-    vorbis_block_clear(&vb);
     vorbis_dsp_clear(&vd);
+    vorbis_block_clear(&vb);
     vorbis_comment_clear(&vc);
     vorbis_info_clear(&vi);
 
@@ -143,6 +145,7 @@
             int eos = 0;
             while (!eos)
             {
+                ogg_page og;
                 int result = ogg_stream_pageout(&os, &og);
                 if (!result)
                     break;
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/vorbisencoder.h mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/vorbisencoder.h
--- mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/vorbisencoder.h	2018-10-12 10:53:57.160000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/vorbisencoder.h	2018-10-12 10:53:59.336666666 +0200
@@ -15,7 +15,6 @@
     int addSamples(int16_t *bytes, unsigned int len);
 
   private:
-    ogg_page og;
     ogg_packet op;
     long packetsdone;
     long bytes_written;
