diff -Naur mythtv-master-20191227-gf15b4f0-old/mythplugins/mythmusic/mythmusic/flacencoder.cpp mythtv-master-20191227-gf15b4f0-new/mythplugins/mythmusic/mythmusic/flacencoder.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythplugins/mythmusic/mythmusic/flacencoder.cpp	2019-12-27 12:19:51.740000000 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythplugins/mythmusic/mythmusic/flacencoder.cpp	2019-12-27 12:19:52.836666667 +0100
@@ -29,6 +29,11 @@
                          MusicMetadata *metadata)
            : Encoder(outfile, qualitylevel, metadata)
 {
+    if (m_out)
+    {
+        fclose(m_out);
+        m_out = NULL;
+    }
     bool streamable_subset = true;
     bool do_mid_side = true;
     bool loose_mid_side = false;
@@ -73,6 +78,8 @@
         LOG(VB_GENERAL, LOG_ERR,
             QString("Error initializing FLAC encoder. Got return code: %1")
                 .arg(ret));
+        encoder_delete(encoder);
+        encoder = NULL;
     }
 
     for (int i = 0; i < NUM_CHANNELS; i++)
@@ -99,6 +106,9 @@
 
 int FlacEncoder::addSamples(int16_t *bytes, unsigned int length)
 {
+    if (!encoder)
+        return EENCODEERROR;
+
     unsigned int index = 0;
 
     length /= sizeof(int16_t);
diff -Naur mythtv-master-20191227-gf15b4f0-old/mythplugins/mythmusic/mythmusic/flacencoder.h mythtv-master-20191227-gf15b4f0-new/mythplugins/mythmusic/mythmusic/flacencoder.h
--- mythtv-master-20191227-gf15b4f0-old/mythplugins/mythmusic/mythmusic/flacencoder.h	2019-12-27 12:19:51.740000000 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythplugins/mythmusic/mythmusic/flacencoder.h	2019-12-27 12:19:52.836666667 +0100
@@ -85,6 +85,7 @@
     FlacEncoder(const QString &outfile, int qualitylevel, MusicMetadata *metadata);
    ~FlacEncoder() override;
     int addSamples(int16_t *bytes, unsigned int len) override; // Encoder
+    virtual bool isValid() { return !!encoder; }
 
   private:
     FLAC_ENCODER *m_encoder    {nullptr};
diff -Naur mythtv-master-20191227-gf15b4f0-old/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp mythtv-master-20191227-gf15b4f0-new/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp
--- mythtv-master-20191227-gf15b4f0-old/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp	2019-12-27 12:19:51.740000000 +0100
+++ mythtv-master-20191227-gf15b4f0-new/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp	2019-12-27 12:19:52.836666667 +0100
@@ -90,6 +90,11 @@
 VorbisEncoder::~VorbisEncoder()
 {
     VorbisEncoder::addSamples(nullptr, 0); //flush
+    if (m_out)
+    {
+        fclose(m_out);
+        m_out = NULL;
+    }
     ogg_stream_clear(&m_os);
     vorbis_block_clear(&m_vb);
     vorbis_dsp_clear(&m_vd);
