diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/flacencoder.cpp mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/flacencoder.cpp
--- mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/flacencoder.cpp	2018-10-12 10:53:58.730000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/flacencoder.cpp	2018-10-12 10:54:01.240000000 +0200
@@ -29,6 +29,12 @@
                          MusicMetadata *metadata)
            : Encoder(outfile, qualitylevel, metadata)
 {
+    if (m_out)
+    {
+        fclose(m_out);
+        m_out = NULL;
+    }
+
     sampleindex = 0;
 
     bool streamable_subset = true;
@@ -75,6 +81,8 @@
         LOG(VB_GENERAL, LOG_ERR,
             QString("Error initializing FLAC encoder. Got return code: %1")
                 .arg(ret));
+        encoder_delete(encoder);
+        encoder = NULL;
     }
 
     for (int i = 0; i < NUM_CHANNELS; i++)
@@ -101,6 +109,9 @@
 
 int FlacEncoder::addSamples(int16_t *bytes, unsigned int length)
 {
+    if (!encoder)
+        return EENCODEERROR;
+
     unsigned int index = 0;
 
     length /= sizeof(int16_t);
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/flacencoder.h mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/flacencoder.h
--- mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/flacencoder.h	2018-10-12 10:53:58.730000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/flacencoder.h	2018-10-12 10:54:01.240000000 +0200
@@ -85,6 +85,7 @@
     FlacEncoder(const QString &outfile, int qualitylevel, MusicMetadata *metadata);
    ~FlacEncoder();
     int addSamples(int16_t *bytes, unsigned int len);
+    virtual bool isValid() { return !!encoder; }
 
   private:
     FLAC_ENCODER *encoder;
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp
--- mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp	2018-10-12 10:53:59.336666666 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp	2018-10-12 10:54:01.240000000 +0200
@@ -98,6 +98,12 @@
 VorbisEncoder::~VorbisEncoder()
 {
     addSamples(nullptr, 0); //flush
+    if (m_out)
+    {
+        fclose(m_out);
+        m_out = NULL;
+    }
+
     ogg_stream_clear(&os);
     vorbis_dsp_clear(&vd);
     vorbis_block_clear(&vb);
