diff -Naur mythtv-master-20190328-gdcb0847-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-master-20190328-gdcb0847-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-master-20190328-gdcb0847-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2019-03-29 08:41:14.833333334 +0100
+++ mythtv-master-20190328-gdcb0847-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2019-03-29 08:41:16.986666667 +0100
@@ -749,11 +749,24 @@
 
 void Ripper::ScanFinished()
 {
+    m_scanThread->wait();
     delete m_scanThread;
     m_scanThread = nullptr;
 
     m_tracks->clear();
 
+    if (m_scanThread)
+    {
+        m_scanThread->wait();
+        delete m_scanThread;
+    }
+
+    if (m_ejectThread)
+    {
+        m_ejectThread->wait();
+        delete m_ejectThread;
+    }
+
     if (m_decoder)
     {
         MusicMetadata *metadata;
@@ -1153,6 +1166,7 @@
 
 void Ripper::EjectFinished()
 {
+    m_ejectThread->wait();
     delete m_ejectThread;
     m_ejectThread = nullptr;
 
