diff -Naur mythtv-master-20200109-gc116014-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-master-20200109-gc116014-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-master-20200109-gc116014-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2020-01-09 22:06:11.440000001 +0100
+++ mythtv-master-20200109-gc116014-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2020-01-09 22:06:13.533333335 +0100
@@ -754,11 +754,24 @@
 
 void Ripper::ScanFinished()
 {
+    m_scanThread->wait();
     delete m_scanThread;
     m_scanThread = nullptr;
 
     m_tracks->clear();
 
+    if (m_scanThread)
+    {
+        m_scanThread->wait();
+        delete m_scanThread;
+    }
+
+    if (m_ejectThread)
+    {
+        m_ejectThread->wait();
+        delete m_ejectThread;
+    }
+
     if (m_decoder)
     {
         bool isCompilation = false;
@@ -1152,6 +1165,7 @@
 
 void Ripper::EjectFinished()
 {
+    m_ejectThread->wait();
     delete m_ejectThread;
     m_ejectThread = nullptr;
 
