diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-master-20181012-g4651002-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2018-10-12 10:54:01.140000000 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2018-10-12 10:54:04.706666666 +0200
@@ -597,6 +597,18 @@
     QString command = "rm -f " + GetConfDir() + "/tmp/RipTemp/*";
     myth_system(command);
 
+    if (m_scanThread)
+    {
+        m_scanThread->wait();
+        delete m_scanThread;
+    }
+
+    if (m_ejectThread)
+    {
+        m_ejectThread->wait();
+        delete m_ejectThread;
+    }
+
     if (m_decoder)
         delete m_decoder;
 
@@ -795,6 +807,7 @@
 
 void Ripper::ScanFinished()
 {
+    m_scanThread->wait();
     delete m_scanThread;
     m_scanThread = nullptr;
 
@@ -1201,6 +1214,7 @@
 
 void Ripper::EjectFinished()
 {
+    m_ejectThread->wait();
     delete m_ejectThread;
     m_ejectThread = nullptr;
 
