diff -Naur mythtv-master-20200409-g0fc7abb082-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-master-20200409-g0fc7abb082-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-master-20200409-g0fc7abb082-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2020-04-09 12:26:39.427586598 +0200
+++ mythtv-master-20200409-g0fc7abb082-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2020-04-09 12:26:41.334253195 +0200
@@ -754,11 +754,24 @@
 
 void Ripper::ScanFinished()
 {
+    m_scanThread->wait();
     delete m_scanThread;
     m_scanThread = nullptr;
 
     m_tracks->clear();
 
+    if (m_scanThread)
+    {
+        m_scanThread->wait();
+        delete m_scanThread;
+    }
+
+    if (m_ejectThread)
+    {
+        m_ejectThread->wait();
+        delete m_ejectThread;
+    }
+
     if (m_decoder)
     {
         bool isCompilation = false;
@@ -1152,6 +1165,7 @@
 
 void Ripper::EjectFinished()
 {
+    m_ejectThread->wait();
     delete m_ejectThread;
     m_ejectThread = nullptr;
 
