diff -Naur mythtv-master-20210424-g69d12b67b6-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-master-20210424-g69d12b67b6-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-master-20210424-g69d12b67b6-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2021-04-24 16:59:05.499999044 +0200
+++ mythtv-master-20210424-g69d12b67b6-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2021-04-24 16:59:10.163332378 +0200
@@ -756,11 +756,24 @@
 
 void Ripper::ScanFinished()
 {
+    m_scanThread->wait();
     delete m_scanThread;
     m_scanThread = nullptr;
 
     m_tracks->clear();
 
+    if (m_scanThread)
+    {
+        m_scanThread->wait();
+        delete m_scanThread;
+    }
+
+    if (m_ejectThread)
+    {
+        m_ejectThread->wait();
+        delete m_ejectThread;
+    }
+
     if (m_decoder)
     {
         bool isCompilation = false;
@@ -1153,6 +1166,7 @@
 
 void Ripper::EjectFinished()
 {
+    m_ejectThread->wait();
     delete m_ejectThread;
     m_ejectThread = nullptr;
 
