diff -Naur mythtv-master-20210225-gc0426e3fea-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-master-20210225-gc0426e3fea-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-master-20210225-gc0426e3fea-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2021-03-04 20:28:43.063332909 +0100
+++ mythtv-master-20210225-gc0426e3fea-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2021-03-04 20:28:50.116666238 +0100
@@ -756,11 +756,24 @@
 
 void Ripper::ScanFinished()
 {
+    m_scanThread->wait();
     delete m_scanThread;
     m_scanThread = nullptr;
 
     m_tracks->clear();
 
+    if (m_scanThread)
+    {
+        m_scanThread->wait();
+        delete m_scanThread;
+    }
+
+    if (m_ejectThread)
+    {
+        m_ejectThread->wait();
+        delete m_ejectThread;
+    }
+
     if (m_decoder)
     {
         bool isCompilation = false;
@@ -1154,6 +1167,7 @@
 
 void Ripper::EjectFinished()
 {
+    m_ejectThread->wait();
     delete m_ejectThread;
     m_ejectThread = nullptr;
 
