diff -Naur mythtv-master-20191212-g4fddad7-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-master-20191212-g4fddad7-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-master-20191212-g4fddad7-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2019-12-12 19:16:54.053333335 +0100
+++ mythtv-master-20191212-g4fddad7-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2019-12-12 19:16:55.190000002 +0100
@@ -750,11 +750,24 @@
 
 void Ripper::ScanFinished()
 {
+    m_scanThread->wait();
     delete m_scanThread;
     m_scanThread = nullptr;
 
     m_tracks->clear();
 
+    if (m_scanThread)
+    {
+        m_scanThread->wait();
+        delete m_scanThread;
+    }
+
+    if (m_ejectThread)
+    {
+        m_ejectThread->wait();
+        delete m_ejectThread;
+    }
+
     if (m_decoder)
     {
         bool isCompilation = false;
@@ -1148,6 +1161,7 @@
 
 void Ripper::EjectFinished()
 {
+    m_ejectThread->wait();
     delete m_ejectThread;
     m_ejectThread = nullptr;
 
