diff -Naur mythtv-master-20190316-g1a44f54-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-master-20190316-g1a44f54-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-master-20190316-g1a44f54-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2019-03-16 10:54:02.850000001 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2019-03-16 10:54:04.760000000 +0100
@@ -552,6 +552,18 @@
     QString command = "rm -f " + GetConfDir() + "/tmp/RipTemp/*";
     myth_system(command);
 
+    if (m_scanThread)
+    {
+        m_scanThread->wait();
+        delete m_scanThread;
+    }
+
+    if (m_ejectThread)
+    {
+        m_ejectThread->wait();
+        delete m_ejectThread;
+    }
+
     if (m_decoder)
         delete m_decoder;
 
@@ -750,6 +762,7 @@
 
 void Ripper::ScanFinished()
 {
+    m_scanThread->wait();
     delete m_scanThread;
     m_scanThread = nullptr;
 
@@ -1156,6 +1169,7 @@
 
 void Ripper::EjectFinished()
 {
+    m_ejectThread->wait();
     delete m_ejectThread;
     m_ejectThread = nullptr;
 
