diff -Naur mythtv-master-20200116-gdd96d68-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-master-20200116-gdd96d68-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-master-20200116-gdd96d68-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2020-01-16 10:17:42.516666668 +0100
+++ mythtv-master-20200116-gdd96d68-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2020-01-16 10:17:43.610000001 +0100
@@ -754,11 +754,24 @@
 
 void Ripper::ScanFinished()
 {
+    m_scanThread->wait();
     delete m_scanThread;
     m_scanThread = nullptr;
 
     m_tracks->clear();
 
+    if (m_scanThread)
+    {
+        m_scanThread->wait();
+        delete m_scanThread;
+    }
+
+    if (m_ejectThread)
+    {
+        m_ejectThread->wait();
+        delete m_ejectThread;
+    }
+
     if (m_decoder)
     {
         bool isCompilation = false;
@@ -1152,6 +1165,7 @@
 
 void Ripper::EjectFinished()
 {
+    m_ejectThread->wait();
     delete m_ejectThread;
     m_ejectThread = nullptr;
 
