diff --speed-large-files --no-dereference --minimal -Naur mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/bumpscope.h mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/bumpscope.h
--- mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/bumpscope.h	2022-06-10 21:56:31.000000000 +0200
+++ mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/bumpscope.h	2022-06-13 16:09:20.824873226 +0200
@@ -2,6 +2,7 @@
 #define BUMPSCOPE
 
 #include "mainvisual.h"
+#include "config.h"
 
 #include <vector>
 
diff --speed-large-files --no-dereference --minimal -Naur mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/decoder.h mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/decoder.h
--- mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/decoder.h	2022-06-10 21:56:31.000000000 +0200
+++ mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/decoder.h	2022-06-13 16:09:59.548206843 +0200
@@ -11,6 +11,7 @@
 #include <QWaitCondition>
 
 // MythTV
+#include "config.h"
 #include <libmythbase/mthread.h>
 #include <libmythbase/mythobservable.h>
 
diff --speed-large-files --no-dereference --minimal -Naur mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/inlines.h mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/inlines.h
--- mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/inlines.h	2022-06-10 21:56:31.000000000 +0200
+++ mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/inlines.h	2022-06-13 16:10:35.284873762 +0200
@@ -7,6 +7,8 @@
 #ifndef INLINES_H
 #define INLINES_H
 
+#include "config.h"
+
 // *fast* convenience functions
 
 static inline void stereo16_from_stereopcm8(short *l,
diff --speed-large-files --no-dereference --minimal -Naur mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/mythgoom.h mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/mythgoom.h
--- mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/mythgoom.h	2022-06-10 21:56:31.000000000 +0200
+++ mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/mythgoom.h	2022-06-13 16:11:05.038207325 +0200
@@ -2,6 +2,7 @@
 #define MYTHGOOM
 
 #include "mainvisual.h"
+#include "config.h"
 
 class Goom : public VisualBase
 {
diff --speed-large-files --no-dereference --minimal -Naur mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/mythmusic.cpp mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/mythmusic.cpp
--- mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/mythmusic.cpp	2022-06-13 16:22:45.768212405 +0200
+++ mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/mythmusic.cpp	2022-06-13 16:11:35.274874211 +0200
@@ -38,6 +38,7 @@
 #include "musicdata.h"
 #include "musicdbcheck.h"
 #include "musicplayer.h"
+#include "config.h"
 #include "playersettings.h"
 #include "playlistcontainer.h"
 #include "playlisteditorview.h"
diff --speed-large-files --no-dereference --minimal -Naur mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/synaesthesia.cpp mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/synaesthesia.cpp
--- mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/synaesthesia.cpp	2022-06-10 21:56:31.000000000 +0200
+++ mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/synaesthesia.cpp	2022-06-13 16:12:07.488207780 +0200
@@ -21,6 +21,7 @@
 // MythTV
 #include <libmythbase/compat.h>
 #include <libmythbase/mythlogging.h>
+#include "config.h"
 
 // MythMusic
 #include "mainvisual.h"
diff --speed-large-files --no-dereference --minimal -Naur mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/synaesthesia.h mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/synaesthesia.h
--- mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/synaesthesia.h	2022-06-10 21:56:31.000000000 +0200
+++ mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/synaesthesia.h	2022-06-13 16:12:33.814874634 +0200
@@ -10,6 +10,7 @@
 // MythMusic
 #include "mainvisual.h"
 #include "polygon.h"
+#include "config.h"
 
 class QImage;
 
diff --speed-large-files --no-dereference --minimal -Naur mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/visualize.h mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/visualize.h
--- mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/visualize.h	2022-06-10 21:56:31.000000000 +0200
+++ mythtv-e2748d6e2b4b293fa10e8d6fa714260c39bee9cd/mythplugins/mythmusic/mythmusic/visualize.h	2022-06-13 16:12:59.151541480 +0200
@@ -23,6 +23,7 @@
 #include <libmyth/visual.h>
 #include <libmythmetadata/musicmetadata.h>
 #include <libmythbase/mythbaseexp.h>
+#include "config.h"
 
 // MythMusic headers
 #include "constants.h"
