diff -Naur mythtv-master-20190320-gb121bce-old/mythplugins/mythweather/mythweather/weatherScreen.cpp mythtv-master-20190320-gb121bce-new/mythplugins/mythweather/mythweather/weatherScreen.cpp
--- mythtv-master-20190320-gb121bce-old/mythplugins/mythweather/mythweather/weatherScreen.cpp	2019-03-20 22:46:29.123333334 +0100
+++ mythtv-master-20190320-gb121bce-new/mythplugins/mythweather/mythweather/weatherScreen.cpp	2019-03-20 22:46:30.456666667 +0100
@@ -100,15 +100,13 @@
 bool WeatherScreen::prepareScreen(bool checkOnly)
 {
     QMap<QString, QString>::iterator itr = m_dataValueMap.begin();
-    while (itr != m_dataValueMap.end())
+    for ( ; itr != m_dataValueMap.end(); ++itr)
     {
         QString name = itr.key();
         MythUIType *widget = GetChild(name);
 
         if (!widget)
         {
-            LOG(VB_GENERAL, LOG_ERR, "Widget not found " + itr.key());
-
             if (name == "copyright")
             {
                 LOG(VB_GENERAL, LOG_WARNING,
@@ -123,14 +121,17 @@
                         .arg(m_name));
                 return false;
             }
-        }
+            else if (name != "updatetime")
+            {
+                LOG(VB_GENERAL, LOG_ERR, "Widget not found " + itr.key());
+            }
 
-        if( !widget || checkOnly )
-        {
-            ++itr;
             continue;
         }
 
+        if (checkOnly)
+            continue;
+
         if (dynamic_cast<MythUIText *>(widget))
         {
             ((MythUIText *) widget)->SetText(itr.value());
@@ -142,7 +143,6 @@
         }
 
         prepareWidget(widget);
-        ++itr;
     }
 
     m_prepared = true;
