diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythweather/mythweather/weatherScreen.cpp mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythweather/mythweather/weatherScreen.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythweather/mythweather/weatherScreen.cpp	2021-05-18 16:13:27.823333270 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythweather/mythweather/weatherScreen.cpp	2021-05-18 16:13:30.243333270 +0200
@@ -96,15 +96,13 @@
 bool WeatherScreen::prepareScreen(bool checkOnly)
 {
     QMap<QString, QString>::iterator itr = m_dataValueMap.begin();
-    while (itr != m_dataValueMap.end())
+    for ( ; itr != m_dataValueMap.end(); ++itr)
     {
         QString name = itr.key();
         MythUIType *widget = GetChild(name);
 
         if (!widget)
         {
-            LOG(VB_GENERAL, LOG_ERR, "Widget not found " + itr.key());
-
             if (name == "copyright")
             {
                 LOG(VB_GENERAL, LOG_WARNING,
@@ -119,13 +117,14 @@
                         .arg(m_name));
                 return false;
             }
+            else if (name != "updatetime")
+            {
+                LOG(VB_GENERAL, LOG_ERR, "Widget not found " + itr.key());
+            }
         }
 
-        if( !widget || checkOnly )
-        {
-            ++itr;
+        if(checkOnly)
             continue;
-        }
 
         if (auto *w2 = dynamic_cast<MythUIText *>(widget))
         {
@@ -138,7 +137,6 @@
         }
 
         prepareWidget(widget);
-        ++itr;
     }
 
     m_prepared = true;
