diff -Naur mythtv-master-20201001-g2a40eb3-old/mythplugins/mythweather/mythweather/weatherScreen.cpp mythtv-master-20201001-g2a40eb3-new/mythplugins/mythweather/mythweather/weatherScreen.cpp
--- mythtv-master-20201001-g2a40eb3-old/mythplugins/mythweather/mythweather/weatherScreen.cpp	2020-10-02 13:03:15.753333333 +0200
+++ mythtv-master-20201001-g2a40eb3-new/mythplugins/mythweather/mythweather/weatherScreen.cpp	2020-10-02 13:03:16.990000000 +0200
@@ -97,15 +97,13 @@
 bool WeatherScreen::prepareScreen(bool checkOnly)
 {
     QMap<QString, QString>::iterator itr = m_dataValueMap.begin();
-    while (itr != m_dataValueMap.end())
+    for ( ; itr != m_dataValueMap.end(); ++itr)
     {
         QString name = itr.key();
         MythUIType *widget = GetChild(name);
 
         if (!widget)
         {
-            LOG(VB_GENERAL, LOG_ERR, "Widget not found " + itr.key());
-
             if (name == "copyright")
             {
                 LOG(VB_GENERAL, LOG_WARNING,
@@ -120,13 +118,14 @@
                         .arg(m_name));
                 return false;
             }
+            else if (name != "updatetime")
+            {
+                LOG(VB_GENERAL, LOG_ERR, "Widget not found " + itr.key());
+            }
         }
 
-        if( !widget || checkOnly )
-        {
-            ++itr;
+        if(checkOnly)
             continue;
-        }
 
         if (auto *w2 = dynamic_cast<MythUIText *>(widget))
         {
@@ -139,7 +138,6 @@
         }
 
         prepareWidget(widget);
-        ++itr;
     }
 
     m_prepared = true;
