diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan_animaps.pl mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan_animaps.pl
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan_animaps.pl	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan_animaps.pl	2018-10-12 10:54:12.793333333 +0200
@@ -22,7 +22,6 @@
         '/usr/local/share/mythtv/mythweather/scripts/ca_envcan';
 
 use Getopt::Std;
-use LWP::Simple;
 use Date::Manip;
 use ENVCANMapSearch;
 use Image::Magick;
@@ -86,8 +85,8 @@
 my $desc    = $results->[0]->{satellite}; 
 
 # Get HTML and find image list
-my $response = get $results->[0]->{animated_url};
-die unless defined $response;
+my $url = $results->[0]->{animated_url};
+my $response = `curl --header "Accept-Language: en" $url 2>/dev/null`;
 
 my @image_list;
 my $size;
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan_maps.pl mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan_maps.pl
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan_maps.pl	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan_maps.pl	2018-10-12 10:54:12.793333333 +0200
@@ -23,7 +23,6 @@
         '/usr/local/share/mythtv/mythweather/scripts/ca_envcan';
 
 use Getopt::Std;
-use LWP::Simple;
 use Date::Manip;
 use ENVCANMapSearch;
 use Image::Magick;
@@ -86,8 +85,8 @@
 my $desc    = $results->[0]->{satellite}; 
 
 # Get HTML and find image list
-my $response = get $results->[0]->{animated_url};
-die unless defined $response;
+my $url = $results->[0]->{animated_url};
+my $response = `curl --header "Accept-Language: en" $url 2>/dev/null`;
 
 my @image_list;
 my $size;
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan.pl mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan.pl
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan.pl	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/ca_envcan/envcan.pl	2018-10-12 10:54:12.796666667 +0200
@@ -17,7 +17,6 @@
         '/usr/share/mythtv/mythweather/scripts/ca_envcan', 
         '/usr/local/share/mythtv/mythweather/scripts/ca_envcan';
 
-use LWP::Simple;
 use Date::Manip;
 use Getopt::Std;
 use ENVCANLocation;
@@ -113,8 +112,8 @@
 # no cache, grab from the web
 if ($getData) {
     my $base_url = 'http://www.weatheroffice.gc.ca/rss/city/';
-    my $response = get $base_url . $loc .'_e.xml';
-    die unless defined $response;
+    my $url = $base_url . $loc .'_e.xml';
+    my $response = `curl --header "Accept-Language: en" $url 2>/dev/null`;
 
     %results = ENVCANParser::doParse($response, @types);
     $results{'station_id'} = $loc;
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/no_yrno/yrnoxml.pl mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/no_yrno/yrnoxml.pl
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/no_yrno/yrnoxml.pl	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/no_yrno/yrnoxml.pl	2018-10-12 10:54:12.796666667 +0200
@@ -6,7 +6,6 @@
 
 use utf8;
 use encoding 'utf8';
-use LWP::UserAgent;
 use Getopt::Std;
 use URI::Escape;
 use XML::XPath;
@@ -246,18 +245,10 @@
         log_print( $logdir, "cached in $cachefile\n" );
     } else {
         log_print( $logdir, "$url\ncaching to $cachefile\n" );
-        my $ua = LWP::UserAgent->new;
-        $ua->timeout(30);
-        $ua->env_proxy;
-        $ua->default_header('Accept-Language' => "en");
-
-        my $response = $ua->get($url);
-        if ( !$response->is_success ) {
-            die $response->status_line;
-        }
+        my $response = `curl --header "Accept-Language: en" $url 2>/dev/null`;
 
         open OF, ">:utf8", $cachefile or die "Can't open $cachefile: $!\n";
-        print OF $response->content;
+        print OF $response;
         close OF;
     }
 
@@ -279,20 +270,10 @@
         log_print( $logdir, "cached in $cachefile\n" );
     } else {
         log_print( $logdir, "$url\ncaching to $cachefile\n" );
-        my $accept = "application/json, text/javascript, */*; q=0.01";
-        my $ua = LWP::UserAgent->new;
-        $ua->timeout(30);
-        $ua->env_proxy;
-        $ua->default_header('Accept' => $accept);
-        $ua->default_header('Accept-Language' => "en");
-
-        my $response = $ua->get($url);
-        if ( !$response->is_success ) {
-            die $response->status_line;
-        }
+        my $response = `curl --header "application/json, text/javascript, */*; q=0.01" $url 2>/dev/null`;
 
         open OF, ">:utf8", $cachefile or die "Can't open $cachefile: $!\n";
-        print OF $response->content;
+        print OF $response;
         close OF;
     }
 
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/uk_metoffice/MetOffCommon.pm mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/uk_metoffice/MetOffCommon.pm
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/uk_metoffice/MetOffCommon.pm	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/uk_metoffice/MetOffCommon.pm	2018-10-12 10:54:12.796666667 +0200
@@ -8,8 +8,6 @@
 
 use utf8;
 use encoding 'utf8';
-use LWP::UserAgent;
-use LWP::Simple;
 use XML::Simple;
 use URI::Escape;
 use POSIX qw(strftime);
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/uk_metoffice/metoffice_fivedayapi.pl mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/uk_metoffice/metoffice_fivedayapi.pl
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/uk_metoffice/metoffice_fivedayapi.pl	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/uk_metoffice/metoffice_fivedayapi.pl	2018-10-12 10:54:12.796666667 +0200
@@ -15,7 +15,6 @@
         '/usr/local/share/mythtv/mythweather/scripts/uk_metoffice';
 
 use XML::Simple;
-use LWP::Simple;
 use Getopt::Std;
 use File::Path;
 use Switch;
@@ -131,8 +130,7 @@
 }
 
 
-my $response = get $url;
-die unless defined $response;
+my $response = `curl --header "Accept-Language: en" $url 2>/dev/null`;
 
 my $xml = XMLin($response);
 
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/uk_metoffice/metoffice_threehourapi.pl mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/uk_metoffice/metoffice_threehourapi.pl
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/uk_metoffice/metoffice_threehourapi.pl	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/uk_metoffice/metoffice_threehourapi.pl	2018-10-12 10:54:12.796666667 +0200
@@ -15,7 +15,6 @@
         '/usr/local/share/mythtv/mythweather/scripts/uk_metoffice';
 
 use XML::Simple;
-use LWP::Simple;
 use Getopt::Std;
 use File::Path;
 use Switch;
@@ -118,8 +117,7 @@
 }
 
 
-my $response = get $url;
-die unless defined $response;
+my $response = `curl --header "Accept-Language: en" $url 2>/dev/null`;
 
 my $xml = XMLin($response);
 
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/us_nws/nws-alert.pl mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/us_nws/nws-alert.pl
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/us_nws/nws-alert.pl	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/us_nws/nws-alert.pl	2018-10-12 10:54:12.796666667 +0200
@@ -15,7 +15,6 @@
 use Date::Manip::TZ;
 use Data::Dumper;
 use Getopt::Std;
-use LWP::Simple;
 
 my $alerts;
 my $currAlert;
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/us_nws/nwsxml.pl mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/us_nws/nwsxml.pl
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/us_nws/nwsxml.pl	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/us_nws/nwsxml.pl	2018-10-12 10:54:12.796666667 +0200
@@ -10,7 +10,6 @@
         '/usr/local/share/mythtv/mythweather/scripts/us_nws';
 
 use XML::Simple;
-use LWP::Simple;
 use Data::Dumper;
 use Getopt::Std;
 use NWSLocation;
@@ -83,8 +82,7 @@
 
 my $base_url = 'http://w1.weather.gov/xml/current_obs/';
 my $url = $base_url . $loc . '.xml';
-my $response = get $url;
-die unless defined $response;
+my $response = `curl --header "Accept-Language: en" $url 2>/dev/null`;
 
 print "copyrightlogo::none\n";
 
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground-animaps.pl mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground-animaps.pl
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground-animaps.pl	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground-animaps.pl	2018-10-12 10:54:12.796666667 +0200
@@ -14,7 +14,6 @@
 
 use utf8;
 use encoding 'utf8';
-use LWP::UserAgent;
 use Getopt::Std;
 use URI::Escape;
 use POSIX qw(strftime);
@@ -160,18 +159,10 @@
         log_print( $logdir, "cached in $cachefile\n" );
     } else {
         log_print( $logdir, "$url\ncaching to $cachefile\n" );
-        my $ua = LWP::UserAgent->new;
-        $ua->timeout(30);
-        $ua->env_proxy;
-        $ua->default_header('Accept-Language' => "en");
-
-        my $response = $ua->get($url);
-        if ( !$response->is_success ) {
-            die $response->status_line;
-        }
+        my $response = `curl --header "Accept-Language: en" $url 2>/dev/null`;
 
         open OF, ">", $cachefile or die "Can't open $cachefile: $!\n";
-        print OF $response->content;
+        print OF $response;
         close OF;
     }
 
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground-maps.pl mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground-maps.pl
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground-maps.pl	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground-maps.pl	2018-10-12 10:54:12.796666667 +0200
@@ -14,7 +14,6 @@
 
 use utf8;
 use encoding 'utf8';
-use LWP::UserAgent;
 use Getopt::Std;
 use URI::Escape;
 use POSIX qw(strftime);
@@ -158,18 +157,10 @@
         log_print( $logdir, "cached in $cachefile\n" );
     } else {
         log_print( $logdir, "$url\ncaching to $cachefile\n" );
-        my $ua = LWP::UserAgent->new;
-        $ua->timeout(30);
-        $ua->env_proxy;
-        $ua->default_header('Accept-Language' => "en");
-
-        my $response = $ua->get($url);
-        if ( !$response->is_success ) {
-            die $response->status_line;
-        }
+        my $response = `curl --header "Accept-Language: en" $url 2>/dev/null`;
 
         open OF, ">", $cachefile or die "Can't open $cachefile: $!\n";
-        print OF $response->content;
+        print OF $response;
         close OF;
     }
 
diff -Naur mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground.pl mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground.pl
--- mythtv-master-20181012-g4651002-old/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground.pl	2018-10-12 10:54:10.223333333 +0200
+++ mythtv-master-20181012-g4651002-new/mythplugins/mythweather/mythweather/scripts/wunderground/wunderground.pl	2018-10-12 10:54:12.796666667 +0200
@@ -6,7 +6,6 @@
 
 use utf8;
 use encoding 'utf8';
-use LWP::UserAgent;
 use Getopt::Std;
 use URI::Escape;
 use POSIX qw(strftime);
@@ -206,18 +205,10 @@
         log_print( $logdir, "cached in $cachefile\n" );
     } else {
         log_print( $logdir, "$url\ncaching to $cachefile\n" );
-        my $ua = LWP::UserAgent->new;
-        $ua->timeout(30);
-        $ua->env_proxy;
-        $ua->default_header('Accept-Language' => "en");
-
-        my $response = $ua->get($url);
-        if ( !$response->is_success ) {
-            die $response->status_line;
-        }
+        my $response = `curl --header "Accept-Language: en" $url 2>/dev/null`;
 
         open OF, ">", $cachefile or die "Can't open $cachefile: $!\n";
-        print OF $response->content;
+        print OF $response;
         close OF;
     }
 
