diff --speed-large-files --no-dereference --minimal -Naur mythtv-6034adfa56366841ba9e62eee7717b0e68b2b633/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-6034adfa56366841ba9e62eee7717b0e68b2b633/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-6034adfa56366841ba9e62eee7717b0e68b2b633/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2022-08-06 22:14:28.000000000 +0200
+++ mythtv-6034adfa56366841ba9e62eee7717b0e68b2b633/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2022-08-15 15:37:10.339997597 +0200
@@ -845,7 +845,7 @@
 
     ADD_STR(outStr, "OK");
 
-    std::string sql("SELECT E.Id, E.Name, M.Id AS MonitorID, M.Name AS MonitorName, E.StartTime,  "
+    std::string sql("SELECT E.Id, E.Name, M.Id AS MonitorID, M.Name AS MonitorName, E.StartDateTime,  "
             "E.Length, M.Width, M.Height, M.DefaultRate, M.DefaultScale "
             "from Events as E inner join Monitors as M on E.MonitorId = M.Id ");
 
@@ -854,13 +854,13 @@
         sql += "WHERE M.Name = '" + monitor + "' ";
 
         if (date != "<ANY>")
-            sql += "AND DATE(E.StartTime) = DATE('" + date + "') ";
+            sql += "AND DATE(E.StartDateTime) = DATE('" + date + "') ";
     }
     else
     {
         if (date != "<ANY>")
         {
-            sql += "WHERE DATE(E.StartTime) = DATE('" + date + "') ";
+            sql += "WHERE DATE(E.StartDateTime) = DATE('" + date + "') ";
 
             if (!includeContinuous)
                 sql += "AND Cause != 'Continuous' ";
@@ -871,9 +871,9 @@
     }
 
     if (oldestFirst)
-        sql += "ORDER BY E.StartTime ASC";
+        sql += "ORDER BY E.StartDateTime ASC";
     else
-        sql += "ORDER BY E.StartTime DESC";
+        sql += "ORDER BY E.StartDateTime DESC";
 
     if (mysql_query(&g_dbConn, sql.c_str()))
     {
@@ -934,16 +934,16 @@
 
     ADD_STR(outStr, "OK");
 
-    std::string sql("SELECT DISTINCT DATE(E.StartTime) "
+    std::string sql("SELECT DISTINCT DATE(E.StartDateTime) "
             "from Events as E inner join Monitors as M on E.MonitorId = M.Id ");
 
     if (monitor != "<ANY>")
         sql += "WHERE M.Name = '" + monitor + "' ";
 
     if (oldestFirst)
-        sql += "ORDER BY E.StartTime ASC";
+        sql += "ORDER BY E.StartDateTime ASC";
     else
-        sql += "ORDER BY E.StartTime DESC";
+        sql += "ORDER BY E.StartDateTime DESC";
 
     if (mysql_query(&g_dbConn, sql.c_str()))
     {
