diff -Naur mythtv-master-20200625-gc3e17e6-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20200625-gc3e17e6-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20200625-gc3e17e6-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2020-06-25 14:43:44.000000000 +0200
+++ mythtv-master-20200625-gc3e17e6-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2020-06-30 20:45:49.120000001 +0200
@@ -42,6 +42,7 @@
 
 #ifdef __APPLE__
 #define MSG_NOSIGNAL 0  // Apple also has SO_NOSIGPIPE?
+#include <array>
 #endif
 
 #include "zmserver.h"
