From d573f80cb621d93fc3dca8a13685876e8f3d45c0 Mon Sep 17 00:00:00 2001
From: Roger Siddons <dizygotheca@ntlworld.com>
Date: Sun, 1 Nov 2015 17:29:57 +0000
Subject: [PATCH] Gallery: Fix hostname is invalid SQL


diff --git a/mythtv/libs/libmythmetadata/imagemanager.cpp b/mythtv/libs/libmythmetadata/imagemanager.cpp
index 9279a84..ac563c2 100644
--- a/mythtv/libs/libmythmetadata/imagemanager.cpp
+++ b/mythtv/libs/libmythmetadata/imagemanager.cpp
@@ -1020,7 +1020,7 @@ ImageDbSg::ImageDbSg() : ImageDb(DB_TABLE)
  \brief Local database constructor
 */
 ImageDbLocal::ImageDbLocal()
-    : ImageDb(DB_TABLE "_" + gCoreContext->GetHostName()),
+    : ImageDb(QString("`%1_%2`").arg(DB_TABLE, gCoreContext->GetHostName())),
       m_DbExists(false)
 {
     // Remove any table leftover from a previous FE crash
-- 
2.1.4

