diff -Naur binutils-2.23.1-old/bfd/configure binutils-2.23.1-new/bfd/configure
--- binutils-2.23.1-old/bfd/configure	2012-11-13 06:17:40.000000000 -0800
+++ binutils-2.23.1-new/bfd/configure	2012-11-14 07:47:25.000000000 -0800
@@ -10322,7 +10322,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.23.1-old/binutils/configure binutils-2.23.1-new/binutils/configure
--- binutils-2.23.1-old/binutils/configure	2012-11-05 08:27:34.000000000 -0800
+++ binutils-2.23.1-new/binutils/configure	2012-11-14 07:47:25.000000000 -0800
@@ -10122,7 +10122,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.23.1-old/gas/configure binutils-2.23.1-new/gas/configure
--- binutils-2.23.1-old/gas/configure	2012-06-17 21:43:06.000000000 -0700
+++ binutils-2.23.1-new/gas/configure	2012-11-14 07:47:25.000000000 -0800
@@ -10105,7 +10105,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.23.1-old/gprof/configure binutils-2.23.1-new/gprof/configure
--- binutils-2.23.1-old/gprof/configure	2012-02-06 05:42:32.000000000 -0800
+++ binutils-2.23.1-new/gprof/configure	2012-11-14 07:47:25.000000000 -0800
@@ -10035,7 +10035,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.23.1-old/ld/configure binutils-2.23.1-new/ld/configure
--- binutils-2.23.1-old/ld/configure	2012-09-04 05:53:47.000000000 -0700
+++ binutils-2.23.1-new/ld/configure	2012-11-14 07:47:25.000000000 -0800
@@ -11095,7 +11095,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.23.1-old/libiberty/Makefile.in binutils-2.23.1-new/libiberty/Makefile.in
--- binutils-2.23.1-old/libiberty/Makefile.in	2012-04-27 11:03:26.000000000 -0700
+++ binutils-2.23.1-new/libiberty/Makefile.in	2012-11-14 07:47:25.000000000 -0800
@@ -352,7 +352,7 @@
 # multilib-specific flags, it's overridden by FLAGS_TO_PASS from the
 # default multilib, so we have to take CFLAGS into account as well,
 # since it will be passed the multilib flags.
-MULTIOSDIR = `$(CC) $(CFLAGS) -print-multi-os-directory`
+MULTIOSDIR = '../lib'
 install_to_libdir: all
 	${mkinstalldirs} $(DESTDIR)$(libdir)/$(MULTIOSDIR)
 	$(INSTALL_DATA) $(TARGETLIB) $(DESTDIR)$(libdir)/$(MULTIOSDIR)/$(TARGETLIB)n
diff -Naur binutils-2.23.1-old/libtool.m4 binutils-2.23.1-new/libtool.m4
--- binutils-2.23.1-old/libtool.m4	2011-02-13 13:00:08.000000000 -0800
+++ binutils-2.23.1-new/libtool.m4	2012-11-14 07:47:25.000000000 -0800
@@ -2043,7 +2043,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.23.1-old/opcodes/configure binutils-2.23.1-new/opcodes/configure
--- binutils-2.23.1-old/opcodes/configure	2012-09-04 07:21:07.000000000 -0700
+++ binutils-2.23.1-new/opcodes/configure	2012-11-14 07:47:25.000000000 -0800
@@ -10053,7 +10053,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
