diff -Naur binutils-2.34-old/bfd/Makefile.in binutils-2.34-new/bfd/Makefile.in
--- binutils-2.34-old/bfd/Makefile.in	2020-02-01 12:49:13.000000000 +0100
+++ binutils-2.34-new/bfd/Makefile.in	2020-03-30 12:55:57.557246772 +0200
@@ -460,7 +460,7 @@
 ACLOCAL_AMFLAGS = -I . -I .. -I ../config
 INCDIR = $(srcdir)/../include
 CSEARCH = -I. -I$(srcdir) -I$(INCDIR)
-SUBDIRS = doc po
+SUBDIRS = 
 bfddocdir = doc
 libbfd_la_LDFLAGS = $(am__append_1) -release `cat libtool-soversion` \
 	@SHARED_LDFLAGS@ $(am__empty)
diff -Naur binutils-2.34-old/binutils/Makefile.in binutils-2.34-new/binutils/Makefile.in
--- binutils-2.34-old/binutils/Makefile.in	2020-02-01 12:50:05.000000000 +0100
+++ binutils-2.34-new/binutils/Makefile.in	2020-03-30 12:56:42.390577403 +0200
@@ -560,7 +560,7 @@
 zlibinc = @zlibinc@
 AUTOMAKE_OPTIONS = dejagnu no-dist foreign subdir-objects
 ACLOCAL_AMFLAGS = -I .. -I ../config -I ../bfd
-SUBDIRS = doc po
+SUBDIRS = 
 tooldir = $(exec_prefix)/$(target_alias)
 
 # Automake 1.10+ disables lex and yacc output file regeneration if
diff -Naur binutils-2.34-old/gprof/Makefile.in binutils-2.34-new/gprof/Makefile.in
--- binutils-2.34-old/gprof/Makefile.in	2020-02-01 12:52:21.000000000 +0100
+++ binutils-2.34-new/gprof/Makefile.in	2020-03-30 12:56:19.080578812 +0200
@@ -440,7 +440,7 @@
 ACLOCAL_AMFLAGS = -I .. -I ../config -I ../bfd
 TEXINFO_TEX = ../texinfo/texinfo.tex
 SUFFIXES = .c .m
-SUBDIRS = po
+SUBDIRS = 
 BASEDIR = $(srcdir)/..
 BFDDIR = $(BASEDIR)/bfd
 INCDIR = $(BASEDIR)/include
