diff -Naur binutils-2.44.old/gprofng/src/Makefile.am binutils-2.44.new/gprofng/src/Makefile.am
--- binutils-2.44.old/gprofng/src/Makefile.am	2025-02-02 01:00:00.000000000 +0100
+++ binutils-2.44.new/gprofng/src/Makefile.am	2025-04-18 19:32:07.184716678 +0200
@@ -180,9 +180,9 @@
 install-exec-local:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)
 	rm -f $(DESTDIR)$(bindir)/gp-{archive,collect-app,display-html,display-src,display-text}
-	ln -s gprofng-archive $(DESTDIR)$(bindir)/gp-archive
-	ln -s gprofng-collect-app $(DESTDIR)$(bindir)/gp-collect-app
-	ln -s gprofng-display-html $(DESTDIR)$(bindir)/gp-display-html
-	ln -s gprofng-display-src $(DESTDIR)$(bindir)/gp-display-src
-	ln -s gprofng-display-text $(DESTDIR)$(bindir)/gp-display-text
+	ln -sf gprofng-archive $(DESTDIR)$(bindir)/gp-archive
+	ln -sf gprofng-collect-app $(DESTDIR)$(bindir)/gp-collect-app
+	ln -sf gprofng-display-html $(DESTDIR)$(bindir)/gp-display-html
+	ln -sf gprofng-display-src $(DESTDIR)$(bindir)/gp-display-src
+	ln -sf gprofng-display-text $(DESTDIR)$(bindir)/gp-display-text
 
diff -Naur binutils-2.44.old/gprofng/src/Makefile.in binutils-2.44.new/gprofng/src/Makefile.in
--- binutils-2.44.old/gprofng/src/Makefile.in	2025-02-02 01:00:00.000000000 +0100
+++ binutils-2.44.new/gprofng/src/Makefile.in	2025-04-18 19:32:20.932869766 +0200
@@ -1120,11 +1120,11 @@
 install-exec-local:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)
 	rm -f $(DESTDIR)$(bindir)/gp-{archive,collect-app,display-html,display-src,display-text}
-	ln -s gprofng-archive $(DESTDIR)$(bindir)/gp-archive
-	ln -s gprofng-collect-app $(DESTDIR)$(bindir)/gp-collect-app
-	ln -s gprofng-display-html $(DESTDIR)$(bindir)/gp-display-html
-	ln -s gprofng-display-src $(DESTDIR)$(bindir)/gp-display-src
-	ln -s gprofng-display-text $(DESTDIR)$(bindir)/gp-display-text
+	ln -sf gprofng-archive $(DESTDIR)$(bindir)/gp-archive
+	ln -sf gprofng-collect-app $(DESTDIR)$(bindir)/gp-collect-app
+	ln -sf gprofng-display-html $(DESTDIR)$(bindir)/gp-display-html
+	ln -sf gprofng-display-src $(DESTDIR)$(bindir)/gp-display-src
+	ln -sf gprofng-display-text $(DESTDIR)$(bindir)/gp-display-text
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
