diff -Naur binutils-2.25.1-old/bfd/configure binutils-2.25.1-new/bfd/configure
--- binutils-2.25.1-old/bfd/configure	2015-07-21 06:42:55.000000000 -0700
+++ binutils-2.25.1-new/bfd/configure	2015-07-25 23:26:39.100000221 -0700
@@ -10339,7 +10339,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.25.1-old/binutils/configure binutils-2.25.1-new/binutils/configure
--- binutils-2.25.1-old/binutils/configure	2015-07-21 06:42:56.000000000 -0700
+++ binutils-2.25.1-new/binutils/configure	2015-07-25 23:26:39.103000221 -0700
@@ -10137,7 +10137,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.25.1-old/gas/configure binutils-2.25.1-new/gas/configure
--- binutils-2.25.1-old/gas/configure	2015-07-21 06:42:55.000000000 -0700
+++ binutils-2.25.1-new/gas/configure	2015-07-25 23:26:39.106000221 -0700
@@ -9885,7 +9885,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.25.1-old/gprof/configure binutils-2.25.1-new/gprof/configure
--- binutils-2.25.1-old/gprof/configure	2015-07-21 06:42:59.000000000 -0700
+++ binutils-2.25.1-new/gprof/configure	2015-07-25 23:26:39.117000221 -0700
@@ -9815,7 +9815,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.25.1-old/ld/configure binutils-2.25.1-new/ld/configure
--- binutils-2.25.1-old/ld/configure	2015-07-21 06:42:58.000000000 -0700
+++ binutils-2.25.1-new/ld/configure	2015-07-25 23:26:39.120000221 -0700
@@ -10630,7 +10630,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.25.1-old/libiberty/Makefile.in binutils-2.25.1-new/libiberty/Makefile.in
--- binutils-2.25.1-old/libiberty/Makefile.in	2014-10-14 00:32:04.000000000 -0700
+++ binutils-2.25.1-new/libiberty/Makefile.in	2015-07-25 23:26:39.121000221 -0700
@@ -361,7 +361,7 @@
 # multilib-specific flags, it's overridden by FLAGS_TO_PASS from the
 # default multilib, so we have to take CFLAGS into account as well,
 # since it will be passed the multilib flags.
-MULTIOSDIR = `$(CC) $(CFLAGS) -print-multi-os-directory`
+MULTIOSDIR = '../lib'
 install_to_libdir: all
 	if test -n "${target_header_dir}"; then \
 		${mkinstalldirs} $(DESTDIR)$(libdir)/$(MULTIOSDIR); \
diff -Naur binutils-2.25.1-old/libtool.m4 binutils-2.25.1-new/libtool.m4
--- binutils-2.25.1-old/libtool.m4	2013-11-04 07:33:40.000000000 -0800
+++ binutils-2.25.1-new/libtool.m4	2015-07-25 23:26:39.122000221 -0700
@@ -2056,7 +2056,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.25.1-old/opcodes/configure binutils-2.25.1-new/opcodes/configure
--- binutils-2.25.1-old/opcodes/configure	2015-07-21 06:42:57.000000000 -0700
+++ binutils-2.25.1-new/opcodes/configure	2015-07-25 23:26:39.124000221 -0700
@@ -10064,7 +10064,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
