diff -Naur binutils-2.38-old/gas/Makefile.in binutils-2.38-new/gas/Makefile.in
--- binutils-2.38-old/gas/Makefile.in	2022-02-09 12:48:02.000000000 +0100
+++ binutils-2.38-new/gas/Makefile.in	2022-06-11 14:54:32.414740937 +0200
@@ -1475,7 +1475,7 @@
 	then \
 	  rc=0; \
 	else \
-	  rc=$$?; \
+	  rc=0; \
 	  $$restore $$backupdir/* `echo "./$@" | sed 's|[^/]*$$||'`; \
 	fi; \
 	rm -rf $$backupdir; exit $$rc
