From 25c41cfea709d1d9e38416c703b2593ec3a63f34 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Sat, 8 Aug 2020 09:46:02 +0000
Subject: [PATCH 21/22] Revert "WIP: h264 field reference"

---
 libavcodec/v4l2_request_h264.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/libavcodec/v4l2_request_h264.c b/libavcodec/v4l2_request_h264.c
index 5b0f21a..d28ed07 100644
--- a/libavcodec/v4l2_request_h264.c
+++ b/libavcodec/v4l2_request_h264.c
@@ -116,8 +116,7 @@ static uint8_t get_dpb_index(struct v4l2_ctrl_h264_decode_params *decode, const
         struct v4l2_h264_dpb_entry *entry = &decode->dpb[i];
         if ((entry->flags & V4L2_H264_DPB_ENTRY_FLAG_VALID) &&
             entry->reference_ts == timestamp)
-            // TODO: signal reference type, possible using top 2 bits
-            return i | ((ref->reference & 3) << 6);
+            return i;
     }
 
     return 0;
-- 
2.7.1

