diff -Naur ifplugd-0.28-old/src/interface.c ifplugd-0.28-new/src/interface.c
--- ifplugd-0.28-old/src/interface.c	2005-01-04 16:25:31.000000000 -0800
+++ ifplugd-0.28-new/src/interface.c	2009-02-28 07:09:49.000000000 -0800
@@ -22,9 +22,9 @@
 #include <config.h>
 #endif
 
+#include <sys/types.h>
 #include <linux/sockios.h>
 #include <linux/if_ether.h>
-#include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <linux/if.h>
