diff -Naur 1.old/main.c 1.new/main.c
--- 1.old/main.c	2019-11-15 10:29:05.000000000 +0100
+++ 1.new/main.c	2020-02-05 10:33:17.399999997 +0100
@@ -825,7 +825,7 @@
 	c->coded_width = frame_width;
         c->get_format = get_format;
 
-	av_dict_set(&opts, "num_capture_buffers", "16", 0);
+	av_dict_set(&opts, "num_capture_buffers", "20", 0);
 
 	/* open it */
 	if (avcodec_open2(c, codec, &opts) < 0) {
