From 8ac55e8f8a6b108f6a661a0faa62064fd82676ba Mon Sep 17 00:00:00 2001
From: Robert Siebert <rsiebert@miroku.no-ip.com>
Date: Wed, 20 Mar 2013 11:13:12 +0100
Subject: [PATCH 045/760] Made the new image service API known to the backend.

Signed-off-by: Stuart Morgan <smorgan@mythtv.org>
---
 mythtv/programs/mythbackend/mediaserver.cpp |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mythtv/programs/mythbackend/mediaserver.cpp b/mythtv/programs/mythbackend/mediaserver.cpp
index 19f49f0..357a61f 100644
--- a/mythtv/programs/mythbackend/mediaserver.cpp
+++ b/mythtv/programs/mythbackend/mediaserver.cpp
@@ -27,6 +27,7 @@
 #include "serviceHosts/channelServiceHost.h"
 #include "serviceHosts/videoServiceHost.h"
 #include "serviceHosts/captureServiceHost.h"
+#include "serviceHosts/imageServiceHost.h"
 
 #ifdef USING_LIBDNS_SD
 #include "bonjourregister.h"
@@ -122,6 +123,7 @@ void MediaServer::Init(bool bIsMaster, bool bDisableUPnp /* = false */)
     m_pHttpServer->RegisterExtension( new ChannelServiceHost( m_sSharePath ));
     m_pHttpServer->RegisterExtension( new VideoServiceHost  ( m_sSharePath ));
     m_pHttpServer->RegisterExtension( new CaptureServiceHost( m_sSharePath ));
+    m_pHttpServer->RegisterExtension( new ImageServiceHost  ( m_sSharePath ));
 
     QString sIP = g_pConfig->GetValue( "BackendServerIP"  , ""   );
     if (sIP.isEmpty())
@@ -158,6 +160,8 @@ void MediaServer::Init(bool bIsMaster, bool bDisableUPnp /* = false */)
          pEngine->scriptValueFromQMetaObject< ScriptableVideo   >() );
      pEngine->globalObject().setProperty("Capture"  ,
          pEngine->scriptValueFromQMetaObject< ScriptableCapture   >() );
+     pEngine->globalObject().setProperty("Image"  ,
+         pEngine->scriptValueFromQMetaObject< Image             >() );
 
     // ------------------------------------------------------------------
 
-- 
1.7.10.2

