diff -Naur mythtv-0.27-20140303-gcb3b784-old/mythtv/libs/libmythmetadata/libmythmetadata.pro mythtv-0.27-20140303-gcb3b784-new/mythtv/libs/libmythmetadata/libmythmetadata.pro
--- mythtv-0.27-20140303-gcb3b784-old/mythtv/libs/libmythmetadata/libmythmetadata.pro	2014-03-04 18:53:16.000000000 +0100
+++ mythtv-0.27-20140303-gcb3b784-new/mythtv/libs/libmythmetadata/libmythmetadata.pro	2014-03-05 13:22:32.773877453 +0100
@@ -25,6 +25,7 @@
 HEADERS += musicmetadata.h musicutils.h metaio.h metaiotaglib.h
 HEADERS += metaioflacvorbis.h metaioavfcomment.h metaiomp4.h
 HEADERS += metaiowavpack.h metaioid3.h metaiooggvorbis.h musicfilescanner.h
+HEADERS += imagemetadata.h imageutils.h imagescan.h imagescanthread.h
 
 SOURCES += cleanup.cpp  dbaccess.cpp  dirscan.cpp  globals.cpp
 SOURCES += parentalcontrols.cpp  videoscan.cpp  videoutils.cpp
@@ -35,6 +36,7 @@
 SOURCES += musicmetadata.cpp musicutils.cpp metaio.cpp metaiotaglib.cpp
 SOURCES += metaioflacvorbis.cpp metaioavfcomment.cpp metaiomp4.cpp
 SOURCES += metaiowavpack.cpp metaioid3.cpp metaiooggvorbis.cpp musicfilescanner.cpp
+SOURCES += imagemetadata.cpp imageutils.cpp imagescan.cpp imagescanthread.cpp
 
 INCLUDEPATH += ../libmythbase ../libmythtv
 INCLUDEPATH += ../.. ../ ./ ../libmythupnp ../libmythui
@@ -86,6 +88,7 @@
 inc.files += metaio.h metaiotaglib.h
 inc.files += metaioflacvorbis.h metaioavfcomment.h metaiomp4.h
 inc.files += metaiowavpack.h metaioid3.h metaiooggvorbis.h musicfilescanner.h
+inc.files += imagemetadata.h imageutils.h imagescan.h imagescanthread.h
 
 INSTALLS += inc
 
@@ -99,4 +102,4 @@
 
 include ( ../libs-targetfix.pro )
 
-LIBS += $$EXTRA_LIBS $$LATE_LIBS
+LIBS += $$EXTRA_LIBS $$LATE_LIBS -lexiv2
