From 52508e1388019de86a9f03dde3261977929e903c Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Thu, 19 Sep 2013 20:51:55 +0100
Subject: [PATCH 062/760] Fix a manual merge mistake

---
 mythtv/configure |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mythtv/configure b/mythtv/configure
index 5d208e9..430b6ab 100755
--- a/mythtv/configure
+++ b/mythtv/configure
@@ -5382,8 +5382,8 @@ EOF
 enabled lamemp3 && check_lib2 lame/lame.h lame_init -lmp3lame -lm ||
     die "ERROR! You must have the Lame MP3 encoding library installed to compile MythTV."
 
-+enabled exiv2 && $(pkg-config --exists exiv2) ||
-+    die "ERROR! You must have the Exiv2 image tag reader library installed to compile MythTV."
+enabled exiv2 && $(pkg-config --exists exiv2) ||
+    die "ERROR! You must have the Exiv2 image tag reader library installed to compile MythTV."
     
 # Qt dependency checks
 if test $target_os = mingw32 ; then
-- 
1.7.10.2

