From d1d655fe812a8d71a715e66f172ad2df8151ff0c Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Wed, 9 Oct 2013 20:01:49 +0100
Subject: [PATCH 160/760] Fix mask image scaling after dfd97aa0

---
 mythtv/libs/libmythui/mythuiimage.cpp |   12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/mythtv/libs/libmythui/mythuiimage.cpp b/mythtv/libs/libmythui/mythuiimage.cpp
index dbe5d2a..3dc5c53 100644
--- a/mythtv/libs/libmythui/mythuiimage.cpp
+++ b/mythtv/libs/libmythui/mythuiimage.cpp
@@ -1436,7 +1436,19 @@ bool MythUIImage::ParseElement(
 
         MythImage *newMaskImage = GetPainter()->GetFormatImage();
         if (newMaskImage->Load(maskfile))
+        {
+            float wmult; // Width multipler
+            float hmult; // Height multipler
+            GetMythUI()->GetScreenSettings(wmult, hmult);
+            if (wmult != 1.0f || hmult != 1.0f)
+            {
+                int width = newMaskImage->size().width() * wmult;
+                int height = newMaskImage->size().height() * hmult;
+                newMaskImage->Resize(QSize(width, height));
+            }
+
             m_imageProperties.SetMaskImage(newMaskImage);
+        }
         else
             m_imageProperties.SetMaskImage(NULL);
         newMaskImage->DecrRef();
-- 
1.7.10.2

