diff -Naur mythtv-0.27-build-old/mythtv/bindings/perl/MythTV.pm mythtv-0.27-build-new/mythtv/bindings/perl/MythTV.pm
--- mythtv-0.27-build-old/mythtv/bindings/perl/MythTV.pm	2014-03-08 21:18:31.000000000 +0100
+++ mythtv-0.27-build-new/mythtv/bindings/perl/MythTV.pm	2014-03-08 21:22:02.654147533 +0100
@@ -376,11 +376,11 @@
             or die "Can't set timezone: $!\n\n";
 
     # Check for supported schema version
-        $self->{'schema_version'} = $self->backend_setting('DBSchemaVer');
-        if( $self->{'schema_version'} != $SCHEMA_VERSION ) {
-            die "Database schema $self->{'schema_version'} not supported.\n" .
-                "Bindings support schema version $SCHEMA_VERSION\n";
-        }
+#        $self->{'schema_version'} = $self->backend_setting('DBSchemaVer');
+#        if( $self->{'schema_version'} != $SCHEMA_VERSION ) {
+#            die "Database schema $self->{'schema_version'} not supported.\n" .
+#                "Bindings support schema version $SCHEMA_VERSION\n";
+#        }
 
     # Load the master host and port
         $self->{'master_host'} = $self->backend_setting('MasterServerIP');
