diff --speed-large-files --no-dereference --minimal -Naur u-boot-eed05a18e50492ec20b16f97ace8d5d9606bb9f9/drivers/fastboot/Kconfig u-boot-eed05a18e50492ec20b16f97ace8d5d9606bb9f9/drivers/fastboot/Kconfig
--- u-boot-eed05a18e50492ec20b16f97ace8d5d9606bb9f9/drivers/fastboot/Kconfig	2024-01-04 07:14:35.000000000 +0100
+++ u-boot-eed05a18e50492ec20b16f97ace8d5d9606bb9f9/drivers/fastboot/Kconfig	2024-01-19 09:35:55.513718815 +0100
@@ -55,6 +55,8 @@
 	default 0x800800 if ROCKCHIP_RK3288 || ROCKCHIP_RK3329 || \
 				ROCKCHIP_RK3399
 	default 0x280000 if ROCKCHIP_RK3368
+	default 0xc00800 if ROCKCHIP_RK3566 || ROCKCHIP_RK3568 || \
+				ROCKCHIP_RK3588
 	default 0x100000 if ARCH_ZYNQMP
 	default 0 if SANDBOX
 	help
