diff -Naur qt-everywhere-src-5.11.1-old/qtlocation/src/plugins/geoservices/geoservices.pro qt-everywhere-src-5.11.1-new/qtlocation/src/plugins/geoservices/geoservices.pro
--- qt-everywhere-src-5.11.1-old/qtlocation/src/plugins/geoservices/geoservices.pro	2018-06-12 20:44:12.000000000 +0200
+++ qt-everywhere-src-5.11.1-new/qtlocation/src/plugins/geoservices/geoservices.pro	2018-09-03 17:13:05.319999972 +0200
@@ -8,11 +8,11 @@
 qtConfig(geoservices_itemsoverlay): SUBDIRS += itemsoverlay
 qtConfig(geoservices_osm): SUBDIRS += osm
 
-qtConfig(geoservices_mapboxgl) {
-    !exists(../../3rdparty/mapbox-gl-native/mapbox-gl-native.pro) {
-        warning("Submodule mapbox-gl-native does not exist. Run 'git submodule update --init' on qtlocation.")
-    } else {
-        SUBDIRS += mapboxgl ../../3rdparty/mapbox-gl-native
-        mapboxgl.depends = ../../3rdparty/mapbox-gl-native
-    }
-}
+#qtConfig(geoservices_mapboxgl) {
+#    !exists(../../3rdparty/mapbox-gl-native/mapbox-gl-native.pro) {
+#        warning("Submodule mapbox-gl-native does not exist. Run 'git submodule update --init' on qtlocation.")
+#    } else {
+#        SUBDIRS += mapboxgl ../../3rdparty/mapbox-gl-native
+#        mapboxgl.depends = ../../3rdparty/mapbox-gl-native
+#    }
+#}
