diff -Naur gperf-3.0.4-old/lib/configure.ac gperf-3.0.4-new/lib/configure.ac
--- gperf-3.0.4-old/lib/configure.ac	2009-01-14 16:24:31.000000000 -0800
+++ gperf-3.0.4-new/lib/configure.ac	2010-06-18 14:49:58.000000000 -0700
@@ -35,6 +35,8 @@
                       dnl sets variable CXXCPP
 AC_PROG_RANLIB
                       dnl sets variable RANLIB
+AC_CHECK_TOOL(AR, ar, false)
+                      dnl sets varialbe AR
 CL_PROG_INSTALL
                       dnl sets variables INSTALL, INSTALL_DATA, INSTALL_PROGRAM
 dnl
diff -Naur gperf-3.0.4-old/lib/Makefile.in gperf-3.0.4-new/lib/Makefile.in
--- gperf-3.0.4-old/lib/Makefile.in	2008-08-23 11:52:48.000000000 -0700
+++ gperf-3.0.4-new/lib/Makefile.in	2010-06-18 14:50:54.000000000 -0700
@@ -41,7 +41,7 @@
 # Both C and C++ compiler
 OBJEXT = @OBJEXT@
 # Other
-AR = ar
+AR = @AR@
 AR_FLAGS = rc
 RANLIB = @RANLIB@
 MV = mv
