diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythupnp/ssdp.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythupnp/ssdp.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythupnp/ssdp.cpp	2021-05-18 16:11:31.379999938 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythupnp/ssdp.cpp	2021-05-18 16:11:34.569999938 +0200
@@ -155,21 +155,21 @@
         // ------------------------------------------------------------------
 
         LOG(VB_UPNP, LOG_INFO,
-            "SSDP::EnableNotifications() - sending NTS_byebye");
-        m_pNotifyTask->SetNTS( NTS_byebye );
-        m_pNotifyTask->Execute( nullptr );
-
-        m_bAnnouncementsEnabled = true;
+            "SSDP::EnableNotifications() - sending NTS_byebye2");
+        m_pNotifyTask->SetNTS( NTS_byebye2 );
+    }
+    else
+    {
+        LOG(VB_UPNP, LOG_INFO,
+            "SSDP::EnableNotifications() - sending NTS_alive");
+        m_pNotifyTask->SetNTS( NTS_alive );
     }
 
+    m_bAnnouncementsEnabled = true;
+
     // ------------------------------------------------------------------
     // Add Announcement Task to the Queue
     // ------------------------------------------------------------------
-
-    LOG(VB_UPNP, LOG_INFO, "SSDP::EnableNotifications() - sending NTS_alive");
-
-    m_pNotifyTask->SetNTS( NTS_alive );
-
     TaskQueue::Instance()->AddTask(m_pNotifyTask);
 
     LOG(VB_UPNP, LOG_INFO,
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythupnp/upnptasknotify.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythupnp/upnptasknotify.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythupnp/upnptasknotify.cpp	2021-05-18 16:11:31.379999938 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythupnp/upnptasknotify.cpp	2021-05-18 16:11:34.586666605 +0200
@@ -165,6 +167,11 @@
 
     if (m_eNTS == NTS_alive) 
         pQueue->AddTask( (m_nMaxAge / 2), (Task *)this  );
+    else if (m_eNTS == NTS_byebye2)
+    {
+        m_eNTS = NTS_alive;
+        pQueue->AddTask( this );
+    }
 
     m_mutex.unlock();
 
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythupnp/upnptasknotify.h mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythupnp/upnptasknotify.h
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythupnp/upnptasknotify.h	2021-05-18 16:11:31.379999938 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythupnp/upnptasknotify.h	2021-05-18 16:11:34.586666605 +0200
@@ -38,7 +38,8 @@
 enum UPnpNotifyNTS
 {
     NTS_alive   = 0,
-    NTS_byebye  = 1
+    NTS_byebye  = 1,
+    NTS_byebye2
 };
 
 /////////////////////////////////////////////////////////////////////////////
@@ -89,6 +90,7 @@
             {
                 case NTS_alive : return( "ssdp:alive"  );
                 case NTS_byebye: return( "ssdp:byebye" );
+                case NTS_byebye2: return( "ssdp:byebye" );
             }
             return( "unknown" );
         }
