diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2021-05-18 16:12:37.406666604 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2021-05-18 16:12:40.566666604 +0200
@@ -779,6 +779,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
