diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythbase/mythmedia.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythbase/mythmedia.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythbase/mythmedia.cpp	2021-05-18 16:12:41.953333271 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythbase/mythmedia.cpp	2021-05-18 16:12:45.056666604 +0200
@@ -135,8 +135,16 @@
         // Is there a better way to do this?
         if (QFile(PATHTO_PMOUNT).exists() && QFile(PATHTO_PUMOUNT).exists())
         {
-            MountCommand = QString("%1 %2")
-                .arg((DoMount) ? PATHTO_PMOUNT : PATHTO_PUMOUNT, m_devicePath);
+            if(DoMount)
+            {
+                MountCommand = QString("%1 %2 /media/disk")
+                    .arg((DoMount) ? PATHTO_PMOUNT : PATHTO_PUMOUNT, m_devicePath);
+            }
+            else
+            {
+                MountCommand = QString("%1 %2")
+                    .arg((DoMount) ? PATHTO_PMOUNT : PATHTO_PUMOUNT, m_devicePath);
+            }
         }
         else
         {
@@ -144,12 +152,12 @@
                 .arg((DoMount) ? PATHTO_MOUNT : PATHTO_UNMOUNT, m_devicePath);
         }
 
-        LOG(VB_MEDIA, LOG_INFO, QString("Executing '%1'").arg(MountCommand));
+        LOG(VB_GENERAL, LOG_INFO, QString("Executing '%1'").arg(MountCommand));
         int ret = myth_system(MountCommand, kMSDontBlockInputDevs);
         if (ret !=  GENERIC_EXIT_OK)
         {
             usleep(300000);
-            LOG(VB_MEDIA, LOG_INFO, QString("Retrying '%1'").arg(MountCommand));
+            LOG(VB_GENERAL, LOG_INFO, QString("Retrying '%1'").arg(MountCommand));
             ret = myth_system(MountCommand, kMSDontBlockInputDevs);
         }
         if (ret == GENERIC_EXIT_OK)
@@ -171,7 +179,7 @@
                 }
                 if (!findMountPath())
                 {
-                    LOG(VB_MEDIA, LOG_ERR, "performMountCmd() attempted to"
+                    LOG(VB_GENERAL, LOG_ERR, "performMountCmd() attempted to"
                                            " find mounted media, but failed?");
                     return false;
                 }
