diff -Naur ratpoison-1.4.8-old/src/main.c ratpoison-1.4.8-new/src/main.c
--- ratpoison-1.4.8-old/src/main.c	2014-06-14 16:38:35.000000000 +0200
+++ ratpoison-1.4.8-new/src/main.c	2017-01-26 21:18:11.036666662 +0100
@@ -620,8 +620,8 @@
       if (!XSetLocaleModifiers (""))
 	PRINT_ERROR (("Couldn't set X locale modifiers.\n"));
     }
-  else
-    PRINT_ERROR (("X doesn't seem to support your locale.\n"));
+//  else
+//    PRINT_ERROR (("X doesn't seem to support your locale.\n"));
 
   /* Parse the arguments */
   myargv = argv;
