diff --speed-large-files --no-dereference --minimal -Naur monitorix-3.15.0/monitorix.conf monitorix-3.15.0/monitorix.conf
--- monitorix-3.15.0/monitorix.conf	2022-12-01 08:22:02.000000000 +0100
+++ monitorix-3.15.0/monitorix.conf	2022-12-09 15:02:04.876663649 +0100
@@ -22,18 +22,18 @@
 enable_rrd_lock = n
 include_dir = /etc/monitorix/conf.d
 
-base_dir = /var/lib/monitorix/www/
+base_dir = /srv/www/monitorix/
 base_lib = /var/lib/monitorix/
 base_url = /monitorix
-base_cgi = /monitorix-cgi
+base_cgi = /monitorix/cgi
 
 <httpd_builtin>
-	enabled = y
+	enabled = n
 	host =
 	port = 8080
 	user = nobody
 	group = nobody
-	log_file = /var/log/monitorix-httpd
+	log_file = /var/log/monitorix-httpd.log
 	hosts_deny =
 	hosts_allow =
 	autocheck_responsiveness = y
@@ -48,7 +48,7 @@
 
 # Log files pathnames
 # -----------------------------------------------------------------------------
-log_file	= /var/log/monitorix
+log_file	= /var/log/monitorix.log
 secure_log	= /var/log/secure
 mail_log	= /var/log/maillog
 milter_gl	= /var/milter-greylist/greylist.db
@@ -84,7 +84,7 @@
 	intelrapl	= n
 	hptemp		= n
 	lmsens		= n
-	gensens		= n
+	gensens		= y
 	ipmi		= n
 	ambsens		= n
 	amdgpu		= n
@@ -94,17 +94,17 @@
 	nvme		= n
 	fs		= y
 	zfs		= n
-	du		= n
+	du		= y
 	net		= y
-	netstat		= y
+	netstat		= n
 	tinyproxy	= n
 	tc		= n
 	libvirt		= n
-	process		= n
-	serv		= y
+	process		= y
+	serv		= n
 	mail		= n
-	port		= y
-	user		= y
+	port		= n
+	user		= n
 	ftp		= n
 	apache		= n
 	nginx		= n
@@ -131,7 +131,7 @@
 	apcupsd		= n
 	nut		= n
 	wowza		= n
-	int		= y
+	int		= n
 	verlihub	= n
 </graph_enable>
 
@@ -295,31 +295,31 @@
 	<list>
 		0 = temp0
 		1 = cpu0
-		2 = bat0
+		#@@GPU_GENSYS_ENABLE@@
 	</list>
 	<title>
-		0 = Temperatures
-		1 = CPU frequency
-		2 = Battery status
+		0 = CPU Temp
+		1 = CPU Clock
+		2 = GPU Clock
 	</title>
 	<desc>
 		temp0 = /sys/devices/virtual/thermal/thermal_zone0/temp
 		cpu0 = /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq
-		bat0 = /sys/class/power_supply/BAT0/capacity
+		#@@GPU_FREQ_DESC@@
 	</desc>
 	<unit>
 		temp0 = 1000
 		cpu0 = 0.001
-		bat0 = 1
+		gpu0 = 1
 	</unit>
 	<map>
 		temp0 = Temperature Zone 0
-		cpu0 = CPU0 frequency
-		bat0 = Battery 0
+		cpu0 = CPU Clock
+		gpu0 = GPU Clock
 	</map>
 	<alerts>
 	</alerts>
-	rigid = 0, 0, 2
+	rigid = 0, 0, 0
 	limit = 100, 100, 100
 </gensens>
 
@@ -455,7 +455,7 @@
 # -----------------------------------------------------------------------------
 <disk>
 	<list>
-		0 = /dev/sda, /dev/sdb, /dev/sdc
+		0 = /home/minimyth, /var/log, /var/tmp
 	</list>
 	<desc>
 	</desc>
@@ -498,7 +498,7 @@
 # -----------------------------------------------------------------------------
 <fs>
 	<list>
-		0 = /, swap, /boot
+		0 = swap
 	</list>
 	<desc>
 	</desc>
@@ -526,12 +526,12 @@
 <du>
 	list = System, Users
 	<desc>
-		0 = /var/spool/mail, /var/spool/mqueue, /etc, /var/ftp, /tmp
-		1 = /home/ace, /home/gene, /home/paul, /home/peter
+		0 = /etc, /var/log, /var/tmp
+		1 = /home/minimyth
 	</desc>
 	<type>
 		0 = size
-		1 = files
+		1 = size
 	</type>
 	<dirmap>
 		/var/spool/mail = Mail boxes
@@ -547,9 +547,10 @@
 # -----------------------------------------------------------------------------
 <net>
 	max = 10
-	list = eth0
+	list = eth0, wlan0
 	<desc>
-		eth0 = FastEthernet LAN, 0, 10000000
+		eth0 = LAN, 0, 1000000
+		wlan0 = WiFi, 0, 1000000
 	</desc>
 	gateway = eth0
 </net>
@@ -611,12 +612,14 @@
 # -----------------------------------------------------------------------------
 <process>
 	<list>
-		0 = httpd, sshd, ntpd, mysqld, proftpd, clamd, imap, sendmail, named, smbd
+		0 = mythfrontend, Xorg, weston, lighttpd, monitorix
 	</list>
 	<desc>
-		httpd = Apache
-		imap = Dovecot
-		named = Bind
+		mythfrontend = MythTV frontend
+		Xorg = Xorg server
+		weston = Weston server
+		lighttpd = HTTP server
+		monitorix = monitorix
 	</desc>
 	rigid = 2, 0, 0, 0, 0, 0, 0, 0
 	limit = 100, 1000, 1000, 1000, 1000, 1000, 1000, 1000
