diff -Naur libg15-1.2.7-old/configure.in libg15-1.2.7-new/configure.in
--- libg15-1.2.7-old/configure.in	2008-11-24 19:49:10.000000000 -0800
+++ libg15-1.2.7-new/configure.in	2010-08-27 09:50:07.000000000 -0700
@@ -34,7 +34,7 @@
 AC_CHECK_LIB([usb], [main], ,AC_MSG_ERROR(["libusb not found. please install it"]))
 
 dnl Some versions of libusb do not honour timeout and block. Suns' version is one of them.
-if test "x$libusb_blocks" == "xtrue"; then
+if test "x$libusb_blocks" = "xtrue"; then
       AC_DEFINE([LIBUSB_BLOCKS], [1],[Define if libusb implementation blocks on read or write])
 fi
 
