diff -Naur linux-5.9-rc3-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts linux-5.9-rc3-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
--- linux-5.9-rc3-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2020-09-28 15:23:34.273333332 +0200
+++ linux-5.9-rc3-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2020-09-28 15:09:07.809999976 +0200
@@ -145,6 +145,7 @@
 	vqmmc-supply = <&reg_bldo2>;
 	non-removable;
 	cap-mmc-hw-reset;
+	mmc-hs200-1_8v;
 	bus-width = <8>;
 	status = "okay";
 };
diff -Naur linux-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts linux-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
--- linux-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2021-06-17 13:40:32.566645346 +0200
+++ linux-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2021-06-17 16:18:10.359978561 +0200
@@ -260,7 +260,7 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-ramp-delay = <2500>;
 				regulator-name = "vdd-cpu";
 			};
