diff -Naur linux-old/arch/arm64/boot/dts/rockchip/rk3399.dtsi linux-new/arch/arm64/boot/dts/rockchip/rk3399.dtsi
--- linux-old/arch/arm64/boot/dts/rockchip/rk3399.dtsi	2021-04-12 19:24:43.003331931 +0200
+++ linux-new/arch/arm64/boot/dts/rockchip/rk3399.dtsi	2021-04-12 19:29:18.379998611 +0200
@@ -1294,6 +1294,7 @@
 		clock-names = "aclk", "iface";
 		power-domains = <&power RK3399_PD_VDU>;
 		#iommu-cells = <0>;
+		rockchip,disable-mmu-reset;
 	};
 
 	iep: iep@ff670000 {
--- linux-old/arch/arm64/boot/dts/rockchip/rk3328.dtsi	2021-04-12 14:00:05.626665491 +0200
+++ linux-new/arch/arm64/boot/dts/rockchip/rk3328.dtsi	2021-04-12 10:13:48.353332277 +0200
@@ -750,6 +731,7 @@
 		clock-names = "aclk", "iface";
 		#iommu-cells = <0>;
 		power-domains = <&power RK3328_PD_VIDEO>;
+		rockchip,disable-mmu-reset;
 	};
 
 	vop: vop@ff370000 {
