diff -Naur libtool-2.4.6-old/configure libtool-2.4.6-new/configure
--- libtool-2.4.6-old/configure	2015-02-15 17:14:34.000000000 +0100
+++ libtool-2.4.6-new/configure	2020-10-06 12:07:53.789999988 +0200
@@ -2784,7 +2784,7 @@
 ## Automake Initialisation. ##
 ## ------------------------ ##
 
-am__api_version='1.15'
+am__api_version='1.16'
 
 # Find a good install program.  We prefer a C program (faster),
 # so one script is as good as another.  But avoid the broken or
