diff -Naur libtool-2.4.2-old/libltdl/m4/libtool.m4 libtool-2.4.2-new/libltdl/m4/libtool.m4
--- libtool-2.4.2-old/libltdl/m4/libtool.m4	2011-10-17 03:17:05.000000000 -0700
+++ libtool-2.4.2-new/libltdl/m4/libtool.m4	2011-11-02 17:19:15.000000000 -0700
@@ -6980,6 +6980,9 @@
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 *\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
 esac
+case "$CC $CFLAGS " in #(
+*\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
+esac
 
 dnl Parse the compiler output and extract the necessary
 dnl objects, libraries and library flags.
