diff --speed-large-files --no-dereference --minimal -Naur linux-6.2.10/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux-6.2.10/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux-6.2.10/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2023-04-10 12:20:25.606636540 +0200
+++ linux-6.2.10/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2023-04-10 18:13:58.443298021 +0200
@@ -364,6 +364,11 @@
 				function = "uart0";
 			};
 
+			uart5_ph_pins: uart5-ph-pins {
+				pins = "PH2", "PH3";
+				function = "uart5";
+			};
+
 			/omit-if-no-ref/
 			uart1_pins: uart1-pins {
 				pins = "PG6", "PG7";
@@ -1236,6 +1241,25 @@
 			nvmem-cell-names = "calibration";
 			#thermal-sensor-cells = <1>;
 		};
+
+		sunxi-info {
+			compatible = "allwinner,sun50i-h6-sys-info";
+			status = "okay";
+		};
+
+		addr_mgt: addr-mgt {
+			compatible = "allwinner,sunxi-addr_mgt";
+			type_addr_wifi = <0x2>;
+			type_addr_bt = <0x2>;
+			type_addr_eth = <0x2>;
+			status = "okay";
+		};
+
+		dump_reg: dump_reg@20000 {
+			compatible = "allwinner,sunxi-dump-reg";
+			reg = <0x0 0x03001000 0x0 0x0f20>;
+			status = "okay";
+		};
 	};
 
 	thermal-zones {
