diff --speed-large-files --no-dereference --minimal -Naur linux-6.4.11/drivers/clk/rockchip/clk-rk3528.c linux-6.4.11/drivers/clk/rockchip/clk-rk3528.c
--- linux-6.4.11/drivers/clk/rockchip/clk-rk3528.c	2023-08-22 13:19:01.206715340 +0200
+++ linux-6.4.11/drivers/clk/rockchip/clk-rk3528.c	2023-08-22 11:59:56.395891328 +0200
@@ -287,10 +287,10 @@
 	COMPOSITE_NOMUX(CLK_MATRIX_600M_SRC, "clk_600m_src", "gpll", CLK_IS_CRITICAL,
 	                RK3528_CLKSEL_CON(4), 0, 5, DFLAGS,
 	                RK3528_CLKGATE_CON(0), 11, GFLAGS),
-	COMPOSITE(DCLK_VOP_SRC0, "dclk_vop_src0", mux_gpll_cpll_p, 0,
+	COMPOSITE(DCLK_VOP_SRC0, "dclk_vop_src0", mux_gpll_cpll_p, CLK_IS_CRITICAL,
 	          RK3528_CLKSEL_CON(32), 10, 1, MFLAGS, 2, 8, DFLAGS,
 	          RK3528_CLKGATE_CON(3), 7, GFLAGS),
-	COMPOSITE(DCLK_VOP_SRC1, "dclk_vop_src1", mux_gpll_cpll_p, 0,
+	COMPOSITE(DCLK_VOP_SRC1, "dclk_vop_src1", mux_gpll_cpll_p, CLK_IS_CRITICAL,
 	          RK3528_CLKSEL_CON(33), 8, 1, MFLAGS, 0, 8, DFLAGS,
 	          RK3528_CLKGATE_CON(3), 8, GFLAGS),
 	COMPOSITE_NOMUX(CLK_HSM, "clk_hsm", "xin24m", 0,
@@ -792,9 +792,9 @@
 	     RK3528_CLKGATE_CON(39), 13, GFLAGS),
 	GATE(PCLK_CRU, "pclk_cru", "pclk_vo_root", CLK_IS_CRITICAL,
 	     RK3528_CLKGATE_CON(39), 15, GFLAGS),
-	GATE(PCLK_HDMI, "pclk_hdmi", "pclk_vo_root", 0,
+	GATE(PCLK_HDMI, "pclk_hdmi", "pclk_vo_root", CLK_IS_CRITICAL,
 	     RK3528_CLKGATE_CON(40), 6, GFLAGS),
-	GATE(PCLK_HDMIPHY, "pclk_hdmiphy", "pclk_vo_root", 0,
+	GATE(PCLK_HDMIPHY, "pclk_hdmiphy", "pclk_vo_root", CLK_IS_CRITICAL,
 	     RK3528_CLKGATE_CON(40), 14, GFLAGS),
 	GATE(PCLK_HDCP, "pclk_hdcp", "pclk_vo_root", 0,
 	     RK3528_CLKGATE_CON(41), 2, GFLAGS),
@@ -852,7 +852,7 @@
 
 	GATE(MCLK_SAI_I2S3, "mclk_sai_i2s3", "mclk_i2s3_8ch_sai_src", 0,
 	     RK3528_CLKGATE_CON(42), 2, GFLAGS),
-	COMPOSITE_NODIV(DCLK_VOP0, "dclk_vop0", dclk_vop0_p, CLK_SET_RATE_PARENT | CLK_SET_RATE_NO_REPARENT,
+	COMPOSITE_NODIV(DCLK_VOP0, "dclk_vop0", dclk_vop0_p, CLK_IS_CRITICAL | CLK_SET_RATE_PARENT | CLK_SET_RATE_NO_REPARENT,
 	                RK3528_CLKSEL_CON(84), 0, 1, MFLAGS,
 	                RK3528_CLKGATE_CON(40), 3, GFLAGS),
 	GATE(DCLK_VOP1, "dclk_vop1", "dclk_vop_src1", CLK_SET_RATE_PARENT,
@@ -862,14 +862,14 @@
 	GATE(DCLK_4X_CVBS, "dclk_4x_cvbs", "dclk_vop1", 0,
 	     RK3528_CLKGATE_CON(41), 5, GFLAGS),
 
-	FACTOR_GATE(CLK_SFR_HDMI, "clk_sfr_hdmi", "dclk_vop_src1", 0, 1, 4,
+	FACTOR_GATE(CLK_SFR_HDMI, "clk_sfr_hdmi", "dclk_vop_src1", CLK_IS_CRITICAL, 1, 4,
 	            RK3528_CLKGATE_CON(40), 7, GFLAGS),
 
 	GATE(CLK_SPDIF_HDMI, "clk_spdif_hdmi", "mclk_spdif_src", 0,
 	     RK3528_CLKGATE_CON(40), 10, GFLAGS),
 	GATE(MCLK_SPDIF, "mclk_spdif", "mclk_spdif_src", 0,
 	     RK3528_CLKGATE_CON(37), 15, GFLAGS),
-	GATE(CLK_CEC_HDMI, "clk_cec_hdmi", "clk_32k", 0,
+	GATE(CLK_CEC_HDMI, "clk_cec_hdmi", "clk_32k", CLK_IS_CRITICAL,
 	     RK3528_CLKGATE_CON(40), 8, GFLAGS),
 #if 0
 	GATE(CLK_USBHOST_OHCI, "clk_usbhost_ohci", "clk_usbhost_ohci_io", 0,
