diff --speed-large-files --no-dereference --minimal -Naur bluez-5.83/src/shared/shell.c bluez-5.83/src/shared/shell.c
--- bluez-5.83/src/shared/shell.c	2025-06-02 09:37:58.000000000 +0200
+++ bluez-5.83/src/shared/shell.c	2025-06-06 09:27:57.125023363 +0200
@@ -1432,7 +1432,7 @@
 	if (data.history[0] != '\0')
 		write_history(data.history);
 
-	rl_message("%s", "");
+	rl_message("");
 	rl_callback_handler_remove();
 }
 
