diff --speed-large-files --no-dereference --minimal -Naur bluez-5.82/src/shared/shell.c bluez-5.82/src/shared/shell.c
--- bluez-5.82/src/shared/shell.c	2025-04-01 15:52:58.000000000 +0200
+++ bluez-5.82/src/shared/shell.c	2025-05-08 09:59:56.247735525 +0200
@@ -1426,7 +1426,7 @@
 	if (data.history[0] != '\0')
 		write_history(data.history);
 
-	rl_message("");
+	rl_message();
 	rl_callback_handler_remove();
 }
 
