diff -Naur bluez-5.50-old/tools/l2test.c bluez-5.50-new/tools/l2test.c
--- bluez-5.50-old/tools/l2test.c	2016-10-28 14:57:27.000000000 +0200
+++ bluez-5.50-new/tools/l2test.c	2019-06-04 09:04:12.230000001 +0200
@@ -48,6 +48,7 @@
 #include "lib/l2cap.h"
 
 #include "src/shared/util.h"
+#include <linux/sockios.h>
 
 #define NIBBLE_TO_ASCII(c)  ((c) < 0x0a ? (c) + 0x30 : (c) + 0x57)
 
diff -Naur bluez-5.50-old/tools/rctest.c bluez-5.50-new/tools/rctest.c
--- bluez-5.50-old/tools/rctest.c	2015-09-04 03:19:36.000000000 +0200
+++ bluez-5.50-new/tools/rctest.c	2019-06-04 09:03:20.969999978 +0200
@@ -48,6 +48,7 @@
 #include "lib/sdp_lib.h"
 
 #include "src/shared/util.h"
+#include <linux/sockios.h>
 
 /* Test modes */
 enum {
