diff -Naur bluez-5.60-old/profiles/audio/media.c bluez-5.60-new/profiles/audio/media.c
--- bluez-5.60-old/profiles/audio/media.c	2021-07-07 15:45:55.000000000 +0200
+++ bluez-5.60-new/profiles/audio/media.c	2021-08-03 14:05:59.003332344 +0200
@@ -1280,7 +1280,7 @@
 
 	return media_player_send(mp, "Stop");
 }
-
+/*
 static bool pause(void *user_data)
 {
 	struct media_player *mp = user_data;
@@ -1292,7 +1292,7 @@
 
 	return media_player_send(mp, "Pause");
 }
-
+*/
 static bool next(void *user_data)
 {
 	struct media_player *mp = user_data;
