diff --speed-large-files --no-dereference --minimal -Naur linux-6.10.5/drivers/leds/trigger/ledtrig-disk.c linux-6.10.5/drivers/leds/trigger/ledtrig-disk.c
--- linux-6.10.5/drivers/leds/trigger/ledtrig-disk.c	2024-08-14 15:34:38.000000000 +0200
+++ linux-6.10.5/drivers/leds/trigger/ledtrig-disk.c	2024-08-16 16:47:47.785129586 +0200
@@ -11,7 +11,7 @@
 #include <linux/init.h>
 #include <linux/leds.h>
 
-#define BLINK_DELAY 30
+#define BLINK_DELAY 1
 
 DEFINE_LED_TRIGGER(ledtrig_disk);
 DEFINE_LED_TRIGGER(ledtrig_disk_read);
@@ -22,10 +22,10 @@
 	led_trigger_blink_oneshot(ledtrig_disk, BLINK_DELAY, BLINK_DELAY, 0);
 	if (write)
 		led_trigger_blink_oneshot(ledtrig_disk_write,
-					  BLINK_DELAY, BLINK_DELAY, 0);
+					  BLINK_DELAY, 0, 0);
 	else
 		led_trigger_blink_oneshot(ledtrig_disk_read,
-					  BLINK_DELAY, BLINK_DELAY, 0);
+					  BLINK_DELAY, 0, 0);
 }
 EXPORT_SYMBOL(ledtrig_disk_activity);
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10.5/drivers/mmc/core/core.c linux-6.10.5/drivers/mmc/core/core.c
--- linux-6.10.5/drivers/mmc/core/core.c	2024-08-14 15:34:38.000000000 +0200
+++ linux-6.10.5/drivers/mmc/core/core.c	2024-08-16 16:50:46.978463514 +0200
@@ -173,8 +173,10 @@
 	if (!err || !cmd->retries || mmc_card_removed(host->card)) {
 		mmc_should_fail_request(host, mrq);
 
-		if (!host->ongoing_mrq)
+		if (!host->ongoing_mrq) {
 			led_trigger_event(host->led, LED_OFF);
+			ledtrig_disk_activity(true);
+		}
 
 		if (mrq->sbc) {
 			pr_debug("%s: req done <CMD%u>: %d: %08x %08x %08x %08x\n",
