diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.2/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts linux-6.6.2/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
--- linux-6.6.2/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2023-11-21 13:30:00.583382576 +0100
+++ linux-6.6.2/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2023-11-21 13:27:24.200048739 +0100
@@ -62,7 +62,7 @@
 
 	sound-spdif {
 		compatible = "simple-audio-card";
-		simple-audio-card,name = "sun50i-h6-spdif";
+		simple-audio-card,name = "SPDIF";
 
 		simple-audio-card,cpu {
 			sound-dai = <&spdif>;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi linux-6.6.2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi
--- linux-6.6.2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi	2023-11-21 13:30:01.200049246 +0100
+++ linux-6.6.2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi	2023-11-21 13:28:15.713382240 +0100
@@ -53,7 +53,7 @@
 
 	sound-spdif {
 		compatible = "simple-audio-card";
-		simple-audio-card,name = "sun50i-h6-spdif";
+		simple-audio-card,name = "SPDIF";
 
 		simple-audio-card,cpu {
 			sound-dai = <&spdif>;
