diff --speed-large-files --no-dereference --minimal -Naur linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-04-29 16:11:03.456956989 +0200
+++ linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-04-29 16:02:28.576438253 +0200
@@ -85,7 +85,7 @@
 		 * to propagate this into the DTB handed to kernels.
 		 */
 		secmon@40000000 {
-			reg = <0x0 0x40000000 0x0 0x40000>;
+			reg = <0x0 0x40000000 0x0 0x80000>;
 			no-map;
 		};
 	};
@@ -307,6 +307,7 @@
 				function = "i2c0";
 			};
 
+			/omit-if-no-ref/
 			i2c3_ph_pins: i2c3-ph-pins {
 				pins = "PH4", "PH5";
 				function = "i2c3";
@@ -367,6 +368,7 @@
 				function = "spi1";
 			};
 
+			/omit-if-no-ref/
 			spdif_tx_pin: spdif-tx-pin {
 				pins = "PH4";
 				function = "spdif";
@@ -389,7 +391,6 @@
 				function = "uart1";
 			};
 
-			/omit-if-no-ref/
 			x32clk_fanout_pin: x32clk-fanout-pin {
 				pins = "PG10";
 				function = "clock";
@@ -1146,13 +1147,12 @@
 		};
 
 		rtc: rtc@7000000 {
-			compatible = "allwinner,sun50i-h616-rtc";
+			compatible = "allwinner,sun50i-h616-rtc",
+				         "allwinner,sun50i-h6-rtc";
 			reg = <0x07000000 0x400>;
-			interrupts = <GIC_SPI 104 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&r_ccu CLK_R_APB1_RTC>, <&osc24M>,
-				 <&ccu CLK_PLL_SYSTEM_32K>;
-			clock-names = "bus", "hosc",
-				      "pll-32k";
+			interrupts = <GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>,
+				         <GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH>;
+			clock-output-names = "osc32k", "osc32k-out", "iosc";
 			#clock-cells = <1>;
 		};
 
