diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc3/arch/arm64/boot/dts/allwinner/sun50i-h313-tanix-tx1.dts linux-6.10-rc3/arch/arm64/boot/dts/allwinner/sun50i-h313-tanix-tx1.dts
--- linux-6.10-rc3/arch/arm64/boot/dts/allwinner/sun50i-h313-tanix-tx1.dts	2024-06-09 23:19:43.000000000 +0200
+++ linux-6.10-rc3/arch/arm64/boot/dts/allwinner/sun50i-h313-tanix-tx1.dts	2024-06-14 17:32:59.271305076 +0200
@@ -1,15 +1,14 @@
-// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+// SPDX-License-Identifier: (GPL-2.0+ or MIT)
 /*
- * Copyright (C) 2024 Arm Ltd.
+ * Author: piotr.oniszczuk@gmail.com
  */
 
 /dts-v1/;
 
 #include "sun50i-h616.dtsi"
-
+#include "sun50i-h313-cpu-opp.dtsi"
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
-#include <dt-bindings/input/linux-event-codes.h>
 #include <dt-bindings/leds/common.h>
 
 / {
@@ -17,82 +16,134 @@
 	compatible = "oranth,tanix-tx1", "allwinner,sun50i-h616";
 
 	aliases {
+		mmc1 = &mmc1;
+		mmc2 = &mmc2;
 		serial0 = &uart0;
-		ethernet0 = &sdio_wifi;
+		ethernet0 = &wlan;
 	};
 
 	chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
-	gpio-keys {
-		compatible = "gpio-keys";
+	connector {
+		compatible = "hdmi-connector";
+		type = "d";
 
-		key {
-			label = "hidden";
-			linux,code = <BTN_0>;
-			gpios = <&pio 7 9 GPIO_ACTIVE_LOW>; /* PH9 */
+		port {
+			hdmi_con_in: endpoint {
+				remote-endpoint = <&hdmi_out_con>;
+			};
 		};
 	};
 
 	leds {
 		compatible = "gpio-leds";
 
-		led-0 {
-			function = LED_FUNCTION_POWER;
-			color = <LED_COLOR_ID_BLUE>;
+		led-red {
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			color = <LED_COLOR_ID_RED>;
 			gpios = <&pio 7 6 GPIO_ACTIVE_HIGH>; /* PH6 */
-			default-state = "on";
+			linux,default-trigger = "disk-activity";
 		};
 	};
 
-	wifi_pwrseq: pwrseq {
-		compatible = "mmc-pwrseq-simple";
-		clocks = <&rtc CLK_OSC32K_FANOUT>;
-		clock-names = "ext_clock";
-		pinctrl-0 = <&x32clk_fanout_pin>;
-		pinctrl-names = "default";
-		reset-gpios = <&pio 6 18 GPIO_ACTIVE_LOW>; /* PG18 */
-	};
-
 	reg_vcc5v: vcc5v {
-		/* board wide 5V supply directly from the DC input */
+		/* board wide 5V supply directly from the USB-C socket */
 		compatible = "regulator-fixed";
 		regulator-name = "vcc-5v";
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
 		regulator-always-on;
 	};
+
+	reg_vcc_wifi: reg_vcc_wifi {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc-wifi";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&pio 6 18 GPIO_ACTIVE_HIGH>; /* PG18 WL_REG_ON */
+		regulator-always-on;
+		enable-active-high;
+		status = "okay";
+	};
+
+	wifi_pwrseq: wifi_pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		clocks = <&rtc CLK_OSC32K_FANOUT>;
+		clock-names = "ext_clock";
+		pinctrl-names = "default";
+		pinctrl-0 = <&x32clk_fanout_pin>; /* PG10 with MUX3 set */
+	};
+
+	sound-spdif {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "SPDIF";
+
+		simple-audio-card,cpu {
+			sound-dai = <&spdif>;
+		};
+
+		simple-audio-card,codec {
+			sound-dai = <&spdif_out>;
+		};
+	};
+
+	spdif_out: spdif-out {
+		#sound-dai-cells = <0>;
+		compatible = "linux,spdif-dit";
+	};
 };
 
 &cpu0 {
 	cpu-supply = <&reg_dcdc2>;
+	status = "okay";
+};
+
+&de {
+	status = "okay";
 };
 
 &ehci0 {
 	status = "okay";
 };
 
-&ir {
+&hdmi {
+	hvcc-supply = <&reg_aldo1>;
+	status = "okay";
+};
+
+&hdmi_out {
+	hdmi_out_con: endpoint {
+		remote-endpoint = <&hdmi_con_in>;
+	};
+};
+
+&gpu {
+	mali-supply = <&reg_dcdc1>;
 	status = "okay";
 };
 
 &mmc1 {
 	vmmc-supply = <&reg_dldo1>;
-	vqmmc-supply = <&reg_aldo1>;
+	vqmmc-supply = <&reg_vcc_wifi>;
 	mmc-pwrseq = <&wifi_pwrseq>;
 	bus-width = <4>;
 	non-removable;
 	status = "okay";
 
-	sdio_wifi: wifi@1 {
+	wlan: wifi@1 {
+		compatible = "smartchip,s9083s";
 		reg = <1>;
+		interrupt-parent = <&pio>;
+		interrupts = <6 15 IRQ_TYPE_EDGE_RISING>; /* PG15 WL_HOSTWAKE*/
+		interrupt-names = "host-wake";
+		status = "okay";
 	};
 };
 
 &mmc2 {
 	vmmc-supply = <&reg_dldo1>;
-	vqmmc-supply = <&reg_aldo1>;
 	bus-width = <8>;
 	non-removable;
 	max-frequency = <100000000>;
@@ -105,55 +156,29 @@
 	status = "okay";
 };
 
-&pio {
-	vcc-pc-supply = <&reg_aldo1>;
-	vcc-pf-supply = <&reg_dldo1>;
-	vcc-pg-supply = <&reg_aldo1>;
-	vcc-ph-supply = <&reg_dldo1>;
-	vcc-pi-supply = <&reg_dldo1>;
-};
-
 &r_i2c {
 	status = "okay";
 
-	axp313: pmic@36 {
+	axp313a: pmic@36 {
 		compatible = "x-powers,axp313a";
 		reg = <0x36>;
-		#interrupt-cells = <1>;
-		interrupt-controller;
-
+		wakeup-source;
 		vin1-supply = <&reg_vcc5v>;
 		vin2-supply = <&reg_vcc5v>;
 		vin3-supply = <&reg_vcc5v>;
 
 		regulators {
-			/* Supplies VCC-PLL, so needs to be always on. */
-			reg_aldo1: aldo1 {
-				regulator-always-on;
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <1800000>;
-				regulator-name = "vcc1v8";
-			};
-
-			/* Supplies VCC-IO, so needs to be always on. */
-			reg_dldo1: dldo1 {
-				regulator-always-on;
-				regulator-min-microvolt = <3300000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-name = "vcc3v3";
-			};
-
 			reg_dcdc1: dcdc1 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <990000>;
-				regulator-name = "vdd-gpu-sys";
+				regulator-max-microvolt = <1160000>;
+				regulator-name = "vdd-gpu";
 			};
 
 			reg_dcdc2: dcdc2 {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1120000>;
+				regulator-max-microvolt = <1160000>;
 				regulator-name = "vdd-cpu";
 			};
 
@@ -163,6 +188,20 @@
 				regulator-max-microvolt = <1200000>;
 				regulator-name = "vdd-dram";
 			};
+
+			reg_aldo1: aldo1 {
+				regulator-always-on;
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-name = "vcc1v8";
+			};
+
+			reg_dldo1: dldo1 {
+				regulator-always-on;
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-name = "vcc3v3";
+			};
 		};
 	};
 };
@@ -173,11 +212,56 @@
 	status = "okay";
 };
 
+//&uart1 {
+//	pinctrl-names = "default";
+//	pinctrl-0 = <&uart1_pins>, <&uart1_rts_cts_pins>;
+//	uart-has-rtscts;
+//	status = "okay";
+
+//	bluetooth {
+//		compatible = "brcm,bcm43438-bt";
+//		clocks = <&rtc 1>;
+//		clock-names = "lpo";
+//		vbat-supply = <&reg_dldo1>;
+//		vddio-supply = <&reg_dldo1>;
+//		device-wakeup-gpios = <&r_pio 0 6 GPIO_ACTIVE_HIGH>; /* PL6 */
+//		host-wakeup-gpios = <&r_pio 0 5 GPIO_ACTIVE_HIGH>; /* PL5 */
+//		shutdown-gpios = <&r_pio 0 4 GPIO_ACTIVE_HIGH>; /* PL4 */
+//	};
+//};
+
 &usbotg {
-	dr_mode = "host";       /* USB A type receptable */
+	dr_mode = "host";
 	status = "okay";
 };
 
 &usbphy {
 	status = "okay";
 };
+
+&ir {
+	linux,rc-map-name = "rc-beelink-gs1";
+	status = "okay";
+};
+
+&codec {
+	allwinner,audio-routing =
+	        "Line Out", "LINEOUT";
+	status = "okay";
+};
+
+&ahub_dam_plat {
+	status = "okay";
+};
+
+&ahub_dam_mach {
+	status = "okay";
+};
+
+&ahub1_plat {
+	status = "okay";
+};
+
+&ahub1_mach {
+	status = "okay";
+};
