diff --speed-large-files --no-dereference --minimal -Naur linux-6.8-rc6/arch/arm/boot/dts/broadcom/bcm2711.dtsi linux-6.8-rc6/arch/arm/boot/dts/broadcom/bcm2711.dtsi
--- linux-6.8-rc6/arch/arm/boot/dts/broadcom/bcm2711.dtsi	2024-02-26 22:15:15.883898537 +0100
+++ linux-6.8-rc6/arch/arm/boot/dts/broadcom/bcm2711.dtsi	2024-02-26 22:06:52.850576032 +0100
@@ -31,6 +31,12 @@
 		clock-output-names = "108MHz-clock";
 	};
 
+	hevc_clk: hevc_clk {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <500000000>;
+	};
+
 	soc {
 		/*
 		 * Defined ranges:
@@ -644,6 +650,31 @@
 
 			dma-channel-mask = <0x3000>;
 		};
+
+		rpivid: codec@7eb10000 {
+			compatible = "raspberrypi,rpivid-vid-decoder";
+			#address-cells = <0x2>;
+			#size-cells = <0x2>;
+			reg = <0x0 0x7eb10000 0x1000>,
+			      <0x0 0x7eb00000 0x10000>;
+			reg-names = "intc","hevc";
+			interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&hevc_clk>;
+			clock-names = "hevc";
+		};
+
+		hevc-decoder@7eb00000 {
+			status = "disabled";
+		};
+		rpivid-local-intc@7eb10000 {
+			status = "disabled";
+		};
+		h264-decoder@7eb20000 {
+			status = "disabled";
+		};
+		vp9-decoder@7eb30000 {
+			status = "disabled";
+		};
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.8-rc6/arch/arm64/boot/dts/broadcom/bcm2712.dtsi linux-6.8-rc6/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
--- linux-6.8-rc6/arch/arm64/boot/dts/broadcom/bcm2712.dtsi	2024-02-26 22:15:15.957231869 +0100
+++ linux-6.8-rc6/arch/arm64/boot/dts/broadcom/bcm2712.dtsi	2024-02-26 22:13:39.407233946 +0100
@@ -1072,6 +1072,20 @@
 			reg = <0x10 0x00400018  0x0 0x18>;
 		};
 
+		rpivid: codec@800000 {
+			compatible = "raspberrypi,rpivid-vid-decoder";
+			reg = <0x10 0x00800000  0x0 0x10000>, /* HEVC */
+			      <0x10 0x00840000  0x0 0x1000>;  /* INTC */
+			reg-names = "hevc","intc";
+
+			interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
+
+			clocks = <&firmware_clocks 11>;
+			clock-names = "hevc";
+			//iommus = <&iommu2>;
+			status = "okay";
+		};
+
 		sdio1: mmc@fff000 {
 			compatible = "brcm,bcm2712-sdhci";
 			reg = <0x10 0x00fff000  0x0 0x260>,
