diff -Naur mock-old/setup.py mock-new/setup.py
--- mock-old/setup.py	2015-07-15 02:06:46.000000000 +0200
+++ mock-new/setup.py	2016-03-22 21:42:19.000000000 +0100
@@ -6,12 +6,12 @@
 
 # Setuptools 17.1 is required, and setup_requires cannot upgrade setuptools
 # in-place, nor trigger the use of a newer version. Abort cleanly up-front.
-setuptools_required = StrictVersion("17.1")
-setuptools_installed = StrictVersion(setuptools.__version__)
-if setuptools_installed < setuptools_required:
-    sys.stderr.write(
-        "mock requires setuptools>=17.1. Aborting installation\n")
-    sys.exit(1)
+#setuptools_required = StrictVersion("17.1")
+#setuptools_installed = StrictVersion(setuptools.__version__)
+#if setuptools_installed < setuptools_required:
+#    sys.stderr.write(
+#        "mock requires setuptools>=17.1. Aborting installation\n")
+#    sys.exit(1)
 
 setuptools.setup(
     setup_requires=['pbr>=1.3'],
