diff -Naur linux-5.4-rc1-old/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi linux-5.4-rc1-new/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi
--- linux-5.4-rc1-old/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi	2019-09-30 19:35:40.000000000 +0200
+++ linux-5.4-rc1-new/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi	2019-10-04 18:21:18.796666664 +0200
@@ -2177,6 +2177,29 @@
 			};
 		};
 
+		vdec: video-decoder@ff620000 {
+			compatible = "amlogic,g12a-vdec";
+			reg = <0x0 0xff620000 0x0 0x10000>,
+			      <0x0 0xffd0e180 0x0 0xe4>;
+			reg-names = "dos", "esparser";
+
+			interrupts = <GIC_SPI 44 IRQ_TYPE_EDGE_RISING>,
+				     <GIC_SPI 32 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "vdec", "esparser";
+
+			amlogic,ao-sysctrl = <&rti>;
+			amlogic,canvas = <&canvas>;
+
+			clocks = <&clkc CLKID_PARSER>,
+				 <&clkc CLKID_DOS>,
+				 <&clkc CLKID_VDEC_1>,
+				 <&clkc CLKID_VDEC_HEVC>,
+				 <&clkc CLKID_VDEC_HEVCF>;
+			clock-names = "dos_parser", "dos", "vdec_1", "vdec_hevc", "vdec_hevcf";
+			resets = <&reset RESET_PARSER>;
+			reset-names = "esparser";
+		};
+
 		gic: interrupt-controller@ffc01000 {
 			compatible = "arm,gic-400";
 			reg = <0x0 0xffc01000 0 0x1000>,
