diff -Naur linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi
--- linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi	2019-09-05 21:30:46.150000007 +0200
+++ linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi	2019-09-05 21:33:24.759999997 +0200
@@ -300,6 +300,13 @@
 				drive-strength = <40>;
 			};
 
+			ext_rmii_pins: rmii_pins {
+				pins = "PA0", "PA1", "PA2", "PA3", "PA4",
+				       "PA5", "PA6", "PA7", "PA8", "PA9";
+				function = "emac";
+				drive-strength = <40>;
+			};
+
 			hdmi_pins: hdmi-pins {
 				pins = "PH8", "PH9", "PH10";
 				function = "hdmi";

