diff -Naur linux-5.4-rc1-old/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts linux-5.4-rc1-new/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts
--- linux-5.4-rc1-old/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts	2019-09-30 19:35:40.000000000 +0200
+++ linux-5.4-rc1-new/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts	2019-10-04 18:15:20.036666666 +0200
@@ -14,6 +14,7 @@
 	compatible = "oranth,tanix-tx6", "allwinner,sun50i-h6";
 
 	aliases {
+		ethernet0 = &emac;
 		serial0 = &uart0;
 	};
 
@@ -45,6 +46,10 @@
 	status = "okay";
 };
 
+&ac200_pwm_clk {
+	status = "okay";
+};
+
 &ehci0 {
 	status = "okay";
 };
@@ -53,6 +58,18 @@
 	status = "okay";
 };
 
+&dwc3 {
+	status = "okay";
+};
+
+&emac {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ext_rmii_pins>;
+	phy-mode = "rmii";
+	phy-handle = <&ext_rmii_phy>;
+	status = "okay";
+};
+
 &hdmi {
 	status = "okay";
 };
@@ -63,6 +80,17 @@
 	};
 };
 
+&i2c3 {
+	status = "okay";
+};
+
+&mdio {
+	ext_rmii_phy: ethernet-phy@1 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <1>;
+	};
+};
+
 &mmc0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&mmc0_pins>;
@@ -72,6 +100,18 @@
 	status = "okay";
 };
 
+&mmc2 {
+	vmmc-supply = <&reg_vcc3v3>;
+	non-removable;
+	cap-mmc-hw-reset;
+	bus-width = <8>;
+	status = "okay";
+};
+
+&pwm {
+	status = "okay";
+};
+
 &ohci0 {
 	status = "okay";
 };
@@ -98,3 +138,7 @@
 &usb2phy {
 	status = "okay";
 };
+
+&usb3phy {
+	status = "okay";
+};
