diff -Naur linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
--- linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2019-09-08 17:20:39.023333340 +0200
+++ linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2019-09-08 17:39:45.633333301 +0200
@@ -220,7 +220,7 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
 				regulator-name = "vdd-cpu";
 			};
 
@@ -267,6 +267,10 @@
 	vcc-pm-supply = <&reg_aldo1>;
 };
 
+&cpu0 {
+	cpu-supply = <&reg_dcdca>;
+};
+
 &uart0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart0_ph_pins>;
diff -Naur linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi
--- linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi	2019-09-08 17:25:03.736666656 +0200
+++ linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi	2019-09-08 17:51:58.830000004 +0200
@@ -39,6 +39,7 @@
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
+			operating-points-v2 = <&cpu_opp_table>;
 			#cooling-cells = <2>;
 		};
 
@@ -49,6 +50,7 @@
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
+			operating-points-v2 = <&cpu_opp_table>;
 			#cooling-cells = <2>;
 		};
 
@@ -59,6 +61,7 @@
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
+			operating-points-v2 = <&cpu_opp_table>;
 			#cooling-cells = <2>;
 		};
 
@@ -69,10 +72,64 @@
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
+			operating-points-v2 = <&cpu_opp_table>;
 			#cooling-cells = <2>;
 		};
 	};
 
+	cpu_opp_table: opp_table {
+		compatible = "operating-points-v2";
+		opp-shared;
+
+		opp@480000000 {
+			opp-hz = /bits/ 64 <480000000>;
+			opp-microvolt = <880000 880000 880000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@720000000 {
+			opp-hz = /bits/ 64 <720000000>;
+			opp-microvolt = <880000 880000 880000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@816000000 {
+			opp-hz = /bits/ 64 <816000000>;
+			opp-microvolt = <880000 880000 880000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@888000000 {
+			opp-hz = /bits/ 64 <888000000>;
+			opp-microvolt = <880000 880000 880000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@1080000000 {
+			opp-hz = /bits/ 64 <1080000000>;
+			opp-microvolt = <940000 940000 940000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@1320000000 {
+			opp-hz = /bits/ 64 <1320000000>;
+			opp-microvolt = <1000000 1000000 1000000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@1488000000 {
+			opp-hz = /bits/ 64 <1488000000>;
+			opp-microvolt = <1060000 1060000 1060000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@1800000000 {
+			opp-hz = /bits/ 64 <1800000000>;
+			opp-microvolt = <1160000 1160000 1160000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+	};
+
 	de: display-engine {
 		compatible = "allwinner,sun50i-h6-display-engine";
 		allwinner,pipelines = <&mixer0>;
diff -Naur linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts
--- linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2019-09-08 17:20:37.610000007 +0200
+++ linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2019-09-08 17:40:08.773333327 +0200
@@ -172,7 +172,7 @@
 			reg_dcdcc: dcdcc {
 				regulator-enable-ramp-delay = <32000>;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
 				regulator-name = "vdd-gpu";
 			};
 
@@ -197,6 +197,10 @@
 	};
 };
 
+&cpu0 {
+	cpu-supply = <&reg_dcdca>;
+};
+
 &uart0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart0_ph_pins>;
