diff -Naur linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
--- linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2019-09-06 16:53:47.263333340 +0200
+++ linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2019-09-06 16:58:42.376666656 +0200
@@ -291,3 +291,7 @@
 		vbus-supply = <&reg_vcc5v>;
 	};
 };
+
+&ths {
+	status = "okay";
+};
diff -Naur linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts
--- linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts	2019-09-06 16:53:47.446666674 +0200
+++ linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts	2019-09-06 16:58:51.473333327 +0200
@@ -168,3 +168,7 @@
 	mali-supply = <&reg_dcdcc>;
 	status = "okay";
 };
+
+&ths {
+	status = "okay";
+};
diff -Naur linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-qplus.dts linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-qplus.dts
--- linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-qplus.dts	2019-09-06 16:53:47.490000007 +0200
+++ linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-qplus.dts	2019-09-06 16:59:08.593333340 +0200
@@ -154,3 +154,7 @@
 &r_ir {
 	status = "okay";
 };
+
+&ths {
+	status = "okay";
+};
diff -Naur linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts
--- linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts	2019-09-06 16:53:47.403333341 +0200
+++ linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts	2019-09-06 16:59:19.136666663 +0200
@@ -151,3 +151,7 @@
 &usb3phy {
 	status = "okay";
 };
+
+&ths {
+	status = "okay";
+};
