diff --speed-large-files --no-dereference --minimal -Naur pseudo-1.9.0/ports/linux/xattr/portdefs.h work/build.d/pseudo-1.9.0/ports/linux/xattr/portdefs.h
--- pseudo-1.9.0/ports/linux/xattr/portdefs.h	2018-01-20 17:04:14.000000000 +0100
+++ work/build.d/pseudo-1.9.0/ports/linux/xattr/portdefs.h	2022-06-10 16:25:35.895908864 +0200
@@ -1,2 +1,6 @@
-#include <attr/xattr.h>
+#include <sys/xattr.h>
 #include <stdint.h>
+
+#ifndef ENOATTR
+# define ENOATTR ENODATA
+#endif
diff --speed-large-files --no-dereference --minimal -Naur pseudo-1.9.0/pseudo_ipc.h work/build.d/pseudo-1.9.0/pseudo_ipc.h
--- pseudo-1.9.0/pseudo_ipc.h	2018-01-20 17:04:14.000000000 +0100
+++ work/build.d/pseudo-1.9.0/pseudo_ipc.h	2022-06-10 16:37:11.469247287 +0200
@@ -40,7 +40,7 @@
 	char path[];
 } pseudo_msg_t;
 
-enum {
+extern enum {
 	PSA_EXEC = 1,
 	PSA_WRITE = (PSA_EXEC << 1),
 	PSA_READ = (PSA_WRITE << 1),
diff --speed-large-files --no-dereference --minimal -Naur pseudo-1.9.0/templatefile.pyc work/build.d/pseudo-1.9.0/templatefile.pyc
--- pseudo-1.9.0/templatefile.pyc	1970-01-01 01:00:00.000000000 +0100
+++ work/build.d/pseudo-1.9.0/templatefile.pyc	2022-06-10 16:31:41.125911534 +0200
