diff -Naur pseudo-1.9.0-old/ports/linux/portdefs.h pseudo-1.9.0-new/ports/linux/portdefs.h
--- pseudo-1.9.0-old/ports/linux/portdefs.h	2018-01-20 08:04:14.000000000 -0800
+++ pseudo-1.9.0-new/ports/linux/portdefs.h	2021-04-29 06:29:46.722571909 -0700
@@ -26,4 +26,18 @@
 GLIBC_COMPAT_SYMBOL(memcpy,2.0);
 #endif
 
+#ifndef _STAT_VER
+#if defined (__aarch64__)
+#define _STAT_VER 0
+#elif defined (__x86_64__)
+#define _STAT_VER 1
+#else
+#define _STAT_VER 3
+#endif
+#endif
+
+#ifndef _MKNOD_VER
+#define _MKNOD_VER 0
+#endif
+
 #include <linux/capability.h>
