diff -Naur zlib-1.2.5-old/zlib.h zlib-1.2.5-new/zlib.h
--- zlib-1.2.5-old/zlib.h	2010-04-19 21:12:48.000000000 -0700
+++ zlib-1.2.5-new/zlib.h	2011-11-05 08:24:30.000000000 -0700
@@ -1578,7 +1578,7 @@
 #  define gzoffset gzoffset64
 #  define adler32_combine adler32_combine64
 #  define crc32_combine crc32_combine64
-#  ifdef _LARGEFILE64_SOURCE
+#  ifndef _LARGEFILE64_SOURCE
      ZEXTERN gzFile ZEXPORT gzopen64 OF((const char *, const char *));
      ZEXTERN z_off_t ZEXPORT gzseek64 OF((gzFile, z_off_t, int));
      ZEXTERN z_off_t ZEXPORT gztell64 OF((gzFile));
