diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc7/drivers/net/wireless/realtek/rtw88/main.c linux-6.16-rc7/drivers/net/wireless/realtek/rtw88/main.c
--- linux-6.16-rc7/drivers/net/wireless/realtek/rtw88/main.c	2025-07-21 00:18:33.000000000 +0200
+++ linux-6.16-rc7/drivers/net/wireless/realtek/rtw88/main.c	2025-07-23 15:05:06.233406252 +0200
@@ -995,6 +995,8 @@
 		hal->rf_path_num = 1;
 		if (!chip->fix_rf_phy_num)
 			hal->rf_phy_num = hal->rf_path_num;
+		printk("rtw88: %s:%d hal->rf_phy_num=%d hal->rf_path_num=%d hw_ant_num=%d\n",
+			__func__, __LINE__, hal->rf_phy_num, hal->rf_path_num, hw_ant_num);
 		hal->antenna_tx = BB_PATH_A;
 		hal->antenna_rx = BB_PATH_A;
 		break;
@@ -1854,6 +1856,9 @@
 	hal->chip_version = rtw_read32(rtwdev, REG_SYS_CFG1);
 	hal->cut_version = BIT_GET_CHIP_VER(hal->chip_version);
 	hal->mp_chip = (hal->chip_version & BIT_RTL_ID) ? 0 : 1;
+	printk("rtw88: %s:%d hal->chip_version=0x%x\n",
+	       __func__, __LINE__, hal->chip_version);
+
 	if (hal->chip_version & BIT_RF_TYPE_ID) {
 		hal->rf_type = RF_2T2R;
 		hal->rf_path_num = 2;
@@ -1867,7 +1872,8 @@
 	}
 	hal->rf_phy_num = chip->fix_rf_phy_num ? chip->fix_rf_phy_num :
 			  hal->rf_path_num;
-
+	printk("rtw88: %s:%d hal->rf_phy_num=%d hal->rf_path_num=%d\n",
+		__func__, __LINE__, hal->rf_phy_num, hal->rf_path_num);
 	efuse->physical_size = chip->phy_efuse_size;
 	efuse->logical_size = chip->log_efuse_size;
 	efuse->protect_size = chip->ptct_efuse_size;
@@ -1941,6 +1947,9 @@
 	for (i = 0; i < HW_FEATURE_LEN; i++)
 		hw_feature[i] = rtw_read8(rtwdev, REG_C2HEVT + 2 + i);
 
+	printk("rtw88: %s:%d hw_feature = %*ph\n",
+		__func__, __LINE__, HW_FEATURE_LEN, hw_feature);
+
 	rtw_write8(rtwdev, REG_C2HEVT, 0);
 
 	bw = GET_EFUSE_HW_CAP_BW(hw_feature);
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc7/drivers/net/wireless/realtek/rtw88/sdio.c linux-6.16-rc7/drivers/net/wireless/realtek/rtw88/sdio.c
--- linux-6.16-rc7/drivers/net/wireless/realtek/rtw88/sdio.c	2025-07-21 00:18:33.000000000 +0200
+++ linux-6.16-rc7/drivers/net/wireless/realtek/rtw88/sdio.c	2025-07-23 14:41:07.823401331 +0200
@@ -144,6 +144,10 @@
 
 static bool rtw_sdio_use_direct_io(struct rtw_dev *rtwdev, u32 addr)
 {
+	if (!rtw_sdio_is_bus_addr(addr) &&
+	    !test_bit(RTW_FLAG_POWERON, rtwdev->flags))
+		return false;
+
 	return !rtw_sdio_is_sdio30_supported(rtwdev) ||
 		rtw_sdio_is_bus_addr(addr);
 }
