diff -Naur linux-5.8-rc2-old/drivers/gpu/drm/lima/lima_devfreq.c linux-5.8-rc2-new/drivers/gpu/drm/lima/lima_devfreq.c
--- linux-5.8-rc2-old/drivers/gpu/drm/lima/lima_devfreq.c	2020-07-15 15:20:12.833333333 +0200
+++ linux-5.8-rc2-new/drivers/gpu/drm/lima/lima_devfreq.c	2020-07-15 15:05:27.139999988 +0200
@@ -178,7 +178,7 @@
 	ldevfreq->gov_data.downdifferential = 5;
 
 	devfreq = devm_devfreq_add_device(dev, &lima_devfreq_profile,
-					  DEVFREQ_GOV_SIMPLE_ONDEMAND,
+					  DEVFREQ_GOV_PERFORMANCE,
 					  &ldevfreq->gov_data);
 	if (IS_ERR(devfreq)) {
 		dev_err(dev, "Couldn't initialize GPU devfreq\n");
