diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.5/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts linux-6.14.5/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
--- linux-6.14.5/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2025-05-02 08:02:16.000000000 +0200
+++ linux-6.14.5/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2025-05-03 08:22:28.721085186 +0200
@@ -6,8 +6,8 @@
 #include "sun50i-h6.dtsi"
 #include "sun50i-h6-cpu-opp.dtsi"
 #include "sun50i-h6-gpu-opp.dtsi"
-
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/leds/common.h>
 
 / {
 	model = "Beelink GS1";
@@ -45,9 +45,10 @@
 		compatible = "gpio-leds";
 
 		led {
-			label = "beelink:white:power";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			label = "beelink:white:disk";
 			gpios = <&r_pio 0 4 GPIO_ACTIVE_HIGH>; /* PL4 */
-			default-state = "on";
+			linux,default-trigger = "disk-activity";
 		};
 	};
 
@@ -119,6 +120,14 @@
 	};
 };
 
+&sound_hdmi {
+	status = "okay";
+};
+
+&i2s1 {
+	status = "okay";
+};
+
 &mdio {
 	ext_rgmii_phy: ethernet-phy@1 {
 		compatible = "ethernet-phy-ieee802.3-c22";
@@ -139,6 +148,7 @@
 	vqmmc-supply = <&reg_bldo2>;
 	non-removable;
 	cap-mmc-hw-reset;
+	mmc-hs200-1_8v;
 	bus-width = <8>;
 	status = "okay";
 };
@@ -257,7 +267,7 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1160000>;
+				regulator-max-microvolt = <1200000>;
 				regulator-ramp-delay = <2500>;
 				regulator-name = "vdd-cpu";
 			};
