diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.5/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi linux-6.14.5/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi
--- linux-6.14.5/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi	2025-05-02 21:33:08.044358526 +0200
+++ linux-6.14.5/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi	2025-05-02 21:32:46.301025659 +0200
@@ -5,6 +5,7 @@
 
 #include "sun50i-h6.dtsi"
 #include "sun50i-h6-cpu-opp.dtsi"
+#include "sun50i-h6-gpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
@@ -29,14 +30,6 @@
 		};
 	};
 
-	/* used for FD650 LED display driver */
-	i2c {
-		compatible = "i2c-gpio";
-		sda-gpios = <&pio 7 6 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>; /* PH6 */
-		scl-gpios = <&pio 7 5 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>; /* PH5 */
-		i2c-gpio,delay-us = <5>;
-	};
-
 	reg_vcc1v8: regulator-vcc1v8 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc1v8";
@@ -53,7 +53,7 @@
 
 	sound-spdif {
 		compatible = "simple-audio-card";
-		simple-audio-card,name = "sun50i-h6-spdif";
+		simple-audio-card,name = "SPDIF";
 
 		simple-audio-card,cpu {
 			sound-dai = <&spdif>;
