
diff --speed-large-files --no-dereference --minimal -Naur linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-11-16 13:04:49.558353822 +0100
+++ linux-6.12-rc7/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-11-16 13:02:52.181686761 +0100
@@ -1344,4 +1371,56 @@
 			};
 		};
 	};
+
+	gpu_opp_table: gpu-opp-table {
+		compatible = "operating-points-v2";
+
+		opp-420000000 {
+			opp-hz = /bits/ 64 <420000000>;
+			opp-microvolt = <900000>;
+		};
+
+		opp-456000000 {
+			opp-hz = /bits/ 64 <456000000>;
+			opp-microvolt = <900000>;
+		};
+
+		opp-504000000 {
+			opp-hz = /bits/ 64 <504000000>;
+			opp-microvolt = <900000>;
+		};
+
+		opp-552000000 {
+			opp-hz = /bits/ 64 <552000000>;
+			opp-microvolt = <900000>;
+		};
+
+		opp-600000000 {
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <900000>;
+		};
+
+		opp-648000000 {
+			opp-hz = /bits/ 64 <648000000>;
+			opp-microvolt = <960000>;
+		};
+
+		opp-696000000 {
+			opp-hz = /bits/ 64 <696000000>;
+			opp-microvolt = <1020000>;
+			status = "disabled";
+		};
+
+		opp-744000000 {
+			opp-hz = /bits/ 64 <744000000>;
+			opp-microvolt = <1080000>;
+			status = "disabled";
+		};
+
+		opp-796000000 {
+			opp-hz = /bits/ 64 <796000000>;
+			opp-microvolt = <1080000>;
+			status = "disabled";
+		};
+	};
 };
