diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/Makefile linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/Makefile
--- linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/Makefile	2025-05-02 10:58:31.837774736 +0200
+++ linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/Makefile	2025-05-02 11:03:31.037768299 +0200
@@ -12,6 +12,14 @@
 	overclock-h616-cpu-to-1.800ghz.dtbo \
 	overclock-h616-cpu-to-1.896ghz.dtbo \
 	overclock-h616-cpu-to-1.992ghz.dtbo \
+	overclock-h313-gpu-to-696mhz.dtbo \
+	overclock-h313-gpu-to-744mhz.dtbo \
+	overclock-h313-gpu-to-792mhz.dtbo \
+	overclock-h313-gpu-to-840mhz.dtbo \
+	overclock-h616-gpu-to-696mhz.dtbo \
+	overclock-h616-gpu-to-744mhz.dtbo \
+	overclock-h616-gpu-to-792mhz.dtbo \
+	overclock-h616-gpu-to-840mhz.dtbo \
 
 targets += $(dtb-y)
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-696mhz.dtso linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-696mhz.dtso
--- linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-696mhz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-696mhz.dtso	2025-05-02 10:55:18.121112215 +0200
@@ -0,0 +1,10 @@
+/dts-v1/;
+/plugin/;
+
+&{/gpu-opp-table} {
+	opp-696000000 {
+		opp-hz = /bits/ 64 <696000000>;
+		opp-microvolt = <1020000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-744mhz.dtso linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-744mhz.dtso
--- linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-744mhz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-744mhz.dtso	2025-05-02 10:55:00.797779251 +0200
@@ -0,0 +1,16 @@
+/dts-v1/;
+/plugin/;
+
+&{/gpu-opp-table} {
+	opp-696000000 {
+		opp-hz = /bits/ 64 <696000000>;
+		opp-microvolt = <1020000>;
+		status = "okay";
+	};
+
+	opp-744000000 {
+		opp-hz = /bits/ 64 <744000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-792mhz.dtso linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-792mhz.dtso
--- linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-792mhz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-792mhz.dtso	2025-05-02 10:54:48.001112862 +0200
@@ -0,0 +1,22 @@
+/dts-v1/;
+/plugin/;
+
+&{/gpu-opp-table} {
+	opp-696000000 {
+		opp-hz = /bits/ 64 <696000000>;
+		opp-microvolt = <1020000>;
+		status = "okay";
+	};
+
+	opp-744000000 {
+		opp-hz = /bits/ 64 <744000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+
+	opp-792000000 {
+		opp-hz = /bits/ 64 <792000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-840mhz.dtso linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-840mhz.dtso
--- linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-840mhz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h313-gpu-to-840mhz.dtso	2025-05-02 10:54:28.637779933 +0200
@@ -0,0 +1,28 @@
+/dts-v1/;
+/plugin/;
+
+&{/gpu-opp-table} {
+	opp-696000000 {
+		opp-hz = /bits/ 64 <696000000>;
+		opp-microvolt = <1020000>;
+		status = "okay";
+	};
+
+	opp-744000000 {
+		opp-hz = /bits/ 64 <744000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+
+	opp-792000000 {
+		opp-hz = /bits/ 64 <792000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+
+	opp-840000000 {
+		opp-hz = /bits/ 64 <840000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-696mhz.dtso linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-696mhz.dtso
--- linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-696mhz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-696mhz.dtso	2025-05-02 10:55:18.121112215 +0200
@@ -0,0 +1,10 @@
+/dts-v1/;
+/plugin/;
+
+&{/gpu-opp-table} {
+	opp-696000000 {
+		opp-hz = /bits/ 64 <696000000>;
+		opp-microvolt = <1020000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-744mhz.dtso linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-744mhz.dtso
--- linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-744mhz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-744mhz.dtso	2025-05-02 10:55:00.797779251 +0200
@@ -0,0 +1,16 @@
+/dts-v1/;
+/plugin/;
+
+&{/gpu-opp-table} {
+	opp-696000000 {
+		opp-hz = /bits/ 64 <696000000>;
+		opp-microvolt = <1020000>;
+		status = "okay";
+	};
+
+	opp-744000000 {
+		opp-hz = /bits/ 64 <744000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-792mhz.dtso linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-792mhz.dtso
--- linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-792mhz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-792mhz.dtso	2025-05-02 10:54:48.001112862 +0200
@@ -0,0 +1,22 @@
+/dts-v1/;
+/plugin/;
+
+&{/gpu-opp-table} {
+	opp-696000000 {
+		opp-hz = /bits/ 64 <696000000>;
+		opp-microvolt = <1020000>;
+		status = "okay";
+	};
+
+	opp-744000000 {
+		opp-hz = /bits/ 64 <744000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+
+	opp-792000000 {
+		opp-hz = /bits/ 64 <792000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-840mhz.dtso linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-840mhz.dtso
--- linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-840mhz.dtso	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.14.5/arch/arm64/boot/dts/allwinner/overlay/overclock-h616-gpu-to-840mhz.dtso	2025-05-02 10:54:28.637779933 +0200
@@ -0,0 +1,28 @@
+/dts-v1/;
+/plugin/;
+
+&{/gpu-opp-table} {
+	opp-696000000 {
+		opp-hz = /bits/ 64 <696000000>;
+		opp-microvolt = <1020000>;
+		status = "okay";
+	};
+
+	opp-744000000 {
+		opp-hz = /bits/ 64 <744000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+
+	opp-792000000 {
+		opp-hz = /bits/ 64 <792000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+
+	opp-840000000 {
+		opp-hz = /bits/ 64 <840000000>;
+		opp-microvolt = <1080000>;
+		status = "okay";
+	};
+};
