diff --speed-large-files --no-dereference --minimal -Naur linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-base.dtsi linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-base.dtsi
--- linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-base.dtsi	2024-02-24 12:58:17.452839060 +0100
+++ linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-base.dtsi	2024-02-24 12:54:01.019511270 +0100
@@ -10,6 +10,7 @@
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/power/rk3399-power.h>
 #include <dt-bindings/thermal/thermal.h>
+#include <dt-bindings/leds/common.h>
 
 / {
 	compatible = "rockchip,rk3399";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi.dts linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi.dts
--- linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi.dts	2024-02-18 21:56:25.000000000 +0100
+++ linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi.dts	2024-02-24 12:55:46.862842317 +0100
@@ -18,8 +18,8 @@
 
 	aliases {
 		ethernet0 = &gmac;
-		mmc0 = &sdio0;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdio0;
 		mmc2 = &sdhci;
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts
--- linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts	2024-02-18 21:56:25.000000000 +0100
+++ linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts	2024-02-24 12:56:15.936175019 +0100
@@ -16,8 +16,8 @@
 
 	aliases {
 		ethernet0 = &gmac;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
@@ -41,15 +41,15 @@
 			function = LED_FUNCTION_POWER;
 			color = <LED_COLOR_ID_GREEN>;
 			gpios = <&gpio3 RK_PD4 GPIO_ACTIVE_LOW>;
-			linux,default-trigger = "default-on";
+			linux,default-trigger = "heartbeat";
 		};
 
 		/* USER_LED2 */
 		led-1 {
-			function = LED_FUNCTION_STATUS;
 			color = <LED_COLOR_ID_BLUE>;
 			gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "disk-activity";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts
--- linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts	2024-02-24 12:58:17.666172389 +0100
+++ linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts	2024-02-24 12:54:01.022844604 +0100
@@ -13,6 +13,8 @@
 	compatible = "radxa,rockpi4b", "radxa,rockpi4", "rockchip,rk3399";
 
 	aliases {
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio0;
 	};
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts
--- linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts	2024-02-24 12:58:17.666172389 +0100
+++ linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts	2024-02-24 12:54:01.022844604 +0100
@@ -13,6 +13,8 @@
 	compatible = "radxa,rockpi4b-plus", "radxa,rockpi4", "rockchip,rk3399";
 
 	aliases {
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio0;
 	};
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts
--- linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts	2024-02-24 12:58:17.666172389 +0100
+++ linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts	2024-02-24 12:54:01.022844604 +0100
@@ -14,6 +14,8 @@
 	compatible = "radxa,rockpi4c", "radxa,rockpi4", "rockchip,rk3399";
 
 	aliases {
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio0;
 	};
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
--- linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi	2024-02-18 21:56:25.000000000 +0100
+++ linux-6.8-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi	2024-02-24 12:56:42.669507776 +0100
@@ -13,8 +13,8 @@
 / {
 	aliases {
 		ethernet0 = &gmac;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
@@ -35,10 +35,10 @@
 
 		/* USER_LED2 */
 		led-0 {
-			function = LED_FUNCTION_STATUS;
 			color = <LED_COLOR_ID_BLUE>;
 			gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "disk-activity";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.8.2/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts linux-6.8.2/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts
--- linux-6.8.2/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts	2024-03-26 23:23:34.000000000 +0100
+++ linux-6.8.2/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts	2024-03-30 19:27:30.003354436 +0100
@@ -9,10 +9,12 @@
 #include "rk3399-t-opp.dtsi"
 
 / {
-	model = "Radxa ROCK 4SE";
-	compatible = "radxa,rock-4se", "rockchip,rk3399";
+	model = "Radxa ROCK Pi 4SE";
+	compatible = "radxa,rockpi4se", "radxa,rockpi4", "rockchip,rk3399";
 
 	aliases {
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio0;
 	};
 };
