diff --speed-large-files --no-dereference --minimal -Naur linux-6.10.11/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4-lts.dts linux-6.10.11/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4-lts.dts
--- linux-6.10.11/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4-lts.dts	2024-09-29 15:47:14.229221848 +0200
+++ linux-6.10.11/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4-lts.dts	2024-09-29 13:50:10.212108363 +0200
@@ -263,6 +263,14 @@
 		};
 	};
 
+	fan: pwm-fan {
+		compatible = "pwm-fan";
+		cooling-levels = <0 100 150 200 255>;
+		#cooling-cells = <2>;
+		fan-supply = <&vcc5v0_sys>;
+		pwms = <&pwm1 0 50000 0>;
+	};
+
 	sdio_pwrseq: sdio-pwrseq {
 		compatible = "mmc-pwrseq-simple";
 		clocks = <&rk808 1>;
@@ -1094,6 +1102,19 @@
 };
 
 &cpu_thermal {
+	trips {
+		cpu_warm: cpu_warm {
+			temperature = <55000>;
+			hysteresis = <2000>;
+			type = "active";
+		};
+
+		cpu_hot: cpu_hot {
+			temperature = <65000>;
+			hysteresis = <2000>;
+			type = "active";
+		};
+	};
 
 	cooling-maps {
 
@@ -1115,6 +1136,14 @@
 				<&cpu_b1 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
 		};
 
-	};
+		map2 {
+			trip = <&cpu_warm>;
+			cooling-device = <&fan THERMAL_NO_LIMIT 1>;
+		};
 
+		map3 {
+			trip = <&cpu_hot>;
+			cooling-device = <&fan 2 THERMAL_NO_LIMIT>;
+		};
+	};
 };
