diff --speed-large-files --no-dereference --minimal -Naur linux-6.15/arch/arm64/boot/dts/rockchip/Makefile linux-6.15/arch/arm64/boot/dts/rockchip/Makefile
--- linux-6.15/arch/arm64/boot/dts/rockchip/Makefile	2025-05-29 10:37:30.848300899 +0200
+++ linux-6.15/arch/arm64/boot/dts/rockchip/Makefile	2025-05-29 10:29:23.628311239 +0200
@@ -110,6 +110,7 @@
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-radxa-cm3-io.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-radxa-zero-3e.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-radxa-zero-3w.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-radxa-zero-3w-ap6212.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-roc-pc.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-x96_x6.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-urve-pi.dtb
diff --speed-large-files --no-dereference --minimal -Naur linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3.dtsi linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3.dtsi
--- linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3.dtsi	2025-05-26 01:09:23.000000000 +0200
+++ linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3.dtsi	2025-05-29 15:39:19.704579082 +0200
@@ -4,6 +4,7 @@
 #include <dt-bindings/leds/common.h>
 #include <dt-bindings/soc/rockchip,vop2.h>
 #include "rk3566t.dtsi"
+#include <dt-bindings/leds/common.h>
 
 / {
 	chosen {
@@ -29,9 +30,9 @@
 		led-green {
 			color = <LED_COLOR_ID_GREEN>;
 			default-state = "on";
-			function = LED_FUNCTION_HEARTBEAT;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			gpios = <&gpio0 RK_PA0 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "disk-activity";
 		};
 	};
 
@@ -461,7 +462,14 @@
 	vccio3-supply = <&vccio_sd>;
 	vccio4-supply = <&vcc_1v8>;
 	vccio5-supply = <&vcc_3v3>;
-	vccio6-supply = <&vcc_3v3>;
+
+	/* by v1.11 schematic, vccio6-supply should be vcc_3v3, but
+	   3v3 results with v.low speeds on aic8800 wlan. With vcc1v8_dvp
+	   however aic8800 seems work as expected (in my case: 2.4G gives
+	   40-45Mb/s while 5G gives 170-240Mb/s.
+	*/
+	vccio6-supply = <&vcc1v8_dvp>;
+	
 	vccio7-supply = <&vcc_3v3>;
 	status = "okay";
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-ap6212.dts linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-ap6212.dts
--- linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-ap6212.dts	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w-ap6212.dts	2025-05-29 16:06:45.091210398 +0200
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ *
+ * Copyright (c) 2023 Piotr Oniszczuk <piotr.oniszczuk@gmail.com>
+ *
+ */
+
+/dts-v1/;
+
+#include "rk3566-radxa-zero-3.dtsi"
+
+/ {
+	model = "Radxa ZERO 3W (AP6212)";
+	compatible ="radxa,zero-3w", "radxa,zero-3", "rockchip,rk3566";
+
+	aliases {
+		mmc0 = &sdmmc0;
+		mmc1 = &sdhci;
+		ethernet0 = &sdmmc1;
+	};
+
+	sdio_pwrseq: sdio-pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		clocks = <&rk817 1>;
+		clock-names = "ext_clock";
+		pinctrl-names = "default";
+		pinctrl-0 = <&wifi_reg_on_h>;
+		post-power-on-delay-ms = <100>;
+		power-off-delay-us = <5000000>;
+		reset-gpios = <&gpio0 RK_PC0 GPIO_ACTIVE_LOW>;
+	};
+
+	board_antenna: board-antenna {
+		status = "okay";
+		compatible = "regulator-fixed";
+		enable-active-low;
+		gpio = <&gpio3 RK_PD3 GPIO_ACTIVE_LOW>;
+		regulator-always-on;
+		regulator-boot-on;
+		pinctrl-0 = <&ant_1>;
+		pinctrl-names = "default";
+		regulator-name = "board_antenna";
+	};
+};
+
+&pinctrl {
+
+	wifi {
+		wifi_reg_on_h: wifi-reg-on-h {
+			rockchip,pins = <0 RK_PC0 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+		wifi_wake_host_h: wifi-wake-host-h {
+			rockchip,pins = <0 RK_PB7 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
+	bt {
+		bt_enable_h: bt-enable-h {
+			rockchip,pins = <0 RK_PC1 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+		bt_host_wake_l: bt-host-wake-l {
+			rockchip,pins = <0 RK_PB3 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+		bt_wake_l: bt-wake-l {
+			rockchip,pins = <0 RK_PB4 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
+	sdmmc1 {
+		/omit-if-no-ref/
+		sdmmc1_bus4: sdmmc1-bus4 {
+			rockchip,pins =
+				/* sdmmc1_d0 */
+				<2 RK_PA3 1 &pcfg_pull_up_drv_level_12>,
+				/* sdmmc1_d1 */
+				<2 RK_PA4 1 &pcfg_pull_up_drv_level_12>,
+				/* sdmmc1_d2 */
+				<2 RK_PA5 1 &pcfg_pull_up_drv_level_12>,
+				/* sdmmc1_d3 */
+				<2 RK_PA6 1 &pcfg_pull_up_drv_level_12>;
+		};
+
+		/omit-if-no-ref/
+		sdmmc1_clk: sdmmc1-clk {
+			rockchip,pins =
+				/* sdmmc1_clk */
+				<2 RK_PB0 1 &pcfg_pull_up_drv_level_12>;
+		};
+
+		/omit-if-no-ref/
+		sdmmc1_cmd: sdmmc1-cmd {
+			rockchip,pins =
+				/* sdmmc1_cmd */
+				<2 RK_PA7 1 &pcfg_pull_up_drv_level_12>;
+		};
+
+		/omit-if-no-ref/
+		sdmmc1_det: sdmmc1-det {
+			rockchip,pins =
+				/* sdmmc1_det */
+				<2 RK_PB2 1 &pcfg_pull_up>;
+		};
+
+		/omit-if-no-ref/
+		sdmmc1_pwren: sdmmc1-pwren {
+			rockchip,pins =
+				/* sdmmc1_pwren */
+				<2 RK_PB1 1 &pcfg_pull_none>;
+		};
+	};
+
+	antenna {
+		ant_1: ant-1 {
+			rockchip,pins = <3 RK_PD3 RK_FUNC_GPIO &pcfg_pull_down>;
+		};
+	};
+};
+
+&sdhci {
+	bus-width = <8>;
+	cap-mmc-highspeed;
+	max-frequency = <200000000>;
+	mmc-hs200-1_8v;
+	no-sd;
+	no-sdio;
+	non-removable;
+	pinctrl-names = "default";
+	pinctrl-0 = <&emmc_bus8 &emmc_clk &emmc_cmd &emmc_datastrobe>;
+	vmmc-supply = <&vcc_3v3>;
+	vqmmc-supply = <&vcc_1v8>;
+	status = "okay";
+};
+
+&sdmmc1 {
+	max-frequency = <150000000>;
+	bus-width = <4>;
+	cap-sd-highspeed;
+	cap-sdio-irq;
+	keep-power-in-suspend;
+	mmc-pwrseq = <&sdio_pwrseq>;
+	no-mmc;
+	no-sd;
+	non-removable;
+	pinctrl-names = "default";
+	pinctrl-0 = <&sdmmc1_bus4 &sdmmc1_clk &sdmmc1_cmd>;
+	sd-uhs-sdr104;
+	vmmc-supply = <&vcc_3v3>;
+	vqmmc-supply = <&vcc_1v8>;
+	status = "okay";
+};
+
+&uart1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart1m0_xfer &uart1m0_ctsn &uart1m0_rtsn>;
+
+	bluetooth {
+		compatible = "brcm,bcm43438-bt";
+		clocks = <&rk817 1>;
+		clock-names = "lpo";
+		device-wakeup-gpios = <&gpio0 RK_PB4 GPIO_ACTIVE_HIGH>;
+		host-wakeup-gpios = <&gpio0 RK_PB3 GPIO_ACTIVE_HIGH>;
+		shutdown-gpios = <&gpio0 RK_PC1 GPIO_ACTIVE_HIGH>;
+		max-speed = <1500000>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&bt_host_wake_l &bt_wake_l &bt_enable_h>;
+		vbat-supply = <&vcc3v3_sys>;
+		vddio-supply = <&vcc_1v8>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w.dts linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w.dts
--- linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w.dts	2025-05-26 01:09:23.000000000 +0200
+++ linux-6.15/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w.dts	2025-05-29 16:07:03.014543338 +0200
@@ -5,13 +5,13 @@
 #include "rk3566-radxa-zero-3.dtsi"
 
 / {
-	model = "Radxa ZERO 3W";
+	model = "Radxa ZERO 3W (AIC8800)";
 	compatible = "radxa,zero-3w", "rockchip,rk3566";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc0;
-		mmc2 = &sdmmc1;
+		mmc0 = &sdmmc0;
+		mmc1 = &sdhci;
+		ethernet0 = &sdmmc1;
 	};
 
 	sdio_pwrseq: sdio-pwrseq {
@@ -24,22 +24,21 @@
 		power-off-delay-us = <5000000>;
 		reset-gpios = <&gpio0 RK_PC0 GPIO_ACTIVE_LOW>;
 	};
+
+	board_antenna: board-antenna {
+		status = "okay";
+		compatible = "regulator-fixed";
+		enable-active-low;
+		gpio = <&gpio3 RK_PD3 GPIO_ACTIVE_LOW>;
+		regulator-always-on;
+		regulator-boot-on;
+		pinctrl-0 = <&ant_1>;
+		pinctrl-names = "default";
+		regulator-name = "board_antenna";
+	};
 };
 
 &pinctrl {
-	bluetooth {
-		bt_reg_on_h: bt-reg-on-h {
-			rockchip,pins = <0 RK_PC1 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-
-		bt_wake_host_h: bt-wake-host-h {
-			rockchip,pins = <0 RK_PB3 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-
-		host_wake_bt_h: host-wake-bt-h {
-			rockchip,pins = <0 RK_PB4 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
 
 	wifi {
 		wifi_reg_on_h: wifi-reg-on-h {
@@ -50,6 +49,55 @@
 			rockchip,pins = <0 RK_PB7 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	sdmmc1 {
+		/omit-if-no-ref/
+		sdmmc1_bus4: sdmmc1-bus4 {
+			rockchip,pins =
+				/* sdmmc1_d0 */
+				<2 RK_PA3 1 &pcfg_pull_up_drv_level_12>,
+				/* sdmmc1_d1 */
+				<2 RK_PA4 1 &pcfg_pull_up_drv_level_12>,
+				/* sdmmc1_d2 */
+				<2 RK_PA5 1 &pcfg_pull_up_drv_level_12>,
+				/* sdmmc1_d3 */
+				<2 RK_PA6 1 &pcfg_pull_up_drv_level_12>;
+		};
+
+		/omit-if-no-ref/
+		sdmmc1_clk: sdmmc1-clk {
+			rockchip,pins =
+				/* sdmmc1_clk */
+				<2 RK_PB0 1 &pcfg_pull_up_drv_level_12>;
+		};
+
+		/omit-if-no-ref/
+		sdmmc1_cmd: sdmmc1-cmd {
+			rockchip,pins =
+				/* sdmmc1_cmd */
+				<2 RK_PA7 1 &pcfg_pull_up_drv_level_12>;
+		};
+
+		/omit-if-no-ref/
+		sdmmc1_det: sdmmc1-det {
+			rockchip,pins =
+				/* sdmmc1_det */
+				<2 RK_PB2 1 &pcfg_pull_up>;
+		};
+
+		/omit-if-no-ref/
+		sdmmc1_pwren: sdmmc1-pwren {
+			rockchip,pins =
+				/* sdmmc1_pwren */
+				<2 RK_PB1 1 &pcfg_pull_none>;
+		};
+	};
+
+	antenna {
+		ant_1: ant-1 {
+			rockchip,pins = <3 RK_PD3 RK_FUNC_GPIO &pcfg_pull_down>;
+		};
+	};
 };
 
 &sdhci {
@@ -68,6 +116,7 @@
 };
 
 &sdmmc1 {
+	max-frequency = <150000000>;
 	bus-width = <4>;
 	cap-sd-highspeed;
 	cap-sdio-irq;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc5/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts linux-6.16-rc5/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts
--- linux-6.16-rc5/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts	2025-07-06 23:10:26.000000000 +0200
+++ linux-6.16-rc5/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts	2025-07-12 14:09:03.818114773 +0200
@@ -15,31 +15,36 @@
 };
 
 &gmac1 {
+	phy-mode = "rgmii";
+	clock_in_out = "output";
+
+	snps,reset-gpio = <&gpio3 RK_PC0 GPIO_ACTIVE_LOW>;
+	snps,reset-active-low;
+	snps,reset-delays-us = <0 20000 100000>;
+
 	assigned-clocks = <&cru SCLK_GMAC1_RX_TX>, <&cru SCLK_GMAC1>;
-	assigned-clock-parents = <&cru SCLK_GMAC1_RGMII_SPEED>, <&cru CLK_MAC1_2TOP>;
-	clock_in_out = "input";
-	phy-handle = <&rgmii_phy1>;
-	phy-mode = "rgmii-id";
-	phy-supply = <&vcc_3v3>;
+	assigned-clock-parents = <&cru SCLK_GMAC1_RGMII_SPEED>;
+	assigned-clock-rates = <0>, <125000000>;
+
 	pinctrl-names = "default";
 	pinctrl-0 = <&gmac1m1_miim
-		     &gmac1m1_tx_bus2
-		     &gmac1m1_rx_bus2
-		     &gmac1m1_rgmii_clk
-		     &gmac1m1_rgmii_bus
-		     &gmac1m1_clkinout>;
+		    	 &gmac1m1_tx_bus2
+		     	 &gmac1m1_rx_bus2
+		     	 &gmac1m1_rgmii_clk
+		     	 &gmac1m1_rgmii_bus>;
+
+	tx_delay = <0x47>;
+	rx_delay = <0x27>;
+
+	phy-handle = <&rgmii_phy1>;
 	status = "okay";
 };
 
 &mdio1 {
-	rgmii_phy1: ethernet-phy@1 {
+	rgmii_phy1: phy@0 {
 		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <1>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&gmac1_rstn>;
-		reset-assert-us = <20000>;
-		reset-deassert-us = <50000>;
-		reset-gpios = <&gpio3 RK_PC0 GPIO_ACTIVE_LOW>;
+		reg = <0x0>;
+		status = "okay";
 	};
 };
 
