diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc2/drivers/mmc/host/Kconfig linux-6.16-rc2/drivers/mmc/host/Kconfig
--- linux-6.16-rc2/drivers/mmc/host/Kconfig	2025-06-15 22:49:41.000000000 +0200
+++ linux-6.16-rc2/drivers/mmc/host/Kconfig	2025-06-20 14:59:56.271360415 +0200
@@ -1055,7 +1055,9 @@
 	tristate "Broadcom SDIO/SD/MMC support"
 	depends on ARCH_BRCMSTB || ARCH_BCM2835 || BMIPS_GENERIC || COMPILE_TEST
 	depends on MMC_SDHCI_PLTFM
+	select MMC_SDHCI_IO_ACCESSORS
 	select MMC_CQHCI
+	select OF_DYNAMIC
 	default ARCH_BRCMSTB || BMIPS_GENERIC
 	help
 	  This selects support for the SDIO/SD/MMC Host Controller on
