diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm2711-rpi-4-b.dts linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm2711-rpi-4-b.dts
--- linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm2711-rpi-4-b.dts	2025-06-22 22:30:08.000000000 +0200
+++ linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm2711-rpi-4-b.dts	2025-06-26 12:39:49.851692237 +0200
@@ -158,10 +158,11 @@
 
 &leds {
 	led_pwr: led-pwr {
-		label = "PWR";
+		label = "DISK";
 		gpios = <&expgpio 2 GPIO_ACTIVE_LOW>;
-		default-state = "keep";
-		linux,default-trigger = "default-on";
+		default-state = "off";
+		function = LED_FUNCTION_DISK_ACTIVITY;
+		linux,default-trigger = "disk-activity";
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm283x-rpi-led-deprecated.dtsi linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm283x-rpi-led-deprecated.dtsi
--- linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm283x-rpi-led-deprecated.dtsi	2025-06-22 22:30:08.000000000 +0200
+++ linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm283x-rpi-led-deprecated.dtsi	2025-06-26 12:44:30.908352852 +0200
@@ -1,5 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0
 
+#include <dt-bindings/leds/common.h>
+
 / {
 	/*
 	 * This file provides the now deprecated ACT LED to the
@@ -12,6 +14,7 @@
 		led_act: led-act {
 			label = "ACT";
 			default-state = "keep";
+			function = LED_FUNCTION_STATUS;
 			linux,default-trigger = "heartbeat";
 		};
 	};
