diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712.dtsi linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
--- linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712.dtsi	2025-06-25 13:58:16.054763728 +0200
+++ linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712.dtsi	2025-06-25 13:55:51.834766835 +0200
@@ -650,6 +650,20 @@
 				     <GIC_SPI 249 IRQ_TYPE_LEVEL_HIGH>;
 			status = "okay";
 		};
+
+		rpivid: codec@800000 {
+			compatible = "raspberrypi,rpivid-vid-decoder";
+			reg = <0x10 0x00800000  0x0 0x10000>, /* HEVC */
+			      <0x10 0x00840000  0x0 0x1000>;  /* INTC */
+			reg-names = "hevc","intc";
+
+			interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
+
+			clocks = <&firmware_clocks 11>;
+			clock-names = "hevc";
+			//iommus = <&iommu2>;
+			status = "okay";
+		};
 	};
 
 	timer {
