diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm2711.dtsi linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm2711.dtsi
--- linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm2711.dtsi	2025-06-26 09:46:54.558286413 +0200
+++ linux-6.16-rc3/arch/arm/boot/dts/broadcom/bcm2711.dtsi	2025-06-26 08:49:24.488360705 +0200
@@ -1153,6 +1153,10 @@
 };
 
 &cma {
+	compatible = "shared-dma-pool";
+	size = <0x18000000>; /* 384MB */
+	reusable;
+	linux,cma-default;
 	/*
 	 * arm64 reserves the CMA by default somewhere in ZONE_DMA32,
 	 * that's not good enough for the BCM2711 as some devices can
