diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
--- linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts	2025-06-26 09:46:54.604953079 +0200
+++ linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts	2025-06-26 09:57:43.241605790 +0200
@@ -25,6 +25,24 @@
 		reg = <0 0 0 0x28000000>;
 	};
 
+	leds: leds {
+		compatible = "gpio-leds";
+
+		led_pwr: led-pwr {
+			label = "DISK";
+			gpios = <&rp1_gpio 44 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+			linux,default-trigger = "disk-activity";
+		};
+
+		led_act: led-act {
+			label = "ACT";
+			gpios = <&gio_aon 9 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+			linux,default-trigger = "heartbeat";
+		};
+	};
+
 	clocks: clocks {
 		compatible = "simple-bus";
 		#address-cells = <1>;
