diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc7/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts linux-6.16-rc7/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
--- linux-6.16-rc7/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts	2025-07-21 16:42:03.183347028 +0200
+++ linux-6.16-rc7/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts	2025-07-21 16:36:48.970012760 +0200
@@ -43,6 +43,11 @@
 		};
 	};
 
+	usbc_vin2 {
+		compatible = "iio-hwmon";
+		status = "okay";
+	};
+
 	clocks: clocks {
 		compatible = "simple-bus";
 		#address-cells = <1>;
@@ -389,6 +395,10 @@
 	pinctrl-names = "default";
 };
 
+&rp1_adc {
+	status = "okay";
+};
+
 &thermal_trips {
 	cpu_tepid: cpu-tepid {
 		temperature = <50000>;
