diff --speed-large-files --no-dereference --minimal -Naur linux-6.15-rc5/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi linux-6.15-rc5/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi
--- linux-6.15-rc5/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi	2025-05-04 22:55:04.000000000 +0200
+++ linux-6.15-rc5/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi	2025-05-06 16:35:28.177784342 +0200
@@ -14,8 +14,8 @@
 
 / {
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
@@ -85,15 +85,15 @@
 
 		led_blue_pwm: led-1 {
 			color = <LED_COLOR_ID_BLUE>;
-			function = LED_FUNCTION_STATUS;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "disk-activity";
 			max-brightness = <255>;
-			status = "disabled";
 		};
 
 		led_green_pwm: led-2 {
 			color = <LED_COLOR_ID_GREEN>;
-			function = LED_FUNCTION_INDICATOR;
+			function = LED_FUNCTION_HEARTBEAT;
+			linux,default-trigger = "heartbeat";
 			function-enumerator = <2>;
 			max-brightness = <255>;
 		};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.15-rc5/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts linux-6.15-rc5/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
--- linux-6.15-rc5/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2025-05-04 22:55:04.000000000 +0200
+++ linux-6.15-rc5/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2025-05-06 16:35:45.064450640 +0200
@@ -42,6 +42,7 @@
 		gpios = <&gpio4 RK_PB3 GPIO_ACTIVE_LOW>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&ir_receiver_pin>;
+		linux,rc-map-name = "rc-beelink-gs1";
 	};
 
 	rfkill {
