diff --speed-large-files --no-dereference --minimal -Naur linux-6.14-rc5/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts linux-6.14-rc5/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
--- linux-6.14-rc5/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2025-03-02 20:48:20.000000000 +0100
+++ linux-6.14-rc5/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2025-03-09 20:42:19.676698146 +0100
@@ -14,8 +14,8 @@
 
 	aliases {
 		ethernet0 = &gmac1;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	analog-sound {
@@ -47,6 +47,17 @@
 		};
 	};
 
+	dp-con {
+		compatible = "dp-connector";
+		type = "full-size"; // HDMI Type-D via RA620
+
+		port {
+			dp_con_in: endpoint {
+				remote-endpoint = <&dp0_out_con>;
+			};
+		};
+	};
+
 	leds {
 		compatible = "gpio-leds";
 		pinctrl-names = "default";
@@ -54,17 +65,17 @@
 
 		io-led {
 			color = <LED_COLOR_ID_BLUE>;
-			function = LED_FUNCTION_STATUS;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "disk-activity";
 		};
 	};
 
 	fan: pwm-fan {
 		compatible = "pwm-fan";
-		cooling-levels = <0 95 145 195 255>;
+		cooling-levels = <0 96 128 164 196 224 255>;
 		fan-supply = <&vcc_5v0>;
-		pwms = <&pwm3 0 50000 0>;
+		pwms = <&pwm3 0 10000 0>;
 		#cooling-cells = <2>;
 	};
 
@@ -298,6 +309,10 @@
 	};
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
 &gmac1 {
 	clock_in_out = "output";
 	phy-handle = <&rgmii_phy1>;
@@ -313,6 +328,10 @@
 	status = "okay";
 };
 
+&hdmi0_sound {
+	status = "okay";
+};
+
 &hdmi0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&hdmim0_tx0_cec
@@ -338,6 +357,54 @@
 	status = "okay";
 };
 
+&dp0 {
+	status = "okay";
+	pinctrl-0 = <&dp0m1_pins>;
+	pinctrl-names = "default";
+};
+
+&dp0_in {
+	dp0_in_vp2: endpoint {
+		remote-endpoint = <&vp2_out_dp0>;
+	};
+};
+
+&dp0_out {
+	dp0_out_con: endpoint {
+		remote-endpoint = <&dp_con_in>;
+	};
+};
+
+&package_thermal {
+	polling-delay = <1000>;
+
+	trips {
+		package_fan0: package-fan0 {
+			temperature = <55000>;
+			hysteresis = <2000>;
+			type = "active";
+		};
+
+		package_fan1: package-fan1 {
+			temperature = <65000>;
+			hysteresis = <2000>;
+			type = "active";
+		};
+	};
+
+	cooling-maps {
+		map0 {
+			trip = <&package_fan0>;
+			cooling-device = <&fan THERMAL_NO_LIMIT 1>;
+		};
+
+		map1 {
+			trip = <&package_fan1>;
+			cooling-device = <&fan 2 THERMAL_NO_LIMIT>;
+		};
+	};
+};
+
 &mdio1 {
 	rgmii_phy1: ethernet-phy@1 {
 		/* RTL8211F */
@@ -845,3 +912,10 @@
 		remote-endpoint = <&hdmi0_in_vp0>;
 	};
 };
+
+&vp2 {
+	vp2_out_dp0: endpoint@a {
+		reg = <ROCKCHIP_VOP2_EP_DP0>;
+		remote-endpoint = <&dp0_in_vp2>;
+	};
+};
