diff --speed-large-files --no-dereference --minimal -Naur linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dtsi linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dtsi
--- linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dtsi	2025-02-03 00:39:26.000000000 +0100
+++ linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dtsi	2025-02-09 14:02:54.923363555 +0100
@@ -14,6 +14,7 @@
 	aliases {
 		ethernet0 = &gmac1;
 		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
@@ -83,8 +84,8 @@
 
 		led {
 			color = <LED_COLOR_ID_GREEN>;
-			function = LED_FUNCTION_STATUS;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "disk-activity";
 			max-brightness = <255>;
 			pwms = <&pwm0 0 25000 0>;
 		};
@@ -181,6 +182,10 @@
 	status = "okay";
 };
 
+&hdmi0_sound {
+	status = "okay";
+};
+
 &hdmi0 {
 	status = "okay";
 };
@@ -355,6 +360,10 @@
 	status = "okay";
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
 &mdio1 {
 	rgmii_phy1: ethernet-phy@1 {
 		compatible = "ethernet-phy-ieee802.3-c22";
@@ -827,7 +836,7 @@
 };
 
 &usb_host0_xhci {
-	dr_mode = "otg";
+	dr_mode = "host";
 	usb-role-switch;
 	status = "okay";
 
