diff --speed-large-files --no-dereference --minimal -Naur linux-6.16/drivers/clk/sunxi-ng/ccu-sun55i-a523.c linux-6.16/drivers/clk/sunxi-ng/ccu-sun55i-a523.c
--- linux-6.16/drivers/clk/sunxi-ng/ccu-sun55i-a523.c	2025-07-27 23:26:38.000000000 +0200
+++ linux-6.16/drivers/clk/sunxi-ng/ccu-sun55i-a523.c	2025-07-30 16:41:36.813334390 +0200
@@ -167,6 +167,7 @@
 	.n		= _SUNXI_CCU_MULT_MIN(8, 8, 11),
 	.m		= _SUNXI_CCU_DIV(1, 1), /* input divider */
 	.p		= _SUNXI_CCU_DIV(0, 1), /* output divider */
+	.max_rate	= 2520000000,
 	.common		= {
 		.reg		= 0x030,
 		.hw.init	= CLK_HW_INIT_PARENTS_DATA("pll-gpu", osc24M,
@@ -248,6 +249,7 @@
 	.n		= _SUNXI_CCU_MULT_MIN(8, 8, 11),
 	.m		= _SUNXI_CCU_DIV(1, 1), /* input divider */
 	.p		= _SUNXI_CCU_DIV(0, 1), /* output divider */
+	.max_rate	= 2520000000,
 	.common		= {
 		.reg		= 0x058,
 		.hw.init	= CLK_HW_INIT_PARENTS_DATA("pll-ve", osc24M,
@@ -321,6 +323,8 @@
 	.lock		= BIT(28),
 	.n		= _SUNXI_CCU_MULT_MIN(8, 8, 11),
 	.m		= _SUNXI_CCU_DIV(1, 1),	/* input divider */
+	.min_rate = 1200000000,
+	.max_rate = 2520000000,
 	.common		= {
 		.reg		= 0x0080,
 		.hw.init	= CLK_HW_INIT_PARENTS_DATA("pll-npu-4x",
@@ -439,9 +443,9 @@
 
 static SUNXI_CCU_GATE_HWS(bus_g2d_clk, "bus-g2d", ahb_hws, 0x63c, BIT(0), 0);
 
+/* Delete the pll-periph0-800m. If GPU use pll-periph0-800m, gpu will occur job fault */
 static const struct clk_hw *gpu_parents[] = {
 	&pll_gpu_clk.common.hw,
-	&pll_periph0_800M_clk.common.hw,
 	&pll_periph0_600M_clk.hw,
 	&pll_periph0_400M_clk.hw,
 	&pll_periph0_300M_clk.hw,
@@ -1173,6 +1177,39 @@
 				    BIT(23),	/* gate */
 				    0);
 
+static SUNXI_CCU_GATE(usb3_mbus_gate_clk, "usb3-mbus-gate",
+		"osc24M",
+		0x0804, BIT(6), 0);
+
+static SUNXI_CCU_GATE(usb2_ref_clk, "usb2_ref",
+		"osc24M",
+		0x0A80, BIT(31), 0);
+
+static const struct clk_parent_data usb3_ref_parents[] = { { .fw_name = "hosc" }, { .hw = &pll_periph0_200M_clk.hw }, { .hw = &pll_periph1_200M_clk.hw } };
+static SUNXI_CCU_M_DATA_WITH_MUX_GATE(usb3_ref_clk, "usb3-ref",
+		usb3_ref_parents, 0x0A84,
+		0, 5,	/* M */
+		24, 3,	/* mux */
+		BIT(31),	/* gate */
+		0);
+
+static const struct clk_parent_data usb3_suspend_parents[] = { { .fw_name = "losc" }, { .fw_name = "hosc" } };
+static SUNXI_CCU_M_DATA_WITH_MUX_GATE(usb3_suspend_clk, "usb3-suspend",
+		usb3_suspend_parents, 0x0A88,
+		0, 5,	/* M */
+		24, 1,	/* mux */
+		BIT(31),	/* gate */
+		0);
+
+static const struct clk_parent_data npu_parents[] = { { .hw = &pll_periph0_480M_clk.common.hw }, { .hw = &pll_periph0_600M_clk.hw }, { .hw = &pll_periph0_800M_clk.common.hw }, { .hw = &pll_npu_2x_clk.hw } };
+
+static SUNXI_CCU_M_DATA_WITH_MUX_GATE(npu_clk, "npu",
+		npu_parents, 0x06E0,
+		0, 5,	/* M */
+		24, 3,	/* mux */
+		BIT(31),	/* gate */
+		CLK_SET_RATE_PARENT);
+
 /*
  * Contains all clocks that are controlled by a hardware register. They
  * have a (sunxi) .common member, which needs to be initialised by the common
@@ -1340,6 +1377,11 @@
 	&fanout0_clk.common,
 	&fanout1_clk.common,
 	&fanout2_clk.common,
+	&npu_clk.common,
+	&usb3_mbus_gate_clk.common,
+	&usb3_ref_clk.common,
+	&usb2_ref_clk.common,
+	&usb3_suspend_clk.common,
 };
 
 static struct clk_hw_onecell_data sun55i_a523_hw_clks = {
@@ -1524,6 +1566,11 @@
 		[CLK_FANOUT0]		= &fanout0_clk.common.hw,
 		[CLK_FANOUT1]		= &fanout1_clk.common.hw,
 		[CLK_FANOUT2]		= &fanout2_clk.common.hw,
+		[CLK_NPU]		= &npu_clk.common.hw,
+		[CLK_USB3_MBUS_GATE]	= &usb3_mbus_gate_clk.common.hw,
+		[CLK_USB3_REF]		= &usb3_ref_clk.common.hw,
+		[CLK_USB2_REF]		= &usb2_ref_clk.common.hw,
+		[CLK_USB3_SUSPEND]	= &usb3_suspend_clk.common.hw,
 	},
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16/drivers/clk/sunxi-ng/ccu-sun55i-a523.h linux-6.16/drivers/clk/sunxi-ng/ccu-sun55i-a523.h
--- linux-6.16/drivers/clk/sunxi-ng/ccu-sun55i-a523.h	2025-07-27 23:26:38.000000000 +0200
+++ linux-6.16/drivers/clk/sunxi-ng/ccu-sun55i-a523.h	2025-07-30 16:41:36.813334390 +0200
@@ -9,6 +9,6 @@
 #include <dt-bindings/clock/sun55i-a523-ccu.h>
 #include <dt-bindings/reset/sun55i-a523-ccu.h>
 
-#define CLK_NUMBER	(CLK_FANOUT2 + 1)
+#define CLK_NUMBER	(CLK_USB3_SUSPEND + 1)
 
 #endif /* _CCU_SUN55I_A523_H */
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16/include/dt-bindings/clock/sun55i-a523-ccu.h linux-6.16/include/dt-bindings/clock/sun55i-a523-ccu.h
--- linux-6.16/include/dt-bindings/clock/sun55i-a523-ccu.h	2025-07-27 23:26:38.000000000 +0200
+++ linux-6.16/include/dt-bindings/clock/sun55i-a523-ccu.h	2025-07-30 16:41:58.916667789 +0200
@@ -185,5 +185,10 @@
 #define CLK_FANOUT0		176
 #define CLK_FANOUT1		177
 #define CLK_FANOUT2		178
+#define CLK_NPU			179
+#define CLK_USB3_MBUS_GATE	180
+#define CLK_USB3_REF		181
+#define CLK_USB2_REF		182
+#define CLK_USB3_SUSPEND	183
 
 #endif /* _DT_BINDINGS_CLK_SUN55I_A523_CCU_H_ */
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16/include/dt-bindings/clock/sun55iw3-mcu-ccu.h linux-6.16/include/dt-bindings/clock/sun55iw3-mcu-ccu.h
--- linux-6.16/include/dt-bindings/clock/sun55iw3-mcu-ccu.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.16/include/dt-bindings/clock/sun55iw3-mcu-ccu.h	2025-07-30 16:41:58.916667789 +0200
@@ -0,0 +1,58 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright(c) 2020 - 2023 Allwinner Technology Co.,Ltd. All rights reserved. */
+/*
+ * Copyright (c) 2022 rengaomin@allwinnertech.com
+ */
+
+#ifndef _DT_BINDINGS_CLK_SUN55IW3_MCU_CCU_H_
+#define _DT_BINDINGS_CLK_SUN55IW3_MCU_CCU_H_
+
+#define CLK_PLL_MCU_AUDIO1	0
+#define CLK_PLL_MCU_AUDIO1_DIV2	1
+#define CLK_PLL_MCU_AUDIO1_DIV5	2
+#define CLK_PLL_MCU_AUDIO_OUT	3
+#define CLK_DSP_DSP		4
+#define CLK_MCU_I2S0		5
+#define CLK_MCU_I2S1		6
+#define CLK_MCU_I2S2		7
+#define CLK_MCU_I2S3		8
+#define CLK_MCU_I2S3_ASRC	9
+#define CLK_BUS_MCU_I2S0	10
+#define CLK_BUS_MCU_I2S1	11
+#define CLK_BUS_MCU_I2S2	12
+#define CLK_BUS_MCU_I2S3	13
+#define CLK_MCU_OWA_TX		14
+#define CLK_MCU_OWA_RX		15
+#define CLK_BUS_MCU_OWA		16
+#define CLK_MCU_DMIC		17
+#define CLK_BUS_MCU_DMIC	18
+#define CLK_MCU_AUDIO_CODEC_DAC	19
+#define CLK_MCU_AUDIO_CODEC_ADC	20
+#define CLK_BUS_MCU_AUDIO_CODEC	21
+#define CLK_BUS_DSP_MSG		22
+#define CLK_BUS_DSP_CFG		23
+#define CLK_BUS_MCU_NPU_ACLK	24
+#define CLK_BUS_MCU_NPU_HCLK	25
+#define CLK_MCU_TIMER0		27
+#define CLK_MCU_TIMER1		28
+#define CLK_MCU_TIMER2		29
+#define CLK_MCU_TIMER3		30
+#define CLK_MCU_TIMER4		31
+#define CLK_MCU_TIMER5		32
+#define CLK_BUS_MCU_TIMER	33
+#define CLK_BUS_MCU_DMA		34
+#define CLK_BUS_MCU_TZMA0	35
+#define CLK_BUS_MCU_TZMA1	36
+#define CLK_BUS_PUBSRAM		37
+#define CLK_BUS_MCU_MBUS	38
+#define CLK_BUS_MCU_DMA_MBUS	39
+#define CLK_BUS_RV		40
+#define CLK_BUS_RV_CFG		41
+#define CLK_BUS_MCU_RISCV_MSG	42
+#define CLK_MCU_PWM		43
+#define CLK_BUS_MCU_PWM		44
+#define CLK_BUS_MCU_AHB_AUTO	45
+
+#define CLK_MCU_NUMBER		(CLK_BUS_MCU_AHB_AUTO + 1)
+
+#endif /* _DT_BINDINGS_CLK_SUN55IW3_MCU_CCU_H_ */
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16/include/dt-bindings/power/a523-power.h linux-6.16/include/dt-bindings/power/a523-power.h
--- linux-6.16/include/dt-bindings/power/a523-power.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.16/include/dt-bindings/power/a523-power.h	2025-07-30 16:41:58.920001122 +0200
@@ -0,0 +1,18 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+#ifndef __DT_BINDINGS_POWER_A523_H__
+#define __DT_BINDINGS_POWER_A523_H__
+
+#define A523_PD_DSP			0
+#define A523_PD_NPU			1
+#define A523_PD_AUDIO		2
+#define A523_PD_SRAM		3
+#define A523_PD_RISCV		4
+#define A523_PCK_VE			0
+#define A523_PCK_GPU		1
+#define A523_PCK_VI			2
+#define A523_PCK_VO0		3
+#define A523_PCK_VO1		4
+#define A523_PCK_DE			5
+#define A523_PCK_NAND		6
+#define A523_PCK_PCIE		7
+#endif
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16/include/dt-bindings/reset/sun55i-a523-ccu.h linux-6.16/include/dt-bindings/reset/sun55i-a523-ccu.h
--- linux-6.16/include/dt-bindings/reset/sun55i-a523-ccu.h	2025-07-27 23:26:38.000000000 +0200
+++ linux-6.16/include/dt-bindings/reset/sun55i-a523-ccu.h	2025-07-30 16:41:58.920001122 +0200
@@ -84,5 +84,6 @@
 #define RST_BUS_LEDC		75
 #define RST_BUS_CSI		76
 #define RST_BUS_ISP		77
+#define RST_BUS_USB_3		78
 
 #endif /* _DT_BINDINGS_RST_SUN55I_A523_CCU_H_ */
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16/include/dt-bindings/reset/sun55iw3-mcu-ccu.h linux-6.16/include/dt-bindings/reset/sun55iw3-mcu-ccu.h
--- linux-6.16/include/dt-bindings/reset/sun55iw3-mcu-ccu.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.16/include/dt-bindings/reset/sun55iw3-mcu-ccu.h	2025-07-30 16:41:58.920001122 +0200
@@ -0,0 +1,31 @@
+/* SPDX-License-Identifier: (GPL-2.0+ or MIT) */
+/* Copyright(c) 2020 - 2023 Allwinner Technology Co.,Ltd. All rights reserved. */
+/*
+ * Copyright (C) 2022 rengaomin@allwinnertech.com
+ */
+
+#ifndef _DT_BINDINGS_RST_SUN55IW3_MCU_CCU_H_
+#define _DT_BINDINGS_RST_SUN55IW3_MCU_CCU_H_
+
+#define RST_BUS_MCU_I2S3	0
+#define RST_BUS_MCU_I2S2	1
+#define RST_BUS_MCU_I2S1	2
+#define RST_BUS_MCU_I2S0	3
+#define RST_BUS_MCU_OWA		4
+#define RST_BUS_MCU_DMIC	5
+#define RST_BUS_MCU_AUDIO_CODEC	6
+#define RST_BUS_DSP_MSG		7
+#define RST_BUS_DSP_CFG		8
+#define RST_BUS_MCU_NPU		9
+#define RST_BUS_MCU_TIME	10
+#define RST_BUS_DSP		11
+#define RST_BUS_DSP_DBG		12
+#define RST_BUS_MCU_DMA		13
+#define RST_BUS_PUBSRAM		14
+#define RST_BUS_RV		15
+#define RST_BUS_RV_DBG		16
+#define RST_BUS_RV_CFG		17
+#define RST_BUS_MCU_RV_MSG	18
+#define RST_BUS_MCU_PWM		19
+
+#endif /* _DT_BINDINGS_RST_SUN55IW3_MCU_CCU_H_ */
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16/include/dt-bindings/spi/sunxi-spi.h linux-6.16/include/dt-bindings/spi/sunxi-spi.h
--- linux-6.16/include/dt-bindings/spi/sunxi-spi.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.16/include/dt-bindings/spi/sunxi-spi.h	2025-07-30 16:41:58.920001122 +0200
@@ -0,0 +1,18 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+/* Copyright(c) 2020 - 2023 Allwinner Technology Co.,Ltd. All rights reserved. */
+
+#ifndef __DT_SUNXI_SPI_H
+#define __DT_SUNXI_SPI_H
+
+#define SUNXI_SPI_BUS_MASTER	(1 << 0)
+#define SUNXI_SPI_BUS_SLAVE		(1 << 1)
+#define SUNXI_SPI_BUS_DBI		(1 << 2)
+#define SUNXI_SPI_BUS_BIT		(1 << 3)
+#define SUNXI_SPI_BUS_NOR		(1 << 4)
+#define SUNXI_SPI_BUS_NAND		(1 << 5)
+#define SUNXI_SPI_BUS_CAMERA	(1 << 6)
+
+#define SUNXI_SPI_CS_AUTO	0
+#define SUNXI_SPI_CS_SOFT	1
+
+#endif /* __DT_SUNXI_SPI_H */
