From b418acefec3c673c087afb511f0fe6f156a8928c Mon Sep 17 00:00:00 2001
From: iuncuim <iuncuim@gmail.com>
Date: Wed, 27 Aug 2025 16:41:46 +0800
Subject: [PATCH] DRAFT: amr64: a523: make emmc work

Signed-off-by: Mikhail Kalashnikov <iuncuim@gmail.com>
diff --git a/drivers/clk/sunxi-ng/ccu-sun55i-a523.c b/drivers/clk/sunxi-ng/ccu-sun55i-a523.c
index 7bf41e628..216472f85 100644
--- a/drivers/clk/sunxi-ng/ccu-sun55i-a523.c
+++ b/drivers/clk/sunxi-ng/ccu-sun55i-a523.c
@@ -119,7 +119,7 @@ static const struct clk_hw *pll_periph0_150M_hws[] = {
 
 #define SUN55I_A523_PLL_PERIPH1_REG	0x028
 static struct ccu_nm pll_periph1_4x_clk = {
-	.enable		= BIT(27),
+	.enable		= BIT(27) | BIT(30) | BIT(31),
 	.lock		= BIT(28),
 	.n		= _SUNXI_CCU_MULT_MIN(8, 8, 11),
 	.m		= _SUNXI_CCU_DIV(1, 1), /* input divider */
@@ -127,7 +127,8 @@ static struct ccu_nm pll_periph1_4x_clk = {
 		.reg		= 0x028,
 		.hw.init	= CLK_HW_INIT_PARENTS_DATA("pll-periph1-4x",
 							   osc24M, &ccu_nm_ops,
-							   CLK_SET_RATE_GATE),
+							   CLK_SET_RATE_UNGATE |
+							   CLK_IS_CRITICAL),
 	},
 };
 
@@ -664,10 +665,10 @@ static SUNXI_CCU_MP_MUX_GATE_POSTDIV_DUALDIV(mmc1_clk, "mmc1", nand_mmc_parents,
 
 static const struct clk_parent_data mmc2_parents[] = {
 	{ .fw_name = "hosc" },
+	{ .hw = &pll_periph1_600M_clk.hw },
 	{ .hw = &pll_periph0_800M_clk.common.hw },
 	{ .hw = &pll_periph0_600M_clk.hw },
 	{ .hw = &pll_periph1_800M_clk.common.hw },
-	{ .hw = &pll_periph1_600M_clk.hw },
 };
 
 static SUNXI_CCU_MP_MUX_GATE_POSTDIV_DUALDIV(mmc2_clk, "mmc2", mmc2_parents,
-- 
2.50.1

