diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc4/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi linux-6.16-rc4/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
--- linux-6.16-rc4/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi	2025-07-06 19:24:03.074765062 +0200
+++ linux-6.16-rc4/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi	2025-07-06 19:23:38.624765594 +0200
@@ -183,6 +183,32 @@
 				bias-pull-up;
 			};
 
+			rgmii0_25m_pin: rgmii0-25m-pin {
+				pins = "PH13";
+				allwinner,pinmux = <5>;
+				function = "gmac0";
+				drive-strength = <40>;
+				bias-disable;
+			};
+
+			rgmii1_pins: rgmii1-pins {
+				pins = "PJ0", "PJ1", "PJ2", "PJ3", "PJ4",
+				       "PJ5", "PJ6", "PJ7", "PJ8", "PJ9",
+				       "PJ11", "PJ12", "PJ13", "PJ14", "PJ15";
+				allwinner,pinmux = <5>;
+				function = "gmac1";
+				drive-strength = <40>;
+				bias-disable;
+			};
+
+			rgmii1_25m_pin: rgmii1-25m-pin {
+				pins = "PJ10";
+				allwinner,pinmux = <5>;
+				function = "gmac1";
+				drive-strength = <40>;
+				bias-disable;
+			};
+
 			uart0_pb_pins: uart0-pb-pins {
 				pins = "PB9", "PB10";
 				allwinner,pinmux = <2>;
@@ -595,12 +621,12 @@
 			status = "disabled";
 		};
 
-		emac0: ethernet@4500000 {
-			compatible = "allwinner,sun55i-a523-emac0",
+		gmac0: ethernet@4500000 {
+			compatible = "allwinner,sun55i-a523-gmac0",
 				     "allwinner,sun50i-a64-emac";
 			reg = <0x04500000 0x10000>;
-			clocks = <&ccu CLK_BUS_EMAC0>;
-			clock-names = "stmmaceth";
+			clocks = <&ccu CLK_BUS_EMAC0>, <&ccu CLK_EMAC0_25M>;
+			clock-names = "stmmaceth", "emac25m";
 			resets = <&ccu RST_BUS_EMAC0>;
 			reset-names = "stmmaceth";
 			interrupts = <GIC_SPI 46 IRQ_TYPE_LEVEL_HIGH>;
@@ -617,6 +643,52 @@
 			};
 		};
 
+		gmac1: ethernet@4510000 {
+			compatible = "allwinner,sun55i-a523-gmac200",
+				     "snps,dwmac-4.20a";
+			reg = <0x04510000 0x10000>;
+			clocks = <&ccu CLK_BUS_EMAC1>, <&ccu CLK_MBUS_EMAC1>,
+				 <&ccu CLK_EMAC1_25M>;
+			clock-names = "stmmaceth", "mbus", "emac25m";
+			resets = <&ccu RST_BUS_EMAC1>;
+			reset-names = "stmmaceth";
+			interrupts = <GIC_SPI 47 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "macirq";
+			pinctrl-names = "default";
+			pinctrl-0 = <&rgmii1_pins>;
+			power-domains = <&pck600 PD_VO1>;
+			syscon = <&syscon>;
+			snps,fixed-burst;
+			snps,axi-config = <&gmac1_stmmac_axi_setup>;
+			snps,mtl-rx-config = <&gmac1_mtl_rx_setup>;
+			snps,mtl-tx-config = <&gmac1_mtl_tx_setup>;
+			status = "disabled";
+
+			mdio1: mdio {
+				compatible = "snps,dwmac-mdio";
+				#address-cells = <1>;
+				#size-cells = <0>;
+			};
+
+			gmac1_mtl_rx_setup: rx-queues-config {
+				snps,rx-queues-to-use = <1>;
+
+				queue0 {};
+			};
+
+			gmac1_stmmac_axi_setup: stmmac-axi-config {
+				snps,wr_osr_lmt = <0xf>;
+				snps,rd_osr_lmt = <0xf>;
+				snps,blen = <256 128 64 32 16 8 4>;
+			};
+
+			gmac1_mtl_tx_setup: tx_queues-config {
+				snps,tx-queues-to-use = <1>;
+
+				queue0 {};
+			};
+		};
+
 		ppu: power-controller@7001400 {
 			compatible = "allwinner,sun55i-a523-ppu";
 			reg = <0x07001400 0x400>;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc4/arch/arm64/boot/dts/allwinner/sun55i-t527-avaota-a1.dts linux-6.16-rc4/arch/arm64/boot/dts/allwinner/sun55i-t527-avaota-a1.dts
--- linux-6.16-rc4/arch/arm64/boot/dts/allwinner/sun55i-t527-avaota-a1.dts	2025-06-29 22:09:04.000000000 +0200
+++ linux-6.16-rc4/arch/arm64/boot/dts/allwinner/sun55i-t527-avaota-a1.dts	2025-07-06 20:06:50.698043063 +0200
@@ -12,7 +12,7 @@
 	compatible = "yuzukihd,avaota-a1", "allwinner,sun55i-t527";
 
 	aliases {
-		ethernet0 = &emac0;
+		ethernet0 = &gmac0;
 		serial0 = &uart0;
 	};
 
@@ -65,7 +65,7 @@
 	status = "okay";
 };
 
-&emac0 {
+&gmac0 {
 	phy-mode = "rgmii-id";
 	phy-handle = <&ext_rgmii_phy>;
 	phy-supply = <&reg_dcdc4>;
