diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc5/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi linux-6.16-rc5/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
--- linux-6.16-rc5/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi	2025-07-08 10:22:13.248275174 +0200
+++ linux-6.16-rc5/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi	2025-07-08 10:20:58.914943435 +0200
@@ -785,6 +785,21 @@
 			clock-names = "bus", "hosc", "ahb";
 			#clock-cells = <1>;
 		};
+
+		gpu: gpu@1800000 {
+			compatible = "allwinner,sun55i-a523-mali",
+				     "arm,mali-valhall-jm";
+			reg = <0x1800000 0x10000>;
+			interrupts = <GIC_SPI 117 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 118 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 119 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "job", "mmu", "gpu";
+			clocks = <&ccu CLK_GPU>, <&ccu CLK_BUS_GPU>;
+			clock-names = "core", "bus";
+			power-domains = <&pck600 PD_GPU>;
+			resets = <&ccu RST_BUS_GPU>;
+			status = "disabled";
+		};
 	};
 
 	thermal-zones {
