diff --speed-large-files --no-dereference --minimal -Naur linux-6.15/arch/arm64/boot/dts/rockchip/rk3576.dtsi linux-6.15/arch/arm64/boot/dts/rockchip/rk3576.dtsi
--- linux-6.15/arch/arm64/boot/dts/rockchip/rk3576.dtsi	2025-06-04 15:38:18.124568490 +0200
+++ linux-6.15/arch/arm64/boot/dts/rockchip/rk3576.dtsi	2025-06-04 17:49:28.687732032 +0200
@@ -1592,6 +1626,22 @@
 			status = "disabled";
 		};
 
+		sdio: mmc@2a320000 {
+			compatible = "rockchip,rk3576-dw-mshc";
+			reg = <0x0 0x2a320000 0x0 0x4000>;
+			clocks = <&cru HCLK_SDIO>, <&cru CCLK_SRC_SDIO>;
+			clock-names = "biu", "ciu";
+			fifo-depth = <0x100>;
+			interrupts = <GIC_SPI 252 IRQ_TYPE_LEVEL_HIGH>;
+			max-frequency = <200000000>;
+			pinctrl-0 = <&sdmmc1m0_clk &sdmmc1m0_cmd &sdmmc1m0_bus4>;
+			pinctrl-names = "default";
+			power-domains = <&power RK3576_PD_SDGMAC>;
+			resets = <&cru SRST_H_SDIO>;
+			reset-names = "reset";
+			status = "disabled";
+		};
+
 		sdhci: mmc@2a330000 {
 			compatible = "rockchip,rk3576-dwcmshc", "rockchip,rk3588-dwcmshc";
 			reg = <0x0 0x2a330000 0x0 0x10000>;
