diff --speed-large-files --no-dereference --minimal -Naur linux_openvfd-766c6a948836bbf350bd90917fc925079e297e6e/driver/openvfd_drv.c linux_openvfd-766c6a948836bbf350bd90917fc925079e297e6e/driver/openvfd_drv.c
--- linux_openvfd-766c6a948836bbf350bd90917fc925079e297e6e/driver/openvfd_drv.c	2022-01-27 23:09:01.000000000 +0100
+++ linux_openvfd-766c6a948836bbf350bd90917fc925079e297e6e/driver/openvfd_drv.c	2022-03-01 10:52:38.306561225 +0100
@@ -622,11 +622,11 @@
 	int pin = -1;
 	struct gpio_chip *chip;
 	const char *bank_name = vfd_gpio_chip_name;
-	if (!bank_name && gpio[0] < 6) {
+	if (!bank_name && gpio[0] < 7) {
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4,0,0)
-		const char *pin_banks[] = { "banks", "ao-bank", "gpio0", "gpio1", "gpio2", "gpio3" };
+		const char *pin_banks[] = { "banks", "ao-bank", "gpio0", "gpio1", "gpio2", "gpio3", "gpio4" };
 #else
-		const char *pin_banks[] = { "periphs-banks", "aobus-banks", "gpio0", "gpio1", "gpio2", "gpio3" };
+		const char *pin_banks[] = { "periphs-banks", "aobus-banks", "gpio0", "gpio1", "gpio2", "gpio3", "gpio4" };
 #endif
 		bank_name = pin_banks[gpio[0]];
 	}
