diff --speed-large-files --no-dereference --minimal -Naur linux_openvfd-766c6a948836bbf350bd90917fc925079e297e6e/driver/openvfd_drv.c linux_openvfd-766c6a948836bbf350bd90917fc925079e297e6e/driver/openvfd_drv.c
--- linux_openvfd-766c6a948836bbf350bd90917fc925079e297e6e/driver/openvfd_drv.c	2022-03-07 14:56:25.822957036 +0100
+++ linux_openvfd-766c6a948836bbf350bd90917fc925079e297e6e/driver/openvfd_drv.c	2022-03-07 14:52:51.232960910 +0100
@@ -790,7 +790,7 @@
 	if (!verify_module_params(pdata->dev)) {
 		int i;
 		__u8 j;
-		pr_error("Failed to verify VFD configuration file, attempt using device tree as fallback.\n");
+		pr_dbg2("Failed to verify VFD configuration file, attempt using device tree as fallback.\n");
 		get_pin_from_dt(MOD_NAME_CLK, pdev, &pdata->dev->clk_pin);
 		get_pin_from_dt(MOD_NAME_DAT, pdev, &pdata->dev->dat_pin);
 		get_pin_from_dt(MOD_NAME_STB, pdev, &pdata->dev->stb_pin);
