diff --speed-large-files --no-dereference --minimal -Naur u-boot-32640b0ada9344f91e7a407576568782907161cd/arch/arm/dts/rk3528-u-boot.dtsi u-boot-32640b0ada9344f91e7a407576568782907161cd/arch/arm/dts/rk3528-u-boot.dtsi
--- u-boot-32640b0ada9344f91e7a407576568782907161cd/arch/arm/dts/rk3528-u-boot.dtsi	2023-07-25 03:31:46.000000000 +0200
+++ u-boot-32640b0ada9344f91e7a407576568782907161cd/arch/arm/dts/rk3528-u-boot.dtsi	2023-08-18 16:53:26.543405420 +0200
@@ -6,8 +6,10 @@
 
 / {
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		ethernet0 = &gmac0;
+		mmc0 = &sdmmc;
+		mmc1 = &sdio1;
+		mmc2 = &sdhci;
 	};
 
 	chosen {
