diff -Naur lockdev-1.0.3-old/Makefile lockdev-1.0.3-new/Makefile
--- lockdev-1.0.3-old/Makefile	2005-10-03 11:51:53.000000000 -0700
+++ lockdev-1.0.3-new/Makefile	2012-07-03 19:26:03.000000000 -0700
@@ -11,7 +11,7 @@
 MVER	= ${shell expr `pwd` : '.*-\([0-9]*\).[0-9]*'}
 
 static	= ${libname}.a
-shared	= ${libname}.${VER}.so
+shared	= ${libname}.so.${VER}
 soname	= ${libname}.so.${MVER}
 
 # overwritten by caller (e.g.: debian/rules)
@@ -45,8 +45,8 @@
 	cd LockDev && make OPTIMIZE="-O2 -g -Wall"
 	cd LockDev && make test
 
-.PHONY: install install_dev install_dbg install_doc install_run 
-install:	install_dev install_dbg install_doc install_run
+.PHONY: install install_dev install_doc install_run 
+install:	install_dev install_doc install_run
 
 install_dev:	${static} src/lockdev.h
 	install -m755 -d	${libdir}
