diff -Naur polkit-0.102-old/configure polkit-0.102-new/configure
--- polkit-0.102-old/configure	2011-08-01 15:42:12.000000000 +0200
+++ polkit-0.102-new/configure	2011-10-07 22:16:25.000000000 +0200
@@ -14251,14 +14251,9 @@
 if ${ac_cv_file__etc_redhat_release+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r "/etc/redhat-release"; then
-  ac_cv_file__etc_redhat_release=yes
-else
   ac_cv_file__etc_redhat_release=no
 fi
-fi
+
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_file__etc_redhat_release" >&5
 $as_echo "$ac_cv_file__etc_redhat_release" >&6; }
 if test "x$ac_cv_file__etc_redhat_release" = xyes; then :
@@ -14270,14 +14265,8 @@
 if ${ac_cv_file__etc_SuSE_release+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r "/etc/SuSE-release"; then
-  ac_cv_file__etc_SuSE_release=yes
-else
   ac_cv_file__etc_SuSE_release=no
 fi
-fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_file__etc_SuSE_release" >&5
 $as_echo "$ac_cv_file__etc_SuSE_release" >&6; }
 if test "x$ac_cv_file__etc_SuSE_release" = xyes; then :
@@ -14289,14 +14278,8 @@
 if ${ac_cv_file__etc_gentoo_release+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r "/etc/gentoo-release"; then
-  ac_cv_file__etc_gentoo_release=yes
-else
   ac_cv_file__etc_gentoo_release=no
 fi
-fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_file__etc_gentoo_release" >&5
 $as_echo "$ac_cv_file__etc_gentoo_release" >&6; }
 if test "x$ac_cv_file__etc_gentoo_release" = xyes; then :
@@ -14308,14 +14291,8 @@
 if ${ac_cv_file__etc_pardus_release+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
-  as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r "/etc/pardus-release"; then
-  ac_cv_file__etc_pardus_release=yes
-else
   ac_cv_file__etc_pardus_release=no
 fi
-fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_file__etc_pardus_release" >&5
 $as_echo "$ac_cv_file__etc_pardus_release" >&6; }
 if test "x$ac_cv_file__etc_pardus_release" = xyes; then :
