diff --speed-large-files --no-dereference --minimal -Naur arm-trusted-firmware-2.13.0/Makefile arm-trusted-firmware-2.13.0/Makefile
--- arm-trusted-firmware-2.13.0/Makefile	2025-05-22 12:24:05.000000000 +0200
+++ arm-trusted-firmware-2.13.0/Makefile	2025-06-06 14:40:47.064618894 +0200
@@ -196,7 +196,8 @@
 # General warnings
 WARNINGS		:=	-Wall -Wmissing-include-dirs -Wunused	\
 				-Wdisabled-optimization -Wvla -Wshadow	\
-				-Wredundant-decls
+				-Wredundant-decls \
+				-Wno-unterminated-string-initialization
 # stricter warnings
 WARNINGS		+=	-Wextra -Wno-trigraphs
 # too verbose for generic build
