diff -Naur libraw1394-2.1.0-old/configure.ac libraw1394-2.1.0-new/configure.ac
--- libraw1394-2.1.0-old/configure.ac	2012-06-27 13:10:19.000000000 -0700
+++ libraw1394-2.1.0-new/configure.ac	2013-01-03 18:20:32.000000000 -0800
@@ -2,7 +2,7 @@
 
 AC_INIT(libraw1394, 2.1.0)
 AC_CONFIG_SRCDIR(src/raw1394.h)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 dnl Checking if the NEWS file has been updated to reflect the current version.
 AM_INIT_AUTOMAKE(check-news)
