diff --speed-large-files --no-dereference --minimal -Naur elfutils-0.189/libelf/Makefile.in elfutils-0.189/libelf/Makefile.in
--- elfutils-0.189/libelf/Makefile.in	2023-03-03 16:42:30.000000000 +0100
+++ elfutils-0.189/libelf/Makefile.in	2023-04-28 21:39:32.786650172 +0200
@@ -628,7 +628,7 @@
 
 libelf_pic_a_SOURCES = 
 am_libelf_pic_a_OBJECTS = $(libelf_a_SOURCES:.c=.os)
-libelf_so_DEPS = ../lib/libeu.a
+# libelf_so_DEPS = ../lib/libeu.a
 libelf_so_LDLIBS = $(libelf_so_DEPS) -lz $(zstd_LIBS) $(am__append_5)
 libelf_so_LIBS = libelf_pic.a
 EXTRA_DIST = libelf.map
