diff --git a/src/Makefile.housekeeping b/src/Makefile.housekeeping
index 134becd50..4b0aa9e96 100644
--- a/src/Makefile.housekeeping
+++ b/src/Makefile.housekeeping
@@ -183,6 +183,20 @@ WNAPM_TEST = $(CC) -Wno-address-of-packed-member -x c -c /dev/null \
 WNAPM_FLAGS := $(shell $(WNAPM_TEST) && \
	 $(ECHO) '-Wno-address-of-packed-member')
 WORKAROUND_CFLAGS += $(WNAPM_FLAGS)
+
+# gcc 12.1 generates false positive warnings.  Inhibit the warnings.
+WNAB_TEST = $(CC) -Wno-array-bounds -Wno-error=maybe-uninitialized -x c -c /dev/null \
+		   -o /dev/null >/dev/null 2>&1
+WNAB_FLAGS := $(shell $(WNAB_TEST) && \
+		 $(ECHO) '-Wno-array-bounds -Wno-error=maybe-uninitialized')
+WORKAROUND_CFLAGS += $(WNAB_FLAGS)
+
+WNDP_TEST = $(CC) -Wno-dangling-pointer-x -Wno-error=maybe-uninitialized c -c /dev/null \
+		   -o /dev/null >/dev/null 2>&1
+WNDP_FLAGS := $(shell $(WNAB_TEST) && \
+		 $(ECHO) '-Wno-dangling-pointer -Wno-error=maybe-uninitialized')
+WORKAROUND_CFLAGS += $(WNDP_FLAGS)
+
 endif
 
 # Some versions of gas choke on division operators, treating them as
 