diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/i386/core/setjmp.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/i386/core/setjmp.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/i386/core/setjmp.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/i386/core/setjmp.S	2024-12-19 21:10:09.940103697 +0100
@@ -1,8 +1,8 @@
 FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL )
 
 	.text
-	.arch i386
 	.code32
+	.arch i386
 
 	/* Must match jmp_buf structure layout */
 	.struct	0
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/i386/tests/gdbstub_test.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/i386/tests/gdbstub_test.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/i386/tests/gdbstub_test.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/i386/tests/gdbstub_test.S	2024-12-19 21:10:09.940103697 +0100
@@ -1,3 +1,4 @@
+	.code32
 	.arch i386
 
 	.section ".data", "aw", @progbits
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/patch_cf.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/patch_cf.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/patch_cf.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/patch_cf.S	2024-12-19 21:10:09.940103697 +0100
@@ -23,8 +23,8 @@
 FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL )
 
 	.text
-	.arch i386
 	.code16
+	.arch i386
 
 /****************************************************************************
  * Set/clear CF on the stack as appropriate, assumes stack is as it should
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/stack16.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/stack16.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/stack16.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/stack16.S	2024-12-19 21:10:09.940103697 +0100
@@ -1,7 +1,5 @@
 FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL )
 
-	.arch i386
-
 /****************************************************************************
  * Internal stack
  ****************************************************************************
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/stack.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/stack.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/stack.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/core/stack.S	2024-12-19 21:10:09.940103697 +0100
@@ -1,7 +1,5 @@
 FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL )
 
-	.arch i386
-
 #ifdef __x86_64__
 #define STACK_SIZE 8192
 #else
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/drivers/net/undiisr.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/drivers/net/undiisr.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/drivers/net/undiisr.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/drivers/net/undiisr.S	2024-12-19 21:10:09.940103697 +0100
@@ -11,8 +11,8 @@
 #define PIC2_ICR 0xa0
 	
 	.text
-	.arch i386
 	.code16
+	.arch i386
 
 	.section ".text16", "ax", @progbits
 	.globl undiisr
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/interface/pcbios/e820mangler.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/interface/pcbios/e820mangler.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/interface/pcbios/e820mangler.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/interface/pcbios/e820mangler.S	2024-12-19 21:10:09.940103697 +0100
@@ -24,8 +24,8 @@
 FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL )
 
 	.text
-	.arch i386
 	.code16
+	.arch i386
 
 #define SMAP 0x534d4150
 
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/interface/pxe/pxe_entry.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/interface/pxe/pxe_entry.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/interface/pxe/pxe_entry.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/interface/pxe/pxe_entry.S	2024-12-19 21:10:09.943437030 +0100
@@ -26,6 +26,7 @@
 
 #include <librm.h>
 
+	.code16
 	.arch i386
 
 /****************************************************************************
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/bootpart.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/bootpart.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/bootpart.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/bootpart.S	2024-12-19 21:10:09.943437030 +0100
@@ -6,9 +6,9 @@
 #define STACK_SIZE	0x2000
 	
 	.text
+	.code16
 	.arch i386
 	.section ".prefix", "awx", @progbits
-	.code16
 
 /*
  * Find active partition
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/dskprefix.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/dskprefix.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/dskprefix.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/dskprefix.S	2024-12-19 21:10:09.943437030 +0100
@@ -25,10 +25,10 @@
 .equ	SYSSEG, 0x1000			/* system loaded at SYSSEG<<4 */
 
 	.org	0
+	.code16
 	.arch i386
 	.text
 	.section ".prefix", "ax", @progbits
-	.code16
 	.globl	_dsk_start
 _dsk_start:
 
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/exeprefix.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/exeprefix.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/exeprefix.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/exeprefix.S	2024-12-19 21:10:09.943437030 +0100
@@ -37,9 +37,9 @@
 #define PSP_CMDLINE_START 0x81
 
 	.text
+	.code16
 	.arch i386
 	.org 0
-	.code16
 	.section ".prefix", "awx", @progbits
 
 signature:
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/hdprefix.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/hdprefix.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/hdprefix.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/hdprefix.S	2024-12-19 21:10:09.943437030 +0100
@@ -3,9 +3,9 @@
 #include <librm.h>
 
 	.text
+	.code16
 	.arch i386
 	.section ".prefix", "awx", @progbits
-	.code16
 	.org 0
 	.globl	_hd_start
 _hd_start:
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/libprefix.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/libprefix.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/libprefix.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/libprefix.S	2024-12-19 21:10:09.943437030 +0100
@@ -26,6 +26,7 @@
 
 #include <librm.h>
 
+	.code16
 	.arch i386
 
 /* Image compression enabled */
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/lkrnprefix.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/lkrnprefix.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/lkrnprefix.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/lkrnprefix.S	2024-12-19 21:10:09.943437030 +0100
@@ -5,8 +5,8 @@
 #define BZI_LOAD_HIGH_ADDR 0x100000
 
 	.text
-	.arch i386
 	.code16
+	.arch i386
 	.section ".prefix", "ax", @progbits
 	.globl	_lkrn_start
 _lkrn_start:
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/mbr.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/mbr.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/mbr.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/mbr.S	2024-12-19 21:10:09.943437030 +0100
@@ -1,9 +1,9 @@
 FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL )
 
 	.text
+	.code16
 	.arch i386
 	.section ".prefix", "awx", @progbits
-	.code16
 	.org 0
 
 	.globl mbr
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/mromprefix.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/mromprefix.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/mromprefix.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/mromprefix.S	2024-12-19 21:10:09.943437030 +0100
@@ -42,8 +42,8 @@
 #include "pciromprefix.S"
 
 	.text
-	.arch i386
 	.code16
+	.arch i386
 
 /* Obtain access to payload by exposing the expansion ROM BAR at the
  * address currently used by a suitably large memory BAR on the same
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/nbiprefix.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/nbiprefix.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/nbiprefix.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/nbiprefix.S	2024-12-19 21:10:09.943437030 +0100
@@ -3,8 +3,8 @@
 #include <librm.h>
 
 	.text
-	.arch i386
 	.code16
+	.arch i386
 	.section ".prefix", "ax", @progbits
 	.org 0
 
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/nullprefix.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/nullprefix.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/nullprefix.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/nullprefix.S	2024-12-19 21:10:09.943437030 +0100
@@ -2,10 +2,10 @@
 
 	.org	0
 	.text
+	.code16
 	.arch i386
 
 	.section ".prefix", "ax", @progbits
-	.code16
 _prefix:
 
 	.section ".text16", "ax", @progbits
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/pxeprefix.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/pxeprefix.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/pxeprefix.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/pxeprefix.S	2024-12-19 21:10:09.943437030 +0100
@@ -12,9 +12,9 @@
 #define PXE_HACK_EB54			0x0001
 
 	.text
+	.code16
 	.arch i386
 	.org 0
-	.code16
 
 #include <librm.h>
 #include <undi.h>
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/unlzma.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/unlzma.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/unlzma.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/unlzma.S	2024-12-19 21:10:59.853438358 +0100
@@ -44,6 +44,7 @@
  */
 
 	.text
+	.code32
 	.arch i586
 	.section ".prefix.lib", "ax", @progbits
 
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/usbdisk.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/usbdisk.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/usbdisk.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/prefix/usbdisk.S	2024-12-19 21:10:16.340103872 +0100
@@ -3,9 +3,9 @@
 #include <config/console.h>
 
 	.text
+	.code16
 	.arch i386
 	.section ".prefix", "awx", @progbits
-	.code16
 	.org 0
 
 #include "mbr.S"
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/transitions/liba20.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/transitions/liba20.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/transitions/liba20.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/transitions/liba20.S	2024-12-19 21:10:16.340103872 +0100
@@ -24,6 +24,7 @@
 
 FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL )
 
+	.code16
 	.arch i386
 
 /****************************************************************************
diff --speed-large-files --no-dereference --minimal -Naur ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/transitions/libkir.S ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/transitions/libkir.S
--- ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/transitions/libkir.S	2020-12-31 22:03:10.000000000 +0100
+++ ipxe-988d2c13cdf0f0b4140685af35ced70ac5b3283c/src/arch/x86/transitions/libkir.S	2024-12-19 21:10:16.340103872 +0100
@@ -32,9 +32,9 @@
 #define BOCHSBP xchgw %bx, %bx
 
 	.text
+	.code16
 	.arch i386
 	.section ".text16", "awx", @progbits
-	.code16
 	
 /****************************************************************************
  * init_libkir (real-mode or 16:xx protected-mode far call)
