diff --speed-large-files --no-dereference --minimal -Naur db-6.2.32/dist/Makefile.in db-6.2.32/dist/Makefile.in
--- db-6.2.32/dist/Makefile.in	2017-04-13 16:06:13.000000000 +0200
+++ db-6.2.32/dist/Makefile.in	2025-05-08 10:14:03.251050599 +0200
@@ -64,7 +64,7 @@
 ##################################################
 # C API.
 ##################################################
-CFLAGS=		-c $(CPPFLAGS) @CFLAGS@
+CFLAGS=		-c $(CPPFLAGS) @CFLAGS@ -Wno-incompatible-pointer-types
 CC=		@MAKEFILE_CC@
 CCLINK=		@MAKEFILE_CCLINK@ @CFLAGS@
 
