diff -Naur bumblebee-3.2.1-old/Makefile.in bumblebee-3.2.1-new/Makefile.in
--- bumblebee-3.2.1-old/Makefile.in	2013-04-26 17:49:11.000000000 +0200
+++ bumblebee-3.2.1-new/Makefile.in	2016-03-02 11:59:22.436666651 +0100
@@ -336,7 +336,7 @@
 	-e 's|[@]CONF_MODPATH_NVIDIA[@]|$(CONF_MODPATH_NVIDIA)|g' \
 	-e 's|[@]CONF_PIDFILE[@]|$(CONF_PIDFILE)|g'
 
-@HAVE_HELP2MAN_TRUE@man_MANS = bumblebeed.1 optirun.1
+@HAVE_HELP2MAN_TRUE@man_MANS =
 all: all-am
 
 .SUFFIXES:
@@ -1138,14 +1138,9 @@
 	$(do_subst) < $< > $@
 
 @HAVE_HELP2MAN_TRUE@optirun.1: bin/optirun $(bin_optirun_SOURCES)
-@HAVE_HELP2MAN_TRUE@	$(HELP2MAN) --output=$@ --no-info \
-@HAVE_HELP2MAN_TRUE@		--name='run programs on the discrete video card' \
-@HAVE_HELP2MAN_TRUE@		bin/optirun
 
 @HAVE_HELP2MAN_TRUE@bumblebeed.1: bin/bumblebeed $(bin_bumblebeed_SOURCES)
-@HAVE_HELP2MAN_TRUE@	$(HELP2MAN) --output=$@ --no-info \
-@HAVE_HELP2MAN_TRUE@		--name='bumblebee daemon' \
-@HAVE_HELP2MAN_TRUE@		bin/bumblebeed
+
 @HAVE_HELP2MAN_FALSE@	@echo "Warning: help2man not available, no man page is created."
 
 dist-hook:
