diff -Naur nspr-4.10.8-old/nspr/config/nspr-config.in nspr-4.10.8-new/nspr/config/nspr-config.in
--- nspr-4.10.8-old/nspr/config/nspr-config.in	2015-01-22 12:44:59.000000000 -0800
+++ nspr-4.10.8-new/nspr/config/nspr-config.in	2015-08-15 15:44:43.537000027 -0700
@@ -122,11 +122,11 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-    echo -I$includedir
+    echo -I$DESTDIR$includedir
 fi
 
 if test "$echo_libs" = "yes"; then
-      libdirs=-L$libdir
+      libdirs=-L$DESTDIR$libdir
       if test -n "$lib_plds"; then
 	libdirs="$libdirs -lplds${major_version}"
       fi
diff -Naur nspr-4.10.8-old/nspr/configure nspr-4.10.8-new/nspr/configure
--- nspr-4.10.8-old/nspr/configure	2015-01-22 12:44:59.000000000 -0800
+++ nspr-4.10.8-new/nspr/configure	2015-08-15 15:44:43.538000017 -0700
@@ -5408,6 +5408,7 @@
 
 if test "$cross_compiling"  = "yes"; then
     CROSS_COMPILE=1
+    NSINSTALL='nsinstall'
 else
     CROSS_COMPILE=
 fi
diff -Naur nspr-4.10.8-old/nspr/configure.in nspr-4.10.8-new/nspr/configure.in
--- nspr-4.10.8-old/nspr/configure.in	2015-01-22 12:44:59.000000000 -0800
+++ nspr-4.10.8-new/nspr/configure.in	2015-08-15 15:44:43.539000010 -0700
@@ -718,6 +718,7 @@
 
 if test "$cross_compiling"  = "yes"; then
     CROSS_COMPILE=1
+    NSINSTALL='nsinstall'
 else
     CROSS_COMPILE=
 fi
