diff -Naur nspr-4.9.4-old/mozilla/nsprpub/config/nspr-config.in nspr-4.9.4-new/mozilla/nsprpub/config/nspr-config.in
--- nspr-4.9.4-old/mozilla/nsprpub/config/nspr-config.in	2012-03-06 05:13:39.000000000 -0800
+++ nspr-4.9.4-new/mozilla/nsprpub/config/nspr-config.in	2012-11-22 22:05:34.000000000 -0800
@@ -122,11 +122,11 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-    echo -I$includedir
+    echo -I$DESTDIR$includedir
 fi
 
 if test "$echo_libs" = "yes"; then
-      libdirs=-L$libdir
+      libdirs=-L$DESTDIR$libdir
       if test -n "$lib_plds"; then
 	libdirs="$libdirs -lplds${major_version}"
       fi
diff -Naur nspr-4.9.4-old/mozilla/nsprpub/configure nspr-4.9.4-new/mozilla/nsprpub/configure
--- nspr-4.9.4-old/mozilla/nsprpub/configure	2012-10-24 15:19:07.000000000 -0700
+++ nspr-4.9.4-new/mozilla/nsprpub/configure	2012-11-22 22:05:34.000000000 -0800
@@ -2839,6 +2839,7 @@
 
 if test "$cross_compiling"  = "yes"; then
     CROSS_COMPILE=1
+    NSINSTALL='nsinstall'
 else
     CROSS_COMPILE=
 fi
diff -Naur nspr-4.9.4-old/mozilla/nsprpub/configure.in nspr-4.9.4-new/mozilla/nsprpub/configure.in
--- nspr-4.9.4-old/mozilla/nsprpub/configure.in	2012-10-24 15:19:07.000000000 -0700
+++ nspr-4.9.4-new/mozilla/nsprpub/configure.in	2012-11-22 22:05:34.000000000 -0800
@@ -676,6 +676,7 @@
 
 if test "$cross_compiling"  = "yes"; then
     CROSS_COMPILE=1
+    NSINSTALL='nsinstall'
 else
     CROSS_COMPILE=
 fi
