diff --speed-large-files --no-dereference --minimal -Naur u-boot-182cb30084516c3085d2ca5dde3f105f7625d774/dts/upstream/src/arm64/rockchip/rk3399-orangepi.dts u-boot-182cb30084516c3085d2ca5dde3f105f7625d774/dts/upstream/src/arm64/rockchip/rk3399-orangepi.dts
--- u-boot-182cb30084516c3085d2ca5dde3f105f7625d774/dts/upstream/src/arm64/rockchip/rk3399-orangepi.dts	2025-08-04 17:57:50.866669736 +0200
+++ u-boot-182cb30084516c3085d2ca5dde3f105f7625d774/dts/upstream/src/arm64/rockchip/rk3399-orangepi.dts	2025-08-04 17:55:57.760002717 +0200
@@ -303,7 +303,7 @@
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
+				regulator-max-microvolt = <1800000>;
 				regulator-state-mem {
 					regulator-on-in-suspend;
 					regulator-suspend-microvolt = <1800000>;
@@ -315,7 +315,7 @@
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3400000>;
+				regulator-max-microvolt = <1800000>;
 				regulator-state-mem {
 					regulator-off-in-suspend;
 				};
@@ -326,7 +326,7 @@
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3400000>;
+				regulator-max-microvolt = <3000000>;
 				regulator-state-mem {
 					regulator-off-in-suspend;
 				};
@@ -336,8 +336,8 @@
 				regulator-name = "vcc1v8_pmupll";
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <2500000>;
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
 				regulator-state-mem {
 					regulator-on-in-suspend;
 					regulator-suspend-microvolt = <1800000>;
@@ -360,8 +360,8 @@
 				regulator-name = "vcca3v0_codec";
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3400000>;
+				regulator-min-microvolt = <3000000>;
+				regulator-max-microvolt = <3000000>;
 				regulator-state-mem {
 					regulator-off-in-suspend;
 				};
