diff -Naur qt-everywhere-opensource-src-4.8.4-old/src/3rdparty/webkit/Source/WebKit.pri qt-everywhere-opensource-src-4.8.4-new/src/3rdparty/webkit/Source/WebKit.pri
--- qt-everywhere-opensource-src-4.8.4-old/src/3rdparty/webkit/Source/WebKit.pri	2012-11-23 02:10:05.000000000 -0800
+++ qt-everywhere-opensource-src-4.8.4-new/src/3rdparty/webkit/Source/WebKit.pri	2012-12-02 04:54:53.000000000 -0800
@@ -102,7 +102,7 @@
 
 # Treat warnings as errors on x86/Linux/GCC
 linux-g++* {
-    !CONFIG(standalone_package):if(isEqual(QT_ARCH,x86_64)|isEqual(QT_ARCH,i386)): QMAKE_CXXFLAGS += -Werror
+    #!CONFIG(standalone_package):if(isEqual(QT_ARCH,x86_64)|isEqual(QT_ARCH,i386)): QMAKE_CXXFLAGS += -Werror
 
     greaterThan(QT_GCC_MAJOR_VERSION, 3):greaterThan(QT_GCC_MINOR_VERSION, 5) {
         if (!contains(QMAKE_CXXFLAGS, -std=c++0x) && !contains(QMAKE_CXXFLAGS, -std=gnu++0x)) {
