diff -Naur SDL2_net-2.0.0-old/configure.in SDL2_net-2.0.0-new/configure.in
--- SDL2_net-2.0.0-old/configure.in	2013-08-10 10:41:30.000000000 -0700
+++ SDL2_net-2.0.0-new/configure.in	2015-06-23 13:48:25.822000236 -0700
@@ -1,6 +1,8 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(README.txt)
 
+AC_CONFIG_MACRO_DIR([acinclude])
+
 dnl Set various version strings - taken gratefully from the GTk sources
 
 # Making releases:
diff -Naur SDL2_net-2.0.0-old/Makefile.am SDL2_net-2.0.0-new/Makefile.am
--- SDL2_net-2.0.0-old/Makefile.am	2013-08-10 10:41:30.000000000 -0700
+++ SDL2_net-2.0.0-new/Makefile.am	2015-06-23 13:48:49.956000233 -0700
@@ -1,5 +1,7 @@
 # Makefile.am for the SDL sample image loading library and viewer
 
+ACLOCAL_AMFLAGS = -I acinclude
+
 lib_LTLIBRARIES = libSDL2_net.la
 
 libSDL2_netincludedir = $(includedir)/SDL2
