diff -Naur xf86-input-evdev-2.7.2-old/configure.ac xf86-input-evdev-2.7.2-new/configure.ac
--- xf86-input-evdev-2.7.2-old/configure.ac	2012-08-02 23:01:37.000000000 -0700
+++ xf86-input-evdev-2.7.2-new/configure.ac	2012-08-04 15:14:43.000000000 -0700
@@ -69,7 +69,8 @@
 
 # X Server SDK location is required to install evdev header files
 # This location is also relayed in the xorg-evdev.pc file
-sdkdir=`$PKG_CONFIG --variable=sdkdir xorg-server`
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+sdkdir="$PKG_CONFIG_SYSROOT_DIR$($PKG_CONFIG --variable=sdkdir xorg-server)"
 
 # Workaround overriding sdkdir to be able to create a tarball when user has no
 # write permission in sdkdir. See DISTCHECK_CONFIGURE_FLAGS in Makefile.am
