diff --speed-large-files --no-dereference --minimal -Naur build-763eec0c7c29af9f60cca5e98eb1db6031b11976/board_configs.sh build-763eec0c7c29af9f60cca5e98eb1db6031b11976/board_configs.sh
--- build-763eec0c7c29af9f60cca5e98eb1db6031b11976/board_configs.sh	2020-10-12 03:58:20.000000000 +0200
+++ build-763eec0c7c29af9f60cca5e98eb1db6031b11976/board_configs.sh	2022-12-15 19:07:22.863304439 +0100
@@ -90,6 +90,16 @@
 		export CROSS_COMPILE=aarch64-linux-gnu-
 		CHIP="rk3399"
 		;;
+	"orangepi-rk3399")
+		DEFCONFIG=rockchip_linux_defconfig
+		DEFCONFIG_MAINLINE=defconfig
+		UBOOT_DEFCONFIG=orangepi_rk3399_defconfig
+		DTB=rk3399-orangepi.dtb
+		DTB_MAINLINE=rk3399-orangepi.dtb
+		export ARCH=arm64
+		export CROSS_COMPILE=aarch64-linux-gnu-
+		CHIP="rk3399"
+		;;
 	"rockpis")
 		DEFCONFIG=rk3308_linux_defconfig
 		UBOOT_DEFCONFIG=rock-pi-s-rk3308_defconfig
