diff -Naur tar-1.28-old/Makefile.am tar-1.28-new/Makefile.am
--- tar-1.28-old/Makefile.am	2014-07-07 20:08:27.000000000 +0200
+++ tar-1.28-new/Makefile.am	2020-05-10 20:45:42.836292935 +0200
@@ -20,7 +20,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 EXTRA_DIST = ChangeLog.1 Make.rules
-SUBDIRS = doc gnu lib rmt src scripts po tests
+SUBDIRS = gnu lib rmt src scripts tests
 
 dist-hook:
 	$(MAKE) changelog_dir=$(distdir) ChangeLog
