diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythupnp/upnpcds.h mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythupnp/upnpcds.h
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythupnp/upnpcds.h	2021-05-18 16:11:29.096666604 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythupnp/upnpcds.h	2021-05-18 16:11:32.283333271 +0200
@@ -312,7 +312,7 @@
         // Implement UPnpServiceImpl methods that we can
 
         QString GetServiceType() override // UPnpServiceImpl
-            { return "urn:schemas-upnp-org:service:ContentDirectory:4"; }
+            { return "urn:schemas-upnp-org:service:ContentDirectory:1"; }
         QString GetServiceId() override // UPnpServiceImpl
             { return "urn:upnp-org:serviceId:ContentDirectory"; }
         QString GetServiceControlURL() override // UPnpServiceImpl
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythupnp/upnpcmgr.h mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythupnp/upnpcmgr.h
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythupnp/upnpcmgr.h	2021-05-18 16:11:29.093333271 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythupnp/upnpcmgr.h	2021-05-18 16:11:32.286666604 +0200
@@ -66,7 +66,7 @@
         // Implement UPnpServiceImpl methods that we can
 
         QString GetServiceType() override // UPnpServiceImpl
-            { return "urn:schemas-upnp-org:service:ConnectionManager:3"; }
+            { return "urn:schemas-upnp-org:service:ConnectionManager:1"; }
         QString GetServiceId() override // UPnpServiceImpl
             { return "urn:upnp-org:serviceId:ConnectionManager"; }
         QString GetServiceControlURL() override // UPnpServiceImpl
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythbackend/devicemaster.xml mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythbackend/devicemaster.xml
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythbackend/devicemaster.xml	2021-05-18 16:11:28.919999938 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythbackend/devicemaster.xml	2021-05-18 16:11:32.286666604 +0200
@@ -7,7 +7,7 @@
     <device>
         <!-- UDN auto generated -->
         <friendlyName>MythTV AV Media Server</friendlyName>
-        <deviceType>urn:schemas-upnp-org:device:MediaServer:4</deviceType>
+        <deviceType>urn:schemas-upnp-org:device:MediaServer:1</deviceType>
         <manufacturer>MythTV</manufacturer>
         <manufacturerURL>http://www.mythtv.org/</manufacturerURL>
         <modelName>MythTV AV Media Server</modelName>
