diff --speed-large-files --no-dereference --minimal -Naur mythtv-e64f9def649346ebb9d66a43948142d04489930b/mythtv/libs/libmythtv/decoders/avformatdecoder.h mythtv-e64f9def649346ebb9d66a43948142d04489930b/mythtv/libs/libmythtv/decoders/avformatdecoder.h
--- mythtv-e64f9def649346ebb9d66a43948142d04489930b/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2022-09-19 02:23:10.000000000 +0200
+++ mythtv-e64f9def649346ebb9d66a43948142d04489930b/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2022-09-23 12:50:42.483145249 +0200
@@ -71,10 +71,11 @@
     }
     QString toString() const
     {
-        return QString("id(%1) %2Hz %3ch %4bps %5 (profile %6)")
+        return QString("id(%1) %2Hz %3/%7ch %4bps %5 (profile %6)")
             .arg(avcodec_get_name(m_codecId),4).arg(m_sampleRate,6)
             .arg(m_channels,2).arg(AudioOutputSettings::FormatToBits(format),2)
-            .arg((m_doPassthru) ? "pt":"",3).arg(m_codecProfile);
+            .arg((m_doPassthru) ? "pt":"",3).arg(m_codecProfile)
+            .arg(m_originalChannels);
     }
 };
 
