diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmyth/mediamonitor-unix.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmyth/mediamonitor-unix.cpp	2021-05-18 16:12:39.659999937 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmyth/mediamonitor-unix.cpp	2021-05-18 16:12:42.869999937 +0200
@@ -198,7 +198,7 @@
 bool MediaMonitorUnix::CheckMountable(void)
 {
 #if CONFIG_QTDBUS
-    for (int i = 0; i < 10; ++i, usleep(500000))
+    for (int i = 0; i < 3; ++i, usleep(500000))
     {
         // Connect to UDisks.  This can sometimes fail if mythfrontend
         // is started during system init
