diff --speed-large-files --no-dereference --minimal -Naur mythtv-6034adfa56366841ba9e62eee7717b0e68b2b633/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-6034adfa56366841ba9e62eee7717b0e68b2b633/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-6034adfa56366841ba9e62eee7717b0e68b2b633/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2022-08-06 22:14:28.000000000 +0200
+++ mythtv-6034adfa56366841ba9e62eee7717b0e68b2b633/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2022-08-24 12:37:24.409931597 +0200
@@ -1141,6 +1141,7 @@
 void ZMServer::handleGetEventFrame(std::vector<std::string> tokens)
 {
     static FrameData s_buffer {};
+    std::array<char,100> str {};
 
     if (tokens.size() != 5)
     {
@@ -1161,12 +1162,8 @@
     }
 
     std::string outStr;
-
-    ADD_STR(outStr, "OK");
-
-    // try to find the frame file
     std::string filepath;
-    std::string str (100,'\0');
+    std::string frameFile;
 
     if (checkVersion(1, 32, 0))
     {
@@ -1176,10 +1173,8 @@
 
         sscanf(eventTime.data(), "%2d/%2d/%2d", &year, &month, &day);
         sprintf(str.data(), "20%02d-%02d-%02d", year, month, day);
-
-        filepath = g_eventsPath + "/" + monitorID + "/" + str + "/" + eventID + "/";
+        filepath = g_eventsPath + "/" + monitorID + "/" + str.data() + "/" + eventID + "/";
         sprintf(str.data(), m_eventFileFormat.c_str(), frameNo);
-        filepath += str;
     }
     else
     {
@@ -1187,18 +1182,20 @@
         {
             filepath = g_webPath + "/events/" + monitorID + "/" + eventTime + "/";
             sprintf(str.data(), m_eventFileFormat.c_str(), frameNo);
-            filepath += str;
         }
         else
         {
             filepath = g_webPath + "/events/" + monitorID + "/" + eventID + "/";
             sprintf(str.data(), m_eventFileFormat.c_str(), frameNo);
-            filepath += str;
         }
     }
 
+    ADD_STR(outStr, "OK");
+
     int fileSize = 0;
-    FILE *fd = fopen(filepath.c_str(), "r" );
+    frameFile = filepath + str.data();
+
+    FILE *fd = fopen(frameFile.c_str(), "r" );
     if (fd != nullptr)
     {
         fileSize = fread(s_buffer.data(), 1, s_buffer.size(), fd);
@@ -1212,7 +1209,10 @@
     }
 
     if (m_debug)
-        std::cout << "Frame size: " <<  fileSize << std::endl;
+    {
+        std::cout << "File path: " <<  frameFile << std::endl;
+        std::cout << "File size: " <<  fileSize << std::endl;
+    }
 
     // get the file size
     ADD_INT(outStr, fileSize);
@@ -1382,7 +1382,10 @@
     }
 
     if (m_debug)
-        std::cout << "Frame size: " <<  fileSize << std::endl;
+    {
+        std::cout << "File path: " <<  frameFile << std::endl;
+        std::cout << "File size: " <<  fileSize << std::endl;
+    }
 
     // get the file size
     ADD_INT(outStr, fileSize);
