diff -Naur mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/cardutil.cpp	2019-04-08 14:44:46.536666667 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/cardutil.cpp	2019-04-08 14:44:47.850000000 +0200
@@ -2302,7 +2302,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
