diff -Naur mythtv-b50f8b17c1a7a34f59f207625d32c15adda3d8b1-old/mythtv/configure mythtv-b50f8b17c1a7a34f59f207625d32c15adda3d8b1-new/mythtv/configure
--- mythtv-b50f8b17c1a7a34f59f207625d32c15adda3d8b1-old/mythtv/configure	2021-10-05 21:52:35.409248479 +0200
+++ mythtv-b50f8b17c1a7a34f59f207625d32c15adda3d8b1-new/mythtv/configure	2021-10-05 21:46:09.629255435 +0200
@@ -6576,7 +6576,7 @@
     check_ecxx ${qt_inc} <<EOF
 #include <QtCore/QtCore>
 #include <QtGui/qtguiglobal.h>
-#if !defined(QT_OPENGL_ES_2)
+#if !defined(QT_OPENGL_ES_2) & !defined(QT_FEATURE_opengles2)
 No OpenGL ES here!
 #endif
 int x;
