diff -Naur mythtv-master-20190408-g038e171-old/mythtv/contrib/user_jobs/mythlink.pl mythtv-master-20190408-g038e171-new/mythtv/contrib/user_jobs/mythlink.pl
--- mythtv-master-20190408-g038e171-old/mythtv/contrib/user_jobs/mythlink.pl	2019-04-08 14:44:50.953333334 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/contrib/user_jobs/mythlink.pl	2019-04-08 14:44:52.726666667 +0200
@@ -29,7 +29,7 @@
     our ($hostname, $dbh, $sh, $q, $count, $base_dir);
 
 # Default filename format
-    $dformat = '%T %- %Y-%m-%d, %g-%i %A %- %S';
+    $dformat = '%U/%T %- %Y-%m-%d, %g-%i %A %- %S';
 # Default separator character
     $dseparator = '-';
 # Default replacement character
