diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythui/mythmainwindow.cpp	2019-03-16 10:52:50.173333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythui/mythmainwindow.cpp	2019-03-16 10:52:51.363333334 +0100
@@ -2906,12 +2906,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->m_wmult);
+    return (int)(x * d->m_wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->m_hmult);
+    return (int)(y * d->m_hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
