diff -Naur mythtv-master-20190415-gb60b867-new/mythtv/libs/libmythtv/subtitlescreen.cpp mythtv-master-20190415-gb60b867-old/mythtv/libs/libmythtv/subtitlescreen.cpp
--- mythtv-master-20190415-gb60b867-new/mythtv/libs/libmythtv/subtitlescreen.cpp	2019-04-15 10:13:25.000000000 +0200
+++ mythtv-master-20190415-gb60b867-old/mythtv/libs/libmythtv/subtitlescreen.cpp	2019-04-15 10:22:22.943333340 +0200
@@ -485,8 +485,6 @@
         dynamic_cast<MythUIShape *>(baseParent->GetChild(prefix));
     if (!resultBG)
         resultBG = providerBaseShape;
-    else
-        delete providerBaseShape;
     MythFontProperties *testFont = negParent->GetFont(prefix);
     if (!testFont)
         testFont = negFont;
