diff -Naur mythtv-master-20200207-g2869463-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20200207-g2869463-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20200207-g2869463-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2020-02-08 14:18:32.150000002 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2020-02-08 14:18:34.113333335 +0100
@@ -844,7 +844,7 @@
                 tspacket->data()[i++] != 0x00 ||
                 tspacket->data()[i++] != 0x01)
             {
-                LOG(VB_GENERAL, LOG_ERR, LOC +
+                LOG(VB_RECORD, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
                 break;
             }
diff -Naur mythtv-master-20200207-g2869463-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20200207-g2869463-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20200207-g2869463-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-02-08 14:18:32.263333335 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-02-08 14:18:34.113333335 +0100
@@ -351,7 +351,7 @@
     {
         if (insert_crc(m_seen_table_crc, *pmt))
         {
-            LOG(VB_CHANNEL, LOG_DEBUG, LOC +
+            LOG(VB_CHANNEL, LOG_ERR, LOC +
                 QString("Wrong PMT; pmt->ProgramNumber(%1) desired(%2)")
                 .arg(pmt->ProgramNumber()).arg(m_programNumber));
         }
