diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2019-03-16 10:52:58.743333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2019-03-16 10:53:00.396666667 +0100
@@ -843,7 +843,7 @@
                 tspacket->data()[i++] != 0x00 ||
                 tspacket->data()[i++] != 0x01)
             {
-                LOG(VB_GENERAL, LOG_ERR, LOC +
+                LOG(VB_RECORD, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
                 break;
             }
diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-03-16 10:52:58.986666667 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-03-16 10:53:00.396666667 +0100
@@ -350,7 +350,7 @@
     {
         if (insert_crc(m_seen_table_crc, *pmt))
         {
-            LOG(VB_GENERAL, LOG_ERR, LOC +
+            LOG(VB_CHANNEL, LOG_ERR, LOC +
                 QString("Wrong PMT; pmt->pn(%1) desired(%2)")
                 .arg(pmt->ProgramNumber()).arg(m_programNumber));
         }
