diff -Naur mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2019-04-08 14:45:06.683333333 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2019-04-08 14:45:07.873333333 +0200
@@ -842,7 +842,7 @@
                 tspacket->data()[i++] != 0x00 ||
                 tspacket->data()[i++] != 0x01)
             {
-                LOG(VB_GENERAL, LOG_ERR, LOC +
+                LOG(VB_RECORD, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
                 break;
             }
diff -Naur mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-04-08 14:45:06.833333333 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-04-08 14:45:07.876666667 +0200
@@ -350,7 +350,7 @@
     {
         if (insert_crc(m_seen_table_crc, *pmt))
         {
-            LOG(VB_GENERAL, LOG_ERR, LOC +
+            LOG(VB_CHANNEL, LOG_ERR, LOC +
                 QString("Wrong PMT; pmt->pn(%1) desired(%2)")
                 .arg(pmt->ProgramNumber()).arg(m_programNumber));
         }
