diff -Naur mythtv-master-20200207-g2869463-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20200207-g2869463-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20200207-g2869463-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-02-08 14:18:33.943333336 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2020-02-08 14:18:35.496666668 +0100
@@ -334,7 +334,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
