diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-02-17 16:58:10.543333336 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-02-17 16:58:11.976666669 +0100
@@ -298,7 +298,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
