diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/tv_rec.cpp	2019-03-16 10:53:01.810000000 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/tv_rec.cpp	2019-03-16 10:53:03.343333334 +0100
@@ -3920,7 +3920,7 @@
     // Start signal monitoring for devices capable of monitoring
     if (use_sm)
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "Starting Signal Monitor");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Starting Signal Monitor");
         bool error = false;
         if (!SetupSignalMonitor(
                 !antadj, request.flags & kFlagEITScan, livetv | antadj))
@@ -3970,7 +3970,7 @@
                     m_signalMonitorDeadline =
                         m_curRecording->GetRecordingEndTime().addSecs(-10);
 
-                    LOG(VB_CHANNEL, LOG_DEBUG, LOC +
+                    LOG(VB_GENERAL, LOG_DEBUG, LOC +
                         QString("Pre-fail start deadline: %1 "
                                 "Start recording deadline: %2 "
                                 "Good signal deadline: %3")
@@ -4042,7 +4042,7 @@
 
     if (m_signalMonitor->IsAllGood())
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
         if (m_curRecording && (current_time > m_startRecordingDeadline))
         {
             newRecStatus = RecStatus::Failing;
@@ -4131,7 +4131,7 @@
             --m_signalMonitorCheckCnt;
         else
         {
-            LOG(VB_RECORD, LOG_INFO, LOC +
+            LOG(VB_GENERAL, LOG_INFO, LOC +
                 QString("TuningSignalCheck: Still waiting.  Will timeout @ %1")
                 .arg(m_signalMonitorDeadline.toLocalTime()
                      .toString("hh:mm:ss.zzz")));
