diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythtv/decoders/avformatdecoder.h mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythtv/decoders/avformatdecoder.h
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2021-05-18 16:11:36.353333272 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2021-05-18 16:11:38.986666605 +0200
@@ -74,10 +74,11 @@
     }
     QString toString() const
     {
-        return QString("id(%1) %2Hz %3ch %4bps %5 (profile %6)")
+        return QString("id(%1) %2Hz %3/%7ch %4bps %5 (profile %6)")
             .arg(ff_codec_id_string(m_codecId),4).arg(m_sampleRate,6)
             .arg(m_channels,2).arg(AudioOutputSettings::FormatToBits(format),2)
-            .arg((m_doPassthru) ? "pt":"",3).arg(m_codecProfile);
+            .arg((m_doPassthru) ? "pt":"",3).arg(m_codecProfile)
+            .arg(m_originalChannels);
     }
 };
 
