diff -Naur mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/decoders/avformatdecoder.h mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/decoders/avformatdecoder.h
--- mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2019-04-08 14:45:09.570000000 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/decoders/avformatdecoder.h	2019-04-08 14:45:10.896666667 +0200
@@ -75,10 +75,11 @@
     }
     QString toString() const
     {
-        return QString("id(%1) %2Hz %3ch %4bps %5 (profile %6)")
+        return QString("id(%1) %2Hz %3/%7ch %4bps %5 (profile %6)")
             .arg(ff_codec_id_string(codec_id),4).arg(sample_rate,6)
             .arg(channels,2).arg(AudioOutputSettings::FormatToBits(format),2)
-            .arg((do_passthru) ? "pt":"",3).arg(codec_profile);
+            .arg((do_passthru) ? "pt":"",3).arg(codec_profile)
+            .arg(original_channels);
     }
 };
 
