diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/avformatdecoder.h mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/avformatdecoder.h
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/avformatdecoder.h	2019-02-17 16:58:13.266666669 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/avformatdecoder.h	2019-02-17 16:58:14.663333336 +0100
@@ -75,10 +75,11 @@
     }
     QString toString() const
     {
-        return QString("id(%1) %2Hz %3ch %4bps %5 (profile %6)")
+        return QString("id(%1) %2Hz %3/%7ch %4bps %5 (profile %6)")
             .arg(ff_codec_id_string(codec_id),4).arg(sample_rate,6)
             .arg(channels,2).arg(AudioOutputSettings::FormatToBits(format),2)
-            .arg((do_passthru) ? "pt":"",3).arg(codec_profile);
+            .arg((do_passthru) ? "pt":"",3).arg(codec_profile)
+            .arg(original_channels);
     }
 };
 
