diff -Naur mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20201010-g68e85b9-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-10-10 16:07:41.900000000 +0200
+++ mythtv-master-20201010-g68e85b9-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-10-10 16:07:43.403333333 +0200
@@ -516,6 +516,11 @@
 
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
+        LOG(VB_GENERAL, LOG_WARNING, LOC +
+            QString("Wrong SDT; onid %1 tsid %2, desired %3 %4\n"
+                    "\t\t\tRescan your transports.")
+                .arg(m_detectedNetworkID).arg(m_detectedTransportID)
+                .arg(m_networkID).arg(m_transportID) );
         DVBStreamData *dvb = GetDVBStreamData();
         if (dvb)
             dvb->SetVersionSDT(sdt->TSID(), -1, 0);
