diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-02-17 16:58:14.440000002 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-02-17 16:58:16.186666669 +0100
@@ -513,6 +513,11 @@
 
     if (sdt->OriginalNetworkID() != networkID || sdt->TSID() != transportID)
     {
+        LOG(VB_GENERAL, LOG_WARNING, LOC +
+            QString("Wrong SDT; onid %1 tsid %2, desired %3 %4\n"
+                    "\t\t\tRescan your transports.")
+                .arg(m_detectedNetworkID).arg(m_detectedTransportID)
+                .arg(m_networkID).arg(m_transportID) );
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
     }
     else
