diff -Naur mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-04-08 14:45:10.583333333 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2019-04-08 14:45:11.913333333 +0200
@@ -509,6 +509,11 @@
 
     if (sdt->OriginalNetworkID() != m_networkID || sdt->TSID() != m_transportID)
     {
+        LOG(VB_GENERAL, LOG_WARNING, LOC +
+            QString("Wrong SDT; onid %1 tsid %2, desired %3 %4\n"
+                    "\t\t\tRescan your transports.")
+                .arg(m_detectedNetworkID).arg(m_detectedTransportID)
+                .arg(m_networkID).arg(m_transportID) );
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
     }
     else
