diff -Naur mythtv-master-20190408-g038e171-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-master-20190408-g038e171-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-master-20190408-g038e171-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2019-04-08 14:45:11.556666667 +0200
+++ mythtv-master-20190408-g038e171-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2019-04-08 14:45:12.943333333 +0200
@@ -162,6 +162,12 @@
             if (m_meta_album_pos > 0)
                 result["album"] = rx.cap(m_meta_album_pos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_metaFormat) );
+        }
     }
 
     return result;
