diff -Naur mythtv-master-20190105-g3382e1a-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-master-20190105-g3382e1a-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-master-20190105-g3382e1a-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2019-01-05 12:45:06.676666667 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2019-01-05 12:45:07.796666666 +0100
@@ -162,6 +162,12 @@
             if (m_meta_album_pos > 0)
                 result["album"] = rx.cap(m_meta_album_pos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_meta_format) );
+        }
     }
 
     return result;
