diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythbase/mythdb.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythbase/mythdb.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythbase/mythdb.cpp	2021-05-18 16:11:40.326666605 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythbase/mythdb.cpp	2021-05-18 16:11:43.526666605 +0200
@@ -430,6 +430,11 @@
     if (query.exec() && query.next())
     {
         value = query.value(0).toString();
+
+        LOG(VB_DATABASE, LOG_INFO,
+            QString("GetSetting: DB for:'%1' for host:'%2' returns:'%3'")
+            .arg(key).arg(d->m_localhostname).arg(value));
+
     }
     else
     {
@@ -442,6 +447,11 @@
         if (query.exec() && query.next())
         {
             value = query.value(0).toString();
+
+            LOG(VB_DATABASE, LOG_INFO,
+                QString("GetSetting: no:'%1' found for:'%2'. Asking with NULL hostname returns:'%3'")
+                .arg(key).arg(d->m_localhostname).arg(value));
+
         }
     }
 
