diff -Naur mythtv-master-20190316-g1a44f54-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20190316-g1a44f54-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20190316-g1a44f54-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-03-16 10:53:07.343333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-03-16 10:53:09.416666667 +0100
@@ -454,6 +454,11 @@
 
     // get the MMAP path
     m_mmapPath = getZMSetting("ZM_PATH_MAP");
+    if (m_mmapPath == "")
+    {
+        m_mmapPath = "/dev/shm";
+    }
+
     if (m_debug)
     {
         cout << "Memory path directory is: " << m_mmapPath << endl;
