diff -Naur mythtv-master-20190105-g3382e1a-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20190105-g3382e1a-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20190105-g3382e1a-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-01-05 12:45:07.743333334 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-01-05 12:45:09.153333333 +0100
@@ -463,6 +463,11 @@
 
     // get the MMAP path
     m_mmapPath = getZMSetting("ZM_PATH_MAP");
+    if (m_mmapPath == "")
+    {
+        m_mmapPath = "/dev/shm";
+    }
+
     if (m_debug)
     {
         cout << "Memory path directory is: " << m_mmapPath << endl;
