diff -Naur mythtv-master-20190316-g1a44f54-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-master-20190316-g1a44f54-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-master-20190316-g1a44f54-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-03-16 10:53:10.243333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2019-03-16 10:53:12.203333334 +0100
@@ -25,6 +25,7 @@
 #include <sys/stat.h>
 #include <sys/shm.h>
 #include <sys/mman.h>
+#include <algorithm>
 
 #ifdef linux
 #  include <sys/vfs.h>
@@ -1093,9 +1094,13 @@
     int frameNo = atoi(tokens[3].c_str());
     string eventTime(tokens[4]);
 
+    std::replace( eventTime.begin(), eventTime.end(), '/', '-');
+    std::string eventDate = eventTime.substr(0, 8);
+
     if (m_debug)
         cout << "Getting frame " << frameNo << " for event " << eventID
-             << " on monitor " << monitorID  << " event time is " << eventTime << endl;
+             << " on monitor " << monitorID  << " event time is " << eventTime
+             << " event date is " << eventDate << endl;
 
     string outStr("");
 
@@ -1107,7 +1112,7 @@
 
     if (m_useDeepStorage)
     {
-        filepath = g_webPath + "/events/" + monitorID + "/" + eventTime + "/";
+        filepath = g_webPath + "/events/" + monitorID + "/20" + eventDate + "/" + eventID + "/" ;
         sprintf(str, m_eventFileFormat.c_str(), frameNo);
         filepath += str;
     }
@@ -1160,9 +1165,13 @@
     int frameID;
     int frameCount = 0;
 
+    std::replace( eventTime.begin(), eventTime.end(), '/', '-');
+    std::string eventDate = eventTime.substr(0, 8);
+
     if (m_debug)
         cout << "Getting analysis frame " << frameNo << " for event " << eventID
-             << " on monitor " << monitorID << " event time is " << eventTime << endl;
+             << " on monitor " << monitorID << " event time is " << eventTime
+             << " event date is " << eventDate << endl;
 
     // get the 'alarm' frames from the Frames table for this event
     MYSQL_RES *res;
@@ -1240,7 +1249,7 @@
     string frameFile("");
 
     if (m_useDeepStorage)
-        filepath = g_webPath + "/events/" + monitorID + "/" + eventTime + "/";
+        filepath = g_webPath + "/events/" + monitorID + "/20" + eventDate + "/" + eventID + "/" ;
     else
         filepath = g_webPath + "/events/" + monitorID + "/" + eventID + "/";
 
