diff -Naur mythtv-old/mythplugins/mythweb/modules/stream/stream_flv.pl mythtv-new/mythplugins/mythweb/modules/stream/stream_flv.pl
--- mythtv-old/mythplugins/mythweb/modules/stream/stream_flv.pl	2018-01-11 14:34:14.000000000 +0100
+++ mythtv-new/mythplugins/mythweb/modules/stream/stream_flv.pl	2018-01-11 14:25:00.000000000 +0100
@@ -93,19 +93,25 @@
         $aspect = $sh->fetchrow_hashref;
         $sh->finish();
 
-        if( $aspect->{'type'} == 10 ) {
-            $x = $y = 1;
-        } elsif( $aspect->{'type'}== 11 ) {
-            $x = 4; $y = 3;
-        } elsif( $aspect->{'type'}== 12 ) {
-            $x = 16; $y = 9;
-        } elsif( $aspect->{'type'}== 13 ) {
-            $x = 2.21; $y = 1;
-        } elsif( $aspect->{'type'}== 14 ) {
-            $x = $aspect->{'data'}; $y = 1000000;
-        } else {
+        if( $aspect->{'type'} ) {
+            if( $aspect->{'type'} == 10 ) {
+                $x = $y = 1;
+            } elsif( $aspect->{'type'}== 11 ) {
+                $x = 4; $y = 3;
+            } elsif( $aspect->{'type'}== 12 ) {
+                $x = 16; $y = 9;
+            } elsif( $aspect->{'type'}== 13 ) {
+                $x = 2.21; $y = 1;
+            } elsif( $aspect->{'type'}== 14 ) {
+                $x = $aspect->{'data'}; $y = 1000000;
+            } else {
+                $x = 4; $y = 3;
+            }
+        }
+        else {
             $x = 4; $y = 3;
         }
+
     }
     $height = round_even($width * ($y/$x));
 
@@ -124,7 +130,7 @@
                         .' -deinterlace'
                         .' -async 2'
                         .' -ac 2'
-                        .' -ar 11025'
+                        .' -ar 44100'
                         .' -ab '.shell_escape("${abitrate}k")
                         .' -b '.shell_escape("${vbitrate}k")
                         .' /dev/stdout 2>/dev/null |';
