diff --speed-large-files --no-dereference --minimal -Naur mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/libs/libmythtv/tv_play.cpp mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/libs/libmythtv/tv_play.cpp	2024-10-04 04:31:01.000000000 +0200
+++ mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/libs/libmythtv/tv_play.cpp	2024-10-04 12:42:11.493998311 +0200
@@ -459,7 +459,7 @@
 void TV::InitKeys()
 {
     REG_KEY("TV Frontend", ACTION_PLAYBACK, QT_TRANSLATE_NOOP("MythControls",
-            "Play Program"), "P,Media Play");
+            "Play Program"), "Ctrl+P,Media Play");
     REG_KEY("TV Frontend", ACTION_STOP, QT_TRANSLATE_NOOP("MythControls",
             "Stop Program"), "");
     REG_KEY("TV Frontend", ACTION_TOGGLERECORD, QT_TRANSLATE_NOOP("MythControls",
@@ -781,7 +781,7 @@
     REG_KEY("TV Playback", ACTION_SWITCHANGLE, QT_TRANSLATE_NOOP("MythControls",
             "Switch angle"), "");
     REG_KEY("TV Playback", ACTION_OSDNAVIGATION, QT_TRANSLATE_NOOP("MythControls",
-            "OSD Navigation"), "");
+            "OSD Navigation"), "Select");
     REG_KEY("TV Playback", ACTION_ZOOMUP, QT_TRANSLATE_NOOP("MythControls",
             "Zoom mode - shift up"), "");
     REG_KEY("TV Playback", ACTION_ZOOMDOWN, QT_TRANSLATE_NOOP("MythControls",
@@ -885,7 +885,7 @@
 
     /* OSD playback information screen */
     REG_KEY("TV Playback", ACTION_TOGGLEOSDDEBUG,
-            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "");
+            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "Ctrl+I");
 
     /* 3D/Frame compatible/Stereoscopic TV */
     REG_KEY("TV Playback", ACTION_3DNONE,
diff --speed-large-files --no-dereference --minimal -Naur mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/libs/libmythui/mythmainwindow.cpp	2024-10-04 04:31:01.000000000 +0200
+++ mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/libs/libmythui/mythmainwindow.cpp	2024-10-04 12:44:01.367334573 +0200
@@ -804,7 +804,7 @@
     RegisterKey("Global", "PREVIOUS", QT_TRANSLATE_NOOP("MythControls",
         "Move to preview widget"), "Backtab");
     RegisterKey("Global", ACTION_SELECT, QT_TRANSLATE_NOOP("MythControls",
-        "Select"), "Return,Enter,Space");
+        "Select"), "Return,Enter,Space,Select");
     RegisterKey("Global", "BACKSPACE", QT_TRANSLATE_NOOP("MythControls",
         "Backspace"),       "Backspace");
     RegisterKey("Global", "ESCAPE", QT_TRANSLATE_NOOP("MythControls",
@@ -935,7 +935,7 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc,Back");
+        "Display System Exit Prompt"),      "Ctrl+Esc,Back");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
         "System Exit"),                     "");
     RegisterKey("Main Menu",    "STANDBYMODE",QT_TRANSLATE_NOOP("MythControls",
diff --speed-large-files --no-dereference --minimal -Naur mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/globalsettings.cpp	2024-10-04 12:45:49.514004114 +0200
+++ mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/globalsettings.cpp	2024-10-04 12:38:49.170659542 +0200
@@ -1751,7 +1751,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1805,7 +1805,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -1969,7 +1969,7 @@
     gs->setLabel(MainGeneralSettings::tr("Idle time before entering standby "
                                          "mode (minutes)"));
 
-    gs->setValue(90);
+    gs->setValue(0);
 
     gs->setHelpText(MainGeneralSettings::tr("Number of minutes to wait when "
                                             "the frontend is idle before "
@@ -2127,7 +2127,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -3323,7 +3323,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
diff --speed-large-files --no-dereference --minimal -Naur mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/mythfrontend.cpp mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/mythfrontend.cpp
--- mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/mythfrontend.cpp	2024-10-04 12:45:50.047337462 +0200
+++ mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/mythfrontend.cpp	2024-10-04 12:45:10.137336405 +0200
@@ -1749,7 +1749,7 @@
 
      // Gallery keybindings
      REG_KEY("Images", "PLAY", QT_TRANSLATE_NOOP("MythControls",
-         "Start/Stop Slideshow"), "P,Media Play");
+         "Start/Stop Slideshow"), "Ctrl+P,Media Play");
      REG_KEY("Images", "RECURSIVESHOW", QT_TRANSLATE_NOOP("MythControls",
          "Start Recursive Slideshow"), "R");
      REG_KEY("Images", "ROTRIGHT", QT_TRANSLATE_NOOP("MythControls",
diff --speed-large-files --no-dereference --minimal -Naur mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/videoplayersettings.cpp mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/videoplayersettings.cpp
--- mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/videoplayersettings.cpp	2024-10-04 04:31:01.000000000 +0200
+++ mythtv-b96c272e7aa25266340b89f2928144a9a479f82f/mythtv/programs/mythfrontend/videoplayersettings.cpp	2024-10-04 12:38:49.170659542 +0200
@@ -57,7 +57,7 @@
     m_dvdDriveEdit->SetText(gCoreContext->GetSetting("DVDDeviceLocation",
                            "default"));
     m_blurayMountEdit->SetText(gCoreContext->GetSetting("BluRayMountpoint",
-                           "/media/cdrom"));
+                           "/media/disk"));
     m_altPlayerEdit->SetText(gCoreContext->GetSetting(
                            "mythvideo.VideoAlternatePlayer", "Internal"));
 
