diff -Naur mythtv-master-20190316-g1a44f54-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp mythtv-master-20190316-g1a44f54-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp
--- mythtv-master-20190316-g1a44f54-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2019-03-16 10:53:19.303333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2019-03-16 10:53:21.316666667 +0100
@@ -45,7 +45,7 @@
 #ifdef Q_OS_MAC
     gc->setValue(QDir::homePath() + "/Pictures");
 #else
-    gc->setValue("/var/lib/pictures");
+    gc->setValue("/mnt/gallery");
 #endif
     gc->setHelpText(GallerySettings::tr("This directory must exist and "
                                         "MythGallery needs to have read "
diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythtv/tv_play.cpp	2019-03-16 10:53:19.663333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythtv/tv_play.cpp	2019-03-16 10:53:21.320000001 +0100
@@ -508,7 +508,7 @@
 void TV::InitKeys(void)
 {
     REG_KEY("TV Frontend", ACTION_PLAYBACK, QT_TRANSLATE_NOOP("MythControls",
-            "Play Program"), "P");
+            "Play Program"), "Ctrl+P");
     REG_KEY("TV Frontend", ACTION_STOP, QT_TRANSLATE_NOOP("MythControls",
             "Stop Program"), "");
     REG_KEY("TV Frontend", ACTION_TOGGLERECORD, QT_TRANSLATE_NOOP("MythControls",
@@ -928,7 +928,7 @@
 
     /* OSD playback information screen */
     REG_KEY("TV Playback", ACTION_TOGGLEOSDDEBUG,
-            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "");
+            QT_TRANSLATE_NOOP("MythControls", "Toggle OSD playback information"), "Ctrl+I");
 
     /* 3D/Frame compatible/Stereoscopic TV */
     REG_KEY("TV Playback", ACTION_3DNONE,
diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmythui/mythmainwindow.cpp	2019-03-16 10:53:19.650000001 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmythui/mythmainwindow.cpp	2019-03-16 10:53:21.320000001 +0100
@@ -1359,9 +1359,9 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc");
+        "Display System Exit Prompt"),      "");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
-        "System Exit"),                     "");
+        "System Exit"),                     "Ctrl+Esc");
     RegisterKey("Main Menu",    "STANDBYMODE",QT_TRANSLATE_NOOP("MythControls",
         "Enter Standby Mode"),              "");
     RegisterKey("Long Press",    "LONGPRESS1",QT_TRANSLATE_NOOP("MythControls",
diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20190316-g1a44f54-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/programs/mythfrontend/globalsettings.cpp	2019-03-16 10:53:19.363333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/programs/mythfrontend/globalsettings.cpp	2019-03-16 10:53:21.320000001 +0100
@@ -1645,7 +1645,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1717,7 +1717,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -1968,7 +1968,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -3132,7 +3132,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20190316-g1a44f54-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/programs/mythfrontend/main.cpp	2019-03-16 10:53:19.363333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/programs/mythfrontend/main.cpp	2019-03-16 10:53:21.323333334 +0100
@@ -1634,7 +1634,7 @@
 
      // Gallery keybindings
      REG_KEY("Images", "PLAY", QT_TRANSLATE_NOOP("MythControls",
-         "Start/Stop Slideshow"), "P");
+         "Start/Stop Slideshow"), "Ctrl+P");
      REG_KEY("Images", "RECURSIVESHOW", QT_TRANSLATE_NOOP("MythControls",
          "Start Recursive Slideshow"), "R");
      REG_KEY("Images", "ROTRIGHT", QT_TRANSLATE_NOOP("MythControls",
diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/programs/mythfrontend/videoplayersettings.cpp mythtv-master-20190316-g1a44f54-new/mythtv/programs/mythfrontend/videoplayersettings.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/programs/mythfrontend/videoplayersettings.cpp	2019-03-16 10:53:19.363333334 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/programs/mythfrontend/videoplayersettings.cpp	2019-03-16 10:53:21.323333334 +0100
@@ -56,7 +56,7 @@
     m_dvdDriveEdit->SetText(gCoreContext->GetSetting("DVDDeviceLocation",
                            "default"));
     m_blurayMountEdit->SetText(gCoreContext->GetSetting("BluRayMountpoint",
-                           "/media/cdrom"));
+                           "/media/disk"));
     m_altPlayerEdit->SetText(gCoreContext->GetSetting(
                            "mythvideo.VideoAlternatePlayer", "Internal"));
 
