diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/bindings/perl/Makefile mythtv-master-20210518-gc8c226c9b4-new/mythtv/bindings/perl/Makefile
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/bindings/perl/Makefile	2021-05-18 16:11:59.366666605 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/bindings/perl/Makefile	2021-05-18 16:12:02.526666605 +0200
@@ -6,7 +6,7 @@
 
 ifneq ($(PREFIX:/=), /usr)
 	ifneq ($(PREFIX:/=), /usr/local)
-		PREFIX_FLAGS=PREFIX="$(PREFIX)"
+		PREFIX_FLAGS=PREFIX="/usr"
 	endif
 endif
 
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/bindings/python/Makefile mythtv-master-20210518-gc8c226c9b4-new/mythtv/bindings/python/Makefile
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/bindings/python/Makefile	2021-05-18 16:11:59.363333272 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/bindings/python/Makefile	2021-05-18 16:12:02.533333272 +0200
@@ -19,7 +19,7 @@
 distclean: clean
 
 python_build: setup.py
-	$(PYTHON) setup.py build --mythtv-prefix="$(REAL_PREFIX)"
+	$(PYTHON) setup.py build --mythtv-prefix="/usr"
 
 install: setup.py
 	$(PYTHON) setup.py install --skip-build $(ROOT_FLAGS) $(PREFIX_FLAGS)
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/scripts/python_pathfix.py mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/scripts/python_pathfix.py
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/scripts/python_pathfix.py	2021-05-18 16:11:59.233333272 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/scripts/python_pathfix.py	2021-05-18 16:12:02.533333272 +0200
@@ -12,7 +12,7 @@
 def main():
 
     # Pick up default interpreter (ours)
-    python_interpreter = os.path.normcase(sys.executable).encode()
+    python_interpreter = os.path.normcase('/usr/bin/python').encode()
 
     # Don't bother reporting
     verbose = False
