diff --speed-large-files --no-dereference --minimal -Naur mythtv-dced3a83d1de1444798dd91c0cd8ef4741236ceb/mythtv/libs/libmythtv/audio/audiooutputalsa.cpp mythtv-dced3a83d1de1444798dd91c0cd8ef4741236ceb/mythtv/libs/libmythtv/audio/audiooutputalsa.cpp
--- mythtv-dced3a83d1de1444798dd91c0cd8ef4741236ceb/mythtv/libs/libmythtv/audio/audiooutputalsa.cpp	2025-12-14 20:23:09.000000000 +0100
+++ mythtv-dced3a83d1de1444798dd91c0cd8ef4741236ceb/mythtv/libs/libmythtv/audio/audiooutputalsa.cpp	2025-12-17 09:42:19.083362664 +0100
@@ -372,7 +372,7 @@
             case FORMAT_S16:    afmt = SND_PCM_FORMAT_S16;   break;
             // NOLINTNEXTLINE(bugprone-branch-clone)
             case FORMAT_S24LSB: afmt = SND_PCM_FORMAT_S24;   break;
-            case FORMAT_S24:    afmt = SND_PCM_FORMAT_S24;   break;
+            case FORMAT_S24:    afmt = SND_PCM_FORMAT_S32;   break;
             case FORMAT_S32:    afmt = SND_PCM_FORMAT_S32;   break;
             case FORMAT_FLT:    afmt = SND_PCM_FORMAT_FLOAT; break;
             default:         continue;
@@ -469,7 +469,7 @@
         case FORMAT_S16:    format = SND_PCM_FORMAT_S16;   break;
         // NOLINTNEXTLINE(bugprone-branch-clone)
         case FORMAT_S24LSB: format = SND_PCM_FORMAT_S24;   break;
-        case FORMAT_S24:    format = SND_PCM_FORMAT_S24;   break;
+        case FORMAT_S24:    format = SND_PCM_FORMAT_S32;   break;
         case FORMAT_S32:    format = SND_PCM_FORMAT_S32;   break;
         case FORMAT_FLT:    format = SND_PCM_FORMAT_FLOAT; break;
         default:
