diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/external/nv-codec-headers/Makefile mythtv-master-20210518-gc8c226c9b4-new/mythtv/external/nv-codec-headers/Makefile
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/external/nv-codec-headers/Makefile	2021-05-18 16:12:06.156666605 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/external/nv-codec-headers/Makefile	2021-05-18 16:12:09.483333271 +0200
@@ -11,13 +11,13 @@
 endif
 
 install: all
-	$(INSTALL) -m 0755 -d '$(DESTDIR)$(PREFIX)/include/ffnvcodec'
-	$(INSTALL) -m 0644 include/ffnvcodec/*.h '$(DESTDIR)$(PREFIX)/include/ffnvcodec'
-	$(INSTALL) -m 0755 -d '$(DESTDIR)$(PREFIX)/$(LIBDIR)/pkgconfig'
-	$(INSTALL) -m 0644 ffnvcodec.pc '$(DESTDIR)$(PREFIX)/$(LIBDIR)/pkgconfig'
+	$(INSTALL) -m 0755 -d '$(DESTDIR)/usr/include/ffnvcodec'
+	$(INSTALL) -m 0644 include/ffnvcodec/*.h '$(DESTDIR)/usr/include/ffnvcodec'
+	$(INSTALL) -m 0755 -d '$(DESTDIR)/usr/$(LIBDIR)/pkgconfig'
+	$(INSTALL) -m 0644 ffnvcodec.pc '$(DESTDIR)/usr/$(LIBDIR)/pkgconfig'
 
 uninstall:
-	rm -rf '$(DESTDIR)$(PREFIX)/include/ffnvcodec' '$(DESTDIR)$(PREFIX)/$(LIBDIR)/pkgconfig/ffnvcodec.pc'
+	rm -rf '$(DESTDIR)/include/ffnvcodec' '$(DESTDIR)/usr/$(LIBDIR)/pkgconfig/ffnvcodec.pc'
 
 .PHONY: all install uninstall
 
