diff -Naur mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-old/mythtv/configure mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-new/mythtv/configure
--- mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-old/mythtv/configure	2021-11-10 13:55:10.946358519 +0100
+++ mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-new/mythtv/configure	2021-11-10 14:00:37.963019287 +0100
@@ -125,6 +125,7 @@
   --disable-vbox           disable support for V@Box TV Gateway boxes
   --disable-ceton          disable support for Ceton cards
   --disable-satip          disable support for Sat>IP
+  --enable-v4l2request     enable Video4Linux V4l2-request support
   --disable-v4l2           disable Video4Linux support
   --disable-dvb            disable DVB support
   --dvb-path=HDRLOC        location of directory containing
@@ -2007,6 +2008,7 @@
     taglib
     v4l2
     v4l2prime
+    v4l2request
     valgrind
     x11
     libexiv2_external
@@ -2801,6 +2803,7 @@
 enable deprecation_warnings
 enable v4l2
 enable v4l2prime
+disable v4l2request
 enable x11
 disable indevs
 enable libfftw3
@@ -4768,6 +4771,7 @@
         disable dvb
         disable v4l2
         disable v4l2prime
+        disable v4l2request
         disable x11
         # Workaround compile errors from missing u_int/uint def
         CFLAGS=`echo $CFLAGS | sed 's/-D_POSIX_C_SOURCE=200112//'`
@@ -4817,6 +4821,7 @@
         disable symver
         disable v4l2
         disable v4l2prime
+        disable v4l2request
         enable  windows
         disable x11
         disable qtdbus
@@ -7303,6 +7308,10 @@
     ffopts="$ffopts --enable-lto"
 fi
 
+if enabled v4l2request; then
+    ffopts="$ffopts --enable-v4l2_request --enable-libdrm --enable-libudev"
+fi
+
 ffmpeg_extra_cflags="$extra_cflags -w"
 
 ## Call FFmpeg configure here
@@ -7513,7 +7522,7 @@
   if enabled drm; then
     echo "DRM Qt integration        ${qtprivateheaders-no}"
   fi
-  echo "Video4Linux codecs        ${v4l2-no} (DRM ${v4l2prime-no})"
+  echo "Video4Linux codecs        ${v4l2-no} (DRM:${v4l2prime-no} v4l2-request:${v4l2request-no})"
   echo "MMAL decoder support      ${mmal-no}"
   echo "OpenGL                    ${opengl-no} (OpenGLES ${opengles-no})"
   echo "EGL support               ${egl-no}"
diff -Naur mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-old/mythtv/libs/libmythtv/libmythtv.pro mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-new/mythtv/libs/libmythtv/libmythtv.pro
--- mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-old/mythtv/libs/libmythtv/libmythtv.pro	2021-11-08 20:58:02.000000000 +0100
+++ mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-new/mythtv/libs/libmythtv/libmythtv.pro	2021-11-10 14:01:30.069685001 +0100
@@ -525,6 +525,10 @@
         DEFINES += USING_V4L2PRIME
     }
 
+    using_v4l2request {
+        DEFINES += USING_V4L2_REQUEST
+    }
+
     using_vdpau:using_x11 {
         DEFINES += USING_VDPAU
         HEADERS += decoders/mythvdpaucontext.h   decoders/mythvdpauhelper.h
