diff -Naur mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-old/mythtv/configure mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-new/mythtv/configure
--- mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-old/mythtv/configure	2021-11-10 14:00:37.963019287 +0100
+++ mythtv-776fe08d1b42c1339ef7e7653fbd1cd958dbafba-new/mythtv/configure	2021-11-10 14:06:50.409679211 +0100
@@ -105,7 +105,9 @@
   --disable-avx            disable AVX optimizations
   --disable-yasm           disable use of yasm assembler
   --enable-pic             build position-independent code
+  --enable-lto             enable link-time optimizations (LTO)
   --enable-proc-opt        enable processor specific compilation
+  --disable-inlineasm      disable FFmpeg in-line assembler optimizations
   --enable-compdb          enable clang compiler database
 
   --disable-audio-oss      disable OSS audio support
@@ -2009,6 +2011,7 @@
     v4l2
     v4l2prime
     v4l2request
+    inlineasm
     valgrind
     x11
     libexiv2_external
@@ -2749,6 +2752,7 @@
 enable optimizations
 enable swscale_alpha
 enable valgrind_backtrace
+enable inlineasm
 
 sws_max_filter_size_default=256
 set_default sws_max_filter_size
@@ -7308,6 +7312,10 @@
     ffopts="$ffopts --enable-lto"
 fi
 
+if disabled inlineasm; then
+    ffopts="$ffopts --disable-inline-asm"
+fi
+
 if enabled v4l2request; then
     ffopts="$ffopts --enable-v4l2_request --enable-libdrm --enable-libudev"
 fi
@@ -7592,6 +7600,8 @@
 echo "C++ standard supported    ${CPP_STANDARD}"
 echo "Enforce c++11 nullptr     ${enforce_nullptr-no}"
 echo "Enforce shadowed vars     ${enforce_wshadow-no}"
+echo "Link time optimize (LTO)  ${lto-no}"
+echo "FFmpeg in-line asm        ${inlineasm-no}"
 echo ""
 
 # Remove redundant elements from variables
