From bbfd44b1c16603f958c609abc7bce92d96a91ff1 Mon Sep 17 00:00:00 2001
From: Mitch Capper <mitch.capper@gmail.com>
Date: Wed, 4 Dec 2024 03:31:39 -0800
Subject: [PATCH] fix QT>=6.4 segfault by always setting surface type

---
 mythtv/libs/libmythui/opengl/mythrenderopengl.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/mythtv/libs/libmythui/opengl/mythrenderopengl.cpp b/mythtv/libs/libmythui/opengl/mythrenderopengl.cpp
index e0bab337c81..f3f38f61c7d 100644
--- a/mythtv/libs/libmythui/opengl/mythrenderopengl.cpp
+++ b/mythtv/libs/libmythui/opengl/mythrenderopengl.cpp
@@ -541,12 +541,10 @@ void MythRenderOpenGL::SetWidget(QWidget *Widget)
         return;
     }
 
-#ifdef Q_OS_ANDROID
     // Ensure surface type is always OpenGL
     m_window->setSurfaceType(QWindow::OpenGLSurface);
     if (native && native->windowHandle())
         native->windowHandle()->setSurfaceType(QWindow::OpenGLSurface);
-#endif
 
 #ifdef USING_QTWEBENGINE
     auto * globalcontext = QOpenGLContext::globalShareContext();
