diff --speed-large-files --no-dereference --minimal -Naur mythtv-433e7aff2fe00520488f01de71c2344d79e84985/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp mythtv-433e7aff2fe00520488f01de71c2344d79e84985/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp
--- mythtv-433e7aff2fe00520488f01de71c2344d79e84985/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2024-12-23 16:55:24.580017812 +0100
+++ mythtv-433e7aff2fe00520488f01de71c2344d79e84985/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2024-12-23 16:54:56.956683738 +0100
@@ -507,7 +507,7 @@
 
     // N.B. Interop support should already have been checked
     // Create the device context
-    auto * hwdeviceref = MythCodecContext::CreateDevice(AV_HWDEVICE_TYPE_DRM, nullptr);
+    auto * hwdeviceref = MythCodecContext::CreateDevice(AV_HWDEVICE_TYPE_V4L2REQUEST, nullptr);
     if (!hwdeviceref)
         return -1;
 
