diff --speed-large-files --no-dereference --minimal -Naur mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/configure mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/configure
--- mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/configure	2026-02-26 15:15:53.025113362 +0100
+++ mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/configure	2026-02-26 15:14:54.811310427 +0100
@@ -4155,7 +4155,7 @@
 
 avcodec_extralibs="pthreads_extralibs iconv_extralibs dxva2_extralibs liblcevc_dec_extralibs lcms2_extralibs"
 avfilter_extralibs="pthreads_extralibs"
-avutil_extralibs="d3d11va_extralibs d3d12va_extralibs mediacodec_extralibs nanosleep_extralibs pthreads_extralibs v4l2_request_extralibs vaapi_drm_extralibs vaapi_x11_extralibs vaapi_win32_extralibs vdpau_x11_extralibs"
+avutil_extralibs="d3d11va_extralibs d3d12va_extralibs mediacodec_extralibs nanosleep_extralibs pthreads_extralibs vaapi_drm_extralibs vaapi_x11_extralibs vaapi_win32_extralibs vdpau_x11_extralibs"
 
 # programs
 ffmpeg_deps="avcodec avfilter avformat threads"
@@ -7395,7 +7395,6 @@
     check_func_headers "linux/media.h linux/videodev2.h" v4l2_timeval_to_ns
     check_pkg_config libudev libudev libudev.h udev_new
     check_struct linux/videodev2.h "struct v4l2_ctrl_hevc_decode_params" num_delta_pocs_of_ref_rps_idx
-    v4l2_request_extralibs="$libudev_extralibs"
 fi
 
 check_headers sys/videoio.h
diff --speed-large-files --no-dereference --minimal -Naur mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/hwcontext_v4l2request.c mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/hwcontext_v4l2request.c
--- mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/hwcontext_v4l2request.c	2026-02-26 15:15:52.992629885 +0100
+++ mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/hwcontext_v4l2request.c	2026-02-26 15:01:02.765886895 +0100
@@ -114,6 +114,34 @@
 #define V4L2_PIX_FMT_YUV420_10_AFBC_16X16_SPLIT v4l2_fourcc('A', 'S', '0', '1')
 #endif
 
+#ifndef DRM_FORMAT_NV15
+#define DRM_FORMAT_NV15         fourcc_code('N', 'V', '1', '5')
+#endif
+
+#ifndef DRM_FORMAT_NV20
+#define DRM_FORMAT_NV20         fourcc_code('N', 'V', '2', '0')
+#endif
+
+#ifndef DRM_FORMAT_P010
+#define DRM_FORMAT_P010         fourcc_code('P', '0', '1', '0')
+#endif
+
+#ifndef V4L2_PIX_FMT_NV12_32L32
+#define V4L2_PIX_FMT_NV12_32L32 v4l2_fourcc('S', 'T', '1', '2')
+#endif
+
+#ifndef V4L2_PIX_FMT_NV15
+#define V4L2_PIX_FMT_NV15       v4l2_fourcc('N', 'V', '1', '5')
+#endif
+
+#ifndef V4L2_PIX_FMT_NV20
+#define V4L2_PIX_FMT_NV20       v4l2_fourcc('N', 'V', '2', '0')
+#endif
+
+#ifndef V4L2_PIX_FMT_P010
+#define V4L2_PIX_FMT_P010       v4l2_fourcc('P', '0', '1', '0')
+#endif
+
 static const struct {
     uint32_t pixelformat;
     enum AVPixelFormat sw_format;
@@ -123,40 +151,18 @@
 } v4l2request_capture_pixelformats[] = {
     { V4L2_PIX_FMT_NV12, AV_PIX_FMT_NV12, DRM_FORMAT_NV12, DRM_FORMAT_MOD_LINEAR, 8 },
 #if defined(V4L2_PIX_FMT_NV12_32L32)
-    { V4L2_PIX_FMT_NV12_32L32, AV_PIX_FMT_YUV420P, DRM_FORMAT_NV12, DRM_FORMAT_MOD_ALLWINNER_TILED, 8 },
+    { V4L2_PIX_FMT_NV12_32L32, AV_PIX_FMT_NONE, DRM_FORMAT_NV12, DRM_FORMAT_MOD_ALLWINNER_TILED, 8 },
 #endif
 #if defined(V4L2_PIX_FMT_NV15) && defined(DRM_FORMAT_NV15)
-    { V4L2_PIX_FMT_NV15, AV_PIX_FMT_YUV420P10, DRM_FORMAT_NV15, DRM_FORMAT_MOD_LINEAR, 10 },
+    { V4L2_PIX_FMT_NV15, AV_PIX_FMT_NV15, DRM_FORMAT_NV15, DRM_FORMAT_MOD_LINEAR, 10 },
 #endif
     { V4L2_PIX_FMT_NV16, AV_PIX_FMT_NV16, DRM_FORMAT_NV16, DRM_FORMAT_MOD_LINEAR, 8 },
 #if defined(V4L2_PIX_FMT_NV20) && defined(DRM_FORMAT_NV20)
-    { V4L2_PIX_FMT_NV20, AV_PIX_FMT_YUV422P10, DRM_FORMAT_NV20, DRM_FORMAT_MOD_LINEAR, 10 },
+    { V4L2_PIX_FMT_NV20, AV_PIX_FMT_NV20BS, DRM_FORMAT_NV20, DRM_FORMAT_MOD_LINEAR, 10 },
 #endif
 #if defined(V4L2_PIX_FMT_P010) && defined(DRM_FORMAT_P010)
     { V4L2_PIX_FMT_P010, AV_PIX_FMT_P010, DRM_FORMAT_P010, DRM_FORMAT_MOD_LINEAR, 10 },
 #endif
-#if defined(V4L2_PIX_FMT_YUV420_8_AFBC_16X16_SPLIT)
-    {
-        .pixelformat = V4L2_PIX_FMT_YUV420_8_AFBC_16X16_SPLIT,
-        .sw_format = AV_PIX_FMT_YUV420P,
-        .drm_format = DRM_FORMAT_YUV420_8BIT,
-        .format_modifier = DRM_FORMAT_MOD_ARM_AFBC(AFBC_FORMAT_MOD_BLOCK_SIZE_16x16 |
-                                                   AFBC_FORMAT_MOD_SPARSE |
-                                                   AFBC_FORMAT_MOD_SPLIT),
-        .bit_depth = 8,
-    },
-#endif
-#if defined(V4L2_PIX_FMT_YUV420_10_AFBC_16X16_SPLIT)
-    {
-        .pixelformat = V4L2_PIX_FMT_YUV420_10_AFBC_16X16_SPLIT,
-        .sw_format = AV_PIX_FMT_YUV420P10,
-        .drm_format = DRM_FORMAT_YUV420_10BIT,
-        .format_modifier = DRM_FORMAT_MOD_ARM_AFBC(AFBC_FORMAT_MOD_BLOCK_SIZE_16x16 |
-                                                   AFBC_FORMAT_MOD_SPARSE |
-                                                   AFBC_FORMAT_MOD_SPLIT),
-        .bit_depth = 10,
-    },
-#endif
 #if defined(V4L2_PIX_FMT_NV12_COL128) && defined(V4L2_PIX_FMT_NV12_10_COL128)
     {
         .pixelformat = V4L2_PIX_FMT_NV12_COL128,
@@ -205,7 +211,7 @@
     }
 
     desc->nb_layers = 1;
-    layer->nb_planes = 1;
+    layer->nb_planes = 2;
 
     layer->planes[0].object_index = 0;
     layer->planes[0].offset = 0;
@@ -213,16 +219,13 @@
                              format->fmt.pix_mp.plane_fmt[0].bytesperline :
                              format->fmt.pix.bytesperline;
 
-    // AFBC formats only use 1 plane, remaining use 2 planes
-    if ((desc->objects[0].format_modifier >> 56) != DRM_FORMAT_MOD_VENDOR_ARM) {
-        layer->nb_planes = 2;
-        layer->planes[1].object_index = 0;
-        layer->planes[1].offset = layer->planes[0].pitch *
-                                  (V4L2_TYPE_IS_MULTIPLANAR(format->type) ?
-                                   format->fmt.pix_mp.height :
-                                   format->fmt.pix.height);
-        layer->planes[1].pitch = layer->planes[0].pitch;
-    }
+
+    layer->planes[1].object_index = 0;
+    layer->planes[1].offset = layer->planes[0].pitch *
+                              (V4L2_TYPE_IS_MULTIPLANAR(format->type) ?
+                               format->fmt.pix_mp.height :
+                               format->fmt.pix.height);
+    layer->planes[1].pitch = layer->planes[0].pitch;
 
 #if defined(V4L2_PIX_FMT_NV12_COL128) && defined(V4L2_PIX_FMT_NV12_10_COL128)
     // Raspberry Pi formats need special handling
@@ -1136,11 +1139,6 @@
     fmts[0] = hwfc->sw_format;
     fmts[1] = AV_PIX_FMT_NONE;
 
-    if (hwfc->sw_format == AV_PIX_FMT_YUV420P ||
-        hwfc->sw_format == AV_PIX_FMT_YUV420P10 ||
-        hwfc->sw_format == AV_PIX_FMT_YUV422P10)
-        fmts[0] = AV_PIX_FMT_NONE;
-
     *formats = fmts;
     return 0;
 }
@@ -1184,10 +1182,7 @@
     if (!(flags & AV_HWFRAME_MAP_READ))
         return AVERROR(ENOSYS);
 
-    if (hwfc->sw_format == AV_PIX_FMT_NONE ||
-        hwfc->sw_format == AV_PIX_FMT_YUV420P ||
-        hwfc->sw_format == AV_PIX_FMT_YUV420P10 ||
-        hwfc->sw_format == AV_PIX_FMT_YUV422P10)
+    if (hwfc->sw_format == AV_PIX_FMT_NONE)
         return AVERROR(ENOSYS);
     else if (dst->format == AV_PIX_FMT_NONE)
         dst->format = hwfc->sw_format;
diff --speed-large-files --no-dereference --minimal -Naur mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/pixdesc.c mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/pixdesc.c
--- mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/pixdesc.c	2026-02-14 09:51:19.000000000 +0100
+++ mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/pixdesc.c	2026-02-26 15:14:54.812975222 +0100
@@ -588,6 +588,18 @@
         },
         .flags = AV_PIX_FMT_FLAG_PLANAR,
     },
+    [AV_PIX_FMT_NV15] = {
+        .name = "nv15",
+        .nb_components = 3,
+        .log2_chroma_w = 1,
+        .log2_chroma_h = 1,
+        .comp = {
+            { 0, 10,  0, 0, 10 },      /* Y */
+            { 1, 20,  0, 0, 10 },      /* U */
+            { 1, 20, 10, 0, 10 },      /* V */
+        },
+        .flags = AV_PIX_FMT_FLAG_BITSTREAM | AV_PIX_FMT_FLAG_PLANAR,
+    },
     [AV_PIX_FMT_ARGB] = {
         .name = "argb",
         .nb_components = 4,
@@ -2238,6 +2250,18 @@
         },
         .flags = AV_PIX_FMT_FLAG_PLANAR,
     },
+    [AV_PIX_FMT_NV20BS] = {
+        .name = "nv20",
+        .nb_components = 3,
+        .log2_chroma_w = 1,
+        .log2_chroma_h = 0,
+        .comp = {
+            { 0, 10,  0, 0, 10 },      /* Y */
+            { 1, 20,  0, 0, 10 },      /* U */
+            { 1, 20, 10, 0, 10 },      /* V */
+        },
+        .flags = AV_PIX_FMT_FLAG_BITSTREAM | AV_PIX_FMT_FLAG_PLANAR,
+    },
     [AV_PIX_FMT_NV20LE] = {
         .name = "nv20le",
         .nb_components = 3,
@@ -2262,6 +2286,18 @@
         },
         .flags = AV_PIX_FMT_FLAG_PLANAR | AV_PIX_FMT_FLAG_BE,
     },
+    [AV_PIX_FMT_NV30] = {
+        .name = "nv30",
+        .nb_components = 3,
+        .log2_chroma_w = 0,
+        .log2_chroma_h = 0,
+        .comp = {
+            { 0, 10,  0, 0, 10 },      /* Y */
+            { 1, 20,  0, 0, 10 },      /* U */
+            { 1, 20, 10, 0, 10 },      /* V */
+        },
+        .flags = AV_PIX_FMT_FLAG_BITSTREAM | AV_PIX_FMT_FLAG_PLANAR,
+    },
     [AV_PIX_FMT_QSV] = {
         .name = "qsv",
         .flags = AV_PIX_FMT_FLAG_HWACCEL,
diff --speed-large-files --no-dereference --minimal -Naur mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/pixfmt.h mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/pixfmt.h
--- mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/pixfmt.h	2026-02-14 09:51:19.000000000 +0100
+++ mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libavutil/pixfmt.h	2026-02-26 15:14:54.813142532 +0100
@@ -499,6 +499,10 @@
 
     AV_PIX_FMT_OHCODEC, /// hardware decoding through openharmony
 
+    AV_PIX_FMT_NV15,        ///< packed planar YUV 4:2:0, 15bpp, 1 plane for Y and 1 plane for the UV components, which are interleaved (first component U and the following component V), no padding between components
+    AV_PIX_FMT_NV20BS,      ///< packed planar YUV 4:2:2, 20bpp, 1 plane for Y and 1 plane for the UV components, which are interleaved (first component U and the following component V), no padding between components
+    AV_PIX_FMT_NV30,        ///< packed planar YUV 4:4:4, 30bpp, 1 plane for Y and 1 plane for the UV components, which are interleaved (first component U and the following component V), no padding between components
+
     AV_PIX_FMT_NB         ///< number of pixel formats, DO NOT USE THIS if you want to link with shared libav* because the number of formats might differ between versions
 };
 
diff --speed-large-files --no-dereference --minimal -Naur mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libswscale/format.c mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libswscale/format.c
--- mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libswscale/format.c	2026-02-14 09:51:19.000000000 +0100
+++ mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libswscale/format.c	2026-02-26 15:14:54.813233523 +0100
@@ -209,6 +209,9 @@
     [AV_PIX_FMT_XYZ12LE]        = { 1, 1, 1 },
     [AV_PIX_FMT_AYUV64LE]       = { 1, 1},
     [AV_PIX_FMT_AYUV64BE]       = { 1, 1 },
+    [AV_PIX_FMT_NV15]        = { 1, 0 },
+    [AV_PIX_FMT_NV20BS]      = { 1, 0 },
+    [AV_PIX_FMT_NV30]        = { 1, 0 },
     [AV_PIX_FMT_P010LE]         = { 1, 1 },
     [AV_PIX_FMT_P010BE]         = { 1, 1 },
     [AV_PIX_FMT_P012LE]         = { 1, 1 },
diff --speed-large-files --no-dereference --minimal -Naur mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libswscale/input.c mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libswscale/input.c
--- mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libswscale/input.c	2026-02-14 09:51:19.000000000 +0100
+++ mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/libswscale/input.c	2026-02-26 15:14:54.813363823 +0100
@@ -946,6 +946,35 @@
     nvXXtoUV_c(dstV, dstU, src1, width);
 }
 
+static av_always_inline void nv15ToX_c(uint8_t *dst, const uint8_t *src, int pos)
+{
+    int shift = (pos & 3) << 1;
+    pos = (pos * 5) >> 2;
+    AV_WN16(dst, ((src[pos + 0] >> shift) |
+                  (src[pos + 1] << (8 - shift))) & 0x3ff);
+}
+
+static void nv15ToY_c(uint8_t *dst, const uint8_t *src,
+                      const uint8_t *unused1, const uint8_t *unused2,
+                      int width, uint32_t *unused, void *opq)
+{
+    int i;
+    for (i = 0; i < width; i++) {
+        nv15ToX_c(dst + i * 2, src, i);
+    }
+}
+
+static void nv15ToUV_c(uint8_t *dstU, uint8_t *dstV,
+                       const uint8_t *unused0, const uint8_t *src1, const uint8_t *src2,
+                       int width, uint32_t *unused, void *opq)
+{
+    int i;
+    for (i = 0; i < width; i++) {
+        nv15ToX_c(dstU + i * 2, src1, i * 2 + 0);
+        nv15ToX_c(dstV + i * 2, src1, i * 2 + 1);
+    }
+}
+
 #define p01x_uv_wrapper(fmt, shift) \
     static void fmt ## LEToUV ## _c(uint8_t *dstU,                       \
                                        uint8_t *dstV,                    \
@@ -2031,6 +2060,11 @@
     case AV_PIX_FMT_NV20LE:
         *chrToYV12 = nv20LEToUV_c;
         break;
+    case AV_PIX_FMT_NV15:
+    case AV_PIX_FMT_NV20BS:
+    case AV_PIX_FMT_NV30:
+        c->chrToYV12 = nv15ToUV_c;
+        break;
     case AV_PIX_FMT_P010LE:
     case AV_PIX_FMT_P210LE:
     case AV_PIX_FMT_P410LE:
@@ -2619,6 +2653,11 @@
     case AV_PIX_FMT_NV20LE:
         *lumToYV12 = nv20LEToY_c;
         break;
+    case AV_PIX_FMT_NV15:
+    case AV_PIX_FMT_NV20BS:
+    case AV_PIX_FMT_NV30:
+        c->lumToYV12 = nv15ToY_c;
+        break;
     case AV_PIX_FMT_P010LE:
     case AV_PIX_FMT_P210LE:
     case AV_PIX_FMT_P410LE:
diff --speed-large-files --no-dereference --minimal -Naur mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/tests/ref/fate/imgutils mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/tests/ref/fate/imgutils
--- mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/tests/ref/fate/imgutils	2026-02-14 09:51:19.000000000 +0100
+++ mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/tests/ref/fate/imgutils	2026-02-26 15:14:54.813515523 +0100
@@ -306,6 +306,9 @@
 gbrp10msble     planes: 3, linesizes: 128 128 128   0, plane_sizes:  6144  6144  6144     0, plane_offsets:  6144  6144     0, total_size: 18432
 gbrp12msbbe     planes: 3, linesizes: 128 128 128   0, plane_sizes:  6144  6144  6144     0, plane_offsets:  6144  6144     0, total_size: 18432
 gbrp12msble     planes: 3, linesizes: 128 128 128   0, plane_sizes:  6144  6144  6144     0, plane_offsets:  6144  6144     0, total_size: 18432
+nv15            planes: 2, linesizes:  80  80   0   0, plane_sizes:  3840  1920     0     0, plane_offsets:  3840     0     0, total_size: 5760
+nv20            planes: 2, linesizes:  80  80   0   0, plane_sizes:  3840  3840     0     0, plane_offsets:  3840     0     0, total_size: 7680
+nv30            planes: 2, linesizes:  80 160   0   0, plane_sizes:  3840  7680     0     0, plane_offsets:  3840     0     0, total_size: 11520
 
 image_fill_black tests
 yuv420p         total_size:   4608,  black_unknown_crc: 0xd00f6cc6,  black_tv_crc: 0xd00f6cc6,  black_pc_crc: 0x234969af
@@ -559,3 +562,6 @@
 gbrp10msble     total_size:  18432,  black_unknown_crc: 0x00000000,  black_tv_crc: 0x00000000,  black_pc_crc: 0x00000000
 gbrp12msbbe     total_size:  18432,  black_unknown_crc: 0x00000000,  black_tv_crc: 0x00000000,  black_pc_crc: 0x00000000
 gbrp12msble     total_size:  18432,  black_unknown_crc: 0x00000000,  black_tv_crc: 0x00000000,  black_pc_crc: 0x00000000
+nv15            total_size:   5760,  black_unknown_crc: 0x6b5fdb58,  black_tv_crc: 0x6b5fdb58,  black_pc_crc: 0x660a512c
+nv20            total_size:   7680,  black_unknown_crc: 0x171f53da,  black_tv_crc: 0x171f53da,  black_pc_crc: 0xfcf5cda3
+nv30            total_size:  11520,  black_unknown_crc: 0x664d0699,  black_tv_crc: 0x664d0699,  black_pc_crc: 0x4ab5d724
diff --speed-large-files --no-dereference --minimal -Naur mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/tests/ref/fate/sws-pixdesc-query mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/tests/ref/fate/sws-pixdesc-query
--- mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/tests/ref/fate/sws-pixdesc-query	2026-02-14 09:51:19.000000000 +0100
+++ mythtv-a535f9e4136f086f0a73f9f8eeb0cdb2b8423bd6/mythtv/external/FFmpeg/tests/ref/fate/sws-pixdesc-query	2026-02-26 15:14:54.813623473 +0100
@@ -79,8 +79,11 @@
   gray14le
   gray9be
   gray9le
+  nv15
+  nv20
   nv20be
   nv20le
+  nv30
   p010be
   p010le
   p012be
@@ -263,11 +266,14 @@
   ayuv64be
   ayuv64le
   nv12
+  nv15
   nv16
+  nv20
   nv20be
   nv20le
   nv21
   nv24
+  nv30
   nv42
   p010be
   p010le
@@ -395,11 +401,14 @@
 
 isPlanarYUV:
   nv12
+  nv15
   nv16
+  nv20
   nv20be
   nv20le
   nv21
   nv24
+  nv30
   nv42
   p010be
   p010le
@@ -496,11 +505,14 @@
 
 isSemiPlanarYUV:
   nv12
+  nv15
   nv16
+  nv20
   nv20be
   nv20le
   nv21
   nv24
+  nv30
   nv42
   p010be
   p010le
@@ -1011,11 +1023,14 @@
   gbrpf32be
   gbrpf32le
   nv12
+  nv15
   nv16
+  nv20
   nv20be
   nv20le
   nv21
   nv24
+  nv30
   nv42
   p010be
   p010le
