diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2021-05-18 16:12:21.253333272 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2021-05-18 16:12:28.359999938 +0200
@@ -199,8 +199,10 @@
     // best guess currently - this matches the number of capture buffers to the
     // number of output buffers - and hence to the number of video buffers for
     // direct rendering
-    LOG(VB_PLAYBACK, LOG_INFO, LOC + "Setting number of capture buffers to 6");
-    av_opt_set_int(Context->priv_data, "num_capture_buffers", 6, 0);
+    LOG(VB_PLAYBACK, LOG_INFO, LOC + "Setting number of capture buffers to 16");
+    av_opt_set_int(Context->priv_data, "num_capture_buffers", 16, 0);
+    LOG(VB_PLAYBACK, LOG_INFO, LOC + "Setting number of output buffers to 8");
+    av_opt_set_int(Context->priv_data, "num_output_buffers", 8, 0);
 }
 
 /*! \brief Retrieve a frame from CPU memory
