diff --speed-large-files --no-dereference --minimal -Naur mythtv-675d5944ce23c8bbe0629750b5aeac63dac09fca/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c mythtv-675d5944ce23c8bbe0629750b5aeac63dac09fca/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c
--- mythtv-675d5944ce23c8bbe0629750b5aeac63dac09fca/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c	2021-12-14 19:11:45.000000000 +0100
+++ mythtv-675d5944ce23c8bbe0629750b5aeac63dac09fca/mythtv/external/FFmpeg/libavcodec/v4l2_m2m_dec.c	2021-12-16 15:07:55.559640571 +0100
@@ -252,9 +282,13 @@
     V4L2m2mPriv *priv = avctx->priv_data;
     V4L2m2mContext* s = priv->context;
 
+    // Clear any buffered input packet
+    av_packet_unref(&s->buf_pkt);
+
     /* wait for pending buffer references */
     if (atomic_load(&s->refcount))
         while(sem_wait(&s->refsync) == -1 && errno == EINTR);
+
 }
 
 #define OFFSET(x) offsetof(V4L2m2mPriv, x)
