diff --speed-large-files --no-dereference --minimal -Naur mythtv-6705a26e463a07f866ff93bcac955aa7166906cb/mythtv/external/FFmpeg/libavcodec/hevc/hevcdec.h mythtv-6705a26e463a07f866ff93bcac955aa7166906cb/mythtv/external/FFmpeg/libavcodec/hevc/hevcdec.h
--- mythtv-6705a26e463a07f866ff93bcac955aa7166906cb/mythtv/external/FFmpeg/libavcodec/hevc/hevcdec.h	2024-10-15 07:14:12.000000000 +0200
+++ mythtv-6705a26e463a07f866ff93bcac955aa7166906cb/mythtv/external/FFmpeg/libavcodec/hevc/hevcdec.h	2024-10-20 09:21:05.580001953 +0200
@@ -518,6 +518,7 @@
     int temporal_id;  ///< temporal_id_plus1 - 1
     HEVCFrame *cur_frame;
     HEVCFrame *collocated_ref;
+    HEVCFrame DPB[32];
     int poc;
     int poc_tid0;
     int slice_idx; ///< number of the slice being currently decoded
