diff --speed-large-files --no-dereference --minimal -Naur mythtv-f8131413d5fff7f291e0a68bfc28cb1666835a99/mythtv/external/FFmpeg/libavutil/fifo.c mythtv-f8131413d5fff7f291e0a68bfc28cb1666835a99/mythtv/external/FFmpeg/libavutil/fifo.c
--- mythtv-f8131413d5fff7f291e0a68bfc28cb1666835a99/mythtv/external/FFmpeg/libavutil/fifo.c	2022-09-10 15:08:30.000000000 +0200
+++ mythtv-f8131413d5fff7f291e0a68bfc28cb1666835a99/mythtv/external/FFmpeg/libavutil/fifo.c	2022-09-18 14:07:51.556296976 +0200
@@ -29,8 +29,8 @@
 #include "macros.h"
 #include "mem.h"
 
-// by default the FIFO can be auto-grown to 1MB
-#define AUTO_GROW_DEFAULT_BYTES (1024 * 1024)
+// by default the FIFO can be auto-grown to 64MB
+#define AUTO_GROW_DEFAULT_BYTES (64 * 1024 * 1024)
 
 struct AVFifo {
     uint8_t *buffer;
