diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-16 10:53:21.146666667 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-03-16 10:53:22.756666667 +0100
@@ -786,6 +786,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
