diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-01-05 12:45:18.170000001 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-01-05 12:45:19.763333333 +0100
@@ -770,6 +770,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
