diff -Naur mythtv-master-20200207-g2869463-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20200207-g2869463-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20200207-g2869463-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-02-08 14:19:25.996666669 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-02-08 14:19:27.553333336 +0100
@@ -780,6 +780,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
