diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmyth/mediamonitor-unix.cpp mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmyth/mediamonitor-unix.cpp
--- mythtv-master-20190316-g1a44f54-old/mythtv/libs/libmyth/mediamonitor-unix.cpp	2019-03-16 10:53:22.540000001 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/libs/libmyth/mediamonitor-unix.cpp	2019-03-16 10:53:24.346666667 +0100
@@ -196,7 +196,7 @@
 bool MediaMonitorUnix::CheckMountable(void)
 {
 #if CONFIG_QTDBUS
-    for (int i = 0; i < 10; ++i, usleep(500000))
+    for (int i = 0; i < 3; ++i, usleep(500000))
     {
         // Connect to UDisks.  This can sometimes fail if mythfrontend
         // is started during system init
