diff -Naur mythtv-master-20190408-g038e171-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20190408-g038e171-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20190408-g038e171-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-04-08 14:45:25.913333333 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-04-08 14:45:27.210000001 +0200
@@ -463,10 +463,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
