diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20190216-g938606f-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-02-17 16:58:32.433333336 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2019-02-17 16:58:33.996666669 +0100
@@ -463,10 +463,6 @@
     // Sanity check
     if (m_Active)
         return;
-    if (!gCoreContext->GetBoolSetting("MonitorDrives", false)) {
-        LOG(VB_MEDIA, LOG_NOTICE, "MediaMonitor diasabled by user setting.");
-        return;
-    }
 
     if (!m_Thread)
         m_Thread = new MonitorThread(this, m_MonitorPollingInterval);
