diff -Naur mythtv-master-20190408-g038e171-old/mythtv/configure mythtv-master-20190408-g038e171-new/mythtv/configure
--- mythtv-master-20190408-g038e171-old/mythtv/configure	2019-04-08 14:45:27.880000000 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/configure	2019-04-08 14:45:29.286666667 +0200
@@ -5186,7 +5186,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.3"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
