diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythfrontend/mythfrontend.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythfrontend/mythfrontend.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythfrontend/mythfrontend.cpp	2021-05-18 16:12:55.269999937 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythfrontend/mythfrontend.cpp	2021-05-18 16:12:58.469999936 +0200
@@ -827,8 +827,22 @@
     switch (gCoreContext->GetNumSetting("DVDOnInsertDVD", 1))
     {
         case 0 : // Do nothing
-        case 1 : // Display menu (mythdvd)*/
-            break;
+        case 1 : // Display menu (mythdvd)
+        {
+            MythThemedMenuState *menu = dynamic_cast<MythThemedMenuState *>
+                (GetMythMainWindow()->GetMainStack()->GetTopScreen());
+            if (menu)
+            {
+                menu->m_buttonList->MoveToNamedPosition("Videos");
+                QKeyEvent key(QEvent::KeyPress, Qt::Key_Return, Qt::NoModifier);
+                menu->m_buttonList->keyPressEvent(&key);
+                menu = dynamic_cast<MythThemedMenuState *>
+                    (GetMythMainWindow()->GetMainStack()->GetTopScreen());
+                if (menu)
+                    menu->m_buttonList->MoveToNamedPosition("Play a DVD or Blu-ray");
+            }
+             break;
+        }
         case 2 : // play DVD or Blu-ray
             playDisc();
             break;
