diff -Naur mythtv-test-20190716-gcb60da0-old/mythtv/libs/libmythtv/mythv4l2m2mcontext.cpp mythtv-test-20190716-gcb60da0-new/mythtv/libs/libmythtv/mythv4l2m2mcontext.cpp
--- mythtv-test-20190716-gcb60da0-old/mythtv/libs/libmythtv/mythv4l2m2mcontext.cpp	2019-07-16 09:58:12.000000000 +0200
+++ mythtv-test-20190716-gcb60da0-new/mythtv/libs/libmythtv/mythv4l2m2mcontext.cpp	2019-08-09 13:43:51.233333328 +0200
@@ -272,5 +272,6 @@
 
     if (!Codec)
         return !s_supportedV4L2Codecs.isEmpty();
-    return s_supportedV4L2Codecs.contains(Codec);
+    //return s_supportedV4L2Codecs.contains(Codec);
+    return true;
 }
