diff -Naur mythtv-test-20191105-gbe7604c-old/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp mythtv-test-20191105-gbe7604c-new/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp
--- mythtv-test-20191105-gbe7604c-old/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2019-11-05 12:29:01.000000000 +0100
+++ mythtv-test-20191105-gbe7604c-new/mythtv/libs/libmythtv/decoders/mythv4l2m2mcontext.cpp	2019-11-05 12:47:32.846666658 +0100
@@ -215,7 +215,10 @@
             bool splanarm2m = caps & V4L2_CAP_VIDEO_M2M;
 
             if (!(mplanar || mplanarm2m || splanar || splanarm2m))
+            {
+                LOG(VB_PLAYBACK, LOG_INFO, LOC + QString("Device: %1 No mplanar/mplanarm2m/splanar/splanarm2m found. Skiping this device!").arg(v4l2dev.GetDeviceName()));
                 continue;
+            }
 
             v4l2_buf_type capturetype = V4L2_BUF_TYPE_VIDEO_CAPTURE;
             v4l2_buf_type outputtype = V4L2_BUF_TYPE_VIDEO_OUTPUT;
@@ -269,6 +272,12 @@
                             debug.append(avcodec_get_name(codec));
                             break;
                         }
+                        else
+                        {
+                            LOG(VB_GENERAL, LOG_INFO, LOC + QString("V4L2 disabled as %1 has no YUV420/YUV420M/NV12/NV12M support for %2.")
+                                .arg(v4l2dev.GetDeviceName())
+                                .arg(avcodec_get_name(codec)));
+                        }
                         fdesc.index++;
                     }
                 }
