diff -Naur mythtv-master-20190316-g1a44f54-old/mythtv/configure mythtv-master-20190316-g1a44f54-new/mythtv/configure
--- mythtv-master-20190316-g1a44f54-old/mythtv/configure	2019-03-16 10:53:35.980000000 +0100
+++ mythtv-master-20190316-g1a44f54-new/mythtv/configure	2019-03-16 10:53:37.666666667 +0100
@@ -5828,28 +5828,28 @@
 enabled omx_rpi && enable openmax
 
 if  enabled openmax ; then
-    openmax_was_enabled=Y
-fi
-if ! disabled openmax; then
-    test -f /opt/vc/include/IL/OMX_Core.h && enable openmax
-fi
-if ! disabled openmax; then
-    if test -f /opt/vc/include/IL/OMX_Broadcom.h \
-        && test -f /opt/vc/lib/libopenmaxil.so ; then
+#    openmax_was_enabled=Y
+#fi
+#if ! disabled openmax; then
+#    test -f /opt/vc/include/IL/OMX_Core.h && enable openmax
+#fi
+#if ! disabled openmax; then
+#    if test -f /opt/vc/include/IL/OMX_Broadcom.h \
+#        && test -f /opt/vc/lib/libopenmaxil.so ; then
         enable openmax_broadcom
-    elif test -f /usr/include/bellagio/omxcore.h ; then
-        if test -f /usr/lib/libomxil-bellagio.so \
-           -o -f /usr/lib32/libomxil-bellagio.so \
-           -o -f /usr/lib64/libomxil-bellagio.so ; then
+#    elif test -f /usr/include/bellagio/omxcore.h ; then
+#        if test -f /usr/lib/libomxil-bellagio.so \
+#           -o -f /usr/lib32/libomxil-bellagio.so \
+#           -o -f /usr/lib64/libomxil-bellagio.so ; then
             enable openmax
-            enable openmax_bellagio
-            add_extralibs -lomxil-bellagio
-        else
-            disable openmax
-        fi
+#            enable openmax_bellagio
+#            add_extralibs -lomxil-bellagio
+#        else
+#            disable openmax
+#        fi
     else
         disable openmax
-    fi
+#    fi
 fi
 
 if  test "$openmax_was_enabled" = "Y" && disabled openmax ; then
