diff -Naur mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20190408-g038e171-old/mythtv/libs/libmythtv/osd.cpp	2019-04-08 14:45:35.566666667 +0200
+++ mythtv-master-20190408-g038e171-new/mythtv/libs/libmythtv/osd.cpp	2019-04-08 14:45:37.023333334 +0200
@@ -566,7 +566,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
