diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/osd.cpp	2019-02-17 16:58:44.200000002 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/osd.cpp	2019-02-17 16:58:45.756666669 +0100
@@ -565,7 +565,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
