diff -Naur mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-master-20190216-g938606f-old/mythtv/libs/libmythtv/mythplayer.cpp	2019-02-17 16:58:49.753333336 +0100
+++ mythtv-master-20190216-g938606f-new/mythtv/libs/libmythtv/mythplayer.cpp	2019-02-17 16:58:51.470000002 +0100
@@ -5119,13 +5119,13 @@
     if (decoder)
         infoMap["videocodecdesc"] = decoder->GetRawEncodingType();
     infoMap["videowidth"]     = QString::number(width);
-    infoMap["videoheight"]    = QString::number(height);
+    bool interlaced = is_interlaced(m_scan);
+    infoMap["videoheight"]    = QString::number(height) + (interlaced ? "i" : "");
     infoMap["videoframerate"] = QString::number(video_frame_rate, 'f', 2);
 
     if (width < 640)
         return;
 
-    bool interlaced = is_interlaced(m_scan);
     if (width == 1920 || height == 1080 || height == 1088)
         infoMap["videodescrip"] = interlaced ? "HD_1080_I" : "HD_1080_P";
     else if ((width == 1280 || height == 720) && !interlaced)
