diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythmusic/mythmusic/flacencoder.cpp mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythmusic/mythmusic/flacencoder.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythmusic/mythmusic/flacencoder.cpp	2021-05-18 16:13:21.043333270 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythmusic/mythmusic/flacencoder.cpp	2021-05-18 16:13:23.359999937 +0200
@@ -27,6 +27,11 @@
                          MusicMetadata *metadata)
            : Encoder(outfile, qualitylevel, metadata)
 {
+    if (m_out)
+    {
+        fclose(m_out);
+        m_out = NULL;
+    }
     bool streamable_subset = true;
     bool do_mid_side = true;
     bool loose_mid_side = false;
@@ -71,6 +76,8 @@
         LOG(VB_GENERAL, LOG_ERR,
             QString("Error initializing FLAC encoder. Got return code: %1")
                 .arg(ret));
+        encoder_delete(m_encoder);
+        m_encoder = NULL;
     }
 
     for (auto & chan : m_inputIn)
@@ -96,6 +103,9 @@
 
 int FlacEncoder::addSamples(int16_t *bytes, unsigned int length)
 {
+    if (!m_encoder)
+        return EENCODEERROR;
+
     unsigned int index = 0;
 
     length /= sizeof(int16_t);
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythmusic/mythmusic/flacencoder.h mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythmusic/mythmusic/flacencoder.h
--- mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythmusic/mythmusic/flacencoder.h	2021-05-18 16:13:21.043333270 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythmusic/mythmusic/flacencoder.h	2021-05-18 16:13:23.366666603 +0200
@@ -85,6 +85,7 @@
     FlacEncoder(const QString &outfile, int qualitylevel, MusicMetadata *metadata);
    ~FlacEncoder() override;
     int addSamples(int16_t *bytes, unsigned int len) override; // Encoder
+    virtual bool isValid() { return !!m_encoder; }
 
   private:
     FLAC_ENCODER *m_encoder    {nullptr};
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp	2021-05-18 16:13:21.043333270 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythmusic/mythmusic/vorbisencoder.cpp	2021-05-18 16:13:23.366666603 +0200
@@ -89,6 +89,11 @@
 VorbisEncoder::~VorbisEncoder()
 {
     VorbisEncoder::addSamples(nullptr, 0); //flush
+    if (m_out)
+    {
+        fclose(m_out);
+        m_out = NULL;
+    }
     ogg_stream_clear(&m_os);
     vorbis_block_clear(&m_vb);
     vorbis_dsp_clear(&m_vd);
