diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2021-05-18 16:13:23.329999937 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2021-05-18 16:13:25.663333270 +0200
@@ -756,11 +756,24 @@
 
 void Ripper::ScanFinished()
 {
+    m_scanThread->wait();
     delete m_scanThread;
     m_scanThread = nullptr;
 
     m_tracks->clear();
 
+    if (m_scanThread)
+    {
+        m_scanThread->wait();
+        delete m_scanThread;
+    }
+
+    if (m_ejectThread)
+    {
+        m_ejectThread->wait();
+        delete m_ejectThread;
+    }
+
     if (m_decoder)
     {
         bool isCompilation = false;
@@ -1153,6 +1166,7 @@
 
 void Ripper::EjectFinished()
 {
+    m_ejectThread->wait();
     delete m_ejectThread;
     m_ejectThread = nullptr;
 
