diff --git a/mythtv/libs/libmythtv/decoders/avformatdecoder.cpp b/mythtv/libs/libmythtv/decoders/avformatdecoder.cpp
index 1f81cf4bd1..0566dddd41 100644
--- a/mythtv/libs/libmythtv/decoders/avformatdecoder.cpp
+++ b/mythtv/libs/libmythtv/decoders/avformatdecoder.cpp
@@ -5240,7 +5240,16 @@ inline bool AvFormatDecoder::DecoderWillDownmix(const AVCodecContext *ctx)
     if (m_audio->CanDownmix() && AudioOutputUtil::has_optimized_SIMD())
         return false;
     // use ffmpeg only for dolby codecs if we have to
-    return av_opt_find(ctx->priv_data, "downmix", nullptr, 0, 0);
+    //return av_opt_find(ctx->priv_data, "downmix", nullptr, 0, 0);
+    switch (ctx->codec_id)
+    {
+        case AV_CODEC_ID_AC3:
+        case AV_CODEC_ID_TRUEHD:
+        case AV_CODEC_ID_EAC3:
+            return true;
+        default:
+            return false;
+    }
 }
 
 bool AvFormatDecoder::DoPassThrough(const AVCodecParameters *par, bool withProfile)
