From 6e77559bef4f3cabeaa746c1bd2096309c8c94dd Mon Sep 17 00:00:00 2001
From: iuncuim <iuncuim@gmail.com>
Date: Mon, 7 Apr 2025 03:26:59 +0300
Subject: [PATCH 1/2] axp20x: HACK: mfd and multiple axp on the bus

Signed-off-by: Mikhail Kalashnikov <iuncuim@gmail.com>
---
 drivers/mfd/axp20x.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mfd/axp20x.c b/drivers/mfd/axp20x.c
index e9914e8a2..0d102ad0b 100644
--- a/drivers/mfd/axp20x.c
+++ b/drivers/mfd/axp20x.c
@@ -1448,6 +1448,10 @@ int axp20x_device_probe(struct axp20x_dev *axp20x)
 
 	ret = mfd_add_devices(axp20x->dev, PLATFORM_DEVID_NONE, axp20x->cells,
 			      axp20x->nr_cells, NULL, 0, NULL);
+	if (ret) {
+		ret = mfd_add_devices(axp20x->dev, PLATFORM_DEVID_AUTO, axp20x->cells,
+				      axp20x->nr_cells, NULL, 0, NULL);
+	}
 
 	if (ret) {
 		dev_err(axp20x->dev, "failed to add MFD devices: %d\n", ret);
-- 
2.49.0

