diff --speed-large-files --no-dereference --minimal -Naur linux-7.1.3/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi linux-7.1.3/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
--- linux-7.1.3/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi	2026-07-06 17:31:36.424186234 +0200
+++ linux-7.1.3/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi	2026-07-06 17:07:04.196057464 +0200
@@ -386,7 +388,7 @@
 				pins = "PF0" ,"PF1", "PF2", "PF3", "PF4", "PF5";
 				allwinner,pinmux = <2>;
 				function = "mmc0";
-				drive-strength = <30>;
+				drive-strength = <40>;
 				bias-pull-up;
 			};
 
@@ -395,7 +397,7 @@
 				pins = "PG0" ,"PG1", "PG2", "PG3", "PG4", "PG5";
 				allwinner,pinmux = <2>;
 				function = "mmc1";
-				drive-strength = <30>;
+				drive-strength = <40>;
 				bias-pull-up;
 			};
 
@@ -405,7 +407,7 @@
 				       "PC15", "PC16";
 				allwinner,pinmux = <3>;
 				function = "mmc2";
-				drive-strength = <30>;
+				drive-strength = <40>;
 				bias-pull-up;
 			};
 
