diff -Naur glibc-2.15-old/locale/Makefile glibc-2.15-new/locale/Makefile
--- glibc-2.15-old/locale/Makefile	2012-01-01 04:16:32.000000000 -0800
+++ glibc-2.15-new/locale/Makefile	2012-03-22 08:48:37.000000000 -0700
@@ -44,8 +44,8 @@
 		  address telephone measurement identification collate
 aux		= $(categories:%=lc-%) $(categories:%=C-%) SYS_libc C_name \
 		  xlocale localename global-locale coll-lookup
-others		= localedef locale
-#others-static	= localedef locale
+others		= locale
+others-static	= localedef
 install-bin	= localedef locale
 extra-objs	= $(localedef-modules:=.o) $(localedef-aux:=.o) \
 		  $(locale-modules:=.o) $(lib-modules:=.o)
