diff -Naur pcre-8.37-old/Makefile.am pcre-8.37-new/Makefile.am
--- pcre-8.37-old/Makefile.am	2015-06-05 12:02:56.738000231 -0700
+++ pcre-8.37-new/Makefile.am	2015-06-05 12:06:15.206004670 -0700
@@ -207,7 +207,7 @@
 dftables_SOURCES = dftables.c
 
 pcre_chartables.c: dftables$(EXEEXT)
-	./dftables$(EXEEXT) $@
+	pcre-dftables$(EXEEXT) $@
 else
 
 pcre_chartables.c: $(srcdir)/pcre_chartables.c.dist
diff -Naur pcre-8.37-old/Makefile.in pcre-8.37-new/Makefile.in
--- pcre-8.37-old/Makefile.in	2015-06-05 12:02:56.749000223 -0700
+++ pcre-8.37-new/Makefile.in	2015-06-05 12:06:15.206004670 -0700
@@ -3161,7 +3161,7 @@
 	rm -rf _generic
 
 @WITH_REBUILD_CHARTABLES_TRUE@pcre_chartables.c: dftables$(EXEEXT)
-@WITH_REBUILD_CHARTABLES_TRUE@	./dftables$(EXEEXT) $@
+@WITH_REBUILD_CHARTABLES_TRUE@	pcre-dftables$(EXEEXT) $@
 
 @WITH_REBUILD_CHARTABLES_FALSE@pcre_chartables.c: $(srcdir)/pcre_chartables.c.dist
 @WITH_REBUILD_CHARTABLES_FALSE@	rm -f $@
