diff -Naur pcre-8.32-old/Makefile.am pcre-8.32-new/Makefile.am
--- pcre-8.32-old/Makefile.am	2012-12-02 04:21:43.000000000 -0800
+++ pcre-8.32-new/Makefile.am	2012-12-02 04:23:01.000000000 -0800
@@ -201,7 +201,7 @@
 dftables_SOURCES = dftables.c
 
 pcre_chartables.c: dftables$(EXEEXT)
-	./dftables$(EXEEXT) $@
+	pcre-dftables$(EXEEXT) $@
 else
 
 pcre_chartables.c: $(srcdir)/pcre_chartables.c.dist
diff -Naur pcre-8.32-old/Makefile.in pcre-8.32-new/Makefile.in
--- pcre-8.32-old/Makefile.in	2012-12-02 04:21:43.000000000 -0800
+++ pcre-8.32-new/Makefile.in	2012-12-02 04:23:01.000000000 -0800
@@ -2782,7 +2782,7 @@
 	rm -rf _generic
 
 @WITH_REBUILD_CHARTABLES_TRUE@pcre_chartables.c: dftables$(EXEEXT)
-@WITH_REBUILD_CHARTABLES_TRUE@	./dftables$(EXEEXT) $@
+@WITH_REBUILD_CHARTABLES_TRUE@	pcre-dftables$(EXEEXT) $@
 
 @WITH_REBUILD_CHARTABLES_FALSE@pcre_chartables.c: $(srcdir)/pcre_chartables.c.dist
 @WITH_REBUILD_CHARTABLES_FALSE@	rm -f $@
