diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts
--- linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts	2022-10-25 21:25:03.459491359 +0200
+++ linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts	2022-10-25 21:17:40.446166027 +0200
@@ -9,6 +9,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include "sun50i-h6-cpu-opp.dtsi"
+#include "sun50i-h6-gpu-opp.dtsi"
 
 / {
 	model = "Eachlink H6 Mini";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-gpu-opp.dtsi linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-gpu-opp.dtsi
--- linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-gpu-opp.dtsi	2022-10-24 00:27:33.000000000 +0200
+++ linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-gpu-opp.dtsi	2022-10-25 21:23:45.249492767 +0200
@@ -10,66 +10,21 @@
 			opp-microvolt = <810000 810000 1200000>;
 		};
 
-		opp-264000000 {
-			opp-hz = /bits/ 64 <264000000>;
-			opp-microvolt = <810000 810000 1200000>;
-		};
-
 		opp-312000000 {
 			opp-hz = /bits/ 64 <312000000>;
 			opp-microvolt = <810000 810000 1200000>;
 		};
 
-		opp-336000000 {
-			opp-hz = /bits/ 64 <336000000>;
-			opp-microvolt = <810000 810000 1200000>;
-		};
-
-		opp-360000000 {
-			opp-hz = /bits/ 64 <360000000>;
-			opp-microvolt = <820000 820000 1200000>;
-		};
-
-		opp-384000000 {
-			opp-hz = /bits/ 64 <384000000>;
-			opp-microvolt = <830000 830000 1200000>;
-		};
-
 		opp-408000000 {
 			opp-hz = /bits/ 64 <408000000>;
 			opp-microvolt = <840000 840000 1200000>;
 		};
 
-		opp-420000000 {
-			opp-hz = /bits/ 64 <420000000>;
-			opp-microvolt = <850000 850000 1200000>;
-		};
-
-		opp-432000000 {
-			opp-hz = /bits/ 64 <432000000>;
-			opp-microvolt = <860000 860000 1200000>;
-		};
-
-		opp-456000000 {
-			opp-hz = /bits/ 64 <456000000>;
-			opp-microvolt = <870000 870000 1200000>;
-		};
-
 		opp-504000000 {
 			opp-hz = /bits/ 64 <504000000>;
 			opp-microvolt = <890000 890000 1200000>;
 		};
 
-		opp-540000000 {
-			opp-hz = /bits/ 64 <540000000>;
-			opp-microvolt = <910000 910000 1200000>;
-		};
-
-		opp-576000000 {
-			opp-hz = /bits/ 64 <576000000>;
-			opp-microvolt = <930000 930000 1200000>;
-		};
-
 		opp-624000000 {
 			opp-hz = /bits/ 64 <624000000>;
 			opp-microvolt = <950000 950000 1200000>;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts
--- linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2022-10-25 21:25:04.109491348 +0200
+++ linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2022-10-25 21:18:47.036164833 +0200
@@ -5,6 +5,7 @@
 
 #include "sun50i-h6.dtsi"
 #include "sun50i-h6-cpu-opp.dtsi"
+#include "sun50i-h6-gpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi
--- linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi	2022-10-24 00:27:33.000000000 +0200
+++ linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi	2022-10-25 21:18:27.422831852 +0200
@@ -5,6 +5,7 @@
 /dts-v1/;
 
 #include "sun50i-h6.dtsi"
+#include "sun50i-h6-gpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts
--- linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts	2022-10-24 00:27:33.000000000 +0200
+++ linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts	2022-10-25 21:19:02.412831228 +0200
@@ -5,6 +5,7 @@
 
 #include "sun50i-h6.dtsi"
 #include "sun50i-h6-cpu-opp.dtsi"
+#include "sun50i-h6-gpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi
--- linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi	2022-10-25 21:25:03.566158024 +0200
+++ linux-6.1-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi	2022-10-25 21:19:20.299497568 +0200
@@ -5,6 +5,7 @@
 
 #include "sun50i-h6.dtsi"
 #include "sun50i-h6-cpu-opp.dtsi"
+#include "sun50i-h6-gpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
