diff --speed-large-files --no-dereference --minimal -Naur linux-6.0/drivers/gpu/drm/panfrost/panfrost_devfreq.c linux-6.0/drivers/gpu/drm/panfrost/panfrost_devfreq.c
--- linux-6.0/drivers/gpu/drm/panfrost/panfrost_devfreq.c	2022-10-07 18:02:30.187865512 +0200
+++ linux-6.0/drivers/gpu/drm/panfrost/panfrost_devfreq.c	2022-10-07 18:01:28.284534858 +0200
@@ -131,17 +131,6 @@
 		return PTR_ERR(opp);
 
 	panfrost_devfreq_profile.initial_freq = cur_freq;
-
-	/*
-	 * Set the recommend OPP this will enable and configure the regulator
-	 * if any and will avoid a switch off by regulator_late_cleanup()
-	 */
-	ret = dev_pm_opp_set_opp(dev, opp);
-	if (ret) {
-		DRM_DEV_ERROR(dev, "Couldn't set recommended OPP\n");
-		return ret;
-	}
-
 	dev_pm_opp_put(opp);
 
 	/*
