--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c	2022-08-23 09:24:48.023418327 +0000
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c	2022-08-23 10:17:47.908444438 +0000
@@ -32,6 +32,8 @@
 #include <drm/drm_self_refresh_helper.h>
 #include <drm/drm_vblank.h>
 
+#include <uapi/linux/media-bus-format.h>
+
 #ifdef CONFIG_DRM_ANALOGIX_DP
 #include <drm/bridge/analogix_dp.h>
 #endif
--- a/drivers/gpu/drm/rockchip/dw_hdmi-rockchip.c	2022-08-23 09:24:48.023418327 +0000
+++ b/drivers/gpu/drm/rockchip/dw_hdmi-rockchip.c	2022-08-23 10:36:41.755225409 +0000
@@ -17,6 +17,8 @@
 #include <drm/drm_probe_helper.h>
 #include <drm/drm_simple_kms_helper.h>
 
+#include <uapi/linux/media-bus-format.h>
+
 #include "rockchip_drm_drv.h"
 #include "rockchip_drm_vop.h"
 
