diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts
--- linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2022-11-07 09:05:15.685144751 +0100
+++ linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2022-11-07 08:54:11.545139898 +0100
@@ -116,6 +116,18 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&usb30_host_drv>;
 		regulator-name = "vcc_host_5v";
+		regulator-always-on;
+		vin-supply = <&vcc_sys>;
+	};
+
+	vcc_host1_5v: vcc_otg_5v: vcc-host1-5v-regulator {
+		compatible = "regulator-fixed";
+		enable-active-high;
+		gpio = <&gpio0 RK_PD3 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&usb20_host_drv>;
+		regulator-name = "vcc_host1_5v";
+		regulator-always-on;
 		vin-supply = <&vcc_sys>;
 	};
 
@@ -467,6 +479,12 @@
 		};
 	};
 
+	usb2 {
+		usb20_host_drv: usb20-host-drv {
+			rockchip,pins = <1 RK_PD2 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
 	wifi {
 		bt_dis: bt-dis {
 			rockchip,pins = <2 RK_PC5 RK_FUNC_GPIO &pcfg_output_low>;
@@ -545,10 +563,26 @@
 };
 
 &usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
 	dr_mode = "host";
 	status = "okay";
 };
 
+&usb3phy {
+	status = "okay";
+};
+
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
+	status = "okay";
+};
+
 &vop {
 	status = "okay";
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts
--- linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts	2022-11-07 09:05:15.715144752 +0100
+++ linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts	2022-11-07 09:00:54.415142843 +0100
@@ -451,16 +451,32 @@
 	status = "okay";
 };
 
+&usb_host0_ehci {
+	status = "okay";
+};
+
+&usb_host0_ohci {
+	status = "okay";
+};
+
 &usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
 	dr_mode = "host";
 	status = "okay";
 };
 
-&usb_host0_ehci {
+&usb3phy {
 	status = "okay";
 };
 
-&usb_host0_ohci {
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
 	status = "okay";
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts
--- linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts	2022-10-30 23:19:28.000000000 +0100
+++ linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts	2022-11-07 09:00:54.415142843 +0100
@@ -381,16 +381,32 @@
 	status = "okay";
 };
 
+&usb_host0_ehci {
+	status = "okay";
+};
+
+&usb_host0_ohci {
+	status = "okay";
+};
+
 &usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
 	dr_mode = "host";
 	status = "okay";
 };
 
-&usb_host0_ehci {
+&usb3phy {
 	status = "okay";
 };
 
-&usb_host0_ohci {
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
 	status = "okay";
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts
--- linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts	2022-10-30 23:19:28.000000000 +0100
+++ linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts	2022-11-07 09:00:54.415142843 +0100
@@ -380,11 +380,28 @@
 	status = "okay";
 };
 
+&usb_host0_ehci {
+	status = "okay";
+};
+
 &usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
 	dr_mode = "host";
 	status = "okay";
 };
 
-&usb_host0_ehci {
+&usb3phy {
 	status = "okay";
 };
+
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
+	status = "okay";
+};
+
