diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts
--- linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2022-11-06 18:26:03.494759840 +0100
+++ linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2022-11-06 18:22:32.898091640 +0100
@@ -545,10 +545,26 @@
 };
 
 &usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
 	dr_mode = "host";
 	status = "okay";
 };
 
+&usb3phy {
+	status = "okay";
+};
+
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
+	status = "okay";
+};
+
 &vop {
 	status = "okay";
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts
--- linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts	2022-11-06 18:26:03.534759841 +0100
+++ linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts	2022-11-06 18:23:22.254758671 +0100
@@ -451,16 +451,32 @@
 	status = "okay";
 };
 
+&usb_host0_ehci {
+	status = "okay";
+};
+
+&usb_host0_ohci {
+	status = "okay";
+};
+
 &usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
 	dr_mode = "host";
 	status = "okay";
 };
 
-&usb_host0_ehci {
+&usb3phy {
 	status = "okay";
 };
 
-&usb_host0_ohci {
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
 	status = "okay";
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts
--- linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts	2022-10-30 23:19:28.000000000 +0100
+++ linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts	2022-11-06 18:24:30.091425829 +0100
@@ -381,16 +381,32 @@
 	status = "okay";
 };
 
+&usb_host0_ehci {
+	status = "okay";
+};
+
+&usb_host0_ohci {
+	status = "okay";
+};
+
 &usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
 	dr_mode = "host";
 	status = "okay";
 };
 
-&usb_host0_ehci {
+&usb3phy {
 	status = "okay";
 };
 
-&usb_host0_ohci {
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
 	status = "okay";
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts
--- linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts	2022-10-30 23:19:28.000000000 +0100
+++ linux-6.1-rc3/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts	2022-11-06 18:24:12.361425701 +0100
@@ -380,11 +380,28 @@
 	status = "okay";
 };
 
+&usb_host0_ehci {
+	status = "okay";
+};
+
 &usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
 	dr_mode = "host";
 	status = "okay";
 };
 
-&usb_host0_ehci {
+&usb3phy {
 	status = "okay";
 };
+
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
+	status = "okay";
+};
+
