diff --speed-large-files --no-dereference --minimal -Naur linux-6.1.8/arch/arm64/boot/dts/rockchip/Makefile linux-6.1.8/arch/arm64/boot/dts/rockchip/Makefile
--- linux-6.1.8/arch/arm64/boot/dts/rockchip/Makefile	2023-01-25 11:43:09.529836417 +0100
+++ linux-6.1.8/arch/arm64/boot/dts/rockchip/Makefile	2023-01-25 11:41:37.883171402 +0100
@@ -55,6 +55,7 @@
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4b.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4b-plus.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4c.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4se.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock960.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rockpro64-v2.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rockpro64.dtb
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1.8/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4se.dts linux-6.1.8/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4se.dts
--- linux-6.1.8/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4se.dts	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.1.8/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4se.dts	2023-01-24 18:01:48.672799537 +0100
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2019 Akash Gajjar <Akash_Gajjar@mentor.com>
+ * Copyright (c) 2019 Pragnesh Patel <Pragnesh_Patel@mentor.com>
+ */
+
+/dts-v1/;
+#include "rk3399-rock-pi-4.dtsi"
+
+/ {
+	model = "Radxa ROCK Pi 4SE";
+	compatible = "radxa,rockpi4se", "radxa,rockpi4", "rockchip,rk3399";
+
+	aliases {
+		mmc2 = &sdio0;
+	};
+};
+
+&sdio0 {
+	status = "okay";
+
+	brcmf: wifi@1 {
+		compatible = "brcm,bcm4329-fmac";
+		reg = <1>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <RK_PA3 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "host-wake";
+		pinctrl-names = "default";
+		pinctrl-0 = <&wifi_host_wake_l>;
+	};
+};
+
+&uart0 {
+	status = "okay";
+
+	bluetooth {
+		compatible = "brcm,bcm4345c5";
+		clocks = <&rk808 1>;
+		clock-names = "lpo";
+		device-wakeup-gpios = <&gpio2 RK_PD3 GPIO_ACTIVE_HIGH>;
+		host-wakeup-gpios = <&gpio0 RK_PA4 GPIO_ACTIVE_HIGH>;
+		shutdown-gpios = <&gpio0 RK_PB1 GPIO_ACTIVE_HIGH>;
+		max-speed = <1500000>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&bt_host_wake_l &bt_wake_l &bt_enable_h>;
+		vbat-supply = <&vcc3v3_sys>;
+		vddio-supply = <&vcc_1v8>;
+	};
+};
