diff --speed-large-files --no-dereference --minimal -Naur linux-5.18.3/arch/arm64/boot/dts/rockchip/rk356x.dtsi linux-5.18.3/arch/arm64/boot/dts/rockchip/rk356x.dtsi
--- linux-5.18.3/arch/arm64/boot/dts/rockchip/rk356x.dtsi	2022-06-15 11:56:46.908530631 +0200
+++ linux-5.18.3/arch/arm64/boot/dts/rockchip/rk356x.dtsi	2022-06-15 11:55:40.195196819 +0200
@@ -1424,12 +1424,12 @@
 
 			trips {
 				cpu_alert0: cpu_alert0 {
-					temperature = <70000>;
+					temperature = <85000>;
 					hysteresis = <2000>;
 					type = "passive";
 				};
 				cpu_alert1: cpu_alert1 {
-					temperature = <75000>;
+					temperature = <90000>;
 					hysteresis = <2000>;
 					type = "passive";
 				};
