diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc2/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts linux-6.1-rc2/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts
--- linux-6.1-rc2/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2022-10-26 14:32:31.166356362 +0200
+++ linux-6.1-rc2/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2022-10-26 14:26:05.709696630 +0200
@@ -209,6 +209,8 @@
 };
 
 &hdmi {
+	assigned-clocks = <&cru CLK_HDMI_CEC>;
+	assigned-clock-rates = <32768>;
 	avdd-0v9-supply = <&vdda0v9_image>;
 	avdd-1v8-supply = <&vcca1v8_image>;
 	status = "okay";
