diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc4/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts linux-6.1-rc4/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts
--- linux-6.1-rc4/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2022-11-07 09:22:40.898485718 +0100
+++ linux-6.1-rc4/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2022-11-07 09:21:29.768485196 +0100
@@ -209,6 +209,8 @@
 };
 
 &hdmi {
+	assigned-clocks = <&cru CLK_HDMI_CEC>;
+	assigned-clock-rates = <32768>;
 	avdd-0v9-supply = <&vdda0v9_image>;
 	avdd-1v8-supply = <&vcca1v8_image>;
 	status = "okay";
