diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc2/drivers/gpu/drm/vc4/vc4_hdmi.c linux-6.1-rc2/drivers/gpu/drm/vc4/vc4_hdmi.c
--- linux-6.1-rc2/drivers/gpu/drm/vc4/vc4_hdmi.c	2022-10-24 00:27:33.000000000 +0200
+++ linux-6.1-rc2/drivers/gpu/drm/vc4/vc4_hdmi.c	2022-10-24 20:50:21.461640211 +0200
@@ -3321,15 +3321,6 @@
 	unsigned long rate;
 	int ret;
 
-	/*
-	 * The HSM clock is in the HDMI power domain, so we need to set
-	 * its frequency while the power domain is active so that it
-	 * keeps its rate.
-	 */
-	ret = clk_set_min_rate(vc4_hdmi->hsm_clock, HSM_MIN_CLOCK_FREQ);
-	if (ret)
-		return ret;
-
 	ret = clk_prepare_enable(vc4_hdmi->hsm_clock);
 	if (ret)
 		return ret;
