diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc8/arch/arm/boot/dts/bcm2835-rpi.dtsi linux-6.1-rc8/arch/arm/boot/dts/bcm2835-rpi.dtsi
--- linux-6.1-rc8/arch/arm/boot/dts/bcm2835-rpi.dtsi	2022-12-04 23:48:12.000000000 +0100
+++ linux-6.1-rc8/arch/arm/boot/dts/bcm2835-rpi.dtsi	2022-12-10 13:56:42.656544263 +0100
@@ -1,4 +1,5 @@
 #include <dt-bindings/power/raspberrypi-power.h>
+#include <dt-bindings/leds/common.h>
 
 / {
 	leds {
@@ -7,7 +8,8 @@
 		led-act {
 			label = "ACT";
 			default-state = "keep";
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 		};
 	};
 
