diff --speed-large-files --no-dereference --minimal -Naur linux-6.1.6/arch/arm/boot/dts/bcm2711-rpi.dtsi linux-6.1.6/arch/arm/boot/dts/bcm2711-rpi.dtsi
--- linux-6.1.6/arch/arm/boot/dts/bcm2711-rpi.dtsi	2023-01-14 10:33:46.000000000 +0100
+++ linux-6.1.6/arch/arm/boot/dts/bcm2711-rpi.dtsi	2023-01-18 12:43:34.809762842 +0100
@@ -15,6 +15,9 @@
 		ethernet0 = &genet;
 		pcie0 = &pcie0;
 		blconfig = &blconfig;
+		mmc2 = &sdhost;
+		mmc1 = &sdhci;
+		mmc0 = &emmc2;
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1.6/arch/arm/boot/dts/bcm2835-rpi.dtsi linux-6.1.6/arch/arm/boot/dts/bcm2835-rpi.dtsi
--- linux-6.1.6/arch/arm/boot/dts/bcm2835-rpi.dtsi	2023-01-14 10:33:46.000000000 +0100
+++ linux-6.1.6/arch/arm/boot/dts/bcm2835-rpi.dtsi	2023-01-18 12:52:07.669753595 +0100
@@ -1,13 +1,22 @@
 #include <dt-bindings/power/raspberrypi-power.h>
+#include <dt-bindings/leds/common.h>
 
 / {
+	aliases {
+		serial0 = &uart0;
+		serial1 = &uart1;
+		mmc0 = &sdhost;
+		mmc1 = &sdhci;
+	};
+
 	leds {
 		compatible = "gpio-leds";
 
 		led-act {
 			label = "ACT";
 			default-state = "keep";
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 		};
 	};
 
