From af03b9522db56dc49a130c92e918416e408f667f Mon Sep 17 00:00:00 2001
From: Andrew Powers-Holmes <aholmes@omnom.net>
Date: Sat, 21 Jan 2023 00:01:26 +1100
Subject: [PATCH 63/63] add support for enabling vbus to husb311 driver

---
 drivers/usb/typec/tcpm/tcpci_husb311.c | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/drivers/usb/typec/tcpm/tcpci_husb311.c b/drivers/usb/typec/tcpm/tcpci_husb311.c
index 28e26a69fc8..858577a0eb8 100644
--- a/drivers/usb/typec/tcpm/tcpci_husb311.c
+++ b/drivers/usb/typec/tcpm/tcpci_husb311.c
@@ -10,8 +10,10 @@
 #include <linux/i2c.h>
 #include <linux/interrupt.h>
 #include <linux/kernel.h>
+#include <linux/mod_devicetable.h>
 #include <linux/module.h>
 #include <linux/regmap.h>
+#include <linux/regulator/consumer.h>
 #include <linux/usb/tcpm.h>
 #include <linux/usb/tcpci.h>
 
@@ -27,6 +29,7 @@ struct husb311_chip {
 	struct tcpci_data data;
 	struct tcpci *tcpci;
 	struct device *dev;
+	struct regulator *vbus;
 };
 
 static int husb311_write8(struct husb311_chip *chip, unsigned int reg, u8 val)
@@ -76,6 +79,25 @@ static int husb311_init(struct tcpci *tcpci, struct tcpci_data *tdata)
 	return ret;
 }
 
+static int husb311_tcpc_set_vbus(struct tcpci *tcpci, struct tcpci_data *tdata,
+				 bool source, bool sink)
+{
+	struct husb311_chip *chip = tdata_to_husb311(tdata);
+	int ret;
+
+	ret = regulator_is_enabled(chip->vbus);
+	if (ret < 0)
+		return ret;
+
+	if (ret && !source)
+		return regulator_disable(chip->vbus);
+
+	if (!ret && source)
+		return regulator_enable(chip->vbus);
+
+	return 0;
+}
+
 static irqreturn_t husb311_irq(int irq, void *dev_id)
 {
 	struct husb311_chip *chip = dev_id;
@@ -142,6 +164,10 @@ static int husb311_probe(struct i2c_client *client, const struct i2c_device_id *
 	}
 
 	chip->data.init = husb311_init;
+	chip->vbus = devm_regulator_get_optional(chip->dev, "vbus");
+	if (!IS_ERR(chip->vbus))
+		chip->data.set_vbus = husb311_tcpc_set_vbus;
+
 	chip->tcpci = tcpci_register_port(chip->dev, &chip->data);
 	if (IS_ERR(chip->tcpci))
 		return PTR_ERR(chip->tcpci);
-- 
2.36.1

