diff -Naur module-init-tools-3.16-old/Makefile.am module-init-tools-3.16-new/Makefile.am
--- module-init-tools-3.16-old/Makefile.am	2011-06-02 10:55:01.000000000 -0700
+++ module-init-tools-3.16-new/Makefile.am	2011-06-03 20:36:38.000000000 -0700
@@ -36,9 +36,6 @@
 modinfo_LDADD = $(LDADD) libmodtools.a
 modindex_LDADD = $(LDADD) libmodtools.a
 
-MAN5 = depmod.conf.5 depmod.d.5 modprobe.conf.5 modprobe.d.5 \
-	modules.dep.5 modules.dep.bin.5
-MAN8 = depmod.8 insmod.8 lsmod.8 rmmod.8 modprobe.8 modinfo.8
 SGML = $(addprefix doc/,  $(MAN5:%.5=%.sgml) $(MAN8:%.8=%.sgml))
 dist_man_MANS = $(MAN5) $(MAN8)
 # If they haven't overridden mandir, fix it (never /man!)
