diff -Naur RFLibs-v2-old/RFLibs/Aquos.pm RFLibs-v2-new/RFLibs/Aquos.pm
--- RFLibs-v2-old/RFLibs/Aquos.pm	2009-05-25 15:11:07.000000000 -0700
+++ RFLibs-v2-new/RFLibs/Aquos.pm	2009-05-25 15:10:54.000000000 -0700
@@ -191,8 +191,8 @@
 	while(1)
 	{
 		my ($bytes, $read) = $self->{port}->read(1);
-		last if $read == 0;
-		$self->debugprint('WARN', "$read byte(s) -->$bytes<-- cleared from buffer\n");
+		last if $bytes == 0;
+		$self->debugprint('WARN', "$bytes byte(s) -->$read<-- cleared from buffer\n");
 	};
 
 
