diff -Naur rtl8189ES_linux-930d3d7cfe3b25c0cf8a57ca4b79f6aef5b6de4b-old/os_dep/osdep_service.c rtl8189ES_linux-930d3d7cfe3b25c0cf8a57ca4b79f6aef5b6de4b-new/os_dep/osdep_service.c
--- rtl8189ES_linux-930d3d7cfe3b25c0cf8a57ca4b79f6aef5b6de4b-old/os_dep/osdep_service.c	2020-10-03 14:21:01.000000000 +0200
+++ rtl8189ES_linux-930d3d7cfe3b25c0cf8a57ca4b79f6aef5b6de4b-new/os_dep/osdep_service.c	2020-12-28 17:20:20.523333246 +0100
@@ -1952,12 +1952,15 @@
 		ret = PTR_ERR(fp);
 	}
 	else {
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 		oldfs = get_fs(); set_fs(KERNEL_DS);
+#endif
 		
 		if(1!=readFile(fp, &buf, 1))
 			ret = PTR_ERR(fp);
-		
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 		set_fs(oldfs);
+#endif
 		filp_close(fp,NULL);
 	}	
 	return ret;
@@ -1979,10 +1982,13 @@
 	if(path && buf) {
 		if( 0 == (ret=openFile(&fp,path, O_RDONLY, 0)) ){
 			DBG_871X("%s openFile path:%s fp=%p\n",__FUNCTION__, path ,fp);
-
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 			oldfs = get_fs(); set_fs(KERNEL_DS);
+#endif
 			ret=readFile(fp, buf, sz);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 			set_fs(oldfs);
+#endif
 			closeFile(fp);
 			
 			DBG_871X("%s readFile, ret:%d\n",__FUNCTION__, ret);
@@ -2013,10 +2019,13 @@
 	if(path && buf) {
 		if( 0 == (ret=openFile(&fp, path, O_CREAT|O_WRONLY, 0666)) ) {
 			DBG_871X("%s openFile path:%s fp=%p\n",__FUNCTION__, path ,fp);
-
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 			oldfs = get_fs(); set_fs(KERNEL_DS);
+#endif
 			ret=writeFile(fp, buf, sz);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 			set_fs(oldfs);
+#endif
 			closeFile(fp);
 
 			DBG_871X("%s writeFile, ret:%d\n",__FUNCTION__, ret);
