diff --speed-large-files --no-dereference --minimal -Naur xmlstarlet-1.6.1/src/xml_pyx.c xmlstarlet-1.6.1/src/xml_pyx.c
--- xmlstarlet-1.6.1/src/xml_pyx.c	2013-07-01 04:58:36.000000000 +0200
+++ xmlstarlet-1.6.1/src/xml_pyx.c	2025-05-10 13:07:25.301005456 +0200
@@ -21,6 +21,14 @@
 
 #include "xmlstar.h"
 
+#ifndef ATTRIBUTE_UNUSED
+#  if (defined(__GNUC__) && __GNUC__ >= 3) || defined(__clang__)
+#    define ATTRIBUTE_UNUSED __attribute__((unused))
+#  else
+#    define ATTRIBUTE_UNUSED /* empty */
+#  endif
+#endif
+
 /**
  *  Output newline and tab characters as escapes
  *  Required both for attribute values and character data (#PCDATA)
