diff -Naur MySQL-python-1.2.5-old/_mysql.c MySQL-python-1.2.5-new/_mysql.c
--- MySQL-python-1.2.5-old/_mysql.c	2014-01-02 13:52:50.000000000 +0100
+++ MySQL-python-1.2.5-new/_mysql.c	2021-09-20 19:18:57.116622885 +0200
@@ -2002,7 +2002,10 @@
 	int r, reconnect = -1;
 	if (!PyArg_ParseTuple(args, "|I", &reconnect)) return NULL;
 	check_connection(self);
-	if ( reconnect != -1 ) self->connection.reconnect = reconnect;
+	if (reconnect != -1) {
+		my_bool recon = reconnect;
+		mysql_options(&self->connection, MYSQL_OPT_RECONNECT, &recon);
+	};
 	Py_BEGIN_ALLOW_THREADS
 	r = mysql_ping(&(self->connection));
 	Py_END_ALLOW_THREADS
