diff --speed-large-files --no-dereference --minimal -Naur rtw88-fa96fd4c014fa528d1fa50318e97aa71bf4f473c/tx.c rtw88-fa96fd4c014fa528d1fa50318e97aa71bf4f473c/tx.c
--- rtw88-fa96fd4c014fa528d1fa50318e97aa71bf4f473c/tx.c	2025-07-13 18:23:13.000000000 +0200
+++ rtw88-fa96fd4c014fa528d1fa50318e97aa71bf4f473c/tx.c	2025-07-19 13:03:17.710012545 +0200
@@ -198,7 +198,9 @@
 void rtw_tx_report_purge_timer(void *cntx)
 #endif
 {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 16, 0)
+	struct rtw_dev *rtwdev = timer_container_of(rtwdev, t, tx_report.purge_timer);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0)
 	struct rtw_dev *rtwdev = from_timer(rtwdev, t, tx_report.purge_timer);
 #else
 	struct rtw_dev *rtwdev = (struct rtw_dev *)cntx;
