diff --speed-large-files --no-dereference --minimal -Naur rtw88-fa96fd4c014fa528d1fa50318e97aa71bf4f473c/sdio.c rtw88-fa96fd4c014fa528d1fa50318e97aa71bf4f473c/sdio.c
--- rtw88-fa96fd4c014fa528d1fa50318e97aa71bf4f473c/sdio.c	2025-07-13 18:23:13.000000000 +0200
+++ rtw88-fa96fd4c014fa528d1fa50318e97aa71bf4f473c/sdio.c	2025-07-24 12:57:39.813346641 +0200
@@ -144,6 +144,10 @@
 
 static bool rtw_sdio_use_direct_io(struct rtw_dev *rtwdev, u32 addr)
 {
+	if (!rtw_sdio_is_bus_addr(addr) &&
+	    !test_bit(RTW_FLAG_POWERON, rtwdev->flags))
+		return false;
+
 	return !rtw_sdio_is_sdio30_supported(rtwdev) ||
 		rtw_sdio_is_bus_addr(addr);
 }
