diff -Naur spdlog.old/spdlogConfigTargets.cmake spdlog.new/spdlogConfigTargets.cmake
--- spdlog.old/spdlogConfigTargets.cmake	2025-06-24 16:58:02.000000000 +0200
+++ spdlog.new/spdlogConfigTargets.cmake	2025-06-24 16:44:50.374814619 +0200
@@ -47,7 +47,10 @@
 
 
 # The installation prefix configured by this project.
-set(_IMPORT_PREFIX "/usr")
+get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
 
 # Create imported target spdlog::spdlog
 add_library(spdlog::spdlog SHARED IMPORTED)
diff -Naur spdlog.old/spdlogConfigTargets-release.cmake spdlog.new/spdlogConfigTargets-release.cmake
--- spdlog.old/spdlogConfigTargets-release.cmake	2025-06-24 16:58:02.000000000 +0200
+++ spdlog.new/spdlogConfigTargets-release.cmake	2025-06-24 16:46:06.981479641 +0200
@@ -8,12 +8,12 @@
 # Import target "spdlog::spdlog" for configuration "Release"
 set_property(TARGET spdlog::spdlog APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
 set_target_properties(spdlog::spdlog PROPERTIES
-  IMPORTED_LOCATION_RELEASE "/usr/lib/libspdlog.so.1.15.3"
+  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/libspdlog.so.1.15.3"
   IMPORTED_SONAME_RELEASE "libspdlog.so.1.15"
   )
 
 list(APPEND _cmake_import_check_targets spdlog::spdlog )
-list(APPEND _cmake_import_check_files_for_spdlog::spdlog "/usr/lib/libspdlog.so.1.15.3" )
+list(APPEND _cmake_import_check_files_for_spdlog::spdlog "/${_IMPORT_PREFIX}/lib/libspdlog.so.1.15.3" )
 
 # Commands beyond this point should not need to know the version.
 set(CMAKE_IMPORT_FILE_VERSION)
