diff --speed-large-files --no-dereference --minimal -Naur u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/include/net.h u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/include/net.h
--- u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/include/net.h	2021-06-26 14:56:24.000000000 +0200
+++ u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/include/net.h	2022-02-04 13:35:51.629932926 +0100
@@ -815,14 +815,12 @@
  */
 static inline void net_random_ethaddr(uchar *addr)
 {
-	int i;
-	unsigned int seed = get_timer(0);
-
-	for (i = 0; i < 6; i++)
-		addr[i] = rand_r(&seed);
-
-	addr[0] &= 0xfe;	/* clear multicast bit */
-	addr[0] |= 0x02;	/* set local assignment bit (IEEE802) */
+	addr[0] = 0x2e;
+	addr[1] = 0x42;
+	addr[2] = 0x53;
+	addr[3] = 0xa5;
+	addr[4] = 0xd8;
+	addr[5] = 0x1b;
 }
 
 /* Convert an IP address to a string */
