diff --speed-large-files --no-dereference --minimal -Naur u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/command.c u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/command.c
--- u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/command.c	2023-04-13 05:19:49.000000000 +0200
+++ u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/command.c	2023-05-03 11:18:21.653229318 +0200
@@ -501,10 +501,10 @@
 	return result;
 }
 
-enum command_ret_t cmd_process(int flag, int argc, char * const argv[],
+int cmd_process(int flag, int argc, char * const argv[],
 			       int *repeatable, ulong *ticks)
 {
-	enum command_ret_t rc = CMD_RET_SUCCESS;
+	int rc = CMD_RET_SUCCESS;
 	cmd_tbl_t *cmdtp;
 
 	/* Look up command in command table */
diff --speed-large-files --no-dereference --minimal -Naur u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/edid.c u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/edid.c
--- u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/edid.c	2023-04-13 05:19:49.000000000 +0200
+++ u-boot-a2a9f1df8fa56c38c776dec94915cc80fa66f9c8/common/edid.c	2023-05-03 11:21:13.103224246 +0200
@@ -3579,7 +3579,7 @@
 {
 	const u8 *cea = drm_find_cea_extension(edid);
 	const u8 *db, *hdmi = NULL, *video = NULL;
-	u8 dbl, hdmi_len, video_len = 0;
+	u8 dbl, hdmi_len = 0, video_len = 0;
 	int modes = 0;
 
 	if (cea && cea_revision(cea) >= 3) {
