diff --speed-large-files --no-dereference --minimal -Naur xscreensaver-6.13/Makefile.in xscreensaver-6.13/Makefile.in
--- xscreensaver-6.13/Makefile.in	2025-12-02 20:47:25.359670995 +0100
+++ xscreensaver-6.13/Makefile.in	2025-12-28 10:49:39.513429438 +0100
@@ -7,7 +7,7 @@
 
 SHELL		= /bin/sh
 SUBDIRS 	= utils jwxyz hacks/images hacks hacks/glx hacks/fonts \
-		  driver po
+		  driver
 SUBDIRS2	= $(SUBDIRS) OSX android
 TARFILES	= README README.hacking INSTALL \
 		  configure configure.ac Makefile.in config.h.in \
@@ -61,7 +61,6 @@
 distdepend::
 	@$(MAKE2) update_spec_version
 	@$(MAKE_SUBDIR2)
-	@cd po ; $(MAKE2) update-po
 
 TAGS:: tags
 tags::
