diff -Naur xscreensaver-5.20-old/configure.in xscreensaver-5.20-new/configure.in
--- xscreensaver-5.20-old/configure.in	2012-06-30 15:50:43.000000000 -0700
+++ xscreensaver-5.20-new/configure.in	2012-10-20 05:06:36.000000000 -0700
@@ -1224,7 +1224,7 @@
 #
 ###############################################################################
 
-AC_PROG_INTLTOOL
+# AC_PROG_INTLTOOL # http://bugs.gentoo.org/show_bug.cgi?id=240958
 GETTEXT_PACKAGE=xscreensaver
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE",
 		   [This is the name of the gettext package to use.])
diff -Naur xscreensaver-5.20-old/driver/Makefile.in xscreensaver-5.20-new/driver/Makefile.in
--- xscreensaver-5.20-old/driver/Makefile.in	2012-10-06 13:20:04.000000000 -0700
+++ xscreensaver-5.20-new/driver/Makefile.in	2012-10-20 05:06:36.000000000 -0700
@@ -22,7 +22,7 @@
 mansuffix	= 1
 manNdir		= $(mandir)/man$(mansuffix)
 
-INTLTOOL_MERGE	= @INTLTOOL_MERGE@
+#INTLTOOL_MERGE	= @INTLTOOL_MERGE@ # http://bugs.gentoo.org/show_bug.cgi?id=240958
 
 GTK_DATADIR	= @GTK_DATADIR@
 GTK_APPDIR	= $(GTK_DATADIR)/applications
@@ -494,20 +494,19 @@
 
 # screensaver-properties.desktop
 # into /usr/share/applications/
-install-gnome:: screensaver-properties.desktop
+install-gnome::
 	@if [ "$(GTK_DATADIR)" != "" ]; then				      \
 	   if [ ! -d "$(install_prefix)$(GTK_APPDIR)" ]; then		      \
 	     echo $(INSTALL_DIRS) "$(install_prefix)$(GTK_APPDIR)"	     ;\
 		  $(INSTALL_DIRS) "$(install_prefix)$(GTK_APPDIR)"	     ;\
 	   fi								     ;\
 	   name2=xscreensaver-properties.desktop			     ;\
-	   echo $(INSTALL_DATA) screensaver-properties.desktop		      \
+	   echo $(INSTALL_DATA) screensaver-properties.desktop.in		      \
 		  $(install_prefix)$(GTK_APPDIR)/$$name2		     ;\
-		$(INSTALL_DATA) screensaver-properties.desktop		      \
+		$(INSTALL_DATA) screensaver-properties.desktop.in		      \
 		  $(install_prefix)$(GTK_APPDIR)/$$name2		     ;\
 	 fi
 
-
 # xscreensaver.xpm
 # into /usr/share/pixmaps/
 install-gnome:: $(LOGO)
@@ -772,7 +771,7 @@
 XScreenSaver_Xm_ad.h: XScreenSaver-Xm.ad
 	$(SHELL) $(UTILS_SRC)/ad2c XScreenSaver-Xm.ad > XScreenSaver_Xm_ad.h
 
-@INTLTOOL_DESKTOP_RULE@
+# @INTLTOOL_DESKTOP_RULE@ # http://bugs.gentoo.org/show_bug.cgi?id=240958
 
 # The executables linked in this directory.
 #
