diff -Naur xscreensaver-5.36-old/driver/prefs.c xscreensaver-5.36-new/driver/prefs.c
--- xscreensaver-5.36-old/driver/prefs.c	2016-05-24 05:08:27.000000000 +0200
+++ xscreensaver-5.36-new/driver/prefs.c	2018-03-12 15:25:51.223333330 +0100
@@ -1663,6 +1663,7 @@
 Bool
 decrepit_p (void)
 {
+    return 0;
   /* If you are in here because you're planning on disabling this warning
      before redistributing my software, please don't.
 
