diff -Naur xscreensaver-5.20-old/Makefile.in xscreensaver-5.20-new/Makefile.in
--- xscreensaver-5.20-old/Makefile.in	2012-10-20 05:06:26.000000000 -0700
+++ xscreensaver-5.20-new/Makefile.in	2012-10-20 05:07:14.000000000 -0700
@@ -6,7 +6,7 @@
 VPATH		= @srcdir@
 
 SHELL		= /bin/sh
-SUBDIRS 	= utils driver hacks hacks/glx po
+SUBDIRS 	= utils driver hacks hacks/glx
 #SUBDIRS 	= utils driver hacks hacks/glx
 SUBDIRS2	= $(SUBDIRS) OSX
 TARFILES	= README README.hacking README.VMS INSTALL \
