diff --speed-large-files --no-dereference --minimal -Naur iotop-0.6/setup.py iotop-0.6/setup.py
--- iotop-0.6/setup.py	2013-05-27 00:44:18.000000000 +0200
+++ iotop-0.6/setup.py	2024-09-17 21:24:55.386736601 +0200
@@ -4,13 +4,6 @@
 from distutils.command import install as distutils_install
 from iotop.version import VERSION
 
-# Dirty hack to make setup.py install the iotop script to sbin/ instead of bin/
-# while still honoring the choice of installing into local/ or not.
-if hasattr(distutils_install, 'INSTALL_SCHEMES'):
-    for d in distutils_install.INSTALL_SCHEMES.itervalues():
-        if d.get('scripts', '').endswith('/bin'):
-            d['scripts'] = d['scripts'][:-len('/bin')] + '/sbin'
-
 setup(name='iotop',
       version=VERSION,
       description='Per process I/O bandwidth monitor',
