diff -Naur xosd-2.2.14-old/libxosd.m4 xosd-2.2.14-new/libxosd.m4
--- xosd-2.2.14-old/libxosd.m4	2004-08-19 15:39:40.000000000 -0700
+++ xosd-2.2.14-new/libxosd.m4	2010-08-25 10:33:42.000000000 -0700
@@ -6,7 +6,7 @@
 dnl AM_PATH_LIBXOSD([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
 dnl Test to see if libxosd is installed, and define LIBXOSD_CFLAGS, LIBXOSDLIBS
 dnl
-AC_DEFUN(AM_PATH_LIBXOSD,
+AC_DEFUN([AM_PATH_LIBXOSD],
 [dnl
 dnl Get the cflags and libraries from the xosd-config script
 dnl
