diff -Naur xosd-2.2.14-old/script/xosd-config.in xosd-2.2.14-new/script/xosd-config.in
--- xosd-2.2.14-old/script/xosd-config.in	2004-08-19 15:39:31.000000000 -0700
+++ xosd-2.2.14-new/script/xosd-config.in	2012-04-25 22:28:28.000000000 -0700
@@ -209,8 +209,8 @@
 ## Libraries
 ##
 
-includes="-I@includedir@"
-libs="-L@libdir@"
+includes="-I$DESTDIR@includedir@"
+libs="-L$DESTDIR@libdir@"
 
 af_cflags="$af_cflags $includes"
 af_libs="$libs -l$LIB_NAME $af_libs"
