diff -Naur xosd-2.2.14-old/src/libxosd/xosd.c xosd-2.2.14-new/src/libxosd/xosd.c
--- xosd-2.2.14-old/src/libxosd/xosd.c	2004-11-17 11:27:11.000000000 -0800
+++ xosd-2.2.14-new/src/libxosd/xosd.c	2008-06-20 18:35:05.000000000 -0700
@@ -24,7 +24,7 @@
 #define XOFFSET 10
 
 const char *osd_default_font =
-  "-misc-fixed-medium-r-semicondensed--*-*-*-*-c-*-*-*";
+  "-misc-liberation sans-medium-r-normal--*-360-*-*-p-*-*-*";
 #if 0
 "-adobe-helvetica-bold-r-*-*-10-*";
 #endif
diff -Naur xosd-2.2.14-old/src/osd_cat.c xosd-2.2.14-new/src/osd_cat.c
--- xosd-2.2.14-old/src/osd_cat.c	2004-08-26 15:14:20.000000000 -0700
+++ xosd-2.2.14-new/src/osd_cat.c	2008-06-20 18:37:47.000000000 -0700
@@ -55,7 +55,7 @@
 char *shadow_colour = NULL;
 int delay = 5;
 int forcewait = 0;
-xosd_pos pos = XOSD_top;
+xosd_pos pos = XOSD_middle;
 int voffset = 0;
 int hoffset = 0;
 int shadow = 0;
@@ -63,7 +63,7 @@
 struct timeval old_age, new_age;
 int screen_line = 0;
 int lines = 5;
-xosd_align align = XOSD_left;
+xosd_align align = XOSD_center;
 
 int
 main(int argc, char *argv[])
