diff -Naur Device-SerialPort-1.04-old/Makefile.PL Device-SerialPort-1.04-new/Makefile.PL
--- Device-SerialPort-1.04-old/Makefile.PL	2007-07-20 10:27:38.000000000 -0700
+++ Device-SerialPort-1.04-new/Makefile.PL	2009-05-24 19:37:21.000000000 -0700
@@ -3,6 +3,7 @@
 use warnings;
 use ExtUtils::MakeMaker;
 
+my $CONFIGURE_ARGS;
 my @SERIALS;
 
 # Scan for argument we know about
@@ -13,6 +14,9 @@
 		if ($key eq "TESTPORT") {
 			push(@SERIALS,$value);
 		}
+		elsif ($key eq "CONFIGURE_ARGS") {
+			$CONFIGURE_ARGS=$value;
+		}
 		else {
 			push(@passargs,$arg);
 		}
@@ -134,7 +138,7 @@
 
 # Prepare config.h
 print "Running ./configure ...\n";
-system("./configure");
+system("./configure $CONFIGURE_ARGS");
 if ($? != 0) {
         die <<EOM;
 The 'configure' script didn't like something.  It should have told you
