diff -Naur weston-9.0.0-old/compositor/main.c weston-9.0.0-new/compositor/main.c
--- weston-9.0.0-old/compositor/main.c	2020-09-04 12:21:49.492731600 +0200
+++ weston-9.0.0-new/compositor/main.c	2021-04-23 10:10:12.063332833 +0200
@@ -3390,8 +3390,8 @@
 
 	for (i = 1; i < argc; i++)
 		weston_log("fatal: unhandled option: %s\n", argv[i]);
-	if (argc > 1)
-		goto out;
+	//if (argc > 1)
+	//	goto out;
 
 	weston_compositor_wake(wet.compositor);
 
