diff -Naur weston-9.0.0-old/libweston/backend-drm/drm.c weston-9.0.0-new/libweston/backend-drm/drm.c
--- weston-9.0.0-old/libweston/backend-drm/drm.c	2020-09-04 12:21:49.549400300 +0200
+++ weston-9.0.0-new/libweston/backend-drm/drm.c	2021-02-07 10:41:01.949999156 +0100
@@ -2929,8 +2929,8 @@
 	if (udev_input_init(&b->input,
 			    compositor, b->udev, seat_id,
 			    config->configure_device) < 0) {
-		weston_log("failed to create input devices\n");
-		goto err_sprite;
+		weston_log("failed to create input devices. Continuing with no input devices ...\n");
+		//goto err_sprite;
 	}
 
 	if (drm_backend_create_heads(b, drm_device) < 0) {
