diff -Naur flex-2.5.35-old/config.h flex-2.5.35-new/config.h
--- flex-2.5.35-old/config.h	2011-10-07 18:50:41.000000000 +0200
+++ flex-2.5.35-new/config.h	2011-10-07 18:53:00.000000000 +0200
@@ -183,13 +183,13 @@
 /* #undef const */
 
 /* Define to rpl_malloc if the replacement function should be used. */
-#define malloc rpl_malloc
+//#define malloc rpl_malloc
 
 /* Define to `int' if <sys/types.h> does not define. */
 /* #undef pid_t */
 
 /* Define to rpl_realloc if the replacement function should be used. */
-#define realloc rpl_realloc
+//#define realloc rpl_realloc
 
 /* Define to `unsigned' if <sys/types.h> does not define. */
 /* #undef size_t */
