diff -Naur pycurl-7.19.0-old/setup.py pycurl-7.19.0-new/setup.py
--- pycurl-7.19.0-old/setup.py	2010-09-15 17:04:44.000000000 -0700
+++ pycurl-7.19.0-new/setup.py	2010-09-15 17:06:59.000000000 -0700
@@ -97,8 +97,7 @@
         else:
             extra_compile_args.append(e)
     libs = split_quoted(
-        os.popen("'%s' --libs" % CURL_CONFIG).read()+\
-        os.popen("'%s' --static-libs" % CURL_CONFIG).read())
+        os.popen("'%s' --libs" % CURL_CONFIG).read())
     for e in libs:
         if e[:2] == "-l":
             libraries.append(e[2:])
