diff -Naur tinyxml2.old/tinyxml2-shared-targets.cmake tinyxml2.new/tinyxml2-shared-targets.cmake
--- tinyxml2.old/tinyxml2-shared-targets.cmake	2024-01-24 19:26:07.000000000 +0100
+++ tinyxml2.new/tinyxml2-shared-targets.cmake	2024-01-24 19:31:20.002560917 +0100
@@ -46,8 +46,11 @@
 unset(_cmake_expected_targets)
 
 
-# The installation prefix configured by this project.
-set(_IMPORT_PREFIX "/usr")
+# Compute the installation prefix relative to this file.
+get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
 
 # Create imported target tinyxml2::tinyxml2
 add_library(tinyxml2::tinyxml2 SHARED IMPORTED)
diff -Naur tinyxml2.old/tinyxml2-shared-targets-noconfig.cmake tinyxml2.new/tinyxml2-shared-targets-noconfig.cmake
--- tinyxml2.old/tinyxml2-shared-targets-noconfig.cmake	2024-01-24 19:26:07.000000000 +0100
+++ tinyxml2.new/tinyxml2-shared-targets-noconfig.cmake	2024-01-24 19:32:42.225892479 +0100
@@ -8,12 +8,12 @@
 # Import target "tinyxml2::tinyxml2" for configuration ""
 set_property(TARGET tinyxml2::tinyxml2 APPEND PROPERTY IMPORTED_CONFIGURATIONS NOCONFIG)
 set_target_properties(tinyxml2::tinyxml2 PROPERTIES
-  IMPORTED_LOCATION_NOCONFIG "/usr/lib/libtinyxml2.so.9.0.0"
+  IMPORTED_LOCATION_NOCONFIG "${_IMPORT_PREFIX}/lib/libtinyxml2.so.9.0.0"
   IMPORTED_SONAME_NOCONFIG "libtinyxml2.so.9"
   )
 
 list(APPEND _cmake_import_check_targets tinyxml2::tinyxml2 )
-list(APPEND _cmake_import_check_files_for_tinyxml2::tinyxml2 "/usr/lib/libtinyxml2.so.9.0.0" )
+list(APPEND _cmake_import_check_files_for_tinyxml2::tinyxml2 "${_IMPORT_PREFIX}/lib/libtinyxml2.so.9.0.0" )
 
 # Commands beyond this point should not need to know the version.
 set(CMAKE_IMPORT_FILE_VERSION)
