diff -Naur zoneinfo-2012j-old/Makefile zoneinfo-2012j-new/Makefile
--- zoneinfo-2012j-old/Makefile	2012-11-12 18:34:49.000000000 -0800
+++ zoneinfo-2012j-new/Makefile	2012-11-14 12:21:22.000000000 -0800
@@ -384,9 +384,8 @@
 
 $(TZLIB):	$(LIBOBJS)
 		-mkdir $(TOPDIR) $(LIBDIR)
-		ar ru $@ $(LIBOBJS)
-		if [ -x /usr/ucb/ranlib ] || [ -x /usr/bin/ranlib ]; \
-			then ranlib $@ ; fi
+		$(AR) ru $@ $(LIBOBJS)
+		$(RANLIB) $@
 
 date:		$(DATEOBJS)
 		$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $(DATEOBJS) $(LDLIBS)
