diff --speed-large-files --no-dereference --minimal -Naur parted-3.6/doc/pt_BR/Makefile.in parted-3.6/doc/pt_BR/Makefile.in
--- parted-3.6/doc/pt_BR/Makefile.in	2023-04-11 00:37:53.000000000 +0200
+++ parted-3.6/doc/pt_BR/Makefile.in	2025-05-10 14:59:45.974194031 +0200
@@ -1575,7 +1575,7 @@
 localstatedir = @localstatedir@
 localstatedir_c = @localstatedir_c@
 localstatedir_c_make = @localstatedir_c_make@
-mandir = $(mandir)/$(lang)
+mandir = @mandir@/$(lang)
 mandir_c = @mandir_c@
 mandir_c_make = @mandir_c_make@
 mkdir_p = @mkdir_p@
