diff -Naur libcec-libcec-3.0.1-old/src/libcec/CMakeLists.txt libcec-libcec-3.0.1-new/src/libcec/CMakeLists.txt
--- libcec-libcec-3.0.1-old/src/libcec/CMakeLists.txt	2015-07-03 10:20:49.000000000 -0700
+++ libcec-libcec-3.0.1-new/src/libcec/CMakeLists.txt	2015-07-25 17:27:02.272999571 -0700
@@ -122,7 +122,7 @@
 source_group("Header Files" FILES ${CEC_HEADERS})
 
 # platform and device specific
-include(cmake/SetBuildInfo.cmake)
+#include(cmake/SetBuildInfo.cmake)
 include(cmake/CheckPlatformSupport.cmake)
 
 ## create project groups
