diff -Naur PyXML-0.8.4-old/xml/xpath/ParsedAbbreviatedAbsoluteLocationPath.py PyXML-0.8.4-new/xml/xpath/ParsedAbbreviatedAbsoluteLocationPath.py
--- PyXML-0.8.4-old/xml/xpath/ParsedAbbreviatedAbsoluteLocationPath.py	2003-03-11 06:01:34.000000000 -0800
+++ PyXML-0.8.4-new/xml/xpath/ParsedAbbreviatedAbsoluteLocationPath.py	2010-06-28 15:51:53.000000000 -0700
@@ -24,8 +24,8 @@
         self._rel = rel
         nt = ParsedNodeTest.ParsedNodeTest('node', '')
         ppl = ParsedPredicateList.ParsedPredicateList([])
-        as = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
-        self._step = ParsedStep.ParsedStep(as, nt, ppl)
+        asp = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
+        self._step = ParsedStep.ParsedStep(asp, nt, ppl)
         return
 
     def evaluate(self, context):
diff -Naur PyXML-0.8.4-old/xml/xpath/ParsedAbbreviatedRelativeLocationPath.py PyXML-0.8.4-new/xml/xpath/ParsedAbbreviatedRelativeLocationPath.py
--- PyXML-0.8.4-old/xml/xpath/ParsedAbbreviatedRelativeLocationPath.py	2003-03-11 06:01:34.000000000 -0800
+++ PyXML-0.8.4-new/xml/xpath/ParsedAbbreviatedRelativeLocationPath.py	2010-06-28 15:51:53.000000000 -0700
@@ -28,8 +28,8 @@
         self._right = right
         nt = ParsedNodeTest.ParsedNodeTest('node','')
         ppl = ParsedPredicateList.ParsedPredicateList([])
-        as = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
-        self._middle = ParsedStep.ParsedStep(as, nt, ppl)
+        asp = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
+        self._middle = ParsedStep.ParsedStep(asp, nt, ppl)
 
     def evaluate(self, context):
         res = []
