diff -Naur linux-5.8-rc1-old/arch/arm64/kernel/head.S linux-5.8-rc1-new/arch/arm64/kernel/head.S
--- linux-5.8-rc1-old/arch/arm64/kernel/head.S	2020-06-14 21:45:04.000000000 +0200
+++ linux-5.8-rc1-new/arch/arm64/kernel/head.S	2020-06-18 12:27:31.419999989 +0200
@@ -43,7 +43,7 @@
 #elif (PAGE_OFFSET & 0x1fffff) != 0
 #error PAGE_OFFSET must be at least 2MB aligned
 #elif TEXT_OFFSET > 0x1fffff
-#error TEXT_OFFSET must be less than 2MB
+//#error TEXT_OFFSET must be less than 2MB
 #endif
 
 /*
diff -Naur linux-5.8-rc1-old/arch/arm64/Makefile linux-5.8-rc1-new/arch/arm64/Makefile
--- linux-5.8-rc1-old/arch/arm64/Makefile	2020-06-14 21:45:04.000000000 +0200
+++ linux-5.8-rc1-new/arch/arm64/Makefile	2020-06-18 12:30:25.479999981 +0200
@@ -126,7 +126,7 @@
 		 int(2 * 1024 * 1024 / (2 ^ $(CONFIG_ARM64_PAGE_SHIFT)) * \
 		 rand()) * (2 ^ $(CONFIG_ARM64_PAGE_SHIFT))}")
 else
-TEXT_OFFSET := 0x0
+TEXT_OFFSET := 0x01080000
 endif
 
 ifeq ($(CONFIG_KASAN_SW_TAGS), y)
