diff --speed-large-files --no-dereference --minimal -Naur linux-5.17-rc8/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c linux-5.17-rc8/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
--- linux-5.17-rc8/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c	2022-03-13 21:23:37.000000000 +0100
+++ linux-5.17-rc8/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c	2022-03-15 16:18:04.100453679 +0100
@@ -140,7 +140,7 @@
 		return err;
 	}
 
-	err = firmware_request_nowarn(&clm, clm_name, bus->dev);
+	err = request_firmware_direct(&clm, clm_name, bus->dev);
 	if (err) {
 		brcmf_info("no clm_blob available (err=%d), device may have limited channels available\n",
 			   err);
