diff --speed-large-files --no-dereference --minimal -Naur linux-5.17.4/sound/soc/codecs/hdmi-codec.c linux-5.17.4/sound/soc/codecs/hdmi-codec.c
--- linux-5.17.4/sound/soc/codecs/hdmi-codec.c	2022-04-22 18:17:27.959815428 +0200
+++ linux-5.17.4/sound/soc/codecs/hdmi-codec.c	2022-04-22 18:14:55.313149668 +0200
@@ -464,12 +464,14 @@
 {
 	struct hdmi_codec_priv *hcp = snd_soc_dai_get_drvdata(dai);
 
+#if ! defined(CONFIG_SND_SOC_SUN50I_HDMI)
 	hcp->chmap_idx = HDMI_CODEC_CHMAP_IDX_UNKNOWN;
 	hcp->hcd.ops->audio_shutdown(dai->dev->parent, hcp->hcd.data);
 
 	mutex_lock(&hcp->lock);
 	hcp->busy = false;
 	mutex_unlock(&hcp->lock);
+#endif
 }
 
 static int hdmi_codec_fill_codec_params(struct snd_soc_dai *dai,
@@ -1033,6 +1035,15 @@
 			__func__, ret);
 		return ret;
 	}
+
+#if defined(CONFIG_SND_SOC_SUN50I_HDMI)
+	if (hcp->hcd.ops->audio_startup) {
+		ret = hcp->hcd.ops->audio_startup(dev, hcp->hcd.data);
+		if (ret)
+			return 0;
+	}
+#endif
+
 	return 0;
 }
 
