diff --speed-large-files --no-dereference --minimal -Naur linux-5.17.2/sound/soc/codecs/hdmi-codec.c linux-5.17.2/sound/soc/codecs/hdmi-codec.c
--- linux-5.17.2/sound/soc/codecs/hdmi-codec.c	2022-04-14 12:44:19.179907544 +0200
+++ linux-5.17.2/sound/soc/codecs/hdmi-codec.c	2022-04-14 12:37:11.389915096 +0200
@@ -425,6 +425,7 @@
 	bool tx = substream->stream == SNDRV_PCM_STREAM_PLAYBACK;
 	int ret = 0;
 
+#if ! defined(CONFIG_SND_SOC_SUN50I_HDMI)
 	mutex_lock(&hcp->lock);
 	if (hcp->busy) {
 		dev_err(dai->dev, "Only one simultaneous stream supported!\n");
@@ -437,6 +438,7 @@
 		if (ret)
 			goto err;
 	}
+#endif
 
 	if (tx && hcp->hcd.ops->get_eld) {
 		ret = hcp->hcd.ops->get_eld(dai->dev->parent, hcp->hcd.data,
@@ -464,12 +466,14 @@
 {
 	struct hdmi_codec_priv *hcp = snd_soc_dai_get_drvdata(dai);
 
+#if ! defined(CONFIG_SND_SOC_SUN50I_HDMI)
 	hcp->chmap_idx = HDMI_CODEC_CHMAP_IDX_UNKNOWN;
 	hcp->hcd.ops->audio_shutdown(dai->dev->parent, hcp->hcd.data);
 
 	mutex_lock(&hcp->lock);
 	hcp->busy = false;
 	mutex_unlock(&hcp->lock);
+#endif
 }
 
 static int hdmi_codec_fill_codec_params(struct snd_soc_dai *dai,
@@ -1033,6 +1037,15 @@
 			__func__, ret);
 		return ret;
 	}
+
+#if defined(CONFIG_SND_SOC_SUN50I_HDMI)
+	if (hcp->hcd.ops->audio_startup) {
+		ret = hcp->hcd.ops->audio_startup(dev, hcp->hcd.data);
+		if (ret)
+			return 0;
+	}
+#endif
+
 	return 0;
 }
 
diff --speed-large-files --no-dereference --minimal -Naur linux-5.17.2/sound/soc/soc-utils.c linux-5.17.2/sound/soc/soc-utils.c
--- linux-5.17.2/sound/soc/soc-utils.c	2022-04-08 13:59:05.000000000 +0200
+++ linux-5.17.2/sound/soc/soc-utils.c	2022-04-14 12:37:11.389915096 +0200
@@ -76,10 +76,12 @@
 	 * If there are other components associated with rtd, we shouldn't
 	 * override their hwparams
 	 */
+#if ! defined(CONFIG_SND_SOC_SUN50I_HDMI)
 	for_each_rtd_components(rtd, i, component) {
 		if (component->driver == &dummy_platform)
 			return 0;
 	}
+#endif
 
 	/* BE's dont need dummy params */
 	if (!rtd->dai_link->no_pcm)
