diff --speed-large-files --no-dereference --minimal -Naur linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts
--- linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2022-03-06 12:32:16.173157193 +0100
+++ linux-5.17-rc6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2022-03-06 12:27:29.149829041 +0100
@@ -149,6 +149,18 @@
 		compatible = "linux,spdif-dit";
 		#sound-dai-cells = <0>;
 	};
+
+	openvfd {
+		compatible = "open,vfd";
+		dev_name = "openvfd";
+		openvfd_gpio_dat = <&gpio2 RK_PC7 GPIO_ACTIVE_HIGH>; /* PC7 */
+		openvfd_gpio_clk = <&gpio2 RK_PC4 GPIO_ACTIVE_HIGH>; /* PC4 */
+		openvfd_gpio_stb = <&gpio2 RK_PC2 GPIO_ACTIVE_LOW>; /* PC2 */
+		openvfd_chars = [02 04 03 01 00];
+		openvfd_dot_bits = [00 02 04 03 01 05 06];
+		openvfd_display_type = <0x02000009>;
+		status = "okay";
+	};
 };
 
 &analog_sound {
