diff --speed-large-files --no-dereference --minimal -Naur linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
--- linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts	2022-02-27 12:33:34.936642695 +0100
+++ linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts	2022-02-27 12:24:44.176652291 +0100
@@ -499,7 +499,6 @@
 			};
 
 			vccio_sd: LDO_REG5 {
-				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
@@ -692,7 +691,7 @@
 
 /* sata1 is muxed with the usb3 port */
 &sata1 {
-	status = "okay";
+	status = "disabled";
 };
 
 /* sata2 is muxed with the pcie2 slot*/
@@ -705,6 +704,8 @@
 	bus-width = <8>;
 	mmc-hs200-1_8v;
 	non-removable;
+	pinctrl-names = "default";
+	pinctrl-0 = <&emmc_bus8 &emmc_clk &emmc_cmd &emmc_datastrobe>;
 	vmmc-supply = <&vcc_3v3>;
 	vqmmc-supply = <&vcc_1v8>;
 	status = "okay";
@@ -740,15 +741,16 @@
 };
 
 &sfc {
-	#address-cells = <1>;
-	#size-cells = <0>;
-	status = "disabled";
+	pinctrl-0 = <&fspi_pins>;
+	pinctrl-names = "default";
+	rockchip,sfc-no-dma;
+	status = "okay";
 
 	flash@0 {
 		compatible = "jedec,spi-nor";
 		reg = <0>;
-		spi-max-frequency = <108000000>;
-		spi-rx-bus-width = <4>;
+		spi-max-frequency = <50000000>;
+		spi-rx-bus-width = <1>;
 		spi-tx-bus-width = <1>;
 	};
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts
--- linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2022-02-27 12:33:35.056642693 +0100
+++ linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2022-02-27 12:30:38.869979200 +0100
@@ -231,6 +231,8 @@
 
 			vdd_gpu: DCDC_REG2 {
 				regulator-name = "vdd_gpu";
+				regulator-always-on;
+				regulator-boot-on;
 				regulator-min-microvolt = <900000>;
 				regulator-max-microvolt = <1350000>;
 				regulator-init-microvolt = <900000>;
@@ -440,7 +442,7 @@
 &pcie2x1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pcie_reset_h>;
-	reset-gpios = <&gpio0 RK_PB6 GPIO_ACTIVE_HIGH>;
+	reset-gpios = <&gpio1 RK_PB2 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 	vpcie3v3-supply = <&vcc3v3_pcie>;
 };
@@ -513,6 +515,11 @@
 	vccio7-supply = <&vcc_3v3>;
 };
 
+&saradc {
+	vref-supply = <&vcca_1v8>;
+	status = "okay";
+};
+
 &sdhci {
 	bus-width = <8>;
 	mmc-hs200-1_8v;
@@ -543,22 +550,24 @@
 	mmc-pwrseq = <&sdio_pwrseq>;
 	non-removable;
 	pinctrl-names = "default";
-	pinctrl-0 = <&sdmmc1_bus4 &sdmmc1_cmd &sdmmc1_clk &uart9m1_xfer &uart8m1_xfer>;
+	pinctrl-0 = <&sdmmc1_bus4 &sdmmc1_cmd &sdmmc1_clk>;
 	vmmc-supply = <&vcc3v3_sys>;
 	vqmmc-supply = <&vcca1v8_pmu>;
 	status = "okay";
 };
 
-&sdmmc2 {
-	cap-sd-highspeed;
-	cap-sdio-irq;
-	bus-width = <4>;
-	vmmc-supply = <&vcc_3v3>;
-	vqmmc-supply = <&vcca1v8_pmu>;
+&sfc {
+	pinctrl-0 = <&fspi_pins>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&sdmmc2m0_bus4 &sdmmc2m0_cmd &sdmmc2m0_clk>;
-	sd-uhs-sdr104;
 	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <24000000>;
+		spi-rx-bus-width = <4>;
+		spi-tx-bus-width = <1>;
+	};
 };
 
 &tsadc {
diff --speed-large-files --no-dereference --minimal -Naur linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-x96_x6.dts linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-x96_x6.dts
--- linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-x96_x6.dts	2022-02-27 12:33:35.116642692 +0100
+++ linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk3566-x96_x6.dts	2022-02-27 12:31:57.106644452 +0100
@@ -411,6 +411,20 @@
 	status = "okay";
 };
 
+&sfc {
+	pinctrl-0 = <&fspi_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <24000000>;
+		spi-rx-bus-width = <4>;
+		spi-tx-bus-width = <1>;
+	};
+};
+
 &spdif {
 	status = "okay";
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk356x.dtsi linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk356x.dtsi
--- linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk356x.dtsi	2022-02-27 12:33:34.869976029 +0100
+++ linux-5.17-rc5/arch/arm64/boot/dts/rockchip/rk356x.dtsi	2022-02-27 12:24:44.179985624 +0100
@@ -984,8 +984,10 @@
 		interrupts = <GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&cru SCLK_SFC>, <&cru HCLK_SFC>;
 		clock-names = "clk_sfc", "hclk_sfc";
-		pinctrl-0 = <&fspi_pins>;
-		pinctrl-names = "default";
+		assigned-clocks = <&cru SCLK_SFC>;
+		assigned-clock-rates = <100000000>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		status = "disabled";
 	};
 
