diff -Naur libgpg-error-1.12-old/src/Makefile.am libgpg-error-1.12-new/src/Makefile.am
--- libgpg-error-1.12-old/src/Makefile.am	2013-06-17 06:51:40.000000000 -0700
+++ libgpg-error-1.12-new/src/Makefile.am	2015-07-12 18:01:54.178000232 -0700
@@ -140,7 +140,7 @@
 # It is correct to use $(CPP).  We want the host's idea of the error codes.
 mkerrcodes.h: Makefile mkerrcodes.awk $(gpg_extra_headers)
 	$(AWK) -f $(srcdir)/mkerrcodes1.awk $(srcdir)/errnos.in >_$@
-	$(CPP) $(CPPFLAGS) $(extra_cppflags) _$@ | grep GPG_ERR_ | \
+	$(CPP) $(CPPFLAGS) $(extra_cppflags) -P _$@ | grep GPG_ERR_ | \
                $(AWK) -f $(srcdir)/mkerrcodes.awk >$@
 	-rm _$@
 
diff -Naur libgpg-error-1.12-old/src/Makefile.in libgpg-error-1.12-new/src/Makefile.in
--- libgpg-error-1.12-old/src/Makefile.in	2013-06-23 21:42:26.000000000 -0700
+++ libgpg-error-1.12-new/src/Makefile.in	2015-07-12 18:01:54.179000232 -0700
@@ -979,7 +979,7 @@
 # It is correct to use $(CPP).  We want the host's idea of the error codes.
 mkerrcodes.h: Makefile mkerrcodes.awk $(gpg_extra_headers)
 	$(AWK) -f $(srcdir)/mkerrcodes1.awk $(srcdir)/errnos.in >_$@
-	$(CPP) $(CPPFLAGS) $(extra_cppflags) _$@ | grep GPG_ERR_ | \
+	$(CPP) $(CPPFLAGS) $(extra_cppflags) -P _$@ | grep GPG_ERR_ | \
                $(AWK) -f $(srcdir)/mkerrcodes.awk >$@
 	-rm _$@
 
