diff -Naur mplayer-resumer-20060503-old/mplayer-resumer.pl mplayer-resumer-20060503-new/mplayer-resumer.pl
--- mplayer-resumer-20060503-old/mplayer-resumer.pl	2008-05-25 09:37:44.000000000 -0700
+++ mplayer-resumer-20060503-new/mplayer-resumer.pl	2008-05-25 10:34:48.000000000 -0700
@@ -3,6 +3,7 @@
 use Shell;
 use strict;
 use POSIX qw(floor);
+use Cwd qw(abs_path);
 
 # Written by Bob Igo from the MythTV Store at http://MythiC.TV
 # Email: bob@stormlogic.com
@@ -101,17 +102,27 @@
 # For $pathname, return $path.$filename.resume
 sub get_resume_filename () {
     my($pathname)=@_;
-    
-    my $idx = rindex($pathname,"/");
 
-    if ($idx == -1) { # There was no "/" in $pathname
-	return ".".$pathname.".resume";
-    } else {
-	# Now we need to split out the path from the filename.
-	my $path = substr($pathname,0,$idx+1);
-	my $filename = substr($pathname,$idx+1);
-	return "$path.$filename.resume";
+    # Convert to absolute path.
+    $pathname=abs_path($pathname);
+    die "File does not exist: $pathname" if (! -e $pathname);
+
+    # Determine and create the mplayer-resumer configuration directory.
+    my $home=$ENV{'HOME'};
+    if ($home ne "") {
+        $home=abs_path($home);
     }
+    die "Could not determine home directory" if ($home eq "");
+    die "Home directory does not exist: $home" if (! -e $home);
+
+    my $conf="$home/.mplayer-resumer";
+    mkdir("$conf", 0755) if (! -e $conf);
+    die "Could not create mplayer-resumer configuration directory: $conf" if (! -d $conf);
+
+    $pathname=~s/\//~/g;
+    $pathname="$conf/$pathname";
+
+    return $pathname
 }
 
 # Calls mplayer and returns the last known video position
