diff --speed-large-files --no-dereference --minimal -Naur Python-3.9.14/configure.ac Python-3.9.14/configure.ac
--- Python-3.9.14/configure.ac	2022-09-06 19:26:16.000000000 +0200
+++ Python-3.9.14/configure.ac	2022-10-21 23:08:28.689618823 +0200
@@ -893,7 +893,7 @@
 
 if test x$PLATFORM_TRIPLET != x && test x$MULTIARCH != x; then
   if test x$PLATFORM_TRIPLET != x$MULTIARCH; then
-    AC_MSG_ERROR([internal configure error for the platform triplet, please file a bug report])
+    AC_MSG_RESULT([internal configure error for the platform triplet, please file a bug report])
   fi
 elif test x$PLATFORM_TRIPLET != x && test x$MULTIARCH = x; then
   MULTIARCH=$PLATFORM_TRIPLET
