diff -Naur Python-3.7.1-old/configure.ac Python-3.7.1-new/configure.ac
--- Python-3.7.1-old/configure.ac	2018-10-20 08:04:19.000000000 +0200
+++ Python-3.7.1-new/configure.ac	2020-04-13 17:20:41.925270803 +0200
@@ -882,7 +882,7 @@
 
 if test x$PLATFORM_TRIPLET != x && test x$MULTIARCH != x; then
   if test x$PLATFORM_TRIPLET != x$MULTIARCH; then
-    AC_MSG_ERROR([internal configure error for the platform triplet, please file a bug report])
+    AC_MSG_RESULT([internal configure error for the platform triplet, please file a bug report])
   fi
 elif test x$PLATFORM_TRIPLET != x && test x$MULTIARCH = x; then
   MULTIARCH=$PLATFORM_TRIPLET
