diff --speed-large-files --no-dereference --minimal -Naur mpv-e48ac7ce08462f5e33af6ef9deeac6fa87eef01e/meson.build mpv-e48ac7ce08462f5e33af6ef9deeac6fa87eef01e/meson.build
--- mpv-e48ac7ce08462f5e33af6ef9deeac6fa87eef01e/meson.build	2025-03-25 18:33:27.000000000 +0100
+++ mpv-e48ac7ce08462f5e33af6ef9deeac6fa87eef01e/meson.build	2025-06-20 17:00:52.127871195 +0200
@@ -581,7 +581,6 @@
 docutils_wrapper = find_program(join_paths(tools_directory, 'docutils-wrapper.py'))
 file2string = find_program(join_paths(tools_directory, 'file2string.py'))
 matroska = find_program(join_paths(tools_directory, 'matroska.py'))
-mpv_desktop = find_program(join_paths(tools_directory, 'gen-mpv-desktop.py'))
 
 ebml_defs = custom_target('ebml_defs.inc',
     output: 'ebml_defs.inc',
@@ -1841,14 +1840,6 @@
                 command: [mpv, '--no-config','--list-protocols'],
                 capture: true,
             )
-            mpv_desktop_path = join_paths(source_root, 'etc', 'mpv.desktop')
-            custom_target('mpv.desktop',
-                input: mpv_protocols,
-                output: 'mpv.desktop',
-                command: [mpv_desktop, mpv_desktop_path, '@INPUT@', '@OUTPUT@'],
-                install: true,
-                install_dir: join_paths(datadir, 'applications'),
-            )
         else
             install_data('etc/mpv.desktop', install_dir: join_paths(datadir, 'applications'))
         endif
