diff --speed-large-files --no-dereference --minimal -Naur mpv-a0fba7be57f3822d967b04f0f6b6d6341e7516e7/meson.build mpv-a0fba7be57f3822d967b04f0f6b6d6341e7516e7/meson.build
--- mpv-a0fba7be57f3822d967b04f0f6b6d6341e7516e7/meson.build	2024-09-23 19:36:41.000000000 +0200
+++ mpv-a0fba7be57f3822d967b04f0f6b6d6341e7516e7/meson.build	2025-01-09 21:29:16.077905985 +0100
@@ -586,7 +586,6 @@
 docutils_wrapper = find_program(join_paths(tools_directory, 'docutils-wrapper.py'))
 file2string = find_program(join_paths(tools_directory, 'file2string.py'))
 matroska = find_program(join_paths(tools_directory, 'matroska.py'))
-mpv_desktop = find_program(join_paths(tools_directory, 'gen-mpv-desktop.py'))
 
 ebml_defs = custom_target('ebml_defs.inc',
     output: 'ebml_defs.inc',
@@ -1827,21 +1826,6 @@
             command: [osxbundle, mpv.full_path(), '@SOURCE_ROOT@'],
         )
     endif
-
-    if not win32 and not darwin
-        if meson.can_run_host_binaries()
-            mpv_desktop_path = join_paths(source_root, 'etc', 'mpv.desktop')
-            custom_target('mpv.desktop',
-                depends: mpv,
-                output: 'mpv.desktop',
-                command: [mpv_desktop, mpv_desktop_path, mpv.full_path(), '@OUTPUT@'],
-                install: true,
-                install_dir: join_paths(datadir, 'applications'),
-            )
-        else
-            install_data('etc/mpv.desktop', install_dir: join_paths(datadir, 'applications'))
-        endif
-    endif
 endif
 
 if get_option('tests')
