diff -Naur fceu-old/configure.ac fceu-new/configure.ac
--- fceu-old/configure.ac	2004-09-11 14:32:25.000000000 -0700
+++ fceu-new/configure.ac	2006-12-25 20:52:14.000000000 -0800
@@ -68,9 +68,9 @@
 AC_DEFINE([PSS_STYLE],[1])
 AC_DEFINE([NETWORK],[1])
 
-AC_CHECK_HEADER([sys/soundcard.h],
-        AM_CONDITIONAL(USE_SEXYAL, true)
-	AC_DEFINE([USE_SEXYAL]),[])
+dnl AC_CHECK_HEADER([sys/soundcard.h],
+dnl        AM_CONDITIONAL(USE_SEXYAL, true)
+dnl	AC_DEFINE([USE_SEXYAL]),[])
 fi
 
 dnl if expr x"$target" : 'x.*darwin' > /dev/null; then
