diff -Naur dbus-glib-0.104-old/dbus/Makefile.am dbus-glib-0.104-new/dbus/Makefile.am
--- dbus-glib-0.104-old/dbus/Makefile.am	2015-02-09 15:26:34.000000000 +0100
+++ dbus-glib-0.104-new/dbus/Makefile.am	2020-04-14 21:42:08.383852154 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = . examples
+SUBDIRS = . 
 
 INCLUDES = \
 	-I$(top_srcdir)						\
diff -Naur dbus-glib-0.104-old/Makefile.am dbus-glib-0.104-new/Makefile.am
--- dbus-glib-0.104-old/Makefile.am	2015-02-09 15:26:34.000000000 +0100
+++ dbus-glib-0.104-new/Makefile.am	2020-04-14 21:37:49.566099823 +0200
@@ -1,8 +1,8 @@
 ACLOCAL_AMFLAGS = -I m4
 GLIB_PC=dbus-glib-1.pc
 
-SUBDIRS=dbus tools test doc
-DIST_SUBDIRS=dbus tools test doc m4
+SUBDIRS=dbus tools 
+DIST_SUBDIRS=dbus tools m4
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = $(GLIB_PC)
diff -Naur dbus-glib-0.104-old/tools/Makefile.am dbus-glib-0.104-new/tools/Makefile.am
--- dbus-glib-0.104-old/tools/Makefile.am	2015-02-09 15:22:46.000000000 +0100
+++ dbus-glib-0.104-new/tools/Makefile.am	2020-04-14 21:51:47.448926602 +0200
@@ -4,7 +4,7 @@
 libdbus_glibdir = $(includedir)/dbus-1.0/dbus
 
 dbus-glib-bindings.h: $(top_srcdir)/dbus-bus-introspect.xml $(top_builddir)/dbus/dbus-binding-tool$(EXEEXT)
-	$(DBUS_BINDING_TOOL) --mode=glib-client --prefix=dbus_bus --output=dbus-glib-bindings.h $(top_srcdir)/dbus-bus-introspect.xml
+	@touch dbus-glib-bindings.h
 
 BUILT_SOURCES = dbus-glib-bindings.h
 
