diff -Naur x11vnc-0.9.13-old/configure.ac x11vnc-0.9.13-new/configure.ac
--- x11vnc-0.9.13-old/configure.ac	2011-08-10 15:34:57.000000000 -0700
+++ x11vnc-0.9.13-new/configure.ac	2013-01-12 15:50:21.000000000 -0800
@@ -333,9 +333,9 @@
 		if test "x$ld_minus_R" = "xno"; then
 			:
 		elif test "x$GCC" = "xyes"; then
-			LDFLAGS="$LDFLAGS -Xlinker -R$with_ssl/lib"
+			LDFLAGS="$LDFLAGS"
 		else
-			LDFLAGS="$LDFLAGS -R$with_ssl/lib"
+			LDFLAGS="$LDFLAGS"
 		fi
 	fi
 	AC_CHECK_LIB(crypto, RAND_file_name,
@@ -450,9 +450,9 @@
 		if test "x$ld_minus_R" = "xno"; then
 			:
 		elif test "x$GCC" = "xyes"; then
-			rflag="-Xlinker -R$with_system_libvncserver/lib"
+			rflag=""
 		else
-			rflag="-R$with_system_libvncserver/lib"
+			rflag=""
 		fi
 		cmd="$with_system_libvncserver/bin/libvncserver-config"
 		if $cmd --version 1>/dev/null 2>&1; then
@@ -487,9 +487,9 @@
 		if test "x$ld_minus_R" = "xno"; then
 			:
 		elif test "x$GCC" = "xyes"; then
-			rflag=" -Xlinker -R$rprefix/lib "
+			rflag=""
 		else
-			rflag=" -R$rprefix/lib "
+			rflag=""
 		fi
 		cvers=`libvncserver-config --version 2>/dev/null`
 		cscore=`echo "$cvers" | tr '.' ' ' | awk '{print 10000 * $1 + 100 * $2 + $3}'`
@@ -545,9 +545,9 @@
 			:
 		elif test "x$GCC" = "xyes"; then
 			# this is not complete... in general a rat's nest.
-			LDFLAGS="$LDFLAGS -Xlinker -R$with_jpeg/lib"
+			LDFLAGS="$LDFLAGS"
 		else
-			LDFLAGS="$LDFLAGS -R$with_jpeg/lib"
+			LDFLAGS="$LDFLAGS"
 		fi
 	fi
 	AC_CHECK_HEADER(jpeglib.h, HAVE_JPEGLIB_H="true")
@@ -593,9 +593,9 @@
 		if test "x$ld_minus_R" = "xno"; then
 			:
 		elif test "x$GCC" = "xyes"; then
-			LDFLAGS="$LDFLAGS -Xlinker -R$with_zlib/lib"
+			LDFLAGS="$LDFLAGS"
 		else
-			LDFLAGS="$LDFLAGS -R$with_zlib/lib"
+			LDFLAGS="$LDFLAGS"
 		fi
 	fi
 	AC_CHECK_HEADER(zlib.h, HAVE_ZLIB_H="true")
