diff -Naur x11vnc-0.9.13-old/configure.ac x11vnc-0.9.13-new/configure.ac
--- x11vnc-0.9.13-old/configure.ac	2011-08-10 15:34:57.000000000 -0700
+++ x11vnc-0.9.13-new/configure.ac	2013-01-03 19:31:27.000000000 -0800
@@ -1,7 +1,7 @@
 # Process this file with autoconf to produce a configure script.
 AC_INIT(x11vnc, 0.9.13, http://sourceforge.net/projects/libvncserver)
 AM_INIT_AUTOMAKE(x11vnc, 0.9.13)
-AM_CONFIG_HEADER(rfbconfig.h)
+AC_CONFIG_HEADERS(rfbconfig.h)
 AX_PREFIX_CONFIG_H([rfb/rfbconfig.h])
 
 # Checks for programs.
