diff -Naur x11vnc-0.9.13-old/configure.ac x11vnc-0.9.13-new/configure.ac
--- x11vnc-0.9.13-old/configure.ac	2011-08-10 15:34:57.000000000 -0700
+++ x11vnc-0.9.13-new/configure.ac	2011-09-13 21:09:32.000000000 -0700
@@ -15,7 +15,7 @@
 test "x$GCC" = "xyes" && CFLAGS="$CFLAGS -Wall"
 AC_PROG_MAKE_SET
 AC_PROG_RANLIB
-AC_PATH_PROG([AR], [ar], [/usr/bin/ar],
+AC_PATH_TOOL([AR], [ar], [/usr/bin/ar],
                   [$PATH:/usr/ccs/bin])
 
 # Options
