diff -Naur NVIDIA-Linux-x86_64-340.104-old/kernel/nv-drm.c NVIDIA-Linux-x86_64-340.104-new/kernel/nv-drm.c
--- NVIDIA-Linux-x86_64-340.104-old/kernel/nv-drm.c	2017-09-15 01:16:51.000000000 +0200
+++ NVIDIA-Linux-x86_64-340.104-new/kernel/nv-drm.c	2017-11-15 15:30:19.326666665 +0100
@@ -173,7 +173,7 @@
 {
     int ret = 0;
 #if defined(NV_DRM_AVAILABLE)
-    ret = drm_pci_init(&nv_drm_driver, pci_driver);
+    ret = drm_legacy_pci_init(&nv_drm_driver, pci_driver);
 #endif
     return ret;
 }
@@ -183,7 +183,7 @@
 )
 {
 #if defined(NV_DRM_AVAILABLE)
-    drm_pci_exit(&nv_drm_driver, pci_driver);
+    drm_legacy_pci_exit(&nv_drm_driver, pci_driver);
 #endif
 }
 
diff -Naur NVIDIA-Linux-x86_64-340.104-old/kernel/uvm/nvidia_uvm_lite.c NVIDIA-Linux-x86_64-340.104-new/kernel/uvm/nvidia_uvm_lite.c
--- NVIDIA-Linux-x86_64-340.104-old/kernel/uvm/nvidia_uvm_lite.c	2017-09-15 01:16:51.000000000 +0200
+++ NVIDIA-Linux-x86_64-340.104-new/kernel/uvm/nvidia_uvm_lite.c	2017-11-15 15:30:19.326666665 +0100
@@ -818,8 +818,15 @@
 }
 
 #if defined(NV_VM_OPERATIONS_STRUCT_HAS_FAULT)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0)
 int _fault(struct vm_area_struct *vma, struct vm_fault *vmf)
+#else
+int _fault(struct vm_fault *vmf)
+#endif
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
+    struct vm_area_struct *vma = vmf->vma;
+#endif
 #if defined(NV_VM_FAULT_HAS_ADDRESS)
     unsigned long vaddr = vmf->address;
 #else
@@ -866,7 +873,11 @@
 // it's dealing with anonymous mapping (see handle_pte_fault).
 //
 #if defined(NV_VM_OPERATIONS_STRUCT_HAS_FAULT)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0)
 int _sigbus_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
+#else
+int _sigbus_fault(struct vm_fault *vmf)
+#endif
 {
     vmf->page = NULL;
     return VM_FAULT_SIGBUS;
