From b70e46212640205e006bb1db3066ef7c68d78dff Mon Sep 17 00:00:00 2001
From: Qiang Yu <yuq825@gmail.com>
Date: Wed, 27 Feb 2019 21:32:16 +0800
Subject: [PATCH 5/5] drm/lima: restrict lima to tested arch

Signed-off-by: Qiang Yu <yuq825@gmail.com>
---
 drivers/gpu/drm/lima/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/lima/Kconfig b/drivers/gpu/drm/lima/Kconfig
index 7cf061c..260a3b4 100644
--- a/drivers/gpu/drm/lima/Kconfig
+++ b/drivers/gpu/drm/lima/Kconfig
@@ -4,6 +4,7 @@
 config DRM_LIMA
        tristate "LIMA (DRM support for ARM Mali 400/450 GPU)"
        depends on DRM
+       depends on ARM || ARM64 || COMPILE_TEST
        select DRM_SCHED
        help
          DRM driver for ARM Mali 400/450 GPUs.
-- 
2.7.1

