diff --speed-large-files --no-dereference --minimal -Naur linux-6.8.3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts linux-6.8.3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts
--- linux-6.8.3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2024-04-03 17:19:43.053406631 +0200
+++ linux-6.8.3/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2024-04-03 17:10:51.243404849 +0200
@@ -443,6 +443,7 @@
 	bus-width = <4>;
 	cap-mmc-highspeed;
 	cap-sd-highspeed;
+	max-frequency = <35000000>;
 	disable-wp;
 	pinctrl-names = "default";
 	pinctrl-0 = <&sdmmc0_clk &sdmmc0_cmd &sdmmc0_dectn &sdmmc0_bus4>;
