diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc4/drivers/net/wireless/aic8800_usb/aic8800_fdrv/Makefile linux-6.10-rc4/drivers/net/wireless/aic8800_usb/aic8800_fdrv/Makefile
--- linux-6.10-rc4/drivers/net/wireless/aic8800_usb/aic8800_fdrv/Makefile	2024-06-19 12:21:24.934574091 +0200
+++ linux-6.10-rc4/drivers/net/wireless/aic8800_usb/aic8800_fdrv/Makefile	2024-06-19 12:17:09.987912904 +0200
@@ -104,7 +104,7 @@
 CONFIG_USE_P2P0 = n
 CONFIG_ONE_TXQ = n
 CONFIG_PER_STA_FC = n
-CONFIG_PREALLOC_RX_SKB ?= n
+CONFIG_PREALLOC_RX_SKB ?= y
 CONFIG_PREALLOC_TXQ ?= y
 CONFIG_USE_WIRELESS_EXT = y
 CONFIG_DPD = y
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc4/drivers/net/wireless/aic8800_usb/aic_load_fw/Makefile linux-6.10-rc4/drivers/net/wireless/aic8800_usb/aic_load_fw/Makefile
--- linux-6.10-rc4/drivers/net/wireless/aic8800_usb/aic_load_fw/Makefile	2024-06-19 12:21:24.901240758 +0200
+++ linux-6.10-rc4/drivers/net/wireless/aic8800_usb/aic_load_fw/Makefile	2024-06-19 12:17:09.941246238 +0200
@@ -10,7 +10,7 @@
 
 # Need to set fw path in BOARD_KERNEL_CMDLINE
 CONFIG_USE_FW_REQUEST ?= n
-CONFIG_PREALLOC_RX_SKB ?= n
+CONFIG_PREALLOC_RX_SKB ?= y
 CONFIG_PREALLOC_TXQ ?= y
 
 # Platform support list
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc4/drivers/net/wireless/aic8800_usb/Makefile linux-6.10-rc4/drivers/net/wireless/aic8800_usb/Makefile
--- linux-6.10-rc4/drivers/net/wireless/aic8800_usb/Makefile	2024-06-19 12:21:24.937907423 +0200
+++ linux-6.10-rc4/drivers/net/wireless/aic8800_usb/Makefile	2024-06-19 12:17:09.987912904 +0200
@@ -6,7 +6,7 @@
 
 ########## config option ##########
 export CONFIG_USE_FW_REQUEST = n
-export CONFIG_PREALLOC_RX_SKB = n
+export CONFIG_PREALLOC_RX_SKB = y
 export CONFIG_PREALLOC_TXQ = y
 ###################################
 
