diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts
--- linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2024-06-26 14:44:50.237623091 +0200
+++ linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2024-06-26 14:38:30.234297929 +0200
@@ -14,6 +14,7 @@
 
 	aliases {
 		ethernet0 = &gmac1;
+		ethernet1 = &sdmmc1;
 		mmc0 = &sdmmc0;
 		mmc1 = &sdhci;
 		mmc2 = &sdmmc1;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-rock-3c.dts linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-rock-3c.dts
--- linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-rock-3c.dts	2024-06-26 14:44:50.200956424 +0200
+++ linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-rock-3c.dts	2024-06-26 14:38:56.904297350 +0200
@@ -20,6 +20,7 @@
 
 	aliases {
 		ethernet0 = &gmac1;
+		ethernet1 = &sdmmc1;
 		mmc0 = &sdmmc0;
 		mmc1 = &sdhci;
 		mmc2 = &sdmmc1;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-urve-pi.dts linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-urve-pi.dts
--- linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-urve-pi.dts	2024-06-26 14:44:50.274289756 +0200
+++ linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-urve-pi.dts	2024-06-26 14:39:41.000963079 +0200
@@ -19,6 +19,7 @@
 
 	aliases {
 		ethernet0 = &gmac1;
+		ethernet1 = &sdmmc1;
 		mmc0 = &sdmmc0;
 		mmc1 = &sdmmc1;
 		mmc2 = &sdhci;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-x96_x6.dts linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-x96_x6.dts
--- linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-x96_x6.dts	2024-06-26 14:44:50.020956428 +0200
+++ linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3566-x96_x6.dts	2024-06-26 14:40:17.784295626 +0200
@@ -19,6 +19,7 @@
 
 	aliases {
 		ethernet0 = &gmac1;
+		ethernet1 = &sdmmc1;
 		mmc0 = &sdmmc0;
 		mmc1 = &sdmmc1;
 		mmc2 = &sdhci;
