diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s.dtsi linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s.dtsi
--- linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s.dtsi	2024-07-04 17:14:01.277680780 +0200
+++ linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s.dtsi	2024-07-02 10:35:07.194707545 +0200
@@ -1472,6 +1472,7 @@
 		rockchip,vo1_grf = <&vo1_grf>;
 		phys = <&hdptxphy_hdmi0>;
 		phy-names = "hdmi";
+		#sound-dai-cells = <0>;
 		status = "disabled";
 
 		ports {
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s.dtsi linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s.dtsi
--- linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s.dtsi	2024-07-05 16:18:16.167679572 +0200
+++ linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s.dtsi	2024-07-05 12:56:05.724606770 +0200
@@ -1455,12 +1455,20 @@
 			 <&cru CLK_HDMITX0_EARC>,
 			 <&cru CLK_HDMITX0_REF>,
 			 <&cru MCLK_I2S5_8CH_TX>,
+			 <&cru DCLK_VOP0>,
+			 <&cru DCLK_VOP1>,
+			 <&cru DCLK_VOP2>,
+			 <&cru DCLK_VOP3>,
 			 <&cru HCLK_VO1>;
 		clock-names = "pclk",
 			      "hdp",
 			      "earc",
 			      "ref",
 			      "aud",
+			      "dclk_vp0",
+			      "dclk_vp1",
+			      "dclk_vp2",
+			      "dclk_vp3",
 			      "hclk_vo1";
 		resets = <&cru SRST_HDMITX0_REF>, <&cru SRST_HDMIHDP0>;
 		reset-names = "ref", "hdp";
