diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc7/arch/arm64/boot/dts/rockchip/rk3588s.dtsi linux-6.10-rc7/arch/arm64/boot/dts/rockchip/rk3588s.dtsi
--- linux-6.10-rc7/arch/arm64/boot/dts/rockchip/rk3588s.dtsi	2024-07-13 10:06:29.388060113 +0200
+++ linux-6.10-rc7/arch/arm64/boot/dts/rockchip/rk3588s.dtsi	2024-07-13 10:04:37.711395855 +0200
@@ -1481,6 +1481,7 @@
 		phys = <&hdptxphy_hdmi0>;
 		phy-names = "hdmi";
 		#sound-dai-cells = <0>;
+		cec-enable;
 		status = "disabled";
 
 		ports {
