diff --speed-large-files --no-dereference --minimal -Naur linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
--- linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-05-28 17:44:31.029419125 +0200
+++ linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-05-28 17:43:02.411542913 +0200
@@ -7,6 +7,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/leds/common.h>
+#include <dt-bindings/soc/rockchip,vop2.h>
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/usb/pd.h>
@@ -992,3 +993,41 @@
 &usb_host1_xhci {
 	status = "okay";
 };
+
+&hdmi0 {
+	status = "okay";
+};
+
+&display_subsystem {
+	clocks = <&hdptxphy_hdmi0>;
+	clock-names = "hdmi0_phy_pll";
+};
+
+&hdmi0 {
+	status = "okay";
+};
+
+&hdmi0_in {
+	hdmi0_in_vp0: endpoint {
+		remote-endpoint = <&vp0_out_hdmi0>;
+	};
+};
+
+&hdptxphy_hdmi0 {
+	status = "okay";
+};
+
+&vop_mmu {
+	status = "okay";
+};
+
+&vop {
+	status = "okay";
+};
+
+&vp0 {
+	vp0_out_hdmi0: endpoint@ROCKCHIP_VOP2_EP_HDMI0 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI0>;
+		remote-endpoint = <&hdmi0_in_vp0>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
--- linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2024-05-28 17:44:31.029419125 +0200
+++ linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2024-05-28 17:43:00.350511920 +0200
@@ -4,6 +4,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/leds/common.h>
+#include <dt-bindings/soc/rockchip,vop2.h>
 #include "rk3588.dtsi"
 
 / {
@@ -927,3 +928,37 @@
 &usb_host2_xhci {
 	status = "okay";
 };
+
+&display_subsystem {
+	clocks = <&hdptxphy_hdmi0>;
+	clock-names = "hdmi0_phy_pll";
+};
+
+&hdmi0 {
+	status = "okay";
+};
+
+&hdmi0_in {
+	hdmi0_in_vp0: endpoint {
+		remote-endpoint = <&vp0_out_hdmi0>;
+	};
+};
+
+&hdptxphy_hdmi0 {
+	status = "okay";
+};
+
+&vop_mmu {
+	status = "okay";
+};
+
+&vop {
+	status = "okay";
+};
+
+&vp0 {
+	vp0_out_hdmi0: endpoint@ROCKCHIP_VOP2_EP_HDMI0 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI0>;
+		remote-endpoint = <&hdmi0_in_vp0>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts
--- linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2024-05-28 17:44:31.032752460 +0200
+++ linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2024-05-28 17:42:55.813353091 +0200
@@ -4,6 +4,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/leds/common.h>
+#include <dt-bindings/soc/rockchip,vop2.h>
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/usb/pd.h>
@@ -845,3 +846,38 @@
 	dr_mode = "host";
 	status = "okay";
 };
+
+&display_subsystem {
+	clocks = <&hdptxphy_hdmi0>;
+	clock-names = "hdmi0_phy_pll";
+};
+
+&hdmi0 {
+	status = "okay";
+};
+
+&hdmi0_in {
+	hdmi0_in_vp0: endpoint {
+		remote-endpoint = <&vp0_out_hdmi0>;
+	};
+};
+
+&hdptxphy_hdmi0 {
+	status = "okay";
+};
+
+&vop_mmu {
+	status = "okay";
+};
+
+&vop {
+	status = "okay";
+};
+
+&vp0 {
+	vp0_out_hdmi0: endpoint@ROCKCHIP_VOP2_EP_HDMI0 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI0>;
+		remote-endpoint = <&hdmi0_in_vp0>;
+	};
+};
+
diff --speed-large-files --no-dereference --minimal -Naur linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
--- linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2024-05-28 17:44:31.032752460 +0200
+++ linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2024-05-28 17:42:58.356288237 +0200
@@ -4,6 +4,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/leds/common.h>
+#include <dt-bindings/soc/rockchip,vop2.h>
 #include <dt-bindings/pinctrl/rockchip.h>
 #include "rk3588s.dtsi"
 
@@ -771,3 +772,37 @@
 &usb_host2_xhci {
 	status = "okay";
 };
+
+&display_subsystem {
+	clocks = <&hdptxphy_hdmi0>;
+	clock-names = "hdmi0_phy_pll";
+};
+
+&hdmi0 {
+	status = "okay";
+};
+
+&hdmi0_in {
+	hdmi0_in_vp0: endpoint {
+		remote-endpoint = <&vp0_out_hdmi0>;
+	};
+};
+
+&hdptxphy_hdmi0 {
+	status = "okay";
+};
+
+&vop_mmu {
+	status = "okay";
+};
+
+&vop {
+	status = "okay";
+};
+
+&vp0 {
+	vp0_out_hdmi0: endpoint@ROCKCHIP_VOP2_EP_HDMI0 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI0>;
+		remote-endpoint = <&hdmi0_in_vp0>;
+	};
+};
