diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
--- linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-06-17 18:59:15.374682346 +0200
+++ linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-06-17 18:56:12.611352922 +0200
@@ -506,6 +506,10 @@
 	status = "okay";
 };
 
+&rkvdec0 {
+	status = "okay";
+};
+
 &saradc {
 	vref-supply = <&vcc_1v8_s0>;
 	status = "okay";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
--- linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2024-06-17 18:59:15.408015679 +0200
+++ linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2024-06-17 18:54:29.184688452 +0200
@@ -468,6 +468,10 @@
 	status = "okay";
 };
 
+&rkvdec0 {
+	status = "okay";
+};
+
 &saradc {
 	vref-supply = <&avcc_1v8_s0>;
 	status = "okay";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts
--- linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2024-06-17 18:59:15.374682346 +0200
+++ linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2024-06-17 18:54:29.184688452 +0200
@@ -395,6 +395,10 @@
 	};
 };
 
+&rkvdec0 {
+	status = "okay";
+};
+
 &saradc {
 	vref-supply = <&avcc_1v8_s0>;
 	status = "okay";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
--- linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2024-06-17 18:59:15.408015679 +0200
+++ linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2024-06-17 18:57:06.498018433 +0200
@@ -361,6 +361,10 @@
 	status = "okay";
 };
 
+&rkvdec0 {
+	status = "okay";
+};
+
 &saradc {
 	vref-supply = <&avcc_1v8_s0>;
 	status = "okay";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts
--- linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts	2024-06-17 18:59:15.338015680 +0200
+++ linux-6.10-rc3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts	2024-06-17 18:57:29.654684618 +0200
@@ -383,6 +383,10 @@
 	status = "okay";
 };
 
+&rkvdec0 {
+	status = "okay";
+};
+
 &saradc {
 	vref-supply = <&avcc_1v8_s0>;
 	status = "okay";
