diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts
--- linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts	2024-07-04 17:14:01.564347442 +0200
+++ linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts	2024-07-04 16:40:25.794390749 +0200
@@ -137,6 +137,16 @@
 		regulator-max-microvolt = <1100000>;
 		vin-supply = <&vcc5v0_sys>;
 	};
+
+	hdmi0_sound: hdmi0-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi0";
+		rockchip,cpu = <&i2s5_8ch>;
+		rockchip,codec = <&hdmi0>;
+		rockchip,jack-det;
+	};
 };
 
 &combphy2_psu {
@@ -298,6 +308,10 @@
 	};
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
 &combphy0_ps {
 	status = "okay";
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
--- linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-07-05 16:48:50.617640029 +0200
+++ linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-07-05 16:45:26.120977768 +0200
@@ -159,6 +159,16 @@
 		};
 	};
 
+	hdmi0_sound: hdmi0-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi0";
+		rockchip,cpu = <&i2s5_8ch>;
+		rockchip,codec = <&hdmi0>;
+		rockchip,jack-det;
+	};
+
 	vcc3v3_pcie30: vcc3v3-pcie30-regulator {
 		compatible = "regulator-fixed";
 		enable-active-high;
@@ -424,6 +434,10 @@
 	status = "okay";
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
 /* phy1 - M.KEY socket */
 &pcie2x1l0 {
 	reset-gpios = <&gpio4 RK_PA5 GPIO_ACTIVE_HIGH>;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
--- linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2024-07-05 16:48:50.617640029 +0200
+++ linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2024-07-05 16:45:52.057643876 +0200
@@ -39,6 +39,16 @@
 		pinctrl-0 = <&hp_detect>;
 	};
 
+	hdmi0_sound: hdmi0-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi0";
+		rockchip,cpu = <&i2s5_8ch>;
+		rockchip,codec = <&hdmi0>;
+		rockchip,jack-det;
+	};
+
 	leds {
 		compatible = "gpio-leds";
 		pinctrl-names = "default";
@@ -354,6 +364,10 @@
 	};
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
 &package_thermal {
 	polling-delay = <1000>;
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts
--- linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2024-07-05 16:48:50.617640029 +0200
+++ linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2024-07-05 16:46:18.547643305 +0200
@@ -86,6 +86,16 @@
 		};
 	};
 
+	hdmi0_sound: hdmi0-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi0";
+		rockchip,cpu = <&i2s5_8ch>;
+		rockchip,codec = <&hdmi0>;
+		rockchip,jack-det;
+	};
+
 	vbus_typec: vbus-typec-regulator {
 		compatible = "regulator-fixed";
 		enable-active-high;
@@ -349,6 +359,10 @@
 	status = "okay";
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
 &mdio1 {
 	rgmii_phy1: ethernet-phy@1 {
 		compatible = "ethernet-phy-ieee802.3-c22";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
--- linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2024-07-05 16:48:50.617640029 +0200
+++ linux-6.10-rc6/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2024-07-05 16:46:39.654309517 +0200
@@ -32,6 +32,16 @@
 		dais = <&i2s0_8ch_p0>;
 	};
 
+	hdmi0_sound: hdmi0-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi0";
+		rockchip,cpu = <&i2s5_8ch>;
+		rockchip,codec = <&hdmi0>;
+		rockchip,jack-det;
+	};
+
 	chosen {
 		stdout-path = "serial2:1500000n8";
 	};
@@ -274,6 +284,10 @@
 	};
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
 &gmac1 {
 	clock_in_out = "output";
 	phy-handle = <&rgmii_phy1>;
