diff --git a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
index 81377dc4b583..8d34c616e672 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
@@ -281,6 +281,11 @@ &gmac1_rgmii_clk
 	status = "okay";
 };
 
+&gpu {
+	mali-supply = <&vdd_gpu_s0>;
+	status = "okay";
+};
+
 &mdio1 {
 	rgmii_phy1: ethernet-phy@1 {
 		/* RTL8211F */
@@ -433,6 +438,7 @@ rk806_dvs3_null: dvs3-null-pins {
 		regulators {
 			vdd_gpu_s0: vdd_gpu_mem_s0: dcdc-reg1 {
 				regulator-name = "vdd_gpu_s0";
+				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <550000>;
 				regulator-max-microvolt = <950000>;
-- 
Armbian

diff --speed-large-files --no-dereference --minimal -Naur linux-6.9.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts linux-6.9.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
--- linux-6.9.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-05-31 17:31:22.482893334 +0200
+++ linux-6.9.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-05-31 17:25:44.336233951 +0200
@@ -277,6 +277,11 @@
 	mem-supply = <&vdd_cpu_lit_mem_s0>;
 };
 
+&gpu {
+	mali-supply = <&vdd_gpu_s0>;
+	status = "okay";
+};
+
 &i2c0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c0m2_xfer>;
@@ -600,6 +605,7 @@
 		regulators {
 			vdd_gpu_s0: dcdc-reg1 {
 				regulator-name = "vdd_gpu_s0";
+				regulator-always-on;
 				regulator-boot-on;
 				regulator-enable-ramp-delay = <400>;
 				regulator-min-microvolt = <550000>;
