diff -Naur iguanaIR-1.0.3-old/autoconf/ac_python_devel.m4 iguanaIR-1.0.3-new/autoconf/ac_python_devel.m4
--- iguanaIR-1.0.3-old/autoconf/ac_python_devel.m4	2012-09-01 09:04:31.000000000 -0700
+++ iguanaIR-1.0.3-new/autoconf/ac_python_devel.m4	2012-09-01 09:06:56.000000000 -0700
@@ -175,7 +175,7 @@
 		py_version=`$PYTHON -c "from distutils.sysconfig import *; \
 			from string import join; \
 			print join(get_config_vars('VERSION'))"`
-		if test "$py_version" == "[None]"; then
+		if test "$py_version" = "[None]"; then
 			if test -n "$PYTHON_VERSION"; then
 				py_version=$PYTHON_VERSION
 			else
diff -Naur iguanaIR-1.0.3-old/autoconf/configure.ac iguanaIR-1.0.3-new/autoconf/configure.ac
--- iguanaIR-1.0.3-old/autoconf/configure.ac	2012-09-01 09:04:31.000000000 -0700
+++ iguanaIR-1.0.3-new/autoconf/configure.ac	2012-09-01 09:06:56.000000000 -0700
@@ -124,7 +124,7 @@
     else
       AC_ARG_ENABLE(no-threads,
           [  --disable-no-threads enable the "no threads" option ],
-          [ if test "x$enableval" == "xno"; then
+          [ if test "x$enableval" = "xno"; then
                 LIBUSB_THREADS_OPTIONAL=no
             fi
           ]
@@ -156,7 +156,7 @@
 # Allow the user to override python detection
 AC_ARG_ENABLE(python,
   [  --disable-python do build the Python interface.],
-  [ if test x$enableval == xno; then
+  [ if test "x$enableval" = "xno"; then
         USE_PYTHON=no
     fi ],
 )
