diff -Naur iguanaIR-1.0.3-old/autoconf/configure.ac iguanaIR-1.0.3-new/autoconf/configure.ac
--- iguanaIR-1.0.3-old/autoconf/configure.ac	2012-09-01 09:06:56.000000000 -0700
+++ iguanaIR-1.0.3-new/autoconf/configure.ac	2012-09-01 09:07:24.000000000 -0700
@@ -59,7 +59,8 @@
 AC_CHECK_FUNCS([floor gettimeofday memset mkdir select socket strchr strdup strerror strrchr strspn strstr mach_absolute_time])
 
 # Some default environment variable values
-CFLAGS="-Wall -pedantic"
+AX_CFLAGS_GCC_OPTION(-Wall)
+AX_CFLAGS_GCC_OPTION(-pedantic)
 
 # use -Wextra when possible
 AX_CFLAGS_GCC_OPTION(-Wextra)
@@ -77,8 +78,6 @@
 if test x$ENABLE_DEBUG_FLAGS = xyes; then
   CPPFLAGS="$CPPFLAGS -D_DEBUG"
   CFLAGS="$CFLAGS -g"
-else
-  CFLAGS="$CFLAGS -O2"
 fi
 
 # Allow the user to override clock_gettime detection
diff -Naur iguanaIR-1.0.3-old/Makefile.in iguanaIR-1.0.3-new/Makefile.in
--- iguanaIR-1.0.3-old/Makefile.in	2012-09-01 09:06:38.000000000 -0700
+++ iguanaIR-1.0.3-new/Makefile.in	2012-09-01 09:07:24.000000000 -0700
@@ -20,7 +20,7 @@
 igclient igdaemon: $(LIBNAME)
 
 $(LIBNAME): iguanaIR.o support.o dataPackets.o compat-unix.o pipes.o
-	$(CC) $(SHAREARG) $(LINKARGS) $(LDFLAGS) $^ -o $@
+	$(CC) $(SHAREARG) $(LINKARGS) $(CFLAGS) $(LDFLAGS) $^ -o $@
 
 drivers:
 	$(MAKE) -C $(srcdir)/drivers prefix=$(prefix)
@@ -33,7 +33,7 @@
               $(LDFLAGS) $^ -lpopt -lpthread -ldl $(LIBNAME) -o $@
 
 igclient: client.o list.o support.o compat-unix.o
-	$(CC) $(LDFLAGS) $^ -lpopt $(LIBNAME) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lpopt $(LIBNAME) -o $@
 
 %.o: %.c Makefile iguanaIR.h
 	$(CC) -c $(CPPFLAGS) $(CFLAGS) $< -o $@
@@ -76,7 +76,7 @@
 	$(CC) -c $(CPPFLAGS) @PYTHON_CPPFLAGS@ $(CFLAGS) -Wno-missing-field-initializers -Wno-unused -Wno-long-long -Wno-strict-aliasing $< -o $@
 
 _iguanaIR.$(SHAREEXT): iguanaIR_wrap.o
-	$(CC) $(LDFLAGS) @PYTHON_LDFLAGS@ $(SHAREARG) $^ $(LIBNAME) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) @PYTHON_LDFLAGS@ $(SHAREARG) $^ $(LIBNAME) -o $@
 endif
 
 # change ownership unless we're building an rpm
