diff -Naur iguanaIR-1.0.3-old/Makefile.in iguanaIR-1.0.3-new/Makefile.in
--- iguanaIR-1.0.3-old/Makefile.in	2012-09-01 09:08:05.000000000 -0700
+++ iguanaIR-1.0.3-new/Makefile.in	2012-09-05 15:45:15.000000000 -0700
@@ -30,16 +30,16 @@
           list.o protocol-versions.o pipes.o dataPackets.o compat-unix.o \
           support.c $(OSSUPPORT)
 	$(CC) $(EXPORTARGS) $(CPPFLAGS) $(CFLAGS) -DSUPPORT_EXPORTS \
-              $(LDFLAGS) $^ -lpopt -lpthread -ldl $(LIBNAME) -o $@
+              $(LDFLAGS) -flto -fwhole-program $^ -lpopt -lpthread -ldl $(LIBNAME) -o $@
 
 igclient: client.o list.o support.o compat-unix.o
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -lpopt $(LIBNAME) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) -flto -fwhole-program $^ -lpopt $(LIBNAME) -o $@
 
 %.o: %.c Makefile iguanaIR.h
-	$(CC) -c $(CPPFLAGS) $(CFLAGS) $< -o $@
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) -flto $< -o $@
 
 driver.o: driver.c Makefile iguanaIR.h
-	$(CC) -c $(CPPFLAGS) $(CFLAGS) -Wno-strict-aliasing $< -o $@
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) -flto -Wno-strict-aliasing $< -o $@
 
 # need a special dependency for config.h
 compat-unix.o: compat-unix.c config.h
