diff -Naur linux-5.9.6-old/drivers/regulator/core.c linux-5.9.6-old-new/drivers/regulator/core.c
--- linux-5.9.6-old/drivers/regulator/core.c	2020-11-05 17:50:14.000000000 +0100
+++ linux-5.9.6-old-new/drivers/regulator/core.c	2020-11-06 12:40:33.166666659 +0100
@@ -1134,6 +1134,9 @@
 		}
 
 		if (current_uV < 0) {
+			if (current_uV == -EINVAL && rdev->supply_name)
+				return -EPROBE_DEFER;
+
 			rdev_err(rdev,
 				 "failed to get the current voltage(%d)\n",
 				 current_uV);
