diff -Naur linux-5.9.1-old/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi linux-5.9.1-new/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi
--- linux-5.9.1-old/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi	2020-11-09 10:48:20.439999964 +0100
+++ linux-5.9.1-new/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi	2020-11-09 10:42:40.573333332 +0100
@@ -224,8 +224,8 @@
 				      "timing-adjustment";
 			rx-fifo-depth = <4096>;
 			tx-fifo-depth = <2048>;
-			resets = <&reset RESET_ETHERNET>;
-			reset-names = "stmmaceth";
+//			resets = <&reset RESET_ETHERNET>;
+//			reset-names = "stmmaceth";
 			status = "disabled";
 
 			mdio0: mdio {
