diff -Naur linux-5.10-rc3-old/drivers/gpu/drm/v3d/Kconfig linux-5.10-rc3-new/drivers/gpu/drm/v3d/Kconfig
--- linux-5.10-rc3-old/drivers/gpu/drm/v3d/Kconfig	2020-11-09 01:10:16.000000000 +0100
+++ linux-5.10-rc3-new/drivers/gpu/drm/v3d/Kconfig	2020-11-13 17:33:35.733333332 +0100
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0-only
 config DRM_V3D
 	tristate "Broadcom V3D 3.x and newer"
-	depends on ARCH_BCM || ARCH_BCMSTB || COMPILE_TEST
+	depends on ARCH_BCM || ARCH_BRCMSTB || COMPILE_TEST
 	depends on DRM
 	depends on COMMON_CLK
 	depends on MMU
