From 7aadebe8cd08e244d333662b3f2da36f68739399 Mon Sep 17 00:00:00 2001
From: Jorge Ramirez-Ortiz <jramirez@baylibre.com>
Date: Mon, 30 Apr 2018 23:48:11 +0200
Subject: [PATCH 08/13] libavcodec: v4l2m2m: DRM: tidy up code

Signed-off-by: Jorge Ramirez-Ortiz <jramirez@baylibre.com>
---
 libavcodec/v4l2_context.c | 12 +++---------
 libavcodec/v4l2_m2m.c     |  4 +++-
 libavcodec/v4l2_m2m_dec.c |  6 ++++++
 3 files changed, 12 insertions(+), 10 deletions(-)

diff --git a/libavcodec/v4l2_context.c b/libavcodec/v4l2_context.c
index 641514a..ae90f0d 100644
--- a/libavcodec/v4l2_context.c
+++ b/libavcodec/v4l2_context.c
@@ -403,16 +403,10 @@ static int v4l2_release_buffers(V4L2Context* ctx)
         for (j = 0; j < buffer->num_planes; j++) {
             struct V4L2Plane_info *p = &buffer->plane_info[j];
 
-            if (V4L2_TYPE_IS_MULTIPLANAR(buffer->buf.type)) {
-                if (buffer->drm_frame.objects[i].fd >= 0) {
+            if (ctx_to_m2mctx(ctx)->output_drm) {
+                /* use the DRM frame to close */
+                if (buffer->drm_frame.objects[i].fd >= 0)
                     close(buffer->drm_frame.objects[i].fd);
-                    buffer->drm_frame.objects[i].fd = -1;
-                }
-            } else {
-                if (buffer->drm_frame.objects[0].fd >= 0) {
-                    close(buffer->drm_frame.objects[0].fd);
-                    buffer->drm_frame.objects[0].fd = -1;
-                }
             }
 
             if (p->mm_addr && p->length)
diff --git a/libavcodec/v4l2_m2m.c b/libavcodec/v4l2_m2m.c
index 427e165..7896326 100644
--- a/libavcodec/v4l2_m2m.c
+++ b/libavcodec/v4l2_m2m.c
@@ -159,7 +159,9 @@ static int v4l2_configure_contexts(V4L2m2mContext* s)
         goto error;
     }
 
-    /* decoder's buffers need to be updated at a later stage */
+    /* decoder's capture buffers are updated during v4l2_try_start once we find
+     * the valid format.
+     */
     if (!av_codec_is_decoder(s->avctx->codec)) {
         ret = ff_v4l2_context_init(&s->capture);
         if (ret) {
diff --git a/libavcodec/v4l2_m2m_dec.c b/libavcodec/v4l2_m2m_dec.c
index 8b4ada8..cd5569e 100644
--- a/libavcodec/v4l2_m2m_dec.c
+++ b/libavcodec/v4l2_m2m_dec.c
@@ -184,6 +184,12 @@ static av_cold int v4l2_decode_init(AVCodecContext *avctx)
     capture->av_codec_id = AV_CODEC_ID_RAWVIDEO;
     capture->av_pix_fmt = avctx->pix_fmt;
 
+    /* the client requests the codec to generate DRM frames:
+     *   - data[0] will therefore point to the returned AVDRMFrameDescriptor
+     *       check the ff_v4l2_buffer_to_avframe conversion function.
+     *   - the DRM frame format is passed in the DRM frame descriptor layer.
+     *       check the v4l2_get_drm_frame function.
+     */
     if (avctx->pix_fmt == AV_PIX_FMT_DRM_PRIME)
         s->output_drm = 1;
 
-- 
2.7.1

