From 0e9e33fd9986cab5a2dc7508c253b960a713a4d6 Mon Sep 17 00:00:00 2001
From: Jorge Ramirez-Ortiz <jorge.ramirez-ortiz@linaro.org>
Date: Tue, 1 May 2018 11:07:47 +0000
Subject: [PATCH 10/13] libavcodec: v4l2m2m: dec: declare HW config

Signed-off-by: Jorge Ramirez-Ortiz <jorge.ramirez-ortiz@linaro.org>
---
 libavcodec/v4l2_m2m_dec.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/libavcodec/v4l2_m2m_dec.c b/libavcodec/v4l2_m2m_dec.c
index cd5569e..2b33bad 100644
--- a/libavcodec/v4l2_m2m_dec.c
+++ b/libavcodec/v4l2_m2m_dec.c
@@ -23,6 +23,8 @@
 
 #include <linux/videodev2.h>
 #include <sys/ioctl.h>
+
+#include "libavutil/hwcontext.h"
 #include "libavutil/hwcontext_drm.h"
 #include "libavutil/pixfmt.h"
 #include "libavutil/pixdesc.h"
@@ -30,6 +32,9 @@
 #include "libavcodec/avcodec.h"
 #include "libavcodec/decode.h"
 
+#include "libavcodec/hwaccel.h"
+#include "libavcodec/internal.h"
+
 #include "v4l2_context.h"
 #include "v4l2_m2m.h"
 #include "v4l2_fmt.h"
@@ -212,6 +217,11 @@ static const AVOption options[] = {
     { NULL},
 };
 
+static const AVCodecHWConfigInternal *v4l2_m2m_hw_configs[] = {
+    HW_CONFIG_INTERNAL(DRM_PRIME),
+    NULL
+};
+
 #define M2MDEC(NAME, LONGNAME, CODEC, bsf_name) \
 static const AVClass v4l2_m2m_ ## NAME ## _dec_class = {\
     .class_name = #NAME "_v4l2_m2m_decoder",\
@@ -235,6 +245,7 @@ static const AVOption options[] = {
     .pix_fmts       = (const enum AVPixelFormat[]) { AV_PIX_FMT_DRM_PRIME, \
                                                      AV_PIX_FMT_NONE}, \
     .bsfs           = bsf_name, \
+    .hw_configs     = v4l2_m2m_hw_configs, \
     .capabilities   = AV_CODEC_CAP_HARDWARE | AV_CODEC_CAP_DELAY | \
                       AV_CODEC_CAP_AVOID_PROBING, \
     .wrapper_name   = "v4l2m2m", \
-- 
2.7.1

