diff -Naur ffmpeg-old/configure ffmpeg-new/configure
--- ffmpeg-old/configure	2020-07-11 00:26:17.000000000 +0200
+++ ffmpeg-new/configure	2020-09-23 12:45:30.833333334 +0200
@@ -6903,7 +6931,6 @@
 elif enabled gcc; then
     check_optflags -fno-tree-vectorize
     check_cflags -Werror=format-security
-    check_cflags -Werror=implicit-function-declaration
     check_cflags -Werror=missing-prototypes
     check_cflags -Werror=return-type
     check_cflags -Werror=vla
@@ -6944,7 +6971,6 @@
         check_cflags -mstack-alignment=16
     fi
     check_cflags -Qunused-arguments
-    check_cflags -Werror=implicit-function-declaration
     check_cflags -Werror=missing-prototypes
     check_cflags -Werror=return-type
 elif enabled cparser; then
