diff -Naur ffmpeg-1.1.1-old/libavutil/common.h ffmpeg-1.1.1-new/libavutil/common.h
--- ffmpeg-1.1.1-old/libavutil/common.h	2013-01-21 19:25:12.000000000 -0800
+++ ffmpeg-1.1.1-new/libavutil/common.h	2013-01-21 19:26:14.000000000 -0800
@@ -26,6 +26,11 @@
 #ifndef AVUTIL_COMMON_H
 #define AVUTIL_COMMON_H
 
+#ifndef __STDC_CONSTANT_MACROS
+#  define __STDC_CONSTANT_MACROS
+#endif
+#include <stdint.h>
+
 #include <ctype.h>
 #include <errno.h>
 #include <inttypes.h>
