diff -Naur ffmpeg-2.1-old/libavutil/common.h ffmpeg-2.1-new/libavutil/common.h
--- ffmpeg-2.1-old/libavutil/common.h	2013-10-28 01:58:06.000000000 +0100
+++ ffmpeg-2.1-new/libavutil/common.h	2013-11-05 15:24:13.805317732 +0100
@@ -26,6 +26,11 @@
 #ifndef AVUTIL_COMMON_H
 #define AVUTIL_COMMON_H
 
+#ifndef __STDC_CONSTANT_MACROS
+#  define __STDC_CONSTANT_MACROS
+#endif
+#include <stdint.h>
+
 #include <errno.h>
 #include <inttypes.h>
 #include <limits.h>
