diff -Naur linux-5.2-rc5-old/drivers/gpu/drm/panfrost/panfrost_gpu.c linux-5.2-rc50-new/drivers/gpu/drm/panfrost/panfrost_gpu.c
--- linux-5.2-rc5-old/drivers/gpu/drm/panfrost/panfrost_gpu.c	2019-06-16 20:49:45.000000000 +0200
+++ linux-5.2-rc50-new/drivers/gpu/drm/panfrost/panfrost_gpu.c	2019-06-22 15:03:21.989999993 +0200
@@ -7,7 +7,9 @@
 #include <linux/delay.h>
 #include <linux/dma-mapping.h>
 #include <linux/interrupt.h>
+#include <linux/reset.h>
 #include <linux/io.h>
+#include <linux/of.h>
 #include <linux/iopoll.h>
 #include <linux/platform_device.h>
 
@@ -55,7 +57,16 @@
 
 	gpu_write(pfdev, GPU_INT_MASK, 0);
 	gpu_write(pfdev, GPU_INT_CLEAR, GPU_IRQ_RESET_COMPLETED);
-	gpu_write(pfdev, GPU_CMD, GPU_CMD_SOFT_RESET);
+
+	if (of_device_is_compatible(pfdev->dev->of_node, "amlogic,meson-gxm-mali")) {
+		reset_control_assert(pfdev->rstc);
+		udelay(10);
+		reset_control_deassert(pfdev->rstc);
+
+		gpu_write(pfdev, GPU_PWR_KEY, 0x2968A819);
+		gpu_write(pfdev, GPU_PWR_OVERRIDE1, 0xfff | (0x20 << 16));
+	} else
+		gpu_write(pfdev, GPU_CMD, GPU_CMD_SOFT_RESET);
 
 	ret = readl_relaxed_poll_timeout(pfdev->iomem + GPU_INT_RAWSTAT,
 		val, val & GPU_IRQ_RESET_COMPLETED, 100, 10000);
diff -Naur linux-5.2-rc5-old/drivers/gpu/drm/panfrost/panfrost_regs.h linux-5.2-rc50-new/drivers/gpu/drm/panfrost/panfrost_regs.h
--- linux-5.2-rc5-old/drivers/gpu/drm/panfrost/panfrost_regs.h	2019-06-16 20:49:45.000000000 +0200
+++ linux-5.2-rc50-new/drivers/gpu/drm/panfrost/panfrost_regs.h	2019-06-22 15:05:14.929999994 +0200
@@ -46,6 +46,9 @@
 #define   GPU_CMD_SOFT_RESET		0x01
 #define GPU_STATUS			0x34
 #define GPU_LATEST_FLUSH_ID		0x38
+#define GPU_PWR_KEY			0x050	/* (WO) Power manager key register */
+#define GPU_PWR_OVERRIDE0		0x054	/* (RW) Power manager override settings */
+#define GPU_PWR_OVERRIDE1		0x058	/* (RW) Power manager override settings */
 #define GPU_FAULT_STATUS		0x3C
 #define GPU_FAULT_ADDRESS_LO		0x40
 #define GPU_FAULT_ADDRESS_HI		0x44
