diff -Naur make-3.82-old/configure make-3.82-new/configure
--- make-3.82-old/configure	2010-07-28 07:41:51.000000000 +0200
+++ make-3.82-new/configure	2018-12-27 16:45:59.819999993 +0100
@@ -9130,7 +9130,7 @@
 #define GLOB_INTERFACE_VERSION 1
 #if !defined _LIBC && defined __GNU_LIBRARY__ && __GNU_LIBRARY__ > 1
 # include <gnu-versions.h>
-# if _GNU_GLOB_INTERFACE_VERSION == GLOB_INTERFACE_VERSION
+# if _GNU_GLOB_INTERFACE_VERSION >= GLOB_INTERFACE_VERSION
    gnu glob
 # endif
 #endif
