diff --speed-large-files --no-dereference --minimal -Naur mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythplugins/cmake/MythPluginsFindPackages.cmake mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythplugins/cmake/MythPluginsFindPackages.cmake
--- mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythplugins/cmake/MythPluginsFindPackages.cmake	2025-03-28 21:06:21.000000000 +0100
+++ mythtv-8f72065aaa27c414ef04f0d93bf6f44c5005efff/mythplugins/cmake/MythPluginsFindPackages.cmake	2025-04-01 13:09:44.453411643 +0200
@@ -57,21 +57,26 @@
   Python3 3.6
   COMPONENTS Interpreter
   REQUIRED)
-if(TARGET Python3::Interpreter)
-  find_package(Python3Modules OPTIONAL_COMPONENTS MythTV lxml oauth pycurl
-                                                  urllib xml)
-endif()
 
 if(NOT TARGET Python3::Interpreter)
   message(STATUS "Missing python interpreter. Disabling netvision.")
-elseif(Python3Modules_COMPONENTS_MISSING)
-  message(STATUS "Missing some python modules. Disabling netvision.")
+elseif(NOT CMAKE_CROSSCOMPILING)
+  if(TARGET Python3::Interpreter)
+    find_package(Python3Modules OPTIONAL_COMPONENTS MythTV lxml oauth pycurl
+                                                  urllib xml)
+  endif()
+  if(Python3Modules_COMPONENTS_MISSING)
+    message(STATUS "Missing some python modules. Disabling netvision.")
+  endif()
 else()
   set(ENABLE_NETVISION TRUE)
 endif()
 
 find_package(Perl)
-if(PERL_FOUND)
+
+if(NOT PERL_FOUND)
+  message(STATUS "Missing perl interpreter. Disabling mythweather.")
+elseif(NOT CMAKE_CROSSCOMPILING)
   find_package(
     PerlModules
     OPTIONAL_COMPONENTS
@@ -82,11 +87,9 @@
       SOAP::Lite
       XML::Simple
       XML::XPath)
-endif()
-if(NOT PERL_FOUND)
-  message(STATUS "Missing perl interpreter. Disabling mythweather.")
-elseif(PerlModules_COMPONENTS_MISSING)
-  message(STATUS "Missing some perl modules. Disabling mythweather.")
+  if(PerlModules_COMPONENTS_MISSING)
+    message(STATUS "Missing some perl modules. Disabling mythweather.")
+  endif()
 else()
   set(ENABLE_WEATHER TRUE)
 endif()
