diff --speed-large-files --no-dereference --minimal -Naur mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/distros/mythtv_data/data_mythtv.py mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/distros/mythtv_data/data_mythtv.py
--- mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/distros/mythtv_data/data_mythtv.py	2022-12-04 02:18:33.000000000 +0100
+++ mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/distros/mythtv_data/data_mythtv.py	2022-12-12 20:14:48.073326053 +0100
@@ -333,7 +333,7 @@
                     cursor.execute("""SELECT DISTINCT(profileid)
                                     FROM displayprofiles
                                     WHERE profilegroupid=%s""",
-                                    profilegroupid)
+                                    (profilegroupid,))
                     for profileid in cursor.fetchall():
                         yield cls(profilegroupid, profileid[0], db)
 
@@ -349,7 +349,6 @@
             d.deint_pri = profile.pref_deint0
             d.deint_sec = profile.pref_deint1
             d.renderer  = profile.pref_videorenderer
-            d.filters   = profile.pref_filters
             data.profiles.append(d)
 
         return {'playbackprofile':data}
@@ -487,8 +486,9 @@
     def get_data(self,gate):
         self._data = OrdDict()
         for func in (self.ProcessVersion,
-                     self.ProcessPrograms,
-                     self.ProcessHistorical,
+                     # 2 below are commented-out as causing long CPU hog on small systems
+                     #self.ProcessPrograms,
+                     #self.ProcessHistorical,
                      self.ProcessSource,
                      self.ProcessTimeZone,
                      self.ProcessStorage,
diff --speed-large-files --no-dereference --minimal -Naur mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/os_detect.py mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/os_detect.py
--- mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/os_detect.py	2022-12-04 02:18:33.000000000 +0100
+++ mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/os_detect.py	2022-12-12 20:05:39.471537509 +0100
@@ -246,6 +246,7 @@
     solaris     = OSWithFile('/etc/release')
     sunjds      = OSWithFile('/etc/sun-release')
     pldlinux    = OSWithFile('/etc/pld-release')
+    minimyth2   = OSWithFile('/etc/minimyth2-release')
 
     @OSWithFile('/etc/SuSE-release')
     def suselinux(self, text):
diff --speed-large-files --no-dereference --minimal -Naur mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/smolt.py mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/smolt.py
--- mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/smolt.py	2022-12-04 02:18:33.000000000 +0100
+++ mythtv-7ebfa0f03c6ae25496150f807251ccf97f29b038/mythtv/programs/scripts/hardwareprofile/smolt.py	2022-12-12 20:01:31.175846307 +0100
@@ -264,7 +264,10 @@
             try:
                 self.systemVendor = cat('/sys/devices/virtual/dmi/id/sys_vendor')[0].strip()
             except:
-                self.systemVendor = 'Unknown'
+                try:
+                    self.systemVendor = cat('/sys/firmware/devicetree/base/model')[0].strip()
+                except:
+                    self.systemVendor = 'Unknown'
         else:
             self.systemVendor = WITHHELD_MAGIC_STRING
 
@@ -272,7 +275,10 @@
             try:
                 self.systemModel = cat('/sys/devices/virtual/dmi/id/product_name')[0].strip() + ' ' + cat('/sys/devices/virtual/dmi/id/product_version')[0].strip()
             except:
-                self.systemModel = 'Unknown'
+                try:
+                    self.systemModel = cat('/sys/firmware/devicetree/base/model')[0].strip()
+                except:
+                    self.systemModel = 'Unknown'
             #hostInfo was removed with the hal restructure
             #if not self.systemModel:
                 #self.systemModel = hostInfo.get('system.hardware.product')
@@ -288,7 +294,7 @@
                 formfactor_id = int(cat('/sys/devices/virtual/dmi/id/chassis_type')[0].strip())
                 self.formfactor = FORMFACTOR_LIST[formfactor_id]
             except:
-                self.formfactor = 'Unknown'
+                self.formfactor = "Lunch Box"
         else:
             self.formfactor = WITHHELD_MAGIC_STRING
 
@@ -976,7 +982,6 @@
 
     cpulist = open("/proc/cpuinfo", "r").read()
     uname = os.uname()[4].lower()
-
     # This thing should return a hwdict that has the following
     # members:
     #
@@ -1175,6 +1180,40 @@
         hwdict['speed'] = speed
         hwdict['system'] = system
 
+    elif uname in ["aarch64"]:
+        tmpdict = {}
+        count = 0
+        for cpu in cpulist.split("processor"):
+            if not len(cpu):
+                continue
+            count = count + 1
+            # count the rest
+            if count > 1:
+                continue
+            for cpu_attr in cpu.split("\n"):
+                if not len(cpu_attr):
+                    continue
+                vals = cpu_attr.split(":")
+                if len(vals) != 2:
+                    # XXX: make at least some effort to recover this data...
+                    continue
+                name, value = vals[0].strip(), vals[1].strip()
+                tmpdict[name.lower()] = value.lower()
+
+        system = ''
+        hwdict['platform']      = uname
+        hwdict['count']         = count
+        hwdict['type']          = get_entry(tmpdict, 'CPU implementer')
+        hwdict['model']         = get_entry(tmpdict, 'CPU architecture')
+        hwdict['cpu_family']    = get_entry(tmpdict, 'CPU part')
+        hwdict['model_ver']     = get_entry(tmpdict, 'CPU variant')
+        hwdict['model_rev']     = get_entry(tmpdict, 'CPU revision')
+        hwdict['model_number']  = get_entry(tmpdict, 'CPU part')
+        hwdict['bogomips']      = get_entry(tmpdict, 'BogoMIPS')
+        hwdict['speed']         = -1
+        hwdict['other']         = get_entry(tmpdict, 'Features')
+
+
     else:
         # XXX: expand me. Be nice to others
         hwdict['platform']      = uname
@@ -1219,6 +1258,8 @@
 def read_memory():
     un = os.uname()
     kernel = un[2]
+    if kernel[:2] == "6.":
+        return read_memory_2_6()
     if kernel[:2] == "5.":
         return read_memory_2_6()
     if kernel[:2] == "4.":
