diff -Naur linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi
--- linux-5.3-rc7-old/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi	2019-09-06 16:53:47.710000007 +0200
+++ linux-5.3-rc7-new/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi	2019-09-06 15:56:32.883333338 +0200
@@ -36,6 +37,7 @@
 			device_type = "cpu";
 			reg = <0>;
 			enable-method = "psci";
+			#cooling-cells = <2>;
 		};
 
 		cpu1: cpu@1 {
@@ -43,6 +45,7 @@
 			device_type = "cpu";
 			reg = <1>;
 			enable-method = "psci";
+			#cooling-cells = <2>;
 		};
 
 		cpu2: cpu@2 {
@@ -50,6 +53,7 @@
 			device_type = "cpu";
 			reg = <2>;
 			enable-method = "psci";
+			#cooling-cells = <2>;
 		};
 
 		cpu3: cpu@3 {
@@ -57,6 +61,7 @@
 			device_type = "cpu";
 			reg = <3>;
 			enable-method = "psci";
+			#cooling-cells = <2>;
 		};
 	};
 
