diff --speed-large-files --no-dereference --minimal -Naur u-boot-19cefea3b8a7e22fa29b39367b04a8ab90c5c765/drivers/mmc/rockchip_dw_mmc.c u-boot-19cefea3b8a7e22fa29b39367b04a8ab90c5c765/drivers/mmc/rockchip_dw_mmc.c
--- u-boot-19cefea3b8a7e22fa29b39367b04a8ab90c5c765/drivers/mmc/rockchip_dw_mmc.c	2022-04-21 10:31:50.000000000 +0200
+++ u-boot-19cefea3b8a7e22fa29b39367b04a8ab90c5c765/drivers/mmc/rockchip_dw_mmc.c	2022-06-15 09:47:33.305140782 +0200
@@ -42,7 +42,7 @@
 {
 	struct rockchip_dwmmc_priv *priv = dev_get_priv(dev);
 
-	if (!priv || !&priv->clk)
+	if (!priv)
 		return 0;
 
 	if (!memcmp(dev->name, "dwmmc", strlen("dwmmc")))
