diff --speed-large-files --no-dereference --minimal -Naur ntp-4.2.8p18/include/l_stdlib.h ntp-4.2.8p18/include/l_stdlib.h
--- ntp-4.2.8p18/include/l_stdlib.h	2020-06-23 11:17:30.000000000 +0200
+++ ntp-4.2.8p18/include/l_stdlib.h	2026-05-16 11:56:36.092373350 +0200
@@ -222,7 +222,7 @@
 #endif
 
 #ifndef HAVE_MEMCHR
-extern void *memchr(const void *s, int c, size_t n);
+extern void *(memchr)(const void *s, int c, size_t n);
 #endif
 
 #ifndef HAVE_STRNLEN
