diff --speed-large-files --no-dereference --minimal -Naur ntp-4.2.8p18/ntpd/ntp_control.c ntp-4.2.8p18/ntpd/ntp_control.c
--- ntp-4.2.8p18/ntpd/ntp_control.c	2024-05-07 13:21:15.000000000 +0200
+++ ntp-4.2.8p18/ntpd/ntp_control.c	2026-05-16 12:26:11.983307410 +0200
@@ -3664,16 +3664,16 @@
 	}
 
 	MD5Init(&ctx);
-	MD5Update(&ctx, salt, sizeof(salt));
-	MD5Update(&ctx, &ts_i, sizeof(ts_i));
-	MD5Update(&ctx, &ts_f, sizeof(ts_f));
+	MD5Update(&ctx, (const unsigned char *)salt, sizeof(salt));
+	MD5Update(&ctx, (const unsigned char *)&ts_i, sizeof(ts_i));
+	MD5Update(&ctx, (const unsigned char *)&ts_f, sizeof(ts_f));
 	if (IS_IPV4(addr)) {
-		MD5Update(&ctx, &SOCK_ADDR4(addr), sizeof(SOCK_ADDR4(addr)));
+		MD5Update(&ctx, (const unsigned char *)&SOCK_ADDR4(addr), sizeof(SOCK_ADDR4(addr)));
 	} else {
-		MD5Update(&ctx, &SOCK_ADDR6(addr), sizeof(SOCK_ADDR6(addr)));
+		MD5Update(&ctx, (const unsigned char *)&SOCK_ADDR6(addr), sizeof(SOCK_ADDR6(addr)));
 	}
-	MD5Update(&ctx, &NSRCPORT(addr), sizeof(NSRCPORT(addr)));
-	MD5Update(&ctx, salt, sizeof(salt));
+	MD5Update(&ctx, (const unsigned char *)&NSRCPORT(addr), sizeof(NSRCPORT(addr)));
+	MD5Update(&ctx, (const unsigned char *)salt, sizeof(salt));
 	MD5Final(d.digest, &ctx);
 
 	return d.extract;
