diff -Naur ntp-4.2.8p15-old/libntp/work_thread.c ntp-4.2.8p15-new/libntp/work_thread.c
--- ntp-4.2.8p15-old/libntp/work_thread.c	2018-08-14 13:51:06.000000000 +0200
+++ ntp-4.2.8p15-new/libntp/work_thread.c	2021-08-04 10:32:59.196665089 +0200
@@ -41,7 +41,7 @@
 #ifndef THREAD_MINSTACKSIZE
 # define THREAD_MINSTACKSIZE	(64U * 1024)
 #endif
-#ifndef __sun
+#ifdef __sun
 #if defined(PTHREAD_STACK_MIN) && THREAD_MINSTACKSIZE < PTHREAD_STACK_MIN
 # undef THREAD_MINSTACKSIZE
 # define THREAD_MINSTACKSIZE PTHREAD_STACK_MIN
