diff -Naur ntp-4.2.6p5-old/ntpdc/Makefile.am ntp-4.2.6p5-new/ntpdc/Makefile.am
--- ntp-4.2.6p5-old/ntpdc/Makefile.am	2012-01-09 22:06:08.000000000 -0800
+++ ntp-4.2.6p5-new/ntpdc/Makefile.am	2012-01-09 22:07:04.000000000 -0800
@@ -55,7 +55,7 @@
 ntpdc-layout.o: nl.c
 
 layout.here: ntpdc-layout
-	./ntpdc-layout > $@
+	ntp-ntpdc-layout > $@
 
 check-layout: ntpdc-layout $(srcdir)/layout.std layout.here
 	cmp $(srcdir)/layout.std layout.here && echo stamp > $@
diff -Naur ntp-4.2.6p5-old/ntpdc/Makefile.in ntp-4.2.6p5-new/ntpdc/Makefile.in
--- ntp-4.2.6p5-old/ntpdc/Makefile.in	2012-01-09 22:06:08.000000000 -0800
+++ ntp-4.2.6p5-new/ntpdc/Makefile.in	2012-01-09 22:07:04.000000000 -0800
@@ -761,7 +761,7 @@
 ntpdc-layout.o: nl.c
 
 layout.here: ntpdc-layout
-	./ntpdc-layout > $@
+	ntp-ntpdc-layout > $@
 
 check-layout: ntpdc-layout $(srcdir)/layout.std layout.here
 	cmp $(srcdir)/layout.std layout.here && echo stamp > $@
