diff -Naur ntp-4.2.6p5-old/include/ntp_syscall.h ntp-4.2.6p5-new/include/ntp_syscall.h
--- ntp-4.2.6p5-old/include/ntp_syscall.h	2009-12-08 23:36:37.000000000 -0800
+++ ntp-4.2.6p5-new/include/ntp_syscall.h	2012-01-09 22:06:22.000000000 -0800
@@ -14,6 +14,14 @@
 # include <sys/timex.h>
 #endif
 
+#if defined(ADJ_NANO) && !defined(MOD_NANO)
+#define MOD_NANO ADJ_NANO
+#endif
+
+#if defined(ADJ_TAI) && !defined(MOD_TAI)
+#define MOD_TAI ADJ_TAI
+#endif
+
 #ifndef NTP_SYSCALLS_LIBC
 #ifdef NTP_SYSCALLS_STD
 # define ntp_adjtime(t)		syscall(SYS_ntp_adjtime, (t))
