diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328.dtsi linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328.dtsi
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328.dtsi	2023-05-11 13:12:04.399592008 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328.dtsi	2023-05-11 13:06:38.482935009 +0200
@@ -11,6 +11,7 @@
 #include <dt-bindings/power/rk3328-power.h>
 #include <dt-bindings/soc/rockchip,boot-mode.h>
 #include <dt-bindings/thermal/thermal.h>
+#include <dt-bindings/leds/common.h>
 
 / {
 	compatible = "rockchip,rk3328";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts	2023-05-11 13:12:04.439592007 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts	2023-05-11 13:06:38.482935009 +0200
@@ -181,6 +181,7 @@
 
 		user_led: led-1 {
 			label = "firefly:yellow:user";
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			linux,default-trigger = "mmc1";
 			gpios = <&rk805 0 GPIO_ACTIVE_LOW>;
 			default-state = "off";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts	2023-05-11 13:12:04.439592007 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts	2023-05-11 13:06:38.482935009 +0200
@@ -80,6 +80,7 @@
 
 		power_led: led-0 {
 			gpios = <&rk805 1 GPIO_ACTIVE_LOW>;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			linux,default-trigger = "mmc0";
 		};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts	2023-05-11 13:12:04.439592007 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts	2023-05-11 13:06:38.482935009 +0200
@@ -58,7 +58,8 @@
 		led-0 {
 			color = <LED_COLOR_ID_BLUE>;
 			gpios = <&gpio3 RK_PA5 GPIO_ACTIVE_LOW>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399.dtsi linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399.dtsi
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399.dtsi	2023-05-11 13:12:04.159592015 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399.dtsi	2023-05-11 13:06:38.482935009 +0200
@@ -10,6 +10,7 @@
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/power/rk3399-power.h>
 #include <dt-bindings/thermal/thermal.h>
+#include <dt-bindings/leds/common.h>
 
 / {
 	compatible = "rockchip,rk3399";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-eaidk-610.dts linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-eaidk-610.dts
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-eaidk-610.dts	2023-05-01 01:32:26.000000000 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-eaidk-610.dts	2023-05-11 13:06:38.482935009 +0200
@@ -105,7 +105,8 @@
 
 		user_led: led-1 {
 			label = "read:user";
-			default-state = "off";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 			gpios = <&gpio0 RK_PB4 GPIO_ACTIVE_HIGH>;
 		};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-ficus.dts linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-ficus.dts
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-ficus.dts	2023-05-01 01:32:26.000000000 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-ficus.dts	2023-05-11 13:06:38.482935009 +0200
@@ -40,6 +40,7 @@
 		user_led2: led-2 {
 			label = "red:user2";
 			gpios = <&gpio4 26 0>;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			linux,default-trigger = "mmc0";
 		};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-khadas-edge.dtsi linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-khadas-edge.dtsi
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-khadas-edge.dtsi	2023-05-01 01:32:26.000000000 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-khadas-edge.dtsi	2023-05-11 13:06:38.482935009 +0200
@@ -166,7 +166,8 @@
 
 		user_led: led-1 {
 			label = "user_led";
-			default-state = "off";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 			gpios = <&gpio4 RK_PD0 GPIO_ACTIVE_HIGH>;
 		};
 	};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-nanopi4.dtsi linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-nanopi4.dtsi
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-nanopi4.dtsi	2023-05-01 01:32:26.000000000 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-nanopi4.dtsi	2023-05-11 13:06:38.482935009 +0200
@@ -128,7 +128,8 @@
 		status_led: led-0 {
 			gpios = <&gpio0 RK_PB5 GPIO_ACTIVE_HIGH>;
 			label = "status_led";
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-puma.dtsi linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-puma.dtsi
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-puma.dtsi	2023-05-01 01:32:26.000000000 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-puma.dtsi	2023-05-11 13:06:38.482935009 +0200
@@ -20,7 +20,8 @@
 		module_led: led-0 {
 			label = "module_led";
 			gpios = <&gpio2 RK_PD1 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 			panic-indicator;
 		};
 	};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts	2023-05-01 01:32:26.000000000 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts	2023-05-11 13:08:30.756265003 +0200
@@ -15,8 +15,8 @@
 	compatible = "radxa,rock-4c-plus", "rockchip,rk3399";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
@@ -40,15 +40,15 @@
 			function = LED_FUNCTION_POWER;
 			color = <LED_COLOR_ID_GREEN>;
 			gpios = <&gpio3 RK_PD4 GPIO_ACTIVE_LOW>;
-			linux,default-trigger = "default-on";
+			linux,default-trigger = "heartbeat";
 		};
 
 		/* USER_LED2 */
 		led-1 {
-			function = LED_FUNCTION_STATUS;
 			color = <LED_COLOR_ID_BLUE>;
 			gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock960.dts linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock960.dts
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock960.dts	2023-05-01 01:32:26.000000000 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock960.dts	2023-05-11 13:06:38.482935009 +0200
@@ -30,6 +30,7 @@
 		user_led2: led-2 {
 			label = "green:user2";
 			gpios = <&gpio4 RK_PC6 0>;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			linux,default-trigger = "mmc0";
 		};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts	2023-05-11 13:12:04.516258672 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts	2023-05-11 13:10:07.289595479 +0200
@@ -12,6 +12,8 @@
 	compatible = "radxa,rockpi4b", "radxa,rockpi4", "rockchip,rk3399";
 
 	aliases {
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio0;
 	};
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts	2023-05-11 13:12:04.516258672 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts	2023-05-11 13:09:52.516262589 +0200
@@ -13,6 +13,8 @@
 	compatible = "radxa,rockpi4b-plus", "radxa,rockpi4", "rockchip,rk3399";
 
 	aliases {
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio0;
 	};
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts	2023-05-11 13:12:04.516258672 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts	2023-05-11 13:10:20.359595108 +0200
@@ -13,6 +13,8 @@
 	compatible = "radxa,rockpi4c", "radxa,rockpi4", "rockchip,rk3399";
 
 	aliases {
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio0;
 	};
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi	2023-05-01 01:32:26.000000000 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi	2023-05-11 13:08:00.106265926 +0200
@@ -13,8 +13,8 @@
 
 / {
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
@@ -35,10 +35,10 @@
 
 		/* USER_LED2 */
 		led-0 {
-			function = LED_FUNCTION_STATUS;
 			color = <LED_COLOR_ID_BLUE>;
 			gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 		};
 	};
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4se.dts linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4se.dts
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4se.dts	2023-05-11 13:12:04.476258672 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4se.dts	2023-05-11 13:10:37.502927934 +0200
@@ -12,6 +12,8 @@
 	compatible = "radxa,rockpi4se", "radxa,rockpi4", "rockchip,rk3399";
 
 	aliases {
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio0;
 	};
 };
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-roc-pc.dtsi linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-roc-pc.dtsi
--- linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-roc-pc.dtsi	2023-05-01 01:32:26.000000000 +0200
+++ linux-6.3.1/arch/arm64/boot/dts/rockchip/rk3399-roc-pc.dtsi	2023-05-11 13:06:38.482935009 +0200
@@ -92,7 +92,7 @@
 		yellow_led: led-2 {
 			label = "yellow:yellow-led";
 			gpios = <&gpio0 RK_PA2 GPIO_ACTIVE_HIGH>;
-			default-state = "off";
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			linux,default-trigger = "mmc1";
 		};
 	};
