diff --speed-large-files --no-dereference --minimal -Naur dtc-1.7.2/libfdt/fdt_overlay.c dtc-1.7.2/libfdt/fdt_overlay.c
--- dtc-1.7.2/libfdt/fdt_overlay.c	2024-11-06 03:01:37.000000000 +0000
+++ dtc-1.7.2/libfdt/fdt_overlay.c	2026-03-03 13:53:07.440811801 +0000
@@ -421,7 +421,7 @@
 		value += fixup_len + 1;
 
 		path = fixup_str;
-		sep = memchr(fixup_str, ':', fixup_len);
+		sep = (char *)(uintptr_t) memchr(fixup_str, ':', fixup_len);
 		if (!sep || *sep != ':')
 			return -FDT_ERR_BADOVERLAY;
 
@@ -431,7 +431,7 @@
 
 		fixup_len -= path_len + 1;
 		name = sep + 1;
-		sep = memchr(name, ':', fixup_len);
+		sep = (char *)(uintptr_t) memchr(name, ':', fixup_len);
 		if (!sep || *sep != ':')
 			return -FDT_ERR_BADOVERLAY;
 
diff --speed-large-files --no-dereference --minimal -Naur dtc-1.7.2/fdtput.c dtc-1.7.2/fdtput.c
--- dtc-1.7.2/fdtput.c	2024-11-06 03:01:37.000000000 +0000
+++ dtc-1.7.2/fdtput.c	2026-03-03 14:03:03.166268847 +0000
@@ -232,7 +232,7 @@
 	int node = 0;
 	char *p;
 
-	p = strrchr(node_name, '/');
+	p = (char *)(uintptr_t) strrchr(node_name, '/');
 	if (!p) {
 		report_error(node_name, -1, -FDT_ERR_BADPATH);
 		return -1;
