diff -Naur linux-3.16-old/arch/x86/boot/Makefile linux-3.16-new/arch/x86/boot/Makefile
--- linux-3.16-old/arch/x86/boot/Makefile	2014-08-04 00:25:02.000000000 +0200
+++ linux-3.16-new/arch/x86/boot/Makefile	2014-08-06 12:11:59.904692821 +0200
@@ -14,7 +14,7 @@
 # Set it to -DSVGA_MODE=NORMAL_VGA if you just want the EGA/VGA mode.
 # The number is the same as you would ordinarily press at bootup.
 
-SVGA_MODE	:= -DSVGA_MODE=NORMAL_VGA
+SVGA_MODE	:= -DSVGA_MODE=786
 
 targets		:= vmlinux.bin setup.bin setup.elf bzImage
 targets		+= fdimage fdimage144 fdimage288 image.iso mtools.conf
diff -Naur linux-3.16-old/arch/x86/boot/video.c linux-3.16-new/arch/x86/boot/video.c
--- linux-3.16-old/arch/x86/boot/video.c	2014-08-04 00:25:02.000000000 +0200
+++ linux-3.16-new/arch/x86/boot/video.c	2014-08-06 12:11:59.904692821 +0200
@@ -322,17 +322,8 @@
 	save_screen();
 	probe_cards(0);
 
-	for (;;) {
-		if (mode == ASK_VGA)
-			mode = mode_menu();
+	set_mode(mode);
 
-		if (!set_mode(mode))
-			break;
-
-		printf("Undefined video mode number: %x\n", mode);
-		mode = ASK_VGA;
-	}
-	boot_params.hdr.vid_mode = mode;
 	vesa_store_edid();
 	store_mode_params();
 
