diff -Naur mythtv-master-20200207-g2869463-old/mythtv/libs/libmythbase/mythcorecontext.cpp mythtv-master-20200207-g2869463-new/mythtv/libs/libmythbase/mythcorecontext.cpp
--- mythtv-master-20200207-g2869463-old/mythtv/libs/libmythbase/mythcorecontext.cpp	2020-02-07 17:35:22.000000000 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/libs/libmythbase/mythcorecontext.cpp	2020-02-08 14:18:04.680000002 +0100
@@ -311,7 +311,8 @@
     proto_mismatch = false;
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!CheckProtoVersion(serverSock, timeout_in_ms, true))
+    if (!gCoreContext->GetNumSetting("IgnoreProtoVerMismatch", false)
+        && !CheckProtoVersion(serverSock, timeout_in_ms, true))
     {
         proto_mismatch = true;
         return false;
diff -Naur mythtv-master-20200207-g2869463-old/mythtv/libs/libmythbase/remotefile.cpp mythtv-master-20200207-g2869463-new/mythtv/libs/libmythbase/remotefile.cpp
--- mythtv-master-20200207-g2869463-old/mythtv/libs/libmythbase/remotefile.cpp	2020-02-07 17:35:22.000000000 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/libs/libmythbase/remotefile.cpp	2020-02-08 14:18:04.680000002 +0100
@@ -160,7 +160,8 @@
     QStringList strlist;
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!gCoreContext->CheckProtoVersion(lsock, 5000))
+    if (!gCoreContext->GetNumSetting("IgnoreProtoVerMismatch", false)
+        && !gCoreContext->CheckProtoVersion(lsock, 5000))
     {
         LOG(VB_GENERAL, LOG_ERR, loc +
             QString("Failed validation to server %1:%2").arg(host).arg(port));
diff -Naur mythtv-master-20200207-g2869463-old/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp mythtv-master-20200207-g2869463-new/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp
--- mythtv-master-20200207-g2869463-old/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp	2020-02-07 17:35:21.000000000 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp	2020-02-08 14:18:04.680000002 +0100
@@ -45,7 +45,8 @@
     }
 
 #ifndef IGNORE_PROTO_VER_MISMATCH
-    if (!m_socket->Validate())
+    if (!gCoreContext->GetNumSetting("IgnoreProtoVerMismatch", false)
+        && !m_socket->Validate())
     {
         LOG(VB_GENERAL, LOG_NOTICE, "Unable to confirm protocol version with backend.");
         m_socket->DecrRef();
diff -Naur mythtv-master-20200207-g2869463-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20200207-g2869463-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20200207-g2869463-old/mythtv/programs/mythfrontend/main.cpp	2020-02-07 17:35:18.000000000 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/programs/mythfrontend/main.cpp	2020-02-08 14:18:04.680000002 +0100
@@ -2050,7 +2050,8 @@
             return GENERIC_EXIT_NO_THEME;
     }
 
-    if (!UpgradeTVDatabaseSchema(false))
+    if (!gCoreContext->GetNumSetting("IgnoreSchemaVerMismatch", false)
+        && !UpgradeTVDatabaseSchema(false))
     {
         LOG(VB_GENERAL, LOG_ERR,
             "Couldn't upgrade database to new schema, exiting.");
