diff -Naur mythtv-master-20150827-g4a81075-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20150827-g4a81075-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20150827-g4a81075-old/mythtv/libs/libmythtv/cardutil.cpp	2015-08-27 20:29:30.000000000 +0200
+++ mythtv-master-20150827-g4a81075-new/mythtv/libs/libmythtv/cardutil.cpp	2015-08-27 21:10:12.749406004 +0200
@@ -1715,7 +1715,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
diff -Naur mythtv-master-20150827-g4a81075-old/mythtv/programs/mythfrontend/statusbox.cpp mythtv-master-20150827-g4a81075-new/mythtv/programs/mythfrontend/statusbox.cpp
--- mythtv-master-20150827-g4a81075-old/mythtv/programs/mythfrontend/statusbox.cpp	2015-08-27 20:29:49.000000000 +0200
+++ mythtv-master-20150827-g4a81075-new/mythtv/programs/mythfrontend/statusbox.cpp	2015-08-27 21:15:15.081839976 +0200
@@ -862,7 +862,7 @@
 
         QString tun = tr("Tuner %1 %2 %3");
         QString devlabel = CardUtil::GetDeviceLabel(
-            query.value(1).toString(), query.value(2).toString());
+            query.value(0).toInt());
 
         QString shorttuner = tun.arg(cardid).arg("").arg(status);
         QString longtuner = tun.arg(cardid).arg(devlabel).arg(status);
