diff -Naur mythtv-0.27-20130909-ga2aa581-old/mythplugins/mythweb/modules/tv/tmpl/default/detail.php mythtv-0.27-20130909-ga2aa581-new/mythplugins/mythweb/modules/tv/tmpl/default/detail.php
--- mythtv-0.27-20130909-ga2aa581-old/mythplugins/mythweb/modules/tv/tmpl/default/detail.php	2013-09-09 07:37:33.000000000 +0200
+++ mythtv-0.27-20130909-ga2aa581-new/mythplugins/mythweb/modules/tv/tmpl/default/detail.php	2013-09-10 20:01:56.947525426 +0200
@@ -853,7 +853,7 @@
             </div>
 
         <?php
             flush();
-            $frontends = MythFrontend::findFrontends();
+//            $frontends = MythFrontend::findFrontends();
             if (is_array($frontends)) {
                 echo '<div class="x-frontends">'.t('Play Recording on Frontend').':<ul>';
                     foreach ($frontends as $frontend)
