diff -Naur mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/osd.cpp	2015-04-30 10:55:18.574359835 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/osd.cpp	2015-04-30 10:55:52.980887393 +0200
@@ -528,7 +528,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
