From c2e2844a8106731a4217d8f090bf48f27c8d6f1c Mon Sep 17 00:00:00 2001
From: Roger Siddons <dizygotheca@ntlworld.com>
Date: Sun, 11 Oct 2015 14:57:12 +0100
Subject: [PATCH 13/13] Minimyth2: Use remotecache for local gallery images


diff --git a/mythtv/libs/libmythui/mythuihelper.cpp b/mythtv/libs/libmythui/mythuihelper.cpp
index 1443dad..49b179e 100644
--- a/mythtv/libs/libmythui/mythuihelper.cpp
+++ b/mythtv/libs/libmythui/mythuihelper.cpp
@@ -751,8 +751,9 @@ bool MythUIHelper::IsImageInCache(const QString &url)
 
 QString MythUIHelper::GetThemeCacheDir(const QString &mrl)
 {
-    // Gallery thumbnails ("-Photographs-") and images ("-Photographs@") are cached separately
-    if (mrl.contains("-Photographs"))
+    // Gallery SG thumbnails ("-Photographs-"), SG slides ("-Photographs@") and
+    // local thumbs/slides (absolute path) are cached separately
+    if (mrl.startsWith('-') || mrl.contains("-Photographs"))
         return GetConfDir() + "/cache/remotecache/";
 
     QString cachedirname = GetConfDir() + "/cache/themecache/";
-- 
2.1.4

