diff -Naur mythtv-master-20160429-g24c5575-old/mythplugins/mythgallery/mythgallery/main.cpp mythtv-master-20160429-g24c5575-new/mythplugins/mythgallery/mythgallery/main.cpp
--- mythtv-master-20160429-g24c5575-old/mythplugins/mythgallery/mythgallery/main.cpp	2016-04-29 16:01:09.000000000 +0200
+++ mythtv-master-20160429-g24c5575-new/mythplugins/mythgallery/mythgallery/main.cpp	2016-04-29 16:55:06.540000000 +0200
@@ -137,9 +137,10 @@
         "Mark image"), "T");
     REG_KEY("Gallery", "FULLSCREEN", QT_TRANSLATE_NOOP("MythControls",
         "Toggle scale to fullscreen/scale to fit"), "W");
-    REG_MEDIA_HANDLER(QT_TRANSLATE_NOOP("MythControls",
-        "MythGallery Media Handler 1/3"), "", "", handleMedia,
-        MEDIATYPE_DATA | MEDIATYPE_MIXED, QString::null);
+    REG_MEDIA_HANDLER(
+        QT_TRANSLATE_NOOP("MythControls", "MythGallery Media Handler 1/3"),
+        QT_TRANSLATE_NOOP("MythControls", "MythGallery mixed data"),
+        "", handleMedia, MEDIATYPE_DATA | MEDIATYPE_MIXED, QString::null);
     QString filt;
     Q_FOREACH(QString format, GalleryUtil::GetImageFilter())
     {
@@ -149,9 +150,10 @@
         else
             filt += "," + format;
     }
-    REG_MEDIA_HANDLER(QT_TRANSLATE_NOOP("MythControls",
-        "MythGallery Media Handler 2/3"), "", "", handleMedia,
-        MEDIATYPE_MGALLERY, filt);
+    REG_MEDIA_HANDLER(
+        QT_TRANSLATE_NOOP("MythControls", "MythGallery Media Handler 2/3"),
+        QT_TRANSLATE_NOOP("MythControls", "MythGallery images"),
+        "", handleMedia, MEDIATYPE_MGALLERY, filt);
     filt.clear();
     Q_FOREACH(QString format, GalleryUtil::GetMovieFilter())
     {
@@ -161,9 +163,10 @@
         else
             filt += "," + format;
     }
-    REG_MEDIA_HANDLER(QT_TRANSLATE_NOOP("MythControls",
-        "MythGallery Media Handler 3/3"), "", "", handleMedia,
-        MEDIATYPE_MVIDEO, filt);
+    REG_MEDIA_HANDLER(
+        QT_TRANSLATE_NOOP("MythControls", "MythGallery Media Handler 3/3"),
+        QT_TRANSLATE_NOOP("MythControls", "MythGallery movies"),
+        "", handleMedia, MEDIATYPE_MVIDEO, filt);
 }
 
 int mythplugin_init(const char *libversion)
diff -Naur mythtv-master-20160429-g24c5575-old/mythplugins/mythmusic/mythmusic/main.cpp mythtv-master-20160429-g24c5575-new/mythplugins/mythmusic/mythmusic/main.cpp
--- mythtv-master-20160429-g24c5575-old/mythplugins/mythmusic/mythmusic/main.cpp	2016-04-29 16:53:17.000000000 +0200
+++ mythtv-master-20160429-g24c5575-new/mythplugins/mythmusic/mythmusic/main.cpp	2016-04-29 16:55:06.540000000 +0200
@@ -870,15 +870,17 @@
     REG_KEY("Music", "ADDTRACKSANDPLAY",              QT_TRANSLATE_NOOP("MythControls",
         "Add selected tracks to playlist and start playback"), "");
 
-    REG_MEDIA_HANDLER(QT_TRANSLATE_NOOP("MythControls",
-        "MythMusic Media Handler 1/2"), "", "", handleCDMedia,
-        MEDIATYPE_AUDIO | MEDIATYPE_MIXED, QString::null);
+    REG_MEDIA_HANDLER(
+        QT_TRANSLATE_NOOP("MythControls", "MythMusic Media Handler 1/2"),
+        QT_TRANSLATE_NOOP("MythControls", "MythMusic audio CD"),
+        "", handleCDMedia, MEDIATYPE_AUDIO | MEDIATYPE_MIXED, QString::null);
     QString filt = MetaIO::ValidFileExtensions;
     filt.replace('|',',');
     filt.remove('.');
-    REG_MEDIA_HANDLER(QT_TRANSLATE_NOOP("MythControls",
-        "MythMusic Media Handler 2/2"), "", "", handleMedia,
-        MEDIATYPE_MMUSIC, filt);
+    REG_MEDIA_HANDLER(
+        QT_TRANSLATE_NOOP("MythControls", "MythMusic Media Handler 2/2"),
+        QT_TRANSLATE_NOOP("MythControls", "MythMusic audio files"),
+         "", handleMedia, MEDIATYPE_MMUSIC, filt);
 }
 
 int mythplugin_init(const char *libversion)
diff -Naur mythtv-master-20160429-g24c5575-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-master-20160429-g24c5575-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-master-20160429-g24c5575-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2016-04-29 16:53:16.000000000 +0200
+++ mythtv-master-20160429-g24c5575-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2016-04-29 16:55:06.540000000 +0200
@@ -664,10 +664,33 @@
         return;
     }
 
-
-    // TODO - Generate a dialog, add buttons for each description,
-    // if user didn't cancel, selected = handlers.at(choice);
     int selected = 0;
+    if (handlers.size() > 1)
+    {
+        QStringList buttonmsgs;
+        for (QList<MHData>::const_iterator it = handlers.begin(); it != handlers.end(); ++it)
+            buttonmsgs << ((!it->description.isEmpty()) ? it->description : it->destination);
+        buttonmsgs << tr("Cancel");
+
+        const DialogCode cancelbtn = DialogCode(
+            int(kDialogCodeButton0) + buttonmsgs.size() - 1);
+
+        DialogCode ret = MythPopupBox::ShowButtonPopup(GetMythMainWindow(),
+                                tr("Media Handler Selection"),
+                                tr("The new media contains mixed content "
+                                   "that can be rendered in different ways. "
+                                   "Select your preferred method."),
+                                buttonmsgs, cancelbtn);
+        if (kDialogCodeRejected == ret || cancelbtn == ret)
+        {
+            LOG(VB_MEDIA, LOG_INFO, "User cancelled media handler selection");
+            return;
+        }
+
+        selected = MythDialog::CalcItemIndex(ret);
+        LOG(VB_MEDIA, LOG_NOTICE, QString("User selected '%1'")
+            .arg(handlers.at(selected).destination) );
+    }
 
     handlers.at(selected).callback(pMedia);
 }
diff -Naur mythtv-master-20160429-g24c5575-old/mythtv/libs/libmythbase/mythmedia.cpp mythtv-master-20160429-g24c5575-new/mythtv/libs/libmythbase/mythmedia.cpp
--- mythtv-master-20160429-g24c5575-old/mythtv/libs/libmythbase/mythmedia.cpp	2016-04-29 16:01:21.000000000 +0200
+++ mythtv-master-20160429-g24c5575-new/mythtv/libs/libmythbase/mythmedia.cpp	2016-04-29 16:55:06.540000000 +0200
@@ -237,7 +237,7 @@
         else
         {
             LOG(VB_MEDIA, LOG_NOTICE, QString(
-                    "DetectMediaType(this=0x%1) unknown file type %1")
+                    "DetectMediaType(this=0x%1) unknown file type %2")
                 .arg(quintptr(this),0,16).arg(it.key()));
         }
     }
diff -Naur mythtv-master-20160429-g24c5575-old/mythtv/programs/mythfrontend/main.cpp mythtv-master-20160429-g24c5575-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-master-20160429-g24c5575-old/mythtv/programs/mythfrontend/main.cpp	2016-04-29 16:53:18.000000000 +0200
+++ mythtv-master-20160429-g24c5575-new/mythtv/programs/mythfrontend/main.cpp	2016-04-29 16:55:06.543333333 +0200
@@ -1554,9 +1554,12 @@
 {
     REG_MEDIAPLAYER("Internal", QT_TRANSLATE_NOOP("MythControls",
         "MythTV's native media player."), internal_play_media);
-    REG_MEDIA_HANDLER(QT_TRANSLATE_NOOP("MythControls",
-        "MythDVD DVD Media Handler"), "", "", handleDVDMedia,
-        MEDIATYPE_DVD, QString::null);
+
+    REG_MEDIA_HANDLER(
+        QT_TRANSLATE_NOOP("MythControls", "MythDVD DVD Media Handler"),
+        QT_TRANSLATE_NOOP("MythControls", "MythDVD media"),
+        "", handleDVDMedia, MEDIATYPE_DVD, QString::null);
+
     REG_MEDIA_HANDLER(QT_TRANSLATE_NOOP("MythControls",
         "MythImage Media Handler 1/2"), "", "", handleGalleryMedia,
         MEDIATYPE_DATA | MEDIATYPE_MIXED, QString::null);
