diff -Naur mythtv-master-20160811-ge829c96-old/mythtv/libs/libmyth/mythwidgets.cpp mythtv-master-20160811-ge829c96-new/mythtv/libs/libmyth/mythwidgets.cpp
--- mythtv-master-20160811-ge829c96-old/mythtv/libs/libmyth/mythwidgets.cpp	2016-08-11 09:33:44.000000000 +0200
+++ mythtv-master-20160811-ge829c96-new/mythtv/libs/libmyth/mythwidgets.cpp	2016-08-31 12:54:03.623333325 +0200
@@ -22,6 +22,10 @@
 {
     setObjectName(name);
     setEditable(rw);
+#ifdef Q_OS_MAC
+    // Qt 5 changed OS X combobox behaviour
+    setFocusPolicy(Qt::WheelFocus);
+#endif
 }
 
 MythComboBox::~MythComboBox()
