diff -Naur mythtv-master-20170809-ga1823f3-old/mythtv/libs/libmythtv/subtitlescreen.cpp mythtv-master-20170809-ga1823f3-new/mythtv/libs/libmythtv/subtitlescreen.cpp
--- mythtv-master-20170809-ga1823f3-old/mythtv/libs/libmythtv/subtitlescreen.cpp	2017-08-09 10:02:38.000000000 +0200
+++ mythtv-master-20170809-ga1823f3-new/mythtv/libs/libmythtv/subtitlescreen.cpp	2017-08-09 12:46:07.503333328 +0200
@@ -2150,11 +2150,7 @@
         //    playPos = (uint64_t)
         //        ((currentFrame->frameNumber / video_frame_rate) * 1000);
         //else
-        //playPos = m_player->GetDecoder()->NormalizeVideoTimecode(currentFrame->timecode);
-        //
-        // Change of plans.  Just ask the player how many milliseconds
-        // have been played so far.
-        playPos = m_player->GetSecondsPlayed(false, 1);
+        playPos = m_player->GetDecoder()->NormalizeVideoTimecode(currentFrame->timecode);
     }
     playPos -= playPosAdj;
     if (playPos != 0)
