diff --git a/mythtv/libs/libmythui/mythpainter.cpp b/mythtv/libs/libmythui/mythpainter.cpp
index 0a96b97..5dd1114 100644
--- a/mythtv/libs/libmythui/mythpainter.cpp
+++ b/mythtv/libs/libmythui/mythpainter.cpp
@@ -24,8 +24,8 @@ MythPainter::MythPainter()
     m_showBorders(false), m_showNames(false)
 {
     SetMaximumCacheSizes(
-        gCoreContext->GetNumSetting("UIPainterMaxCacheHW",64),
-        gCoreContext->GetNumSetting("UIPainterMaxCacheSW",48) );
+        gCoreContext->GetNumSetting("UIPainterMaxCacheHW",256),
+        gCoreContext->GetNumSetting("UIPainterMaxCacheSW",192) );
 }
 
 void MythPainter::Teardown(void)
diff --git a/mythtv/programs/mythfrontend/globalsettings.cpp b/mythtv/programs/mythfrontend/globalsettings.cpp
index b90d080..ada00bf 100644
--- a/mythtv/programs/mythfrontend/globalsettings.cpp
+++ b/mythtv/programs/mythfrontend/globalsettings.cpp
@@ -2082,7 +2082,7 @@ static HostComboBox *AdjustFill()
 
 static HostSpinBox *GuiWidth()
 {
-    HostSpinBox *gs = new HostSpinBox("GuiWidth", 0, 1920, 8, true);
+    HostSpinBox *gs = new HostSpinBox("GuiWidth", 0, 3840, 8, true);
 
     gs->setLabel(AppearanceSettings::tr("GUI width (pixels)"));
 
@@ -2098,7 +2098,7 @@ static HostSpinBox *GuiWidth()
 
 static HostSpinBox *GuiHeight()
 {
-    HostSpinBox *gs = new HostSpinBox("GuiHeight", 0, 1600, 8, true);
+    HostSpinBox *gs = new HostSpinBox("GuiHeight", 0, 2160, 8, true);
 
     gs->setLabel(AppearanceSettings::tr("GUI height (pixels)"));
 
@@ -2128,7 +2128,7 @@ static HostSpinBox *GuiOffsetX()
 
 static HostSpinBox *GuiOffsetY()
 {
-    HostSpinBox *gs = new HostSpinBox("GuiOffsetY", -1600, 1600, 8, true);
+    HostSpinBox *gs = new HostSpinBox("GuiOffsetY", -2160, 2160, 8, true);
 
     gs->setLabel(AppearanceSettings::tr("GUI Y offset"));
 
@@ -2202,7 +2202,7 @@ static HostCheckBox *UseVideoModes()
 static HostSpinBox *VidModeWidth(int idx)
 {
     HostSpinBox *gs = new HostSpinBox(QString("VidModeWidth%1").arg(idx),
-                                            0, 1920, 16, true);
+                                            0, 3840, 16, true);
 
     gs->setLabel((idx<1) ? VideoModeSettings::tr("In X", "Video mode width")
                          : "");
@@ -2220,7 +2220,7 @@ static HostSpinBox *VidModeWidth(int idx)
 static HostSpinBox *VidModeHeight(int idx)
 {
     HostSpinBox *gs = new HostSpinBox(QString("VidModeHeight%1").arg(idx),
-                                            0, 1080, 16, true);
+                                            0, 2180, 16, true);
 
     gs->setLabel((idx<1) ? VideoModeSettings::tr("In Y", "Video mode height")
                          : "");

