diff -Naur libcap-ng-0.7.3-old/configure.ac libcap-ng-0.7.3-new/configure.ac
--- libcap-ng-0.7.3-old/configure.ac	2012-11-09 11:59:13.000000000 -0800
+++ libcap-ng-0.7.3-new/configure.ac	2012-12-29 09:02:29.000000000 -0800
@@ -31,7 +31,7 @@
 AC_REVISION($Revision: 1.3 $)dnl
 AC_INIT(libcap-ng,0.7.3)
 AC_PREREQ(2.12)dnl
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIR([m4])
 
 echo Configuring libcap-ng $VERSION
