diff --speed-large-files --no-dereference --minimal -Naur llvm-16.0.1.src/CMakeLists.txt llvm-16.0.1.src/CMakeLists.txt
--- llvm-16.0.1.src/CMakeLists.txt	2023-04-05 05:36:38.000000000 +0200
+++ llvm-16.0.1.src/CMakeLists.txt	2023-04-29 14:44:29.753175727 +0200
@@ -10,8 +10,8 @@
 endif()
 
 set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/../cmake)
-include(${LLVM_COMMON_CMAKE_UTILS}/Modules/CMakePolicy.cmake
-  NO_POLICY_SCOPE)
+#include(${LLVM_COMMON_CMAKE_UTILS}/Modules/CMakePolicy.cmake
+#  NO_POLICY_SCOPE)
 
 set(CMAKE_BUILD_WITH_INSTALL_NAME_DIR ON)
 
