diff -Naur llvm-3.6.2.src-old/autoconf/configure.ac llvm-3.6.2.src-new/autoconf/configure.ac
--- llvm-3.6.2.src-old/autoconf/configure.ac	2015-06-01 07:30:22.000000000 -0700
+++ llvm-3.6.2.src-new/autoconf/configure.ac	2015-07-19 10:01:18.386000003 -0700
@@ -1423,21 +1423,21 @@
 
   AC_PATH_PROG([PYTHON], [python python2 python27],
                [AC_MSG_RESULT([not found])
-                AC_MSG_ERROR([could not find python 2.7 or higher])])
+                AC_MSG_ERROR([could not find python 2.6 or higher])])
 fi
 
-AC_MSG_CHECKING([for python >= 2.7])
+AC_MSG_CHECKING([for python >= 2.6])
 ac_python_version=`$PYTHON -V 2>&1 | cut -d' ' -f2`
 ac_python_version_major=`echo $ac_python_version | cut -d'.' -f1`
 ac_python_version_minor=`echo $ac_python_version | cut -d'.' -f2`
 ac_python_version_patch=`echo $ac_python_version | cut -d'.' -f3`
 if test "$ac_python_version_major" -gt "2" || \
    (test "$ac_python_version_major" -eq "2" && \
-    test "$ac_python_version_minor" -ge "7") ; then
+    test "$ac_python_version_minor" -ge "6") ; then
   AC_MSG_RESULT([$PYTHON ($ac_python_version)])
 else
   AC_MSG_RESULT([not found])
-  AC_MSG_FAILURE([found python $ac_python_version ($PYTHON); required >= 2.7])
+  AC_MSG_FAILURE([found python $ac_python_version ($PYTHON); required >= 2.6])
 fi
 
 dnl===-----------------------------------------------------------------------===
diff -Naur llvm-3.6.2.src-old/CMakeLists.txt llvm-3.6.2.src-new/CMakeLists.txt
--- llvm-3.6.2.src-old/CMakeLists.txt	2015-06-01 07:30:22.000000000 -0700
+++ llvm-3.6.2.src-new/CMakeLists.txt	2015-07-19 10:00:13.799000003 -0700
@@ -346,7 +346,7 @@
 # Verify that we can find a Python 2 interpreter.  Python 3 is unsupported.
 # FIXME: We should support systems with only Python 3, but that requires work
 # on LLDB.
-set(Python_ADDITIONAL_VERSIONS 2.7)
+set(Python_ADDITIONAL_VERSIONS 2.6)
 include(FindPythonInterp)
 if( NOT PYTHONINTERP_FOUND )
   message(FATAL_ERROR
@@ -355,8 +355,8 @@
 Please install Python or specify the PYTHON_EXECUTABLE CMake variable.")
 endif()
 
-if( ${PYTHON_VERSION_STRING} VERSION_LESS 2.7 )
-  message(FATAL_ERROR "Python 2.7 or newer is required")
+if( ${PYTHON_VERSION_STRING} VERSION_LESS 2.6 )
+  message(FATAL_ERROR "Python 2.6 or newer is required")
 endif()
 
 ######
diff -Naur llvm-3.6.2.src-old/configure llvm-3.6.2.src-new/configure
--- llvm-3.6.2.src-old/configure	2015-06-01 07:30:22.000000000 -0700
+++ llvm-3.6.2.src-new/configure	2015-07-19 10:00:44.840000003 -0700
@@ -8188,8 +8188,8 @@
 
   test -z "$ac_cv_path_PYTHON" && ac_cv_path_PYTHON="{ echo "$as_me:$LINENO: result: not found" >&5
 echo "${ECHO_T}not found" >&6; }
-                { { echo "$as_me:$LINENO: error: could not find python 2.7 or higher" >&5
-echo "$as_me: error: could not find python 2.7 or higher" >&2;}
+                { { echo "$as_me:$LINENO: error: could not find python 2.6 or higher" >&5
+echo "$as_me: error: could not find python 2.6 or higher" >&2;}
    { (exit 1); exit 1; }; }"
   ;;
 esac
@@ -8206,23 +8206,23 @@
 
 fi
 
-{ echo "$as_me:$LINENO: checking for python >= 2.7" >&5
-echo $ECHO_N "checking for python >= 2.7... $ECHO_C" >&6; }
+{ echo "$as_me:$LINENO: checking for python >= 2.6" >&5
+echo $ECHO_N "checking for python >= 2.6... $ECHO_C" >&6; }
 ac_python_version=`$PYTHON -V 2>&1 | cut -d' ' -f2`
 ac_python_version_major=`echo $ac_python_version | cut -d'.' -f1`
 ac_python_version_minor=`echo $ac_python_version | cut -d'.' -f2`
 ac_python_version_patch=`echo $ac_python_version | cut -d'.' -f3`
 if test "$ac_python_version_major" -gt "2" || \
    (test "$ac_python_version_major" -eq "2" && \
-    test "$ac_python_version_minor" -ge "7") ; then
+    test "$ac_python_version_minor" -ge "6") ; then
   { echo "$as_me:$LINENO: result: $PYTHON ($ac_python_version)" >&5
 echo "${ECHO_T}$PYTHON ($ac_python_version)" >&6; }
 else
   { echo "$as_me:$LINENO: result: not found" >&5
 echo "${ECHO_T}not found" >&6; }
-  { { echo "$as_me:$LINENO: error: found python $ac_python_version ($PYTHON); required >= 2.7
+  { { echo "$as_me:$LINENO: error: found python $ac_python_version ($PYTHON); required >= 2.6
 See \`config.log' for more details." >&5
-echo "$as_me: error: found python $ac_python_version ($PYTHON); required >= 2.7
+echo "$as_me: error: found python $ac_python_version ($PYTHON); required >= 2.6
 See \`config.log' for more details." >&2;}
    { (exit 1); exit 1; }; }
 fi
