diff -Naur llvm-3.4-old/lib/Target/R600/TargetInfo/AMDGPUTargetInfo.cpp llvm-3.4-new/lib/Target/R600/TargetInfo/AMDGPUTargetInfo.cpp
--- llvm-3.4-old/lib/Target/R600/TargetInfo/AMDGPUTargetInfo.cpp	2013-06-07 22:28:43.000000000 +0200
+++ llvm-3.4-new/lib/Target/R600/TargetInfo/AMDGPUTargetInfo.cpp	2014-02-25 14:32:40.000000000 +0100
@@ -11,7 +11,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include "AMDGPUTargetMachine.h"
+#include "../AMDGPUTargetMachine.h"
 #include "llvm/Support/TargetRegistry.h"
 
 using namespace llvm;
