diff -Naur llvm-3.7.0.src-old/lib/Target/AMDGPU/TargetInfo/AMDGPUTargetInfo.cpp llvm-3.7.0.src-new/lib/Target/AMDGPU/TargetInfo/AMDGPUTargetInfo.cpp
--- llvm-3.7.0.src-old/lib/Target/AMDGPU/TargetInfo/AMDGPUTargetInfo.cpp	2015-06-13 05:28:10.000000000 +0200
+++ llvm-3.7.0.src-new/lib/Target/AMDGPU/TargetInfo/AMDGPUTargetInfo.cpp	2015-09-02 21:28:42.000000000 +0200
@@ -11,7 +11,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include "AMDGPUTargetMachine.h"
+#include "../AMDGPUTargetMachine.h"
 #include "llvm/Support/TargetRegistry.h"
 
 using namespace llvm;
diff -Naur llvm-3.7.0.src-old/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.h llvm-3.7.0.src-new/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.h
--- llvm-3.7.0.src-old/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.h	2015-06-26 23:58:31.000000000 +0200
+++ llvm-3.7.0.src-new/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.h	2015-09-02 21:51:28.000000000 +0200
@@ -10,7 +10,7 @@
 #ifndef LLVM_LIB_TARGET_AMDGPU_UTILS_AMDGPUBASEINFO_H
 #define LLVM_LIB_TARGET_AMDGPU_UTILS_AMDGPUBASEINFO_H
 
-#include "AMDKernelCodeT.h"
+#include "../AMDKernelCodeT.h"
 
 namespace llvm {
 
