diff -Naur llvm-3.6.2.src-old/Makefile.rules llvm-3.6.2.src-new/Makefile.rules
--- llvm-3.6.2.src-old/Makefile.rules	2015-07-19 09:59:18.435000003 -0700
+++ llvm-3.6.2.src-new/Makefile.rules	2015-07-19 10:02:42.783000003 -0700
@@ -651,7 +651,9 @@
   endif
   ifneq ($(HOST_OS), Darwin)
     ifdef TOOLNAME
-      LD.Flags += $(RPATH) -Wl,'$$ORIGIN/../lib'
+      ifneq ($(TOOLNAME),llvm-config)
+        LD.Flags += $(RPATH) -Wl,'$$ORIGIN/../lib'
+      endif
     endif
   else
     ifneq ($(DARWIN_MAJVERS),4)
