diff -Naur llvm-3.7.0.src-old/Makefile.config.in llvm-3.7.0.src-new/Makefile.config.in
--- llvm-3.7.0.src-old/Makefile.config.in	2015-09-02 13:30:25.000000000 +0200
+++ llvm-3.7.0.src-new/Makefile.config.in	2015-09-02 13:51:45.794464335 +0200
@@ -58,11 +58,6 @@
 PROJ_SRC_ROOT   := $(LLVM_SRC_ROOT)
 PROJ_SRC_DIR    := $(LLVM_SRC_ROOT)$(patsubst $(PROJ_OBJ_ROOT)%,%,$(PROJ_OBJ_DIR))
 
-# See: http://lists.llvm.org/pipermail/llvm-commits/Week-of-Mon-20150323/268067.html
-ifeq ($(LLVM_SRC_ROOT), $(LLVM_OBJ_ROOT))
-  $(error In-source builds are not allowed. Please configure from a separate build directory!)
-endif
-
 ifneq ($(CLANG_SRC_ROOT),)
   CLANG_SRC_ROOT:= $(call realpath, $(CLANG_SRC_ROOT))
   PROJ_SRC_DIR  := $(patsubst $(LLVM_SRC_ROOT)/tools/clang%,$(CLANG_SRC_ROOT)%,$(PROJ_SRC_DIR))
diff -Naur llvm-3.7.0.src-old/autoconf/configure.ac llvm-3.7.0.src-new/autoconf/configure.ac
--- llvm-3.7.0.src-old/autoconf/configure.ac	2015-09-02 13:30:25.000000000 +0200
+++ llvm-3.7.0.src-new/autoconf/configure.ac	2015-09-02 13:50:35.704710248 +0200
@@ -75,7 +75,15 @@
 
 dnl Quit if it is an in-source build
 if test ${srcdir} == "." ; then
-  AC_MSG_ERROR([In-source builds are not allowed. Please configure from a separate build directory!])
+  AC_MSG_WARN([**************************************************************************************])
+  AC_MSG_WARN([*                                                                                    *])
+  AC_MSG_WARN([*                                   WARNING                                          *])
+  AC_MSG_WARN([*                                                                                    *])
+  AC_MSG_WARN([*                        In-source builds are deprecated.                            *])
+  AC_MSG_WARN([*                                                                                    *])
+  AC_MSG_WARN([*               Please configure from a separate build directory!                    *])
+  AC_MSG_WARN([*                                                                                    *])
+  AC_MSG_WARN([**************************************************************************************])
 fi
 
 dnl Default to empty (i.e. assigning the null string to) CFLAGS and CXXFLAGS,
diff -Naur llvm-3.7.0.src-old/configure llvm-3.7.0.src-new/configure
--- llvm-3.7.0.src-old/configure	2015-07-15 01:54:34.000000000 +0200
+++ llvm-3.7.0.src-new/configure	2015-09-02 13:50:35.708043570 +0200
@@ -2033,9 +2033,24 @@
 fi
 
 if test ${srcdir} == "." ; then
-  { { echo "$as_me:$LINENO: error: In-source builds are not allowed. Please configure from a separate build directory!" >&5
-echo "$as_me: error: In-source builds are not allowed. Please configure from a separate build directory!" >&2;}
-   { (exit 1); exit 1; }; }
+  { echo "$as_me:$LINENO: WARNING: **************************************************************************************" >&5
+echo "$as_me: WARNING: **************************************************************************************" >&2;}
+  { echo "$as_me:$LINENO: WARNING: *                                                                                    *" >&5
+echo "$as_me: WARNING: *                                                                                    *" >&2;}
+  { echo "$as_me:$LINENO: WARNING: *                                   WARNING                                          *" >&5
+echo "$as_me: WARNING: *                                   WARNING                                          *" >&2;}
+  { echo "$as_me:$LINENO: WARNING: *                                                                                    *" >&5
+echo "$as_me: WARNING: *                                                                                    *" >&2;}
+  { echo "$as_me:$LINENO: WARNING: *                        In-source builds are deprecated.                            *" >&5
+echo "$as_me: WARNING: *                        In-source builds are deprecated.                            *" >&2;}
+  { echo "$as_me:$LINENO: WARNING: *                                                                                    *" >&5
+echo "$as_me: WARNING: *                                                                                    *" >&2;}
+  { echo "$as_me:$LINENO: WARNING: *               Please configure from a separate build directory!                    *" >&5
+echo "$as_me: WARNING: *               Please configure from a separate build directory!                    *" >&2;}
+  { echo "$as_me:$LINENO: WARNING: *                                                                                    *" >&5
+echo "$as_me: WARNING: *                                                                                    *" >&2;}
+  { echo "$as_me:$LINENO: WARNING: **************************************************************************************" >&5
+echo "$as_me: WARNING: **************************************************************************************" >&2;}
 fi
 
 : ${CFLAGS=}
