diff --speed-large-files --no-dereference --minimal -Naur linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-04-26 17:26:22.246973123 +0200
+++ linux-6.9-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-04-26 17:25:03.846685460 +0200
@@ -290,6 +290,13 @@
 				drive-strength = <40>;
 			};
 
+			rmii_pins: rmii-pins {
+				pins = "PA0", "PA1", "PA2", "PA3", "PA4",
+				       "PA5", "PA6", "PA7", "PA8", "PA9";
+				function = "emac1";
+				drive-strength = <40>;
+			};
+
 			i2c0_pins: i2c0-pins {
 				pins = "PI6", "PI7";
 				function = "i2c0";
@@ -895,6 +914,25 @@
 				compatible = "snps,dwmac-mdio";
 				#address-cells = <1>;
 				#size-cells = <0>;
+			};
+		};
+		emac1: ethernet@5030000 {
+			compatible = "allwinner,sun50i-h616-emac0",
+				     "allwinner,sun50i-a64-emac";
+			reg = <0x05030000 0x10000>;
+			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "macirq";
+			resets = <&ccu RST_BUS_EMAC1>;
+			reset-names = "stmmaceth";
+			clocks = <&ccu CLK_BUS_EMAC1>;
+			clock-names = "stmmaceth";
+			syscon = <&syscon 1>;
+			status = "disabled";
+
+			mdio1: mdio {
+				compatible = "snps,dwmac-mdio";
+				#address-cells = <1>;
+				#size-cells = <0>;
 			};
 		};
 
