diff --speed-large-files --no-dereference --minimal -Naur linux-6.10.11/arch/arm64/boot/dts/rockchip/rk3328-a1.dts linux-6.10.11/arch/arm64/boot/dts/rockchip/rk3328-a1.dts
--- linux-6.10.11/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2024-09-27 17:05:13.010492210 +0200
+++ linux-6.10.11/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2024-09-27 16:48:31.910465552 +0200
@@ -140,7 +140,6 @@
 	bus-width = <8>;
 	cap-mmc-highspeed;
 	mmc-ddr-1_8v;
-	mmc-hs200-1_8v;
 	no-sd;
 	no-sdio;
 	non-removable;
