diff --speed-large-files --no-dereference --minimal -Naur linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
--- linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts	2023-04-28 11:07:29.873151074 +0200
+++ linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts	2023-04-28 11:04:10.996491672 +0200
@@ -6,6 +6,7 @@
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/soc/rockchip,vop2.h>
 #include "rk3566.dtsi"
+#include <dt-bindings/leds/common.h>
 
 / {
 	model = "Pine64 RK3566 Quartz64-A Board";
@@ -72,7 +73,8 @@
 
 		led-work {
 			label = "work-led";
-			default-state = "off";
+			default-state = "on";
+			linux,default-trigger = "heartbeat";
 			gpios = <&gpio0 RK_PD3 GPIO_ACTIVE_HIGH>;
 			pinctrl-names = "default";
 			pinctrl-0 = <&work_led_enable_h>;
@@ -81,9 +83,9 @@
 
 		led-diy {
 			label = "diy-led";
-			default-state = "on";
 			gpios = <&gpio0 RK_PD4 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "disk-activity";
 			pinctrl-names = "default";
 			pinctrl-0 = <&diy_led_enable_h>;
 			retain-state-suspended;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts
--- linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2023-04-28 11:07:30.026484402 +0200
+++ linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2023-04-28 11:04:10.996491672 +0200
@@ -6,6 +6,7 @@
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/soc/rockchip,vop2.h>
 #include "rk3566.dtsi"
+#include <dt-bindings/leds/common.h>
 
 / {
 	model = "Pine64 RK3566 Quartz64-B Board";
@@ -45,9 +46,9 @@
 
 		led-user {
 			label = "user-led";
-			default-state = "on";
 			gpios = <&gpio0 RK_PA0 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "disk-activity";
 			pinctrl-names = "default";
 			pinctrl-0 = <&user_led_enable_h>;
 			retain-state-suspended;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-roc-pc.dts linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-roc-pc.dts
--- linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-roc-pc.dts	2023-04-23 21:02:52.000000000 +0200
+++ linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-roc-pc.dts	2023-04-28 11:04:10.996491672 +0200
@@ -6,6 +6,7 @@
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/soc/rockchip,vop2.h>
 #include "rk3566.dtsi"
+#include <dt-bindings/leds/common.h>
 
 / {
 	model = "Firefly Station M2";
@@ -46,7 +47,8 @@
 			label = "user-led";
 			default-state = "on";
 			gpios = <&gpio0 RK_PD3 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "disk-activity";
 			pinctrl-names = "default";
 			pinctrl-0 = <&user_led_enable_h>;
 			retain-state-suspended;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-soquartz.dtsi linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-soquartz.dtsi
--- linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-soquartz.dtsi	2023-04-23 21:02:52.000000000 +0200
+++ linux-6.3/arch/arm64/boot/dts/rockchip/rk3566-soquartz.dtsi	2023-04-28 11:04:10.996491672 +0200
@@ -6,6 +6,7 @@
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/soc/rockchip,vop2.h>
 #include "rk3566.dtsi"
+#include <dt-bindings/leds/common.h>
 
 / {
 	model = "Pine64 RK3566 SoQuartz SOM";
@@ -56,8 +57,9 @@
 
 		led_work: led-work {
 			label = "work-led";
-			default-state = "off";
 			gpios = <&gpio0 RK_PC0 GPIO_ACTIVE_LOW>;
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "disk-activity";
 			pinctrl-names = "default";
 			pinctrl-0 = <&work_led_enable_h>;
 			retain-state-suspended;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts linux-6.3/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts
--- linux-6.3/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts	2023-04-23 21:02:52.000000000 +0200
+++ linux-6.3/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts	2023-04-28 11:04:10.996491672 +0200
@@ -51,9 +51,9 @@
 
 		led_work: led-0 {
 			gpios = <&gpio0 RK_PC0 GPIO_ACTIVE_HIGH>;
-			function = LED_FUNCTION_HEARTBEAT;
 			color = <LED_COLOR_ID_BLUE>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "disk-activity";
 			pinctrl-names = "default";
 			pinctrl-0 = <&led_work_en>;
 		};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts linux-6.3/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts
--- linux-6.3/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts	2023-04-28 11:07:30.063151069 +0200
+++ linux-6.3/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts	2023-04-28 11:05:29.149822157 +0200
@@ -13,8 +13,8 @@
 
 	aliases {
 		ethernet0 = &gmac1;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc0;
+		mmc1 = &sdhci;
+		mmc0 = &sdmmc0;
 		mmc2 = &sdmmc2;
 	};
 
@@ -44,9 +44,9 @@
 
 		led_user: led-0 {
 			gpios = <&gpio0 RK_PB7 GPIO_ACTIVE_HIGH>;
-			function = LED_FUNCTION_HEARTBEAT;
 			color = <LED_COLOR_ID_BLUE>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "disk-activity";
 			pinctrl-names = "default";
 			pinctrl-0 = <&led_user_en>;
 		};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3/arch/arm64/boot/dts/rockchip/rk356x.dtsi linux-6.3/arch/arm64/boot/dts/rockchip/rk356x.dtsi
--- linux-6.3/arch/arm64/boot/dts/rockchip/rk356x.dtsi	2023-04-28 11:07:29.986484405 +0200
+++ linux-6.3/arch/arm64/boot/dts/rockchip/rk356x.dtsi	2023-04-28 11:04:10.996491672 +0200
@@ -43,6 +43,8 @@
 		spi1 = &spi1;
 		spi2 = &spi2;
 		spi3 = &spi3;
+		mmc0 = &sdmmc0;
+		mmc1 = &sdhci;
 	};
 
 	cpus {
