diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.12/drivers/gpu/drm/drm_connector.c linux-6.12.12/drivers/gpu/drm/drm_connector.c
--- linux-6.12.12/drivers/gpu/drm/drm_connector.c	2025-02-05 12:32:20.456689591 +0100
+++ linux-6.12.12/drivers/gpu/drm/drm_connector.c	2025-02-05 12:21:27.706687409 +0100
@@ -507,9 +507,6 @@
 	if (!supported_formats || !(supported_formats & BIT(HDMI_COLORSPACE_RGB)))
 		return -EINVAL;
 
-	if (connector->ycbcr_420_allowed != !!(supported_formats & BIT(HDMI_COLORSPACE_YUV420)))
-		return -EINVAL;
-
 	if (!(max_bpc == 8 || max_bpc == 10 || max_bpc == 12))
 		return -EINVAL;
 
