diff --speed-large-files --no-dereference --minimal -Naur linux-6.11.2/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi linux-6.11.2/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
--- linux-6.11.2/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2024-10-05 10:17:53.567298429 +0200
+++ linux-6.11.2/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2024-10-05 09:49:52.282394471 +0200
@@ -1270,8 +1308,21 @@
 			 <&cru CLK_HDMITX0_EARC>,
 			 <&cru CLK_HDMITX0_REF>,
 			 <&cru MCLK_I2S5_8CH_TX>,
+			 <&cru DCLK_VOP0>,
+			 <&cru DCLK_VOP1>,
+			 <&cru DCLK_VOP2>,
+			 <&cru DCLK_VOP3>,
 			 <&cru HCLK_VO1>;
-		clock-names = "pclk", "hdp", "earc", "ref", "aud", "hclk_vo1";
+		clock-names = "pclk",
+			    	  "hdp",
+			    	  "earc",
+			    	  "ref",
+			    	  "aud",
+			    	  "dclk_vp0",
+			    	  "dclk_vp1",
+			    	  "dclk_vp2",
+			    	  "dclk_vp3",
+			    	  "hclk_vo1";
 		interrupts = <GIC_SPI 169 IRQ_TYPE_LEVEL_HIGH 0>,
 			     <GIC_SPI 170 IRQ_TYPE_LEVEL_HIGH 0>,
 			     <GIC_SPI 171 IRQ_TYPE_LEVEL_HIGH 0>,
@@ -1287,6 +1338,7 @@
 		reset-names = "ref", "hdp";
 		rockchip,grf = <&sys_grf>;
 		rockchip,vo-grf = <&vo1_grf>;
+		#sound-dai-cells = <0>;
 		status = "disabled";
 
 		ports {
