diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.11/arch/arm64/boot/dts/rockchip/rk356x.dtsi linux-6.12.11/arch/arm64/boot/dts/rockchip/rk356x.dtsi
--- linux-6.12.11/arch/arm64/boot/dts/rockchip/rk356x.dtsi	2025-01-29 15:13:34.801138187 +0100
+++ linux-6.12.11/arch/arm64/boot/dts/rockchip/rk356x.dtsi	2025-01-29 15:04:22.464456641 +0100
@@ -294,6 +294,21 @@
 		};
 	};
 
+	system_sram: sram@fdcc0000 {
+		compatible = "mmio-sram";
+		reg = <0x0 0xfdcc0000 0x0 0xb000>;
+		ranges = <0x0 0x0 0xfdcc0000 0xb000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/* start address and size should be 4k algin */
+		vdec_sram: vdec-sram@0 {
+			compatible = "mmio-sram";
+			reg = <0x0 0xb000>;
+			pool;
+		};
+	};
+
 	sata1: sata@fc400000 {
 		compatible = "rockchip,rk3568-dwc-ahci", "snps,dwc-ahci";
 		reg = <0 0xfc400000 0 0x1000>;
@@ -702,6 +717,46 @@
 		#iommu-cells = <0>;
 	};
 
+	vdec: video-decoder@ffdf80200 {
+		compatible = "rockchip,rk3568-vdec";
+		//rk3588 has also cache in reg/reg-names but rk3566 bsp don't ...
+		reg = <0x0 0xfdf80100 0x0 0x100>, <0x0 0xfdf80200 0x0 0x400>;
+		reg-names = "link", "function";
+		interrupts = <GIC_SPI 91 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "irq_vdec";
+		clocks = <&cru ACLK_RKVDEC>, <&cru HCLK_RKVDEC>,
+		         <&cru CLK_RKVDEC_CA>, <&cru CLK_RKVDEC_CORE>,
+		         <&cru CLK_RKVDEC_HEVC_CA>;
+		clock-names = "axi", "ahb",
+		              "cabac", "core",
+		              "hevc_cabac";
+		assigned-clocks = <&cru ACLK_RKVDEC>, <&cru HCLK_RKVDEC>,
+		                  <&cru CLK_RKVDEC_CA>, <&cru CLK_RKVDEC_CORE>,
+		                  <&cru CLK_RKVDEC_HEVC_CA>;
+		assigned-clock-rates = <396000000>, <0>, <396000000>,
+		                       <396000000>, <600000000>;
+		resets = <&cru SRST_A_RKVDEC>, <&cru SRST_H_RKVDEC>,
+		         <&cru SRST_RKVDEC_CA>, <&cru SRST_RKVDEC_CORE>,
+		         <&cru SRST_RKVDEC_HEVC_CA>;
+		reset-names = "axi", "ahb",
+		              "cabac", "core",
+		              "hevc_cabac";
+		power-domains = <&power RK3568_PD_RKVDEC>;
+		iommus = <&vdec_mmu>;
+		sram = <&vdec_sram>;
+	};
+
+	vdec_mmu: iommu@fdf80800 {
+		compatible = "rockchip,rk3568-iommu";
+		reg = <0x0 0xfdf80800 0x0 0x40>, <0x0 0xfdf80840 0x0 0x40>;
+		interrupts = <GIC_SPI 92 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "rkvdec_mmu";
+		clocks = <&cru ACLK_RKVDEC>, <&cru HCLK_RKVDEC>;
+		clock-names = "aclk", "iface";
+		#iommu-cells = <0>;
+		power-domains = <&power RK3568_PD_RKVDEC>;
+	};
+
 	rga: rga@fdeb0000 {
 		compatible = "rockchip,rk3568-rga", "rockchip,rk3288-rga";
 		reg = <0x0 0xfdeb0000 0x0 0x180>;
