diff --speed-large-files --no-dereference --minimal -Naur linux-6.11.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts linux-6.11.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts
--- linux-6.11.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts	2024-10-04 16:38:57.000000000 +0200
+++ linux-6.11.2/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts	2024-10-05 13:19:40.240492859 +0200
@@ -9,6 +9,7 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/leds/common.h>
+#include <dt-bindings/soc/rockchip,vop2.h>
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/pwm/pwm.h>
 #include "dt-bindings/usb/pd.h"
@@ -19,8 +20,8 @@
 	compatible = "radxa,rock-5-itx", "rockchip,rk3588";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 		mmc2 = &sdio;
 	};
 
@@ -56,6 +57,16 @@
 			  "Headphone", "Headphones";
 	};
 
+//	hdmi1_sound: hdmi1-sound {
+//		status = "okay";
+//		compatible = "rockchip,hdmi";
+//		rockchip,mclk-fs = <128>;
+//		rockchip,card-name = "rockchip-hdmi1";
+//		rockchip,cpu = <&i2s6_8ch>;
+//		rockchip,codec = <&hdmi1>;
+//		rockchip,jack-det;
+//	};
+
 	gpio-leds {
 		compatible = "gpio-leds";
 		pinctrl-names = "default";
@@ -63,7 +74,7 @@
 
 		power-led1 {
 			gpios = <&gpio0 RK_PB7 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "default-on";
+			linux,default-trigger = "disk-activity";
 		};
 
 		hdd-led2 {
@@ -75,7 +86,7 @@
 	fan0: pwm-fan {
 		compatible = "pwm-fan";
 		#cooling-cells = <2>;
-		cooling-levels = <0 64 128 192 255>;
+		cooling-levels = <0 96 128 164 196 224 255>;
 		fan-supply = <&vcc12v_dcin>;
 		pwms = <&pwm14 0 10000 0>;
 	};
@@ -448,6 +459,10 @@
 	};
 };
 
+&i2s6_8ch {
+	status = "okay";
+};
+
 &package_thermal {
 	polling-delay = <1000>;
 
@@ -687,6 +702,7 @@
 	broken-cd;
 	bus-width = <4>;
 	cap-sdio-irq;
+	disable-wp;
 	keep-power-in-suspend;
 	max-frequency = <150000000>;
 	mmc-pwrseq = <&sdio_pwrseq>;
@@ -1175,3 +1191,40 @@
 	rockchip,dp-lane-mux = <2 3>;
 	status = "okay";
 };
+/*
+&display_subsystem {
+	clocks = <&hdptxphy_hdmi1>;
+	clock-names = "hdmi1_phy_pll";
+};
+
+&hdmi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&hdmim0_tx1_cec &hdmim0_tx1_hpd &hdmim1_tx1_scl &hdmim1_tx1_sda>;
+	status = "okay";
+};
+
+&hdmi1_in {
+	hdmi1_in_vp0: endpoint {
+		remote-endpoint = <&vp0_out_hdmi1>;
+	};
+};
+
+&hdptxphy_hdmi1 {
+	status = "okay";
+};
+
+&vop_mmu {
+	status = "okay";
+};
+
+&vop {
+	status = "okay";
+};
+
+&vp0 {
+	vp0_out_hdmi1: endpoint@ROCKCHIP_VOP2_EP_HDMI1 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI1>;
+		remote-endpoint = <&hdmi1_in_vp0>;
+	};
+};
+*/
