diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.1/arch/arm64/boot/dts/rockchip/rk3588-nanopc-t6.dtsi linux-6.12.1/arch/arm64/boot/dts/rockchip/rk3588-nanopc-t6.dtsi
--- linux-6.12.1/arch/arm64/boot/dts/rockchip/rk3588-nanopc-t6.dtsi	2024-11-22 15:30:26.000000000 +0100
+++ linux-6.12.1/arch/arm64/boot/dts/rockchip/rk3588-nanopc-t6.dtsi	2024-11-24 17:06:07.618348337 +0100
@@ -11,6 +11,7 @@
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/usb/pd.h>
+#include <dt-bindings/soc/rockchip,vop2.h>
 #include "rk3588.dtsi"
 
 / {
@@ -18,8 +19,8 @@
 	compatible = "friendlyarm,nanopc-t6", "rockchip,rk3588";
 
 	aliases {
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	adc-keys-0 {
@@ -53,7 +54,7 @@
 		sys_led: led-0 {
 			gpios = <&gpio2 RK_PB7 GPIO_ACTIVE_HIGH>;
 			label = "system-led";
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "disk-activity";
 			pinctrl-names = "default";
 			pinctrl-0 = <&sys_led_pin>;
 		};
@@ -61,12 +62,23 @@
 		usr_led: led-1 {
 			gpios = <&gpio2 RK_PC0 GPIO_ACTIVE_HIGH>;
 			label = "user-led";
+			linux,default-trigger = "heartbeat";
 			pinctrl-names = "default";
 			pinctrl-0 = <&usr_led_pin>;
 		};
 	};
 
-	sound {
+	hdmi0_sound: hdmi0-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi0";
+		rockchip,cpu = <&i2s5_8ch>;
+		rockchip,codec = <&hdmi0>;
+		rockchip,jack-det;
+	};
+
+	analog_sound {
 		compatible = "simple-audio-card";
 		pinctrl-names = "default";
 		pinctrl-0 = <&hp_det>;
@@ -485,6 +497,10 @@
 	};
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
 &pcie2x1l0 {
 	reset-gpios = <&gpio4 RK_PB3 GPIO_ACTIVE_HIGH>;
 	vpcie3v3-supply = <&vcc_3v3_pcie20>;
@@ -1039,3 +1055,37 @@
 &usb_host1_ohci {
 	status = "okay";
 };
+
+&display_subsystem {
+	clocks = <&hdptxphy_hdmi0>;
+	clock-names = "hdmi0_phy_pll";
+};
+
+&hdmi0 {
+	status = "okay";
+};
+
+&hdmi0_in {
+	hdmi0_in_vp0: endpoint {
+		remote-endpoint = <&vp0_out_hdmi0>;
+	};
+};
+
+&hdptxphy_hdmi0 {
+	status = "okay";
+};
+
+&vop_mmu {
+	status = "okay";
+};
+
+&vop {
+	status = "okay";
+};
+
+&vp0 {
+	vp0_out_hdmi0: endpoint@ROCKCHIP_VOP2_EP_HDMI0 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI0>;
+		remote-endpoint = <&hdmi0_in_vp0>;
+	};
+};
