diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-nanopc-t6.dtsi linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-nanopc-t6.dtsi
--- linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-nanopc-t6.dtsi	2024-12-09 15:39:00.763923887 +0100
+++ linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-nanopc-t6.dtsi	2024-12-09 15:36:12.672962569 +0100
@@ -70,6 +70,28 @@
 		};
 	};
 
+	hdmi0-con {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi0_con_in: endpoint {
+				remote-endpoint = <&hdmi0_out_con>;
+			};
+		};
+	};
+
+	hdmi1-con {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi1_con_in: endpoint {
+				remote-endpoint = <&hdmi1_out_con>;
+			};
+		};
+	};
+
 	hdmi0_sound: hdmi0-sound {
 		status = "okay";
 		compatible = "rockchip,hdmi";
@@ -80,6 +102,16 @@
 		rockchip,jack-det;
 	};
 
+	hdmi1_sound: hdmi1-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi1";
+		rockchip,cpu = <&i2s6_8ch>;
+		rockchip,codec = <&hdmi1>;
+		rockchip,jack-det;
+	};
+
 	analog_sound {
 		compatible = "simple-audio-card";
 		pinctrl-names = "default";
@@ -525,6 +557,10 @@
 	status = "okay";
 };
 
+&i2s6_8ch {
+	status = "okay";
+};
+
 &pcie2x1l0 {
 	reset-gpios = <&gpio4 RK_PB3 GPIO_ACTIVE_HIGH>;
 	vpcie3v3-supply = <&vcc_3v3_pcie20>;
@@ -1140,6 +1176,12 @@
 	};
 };
 
+&hdmi0_out {
+	hdmi0_out_con: endpoint {
+		remote-endpoint = <&hdmi0_con_in>;
+	};
+};
+
 &hdptxphy_hdmi0 {
 	status = "okay";
 };
@@ -1158,3 +1200,30 @@
 		remote-endpoint = <&hdmi0_in_vp0>;
 	};
 };
+
+&hdmi1 {
+	status = "okay";
+};
+
+&hdmi1_in {
+	hdmi1_in_vp1: endpoint {
+		remote-endpoint = <&vp1_out_hdmi1>;
+	};
+};
+
+&hdmi1_out {
+	hdmi1_out_con: endpoint {
+		remote-endpoint = <&hdmi1_con_in>;
+	};
+};
+
+&hdptxphy1 {
+	status = "okay";
+};
+
+&vp1 {
+	vp1_out_hdmi1: endpoint@ROCKCHIP_VOP2_EP_HDMI1 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI1>;
+		remote-endpoint = <&hdmi1_in_vp1>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-max.dts linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-max.dts
--- linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-max.dts	2024-12-09 15:39:00.733923886 +0100
+++ linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-max.dts	2024-12-09 15:36:19.332823081 +0100
@@ -43,6 +43,26 @@
 		enable-gpios = <&gpio4 RK_PB0 GPIO_ACTIVE_HIGH>;
 	};
 
+	hdmi0_sound: hdmi0-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi0";
+		rockchip,cpu = <&i2s5_8ch>;
+		rockchip,codec = <&hdmi0>;
+		rockchip,jack-det;
+	};
+
+	hdmi1_sound: hdmi1-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi1";
+		rockchip,cpu = <&i2s6_8ch>;
+		rockchip,codec = <&hdmi1>;
+		rockchip,jack-det;
+	};
+
 	analog-sound {
 		compatible = "simple-audio-card";
 		pinctrl-names = "default";
@@ -98,6 +118,17 @@
 		};
 	};
 
+	hdmi1-con {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi1_con_in: endpoint {
+				remote-endpoint = <&hdmi1_out_con>;
+			};
+		};
+	};
+
 	pwm-leds {
 		compatible = "pwm-leds";
 
@@ -359,6 +390,14 @@
 	status = "okay";
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
+&i2s6_8ch {
+	status = "okay";
+};
+
 /* phy2 */
 &pcie2x1l1 {
 	reset-gpios = <&gpio4 RK_PD4 GPIO_ACTIVE_HIGH>;
@@ -885,3 +924,30 @@
 		remote-endpoint = <&hdmi0_in_vp0>;
 	};
 };
+
+&hdmi1 {
+	status = "okay";
+};
+
+&hdmi1_in {
+	hdmi1_in_vp1: endpoint {
+		remote-endpoint = <&vp1_out_hdmi1>;
+	};
+};
+
+&hdmi1_out {
+	hdmi1_out_con: endpoint {
+		remote-endpoint = <&hdmi1_con_in>;
+	};
+};
+
+&hdptxphy1 {
+	status = "okay";
+};
+
+&vp1 {
+	vp1_out_hdmi1: endpoint@ROCKCHIP_VOP2_EP_HDMI1 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI1>;
+		remote-endpoint = <&hdmi1_in_vp1>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
--- linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-12-09 15:39:00.717257219 +0100
+++ linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2024-12-09 15:36:15.824732543 +0100
@@ -114,6 +114,28 @@
 		shutdown-gpios = <&gpio0 RK_PC4 GPIO_ACTIVE_HIGH>;
 	};
 
+	hdmi0-con {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi0_con_in: endpoint {
+				remote-endpoint = <&hdmi0_out_con>;
+			};
+		};
+	};
+
+	hdmi1-con {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi1_con_in: endpoint {
+				remote-endpoint = <&hdmi1_out_con>;
+			};
+		};
+	};
+
 	sound {
 		compatible = "simple-audio-card";
 		pinctrl-names = "default";
@@ -177,6 +199,16 @@
 		rockchip,jack-det;
 	};
 
+	hdmi1_sound: hdmi1-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi1";
+		rockchip,cpu = <&i2s6_8ch>;
+		rockchip,codec = <&hdmi1>;
+		rockchip,jack-det;
+	};
+
 	vbus_typec: vbus-typec-regulator {
 		compatible = "regulator-fixed";
 		enable-active-high;
@@ -448,6 +480,10 @@
 	status = "okay";
 };
 
+&i2s6_8ch {
+	status = "okay";
+};
+
 /* phy1 - M.KEY socket */
 &pcie2x1l0 {
 	reset-gpios = <&gpio4 RK_PA5 GPIO_ACTIVE_HIGH>;
@@ -1057,6 +1093,12 @@
 	};
 };
 
+&hdmi0_out {
+	hdmi0_out_con: endpoint {
+		remote-endpoint = <&hdmi0_con_in>;
+	};
+};
+
 &hdptxphy_hdmi0 {
 	status = "okay";
 };
@@ -1080,3 +1122,30 @@
 		remote-endpoint = <&hdmi0_in_vp0>;
 	};
 };
+
+&hdmi1 {
+	status = "okay";
+};
+
+&hdmi1_in {
+	hdmi1_in_vp1: endpoint {
+		remote-endpoint = <&vp1_out_hdmi1>;
+	};
+};
+
+&hdmi1_out {
+	hdmi1_out_con: endpoint {
+		remote-endpoint = <&hdmi1_con_in>;
+	};
+};
+
+&hdptxphy1 {
+	status = "okay";
+};
+
+&vp1 {
+	vp1_out_hdmi1: endpoint@ROCKCHIP_VOP2_EP_HDMI1 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI1>;
+		remote-endpoint = <&hdmi1_in_vp1>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
--- linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2024-12-09 15:39:00.610590550 +0100
+++ linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2024-12-09 15:36:27.226434975 +0100
@@ -39,6 +39,28 @@
 		pinctrl-0 = <&hp_detect>;
 	};
 
+	hdmi0-con {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi0_con_in: endpoint {
+				remote-endpoint = <&hdmi0_out_con>;
+			};
+		};
+	};
+
+	hdmi1-con {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi1_con_in: endpoint {
+				remote-endpoint = <&hdmi1_out_con>;
+			};
+		};
+	};
+
 	hdmi0_sound: hdmi0-sound {
 		status = "okay";
 		compatible = "rockchip,hdmi";
@@ -49,6 +71,16 @@
 		rockchip,jack-det;
 	};
 
+	hdmi1_sound: hdmi1-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi1";
+		rockchip,cpu = <&i2s6_8ch>;
+		rockchip,codec = <&hdmi1>;
+		rockchip,jack-det;
+	};
+
 	leds {
 		compatible = "gpio-leds";
 		pinctrl-names = "default";
@@ -375,6 +407,10 @@
 	status = "okay";
 };
 
+&i2s6_8ch {
+	status = "okay";
+};
+
 &package_thermal {
 	polling-delay = <1000>;
 
@@ -1013,6 +1049,12 @@
 	};
 };
 
+&hdmi0_out {
+	hdmi0_out_con: endpoint {
+		remote-endpoint = <&hdmi0_con_in>;
+	};
+};
+
 &hdptxphy_hdmi0 {
 	status = "okay";
 };
@@ -1031,3 +1073,32 @@
 		remote-endpoint = <&hdmi0_in_vp0>;
 	};
 };
+
+&hdmi1 {
+	pinctrl-0 = <&hdmim0_tx1_cec &hdmim0_tx1_hpd
+		     &hdmim1_tx1_scl &hdmim1_tx1_sda>;
+	status = "okay";
+};
+
+&hdmi1_in {
+	hdmi1_in_vp1: endpoint {
+		remote-endpoint = <&vp1_out_hdmi1>;
+	};
+};
+
+&hdmi1_out {
+	hdmi1_out_con: endpoint {
+		remote-endpoint = <&hdmi1_con_in>;
+	};
+};
+
+&hdptxphy1 {
+	status = "okay";
+};
+
+&vp1 {
+	vp1_out_hdmi1: endpoint@ROCKCHIP_VOP2_EP_HDMI1 {
+		reg = <ROCKCHIP_VOP2_EP_HDMI1>;
+		remote-endpoint = <&hdmi1_in_vp1>;
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts
--- linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts	2024-12-09 15:39:00.627257217 +0100
+++ linux-6.12.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5-itx.dts	2024-12-09 15:36:23.022127737 +0100
@@ -57,6 +57,17 @@
 			  "Headphone", "Headphones";
 	};
 
+	hdmi1-con {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi1_con_in: endpoint {
+				remote-endpoint = <&hdmi1_out_con>;
+			};
+		};
+	};
+
 	hdmi1_sound: hdmi1-sound {
 		status = "okay";
 		compatible = "rockchip,hdmi";
