From bddc3873ca41bf2d553488e088ceeee281227a42 Mon Sep 17 00:00:00 2001
From: Andre Przywara <andre.przywara@arm.com>
Date: Tue, 1 Oct 2024 01:41:50 +0100
Subject: [PATCH 07/37] disable unused cores debug

---
 drivers/clk/clk.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/clk/clk.c b/drivers/clk/clk.c
index d02451f951c..5390ff9ec85 100644
--- a/drivers/clk/clk.c
+++ b/drivers/clk/clk.c
@@ -1506,6 +1506,8 @@ static void __init clk_disable_unused_subtree(struct clk_core *core)
 	 * back to .disable
 	 */
 	if (clk_core_is_enabled(core)) {
+		dev_info(core->dev, "Powering off unused clock %s\n",
+			 core->name);
 		trace_clk_disable(core);
 		if (core->ops->disable_unused)
 			core->ops->disable_unused(core->hw);
-- 
2.46.0

