diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.8/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi linux-6.12.8/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
--- linux-6.12.8/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi	2025-01-08 11:36:21.714573325 +0100
+++ linux-6.12.8/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi	2025-01-08 11:08:58.677862567 +0100
@@ -163,6 +163,25 @@
 			#size-cells = <0>;
 		};
 
+		mmc1: mmc@4021000 {
+			compatible = "allwinner,sun55i-a523-mmc",
+				     "allwinner,sun20i-d1-mmc";
+			reg = <0x04021000 0x1000>;
+			clocks = <&ccu CLK_BUS_MMC1>, <&ccu CLK_MMC1>;
+			clock-names = "ahb", "mmc";
+			resets = <&ccu RST_BUS_MMC1>;
+			reset-names = "ahb";
+			interrupts = <GIC_SPI 41 IRQ_TYPE_LEVEL_HIGH>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&mmc1_pins>;
+			status = "disabled";
+
+			max-frequency = <150000000>;
+			cap-sdio-irq;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+
 		mmc2: mmc@4022000 {
 			compatible = "allwinner,sun55i-a523-mmc",
 				     "allwinner,sun20i-d1-mmc";
