diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.4/arch/arm64/boot/dts/allwinner/sun55i-h728-x96q-pro-plus.dts linux-6.12.4/arch/arm64/boot/dts/allwinner/sun55i-h728-x96q-pro-plus.dts
--- linux-6.12.4/arch/arm64/boot/dts/allwinner/sun55i-h728-x96q-pro-plus.dts	2024-12-18 16:00:44.520522855 +0100
+++ linux-6.12.4/arch/arm64/boot/dts/allwinner/sun55i-h728-x96q-pro-plus.dts	2024-12-18 16:48:49.613932120 +0100
@@ -279,3 +279,22 @@
 &usbphy {
 	status = "okay";
 };
+
+&gmac1 {
+	phy-mode = "rgmii";
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&gmac1_pins_default>;
+	pinctrl-1 = <&gmac1_pins_sleep>;
+	aw,soc-phy25m;
+	tx-delay = <2>;
+	rx-delay = <5>;
+	dwmac3v3-supply = <&reg_cldo3>;
+	//in vendor dt it is clod4 phy3v3-supply = <&reg_cldo4>;
+	status = "okay";
+
+	mdio1: mdio1@1 {
+		gmac1_phy0: ethernet-phy@1 {
+			reset-gpios = <&pio 9 16 GPIO_ACTIVE_LOW>; /* PJ16 */
+		};
+	};
+};
