diff -Naur urlgrabber-3.9.1-old/setup.py urlgrabber-3.9.1-new/setup.py
--- urlgrabber-3.9.1-old/setup.py	2009-02-23 13:37:16.000000000 -0800
+++ urlgrabber-3.9.1-new/setup.py	2011-03-03 15:17:20.000000000 -0800
@@ -41,5 +41,5 @@
     #print '%-20s -> %s' % (k, config[k])
     if k.startswith('_'): del config[k]
 
-  from distutils.core import setup
+  from setuptools import setup
   setup(**config)
