diff -Naur glibc-2.15-old/Makefile.in glibc-2.15-new/Makefile.in
--- glibc-2.15-old/Makefile.in	2012-03-22 08:39:54.000000000 -0700
+++ glibc-2.15-new/Makefile.in	2012-03-22 08:40:46.000000000 -0700
@@ -6,6 +6,9 @@
 all .DEFAULT:
 	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" -C $(srcdir) objdir=`pwd` $@
 
+lib:
+	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" CVSOPTS="$(CVSOPTS)" -C $(srcdir) objdir=`pwd` $@
+
 install:
 	LANGUAGE=C LC_ALL=C; export LANGUAGE LC_ALL; \
 	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" -C $(srcdir) objdir=`pwd` $@
