diff -Naur glibc-2.15-old/Makefile.in glibc-2.15-new/Makefile.in
--- glibc-2.15-old/Makefile.in	2012-03-22 08:41:21.000000000 -0700
+++ glibc-2.15-new/Makefile.in	2012-03-22 08:42:14.000000000 -0700
@@ -12,6 +12,9 @@
 locale: lib
 	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" CVSOPTS="$(CVSOPTS)" -C $(srcdir)/locale objdir=`pwd`
 
+sunrpc: lib
+	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" CVSOPTS="$(CVSOPTS)" -C $(srcdir)/sunrpc objdir=`pwd`
+
 install:
 	LANGUAGE=C LC_ALL=C; export LANGUAGE LC_ALL; \
 	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" -C $(srcdir) objdir=`pwd` $@
