diff -Naur glibc-2.17-old/sunrpc/Makefile glibc-2.17-new/sunrpc/Makefile
--- glibc-2.17-old/sunrpc/Makefile	2012-12-24 19:02:13.000000000 -0800
+++ glibc-2.17-new/sunrpc/Makefile	2013-01-04 10:45:33.000000000 -0800
@@ -52,7 +52,7 @@
 				    des_crypt.h)
 headers-not-in-tirpc = $(addprefix rpc/,key_prot.h rpc_des.h) \
 		       $(rpcsvc:%=rpcsvc/%) rpcsvc/bootparam.h
-headers = rpc/netdb.h
+headers = rpc/key_prot.h rpc/netdb.h
 install-others = $(inst_sysconfdir)/rpc
 generated = $(rpcsvc:%.x=rpcsvc/%.h) $(rpcsvc:%.x=x%.c) $(rpcsvc:%.x=x%.stmp) \
 	    $(rpcsvc:%.x=rpcsvc/%.stmp) rpcgen
