diff -Naur glibc-2.17-old/configure glibc-2.17-new/configure
--- glibc-2.17-old/configure	2013-01-04 10:48:58.000000000 -0800
+++ glibc-2.17-new/configure	2013-01-04 10:49:51.000000000 -0800
@@ -4680,11 +4680,11 @@
     path_binutils=`(cd $path_binutils; pwd) | sed 's%/*$%/%'`
     CC="$CC -B$path_binutils"
 fi
-AS=`$CC -print-prog-name=as`
-LD=`$CC -print-prog-name=ld`
-AR=`$CC -print-prog-name=ar`
-
-OBJDUMP=`$CC -print-prog-name=objdump`
+AS=${AS}
+LD=${LD}
+AR=${AR}
+  
+OBJDUMP=${OBJDUMP}
 
 OBJCOPY=`$CC -print-prog-name=objcopy`
 
