diff -Naur glibc-2.31-old/configure glibc-2.31-new/configure
--- glibc-2.31-old/configure	2020-05-09 16:31:10.223333333 +0200
+++ glibc-2.31-new/configure	2020-05-09 14:43:32.003333328 +0200
@@ -5864,7 +5864,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $libc_cv_as_noexecstack" >&5
 $as_echo "$libc_cv_as_noexecstack" >&6; }
 if test $libc_cv_as_noexecstack = yes; then
-  ASFLAGS_config="$ASFLAGS_config -Wa,--noexecstack"
+  ASFLAGS_config="$ASFLAGS_config -Wa,--noexecstack $CFLAGS"
 fi
 
 
