diff -Naur glibc-2.31-old/configure glibc-2.31-new/configure
--- glibc-2.31-old/configure	2020-04-13 13:32:00.809999060 +0200
+++ glibc-2.31-new/configure	2020-04-13 13:33:48.439999043 +0200
@@ -4292,7 +4291,9 @@
   as_fn_error $? "The $machine is not supported." "$LINENO" 5
 fi
 if test -z "$submachine_used" && test -n "$submachine"; then
-  as_fn_error $? "The $submachine subspecies of $host_cpu is not supported." "$LINENO" 5
+  echo "submachine_used:$submachine_used"
+  echo "submachine:$submachine"
+#  as_fn_error $? "The $submachine subspecies of $host_cpu is not supported." "$LINENO" 5
 fi
 
 
