diff -Naur glibc-2.12.2-old/Makefile.in glibc-2.12.2-new/Makefile.in
--- glibc-2.12.2-old/Makefile.in	2011-07-15 12:35:45.000000000 -0700
+++ glibc-2.12.2-new/Makefile.in	2011-07-15 12:36:32.000000000 -0700
@@ -14,6 +14,9 @@
 lib:
 	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" CVSOPTS="$(CVSOPTS)" -C $(srcdir) objdir=`pwd` $@
 
+locale: lib
+	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" CVSOPTS="$(CVSOPTS)" -C $(srcdir)/locale objdir=`pwd`
+
 install:
 	LANGUAGE=C LC_ALL=C; export LANGUAGE LC_ALL; \
 	$(MAKE) -r PARALLELMFLAGS="$(PARALLELMFLAGS)" CVSOPTS="$(CVSOPTS)" -C $(srcdir) objdir=`pwd` $@
