diff -Naur glibc-2.15-old/sunrpc/Makefile glibc-2.15-new/sunrpc/Makefile
--- glibc-2.15-old/sunrpc/Makefile	2012-01-01 04:16:32.000000000 -0800
+++ glibc-2.15-new/sunrpc/Makefile	2012-03-22 08:36:40.000000000 -0700
@@ -53,7 +53,7 @@
 				    des_crypt.h)
 headers-not-in-tirpc = $(addprefix rpc/,key_prot.h rpc_des.h) \
 		       $(rpcsvc:%=rpcsvc/%) rpcsvc/bootparam.h
-headers = rpc/netdb.h
+headers = rpc/key_prot.h rpc/netdb.h
 install-others = $(inst_sysconfdir)/rpc
 generated = $(rpcsvc:%.x=rpcsvc/%.h) $(rpcsvc:%.x=x%.c) $(rpcsvc:%.x=x%.stmp) \
 	    $(rpcsvc:%.x=rpcsvc/%.stmp) rpcgen
