diff --speed-large-files --no-dereference --minimal -Naur mc-4.8.33/Makefile.am mc-4.8.33/Makefile.am
--- mc-4.8.33/Makefile.am	2025-01-23 10:33:38.000000000 +0100
+++ mc-4.8.33/Makefile.am	2025-04-10 17:02:13.506781361 +0200
@@ -1,7 +1,7 @@
 ## Process this file with automake to create Makefile.in.
 AUTOMAKE_OPTIONS = 1.14
 
-SUBDIRS = po lib src doc contrib misc
+SUBDIRS = lib src doc contrib misc
 
 if HAVE_TESTS
     SUBDIRS += tests
