
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/base/task/thread_pool.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/base/task/thread_pool.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/base/task/thread_pool.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/base/task/thread_pool.h	2025-05-09 16:24:40.931103661 +0200
@@ -15,9 +15,9 @@
 // dependency. Some of the includes below result in an extended version of this
 // circular dependency. These forward-declarations are temporarily required for
 // the duration of the migration.
-enum class TaskPriority : uint8_t;
-enum class TaskShutdownBehavior : uint8_t;
-enum class ThreadPolicy : uint8_t;
+//enum class TaskPriority : uint8_t;
+//enum class TaskShutdownBehavior : uint8_t;
+//enum class ThreadPolicy : uint8_t;
 struct MayBlock;
 struct WithBaseSyncPrimitives;
 class TaskTraits;
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/gpu/command_buffer/common/skia_utils.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/gpu/command_buffer/common/skia_utils.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/gpu/command_buffer/common/skia_utils.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/gpu/command_buffer/common/skia_utils.h	2025-05-08 20:53:32.674492939 +0200
@@ -5,6 +5,7 @@
 #ifndef GPU_COMMAND_BUFFER_COMMON_SKIA_UTILS_H_
 #define GPU_COMMAND_BUFFER_COMMON_SKIA_UTILS_H_
 
+#include <cstdint>
 #include <memory>
 
 #include "base/optional.h"
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h	2025-05-08 20:53:32.674492939 +0200
@@ -5,6 +5,7 @@
 #ifndef GPU_COMMAND_BUFFER_SERVICE_GPU_FENCE_MANAGER_H_
 #define GPU_COMMAND_BUFFER_SERVICE_GPU_FENCE_MANAGER_H_
 
+#include <cstdint>
 #include <memory>
 #include <vector>
 

diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/net/tools/huffman_trie/trie_entry.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/net/tools/huffman_trie/trie_entry.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/net/tools/huffman_trie/trie_entry.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/net/tools/huffman_trie/trie_entry.h	2025-05-09 16:31:27.727761429 +0200
@@ -8,6 +8,7 @@
 #include <memory>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 namespace net {
 
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fpdfapi/page/cpdf_function.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fpdfapi/page/cpdf_function.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fpdfapi/page/cpdf_function.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fpdfapi/page/cpdf_function.h	2025-05-08 20:53:42.311159398 +0200
@@ -7,6 +7,7 @@
 #ifndef CORE_FPDFAPI_PAGE_CPDF_FUNCTION_H_
 #define CORE_FPDFAPI_PAGE_CPDF_FUNCTION_H_
 
+#include <cstdint>
 #include <memory>
 #include <set>
 #include <vector>
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fxcodec/jbig2/JBig2_DocumentContext.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fxcodec/jbig2/JBig2_DocumentContext.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fxcodec/jbig2/JBig2_DocumentContext.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fxcodec/jbig2/JBig2_DocumentContext.h	2025-05-08 20:53:42.311159398 +0200
@@ -7,6 +7,7 @@
 #ifndef CORE_FXCODEC_JBIG2_JBIG2_DOCUMENTCONTEXT_H_
 #define CORE_FXCODEC_JBIG2_JBIG2_DOCUMENTCONTEXT_H_
 
+#include <cstdint>
 #include <list>
 #include <memory>
 #include <utility>
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fxcodec/jpeg/jpegmodule.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fxcodec/jpeg/jpegmodule.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fxcodec/jpeg/jpegmodule.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/core/fxcodec/jpeg/jpegmodule.h	2025-05-08 20:53:42.311159398 +0200
@@ -7,6 +7,7 @@
 #ifndef CORE_FXCODEC_JPEG_JPEGMODULE_H_
 #define CORE_FXCODEC_JPEG_JPEGMODULE_H_
 
+#include <cstdint>
 #include <memory>
 
 #include "build/build_config.h"
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/third_party/base/span.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/third_party/base/span.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/third_party/base/span.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/pdfium/third_party/base/span.h	2025-05-08 20:53:42.311159398 +0200
@@ -9,6 +9,7 @@
 
 #include <algorithm>
 #include <array>
+#include <cstdint>
 #include <iterator>
 #include <type_traits>
 #include <utility>
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/tracing/core/slice.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/tracing/core/slice.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/tracing/core/slice.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/tracing/core/slice.h	2025-05-08 20:53:27.951159705 +0200
@@ -20,6 +20,7 @@
 #include <stddef.h>
 #include <string.h>
 
+#include <cstdint>
 #include <memory>
 #include <string>
 #include <vector>
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/tracing_backend.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/tracing_backend.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/tracing_backend.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/include/perfetto/tracing/tracing_backend.h	2025-05-08 20:53:27.951159705 +0200
@@ -17,6 +17,7 @@
 #ifndef INCLUDE_PERFETTO_TRACING_TRACING_BACKEND_H_
 #define INCLUDE_PERFETTO_TRACING_TRACING_BACKEND_H_
 
+#include <cstdint>
 #include <memory>
 #include <string>
 
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/gzip/gzip_utils.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/gzip/gzip_utils.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/gzip/gzip_utils.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/gzip/gzip_utils.h	2025-05-08 20:53:23.284493138 +0200
@@ -17,6 +17,7 @@
 #ifndef SRC_TRACE_PROCESSOR_IMPORTERS_GZIP_GZIP_UTILS_H_
 #define SRC_TRACE_PROCESSOR_IMPORTERS_GZIP_GZIP_UTILS_H_
 
+#include <cstdint>
 #include <memory>
 
 struct z_stream_s;
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/fec_controller.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/fec_controller.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/fec_controller.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/fec_controller.h	2025-05-08 20:53:17.741159921 +0200
@@ -11,6 +11,7 @@
 #ifndef API_FEC_CONTROLLER_H_
 #define API_FEC_CONTROLLER_H_
 
+#include <cstdint>
 #include <memory>
 #include <vector>
 
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/network_state_predictor.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/network_state_predictor.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/network_state_predictor.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/network_state_predictor.h	2025-05-08 20:53:17.741159921 +0200
@@ -11,6 +11,7 @@
 #ifndef API_NETWORK_STATE_PREDICTOR_H_
 #define API_NETWORK_STATE_PREDICTOR_H_
 
+#include <cstdint>
 #include <memory>
 #include <vector>
 
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/task_queue/task_queue_base.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/task_queue/task_queue_base.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/task_queue/task_queue_base.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/api/task_queue/task_queue_base.h	2025-05-08 20:53:17.741159921 +0200
@@ -10,6 +10,7 @@
 #ifndef API_TASK_QUEUE_TASK_QUEUE_BASE_H_
 #define API_TASK_QUEUE_TASK_QUEUE_BASE_H_
 
+#include <cstdint>
 #include <memory>
 
 #include "api/task_queue/queued_task.h"
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/common_video/h264/pps_parser.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/common_video/h264/pps_parser.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/common_video/h264/pps_parser.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/common_video/h264/pps_parser.h	2025-05-08 20:53:17.741159921 +0200
@@ -11,6 +11,7 @@
 #ifndef COMMON_VIDEO_H264_PPS_PARSER_H_
 #define COMMON_VIDEO_H264_PPS_PARSER_H_
 
+#include <cstdint>
 #include "absl/types/optional.h"
 
 namespace rtc {
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/common_video/h264/sps_parser.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/common_video/h264/sps_parser.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/common_video/h264/sps_parser.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/common_video/h264/sps_parser.h	2025-05-08 20:53:17.741159921 +0200
@@ -11,6 +11,7 @@
 #ifndef COMMON_VIDEO_H264_SPS_PARSER_H_
 #define COMMON_VIDEO_H264_SPS_PARSER_H_
 
+#include <cstdint>
 #include "absl/types/optional.h"
 
 namespace rtc {
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/include/module_common_types_public.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/include/module_common_types_public.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/include/module_common_types_public.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/include/module_common_types_public.h	2025-05-08 20:53:17.741159921 +0200
@@ -11,6 +11,7 @@
 #ifndef MODULES_INCLUDE_MODULE_COMMON_TYPES_PUBLIC_H_
 #define MODULES_INCLUDE_MODULE_COMMON_TYPES_PUBLIC_H_
 
+#include <cstdint>
 #include <limits>
 
 #include "absl/types/optional.h"
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/rtp_rtcp/source/receive_statistics_impl.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/rtp_rtcp/source/receive_statistics_impl.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/rtp_rtcp/source/receive_statistics_impl.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/rtp_rtcp/source/receive_statistics_impl.h	2025-05-08 20:53:17.741159921 +0200
@@ -12,6 +12,7 @@
 #define MODULES_RTP_RTCP_SOURCE_RECEIVE_STATISTICS_IMPL_H_
 
 #include <algorithm>
+#include <cstdint>
 #include <map>
 #include <vector>
 
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/video_coding/timestamp_map.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/video_coding/timestamp_map.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/video_coding/timestamp_map.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/modules/video_coding/timestamp_map.h	2025-05-08 20:53:17.741159921 +0200
@@ -11,6 +11,7 @@
 #ifndef MODULES_VIDEO_CODING_TIMESTAMP_MAP_H_
 #define MODULES_VIDEO_CODING_TIMESTAMP_MAP_H_
 
+#include <cstdint>
 #include <memory>
 
 namespace webrtc {
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/video/stats_counter.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/video/stats_counter.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/video/stats_counter.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/webrtc/video/stats_counter.h	2025-05-08 20:53:17.741159921 +0200
@@ -11,6 +11,7 @@
 #ifndef VIDEO_STATS_COUNTER_H_
 #define VIDEO_STATS_COUNTER_H_
 
+#include <cstdint>
 #include <memory>
 #include <string>
 
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/woff2/include/woff2/output.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/woff2/include/woff2/output.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/woff2/include/woff2/output.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/third_party/woff2/include/woff2/output.h	2025-05-08 20:53:37.497826171 +0200
@@ -10,6 +10,7 @@
 #define WOFF2_WOFF2_OUT_H_
 
 #include <algorithm>
+#include <cstdint>
 #include <cstring>
 #include <memory>
 #include <string>
