diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/bookmarks/browser/base_bookmark_model_observer.h qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/bookmarks/browser/base_bookmark_model_observer.h
--- qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/bookmarks/browser/base_bookmark_model_observer.h	2021-02-19 17:41:59.000000000 +0100
+++ qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/bookmarks/browser/base_bookmark_model_observer.h	2022-03-20 20:47:11.353541629 +0100
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_BOOKMARKS_BROWSER_BASE_BOOKMARK_MODEL_OBSERVER_H_
 #define COMPONENTS_BOOKMARKS_BROWSER_BASE_BOOKMARK_MODEL_OBSERVER_H_
 
+#include <cstddef>
 #include "base/macros.h"
 #include "components/bookmarks/browser/bookmark_model_observer.h"
 
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/bookmarks/browser/bookmark_expanded_state_tracker.cc qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/bookmarks/browser/bookmark_expanded_state_tracker.cc
--- qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/bookmarks/browser/bookmark_expanded_state_tracker.cc	2021-02-19 17:41:59.000000000 +0100
+++ qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/bookmarks/browser/bookmark_expanded_state_tracker.cc	2022-03-20 20:47:11.353541629 +0100
@@ -4,6 +4,7 @@
 
 #include "components/bookmarks/browser/bookmark_expanded_state_tracker.h"
 
+#include <cstddef>
 #include <stdint.h>
 
 #include <utility>
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/federated_learning/floc_constants.cc qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/federated_learning/floc_constants.cc
--- qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/federated_learning/floc_constants.cc	2021-02-19 17:41:59.000000000 +0100
+++ qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/components/federated_learning/floc_constants.cc	2022-03-20 20:47:11.353541629 +0100
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <limits>
 #include "components/federated_learning/floc_constants.h"
 
 namespace federated_learning {
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/net/dns/dns_server_iterator.h qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/net/dns/dns_server_iterator.h
--- qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/net/dns/dns_server_iterator.h	2021-02-19 17:41:59.000000000 +0100
+++ qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/net/dns/dns_server_iterator.h	2022-03-20 20:47:11.353541629 +0100
@@ -7,6 +7,7 @@
 
 #include <cstddef>
 #include <vector>
+#include <cstddef>
 
 #include "net/base/net_export.h"
 #include "net/dns/public/secure_dns_mode.h"
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/abseil-cpp/absl/debugging/failure_signal_handler.cc qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/abseil-cpp/absl/debugging/failure_signal_handler.cc
--- qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/abseil-cpp/absl/debugging/failure_signal_handler.cc	2021-02-19 17:41:59.000000000 +0100
+++ qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/abseil-cpp/absl/debugging/failure_signal_handler.cc	2022-03-20 20:46:30.083540267 +0100
@@ -135,7 +135,7 @@
 #else
   const size_t page_mask = sysconf(_SC_PAGESIZE) - 1;
 #endif
-  size_t stack_size = (std::max(SIGSTKSZ, 65536) + page_mask) & ~page_mask;
+  size_t stack_size = (std::max<size_t>(SIGSTKSZ, 65536) + page_mask) & ~page_mask;
 #if defined(ABSL_HAVE_ADDRESS_SANITIZER) || \
     defined(ABSL_HAVE_MEMORY_SANITIZER) || defined(ABSL_HAVE_THREAD_SANITIZER)
   // Account for sanitizer instrumentation requiring additional stack space.
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc
--- qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc	2021-02-19 17:41:59.000000000 +0100
+++ qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc	2022-03-20 20:47:11.353541629 +0100
@@ -35,6 +35,7 @@
 
 #include "absl/synchronization/internal/graphcycles.h"
 
+#include <limits>
 #include <algorithm>
 #include <array>
 #include "absl/base/internal/hide_ptr.h"
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/client/linux/handler/exception_handler.cc qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/client/linux/handler/exception_handler.cc
--- qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/client/linux/handler/exception_handler.cc	2021-02-19 17:41:59.000000000 +0100
+++ qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/client/linux/handler/exception_handler.cc	2022-03-20 20:46:19.373539913 +0100
@@ -138,7 +138,7 @@
   // SIGSTKSZ may be too small to prevent the signal handlers from overrunning
   // the alternative stack. Ensure that the size of the alternative stack is
   // large enough.
-  static const unsigned kSigStackSize = std::max(16384, SIGSTKSZ);
+  static const unsigned kSigStackSize = std::max<size_t>(16384, SIGSTKSZ);
 
   // Only set an alternative stack if there isn't already one, or if the current
   // one is too small.
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/containers/string_pool.h qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/containers/string_pool.h
--- qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/containers/string_pool.h	2021-02-19 17:41:59.000000000 +0100
+++ qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/containers/string_pool.h	2022-03-20 20:47:11.353541629 +0100
@@ -17,6 +17,7 @@
 #ifndef SRC_TRACE_PROCESSOR_CONTAINERS_STRING_POOL_H_
 #define SRC_TRACE_PROCESSOR_CONTAINERS_STRING_POOL_H_
 
+#include <limits>
 #include <stddef.h>
 #include <stdint.h>
 
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/tint/src/src/writer/hlsl/generator_impl.cc qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/tint/src/src/writer/hlsl/generator_impl.cc
--- qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/tint/src/src/writer/hlsl/generator_impl.cc	2021-02-19 17:41:59.000000000 +0100
+++ qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/tint/src/src/writer/hlsl/generator_impl.cc	2022-03-20 20:47:11.353541629 +0100
@@ -14,6 +14,7 @@
 
 #include "src/writer/hlsl/generator_impl.h"
 
+#include <limits>
 #include <sstream>
 
 #include "spirv/unified1/GLSL.std.450.h"
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/tint/src/src/writer/msl/generator_impl.cc qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/tint/src/src/writer/msl/generator_impl.cc
--- qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/tint/src/src/writer/msl/generator_impl.cc	2021-02-19 17:41:59.000000000 +0100
+++ qt-everywhere-src-5.15.3/qtwebengine/src/3rdparty/chromium/third_party/tint/src/src/writer/msl/generator_impl.cc	2022-03-20 20:47:11.353541629 +0100
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <limits>
+
 #include "src/writer/msl/generator_impl.h"
 
 #include "spirv/unified1/GLSL.std.450.h"
