diff -Naur qt-everywhere-src-5.15.2-old/qtbase/src/platformsupport/devicediscovery/qdevicediscovery_udev.cpp qt-everywhere-src-5.15.2-new/qtbase/src/platformsupport/devicediscovery/qdevicediscovery_udev.cpp
--- qt-everywhere-src-5.15.2-old/qtbase/src/platformsupport/devicediscovery/qdevicediscovery_udev.cpp	2020-10-27 09:02:11.000000000 +0100
+++ qt-everywhere-src-5.15.2-new/qtbase/src/platformsupport/devicediscovery/qdevicediscovery_udev.cpp	2021-11-29 15:54:07.132942062 +0100
@@ -146,11 +146,15 @@
             if (m_types & Device_DRM_PrimaryGPU) {
                 udev_device *pci = udev_device_get_parent_with_subsystem_devtype(udevice, "pci", 0);
                 if (pci) {
-                    if (qstrcmp(udev_device_get_sysattr_value(pci, "boot_vga"), "1") == 0)
+                    if (qstrcmp(udev_device_get_sysattr_value(pci, "boot_vga"), "1") == 0) {
+                        qCDebug(lcDD) << "Found matching DRM device:" << devices;
                         devices << candidate;
+                    }
+                } else {
+                    qCDebug(lcDD) << "Found matching DRM device:" << devices;
+                    devices << candidate;
                 }
-            } else
-                devices << candidate;
+            }
         }
 
         udev_device_unref(udevice);
