diff -Naur qt-everywhere-src-5.15.2-old/qtbase/src/corelib/global/qglobal.h qt-everywhere-src-5.15.2-new/qtbase/src/corelib/global/qglobal.h
--- qt-everywhere-src-5.15.2-old/qtbase/src/corelib/global/qglobal.h	2020-10-27 09:02:11.000000000 +0100
+++ qt-everywhere-src-5.15.2-new/qtbase/src/corelib/global/qglobal.h	2021-08-04 13:34:41.853331721 +0200
@@ -45,6 +45,7 @@
 #  include <type_traits>
 #  include <cstddef>
 #  include <utility>
+#  include <limits>
 #endif
 #ifndef __ASSEMBLER__
 #  include <assert.h>
