--- qtwayland-everywhere-src-5.13.2/qtwayland/src/hardwareintegration/compositor/linux-dmabuf-unstable-v1/linuxdmabuf.h~  2019-10-15 06:21:15.000000000 +0200
+++ qtwayland-everywhere-src-5.13.2/qtwayland/src/hardwareintegration/compositor/linux-dmabuf-unstable-v1/linuxdmabuf.h   2019-11-21 11:05:54.735760060 +0100
@@ -53,6 +53,7 @@
 
 #include <EGL/egl.h>
 #include <EGL/eglext.h>
+#include <EGL/eglmesaext.h>
 
 // compatibility with libdrm <= 2.4.74
 #ifndef DRM_FORMAT_RESERVED
 