diff -Naur qt-everywhere-src-5.10.1-old/qtbase/mkspecs/common/gcc-base.conf qt-everywhere-src-5.10.1-new/qtbase/mkspecs/common/gcc-base.conf
--- qt-everywhere-src-5.10.1-old/qtbase/mkspecs/common/gcc-base.conf	2018-02-08 19:24:48.000000000 +0100
+++ qt-everywhere-src-5.10.1-new/qtbase/mkspecs/common/gcc-base.conf	2018-06-28 15:48:29.069273214 +0200
@@ -46,7 +46,7 @@
 QMAKE_CFLAGS_SHLIB         += $$QMAKE_CFLAGS_PIC
 QMAKE_CFLAGS_STATIC_LIB    += $$QMAKE_CFLAGS_PIC
 QMAKE_CFLAGS_APP           += $$QMAKE_CFLAGS_PIC
-QMAKE_CFLAGS_ISYSTEM        = -isystem
+QMAKE_CFLAGS_ISYSTEM        = 
 QMAKE_CFLAGS_YACC          += -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_HIDESYMS      += -fvisibility=hidden
 QMAKE_CFLAGS_EXCEPTIONS_OFF += -fno-exceptions
