diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.17/qtwebengine/src/core/config/linux.pri qt-everywhere-src-5.15.17/qtwebengine/src/core/config/linux.pri
--- qt-everywhere-src-5.15.17/qtwebengine/src/core/config/linux.pri	2024-04-03 11:31:09.000000000 +0200
+++ qt-everywhere-src-5.15.17/qtwebengine/src/core/config/linux.pri	2025-05-23 09:56:16.128323204 +0200
@@ -20,6 +20,10 @@
         gn_args += use_alsa=false
     }
 
+    qtConfig(webengine-system-libevent) {
+        qtConfig(webengine-system-libevent): gn_args += use_system_libevent=true
+    }
+
     !packagesExist(libpci): gn_args += use_libpci=false
 
     qtConfig(webengine-ozone-x11) {
@@ -33,7 +37,6 @@
 
     qtConfig(webengine-webrtc): qtConfig(webengine-webrtc-pipewire): gn_args += rtc_use_pipewire=true
 
-    qtConfig(webengine-system-libevent): gn_args += use_system_libevent=true
     qtConfig(webengine-system-libwebp):  gn_args += use_system_libwebp=true
     qtConfig(webengine-system-libxml2):  gn_args += use_system_libxml=true use_system_libxslt=true
     qtConfig(webengine-system-opus):     gn_args += use_system_opus=true
