diff -Naur qt5-old/qtdeclarative/src/3rdparty/masm/stubs/wtf/Optional.h qt5-new/qtdeclarative/src/3rdparty/masm/stubs/wtf/Optional.h
--- qt5-old/qtdeclarative/src/3rdparty/masm/stubs/wtf/Optional.h	2018-11-19 15:39:18.000000000 +0100
+++ qt5-new/qtdeclarative/src/3rdparty/masm/stubs/wtf/Optional.h	2018-12-25 11:44:37.096666662 +0100
@@ -84,7 +84,7 @@
 
 #endif
 
-#if __cplusplus < 201402L && !QT_CONFIG(cxx14_make_unique)
+#if __cplusplus < 201402L && !defined(__cpp_lib_make_unique) && !defined(Q_CC_MSVC) && !defined(Q_CC_GHS)
 
 namespace std {
     template<typename T, class ...Args>
