diff -Naur qt-everywhere-opensource-src-5.4.1-old/qtwebkit/Source/JavaScriptCore/runtime/JSObject.cpp qt-everywhere-opensource-src-5.4.1-new/qtwebkit/Source/JavaScriptCore/runtime/JSObject.cpp
--- qt-everywhere-opensource-src-5.4.1-old/qtwebkit/Source/JavaScriptCore/runtime/JSObject.cpp	2015-02-17 05:57:11.000000000 +0100
+++ qt-everywhere-opensource-src-5.4.1-new/qtwebkit/Source/JavaScriptCore/runtime/JSObject.cpp	2016-02-28 10:38:12.000000000 +0100
@@ -1908,7 +1908,10 @@
         CRASH();
     }
 }
-
+// Used in JSArray.cpp so we must instantiate explicit
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<Int32Shape>(ExecState* exec, unsigned i, JSValue value);
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<DoubleShape>(ExecState* exec, unsigned i, JSValue value);
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<ContiguousShape>(ExecState* exec, unsigned i, JSValue value);
 void JSObject::putByIndexBeyondVectorLengthWithArrayStorage(ExecState* exec, unsigned i, JSValue value, bool shouldThrow, ArrayStorage* storage)
 {
     VM& vm = exec->vm();
