diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_key_util.cc qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_key_util.cc
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_key_util.cc	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_key_util.cc	2025-04-14 19:30:42.123736093 +0200
@@ -4,10 +4,10 @@
 
 #include "crypto/nss_key_util.h"
 
-#include <cryptohi.h>
-#include <keyhi.h>
-#include <pk11pub.h>
-#include <secmod.h>
+#include <nss/cryptohi.h>
+#include <nss/keyhi.h>
+#include <nss/pk11pub.h>
+#include <nss/secmod.h>
 #include <stdint.h>
 
 #include <memory>
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_key_util.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_key_util.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_key_util.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_key_util.h	2025-04-14 19:19:24.894700782 +0200
@@ -5,7 +5,7 @@
 #ifndef CRYPTO_NSS_KEY_UTIL_H_
 #define CRYPTO_NSS_KEY_UTIL_H_
 
-#include <secoidt.h>
+#include <nss/secoidt.h>
 #include <stdint.h>
 
 #include <vector>
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_util.cc qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_util.cc
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_util.cc	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_util.cc	2025-04-14 19:35:33.029711035 +0200
@@ -4,13 +4,13 @@
 
 #include "crypto/nss_util.h"
 
-#include <nss.h>
-#include <pk11pub.h>
+#include <nss/nss.h>
+#include <nss/pk11pub.h>
 #include <plarena.h>
 #include <prerror.h>
 #include <prinit.h>
 #include <prtime.h>
-#include <secmod.h>
+#include <nss/secmod.h>
 
 #include <memory>
 #include <utility>
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_util_internal.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_util_internal.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_util_internal.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/nss_util_internal.h	2025-04-14 19:34:42.227425975 +0200
@@ -5,7 +5,7 @@
 #ifndef CRYPTO_NSS_UTIL_INTERNAL_H_
 #define CRYPTO_NSS_UTIL_INTERNAL_H_
 
-#include <secmodt.h>
+#include <nss/secmodt.h>
 
 #include <string>
 
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/scoped_nss_types.h qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/scoped_nss_types.h
--- qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/scoped_nss_types.h	2023-11-09 13:02:51.000000000 +0100
+++ qt-everywhere-src-5.15.16/qtwebengine/src/3rdparty/chromium/crypto/scoped_nss_types.h	2025-04-14 19:23:21.117919517 +0200
@@ -5,10 +5,10 @@
 #ifndef CRYPTO_SCOPED_NSS_TYPES_H_
 #define CRYPTO_SCOPED_NSS_TYPES_H_
 
-#include <certt.h>
-#include <keyhi.h>
-#include <nss.h>
-#include <pk11pub.h>
+#include <nss/certt.h>
+#include <nss/keyhi.h>
+#include <nss/nss.h>
+#include <nss/pk11pub.h>
 #include <plarena.h>
 
 #include <memory>
