diff -Naur vcdimager-0.7.24-old/configure.ac vcdimager-0.7.24-new/configure.ac
--- vcdimager-0.7.24-old/configure.ac	2011-03-17 16:43:53.000000000 -0700
+++ vcdimager-0.7.24-new/configure.ac	2012-12-30 15:30:02.000000000 -0800
@@ -31,8 +31,8 @@
 AC_SUBST(SPEC_VERSION)
 
 dnl AM_ACLOCAL_INCLUDE(.)
-dnl AM_CONFIG_HEADER(include/libvcd/acconfig.h)
-AM_CONFIG_HEADER(config.h)
+dnl AC_CONFIG_HEADERS(include/libvcd/acconfig.h)
+AC_CONFIG_HEADERS(config.h)
 
 LIBVCD_VERSION_NUM=RELEASE_NUM
 AC_SUBST(LIBVCD_VERSION_NUM)
@@ -117,7 +117,6 @@
 fi
 
 AC_PROG_CPP
-AM_PROG_CC_STDC
 AC_PROG_AWK
 dnl AM_PROG_LIBTOOL tests whether we have GNU ld
 dnl this must come before checking --with-versioned-libs
