diff -Naur vcdimager-0.7.24-old/configure.ac vcdimager-0.7.24-new/configure.ac
--- vcdimager-0.7.24-old/configure.ac	2011-03-17 16:43:53.000000000 -0700
+++ vcdimager-0.7.24-new/configure.ac	2012-01-17 07:30:34.000000000 -0800
@@ -336,6 +336,7 @@
 AC_SUBST(LIBVCD_CFLAGS)
 AC_SUBST(LIBVCD_LIBS)
 AC_SUBST(LIBVCDINFO_LIBS)
+AC_SUBST(NM)
 
 AC_CONFIG_FILES([ \
 	Makefile \
diff -Naur vcdimager-0.7.24-old/lib/Makefile.am vcdimager-0.7.24-new/lib/Makefile.am
--- vcdimager-0.7.24-old/lib/Makefile.am	2005-07-10 05:20:03.000000000 -0700
+++ vcdimager-0.7.24-new/lib/Makefile.am	2012-01-17 07:30:34.000000000 -0800
@@ -125,7 +125,7 @@
 	echo 'VCDINFO_$(libvcdinfo_la_MAJOR) {' > $@
 	echo "  global:" >> $@
 	objs=`for obj in $(libvcdinfo_la_OBJECTS); do sed -ne "s/^pic_object='\(.*\)'$$/\1/p" $$obj; done`; \
-	nm $${objs} | sed -n -e 's/^.*[ 	][ABCDGIRSTW][ABCDGIRSTW]*[ 	][ 	]*\([_A-Za-z][_A-Za-z0-9]*\)$$/    \1;/p' | sort | uniq >> $@
+	$(NM) $${objs} | sed -n -e 's/^.*[ 	][ABCDGIRSTW][ABCDGIRSTW]*[ 	][ 	]*\([_A-Za-z][_A-Za-z0-9]*\)$$/    \1;/p' | sort | uniq >> $@
 	echo '};' >> $@
 
 MOSTLYCLEANFILES = libvcdinfo.la.ver
