diff -Naur vcdimager-0.7.24-old/configure.ac vcdimager-0.7.24-new/configure.ac
--- vcdimager-0.7.24-old/configure.ac	2012-01-17 07:30:34.000000000 -0800
+++ vcdimager-0.7.24-new/configure.ac	2012-01-17 07:31:11.000000000 -0800
@@ -247,7 +247,7 @@
 ], bf_lsbf=0, AC_MSG_ERROR([unsupported bitfield ordering]))
   ],
   [case "$host" in
-     *-*-mingw32* | *-*-cygwin*)
+     i?86-* | x86_64-* | *-*-mingw32* | *-*-cygwin*)
        bf_lsbf=1
      ;;
      *)
