diff --speed-large-files --no-dereference --minimal -Naur connman-1.44/client/input.c connman-1.44/client/input.c
--- connman-1.44/client/input.c	2015-12-28 04:05:49.000000000 +0100
+++ connman-1.44/client/input.c	2025-05-10 20:18:12.154498447 +0200
@@ -273,7 +273,7 @@
 		__connmanctl_monitor_completions(NULL);
 
 		rl_callback_handler_remove();
-		rl_message("");
+		rl_message();
 	}
 
 	dbus_connection_unref(connection);
