diff -Naur cwiid-201-old/wminput/uinput.c cwiid-201-new/wminput/uinput.c
--- cwiid-201-old/wminput/uinput.c	2010-12-08 06:38:55.000000000 -0800
+++ cwiid-201-new/wminput/uinput.c	2010-12-08 06:39:14.000000000 -0800
@@ -130,6 +130,9 @@
 		}			
 	}
 	for (i=0; i < CONF_AXIS_COUNT; i++) {
+		if (conf->amap[i].axis_type != EV_ABS) {
+			continue;
+		}
 		if (conf->amap[i].active) {
 			if (ioctl(conf->fd, UI_SET_EVBIT, conf->amap[i].axis_type) < 0) {
 				wminput_err("error uinput ioctl");
@@ -148,16 +151,16 @@
 			  ((conf->amap[i].action == ABS_X) ||
 			   (conf->amap[i].action == ABS_Y) ||
 			   (conf->amap[i].action == ABS_Z))) {
-				if (ioctl(conf->fd, UI_SET_EVBIT, EV_REL) < 0) {
-					wminput_err("error uinput ioctl");
-					close(conf->fd);
-					return -1;
-				}
-				if (ioctl(conf->fd, UI_SET_RELBIT, conf->amap[i].action) < 0) {
-					wminput_err("error uinput ioctl");
-					close(conf->fd);
-					return -1;
-				}
+//				if (ioctl(conf->fd, UI_SET_EVBIT, EV_REL) < 0) {
+//					wminput_err("error uinput ioctl");
+//					close(conf->fd);
+//					return -1;
+//				}
+//				if (ioctl(conf->fd, UI_SET_RELBIT, conf->amap[i].action) < 0) {
+//					wminput_err("error uinput ioctl");
+//					close(conf->fd);
+//					return -1;
+//				}
 			}
 		}
 	}
@@ -174,6 +177,9 @@
 				}
 			}
 			for (j=0; j < conf->plugins[i].info->axis_count; j++) {
+				if (conf->plugins[i].amap[j].axis_type != EV_ABS) {
+					continue;
+				}
 				if (conf->plugins[i].amap[j].active) {
 					if (ioctl(conf->fd, UI_SET_EVBIT,
 					          conf->plugins[i].amap[j].axis_type) < 0) {
@@ -195,17 +201,17 @@
 					  ((conf->plugins[i].amap[j].action == ABS_X) ||
 					   (conf->plugins[i].amap[j].action == ABS_Y) ||
 					   (conf->plugins[i].amap[j].action == ABS_Z))) {
-						if (ioctl(conf->fd, UI_SET_EVBIT, EV_REL) < 0) {
-							wminput_err("error uinput ioctl");
-							close(conf->fd);
-							return -1;
-						}
-						if (ioctl(conf->fd, UI_SET_RELBIT,
-						          conf->plugins[i].amap[j].action) < 0) {
-							wminput_err("error uinput ioctl");
-							close(conf->fd);
-							return -1;
-						}
+//						if (ioctl(conf->fd, UI_SET_EVBIT, EV_REL) < 0) {
+//							wminput_err("error uinput ioctl");
+//							close(conf->fd);
+//							return -1;
+//						}
+//						if (ioctl(conf->fd, UI_SET_RELBIT,
+//						          conf->plugins[i].amap[j].action) < 0) {
+//							wminput_err("error uinput ioctl");
+//							close(conf->fd);
+//							return -1;
+//						}
 					}
 				}
 			}
