diff -Naur cwiid-201-old/wminput/main.c cwiid-201-new/wminput/main.c
--- cwiid-201-old/wminput/main.c	2009-07-17 09:12:15.000000000 -0700
+++ cwiid-201-new/wminput/main.c	2009-07-17 09:12:43.000000000 -0700
@@ -289,7 +289,7 @@
 			}
 			/* TODO: avoid continuously calling cwiid_open */
 			cwiid_set_err(cwiid_err_connect);
-			while (!(wiimote = cwiid_open(&current_bdaddr, CWIID_FLAG_MESG_IFC)));
+			while (!(wiimote = cwiid_open(&current_bdaddr, CWIID_FLAG_MESG_IFC))) sleep(1);
 			cwiid_set_err(cwiid_err_default);
 		}
 		else {
