diff -Naur cwiid-201-old/defs.mak.in cwiid-201-new/defs.mak.in
--- cwiid-201-old/defs.mak.in	2009-07-17 09:10:55.000000000 -0700
+++ cwiid-201-new/defs.mak.in	2009-07-17 09:11:45.000000000 -0700
@@ -28,8 +28,8 @@
 	ROOTDIR = $(DESTDIR:%/=%)
 endif
 
-CWIID_CONFIG_DIR = $(ROOTDIR)@CWIID_CONFIG_DIR@
-CWIID_PLUGINS_DIR = $(ROOTDIR)@CWIID_PLUGINS_DIR@
+CWIID_CONFIG_DIR = @CWIID_CONFIG_DIR@
+CWIID_PLUGINS_DIR = @CWIID_PLUGINS_DIR@
 
 DEBUGFLAGS = -g
 WARNFLAGS = -Wall -W
diff -Naur cwiid-201-old/Makefile.in cwiid-201-new/Makefile.in
--- cwiid-201-old/Makefile.in	2009-07-17 09:10:55.000000000 -0700
+++ cwiid-201-new/Makefile.in	2009-07-17 09:11:45.000000000 -0700
@@ -26,27 +26,27 @@
 endif
 
 ifeq ($(MAKECMDGOALS),install)
-$(BIN_DIRS) $(LIB_DIRS): $(CWIID_CONFIG_DIR) $(CWIID_PLUGINS_DIR)
+$(BIN_DIRS) $(LIB_DIRS): $(ROOTDIR)$(CWIID_CONFIG_DIR) $(ROOTDIR)$(CWIID_PLUGINS_DIR)
 endif
 
 $(SUB_DIRS):
 	$(MAKE) $(TARGET) -C $@
 
-$(CWIID_CONFIG_DIR):
-	install -d $(CWIID_CONFIG_DIR)
+$(ROOTDIR)$(CWIID_CONFIG_DIR):
+	install -d $(ROOTDIR)$(CWIID_CONFIG_DIR)
 
-$(CWIID_PLUGINS_DIR):
-	install -d $(CWIID_PLUGINS_DIR)
+$(ROOTDIR)$(CWIID_PLUGINS_DIR):
+	install -d $(ROOTDIR)$(CWIID_PLUGINS_DIR)
 
 distclean:
 	rm -rf Makefile config.log config.status autom4te.cache \
 		defs.mak $(COMMON)/include/lib.mak $(COMMON)/include/config.h
 
 uninstall:
-	rm -r $(CWIID_PLUGINS_DIR); true
+	rm -r $(ROOTDIR)$(CWIID_PLUGINS_DIR); true
 
 uninstall_config:
-	rm -rf $(CWIID_CONFIG_DIR)
+	rm -rf $(ROOTDIR)$(CWIID_CONFIG_DIR)
 
 .PHONY: all install clean distclean uninstall uninstall_config $(SUB_DIRS)
 
diff -Naur cwiid-201-old/wminput/Makefile.in cwiid-201-new/wminput/Makefile.in
--- cwiid-201-old/wminput/Makefile.in	2009-07-17 09:10:55.000000000 -0700
+++ cwiid-201-new/wminput/Makefile.in	2009-07-17 09:11:45.000000000 -0700
@@ -41,16 +41,16 @@
 	$(MAKE) $(MAKECMDGOALS) -C plugins
 
 install_configs:
-	install -d $(WMINPUT_CONFIG_DIR)
-	install configs/* $(WMINPUT_CONFIG_DIR)
-	ln -sf acc_led $(WMINPUT_CONFIG_DIR)/default
+	install -d $(ROOTDIR)$(WMINPUT_CONFIG_DIR)
+	install configs/* $(ROOTDIR)$(WMINPUT_CONFIG_DIR)
+	ln -sf acc_led $(ROOTDIR)$(WMINPUT_CONFIG_DIR)/default
 
 #uninstall_configs:
-#	rm -f $(WMINPUT_CONFIG_DIR)/default
+#	rm -f $(ROOTDIR)$(WMINPUT_CONFIG_DIR)/default
 #	for c in `ls configs`; do				\
-#		rm -f $(WMINPUT_CONFIG_DIR)/$$c;	\
+#		rm -f $(ROOTDIR)$(WMINPUT_CONFIG_DIR)/$$c;	\
 #	done
-#	rmdir $(WMINPUT_CONFIG_DIR); true
+#	rmdir $(ROOTDIR)$(WMINPUT_CONFIG_DIR); true
 
 action_enum.c: action_enum.awk action_enum.txt
 	$(AWK) -f action_enum.awk action_enum.txt > action_enum.c
diff -Naur cwiid-201-old/wminput/plugins/acc/Makefile.in cwiid-201-new/wminput/plugins/acc/Makefile.in
--- cwiid-201-old/wminput/plugins/acc/Makefile.in	2009-07-17 09:10:55.000000000 -0700
+++ cwiid-201-new/wminput/plugins/acc/Makefile.in	2009-07-17 09:11:45.000000000 -0700
@@ -6,7 +6,7 @@
 SOURCES = acc.c
 CFLAGS += -I@top_builddir@/wminput -I@top_builddir@/libcwiid
 LDLIBS += -lm
-INST_DIR = $(CWIID_PLUGINS_DIR)
+INST_DIR = $(ROOTDIR)$(CWIID_PLUGINS_DIR)
 
 include $(COMMON)/include/plugin.mak
 
diff -Naur cwiid-201-old/wminput/plugins/ir_ptr/Makefile.in cwiid-201-new/wminput/plugins/ir_ptr/Makefile.in
--- cwiid-201-old/wminput/plugins/ir_ptr/Makefile.in	2009-07-17 09:11:06.000000000 -0700
+++ cwiid-201-new/wminput/plugins/ir_ptr/Makefile.in	2009-07-17 09:11:45.000000000 -0700
@@ -6,7 +6,7 @@
 SOURCES = ir_ptr.c
 CFLAGS += -I@top_builddir@/wminput -I@top_builddir@/libcwiid
 LDLIBS += -lm
-INST_DIR = $(CWIID_PLUGINS_DIR)
+INST_DIR = $(ROOTDIR)$(CWIID_PLUGINS_DIR)
 
 include $(COMMON)/include/plugin.mak
 
diff -Naur cwiid-201-old/wminput/plugins/led/Makefile.in cwiid-201-new/wminput/plugins/led/Makefile.in
--- cwiid-201-old/wminput/plugins/led/Makefile.in	2009-07-17 09:11:06.000000000 -0700
+++ cwiid-201-new/wminput/plugins/led/Makefile.in	2009-07-17 09:11:45.000000000 -0700
@@ -6,7 +6,7 @@
 SOURCES = led.c
 CFLAGS += -I@top_builddir@/wminput -I@top_builddir@/libcwiid
 LDLIBS += -lm
-INST_DIR = $(CWIID_PLUGINS_DIR)
+INST_DIR = $(ROOTDIR)$(CWIID_PLUGINS_DIR)
 
 include $(COMMON)/include/plugin.mak
 
diff -Naur cwiid-201-old/wminput/plugins/nunchuk_acc/Makefile.in cwiid-201-new/wminput/plugins/nunchuk_acc/Makefile.in
--- cwiid-201-old/wminput/plugins/nunchuk_acc/Makefile.in	2009-07-17 09:10:55.000000000 -0700
+++ cwiid-201-new/wminput/plugins/nunchuk_acc/Makefile.in	2009-07-17 09:11:45.000000000 -0700
@@ -6,7 +6,7 @@
 SOURCES = nunchuk_acc.c
 CFLAGS += -I@top_builddir@/wminput -I@top_builddir@/libcwiid
 LDLIBS += -lm
-INST_DIR = $(CWIID_PLUGINS_DIR)
+INST_DIR = $(ROOTDIR)$(CWIID_PLUGINS_DIR)
 
 include $(COMMON)/include/plugin.mak
 
diff -Naur cwiid-201-old/wminput/plugins/nunchuk_stick2btn/Makefile.in cwiid-201-new/wminput/plugins/nunchuk_stick2btn/Makefile.in
--- cwiid-201-old/wminput/plugins/nunchuk_stick2btn/Makefile.in	2009-07-17 09:10:55.000000000 -0700
+++ cwiid-201-new/wminput/plugins/nunchuk_stick2btn/Makefile.in	2009-07-17 09:11:45.000000000 -0700
@@ -6,7 +6,7 @@
 SOURCES = nunchuk_stick2btn.c
 CFLAGS += -I@top_builddir@/wminput -I@top_builddir@/libcwiid
 LDLIBS += -lm
-INST_DIR = $(CWIID_PLUGINS_DIR)
+INST_DIR = $(ROOTDIR)$(CWIID_PLUGINS_DIR)
 
 include $(COMMON)/include/plugin.mak
 
