diff -Naur cryptography-2.9.2-old/setup.py cryptography-2.9.2-new/setup.py
--- cryptography-2.9.2-old/setup.py	2020-04-23 00:26:51.000000000 +0200
+++ cryptography-2.9.2-new/setup.py	2020-06-02 19:44:21.903333317 +0200
@@ -18,14 +18,14 @@
 from setuptools.command.install import install
 
 
-if (
-    pkg_resources.parse_version(setuptools.__version__) <
-    pkg_resources.parse_version("18.5")
-):
-    raise RuntimeError(
-        "cryptography requires setuptools 18.5 or newer, please upgrade to a "
-        "newer version of setuptools"
-    )
+#if (
+#    pkg_resources.parse_version(setuptools.__version__) <
+#    pkg_resources.parse_version("18.5")
+#):
+#    raise RuntimeError(
+#        "cryptography requires setuptools 18.5 or newer, please upgrade to a "
+#        "newer version of setuptools"
+#    )
 
 base_dir = os.path.dirname(__file__)
 src_dir = os.path.join(base_dir, "src")
