diff -Naur makemkv-bin-1.9.0-old/Makefile makemkv-bin-1.9.0-new/Makefile
--- makemkv-bin-1.9.0-old/Makefile	2014-11-16 16:38:40.000000000 +0100
+++ makemkv-bin-1.9.0-new/Makefile	2014-12-17 15:35:45.569688306 +0100
@@ -29,8 +29,8 @@
 
 tmp/eula_accepted:
 	@mkdir -p tmp
-	@/bin/bash src/ask_eula.sh src/eula_en_linux.txt
-	@echo accepted > $@
+	@#/bin/bash src/ask_eula.sh src/eula_en_linux.txt
+	@#echo accepted > $@
 
 install: tmp/eula_accepted bin/$(ARCH)/makemkvcon
 	rm -f $(DESTDIR)/usr/bin/makemkvcon
