diff -Naur makemkv-oss-1.12.2-old/libffabi/src/internal.h makemkv-oss-1.12.2-new/libffabi/src/internal.h
--- makemkv-oss-1.12.2-old/libffabi/src/internal.h	2018-04-28 03:53:47.000000000 +0200
+++ makemkv-oss-1.12.2-new/libffabi/src/internal.h	2018-05-03 11:26:10.000000000 +0200
@@ -62,9 +62,10 @@
 
 #include "ffabicfg.h"
 
+#define FFABI_AVCODEC_OLD_API 1
+
 #ifndef FFABI_HAVE_AVCODEC_SEND_FRAME
 
-#define FFABI_AVCODEC_OLD_API 1
 
 #ifndef FFABI_HAVE_AVCODECID
 #define AVCodecID CodecID
