diff -Naur u-boot-sunxi-5f19c9302133cda54d5d1a6b1caa400260de9192-old/arch/arm/mach-sunxi/dram_sun50i_h6.c u-boot-sunxi-5f19c9302133cda54d5d1a6b1caa400260de9192-new/arch/arm/mach-sunxi/dram_sun50i_h6.c
--- u-boot-sunxi-5f19c9302133cda54d5d1a6b1caa400260de9192-old/arch/arm/mach-sunxi/dram_sun50i_h6.c	2019-10-25 11:10:34.000000000 +0200
+++ u-boot-sunxi-5f19c9302133cda54d5d1a6b1caa400260de9192-new/arch/arm/mach-sunxi/dram_sun50i_h6.c	2020-03-09 09:52:34.729999997 +0100
@@ -5,6 +5,8 @@
  *
  * SPDX-License-Identifier:	GPL-2.0+
  */
+#define DEBUG 1
+
 #include <common.h>
 #include <asm/io.h>
 #include <asm/arch/clock.h>
