diff -Naur eject-old/eject.c eject-new/eject.c
--- eject-old/eject.c	2020-03-30 18:34:43.052688403 +0200
+++ eject-new/eject.c	2020-03-30 18:32:07.149364447 +0200
@@ -71,6 +71,7 @@
 #include <scsi/sg.h>
 #include <scsi/scsi_ioctl.h>
 #include <sys/time.h>
+#include <sys/sysmacros.h>
 
 /* Used by the ToggleTray() function. If ejecting the tray takes this
  * time or less, the tray was probably already ejected, so we close it
