diff -Naur eject-2.1.5-old/eject.c eject-2.1.5-new/eject.c
--- eject-2.1.5-old/eject.c	2006-02-10 17:54:38.000000000 -0800
+++ eject-2.1.5-new/eject.c	2007-03-28 21:15:01.000000000 -0700
@@ -543,10 +543,11 @@
 
 #ifdef CDROMCLOSETRAY
 
-	/* Try to open the CDROM tray and measure the time therefor
-	 * needed.  In my experience the function needs less than 0.05
+	/* Try to open the CDROM tray and measure the time needed.
+	 * In my experience the function needs less than 0.05
 	 * seconds if the tray was already open, and at least 1.5 seconds
-	 * if it was closed.  */
+	 * if it was closed.
+	 */
 	gettimeofday(&time_start, NULL);
 	
 	/* Send the CDROMEJECT command to the device. */
@@ -813,10 +814,17 @@
 }
 
 
-/* Open a device file. */
+/* Open a device file. Try opening first read/write, and if that fails then read only. */
 static int OpenDevice(const char *fullName)
 {
-	int fd = open(fullName, O_RDONLY|O_NONBLOCK);
+	int fd;
+
+	fd = open(fullName, O_RDWR|O_NONBLOCK);
+	if (fd != -1) {
+		return fd;
+	}
+	
+	fd = open(fullName, O_RDONLY|O_NONBLOCK);
 	if (fd == -1) {
 		fprintf(stderr, _("%s: unable to open `%s'\n"), programName, fullName);
 		exit(1);
