diff -Naur fmt.old/fmt-targets.cmake fmt.new/fmt-targets.cmake
--- fmt.old/fmt-targets.cmake	2024-01-24 18:54:26.000000000 +0100
+++ fmt.new/fmt-targets.cmake	2024-01-24 17:41:30.419369643 +0100
@@ -46,8 +46,11 @@
 unset(_cmake_expected_targets)
 
 
-# The installation prefix configured by this project.
-set(_IMPORT_PREFIX "/usr")
+# Compute the installation prefix relative to this file.
+get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
 
 # Create imported target fmt::fmt
 add_library(fmt::fmt SHARED IMPORTED)
diff -Naur fmt.old/fmt-targets-release.cmake fmt.new/fmt-targets-release.cmake
--- fmt.old/fmt-targets-release.cmake	2024-01-24 18:54:26.000000000 +0100
+++ fmt.new/fmt-targets-release.cmake	2024-01-24 18:49:52.552614572 +0100
@@ -8,12 +8,12 @@
 # Import target "fmt::fmt" for configuration "Release"
 set_property(TARGET fmt::fmt APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
 set_target_properties(fmt::fmt PROPERTIES
-  IMPORTED_LOCATION_RELEASE "/usr/lib/libfmt.so.10.2.0"
+  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/libfmt.so.10.2.0"
   IMPORTED_SONAME_RELEASE "libfmt.so.10"
   )
 
 list(APPEND _cmake_import_check_targets fmt::fmt )
-list(APPEND _cmake_import_check_files_for_fmt::fmt "/usr/lib/libfmt.so.10.2.0" )
+list(APPEND _cmake_import_check_files_for_fmt::fmt "${_IMPORT_PREFIX}/lib/libfmt.so.10.2.0" )
 
 # Commands beyond this point should not need to know the version.
 set(CMAKE_IMPORT_FILE_VERSION)
