diff -Naur stella-3.9.3-old/configure stella-3.9.3-new/configure
--- stella-3.9.3-old/configure	2013-04-17 08:00:32.000000000 -0700
+++ stella-3.9.3-new/configure	2015-06-22 15:11:09.781000601 -0700
@@ -31,10 +31,10 @@
 _build_profile=no
 
 # more defaults
-_ranlib=ranlib
+_ranlib="${RANLIB:-'ranlib'}"
 _install=install
-_ar="ar cru"
-_strip=strip
+_ar="${AR:-'ar'} cru"
+_strip="${STRIP:-'strip'}"
 _mkdir="mkdir -p"
 _echo=printf
 _cat=cat
@@ -114,7 +114,7 @@
 	# In cross-compiling mode, we cannot run the result
 	eval "$1 $LDFLAGS -o tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp 2> /dev/null" && rm -f tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp
 else
-	eval "$1 $LDFLAGS -o tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp 2> /dev/null" && eval "./tmp_cxx_compiler 2> /dev/null" && rm -f tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp
+	eval "$1 $LDFLAGS -o tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp 2> /dev/null" && rm -f tmp_cxx_compiler$EXEEXT tmp_cxx_compiler.cpp
 fi
 }
 
diff -Naur stella-3.9.3-old/Makefile stella-3.9.3-new/Makefile
--- stella-3.9.3-old/Makefile	2014-01-17 15:34:03.000000000 -0800
+++ stella-3.9.3-new/Makefile	2015-06-22 15:11:09.781000601 -0700
@@ -172,7 +172,7 @@
 
 install: all
 	$(INSTALL) -d "$(DESTDIR)$(BINDIR)"
-	$(INSTALL) -c -s -m 755 "$(srcdir)/stella$(EXEEXT)" "$(DESTDIR)$(BINDIR)/stella$(EXEEXT)"
+	$(INSTALL) -c -m 755 "$(srcdir)/stella$(EXEEXT)" "$(DESTDIR)$(BINDIR)/stella$(EXEEXT)"
 	$(INSTALL) -d "$(DESTDIR)$(DOCDIR)"
 	$(INSTALL) -c -m 644 "$(srcdir)/Announce.txt" "$(srcdir)/Changes.txt" "$(srcdir)/Copyright.txt" "$(srcdir)/License.txt" "$(srcdir)/README-SDL.txt" "$(srcdir)/Readme.txt" "$(srcdir)/Todo.txt" "$(srcdir)/docs/index.html" "$(srcdir)/docs/debugger.html" "$(DESTDIR)$(DOCDIR)/"
 	$(INSTALL) -d "$(DESTDIR)$(DOCDIR)/graphics"
