diff -Naur poky-yocto-2.7.3-old/scripts/lib/wic/plugins/source/bootimg-efi.py poky-yocto-2.7.3-new/scripts/lib/wic/plugins/source/bootimg-efi.py
--- poky-yocto-2.7.3-old/scripts/lib/wic/plugins/source/bootimg-efi.py	2020-02-21 11:35:31.000000000 +0100
+++ poky-yocto-2.7.3-new/scripts/lib/wic/plugins/source/bootimg-efi.py	2020-03-16 10:26:54.689999981 +0100
@@ -227,6 +227,18 @@
             cp_cmd = "cp %s %s/" % (startup, hdddir)
             exec_cmd(cp_cmd, True)
 
+        minimyth_conf = os.path.join(staging_kernel_dir, "minimyth.conf")
+        if os.path.exists(minimyth_conf):
+            cmd = ("install -m 0644 %s/minimyth.conf %s/minimyth.conf" %
+                   (staging_kernel_dir, hdddir))
+            exec_cmd(cmd)
+
+        rootfs_archive = os.path.join(staging_kernel_dir, "rootfs")
+        if os.path.exists(rootfs_archive):
+            cmd = ("install -m 0644 %s/rootfs %s/rootfs" %
+                   (staging_kernel_dir, hdddir))
+            exec_cmd(cmd)
+
         du_cmd = "du -bks %s" % hdddir
         out = exec_cmd(du_cmd)
         blocks = int(out.split()[0])
diff -Naur poky-yocto-2.7.3-old/scripts/lib/wic/plugins/source/bootimg-pcbios.py poky-yocto-2.7.3-new/scripts/lib/wic/plugins/source/bootimg-pcbios.py
--- poky-yocto-2.7.3-old/scripts/lib/wic/plugins/source/bootimg-pcbios.py	2020-02-21 11:35:31.000000000 +0100
+++ poky-yocto-2.7.3-new/scripts/lib/wic/plugins/source/bootimg-pcbios.py	2020-03-16 10:24:09.639999987 +0100
@@ -163,6 +163,18 @@
         for install_cmd in cmds:
             exec_cmd(install_cmd)
 
+        minimyth_conf = os.path.join(staging_kernel_dir, "minimyth.conf")
+        if os.path.exists(minimyth_conf):
+            cmd = ("install -m 0644 %s/minimyth.conf %s/minimyth.conf" %
+                  (staging_kernel_dir, hdddir))
+            exec_cmd(cmd)
+
+        rootfs_archive = os.path.join(staging_kernel_dir, "rootfs")
+        if os.path.exists(rootfs_archive):
+            cmds = ("install -m 0644 %s/rootfs %s/rootfs" %
+                   (staging_kernel_dir, hdddir))
+            exec_cmd(cmd)
+
         du_cmd = "du -bks %s" % hdddir
         out = exec_cmd(du_cmd)
         blocks = int(out.split()[0])
