diff -Naur poky-yocto-2.7.3-old/meta/classes/base.bbclass poky-yocto-2.7.3-new/meta/classes/base.bbclass
--- poky-yocto-2.7.3-old/meta/classes/base.bbclass	2020-02-21 11:35:31.000000000 +0100
+++ poky-yocto-2.7.3-new/meta/classes/base.bbclass	2020-03-13 15:38:05.740000000 +0100
@@ -136,7 +136,7 @@
     desttool = os.path.join(dest, "python")
     if not os.path.exists(desttool):
         srctool = "python2"
-        os.symlink(srctool, desttool)
+#        os.symlink(srctool, desttool)
 
     if notfound and fatal:
         bb.fatal("The following required tools (as specified by HOSTTOOLS) appear to be unavailable in PATH, please install them in order to proceed:\n  %s" % " ".join(notfound))
