diff -Naur poky-yocto-2.7.3-old/scripts/lib/wic/plugins/imager/direct.py poky-yocto-2.7.3-new/scripts/lib/wic/plugins/imager/direct.py
--- poky-yocto-2.7.3-old/scripts/lib/wic/plugins/imager/direct.py	2020-02-21 11:35:31.000000000 +0100
+++ poky-yocto-2.7.3-new/scripts/lib/wic/plugins/imager/direct.py	2020-10-29 09:06:51.753333316 +0100
@@ -317,8 +317,9 @@
         self.min_size = 0 # Minimum required disk size to fit
                           # all partitions (in bytes)
         self.ptable_format = ptable_format  # Partition table format
-        # Disk system identifier
-        self.identifier = random.SystemRandom().randint(1, 0xffffffff)
+        # Disk system identifier. It is declared as constant and should correspond to
+        # PARTUUID used by init scripts.
+        self.identifier = 0x346d1a6a
 
         self.partitions = partitions
         self.partimages = []
