diff --speed-large-files --no-dereference --minimal -Naur poky-yocto-2.7.3/bitbake/lib/bb/compat.py poky-yocto-2.7.3/bitbake/lib/bb/compat.py
--- poky-yocto-2.7.3/bitbake/lib/bb/compat.py	2020-02-21 11:35:31.000000000 +0100
+++ poky-yocto-2.7.3/bitbake/lib/bb/compat.py	2024-02-05 14:38:10.496185312 +0100
@@ -4,7 +4,8 @@
 
 """Code pulled from future python versions, here for compatibility"""
 
-from collections import MutableMapping, KeysView, ValuesView, ItemsView, OrderedDict
+from collections.abc import MutableMapping, KeysView, ValuesView, ItemsView
+from collections import OrderedDict
 from functools import total_ordering
 
 
diff --speed-large-files --no-dereference --minimal -Naur poky-yocto-2.7.3/bitbake/lib/bb/data_smart.py poky-yocto-2.7.3/bitbake/lib/bb/data_smart.py
--- poky-yocto-2.7.3/bitbake/lib/bb/data_smart.py	2020-02-21 11:35:31.000000000 +0100
+++ poky-yocto-2.7.3/bitbake/lib/bb/data_smart.py	2024-02-05 14:41:34.996180898 +0100
@@ -19,7 +19,7 @@
 # Based on functions from the base bb module, Copyright 2003 Holger Schurig
 
 import copy, re, sys, traceback
-from collections import MutableMapping
+from collections.abc import MutableMapping
 import logging
 import hashlib
 import bb, bb.codeparser
diff --speed-large-files --no-dereference --minimal -Naur poky-yocto-2.7.3/bitbake/lib/bb/persist_data.py poky-yocto-2.7.3/bitbake/lib/bb/persist_data.py
--- poky-yocto-2.7.3/bitbake/lib/bb/persist_data.py	2020-02-21 11:35:31.000000000 +0100
+++ poky-yocto-2.7.3/bitbake/lib/bb/persist_data.py	2024-02-05 14:41:15.236181326 +0100
@@ -17,7 +17,7 @@
 import sys
 import warnings
 from bb.compat import total_ordering
-from collections import Mapping
+from collections.abc import Mapping
 import sqlite3
 import contextlib
 
@@ -29,7 +29,7 @@
 logger = logging.getLogger("BitBake.PersistData")
 
 @total_ordering
-class SQLTable(collections.MutableMapping):
+class SQLTable(collections.abc.MutableMapping):
     class _Decorators(object):
         @staticmethod
         def retry(*, reconnect=True):
diff --speed-large-files --no-dereference --minimal -Naur poky-yocto-2.7.3/scripts/lib/wic/partition.py poky-yocto-2.7.3/scripts/lib/wic/partition.py
--- poky-yocto-2.7.3/scripts/lib/wic/partition.py	2024-02-05 15:09:45.479477818 +0100
+++ poky-yocto-2.7.3/scripts/lib/wic/partition.py	2024-02-05 15:05:23.009483461 +0100
@@ -245,7 +245,7 @@
         out = exec_cmd(du_cmd)
         actual_rootfs_size = int(out.split()[0])
 
-        rootfs_size = self.get_rootfs_size(actual_rootfs_size)
+        rootfs_size = int(self.get_rootfs_size(actual_rootfs_size))
 
         with open(rootfs, 'w') as sparse:
             os.ftruncate(sparse.fileno(), rootfs_size * 1024)
