diff -Naur poky-yocto-2.7.3.old/scripts/lib/wic/plugins/imager/direct.py poky-yocto-2.7.3.new/scripts/lib/wic/plugins/imager/direct.py
--- poky-yocto-2.7.3.old/scripts/lib/wic/plugins/imager/direct.py	2022-01-03 19:05:33.746056363 +0100
+++ poky-yocto-2.7.3.new/scripts/lib/wic/plugins/imager/direct.py	2022-01-03 18:59:48.229395925 +0100
@@ -440,8 +440,8 @@
                     raise WicError("Could not place %s%s at offset %d with sector size %d" % (part.disk, self.numpart, part.offset, self.sector_size))
 
                 delta = offset - self.offset
-                if delta < 0:
-                    raise WicError("Could not place %s%s at offset %d: next free sector is %d (delta: %d)" % (part.disk, self.numpart, part.offset, self.offset, delta))
+                #if delta < 0:
+                #    raise WicError("Could not place %s%s at offset %d: next free sector is %d (delta: %d)" % (part.disk, self.numpart, part.offset, self.offset, delta))
 
                 logger.debug("Skipping %d sectors to place %s%s at offset %dK",
                              delta, part.disk, self.numpart, part.offset)
@@ -532,6 +532,8 @@
                 parted_fs_type = "linux-swap"
             elif part.fstype == "vfat":
                 parted_fs_type = "fat32"
+            elif part.fstype == "efi":
+                parted_fs_type = "efi"
             elif part.fstype == "msdos":
                 parted_fs_type = "fat16"
                 if not part.system_id:
