diff -Naur bison-3.7.6-old/Makefile.in bison-3.7.6-new/Makefile.in
--- bison-3.7.6-old/Makefile.in	2021-03-07 21:16:37.000000000 -0800
+++ bison-3.7.6-new/Makefile.in	2021-04-29 05:10:30.238336017 -0700
@@ -8351,6 +8351,8 @@
 	@: > doc/$(am__dirstamp)
 
 $(srcdir)/doc/bison.info: doc/bison.texi $(srcdir)/doc/version.texi $(doc_bison_TEXINFOS)
+
+$(srcdir)/doc/bison.info1: doc/bison.texi $(srcdir)/doc/version.texi $(doc_bison_TEXINFOS)
 	$(AM_V_MAKEINFO)restore=: && backupdir="$(am__leading_dot)am$$$$" && \
 	am__cwd=`pwd` && $(am__cd) $(srcdir) && \
 	rm -rf $$backupdir && mkdir $$backupdir && \
