diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-g12/DisableHDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-g12/DisableHDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-g12/DisableHDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-g12/DisableHDMI.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1 @@
+# Disabling HDMI on G12 SoC
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-g12/DisableSPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-g12/DisableSPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-g12/DisableSPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-g12/DisableSPDIF.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1 @@
+# Disable SPDIF on G12 SoC
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-g12/EnableHDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-g12/EnableHDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-g12/EnableHDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-g12/EnableHDMI.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1,4 @@
+# Sequence to enable HDMI on G12
+cset "name='FRDDR_A SINK 1 SEL' OUT 1"
+cset "name='TDMOUT_B SRC SEL' IN 0"
+cset "name='TOHDMITX I2S SRC' I2S B"
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-g12/EnableSPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-g12/EnableSPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-g12/EnableSPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-g12/EnableSPDIF.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1,3 @@
+# Sequence to enable SPDIF on G12 SoC
+cset "name='FRDDR_B SINK 1 SEL' OUT 3"
+cset "name='SPDIFOUT SRC SEL' IN 1"
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-gx/DisableHDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-gx/DisableHDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-gx/DisableHDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-gx/DisableHDMI.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1 @@
+# Disabling HDMI on GX SoC
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-gx/DisableSPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-gx/DisableSPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-gx/DisableSPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-gx/DisableSPDIF.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1 @@
+# Disable SPDIF on GX SoC
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-gx/EnableHDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-gx/EnableHDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-gx/EnableHDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-gx/EnableHDMI.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1,2 @@
+# Sequence to enable HDMI on GX
+cset "name='AIU HDMI CTRL SRC' I2S"
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-gx/EnableSPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-gx/EnableSPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/codecs/amlogic-gx/EnableSPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/codecs/amlogic-gx/EnableSPDIF.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1,2 @@
+# Sequence to enable SPDIF on GX SoC
+cset "name='AIU SPDIF SRC SEL' SPDIF"
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-SEI510/G12A-SEI510.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-SEI510/G12A-SEI510.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-SEI510/G12A-SEI510.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-SEI510/G12A-SEI510.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic G12A SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on G12A SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-SEI510/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-SEI510/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-SEI510/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-SEI510/HDMI.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-TANIX-TX5MAX/G12A-TANIX-TX5MAX.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-TANIX-TX5MAX/G12A-TANIX-TX5MAX.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-TANIX-TX5MAX/G12A-TANIX-TX5MAX.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-TANIX-TX5MAX/G12A-TANIX-TX5MAX.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic G12A SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on G12A SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on G12A SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-TANIX-TX5MAX/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-TANIX-TX5MAX/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-TANIX-TX5MAX/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-TANIX-TX5MAX/HDMI.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-TANIX-TX5MAX/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-TANIX-TX5MAX/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-TANIX-TX5MAX/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-TANIX-TX5MAX/SPDIF.conf	2020-06-22 10:09:54.943333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-U200/G12A-U200.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-U200/G12A-U200.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-U200/G12A-U200.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-U200/G12A-U200.conf	2020-06-22 10:09:54.946666666 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic G12A SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on G12A SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on G12A SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-U200/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-U200/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-U200/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-U200/HDMI.conf	2020-06-22 10:09:54.946666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-U200/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-U200/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-U200/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-U200/SPDIF.conf	2020-06-22 10:09:54.946666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-X96-MAX/G12A-X96-MAX.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-X96-MAX/G12A-X96-MAX.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-X96-MAX/G12A-X96-MAX.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-X96-MAX/G12A-X96-MAX.conf	2020-06-22 10:09:54.946666666 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic G12A SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on G12A SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on G12A SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-X96-MAX/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-X96-MAX/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-X96-MAX/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-X96-MAX/HDMI.conf	2020-06-22 10:09:54.946666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-X96-MAX/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-X96-MAX/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12A-X96-MAX/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12A-X96-MAX/SPDIF.conf	2020-06-22 10:09:54.946666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-GTKING/G12B-GTKING.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-GTKING/G12B-GTKING.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-GTKING/G12B-GTKING.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-GTKING/G12B-GTKING.conf	2020-06-22 10:09:54.953333333 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic G12B SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on G12B SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on G12B SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-GTKING/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-GTKING/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-GTKING/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-GTKING/HDMI.conf	2020-06-22 10:09:54.953333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-GTKING/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-GTKING/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-GTKING/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-GTKING/SPDIF.conf	2020-06-22 10:09:54.953333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-GTKING-PRO/G12B-GTKING-PRO.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-GTKING-PRO/G12B-GTKING-PRO.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-GTKING-PRO/G12B-GTKING-PRO.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-GTKING-PRO/G12B-GTKING-PRO.conf	2020-06-22 10:09:54.960000000 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic G12B SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on G12B SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-GTKING-PRO/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-GTKING-PRO/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-GTKING-PRO/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-GTKING-PRO/HDMI.conf	2020-06-22 10:09:54.960000000 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-KHADAS-VIM3/G12B-KHADAS-VIM3.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-KHADAS-VIM3/G12B-KHADAS-VIM3.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-KHADAS-VIM3/G12B-KHADAS-VIM3.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-KHADAS-VIM3/G12B-KHADAS-VIM3.conf	2020-06-22 10:09:54.966666666 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic G12B SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on G12B SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-KHADAS-VIM3/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-KHADAS-VIM3/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-KHADAS-VIM3/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-KHADAS-VIM3/HDMI.conf	2020-06-22 10:09:54.966666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-ODROID-N2/G12B-ODROID-N2.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-ODROID-N2/G12B-ODROID-N2.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-ODROID-N2/G12B-ODROID-N2.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-ODROID-N2/G12B-ODROID-N2.conf	2020-06-22 10:09:54.966666666 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic G12B SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on G12B SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-ODROID-N2/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-ODROID-N2/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-ODROID-N2/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-ODROID-N2/HDMI.conf	2020-06-22 10:09:54.966666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-UGOOS-AM6/G12B-UGOOS-AM6.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-UGOOS-AM6/G12B-UGOOS-AM6.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-UGOOS-AM6/G12B-UGOOS-AM6.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-UGOOS-AM6/G12B-UGOOS-AM6.conf	2020-06-22 10:09:54.973333333 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic G12B SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on G12B SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on G12B SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-UGOOS-AM6/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-UGOOS-AM6/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-UGOOS-AM6/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-UGOOS-AM6/HDMI.conf	2020-06-22 10:09:54.973333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-UGOOS-AM6/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-UGOOS-AM6/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/G12B-UGOOS-AM6/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/G12B-UGOOS-AM6/SPDIF.conf	2020-06-22 10:09:54.973333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-NANOPI-K2/GXBB-NANOPI-K2.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-NANOPI-K2/GXBB-NANOPI-K2.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-NANOPI-K2/GXBB-NANOPI-K2.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-NANOPI-K2/GXBB-NANOPI-K2.conf	2020-06-22 10:09:54.973333333 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic GXBB SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXBB SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-NANOPI-K2/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-NANOPI-K2/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-NANOPI-K2/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-NANOPI-K2/HDMI.conf	2020-06-22 10:09:54.973333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-NEXBOX-A95X/GXBB-NEXBOX-A95X.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-NEXBOX-A95X/GXBB-NEXBOX-A95X.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-NEXBOX-A95X/GXBB-NEXBOX-A95X.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-NEXBOX-A95X/GXBB-NEXBOX-A95X.conf	2020-06-22 10:09:54.980000000 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic GXBB SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXBB SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-NEXBOX-A95X/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-NEXBOX-A95X/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-NEXBOX-A95X/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-NEXBOX-A95X/HDMI.conf	2020-06-22 10:09:54.980000000 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-ODROID-C2/GXBB-ODROID-C2.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-ODROID-C2/GXBB-ODROID-C2.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-ODROID-C2/GXBB-ODROID-C2.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-ODROID-C2/GXBB-ODROID-C2.conf	2020-06-22 10:09:54.986666666 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic GXBB SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXBB SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-ODROID-C2/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-ODROID-C2/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-ODROID-C2/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-ODROID-C2/HDMI.conf	2020-06-22 10:09:54.986666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-P200/GXBB-P200.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-P200/GXBB-P200.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-P200/GXBB-P200.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-P200/GXBB-P200.conf	2020-06-22 10:09:54.993333333 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic GX SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GX SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on GX SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-P200/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-P200/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-P200/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-P200/HDMI.conf	2020-06-22 10:09:54.993333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-P200/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-P200/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-P200/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-P200/SPDIF.conf	2020-06-22 10:09:54.993333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-P201/GXBB-P201.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-P201/GXBB-P201.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-P201/GXBB-P201.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-P201/GXBB-P201.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic GXBB SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXBB SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-P201/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-P201/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-P201/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-P201/HDMI.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-VEGA-S95/GXBB-VEGA-S95.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-VEGA-S95/GXBB-VEGA-S95.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-VEGA-S95/GXBB-VEGA-S95.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-VEGA-S95/GXBB-VEGA-S95.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic GXBB SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXBB SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on GXBB SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-VEGA-S95/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-VEGA-S95/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-VEGA-S95/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-VEGA-S95/HDMI.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-VEGA-S95/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-VEGA-S95/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-VEGA-S95/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-VEGA-S95/SPDIF.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-WETEK-HUB/GXBB-WETEK-HUB.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-WETEK-HUB/GXBB-WETEK-HUB.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-WETEK-HUB/GXBB-WETEK-HUB.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-WETEK-HUB/GXBB-WETEK-HUB.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic GXBB SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXBB SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-WETEK-HUB/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-WETEK-HUB/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-WETEK-HUB/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-WETEK-HUB/HDMI.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-WETEK-PLAY2/GXBB-WETEK-PLAY2.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-WETEK-PLAY2/GXBB-WETEK-PLAY2.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-WETEK-PLAY2/GXBB-WETEK-PLAY2.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-WETEK-PLAY2/GXBB-WETEK-PLAY2.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic GXBB SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXBB SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on GXBB SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-WETEK-PLAY2/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-WETEK-PLAY2/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-WETEK-PLAY2/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-WETEK-PLAY2/HDMI.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-WETEK-PLAY2/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-WETEK-PLAY2/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXBB-WETEK-PLAY2/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXBB-WETEK-PLAY2/SPDIF.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S805X-AC/GXL-LIBRETECH-S805X-AC.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S805X-AC/GXL-LIBRETECH-S805X-AC.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S805X-AC/GXL-LIBRETECH-S805X-AC.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S805X-AC/GXL-LIBRETECH-S805X-AC.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic GXL SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXL SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S805X-AC/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S805X-AC/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S805X-AC/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S805X-AC/HDMI.conf	2020-06-22 10:09:54.996666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S905X-CC/GXL-LIBRETECH-S905X-CC.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S905X-CC/GXL-LIBRETECH-S905X-CC.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S905X-CC/GXL-LIBRETECH-S905X-CC.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S905X-CC/GXL-LIBRETECH-S905X-CC.conf	2020-06-22 10:09:55.000000000 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic GXL SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXL SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S905X-CC/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S905X-CC/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S905X-CC/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S905X-CC/HDMI.conf	2020-06-22 10:09:55.000000000 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S9XX-PC/GXL-LIBRETECH-S9XX-PC.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S9XX-PC/GXL-LIBRETECH-S9XX-PC.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S9XX-PC/GXL-LIBRETECH-S9XX-PC.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S9XX-PC/GXL-LIBRETECH-S9XX-PC.conf	2020-06-22 10:09:55.003333333 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic GXL SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXL SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S9XX-PC/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S9XX-PC/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-LIBRETECH-S9XX-PC/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-LIBRETECH-S9XX-PC/HDMI.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-S905X-P212/GXL-S905X-P212.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-S905X-P212/GXL-S905X-P212.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-S905X-P212/GXL-S905X-P212.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-S905X-P212/GXL-S905X-P212.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic GXL SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXL SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-S905X-P212/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-S905X-P212/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXL-S905X-P212/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXL-S905X-P212/HDMI.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-BEELINK-GT1/GXM-BEELINK-GT1.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-BEELINK-GT1/GXM-BEELINK-GT1.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-BEELINK-GT1/GXM-BEELINK-GT1.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-BEELINK-GT1/GXM-BEELINK-GT1.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic GXM SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXM SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on GXM SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-BEELINK-GT1/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-BEELINK-GT1/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-BEELINK-GT1/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-BEELINK-GT1/HDMI.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-BEELINK-GT1/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-BEELINK-GT1/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-BEELINK-GT1/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-BEELINK-GT1/SPDIF.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-KHADAS-VIM2/GXM-KHADAS-VIM2.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-KHADAS-VIM2/GXM-KHADAS-VIM2.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-KHADAS-VIM2/GXM-KHADAS-VIM2.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-KHADAS-VIM2/GXM-KHADAS-VIM2.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "Amlogic GXM SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXM SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-KHADAS-VIM2/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-KHADAS-VIM2/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-KHADAS-VIM2/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-KHADAS-VIM2/HDMI.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-NEXBOX-A1/GXM-NEXBOX-A1.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-NEXBOX-A1/GXM-NEXBOX-A1.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-NEXBOX-A1/GXM-NEXBOX-A1.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-NEXBOX-A1/GXM-NEXBOX-A1.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic GXM SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXM SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on GXM SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-NEXBOX-A1/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-NEXBOX-A1/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-NEXBOX-A1/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-NEXBOX-A1/HDMI.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-NEXBOX-A1/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-NEXBOX-A1/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-NEXBOX-A1/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-NEXBOX-A1/SPDIF.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-RBOX-PRO/GXM-RBOX-PRO.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-RBOX-PRO/GXM-RBOX-PRO.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-RBOX-PRO/GXM-RBOX-PRO.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-RBOX-PRO/GXM-RBOX-PRO.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic GXM SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXM SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on GXM SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-RBOX-PRO/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-RBOX-PRO/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-RBOX-PRO/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-RBOX-PRO/HDMI.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-RBOX-PRO/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-RBOX-PRO/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-RBOX-PRO/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-RBOX-PRO/SPDIF.conf	2020-06-22 10:09:55.006666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-WETEK-CORE2/GXM-WETEK-CORE2.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-WETEK-CORE2/GXM-WETEK-CORE2.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-WETEK-CORE2/GXM-WETEK-CORE2.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-WETEK-CORE2/GXM-WETEK-CORE2.conf	2020-06-22 10:09:55.013333333 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic GXM SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GXM SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on GXM SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-WETEK-CORE2/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-WETEK-CORE2/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-WETEK-CORE2/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-WETEK-CORE2/HDMI.conf	2020-06-22 10:09:55.013333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-WETEK-CORE2/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-WETEK-CORE2/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GXM-WETEK-CORE2/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GXM-WETEK-CORE2/SPDIF.conf	2020-06-22 10:09:55.013333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GX-P230-Q200/GX-P230-Q200.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GX-P230-Q200/GX-P230-Q200.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GX-P230-Q200/GX-P230-Q200.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GX-P230-Q200/GX-P230-Q200.conf	2020-06-22 10:09:55.020000000 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic GX SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on GX SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on GX SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GX-P230-Q200/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GX-P230-Q200/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GX-P230-Q200/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GX-P230-Q200/HDMI.conf	2020-06-22 10:09:55.020000000 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GX-P230-Q200/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GX-P230-Q200/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/GX-P230-Q200/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/GX-P230-Q200/SPDIF.conf	2020-06-22 10:09:55.020000000 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-gx/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-gx/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-A95XF3-AIR/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-A95XF3-AIR/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-A95XF3-AIR/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-A95XF3-AIR/HDMI.conf	2020-06-22 10:09:55.026666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-A95XF3-AIR/SM1-A95XF3-AIR.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-A95XF3-AIR/SM1-A95XF3-AIR.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-A95XF3-AIR/SM1-A95XF3-AIR.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-A95XF3-AIR/SM1-A95XF3-AIR.conf	2020-06-22 10:09:55.026666666 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic SM1 SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on SM1 SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on SM1 SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-A95XF3-AIR/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-A95XF3-AIR/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-A95XF3-AIR/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-A95XF3-AIR/SPDIF.conf	2020-06-22 10:09:55.026666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-KHADAS-VIM3L/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-KHADAS-VIM3L/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-KHADAS-VIM3L/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-KHADAS-VIM3L/HDMI.conf	2020-06-22 10:09:55.033333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-KHADAS-VIM3L/SM1-KHADAS-VIM3L.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-KHADAS-VIM3L/SM1-KHADAS-VIM3L.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-KHADAS-VIM3L/SM1-KHADAS-VIM3L.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-KHADAS-VIM3L/SM1-KHADAS-VIM3L.conf	2020-06-22 10:09:55.033333333 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic SM1 SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on SM1 SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on SM1 SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-KHADAS-VIM3L/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-KHADAS-VIM3L/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-KHADAS-VIM3L/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-KHADAS-VIM3L/SPDIF.conf	2020-06-22 10:09:55.033333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-ODROID-C4/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-ODROID-C4/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-ODROID-C4/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-ODROID-C4/HDMI.conf	2020-06-22 10:09:55.033333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-ODROID-C4/SM1-ODROID-C4.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-ODROID-C4/SM1-ODROID-C4.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-ODROID-C4/SM1-ODROID-C4.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-ODROID-C4/SM1-ODROID-C4.conf	2020-06-22 10:09:55.033333333 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic SM1 SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on SM1 SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on SM1 SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-ODROID-C4/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-ODROID-C4/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-ODROID-C4/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-ODROID-C4/SPDIF.conf	2020-06-22 10:09:55.033333333 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-SEI610/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-SEI610/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-SEI610/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-SEI610/HDMI.conf	2020-06-22 10:09:55.040000000 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-SEI610/SM1-SEI610.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-SEI610/SM1-SEI610.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-SEI610/SM1-SEI610.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-SEI610/SM1-SEI610.conf	2020-06-22 10:09:55.040000000 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic SM1 SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on SM1 SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on SM1 SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-SEI610/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-SEI610/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-SEI610/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-SEI610/SPDIF.conf	2020-06-22 10:09:55.040000000 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-X96-AIR/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-X96-AIR/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-X96-AIR/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-X96-AIR/HDMI.conf	2020-06-22 10:09:55.046666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-X96-AIR/SM1-X96-AIR.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-X96-AIR/SM1-X96-AIR.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-X96-AIR/SM1-X96-AIR.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-X96-AIR/SM1-X96-AIR.conf	2020-06-22 10:09:55.046666666 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic G12 SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on G12 SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on G12 SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-X96-AIR/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-X96-AIR/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-X96-AIR/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-X96-AIR/SPDIF.conf	2020-06-22 10:09:55.046666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableSPDIF.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-TX5-PLUS/HDMI.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-TX5-PLUS/HDMI.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-TX5-PLUS/HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-TX5-PLUS/HDMI.conf	2020-06-22 10:09:55.046666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "HDMI"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableHDMI.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableHDMI.conf>
+	]
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-TX5-PLUS/SM1-TX5-PLUS.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-TX5-PLUS/SM1-TX5-PLUS.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-TX5-PLUS/SM1-TX5-PLUS.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-TX5-PLUS/SM1-TX5-PLUS.conf	2020-06-22 10:09:55.046666666 +0200
@@ -0,0 +1,10 @@
+Syntax 2
+Comment "Amlogic G12 SoC Audio Device"
+SectionUseCase."HDMI" {
+	File "HDMI.conf"
+	Comment "HDMI on G12 SoC"
+}
+SectionUseCase."SPDIF" {
+	File "SPDIF.conf"
+	Comment "SPDIF on G12 SoC"
+}
diff -Naur alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-TX5-PLUS/SPDIF.conf alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-TX5-PLUS/SPDIF.conf
--- alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-old/ucm2/SM1-TX5-PLUS/SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-ucm-conf-4c1189fdcb4819df6c36cf490d17f532f82c8bbe-new/ucm2/SM1-TX5-PLUS/SPDIF.conf	2020-06-22 10:09:55.046666666 +0200
@@ -0,0 +1,22 @@
+SectionVerb {
+	Value {
+		TQ "SPDIF"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+	}
+
+	EnableSequence [
+		<codecs/amlogic-g12/EnableSPDIF.conf>
+	]
+
+	DisableSequence [
+		<codecs/amlogic-g12/DisableSPDIF.conf>
+	]
+}
