diff -Naur linux-3.7.3-old/drivers/ata/libata-core.c linux-3.7.3-new/drivers/ata/libata-core.c
--- linux-3.7.3-old/drivers/ata/libata-core.c	2013-01-19 16:32:08.000000000 -0800
+++ linux-3.7.3-new/drivers/ata/libata-core.c	2013-01-19 16:32:40.000000000 -0800
@@ -4390,6 +4390,12 @@
 			     "simplex DMA is claimed by other device, disabling DMA\n");
 	}
 
+	if (ata_id_is_cfa(dev->id)) {
+		xfer_mask &= ~(ATA_MASK_MWDMA | ATA_MASK_UDMA);
+		ata_dev_printk(dev, KERN_WARNING,
+			       "device is a CFA, disabling DMA\n");
+	}
+
 	if (ap->flags & ATA_FLAG_NO_IORDY)
 		xfer_mask &= ata_pio_mask_no_iordy(dev);
 
diff -Naur linux-3.7.3-old/drivers/ide/ide-dma.c linux-3.7.3-new/drivers/ide/ide-dma.c
--- linux-3.7.3-old/drivers/ide/ide-dma.c	2013-01-19 16:32:08.000000000 -0800
+++ linux-3.7.3-new/drivers/ide/ide-dma.c	2013-01-19 16:32:40.000000000 -0800
@@ -226,6 +226,12 @@
 				    drive->name, (char *)&id[ATA_ID_PROD]);
 		return blacklist;
 	}
+	if (id->config == 0x848a) {
+		printk(KERN_WARNING "%s: Disabling (U)DMA for %s (CFA)\n",
+				    drive->name, id->model);
+		return 1;
+	}
+
 	return 0;
 }
 EXPORT_SYMBOL(__ide_dma_bad_drive);
