diff -Naur linux-3.7.3-old/drivers/net/ethernet/via/via-rhine.c linux-3.7.3-new/drivers/net/ethernet/via/via-rhine.c
--- linux-3.7.3-old/drivers/net/ethernet/via/via-rhine.c	2013-01-19 16:31:24.000000000 -0800
+++ linux-3.7.3-new/drivers/net/ethernet/via/via-rhine.c	2013-01-19 16:31:55.000000000 -0800
@@ -55,7 +55,7 @@
 
 /* Work-around for broken BIOSes: they are unable to get the chip back out of
    power state D3 so PXE booting fails. bootparam(7): via-rhine.avoid_D3=1 */
-static bool avoid_D3;
+static int avoid_D3 = 0;
 
 /*
  * In case you are looking for 'options[]' or 'full_duplex[]', they
@@ -129,7 +129,7 @@
 
 module_param(debug, int, 0);
 module_param(rx_copybreak, int, 0);
-module_param(avoid_D3, bool, 0);
+module_param(avoid_D3, bool, 0644);
 MODULE_PARM_DESC(debug, "VIA Rhine debug message flags");
 MODULE_PARM_DESC(rx_copybreak, "VIA Rhine copy breakpoint for copy-only-tiny-frames");
 MODULE_PARM_DESC(avoid_D3, "Avoid power state D3 (work-around for broken BIOSes)");
