diff -Naur linux-5.3-rc6-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts linux-5.3-rc6-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
--- linux-5.3-rc6-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2019-09-02 16:26:43.473333341 +0200
+++ linux-5.3-rc6-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2019-09-02 16:35:01.436666672 +0200
@@ -25,6 +25,7 @@
 	connector {
 		compatible = "hdmi-connector";
 		type = "a";
+		ddc-supply = <&reg_ddc>;
 
 		port {
 			hdmi_con_in: endpoint {
@@ -43,6 +44,15 @@
 		};
 	};
 
+	reg_ddc: ddc-io {
+		compatible = "regulator-fixed";
+		regulator-name = "ddc-io";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		gpio = <&pio 7 2 GPIO_ACTIVE_HIGH>; /* PH2 */
+	};
+
 	reg_vcc5v: vcc5v {
 		/* board wide 5V supply directly from the DC jack */
 		compatible = "regulator-fixed";
