diff --speed-large-files --no-dereference --minimal -Naur u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-orangepi.dts u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-orangepi.dts
--- u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-orangepi.dts	2025-02-12 00:02:34.000000000 +0100
+++ u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-orangepi.dts	2025-02-20 09:18:06.668257325 +0100
@@ -17,9 +17,9 @@
 
 	aliases {
 		ethernet0 = &gmac;
-		mmc0 = &sdio0;
-		mmc1 = &sdmmc;
-		mmc2 = &sdhci;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
+		mmc2 = &sdio0;
 	};
 
 	chosen {
@@ -233,20 +233,6 @@
 	};
 };
 
-&gpu {
-	mali-supply = <&vdd_gpu>;
-	status = "okay";
-};
-
-&hdmi {
-	ddc-i2c-bus = <&i2c3>;
-	status = "okay";
-};
-
-&hdmi_sound {
-	status = "okay";
-};
-
 &i2c0 {
 	clock-frequency = <400000>;
 	i2c-scl-rising-time-ns = <168>;
@@ -877,19 +863,3 @@
 	status = "okay";
 	dr_mode = "host";
 };
-
-&vopb {
-	status = "okay";
-};
-
-&vopb_mmu {
-	status = "okay";
-};
-
-&vopl {
-	status = "okay";
-};
-
-&vopl_mmu {
-	status = "okay";
-};
diff --speed-large-files --no-dereference --minimal -Naur u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-4c-plus.dts u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-4c-plus.dts
--- u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-4c-plus.dts	2025-02-12 00:02:34.000000000 +0100
+++ u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-4c-plus.dts	2025-02-20 09:18:06.668257325 +0100
@@ -15,8 +15,8 @@
 
 	aliases {
 		ethernet0 = &gmac;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
@@ -185,24 +185,6 @@
 	status = "okay";
 };
 
-&gpu {
-	mali-supply = <&vdd_gpu>;
-	status = "okay";
-};
-
-&hdmi {
-	avdd-0v9-supply = <&vcc_0v9_s0>;
-	avdd-1v8-supply = <&vcc_1v8_s0>;
-	ddc-i2c-bus = <&i2c3>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&hdmi_cec>;
-	status = "okay";
-};
-
-&hdmi_sound {
-	status = "okay";
-};
-
 &i2c0 {
 	status = "okay";
 	i2c-scl-falling-time-ns = <30>;
@@ -483,10 +465,6 @@
 		<3 RK_PD7 1 &pcfg_pull_none>;
 };
 
-&i2s2 {
-	status = "okay";
-};
-
 &io_domains {
 	audio-supply = <&vcc_1v8_s0>;
 	bt656-supply = <&vcc_3v0_s0>;
@@ -736,19 +714,3 @@
 	status = "okay";
 	dr_mode = "host";
 };
-
-&vopb {
-	status = "okay";
-};
-
-&vopb_mmu {
-	status = "okay";
-};
-
-&vopl {
-	status = "okay";
-};
-
-&vopl_mmu {
-	status = "okay";
-};
diff --speed-large-files --no-dereference --minimal -Naur u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-pi-4c.dts u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-pi-4c.dts
--- u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-pi-4c.dts	2025-02-12 00:02:34.000000000 +0100
+++ u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-pi-4c.dts	2025-02-20 09:18:41.264924098 +0100
@@ -18,13 +18,6 @@
 	};
 };
 
-&es8316 {
-	pinctrl-0 = <&hp_detect &hp_int>;
-	pinctrl-names = "default";
-	interrupt-parent = <&gpio1>;
-	interrupts = <RK_PA1 IRQ_TYPE_LEVEL_HIGH>;
-};
-
 &sdio0 {
 	status = "okay";
 
@@ -39,10 +32,6 @@
 	};
 };
 
-&sound {
-	hp-det-gpios = <&gpio1 RK_PA0 GPIO_ACTIVE_HIGH>;
-};
-
 &spi1 {
 	status = "okay";
 
diff --speed-large-files --no-dereference --minimal -Naur u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-pi-4.dtsi u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-pi-4.dtsi
--- u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-pi-4.dtsi	2025-02-12 00:02:34.000000000 +0100
+++ u-boot-rockchip-228e9e5b3502ec0e3aac3fae38d9d99f77e9ede1/dts/upstream/src/arm64/rockchip/rk3399-rock-pi-4.dtsi	2025-02-20 09:19:01.361590830 +0100
@@ -11,8 +11,8 @@
 / {
 	aliases {
 		ethernet0 = &gmac;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
@@ -49,29 +49,6 @@
 		reset-gpios = <&gpio0 RK_PB2 GPIO_ACTIVE_LOW>;
 	};
 
-	sound: sound {
-		compatible = "audio-graph-card";
-		label = "Analog";
-		dais = <&i2s0_p0>;
-	};
-
-	sound-dit {
-		compatible = "audio-graph-card";
-		label = "SPDIF";
-		dais = <&spdif_p0>;
-	};
-
-	spdif-dit {
-		compatible = "linux,spdif-dit";
-		#sound-dai-cells = <0>;
-
-		port {
-			dit_p0_0: endpoint {
-				remote-endpoint = <&spdif_p0_0>;
-			};
-		};
-	};
-
 	vbus_typec: regulator-vbus-typec {
 		compatible = "regulator-fixed";
 		enable-active-high;
@@ -212,24 +189,6 @@
 	status = "okay";
 };
 
-&gpu {
-	mali-supply = <&vdd_gpu>;
-	status = "okay";
-};
-
-&hdmi {
-	avdd-0v9-supply = <&vcca0v9_hdmi>;
-	avdd-1v8-supply = <&vcca1v8_hdmi>;
-	ddc-i2c-bus = <&i2c3>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&hdmi_cec>;
-	status = "okay";
-};
-
-&hdmi_sound {
-	status = "okay";
-};
-
 &i2c0 {
 	clock-frequency = <400000>;
 	i2c-scl-rising-time-ns = <168>;
@@ -458,26 +417,6 @@
 	};
 };
 
-&i2c1 {
-	i2c-scl-rising-time-ns = <300>;
-	i2c-scl-falling-time-ns = <15>;
-	status = "okay";
-
-	es8316: codec@11 {
-		compatible = "everest,es8316";
-		reg = <0x11>;
-		clocks = <&cru SCLK_I2S_8CH_OUT>;
-		clock-names = "mclk";
-		#sound-dai-cells = <0>;
-
-		port {
-			es8316_p0_0: endpoint {
-				remote-endpoint = <&i2s0_p0_0>;
-			};
-		};
-	};
-};
-
 &i2c3 {
 	i2c-scl-rising-time-ns = <450>;
 	i2c-scl-falling-time-ns = <15>;
@@ -490,31 +429,6 @@
 	status = "okay";
 };
 
-&i2s0 {
-	pinctrl-0 = <&i2s0_2ch_bus>;
-	pinctrl-1 = <&i2s0_2ch_bus_bclk_off>;
-	rockchip,capture-channels = <2>;
-	rockchip,playback-channels = <2>;
-	status = "okay";
-
-	i2s0_p0: port {
-		i2s0_p0_0: endpoint {
-			dai-format = "i2s";
-			mclk-fs = <256>;
-			remote-endpoint = <&es8316_p0_0>;
-		};
-	};
-};
-
-&i2s1 {
-	rockchip,playback-channels = <2>;
-	rockchip,capture-channels = <2>;
-};
-
-&i2s2 {
-	status = "okay";
-};
-
 &io_domains {
 	audio-supply = <&vcca1v8_codec>;
 	bt656-supply = <&vcc_3v0>;
@@ -680,15 +594,6 @@
 	status = "okay";
 };
 
-&spdif {
-
-	spdif_p0: port {
-		spdif_p0_0: endpoint {
-			remote-endpoint = <&dit_p0_0>;
-		};
-	};
-};
-
 &tcphy0 {
 	status = "okay";
 };
@@ -774,19 +679,3 @@
 	status = "okay";
 	dr_mode = "host";
 };
-
-&vopb {
-	status = "okay";
-};
-
-&vopb_mmu {
-	status = "okay";
-};
-
-&vopl {
-	status = "okay";
-};
-
-&vopl_mmu {
-	status = "okay";
-};
