diff -Naur NVIDIA-Linux-x86_64-340.101-old/kernel/nv-drm.c NVIDIA-Linux-x86_64-340.101-new/kernel/nv-drm.c
--- NVIDIA-Linux-x86_64-340.101-old/kernel/nv-drm.c	2016-12-01 23:48:28.000000000 +0100
+++ NVIDIA-Linux-x86_64-340.101-new/kernel/nv-drm.c	2016-12-18 18:04:44.016666639 +0100
@@ -115,7 +115,11 @@
 };
 
 static struct drm_driver nv_drm_driver = {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+    .driver_features = DRIVER_GEM | DRIVER_PRIME | DRIVER_LEGACY,
+#else
     .driver_features = DRIVER_GEM | DRIVER_PRIME,
+#endif
     .load = nv_drm_load,
     .unload = nv_drm_unload,
     .fops = &nv_drm_fops,
