diff -Naur libcdio-paranoia-10.2+0.90-old/include/cdio/paranoia/paranoia.h libcdio-paranoia-10.2+0.90-new/include/cdio/paranoia/paranoia.h
--- libcdio-paranoia-10.2+0.90-old/include/cdio/paranoia/paranoia.h	2012-03-18 16:17:50.000000000 -0700
+++ libcdio-paranoia-10.2+0.90-new/include/cdio/paranoia/paranoia.h	2012-11-15 06:24:12.000000000 -0800
@@ -15,7 +15,7 @@
 #ifndef CDIO__PARANOIA__PARANOIA_H_
 #define CDIO__PARANOIA__PARANOIA_H_
 
-#include <cdio/paranoia/cdda.h>
+#include <cdio/cdda.h>
 
 /*! Paranoia likes to work with 16-bit numbers rather than
     (possibly byte-swapped) bytes. So there are this many 
