From 98eff3aab138ccbd3153e4174aaead2ae6a56a88 Mon Sep 17 00:00:00 2001
From: Jerome Brunet <jbrunet@baylibre.com>
Date: Fri, 10 Jan 2020 14:02:09 +0100
Subject: [PATCH 86/90] WIP: ASoC: meson-gx: hardcode codec to codec routes

---
 sound/soc/meson/aiu-i2s-encode.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sound/soc/meson/aiu-i2s-encode.c b/sound/soc/meson/aiu-i2s-encode.c
index c8c67cf..f99d42a 100644
--- a/sound/soc/meson/aiu-i2s-encode.c
+++ b/sound/soc/meson/aiu-i2s-encode.c
@@ -165,6 +165,13 @@ static int aiu_i2s_encode_set_clocks(struct snd_soc_component *component,
 				      AIU_CLK_CTRL_MORE_HDMI_AMCLK,
 				      AIU_CLK_CTRL_MORE_HDMI_AMCLK);
 
+	/* REMOVE ME: HARD CODED TEST */
+	/* Set HDMI path */
+	snd_soc_component_write(component, 0xa8, 0x22);
+
+	/* Internal DAC Path */
+	snd_soc_component_write(component, 0xb0, 0x8058);
+
 	return 0;
 }
 
-- 
2.7.1

