diff -Naur linux-5.6-old/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts linux-5.6-new/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts
--- linux-5.6-old/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts	2020-02-26 21:36:40.230000002 +0100
+++ linux-5.6-new/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts	2020-02-26 21:24:05.526666655 +0100
@@ -707,3 +707,12 @@
 &vopl_mmu {
 	status = "okay";
 };
+
+&vdec_mmu {
+	status = "okay";
+};
+
+&vdec {
+	status = "okay";
+};
+
