diff -Naur linux-5.4-rc2-old/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi linux-5.4-rc2-new/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi
--- linux-5.4-rc2-old/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi	2019-10-07 17:13:24.183333335 +0200
+++ linux-5.4-rc2-new/arch/arm64/boot/dts/allwinner/sun50i-h6.dtsi	2019-10-07 16:32:33.270000005 +0200
@@ -39,6 +39,7 @@
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
+			operating-points-v2 = <&cpu_opp_table>;
 			#cooling-cells = <2>;
 		};
 
@@ -49,6 +50,7 @@
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
+			operating-points-v2 = <&cpu_opp_table>;
 			#cooling-cells = <2>;
 		};
 
@@ -59,6 +61,7 @@
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
+			operating-points-v2 = <&cpu_opp_table>;
 			#cooling-cells = <2>;
 		};
 
@@ -69,10 +72,81 @@
 			enable-method = "psci";
 			clocks = <&ccu CLK_CPUX>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
+			operating-points-v2 = <&cpu_opp_table>;
 			#cooling-cells = <2>;
 		};
 	};
 
+	cpu_opp_table: opp_table {
+		compatible = "allwinner,sun50i-h6-operating-points";
+		nvmem-cells = <&speedbin_efuse>;
+		opp-shared;
+
+		opp@480000000 {
+			opp-hz = /bits/ 64 <480000000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-microvolt-speed0 = <880000>;
+			opp-microvolt-speed1 = <820000>;
+			opp-microvolt-speed2 = <800000>;
+		};
+
+		opp@720000000 {
+			opp-hz = /bits/ 64 <720000000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-microvolt-speed0 = <880000>;
+			opp-microvolt-speed1 = <820000>;
+			opp-microvolt-speed2 = <800000>;
+		};
+
+		opp@816000000 {
+			opp-hz = /bits/ 64 <816000000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-microvolt-speed0 = <880000>;
+			opp-microvolt-speed1 = <820000>;
+			opp-microvolt-speed2 = <800000>;
+		};
+
+		opp@888000000 {
+			opp-hz = /bits/ 64 <888000000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-microvolt-speed0 = <940000>;
+			opp-microvolt-speed1 = <820000>;
+			opp-microvolt-speed2 = <800000>;
+		};
+
+		opp@1080000000 {
+			opp-hz = /bits/ 64 <1080000000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-microvolt-speed0 = <1060000>;
+			opp-microvolt-speed1 = <880000>;
+			opp-microvolt-speed2 = <840000>;
+		};
+
+		opp@1320000000 {
+			opp-hz = /bits/ 64 <1320000000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-microvolt-speed0 = <1160000>;
+			opp-microvolt-speed1 = <940000>;
+			opp-microvolt-speed2 = <900000>;
+		};
+
+		opp@1488000000 {
+			opp-hz = /bits/ 64 <1488000000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-microvolt-speed0 = <1160000>;
+			opp-microvolt-speed1 = <1000000>;
+			opp-microvolt-speed2 = <960000>;
+		};
+
+		opp@1800000000 {
+			opp-hz = /bits/ 64 <1800000000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+			opp-microvolt-speed0 = <1160000>;
+			opp-microvolt-speed1 = <1160000>;
+			opp-microvolt-speed2 = <1160000>;
+		};
+	};
+
 	de: display-engine {
 		compatible = "allwinner,sun50i-h6-display-engine";
 		allwinner,pipelines = <&mixer0>;
@@ -267,6 +341,10 @@
 			#address-cells = <1>;
 			#size-cells = <1>;
 
+			speedbin_efuse: speed@1c {
+				reg = <0x1c 4>;
+			};
+
 			ephy_calib: ephy_calib@2c {
 				reg = <0x2c 0x2>;
 			};
