diff -Naur urlgrabber-3.9.1-old/setup.py urlgrabber-3.9.1-new/setup.py
--- urlgrabber-3.9.1-old/setup.py	2011-03-03 15:17:20.000000000 -0800
+++ urlgrabber-3.9.1-new/setup.py	2011-03-03 16:12:44.000000000 -0800
@@ -1,16 +1,16 @@
 # urlgrabber distutils setup
-import re as _re
-import urlgrabber as _urlgrabber
+#import re as _re
+#import urlgrabber as _urlgrabber
 
 name = "urlgrabber"
 description = "A high-level cross-protocol url-grabber"
-long_description = _urlgrabber.__doc__
+#long_description = _urlgrabber.__doc__
 license = "LGPL"
-version = _urlgrabber.__version__
-_authors = _re.split(r',\s+', _urlgrabber.__author__)
-author       = ', '.join([_re.sub(r'\s+<.*',        r'', _) for _ in _authors])
-author_email = ', '.join([_re.sub(r'(^.*<)|(>.*$)', r'', _) for _ in _authors])
-url = _urlgrabber.__url__
+version = "3.9.1"
+#_authors = _re.split(r',\s+', _urlgrabber.__author__)
+#author       = ', '.join([_re.sub(r'\s+<.*',        r'', _) for _ in _authors])
+#author_email = ', '.join([_re.sub(r'(^.*<)|(>.*$)', r'', _) for _ in _authors])
+#url = _urlgrabber.__url__
 
 packages = ['urlgrabber']
 package_dir = {'urlgrabber':'urlgrabber'}
