diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts	2023-12-07 09:39:40.666634856 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts	2023-12-07 09:36:27.966639016 +0100
@@ -9,6 +9,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include "sun50i-h6-cpu-opp.dtsi"
+#include "sun50i-h6-gpu-opp.dtsi"
 
 / {
 	model = "Eachlink H6 Mini";
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2023-12-07 09:39:41.253301510 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2023-12-07 09:36:27.966639016 +0100
@@ -5,6 +5,7 @@
 
 #include "sun50i-h6.dtsi"
 #include "sun50i-h6-cpu-opp.dtsi"
+#include "sun50i-h6-gpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi	2023-12-03 07:33:10.000000000 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi	2023-12-07 09:36:27.966639016 +0100
@@ -5,6 +5,7 @@
 /dts-v1/;
 
 #include "sun50i-h6.dtsi"
+#include "sun50i-h6-gpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts	2023-12-03 07:33:10.000000000 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts	2023-12-07 09:36:27.966639016 +0100
@@ -5,6 +5,7 @@
 
 #include "sun50i-h6.dtsi"
 #include "sun50i-h6-cpu-opp.dtsi"
+#include "sun50i-h6-gpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi
--- linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi	2023-12-07 09:39:40.849968185 +0100
+++ linux-6.6.4/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix.dtsi	2023-12-07 09:36:27.966639016 +0100
@@ -5,6 +5,7 @@
 
 #include "sun50i-h6.dtsi"
 #include "sun50i-h6-cpu-opp.dtsi"
+#include "sun50i-h6-gpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
