diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2023-11-15 19:11:29.293341522 +0100
+++ linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2023-11-15 18:53:47.160004623 +0100
@@ -384,6 +384,18 @@
 				pins = "PG10";
 				function = "clock";
 			};
+
+			/omit-if-no-ref/
+			spdif_pins: spdif-pins {
+				pins = "PH2", "PH4"; // MCLK, SPDIF-TX
+				function = "spdif";
+			};
+
+			/omit-if-no-ref/
+			spdif_tx_pin: spdif-tx-pin {
+				pins = "PH4";
+				function = "spdif";
+			};
 		};
 
 		gic: interrupt-controller@3021000 {
@@ -707,6 +719,22 @@
 			};
 		};
 
+		spdif: spdif@5093000 {
+			#sound-dai-cells = <0>;
+			// H6 spdif is compatible as long as only tx is used. Now sun4i-spdif module only supports tx.
+			compatible = "allwinner,sun50i-h616-spdif", "allwinner,sun50i-h6-spdif";
+			reg = <0x05093000 0x400>;
+			interrupts = <GIC_SPI 21 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&ccu CLK_BUS_SPDIF>, <&ccu CLK_SPDIF>;
+			clock-names = "apb", "spdif";
+			resets = <&ccu RST_BUS_SPDIF>;
+			dmas = <&dma 2>;
+			dma-names = "tx";
+			pinctrl-names = "default";
+			pinctrl-0 = <&spdif_tx_pin>;
+			status = "disabled";
+		};
+
 		uart0: serial@5000000 {
 			compatible = "snps,dw-apb-uart";
 			reg = <0x05000000 0x400>;
