diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero.dtsi linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero.dtsi
--- linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero.dtsi	2023-11-15 19:11:29.330008189 +0100
+++ linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero.dtsi	2023-11-15 18:59:46.420005854 +0100
@@ -102,6 +102,24 @@
 		reset-gpios = <&pio 6 18 GPIO_ACTIVE_LOW>; /* PG18 */
 		post-power-on-delay-ms = <200>;
 	};
+
+	sound-spdif {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "On-board-SPDIF";
+
+		simple-audio-card,cpu {
+			sound-dai = <&spdif>;
+		};
+
+		simple-audio-card,codec {
+			sound-dai = <&spdif_out>;
+		};
+	};
+
+	spdif_out: spdif-out {
+		#sound-dai-cells = <0>;
+		compatible = "linux,spdif-dit";
+	};
 };
 
 &ehci1 {
@@ -250,3 +268,7 @@
 	linux,rc-map-name = "rc-beelink-gs1";
 	status = "okay";
 };
+
+&spdif {
+	status = "okay";
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts
--- linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts	2023-11-15 19:11:29.370008189 +0100
+++ linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h616-tanix-tx6s.dts	2023-11-15 19:02:34.663339755 +0100
@@ -77,6 +77,24 @@
 		pinctrl-0 = <&clk_losc>; /* PG10 with MUX3 set */
 	};
 
+	sound-spdif {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "On-board-SPDIF";
+
+		simple-audio-card,cpu {
+			sound-dai = <&spdif>;
+		};
+
+		simple-audio-card,codec {
+			sound-dai = <&spdif_out>;
+		};
+	};
+
+	spdif_out: spdif-out {
+		#sound-dai-cells = <0>;
+		compatible = "linux,spdif-dit";
+	};
+
 	openvfd {
 		compatible = "open,vfd";
 		dev_name = "openvfd";
@@ -369,3 +387,6 @@
 	status = "okay";
 };
 
+&spdif {
+	status = "okay";
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h618-vontar-h618.dts linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h618-vontar-h618.dts
--- linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h618-vontar-h618.dts	2023-11-15 19:11:29.520008190 +0100
+++ linux-6.6.1/arch/arm64/boot/dts/allwinner/sun50i-h618-vontar-h618.dts	2023-11-15 18:59:30.110005796 +0100
@@ -77,6 +77,24 @@
 		pinctrl-0 = <&clk_losc>; /* PG10 with MUX3 set */
 	};
 
+	sound-spdif {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "On-board-SPDIF";
+
+		simple-audio-card,cpu {
+			sound-dai = <&spdif>;
+		};
+
+		simple-audio-card,codec {
+			sound-dai = <&spdif_out>;
+		};
+	};
+
+	spdif_out: spdif-out {
+		#sound-dai-cells = <0>;
+		compatible = "linux,spdif-dit";
+	};
+
 	openvfd {
 		compatible = "open,vfd";
 		dev_name = "openvfd";
@@ -312,3 +330,6 @@
 	status = "okay";
 };
 
+&spdif {
+	status = "okay";
+};
