diff --speed-large-files --no-dereference --minimal -Naur linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts
--- linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2023-11-04 12:07:27.186675237 +0100
+++ linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-a1.dts	2023-11-04 12:05:38.403341544 +0100
@@ -116,6 +116,18 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&usb30_host_drv>;
 		regulator-name = "vcc_host_5v";
+		regulator-always-on;
+		vin-supply = <&vcc_sys>;
+	};
+
+	vcc_host1_5v: vcc_otg_5v: vcc-host1-5v-regulator {
+		compatible = "regulator-fixed";
+		enable-active-high;
+		gpio = <&gpio0 RK_PD3 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&usb20_host_drv>;
+		regulator-name = "vcc_host1_5v";
+		regulator-always-on;
 		vin-supply = <&vcc_sys>;
 	};
 
@@ -449,6 +461,12 @@
 		};
 	};
 
+	usb2 {
+		usb20_host_drv: usb20-host-drv {
+			rockchip,pins = <1 RK_PD2 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
 	wifi {
 		bt_dis: bt-dis {
 			rockchip,pins = <2 RK_PC5 RK_FUNC_GPIO &pcfg_output_low>;
@@ -528,3 +546,24 @@
 &vop_mmu {
 	status = "okay";
 };
+
+&usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
+	dr_mode = "host";
+	status = "okay";
+};
+
+&usb3phy {
+	status = "okay";
+};
+
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
+	status = "okay";
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts
--- linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts	2023-11-04 12:07:27.373341905 +0100
+++ linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts	2023-11-04 12:02:12.900007528 +0100
@@ -436,16 +436,32 @@
 	status = "okay";
 };
 
+&usb_host0_ehci {
+	status = "okay";
+};
+
+&usb_host0_ohci {
+	status = "okay";
+};
+
 &usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
 	dr_mode = "host";
 	status = "okay";
 };
 
-&usb_host0_ehci {
+&usb3phy {
 	status = "okay";
 };
 
-&usb_host0_ohci {
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
 	status = "okay";
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts
--- linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts	2023-11-04 12:07:27.373341905 +0100
+++ linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts	2023-11-04 12:02:12.900007528 +0100
@@ -372,16 +372,32 @@
 	status = "okay";
 };
 
+&usb_host0_ehci {
+	status = "okay";
+};
+
+&usb_host0_ohci {
+	status = "okay";
+};
+
 &usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
 	dr_mode = "host";
 	status = "okay";
 };
 
-&usb_host0_ehci {
+&usb3phy {
 	status = "okay";
 };
 
-&usb_host0_ohci {
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
 	status = "okay";
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts
--- linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts	2023-11-04 12:07:27.373341905 +0100
+++ linux-6.6/arch/arm64/boot/dts/rockchip/rk3328-rock-pi-e.dts	2023-11-04 12:02:12.900007528 +0100
@@ -381,11 +381,28 @@
 	status = "okay";
 };
 
+&usb_host0_ehci {
+	status = "okay";
+};
+
 &usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
 	dr_mode = "host";
 	status = "okay";
 };
 
-&usb_host0_ehci {
+&usb3phy {
 	status = "okay";
 };
+
+&usb3phy_utmi {
+	status = "okay";
+};
+
+&usb3phy_pipe {
+	status = "okay";
+};
+
