diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
--- linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2023-11-09 14:45:19.576697893 +0100
+++ linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts	2023-11-09 14:43:41.720030892 +0100
@@ -1227,3 +1227,32 @@
 &wdt {
 	status = "okay";
 };
+
+&display_subsystem {
+    clocks = <&hdptxphy_hdmi_clk0>;
+    clock-names = "hdmi0_phy_pll";
+};
+
+&hdmi0 {
+    status = "okay";
+};
+
+&hdmi0_in_vp0 {
+    status = "okay";
+};
+
+&hdptxphy_hdmi0 {
+    status = "okay";
+};
+
+&hdptxphy_hdmi_clk0 {
+    status = "okay";
+};
+
+&vop_mmu {
+    status = "okay";
+};
+
+&vop {
+    status = "okay";
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
--- linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2023-11-09 14:45:19.686697893 +0100
+++ linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2023-11-09 14:43:50.613364259 +0100
@@ -896,3 +896,32 @@
 &usb_host2_xhci {
 	status = "okay";
 };
+
+&display_subsystem {
+    clocks = <&hdptxphy_hdmi_clk0>;
+    clock-names = "hdmi0_phy_pll";
+};
+
+&hdmi0 {
+    status = "okay";
+};
+
+&hdmi0_in_vp0 {
+    status = "okay";
+};
+
+&hdptxphy_hdmi0 {
+    status = "okay";
+};
+
+&hdptxphy_hdmi_clk0 {
+    status = "okay";
+};
+
+&vop_mmu {
+    status = "okay";
+};
+
+&vop {
+    status = "okay";
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts
--- linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2023-11-09 14:45:19.613364560 +0100
+++ linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts	2023-11-09 14:44:00.853364299 +0100
@@ -1277,3 +1277,32 @@
 &wdt {
 	status = "okay";
 };
+
+&display_subsystem {
+    clocks = <&hdptxphy_hdmi_clk0>;
+    clock-names = "hdmi0_phy_pll";
+};
+
+&hdmi0 {
+    status = "okay";
+};
+
+&hdmi0_in_vp0 {
+    status = "okay";
+};
+
+&hdptxphy_hdmi0 {
+    status = "okay";
+};
+
+&hdptxphy_hdmi_clk0 {
+    status = "okay";
+};
+
+&vop_mmu {
+    status = "okay";
+};
+
+&vop {
+    status = "okay";
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
--- linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2023-11-09 14:45:19.650031227 +0100
+++ linux-6.6.1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2023-11-09 14:44:08.053364323 +0100
@@ -796,3 +796,32 @@
 &usb_host2_xhci {
 	status = "okay";
 };
+
+&display_subsystem {
+    clocks = <&hdptxphy_hdmi_clk0>;
+    clock-names = "hdmi0_phy_pll";
+};
+
+&hdmi0 {
+    status = "okay";
+};
+
+&hdmi0_in_vp0 {
+    status = "okay";
+};
+
+&hdptxphy_hdmi0 {
+    status = "okay";
+};
+
+&hdptxphy_hdmi_clk0 {
+    status = "okay";
+};
+
+&vop_mmu {
+    status = "okay";
+};
+
+&vop {
+    status = "okay";
+};
