diff -Naur glu-9.0.0-20160610-0bf42e4-old/src/libutil/error.c glu-9.0.0-20160610-0bf42e4-new/src/libutil/error.c
--- glu-9.0.0-20160610-0bf42e4-old/src/libutil/error.c	2018-07-13 11:58:44.000000000 +0200
+++ glu-9.0.0-20160610-0bf42e4-new/src/libutil/error.c	2019-07-10 18:11:16.263333334 +0200
@@ -31,7 +31,7 @@
 #include "gluos.h"
 #include "gluint.h"
 #include <GL/glu.h>
-
+#include <stddef.h>
 
 struct token_string
 {
