diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/commandlineparser.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/commandlineparser.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/commandlineparser.cpp	2014-02-05 16:35:05.332966036 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/commandlineparser.cpp	2014-02-05 16:35:14.642924534 +0100
@@ -149,6 +149,11 @@
         << add("--print-notification-template", "printntemplate", false,
                 "Print the template to be sent to the frontend", "")
                 ->SetGroup("Messaging")
+
+        // musicmetautils.cpp
+        << add("--updatemeta", "updatemeta", false,
+                "Update a music tracks database record and tag with new metadata", "")
+                ->SetGroup("Metadata Reading/Writing")
         );
 
     // mpegutils.cpp
@@ -203,6 +208,28 @@
     add("--fixseektable", "fixseektable", false, "(optional) fix the seektable if missing for a recording", "")
         ->SetChildOf("checkrecordings");
 
+    // musicmetautils.cpp
+    add("--songid", "songid", "", "ID of track to update", "")
+            ->SetChildOf("updatemeta");
+    add("--title", "title", "", "(optional) Title of track", "")
+            ->SetChildOf("updatemeta");
+    add("--artist", "artist", "", "(optional) Artist of track", "")
+            ->SetChildOf("updatemeta");
+    add("--album", "album", "", "(optional) Album of track", "")
+            ->SetChildOf("updatemeta");
+    add("--genre", "genre", "", "(optional) Genre of track", "")
+            ->SetChildOf("updatemeta");
+    add("--trackno", "trackno", "", "(optional) Track No. of track", "")
+            ->SetChildOf("updatemeta");
+    add("--year", "year", "", "(optional) Year of track", "")
+            ->SetChildOf("updatemeta");
+    add("--rating", "rating", "", "(optional) Rating of track", "")
+            ->SetChildOf("updatemeta");
+    add("--playcount", "playcount", "", "(optional) Playcount of track", "")
+            ->SetChildOf("updatemeta");
+    add("--lastplayed", "lastplayed", "", "(optional) Last played of track", "")
+            ->SetChildOf("updatemeta");
+
     // Generic Options used by more than one utility
     addRecording();
     addInFile(true);
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/main.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/main.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/main.cpp	2014-02-05 16:35:05.332966036 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/main.cpp	2014-02-05 16:35:14.782923900 +0100
@@ -22,6 +22,7 @@
 #include "markuputils.h"
 #include "messageutils.h"
 #include "recordingutils.h"
+#include "musicmetautils.h"
 #include "signalhandling.h"
 
 
@@ -107,6 +108,7 @@
     registerMarkupUtils(utilMap);
     registerMessageUtils(utilMap);
     registerRecordingUtils(utilMap);
+    registerMusicUtils(utilMap);
 
     bool cmdFound = false;
     int cmdResult = GENERIC_EXIT_OK;
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/musicmetautils.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/musicmetautils.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/musicmetautils.cpp	1970-01-01 01:00:00.000000000 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/musicmetautils.cpp	2014-02-05 16:35:14.782923900 +0100
@@ -0,0 +1,81 @@
+// libmyth* headers
+#include "exitcodes.h"
+#include "mythlogging.h"
+#include "musicmetadata.h"
+#include "metaio.h"
+#include "mythcontext.h"
+
+// mythutils headers
+#include "commandlineparser.h"
+#include "musicmetautils.h"
+
+static int UpdateMeta(const MythUtilCommandLineParser &cmdline)
+{
+    bool ok = true;
+    int result = GENERIC_EXIT_OK;
+
+    if (cmdline.toString("songid").isEmpty())
+    {
+        LOG(VB_GENERAL, LOG_ERR, "Missing --songid option");
+        return GENERIC_EXIT_INVALID_CMDLINE;
+    }
+    int songID = cmdline.toInt("songid");
+
+    MusicMetadata *mdata = MusicMetadata::createFromID(songID);
+    if (!mdata)
+    {
+        LOG(VB_GENERAL, LOG_ERR, QString("Cannot find metadata for trackid: %1").arg(songID));
+        return GENERIC_EXIT_NOT_OK;
+    }
+
+    if (!cmdline.toString("title").isEmpty())
+        mdata->setTitle(cmdline.toString("title"));
+
+    if (!cmdline.toString("artist").isEmpty())
+        mdata->setArtist(cmdline.toString("artist"));
+
+    if (!cmdline.toString("album").isEmpty())
+        mdata->setAlbum(cmdline.toString("album"));
+
+    if (!cmdline.toString("genre").isEmpty())
+        mdata->setGenre(cmdline.toString("genre"));
+
+    if (!cmdline.toString("trackno").isEmpty())
+        mdata->setTrack(cmdline.toInt("trackno"));
+
+    if (!cmdline.toString("year").isEmpty())
+        mdata->setYear(cmdline.toInt("year"));
+
+    if (!cmdline.toString("rating").isEmpty())
+        mdata->setRating(cmdline.toInt("rating"));
+
+    if (!cmdline.toString("playcount").isEmpty())
+        mdata->setPlaycount(cmdline.toInt("playcount"));
+
+    if (!cmdline.toString("lastplayed").isEmpty())
+        mdata->setLastPlay(cmdline.toDateTime("lastplayed"));
+
+    mdata->dumpToDatabase();
+
+    MetaIO *tagger = mdata->getTagger();
+    if (tagger)
+    {
+        ok = tagger->write(mdata);
+
+        if (!ok)
+            LOG(VB_GENERAL, LOG_ERR, QString("Failed to write to tag for trackid: %1").arg(songID));
+    }
+
+    // tell any clients that the metadata for this track has changed
+    gCoreContext->SendMessage(QString("MUSIC_METADATA_CHANGED %1").arg(songID));
+
+    if (!ok)
+        result = GENERIC_EXIT_NOT_OK;
+
+    return result;
+}
+
+void registerMusicUtils(UtilMap &utilMap)
+{
+    utilMap["updatemeta"] = &UpdateMeta;
+}
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/musicmetautils.h mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/musicmetautils.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/musicmetautils.h	1970-01-01 01:00:00.000000000 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/musicmetautils.h	2014-02-05 16:35:14.782923900 +0100
@@ -0,0 +1,3 @@
+#include "mythutil.h"
+
+void registerMusicUtils(UtilMap &utilMap);
\ No newline at end of file
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/mythutil.pro mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/mythutil.pro
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/mythutil.pro	2014-02-05 16:35:05.336299354 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/mythutil.pro	2014-02-05 16:35:14.782923900 +0100
@@ -21,9 +21,9 @@
 # Input
 HEADERS += mythutil.h commandlineparser.h
 HEADERS += backendutils.h fileutils.h jobutils.h markuputils.h
-HEADERS += messageutils.h mpegutils.h recordingutils.h
+HEADERS += messageutils.h mpegutils.h recordingutils.h musicmetautils.h
 SOURCES += main.cpp mythutil.cpp commandlineparser.cpp
 SOURCES += backendutils.cpp fileutils.cpp jobutils.cpp markuputils.cpp
-SOURCES += messageutils.cpp mpegutils.cpp recordingutils.cpp
+SOURCES += messageutils.cpp mpegutils.cpp recordingutils.cpp musicmetautils.cpp
 
 mingw: LIBS += -lwinmm -lws2_32
