diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/playlist.cpp mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/playlist.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/playlist.cpp	2014-02-05 16:35:13.472929460 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/playlist.cpp	2014-02-05 16:35:20.929562623 +0100
@@ -81,7 +81,7 @@
 
         changed();
 
-        if (update_display)
+        if (update_display && isActivePlaylist())
             gPlayer->activePlaylistChanged(trackID, false);
     }
     else
@@ -125,7 +125,8 @@
 
     changed();
 
-    gPlayer->activePlaylistChanged(trackID, true);
+    if (isActivePlaylist())
+        gPlayer->activePlaylistChanged(trackID, true);
 }
 
 void Playlist::moveTrackUpDown(bool flag, int where_its_at)
@@ -650,7 +651,8 @@
     if (badTrack)
         changed();
 
-    gPlayer->activePlaylistChanged(-1, false);
+    if (isActivePlaylist())
+        gPlayer->activePlaylistChanged(-1, false);
 }
 
 void Playlist::fillSonglistFromQuery(QString whereClause,
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/playlist.h mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/playlist.h
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/playlist.h	2014-02-05 16:35:13.472929460 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/playlist.h	2014-02-05 16:35:21.032895496 +0100
@@ -9,6 +9,7 @@
 #include <musicmetadata.h>
 
 // mythmusic
+#include "playlistcontainer.h"
 #include "musicplayer.h"
 
 class PlaylistContainer;
@@ -102,6 +103,8 @@
     QString getName(void) { return m_name; } 
     void    setName(QString a_name) { m_name = a_name; }
 
+    bool isActivePlaylist(void) { return m_name == DEFAULT_PLAYLIST_NAME; }
+
     int     getID(void) { return m_playlistid; }
     void    setID(int x) { m_playlistid = x; }
 
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/playlistcontainer.cpp mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/playlistcontainer.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/playlistcontainer.cpp	2014-02-05 16:35:13.486262732 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/playlistcontainer.cpp	2014-02-05 16:35:21.032895496 +0100
@@ -1,9 +1,13 @@
+// MythTV
 #include <mythcontext.h>
 #include <mythdb.h>
 #include <compat.h>
+#include <mythlogging.h>
 
+// mythmusic
+#include "playlist.h"
 #include "playlistcontainer.h"
-#include "mythlogging.h"
+
 
 PlaylistLoadingThread::PlaylistLoadingThread(PlaylistContainer *parent_ptr,
                                              AllMusic *all_music_ptr) :
@@ -82,9 +86,9 @@
 
     m_allPlaylists = new QList<Playlist*>;
 
-    m_activePlaylist->loadPlaylist("default_playlist_storage", m_myHost);
+    m_activePlaylist->loadPlaylist(DEFAULT_PLAYLIST_NAME, m_myHost);
 
-    m_streamPlaylist->loadPlaylist("stream_playlist", m_myHost);
+    m_streamPlaylist->loadPlaylist(DEFAULT_STREAMLIST_NAME, m_myHost);
 
     MSqlQuery query(MSqlQuery::InitCon());
     query.prepare("SELECT playlist_id FROM music_playlists "
@@ -93,9 +97,9 @@
                   " AND playlist_name != :STREAM "
                   " AND (hostname = '' OR hostname = :HOST) "
                   "ORDER BY playlist_name;");
-    query.bindValue(":DEFAULT", "default_playlist_storage");
+    query.bindValue(":DEFAULT", DEFAULT_PLAYLIST_NAME);
     query.bindValue(":BACKUP", "backup_playlist_storage");
-    query.bindValue(":STREAM", "stream_playlist");
+    query.bindValue(":STREAM", DEFAULT_STREAMLIST_NAME);
     query.bindValue(":HOST", m_myHost);
 
     if (!query.exec())
@@ -174,8 +178,8 @@
             (*it)->savePlaylist((*it)->getName(), m_myHost);
     }
 
-    m_activePlaylist->savePlaylist("default_playlist_storage", m_myHost);
-    m_streamPlaylist->savePlaylist("stream_playlist", m_myHost);
+    m_activePlaylist->savePlaylist(DEFAULT_PLAYLIST_NAME, m_myHost);
+    m_streamPlaylist->savePlaylist(DEFAULT_STREAMLIST_NAME, m_myHost);
 }
 
 void PlaylistContainer::createNewPlaylist(QString name)
@@ -274,7 +278,7 @@
 
 bool PlaylistContainer::nameIsUnique(QString a_name, int which_id)
 {
-    if (a_name == "default_playlist_storage")
+    if (a_name == DEFAULT_PLAYLIST_NAME)
         return false;
 
     QList<Playlist*>::iterator it = m_allPlaylists->begin();
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/playlistcontainer.h mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/playlistcontainer.h
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/playlistcontainer.h	2014-02-05 16:35:13.486262732 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/playlistcontainer.h	2014-02-05 16:35:21.249561185 +0100
@@ -1,11 +1,19 @@
 #ifndef _PLAYLIST_CONTAINER_H_
 #define _PLAYLIST_CONTAINER_H_
 
+// qt
 #include <QCoreApplication>
 
+// MythTV
 #include "mthread.h"
 
-#include "playlist.h"
+
+#define DEFAULT_PLAYLIST_NAME     "default_playlist_storage"
+#define DEFAULT_STREAMLIST_NAME   "stream_playlist"
+
+class PlaylistContainer;
+class Playlist;
+class AllMusic;
 
 class PlaylistLoadingThread : public MThread
 {
