diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/commandlineparser.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/commandlineparser.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/commandlineparser.cpp	2014-02-05 16:35:26.629537015 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/commandlineparser.cpp	2014-02-05 16:35:36.132827648 +0100
@@ -151,6 +151,10 @@
                 ->SetGroup("Messaging")
 
         // musicmetautils.cpp
+        << add("--scanmusic", "scanmusic", false,
+                "Scan the 'Music' Storage Group for music files", "")
+                ->SetGroup("Music Scanning")
+
         << add("--updatemeta", "updatemeta", false,
                 "Update a music tracks database record and tag with new metadata", "")
                 ->SetGroup("Metadata Reading/Writing")
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/musicmetautils.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/musicmetautils.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/musicmetautils.cpp	2014-02-05 16:35:26.629537015 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/musicmetautils.cpp	2014-02-05 16:35:36.516159255 +0100
@@ -4,6 +4,8 @@
 #include "musicmetadata.h"
 #include "metaio.h"
 #include "mythcontext.h"
+#include "musicfilescanner.h"
+#include "musicutils.h"
 
 // mythutils headers
 #include "commandlineparser.h"
@@ -75,7 +77,18 @@
     return result;
 }
 
+static int ScanMusic(const MythUtilCommandLineParser &cmdline)
+{
+    MusicFileScanner *fscan = new MusicFileScanner();
+    QString musicDir = getMusicDirectory();
+    fscan->SearchDir(musicDir);
+    delete fscan;
+
+    return GENERIC_EXIT_OK;
+}
+
 void registerMusicUtils(UtilMap &utilMap)
 {
     utilMap["updatemeta"] = &UpdateMeta;
+    utilMap["scanmusic"] = &ScanMusic;
 }
