diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:35:50.176097886 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:36:05.366030829 +0100
@@ -179,31 +179,6 @@
                         query);
 }
 
-int MusicMetadata::compare(const MusicMetadata *other) const
-{
-    if (m_format == "cast")
-    {
-        int artist_cmp = Artist().toLower().localeAwareCompare(
-            other->Artist().toLower());
-
-        if (artist_cmp == 0)
-            return Title().toLower().localeAwareCompare(
-                other->Title().toLower());
-
-        return artist_cmp;
-    }
-    else
-    {
-        int track_cmp = Track() - other->Track();
-
-        if (track_cmp == 0)
-            return Title().toLower().localeAwareCompare(
-                other->Title().toLower());
-
-        return track_cmp;
-    }
-}
-
 // static
 MusicMetadata *MusicMetadata::createFromFilename(const QString &filename)
 {
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.h	2014-02-05 16:35:50.176097886 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.h	2014-02-05 16:36:05.366030829 +0100
@@ -252,7 +252,6 @@
     void persist(void);
     void UpdateModTime(void) const;
     bool hasChanged() const { return m_changed; }
-    int  compare(const MusicMetadata *other) const;
 
     // static functions
     static MusicMetadata *createFromFilename(const QString &filename);
