diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/storagegroup.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/storagegroup.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/storagegroup.cpp	2014-02-05 16:36:18.952635258 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/storagegroup.cpp	2014-02-05 16:36:33.179238004 +0100
@@ -30,6 +30,8 @@
     <<  QT_TRANSLATE_NOOP("(StorageGroups)", "Fanart")
     <<  QT_TRANSLATE_NOOP("(StorageGroups)", "Screenshots")
     <<  QT_TRANSLATE_NOOP("(StorageGroups)", "Banners")
+    <<  QT_TRANSLATE_NOOP("(StorageGroups)", "Music")
+    <<  QT_TRANSLATE_NOOP("(StorageGroups)", "MusicArt")
     ;
 
 /****************************************************************************/
