diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythbackend/mainserver.cpp	2014-02-05 16:36:31.445911817 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythbackend/mainserver.cpp	2014-02-05 16:36:39.269210049 +0100
@@ -793,6 +793,10 @@
     {
         HandleScanVideos(pbs);
     }
+    else if (command == "SCAN_MUSIC")
+    {
+        HandleScanMusic(pbs);
+    }
     else if (command == "ALLOW_SHUTDOWN")
     {
         if (tokens.size() != 1)
@@ -5111,6 +5115,22 @@
 
     if (pbssock)
         SendResponse(pbssock, retlist);
+}
+
+void MainServer::HandleScanMusic(PlaybackSock *pbs)
+{
+    MythSocket *pbssock = pbs->getSocket();
+
+    QStringList retlist;
+
+    QScopedPointer<MythSystem> cmd(MythSystem::Create("mythutil --scanmusic",
+                                                      kMSAutoCleanup | kMSRunBackground |
+                                                      kMSDontDisableDrawing | kMSProcessEvents |
+                                                      kMSDontBlockInputDevs));
+    retlist << "OK";
+
+    if (pbssock)
+        SendResponse(pbssock, retlist);
 }
 
 void MainServer::HandleFileTransferQuery(QStringList &slist,
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythbackend/mainserver.h mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythbackend/mainserver.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythbackend/mainserver.h	2014-02-05 16:36:31.445911817 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythbackend/mainserver.h	2014-02-05 16:36:40.219210830 +0100
@@ -216,6 +216,7 @@
     void HandleSettingQuery(QStringList &tokens, PlaybackSock *pbs);
     void HandleSetSetting(QStringList &tokens, PlaybackSock *pbs);
     void HandleScanVideos(PlaybackSock *pbs);
+    void HandleScanMusic(PlaybackSock *pbs);
     void HandleVersion(MythSocket *socket, const QStringList &slist);
     void HandleBackendRefresh(MythSocket *socket);
     void HandleQueryLoad(PlaybackSock *pbs);
