diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythui/mythimage.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythui/mythimage.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythui/mythimage.cpp	2014-02-05 16:36:55.599136791 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythui/mythimage.cpp	2014-02-05 16:37:01.289111647 +0100
@@ -258,19 +258,30 @@
 
 bool MythImage::Load(const QString &filename)
 {
+    if (filename.isEmpty())
+        return false;
+
     QImage *im = NULL;
     if (filename.startsWith("myth://"))
     {
-        im = new QImage();
-        RemoteFile *rf = new RemoteFile(filename, false, false, 0);
-
-        QByteArray data;
-        bool ret = rf->SaveAs(data);
-
-        delete rf;
-
-        if (ret)
-            im->loadFromData(data);
+        // Attempting a file transfer on a file that doesn't exist throws
+        // a lot of noisy warnings on the backend and frontend, so to avoid
+        // that first check it's there
+        if (RemoteFile::Exists(filename))
+        {
+            RemoteFile *rf = new RemoteFile(filename, false, false, 0);
+
+            QByteArray data;
+            bool ret = rf->SaveAs(data);
+
+            delete rf;
+
+            if (ret)
+            {
+                im = new QImage();
+                im->loadFromData(data);
+            }
+        }
 #if 0
         else
             LOG(VB_GENERAL, LOG_ERR,
@@ -283,10 +294,12 @@
                 (filename.startsWith("https://")) ||
                 (filename.startsWith("ftp://")))
     {
-        im = new QImage();
         QByteArray data;
         if (GetMythDownloadManager()->download(filename, &data))
+        {
+            im = new QImage();
             im->loadFromData(data);
+        }
     }
     else
     {
@@ -296,6 +309,12 @@
         im = new QImage(absoluteUrl);
     }
 
+    if (im && im->isNull())
+    {
+        delete im;
+        im = NULL;
+    }
+
     SetFileName(filename);
     if (im)
     {
@@ -303,6 +322,8 @@
         delete im;
         return true;
     }
+    else
+        LOG(VB_GUI, LOG_WARNING, QString("MythImage::Load(%1) failed").arg(filename));
 
     return false;
 }
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythui/mythuihelper.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythui/mythuihelper.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythui/mythuihelper.cpp	2014-02-05 16:36:58.255791984 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythui/mythuihelper.cpp	2014-02-05 16:37:01.289111647 +0100
@@ -1522,6 +1522,11 @@
         QString cachefilepath = GetThemeCacheDir() + '/' + label;
         QFileInfo cacheFileInfo(cachefilepath);
 
+        // If the file isn't in the disk cache, then we don't want to bother
+        // checking the last modified times of the original
+        if (!cacheFileInfo.exists())
+            return NULL;
+
         // Now compare the time on the source versus our cached copy
         QDateTime srcLastModified;
 
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythbackend/mainserver.cpp	2014-02-05 16:36:53.192481418 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythbackend/mainserver.cpp	2014-02-05 16:37:01.289111647 +0100
@@ -6216,7 +6216,7 @@
             else
             {
                 LOG(VB_GENERAL, LOG_ERR, QString("ERROR: LocalFilePath "
-                    "unable to find local path for '%1'.") .arg(opath));
+                    "unable to find local path for '%1'.") .arg(url.toString()));
                 lpath = "";
             }
 
