diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaio.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaio.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaio.h	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaio.h	2014-02-05 16:37:10.185738400 +0100
@@ -25,7 +25,7 @@
     * \param mdata A pointer to a MusicMetadata object
     * \returns Boolean to indicate success/failure.
     */
-    virtual bool write(const MusicMetadata* mdata) = 0;
+    virtual bool write(MusicMetadata* mdata) = 0;
 
     /*!
     * \brief Writes rating and playcount back to a file
@@ -33,7 +33,7 @@
     * \param mdata A pointer to a MusicMetadata object
     * \returns Boolean to indicate success/failure.
     */
-    virtual bool writeVolatileMetadata(const MusicMetadata* mdata)
+    virtual bool writeVolatileMetadata(MusicMetadata* mdata)
     {
         (void)mdata;
         return false;
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioavfcomment.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioavfcomment.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioavfcomment.cpp	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioavfcomment.cpp	2014-02-05 16:37:10.185738400 +0100
@@ -26,7 +26,7 @@
 /*!
  * \copydoc MetaIO::write()
  */
-bool MetaIOAVFComment::write(const MusicMetadata* /*mdata*/)
+bool MetaIOAVFComment::write(MusicMetadata* /*mdata*/)
 {
     // Wont implement....
     return false;
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioavfcomment.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioavfcomment.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioavfcomment.h	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioavfcomment.h	2014-02-05 16:37:10.185738400 +0100
@@ -22,7 +22,7 @@
     MetaIOAVFComment(void);
     virtual ~MetaIOAVFComment(void);
 
-    bool write(const MusicMetadata* mdata);
+    bool write(MusicMetadata* mdata);
     MusicMetadata* read(const QString &filename);
 
 private:
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp	2014-02-05 16:37:10.185738400 +0100
@@ -47,7 +47,7 @@
 /*!
  * \copydoc MetaIO::write()
  */
-bool MetaIOFLACVorbis::write(const MusicMetadata* mdata)
+bool MetaIOFLACVorbis::write(MusicMetadata* mdata)
 {
     if (!mdata)
         return false;
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioflacvorbis.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioflacvorbis.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioflacvorbis.h	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioflacvorbis.h	2014-02-05 16:37:10.185738400 +0100
@@ -24,7 +24,7 @@
     MetaIOFLACVorbis(void);
     virtual ~MetaIOFLACVorbis(void);
 
-    bool write(const MusicMetadata* mdata);
+    bool write(MusicMetadata* mdata);
     bool writeAlbumArt(const QString &filename, const AlbumArtImage *albumart);
     bool removeAlbumArt(const QString &filename, const AlbumArtImage *albumart);
 
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioid3.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioid3.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioid3.cpp	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioid3.cpp	2014-02-05 16:37:10.185738400 +0100
@@ -156,7 +156,7 @@
 /*!
  * \copydoc MetaIO::write()
  */
-bool MetaIOID3::write(const MusicMetadata* mdata)
+bool MetaIOID3::write(MusicMetadata* mdata)
 {
     if (!OpenFile(mdata->Filename(), true))
         return false;
@@ -842,7 +842,7 @@
     return true;
 }
 
-bool MetaIOID3::writeVolatileMetadata(const MusicMetadata* mdata)
+bool MetaIOID3::writeVolatileMetadata(MusicMetadata* mdata)
 {
     QString filename = mdata->Filename();
     int rating = mdata->Rating();
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioid3.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioid3.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioid3.h	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioid3.h	2014-02-05 16:37:10.185738400 +0100
@@ -36,8 +36,8 @@
     MetaIOID3(void);
     virtual ~MetaIOID3(void);
 
-    virtual bool write(const MusicMetadata* mdata);
-    bool writeVolatileMetadata(const MusicMetadata* mdata);
+    virtual bool write(MusicMetadata* mdata);
+    bool writeVolatileMetadata(MusicMetadata* mdata);
 
     bool writeAlbumArt(const QString &filename, const AlbumArtImage *albumart);
     bool removeAlbumArt(const QString &filename, const AlbumArtImage *albumart);
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiomp4.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiomp4.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiomp4.cpp	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiomp4.cpp	2014-02-05 16:37:10.185738400 +0100
@@ -27,7 +27,7 @@
 /*!
  * \copydoc MetaIO::write()
  */
-bool MetaIOMP4::write(const MusicMetadata* mdata)
+bool MetaIOMP4::write(MusicMetadata* mdata)
 {
     if (!mdata)
         return false;
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiomp4.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiomp4.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiomp4.h	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiomp4.h	2014-02-05 16:37:10.185738400 +0100
@@ -19,7 +19,7 @@
     MetaIOMP4(void);
     virtual ~MetaIOMP4(void);
 
-    bool write(const MusicMetadata* mdata);
+    bool write(MusicMetadata* mdata);
     MusicMetadata* read(const QString &filename);
 
   private:
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiooggvorbis.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiooggvorbis.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiooggvorbis.cpp	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiooggvorbis.cpp	2014-02-05 16:37:10.185738400 +0100
@@ -41,7 +41,7 @@
 /*!
  * \copydoc MetaIO::write()
  */
-bool MetaIOOggVorbis::write(const MusicMetadata* mdata)
+bool MetaIOOggVorbis::write(MusicMetadata* mdata)
 {
     if (!mdata)
         return false;
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiooggvorbis.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiooggvorbis.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiooggvorbis.h	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiooggvorbis.h	2014-02-05 16:37:10.189071719 +0100
@@ -24,7 +24,7 @@
     MetaIOOggVorbis(void);
     ~MetaIOOggVorbis(void);
 
-    bool write(const MusicMetadata* mdata);
+    bool write(MusicMetadata* mdata);
     MusicMetadata* read(const QString &filename);
 
   private:
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiotaglib.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiotaglib.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiotaglib.h	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiotaglib.h	2014-02-05 16:37:10.189071719 +0100
@@ -23,7 +23,7 @@
     MetaIOTagLib(void);
     virtual ~MetaIOTagLib(void);
 
-    virtual bool write(const MusicMetadata* mdata) = 0;
+    virtual bool write(MusicMetadata* mdata) = 0;
     virtual MusicMetadata* read(const QString &filename) = 0;
 
   protected:
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiowavpack.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiowavpack.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiowavpack.cpp	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiowavpack.cpp	2014-02-05 16:37:10.189071719 +0100
@@ -43,7 +43,7 @@
 /*!
 * \copydoc MetaIO::write()
 */
-bool MetaIOWavPack::write(const MusicMetadata* mdata)
+bool MetaIOWavPack::write(MusicMetadata* mdata)
 {
     if (!mdata)
         return false;
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiowavpack.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiowavpack.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaiowavpack.h	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaiowavpack.h	2014-02-05 16:37:10.189071719 +0100
@@ -26,7 +26,7 @@
     MetaIOWavPack(void);
     virtual ~MetaIOWavPack(void);
 
-    bool write(const MusicMetadata* mdata);
+    bool write(MusicMetadata* mdata);
     MusicMetadata* read(const QString &filename);
 
 private:
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:37:10.189071719 +0100
@@ -29,14 +29,14 @@
 
 static QString thePrefix = "the ";
 
-bool operator==(const MusicMetadata& a, const MusicMetadata& b)
+bool operator==(MusicMetadata& a, MusicMetadata& b)
 {
     if (a.Filename() == b.Filename())
         return true;
     return false;
 }
 
-bool operator!=(const MusicMetadata& a, const MusicMetadata& b)
+bool operator!=(MusicMetadata& a, MusicMetadata& b)
 {
     if (a.Filename() != b.Filename())
         return true;
@@ -695,35 +695,50 @@
     return m_formattedtitle;
 }
 
-QString MusicMetadata::Filename(bool find) const
+QString MusicMetadata::Filename(bool find)
 {
     // if not asked to find the file just return the raw filename from the DB
     if (find == false)
         return m_filename;
 
+    if (!m_actualFilename.isEmpty())
+        return m_actualFilename;
+
     // check for a cd track
     if (m_filename.endsWith(".cda"))
+    {
+        m_actualFilename = m_filename;
         return m_filename;
+    }
 
     // check for http urls etc
     if (m_filename.contains("://"))
+    {
+        m_actualFilename = m_filename;
         return m_filename;
+    }
 
     // first check to see if the filename is complete
     if (QFile::exists(m_filename))
+    {
+        m_actualFilename = m_filename;
         return m_filename;
+    }
 
     // next try appending the start directory
     if (QFile::exists(getMusicDirectory() + m_filename))
-        return getMusicDirectory() + m_filename;
+    {
+        m_actualFilename = getMusicDirectory() + m_filename;
+        return m_actualFilename;
+    }
 
     // maybe it's in our 'Music' storage group
-    //FIXME: this is just looking on the master BE
-    QString filename = gCoreContext->GenMythURL(gCoreContext->GetSetting("MasterServerIP"),
-                                                gCoreContext->GetNumSetting("MasterServerPort"),
-                                                m_filename, "Music");
-    if (RemoteFile::Exists(filename))
+    QString filename = RemoteFile::FindFile(m_filename, "", "Music");
+    if (!filename.isEmpty())
+    {
+        m_actualFilename = filename;
         return filename;
+    }
 
     // not found
     LOG(VB_GENERAL, LOG_ERR, QString("MusicMetadata: Asked to get the filename for a track but no file found: %1")
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.h	2014-02-05 16:36:58.972455448 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.h	2014-02-05 16:37:10.389071041 +0100
@@ -195,7 +195,7 @@
     bool isDBTrack(void) const { return ID_TO_REPO(m_id) == RT_Database; }
     bool isRadio(void) const { return ID_TO_REPO(m_id) == RT_Radio; }
 
-    QString Filename(bool find = true) const;
+    QString Filename(bool find = true);
     void setFilename(const QString &lfilename) { m_filename = lfilename; }
 
     uint64_t FileSize() const;
@@ -304,7 +304,8 @@
     AlbumArtImages *m_albumArt;
 
     IdType   m_id;
-    QString  m_filename;
+    QString  m_filename;       // file name as stored in the DB
+    QString  m_actualFilename; // actual URL of the file if found
     uint64_t m_fileSize;
     bool     m_changed;
 
@@ -326,8 +327,8 @@
     static QString m_formatcompilationcdtrack;
 };
 
-bool operator==(const MusicMetadata& a, const MusicMetadata& b);
-bool operator!=(const MusicMetadata& a, const MusicMetadata& b);
+bool operator==(MusicMetadata& a, MusicMetadata& b);
+bool operator!=(MusicMetadata& a, MusicMetadata& b);
 
 Q_DECLARE_METATYPE(MusicMetadata *)
 
