diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:37:19.902361421 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:37:25.712335594 +0100
@@ -761,7 +761,10 @@
 
         QUrl url(mythUrl);
         if (url.host() != m_hostname)
+        {
+            m_hostname = url.host();
             saveHostname();
+        }
 
         return mythUrl;
     }
