diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/remotefile.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/remotefile.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/remotefile.cpp	2014-02-05 16:37:26.842330220 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/remotefile.cpp	2014-02-05 16:37:38.772276627 +0100
@@ -933,7 +933,7 @@
     strList << "QUERY_SG_FILEQUERY" << hostName << storageGroup << filename;
     if (gCoreContext->SendReceiveStringList(strList))
     {
-        if (strList.size() > 0 && strList[0] != "EMPTY LIST")
+        if (strList.size() > 0 && strList[0] != "EMPTY LIST" && !strList[0].startsWith("SLAVE UNREACHABLE"))
             return gCoreContext->GenMythURL(hostName, 0, filename, storageGroup);
     }
 
@@ -965,7 +965,7 @@
 
         if (gCoreContext->SendReceiveStringList(strList))
         {
-            if (strList.size() > 0 && strList[0] != "EMPTY LIST")
+            if (strList.size() > 0 && strList[0] != "EMPTY LIST" && !strList[0].startsWith("SLAVE UNREACHABLE"))
                 return gCoreContext->GenMythURL(hostName, 0, filename, storageGroup);
         }
     }
