diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/dbcheck.cpp mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/dbcheck.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/dbcheck.cpp	2014-02-05 16:37:38.002280089 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/dbcheck.cpp	2014-02-05 16:37:45.775578454 +0100
@@ -12,7 +12,7 @@
 
 #include "dbcheck.h"
 
-const QString currentDatabaseVersion = "1021";
+const QString currentDatabaseVersion = "1022";
 
 static bool doUpgradeMusicDatabaseSchema(QString &dbver);
 
@@ -890,5 +890,17 @@
             return false;
     }
 
+    if (dbver == "1021")
+    {
+        const QString updates[] = {
+"ALTER TABLE music_albumart ADD COLUMN hostname VARCHAR(255) NOT NULL default '';",
+QString("UPDATE music_albumart SET hostname = '%1';").arg(gCoreContext->GetMasterHostName()),
+""
+};
+
+        if (!performActualUpdate(updates, "1022", dbver))
+            return false;
+    }
+
     return true;
 }
