diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicfilescanner.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicfilescanner.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicfilescanner.cpp	2014-02-05 16:38:05.725488858 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicfilescanner.cpp	2014-02-05 16:38:17.135437600 +0100
@@ -498,11 +498,14 @@
  */
 void MusicFileScanner::UpdateFileInDB(const QString &filename, const QString &startDir)
 {
+    QString dbFilename = filename;
+    dbFilename.remove(0, startDir.length());
+
     QString directory = filename;
     directory.remove(0, startDir.length());
     directory = directory.section( '/', 0, -2);
 
-    MusicMetadata *db_meta   = MetaIO::getMetadata(filename);
+    MusicMetadata *db_meta   = MetaIO::getMetadata(dbFilename);
     MusicMetadata *disk_meta = MetaIO::readMetadata(filename);
 
     if (db_meta && disk_meta)
