diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythui/mythimage.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythui/mythimage.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythui/mythimage.cpp	2014-02-05 16:38:24.028739957 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythui/mythimage.cpp	2014-02-05 16:38:40.212000583 +0100
@@ -267,9 +267,11 @@
         // Attempting a file transfer on a file that doesn't exist throws
         // a lot of noisy warnings on the backend and frontend, so to avoid
         // that first check it's there
-        if (RemoteFile::Exists(filename))
+        QUrl url(filename);
+        QString mythUrl = RemoteFile::FindFile(url.path(), url.host(), url.userName());
+        if (!mythUrl.isEmpty())
         {
-            RemoteFile *rf = new RemoteFile(filename, false, false, 0);
+            RemoteFile *rf = new RemoteFile(mythUrl, false, false, 0);
 
             QByteArray data;
             bool ret = rf->SaveAs(data);
