diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:38:32.452035443 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:38:45.815309549 +0100
@@ -1660,7 +1660,7 @@
                 image->filename = logoUrl;
                 image->imageType = IT_FRONTCOVER;
                 image->embedded = false;
-
+                image->hostname = "";
                 m_imageList.push_back(image);
             }
         }
@@ -1676,7 +1676,7 @@
         MSqlQuery query(MSqlQuery::InitCon());
         query.prepare("SELECT albumart_id, CONCAT_WS('/', music_directories.path, "
                 "music_albumart.filename), music_albumart.filename, music_albumart.imagetype, "
-                "music_albumart.embedded "
+                "music_albumart.embedded, music_albumart.hostname "
                 "FROM music_albumart "
                 "LEFT JOIN music_directories ON "
                 "music_directories.directory_id = music_albumart.directory_id "
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.h	2014-02-05 16:38:32.452035443 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.h	2014-02-05 16:38:46.761971326 +0100
@@ -39,13 +39,14 @@
 {
   public:
     AlbumArtImage(void) :
-            id(0), filename(""), imageType(IT_UNKNOWN),
+            id(0), filename(""), hostname(""), imageType(IT_UNKNOWN),
             description(""), embedded(false) {}
     AlbumArtImage(AlbumArtImage *image) :
-            id(image->id), filename(image->filename), imageType(image->imageType),
+            id(image->id), filename(image->filename), hostname(image->hostname), imageType(image->imageType),
             description(image->description), embedded(image->embedded) {}
      int       id;
      QString   filename;
+     QString   hostname;
      ImageType imageType;
      QString   description;
      bool      embedded;
