diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/musicmetautils.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/musicmetautils.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/programs/mythutil/musicmetautils.cpp	2014-02-05 16:38:43.731984765 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/programs/mythutil/musicmetautils.cpp	2014-02-05 16:38:53.398607993 +0100
@@ -174,15 +174,11 @@
     MusicFileScanner *fscan = new MusicFileScanner();
     QStringList dirList;
 
-#if 0
     if (!StorageGroup::FindDirs("Music", gCoreContext->GetHostName(), &dirList))
     {
         LOG(VB_GENERAL, LOG_ERR, "Failed to find any directories in the 'Music' storage group");
         return GENERIC_EXIT_NOT_OK;
     }
-#else
-    dirList << getMusicDirectory();
-#endif
 
     fscan->SearchDirs(dirList);
     delete fscan;
