diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/musicdata.cpp mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/musicdata.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/musicdata.cpp	2014-02-05 16:38:51.808615139 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/musicdata.cpp	2014-02-05 16:39:00.101911215 +0100
@@ -8,7 +8,6 @@
 #include <musicmetadata.h>
 #include <musicfilescanner.h>
 #include <musicutils.h>
-#include "mythsystemlegacy.h"
 
 // mythmusic
 #include "musicdata.h"
@@ -52,16 +51,8 @@
 
 void MusicData::scanMusic (void)
 {
-    // TODO once we switch to storage groups we should use the myth protocol
-    // to ask the master BE to run the scan on all hosts with a valid directory
-    // set for the 'Music' storage group
-    //
-    // for the moment we just run mythutil --scanmusic on this host
-
-    QScopedPointer<MythSystem> cmd(MythSystem::Create("mythutil --scanmusic",
-                                                      kMSAutoCleanup | kMSRunBackground |
-                                                      kMSDontDisableDrawing | kMSProcessEvents |
-                                                      kMSDontBlockInputDevs));
+    gCoreContext->SendReceiveStringList(QStringList() << "SCAN_MUSIC");
+    LOG(VB_GENERAL, LOG_INFO, "Requested a music file scan");
 }
 
 /// reload music after a scan, rip or import
