diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/remotefile.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/remotefile.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/remotefile.cpp	2014-02-05 16:38:54.755268566 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/remotefile.cpp	2014-02-05 16:39:06.401882976 +0100
@@ -512,6 +512,59 @@
     return result;
 }
 
+bool RemoteFile::CopyFile (const QString& src, const QString& dst)
+{
+    const int readSize = 16 * 1024;
+    char *buf = new char[readSize];
+    if (!buf)
+    {
+        LOG(VB_GENERAL, LOG_ERR, "RemoteFile::CopyFile: ERROR, unable to allocate copy buffer");
+        return false;
+    }
+
+    RemoteFile srcFile(src, false);
+    if (!srcFile.isOpen())
+    {
+         LOG(VB_GENERAL, LOG_ERR,
+             QString("RemoteFile::CopyFile: Failed to open file (%1) for reading.").arg(src));
+         delete[] buf;
+         return false;
+    }
+
+    RemoteFile dstFile(dst, true);
+    if (!dstFile.isOpen())
+    {
+         LOG(VB_GENERAL, LOG_ERR,
+             QString("RemoteFile::CopyFile: Failed to open file (%1) for writing.").arg(dst));
+         srcFile.Close();
+         delete[] buf;
+         return false;
+    }
+
+    int srcLen, dstLen;
+
+    while ((srcLen = srcFile.Read(buf, sizeof(buf))) > 0)
+    {
+        dstLen = dstFile.Write(buf, srcLen);
+
+        if (dstLen == -1 || srcLen != dstLen)
+        {
+            LOG(VB_GENERAL, LOG_ERR,
+                "RemoteFile::CopyFile:: Error while trying to write to destination file.");
+            srcFile.Close();
+            dstFile.Close();
+            delete[] buf;
+            return false;
+        }
+    }
+
+    srcFile.Close();
+    dstFile.Close();
+    delete[] buf;
+
+    return true;
+}
+
 void RemoteFile::Reset(void)
 {
     if (isLocal())
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/remotefile.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/remotefile.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/remotefile.h	2014-02-05 16:38:54.755268566 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/remotefile.h	2014-02-05 16:39:06.821881019 +0100
@@ -33,6 +33,7 @@
     static QString GetFileHash(const QString &url);
     static QDateTime LastModified(const QString &url);
     static QString FindFile(const QString &filename, const QString &host, const QString &storageGroup);
+    static bool CopyFile(const QString &src, const QString &dest);
 
     int Write(const void *data, int size);
     int Read(void *data, int size);
