diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/remotefile.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/remotefile.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/remotefile.cpp	2014-02-05 16:39:06.401882976 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/remotefile.cpp	2014-02-05 16:39:14.531846375 +0100
@@ -514,6 +514,13 @@
 
 bool RemoteFile::CopyFile (const QString& src, const QString& dst)
 {
+    // sanity check
+    if (src == dst)
+    {
+        LOG(VB_GENERAL, LOG_ERR, "RemoteFile::CopyFile: Cannot copy a file to itself");
+        return false;
+    }
+
     const int readSize = 16 * 1024;
     char *buf = new char[readSize];
     if (!buf)
