diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/generalsettings.cpp mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/generalsettings.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/generalsettings.cpp	2014-02-05 16:39:20.681818768 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/generalsettings.cpp	2014-02-05 16:39:29.501779158 +0100
@@ -9,7 +9,7 @@
 
 GeneralSettings::GeneralSettings(MythScreenStack *parent, const char *name)
         : MythScreenType(parent, name),
-        m_musicLocation(NULL), m_musicAudioDevice(NULL),
+        m_musicAudioDevice(NULL),
         m_musicDefaultUpmix(NULL), m_musicCDDevice(NULL),
         m_nonID3FileNameFormat(NULL), m_ignoreID3Tags(NULL),
         m_allowTagWriting(NULL), m_saveButton(NULL),
@@ -30,7 +30,6 @@
     if (!LoadWindowFromXML("musicsettings-ui.xml", "generalsettings", this))
         return false;
 
-    UIUtilE::Assign(this, m_musicLocation, "musiclocation", &err);
     UIUtilE::Assign(this, m_musicAudioDevice, "musicaudiodevice", &err);
     UIUtilE::Assign(this, m_musicDefaultUpmix, "musicdefaultupmix", &err);
     UIUtilE::Assign(this, m_musicCDDevice, "musiccddevice", &err);
@@ -46,7 +45,6 @@
         return false;
     }
 
-    m_musicLocation->SetText(gCoreContext->GetSetting("MusicLocation"));
     m_musicAudioDevice->SetText(gCoreContext->GetSetting("MusicAudioDevice"));
 
     int loadMusicDefaultUpmix = gCoreContext->GetNumSetting("MusicDefaultUpmix", 0);
@@ -68,9 +66,6 @@
     connect(m_saveButton, SIGNAL(Clicked()), this, SLOT(slotSave()));
     connect(m_cancelButton, SIGNAL(Clicked()), this, SLOT(Close()));
 
-    m_musicLocation->SetHelpText(tr("This directory must exist, and the user "
-                 "running MythMusic needs to have write permission "
-                 "to the directory."));
     m_musicAudioDevice->SetHelpText(tr("Audio Device used for playback. 'default' "
                  "will use the device specified in MythTV"));
     m_musicDefaultUpmix->SetHelpText(tr("MythTV can upconvert stereo tracks to 5.1 audio. "
@@ -97,25 +92,13 @@
 
     BuildFocusList();
 
-    SetFocusWidget(m_musicLocation);
+    SetFocusWidget(m_musicCDDevice);
 
     return true;
 }
 
 void GeneralSettings::slotSave(void)
 {
-    // get the starting directory from the settings and remove all multiple
-    // directory separators "/" and resolves any "." or ".." in the path.
-    QString dir = m_musicLocation->GetText();
-
-    if (!dir.isEmpty())
-    {
-        dir = QDir::cleanPath(dir);
-        if (!dir.endsWith("/"))
-            dir += "/";
-    }
-
-    gCoreContext->SaveSetting("MusicLocation", dir);
     gCoreContext->SaveSetting("CDDevice", m_musicCDDevice->GetText());
     gCoreContext->SaveSetting("MusicAudioDevice", m_musicAudioDevice->GetText());
     gCoreContext->SaveSetting("NonID3FileNameFormat", m_nonID3FileNameFormat->GetText());
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/generalsettings.h mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/generalsettings.h
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/generalsettings.h	2014-02-05 16:39:20.685152078 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/generalsettings.h	2014-02-05 16:39:29.501779158 +0100
@@ -20,7 +20,6 @@
     bool Create(void);
 
   private:
-    MythUITextEdit     *m_musicLocation;
     MythUITextEdit     *m_musicAudioDevice;
     MythUICheckBox     *m_musicDefaultUpmix;
     MythUITextEdit     *m_musicCDDevice;
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/theme/default/musicsettings-ui.xml mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/theme/default/musicsettings-ui.xml
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/theme/default/musicsettings-ui.xml	2014-02-05 16:39:20.688485389 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/theme/default/musicsettings-ui.xml	2014-02-05 16:39:29.511779118 +0100
@@ -11,15 +11,6 @@
             <value>General Settings</value>
         </textarea>
 
-        <textarea name="musiclocation_label" from="basetextarea">
-            <area>20,40,380,50</area>
-            <align>right,vcenter</align>
-            <value>Directory to hold music:</value>
-        </textarea> 
-        <textedit name="musiclocation" from="basetextedit">
-            <position>410,40</position>
-        </textedit>
-
         <textarea name="musiccddevice_label" from="basetextarea">
             <area>20,100,380,50</area>
             <align>right,vcenter</align>
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/theme/default-wide/musicsettings-ui.xml mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/theme/default-wide/musicsettings-ui.xml
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/theme/default-wide/musicsettings-ui.xml	2014-02-05 16:39:20.688485389 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/theme/default-wide/musicsettings-ui.xml	2014-02-05 16:39:29.538445653 +0100
@@ -11,15 +11,6 @@
             <value>General Settings</value>
         </textarea>
 
-        <textarea name="musiclocation_label" from="basetextarea">
-            <area>140,100,500,50</area>
-            <align>right,vcenter</align>
-            <value>Directory to hold music:</value>
-        </textarea> 
-        <textedit name="musiclocation" from="basetextedit">
-            <area>650,100,550,50</area>
-        </textedit>
-
         <textarea name="musiccddevice_label" from="basetextarea">
             <area>140,160,500,50</area>
             <align>right,vcenter</align>
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/themes/Terra/musicsettings-ui.xml mythtv-0.27-20140204-g1b7404d-new/mythtv/themes/Terra/musicsettings-ui.xml
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/themes/Terra/musicsettings-ui.xml	2014-02-05 16:39:18.161830068 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/themes/Terra/musicsettings-ui.xml	2014-02-05 16:39:29.551778886 +0100
@@ -10,15 +10,6 @@
             <value>General Settings</value>
         </textarea>
 
-        <textarea name="musiclocation_label" from="basetextarea">
-            <area>140,100,500,50</area>
-            <align>right,vcenter</align>
-            <value>Directory to hold music:</value>
-        </textarea> 
-        <textedit name="musiclocation" from="basetextedit">
-            <area>650,100,550,50</area>
-        </textedit>
-
         <textarea name="musiccddevice_label" from="basetextarea">
             <area>140,160,500,50</area>
             <align>right,vcenter</align>
