diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaio.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaio.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaio.cpp	2014-02-05 16:39:33.915092631 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaio.cpp	2014-02-05 16:39:44.485045117 +0100
@@ -102,7 +102,8 @@
 // static
 MusicMetadata* MetaIO::getMetadata(const QString &filename)
 {
-
+    //FIXME: we need a relative path for createFromFilename()
+    //       but readMetadata() needs an absolute path
     MusicMetadata *mdata = MusicMetadata::createFromFilename(filename);
     if (mdata)
         return mdata;
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:39:33.915092631 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:39:44.485045117 +0100
@@ -210,10 +210,6 @@
     // find the trackid for this filename
     QString sqldir = filename.section('/', 0, -2);
 
-    // Filename is the absolute path, we want the relative path
-    if (sqldir.startsWith(getMusicDirectory()))
-        sqldir.remove(0, getMusicDirectory().length());
-
     QString sqlfilename = filename.section('/', -1);
 
     MSqlQuery query(MSqlQuery::InitCon());
@@ -237,7 +233,6 @@
             QString("MusicMetadata::createFromFilename: Could not find '%1'")
                 .arg(filename));
         return NULL;
-
     }
 
     int songID = query.value(0).toInt();
@@ -746,13 +741,6 @@
         return m_filename;
     }
 
-    // next try appending the start directory
-    if (QFile::exists(getMusicDirectory() + m_filename))
-    {
-        m_actualFilename = getMusicDirectory() + m_filename;
-        return m_actualFilename;
-    }
-
     // maybe it's in our 'Music' storage group
     QString mythUrl = RemoteFile::FindFile(m_filename, m_hostname, "Music");
     if (!mythUrl.isEmpty())
@@ -1701,8 +1689,6 @@
                         image->filename = gCoreContext->GenMythURL(url.host(), url.port(),
                                                                    QString("AlbumArt/") + query.value(1).toString(),
                                                                    "MusicArt");
-                    else
-                        image->filename = GetConfDir() + "/MythMusic/AlbumArt/" + query.value(1).toString();
                 }
                 else
                 {
@@ -1710,8 +1696,6 @@
                         image->filename =  gCoreContext->GenMythURL(url.host(), url.port(),
                                                                     query.value(1).toString(),
                                                                     "Music");
-                    else
-                        image->filename = getMusicDirectory() + query.value(1).toString();
                 }
 
                 image->imageType = (ImageType) query.value(3).toInt();
