diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/musicplayer.cpp mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/musicplayer.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/musicplayer.cpp	2014-02-05 16:39:44.238379578 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/musicplayer.cpp	2014-02-05 16:39:51.001682513 +0100
@@ -694,13 +694,6 @@
         }
         else if (me->Message().startsWith("MUSIC_SETTINGS_CHANGED"))
         {
-            QString startdir = gCoreContext->GetSetting("MusicLocation");
-            startdir = QDir::cleanPath(startdir);
-            if (!startdir.isEmpty() && !startdir.endsWith("/"))
-                startdir += "/";
-
-            setMusicDirectory(startdir);
-
             loadSettings();
         }
         else if (me->Message().startsWith("MUSIC_METADATA_CHANGED"))
