diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2014-02-05 16:39:50.601683967 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2014-02-05 16:39:59.008313214 +0100
@@ -303,20 +303,20 @@
                     if (encodertype == "mp3")
                     {
                         outfile += ".mp3";
-                        encoder.reset(new LameEncoder(getMusicDirectory() + outfile, m_quality,
+                        encoder.reset(new LameEncoder(/*getMusicDirectory() + */outfile, m_quality,
                                                       titleTrack, mp3usevbr));
                     }
                     else // ogg
                     {
                         outfile += ".ogg";
-                        encoder.reset(new VorbisEncoder(getMusicDirectory() + outfile, m_quality,
+                        encoder.reset(new VorbisEncoder(/*getMusicDirectory() + */outfile, m_quality,
                                                         titleTrack));
                     }
                 }
                 else
                 {
                     outfile += ".flac";
-                    encoder.reset(new FlacEncoder(getMusicDirectory() + outfile, m_quality,
+                    encoder.reset(new FlacEncoder(/*getMusicDirectory() + */outfile, m_quality,
                                                   titleTrack));
                 }
 
@@ -823,7 +823,7 @@
     if (query.next())
     {
         int trackID = query.value(0).toInt();
-        QString filename = getMusicDirectory() + query.value(1).toString();
+        QString filename = /*getMusicDirectory() +*/ query.value(1).toString();
 
         // delete file
         if (!QFile::remove(filename))
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/importmusic.cpp mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/importmusic.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythplugins/mythmusic/mythmusic/importmusic.cpp	2014-02-05 16:39:50.601683967 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythplugins/mythmusic/mythmusic/importmusic.cpp	2014-02-05 16:39:59.008313214 +0100
@@ -404,7 +404,7 @@
 
         // get the save filename
         QString origFilename = meta->Filename();
-        QString saveFilename = filenameFromMetadata(meta, false);
+        QString saveFilename = filenameFromMetadata(meta);
         QString fullFilename;
 
         QUrl url(m_musicStorageDir);
@@ -1164,7 +1164,7 @@
         m_coverartImage->SetFilename(m_filelist[m_currentFile]);
         m_coverartImage->Load();
 
-        QString saveFilename = /*getMusicDirectory() + */filenameFromMetadata(m_metadata, false);
+        QString saveFilename = /*getMusicDirectory() + */filenameFromMetadata(m_metadata);
         QFileInfo fi(saveFilename);
         QString saveDir = fi.absolutePath();
 
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicutils.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicutils.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicutils.cpp	2014-02-05 16:39:45.555040271 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicutils.cpp	2014-02-05 16:39:59.008313214 +0100
@@ -21,26 +21,6 @@
 #include "musicmetadata.h"
 #include "musicutils.h"
 
-static QString musicDirectory;
-
-QString getMusicDirectory(void)
-{
-    if (musicDirectory.isEmpty())
-    {
-        musicDirectory = gCoreContext->GetSetting("MusicLocation");
-        musicDirectory = QDir::cleanPath(musicDirectory);
-        if (!musicDirectory.isEmpty() && !musicDirectory.endsWith("/"))
-            musicDirectory += "/";
-    }
-
-    return musicDirectory;
-}
-
-void setMusicDirectory(const QString &musicDir)
-{
-    musicDirectory = musicDir;
-}
-
 static QRegExp badChars = QRegExp("(/|\\\\|:|\'|\"|\\?|\\|)");
 
 QString fixFilename(const QString &filename)
@@ -182,9 +162,8 @@
     return token;
 }
 
-QString filenameFromMetadata(MusicMetadata *track, bool createDir)
+QString filenameFromMetadata(MusicMetadata *track)
 {
-    QDir directoryQD(getMusicDirectory());
     QString filename;
     QString fntempl = gCoreContext->GetSetting("FilenameTemplate");
     bool no_ws = gCoreContext->GetNumSetting("NoWhitespace", 0);
@@ -242,14 +221,6 @@
         LOG(VB_GENERAL, LOG_ERR, "Invalid file storage definition.");
     }
 
-    if (createDir)
-    {
-        QFileInfo fi(filename);
-        if (!directoryQD.mkpath(getMusicDirectory() + fi.path()))
-            LOG(VB_GENERAL, LOG_ERR,
-                QString("filenameFromMetadata: Failed to create directory path: '%1'").arg(getMusicDirectory() + filename));
-    }
-
     return filename;
 }
 
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicutils.h mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicutils.h
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicutils.h	2014-02-05 16:39:45.555040271 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicutils.h	2014-02-05 16:39:59.008313214 +0100
@@ -12,12 +12,6 @@
 #undef TStringToQString
 #define TStringToQString(s) QString::fromUtf8(s.toCString(true))
 
-/// get music directory for this host
-META_PUBLIC QString getMusicDirectory(void);
-
-// get music directory for this host
-META_PUBLIC void setMusicDirectory(const QString &musicDir);
-
 /// remove any bad filename characters
 META_PUBLIC QString fixFilename(const QString &filename);
 
@@ -28,7 +22,7 @@
 META_PUBLIC uint calcTrackLength(const QString &musicFile);
 
 /// create a filename using the template in the settings and a MusicMetadata object
-META_PUBLIC QString filenameFromMetadata(MusicMetadata *track, bool createDir = true);
+META_PUBLIC QString filenameFromMetadata(MusicMetadata *track);
 
 /// try to find a track in the db using the given artist, album and title
 META_PUBLIC bool isNewTune(const QString &artist, const QString &album, const QString &title);
