diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp	2014-02-05 16:40:30.848273201 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioflacvorbis.cpp	2014-02-05 16:40:41.608332863 +0100
@@ -282,6 +282,7 @@
                 art->description = TStringToQString(pic->description());
 
             art->embedded = true;
+            art->hostname = gCoreContext->GetHostName();
 
             QString ext = getExtFromMimeType(
                                 TStringToQString(pic->mimeType()).toLower());
diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioid3.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioid3.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/metaioid3.cpp	2014-02-05 16:40:30.848273201 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/metaioid3.cpp	2014-02-05 16:40:41.608332863 +0100
@@ -5,6 +5,7 @@
 
 // Libmythbase
 #include <mythlogging.h>
+#include <mythcorecontext.h>
 
 // Taglib
 #include <flacfile.h>
@@ -485,6 +486,7 @@
                 art->description = TStringToQString(frame->description());
 
             art->embedded = true;
+            art->hostname = gCoreContext->GetHostName();
 
             QString ext = getExtFromMimeType(
                                 TStringToQString(frame->mimeType()).toLower());
