diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:40:48.221705288 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-02-05 16:40:55.988416175 +0100
@@ -1028,7 +1028,6 @@
                             .arg(Hostname())
                             .arg(ID())
                             .arg(AlbumArtImages::getTypeFilename(albumart_image->imageType));
-                    LOG(VB_GENERAL, LOG_INFO, QString("MusicMetadata::getAlbumArtFile: Sending '%1'").arg(slist.join(" ")));
                     gCoreContext->SendReceiveStringList(slist);
                 }
             }
