diff -Naur mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/remotefile.cpp mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/remotefile.cpp
--- mythtv-0.27-20140204-g1b7404d-old/mythtv/libs/libmythbase/remotefile.cpp	2014-02-05 16:40:51.245055850 +0100
+++ mythtv-0.27-20140204-g1b7404d-new/mythtv/libs/libmythbase/remotefile.cpp	2014-02-05 16:41:03.265124203 +0100
@@ -521,7 +521,7 @@
         return false;
     }
 
-    const int readSize = 16 * 1024;
+    const int readSize = 2 * 1024 * 1024;
     char *buf = new char[readSize];
     if (!buf)
     {
@@ -550,7 +550,7 @@
 
     int srcLen, dstLen;
 
-    while ((srcLen = srcFile.Read(buf, sizeof(buf))) > 0)
+    while ((srcLen = srcFile.Read(buf, readSize)) > 0)
     {
         dstLen = dstFile.Write(buf, srcLen);
 
