From 101b07e3cc3c412004a37883f22fe40aaa4bb351 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sat, 8 Feb 2014 19:59:44 +0000
Subject: [PATCH 5/6] Missed a file in 1238ba43bcf

---
 mythtv/programs/mythutil/musicmetautils.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/programs/mythutil/musicmetautils.cpp b/mythtv/programs/mythutil/musicmetautils.cpp
index bf92a10..292b1fa 100644
--- a/mythtv/programs/mythutil/musicmetautils.cpp
+++ b/mythtv/programs/mythutil/musicmetautils.cpp
@@ -66,7 +66,7 @@ static int UpdateMeta(const MythUtilCommandLineParser &cmdline)
     MetaIO *tagger = mdata->getTagger();
     if (tagger)
     {
-        ok = tagger->write(mdata);
+        ok = tagger->write(mdata->getLocalFilename(), mdata);
 
         if (!ok)
             LOG(VB_GENERAL, LOG_ERR, QString("Failed to write to tag for trackid: %1").arg(songID));
-- 
1.7.10

