From ad113b1f1cf02c21b2e4baeb12ca7917c7568230 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 2 Mar 2014 16:35:31 +0000
Subject: [PATCH 1/2] AlbumArtImages: update the id after dumping to the
 database

---
 mythtv/libs/libmythmetadata/musicmetadata.cpp |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mythtv/libs/libmythmetadata/musicmetadata.cpp b/mythtv/libs/libmythmetadata/musicmetadata.cpp
index c025ef0..7cd739c 100644
--- a/mythtv/libs/libmythmetadata/musicmetadata.cpp
+++ b/mythtv/libs/libmythmetadata/musicmetadata.cpp
@@ -2033,5 +2033,10 @@ void AlbumArtImages::dumpToDatabase(void)
         if (!query.exec())
             MythDB::DBError("AlbumArtImages::dumpToDatabase - "
                             "add/update music_albumart", query);
+        else
+        {
+            if (image->id <= 0)
+                image->id = query.lastInsertId().toInt();
+        }
     }
 }
-- 
1.7.10.2

