diff -Naur mythtv-0.27-20140316-g30d86b1-old/mythplugins/mythmusic/mythmusic/musicplayer.cpp mythtv-0.27-20140316-g30d86b1-new/mythplugins/mythmusic/mythmusic/musicplayer.cpp
--- mythtv-0.27-20140316-g30d86b1-old/mythplugins/mythmusic/mythmusic/musicplayer.cpp	2014-03-17 07:59:18.000000000 +0100
+++ mythtv-0.27-20140316-g30d86b1-new/mythplugins/mythmusic/mythmusic/musicplayer.cpp	2014-03-17 08:01:25.243485030 +0100
@@ -1297,12 +1297,12 @@
             if (GetMythDB()->GetNumSetting("AllowTagWriting", 0) == 1)
             {
                 QStringList strList;
-                strList << QString("MUSIC_TAG_UPDATE_VOLATILE %1 %2 %3 %4 %5")
-                                   .arg(getCurrentMetadata()->Hostname())
-                                   .arg(getCurrentMetadata()->ID())
-                                   .arg(getCurrentMetadata()->Rating())
-                                   .arg(getCurrentMetadata()->Playcount())
-                                   .arg(getCurrentMetadata()->LastPlay().toString(Qt::ISODate));
+                strList << QString("MUSIC_TAG_UPDATE_VOLATILE")
+                        << getCurrentMetadata()->Hostname()
+                        << QString::number(getCurrentMetadata()->ID())
+                        << QString::number(getCurrentMetadata()->Rating())
+                        << QString::number(getCurrentMetadata()->Playcount())
+                        << getCurrentMetadata()->LastPlay().toString(Qt::ISODate);
                 SendStringListThread *thread = new SendStringListThread(strList);
                 MThreadPool::globalInstance()->start(thread, "UpdateVolatile");
             }
diff -Naur mythtv-0.27-20140316-g30d86b1-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-0.27-20140316-g30d86b1-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-0.27-20140316-g30d86b1-old/mythtv/programs/mythbackend/mainserver.cpp	2014-03-17 07:59:19.000000000 +0100
+++ mythtv-0.27-20140316-g30d86b1-new/mythtv/programs/mythbackend/mainserver.cpp	2014-03-17 08:08:08.997238420 +0100
@@ -801,10 +801,10 @@
     }
     else if (command == "MUSIC_TAG_UPDATE_VOLATILE")
     {
-        if (tokens.size() != 6)
+        if (listline.size() != 6)
             LOG(VB_GENERAL, LOG_ERR, "Bad MUSIC_TAG_UPDATE_VOLATILE");
         else
-            HandleMusicTagUpdateVolatile(tokens, pbs);
+            HandleMusicTagUpdateVolatile(listline, pbs);
     }
     else if (command == "MUSIC_CALC_TRACK_LENGTH")
     {
@@ -5257,8 +5257,7 @@
         if (slave)
         {
             LOG(VB_GENERAL, LOG_INFO, QString("HandleMusicTagUpdateVolatile: asking slave '%1' to update the metadata").arg(hostname));
-            strlist << slist.join(" ");
-            strlist = slave->ForwardRequest(strlist);
+            strlist = slave->ForwardRequest(slist);
             slave->DecrRef();
 
             if (pbssock)
