diff -Naur mythtv-0.27-20140316-g30d86b1-old/mythplugins/mythmusic/mythmusic/editmetadata.cpp mythtv-0.27-20140316-g30d86b1-new/mythplugins/mythmusic/mythmusic/editmetadata.cpp
--- mythtv-0.27-20140316-g30d86b1-old/mythplugins/mythmusic/mythmusic/editmetadata.cpp	2014-03-17 07:59:18.000000000 +0100
+++ mythtv-0.27-20140316-g30d86b1-new/mythplugins/mythmusic/mythmusic/editmetadata.cpp	2014-03-17 08:10:43.738685513 +0100
@@ -225,9 +225,9 @@
     if (GetMythDB()->GetNumSetting("AllowTagWriting", 0))
     {
         QStringList strList;
-        strList << QString("MUSIC_TAG_UPDATE_METADATA %1 %2")
-                            .arg(m_metadata->Hostname())
-                            .arg(m_metadata->ID());
+        strList << "MUSIC_TAG_UPDATE_METADATA %1 %2"
+                << m_metadata->Hostname()
+                << QString::number(m_metadata->ID());
 
         SendStringListThread *thread = new SendStringListThread(strList);
         MThreadPool::globalInstance()->start(thread, "UpdateMetadata");
@@ -784,9 +784,11 @@
             }
             else if (resulttext == tr("Check Track Length"))
             {
-                QString command = QString("MUSIC_CALC_TRACK_LENGTH %1 %2")
-                                          .arg(m_metadata->Hostname()).arg(m_metadata->ID());
-                QStringList strList(command);
+                QStringList strList;
+                strList << "MUSIC_CALC_TRACK_LENGTH"
+                        << m_metadata->Hostname()
+                        << QString::number(m_metadata->ID());
+
                 SendStringListThread *thread = new SendStringListThread(strList);
                 MThreadPool::globalInstance()->start(thread, "Send MUSIC_CALC_TRACK_LENGTH");
 
diff -Naur mythtv-0.27-20140316-g30d86b1-old/mythtv/libs/libmythmetadata/musicmetadata.cpp mythtv-0.27-20140316-g30d86b1-new/mythtv/libs/libmythmetadata/musicmetadata.cpp
--- mythtv-0.27-20140316-g30d86b1-old/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-03-17 07:59:18.000000000 +0100
+++ mythtv-0.27-20140316-g30d86b1-new/mythtv/libs/libmythmetadata/musicmetadata.cpp	2014-03-17 08:10:43.738685513 +0100
@@ -1092,10 +1092,10 @@
                 if (albumart_image->embedded)
                 {
                     QStringList slist;
-                    slist << QString("MUSIC_TAG_GETIMAGE %1 %2 %3")
-                            .arg(Hostname())
-                            .arg(ID())
-                            .arg(albumart_image->imageType);
+                    slist << "MUSIC_TAG_GETIMAGE"
+                          << Hostname()
+                          << QString::number(ID())
+                          << QString::number(albumart_image->imageType);
                     gCoreContext->SendReceiveStringList(slist);
                 }
             }
@@ -1790,9 +1790,10 @@
         busy = NULL;
     }
 
-    QStringList strList(QString("MUSIC_FIND_ALBUMART %1 %2 1")
-                                .arg(m_parent->Hostname())
-                                .arg(m_parent->ID()));
+    QStringList strList;
+    strList << "MUSIC_FIND_ALBUMART"
+            << m_parent->Hostname()
+            << QString::number(m_parent->ID());
 
     AlbumArtScannerThread *scanThread = new AlbumArtScannerThread(strList);
     scanThread->start();
diff -Naur mythtv-0.27-20140316-g30d86b1-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-0.27-20140316-g30d86b1-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-0.27-20140316-g30d86b1-old/mythtv/programs/mythbackend/mainserver.cpp	2014-03-17 08:08:08.000000000 +0100
+++ mythtv-0.27-20140316-g30d86b1-new/mythtv/programs/mythbackend/mainserver.cpp	2014-03-17 11:14:58.935879064 +0100
@@ -808,31 +808,31 @@
     }
     else if (command == "MUSIC_CALC_TRACK_LENGTH")
     {
-        if (tokens.size() != 3)
+        if (listline.size() != 3)
             LOG(VB_GENERAL, LOG_ERR, "Bad MUSIC_CALC_TRACK_LENGTH");
         else
-            HandleMusicCalcTrackLen(tokens, pbs);
+            HandleMusicCalcTrackLen(listline, pbs);
     }
     else if (command == "MUSIC_TAG_UPDATE_METADATA")
     {
-        if (tokens.size() != 3)
+        if (listline.size() != 3)
             LOG(VB_GENERAL, LOG_ERR, "Bad MUSIC_TAG_UPDATE_METADATA");
         else
-            HandleMusicTagUpdateMetadata(tokens, pbs);
+            HandleMusicTagUpdateMetadata(listline, pbs);
     }
     else if (command == "MUSIC_FIND_ALBUMART")
     {
-        if (tokens.size() != 4)
+        if (listline.size() != 4)
             LOG(VB_GENERAL, LOG_ERR, "Bad MUSIC_FIND_ALBUMART");
         else
-            HandleMusicFindAlbumArt(tokens, pbs);
+            HandleMusicFindAlbumArt(listline, pbs);
     }
     else if (command == "MUSIC_TAG_GETIMAGE")
     {
-        if (tokens.size() < 4)
+        if (listline.size() < 4)
             LOG(VB_GENERAL, LOG_ERR, "Bad MUSIC_TAG_GETIMAGE");
         else
-            HandleMusicTagGetImage(tokens, pbs);
+            HandleMusicTagGetImage(listline, pbs);
     }
     else if (command == "MUSIC_TAG_ADDIMAGE")
     {
@@ -5319,8 +5319,7 @@
         if (slave)
         {
             LOG(VB_GENERAL, LOG_INFO, QString("HandleMusicCalcTrackLen: asking slave '%1' to update the track length").arg(hostname));
-            strlist << slist.join(" ");
-            strlist = slave->ForwardRequest(strlist);
+            strlist = slave->ForwardRequest(slist);
             slave->DecrRef();
 
             if (pbssock)
@@ -5381,8 +5380,7 @@
         {
             LOG(VB_GENERAL, LOG_INFO, QString("HandleMusicTagUpdateMetadata: asking slave '%1' "
                                               "to update the metadata").arg(hostname));
-            strlist << slist.join(" ");
-            strlist = slave->ForwardRequest(strlist);
+            strlist = slave->ForwardRequest(slist);
             slave->DecrRef();
 
             if (pbssock)
@@ -5468,8 +5466,7 @@
         {
             LOG(VB_GENERAL, LOG_INFO, QString("HandleMusicFindAlbumArt: asking slave '%1' "
                                               "to update the albumart").arg(hostname));
-            strlist << slist.join(" ");
-            strlist = slave->ForwardRequest(strlist);
+            strlist = slave->ForwardRequest(slist);
             slave->DecrRef();
 
             if (pbssock)
@@ -5614,51 +5611,43 @@
     QString songid = slist[2];
     QString imagetype = slist[3];
 
-    QStringList paramList;
-    paramList.append(QString("--songid='%1'").arg(songid));
-    paramList.append(QString("--imagetype='%1'").arg(imagetype));
-
-    QString command = "mythutil --extractimage " + paramList.join(" ");
-
-    if (ismaster)
+    if (ismaster && hostname != gCoreContext->GetHostName())
     {
 
-        if (hostname == gCoreContext->GetHostName())
+        // forward the request to the slave BE
+        PlaybackSock *slave = GetMediaServerByHostname(hostname);
+        if (slave)
         {
-            // this is the master BE
-            LOG(VB_GENERAL, LOG_INFO, QString("HandleMusicTagGetImage: running %1 on master BE '%2'").arg(command).arg(hostname));
+            LOG(VB_GENERAL, LOG_INFO, LOC +
+                QString("HandleMusicTagGetImage: asking slave '%1' to "
+                        "extract the image").arg(hostname));
+            strlist = slave->ForwardRequest(slist);
+            slave->DecrRef();
 
-            QScopedPointer<MythSystem> cmd(MythSystem::Create(command,
-                                                              kMSAutoCleanup | kMSRunBackground |
-                                                              kMSDontDisableDrawing | kMSProcessEvents |
-                                                              kMSDontBlockInputDevs));
+            if (pbssock)
+                SendResponse(pbssock, strlist);
+
+            return;
         }
         else
         {
-            // forward the request to the slave BE
-            PlaybackSock *slave = GetMediaServerByHostname(hostname);
-            if (slave)
-            {
-                LOG(VB_GENERAL, LOG_INFO, QString("HandleMusicTagGetImage: asking slave '%1' to extract the image").arg(hostname));
-                strlist << slist.join(" ");
-                slave->ForwardRequest(strlist);
-                slave->DecrRef();
-            }
-            else
-            {
-                LOG(VB_GENERAL, LOG_INFO,
-                    QString("HandleMusicTagGetImage: Failed to grab slave socket on '%1'").arg(hostname));
-            }
+            LOG(VB_GENERAL, LOG_INFO, LOC +
+                QString("HandleMusicTagGetImage: Failed to grab slave "
+                        "socket on '%1'").arg(hostname));
         }
     }
     else
     {
-        // must be a slave run mythutil to extract the image
-        LOG(VB_GENERAL, LOG_INFO, QString("HandleMusicTagGetImage: running %1 on slave BE '%2'").arg(command).arg(gCoreContext->GetHostName()));
+        QStringList paramList;
+        paramList.append(QString("--songid='%1'").arg(songid));
+        paramList.append(QString("--imagetype='%1'").arg(imagetype));
+
+        QString command = "mythutil --extractimage " + paramList.join(" ");
+
         QScopedPointer<MythSystem> cmd(MythSystem::Create(command,
-                                                          kMSAutoCleanup | kMSRunBackground |
-                                                          kMSDontDisableDrawing | kMSProcessEvents |
-                                                          kMSDontBlockInputDevs));
+                                        kMSAutoCleanup | kMSRunBackground |
+                                        kMSDontDisableDrawing | kMSProcessEvents |
+                                        kMSDontBlockInputDevs));
     }
 
     strlist << "OK";
