From 1153fae00c7eef9227ba005d2c0421c5fb4d42e5 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 16 Mar 2014 18:27:03 +0000
Subject: [PATCH 3/5] Silence a couple of compiler warnings

---
 mythplugins/mythmusic/mythmusic/editmetadata.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mythplugins/mythmusic/mythmusic/editmetadata.cpp b/mythplugins/mythmusic/mythmusic/editmetadata.cpp
index 8bc3fa4..3320017 100644
--- a/mythplugins/mythmusic/mythmusic/editmetadata.cpp
+++ b/mythplugins/mythmusic/mythmusic/editmetadata.cpp
@@ -1208,12 +1208,12 @@ void EditAlbumartDialog::customEvent(QEvent *event)
             {
                 if (tokens.size() >= 2)
                 {
-                    int songID = tokens[1].toInt();
+                    MusicMetadata::IdType songID = (MusicMetadata::IdType)tokens[1].toInt();
 
                     if (m_metadata->ID() == songID)
                     {
                         // force all the image to reload
-                        for (int x = 0; x < m_metadata->getAlbumArtImages()->getImageCount(); x++)
+                        for (uint x = 0; x < m_metadata->getAlbumArtImages()->getImageCount(); x++)
                             removeCachedImage(m_metadata->getAlbumArtImages()->getImageAt(x));
 
                         updateImageGrid();
-- 
1.7.10.2

