From 0787da3c22a9651b147d2e788a59866353bc9e81 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 16 Mar 2014 23:19:03 +0000
Subject: [PATCH 04/11] AlbumArtImage: Fix leak in findImages. Fixes Coverity
 ID #1192757

---
 mythtv/libs/libmythmetadata/musicmetadata.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/mythtv/libs/libmythmetadata/musicmetadata.cpp b/mythtv/libs/libmythmetadata/musicmetadata.cpp
index fc669af..f0478ca 100644
--- a/mythtv/libs/libmythmetadata/musicmetadata.cpp
+++ b/mythtv/libs/libmythmetadata/musicmetadata.cpp
@@ -1841,6 +1841,8 @@ void AlbumArtImages::scanForImages()
 
         image->hostname = strList[x + 5];
         addImage(image);
+
+        delete image;
     }
 }
 
-- 
1.7.10.2

