diff -Naur mythtv-0.27-build-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-0.27-build-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-0.27-build-old/mythtv/programs/mythbackend/mainserver.cpp	2014-03-25 19:07:38.000000000 +0100
+++ mythtv-0.27-build-new/mythtv/programs/mythbackend/mainserver.cpp	2014-03-25 19:14:18.547905056 +0100
@@ -3448,10 +3448,12 @@
 void MainServer::HandleSGFileQuery(QStringList &sList,
                                      PlaybackSock *pbs)
 {
+//format: QUERY_SG_FILEQUERY <host> <storagegroup> <filename> <allowfallback (optional)>
+
     MythSocket *pbssock = pbs->getSocket();
     QStringList strList;
 
-    if (sList.size() != 4)
+    if (sList.size() < 4)
     {
         LOG(VB_GENERAL, LOG_ERR,
             QString("HandleSGFileQuery: Invalid Request. %1")
@@ -3465,6 +3467,11 @@
     QString groupname = sList.at(2);
     QString filename = sList.at(3);
 
+    bool allowFallback = true;
+    if (sList.size() >= 5)
+        allowFallback = (sList.at(4).toInt() > 0);
+    LOG(VB_FILE, LOG_ERR, QString("HandleSGFileQuery - allowFallback: %1").arg(allowFallback));
+
     bool slaveUnreachable = false;
 
     LOG(VB_FILE, LOG_INFO, QString("HandleSGFileQuery: %1")
@@ -3475,7 +3482,7 @@
         (wantHost == gCoreContext->GetSetting("BackendServerIP6")))
     {
         LOG(VB_FILE, LOG_INFO, "HandleSGFileQuery: Getting local info");
-        StorageGroup sg(groupname, gCoreContext->GetHostName());
+        StorageGroup sg(groupname, gCoreContext->GetHostName(), allowFallback);
         strList = sg.GetFileInfo(filename);
     }
     else
