From 68ffe67a9ded86c393ff829624e9a42a60229abf Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Tue, 25 Mar 2014 15:51:05 +0000
Subject: [PATCH 4/6] MythMusic: fix filename for embedded images after a file
 scan

---
 mythtv/libs/libmythmetadata/musicmetadata.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/mythtv/libs/libmythmetadata/musicmetadata.cpp b/mythtv/libs/libmythmetadata/musicmetadata.cpp
index 5bfee9c..6d39fd6 100644
--- a/mythtv/libs/libmythmetadata/musicmetadata.cpp
+++ b/mythtv/libs/libmythmetadata/musicmetadata.cpp
@@ -991,6 +991,8 @@ void MusicMetadata::setEmbeddedAlbumArt(AlbumArtList &albumart)
 
     for (int x = 0; x < albumart.size(); x++)
     {
+        AlbumArtImage *image = albumart.at(x);
+        image->filename = QString("%1-%2").arg(m_id).arg(image->filename);
         m_albumArt->addImage(albumart.at(x));
     }
 
-- 
1.7.10.2

