From 7f1f206f917008d57e9e5714c05af19fa2fe2e2d Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Wed, 2 Apr 2014 15:06:57 +0100
Subject: [PATCH 09/10] AlbumArtImages: add some additional debug messages to
 scanForImages()

---
 mythtv/libs/libmythmetadata/musicmetadata.cpp |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/mythtv/libs/libmythmetadata/musicmetadata.cpp b/mythtv/libs/libmythmetadata/musicmetadata.cpp
index 6d39fd6..660ae22 100644
--- a/mythtv/libs/libmythmetadata/musicmetadata.cpp
+++ b/mythtv/libs/libmythmetadata/musicmetadata.cpp
@@ -1867,6 +1867,16 @@ void AlbumArtImages::scanForImages()
         }
 
         image->hostname = strList[x + 5];
+
+        LOG(VB_FILE, LOG_INFO, "AlbumArtImages::scanForImages found image");
+        LOG(VB_FILE, LOG_INFO, QString("ID: %1").arg(image->id));
+        LOG(VB_FILE, LOG_INFO, QString("ImageType: %1").arg(image->imageType));
+        LOG(VB_FILE, LOG_INFO, QString("Embedded: %1").arg(image->embedded));
+        LOG(VB_FILE, LOG_INFO, QString("Description: %1").arg(image->description));
+        LOG(VB_FILE, LOG_INFO, QString("Filename: %1").arg(image->filename));
+        LOG(VB_FILE, LOG_INFO, QString("Hostname: %1").arg(image->hostname));
+        LOG(VB_FILE, LOG_INFO, "-------------------------------");
+
         addImage(image);
 
         delete image;
-- 
1.7.10.2

