diff --speed-large-files --no-dereference --minimal -Naur a/src/gn/desc_builder.cc b/src/gn/desc_builder.cc
--- a/src/gn/desc_builder.cc	2023-05-17 16:26:19.823000000 +0200
+++ b/src/gn/desc_builder.cc	2023-05-17 17:23:32.122629872 +0200
@@ -167,7 +167,7 @@
     base::ListValue res;
     for (const auto& v : vector)
       res.GetList().emplace_back(ToBaseValue(v));
-    return std::move(res);
+    return res;
   }
 
   base::Value ToBaseValue(const Scope* scope) {
@@ -176,7 +176,7 @@
     scope->GetCurrentScopeValues(&map);
     for (const auto& v : map)
       res.SetKey(v.first, ToBaseValue(v.second));
-    return std::move(res);
+    return res;
   }
 
   base::Value ToBaseValue(const Value& val) {
