diff --speed-large-files --no-dereference --minimal -Naur linux-6.17-rc4/drivers/gpu/drm/sun4i/sun8i_dw_hdmi.c linux-6.17-rc4/drivers/gpu/drm/sun4i/sun8i_dw_hdmi.c
--- linux-6.17-rc4/drivers/gpu/drm/sun4i/sun8i_dw_hdmi.c	2025-09-08 10:42:56.404468248 +0200
+++ linux-6.17-rc4/drivers/gpu/drm/sun4i/sun8i_dw_hdmi.c	2025-09-08 10:38:39.664467371 +0200
@@ -189,6 +189,11 @@
 	if (!hdmi)
 		return -ENOMEM;
 
+	hdmi = devm_drm_bridge_alloc(&pdev->dev, struct sun8i_dw_hdmi, enc_bridge,
+				     &sun8i_hdmi_enc_bridge_funcs);
+	if (IS_ERR(hdmi))
+		return PTR_ERR(hdmi);
+
 	plat_data = &hdmi->plat_data;
 	hdmi->dev = &pdev->dev;
 	encoder = &hdmi->encoder;
@@ -280,7 +285,6 @@
 
 	hdmi->hdmi_bridge = of_drm_find_bridge(dev->of_node);
 
-	hdmi->enc_bridge.funcs = &sun8i_hdmi_enc_bridge_funcs;
 	hdmi->enc_bridge.type = DRM_MODE_CONNECTOR_HDMIA;
 	hdmi->enc_bridge.interlace_allowed = true;
 
