diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4.dts linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4.dts
--- linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4.dts	2024-06-26 14:44:49.580956438 +0200
+++ linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4.dts	2024-06-26 12:49:55.411104868 +0200
@@ -25,6 +25,7 @@
 		mmc0 = &sdmmc;
 		mmc1 = &sdhci;
 		spi1 = &spi1;
+		ethernet0 = &gmac;
 	};
 
 	clkin_gmac: external-gmac-clock {
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4-lts.dts linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4-lts.dts
--- linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4-lts.dts	2024-06-26 14:44:49.580956438 +0200
+++ linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-orangepi-4-lts.dts	2024-06-26 12:36:13.394455891 +0200
@@ -25,6 +25,7 @@
 		mmc0 = &sdmmc;
 		mmc1 = &sdhci;
 		spi1 = &spi1;
+		ethernet0 = &gmac;
 	};
 
 	clkin_gmac: external-gmac-clock {
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts
--- linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts	2024-06-26 14:44:49.544289773 +0200
+++ linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4c-plus.dts	2024-06-26 14:16:37.657659544 +0200
@@ -16,6 +16,7 @@
 
 	aliases {
 		ethernet0 = &gmac;
+		ethernet1 = &brcmf;
 		mmc0 = &sdmmc;
 		mmc1 = &sdhci;
 	};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts
--- linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts	2024-06-26 14:44:49.544289773 +0200
+++ linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts	2024-06-26 14:17:11.624325477 +0200
@@ -16,6 +16,8 @@
 		mmc0 = &sdmmc;
 		mmc1 = &sdhci;
 		mmc2 = &sdio0;
+		ethernet0 = &gmac;
+		ethernet1 = &brcmf;
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts
--- linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts	2024-06-26 14:44:49.544289773 +0200
+++ linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts	2024-06-26 14:32:42.637638749 +0200
@@ -16,6 +16,8 @@
 		mmc0 = &sdmmc;
 		mmc1 = &sdhci;
 		mmc2 = &sdio0;
+		ethernet0 = &gmac;
+		ethernet1 = &brcmf;
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts
--- linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts	2024-06-26 14:44:49.544289773 +0200
+++ linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts	2024-06-26 14:32:05.854306204 +0200
@@ -16,6 +16,8 @@
 		mmc0 = &sdmmc;
 		mmc1 = &sdhci;
 		mmc2 = &sdio0;
+		ethernet0 = &gmac;
+		ethernet1 = &brcmf;
 	};
 };
 
diff --speed-large-files --no-dereference --minimal -Naur linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts
--- linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts	2024-06-26 14:44:49.544289773 +0200
+++ linux-6.10-rc5/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts	2024-06-26 14:32:56.140971782 +0200
@@ -17,6 +17,8 @@
 		mmc0 = &sdmmc;
 		mmc1 = &sdhci;
 		mmc2 = &sdio0;
+		ethernet0 = &gmac;
+		ethernet1 = &brcmf;
 	};
 };
 
