diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712.dtsi linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
--- linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712.dtsi	2025-06-25 17:49:48.764974084 +0200
+++ linux-6.16-rc3/arch/arm64/boot/dts/broadcom/bcm2712.dtsi	2025-06-25 17:39:34.311654091 +0200
@@ -458,6 +458,37 @@
 			compatible = "brcm,bcm2712-vc6";
 		};
 
+		iommu2: iommu@5100 {
+			/* IOMMU2 for PISP-BE, HEVC; and (unused) H264 accelerators */
+			compatible = "brcm,bcm2712-iommu";
+			reg = <0x10 0x5100  0x0 0x80>;
+			cache = <&iommuc>;
+			#iommu-cells = <0>;
+		};
+
+		iommu4: iommu@5200 {
+			/* IOMMU4 for HVS, MPL/TXP; and (unused) Unicam, PISP-FE, MiniBVN */
+			compatible = "brcm,bcm2712-iommu";
+			reg = <0x10 0x5200  0x0 0x80>;
+			cache = <&iommuc>;
+			#iommu-cells = <0>;
+			#interconnect-cells = <0>;
+		};
+
+		iommu5: iommu@5280 {
+			/* IOMMU5 for PCIe2 (RP1); and (unused) BSTM */
+			compatible = "brcm,bcm2712-iommu";
+			reg = <0x10 0x5280  0x0 0x80>;
+			cache = <&iommuc>;
+			#iommu-cells = <0>;
+			dma-iova-offset = <0x10 0x00000000>; // HACK for RP1 masters over PCIe
+		};
+
+		iommuc: iommuc@5b00 {
+			compatible = "brcm,bcm2712-iommuc";
+			reg = <0x10 0x5b00  0x0 0x80>;
+		};
+
 		dma32: dma@10000 {
 			compatible = "brcm,bcm2712-dma-lite";
 			reg = <0x10 0x00010000 0 0x600>;
@@ -661,7 +692,7 @@
 
 			clocks = <&firmware_clocks 11>;
 			clock-names = "hevc";
-			//iommus = <&iommu2>;
+			iommus = <&iommu2>;
 			status = "okay";
 		};
 	};
@@ -700,5 +731,6 @@
 		interrupt-parent = <&disp_intr>;
 		interrupts = <2>, <9>, <16>;
 		interrupt-names = "ch0-eof", "ch1-eof", "ch2-eof";
+		iommus = <&iommu4>;
 	};
 };
