diff --speed-large-files --no-dereference --minimal -Naur linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec.c linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec.c
--- linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec.c	2025-09-22 10:33:23.463385177 +0200
+++ linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec.c	2025-09-21 17:47:33.616721588 +0200
@@ -593,7 +593,7 @@
 static int rkvdec_enum_framesizes(struct file *file, void *priv,
 				  struct v4l2_frmsizeenum *fsize)
 {
-	struct rkvdec_ctx *ctx = file_to_rkvdec_ctx(file);
+	struct rkvdec_ctx *ctx = fh_to_rkvdec_ctx(priv);
 	const struct rkvdec_coded_fmt_desc *desc;
 
 	if (fsize->index != 0)
@@ -632,7 +632,7 @@
 				  struct v4l2_format *f)
 {
 	struct v4l2_pix_format_mplane *pix_mp = &f->fmt.pix_mp;
-	struct rkvdec_ctx *ctx = file_to_rkvdec_ctx(file);
+	struct rkvdec_ctx *ctx = fh_to_rkvdec_ctx(priv);
 	const struct rkvdec_coded_fmt_desc *coded_desc;
 
 	/*
@@ -665,7 +665,7 @@
 				 struct v4l2_format *f)
 {
 	struct v4l2_pix_format_mplane *pix_mp = &f->fmt.pix_mp;
-	struct rkvdec_ctx *ctx = file_to_rkvdec_ctx(file);
+	struct rkvdec_ctx *ctx = fh_to_rkvdec_ctx(priv);
 	const struct rkvdec_coded_fmt_desc *desc;
 
 	desc = rkvdec_find_coded_fmt_desc(ctx, pix_mp->pixelformat);
@@ -696,7 +696,7 @@
 static int rkvdec_s_capture_fmt(struct file *file, void *priv,
 				struct v4l2_format *f)
 {
-	struct rkvdec_ctx *ctx = file_to_rkvdec_ctx(file);
+	struct rkvdec_ctx *ctx = fh_to_rkvdec_ctx(priv);
 	struct vb2_queue *vq;
 	int ret;
 
@@ -717,7 +717,7 @@
 static int rkvdec_s_output_fmt(struct file *file, void *priv,
 			       struct v4l2_format *f)
 {
-	struct rkvdec_ctx *ctx = file_to_rkvdec_ctx(file);
+	struct rkvdec_ctx *ctx = fh_to_rkvdec_ctx(priv);
 	struct v4l2_m2m_ctx *m2m_ctx = ctx->fh.m2m_ctx;
 	const struct rkvdec_coded_fmt_desc *desc;
 	struct v4l2_format *cap_fmt;
@@ -782,7 +782,7 @@
 static int rkvdec_g_output_fmt(struct file *file, void *priv,
 			       struct v4l2_format *f)
 {
-	struct rkvdec_ctx *ctx = file_to_rkvdec_ctx(file);
+	struct rkvdec_ctx *ctx = fh_to_rkvdec_ctx(priv);
 
 	*f = ctx->coded_fmt;
 	return 0;
@@ -791,7 +791,7 @@
 static int rkvdec_g_capture_fmt(struct file *file, void *priv,
 				struct v4l2_format *f)
 {
-	struct rkvdec_ctx *ctx = file_to_rkvdec_ctx(file);
+	struct rkvdec_ctx *ctx = fh_to_rkvdec_ctx(priv);
 
 	*f = ctx->decoded_fmt;
 	return 0;
@@ -800,7 +800,7 @@
 static int rkvdec_enum_output_fmt(struct file *file, void *priv,
 				  struct v4l2_fmtdesc *f)
 {
-	struct rkvdec_ctx *ctx = file_to_rkvdec_ctx(file);
+	struct rkvdec_ctx *ctx = fh_to_rkvdec_ctx(priv);
 	const struct rkvdec_coded_fmt_desc *desc;
 
 	desc = rkvdec_enum_coded_fmt_desc(ctx, f->index);
@@ -814,7 +814,7 @@
 static int rkvdec_enum_capture_fmt(struct file *file, void *priv,
 				   struct v4l2_fmtdesc *f)
 {
-	struct rkvdec_ctx *ctx = file_to_rkvdec_ctx(file);
+	struct rkvdec_ctx *ctx = fh_to_rkvdec_ctx(priv);
 	u32 fourcc;
 
 	fourcc = rkvdec_enum_decoded_fmt(ctx, f->index, ctx->image_fmt);
@@ -1257,7 +1257,8 @@
 	if (ret)
 		goto err_cleanup_m2m_ctx;
 
-	v4l2_fh_add(&ctx->fh, filp);
+	filp->private_data = &ctx->fh;
+	v4l2_fh_add(&ctx->fh);
 
 	return 0;
 
@@ -1271,9 +1272,9 @@
 
 static int rkvdec_release(struct file *filp)
 {
-	struct rkvdec_ctx *ctx = file_to_rkvdec_ctx(filp);
+	struct rkvdec_ctx *ctx = fh_to_rkvdec_ctx(filp->private_data);
 
-	v4l2_fh_del(&ctx->fh, filp);
+	v4l2_fh_del(&ctx->fh);
 	v4l2_m2m_ctx_release(ctx->fh.m2m_ctx);
 	v4l2_ctrl_handler_free(&ctx->ctrl_hdl);
 	v4l2_fh_exit(&ctx->fh);
diff --speed-large-files --no-dereference --minimal -Naur linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec.h linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec.h
--- linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec.h	2025-09-22 10:33:23.443385176 +0200
+++ linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec.h	2025-09-21 17:47:33.603388254 +0200
@@ -161,9 +161,9 @@
 	void *priv;
 };
 
-static inline struct rkvdec_ctx *file_to_rkvdec_ctx(struct file *filp)
+static inline struct rkvdec_ctx *fh_to_rkvdec_ctx(struct v4l2_fh *fh)
 {
-	return container_of(file_to_v4l2_fh(filp), struct rkvdec_ctx, fh);
+        return container_of(fh, struct rkvdec_ctx, fh);
 }
 
 enum rkvdec_alloc_type {
diff --speed-large-files --no-dereference --minimal -Naur linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec-vp9.c linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec-vp9.c
--- linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec-vp9.c	2025-09-22 10:33:23.446718510 +0200
+++ linux-6.17-rc6/drivers/media/platform/rockchip/rkvdec/rkvdec-vp9.c	2025-09-22 10:34:22.316718725 +0200
@@ -659,7 +659,7 @@
 
 	regs->vp9.reg44.strmd_error_e = 0xe;
 
-	rkvdec_memcpy_toio(rkvdec->regs, regs, MIN(sizeof(*regs), rkvdec->variant->num_regs));
+	rkvdec_memcpy_toio(rkvdec->regs, regs, sizeof(*regs));
 }
 
 static int validate_dec_params(struct rkvdec_ctx *ctx,
