diff --speed-large-files --no-dereference --minimal -Naur linux-6.17/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu381-hevc.c linux-6.17/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu381-hevc.c
--- linux-6.17/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu381-hevc.c	2025-09-30 17:15:16.126716529 +0200
+++ linux-6.17/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu381-hevc.c	2025-09-30 17:08:55.810048453 +0200
@@ -123,10 +123,10 @@
 static void assemble_hw_pps(struct rkvdec_ctx *ctx,
 			    struct rkvdec_hevc_run *run)
 {
-	struct rkvdec_hevc_ctx *h264_ctx = ctx->priv;
+	struct rkvdec_hevc_ctx *h265_ctx = ctx->priv;
 	const struct v4l2_ctrl_hevc_sps *sps = run->sps;
 	const struct v4l2_ctrl_hevc_pps *pps = run->pps;
-	struct rkvdec_hevc_priv_tbl *priv_tbl = h264_ctx->priv_tbl.cpu;
+	struct rkvdec_hevc_priv_tbl *priv_tbl = h265_ctx->priv_tbl.cpu;
 	struct rkvdec_sps_pps_packet *hw_ps;
 	bool tiles_enabled;
 	s32 max_cu_width;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.17/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu383-hevc.c linux-6.17/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu383-hevc.c
--- linux-6.17/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu383-hevc.c	2025-09-30 17:15:16.130049863 +0200
+++ linux-6.17/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu383-hevc.c	2025-09-30 17:09:19.993381859 +0200
@@ -275,11 +275,11 @@
 static void assemble_hw_pps(struct rkvdec_ctx *ctx,
 			    struct rkvdec_hevc_run *run)
 {
-	struct rkvdec_hevc_ctx *h264_ctx = ctx->priv;
+	struct rkvdec_hevc_ctx *h265_ctx = ctx->priv;
 	const struct v4l2_ctrl_hevc_sps *sps = run->sps;
 	const struct v4l2_ctrl_hevc_pps *pps = run->pps;
 	const struct v4l2_ctrl_hevc_decode_params *dec_params = run->decode_params;
-	struct rkvdec_hevc_priv_tbl *priv_tbl = h264_ctx->priv_tbl.cpu;
+	struct rkvdec_hevc_priv_tbl *priv_tbl = h265_ctx->priv_tbl.cpu;
 	struct rkvdec_hevc_sps_pps *hw_ps;
 	bool tiles_enabled;
 	s32 max_cu_width;
