diff --speed-large-files --no-dereference --minimal -Naur linux-6.17.3/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi linux-6.17.3/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
--- linux-6.17.3/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2025-10-15 21:19:24.396715170 +0200
+++ linux-6.17.3/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2025-10-15 21:18:49.570048418 +0200
@@ -2156,6 +2156,7 @@
 			 <&cru SRST_A_EMMC>, <&cru SRST_B_EMMC>,
 			 <&cru SRST_T_EMMC>;
 		reset-names = "core", "bus", "axi", "block", "timer";
+		supports-cqe;
 		status = "disabled";
 	};
 
