diff --speed-large-files --no-dereference --minimal -Naur linux-6.16/arch/arm64/boot/dts/rockchip/rk3588s-nanopi-r6.dtsi linux-6.16/arch/arm64/boot/dts/rockchip/rk3588s-nanopi-r6.dtsi
--- linux-6.16/arch/arm64/boot/dts/rockchip/rk3588s-nanopi-r6.dtsi	2025-07-27 23:26:38.000000000 +0200
+++ linux-6.16/arch/arm64/boot/dts/rockchip/rk3588s-nanopi-r6.dtsi	2025-08-03 10:13:58.060009813 +0200
@@ -63,7 +63,7 @@
 		sys_led: led-0 {
 			label = "sys_led";
 			gpios = <&gpio1 RK_PC1 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "disk-activity";
 			pinctrl-names = "default";
 			pinctrl-0 = <&sys_led_pin>;
 		};
@@ -89,6 +89,12 @@
 		};
 	};
 
+	usbc_vin2_0 {
+		compatible = "iio-hwmon";
+		io-channels = <&saradc 0 &saradc 1 &saradc 2 &saradc 3 &saradc 4 &saradc 5 &saradc 6 &saradc 7>;
+		status = "okay";
+	};
+
 	vcc5v0_sys: regulator-vcc5v0-sys {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc5v0_sys";
@@ -235,6 +241,10 @@
 	status = "okay";
 };
 
+&hdmi0_sound {
+	status = "okay";
+};
+
 &hdmi0 {
 	status = "okay";
 };
@@ -335,6 +345,10 @@
 	};
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
 &mdio1 {
 	rgmii_phy1: ethernet-phy@1 {
 		compatible = "ethernet-phy-id001c.c916";
