diff --speed-large-files --no-dereference --minimal -Naur linux-6.17-rc4/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts linux-6.17-rc4/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts
--- linux-6.17-rc4/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts	2025-09-01 00:33:07.000000000 +0200
+++ linux-6.17-rc4/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts	2025-08-30 17:08:10.447806186 +0200
@@ -1,11 +1,13 @@
 // SPDX-License-Identifier: (GPL-2.0-only OR MIT)
 /*
  * Copyright (C) 2025 Chen-Yu Tsai <wens@csie.org>
+ * additions: piotr.oniszczuk@gmail.com
  */
 
 /dts-v1/;
 
 #include "sun55i-a523.dtsi"
+#include "sun55i-a523-cpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/leds/common.h>
@@ -15,7 +17,12 @@
 	compatible = "xunlong,orangepi-4a", "allwinner,sun55i-t527";
 
 	aliases {
+		mmc0 = &mmc0;
+		mmc1 = &mmc1;
+		mmc2 = &mmc2;
 		serial0 = &uart0;
+		ethernet0 = &gmac1;
+		ethernet1 = &brcmf;
 	};
 
 	chosen {
@@ -37,6 +44,7 @@
 			function = LED_FUNCTION_STATUS;
 			color = <LED_COLOR_ID_GREEN>;
 			gpios = <&pio 3 20 GPIO_ACTIVE_HIGH>; /* PD20 */
+			linux,default-trigger = "disk-activity";
 		};
 	};
 
@@ -67,6 +75,15 @@
 		enable-active-high;
 	};
 
+	reg_pcie_vcc1v8: regulator-pcie-vcc1v8 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc-pcie-1v8";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-always-on;
+		regulator-boot-on;
+	};
+
 	reg_usb_vbus: regulator-usb-vbus {
 		compatible = "regulator-fixed";
 		regulator-name = "usb-vbus";
@@ -87,6 +104,14 @@
 	};
 };
 
+&cpu0 {
+	cpu-supply = <&reg_dcdc1>;
+};
+
+&cpu4 {
+	cpu-supply = <&reg_dcdc1_323>;
+};
+
 &ehci0 {
 	status = "okay";
 };
@@ -95,6 +120,28 @@
 	status = "okay";
 };
 
+&gmac1 {
+	phy-mode = "rgmii-id";
+	phy-handle = <&ext_rgmii_phy>;
+	phy-supply = <&reg_cldo4>;
+
+	tx-internal-delay-ps = <0>;
+	rx-internal-delay-ps = <300>;
+
+	status = "okay";
+};
+
+&mdio1 {
+	ext_rgmii_phy: ethernet-phy@1 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <1>;
+		interrupts-extended = <&pio 8 16 IRQ_TYPE_LEVEL_LOW>; /* PI16 */
+		reset-gpios = <&pio 8 15 GPIO_ACTIVE_LOW>; /* PI15 */
+		reset-assert-us = <10000>;
+		reset-deassert-us = <150000>;
+	};
+};
+
 &gpu {
 	mali-supply = <&reg_dcdc2>;
 	status = "okay";
@@ -108,6 +155,7 @@
 };
 
 &mmc1 {
+	max-frequency = <20000000>;
 	bus-width = <4>;
 	mmc-pwrseq = <&wifi_pwrseq>;
 	non-removable;
@@ -211,6 +259,7 @@
 			};
 
 			reg_aldo2: aldo2 {
+				regulator-always-on;
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
 				regulator-name = "vcc-pe";
@@ -232,6 +281,7 @@
 			};
 
 			reg_bldo1: bldo1 {
+				regulator-always-on;
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
 				regulator-name = "vcc-pg-wifi";
@@ -304,6 +354,7 @@
 		regulators {
 			reg_aldo1_323: aldo1 {
 				/* less capable and shares load with dldo1 */
+				regulator-always-on;
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-name = "vcc-wifi";
@@ -311,6 +362,7 @@
 
 			reg_dldo1_323: dldo1 {
 				/* more capable and shares load with aldo1 */
+				regulator-always-on;
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-name = "vcc-wifi2";
@@ -325,6 +377,9 @@
 			};
 
 			/* DCDC2 is polyphased with DCDC1 */
+			reg_dcdc2_323: dcdc2 {
+				x-powers,force-polyfase-mode;
+			};
 
 			/* Some RISC-V management core related voltage */
 			reg_dcdc3_323: dcdc3 {
@@ -388,3 +443,20 @@
 	usb1_vbus-supply = <&reg_usb_vbus>;
 	status = "okay";
 };
+
+&combophy {
+	resets = <&ccu RST_BUS_PCIE_USB3>;
+	phy_use_sel = <0>; /* 0:PCIE; 1:USB3 */
+	status = "okay";
+};
+
+&pcie {
+	reset-gpios = <&pio 7 11 GPIO_ACTIVE_HIGH>; /* PH 11 */
+	wake-gpios = <&pio 7 12 GPIO_ACTIVE_HIGH>; /* PH 12 */
+	num-lanes  = <1>;
+	clk-freq-100M;
+	pcie3v3-supply = <&reg_pcie_vcc3v3>;
+	pcie1v8-supply = <&reg_pcie_vcc1v8>;
+	status = "okay";
+};
+
