diff -Naur ncurses-5.9-old/Ada95/doc/Makefile.in ncurses-5.9-new/Ada95/doc/Makefile.in
--- ncurses-5.9-old/Ada95/doc/Makefile.in	2011-03-26 12:26:17.000000000 -0700
+++ ncurses-5.9-new/Ada95/doc/Makefile.in	2011-04-05 12:23:24.000000000 -0700
@@ -31,7 +31,7 @@
 #
 # Makefile for AdaCurses manual pages.
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 
 DESTDIR		= @DESTDIR@
diff -Naur ncurses-5.9-old/Ada95/gen/Makefile.in ncurses-5.9-new/Ada95/gen/Makefile.in
--- ncurses-5.9-old/Ada95/gen/Makefile.in	2011-03-26 16:36:30.000000000 -0700
+++ ncurses-5.9-new/Ada95/gen/Makefile.in	2011-04-05 12:23:58.000000000 -0700
@@ -32,7 +32,7 @@
 #
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/Ada95/include/Makefile.in ncurses-5.9-new/Ada95/include/Makefile.in
--- ncurses-5.9-old/Ada95/include/Makefile.in	2011-03-19 16:39:59.000000000 -0700
+++ ncurses-5.9-new/Ada95/include/Makefile.in	2011-04-05 12:23:19.000000000 -0700
@@ -39,7 +39,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/Ada95/Makefile.in ncurses-5.9-new/Ada95/Makefile.in
--- ncurses-5.9-old/Ada95/Makefile.in	2010-11-27 13:45:27.000000000 -0800
+++ ncurses-5.9-new/Ada95/Makefile.in	2011-04-05 12:23:35.000000000 -0700
@@ -32,7 +32,7 @@
 #  Version Control
 #  $Revision: 1.21 $
 #
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/Ada95/samples/Makefile.in ncurses-5.9-new/Ada95/samples/Makefile.in
--- ncurses-5.9-old/Ada95/samples/Makefile.in	2011-03-28 16:45:08.000000000 -0700
+++ ncurses-5.9-new/Ada95/samples/Makefile.in	2011-04-05 12:23:39.000000000 -0700
@@ -32,7 +32,7 @@
 #
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/Ada95/src/Makefile.in ncurses-5.9-new/Ada95/src/Makefile.in
--- ncurses-5.9-old/Ada95/src/Makefile.in	2011-03-31 02:46:16.000000000 -0700
+++ ncurses-5.9-new/Ada95/src/Makefile.in	2011-04-05 12:23:28.000000000 -0700
@@ -32,7 +32,7 @@
 #
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/c++/Makefile.in ncurses-5.9-new/c++/Makefile.in
--- ncurses-5.9-old/c++/Makefile.in	2010-11-27 13:45:27.000000000 -0800
+++ ncurses-5.9-new/c++/Makefile.in	2011-04-05 12:23:10.000000000 -0700
@@ -34,7 +34,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 
 CF_MFLAGS 	= @cf_cv_makeflags@
diff -Naur ncurses-5.9-old/form/Makefile.in ncurses-5.9-new/form/Makefile.in
--- ncurses-5.9-old/form/Makefile.in	2010-11-27 13:45:27.000000000 -0800
+++ ncurses-5.9-new/form/Makefile.in	2011-04-05 12:22:12.000000000 -0700
@@ -42,7 +42,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/include/Makefile.in ncurses-5.9-new/include/Makefile.in
--- ncurses-5.9-old/include/Makefile.in	2010-11-27 13:45:27.000000000 -0800
+++ ncurses-5.9-new/include/Makefile.in	2011-04-05 12:22:08.000000000 -0700
@@ -39,7 +39,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/Makefile.in ncurses-5.9-new/Makefile.in
--- ncurses-5.9-old/Makefile.in	2010-11-27 13:45:27.000000000 -0800
+++ ncurses-5.9-new/Makefile.in	2011-04-05 12:23:05.000000000 -0700
@@ -31,7 +31,7 @@
 #
 # Master Makefile for ncurses library.
 
-SHELL = /bin/sh
+SHELL = @SHELL@
 VPATH		= @srcdir@
 
 DESTDIR=@DESTDIR@
diff -Naur ncurses-5.9-old/man/Makefile.in ncurses-5.9-new/man/Makefile.in
--- ncurses-5.9-old/man/Makefile.in	2010-11-27 13:45:27.000000000 -0800
+++ ncurses-5.9-new/man/Makefile.in	2011-04-05 12:21:40.000000000 -0700
@@ -34,7 +34,7 @@
 # NOTE: When you add or rename a man page, make sure you update both
 # the top-level MANIFEST and any man/*.renames files!
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 
 DESTDIR		= @DESTDIR@
diff -Naur ncurses-5.9-old/menu/Makefile.in ncurses-5.9-new/menu/Makefile.in
--- ncurses-5.9-old/menu/Makefile.in	2010-11-27 13:45:27.000000000 -0800
+++ ncurses-5.9-new/menu/Makefile.in	2011-04-05 12:22:25.000000000 -0700
@@ -42,7 +42,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/misc/Makefile.in ncurses-5.9-new/misc/Makefile.in
--- ncurses-5.9-old/misc/Makefile.in	2011-02-20 17:09:31.000000000 -0800
+++ ncurses-5.9-new/misc/Makefile.in	2011-04-05 12:22:46.000000000 -0700
@@ -42,7 +42,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES :
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/ncurses/Makefile.in ncurses-5.9-new/ncurses/Makefile.in
--- ncurses-5.9-old/ncurses/Makefile.in	2010-11-27 13:45:27.000000000 -0800
+++ ncurses-5.9-new/ncurses/Makefile.in	2011-04-05 12:22:01.000000000 -0700
@@ -45,7 +45,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/panel/Makefile.in ncurses-5.9-new/panel/Makefile.in
--- ncurses-5.9-old/panel/Makefile.in	2010-11-27 13:45:27.000000000 -0800
+++ ncurses-5.9-new/panel/Makefile.in	2011-04-05 12:21:45.000000000 -0700
@@ -42,7 +42,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/progs/Makefile.in ncurses-5.9-new/progs/Makefile.in
--- ncurses-5.9-old/progs/Makefile.in	2010-11-27 13:45:27.000000000 -0800
+++ ncurses-5.9-new/progs/Makefile.in	2011-04-05 12:21:53.000000000 -0700
@@ -42,7 +42,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-old/test/Makefile.in ncurses-5.9-new/test/Makefile.in
--- ncurses-5.9-old/test/Makefile.in	2010-11-28 08:39:40.000000000 -0800
+++ ncurses-5.9-new/test/Makefile.in	2011-04-05 12:21:32.000000000 -0700
@@ -34,7 +34,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 
 @SET_MAKE@
