diff -Naur ncurses-5.9-20150530-old/Ada95/doc/Makefile.in ncurses-5.9-20150530-new/Ada95/doc/Makefile.in
--- ncurses-5.9-20150530-old/Ada95/doc/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/Ada95/doc/Makefile.in	2015-06-24 16:55:43.302001227 -0700
@@ -31,7 +31,7 @@
 #
 # Makefile for AdaCurses manual pages.
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 
 DESTDIR		= @DESTDIR@
diff -Naur ncurses-5.9-20150530-old/Ada95/gen/Makefile.in ncurses-5.9-20150530-new/Ada95/gen/Makefile.in
--- ncurses-5.9-20150530-old/Ada95/gen/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/Ada95/gen/Makefile.in	2015-06-24 16:55:43.302001227 -0700
@@ -32,7 +32,7 @@
 #
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/Ada95/include/Makefile.in ncurses-5.9-20150530-new/Ada95/include/Makefile.in
--- ncurses-5.9-20150530-old/Ada95/include/Makefile.in	2015-06-24 16:32:40.000000000 -0700
+++ ncurses-5.9-20150530-new/Ada95/include/Makefile.in	2015-06-24 16:55:43.302001227 -0700
@@ -39,7 +39,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/Ada95/Makefile.in ncurses-5.9-20150530-new/Ada95/Makefile.in
--- ncurses-5.9-20150530-old/Ada95/Makefile.in	2015-06-24 16:32:40.000000000 -0700
+++ ncurses-5.9-20150530-new/Ada95/Makefile.in	2015-06-24 16:55:43.302001227 -0700
@@ -32,7 +32,7 @@
 #  Version Control
 #  $Revision: 1.21 $
 #
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/Ada95/samples/Makefile.in ncurses-5.9-20150530-new/Ada95/samples/Makefile.in
--- ncurses-5.9-20150530-old/Ada95/samples/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/Ada95/samples/Makefile.in	2015-06-24 16:55:43.302001227 -0700
@@ -32,7 +32,7 @@
 #
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/Ada95/src/Makefile.in ncurses-5.9-20150530-new/Ada95/src/Makefile.in
--- ncurses-5.9-20150530-old/Ada95/src/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/Ada95/src/Makefile.in	2015-06-24 16:55:43.303001171 -0700
@@ -32,7 +32,7 @@
 #
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/c++/Makefile.in ncurses-5.9-20150530-new/c++/Makefile.in
--- ncurses-5.9-20150530-old/c++/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/c++/Makefile.in	2015-06-24 16:55:43.303001171 -0700
@@ -34,7 +34,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 
 CF_MFLAGS 	= @cf_cv_makeflags@
diff -Naur ncurses-5.9-20150530-old/form/Makefile.in ncurses-5.9-20150530-new/form/Makefile.in
--- ncurses-5.9-20150530-old/form/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/form/Makefile.in	2015-06-24 16:55:43.303001171 -0700
@@ -42,7 +42,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/include/Makefile.in ncurses-5.9-20150530-new/include/Makefile.in
--- ncurses-5.9-20150530-old/include/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/include/Makefile.in	2015-06-24 16:55:43.303001171 -0700
@@ -39,7 +39,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/Makefile.in ncurses-5.9-20150530-new/Makefile.in
--- ncurses-5.9-20150530-old/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/Makefile.in	2015-06-24 16:55:43.303001171 -0700
@@ -31,7 +31,7 @@
 #
 # Master Makefile for ncurses library.
 
-SHELL = /bin/sh
+SHELL = @SHELL@
 VPATH		= @srcdir@
 
 DESTDIR=@DESTDIR@
diff -Naur ncurses-5.9-20150530-old/man/Makefile.in ncurses-5.9-20150530-new/man/Makefile.in
--- ncurses-5.9-20150530-old/man/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/man/Makefile.in	2015-06-24 16:55:43.303001171 -0700
@@ -34,7 +34,7 @@
 # NOTE: When you add or rename a man page, make sure you update both
 # the top-level MANIFEST and any man/*.renames files!
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 
 DESTDIR		= @DESTDIR@
diff -Naur ncurses-5.9-20150530-old/menu/Makefile.in ncurses-5.9-20150530-new/menu/Makefile.in
--- ncurses-5.9-20150530-old/menu/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/menu/Makefile.in	2015-06-24 16:55:43.304001114 -0700
@@ -42,7 +42,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/misc/Makefile.in ncurses-5.9-20150530-new/misc/Makefile.in
--- ncurses-5.9-20150530-old/misc/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/misc/Makefile.in	2015-06-24 16:55:43.304001114 -0700
@@ -42,7 +42,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES :
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/ncurses/Makefile.in ncurses-5.9-20150530-new/ncurses/Makefile.in
--- ncurses-5.9-20150530-old/ncurses/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/ncurses/Makefile.in	2015-06-24 16:55:43.304001114 -0700
@@ -45,7 +45,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/panel/Makefile.in ncurses-5.9-20150530-new/panel/Makefile.in
--- ncurses-5.9-20150530-old/panel/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/panel/Makefile.in	2015-06-24 16:55:43.304001114 -0700
@@ -42,7 +42,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/progs/Makefile.in ncurses-5.9-20150530-new/progs/Makefile.in
--- ncurses-5.9-20150530-old/progs/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/progs/Makefile.in	2015-06-24 16:55:43.304001114 -0700
@@ -42,7 +42,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 THIS		= Makefile
 
diff -Naur ncurses-5.9-20150530-old/test/Makefile.in ncurses-5.9-20150530-new/test/Makefile.in
--- ncurses-5.9-20150530-old/test/Makefile.in	2015-06-24 16:35:34.000000000 -0700
+++ ncurses-5.9-20150530-new/test/Makefile.in	2015-06-24 16:55:43.304001114 -0700
@@ -34,7 +34,7 @@
 # turn off _all_ suffix rules; we'll generate our own
 .SUFFIXES:
 
-SHELL		= /bin/sh
+SHELL		= @SHELL@
 VPATH		= @srcdir@
 
 @SET_MAKE@
