From 1475d413f82ce657fc2cbe45f94fb39849d5d2e1 Mon Sep 17 00:00:00 2001
From: Andre Przywara <andre.przywara@arm.com>
Date: Wed, 25 Nov 2020 22:41:11 +0000
Subject: [PATCH 09/13] only do poweroff without PSCI

---
 drivers/power/axp305.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/power/axp305.c b/drivers/power/axp305.c
index 53d4d3f2..326f561a 100644
--- a/drivers/power/axp305.c
+++ b/drivers/power/axp305.c
@@ -66,6 +66,7 @@ int axp_init(void)
 	return ret;
 }
 
+#ifndef CONFIG_PSCI_RESET
 int do_poweroff(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 {
 	pmic_bus_write(AXP305_SHUTDOWN, AXP305_POWEROFF);
@@ -76,3 +77,4 @@ int do_poweroff(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 	/* not reached */
 	return 0;
 }
+#endif
-- 
2.29.2

