From 4ad12320f37c6f3673927fd4cb37e7b1a75aea69 Mon Sep 17 00:00:00 2001
From: Andre Przywara <andre.przywara@arm.com>
Date: Fri, 27 Nov 2020 16:37:07 +0000
Subject: [PATCH 10/13] bring back SPL_FIT support

---
 arch/arm/mach-sunxi/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-sunxi/Kconfig b/arch/arm/mach-sunxi/Kconfig
index ccda05eb..0d11ba39 100644
--- a/arch/arm/mach-sunxi/Kconfig
+++ b/arch/arm/mach-sunxi/Kconfig
@@ -319,6 +319,8 @@ config MACH_SUN50I_H616
 	bool "sun50i (Allwinner H616)"
 	select ARM64
 	select DRAM_SUN50I_H616
+	select FIT
+	select SPL_LOAD_FIT
 	select SUNXI_GEN_SUN6I
 	select SUPPORT_SPL
 
-- 
2.29.2

