From 58d01656415d741c658d74fd9e285fdd1d51a6ef Mon Sep 17 00:00:00 2001
From: Andre Przywara <andre.przywara@arm.com>
Date: Sat, 28 Nov 2020 02:20:57 +0000
Subject: [PATCH 13/13] H616: use TF-A address in DRAM

---
 arch/arm/dts/sunxi-u-boot.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/dts/sunxi-u-boot.dtsi b/arch/arm/dts/sunxi-u-boot.dtsi
index c77cf7ca..96838f43 100644
--- a/arch/arm/dts/sunxi-u-boot.dtsi
+++ b/arch/arm/dts/sunxi-u-boot.dtsi
@@ -3,6 +3,9 @@
 #ifdef CONFIG_MACH_SUN50I_H6
 #define BL31_ADDR 0x104000
 #define  SCP_ADDR 0x114000
+#elif defined(CONFIG_MACH_SUN50I_H616)
+#define BL31_ADDR 0x40004000
+#define  SCP_ADDR 0x028000
 #else
 #define BL31_ADDR  0x44000
 #define  SCP_ADDR  0x50000
-- 
2.29.2

