diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.07/arch/arm/mach-sunxi/Kconfig u-boot-2024.07/arch/arm/mach-sunxi/Kconfig
--- u-boot-2024.07/arch/arm/mach-sunxi/Kconfig	2024-07-22 09:46:42.936733145 +0200
+++ u-boot-2024.07/arch/arm/mach-sunxi/Kconfig	2024-07-22 09:41:33.196724868 +0200
@@ -1125,6 +1125,13 @@
 	  The used address is "bdaddr" if set, and "ethaddr" with the LSB
 	  flipped elsewise.
 
+config PWRLED
+	string "Power led pin"
+	default ""
+	help
+	  Set the pin used to power the led. This takes a string in the format
+	  understood by sunxi_name_to_gpio, e.g. PC12 for pin 1 of port H.
+
 source "board/sunxi/Kconfig"
 
 endif
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.07/board/sunxi/board.c u-boot-2024.07/board/sunxi/board.c
--- u-boot-2024.07/board/sunxi/board.c	2024-07-22 09:46:43.003399812 +0200
+++ u-boot-2024.07/board/sunxi/board.c	2024-07-22 09:46:17.706732454 +0200
@@ -201,7 +201,7 @@
 /* called only from U-Boot proper */
 int board_init(void)
 {
-	__maybe_unused int id_pfr1, ret;
+	__maybe_unused int id_pfr1, ret, pwrled_pin;
 
 	gd->bd->bi_boot_params = (PHYS_SDRAM_0 + 0x100);
 
@@ -238,6 +238,14 @@
 	if (ret)
 		return ret;
 
+	if (CONFIG_PWRLED[0]) {
+		pwrled_pin = sunxi_name_to_gpio(CONFIG_PWRLED);
+		if (pwrled_pin >= 0) {
+			gpio_request(pwrled_pin, "pwrled");
+			gpio_direction_output(pwrled_pin, 1);
+		}
+	}
+
 	eth_init_board();
 
 	return 0;
