diff --speed-large-files --no-dereference --minimal -Naur u-boot-2021.07/drivers/mtd/spi/Kconfig u-boot-2021.07/drivers/mtd/spi/Kconfig
--- u-boot-2021.07/drivers/mtd/spi/Kconfig	2021-07-05 17:11:28.000000000 +0200
+++ u-boot-2021.07/drivers/mtd/spi/Kconfig	2023-11-27 12:18:36.516684574 +0100
@@ -162,6 +162,11 @@
 	  Add support for various XMC (Wuhan Xinxin Semiconductor
 	  Manufacturing Corp.) SPI flash chips (XM25xxx)
 
+config SPI_FLASH_ZBIT
+	bool "ZBIT SPI flash support"
+	help
+	  Add support for Zbit Semiconductor Inc. SPI flash chips (ZB25xxx).
+
 endif
 
 config SPI_FLASH_USE_4K_SECTORS
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2021.07/drivers/mtd/spi/spi-nor-ids.c u-boot-2021.07/drivers/mtd/spi/spi-nor-ids.c
--- u-boot-2021.07/drivers/mtd/spi/spi-nor-ids.c	2021-07-05 17:11:28.000000000 +0200
+++ u-boot-2021.07/drivers/mtd/spi/spi-nor-ids.c	2023-11-27 12:22:34.340018712 +0100
@@ -337,5 +337,10 @@
 	{ INFO("XM25QH64A", 0x207017, 0, 64 * 1024, 128, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ INFO("XM25QH128A", 0x207018, 0, 64 * 1024, 256, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 #endif
+#ifdef CONFIG_SPI_FLASH_ZBIT
+	/* Zbit Semiconductor Inc. */
+	{ INFO("zb25vq128", 0x5e4018, 0, 64 * 1024, 256,
+	       SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+#endif
 	{ },
 };
