diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.01/drivers/mmc/sunxi_mmc.c u-boot-2024.01/drivers/mmc/sunxi_mmc.c
--- u-boot-2024.01/drivers/mmc/sunxi_mmc.c	2024-01-08 16:37:48.000000000 +0100
+++ u-boot-2024.01/drivers/mmc/sunxi_mmc.c	2024-01-29 22:15:17.266567968 +0100
@@ -542,7 +542,7 @@
 	cfg->b_max = CONFIG_SYS_MMC_MAX_BLK_COUNT;
 
 	cfg->f_min = 400000;
-	cfg->f_max = 52000000;
+	cfg->f_max = 12000000;
 
 	if (mmc_resource_init(sdc_no) != 0)
 		return NULL;
