diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.01/dts/upstream/src/arm64/allwinner/sun50i-h616.dtsi u-boot-2024.01/dts/upstream/src/arm64/allwinner/sun50i-h616.dtsi
--- u-boot-2024.01/dts/upstream/src/arm64/allwinner/sun50i-h616.dtsi	2024-03-08 10:53:12.984916466 +0100
+++ u-boot-2024.01/dts/upstream/src/arm64/allwinner/sun50i-h616.dtsi	2024-03-08 10:53:02.154916695 +0100
@@ -324,7 +324,7 @@
 			pinctrl-names = "default";
 			pinctrl-0 = <&mmc2_pins>;
 			status = "disabled";
-			max-frequency = <150000000>;
+			max-frequency = <12000000>;
 			cap-sd-highspeed;
 			cap-mmc-highspeed;
 			mmc-ddr-3_3v;
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.01/drivers/mmc/sunxi_mmc.c u-boot-2024.01/drivers/mmc/sunxi_mmc.c
--- u-boot-2024.01/drivers/mmc/sunxi_mmc.c	2024-01-08 16:37:48.000000000 +0100
+++ u-boot-2024.01/drivers/mmc/sunxi_mmc.c	2024-03-08 10:51:44.771585037 +0100
@@ -542,7 +542,7 @@
 	cfg->b_max = CONFIG_SYS_MMC_MAX_BLK_COUNT;
 
 	cfg->f_min = 400000;
-	cfg->f_max = 52000000;
+	cfg->f_max = 12000000;
 
 	if (mmc_resource_init(sdc_no) != 0)
 		return NULL;
