diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.07/arch/arm/mach-sunxi/clock_sun50i_h6.c u-boot-2024.07/arch/arm/mach-sunxi/clock_sun50i_h6.c
--- u-boot-2024.07/arch/arm/mach-sunxi/clock_sun50i_h6.c	2024-07-23 14:55:29.750440941 +0200
+++ u-boot-2024.07/arch/arm/mach-sunxi/clock_sun50i_h6.c	2024-07-23 14:55:11.850440465 +0200
@@ -110,6 +110,7 @@
 	val |= CCM_CPU_AXI_MUX_PLL_CPUX;
 	writel(val, &ccm->cpu_axi_cfg);
 }
+#endif /* CONFIG_SPL_BUILD */
 
 int clock_twi_onoff(int port, int state)
 {
@@ -138,7 +139,6 @@
 
 	return 0;
 }
-#endif /* CONFIG_SPL_BUILD */
 
 /* PLL_PERIPH0 clock, used by the MMC driver */
 unsigned int clock_get_pll6(void)
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.07/board/sunxi/board.c u-boot-2024.07/board/sunxi/board.c
--- u-boot-2024.07/board/sunxi/board.c	2024-07-23 14:55:29.750440941 +0200
+++ u-boot-2024.07/board/sunxi/board.c	2024-07-23 14:53:01.693770306 +0200
@@ -246,6 +246,13 @@
 		}
 	}
 
+#if CONFIG_IS_ENABLED(DM_I2C)
+	/*
+	 * Temporary workaround for enabling I2C clocks until proper sunxi DM
+	 * clk, reset and pinctrl drivers land.
+	 */
+	i2c_init_board();
+#endif
 	eth_init_board();
 
 	return 0;
