diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.01/arch/arm/dts/sun50i-h618-orangepi-zero3.dts u-boot-2024.01/arch/arm/dts/sun50i-h618-orangepi-zero3.dts
--- u-boot-2024.01/arch/arm/dts/sun50i-h618-orangepi-zero3.dts	2024-01-08 16:37:48.000000000 +0100
+++ u-boot-2024.01/arch/arm/dts/sun50i-h618-orangepi-zero3.dts	2024-01-12 11:04:13.496525419 +0100
@@ -5,11 +5,37 @@
 
 /dts-v1/;
 
+#include <dt-bindings/leds/common.h>
 #include "sun50i-h616-orangepi-zero.dtsi"
 
 / {
 	model = "OrangePi Zero3";
-	compatible = "xunlong,orangepi-zero3", "allwinner,sun50i-h618";
+	compatible = "xunlong,orangepi-zero3", "allwinner,sun50i-h616";
+
+	aliases {
+		ethernet0 = &emac0;
+		serial0 = &uart0;
+		mmc0 = &mmc0;  /* mmc0/mmcblk0 for SD */
+		mmc1 = &mmc1;  /* mmc1/mmcblk1 for SDIO */
+	};
+
+	leds: leds {
+		compatible = "gpio-leds";
+
+		led-red {
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			label = "red_led";
+			gpios = <&pio 2 12 GPIO_ACTIVE_HIGH>; /* PC12 */
+			linux,default-trigger = "mmc0";
+		};
+
+		led-green {
+			function = LED_FUNCTION_STATUS;
+			label = "green_led";
+			gpios = <&pio 2 13 GPIO_ACTIVE_HIGH>; /* PC13 */
+			linux,default-trigger = "heartbeat";
+		};
+	};
 };
 
 &emac0 {
@@ -94,3 +120,70 @@
 	vcc-ph-supply = <&reg_dldo1>;
 	vcc-pi-supply = <&reg_dldo1>;
 };
+
+&spi0  {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi0_pins>, <&spi0_cs0_pin>;
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
+	};
+};
+
+&usbotg {
+	/*
+	 * PHY0 pins are connected to a USB-C socket, but a role switch
+	 * is not implemented: both CC pins are pulled to GND.
+	 * The VBUS pins power the device, so a fixed peripheral mode
+	 * is the best choice.
+	 * The board can be powered via GPIOs, in this case port0 *can*
+	 * act as a host (with a cable/adapter ignoring CC), as VBUS is
+	 * then provided by the GPIOs. Any user of this setup would
+	 * need to adjust the DT accordingly: dr_mode set to "host",
+	 * enabling OHCI0 and EHCI0.
+	 */
+	dr_mode = "peripheral";
+	status = "okay";
+};
+
+&usbphy {
+	usb1_vbus-supply = <&reg_usb1_vbus>;
+	status = "okay";
+};
+
+&ehci0 {
+	status = "disabled";
+};
+
+&ehci1 {
+	status = "okay";
+};
+
+&ehci2 {
+	status = "okay";
+};
+
+&ehci3 {
+	status = "okay";
+};
+
+&ohci0 {
+	status = "disabled";
+};
+
+&ohci1 {
+	status = "okay";
+};
+
+&ohci2 {
+	status = "okay";
+};
+
+&ohci3 {
+	status = "okay";
+};

diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.01-rc6/configs/orangepi_zero3_defconfig u-boot-2024.01-rc6/configs/orangepi_zero3_defconfig
--- u-boot-2024.01-rc6/configs/orangepi_zero3_defconfig	2024-01-03 15:11:33.000000000 +0100
+++ u-boot-2024.01-rc6/configs/orangepi_zero3_defconfig	2024-01-06 14:05:05.134564679 +0100
@@ -6,18 +6,19 @@
 CONFIG_DRAM_SUN50I_H616_DX_DRI=0x0e0e0e0e
 CONFIG_DRAM_SUN50I_H616_CA_DRI=0x0e0e
 CONFIG_DRAM_SUN50I_H616_ODT_EN=0xaaaaeeee
+CONFIG_DRAM_SUN50I_H616_TPR0=0x0
 CONFIG_DRAM_SUN50I_H616_TPR6=0x44000000
-CONFIG_DRAM_SUN50I_H616_TPR10=0x402f6663
-CONFIG_DRAM_SUN50I_H616_TPR11=0x24242624
-CONFIG_DRAM_SUN50I_H616_TPR12=0x0f0f100f
+CONFIG_DRAM_SUN50I_H616_TPR10=0x402f0663
+CONFIG_DRAM_SUN50I_H616_TPR11=0x24242323
+CONFIG_DRAM_SUN50I_H616_TPR12=0x0e0e0e0e
 CONFIG_MACH_SUN50I_H616=y
 CONFIG_SUNXI_DRAM_H616_LPDDR4=y
 CONFIG_DRAM_CLK=792
-CONFIG_USB1_VBUS_PIN="PC16"
 CONFIG_R_I2C_ENABLE=y
 CONFIG_SPL_SPI_SUNXI=y
-# CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
+CONFIG_SYS_MONITOR_LEN=786432
 CONFIG_SPL_I2C=y
+CONFIG_SPL_I2C_SUPPORT=y
 CONFIG_SPL_SYS_I2C_LEGACY=y
 CONFIG_SYS_I2C_MVTWSI=y
 CONFIG_SYS_I2C_SLAVE=0x7f
@@ -25,8 +26,12 @@
 CONFIG_SPI_FLASH_ZBIT=y
 CONFIG_PHY_MOTORCOMM=y
 CONFIG_SUN8I_EMAC=y
-CONFIG_AXP313_POWER=y
 CONFIG_SPI=y
+CONFIG_AXP313_POWER=y
+CONFIG_I2C3_ENABLE=y
+CONFIG_AXP_DCDC3_VOLT=1150
+CONFIG_USB1_VBUS_PIN="PC16"
+CONFIG_PWRLED="PC13"
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_OHCI_HCD=y
 CONFIG_USB_MUSB_GADGET=y
