diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.04/arch/arm/dts/sun50i-h618-orangepi-zero2w.dts u-boot-2024.04/arch/arm/dts/sun50i-h618-orangepi-zero2w.dts
--- u-boot-2024.04/arch/arm/dts/sun50i-h618-orangepi-zero2w.dts	2024-04-02 12:58:58.000000000 +0200
+++ u-boot-2024.04/arch/arm/dts/sun50i-h618-orangepi-zero2w.dts	2024-03-25 10:19:12.671635927 +0100
@@ -12,24 +12,35 @@
 #include <dt-bindings/leds/common.h>
 
 / {
-	model = "OrangePi Zero 2W";
-	compatible = "xunlong,orangepi-zero2w", "allwinner,sun50i-h618";
+	model = "OrangePi Zero2W";
+	compatible = "xunlong,orangepi-zero2w", "allwinner,sun50i-h616";
 
 	aliases {
+		ethernet0 = &emac1;
 		serial0 = &uart0;
+		mmc0 = &mmc0;  /* mmc0/mmcblk0 for SD */
+		mmc1 = &mmc1;  /* mmc1/mmcblk1 for SDIO */
 	};
 
 	chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
-	leds {
+	leds: leds {
 		compatible = "gpio-leds";
 
-		led-0 {
+		led-red {
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			label = "red_led";
+			gpios = <&pio 2 12 GPIO_ACTIVE_HIGH>; /* PC12 */
+			linux,default-trigger = "mmc0";
+		};
+
+		led-green {
 			function = LED_FUNCTION_STATUS;
-			color = <LED_COLOR_ID_GREEN>;
+			label = "green_led";
 			gpios = <&pio 2 13 GPIO_ACTIVE_HIGH>; /* PC13 */
+			linux,default-trigger = "heartbeat";
 		};
 	};
 
@@ -42,98 +53,113 @@
 		regulator-always-on;
 	};
 
-	reg_vcc3v3: vcc3v3 {
-		/* SY8089 DC/DC converter */
+	reg_usb1_vbus: usb1-vbus {
 		compatible = "regulator-fixed";
-		regulator-name = "vcc-3v3";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
+		regulator-name = "usb1-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
 		vin-supply = <&reg_vcc5v>;
-		regulator-always-on;
+		enable-active-high;
+		gpio = <&pio 2 16 GPIO_ACTIVE_HIGH>; /* PC16 */
+		status = "okay";
 	};
 };
 
-&ehci1 {
+&cpu0 {
+	cpu-supply = <&reg_dcdc2>;
 	status = "okay";
 };
 
-/* USB 2 & 3 are on the FPC connector (or the exansion board) */
-
-&mmc0 {
-	cd-gpios = <&pio 5 6 GPIO_ACTIVE_LOW>;	/* PF6 */
-	bus-width = <4>;
-	vmmc-supply = <&reg_vcc3v3>;
+&emac1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rmii_pins>;
+	phy-mode = "rmii";
+	phy-handle = <&rmii_phy>;
+	phy-supply = <&reg_dldo1>;
+	allwinner,rx-delay-ps = <3100>;
+	allwinner,tx-delay-ps = <700>;
 	status = "okay";
 };
 
-&ohci1 {
-	status = "okay";
+&mdio1 {
+	rmii_phy: ethernet-phy@16 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <16>;
+	};
 };
 
-&pio {
-	vcc-pc-supply = <&reg_dldo1>;
-	vcc-pf-supply = <&reg_dldo1>;	/* internally via VCC-IO */
-	vcc-pg-supply = <&reg_aldo1>;
-	vcc-ph-supply = <&reg_dldo1>;	/* internally via VCC-IO */
-	vcc-pi-supply = <&reg_dldo1>;
+&mmc0 {
+	vmmc-supply = <&reg_dldo1>;
+	cd-gpios = <&pio 5 6 GPIO_ACTIVE_LOW>;	/* PF6 */
+	bus-width = <4>;
+	max-frequency = <50000000>;
+	status = "okay";
 };
 
 &r_i2c {
 	status = "okay";
 
-	axp313: pmic@36 {
+	axp313a: pmic@36 {
 		compatible = "x-powers,axp313a";
 		reg = <0x36>;
-		#interrupt-cells = <1>;
-		interrupt-controller;
-		interrupt-parent = <&pio>;
-		interrupts = <2 9 IRQ_TYPE_LEVEL_LOW>;	/* PC9 */
-
-		vin1-supply = <&reg_vcc5v>;
-		vin2-supply = <&reg_vcc5v>;
-		vin3-supply = <&reg_vcc5v>;
+		wakeup-source;
 
-		regulators {
-			/* Supplies VCC-PLL and DRAM */
-			reg_aldo1: aldo1 {
+		regulators{
+			reg_dcdc1: dcdc1 {
+				regulator-name = "axp313a-dcdc1";
+				regulator-min-microvolt = <500000>;
+				regulator-max-microvolt = <1100000>;
+				regulator-step-delay-us = <25>;
+				regulator-final-delay-us = <50>;
 				regulator-always-on;
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <1800000>;
-				regulator-name = "vcc1v8";
 			};
 
-			/* Supplies VCC-IO, so needs to be always on. */
-			reg_dldo1: dldo1 {
+			reg_dcdc2: dcdc2 {
+				regulator-name = "axp313a-dcdc2";
+				regulator-min-microvolt = <500000>;
+				regulator-max-microvolt = <1200000>;
+				regulator-step-delay-us = <25>;
+				regulator-final-delay-us = <50>;
+				regulator-ramp-delay = <200>;
 				regulator-always-on;
-				regulator-min-microvolt = <3300000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-name = "vcc3v3";
 			};
 
-			reg_dcdc1: dcdc1 {
+			reg_dcdc3: dcdc3 {
+				regulator-name = "axp313a-dcdc3";
+				regulator-min-microvolt = <1150000>;
+				regulator-max-microvolt = <1150000>;
+				regulator-step-delay-us = <25>;
+				regulator-final-delay-us = <50>;
 				regulator-always-on;
-				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <990000>;
-				regulator-name = "vdd-gpu-sys";
 			};
 
-			reg_dcdc2: dcdc2 {
+			reg_aldo1: ldo1 {
+				regulator-name = "axp313a-aldo1";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-step-delay-us = <25>;
+				regulator-final-delay-us = <50>;
 				regulator-always-on;
-				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1100000>;
-				regulator-name = "vdd-cpu";
 			};
 
-			reg_dcdc3: dcdc3 {
+			reg_dldo1: ldo2 {
+				regulator-name = "axp313a-dldo1";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-step-delay-us = <25>;
+				regulator-final-delay-us = <50>;
 				regulator-always-on;
-				regulator-min-microvolt = <1100000>;
-				regulator-max-microvolt = <1100000>;
-				regulator-name = "vdd-dram";
 			};
 		};
 	};
 };
 
+&uart0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_ph_pins>;
+	status = "okay";
+};
+
 &spi0  {
 	status = "okay";
 	pinctrl-names = "default";
@@ -148,12 +174,6 @@
 	};
 };
 
-&uart0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart0_ph_pins>;
-	status = "okay";
-};
-
 &usbotg {
 	/*
 	 * PHY0 pins are connected to a USB-C socket, but a role switch
@@ -171,6 +191,38 @@
 };
 
 &usbphy {
-	usb1_vbus-supply = <&reg_vcc5v>;
+	usb1_vbus-supply = <&reg_usb1_vbus>;
+	status = "okay";
+};
+
+&ehci0 {
+	status = "disabled";
+};
+
+&ehci1 {
+	status = "okay";
+};
+
+&ehci2 {
+	status = "okay";
+};
+
+&ehci3 {
+	status = "okay";
+};
+
+&ohci0 {
+	status = "disabled";
+};
+
+&ohci1 {
+	status = "okay";
+};
+
+&ohci2 {
+	status = "okay";
+};
+
+&ohci3 {
 	status = "okay";
 };
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.04/configs/orangepi_zero2w_defconfig u-boot-2024.04/configs/orangepi_zero2w_defconfig
--- u-boot-2024.04/configs/orangepi_zero2w_defconfig	2024-04-02 12:58:58.000000000 +0200
+++ u-boot-2024.04/configs/orangepi_zero2w_defconfig	2024-03-25 10:19:12.674969259 +0100
@@ -6,25 +6,32 @@
 CONFIG_DRAM_SUN50I_H616_DX_DRI=0x0e0e0e0e
 CONFIG_DRAM_SUN50I_H616_CA_DRI=0x0e0e
 CONFIG_DRAM_SUN50I_H616_ODT_EN=0xaaaaeeee
-CONFIG_DRAM_SUN50I_H616_TPR6=0x48808080
-CONFIG_DRAM_SUN50I_H616_TPR10=0x402f6663
-CONFIG_DRAM_SUN50I_H616_TPR11=0x26262524
-CONFIG_DRAM_SUN50I_H616_TPR12=0x100f100f
+CONFIG_DRAM_SUN50I_H616_TPR0=0x0
+CONFIG_DRAM_SUN50I_H616_TPR6=0x44000000
+CONFIG_DRAM_SUN50I_H616_TPR10=0x402f0663
+CONFIG_DRAM_SUN50I_H616_TPR11=0x24242323
+CONFIG_DRAM_SUN50I_H616_TPR12=0x0e0e0e0e
 CONFIG_MACH_SUN50I_H616=y
 CONFIG_SUNXI_DRAM_H616_LPDDR4=y
 CONFIG_DRAM_CLK=792
 CONFIG_R_I2C_ENABLE=y
 CONFIG_SPL_SPI_SUNXI=y
-# CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
+CONFIG_SYS_MONITOR_LEN=786432
 CONFIG_SPL_I2C=y
+CONFIG_SPL_I2C_SUPPORT=y
 CONFIG_SPL_SYS_I2C_LEGACY=y
 CONFIG_SYS_I2C_MVTWSI=y
 CONFIG_SYS_I2C_SLAVE=0x7f
 CONFIG_SYS_I2C_SPEED=400000
-CONFIG_MTD=y
 CONFIG_SPI_FLASH_ZBIT=y
-CONFIG_AXP313_POWER=y
+CONFIG_PHY_REALTEK=y
+CONFIG_SUN8I_EMAC=y
 CONFIG_SPI=y
+CONFIG_AXP313_POWER=y
+CONFIG_I2C3_ENABLE=y
+CONFIG_AXP_DCDC3_VOLT=1150
+CONFIG_USB1_VBUS_PIN="PC16"
+CONFIG_PWRLED="PC13"
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_OHCI_HCD=y
 CONFIG_USB_MUSB_GADGET=y
