diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.04/arch/arm/dts/Makefile u-boot-2024.04/arch/arm/dts/Makefile
--- u-boot-2024.04/arch/arm/dts/Makefile	2024-05-17 18:23:17.766202699 +0200
+++ u-boot-2024.04/arch/arm/dts/Makefile	2024-05-17 18:19:56.509535363 +0200
@@ -845,6 +845,7 @@
 	sun50i-h616-x96-mate.dtb \
 	sun50i-h313-x96q-lpddr3.dtb \
 	sun50i-h313-x96q-ddr3.dtb \
+	sun50i-h313-x96q-ddr3-v5.1.dtb \
 	sun50i-h313-x96q-lpddr3-v1.3.dtb \
 	sun50i-h313-tanix-tx1.dtb \
 	sun50i-h616-tanix-tx6s.dtb \
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.04/arch/arm/dts/sun50i-h313-x96q-ddr3-v5.1.dts u-boot-2024.04/arch/arm/dts/sun50i-h313-x96q-ddr3-v5.1.dts
--- u-boot-2024.04/arch/arm/dts/sun50i-h313-x96q-ddr3-v5.1.dts	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-2024.04/arch/arm/dts/sun50i-h313-x96q-ddr3-v5.1.dts	2024-05-17 18:21:13.469535621 +0200
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: (GPL-2.0+ or MIT)
+/*
+ * Author: piotr.oniszczuk@gmail.com
+ */
+
+/dts-v1/;
+
+#include "sun50i-h616.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	model = "hechuang,x96-q DDR3 v5.1";
+	compatible = "hechuang,x96-q", "allwinner,sun50i-h616";
+
+	aliases {
+		mmc0 = &mmc0;
+		mmc1 = &mmc1;
+		mmc2 = &mmc2;
+		ethernet0 = &emac1;
+		serial0 = &uart0;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led-red {
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			color = <LED_COLOR_ID_RED>;
+			gpios = <&pio 7 6 GPIO_ACTIVE_HIGH>; /* PH6 */
+			linux,default-trigger = "mmc0";
+		};
+	};
+
+	reg_vcc5v: vcc5v {
+		/* board wide 5V supply directly from the USB-C socket */
+		compatible = "regulator-fixed";
+		regulator-name = "vcc-5v";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		regulator-always-on;
+	};
+};
+
+&cpu0 {
+	cpu-supply = <&reg_dcdc2>;
+	status = "okay";
+};
+
+&cpu1 {
+	cpu-supply = <&reg_dcdc2>;
+	status = "okay";
+};
+
+&cpu2 {
+	cpu-supply = <&reg_dcdc2>;
+	status = "okay";
+};
+
+&cpu3 {
+	cpu-supply = <&reg_dcdc2>;
+	status = "okay";
+};
+
+&emac1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rmii_pins>;
+	phy-mode = "rmii";
+	phy-handle = <&rmii_phy>;
+	phy-supply = <&reg_aldo1>;
+	allwinner,rx-delay-ps = <3100>;
+	allwinner,tx-delay-ps = <700>;
+	status = "okay";
+};
+
+&mdio1 {
+	rmii_phy: ethernet-phy@0 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <0>;
+	};
+};
+
+&mmc0 {
+	vmmc-supply = <&reg_dldo2>;
+	broken-cd;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&mmc1 {
+	vmmc-supply = <&reg_dldo2>;
+	broken-cd;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&mmc2 {
+	vmmc-supply = <&reg_dldo2>;
+	bus-width = <8>;
+	non-removable;
+	cap-mmc-hw-reset;
+	status = "okay";
+};
+
+&r_i2c {
+	status = "okay";
+
+	axp313a: pmic@36 {
+		compatible = "x-powers,axp313a";
+		reg = <0x36>;
+		wakeup-source;
+
+		regulators {
+			reg_dcdc1: dcdc1 {
+				regulator-always-on;
+				regulator-min-microvolt = <1160000>;
+				regulator-max-microvolt = <1160000>;
+				regulator-name = "vdd-cpu";
+			};
+
+			reg_dcdc2: dcdc2 {
+				regulator-always-on;
+				regulator-min-microvolt = <1160000>;
+				regulator-max-microvolt = <1160000>;
+				regulator-name = "vdd-gpu-sys";
+			};
+
+			reg_dcdc3: dcdc3 {
+				regulator-always-on;
+				regulator-min-microvolt = <1500000>;
+				regulator-max-microvolt = <1500000>;
+				regulator-name = "vdd-dram";
+			};
+
+			reg_aldo1: ldo1 {
+				regulator-always-on;
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-name = "vcc-sys";
+			};
+
+			reg_dldo2: ldo2 {
+				regulator-always-on;
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-name = "vcc3v3-ext";
+			};
+		};
+	};
+};
+
+&uart0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_ph_pins>;
+	status = "okay";
+};
diff --speed-large-files --no-dereference --minimal -Naur u-boot-2024.04/configs/x96_q_ddr3_v5.1_defconfig u-boot-2024.04/configs/x96_q_ddr3_v5.1_defconfig
--- u-boot-2024.04/configs/x96_q_ddr3_v5.1_defconfig	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-2024.04/configs/x96_q_ddr3_v5.1_defconfig	2024-05-17 18:22:33.406202553 +0200
@@ -0,0 +1,38 @@
+CONFIG_ARM=y
+CONFIG_ARCH_SUNXI=y
+CONFIG_DEFAULT_DEVICE_TREE="sun50i-h313-x96q-ddr3-v5.1"
+CONFIG_SPL=y
+CONFIG_SUNXI_DRAM_H616_DDR3_1333=y
+CONFIG_DRAM_CLK=600
+CONFIG_DRAM_ODT_EN=y
+CONFIG_DRAM_SUN50I_H616_UNKNOWN_FEATURE=y
+CONFIG_DRAM_SUN50I_H616_BIT_DELAY_COMPENSATION=y
+CONFIG_DRAM_SUN50I_H616_READ_CALIBRATION=y
+CONFIG_DRAM_SUN50I_H616_DX_ODT=0x03030303
+CONFIG_DRAM_SUN50I_H616_DX_DRI=0x0e0e0e0e
+CONFIG_DRAM_SUN50I_H616_CA_DRI=0x00001c12
+CONFIG_DRAM_SUN50I_H616_ODT_EN=0x1
+CONFIG_DRAM_SUN50I_H616_TPR0=0xc0000a05
+CONFIG_DRAM_SUN50I_H616_TPR2=0x00000000
+CONFIG_DRAM_SUN50I_H616_TPR10=0x002f0006
+CONFIG_DRAM_SUN50I_H616_TPR11=0xddddcccc
+CONFIG_DRAM_SUN50I_H616_TPR12=0xeddc7564
+CONFIG_MACH_SUN50I_H616=y
+CONFIG_R_I2C_ENABLE=y
+CONFIG_SPL_I2C=y
+CONFIG_SPL_I2C_SUPPORT=y
+CONFIG_SPL_SYS_I2C_LEGACY=y
+CONFIG_SYS_I2C_MVTWSI=y
+CONFIG_SYS_I2C_SLAVE=0x7f
+CONFIG_SYS_I2C_SPEED=100000
+CONFIG_PHY_REALTEK=y
+CONFIG_SUN8I_EMAC=y
+CONFIG_I2C3_ENABLE=y
+CONFIG_AXP313_POWER=y
+# CONFIG_SUNXI_RVBAR_ALTERNATIVE=0x81000040
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_OHCI_HCD=y
+CONFIG_USB_MUSB_GADGET=y
+CONFIG_SUPPORT_EMMC_BOOT=y
+CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR=0x40
+CONFIG_MMC_SUNXI_SLOT_EXTRA=2
