diff --speed-large-files --no-dereference --minimal -Naur u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/board.c u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/board.c
--- u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/board.c	2022-12-12 18:51:28.000000000 +0100
+++ u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/board.c	2023-03-25 13:42:03.408290291 +0100
@@ -9,6 +9,8 @@
  * Some init for sunxi platform.
  */
 
+#define DEBUG
+
 #include <common.h>
 #include <cpu_func.h>
 #include <init.h>
diff --speed-large-files --no-dereference --minimal -Naur u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c
--- u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c	2022-12-12 18:51:28.000000000 +0100
+++ u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/mach-sunxi/dram_sun50i_h616.c	2023-03-25 13:41:44.954957651 +0100
@@ -12,6 +12,9 @@
  * (C) Copyright 2020 Jernej Skrabec <jernej.skrabec@siol.net>
  *
  */
+
+#define DEBUG 1
+
 #include <common.h>
 #include <init.h>
 #include <log.h>
diff --speed-large-files --no-dereference --minimal -Naur u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/board/sunxi/board.c u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/board/sunxi/board.c
--- u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/board/sunxi/board.c	2022-12-12 18:51:28.000000000 +0100
+++ u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/board/sunxi/board.c	2023-03-25 13:42:03.408290291 +0100
@@ -10,6 +10,8 @@
  * Some board init for the Allwinner A10-evb board.
  */
 
+#define DEBUG
+
 #include <common.h>
 #include <clock_legacy.h>
 #include <dm.h>
