diff -Naur linux-3.12.7-old/arch/x86/boot/Makefile linux-3.12.7-new/arch/x86/boot/Makefile
--- linux-3.12.7-old/arch/x86/boot/Makefile	2014-01-09 21:25:44.000000000 +0100
+++ linux-3.12.7-new/arch/x86/boot/Makefile	2014-01-15 08:55:18.018897619 +0100
@@ -14,7 +14,7 @@
 # Set it to -DSVGA_MODE=NORMAL_VGA if you just want the EGA/VGA mode.
 # The number is the same as you would ordinarily press at bootup.
 
-SVGA_MODE	:= -DSVGA_MODE=NORMAL_VGA
+SVGA_MODE	:= -DSVGA_MODE=786
 
 targets		:= vmlinux.bin setup.bin setup.elf bzImage
 targets		+= fdimage fdimage144 fdimage288 image.iso mtools.conf
diff -Naur linux-3.12.7-old/arch/x86/boot/video.c linux-3.12.7-new/arch/x86/boot/video.c
--- linux-3.12.7-old/arch/x86/boot/video.c	2014-01-09 21:25:44.000000000 +0100
+++ linux-3.12.7-new/arch/x86/boot/video.c	2014-01-15 11:42:31.920069528 +0100
@@ -322,17 +322,8 @@
 	save_screen();
 	probe_cards(0);
 
-	for (;;) {
-		if (mode == ASK_VGA)
-			mode = mode_menu();
+	set_mode(mode);
 
-		if (!set_mode(mode))
-			break;
-
-		printf("Undefined video mode number: %x\n", mode);
-		mode = ASK_VGA;
-	}
-	boot_params.hdr.vid_mode = mode;
 	vesa_store_edid();
 	store_mode_params();
 
diff -Naur linux-3.12.7-old/kernel/printk/printk.c linux-3.12.7-new/kernel/printk/printk.c
--- linux-3.12.7-old/kernel/printk/printk.c	2014-01-09 21:25:44.000000000 +0100
+++ linux-3.12.7-new/kernel/printk/printk.c	2014-01-15 08:55:18.022231362 +0100
@@ -59,7 +59,7 @@
 
 /* We show everything that is MORE important than this.. */
 #define MINIMUM_CONSOLE_LOGLEVEL 1 /* Minimum loglevel we let people use */
-#define DEFAULT_CONSOLE_LOGLEVEL 7 /* anything MORE serious than KERN_DEBUG */
+#define DEFAULT_CONSOLE_LOGLEVEL 3 /* anything MORE serious than KERN_ERR */
 
 int console_printk[4] = {
 	DEFAULT_CONSOLE_LOGLEVEL,	/* console_loglevel */
