diff --speed-large-files --no-dereference --minimal -Naur gettext-1.0/gettext-tools/man/Makefile.am gettext-1.0/gettext-tools/man/Makefile.am
--- gettext-1.0/gettext-tools/man/Makefile.am	2025-12-31 00:11:05.000000000 +0100
+++ gettext-1.0/gettext-tools/man/Makefile.am	2026-06-25 10:59:00.183790423 +0200
@@ -146,7 +146,7 @@
 	progname=`echo $@ | sed -e 's/\.in$$//' -e 's/\.1$$//'`; \
 	IN_HELP2MAN=1 $(SHELL) x-to-1 $(UPDATEMODE) "$(PERL)" "$(HELP2MAN)" \
 	  ../misc/$${progname} $(srcdir)/$${progname}.x $@
-	patch $@ < $(srcdir)/po-fetch.1.diff
+#	patch $@ < $(srcdir)/po-fetch.1.diff
 
 po-fetch.1: po-fetch.x ../misc/po-fetch.in po-fetch.1.diff
 
