diff -Naur gettext-0.19.4-old/gettext-tools/misc/gettextize.in gettext-0.19.4-new/gettext-tools/misc/gettextize.in
--- gettext-0.19.4-old/gettext-tools/misc/gettextize.in	2014-12-09 00:43:46.000000000 -0800
+++ gettext-0.19.4-new/gettext-tools/misc/gettextize.in	2015-07-01 23:16:31.050000416 -0700
@@ -1334,10 +1334,10 @@
     5) count="five paragraphs";;
     *) count="$count paragraphs";;
   esac
-  echo "Press Return to acknowledge the previous $count."
-  # Read from /dev/tty, not stdin, so that gettextize cannot be abused by
-  # non-interactive tools.
-  read dummy < /dev/tty
+#  echo "Press Return to acknowledge the previous $count."
+#  # Read from /dev/tty, not stdin, so that gettextize cannot be abused by
+#  # non-interactive tools.
+#  read dummy < /dev/tty
 fi
 
 exit 0
