diff -Naur gettext-0.18.2-old/gettext-tools/misc/gettextize.in gettext-0.18.2-new/gettext-tools/misc/gettextize.in
--- gettext-0.18.2-old/gettext-tools/misc/gettextize.in	2012-12-18 19:56:12.000000000 -0800
+++ gettext-0.18.2-new/gettext-tools/misc/gettextize.in	2012-12-28 10:07:03.000000000 -0800
@@ -1272,10 +1272,10 @@
     5) count="five paragraphs";;
     *) count="$count paragraphs";;
   esac
-  echo "Press Return to acknowledge the previous $count."
-  # Read from /dev/tty, not stdin, so that gettextize cannot be abused by
-  # non-interactive tools.
-  read dummy < /dev/tty
+#  echo "Press Return to acknowledge the previous $count."
+#  # Read from /dev/tty, not stdin, so that gettextize cannot be abused by
+#  # non-interactive tools.
+#  read dummy < /dev/tty
 fi
 
 exit 0
