
this reverts:

ee48e81b26e4293e9dae9834b892839dd2f236fa Mon Sep 17 00:00:00 2001
From: Ella Stanforth <ella@igalia.com>
Date: Tue, 8 Jul 2025 21:58:27 +0100
Subject: [PATCH] v3d: Always lower frag color


diff --speed-large-files --no-dereference --minimal -Naur mesa-40f7bef16c44b2a8a2283df20204c41297989ecd/src/broadcom/compiler/nir_to_vir.c mesa-40f7bef16c44b2a8a2283df20204c41297989ecd/src/broadcom/compiler/nir_to_vir.c
--- mesa-40f7bef16c44b2a8a2283df20204c41297989ecd/src/broadcom/compiler/nir_to_vir.c	2025-10-16 05:56:25.000000000 +0200
+++ mesa-40f7bef16c44b2a8a2283df20204c41297989ecd/src/broadcom/compiler/nir_to_vir.c	2025-10-19 13:11:53.676738622 +0200
@@ -2472,7 +2472,8 @@
 
                 switch (var->data.location) {
                 case FRAG_RESULT_COLOR:
-                        UNREACHABLE("Frag color should be lowered");
+                        for (int i = 0; i < V3D_MAX_DRAW_BUFFERS; i++)
+                                c->output_color_var[i] = var;
                         break;
                 case FRAG_RESULT_DATA0:
                 case FRAG_RESULT_DATA1:
diff --speed-large-files --no-dereference --minimal -Naur mesa-40f7bef16c44b2a8a2283df20204c41297989ecd/src/gallium/drivers/v3d/v3d_program.c mesa-40f7bef16c44b2a8a2283df20204c41297989ecd/src/gallium/drivers/v3d/v3d_program.c
--- mesa-40f7bef16c44b2a8a2283df20204c41297989ecd/src/gallium/drivers/v3d/v3d_program.c	2025-10-16 05:56:25.000000000 +0200
+++ mesa-40f7bef16c44b2a8a2283df20204c41297989ecd/src/gallium/drivers/v3d/v3d_program.c	2025-10-19 13:11:53.676738622 +0200
@@ -361,9 +361,11 @@
         if (s->info.stage == MESA_SHADER_FRAGMENT &&
             s->info.outputs_written & BITFIELD_BIT(FRAG_RESULT_COLOR)) {
                 /* We only support one attachment when doing dual source blending. */
-                unsigned max_rb = s->info.fs.color_is_dual_source ?
-                        1 : V3D_MAX_DRAW_BUFFERS;
-                NIR_PASS(_, s, nir_lower_fragcolor, max_rb);
+                if (s->info.fs.color_is_dual_source)
+                        NIR_PASS(_, s, nir_lower_fragcolor, 1);
+                else if (V3D_DBG(SOFT_BLEND))
+                        NIR_PASS(_, s, nir_lower_fragcolor,
+                                 V3D_MAX_DRAW_BUFFERS);
         }
 
         if (s->info.stage != MESA_SHADER_VERTEX &&
