diff -Naur mesa-mesa-19.1.2-old/src/gallium/drivers/v3d/meson.build mesa-mesa-19.1.2-new/src/gallium/drivers/v3d/meson.build
--- mesa-mesa-19.1.2-old/src/gallium/drivers/v3d/meson.build	2019-07-09 11:09:53.000000000 +0200
+++ mesa-mesa-19.1.2-new/src/gallium/drivers/v3d/meson.build	2019-07-10 17:48:42.723333308 +0200
@@ -77,7 +77,7 @@
 
 v3d_neon_c_args = []
 if host_machine.cpu_family() == 'arm'
-    v3d_neon_c_args = '-mfpu=neon'
+    v3d_neon_c_args = ['-mfpu=neon', '-mfloat-abi=softfp']
 endif
 
 libv3d_neon = static_library(
diff -Naur mesa-mesa-19.1.2-old/src/gallium/drivers/vc4/meson.build mesa-mesa-19.1.2-new/src/gallium/drivers/vc4/meson.build
--- mesa-mesa-19.1.2-old/src/gallium/drivers/vc4/meson.build	2019-07-09 11:09:53.000000000 +0200
+++ mesa-mesa-19.1.2-new/src/gallium/drivers/vc4/meson.build	2019-07-10 17:40:02.546666639 +0200
@@ -91,7 +91,7 @@
     include_directories : [
       inc_src, inc_include, inc_gallium, inc_gallium_aux, inc_broadcom
     ],
-    c_args : '-mfpu=neon',
+    c_args : ['-mfpu=neon', '-mfloat-abi=softfp'],
   )
   vc4_c_args += '-DUSE_ARM_ASM'
 endif
