diff -Naur Socket6-0.29-old/Makefile.PL Socket6-0.29-new/Makefile.PL
--- Socket6-0.29-old/Makefile.PL	2016-04-11 05:27:34.000000000 +0200
+++ Socket6-0.29-new/Makefile.PL	2020-01-04 19:33:13.970000000 +0100
@@ -48,7 +48,7 @@
         # Call './configure' as 'sh ./configure'
         $cmd .= " $Config{sh}";
     }
-    $cmd .= " ./configure";
+    $cmd .= " ./configure socket6_cv_pl_sv_undef=\"yes\" ipv6_cv_can_inet_ntop=\"no\" --host ";
     $cmd .= " --with-perl=$path_perl" if ($path_perl);
     system($cmd);
 }
