diff -Naur sysfsutils-2.1.0-old/configure.ac sysfsutils-2.1.0-new/configure.ac
--- sysfsutils-2.1.0-old/configure.ac	2006-08-07 07:08:00.000000000 +0200
+++ sysfsutils-2.1.0-new/configure.ac	2013-09-27 15:51:29.325355938 +0200
@@ -2,7 +2,7 @@
 AC_INIT(sysfsutils, 2.1.0, linux-diag-devel@lists.sourceforge.net)
 AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([config.h.in])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 
 # Checks for KLIBC support (should be before AC_PROG_LIBTOOL and AC_PROG_CC)
 AC_CHECK_KLIBC
