diff -Naur sysfsutils-2.1.0-old/configure.ac sysfsutils-2.1.0-new/configure.ac
--- sysfsutils-2.1.0-old/configure.ac	2006-08-06 22:08:00.000000000 -0700
+++ sysfsutils-2.1.0-new/configure.ac	2010-08-27 10:03:48.000000000 -0700
@@ -7,7 +7,7 @@
 # Checks for KLIBC support (should be before AC_PROG_LIBTOOL and AC_PROG_CC)
 AC_CHECK_KLIBC
 
-if test "X$KLIBC" == Xyes; then
+if test "X$KLIBC" = Xyes; then
 	# Should disable/enable building of shared libraries ... I am open to better
 	# suggestions though (AC_DISABLE_SHARED and co do not do the right thing in
 	# a situation like this that its runtime dependent) ...
@@ -17,7 +17,7 @@
 	# These ones make KLIBC compiles very loud
 	EXTRA_CFLAGS="-Wshadow -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wnested-externs"
 fi
-AM_CONDITIONAL([KLIBC],[test "X$KLIBC" == Xyes])
+AM_CONDITIONAL([KLIBC],[test "X$KLIBC" = Xyes])
 AC_SUBST([EXTRA_CFLAGS])
 
 # Checks for LIBTOOL
