diff --speed-large-files --no-dereference --minimal -Naur htop-3.4.1/linux/LibSensors.c htop-3.4.1/linux/LibSensors.c
--- htop-3.4.1/linux/LibSensors.c	2025-04-11 04:00:29.000000000 +0200
+++ htop-3.4.1/linux/LibSensors.c	2025-04-21 13:36:30.401144982 +0200
@@ -158,6 +158,9 @@
       { "cpu5_thermal",       0 },
       { "cpu6_thermal",       0 },
       { "cpu7_thermal",       0 },
+      /* Allwinner A523 */
+      { "cpu0_thermal",       0 },
+      { "cpu4_thermal",       0 },
       /* Low priority drivers */
       { "acpitz",             1 },
    };
@@ -262,6 +265,29 @@
             continue;
 
          if (existingCPUs == 8) {
+
+            /* Map temperature values to Allwinner a523 cores
+            *
+            *   slow a55 cores  -> cores 1..4
+            *   fast a55 cores  -> cores 5..8
+            */
+            if (String_eq(chip->prefix, "cpu0_thermal")) {
+               data[1] = temp;
+               data[2] = temp;
+               data[3] = temp;
+               data[4] = temp;
+               coreTempCount += 4;
+               continue;
+            }
+            if (String_eq(chip->prefix, "cpu4_thermal")) {
+               data[5] = temp;
+               data[6] = temp;
+               data[7] = temp;
+               data[8] = temp;
+               coreTempCount += 4;
+               continue;
+            }
+
             /* Map temperature values to Snapdragon 8cx cores */
             if (String_startsWith(chip->prefix, "cpu") && chip->prefix[3] >= '0' && chip->prefix[3] <= '7' && String_eq(chip->prefix + 4, "_thermal")) {
                data[1 + chip->prefix[3] - '0'] = temp;
@@ -269,7 +295,7 @@
                continue;
             }
 
-            /* Map temperature values to Rockchip cores
+            /* Map temperature values to Rockchip 3588 cores
              *
              *   littlecore -> cores 1..4
              *   bigcore0   -> cores 5,6
