diff --speed-large-files --no-dereference --minimal -Naur htop-3.4.1/linux/LinuxMachine.c htop-3.4.1/linux/LinuxMachine.c
--- htop-3.4.1/linux/LinuxMachine.c	2025-04-11 04:00:29.000000000 +0200
+++ htop-3.4.1/linux/LinuxMachine.c	2025-06-04 22:00:01.044968728 +0200
@@ -534,7 +534,7 @@
          continue;
 
       char pathBuffer[64];
-      xSnprintf(pathBuffer, sizeof(pathBuffer), "/sys/devices/system/cpu/cpu%u/cpufreq/scaling_cur_freq", i);
+      xSnprintf(pathBuffer, sizeof(pathBuffer), "/sys/devices/system/cpu/cpu%u/cpufreq/cpuinfo_cur_freq", i);
 
       struct timespec start;
       if (i == 0)
