diff -Naur g15daemon-1.9.5.3-old/configure g15daemon-1.9.5.3-new/configure
--- g15daemon-1.9.5.3-old/configure	2008-02-04 22:01:06.000000000 -0800
+++ g15daemon-1.9.5.3-new/configure	2009-09-18 04:34:40.000000000 -0700
@@ -21963,8 +21963,6 @@
 
 CPPFLAGS="$CPPFLAGS "'-DPLUGINDIR=\"$(G15DAEMON_PLUGIN_DIR)\"'
 
-CPPFLAGS="$CPPFLAGS "'-I$(includedir)'
-
 ac_config_files="$ac_config_files Makefile g15daemon/Makefile libg15daemon_client/Makefile plugins/Makefile"
 
 
diff -Naur g15daemon-1.9.5.3-old/configure.in g15daemon-1.9.5.3-new/configure.in
--- g15daemon-1.9.5.3-old/configure.in	2008-01-30 21:18:41.000000000 -0800
+++ g15daemon-1.9.5.3-new/configure.in	2009-09-18 04:34:28.000000000 -0700
@@ -128,8 +128,6 @@
 AC_SUBST(G15DAEMON_PLUGIN_DIR,["$libdir/g15daemon/$PACKAGE_VERSION/plugins"])
 CPPFLAGS="$CPPFLAGS "'-DPLUGINDIR=\"$(G15DAEMON_PLUGIN_DIR)\"'
 
-CPPFLAGS="$CPPFLAGS "'-I$(includedir)'
-
 AC_CONFIG_FILES([Makefile g15daemon/Makefile libg15daemon_client/Makefile plugins/Makefile])
 
 AC_OUTPUT
