diff -Naur alsa-lib-1.2.4-old/src/conf/cards/aliases.conf alsa-lib-1.2.4-new/src/conf/cards/aliases.conf
--- alsa-lib-1.2.4-old/src/conf/cards/aliases.conf	2020-11-14 15:57:31.916666666 +0100
+++ alsa-lib-1.2.4-new/src/conf/cards/aliases.conf	2020-11-14 16:00:13.846666666 +0100
@@ -57,6 +57,8 @@
 CMI8787 cards.CMI8788
 pistachio cards.pistachio-card
 VC4-HDMI cards.vc4-hdmi
+VC4-HDMI-0 cards.vc4-hdmi-0
+VC4-HDMI-1 cards.vc4-hdmi-1
 
 # Amlogic HDMI
 GXBB-NANOPI-K2 cards.AMLOGIC-HDMI
diff -Naur alsa-lib-1.2.4-old/src/conf/cards/Makefile.am alsa-lib-1.2.4-new/src/conf/cards/Makefile.am
--- alsa-lib-1.2.4-old/src/conf/cards/Makefile.am	2020-11-14 15:57:31.916666666 +0100
+++ alsa-lib-1.2.4-new/src/conf/cards/Makefile.am	2020-11-14 16:02:04.916666666 +0100
@@ -54,6 +54,8 @@
 	USB-Audio.conf \
 	YMF744.conf \
 	vc4-hdmi.conf \
+	vc4-hdmi-0.conf \
+	vc4-hdmi-1.conf \
 	VIA686A.conf \
 	VIA8233.conf \
 	VIA8233A.conf \
diff -Naur alsa-lib-1.2.4-old/src/conf/cards/vc4-hdmi-0.conf alsa-lib-1.2.4-new/src/conf/cards/vc4-hdmi-0.conf
--- alsa-lib-1.2.4-old/src/conf/cards/vc4-hdmi-0.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-lib-1.2.4-new/src/conf/cards/vc4-hdmi-0.conf	2020-11-13 17:05:39.218306875 +0100
@@ -0,0 +1,64 @@
+#
+# Configuration for the vc4-hdmi-0 sound card using software IEC958
+# subframe conversion
+#
+
+<confdir:pcm/front.conf>
+
+vc4-hdmi-0.pcm.front.0 {
+	@args [ CARD ]
+	@args.CARD {
+		type string
+	}
+	type hw
+	card $CARD
+}
+
+# default with dmix
+vc4-hdmi-0.pcm.default {
+	@args [ CARD ]
+	@args.CARD {
+		type string
+	}
+	type asym
+	playback.pcm {
+		type plug
+		slave.pcm {
+			@func concat
+			strings [ "dmix:" $CARD ]
+		}
+	}
+}
+
+<confdir:pcm/iec958.conf>
+
+vc4-hdmi-0.pcm.iec958.0 {
+	@args [ CARD AES0 AES1 AES2 AES3 ]
+	@args.CARD {
+		type string
+	}
+	@args.AES0 {
+		type integer
+	}
+	@args.AES1 {
+		type integer
+	}
+	@args.AES2 {
+		type integer
+	}
+	@args.AES3 {
+		type integer
+	}
+	type iec958
+	slave {
+		format IEC958_SUBFRAME_LE
+		pcm {
+			type plug
+			slave.pcm {
+				type hw
+				card $CARD
+			}
+		}
+	}
+	status [ $AES0 $AES1 $AES2 $AES3 ]
+}
diff -Naur alsa-lib-1.2.4-old/src/conf/cards/vc4-hdmi-1.conf alsa-lib-1.2.4-new/src/conf/cards/vc4-hdmi-1.conf
--- alsa-lib-1.2.4-old/src/conf/cards/vc4-hdmi-1.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-lib-1.2.4-new/src/conf/cards/vc4-hdmi-1.conf	2020-11-13 17:05:53.118308256 +0100
@@ -0,0 +1,64 @@
+#
+# Configuration for the vc4-hdmi-1 sound card using software IEC958
+# subframe conversion
+#
+
+<confdir:pcm/front.conf>
+
+vc4-hdmi-1.pcm.front.0 {
+	@args [ CARD ]
+	@args.CARD {
+		type string
+	}
+	type hw
+	card $CARD
+}
+
+# default with dmix
+vc4-hdmi-1.pcm.default {
+	@args [ CARD ]
+	@args.CARD {
+		type string
+	}
+	type asym
+	playback.pcm {
+		type plug
+		slave.pcm {
+			@func concat
+			strings [ "dmix:" $CARD ]
+		}
+	}
+}
+
+<confdir:pcm/iec958.conf>
+
+vc4-hdmi-1.pcm.iec958.0 {
+	@args [ CARD AES0 AES1 AES2 AES3 ]
+	@args.CARD {
+		type string
+	}
+	@args.AES0 {
+		type integer
+	}
+	@args.AES1 {
+		type integer
+	}
+	@args.AES2 {
+		type integer
+	}
+	@args.AES3 {
+		type integer
+	}
+	type iec958
+	slave {
+		format IEC958_SUBFRAME_LE
+		pcm {
+			type plug
+			slave.pcm {
+				type hw
+				card $CARD
+			}
+		}
+	}
+	status [ $AES0 $AES1 $AES2 $AES3 ]
+}
