diff -Naur alsa-lib-1.1.9-old/src/conf/cards/AMLGX.conf alsa-lib-1.1.9-new/src/conf/cards/AMLGX.conf
--- alsa-lib-1.1.9-old/src/conf/cards/AMLGX.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-lib-1.1.9-new/src/conf/cards/AMLGX.conf	2019-10-22 11:58:37.836666668 +0200
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2018-present Team LibreELEC (https://libreelec.tv)
+
+# Configuration for Amlogic GX devices with HDMI
+
+<confdir:pcm/hdmi.conf>
+
+AMLGX.pcm.hdmi.0 {
+	@args [ CARD AES0 AES1 AES2 AES3 ]
+	@args.CARD { type string }
+	@args.AES0 { type integer }
+	@args.AES1 { type integer }
+	@args.AES2 { type integer }
+	@args.AES3 { type integer }
+	type hooks
+	slave.pcm {
+		type hw
+		card $CARD
+		device 0
+	}
+	hooks.0 {
+		type ctl_elems
+		hook_args [
+			{
+				interface MIXER
+				name "IEC958 Playback Default"
+				lock true
+				preserve true
+				optional true
+				value [ $AES0 $AES1 $AES2 $AES3 ]
+			}
+		]
+	}
+	hint.device 0
+}
diff -Naur alsa-lib-1.1.9-old/src/conf/cards/Makefile.am alsa-lib-1.1.9-new/src/conf/cards/Makefile.am
--- alsa-lib-1.1.9-old/src/conf/cards/Makefile.am	2019-05-10 08:57:21.000000000 +0200
+++ alsa-lib-1.1.9-new/src/conf/cards/Makefile.am	2019-10-22 12:03:30.683333335 +0200
@@ -2,6 +2,7 @@
 alsadir = $(alsaconfigdir)/cards
 cfg_files = aliases.conf \
 	AACI.conf \
+	AMLGX.conf \
 	ATIIXP.conf \
 	ATIIXP-SPDMA.conf \
 	ATIIXP-MODEM.conf \
diff -Naur alsa-lib-1.1.9-old/src/conf/cards/Makefile.in alsa-lib-1.1.9-new/src/conf/cards/Makefile.in
--- alsa-lib-1.1.9-old/src/conf/cards/Makefile.in	2019-05-10 09:10:52.000000000 +0200
+++ alsa-lib-1.1.9-new/src/conf/cards/Makefile.in	2019-10-22 12:03:10.320000002 +0200
@@ -286,7 +286,7 @@
 top_srcdir = @top_srcdir@
 alsaconfigdir = @ALSA_CONFIG_DIR@
 alsadir = $(alsaconfigdir)/cards
-cfg_files = aliases.conf AACI.conf ATIIXP.conf ATIIXP-SPDMA.conf \
+cfg_files = aliases.conf AACI.conf AMLGX.conf ATIIXP.conf ATIIXP-SPDMA.conf \
 	ATIIXP-MODEM.conf AU8810.conf AU8820.conf AU8830.conf \
 	Audigy.conf Audigy2.conf Aureon51.conf Aureon71.conf \
 	CA0106.conf CMI8338.conf CMI8338-SWIEC.conf CMI8738-MC6.conf \
