diff -Naur alsa-lib-1.2.2-old/src/conf/cards/aliases.conf alsa-lib-1.2.2-new/src/conf/cards/aliases.conf
--- alsa-lib-1.2.2-old/src/conf/cards/aliases.conf	2020-02-19 10:35:39.000000000 +0100
+++ alsa-lib-1.2.2-new/src/conf/cards/aliases.conf	2020-06-18 20:34:50.266666646 +0200
@@ -58,6 +58,42 @@
 pistachio cards.pistachio-card
 VC4-HDMI cards.vc4-hdmi
 
+# Amlogic HDMI
+GXBB-NANOPI-K2 cards.AMLOGIC-HDMI
+GXBB-ODROID-C2 cards.AMLOGIC-HDMI
+GXBB-NEXBOX-A95X cards.AMLOGIC-HDMI
+GXBB-P201 cards.AMLOGIC-HDMI
+GXBB-WETEK-HUB cards.AMLOGIC-HDMI
+GXL-LIBRETECH-S805X-AC cards.AMLOGIC-HDMI
+GXL-LIBRETECH-S905X-CC cards.AMLOGIC-HDMI
+GXL-LIBRETECH-S9XX-PC cards.AMLOGIC-HDMI
+GXL-S905X-P212 cards.AMLOGIC-HDMI
+GXM-KHADAS-VIM2 cards.AMLOGIC-HDMI
+G12A-SEI510 cards.AMLOGIC-HDMI
+G12B-GTKING-PRO cards.AMLOGIC-HDMI
+G12B-KHADAS-VIM3 cards.AMLOGIC-HDMI
+G12B-ODROID-N2 cards.AMLOGIC-HDMI
+SM1-A95XF3-AIR cards.AMLOGIC-HDMI
+SM1-KHADAS-VIM3L cards.AMLOGIC-HDMI
+SM1-ODROID-C4 cards.AMLOGIC-HDMI
+SM1-SEI610 cards.AMLOGIC-HDMI
+SM1-X96-AIR cards.AMLOGIC-HDMI
+
+# Amlogic HDMI-SPDIF
+GXBB-P200 cards.AMLOGIC-HDMI-SPDIF
+GXBB-VEGA-S95 cards.AMLOGIC-HDMI-SPDIF
+GXBB-WETEK-PLAY2 cards.AMLOGIC-HDMI-SPDIF
+GX-P230-Q200 cards.AMLOGIC-HDMI-SPDIF
+GXM-NEXBOX-A1 cards.AMLOGIC-HDMI-SPDIF
+GXM-RBOX-PRO cards.AMLOGIC-HDMI-SPDIF
+GXM-WETEK-CORE2 cards.AMLOGIC-HDMI-SPDIF
+GXM-BEELINK-GT1 cards.AMLOGIC-HDMI-SPDIF
+G12A-TANIX-TX5MAX cards.AMLOGIC-HDMI-SPDIF
+G12A-U200 cards.AMLOGIC-HDMI-SPDIF
+G12A-X96-MAX cards.AMLOGIC-HDMI-SPDIF
+G12B-GTKING cards.AMLOGIC-HDMI-SPDIF
+G12B-UGOOS-AM6 cards.AMLOGIC-HDMI-SPDIF
+
 <confdir:pcm/default.conf>
 <confdir:pcm/dmix.conf>
 <confdir:pcm/dsnoop.conf>
diff -Naur alsa-lib-1.2.2-old/src/conf/cards/AMLOGIC-HDMI.conf alsa-lib-1.2.2-new/src/conf/cards/AMLOGIC-HDMI.conf
--- alsa-lib-1.2.2-old/src/conf/cards/AMLOGIC-HDMI.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-lib-1.2.2-new/src/conf/cards/AMLOGIC-HDMI.conf	2020-06-18 20:37:03.119999996 +0200
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2020-present Team LibreELEC (https://libreelec.tv)
+
+# NOTE: IEC958 controls are not currently implemented in the audio
+# driver, so control blocks are currently dummies.
+
+<confdir:pcm/hdmi.conf>
+
+AMLOGIC-HDMI.pcm.hdmi.0 {
+	@args [ CARD AES0 AES1 AES2 AES3 ]
+	@args.CARD { type string }
+	@args.AES0 { type integer }
+	@args.AES1 { type integer }
+	@args.AES2 { type integer }
+	@args.AES3 { type integer }
+	type hooks
+	slave.pcm {
+		type hw
+		card $CARD
+		device 0
+	}
+	hooks.0 {
+		type ctl_elems
+		hook_args [
+			{
+				interface MIXER
+				name "IEC958 Playback Default"
+				lock true
+				preserve true
+				optional true
+				value [ $AES0 $AES1 $AES2 $AES3 ]
+			}
+		]
+	}
+}
diff -Naur alsa-lib-1.2.2-old/src/conf/cards/AMLOGIC-HDMI-SPDIF.conf alsa-lib-1.2.2-new/src/conf/cards/AMLOGIC-HDMI-SPDIF.conf
--- alsa-lib-1.2.2-old/src/conf/cards/AMLOGIC-HDMI-SPDIF.conf	1970-01-01 01:00:00.000000000 +0100
+++ alsa-lib-1.2.2-new/src/conf/cards/AMLOGIC-HDMI-SPDIF.conf	2020-06-18 20:36:28.476666656 +0200
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2020-present Team LibreELEC (https://libreelec.tv)
+
+# NOTE: IEC958 controls are not currently implemented in the audio
+# driver, so control blocks are currently dummies.
+
+<confdir:pcm/hdmi.conf>
+
+AMLOGIC-HDMI-SPDIF.pcm.hdmi.0 {
+	@args [ CARD AES0 AES1 AES2 AES3 ]
+	@args.CARD { type string }
+	@args.AES0 { type integer }
+	@args.AES1 { type integer }
+	@args.AES2 { type integer }
+	@args.AES3 { type integer }
+	type hooks
+	slave.pcm {
+		type hw
+		card $CARD
+		device 0
+	}
+	hooks.0 {
+		type ctl_elems
+		hook_args [
+			{
+				interface MIXER
+				name "IEC958 Playback Default"
+				lock true
+				preserve true
+				optional true
+				value [ $AES0 $AES1 $AES2 $AES3 ]
+			}
+		]
+	}
+}
+
+<confdir:pcm/iec958.conf>
+
+AMLOGIC-HDMI-SPDIF.pcm.iec958.0 {
+	@args [ CARD AES0 AES1 AES2 AES3 ]
+	@args.CARD { type string }
+	@args.AES0 { type integer }
+	@args.AES1 { type integer }
+	@args.AES2 { type integer }
+	@args.AES3 { type integer }
+	type hooks
+	slave.pcm {
+		type hw
+		card $CARD
+		device 1
+	}
+	hooks.0 {
+		type ctl_elems
+		hook_args [
+			{
+				interface MIXER
+				name "IEC958 Playback Default"
+				lock true
+				preserve true
+				optional true
+				value [ $AES0 $AES1 $AES2 $AES3 ]
+			}
+		]
+	}
+}
diff -Naur alsa-lib-1.2.2-old/src/conf/cards/Makefile.am alsa-lib-1.2.2-new/src/conf/cards/Makefile.am
--- alsa-lib-1.2.2-old/src/conf/cards/Makefile.am	2020-02-19 10:35:39.000000000 +0100
+++ alsa-lib-1.2.2-new/src/conf/cards/Makefile.am	2020-06-18 20:35:50.933333327 +0200
@@ -2,6 +2,8 @@
 alsadir = $(alsaconfigdir)/cards
 cfg_files = aliases.conf \
 	AACI.conf \
+	AMLOGIC-HDMI.conf \
+	AMLOGIC-HDMI-SPDIF.conf \
 	ATIIXP.conf \
 	ATIIXP-SPDMA.conf \
 	ATIIXP-MODEM.conf \
