diff --speed-large-files --no-dereference --minimal -Naur alsa-lib-1.2.14/src/conf/alsa.conf alsa-lib-1.2.14/src/conf/alsa.conf
--- alsa-lib-1.2.14/src/conf/alsa.conf	2025-04-14 18:42:04.000000000 +0200
+++ alsa-lib-1.2.14/src/conf/alsa.conf	2025-11-13 10:57:09.685475618 +0100
@@ -96,11 +96,11 @@
 #
 
 # show all name hints also for definitions without hint {} section
-defaults.namehint.showall off
+defaults.namehint.showall on
 # show just basic name hints
 defaults.namehint.basic on
 # show extended name hints
-defaults.namehint.extended off
+defaults.namehint.extended on
 #
 defaults.ctl.card 0
 defaults.pcm.card 0
