diff --speed-large-files --no-dereference --minimal -Naur linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts
--- linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts	2023-02-09 11:28:28.000000000 +0100
+++ linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b.dts	2023-02-13 18:51:29.296369671 +0100
@@ -20,7 +20,6 @@
 	status = "okay";
 
 	brcmf: wifi@1 {
-		compatible = "brcm,bcm4329-fmac";
 		reg = <1>;
 		interrupt-parent = <&gpio0>;
 		interrupts = <RK_PA3 IRQ_TYPE_LEVEL_HIGH>;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts
--- linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts	2023-02-09 11:28:28.000000000 +0100
+++ linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts	2023-02-13 18:51:20.199703331 +0100
@@ -28,7 +28,6 @@
 	status = "okay";
 
 	brcmf: wifi@1 {
-		compatible = "brcm,bcm4329-fmac";
 		reg = <1>;
 		interrupt-parent = <&gpio0>;
 		interrupts = <RK_PA3 GPIO_ACTIVE_HIGH>;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts
--- linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts	2023-02-09 11:28:28.000000000 +0100
+++ linux-6.1.11/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c.dts	2023-02-13 18:51:38.639702661 +0100
@@ -28,7 +28,6 @@
 	status = "okay";
 
 	brcmf: wifi@1 {
-		compatible = "brcm,bcm4329-fmac";
 		reg = <1>;
 		interrupt-parent = <&gpio0>;
 		interrupts = <RK_PA3 IRQ_TYPE_LEVEL_HIGH>;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.8.2/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts linux-6.8.2/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts
--- linux-6.8.2/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts	2024-03-26 23:23:34.000000000 +0100
+++ linux-6.8.2/arch/arm64/boot/dts/rockchip/rk3399-rock-4se.dts	2024-03-30 19:27:30.003354436 +0100
@@ -29,7 +31,6 @@
 	status = "okay";
 
 	brcmf: wifi@1 {
-		compatible = "brcm,bcm4329-fmac";
 		reg = <1>;
 		interrupt-parent = <&gpio0>;
 		interrupts = <RK_PA3 IRQ_TYPE_LEVEL_HIGH>;
