diff --speed-large-files --no-dereference --minimal -Naur linux-6.15.2/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi linux-6.15.2/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
--- linux-6.15.2/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2025-06-13 09:23:04.908340462 +0200
+++ linux-6.15.2/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi	2025-06-13 09:05:04.138363859 +0200
@@ -1145,6 +1145,7 @@
 		clock-names = "aclk", "hclk";
 		iommus = <&vpu121_mmu>;
 		power-domains = <&power RK3588_PD_VDPU>;
+		status = "disabled";
 	};
 
 	vpu121_mmu: iommu@fdb50800 {
