diff --speed-large-files --no-dereference --minimal -Naur linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts
--- linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts	2025-02-03 00:39:26.000000000 +0100
+++ linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts	2025-02-09 12:02:12.810005967 +0100
@@ -17,8 +17,8 @@
 
 	aliases {
 		ethernet0 = &gmac1;
-		mmc0 = &sdhci;
-		mmc1 = &sdmmc;
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
 	};
 
 	chosen {
@@ -55,23 +55,24 @@
 		led-0 {
 			color = <LED_COLOR_ID_GREEN>;
 			default-state = "on";
-			function = LED_FUNCTION_POWER;
+			function = LED_FUNCTION_HEARTBEAT;
 			gpios = <&gpio3 RK_PC4 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "heartbeat";
 		};
 
 		led-1 {
 			color = <LED_COLOR_ID_BLUE>;
 			default-state = "on";
-			function = LED_FUNCTION_HEARTBEAT;
+			function = LED_FUNCTION_DISK_ACTIVITY;
 			gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "disk-activity";
 		};
 	};
 
-	fan: fan {
+	fan: pwm-fan {
 		compatible = "pwm-fan";
 		#cooling-cells = <2>;
-		cooling-levels = <0 24 44 64 128 192 255>;
+		cooling-levels = <0 96 128 164 196 224 255>;
 		fan-supply = <&vcc_5v0>;
 		pwms = <&pwm3 0 10000 0>;
 	};
@@ -257,6 +258,10 @@
 	status = "okay";
 };
 
+&hdmi0_sound {
+	status = "okay";
+};
+
 &hdmi0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&hdmim0_tx0_cec
@@ -404,6 +409,40 @@
 		};
 	};
 };
+
+&i2s5_8ch {
+	status = "okay";
+};
+
+&package_thermal {
+	polling-delay = <1000>;
+
+	trips {
+		package_fan0: package-fan0 {
+			temperature = <55000>;
+			hysteresis = <2000>;
+			type = "active";
+		};
+
+		package_fan1: package-fan1 {
+			temperature = <65000>;
+			hysteresis = <2000>;
+			type = "active";
+		};
+	};
+
+	cooling-maps {
+		map0 {
+			trip = <&package_fan0>;
+			cooling-device = <&fan THERMAL_NO_LIMIT 1>;
+		};
+
+		map1 {
+			trip = <&package_fan1>;
+			cooling-device = <&fan 2 THERMAL_NO_LIMIT>;
+		};
+	};
+};
 
 &mdio1 {
 	rgmii_phy1: ethernet-phy@1 {
