diff --speed-large-files --no-dereference --minimal -Naur linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-nanopi-r6.dtsi linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-nanopi-r6.dtsi
--- linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-nanopi-r6.dtsi	2025-02-03 00:39:26.000000000 +0100
+++ linux-6.14-rc1/arch/arm64/boot/dts/rockchip/rk3588s-nanopi-r6.dtsi	2025-02-09 16:03:25.646675987 +0100
@@ -63,7 +63,7 @@
 		sys_led: led-0 {
 			label = "sys_led";
 			gpios = <&gpio1 RK_PC1 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			linux,default-trigger = "disk-activity";
 			pinctrl-names = "default";
 			pinctrl-0 = <&sys_led_pin>;
 		};
@@ -235,6 +235,10 @@
 	status = "okay";
 };
 
+&hdmi0_sound {
+	status = "okay";
+};
+
 &hdmi0 {
 	status = "okay";
 };
@@ -335,6 +339,10 @@
 	};
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
 &mdio1 {
 	rgmii_phy1: ethernet-phy@1 {
 		compatible = "ethernet-phy-id001c.c916";
