diff --speed-large-files --no-dereference --minimal -Naur linux-6.15/arch/arm64/boot/dts/rockchip/rk3576.dtsi linux-6.15/arch/arm64/boot/dts/rockchip/rk3576.dtsi
--- linux-6.15/arch/arm64/boot/dts/rockchip/rk3576.dtsi	2025-05-26 01:09:23.000000000 +0200
+++ linux-6.15/arch/arm64/boot/dts/rockchip/rk3576.dtsi	2025-06-02 14:29:44.071317027 +0200
@@ -1959,6 +1959,7 @@
 			reg = <0x0 0x2b000000 0x0 0x2000>;
 			clocks = <&cru CLK_PHY_REF_SRC>, <&cru PCLK_HDPTX_APB>;
 			clock-names = "ref", "apb";
+			#clock-cells = <0x00>;
 			resets = <&cru SRST_P_HDPTX_APB>, <&cru SRST_HDPTX_INIT>,
 				 <&cru SRST_HDPTX_CMN>, <&cru SRST_HDPTX_LANE>;
 			reset-names = "apb", "init", "cmn", "lane";
