diff --speed-large-files --no-dereference --minimal -Naur linux-6.15.2/arch/arm64/boot/dts/rockchip/rk3576.dtsi linux-6.15.2/arch/arm64/boot/dts/rockchip/rk3576.dtsi
--- linux-6.15.2/arch/arm64/boot/dts/rockchip/rk3576.dtsi	2025-06-12 21:02:02.754112354 +0200
+++ linux-6.15.2/arch/arm64/boot/dts/rockchip/rk3576.dtsi	2025-06-12 21:01:42.677446113 +0200
@@ -1123,7 +1123,7 @@
 			assigned-clocks = <&cru ACLK_RKVDEC_ROOT>, <&cru CLK_RKVDEC_CORE>,
 					  <&cru ACLK_RKVDEC_ROOT_BAK>, <&cru CLK_RKVDEC_HEVC_CA>;
 			assigned-clock-rates = <600000000>,<600000000>, <500000000>, <1000000000>;
-			iommus = <&rkvdec_mmu>;
+			//iommus = <&rkvdec_mmu>;
 			power-domains = <&power RK3576_PD_VDEC>;
 			sram = <&rkvdec_sram>;
 		};
