diff -Naur mysql-5.1.67-old/config/ac-macros/ssl.m4 mysql-5.1.67-new/config/ac-macros/ssl.m4
--- mysql-5.1.67-old/config/ac-macros/ssl.m4	2012-12-13 02:05:51.000000000 -0800
+++ mysql-5.1.67-new/config/ac-macros/ssl.m4	2012-12-22 07:31:37.000000000 -0800
@@ -114,7 +114,7 @@
   openssl_libs="-L$location/lib/ -lssl -lcrypto"
   MYSQL_CHECK_SSL_DIR([$openssl_includes], [$openssl_libs])
 
-  if test "$mysql_ssl_found" == "no"
+  if test "$mysql_ssl_found" = "no"
   then
     #
     # BUG 764: Compile failure with OpenSSL on Red Hat Linux (krb5.h missing)
@@ -124,7 +124,7 @@
     MYSQL_CHECK_SSL_DIR([$openssl_includes], [$openssl_libs])
   fi
 
-  if test "$mysql_ssl_found" == "no"
+  if test "$mysql_ssl_found" = "no"
   then
     AC_MSG_ERROR([Could not link with SSL libs at $location])
   fi
