diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythui/mythuiscreenbounds.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythui/mythuiscreenbounds.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythui/mythuiscreenbounds.cpp	2021-05-18 16:11:27.086666604 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythui/mythuiscreenbounds.cpp	2021-05-18 16:11:29.916666604 +0200
@@ -223,12 +223,12 @@
 
 int MythUIScreenBounds::NormX(int X) const
 {
-    return qRound(X * m_screenHorizScale);
+    return (int)(X * m_screenHorizScale);
 }
 
 int MythUIScreenBounds::NormY(int Y) const
 {
-    return qRound(Y * m_screenVertScale);
+    return (int)(Y * m_screenVertScale);
 }
 
 void MythUIScreenBounds::GetScalingFactors(float& Horizontal, float& Vertical) const
