diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmyth/audio/audiooutputalsa.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmyth/audio/audiooutputalsa.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmyth/audio/audiooutputalsa.cpp	2021-05-18 16:12:04.443333272 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmyth/audio/audiooutputalsa.cpp	2021-05-18 16:12:07.099999938 +0200
@@ -368,9 +368,8 @@
         {
             case FORMAT_U8:     afmt = SND_PCM_FORMAT_U8;    break;
             case FORMAT_S16:    afmt = SND_PCM_FORMAT_S16;   break;
-            // NOLINTNEXTLINE(bugprone-branch-clone)
             case FORMAT_S24LSB: afmt = SND_PCM_FORMAT_S24;   break;
-            case FORMAT_S24:    afmt = SND_PCM_FORMAT_S24;   break;
+            case FORMAT_S24:    afmt = SND_PCM_FORMAT_S32;   break;
             case FORMAT_S32:    afmt = SND_PCM_FORMAT_S32;   break;
             case FORMAT_FLT:    afmt = SND_PCM_FORMAT_FLOAT; break;
             default:         continue;
@@ -465,9 +464,8 @@
     {
         case FORMAT_U8:     format = SND_PCM_FORMAT_U8;    break;
         case FORMAT_S16:    format = SND_PCM_FORMAT_S16;   break;
-        // NOLINTNEXTLINE(bugprone-branch-clone)
         case FORMAT_S24LSB: format = SND_PCM_FORMAT_S24;   break;
-        case FORMAT_S24:    format = SND_PCM_FORMAT_S24;   break;
+        case FORMAT_S24:    format = SND_PCM_FORMAT_S32;   break;
         case FORMAT_S32:    format = SND_PCM_FORMAT_S32;   break;
         case FORMAT_FLT:    format = SND_PCM_FORMAT_FLOAT; break;
         default:
