diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/configure mythtv-master-20210518-gc8c226c9b4-new/mythtv/configure
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/configure	2021-05-18 16:12:13.069999939 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/configure	2021-05-18 16:12:16.909999938 +0200
@@ -126,6 +126,7 @@
   --disable-ceton          disable support for Ceton cards
   --disable-satip          disable support for Sat>IP
   --disable-v4l2           disable Video4Linux support
+  --enable-v4l2request     enable Video4Linux V4l2-request support
   --disable-ivtv           disable ivtv support (PVR-x50) req. v4l2 support
   --disable-hdpvr          disable HD-PVR support
   --disable-dvb            disable DVB support
@@ -1957,7 +1958,6 @@
     wma_freqs
     wmv2dsp
 "
-
 MYTHTV_CONFIG_LIST='
     backend
     bdjava
@@ -1996,6 +1996,7 @@
     taglib
     v4l2
     v4l2prime
+    v4l2request
     v4l1
     valgrind
     x11
@@ -2797,6 +2798,7 @@
 enable v4l1
 enable v4l2
 enable v4l2prime
+disable v4l2request
 enable x11
 disable indevs
 enable libfftw3
@@ -4764,6 +4766,7 @@
         disable v4l1
         disable v4l2
         disable v4l2prime
+        disable v4l2request
         disable x11
         # Workaround compile errors from missing u_int/uint def
         CFLAGS=`echo $CFLAGS | sed 's/-D_POSIX_C_SOURCE=200112//'`
@@ -4814,6 +4817,7 @@
         disable v4l1
         disable v4l2
         disable v4l2prime
+        disable v4l2request
         enable  windows
         disable x11
         disable qtdbus
@@ -7323,6 +7327,10 @@
     ffopts="$ffopts --enable-lto"
 fi
 
+if enabled v4l2request; then
+    ffopts="$ffopts --enable-v4l2_request --enable-libdrm --enable-libudev"
+fi
+
 ffmpeg_extra_cflags="$extra_cflags -w"
 
 ## Call FFmpeg configure here
@@ -7537,7 +7545,7 @@
   if enabled drm; then
     echo "DRM Qt integration        ${qtprivateheaders-no}"
   fi
-  echo "Video4Linux codecs        ${v4l2-no} (DRM ${v4l2prime-no})"
+  echo "Video4Linux codecs        ${v4l2-no} (DRM:${v4l2prime-no} v4l2-request:${v4l2request-no})"
   echo "MMAL decoder support      ${mmal-no}"
   echo "OpenGL                    ${opengl-no} (OpenGLES ${opengles-no})"
   echo "EGL support               ${egl-no}"
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythtv/libmythtv.pro mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythtv/libmythtv.pro
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythtv/libmythtv.pro	2021-05-18 16:12:13.246666605 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythtv/libmythtv.pro	2021-05-18 16:12:16.909999938 +0200
@@ -525,6 +525,10 @@
         DEFINES += USING_V4L2PRIME
     }
 
+    using_v4l2request {
+        DEFINES += USING_V4L2_REQUEST
+    }
+
     using_vdpau:using_x11 {
         DEFINES += USING_VDPAU
         HEADERS += decoders/mythvdpaucontext.h   decoders/mythvdpauhelper.h
