diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/external/FFmpeg/libavcodec/golomb.h mythtv-master-20210518-gc8c226c9b4-new/mythtv/external/FFmpeg/libavcodec/golomb.h
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/external/FFmpeg/libavcodec/golomb.h	2021-05-18 16:12:27.166666604 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/external/FFmpeg/libavcodec/golomb.h	2021-05-18 16:12:35.966666604 +0200
@@ -91,7 +91,7 @@
         LAST_SKIP_BITS(re, gb, 32 - log);
         CLOSE_READER(re, gb);
         if (log < 7) {
-            av_log(NULL, AV_LOG_ERROR, "Invalid UE golomb code\n");
+            av_log(NULL, AV_LOG_DEBUG, "Invalid UE golomb code\n");
             return AVERROR_INVALIDDATA;
         }
         buf >>= log;
