diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/configure mythtv-master-20210518-gc8c226c9b4-new/mythtv/configure
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/configure	2021-05-18 16:12:34.856666604 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/configure	2021-05-18 16:12:38.253333271 +0200
@@ -105,7 +105,9 @@
   --disable-avx            disable AVX optimizations
   --disable-yasm           disable use of yasm assembler
   --enable-pic             build position-independent code
+  --enable-lto             enable link-time optimizations (LTO)
   --enable-proc-opt        enable processor specific compilation
+  --disable-inlineasm      disable FFmpeg in-line assembler optimizations
   --enable-compdb          enable clang compiler database
 
   --disable-audio-oss      disable OSS audio support
@@ -1997,6 +1999,7 @@
     v4l2
     v4l2prime
     v4l2request
+    inlineasm
     v4l1
     valgrind
     x11
@@ -2741,6 +2744,7 @@
 enable optimizations
 enable swscale_alpha
 enable valgrind_backtrace
+enable inlineasm
 
 sws_max_filter_size_default=256
 set_default sws_max_filter_size
@@ -7326,6 +7330,10 @@
 if enabled lto; then
     ffopts="$ffopts --enable-lto"
 fi
+ 
+if disabled inlineasm; then
+    ffopts="$ffopts --disable-inline-asm"
+fi
 
 if enabled v4l2request; then
     ffopts="$ffopts --enable-v4l2_request --enable-libdrm --enable-libudev"
@@ -7610,6 +7618,8 @@
 echo "C++ standard supported    ${CPP_STANDARD}"
 echo "Enforce c++11 nullptr     ${enforce_nullptr-no}"
 echo "Enforce shadowed vars     ${enforce_wshadow-no}"
+echo "Link time optimize (LTO)  ${lto-no}"
+echo "FFmpeg in-line asm        ${inlineasm-no}"
 echo ""
 
 # Remove redundant elements from variables
