diff -Naur font-misc-misc-1.1.2-old/Makefile.am font-misc-misc-1.1.2-new/Makefile.am
--- font-misc-misc-1.1.2-old/Makefile.am	2010-11-14 15:37:44.000000000 -0800
+++ font-misc-misc-1.1.2-new/Makefile.am	2010-11-15 22:26:14.000000000 -0800
@@ -184,7 +184,7 @@
 JISX0201_PCF_FILES = $(BDF2UCS_JISX0201_FONT_FILES:%=%-JISX0201.1976-0.pcf$(COMPRESS_SUFFIX))
 endif
 
-UTIL_DIR = @MAPFILES_PATH@
+UTIL_DIR = $(DESTDIR)@MAPFILES_PATH@
 
 SUFFIXES = .bdf .pcf$(COMPRESS_SUFFIX) t.bdf \
 	-ISO8859-1.bdf \
diff -Naur font-misc-misc-1.1.2-old/Makefile.in font-misc-misc-1.1.2-new/Makefile.in
--- font-misc-misc-1.1.2-old/Makefile.in	2010-11-15 14:54:03.000000000 -0800
+++ font-misc-misc-1.1.2-new/Makefile.in	2010-11-15 22:26:14.000000000 -0800
@@ -359,7 +359,7 @@
 @KOI8_R_TRUE@KOI8_R_PCF_FILES = $(BDF2UCS_KOI8_FONT_FILES:%=%-KOI8-R.pcf$(COMPRESS_SUFFIX))
 @JISX0201_TRUE@JISX0201_BDF_FILES = $(BDF2UCS_JISX0201_FONT_FILES:%=%-JISX0201.1976-0.bdf)
 @JISX0201_TRUE@JISX0201_PCF_FILES = $(BDF2UCS_JISX0201_FONT_FILES:%=%-JISX0201.1976-0.pcf$(COMPRESS_SUFFIX))
-UTIL_DIR = @MAPFILES_PATH@
+UTIL_DIR = $(DESTDIR)@MAPFILES_PATH@
 SUFFIXES = .bdf .pcf$(COMPRESS_SUFFIX) t.bdf \
 	-ISO8859-1.bdf \
 	-ISO8859-2.bdf \
