diff -Naur xine-ui-0.99.7-old/misc/Makefile.am xine-ui-0.99.7-new/misc/Makefile.am
--- xine-ui-0.99.7-old/misc/Makefile.am	2012-06-10 12:56:45.000000000 -0700
+++ xine-ui-0.99.7-new/misc/Makefile.am	2012-06-10 12:57:21.000000000 -0700
@@ -1,6 +1,6 @@
 include $(top_srcdir)/misc/Makefile.quiet
 
-SUBDIRS = desktops visuals
+SUBDIRS = visuals
 
 # splash-foo/manifest contains names of extra source files, distributed only.
 
