diff -Naur xine-ui-0.99.7-old/m4/_xine.m4 xine-ui-0.99.7-new/m4/_xine.m4
--- xine-ui-0.99.7-old/m4/_xine.m4	2011-11-08 12:49:07.000000000 -0800
+++ xine-ui-0.99.7-new/m4/_xine.m4	2012-06-10 12:55:23.000000000 -0700
@@ -31,47 +31,17 @@
   found_lirc=no
   if test x"$enable_lirc" = xyes; then
     have_lirc=yes
-    PKG_CHECK_MODULES(LIRC, liblircclient0, [found_lirc=yes], [:])
-    if test "$found_lirc" = yes; then
-      LIRC_INCLUDE="$LIRC_CFLAGS"
-    else
      AC_REQUIRE_CPP
      AC_CHECK_LIB(lirc_client,lirc_init,
            [AC_CHECK_HEADER(lirc/lirc_client.h, true, have_lirc=no)], have_lirc=no)
      if test "$have_lirc" = "yes"; then
-        saved_CFLAGS=$CFLAGS
-        saved_LIBS=$LIBS
-        if test x"$LIRC_PREFIX" != "x"; then
-	   LIRC_INCLUDE="-I$LIRC_PREFIX/include"
-        fi
-        AC_MSG_CHECKING([for liblircclient])
-        for type in "$shrext" .a; do
-          for lib in lib32 lib lib64; do
-            for llirc in "$LIRC_PREFIX/$lib" /$lib /usr/$lib /usr/local/$lib; do
-              CFLAGS="$saved_CFLAGS $LIRC_INCLUDE"
-              LIBS="$saved_LIBS $llirc/liblirc_client$type"
-              AC_LINK_IFELSE(
-                [AC_LANG_PROGRAM([#include <lirc/lirc_client.h>], [lirc_init("",0)])],
-                [LIRC_LIBS="$llirc/liblirc_client$type"
-                 found_lirc=yes]
-                AC_DEFINE([HAVE_LIRC],,[Define this if you have LIRC (liblirc_client) installed]),
-              )
-              if test x"$found_lirc" = xyes; then break 3; fi
-            done
-          done
-        done
-        CFLAGS=$saved_CFLAGS
-        LIBS=$saved_LIBS
-        if test x"$found_lirc" = xyes; then
-          AC_MSG_RESULT([$LIRC_LIBS])
-        else
-          AC_MSG_RESULT([no])
-        fi
+        found_lirc=yes
+        [LIRC_LIBS="-llirc_client"]
+        AC_DEFINE([HAVE_LIRC],,[Define this if you have LIRC (liblirc_client) installed])
      else
 	test $given = Y && AC_MSG_ERROR([LIRC client support requested but not available])
 	AC_MSG_RESULT([*** LIRC client support not available, LIRC support will be disabled ***])
      fi
-    fi
   fi
      if test "$found_lirc" = yes; then
 	AC_DEFINE([HAVE_LIRC],,[Define this if you have LIRC (liblirc_client) installed])
