diff -Naur xine-ui-0.99.7-old/configure.ac xine-ui-0.99.7-new/configure.ac
--- xine-ui-0.99.7-old/configure.ac	2012-06-10 12:56:21.000000000 -0700
+++ xine-ui-0.99.7-new/configure.ac	2012-06-10 12:56:57.000000000 -0700
@@ -274,11 +274,8 @@
 dnl
 dnl XTest Extension
 dnl
-PKG_CHECK_MODULES([XTEST], [xtst], [ac_have_xtest="yes"], [
-  AC_CHECK_LIB(Xtst, XTestFakeKeyEvent, 
-             XTEST_LIBS="-lXtst"
-	     ac_have_xtest="yes",, $X_LIBS $X_PRE_LIBS -lXext $X_EXTRA_LIBS)
-])
+ac_have_xtest="no"
+XTEST_LIBS=
 
 if test "x$ac_have_xtest" = "xyes"; then
    AC_DEFINE(HAVE_XTESTEXTENSION,,[Define this if you have libXtst installed])
