diff -Naur xine-ui-0.99.7-old/src/xitk/videowin.c xine-ui-0.99.7-new/src/xitk/videowin.c
--- xine-ui-0.99.7-old/src/xitk/videowin.c	2012-06-10 12:55:52.000000000 -0700
+++ xine-ui-0.99.7-new/src/xitk/videowin.c	2012-06-10 12:56:30.000000000 -0700
@@ -2312,6 +2312,51 @@
     else 
 #endif
     {
+      /* Reset the xscreensaver. Look up the command in PATH only once to save time, */
+      /* assuming its location and permission will not change during run time of xine-ui. */
+      {
+	static char *xssaver_args[] = { "xscreensaver-command", "-deactivate", NULL };
+	static char *xssaver_path   = NULL;
+
+	if(!xssaver_path) {
+	  char *path = getenv("PATH");
+
+	  if(!path)
+	    path = "/usr/local/bin:/usr/bin";
+	  do {
+	    char *p, pbuf[XITK_PATH_MAX + XITK_NAME_MAX + 2];
+	    int   plen;
+
+	    for(p = path; *path && *path != ':'; path++)
+	      ;
+	    if(p == path)
+	      plen = 1, p = ".";
+	    else
+	      plen = path - p;
+	    plen = snprintf(pbuf, sizeof(pbuf), "%.*s/%s", plen, p, xssaver_args[0]);
+	    if (plen <= -1 || plen >= sizeof(pbuf))
+	      xssaver_path = "";
+	    else
+	      xssaver_path = (access(pbuf, X_OK)) ? "" : strdup(pbuf);
+	  } while(!xssaver_path[0] && *path++);
+	}
+	if(xssaver_path[0])
+        {
+          switch(fork())
+          {
+            case -1:
+              break;
+            case  0:
+	      execv(xssaver_path, xssaver_args);
+	      exit(0);
+              break;
+            default:
+              wait(NULL);
+              break;
+          }
+	}
+      }
+
       /* Reset the gnome screensaver. Look up the command in PATH only once to save time, */
       /* assuming its location and permission will not change during run time of xine-ui. */
       {
