diff -Naur xorg-server-1.17.99.901-old/Xi/exevents.c xorg-server-1.17.99.901-new/Xi/exevents.c
--- xorg-server-1.17.99.901-old/Xi/exevents.c	2015-06-30 05:59:31.000000000 +0200
+++ xorg-server-1.17.99.901-new/Xi/exevents.c	2015-09-18 21:37:00.000000000 +0200
@@ -1732,16 +1732,16 @@
     }
 
     /* send KeyPress and KeyRelease events to XACE plugins */
-    if (XaceHookIsSet(XACE_KEY_AVAIL) &&
-            (event->type == ET_KeyPress || event->type == ET_KeyRelease)) {
-        xEvent *core;
-        int count;
-
-        if (EventToCore(ev, &core, &count) == Success && count > 0) {
-            XaceHook(XACE_KEY_AVAIL, core, device, 0);
-            free(core);
-        }
-    }
+    //if (XaceHookIsSet(XACE_KEY_AVAIL) &&
+    //        (event->type == ET_KeyPress || event->type == ET_KeyRelease)) {
+    //    xEvent *core;
+    //    int count;
+    //
+    //    if (EventToCore(ev, &core, &count) == Success && count > 0) {
+    //        XaceHook(XACE_KEY_AVAIL, core, device, 0);
+    //        free(core);
+    //    }
+    //}
 
     if (DeviceEventCallback && !syncEvents.playingEvents) {
         DeviceEventInfoRec eventinfo;
