diff -Naur xorg-server-1.13.2-old/xorg-server.m4 xorg-server-1.13.2-new/xorg-server.m4
--- xorg-server-1.13.2-old/xorg-server.m4	2013-02-02 06:08:06.000000000 -0800
+++ xorg-server-1.13.2-new/xorg-server.m4	2013-02-02 06:09:14.000000000 -0800
@@ -31,7 +31,8 @@
 AC_DEFUN([XORG_DRIVER_CHECK_EXT],[
 	AC_REQUIRE([PKG_PROG_PKG_CONFIG])
 	SAVE_CFLAGS="$CFLAGS"
-	CFLAGS="$CFLAGS -I`$PKG_CONFIG --variable=sdkdir xorg-server`"
+	m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+	CFLAGS="$CFLAGS -I$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=sdkdir xorg-server`"
 	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "xorg-server.h"
 #if !defined $1
