diff --speed-large-files --no-dereference --minimal -Naur xorg-server-21.1.13/configure xorg-server-21.1.13/configure
--- xorg-server-21.1.13/configure	2024-04-12 19:17:08.000000000 +0200
+++ xorg-server-21.1.13/configure	2024-05-05 15:15:54.156955836 +0200
@@ -9761,9 +9761,9 @@
 			CFLAGS="$CFLAGS -Werror=unused-command-line-argument"
 		fi
 
-		CFLAGS="$CFLAGS -Werror=array-bounds"
+		CFLAGS="$CFLAGS"
 
-		{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if $CC supports -Werror=array-bounds" >&5
+		{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if $CC supports -Werror-array-bounds" >&5
 printf %s "checking if $CC supports -Werror=array-bounds... " >&6; }
 		cacheid=xorg_cv_cc_flag__Werror_array_bounds
 		if eval test \${$cacheid+y}
@@ -9798,7 +9798,7 @@
 		{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $supported" >&5
 printf "%s\n" "$supported" >&6; }
 		if test "$supported" = "yes" ; then
-			BASE_CFLAGS="$BASE_CFLAGS -Werror=array-bounds"
+			BASE_CFLAGS="$BASE_CFLAGS"
 			found="yes"
 		fi
 	fi
@@ -11427,7 +11427,7 @@
 			CFLAGS="$CFLAGS -Werror=unused-command-line-argument"
 		fi
 
-		CFLAGS="$CFLAGS -Warray-bounds"
+		CFLAGS="$CFLAGS"
 
 		{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if $CC supports -Warray-bounds" >&5
 printf %s "checking if $CC supports -Warray-bounds... " >&6; }
@@ -11464,7 +11464,7 @@
 		{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $supported" >&5
 printf "%s\n" "$supported" >&6; }
 		if test "$supported" = "yes" ; then
-			BASE_CFLAGS="$BASE_CFLAGS -Warray-bounds"
+			BASE_CFLAGS="$BASE_CFLAGS"
 			found="yes"
 		fi
 	fi
