diff -Naur bdremote-ng-0.5-old/src/event_out.c bdremote-ng-0.5-new/src/event_out.c
--- bdremote-ng-0.5-old/src/event_out.c	2009-12-18 09:44:54.000000000 -0800
+++ bdremote-ng-0.5-new/src/event_out.c	2010-01-02 18:24:18.000000000 -0800
@@ -54,6 +54,14 @@
             {
                ;
             }
+         memset(&event, 0, sizeof(event));
+         event.type  = EV_SYN;
+         event.code  = SYN_REPORT;
+         event.value = 0;
+         if (write(event_out_fd, &event, sizeof(event)) != sizeof(event))
+            {
+               ;
+            }
       }
 #endif
 }
