diff -Naur e2fsprogs-1.42.7-old/lib/et/compile_et.sh.in e2fsprogs-1.42.7-new/lib/et/compile_et.sh.in
--- e2fsprogs-1.42.7-old/lib/et/compile_et.sh.in	2009-08-12 18:39:57.000000000 -0700
+++ e2fsprogs-1.42.7-new/lib/et/compile_et.sh.in	2013-01-29 20:25:44.000000000 -0800
@@ -51,12 +51,14 @@
     rm -f ${BASE}.h.$$
 else
     mv -f ${BASE}.h.$$ ${BASE}.h
-    chmod -w ${BASE}.h
+    chmod a-w ${BASE}.h
 fi
 $AWK -f "${DIR}/et_c.awk" "outfile=${BASE}.c.$$" "outfn=${BASE}.c" "$ROOT.et"
 if test -f ${BASE}.c && cmp -s ${BASE}.c.$$ ${BASE}.c ; then
     rm -f ${BASE}.c.$$
 else
     mv -f ${BASE}.c.$$ ${BASE}.c
-    chmod -w ${BASE}.c
+    chmod a-w ${BASE}.c
 fi
+
+exit 0
diff -Naur e2fsprogs-1.42.7-old/lib/ss/mk_cmds.sh.in e2fsprogs-1.42.7-new/lib/ss/mk_cmds.sh.in
--- e2fsprogs-1.42.7-old/lib/ss/mk_cmds.sh.in	2009-08-12 18:39:57.000000000 -0700
+++ e2fsprogs-1.42.7-new/lib/ss/mk_cmds.sh.in	2013-01-29 20:25:44.000000000 -0800
@@ -57,6 +57,6 @@
 else
 	rm -f "${BASE}.c"
 	mv -f "${TMP}" "${BASE}.c"
-	chmod -w "${BASE}.c"
+	chmod a-w "${BASE}.c"
 	exit 0
 fi
