diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythui/langsettings.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythui/langsettings.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/libs/libmythui/langsettings.cpp	2021-05-18 16:12:02.136666605 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/libs/libmythui/langsettings.cpp	2021-05-18 16:12:04.929999938 +0200
@@ -199,6 +199,7 @@
 
 void LanguageSelection::Save(void)
 {
+    QString hostname           = gCoreContext->GetHostName();
     MythUIButtonListItem *item = m_languageList->GetItemCurrent();
 
     if (!item)
@@ -210,7 +211,7 @@
     }
 
     QString langCode = item->GetData().toString();
-    gCoreContext->SaveSettingOnHost("Language", langCode, nullptr);
+    gCoreContext->SaveSettingOnHost("Language", langCode, hostname);
 
     item = m_countryList->GetItemCurrent();
 
@@ -223,7 +224,7 @@
     }
 
     QString countryCode = item->GetData().toString();
-    gCoreContext->SaveSettingOnHost("Country", countryCode, nullptr);
+    gCoreContext->SaveSettingOnHost("Country", countryCode, hostname);
 
     if (m_language != langCode)
         m_languageChanged = true;
diff -Naur mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-master-20210518-gc8c226c9b4-old/mythtv/programs/mythfrontend/globalsettings.cpp	2021-05-18 16:12:01.556666605 +0200
+++ mythtv-master-20210518-gc8c226c9b4-new/mythtv/programs/mythfrontend/globalsettings.cpp	2021-05-18 16:12:04.929999938 +0200
@@ -3149,9 +3149,9 @@
     return gs;
 }
 
-static GlobalComboBoxSetting *MythLanguage()
+static HostComboBoxSetting *MythLanguage()
 {
-    auto *gc = new GlobalComboBoxSetting("Language");
+    auto *gc = new HostComboBoxSetting("Language");
 
     gc->setLabel(AppearanceSettings::tr("Language"));
 
