diff --speed-large-files --no-dereference --minimal -Naur lirc-0.10.2/lib/lirc_client.c lirc-0.10.2/lib/lirc_client.c
--- lirc-0.10.2/lib/lirc_client.c	2022-09-25 12:03:38.000000000 +0200
+++ lirc-0.10.2/lib/lirc_client.c	2026-02-02 18:34:11.323304984 +0100
@@ -199,7 +199,7 @@
 
 	todo = strlen(ctx->packet);
 	data = ctx->packet;
-	logprintf(LIRC_DEBUG, "lirc_command_run: Sending: %s", data);
+	//logprintf(LIRC_DEBUG, "lirc_command_run: Sending: %s", data);
 	while (todo > 0) {
 		done = write(fd, (void*)data, todo);
 		if (done < 0) {
@@ -222,9 +222,9 @@
 		while (r == EAGAIN);
 		if (!string || strlen(string) == 0)
 			goto bad_packet;
-		logprintf(LIRC_DEBUG,
-			  "lirc_command_run, state: %d, input: \"%s\"\n",
-			  state, string ? string : "(Null)");
+		//logprintf(LIRC_DEBUG,
+		//	  "lirc_command_run, state: %d, input: \"%s\"\n",
+		//	  state, string ? string : "(Null)");
 		switch (state) {
 		case P_BEGIN:
 			if (strcasecmp(string, "BEGIN") != 0)
@@ -268,9 +268,9 @@
 				state = P_N;
 				break;
 			}
-			logprintf(LIRC_DEBUG,
-				  "data: bad packet: %s\n",
-				  string);
+			//logprintf(LIRC_DEBUG,
+			//	  "data: bad packet: %s\n",
+			//	  string);
 			goto bad_packet;
 		case P_N:
 			errno = 0;
