diff -Naur lirc-0.10.0rc1-old/lib/lirc_client.c lirc-0.10.0rc1-new/lib/lirc_client.c
--- lirc-0.10.0rc1-old/lib/lirc_client.c	2017-05-20 07:10:13.000000000 +0200
+++ lirc-0.10.0rc1-new/lib/lirc_client.c	2017-06-08 10:49:11.973333326 +0200
@@ -199,7 +199,7 @@
 
 	todo = strlen(ctx->packet);
 	data = ctx->packet;
-	logprintf(LIRC_DEBUG, "lirc_command_run: Sending: %s", data);
+	//logprintf(LIRC_DEBUG, "lirc_command_run: Sending: %s", data);
 	while (todo > 0) {
 		done = write(fd, (void*)data, todo);
 		if (done < 0) {
@@ -222,9 +222,9 @@
 		while (r == EAGAIN);
 		if (!string || strlen(string) == 0)
 			goto bad_packet;
-		logprintf(LIRC_DEBUG,
-			  "lirc_command_run, state: %d, input: \"%s\"\n",
-			  state, string ? string : "(Null)");
+		//logprintf(LIRC_DEBUG,
+		//	  "lirc_command_run, state: %d, input: \"%s\"\n",
+		//	  state, string ? string : "(Null)");
 		switch (state) {
 		case P_BEGIN:
 			if (strcasecmp(string, "BEGIN") != 0)
