diff -Naur lirc-0.9.0-old/drivers/lirc_atiusb/lirc_atiusb.c lirc-0.9.0-new/drivers/lirc_atiusb/lirc_atiusb.c
--- lirc-0.9.0-old/drivers/lirc_atiusb/lirc_atiusb.c	2016-06-05 17:30:30.000000000 +0200
+++ lirc-0.9.0-new/drivers/lirc_atiusb/lirc_atiusb.c	2016-06-05 17:34:08.000000000 +0200
@@ -85,9 +85,9 @@
 
 /* module parameters */
 #ifdef CONFIG_USB_DEBUG
-static int debug = 1;
+static bool debug = 1;
 #else
-static int debug;
+static bool debug;
 #endif
 #define dprintk(fmt, args...)					\
 	do {							\
@@ -106,7 +106,7 @@
 #define USB_BUFF_LEN 6
 
 static int mask = 0xFFFF;	/* channel acceptance bit mask */
-static int unique;		/* enable channel-specific codes */
+static bool unique;		/* enable channel-specific codes */
 static int repeat = 10;		/* repeat time in 1/100 sec */
 static int emit_updown;		/* send separate press/release codes (rw2) */
 static int emit_modekeys;	/* send keycodes for aux1-4, pc, mouse (rw2) */
diff -Naur lirc-0.9.0-old/drivers/lirc_bt829/lirc_bt829.c lirc-0.9.0-new/drivers/lirc_bt829/lirc_bt829.c
--- lirc-0.9.0-old/drivers/lirc_bt829/lirc_bt829.c	2011-03-25 23:28:18.000000000 +0100
+++ lirc-0.9.0-new/drivers/lirc_bt829/lirc_bt829.c	2016-06-05 17:34:55.000000000 +0200
@@ -64,7 +64,7 @@
 
 #define DRIVER_NAME "lirc_bt829"
 
-static int debug;
+static bool debug;
 #define dprintk(fmt, args...)					\
 	do {							\
 		if (debug)					\
diff -Naur lirc-0.9.0-old/drivers/lirc_dev/lirc_dev.c lirc-0.9.0-new/drivers/lirc_dev/lirc_dev.c
--- lirc-0.9.0-old/drivers/lirc_dev/lirc_dev.c	2016-06-05 17:30:30.000000000 +0200
+++ lirc-0.9.0-new/drivers/lirc_dev/lirc_dev.c	2016-06-05 17:31:05.069999992 +0200
@@ -52,7 +52,7 @@
 #include "lirc_dev.h"
 #endif
 
-static int debug;
+static bool debug;
 #define dprintk(fmt, args...)					\
 	do {							\
 		if (debug)					\
diff -Naur lirc-0.9.0-old/drivers/lirc_igorplugusb/lirc_igorplugusb.c lirc-0.9.0-new/drivers/lirc_igorplugusb/lirc_igorplugusb.c
--- lirc-0.9.0-old/drivers/lirc_igorplugusb/lirc_igorplugusb.c	2011-03-25 23:28:18.000000000 +0100
+++ lirc-0.9.0-new/drivers/lirc_igorplugusb/lirc_igorplugusb.c	2016-06-05 17:35:44.000000000 +0200
@@ -73,9 +73,9 @@
 
 /* debugging support */
 #ifdef CONFIG_USB_DEBUG
-static int debug = 1;
+static bool debug = 1;
 #else
-static int debug;
+static bool debug;
 #endif
 
 #define dprintk(fmt, args...)					\
diff -Naur lirc-0.9.0-old/drivers/lirc_parallel/lirc_parallel.c lirc-0.9.0-new/drivers/lirc_parallel/lirc_parallel.c
--- lirc-0.9.0-old/drivers/lirc_parallel/lirc_parallel.c	2011-03-25 23:28:18.000000000 +0100
+++ lirc-0.9.0-new/drivers/lirc_parallel/lirc_parallel.c	2016-06-05 17:37:06.000000000 +0200
@@ -76,8 +76,8 @@
 
 /*** Global Variables ***/
 
-static int debug;
-static int check_pselecd;
+static bool debug;
+static bool check_pselecd;
 
 unsigned int irq = LIRC_IRQ;
 unsigned int io = LIRC_PORT;
diff -Naur lirc-0.9.0-old/drivers/lirc_serial/lirc_serial.c lirc-0.9.0-new/drivers/lirc_serial/lirc_serial.c
--- lirc-0.9.0-old/drivers/lirc_serial/lirc_serial.c	2016-06-05 17:30:30.000000000 +0200
+++ lirc-0.9.0-new/drivers/lirc_serial/lirc_serial.c	2016-06-05 17:44:20.000000000 +0200
@@ -112,13 +112,13 @@
 static int type;
 static int io;
 static int irq;
-static int iommap;
+static bool iommap;
 static int ioshift;
-static int softcarrier = 1;
-static int share_irq;
-static int debug;
+static bool softcarrier = 1;
+static bool share_irq;
+static bool debug;
 static int sense = -1;	/* -1 = auto, 0 = active high, 1 = active low */
-static int txsense;	/* 0 = active high, 1 = active low */
+static bool txsense;	/* 0 = active high, 1 = active low */
 
 #define dprintk(fmt, args...)					\
 	do {							\
@@ -1321,9 +1321,9 @@
 module_param(share_irq, bool, S_IRUGO);
 MODULE_PARM_DESC(share_irq, "Share interrupts (0 = off, 1 = on)");
 
-module_param(sense, bool, S_IRUGO);
-MODULE_PARM_DESC(sense, "Override autodetection of IR receiver circuit"
-		 " (0 = active high, 1 = active low )");
+//module_param(sense, bool, S_IRUGO);
+//MODULE_PARM_DESC(sense, "Override autodetection of IR receiver circuit"
+//		 " (0 = active high, 1 = active low )");
 
 #ifdef CONFIG_LIRC_SERIAL_TRANSMITTER
 module_param(txsense, bool, S_IRUGO);
diff -Naur lirc-0.9.0-old/drivers/lirc_sir/lirc_sir.c lirc-0.9.0-new/drivers/lirc_sir/lirc_sir.c
--- lirc-0.9.0-old/drivers/lirc_sir/lirc_sir.c	2016-06-05 17:30:30.000000000 +0200
+++ lirc-0.9.0-new/drivers/lirc_sir/lirc_sir.c	2016-06-05 17:45:19.000000000 +0200
@@ -191,7 +191,7 @@
 static unsigned int rx_tail, rx_head;
 static lirc_t tx_buf[WBUF_LEN];
 
-static int debug;
+static bool debug;
 #define dprintk(fmt, args...)						\
 	do {								\
 		if (debug)						\
diff -Naur lirc-0.9.0-old/drivers/lirc_wpc8769l/lirc_wpc8769l.c lirc-0.9.0-new/drivers/lirc_wpc8769l/lirc_wpc8769l.c
--- lirc-0.9.0-old/drivers/lirc_wpc8769l/lirc_wpc8769l.c	2016-06-05 17:30:30.000000000 +0200
+++ lirc-0.9.0-new/drivers/lirc_wpc8769l/lirc_wpc8769l.c	2016-06-05 17:46:43.000000000 +0200
@@ -106,11 +106,11 @@
 static unsigned int irq;
 
 /* Debugging flag. */
-static int debug;
+static bool debug;
 
 /* If true, we skip ACPI autodetection and use the parameter-supplied I/O and
  * IRQ. */
-static int skip_probe;
+static bool skip_probe;
 
 /* Whether the device is open or not. */
 static int lirc_wpc8769l_is_open;
diff -Naur lirc-0.9.0-old/drivers/lirc_xbox/lirc_xbox.c lirc-0.9.0-new/drivers/lirc_xbox/lirc_xbox.c
--- lirc-0.9.0-old/drivers/lirc_xbox/lirc_xbox.c	2016-06-05 17:30:30.000000000 +0200
+++ lirc-0.9.0-new/drivers/lirc_xbox/lirc_xbox.c	2016-06-05 17:48:40.000000000 +0200
@@ -75,9 +75,9 @@
 
 /* module parameters */
 #ifdef CONFIG_USB_DEBUG
-static int debug = 1;
+static bool debug = 1;
 #else
-static int debug;
+static bool debug;
 #endif
 
 #define dprintk(fmt, args...)					\
@@ -93,7 +93,7 @@
 #define USB_BUFF_LEN 6
 
 static int mask = 0xFFFF;	/* channel acceptance bit mask */
-static int unique;		/* enable channel-specific codes */
+static bool unique;		/* enable channel-specific codes */
 static int repeat = 10;		/* repeat time in 1/100 sec */
 static unsigned long repeat_jiffies; /* repeat timeout */
 
