diff -Naur lirc-0.9.0-old/drivers/kcompat.h lirc-0.9.0-new/drivers/kcompat.h
--- lirc-0.9.0-old/drivers/kcompat.h	2011-03-25 23:28:18.000000000 +0100
+++ lirc-0.9.0-new/drivers/kcompat.h	2015-07-08 12:32:45.000000000 +0200
@@ -114,7 +114,7 @@
 /*************************** interrupt.h ******************************/
 /* added in 2.6.18, old defines removed in 2.6.24 */
 #ifndef IRQF_DISABLED
-#define IRQF_DISABLED SA_INTERRUPT
+#define IRQF_DISABLED 0
 #endif
 #ifndef IRQF_SHARED
 #define IRQF_SHARED SA_SHIRQ
