diff -Naur lirc-0.10.1-old/configure.ac lirc-0.10.1-new/configure.ac
--- lirc-0.10.1-old/configure.ac	2017-09-10 10:52:19.000000000 +0200
+++ lirc-0.10.1-new/configure.ac	2018-06-04 19:24:07.071269355 +0200
@@ -43,14 +43,15 @@
 AM_CONDITIONAL(HAVE_DOXYGEN, test x$DOXYGEN = xyes)
 LT_INIT([disable-static])
 
-AM_PATH_PYTHON([3.1],,)
+AM_PATH_PYTHON([2.6],,)
 PKG_CHECK_MODULES([PYTHON],[python-${PYTHON_VERSION}m],,[true])
 test -z "$PYTHON_LIBS" && \
     PKG_CHECK_MODULES([PYTHON], [python-$PYTHON_VERSION],,[true])
 test -n "$PYTHON_LIBS" || \
     AC_MSG_WARN([No python package found (missing devel package?)])
 python_version_nodots=$(echo $PYTHON_VERSION | tr -d '.')
-AM_CONDITIONAL(HAVE_PYTHON35, [test $python_version_nodots -ge 35])
+AM_CONDITIONAL(HAVE_PYTHON35, [test -n "$PYTHON_LIBS" && \
+			       test $python_version_nodots -ge 35])
 CFLAGS="$CFLAGS $PYTHON_CFLAGS"
 
 
diff -Naur lirc-0.10.1-old/Makefile.am lirc-0.10.1-new/Makefile.am
--- lirc-0.10.1-old/Makefile.am	2017-09-10 10:52:19.000000000 +0200
+++ lirc-0.10.1-new/Makefile.am	2018-06-04 19:23:29.541268011 +0200
@@ -104,6 +104,7 @@
 
 libpython               = $(libdir)/python$(PYTHON_VERSION)
 pydir                   = $(libpython)/site-packages/lirc
+if HAVE_PYTHON35
 py_LTLIBRARIES          = python-pkg/lib/_client.la
 python_pkg_lib__client_la_SOURCES = \
                           python-pkg/lirc/_client.c
@@ -116,6 +117,7 @@
                           $(PYTHON_LIBS)
 python_pkg_lib__client_la_LIBADD = \
                           lib/liblirc_client.la
+endif
 
 pylint: .phony
 	$(MAKE) -C tools pylint
