diff -Naur lirc-0.9.0-old/drivers/lirc_imon/lirc_imon.c lirc-0.9.0-new/drivers/lirc_imon/lirc_imon.c
--- lirc-0.9.0-old/drivers/lirc_imon/lirc_imon.c	2011-03-25 15:28:18.000000000 -0700
+++ lirc-0.9.0-new/drivers/lirc_imon/lirc_imon.c	2012-12-06 16:39:59.000000000 -0800
@@ -47,6 +47,8 @@
 #include "drivers/lirc_dev/lirc_dev.h"
 #endif
 
+#undef err
+#define err(f, arg...)  printk(KERN_ERR MOD_NAME": " f "\n" , ## arg)
 
 #define MOD_AUTHOR	"Venky Raju <dev@venky.ws>"
 #define MOD_DESC	"Driver for SoundGraph iMON MultiMedia IR/Display"
diff -Naur lirc-0.9.0-old/drivers/lirc_sasem/lirc_sasem.c lirc-0.9.0-new/drivers/lirc_sasem/lirc_sasem.c
--- lirc-0.9.0-old/drivers/lirc_sasem/lirc_sasem.c	2011-03-25 15:28:18.000000000 -0700
+++ lirc-0.9.0-new/drivers/lirc_sasem/lirc_sasem.c	2012-12-06 16:40:06.000000000 -0800
@@ -58,6 +58,9 @@
 #include "drivers/lirc_dev/lirc_dev.h"
 #endif
 
+#undef err
+#define err(f, arg...)  printk(KERN_ERR MOD_NAME": " f "\n" , ## arg)
+
 
 #define MOD_AUTHOR	"Oliver Stabel <oliver.stabel@gmx.de>, " \
 			"Tim Davies <tim@opensystems.net.au>"
diff -Naur lirc-0.9.0-old/drivers/lirc_ttusbir/lirc_ttusbir.c lirc-0.9.0-new/drivers/lirc_ttusbir/lirc_ttusbir.c
--- lirc-0.9.0-old/drivers/lirc_ttusbir/lirc_ttusbir.c	2011-03-25 15:28:18.000000000 -0700
+++ lirc-0.9.0-new/drivers/lirc_ttusbir/lirc_ttusbir.c	2012-12-06 16:39:24.000000000 -0800
@@ -47,6 +47,9 @@
 #include "drivers/lirc_dev/lirc_dev.h"
 #endif
 
+#undef err
+#define err(f, arg...)  printk(KERN_ERR " lirc_ttusbir: " f "\n" , ## arg)
+
 MODULE_DESCRIPTION("TechnoTrend USB IR device driver for LIRC");
 MODULE_AUTHOR("Stefan Macher (st_maker-lirc@yahoo.de)");
 MODULE_LICENSE("GPL");
