diff -Naur lirc-0.9.0-old/drivers/lirc_atiusb/lirc_atiusb.c lirc-0.9.0-new/drivers/lirc_atiusb/lirc_atiusb.c
--- lirc-0.9.0-old/drivers/lirc_atiusb/lirc_atiusb.c	2011-03-25 15:28:18.000000000 -0700
+++ lirc-0.9.0-new/drivers/lirc_atiusb/lirc_atiusb.c	2011-07-07 06:02:08.000000000 -0700
@@ -1048,7 +1048,7 @@
 		goto new_irctl_failure_check;
 	}
 
-	if (lirc_buffer_init(driver->rbuf, dclen, 1)) {
+	if (lirc_buffer_init(driver->rbuf, dclen, 2)) {
 		mem_failure = 4;
 		goto new_irctl_failure_check;
 	}
