diff -Naur lirc-0.9.0-old/drivers/lirc_serial/lirc_serial.c lirc-0.9.0-new/drivers/lirc_serial/lirc_serial.c
--- lirc-0.9.0-old/drivers/lirc_serial/lirc_serial.c	2013-09-14 00:33:30.000000000 +0200
+++ lirc-0.9.0-new/drivers/lirc_serial/lirc_serial.c	2013-09-14 00:43:12.665237759 +0200
@@ -1102,12 +1102,12 @@
 
 static struct platform_device *lirc_serial_dev;
 
-static int __devinit lirc_serial_probe(struct platform_device *dev)
+static int lirc_serial_probe(struct platform_device *dev)
 {
 	return 0;
 }
 
-static int __devexit lirc_serial_remove(struct platform_device *dev)
+static int lirc_serial_remove(struct platform_device *dev)
 {
 	return 0;
 }
@@ -1158,7 +1158,7 @@
 
 static struct platform_driver lirc_serial_driver = {
 	.probe		= lirc_serial_probe,
-	.remove		= __devexit_p(lirc_serial_remove),
+	.remove         = lirc_serial_remove,
 	.suspend	= lirc_serial_suspend,
 	.resume		= lirc_serial_resume,
 	.driver		= {
diff -Naur lirc-0.9.0-old/drivers/lirc_sir/lirc_sir.c lirc-0.9.0-new/drivers/lirc_sir/lirc_sir.c
--- lirc-0.9.0-old/drivers/lirc_sir/lirc_sir.c	2013-09-14 00:33:30.000000000 +0200
+++ lirc-0.9.0-new/drivers/lirc_sir/lirc_sir.c	2013-09-14 00:47:13.954510667 +0200
@@ -1271,19 +1271,19 @@
 	return 0;
 }
 
-static int __devinit lirc_sir_probe(struct platform_device *dev)
+static int lirc_sir_probe(struct platform_device *dev)
 {
 	return 0;
 }
 
-static int __devexit lirc_sir_remove(struct platform_device *dev)
+static int lirc_sir_remove(struct platform_device *dev)
 {
 	return 0;
 }
 
 static struct platform_driver lirc_sir_driver = {
 	.probe		= lirc_sir_probe,
-	.remove		= __devexit_p(lirc_sir_remove),
+	.remove      = lirc_sir_remove, 
 	.driver		=	{
 		.name	= "lirc_sir",
 		.owner	= THIS_MODULE,
diff -Naur lirc-0.9.0-old/drivers/lirc_wpc8769l/lirc_wpc8769l.c lirc-0.9.0-new/drivers/lirc_wpc8769l/lirc_wpc8769l.c
--- lirc-0.9.0-old/drivers/lirc_wpc8769l/lirc_wpc8769l.c	2013-09-14 00:33:30.000000000 +0200
+++ lirc-0.9.0-new/drivers/lirc_wpc8769l/lirc_wpc8769l.c	2013-09-14 00:49:28.190781971 +0200
@@ -923,12 +923,12 @@
 #ifdef MODULE
 static struct platform_device *lirc_wpc8769l_platform_dev;
 
-static int __devinit lirc_wpc8769l_probe(struct platform_device *dev)
+static int lirc_wpc8769l_probe(struct platform_device *dev)
 {
 	return 0;
 }
 
-static int __devexit lirc_wpc8769l_remove(struct platform_device *dev)
+static int lirc_wpc8769l_remove(struct platform_device *dev)
 {
 	return 0;
 }
@@ -956,7 +956,7 @@
 
 static struct platform_driver lirc_wpc8769l_platform_driver = {
 	.probe		= lirc_wpc8769l_probe,
-	.remove		= __devexit_p(lirc_wpc8769l_remove),
+	.remove         = lirc_wpc8769l_remove,
 	.suspend	= lirc_wpc8769l_suspend,
 	.resume		= lirc_wpc8769l_resume,
 	.driver		= {
