diff -Naur openssh-7.0p1-old/configure.ac openssh-7.0p1-new/configure.ac
--- openssh-7.0p1-old/configure.ac	2015-08-15 16:27:01.029999992 -0700
+++ openssh-7.0p1-new/configure.ac	2015-08-15 16:28:55.038006681 -0700
@@ -1345,27 +1345,27 @@
 
 AC_CHECK_DECLS([GLOB_NOMATCH], , , [#include <glob.h>])
 
-AC_MSG_CHECKING([whether struct dirent allocates space for d_name])
-AC_RUN_IFELSE(
-	[AC_LANG_PROGRAM([[
+AC_CACHE_CHECK(
+	[whether struct dirent allocates space for d_name],
+	[ac_cv_have_space_for_dirent_d_name],
+	[AC_RUN_IFELSE(
+		[AC_LANG_PROGRAM([[
 #include <sys/types.h>
 #include <dirent.h>]],
 	[[
 	struct dirent d;
 	exit(sizeof(d.d_name)<=sizeof(char));
-	]])],
-	[AC_MSG_RESULT([yes])],
-	[
-		AC_MSG_RESULT([no])
-		AC_DEFINE([BROKEN_ONE_BYTE_DIRENT_D_NAME], [1],
-			[Define if your struct dirent expects you to
-			allocate extra space for d_name])
-	],
-	[
-		AC_MSG_WARN([cross compiling: assuming BROKEN_ONE_BYTE_DIRENT_D_NAME])
-		AC_DEFINE([BROKEN_ONE_BYTE_DIRENT_D_NAME])
-	]
-)
+ 		]])],
+ 		[ ac_cv_have_space_for_dirent_d_name="yes" ],
+ 		[ ac_cv_have_space_for_dirent_d_name="no"  ],
+ 		[ ac_cv_have_space_for_dirent_d_name="no"  ]
+	)]
+)
+if test "x$ac_cv_have_space_for_dirent_d_name" = "xno" ; then
+	AC_DEFINE(BROKEN_ONE_BYTE_DIRENT_D_NAME, 1,
+		[Define if your struct dirent expects you to
+		allocate extra space for d_name])
+fi
 
 AC_MSG_CHECKING([for /proc/pid/fd directory])
 if test -d "/proc/$$/fd" ; then
