diff -Naur sway-1.5.1-old/sway/main.c sway-1.5.1-new/sway/main.c
--- sway-1.5.1-old/sway/main.c	2020-11-10 17:47:31.000000000 +0100
+++ sway-1.5.1-new/sway/main.c	2021-01-29 17:25:28.136665623 +0100
@@ -196,11 +196,11 @@
 			return false;
 		}
 	}
-	if (setgid(0) != -1 || setuid(0) != -1) {
+/*	if (setgid(0) != -1 || setuid(0) != -1) {
 		sway_log(SWAY_ERROR, "Unable to drop root (we shouldn't be able to "
 			"restore it after setuid), refusing to start");
 		return false;
-	}
+	} */
 	return true;
 }
 
