diff --speed-large-files --no-dereference --minimal -Naur alsa-utils-1.2.9/alsactl/init_sysdeps.c alsa-utils-1.2.9/alsactl/init_sysdeps.c
--- alsa-utils-1.2.9/alsactl/init_sysdeps.c	2023-05-04 09:16:17.000000000 +0200
+++ alsa-utils-1.2.9/alsactl/init_sysdeps.c	2023-08-01 18:42:16.030786226 +0200
@@ -18,7 +18,7 @@
  */
 
 #if defined(__GLIBC__) && !(defined(__UCLIBC__) && defined(__USE_BSD))
-static size_t strlcpy(char *dst, const char *src, size_t size)
+size_t strlcpy(char *dst, const char *src, size_t size)
 {
 	size_t bytes = 0;
 	char *q = dst;
@@ -37,7 +37,7 @@
 	return bytes;
 }
 
-static size_t strlcat(char *dst, const char *src, size_t size)
+size_t strlcat(char *dst, const char *src, size_t size)
 {
 	size_t bytes = 0;
 	char *q = dst;
