diff -Naur coreutils-8.19-old/Makefile.am coreutils-8.19-new/Makefile.am
--- coreutils-8.19-old/Makefile.am	2012-08-20 09:46:02.000000000 -0700
+++ coreutils-8.19-new/Makefile.am	2012-08-20 09:46:32.000000000 -0700
@@ -17,7 +17,7 @@
 
 ALL_RECURSIVE_TARGETS =
 
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 
 changelog_etc =				\
   ChangeLog-2005			\
diff -Naur coreutils-8.19-old/Makefile.in coreutils-8.19-new/Makefile.in
--- coreutils-8.19-old/Makefile.in	2012-08-20 09:46:02.000000000 -0700
+++ coreutils-8.19-new/Makefile.in	2012-08-20 09:46:32.000000000 -0700
@@ -1694,7 +1694,7 @@
 
 # Some tests always need root privileges, others need them only sometimes.
 ALL_RECURSIVE_TARGETS = install-root check-root distcheck-hook
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 changelog_etc = \
   ChangeLog-2005			\
   ChangeLog-2006			\
