diff --speed-large-files --no-dereference --minimal -Naur u-boot-e5595b468f0c8aaf7493f9085591370c873f178c/lib/avb/libavb/avb_slot_verify.c u-boot-e5595b468f0c8aaf7493f9085591370c873f178c/lib/avb/libavb/avb_slot_verify.c
--- u-boot-e5595b468f0c8aaf7493f9085591370c873f178c/lib/avb/libavb/avb_slot_verify.c	2021-06-03 03:59:03.000000000 +0200
+++ u-boot-e5595b468f0c8aaf7493f9085591370c873f178c/lib/avb/libavb/avb_slot_verify.c	2022-01-01 17:52:35.379924012 +0100
@@ -296,7 +296,7 @@
   bool image_preloaded = false;
   uint8_t* digest;
   size_t digest_len;
-  const char* found;
+  const char* found = NULL;
   uint64_t image_size;
   size_t expected_digest_len = 0;
   uint8_t expected_digest_buf[AVB_SHA512_DIGEST_SIZE];
