diff --speed-large-files --no-dereference --minimal -Naur u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/arch/arm/dts/rk3568.dtsi u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/arch/arm/dts/rk3568.dtsi
--- u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/arch/arm/dts/rk3568.dtsi	2021-06-26 14:56:24.000000000 +0200
+++ u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/arch/arm/dts/rk3568.dtsi	2022-01-26 16:27:34.846207210 +0100
@@ -22,8 +22,8 @@
 	aliases {
 		dsi0 = &dsi0;
 		dsi1 = &dsi1;
-		ethernet0 = &gmac0;
-		ethernet1 = &gmac1;
+		ethernet0 = &gmac1;
+		ethernet1 = &gmac0;
 		gpio0 = &gpio0;
 		gpio1 = &gpio1;
 		gpio2 = &gpio2;
diff --speed-large-files --no-dereference --minimal -Naur u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/arch/arm/dts/rk3568-u-boot.dtsi u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/arch/arm/dts/rk3568-u-boot.dtsi
--- u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/arch/arm/dts/rk3568-u-boot.dtsi	2021-06-26 14:56:24.000000000 +0200
+++ u-boot-rockchip-02ce577f7b7f32eff47bc711680046a7647d3c7a/arch/arm/dts/rk3568-u-boot.dtsi	2022-01-26 16:27:52.199540231 +0100
@@ -6,8 +6,8 @@
 
 / {
 	aliases {
-		ethernet0 = &gmac0;
-		ethernet1 = &gmac1;
+		ethernet0 = &gmac1;
+		ethernet1 = &gmac0;
 		mmc0 = &sdhci;
 		mmc1 = &sdmmc0;
 		mmc2 = &sdmmc1;
