diff --speed-large-files --no-dereference --minimal -Naur Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/base.xml.snow Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/base.xml.snow
--- Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/base.xml.snow	1970-01-01 01:00:00.000000000 +0100
+++ Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/base.xml.snow	2021-12-07 10:57:26.824505644 +0100
@@ -0,0 +1,1819 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE mythuitheme SYSTEM "http://www.mythtv.org/schema/mythuitheme.dtd">
+<mythuitheme>
+
+    <!-- fonts small, medium, large, extra large, extra extra large, extra small -->
+    <fontdef name="basefont" face="Jesaya Free">
+        <pixelsize>12</pixelsize>
+        <color>#FFFFFF</color>
+    </fontdef>
+    <fontdef name="basefont_xs" from="basefont">
+        <pixelsize>24</pixelsize>
+    </fontdef>
+    <fontdef name="basefont_s" from="basefont">
+        <pixelsize>28</pixelsize>
+    </fontdef>
+    <fontdef name="basefont_m" from="basefont">
+        <pixelsize>32</pixelsize>
+    </fontdef>
+    <fontdef name="basefont_l" from="basefont">
+        <pixelsize>36</pixelsize>
+    </fontdef>
+    <fontdef name="basefont_xl" from="basefont">
+        <pixelsize>39</pixelsize>
+        <weight>bold</weight>
+    </fontdef>
+    <!-- colored fonts -->
+    <fontdef name="basefont_l_good" from="basefont_l">
+        <color>#00FF00</color>
+    </fontdef>
+    <fontdef name="basefont_l_bad" from="basefont_l">
+        <color>#FF0000</color>
+    </fontdef>
+    <fontdef name="basefont_l_warn" from="basefont_l">
+        <color>#FFA500</color>
+    </fontdef>
+    <fontdef name="basefont_l_off" from="basefont_l">
+        <color>#7F7F7F</color>
+    </fontdef>
+    <fontdef name="basefont_m_good" from="basefont_m">
+        <color>#00FF00</color>
+    </fontdef>
+    <fontdef name="basefont_m_bad" from="basefont_m">
+        <color>#FF0000</color>
+    </fontdef>
+    <fontdef name="basefont_m_warn" from="basefont_m">
+        <color>#FFA500</color>
+    </fontdef>
+    <fontdef name="basefont_m_off" from="basefont_m">
+        <color>#7F7F7F</color>
+    </fontdef>
+    <fontdef name="basefont_xs_good" from="basefont_xs">
+        <color>#00FF00</color>
+    </fontdef>
+    <fontdef name="basefont_xs_bad" from="basefont_xs">
+        <color>#FF0000</color>
+    </fontdef>
+    <fontdef name="basefont_xs_warn" from="basefont_xs">
+        <color>#FFA500</color>
+    </fontdef>
+
+    <!-- textarea with small, medium, large and extra large font -->
+    <textarea name="basetext">
+        <multiline>no</multiline>
+        <align>left,vcenter</align>
+        <cutdown>yes</cutdown>
+    </textarea>
+    <textarea name="basetext_xs" from="basetext">
+        <font>basefont_xs</font>
+    </textarea>
+    <textarea name="basetext_s" from="basetext">
+        <font>basefont_s</font>
+    </textarea>
+    <textarea name="basetext_m" from="basetext">
+        <font>basefont_m</font>
+    </textarea>
+    <textarea name="basetext_l" from="basetext">
+        <font>basefont_l</font>
+    </textarea>
+    <textarea name="basetext_xl" from="basetext">
+        <font>basefont_xl</font>
+    </textarea>
+
+    <!-- text for titles -->
+    <textarea name="basetitletext" from="basetext_xl"/>
+
+    <textarea name="basedescriptiontext" from="basetext_m">
+        <multiline>yes</multiline>
+        <align>justify</align>
+        <cutdown>yes</cutdown>
+        <alpha>150</alpha>
+    </textarea>
+
+    <!-- text for screen titles -->
+    <textarea name="basepagetitletext">
+        <area>250, 0, -250, 50</area>
+        <font>basefont_l</font>
+        <multiline>no</multiline>
+        <align>allcenter</align>
+        <alpha>150</alpha>
+    </textarea>
+
+    <!-- text for popup window titles -->
+    <textarea name="basepopuptitletext" from="basepagetitletext">
+        <area>0,0,100%,50</area>
+        <alpha>200</alpha>
+    </textarea>
+
+    <!-- help text for settings windows -->
+    <textarea name="basehelptext" from="basetext_s">
+        <area>50%-500,100%-170,1000,70</area>
+        <align>hcenter,bottom</align>
+        <multiline>yes</multiline>
+        <alpha>200</alpha>
+    </textarea>
+
+    <!-- list item text area -->
+    <textarea name="baseitemtext">
+        <font>basefont_l</font>
+        <multiline>no</multiline>
+        <align>left,vcenter</align>
+        <cutdown>yes</cutdown>
+    </textarea>
+
+    <textarea name="baselabeltext" from="basetext_m">
+        <alpha>200</alpha>
+    </textarea>
+
+    <shape name="baseframe">
+        <area>0,0,30,30</area>
+        <type>roundbox</type>
+        <cornerradius>5</cornerradius>
+        <fill color="#000000" alpha="0" />
+        <line color="#FFFFFF" alpha="255" width="2" />
+    </shape>
+
+    <!-- background for preview/screenshot images -->
+    <shape name="basepreviewframe">
+        <type>box</type>
+        <area>0,0,350,505</area>
+        <fill style="gradient">
+            <gradient direction="vertical">
+                <stop position="0"   color="#FFFFFF" alpha="50" />
+                <stop position="30"   color="#FFFFFF" alpha="50" />
+                <stop position="100" color="#FFFFFF" alpha="5" />
+            </gradient>
+        </fill>
+    </shape>
+
+    <!-- backgrounds -->
+    <shape name="basebackground">
+        <area>0,0,100%,100%</area>
+        <type>box</type>
+        <fill color="#000000" alpha="255" />
+        <line color="#000000" alpha="0" width="0" />
+    </shape>
+
+    <imagetype name="baseimagebackground">
+        <area>0,0,100%,100%</area>
+        <procedural name="backimg">
+            <area>0,0,100%,100%</area>
+            <vertexsource>vertex_shader.txt</vertexsource>
+            <fragmentsource>fragment_shader.txt</fragmentsource>
+        </procedural>
+    </imagetype>
+
+    <imagetype name="basedefaultbackground" from="baseimagebackground">
+        <filename>images/Backgrounds/BackgroundDefault.jpg</filename>
+        <alpha>100</alpha>
+    </imagetype>
+
+    <!-- background of popup windows -->
+    <shape name="basepopupbackground" from="baseframe">
+        <fill color="#000000" alpha="255" />
+        <alpha>255</alpha>
+    </shape>
+
+    <!-- horizontal and vertical separators -->
+    <shape name="baseseparator_h">
+        <area>0,0,100%,1</area>
+        <type>box</type>
+        <fill style="gradient">
+            <gradient direction="horizontal">
+                <stop position="0"   color="#FFFFFF" alpha="50" />
+                <stop position="45"  color="#FFFFFF" alpha="200" />
+                <stop position="50"  color="#FFFFFF" alpha="255" />
+                <stop position="55"  color="#FFFFFF" alpha="200" />
+                <stop position="100" color="#FFFFFF" alpha="50" />
+            </gradient>
+        </fill>
+    </shape>
+    <shape name="baseseparator_v">
+        <area>0,0,1,100%</area>
+        <type>box</type>
+        <fill style="gradient">
+            <gradient direction="vertical">
+                <stop position="0"   color="#FFFFFF" alpha="50" />
+                <stop position="45"  color="#FFFFFF" alpha="200" />
+                <stop position="50"  color="#FFFFFF" alpha="255" />
+                <stop position="55"  color="#FFFFFF" alpha="200" />
+                <stop position="100" color="#FFFFFF" alpha="50" />
+            </gradient>
+        </fill>
+    </shape>
+
+    <!-- separator of items in a button list -->
+    <shape name="baseitemseparator_h" from="baseseparator_h">
+        <alpha>50</alpha>
+    </shape>
+
+    <shape name="baseitemseparator_v" from="baseseparator_v">
+        <alpha>50</alpha>
+    </shape>
+
+    <!-- placeholder for missing images like coverart or preview -->
+    <shape name="baseimageplaceholder">
+        <type>box</type>
+        <area>0, 0, 350, 505</area>
+        <fill style="gradient">
+            <gradient direction="vertical">
+                <stop position="0"   color="#FFFFFF" alpha="75" />
+                <stop position="30"  color="#FFFFFF" alpha="75" />
+                <stop position="100" color="#FFFFFF" alpha="25" />
+            </gradient>
+        </fill>
+    </shape>
+
+    <!-- selected background of controls like button, textedit, spinbox -->
+    <shape name="basecontrolbackground" from="basebackground">
+        <area>0,2,100%,-2</area>
+        <fill color="#FFFFFF" alpha="255" />
+        <alpha>25</alpha>
+    </shape>
+
+    <shape name="basebar" from="basebackground">
+        <area>0,0,100%,50</area>
+    </shape>
+
+
+    <!-- the content background -->
+    <group name="basecontent">
+        <area>0, 475, 100%, -50</area>
+        <shape name="contentbar" from="basebar">
+            <area>0, 0, 100%, 100%</area>
+            <alpha>225</alpha>
+        </shape>
+        <shape name="contentsep" from="baseseparator_h">
+            <position>0, 0</position>
+        </shape>
+    </group>
+
+    <!-- points -->
+    <shape name="basepoint_s">
+        <area>0,0,10,10</area>
+        <type>ellipse</type>
+        <fill color="#FFFFFF" alpha="255" />
+        <line color="#000000" alpha="0" width="0" />
+    </shape>
+    <shape name="basepoint_m" from="basepoint_s">
+        <area>0,0,20,20</area>
+    </shape>
+    <shape name="basepoint_l" from="basepoint_s">
+        <area>0,0,30,30</area>
+    </shape>
+
+    <!-- horizontal and vertical lines -->
+    <shape name="baseline_h">
+        <area>0,0,100%,2</area>
+        <type>box</type>
+        <fill color="#FFFFFF" alpha="255" />
+        <line color="#000000" alpha="0" width="0" />
+    </shape>
+    <shape name="baseline_v">
+        <area>0,0,2,100%</area>
+        <type>box</type>
+        <fill color="#FFFFFF" alpha="255" />
+        <line color="#000000" alpha="0" width="0" />
+    </shape>
+
+    <statetype name="basecheckstate">
+        <state type="full">
+            <imagetype name="mark">
+                <filename>images/CheckboxMark.png</filename>
+            </imagetype>
+        </state>
+        <state type="half">
+            <imagetype name="mark">
+                <filename>images/CheckboxMark.png</filename>
+            </imagetype>
+        </state>
+        <state type="off" />
+    </statetype>
+
+    <!-- item list defaults -->
+    <buttonlist name="baseitemlist">
+        <area>0,1000,750,500</area>
+        <searchposition>-1,50</searchposition>
+        <align>top,hcenter</align>
+        <layout>vertical</layout>
+        <wrapstyle>selection</wrapstyle>
+        <arrange>stack</arrange>
+        <buttonarea>0,0,100%,100%</buttonarea>
+        <showarrow>no</showarrow>
+        <statetype name="buttonitem">
+            <area>0,0,100%,55</area>
+            <state name="active">
+                <area>0,0,100%,55</area>
+                <shape name="itemseparator" from="baseitemseparator_h">
+                    <position>0,100%</position>
+                </shape>
+                <textarea name="buttontext" from="baseitemtext">
+                    <area>5,0,-5,100%</area>
+                    <alpha>100</alpha>
+                </textarea>
+            </state>
+            <state name="selectedactive" from="active">
+                <area>0,0,100%,55</area>
+                <imagetype name="buttonarrow">
+                    <position>100%-25,50%-12</position>
+                    <filename>images/ArrowRightBig.png</filename>
+                    <alpha>50</alpha>
+                </imagetype>
+                <shape name="background" from="basecontrolbackground">
+                    <area>0,0,100%,100%</area>
+                </shape>
+                <shape name="itemseparator"/>
+                <textarea name="buttontext">
+                    <scroll direction="horizontal" />
+                    <cutdown>no</cutdown>
+                    <alpha>255</alpha>
+                </textarea>
+                <statetype name="nodetype">
+                    <area>0,0,100%,100%</area>
+                    <state name="subfolder">
+                        <area>0,0,100%,100%</area>
+                        <imagetype name="icon">
+                            <position>100%-25,50%-12</position>
+                            <filename>images/ArrowRightBig.png</filename>
+                            <alpha>50</alpha>
+                        </imagetype>
+                    </state>
+                    <state name="folder" from="subfolder"/>
+                    <state name="upfolder">
+                        <area>0,0,100%,100%</area>
+                        <imagetype name="icon">
+                            <position>100%-30,50%-10</position>
+                            <filename>images/ArrowUpBig.png</filename>
+                            <alpha>50</alpha>
+                        </imagetype>
+                   </state>
+               </statetype>
+            </state>
+            <state name="inactive" from="active"/>
+            <state name="selectedinactive" from="selectedactive">
+                <textarea name="buttontext">
+                    <alpha>100</alpha>
+                </textarea>
+            </state>
+            <state name="disabledactive" from="active"/>
+        </statetype>
+    </buttonlist>
+
+    <!-- item list with check buttons -->
+    <buttonlist name="baseitemchecklist" from="baseitemlist">
+        <area>50%-500,500,1000,-75</area>
+        <statetype name="buttonitem">
+            <area>0,0,100%,55</area>
+            <state name="active">
+                <area>0,0,100%,55</area>
+                <shape name="itemseparator" from="baseitemseparator_h">
+                    <position>0,100%</position>
+                </shape>
+                <statetype name="buttoncheck" from="basecheckstate">
+                    <position>5,11</position>
+                    <imagetype name="frame">
+                        <filename>images/CheckboxFrame.png</filename>
+                    </imagetype>
+                    <alpha>100</alpha>
+                </statetype>
+                <textarea name="buttontext">
+                    <area>60,0,-45,100%</area>
+                    <alpha>100</alpha>
+                </textarea>
+            </state>
+            <state name="selectedactive" from="active">
+                <area>0,0,100%,55</area>
+                <shape name="background" from="basecontrolbackground">
+                    <area>0,0,100%,100%</area>
+                </shape>
+                <shape name="itemseparator"/>
+                <statetype name="buttoncheck">
+                    <alpha>255</alpha>
+                </statetype>
+                <textarea name="buttontext">
+                    <alpha>255</alpha>
+                </textarea>
+            </state>
+            <state name="inactive" from="active"/>
+            <state name="selectedinactive" from="selectedactive"/>
+            <state name="disabledactive" from="active"/>
+        </statetype>
+    </buttonlist>
+
+    <!-- single line button list, used like a combobox -->
+    <buttonlist name="baseselector" from="baseitemlist">
+        <area>0,0,250,40</area>
+        <buttonarea>0,0,100%,40</buttonarea>
+        <layout>horizontal</layout>
+        <statetype name="buttonitem">
+            <area>0,0,100%,40</area>
+            <state name="selectedactive">
+                <area>0,0,100%,40</area>
+                <textarea name="buttontext" from="basetext_m">
+                    <area>5,0,-20,100%</area>
+                    <scroll direction="horizontal" />
+                    <cutdown>no</cutdown>
+                    <alpha>255</alpha>
+                </textarea>
+                <shape name="itemseparator">
+                    <alpha>150</alpha>
+                </shape>
+            </state>
+            <state name="active" from="selectedactive">
+                <area>0,0,100%,40</area>
+                <!-- disable highlighting background when not selected -->
+                <shape name="background">
+                    <alpha>0</alpha>
+                </shape>
+                <textarea name="buttontext" from="basetext_m">
+                    <area>5,0,-20,100%</area>
+                    <alpha>150</alpha>
+                </textarea>
+            </state>
+            <state name="inactive" from="active">
+                <textarea name="buttontext">
+                    <alpha>50</alpha>
+                </textarea>
+            </state>
+            <state name="selectedinactive" from="active">
+                <textarea name="buttontext">
+                    <alpha>100</alpha>
+                </textarea>
+            </state>
+        </statetype>
+        <statetype name="downscrollarrow">
+            <position>100%-15,50%-10</position>
+            <state type="full">
+                <imagetype name="down">
+                    <filename>images/ArrowRight.png</filename>
+                </imagetype>
+            </state>
+            <state type="off">
+                <imagetype name="down">
+                    <filename>images/ArrowRight.png</filename>
+                    <alpha>100</alpha>
+                </imagetype>
+            </state>
+        </statetype>
+        <statetype name="upscrollarrow">
+            <position>100%-15,50%+2</position>
+            <state type="full">
+                <imagetype name="up">
+                    <filename>images/ArrowLeft.png</filename>
+                </imagetype>
+            </state>
+            <state type="off">
+                <imagetype name="up">
+                    <filename>images/ArrowLeft.png</filename>
+                    <alpha>100</alpha>
+                </imagetype>
+            </state>
+        </statetype>
+    </buttonlist>
+
+    <imagetype name="baseicon">
+        <area>0,0,32,32</area>
+        <alpha>200</alpha>
+        <preserveaspect>yes</preserveaspect>
+    </imagetype>
+
+    <imagetype name="basecoverarticon">
+        <area>50%-50,50%-50,100,100</area>
+        <filename>images/icons/Movie.png</filename>
+        <alpha>50</alpha>
+    </imagetype>
+
+    <imagetype name="basebusyanimation">
+        <filepattern low="1" high="8">images/icons/animated/Spinner%1.png</filepattern>
+        <delay>75</delay>
+    </imagetype>
+
+    <!-- progress bar -->
+    <progressbar name="baseprogress">
+        <area>0,0,700,20</area>
+        <layout>horizontal</layout>
+        <style>reveal</style>
+        <shape name="background">
+            <area>0,0,100%,100%</area>
+            <type>roundbox</type>
+            <cornerradius>5</cornerradius>
+            <line color="#FFFFFF" alpha="255" width="1"/>
+            <alpha>200</alpha>
+        </shape>
+        <shape name="progressimage">
+            <area>5,5,-5,-5</area>
+            <type>roundbox</type>
+            <cornerradius>5</cornerradius>
+            <fill color="#FFFFFF" alpha="255" />
+            <alpha>50</alpha>
+        </shape>
+    </progressbar>
+    <progressbar name="baseprogress_s">
+        <area>0,0,100%,12</area>
+        <layout>horizontal</layout>
+        <style>reveal</style>
+        <shape name="background">
+            <area>0,0,100%,100%</area>
+            <type>roundbox</type>
+            <cornerradius>2</cornerradius>
+            <fill color="#323232" alpha="255" />
+        </shape>
+        <shape name="progressimage">
+            <area>2,3,-2,-3</area>
+            <type>roundbox</type>
+            <cornerradius>2</cornerradius>
+            <fill color="#969696" alpha="255" />
+        </shape>
+    </progressbar>
+
+    <!-- scroll bar -->
+    <scrollbar name="basescrollbar_v">
+        <area>100%-15,0,15,100%</area>
+        <layout>vertical</layout>
+        <shape name="background">
+            <area>0,0,100%,100%</area>
+            <type>roundbox</type>
+            <cornerradius>3</cornerradius>
+            <fill color="#323232" alpha="255" />
+        </shape>
+        <shape name="slider">
+            <area>0,0,100%,10</area>
+            <type>roundbox</type>
+            <cornerradius>3</cornerradius>
+            <fill color="#969696" alpha="255" />
+            <line color="#000000" alpha="0" width="3"/>
+        </shape>
+    </scrollbar>
+
+    <scrollbar name="basescrollbar_h">
+        <area>0,100%-15,100%,15</area>
+        <layout>horizontal</layout>
+        <shape name="background">
+            <area>0,0,100%,100%</area>
+            <type>roundbox</type>
+            <cornerradius>3</cornerradius>
+            <fill color="#323232" alpha="255" />
+        </shape>
+        <shape name="slider">
+            <area>0,0,10,100%</area>
+            <type>roundbox</type>
+            <cornerradius>3</cornerradius>
+            <fill color="#969696" alpha="255" />
+            <line color="#000000" alpha="0" width="3"/>
+        </shape>
+    </scrollbar>
+
+    <!-- text underline for text controls like textedit and spinbox -->
+    <group name="basetextunderline">
+        <area>0,0,150,30</area>
+        <shape name="leftborder" from="baseline_v">
+            <area>0,100%-11,2,10</area>
+        </shape>
+        <shape name="rightborder" from="baseline_v">
+            <area>100%-2,100%-11,2,10</area>
+        </shape>
+        <shape name="bottomborder1" from="baseline_h">
+            <area>0,100%-2,100%,2</area>
+        </shape>
+    </group>
+
+    <!-- checkbox -->
+    <checkbox name="basecheckbox">
+        <area>0,0,25,25</area>
+        <statetype name="background">
+            <state name="active">
+                <alpha>100</alpha>
+                <imagetype name="frame">
+                    <filename>images/CheckboxFrame.png</filename>
+                </imagetype>
+            </state>
+            <state name="selected" from="active">
+                <alpha>255</alpha>
+            </state>
+            <state name="disabled" from="active">
+                <alpha>50</alpha>
+            </state>
+        </statetype>
+        <statetype name="checkstate" from="basecheckstate">
+            <alpha>200</alpha>
+        </statetype>
+    </checkbox>
+
+    <!-- button -->
+    <button name="basebutton">
+        <area>0,0,200,40</area>
+        <statetype name="buttonstate">
+            <area>0,0,100%,100%</area>
+            <state name="active">
+                <area>0,0,100%,100%</area>
+                <shape name="background" from="basecontrolbackground">
+                    <alpha>10</alpha>
+                </shape>
+                <shape name="topborder" from="baseseparator_h">
+                    <position>0,0</position>
+                    <alpha>100</alpha>
+                </shape>
+                <shape name="bottomborder" from="baseseparator_h">
+                    <position>0,100%-1</position>
+                    <alpha>100</alpha>
+                </shape>
+                <textarea name="text" from="basetext_m">
+                    <area>0,0,100%,100%</area>
+                    <align>allcenter</align>
+                    <alpha>150</alpha>
+                </textarea>
+            </state>
+            <state name="selected" from="active">
+                <shape name="background" from="basecontrolbackground"/>
+                <shape name="topborder">
+                    <alpha>200</alpha>
+                </shape>
+                <shape name="bottomborder">
+                    <alpha>200</alpha>
+                </shape>
+                <textarea name="text">
+                    <scroll direction="horizontal" />
+                    <cutdown>no</cutdown>
+                    <alpha>255</alpha>
+                </textarea>
+            </state>
+            <state name="disabled" from="active">
+                <textarea name="text">
+                    <alpha>50</alpha>
+                </textarea>
+            </state>
+            <state name="pushed" from="selected"/>
+        </statetype>
+    </button>
+
+    <!-- search button -->
+    <button name="basesearchbutton" from="basebutton">
+        <area>0,0,100,40</area>
+        <statetype name="buttonstate">
+            <area>0,0,100%,100%</area>
+            <state name="active">
+                <textarea name="text">
+                    <alpha>0</alpha>
+                </textarea>
+                <imagetype name="icon" from="baseicon">
+                    <area>50%-16,50%-16,32,32</area>
+                    <filename>images/icons/Search.png</filename>
+                    <alpha>100</alpha>
+                </imagetype>
+            </state>
+            <state name="selected" from="active">
+                <shape name="background" from="basecontrolbackground"/>
+                <imagetype name="icon">
+                    <alpha>255</alpha>
+                </imagetype>
+            </state>
+            <state name="disabled" from="active">
+                <imagetype name="icon">
+                    <alpha>50</alpha>
+                </imagetype>
+            </state>
+            <state name="pushed" from="selected"/>
+        </statetype>
+    </button>
+
+    <!-- file selector back button -->
+    <button name="basebackbutton" from="basebutton">
+        <area>0,0,75,40</area>
+        <statetype name="buttonstate">
+            <area>0,0,100%,100%</area>
+            <state name="active">
+                <textarea name="text">
+                    <alpha>0</alpha>
+                </textarea>
+                <imagetype name="icon" from="baseicon">
+                    <area>50%-12,50%-12,20,24</area>
+                    <filename>images/ArrowLeftBig.png</filename>
+                    <alpha>100</alpha>
+                </imagetype>
+            </state>
+            <state name="selected" from="active">
+                <shape name="background" from="basecontrolbackground"/>
+                <imagetype name="icon">
+                    <alpha>255</alpha>
+                </imagetype>
+            </state>
+            <state name="disabled" from="active">
+                <imagetype name="icon">
+                    <alpha>50</alpha>
+                </imagetype>
+            </state>
+            <state name="pushed" from="selected"/>
+        </statetype>
+    </button>
+
+    <!-- home button -->
+    <button name="basehomebutton" from="basebutton">
+        <area>0,0,75,40</area>
+        <statetype name="buttonstate">
+            <area>0,0,100%,100%</area>
+            <state name="active">
+                <textarea name="text">
+                    <alpha>0</alpha>
+                </textarea>
+                <imagetype name="icon" from="baseicon">
+                    <area>50%-16,50%-15,32,32</area>
+                    <filename>images/icons/Home.png</filename>
+                    <alpha>100</alpha>
+                </imagetype>
+            </state>
+            <state name="selected" from="active">
+                <shape name="background" from="basecontrolbackground"/>
+                <imagetype name="icon">
+                    <alpha>255</alpha>
+                </imagetype>
+            </state>
+            <state name="disabled" from="active">
+                <imagetype name="icon">
+                    <alpha>50</alpha>
+                </imagetype>
+            </state>
+            <state name="pushed" from="selected"/>
+        </statetype>
+    </button>
+
+    <!-- single line textedit -->
+    <textedit name="basetextedit">
+        <area>0,0,300,32</area>
+        <statetype name="background">
+            <area>0,0,100%,100%</area>
+            <state name="active">
+                <area>0,0,100%,100%</area>
+                <group name="textunderline" from="basetextunderline">
+                    <area>0,0,100%,100%</area>
+                    <alpha>100</alpha>
+                </group>
+            </state>
+            <state name="selected" from="active">
+                <shape name="background" from="basecontrolbackground"/>
+            </state>
+            <state name="inactive" from="active">
+            </state>
+        </statetype>
+        <imagetype name="cursor">
+            <filename>images/Cursor.png</filename>
+            <area>0,0,2,25</area>
+        </imagetype>
+        <textarea name="text" from="basetext_m">
+            <area>5,0,-5,100%</area>
+            <cutdown>no</cutdown>
+            <alpha>150</alpha>
+        </textarea>
+    </textedit>
+
+    <!-- multi line textedit -->
+    <textedit name="basetexteditmulti">
+        <area>0,0,300,100</area>
+        <statetype name="background">
+            <area>0,0,100%,100%</area>
+            <state name="active">
+                <area>0,0,100%,100%</area>
+                <group name="textunderline" from="basetextunderline">
+                    <area>0,0,100%,100%</area>
+                    <alpha>100</alpha>
+                </group>
+            </state>
+            <state name="selected" from="active">
+                <shape name="background" from="basecontrolbackground"/>
+            </state>
+            <state name="inactive" from="active">
+            </state>
+        </statetype>
+        <imagetype name="cursor">
+            <filename>images/Cursor.png</filename>
+            <area>0,0,2,25</area>
+        </imagetype>
+        <textarea name="text" from="basetext_m">
+            <area>5,0,-5,100%</area>
+            <multiline>yes</multiline>
+            <cutdown>no</cutdown>
+            <align>left,bottom</align>
+            <alpha>150</alpha>
+        </textarea>
+    </textedit>
+
+    <!-- spinbox -->
+    <spinbox name="basespinbox">
+        <area>0,0,150,30</area>
+        <layout>horizontal</layout>
+        <statetype name="buttonitem">
+            <area>0,0,100%,100%</area>
+            <state name="active">
+                <area>0,0,100%-15,100%</area>
+                <textarea name="buttontext" from="basetext_m">
+                    <area>5,1,-20,-1</area>
+                    <alpha>150</alpha>
+                </textarea>
+                <shape name="line" from="baseitemseparator_h">
+                    <position>0,100%</position>
+                    <alpha>150</alpha>
+                </shape>
+            </state>
+            <state name="selectedactive" from="active">
+                <area>0,0,100%,100%</area>
+                <shape name="background" from="basecontrolbackground"/>
+                <textarea name="buttontext">
+                    <scroll direction="horizontal" />
+                    <cutdown>no</cutdown>
+                    <alpha>255</alpha>
+                </textarea>
+            </state>
+            <state name="selectedinactive" from="active">
+                <area>0,0,100%,100%</area>
+                <textarea name="buttontext">
+                    <alpha>100</alpha>
+                </textarea>
+            </state>
+            <state name="inactive" from="active">
+                <textarea name="buttontext">
+                    <alpha>50</alpha>
+                </textarea>
+            </state>
+        </statetype>
+        <!-- strangely, up is handled by the left key, and down by the right key when in horizontal layout -->
+        <statetype name="downscrollarrow">
+            <position>100%-15,50%-10</position>
+            <state type="full">
+                <imagetype name="down">
+                    <filename>images/ArrowRight.png</filename>
+                </imagetype>
+            </state>
+            <state type="off">
+                <imagetype name="down">
+                    <filename>images/ArrowRight.png</filename>
+                    <alpha>100</alpha>
+                </imagetype>
+            </state>
+        </statetype>
+        <statetype name="upscrollarrow">
+            <position>100%-15,50%+2</position>
+            <state type="full">
+                <imagetype name="up">
+                    <filename>images/ArrowLeft.png</filename>
+                </imagetype>
+            </state>
+            <state type="off">
+                <imagetype name="up">
+                    <filename>images/ArrowLeft.png</filename>
+                    <alpha>100</alpha>
+                </imagetype>
+            </state>
+        </statetype>
+    </spinbox>
+
+    <!-- webbrowser -->
+    <webbrowser name="basebrowser">
+        <area>0,0,100%,100%</area>
+        <browserarea>0,0,100%-30,100%-30</browserarea>
+        <zoom>1.0</zoom>
+        <background color="#000000"/>
+        <scrollbar name="vertscrollbar" from="basescrollbar_v">
+            <area>100%-12,0,12,100%-12</area>
+        </scrollbar>
+        <scrollbar name="horizscrollbar" from="basescrollbar_h">
+            <area>0,100%-12,100%-12,12</area>
+        </scrollbar>
+    </webbrowser>
+
+    <!-- recording job state (recording, commflagging, transcoding) -->
+    <statetype name="basejobstate">
+        <area>0,0,32,32</area>
+        <state name="recording">
+            <imagetype name="animation">
+                <area>0,0,32,32</area>
+                <filepattern low="1" high="8">images/icons/animated/Recording%1.png</filepattern>
+                <delay>75</delay>
+            </imagetype>
+        </state>
+        <state name="commflagging">
+            <imagetype name="animation">
+                <area>0,0,32,32</area>
+                <filepattern low="1" high="8">images/icons/animated/CommFlagging%1.png</filepattern>
+                <delay>75</delay>
+            </imagetype>
+        </state>
+        <state name="transcoding">
+            <imagetype name="animation">
+                <area>0,0,32,32</area>
+                <filepattern low="1" high="8">images/icons/animated/Transcoding%1.png</filepattern>
+                <delay>75</delay>
+            </imagetype>
+        </state>
+    </statetype>
+
+    <statetype name="basewatchedstate">
+        <area>0,0,32,32</area>
+        <state name="yes">
+            <area>0,0,100%,100%</area>
+            <imagetype name="icon" from="baseicon">
+                <position>0,0</position>
+                <filename>images/icons/Watched.png</filename>
+            </imagetype>
+        </state>
+    </statetype>
+
+    <statetype name="baseplayliststate">
+        <area>0,0,32,32</area>
+        <state name="yes">
+            <area>0,0,100%,100%</area>
+            <imagetype name="icon" from="baseicon">
+                <position>0,0</position>
+                <filename>images/icons/Playlist.png</filename>
+            </imagetype>
+        </state>
+    </statetype>
+
+    <statetype name="basestarrating">
+        <area>0,0,175,32</area>
+        <alpha>200</alpha>
+        <state name="0">
+            <imagetype name="0">
+                <area>0,0,175,32</area>
+                <filename>images/icons/Rating00.png</filename>
+            </imagetype>
+        </state>
+        <state name="1">
+            <imagetype name="1">
+                <area>0,0,175,32</area>
+                <filename>images/icons/Rating01.png</filename>
+            </imagetype>
+        </state>
+        <state name="2">
+            <imagetype name="2">
+                <area>0,0,175,32</area>
+                <filename>images/icons/Rating02.png</filename>
+            </imagetype>
+        </state>
+        <state name="3">
+            <imagetype name="3">
+                <area>0,0,175,32</area>
+                <filename>images/icons/Rating03.png</filename>
+            </imagetype>
+        </state>
+        <state name="4">
+            <imagetype name="4">
+                <area>0,0,175,32</area>
+                <filename>images/icons/Rating04.png</filename>
+            </imagetype>
+        </state>
+        <state name="5">
+            <imagetype name="5">
+                <area>0,0,175,32</area>
+                <filename>images/icons/Rating05.png</filename>
+            </imagetype>
+        </state>
+        <state name="6">
+            <imagetype name="6">
+                <area>0,0,175,32</area>
+                <filename>images/icons/Rating06.png</filename>
+            </imagetype>
+        </state>
+        <state name="7">
+            <imagetype name="7">
+                <area>0,0,175,32</area>
+                <filename>images/icons/Rating07.png</filename>
+            </imagetype>
+        </state>
+        <state name="8">
+            <imagetype name="8">
+                <area>0,0,175,32</area>
+                <filename>images/icons/Rating08.png</filename>
+            </imagetype>
+        </state>
+        <state name="9">
+            <imagetype name="9">
+                <area>0,0,175,32</area>
+                <filename>images/icons/Rating09.png</filename>
+            </imagetype>
+        </state>
+        <state name="10">
+            <imagetype name="10">
+                <area>0,0,175,32</area>
+                <filename>images/icons/Rating10.png</filename>
+            </imagetype>
+        </state>
+    </statetype>
+
+    <group name="baseuserrating">
+        <area>0,0,100,32</area>
+        <textarea name="userrating" from="basetext_s">
+            <area>0,0,100%-40,32</area>
+            <alpha>150</alpha>
+            <align>right,vcenter</align>
+        </textarea>
+        <imagetype name="heart" from="baseicon" depends="userrating">
+            <position>100%-32,0</position>
+            <filename>images/icons/Heart.png</filename>
+        </imagetype>
+    </group>
+
+    <!-- icon based video facts -->
+    <group name="basevideoiconfacts">
+        <area>0,0,100%,100%</area>
+        <imagetype name="time" from="baseicon" depends="length">
+            <position>0,100%-122</position>
+            <filename>images/icons/Clock.png</filename>
+        </imagetype>
+        <textarea name="length" from="basetext_s">
+            <area>50,100%-122,-50,32</area>
+            <alpha>150</alpha>
+        </textarea>
+        <imagetype name="calendar" from="baseicon" depends="year">
+            <position>0,100%-77</position>
+            <filename>images/icons/Calendar.png</filename>
+        </imagetype>
+        <textarea name="year" from="basetext_s">
+            <area>50,100%-77,-50,32</area>
+            <alpha>150</alpha>
+        </textarea>
+        <imagetype name="ratinglabel" from="baseicon" depends="rating">
+            <position>0,100%-32</position>
+            <filename>images/icons/ParentalRating.png</filename>
+        </imagetype>
+        <textarea name="rating" from="basetext_s">
+            <area>50,100%-32,-200,32</area>
+            <alpha>150</alpha>
+        </textarea>
+        <statetype name="watchedstate" from="basewatchedstate">
+            <position>100%-32,100%-122</position>
+        </statetype>
+        <group name="userrating" from="baseuserrating">
+            <position>100%-100,100%-77</position>
+        </group>
+        <statetype name="userratingstate" from="basestarrating">
+            <position>100%-175,100%-32</position>
+        </statetype>
+    </group>
+
+    <!-- text based video facts -->
+    <group name="basevideotextfacts">
+        <area>0,0,100%,125</area>
+        <textarea name="genrelabel" from="baselabeltext" depends="genres">
+            <area>0,100%-122,150,30</area>
+            <value>Genres</value>
+        </textarea>
+        <textarea name="genres" from="basetext_s">
+            <area>160,100%-122,100%,30</area>
+            <alpha>150</alpha>
+        </textarea>
+        <textarea name="directorlabel" from="baselabeltext" depends="director">
+            <area>0,100%-77,150,30</area>
+            <value>Director</value>
+        </textarea>
+        <textarea name="director" from="basetext_s">
+            <area>160,100%-77,350,30</area>
+            <alpha>150</alpha>
+        </textarea>
+        <textarea name="castlabel" from="baselabeltext" depends="cast">
+            <area>0,100%-32,150,32</area>
+            <value>Cast</value>
+        </textarea>
+        <textarea name="cast" from="basetext_s">
+            <area>160,100%-32,100%,32</area>
+            <alpha>150</alpha>
+            <scroll direction="left"/>
+            <cutdown>no</cutdown>
+            <align>top,left</align>
+        </textarea>
+    </group>
+
+    <!-- icon based schedule facts -->
+    <group name="basescheduleiconfacts">
+        <area>0,0,100%,120</area>
+        <imagetype name="sender" from="baseicon" depends="longchannel">
+            <position>0,0</position>
+            <filename>images/icons/Sender.png</filename>
+        </imagetype>
+        <textarea name="longchannel" from="basetext_s">
+            <area>50,0,100%,32</area>
+            <alpha>150</alpha>
+        </textarea>
+        <imagetype name="aried" from="baseicon" depends="timedate">
+            <position>0,40</position>
+            <filename>images/icons/Clock.png</filename>
+        </imagetype>
+        <textarea name="timedate" from="basetext_s">
+            <area>50,40,100%,32</area>
+            <alpha>150</alpha>
+            <template>%TIMEDATE% (%LENMINS%)</template>
+        </textarea>
+        <imagetype name="record" from="baseicon" depends="rectypestatus">
+            <position>0,80</position>
+            <filename>images/icons/Record.png</filename>
+        </imagetype>
+        <textarea name="rectypestatus" from="basetext_s">
+            <area>50,80,100%,32</area>
+            <alpha>150</alpha>
+        </textarea>
+    </group>
+
+    <group name="basescheduleiconfactsext">
+        <area>0,0,100%,120</area>
+        <imagetype name="idicon" from="baseicon" depends="programid">
+            <position>0,0</position>
+            <filename>images/icons/Number.png</filename>
+        </imagetype>
+        <textarea name="programid" from="basetext_s">
+            <area>50,0,100%,32</area>
+            <alpha>150</alpha>
+        </textarea>
+        <imagetype name="group" from="baseicon" depends="recgroup">
+            <position>0,40</position>
+            <filename>images/icons/Group.png</filename>
+        </imagetype>
+        <textarea name="recgroup" from="basetext_s">
+            <area>50,40,100%,32</area>
+            <alpha>150</alpha>
+        </textarea>
+        <imagetype name="storageicon" from="baseicon" depends="storagegroup">
+            <position>0,80</position>
+            <filename>images/icons/Storage.png</filename>
+        </imagetype>
+        <textarea name="storagegroup" from="basetext_s">
+            <area>50,80,100%,32</area>
+            <alpha>150</alpha>
+        </textarea>
+    </group>
+
+    <!-- title text for selected item, e.g program, recording, video -->
+    <textarea name="basetitle" from="basetext_l">
+        <area>0,0,100%,50</area>
+        <scroll direction="horizontal"/>
+        <cutdown>no</cutdown>
+        <template>%TITLE| %%~ |00x00| %%~ |SUBTITLE%</template>
+    </textarea>
+
+    <!-- title + description for selected item -->
+    <group name="basetitledesc">
+        <textarea name="title" from="basetitle"/>
+        <textarea name="description" from="basedescriptiontext">
+            <area>0,65,100%,100%</area>
+            <align>justify,top</align>
+        </textarea>
+    </group>
+
+    <!--====================================================================-->
+
+    <!-- default screen header -->
+    <group name="baseheader">
+        <area>0, 0, 100%, 50</area>
+        <shape name="headerbar" from="basebar" />
+        <shape name="headersep" from="baseseparator_h">
+            <position>0,50</position>
+        </shape>
+        <clock name="headerdate">
+            <area>25,0,250,50</area>
+            <font>basefont_m</font>
+            <align>left,vcenter</align>
+            <alpha>100</alpha>
+            <template>%DATE%</template>
+        </clock>
+        <clock name="headertime">
+            <area>1645,0,250,50</area>
+            <font>basefont_m</font>
+            <align>right,vcenter</align>
+            <alpha>100</alpha>
+            <template>%TIME%</template>
+        </clock>
+    </group>
+
+    <!-- default screen footer -->
+    <group name="basefooter">
+        <area>0, 100%-50, 100%, 50</area>
+        <shape name="footerbar" from="basebar">
+            <area>0,0,100%,100%</area>
+        </shape>
+        <shape name="footersep" from="baseseparator_h">
+            <position>0, 0</position>
+        </shape>
+    </group>
+
+    <!-- header for popup windows -->
+    <group name="basepopupheader">
+        <area>0,0,100%,100%</area>
+        <!-- make titlebar 50px higher to accomodate animation overshooting -->
+        <shape name="titlebar" from="basebar">
+            <area>0,0,100%,100</area>
+        </shape>
+        <shape name="frame" from="basepopupbackground">
+            <area>0,0,100%,100%</area>
+            <alpha>240</alpha>
+        </shape>
+        <shape name="titleseparator" from="baseseparator_h">
+            <position>0,99</position>
+        </shape>
+    </group>
+
+    <!-- decoration for popup windows -->
+    <group name="basepopupdecoration">
+        <area>0,0,100%,100%</area>
+        <group name="header" from="basepopupheader">
+            <area>0,-50,100%,100%</area>
+        </group>
+        <!-- content goes here -->
+    </group>
+
+    <!-- decoration for popup windows with buttons -->
+    <group name="basebuttonpopupdecoration" from="basepopupdecoration">
+        <area>0,0,100%,100%</area>
+        <!-- content goes here -->
+        <shape name="buttonseparator" from="baseseparator_h">
+            <area>0,100%-90,100%,1</area>
+        </shape>
+        <!-- buttons go here -->
+    </group>
+
+    <!-- basic window decoration (header + footer) -->
+    <group name="basewindowdecoration">
+        <area>0,0,100%,100%</area>
+        <shape name="background" from="basebackground"/>
+        <group name="pageheader" from="baseheader" />
+        <!-- content goes here -->
+        <group name="pagefooter" from="basefooter" />
+    </group>
+
+    <!-- content window decoration (header + content bar + footer) -->
+    <group name="basecontentdecoration">
+        <area>0,0,100%,100%</area>
+        <group name="pageheader" from="baseheader" />
+        <group name="pagecontent" from="basecontent" />
+        <!-- content goes here -->
+        <group name="pagefooter" from="basefooter" />
+    </group>
+
+    <!-- settings window decoration (header + big footer for buttons) -->
+    <group name="basesettingsdecoration">
+        <area>0,0,100%,100%</area>
+        <imagetype name="pagebackground" from="basedefaultbackground"/>
+        <shape name="contentbar" from="basebar">
+            <area>0, 0, 100%, 100%</area>
+            <alpha>225</alpha>
+        </shape>
+        <group name="pageheader" from="baseheader" />
+        <!-- content goes here -->
+        <group name="pagefooter" from="basefooter">
+            <area>0,100%-90,100%,100%</area>
+        </group>
+        <!-- buttons go here -->
+    </group>
+
+    <!-- editor window decoration (header + content bar + big footer for buttons) -->
+    <group name="baseeditordecoration" from="basecontentdecoration">
+        <group name="pagecontent">
+            <area>0,440,100%,100%</area>
+        </group>
+        <group name="pagefooter">
+            <area>0,100%-90,100%,100%</area>
+        </group>
+    </group>
+
+    <!--====================================================================-->
+
+    <!-- default background window -->
+    <window name="backgroundwindow">
+        <imagetype name="backimg">
+            <filename>images/Backgrounds/BackgroundDefault.jpg</filename>
+        </imagetype>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="videowindow">
+        <area>0,0,100%,100%</area>
+        <shape name="background" from="basebackground"/>
+        <textarea name="message" from="basetext_xl">
+            <area>50%-200,50%-20,400,40</area>
+            <align>center</align>
+            <alpha>200</alpha>
+            <value>Playback starting...</value>
+        </textarea>
+    </window>
+
+    <!--====================================================================-->
+
+    <!-- Black background for OpenMAX videos to prevent bleed through-->
+    <window name="videobackground">
+        <shape name="background" from="basebackground"/>
+    </window>
+
+    <!--====================================================================-->
+
+    <!-- busy dialog -->
+    <window name="MythBusyDialog">
+        <area>-1,0,750,150</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-150" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basepopupdecoration"/>
+        <group name="content">
+            <area>25,75,-25,-25</area>
+            <imagetype name="animation" from="basebusyanimation">
+                <area>0,50%-25,50,50</area>
+            </imagetype>
+            <textarea name="message" from="basetext_m">
+                <area>50,0,100%,100%</area>
+                <align>allcenter</align>
+                <cutdown>yes</cutdown>
+                <multiline>yes</multiline>
+            </textarea>
+        </group>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="MythPopupBox">
+        <area>-1,0,600,400</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-400" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basepopupdecoration"/>
+        <textarea name="title" from="basepopuptitletext"/>
+        <textarea name="messagearea" from="basetext_m">
+            <area>50,70,-50,110</area>
+            <align>left,vcenter</align>
+            <cutdown>yes</cutdown>
+            <multiline>yes</multiline>
+        </textarea>
+        <shape name="separator" from="baseseparator_h">
+            <position>0,195</position>
+        </shape>
+        <buttonlist name="list" from="baseitemlist">
+            <align>vcenter</align>
+            <area>50,210,-50,-25</area>
+        </buttonlist>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="MythSearchListDialog">
+        <area>-1,0,750,240</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-400" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basebuttonpopupdecoration"/>
+        <textarea name="title" from="basepopuptitletext">
+            <value>Search List</value>
+        </textarea>
+        <statetype name="searchstate">
+            <area>100%-65,84,32,32</area>
+            <state name="found">
+                <area>0,0,32,32</area>
+                <imagetype name="foundicon" from="baseicon">
+                    <position>0,0</position>
+                    <alpha>100</alpha>
+                    <filename>images/icons/Watched.png</filename>
+                </imagetype>
+            </state>
+            <state name="notfound">
+                <area>0,0,32,32</area>
+                <imagetype name="notfoundicon" from="baseicon">
+                    <position>0,0</position>
+                    <alpha>100</alpha>
+                    <filename>images/icons/Cross.png</filename>
+                </imagetype>
+            </state>
+        </statetype>
+        <textedit name="searchedit" from="basetextedit">
+            <area>25,80,-25,40</area>
+        </textedit>
+        <!-- bottom part with buttons -->
+        <button name="prevbutton" from="basebutton">
+            <area>50%-225,100%-65,200,40</area>
+            <value>Previous</value>
+        </button>
+        <button name="nextbutton" from="basebutton">
+            <area>50%+25,100%-65,200,40</area>
+            <value>Next</value>
+        </button>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="MythConfirmationDialog">
+        <area>-1,0,800,350</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-400" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basebuttonpopupdecoration"/>
+        <textarea name="title" from="basepopuptitletext"/>
+        <textarea name="message" from="basetext_m">
+            <area>50,75,-50,160</area>
+            <align>left,vcenter</align>
+            <cutdown>yes</cutdown>
+            <multiline>yes</multiline>
+        </textarea>
+        <!-- bottom part with buttons -->
+        <button name="ok" from="basebutton">
+            <area>50%-100,100%-65,200,40</area>
+            <value>Ok</value>
+        </button>
+        <button name="cancel" from="basebutton">
+            <area>50%+150,100%-65,200,40</area>
+            <value>Cancel</value>
+        </button>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="MythDialogBox">
+        <area>-1,0,800,550</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-400" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basepopupdecoration"/>
+        <textarea name="title" from="basepopuptitletext"/>
+        <textarea name="messagearea" from="basetext_m">
+            <area>50,70,-50,160</area>
+            <align>left,vcenter</align>
+            <multiline>yes</multiline>
+            <scroll direction="vertical"/>
+            <cutdown>no</cutdown>
+        </textarea>
+        <shape name="separator" from="baseseparator_h">
+            <position>0,245</position>
+        </shape>
+        <buttonlist name="list" from="baseitemlist">
+            <align>vcenter</align>
+            <area>50,260,-50,-25</area>
+        </buttonlist>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="MythProgressDialog">
+        <area>-1,0,750,270</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-250" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basepopupdecoration"/>
+        <textarea name="message" from="basetext_m">
+            <area>50,75,-50,80</area>
+            <align>center</align>
+            <cutdown>yes</cutdown>
+            <multiline>yes</multiline>
+        </textarea>
+        <textarea name="progresstext" from="basetext_s">
+            <area>50,180,-50,40</area>
+            <alpha>200</alpha>
+            <align>center</align>
+        </textarea>
+        <progressbar name="progressbar" from="baseprogress_s">
+            <area>50,220,-50,12</area>
+        </progressbar>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="MythTextInputDialog">
+        <area>-1,0,750,330</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-250" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basebuttonpopupdecoration"/>
+        <textarea name="message" from="basetext_m">
+            <area>50,75,-50,80</area>
+            <align>center</align>
+            <cutdown>yes</cutdown>
+            <multiline>yes</multiline>
+        </textarea>
+        <textedit name="input" from="basetextedit">
+            <area>50,180,-50,40</area>
+        </textedit>
+        <!-- bottom part with buttons -->
+        <button name="ok" from="basebutton">
+            <area>50%-225,100%-65,200,40</area>
+            <value>Ok</value>
+        </button>
+        <button name="cancel" from="basebutton">
+            <area>50%+25,100%-65,200,40</area>
+            <value>Cancel</value>
+        </button>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="SpinBoxEntryDialog">
+        <area>-1,0,550,330</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-250" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basebuttonpopupdecoration"/>
+        <textarea name="rules" from="basetext_m">
+            <area>50,75,-50,80</area>
+            <align>center</align>
+            <cutdown>yes</cutdown>
+            <multiline>yes</multiline>
+            <template>Values from %LOW% to %HIGH% in increments of %STEP%</template>
+        </textarea>
+        <textedit name="entry" from="basetextedit">
+            <area>50,180,-50,40</area>
+        </textedit>
+        <!-- bottom part with buttons -->
+        <button name="ok" from="basebutton">
+            <area>50%-225,100%-65,200,40</area>
+            <value>Ok</value>
+        </button>
+        <button name="cancel" from="basebutton">
+            <area>50%+25,100%-65,200,40</area>
+            <value>Cancel</value>
+        </button>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="MythSpinBoxDialog">
+        <area>-1,0,750,330</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-250" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basebuttonpopupdecoration"/>
+        <textarea name="message" from="basetext_m">
+            <area>50,75,-50,80</area>
+            <align>center</align>
+            <cutdown>yes</cutdown>
+            <multiline>yes</multiline>
+        </textarea>
+        <spinbox name="input" from="basespinbox">
+            <area>50,180,-50,40</area>
+        </spinbox>
+        <!-- bottom part with buttons -->
+        <button name="ok" from="basebutton">
+            <area>50%-225,100%-65,200,40</area>
+            <value>Ok</value>
+        </button>
+        <button name="cancel" from="basebutton">
+            <area>50%+25,100%-65,200,40</area>
+            <value>Cancel</value>
+        </button>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="MythSearchDialog">
+        <area>-1,0,750,620</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-250" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basebuttonpopupdecoration"/>
+        <textarea name="title" from="basepopuptitletext"/>
+        <!-- content -->
+        <textedit name="input" from="basetextedit">
+            <area>50,100,-50,40</area>
+        </textedit>
+        <buttonlist name="itemlist" from="baseitemlist">
+            <area>50,165,-50,-100</area>
+        </buttonlist>
+        <!-- bottom part with buttons -->
+        <button name="ok" from="basebutton">
+            <area>50%-225,100%-65,200,40</area>
+            <value>Ok</value>
+        </button>
+        <button name="cancel" from="basebutton">
+            <area>50%+25,100%-65,200,40</area>
+            <value>Cancel</value>
+        </button>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="MythFileBrowser">
+        <group name="decoration" from="basesettingsdecoration"/>
+        <textarea name="pagetitle" from="basepagetitletext">
+            <value>File Browser</value>
+        </textarea>
+        <group name="locationbar">
+            <area>100,100,1000,40</area>
+            <textedit name="location" from="basetextedit">
+                <area>0,0,-200,40</area>
+            </textedit>
+            <button name="back" from="basebackbutton">
+                <area>100%-175,0,75,40</area>
+            </button>
+            <button name="home" from="basehomebutton">
+                <area>100%-75,0,75,40</area>
+            </button>
+        </group>
+        <buttonlist name="filelist" from="baseitemlist">
+            <area>100,175,1000,-115</area>
+        </buttonlist>
+        <group name="details">
+            <area>1200,50%-300,-100,600</area>
+            <imagetype name="preview">
+                <area>0,0,100%,-90</area>
+                <preserveaspect>yes</preserveaspect>
+            </imagetype>
+            <textarea name="info" from="basetext_m">
+                <area>0,100%-40,100%,40</area>
+                <align>center</align>
+                <alpha>200</alpha>
+            </textarea>
+        </group>
+        <!-- bottom part with buttons -->
+        <button name="ok" from="basebutton">
+            <area>50%-225,100%-65,200,40</area>
+            <value>Ok</value>
+        </button>
+        <button name="cancel" from="basebutton">
+            <area>50%+25,100%-65,200,40</area>
+            <value>Cancel</value>
+        </button>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="MythTimeInputDialog">
+        <area>-1,0,750,370</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-250" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basebuttonpopupdecoration"/>
+        <textarea name="message" from="basetext_m">
+            <area>50,75,-50,80</area>
+            <align>center</align>
+            <cutdown>yes</cutdown>
+            <multiline>yes</multiline>
+        </textarea>
+        <shape name="separator" from="baseseparator_h">
+            <position>0,180</position>
+        </shape>
+        <buttonlist name="dates" from="baseselector">
+            <area>50%-215,205,200,40</area>
+        </buttonlist>
+        <buttonlist name="times" from="baseselector">
+            <area>50%+15,205,200,40</area>
+        </buttonlist>
+        <!-- bottom part with buttons -->
+        <button name="ok" from="basebutton">
+            <area>50%-100,100%-65,200,40</area>
+            <value>Ok</value>
+        </button>
+    </window>
+
+    <!--====================================================================-->
+
+    <window name="MythMetadataResults">
+        <area>-1,0,900,500</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-400" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basepopupdecoration"/>
+        <textarea name="title" from="basepopuptitletext">
+            <value>Metadata Results</value>
+        </textarea>
+        <buttonlist name="results" from="baseitemlist">
+            <area>50,75,-50,-25</area>
+            <align>vcenter</align>
+            <statetype name="buttonitem">
+                <area>0,0,100%,130</area>
+                <state name="active">
+                    <area>0,0,100%,130</area>
+                    <shape name="itemseparator" from="baseitemseparator_h">
+                        <position>0,100%</position>
+                    </shape>
+                    <imagetype name="icon" from="basecoverarticon" depends="!buttonimage">
+                        <area>5,5,80,-5</area>
+                    </imagetype>
+                    <imagetype name="buttonimage">
+                        <area>5,5,80,-5</area>
+                        <preserveaspect>true</preserveaspect>
+                        <alpha>150</alpha>
+                    </imagetype>
+                    <textarea name="buttontext">
+                        <area>0,0,0,0</area>
+                    </textarea>
+                    <textarea name="title" from="baseitemtext">
+                        <area>100,5,-5,40</area>
+                        <template>%TITLE%% ~ |SUBTITLE%% (|YEAR|)%</template>
+                        <alpha>100</alpha>
+                    </textarea>
+                    <textarea name="description" from="basetext_s">
+                        <area>100,50,-5,-5</area>
+                        <multiline>yes</multiline>
+                        <align>justify</align>
+                        <cutdown>yes</cutdown>
+                        <alpha>100</alpha>
+                    </textarea>
+                </state>
+                <state name="selectedactive" from="active">
+                    <shape name="background" from="basecontrolbackground">
+                        <area>0,0,100%,100%</area>
+                    </shape>
+                    <shape name="itemseparator"/>
+                    <imagetype name="buttonimage">
+                        <alpha>255</alpha>
+                    </imagetype>
+                    <textarea name="title">
+                        <scroll direction="horizontal" />
+                        <cutdown>no</cutdown>
+                        <alpha>255</alpha>
+                    </textarea>
+                    <textarea name="description">
+                        <alpha>150</alpha>
+                    </textarea>
+                </state>
+                <state name="inactive" from="active"/>
+                <state name="selectedinactive" from="selectedactive">
+                    <textarea name="title">
+                        <alpha>100</alpha>
+                    </textarea>
+                    <textarea name="description">
+                        <alpha>100</alpha>
+                    </textarea>
+                </state>
+            </statetype>
+        </buttonlist>
+    </window>
+
+    <!--====================================================================-->
+    
+    <window name="MythArtworkResults">
+        <area>-1,0,1605,450</area>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="-1,-400" end="-1,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <group name="decoration" from="basepopupdecoration"/>
+        <textarea name="title" from="basepopuptitletext">
+            <value>Artwork Search Results</value>
+        </textarea>
+        <buttonlist name="results">
+            <area>50,75,1505,250</area>
+            <buttonarea>0,0,100%,100%</buttonarea>
+            <align>hcenter</align>
+            <layout>horizontal</layout>
+            <scrollstyle>center</scrollstyle>
+            <wrapstyle>items</wrapstyle>
+            <statetype name="buttonitem">
+                <area>0,0,300,350</area>
+                <state name="active">
+                    <area>0,0,300,350</area>
+                    <imagetype name="buttonimage">
+                        <area>1,0,-1,-50</area>
+                        <preserveaspect>true</preserveaspect>
+                        <alpha>200</alpha>
+                    </imagetype>
+                    <imagetype name="icon" from="basecoverarticon" depends="!buttonimage">
+                        <area>0,0,100%,-50</area>
+                    </imagetype>
+                    <shape name="top" from="baseseparator_h">
+                        <position>0,0</position>
+                    </shape>
+                    <shape name="bottom" from="baseseparator_h">
+                        <position>0,100%-50</position>
+                    </shape>
+                    <textarea name="resolution" from="basetext_s">
+                        <area>0,100%-50,100%,100%</area>
+                        <align>center</align>
+                        <cutdown>yes</cutdown>
+                        <alpha>150</alpha>
+                    </textarea>
+                </state>
+                <state name="selectedactive" from="active">
+                    <shape name="background" from="basecontrolbackground">
+                        <area>0,100%-50,100%,100%</area>
+                    </shape>
+                    <imagetype name="buttonimage">
+                        <alpha>255</alpha>
+                    </imagetype>
+                    <shape name="whiteframe" from="baseframe">
+                        <area>0,0,100%,100%</area>
+                        <type>box</type>
+                    </shape>
+                    <textarea name="resolution">
+                        <alpha>255</alpha>
+                    </textarea>
+                </state>
+            </statetype>
+        </buttonlist>
+    </window>
+
+</mythuitheme>
diff --speed-large-files --no-dereference --minimal -Naur Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/fragment_shader.txt Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/fragment_shader.txt
--- Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/fragment_shader.txt	1970-01-01 01:00:00.000000000 +0100
+++ Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/fragment_shader.txt	2021-12-07 11:00:45.911171533 +0100
@@ -0,0 +1,66 @@
+#define cc vec2(0.5,0.5)
+#define smoothness 1.0
+#define SNOW_COL vec4(1.0,1.0,1.0,1.0)
+#define N_LAYERS 6.0
+#define SNOW_ALPHA 0.1
+#define SPEED -0.3
+#define bg vec4(0.8,0.8,0.9,0.0)
+varying highp float v_alpha;
+varying highp float v_time;
+varying highp vec2  v_pos;
+varying highp vec2  v_res;
+highp float smoothCircle(highp vec2 position, highp float relativeSize) {
+    highp float d = distance(cc,position)*2./relativeSize;
+    if (d > 1.0){
+        return 0.0;
+    }
+    else{
+        return clamp(smoothness/d-smoothness,-1.0,1.0);
+    }
+}
+// fake random stolen from https://gist.github.com/patriciogonzalezvivo/670c22f3966e662d2f83
+highp float randF(highp float n){
+    return fract(sin(n) * 43758.5453123);
+}
+// used to determine whether to draw a flake in the grid
+// the magic number is just me mashing the keyboard
+bool rand2d(highp float i, highp float j, highp float probability) {
+        return  (randF(i + j*7.8124861) > probability);
+}
+// create the grid of circles, with a bunch missing, and fanaggle the sizes a bit
+highp float circleGrid(highp vec2 position, highp float spacing, highp float dotSize) {
+    // idx => which dot we are showing in the grid.
+    // check x and y index to see if we should draw it or not.
+    highp float idx = floor(1./spacing * position.x);
+    highp float yIdx = floor(1./spacing * position.y);
+    // much higher than .06 causes the grid to be obvious
+    if (rand2d(idx,yIdx,0.06)) {
+        return 0.0;
+    }
+    // modify the size of the flake a bit
+    highp float relativeSize = (0.5 + 0.5*randF(yIdx))*dotSize / spacing;
+
+    return smoothCircle(vec2(
+        fract(1./spacing*position.x),
+        fract(1./spacing*position.y + yIdx)
+    ),relativeSize);
+}
+
+void main(void) {
+    highp vec2 uv = v_pos / v_res;
+    highp vec2 uvsq = vec2(uv.x* v_res.x/v_res.y, uv.y);
+    highp float amnt = 0.0;
+
+    for (highp float i = 0.0; i < N_LAYERS; i ++){
+        highp float p = 0.5 + ((i+1.) / N_LAYERS)*0.4;
+        // small flakes in bg drawn first with lower speed
+
+        // the only reason we add "i" is so it tweaks the noise called on this position later.
+        highp vec2 fallPosition = vec2((1.0-p) + uvsq.x + i + p*sin(v_time/2000.+i)/4. * SPEED,
+                                 (1.0-p) + i * 3.0 + uvsq.y + v_time/2000.*p/1. * SPEED
+        );
+        amnt = amnt + SNOW_ALPHA * circleGrid(fallPosition, 0.06* p, 0.04* p*p);
+    }
+    gl_FragColor = mix(SNOW_COL,bg,1.0-amnt);
+};
+ 
diff --speed-large-files --no-dereference --minimal -Naur Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/vertex_shader.txt Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/vertex_shader.txt
--- Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/vertex_shader.txt	1970-01-01 01:00:00.000000000 +0100
+++ Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/vertex_shader.txt	2021-12-07 09:40:14.133312940 +0100
@@ -0,0 +1,18 @@
+attribute vec2 a_position;
+varying highp float v_alpha;
+varying highp float v_time;
+varying highp vec2  v_pos;
+varying highp vec2  v_res;
+uniform highp mat4  u_projection;
+uniform highp mat4  u_transform;
+uniform highp float u_time;
+uniform highp float u_alpha;
+uniform highp vec2  u_res;
+void main() {
+    v_pos = a_position;
+    gl_Position = u_projection * u_transform * vec4(a_position, 0.0, 1.0);
+    v_alpha = u_alpha;
+    v_time = u_time;
+    v_res  = u_res;
+}
+ 
