diff --speed-large-files --no-dereference --minimal -Naur u-boot-2ebdba9bbc547b1b0c84069a6487ece01eb4fdce/arch/arm/mach-sunxi/dram_dw_helpers.c u-boot-2ebdba9bbc547b1b0c84069a6487ece01eb4fdce/arch/arm/mach-sunxi/dram_dw_helpers.c
--- u-boot-2ebdba9bbc547b1b0c84069a6487ece01eb4fdce/arch/arm/mach-sunxi/dram_dw_helpers.c	2025-02-05 18:46:09.000000000 +0100
+++ u-boot-2ebdba9bbc547b1b0c84069a6487ece01eb4fdce/arch/arm/mach-sunxi/dram_dw_helpers.c	2025-02-05 20:15:56.440017732 +0100
@@ -117,7 +117,7 @@
 	memcpy((u32*)CFG_SYS_SDRAM_BASE, buffer, sizeof(buffer));
 
 	/* reconfigure to make sure that all active rows are accessible */
-	config->rows = 17;
+	config->rows = 16;
 	mctl_core_init(para, config);
 
 	/* store data again as it might be moved */
@@ -127,7 +127,7 @@
 
 	/* detect row address bits */
 	shift = config->bus_full_width + 4 + config->cols;
-	for (config->rows = 13; config->rows < 17; config->rows++) {
+	for (config->rows = 13; config->rows < 16; config->rows++) {
 		if (mctl_check_pattern(1ULL << (config->rows + shift)))
 			break;
 	}
