diff -Naur libmng-1.0.10-old/makefiles/configure.in libmng-1.0.10-new/makefiles/configure.in
--- libmng-1.0.10-old/makefiles/configure.in	2005-01-30 02:28:00.000000000 -0800
+++ libmng-1.0.10-new/makefiles/configure.in	2012-07-17 04:35:15.000000000 -0700
@@ -15,10 +15,6 @@
 dnl Checks for programs.
 AC_PROG_CC
 AC_ISC_POSIX
-AM_C_PROTOTYPES
-if test "x$U" != "x"; then
-  AC_MSG_ERROR(Compiler not ANSI compliant)
-fi
 AM_PROG_LIBTOOL
 AC_PROG_INSTALL
 
