diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/cardutil.cpp	2014-05-05 11:47:49.039117407 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/cardutil.cpp	2014-05-05 11:48:01.915708617 +0200
@@ -2054,8 +2054,8 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
-                  "FROM capturecard WHERE cardid = :CARDID ");
+    query.prepare("SELECT cardid, displayname "
+                  "FROM cardinput WHERE cardid = :CARDID ");
     query.bindValue(":CARDID", cardid);
 
     if (query.exec() && query.next())
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/statusbox.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/statusbox.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/statusbox.cpp	2014-05-05 11:47:49.495781397 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/statusbox.cpp	2014-05-05 11:48:01.929041861 +0200
@@ -866,7 +866,7 @@
 
         QString tun = tr("Tuner %1 %2 %3");
         QString devlabel = CardUtil::GetDeviceLabel(
-            query.value(1).toString(), query.value(2).toString());
+            query.value(0).toInt());
 
         QString shorttuner = tun.arg(cardid).arg("").arg(status);
         QString longtuner = tun.arg(cardid).arg(devlabel).arg(status);
