diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/scheduler.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/scheduler.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/scheduler.cpp	2014-05-05 11:48:01.132379867 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/scheduler.cpp	2014-05-05 11:48:10.515658234 +0200
@@ -178,8 +178,7 @@
         LOG(VB_GENERAL, LOG_ERR, LOC +
                 "No capture cards are defined in the database.\n\t\t\t"
                 "Perhaps you should re-read the installation instructions?");
-        error = GENERIC_EXIT_SETUP_ERROR;
-        return false;
+        return true;
     }
 
     query.prepare("SELECT sourceid,name FROM videosource ORDER BY sourceid;");
