diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/programdata.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/programdata.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/programdata.cpp	2014-05-05 11:48:04.865691330 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/programdata.cpp	2014-05-05 11:48:20.232271500 +0200
@@ -634,6 +634,25 @@
     return true;
 }
 
+static bool program_exists(MSqlQuery &query, uint chanid, const QDateTime &st)
+{
+    query.prepare(
+        "SELECT title FROM program "
+        "WHERE chanid    = :CHANID AND "
+        "      starttime = :OLDSTART");
+    query.bindValue(":CHANID",   chanid);
+    query.bindValue(":OLDSTART", st);
+    if (!query.exec())
+    {
+        MythDB::DBError("program_exists", query);
+    }
+    if (query.next())
+    {
+        return true;
+    }
+    return false;
+}
+
 static bool change_program(MSqlQuery &query, uint chanid, const QDateTime &st,
                            const QDateTime &new_st, const QDateTime &new_end)
 {
@@ -691,6 +710,12 @@
     else if (prog.starttime < endtime && prog.endtime > endtime)
     {
         // starts during, but ends after our program
+        if (program_exists(query, chanid, endtime))
+        {
+            LOG(VB_GENERAL, LOG_INFO,
+                QString("EIT: Deleting old program '%1'").arg(prog.title));
+            return delete_program(query, chanid, prog.starttime);
+        }
         return change_program(query, chanid, prog.starttime,
                               endtime, prog.endtime);
     }
