diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/diseqc.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/diseqc.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/diseqc.cpp	2014-05-05 11:49:01.002029522 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/diseqc.cpp	2014-05-05 11:49:13.525289680 +0200
@@ -667,6 +667,8 @@
         return false;
     }
 
+    bool resend_cmd = false;
+
 #ifndef USING_DVB
 
     (void) adr;
@@ -693,6 +695,12 @@
 
     LOG(VB_CHANNEL, LOG_INFO, LOC + "Sending DiSEqC Command: " + cmdstr);
 
+    if (repeats >= 10)
+    {
+        repeats = repeats - 10;
+        resend_cmd = true;
+    }
+
     // send the command
     for (uint i = 0; i <= repeats; i++)
     {
@@ -702,7 +710,9 @@
             return false;
         }
 
-        mcmd.msg[0] |= DISEQC_FRM_REPEAT;
+        if (!resend_cmd)
+            mcmd.msg[0] |= DISEQC_FRM_REPEAT;
+
         usleep(DISEQC_SHORT_WAIT);
     }
 
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/diseqcsettings.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/diseqcsettings.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/diseqcsettings.cpp	2014-05-05 11:48:57.558716318 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/diseqcsettings.cpp	2014-05-05 11:49:13.935287252 +0200
@@ -109,12 +109,14 @@
 {
   public:
     DeviceRepeatSetting(DiSEqCDevDevice &device) :
-        SpinBoxSetting(this, 0, 5, 1), m_device(device)
+        SpinBoxSetting(this, 0, 15, 1), m_device(device)
     {
         setLabel(DeviceTree::tr("Repeat Count"));
         QString help = DeviceTree::tr(
-            "Number of times to repeat DiSEqC commands sent to this device. "
-            "Larger values may help with less reliable devices.");
+            "Number of repeat (command with repeat flag ON) or resend (the same command) DiSEqC commands."
+            "If value is higher than 10, command will be resend N-10 times"
+            "If value is lower than 10, command will be repeated N times"
+            "Repeat useful for unreliable DiSEqC equipment; resend useful when unreliable DiSEq equipment has broken/unsuported repeat flag support.");
         setHelpText(help);
     }
 
