diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2014-05-05 11:49:12.495295698 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2014-05-05 11:49:22.271905263 +0200
@@ -297,7 +297,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_rec.cpp	2014-05-05 11:49:12.498629012 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_rec.cpp	2014-05-05 11:49:22.275238577 +0200
@@ -3930,12 +3930,12 @@
     RecStatusType newRecStatus = rsRecording;
     if (signalMonitor->IsAllGood())
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "TuningSignalCheck: Have a good signal");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "TuningSignalCheck: Have a good signal");
     }
     else if (signalMonitor->IsErrored() ||
              MythDate::current() > signalMonitorDeadline)
     {
-        LOG(VB_RECORD, LOG_ERR, LOC + "TuningSignalCheck: SignalMonitor " +
+        LOG(VB_GENERAL, LOG_ERR, LOC + "TuningSignalCheck: Waiting for Signal " +
             (signalMonitor->IsErrored() ? "failed" : "timed out"));
 
         ClearFlags(kFlagNeedToStartRecorder);
