diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2014-05-05 11:49:46.841761862 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp	2014-05-05 11:50:02.158339208 +0200
@@ -853,7 +853,7 @@
             {
                 LOG(VB_GENERAL, LOG_ERR, LOC +
                     "PES start code not found in TS packet with PUSI set");
-                break;
+                //break;
             }
 
             // bounds check
