diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2014-05-05 11:49:59.515021285 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2014-05-05 11:50:14.354934787 +0200
@@ -729,6 +729,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/videodisplayprofile.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/videodisplayprofile.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/videodisplayprofile.cpp	2014-05-05 11:49:59.501688030 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/videodisplayprofile.cpp	2014-05-05 11:50:14.474934089 +0200
@@ -1183,6 +1183,16 @@
                   "");
 }
 
+void VideoDisplayProfile::CreateMiniMythProfiles(const QString &hostname)
+{
+    (void) QObject::tr("MiniMyth", "MiniMyth");
+    DeleteProfileGroup("MiniMyth", hostname);
+    uint groupid = CreateProfileGroup("MiniMyth", hostname);
+    CreateProfile(groupid, 1, ">", 0, 0, "", 0, 0,
+                  "ffmpeg", 1, true, "xv-blit", "softblend", false,
+                  "bobdeint", "onefield", "");
+}
+
 void VideoDisplayProfile::CreateProfiles(const QString &hostname)
 {
     CreateNewProfiles(hostname);
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/videodisplayprofile.h mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/videodisplayprofile.h
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/videodisplayprofile.h	2014-05-05 11:49:59.491688088 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/videodisplayprofile.h	2014-05-05 11:50:14.481600717 +0200
@@ -150,6 +150,7 @@
     static void        CreateProfiles(const QString &hostname);
     static void        CreateNewProfiles(const QString &hostname);
     static void        CreateVDPAUProfiles(const QString &hostname);
+    static void        CreateMiniMythProfiles(const QString &hostname);
     static void        CreateVDAProfiles(const QString &hostname);
     static void        CreateOpenGLProfiles(const QString &hostname);
     static void        CreateVAAPIProfiles(const QString &hostname);
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/globalsettings.cpp	2014-05-05 11:50:00.305016680 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/globalsettings.cpp	2014-05-05 11:50:14.498267286 +0200
@@ -1196,6 +1196,11 @@
     if (profiles.empty())
         return;
 
+    if (!profiles.contains("MiniMyth"))
+    {
+        VideoDisplayProfile::CreateMiniMythProfiles(host);
+        profiles = VideoDisplayProfile::GetProfiles(host);
+    }
     if (!profiles.contains("Normal") &&
         !profiles.contains("High Quality") &&
         !profiles.contains("Slim"))
