diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp mythtv-0.27-20140504-ge2acf0d-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2014-05-05 11:50:35.371479045 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythplugins/mythgallery/mythgallery/gallerysettings.cpp	2014-05-05 11:50:47.158077106 +0200
@@ -45,7 +45,7 @@
 #ifdef Q_OS_MAC
     gc->setValue(QDir::homePath() + "/Pictures");
 #else
-    gc->setValue("/var/lib/pictures");
+    gc->setValue("/mnt/gallery");
 #endif
     gc->setHelpText(GallerySettings::tr("This directory must exist and "
                                         "MythGallery needs to have read "
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythui/mythmainwindow.cpp	2014-05-05 11:50:27.871522716 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythui/mythmainwindow.cpp	2014-05-05 11:50:47.158077106 +0200
@@ -1269,9 +1269,9 @@
         "Go forward to previous page"),     "F");
 
     RegisterKey("Main Menu",    "EXITPROMPT", QT_TRANSLATE_NOOP("MythControls",
-        "Display System Exit Prompt"),      "Esc");
+        "Display System Exit Prompt"),      "");
     RegisterKey("Main Menu",    "EXIT",       QT_TRANSLATE_NOOP("MythControls",
-        "System Exit"),                     "");
+        "System Exit"),                     "Ctrl+Esc");
 }
 
 void MythMainWindow::ReloadKeys()
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/globalsettings.cpp	2014-05-05 11:50:34.191485915 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/globalsettings.cpp	2014-05-05 11:50:47.164743735 +0200
@@ -1630,7 +1630,7 @@
 
     gc->setLabel(OSDSettings::tr("Browse all channels"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(OSDSettings::tr("If enabled, browse mode will show "
                                     "channels on all available recording "
@@ -1702,7 +1702,7 @@
 
     gc->setLabel(PlaybackSettings::tr("Prompt at end of recording"));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     gc->setHelpText(PlaybackSettings::tr("If enabled, a menu will be displayed "
                                          "allowing you to delete the recording "
@@ -1935,7 +1935,7 @@
 
     ge->setLabel(MainGeneralSettings::tr("Screen shot path"));
 
-    ge->setValue("/tmp/");
+    ge->setValue("/mnt/temp/");
 
     ge->setHelpText(MainGeneralSettings::tr("Path to screenshot storage "
                                             "location. Should be writable "
@@ -2988,7 +2988,7 @@
                                             "controlling MythFrontend "
                                             "over the network."));
 
-    gc->setValue(false);
+    gc->setValue(true);
 
     return gc;
 }
