diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/osd.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/osd.cpp	2014-05-05 11:50:40.441449531 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/osd.cpp	2014-05-05 11:50:56.611355440 +0200
@@ -525,7 +525,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
