diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_play.cpp	2014-05-05 11:50:54.041370390 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_play.cpp	2014-05-05 11:51:08.951283677 +0200
@@ -8171,7 +8171,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
