diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythplugins/configure mythtv-0.27-20140504-ge2acf0d-new/mythplugins/configure
--- mythtv-0.27-20140504-ge2acf0d-old/mythplugins/configure	2014-05-05 11:51:18.384562175 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythplugins/configure	2014-05-05 11:51:29.221165875 +0200
@@ -595,13 +595,13 @@
         disable netvision
     }
 
-    check_py_lib MythTV  || disable_netvision "MythTV Python bindings (MythTV)"
-    check_py_lib pycurl  || disable_netvision "Python pycurl library (pycurl)"
+#    check_py_lib MythTV  || disable_netvision "MythTV Python bindings (MythTV)"
+#    check_py_lib pycurl  || disable_netvision "Python pycurl library (pycurl)"
     check_py_lib lxml    || disable_netvision "Python lxml library (lxml)"
     check_py_lib xml     || disable_netvision "Python XML library (xml)"
     check_py_lib urllib  || disable_netvision "Python URL library (urllib)"
     check_py_lib urllib2 || disable_netvision "Python URL library 2 (urllib2)"
-    check_py_lib oauth   || disable_netvision "Python OAuth library (oauth)"
+#    check_py_lib oauth   || disable_netvision "Python OAuth library (oauth)"
 
     disabled netvision && echo "Disabling MythNetvision due to missing dependencies."
 fi
