diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/bindings/python/MythTV/utility/dt.py mythtv-0.27-20140504-ge2acf0d-new/mythtv/bindings/python/MythTV/utility/dt.py
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/bindings/python/MythTV/utility/dt.py	2014-05-05 11:51:39.281107445 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/bindings/python/MythTV/utility/dt.py	2014-05-05 11:51:47.934390538 +0200
@@ -15,7 +15,6 @@
 import re
 import time
 import singleton
-time.tzset()
 
 class basetzinfo( _pytzinfo ):
     """
