diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythplugins/mythgallery/mythgallery/glsingleview.cpp mythtv-0.27-20140504-ge2acf0d-new/mythplugins/mythgallery/mythgallery/glsingleview.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythplugins/mythgallery/mythgallery/glsingleview.cpp	2014-05-05 11:51:46.374399594 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythplugins/mythgallery/mythgallery/glsingleview.cpp	2014-05-05 11:51:57.794333313 +0200
@@ -222,7 +222,10 @@
         ThumbItem *item = m_itemList.at(m_pos);
 
         if (item)
+        {
             GalleryUtil::PlayVideo(item->GetPath());
+            makeCurrent();
+        }
 
         if (!m_slideshow_running && item)
         {
