diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythui/mythnotificationcenter.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythui/mythnotificationcenter.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythui/mythnotificationcenter.cpp	2014-05-05 11:51:51.281037778 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythui/mythnotificationcenter.cpp	2014-05-05 11:52:09.167600665 +0200
@@ -353,6 +353,9 @@
         // if not, default to the main one
         foundtheme = LoadWindowFromXML("notification-ui.xml", theme, this);
     }
+    if (!m_style.isEmpty())
+        LOG(VB_GENERAL, LOG_NOTICE, QString("Custom notification style request: '%1', will load: '%2'")
+            .arg(m_style).arg(theme_attempt));
 
     if (!foundtheme) // If we cannot load the theme for any reason ...
         return false;
@@ -599,6 +602,11 @@
     {
         m_extra = tmp;
     }
+    tmp = data["asst"];
+    if (!(tmp.isNull() && (m_update & kMetaData)))
+    {
+        m_style = tmp;
+    }
     // We need to re-run init
     m_refresh = true;
 }
@@ -1506,6 +1514,7 @@
                                                                  "MythTV") : origin;
     data["asal"] = detail;
     data["asfm"] = extra;
+    data["asst"] = style;
 
     if (type == MythNotification::Error   ||
         type == MythNotification::Warning ||
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythui/mythudplistener.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythui/mythudplistener.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythui/mythudplistener.cpp	2014-05-05 11:51:51.277704465 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythui/mythudplistener.cpp	2014-05-05 11:52:09.167600665 +0200
@@ -114,6 +114,8 @@
     bool error = false;
     int visibility = 0;
     QString type = "normal";
+    QString style = "";
+    MythNotification::Priority priority = MythNotification::kDefault;
 
     QDomNode n = docElem.firstChild();
     while (!n.isNull())
@@ -143,6 +145,8 @@
                 visibility = e.text().toUInt();
             else if (e.tagName() == "type")
                 type = e.text();
+            else if (notification && e.tagName() == "style")
+                style = e.text();
             else if (notification && e.tagName() == "progress")
             {
                 bool ok;
@@ -172,7 +176,7 @@
                                      MythNotification::TypeFromString(type),
                              msg, origin, description, image, extra,
                              progress_text, progress, timeout, fullscreen,
-                             visibility);
+                             visibility, priority, style);
         }
         else
         {
