diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_rec.cpp	2014-05-05 11:52:28.687487493 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_rec.cpp	2014-05-05 11:52:42.617406784 +0200
@@ -3045,7 +3045,8 @@
     {
         LOG(VB_GENERAL, LOG_ERR, LOC +
             QString("Channel: \'%1\' was not found in the database.\n"
-                    "\t\tMost likely, your DefaultTVChannel setting is wrong.\n"
+                    "\t\tMost likely, the 'starting channel' for this "
+                    "Input Connection is invalid.\n"
                     "\t\tCould not toggle favorite.").arg(channum));
         return;
     }
@@ -4574,8 +4575,8 @@
         {
             LOG(VB_GENERAL, LOG_ERR, LOC +
                 QString("Channel: \'%1\' was not found in the database.\n"
-                        "\t\tMost likely, your DefaultTVChannel setting is "
-                        "wrong.\n"
+                        "\t\tMost likely, the 'starting channel' for this "
+                        "Input Connection is invalid.\n"
                         "\t\tCould not start livetv.").arg(channum));
             return false;
         }
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tvremoteutil.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tvremoteutil.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tvremoteutil.cpp	2014-05-05 11:52:28.697487435 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tvremoteutil.cpp	2014-05-05 11:52:43.214067064 +0200
@@ -144,10 +144,10 @@
     return new RemoteEncoder(num, hostname, port);
 }
 
-/**
- * Given a list of recorder numbers, return the first available.
- */
-vector<uint> RemoteRequestFreeRecorderList(const vector<uint> &excluded_cardids)
+static void
+RemoteRequestFreeRecorderInputList(const vector<uint> &excluded_cardids,
+                                   vector<uint> &cardids,
+                                   vector<uint> &inputids)
 {
 #if 0
     vector<uint> list;
@@ -163,7 +163,6 @@
 
     return list;
 #endif
-    vector<uint> result;
     vector<uint> cards = CardUtil::GetLiveTVCardList();
     for (uint i = 0; i < cards.size(); i++)
     {
@@ -171,18 +170,37 @@
             RemoteRequestFreeInputList(cards[i], excluded_cardids);
         for (uint j = 0; j < inputs.size(); j++)
         {
-            if (find(result.begin(),
-                     result.end(),
-                     inputs[j].cardid) == result.end())
-                result.push_back(inputs[j].cardid);
+            if (find(cardids.begin(),
+                     cardids.end(),
+                     inputs[j].cardid) == cardids.end())
+            {
+                cardids.push_back(inputs[j].cardid);
+                inputids.push_back(inputs[j].inputid);
+            }
         }
     }
-    QString msg("RemoteRequestFreeRecorderList returned {");
-    for (uint k = 0; k < result.size(); k++)
-        msg += QString(" %1").arg(result[k]);
+    QString msg("RemoteRequestFreeRecorderInputList returned {");
+    for (uint k = 0; k < cardids.size(); k++)
+        msg += QString(" %1:%2").arg(cardids[k]).arg(inputids[k]);
     msg += "}";
     LOG(VB_CHANNEL, LOG_INFO, msg);
-    return result;
+}
+
+/**
+ * Given a list of recorder numbers, return the first available.
+ */
+vector<uint> RemoteRequestFreeRecorderList(const vector<uint> &excluded_cardids)
+{
+    vector<uint> cardids, inputids;
+    RemoteRequestFreeRecorderInputList(excluded_cardids, cardids, inputids);
+    return cardids;
+}
+
+vector<uint> RemoteRequestFreeInputList(const vector<uint> &excluded_cardids)
+{
+    vector<uint> cardids, inputids;
+    RemoteRequestFreeRecorderInputList(excluded_cardids, cardids, inputids);
+    return inputids;
 }
 
 RemoteEncoder *RemoteRequestFreeRecorderFromList
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tvremoteutil.h mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tvremoteutil.h
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tvremoteutil.h	2014-05-05 11:52:28.680820865 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tvremoteutil.h	2014-05-05 11:52:43.220734446 +0200
@@ -43,6 +43,8 @@
 MTV_PUBLIC RemoteEncoder *RemoteGetExistingRecorder(int recordernum);
 MTV_PUBLIC vector<uint>
 RemoteRequestFreeRecorderList(const vector<uint> &excluded_cardids);
+MTV_PUBLIC vector<uint>
+RemoteRequestFreeInputList(const vector<uint> &excluded_cardids);
 MTV_PUBLIC vector<InputInfo> RemoteRequestFreeInputList(
     uint cardid, const vector<uint> &excluded_cardids);
 MTV_PUBLIC InputInfo RemoteRequestBusyInputID(uint cardid);
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/globalsettings.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/globalsettings.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/globalsettings.cpp	2014-05-05 11:52:29.347483668 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/globalsettings.cpp	2014-05-05 11:52:43.227401638 +0200
@@ -2887,7 +2887,8 @@
 
     ge->setHelpText(EPGSettings::tr("The program guide starts on this channel "
                                     "if it is run from outside of Live TV "
-                                    "mode."));
+                                    "mode. Leave blank to enable Live TV "
+                                    "automatic start channel."));
 
     return ge;
 }
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/guidegrid.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/guidegrid.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/guidegrid.cpp	2014-05-05 11:52:31.647470343 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/guidegrid.cpp	2014-05-05 11:52:43.227401638 +0200
@@ -27,6 +27,7 @@
 #include "remoteutil.h"
 #include "channelutil.h"
 #include "cardutil.h"
+#include "tvremoteutil.h"
 #include "mythuibuttonlist.h"
 #include "mythuiguidegrid.h"
 #include "mythdialogbox.h"
@@ -441,9 +442,22 @@
         return;
     }
 
+    // If chanid/channum are unset, find the channel that would
+    // naturally be selected when Live TV is started.  This depends on
+    // the available tuners, their cardinput.livetvorder values, and
+    // their cardinput.startchan values.
+    QString actualChannum = channum;
+    if (chanid == 0 && channum.isEmpty())
+    {
+        vector<uint> excluded_cardids;
+        vector<uint> inputIDs = RemoteRequestFreeInputList(excluded_cardids);
+        if (!inputIDs.empty())
+            actualChannum = CardUtil::GetStartingChannel(inputIDs[0]);
+    }
+
     MythScreenStack *mainStack = GetMythMainWindow()->GetMainStack();
     GuideGrid *gg = new GuideGrid(mainStack,
-                                  chanid, channum, startTime,
+                                  chanid, actualChannum, startTime,
                                   player, embedVideo, allowFinder,
                                   changrpid);
 
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/main.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/main.cpp	2014-05-05 11:52:29.334150412 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/main.cpp	2014-05-05 11:52:43.514068257 +0200
@@ -364,7 +364,6 @@
 {
     uint chanid = 0;
     QString channum = gCoreContext->GetSetting("DefaultTVChannel");
-    channum = (channum.isEmpty()) ? "3" : channum;
     QDateTime startTime;
     GuideGrid::RunProgramGuide(chanid, channum, startTime, NULL, false, true, -2);
 }
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/progfind.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/progfind.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/progfind.cpp	2014-05-05 11:52:29.340817040 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/progfind.cpp	2014-05-05 11:52:43.524068199 +0200
@@ -406,8 +406,6 @@
     if (m_allowEPG)
     {
         QString startchannel = gCoreContext->GetSetting("DefaultTVChannel");
-        if (startchannel.isEmpty())
-            startchannel = '3';
         uint startchanid = 0;
         QDateTime starttime;
 
