diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-05 11:52:39.784089863 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-05 11:52:53.727342445 +0200
@@ -63,6 +63,7 @@
 #include "mythmiscutil.h"
 #include "icringbuffer.h"
 #include "audiooutput.h"
+#include "cardutil.h"
 
 extern "C" {
 #include "vbitext/vbi.h"
@@ -1018,6 +1019,11 @@
     {
         gCoreContext->SaveSetting(
             "DefaultChanid", player_ctx->playingInfo->GetChanID());
+        int cardid = player_ctx->recorder->GetRecorderNumber();
+        QString channum = player_ctx->playingInfo->GetChanNum();
+        QString inputname;
+        int cardinputid = CardUtil::GetCardInputID(cardid, channum, inputname);
+        CardUtil::SetStartChannel(cardinputid, channum);
     }
 
     return IsErrored() ? -1 : 0;
