diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_play.cpp	2014-05-05 11:52:52.760681376 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_play.cpp	2014-05-05 11:53:03.890616947 +0200
@@ -8381,7 +8381,12 @@
 
 bool TV::IsTunable(const PlayerContext *ctx, uint chanid, bool use_cache)
 {
-    return !IsTunableOn(ctx,chanid,use_cache,true).empty();
+    return !IsTunableOn(this, ctx,chanid,use_cache,true).empty();
+}
+
+bool TV::IsTunable(uint chanid)
+{
+    return !IsTunableOn(NULL, NULL, chanid, false, true).empty();
 }
 
 static QString toCommaList(const QSet<uint> &list)
@@ -8399,8 +8404,17 @@
 QSet<uint> TV::IsTunableOn(
     const PlayerContext *ctx, uint chanid, bool use_cache, bool early_exit)
 {
+    return IsTunableOn(this, ctx, chanid, use_cache, early_exit);
+}
+
+QSet<uint> TV::IsTunableOn(TV *tv,
+    const PlayerContext *ctx, uint chanid, bool use_cache, bool early_exit)
+{
     QSet<uint> tunable_cards;
 
+    if (tv == NULL)
+        use_cache = false;
+
     if (!chanid)
     {
         LOG(VB_CHANNEL, LOG_INFO, LOC +
@@ -8413,7 +8427,7 @@
     mplexid = (32767 == mplexid) ? 0 : mplexid;
 
     vector<uint> excluded_cards;
-    if (ctx->recorder && ctx->pseudoLiveTVState == kPseudoNormalLiveTV)
+    if (ctx && ctx->recorder && ctx->pseudoLiveTVState == kPseudoNormalLiveTV)
         excluded_cards.push_back(ctx->GetCardID());
 
     uint sourceid = ChannelUtil::GetSourceIDForChannel(chanid);
@@ -8450,10 +8464,10 @@
         bool used_cache = false;
         if (use_cache)
         {
-            QMutexLocker locker(&is_tunable_cache_lock);
-            if (is_tunable_cache_inputs.contains(cardids[i]))
+            QMutexLocker locker(&tv->is_tunable_cache_lock);
+            if (tv->is_tunable_cache_inputs.contains(cardids[i]))
             {
-                inputs = is_tunable_cache_inputs[cardids[i]];
+                inputs = tv->is_tunable_cache_inputs[cardids[i]];
                 used_cache = true;
             }
         }
@@ -8461,8 +8475,11 @@
         if (!used_cache)
         {
             inputs = RemoteRequestFreeInputList(cardids[i], excluded_cards);
-            QMutexLocker locker(&is_tunable_cache_lock);
-            is_tunable_cache_inputs[cardids[i]] = inputs;
+            if (tv)
+            {
+                QMutexLocker locker(&tv->is_tunable_cache_lock);
+                tv->is_tunable_cache_inputs[cardids[i]] = inputs;
+            }
         }
 
 #if 0
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_play.h mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_play.h
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_play.h	2014-05-05 11:52:52.757348061 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_play.h	2014-05-05 11:53:03.890616947 +0200
@@ -327,6 +327,7 @@
     static void ReloadKeys(void);
     static void SetFuncPtr(const char *, void *);
     static int  ConfiguredTunerCards(void);
+    static bool IsTunable(uint chanid);
 
     /// \brief Helper class for Sleep Timer code.
     class SleepTimerInfo
@@ -418,9 +419,12 @@
                       int editType = kScheduleProgramGuide);
     bool StartEmbedding(const QRect&);
     void StopEmbedding(void);
-    bool IsTunable(const PlayerContext*, uint chanid, bool use_cache = false);
+    bool IsTunable(const PlayerContext*, uint chanid,
+                   bool use_cache = false);
     QSet<uint> IsTunableOn(const PlayerContext*, uint chanid,
                            bool use_cache, bool early_exit);
+    static QSet<uint> IsTunableOn(TV *tv, const PlayerContext*, uint chanid,
+                                  bool use_cache, bool early_exit);
     void ClearTunableCache(void);
     void ChangeChannel(const PlayerContext*, const ChannelInfoList &options);
     void DrawUnusedRects(void);
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/guidegrid.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/guidegrid.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/guidegrid.cpp	2014-05-05 11:52:53.080679523 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/guidegrid.cpp	2014-05-05 11:53:03.897283575 +0200
@@ -447,7 +447,13 @@
     // the available tuners, their cardinput.livetvorder values, and
     // their cardinput.startchan values.
     QString actualChannum = channum;
-    if (chanid == 0 && channum.isEmpty())
+    if (chanid == 0 && actualChannum.isEmpty())
+    {
+        uint defaultChanid = gCoreContext->GetNumSetting("DefaultChanid", 0);
+        if (defaultChanid && TV::IsTunable(defaultChanid))
+            chanid = defaultChanid;
+    }
+    if (chanid == 0 && actualChannum.isEmpty())
     {
         vector<uint> excluded_cardids;
         vector<uint> inputIDs = RemoteRequestFreeInputList(excluded_cardids);
