diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/guidegrid.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/guidegrid.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/guidegrid.cpp	2014-05-05 11:53:03.897283575 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/guidegrid.cpp	2014-05-05 11:53:13.180563191 +0200
@@ -856,6 +856,17 @@
     return handled;
 }
 
+static bool SelectionIsTunable(const ChannelInfoList &selection)
+{
+    bool isTunable = false;
+    for (int i = 0; i < selection.size(); ++i)
+    {
+        if (TV::IsTunable(selection[i].chanid))
+            return true;
+    }
+    return false;
+}
+
 void GuideGrid::ShowMenu(void)
 {
     QString label = tr("Guide Options");
@@ -870,8 +881,7 @@
 
         if (m_player && (m_player->GetState(-1) == kState_WatchingLiveTV))
             menuPopup->AddButton(tr("Change to Channel"));
-        else
-            // XXX Make sure this channel/program is watchable
+        else if (SelectionIsTunable(GetSelection()))
             menuPopup->AddButton(tr("Watch This Channel"));
 
         menuPopup->AddButton(tr("Record This"));
@@ -1692,6 +1702,12 @@
                 // XXX Do another check that this channel/program is watchable
                 TV::StartTV(NULL, kStartTVNoFlags, GetSelection());
             }
+            else if (resulttext == tr("Watch This Channel"))
+            {
+                ChannelInfoList selection = GetSelection(); 
+                if (SelectionIsTunable(selection)) 
+                    TV::StartTV(NULL, kStartTVNoFlags, selection); 
+            }
             else if (resulttext == tr("Program Details"))
             {
                 details();
