diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/videoouttypes.h mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/videoouttypes.h
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/videoouttypes.h	2014-05-05 11:53:16.207212349 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/videoouttypes.h	2014-05-05 11:53:37.493755931 +0200
@@ -85,7 +85,7 @@
 typedef enum PictureAttribute
 {
     kPictureAttribute_None = 0,
-    kPictureAttribute_MIN = 1,
+    kPictureAttribute_MIN = 0,
     kPictureAttribute_Brightness = 1,
     kPictureAttribute_Contrast,
     kPictureAttribute_Colour,
