diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdread/dvd_input.c mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdread/dvd_input.c
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdread/dvd_input.c	2014-05-05 11:54:31.606776828 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdread/dvd_input.c	2014-05-05 11:54:40.970056168 +0200
@@ -44,7 +44,6 @@
 #define DVDcss_open(a) dvdcss_open((char*)(a))
 #define DVDcss_close   dvdcss_close
 #define DVDcss_seek    dvdcss_seek
-#define DVDcss_title   dvdcss_title
 #define DVDcss_read    dvdcss_read
 #define DVDcss_error   dvdcss_error
 #else
@@ -61,19 +60,19 @@
 #include <CoreFoundation/CFBundle.h>
 #endif
 
-typedef struct dvdcss_s *dvdcss_handle;
-static dvdcss_handle (*DVDcss_open)  (const char *);
-static int           (*DVDcss_close) (dvdcss_handle);
-static int           (*DVDcss_seek)  (dvdcss_handle, int, int);
-static int           (*DVDcss_title) (dvdcss_handle, int);
-static int           (*DVDcss_read)  (dvdcss_handle, void *, int, int);
-static char *        (*DVDcss_error) (dvdcss_handle);
+typedef struct dvdcss_s *dvdcss_t;
+static dvdcss_t (*DVDcss_open)  (const char *);
+static int      (*DVDcss_close) (dvdcss_t);
+static int      (*DVDcss_seek)  (dvdcss_t, int, int);
+static int      (*DVDcss_read)  (dvdcss_t, void *, int, int);
+static char *   (*DVDcss_error) (dvdcss_t);
+#define DVDCSS_SEEK_KEY (1 << 1)
 #endif
 
 /* The DVDinput handle, add stuff here for new input methods. */
 struct dvd_input_s {
   /* libdvdcss handle */
-  dvdcss_handle dvdcss;
+  dvdcss_t dvdcss;
 
   /* dummy file input */
   int fd;
@@ -127,7 +126,7 @@
  */
 static int css_title(dvd_input_t dev, int block)
 {
-  return DVDcss_title(dev->dvdcss, block);
+  return DVDcss_seek(dev->dvdcss, block, DVDCSS_SEEK_KEY);
 }
 
 /**
@@ -277,13 +276,10 @@
 int dvdinput_setup(const char *path)
 {
   void *dvdcss_library = NULL;
-  char **dvdcss_version = NULL;
 
 #ifdef HAVE_DVDCSS_DVDCSS_H
   /* linking to libdvdcss */
   dvdcss_library = &dvdcss_library;  /* Give it some value != NULL */
-  /* the DVDcss_* functions have been #defined at the top */
-  dvdcss_version = &dvdcss_interface_2;
 
 #else
   /* dlopening libdvdcss */
@@ -291,7 +287,7 @@
 #ifdef __APPLE__
   #define CSS_LIB "libdvdcss.2.dylib"
 #elif defined(WIN32)
-  #define CSS_LIB "libdvdcss.dll"
+  #define CSS_LIB "libdvdcss-2.dll"
 #elif defined(__OS2__)
   #define CSS_LIB "dvdcss.dll"
 #else
@@ -332,21 +328,17 @@
 #else
 #define U_S
 #endif
-    DVDcss_open = (dvdcss_handle (*)(const char*))
+    DVDcss_open = (dvdcss_t (*)(const char*))
       dlsym(dvdcss_library, U_S "dvdcss_open");
-    DVDcss_close = (int (*)(dvdcss_handle))
+    DVDcss_close = (int (*)(dvdcss_t))
       dlsym(dvdcss_library, U_S "dvdcss_close");
-    DVDcss_title = (int (*)(dvdcss_handle, int))
-      dlsym(dvdcss_library, U_S "dvdcss_title");
-    DVDcss_seek = (int (*)(dvdcss_handle, int, int))
+    DVDcss_seek = (int (*)(dvdcss_t, int, int))
       dlsym(dvdcss_library, U_S "dvdcss_seek");
-    DVDcss_read = (int (*)(dvdcss_handle, void*, int, int))
+    DVDcss_read = (int (*)(dvdcss_t, void*, int, int))
       dlsym(dvdcss_library, U_S "dvdcss_read");
-    DVDcss_error = (char* (*)(dvdcss_handle))
+    DVDcss_error = (char* (*)(dvdcss_t))
       dlsym(dvdcss_library, U_S "dvdcss_error");
 
-    dvdcss_version = (char **)dlsym(dvdcss_library, U_S "dvdcss_interface_2");
-
     if(dlsym(dvdcss_library, U_S "dvdcss_crack")) {
       fprintf(stderr,
               "libdvdread: Old (pre-0.0.2) version of libdvdcss found.\n"
@@ -354,8 +346,8 @@
               "http://www.videolan.org/\n" );
       dlclose(dvdcss_library);
       dvdcss_library = NULL;
-    } else if(!DVDcss_open  || !DVDcss_close || !DVDcss_title || !DVDcss_seek
-              || !DVDcss_read || !DVDcss_error || !dvdcss_version) {
+    } else if(!DVDcss_open  || !DVDcss_close || !DVDcss_seek
+              || !DVDcss_read || !DVDcss_error) {
       fprintf(stderr,  "libdvdread: Missing symbols in %s, "
               "this shouldn't happen !\n", CSS_LIB);
       dlclose(dvdcss_library);
@@ -371,8 +363,6 @@
     fprintf(stderr, "DVDCSS_METHOD %s\n", psz_method);
     fprintf(stderr, "DVDCSS_VERBOSE %s\n", psz_verbose);
     */
-    fprintf(stderr, "libdvdread: Using libdvdcss version %s for DVD access\n",
-            dvdcss_version ? *dvdcss_version : "");
 
     /* libdvdcss wrapper functions */
     dvdinput_open  = css_open;
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdread/dvd_reader.c mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdread/dvd_reader.c
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdread/dvd_reader.c	2014-05-05 11:54:31.606776828 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdread/dvd_reader.c	2014-05-05 11:54:40.973389481 +0200
@@ -28,6 +28,7 @@
 #include <stdio.h>
 #include <errno.h>
 #include <string.h>
+#include <strings.h>
 #include <ctype.h>
 #include <unistd.h>
 #include <limits.h>
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdread/dvd_udf.c mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdread/dvd_udf.c
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdread/dvd_udf.c	2014-05-05 11:54:31.606776828 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdread/dvd_udf.c	2014-05-05 11:54:40.973389481 +0200
@@ -31,6 +31,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -593,8 +594,10 @@
     } else
       in_cache = 1;
 
-    if(cached_dir == NULL)
+    if(cached_dir == NULL) {
+      free(cached_dir_base);
       return 0;
+    }
 
     p = 0;
 
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdread/ifo_print.c mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdread/ifo_print.c
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdread/ifo_print.c	2014-05-05 11:54:31.606776828 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdread/ifo_print.c	2014-05-05 11:54:40.973389481 +0200
@@ -1189,10 +1189,10 @@
     if(ifohandle->menu_vobu_admap) {
       ifoPrint_VOBU_ADMAP(ifohandle->menu_vobu_admap);
     } else {
-      printf("No Menu VOBU address map present\n");
+      printf("No Menu VOBU Address map present\n");
     }
 
-    printf("\nCell Adress table\n");
+    printf("\nCell Address table\n");
     printf(  "-----------------\n");
     ifoPrint_C_ADT(ifohandle->vts_c_adt);
 
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdread/ifo_read.c mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdread/ifo_read.c
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdread/ifo_read.c	2014-05-05 11:54:31.606776828 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdread/ifo_read.c	2014-05-05 11:54:40.973389481 +0200
@@ -568,6 +568,8 @@
 
   CHECK_ZERO(vmgi_mat->zero_1);
   CHECK_ZERO(vmgi_mat->zero_2);
+  /* DVDs created by VDR-to-DVD device LG RC590M violate the following check with
+   * vmgi_mat->zero_3 = 0x00000000010000000000000000000000000000. */
   CHECK_ZERO(vmgi_mat->zero_3);
   CHECK_ZERO(vmgi_mat->zero_4);
   CHECK_ZERO(vmgi_mat->zero_5);
@@ -1184,8 +1186,22 @@
     fprintf(stderr, "libdvdread: PTT search table too small.\n");
     goto fail;
   }
+
+  if(vts_ptt_srpt->nr_of_srpts == 0) {
+    fprintf(stderr, "libdvdread: Zero entries in PTT search table.\n");
+    goto fail;
+  }
+
   for(i = 0; i < vts_ptt_srpt->nr_of_srpts; i++) {
-    B2N_32(data[i]);
+    /* Transformers 3 has PTT start bytes that point outside the SRPT PTT */
+    uint32_t start = data[i];
+    B2N_32(start);
+    if(start + sizeof(ptt_info_t) > vts_ptt_srpt->last_byte + 1) {
+      /* don't mess with any bytes beyond the end of the allocation */
+      vts_ptt_srpt->nr_of_srpts = i;
+      break;
+    }
+    data[i] = start;
     /* assert(data[i] + sizeof(ptt_info_t) <= vts_ptt_srpt->last_byte + 1);
        Magic Knight Rayearth Daybreak is mastered very strange and has
        Titles with 0 PTTs. They all have a data[i] offsets beyond the end of
@@ -1211,6 +1227,7 @@
        Titles with 0 PTTs. */
     if(n < 0) n = 0;
 
+    /* DVDs created by the VDR-to-DVD device LG RC590M violate the following requirement */
     CHECK_VALUE(n % 4 == 0);
 
     vts_ptt_srpt->title[i].nr_of_ptts = n / 4;
@@ -1293,7 +1310,7 @@
   if(!ifofile->vmgi_mat)
     return 0;
 
-  if(ifofile->vmgi_mat->ptl_mait == NULL)
+  if(ifofile->vmgi_mat->ptl_mait == 0)
     return 1;
 
   if(!DVDFileSeek_(ifofile->file, ifofile->vmgi_mat->ptl_mait * DVD_BLOCK_LEN))
@@ -1374,6 +1391,7 @@
       ifofile->ptl_mait = NULL;
       return 0;
     }
+    memset(pf_temp, 0, info_length);
     if(!(DVDReadBytes(ifofile->file, pf_temp, info_length))) {
       fprintf(stderr, "libdvdread: Unable to read PTL_MAIT table at index %d.\n",i);
       free(pf_temp);
@@ -1436,6 +1454,7 @@
 
   if(ifofile->vtsi_mat->vts_tmapt == 0) { /* optional(?) */
     ifofile->vts_tmapt = NULL;
+    fprintf(stderr,"libdvdread: No VTS_TMAPT available - skipping.\n");
     return 1;
   }
 
