diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/contrib/user_jobs/mythlink.pl mythtv-0.27-20140504-ge2acf0d-new/mythtv/contrib/user_jobs/mythlink.pl
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/contrib/user_jobs/mythlink.pl	2014-05-05 11:55:13.706534206 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/contrib/user_jobs/mythlink.pl	2014-05-05 11:55:26.079796307 +0200
@@ -29,7 +29,7 @@
     our ($hostname, $dbh, $sh, $q, $count, $base_dir);
 
 # Default filename format
-    $dformat = '%T %- %Y-%m-%d, %g-%i %A %- %S';
+    $dformat = '%U/%T %- %Y-%m-%d, %g-%i %A %- %S';
 # Default separator character
     $dseparator = '-';
 # Default replacement character
