diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdnav/navigation.c mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdnav/navigation.c
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdnav/navigation.c	2014-05-05 11:55:21.199824396 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdnav/navigation.c	2014-05-05 11:55:36.376403722 +0200
@@ -169,14 +169,6 @@
   return DVDNAV_STATUS_ERR;
 }
 
-void dvdnav_first_play(dvdnav_t *this) {
-  pthread_mutex_lock(&this->vm_lock);
-  this->started = 0;
-  vm_start(this->vm);
-  this->started = 1;
-  pthread_mutex_unlock(&this->vm_lock);
-}
-
 dvdnav_status_t dvdnav_title_play(dvdnav_t *this, int32_t title) {
   return dvdnav_part_play(this, title, 1);
 }
