diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdnav/dvdnav.h mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdnav/dvdnav.h
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythdvdnav/dvdnav/dvdnav.h	2014-05-05 11:55:32.499759360 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythdvdnav/dvdnav/dvdnav.h	2014-05-05 11:55:47.279674338 +0200
@@ -262,11 +262,6 @@
  *********************************************************************/
 
 /*
- * Play DVD from the Beginning
- */
-void dvdnav_first_play(dvdnav_t *self);
-
-/*
  * Returns the number of titles on the disk.
  */
 dvdnav_status_t dvdnav_get_number_of_titles(dvdnav_t *self, int32_t *titles);
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/DVD/dvdringbuffer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/DVD/dvdringbuffer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/DVD/dvdringbuffer.cpp	2014-05-05 11:55:32.469759533 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/DVD/dvdringbuffer.cpp	2014-05-05 11:55:47.279674338 +0200
@@ -593,7 +593,6 @@
         dvdnav_menu_language_select(m_dvdnav, lang.toLatin1().data());
         dvdnav_audio_language_select(m_dvdnav, lang.toLatin1().data());
         dvdnav_spu_language_select(m_dvdnav, lang.toLatin1().data());
-        dvdnav_first_play(m_dvdnav);
         m_audioStreamsChanged = true;
     }
 
