diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmyth/programinfo.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmyth/programinfo.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmyth/programinfo.cpp	2014-05-05 11:55:39.519718973 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmyth/programinfo.cpp	2014-05-05 11:55:58.496276511 +0200
@@ -1952,6 +1952,7 @@
     dupmethod    = RecordingDupMethodType(query.value(47).toInt());
 
     // ancillary data -- begin
+    programflags = FL_NONE;
     set_flag(programflags, FL_CHANCOMMFREE,
              query.value(30).toInt() == COMM_DETECT_COMMFREE);
     set_flag(programflags, FL_COMMFLAG,
