diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/external/FFmpeg/libavcodec/libx264.c mythtv-0.27-20140504-ge2acf0d-new/mythtv/external/FFmpeg/libavcodec/libx264.c
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/external/FFmpeg/libavcodec/libx264.c	2014-05-05 11:55:52.399644895 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/external/FFmpeg/libavcodec/libx264.c	2014-05-05 11:56:10.752872737 +0200
@@ -289,6 +289,7 @@
     x4->params.rc.f_pb_factor             = avctx->b_quant_factor;
     x4->params.analyse.i_chroma_qp_offset = avctx->chromaoffset;
     if (x4->preset || x4->tune)
+        av_log(avctx, AV_LOG_INFO, "x264 setting preset/tune to: %s/%s.\n", x4->preset, x4->tune);
         if (x264_param_default_preset(&x4->params, x4->preset, x4->tune) < 0) {
             int i;
             av_log(avctx, AV_LOG_ERROR, "Error setting preset/tune %s/%s.\n", x4->preset, x4->tune);
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/avformatwriter.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/avformatwriter.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/avformatwriter.cpp	2014-05-05 11:55:49.879659385 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/avformatwriter.cpp	2014-05-05 11:56:10.752872737 +0200
@@ -545,6 +545,11 @@
         av_opt_set_int(c, "mixed-refs", 1, 0);
         av_opt_set_int(c, "8x8dct", 0, 0);
         av_opt_set_int(c, "weightb", 0, 0);
+
+        const char* preset = m_EncodingPreset.toStdString().c_str();
+        av_opt_set(c->priv_data, "preset", preset, 0);
+        const char* tune = m_EncodingTune.toStdString().c_str();
+        av_opt_set(c->priv_data, "tune", tune, 0);
     }
 
     if(m_ctx->oformat->flags & AVFMT_GLOBALHEADER)
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/filewriterbase.h mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/filewriterbase.h
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/filewriterbase.h	2014-05-05 11:55:49.876326071 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/filewriterbase.h	2014-05-05 11:56:10.776205936 +0200
@@ -40,6 +40,8 @@
     void SetAudioFormat(AudioFormat f)  { m_audioFormat = f; }
     void SetThreadCount(int count)      { m_encodingThreadCount = count; }
     void SetTimecodeOffset(long long o) { m_startingTimecodeOffset = o; }
+    void SetEncodingPreset(QString preset) { m_EncodingPreset = preset; }
+    void SetEncodingTune(QString tune)  { m_EncodingTune = tune; }
 
     long long GetFramesWritten(void)  const { return m_framesWritten; }
     long long GetTimecodeOffset(void) const { return m_startingTimecodeOffset; }
@@ -67,6 +69,8 @@
     int         m_encodingThreadCount;
     long long   m_framesWritten;
     long long   m_startingTimecodeOffset;
+    QString     m_EncodingPreset;
+    QString     m_EncodingTune;
 };
 
 #endif
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythtranscode/transcode.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythtranscode/transcode.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythtranscode/transcode.cpp	2014-05-05 11:55:55.539626842 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythtranscode/transcode.cpp	2014-05-05 11:56:10.829538964 +0200
@@ -556,8 +556,17 @@
             avfw->SetKeyFrameDist(30);
         }
 
-        avfw->SetThreadCount(
-            gCoreContext->GetNumSetting("HTTPLiveStreamThreads", 2));
+        int threads    = gCoreContext->GetNumSetting("HTTPLiveStreamThreads", 2);
+        QString preset = gCoreContext->GetSetting("HTTPLiveStreamPreset", "medium");
+        QString tune   = gCoreContext->GetSetting("HTTPLiveStreamTune", "film");
+
+        LOG(VB_GENERAL, LOG_NOTICE,
+            QString("x264 HLS using: %1 threads, '%2' profile and '%3' tune")
+                .arg(threads).arg(preset).arg(tune));
+
+        avfw->SetThreadCount(threads);
+        avfw->SetEncodingPreset(preset);
+        avfw->SetEncodingTune(tune);
 
         if (avfw2)
             avfw2->SetThreadCount(1);
