diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2014-05-05 11:56:02.122922331 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2014-05-05 11:56:20.936147571 +0200
@@ -1221,7 +1221,11 @@
 
     // matroska demuxer sets the default_duration to avg_frame_rate
     // mov,mp4,m4a,3gp,3g2,mj2 demuxer sets avg_frame_rate
-    if ((QString(ic->iformat->name).contains("matroska") ||
+    if ((QString(ic->iformat->name).contains("matroska")) &&
+        container_fps < 121.0f && container_fps > 3.0f)
+        fps = container_fps;
+    else
+    if ((QString(ic->iformat->name).contains("xmatroska") ||
         QString(ic->iformat->name).contains("mov")) &&
         avg_fps < 121.0f && avg_fps > 3.0f)
         fps = avg_fps;
