diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmyth/programinfo.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmyth/programinfo.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmyth/programinfo.cpp	2014-05-05 11:56:16.422840162 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmyth/programinfo.cpp	2014-05-05 11:56:33.429409164 +0200
@@ -861,6 +861,7 @@
     director = _director;
     programid = _programid;
     inetref = _inetref;
+    year = _year;
 
     QDateTime cur = MythDate::current();
     recstartts = cur.addSecs(((int)_length_in_minutes + 1) * -60);
@@ -1648,7 +1649,7 @@
     progMap["inetref"] = inetref;
     progMap["catType"] = myth_category_type_to_string(catType);
 
-    progMap["year"] = year ? QString::number(year) : "";
+    progMap["year"] = year > 1895 ? QString::number(year) : "";
 
     progMap["partnumber"] = partnumber ? QString::number(partnumber) : "";
     progMap["parttotal"] = parttotal ? QString::number(parttotal) : "";
