diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/HLS/httplivestream.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/HLS/httplivestream.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/HLS/httplivestream.cpp	2014-05-05 11:56:42.596023215 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/HLS/httplivestream.cpp	2014-05-05 11:56:52.732631719 +0200
@@ -412,6 +412,7 @@
 
     file.write(QString(
         "#EXTM3U\n"
+        "#EXT-X-MEDIA:TYPE=VIDEO,GROUP-ID=\"AV\",NAME=\"Main\",DEFAULT=YES,URI=\"%2.m3u8\"\n"
         "#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=%1\n"
         "%2.m3u8\n"
         ).arg((int)((m_bitrate + m_audioBitrate) * 1.1))
@@ -420,6 +421,7 @@
     if (m_audioOnlyBitrate)
     {
         file.write(QString(
+            "#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID=\"AO\",NAME=\"Main\",DEFAULT=NO,URI=\"%2.m3u8\"\n"
             "#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=%1\n"
             "%2.m3u8\n"
             ).arg((int)((m_audioOnlyBitrate) * 1.1))
