diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_rec.cpp	2014-05-05 11:56:58.695930849 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_rec.cpp	2014-05-05 11:57:14.432507283 +0200
@@ -809,7 +809,7 @@
     if (curRec->IsCommercialFree())
         curRec->SaveCommFlagged(COMM_FLAG_COMMFREE);
 
-    AutoRunInitType t = (curRec->QueryRecordingGroup() == "LiveTV") ?
+    AutoRunInitType t = (curRec->GetRecordingGroup() == "LiveTV") ?
         kAutoRunNone : kAutoRunProfile;
     InitAutoRunJobs(curRec, t, NULL, __LINE__);
 
@@ -4625,7 +4625,7 @@
     }
 
     if (!pseudoLiveTVRecording)
-        prog->ApplyRecordRecGroupChange("LiveTV");
+        prog->SetRecordingGroup("LiveTV");
 
     StartedRecording(prog);
 
