diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythtranscode/transcode.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythtranscode/transcode.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythtranscode/transcode.cpp	2014-05-05 11:57:12.985848908 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythtranscode/transcode.cpp	2014-05-05 11:57:26.572437711 +0200
@@ -391,6 +391,24 @@
         newWidth = cmdWidth;
         newHeight = cmdHeight;
 
+        // Absolutely no purpose is served by scaling video up beyond it's
+        // original resolution, quality is degraded, transcoding is
+        // slower and in future we may wish to scale bitrate according to
+        // resolution, so it would also waste bandwidth (when streaming)
+        //
+        // This change could be said to apply for all transcoding, but for now
+        // we're limiting it to HLS where it's uncontroversial
+        if (hlsMode)
+        {
+//             if (newWidth > video_width)
+//                 newWidth = video_width;
+            if (newHeight > video_height)
+            {
+                newHeight = video_height;
+                newWidth = 0;
+            }
+        }
+
         // TODO: is this necessary?  It got commented out, but may still be
         // needed.
         // int actualHeight = (video_height == 1088 ? 1080 : video_height);
