diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythtranscode/transcode.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythtranscode/transcode.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythtranscode/transcode.cpp	2014-05-05 11:57:34.675724622 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythtranscode/transcode.cpp	2014-05-05 11:57:44.065670851 +0200
@@ -638,7 +638,7 @@
         arb->m_audioFrameSize = avfw->GetAudioFrameSize() * arb->m_channels * 2;
 
         GetPlayer()->SetVideoFilters(
-            gCoreContext->GetSetting("HTTPLiveStreamFilters"));
+            gCoreContext->GetSetting("HTTPLiveStreamFilters", "yadif=1:-1:1"));
     }
     else if (fifodir.isEmpty())
     {
