diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/HLS/httplivestream.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/HLS/httplivestream.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/HLS/httplivestream.cpp	2014-05-05 11:57:41.342353098 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/HLS/httplivestream.cpp	2014-05-05 11:57:53.195618582 +0200
@@ -124,7 +124,7 @@
         m_segmentSize = 10;
 
     if (m_audioOnlyBitrate == 0)
-        m_audioOnlyBitrate = 32000;
+        m_audioOnlyBitrate = 64000;
 
     m_sourceHost = gCoreContext->GetHostName();
 
