diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-05 11:58:26.368762157 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-05 11:58:35.438710313 +0200
@@ -2198,6 +2198,10 @@
     // clear the buffering state
     SetBuffering(false);
 
+    // If PiP then release the last shown frame to the decoding queue
+    if (player_ctx->IsPIP())
+        videoOutput->DoneDisplayingFrame(videoOutput->GetLastShownFrame());
+
     // retrieve the next frame
     videoOutput->StartDisplayingFrame();
     VideoFrame *frame = videoOutput->GetLastShownFrame();
@@ -2223,7 +2227,9 @@
     osdLock.unlock();
 
     AVSync(frame, 0);
-    videoOutput->DoneDisplayingFrame(frame);
+    // If PiP then keep this frame for MythPlayer::GetCurrentFrame
+    if (!player_ctx->IsPIP())
+        videoOutput->DoneDisplayingFrame(frame);
 }
 
 void MythPlayer::PreProcessNormalFrame(void)
