diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/programdata.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/programdata.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/programdata.cpp	2014-05-05 11:58:47.045310664 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/programdata.cpp	2014-05-05 11:58:58.655244357 +0200
@@ -28,6 +28,11 @@
     return str.isNull() ? "" : str;
 }
 
+static QVariant denullify(const QDateTime &dt)
+{
+    return dt.isNull() ? QVariant("0000-00-00 00:00:00") : QVariant(dt);
+}
+
 DBPerson::DBPerson(const DBPerson &other) :
     role(other.role), name(other.name)
 {
@@ -1020,7 +1025,7 @@
     query.bindValue(":CATEGORY",    denullify(category));
     query.bindValue(":CATTYPE",     cattype);
     query.bindValue(":STARTTIME",   starttime);
-    query.bindValue(":ENDTIME",     endtime);
+    query.bindValue(":ENDTIME",     denullify(endtime));
     query.bindValue(":CC",
                     subtitleType & SUB_HARDHEAR ? true : false);
     query.bindValue(":STEREO",
@@ -1160,25 +1165,9 @@
                 (*cur)->endts   = (*it)->startts;
                 (*cur)->endtime = (*it)->starttime;
             }
-            else
-            {
-                (*cur)->endtime = (*cur)->starttime;
-                if ((*cur)->endtime < QDateTime(
-                        (*cur)->endtime.date(), QTime(6, 0), Qt::UTC))
-                {
-                    (*cur)->endtime = QDateTime(
-                        (*cur)->endtime.date(), QTime(6, 0), Qt::UTC);
-                }
-                else
-                {
-                    (*cur)->endtime = QDateTime(
-                        (*cur)->endtime.date().addDays(1),
-                        QTime(0, 0), Qt::UTC);
-                }
-
-                (*cur)->endts =
-                    MythDate::toString((*cur)->endtime, MythDate::kFilename);
-            }
+            /* if its the last programme in the file then leave its
+               endtime as 0000-00-00 00:00:00 so we can find it easily in
+               fix_end_times() */
         }
 
         if (it == fixlist.end())
@@ -1315,7 +1304,7 @@
     MSqlQuery query1(MSqlQuery::InitCon()), query2(MSqlQuery::InitCon());
 
     querystr = "SELECT chanid, starttime, endtime FROM program "
-               "WHERE (DATE_FORMAT(endtime,'%H%i') = '0000') "
+               "WHERE endtime = '0000-00-00 00:00:00' "
                "ORDER BY chanid, starttime;";
 
     if (!query1.exec(querystr))
@@ -1332,11 +1321,10 @@
         endtime = query1.value(2).toString();
 
         querystr = QString("SELECT chanid, starttime, endtime FROM program "
-                           "WHERE starttime BETWEEN '%1 00:00:00'"
-                           "AND '%2 23:59:59' AND chanid = '%3' "
+                           "WHERE starttime > '%1' "
+                           "AND chanid = '%2' "
                            "ORDER BY starttime LIMIT 1;")
-                           .arg(endtime.left(10))
-                           .arg(endtime.left(10))
+                           .arg(starttime)
                            .arg(chanid);
 
         if (!query2.exec(querystr))
@@ -1350,10 +1338,9 @@
         {
             count++;
             endtime = query2.value(1).toString();
-            querystr = QString("UPDATE program SET starttime = '%1', "
+            querystr = QString("UPDATE program SET "
                                "endtime = '%2' WHERE (chanid = '%3' AND "
                                "starttime = '%4');")
-                               .arg(starttime)
                                .arg(endtime)
                                .arg(chanid)
                                .arg(starttime);
