diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythmetadata/dbaccess.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythmetadata/dbaccess.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythmetadata/dbaccess.cpp	2014-05-05 11:59:15.185149999 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythmetadata/dbaccess.cpp	2014-05-05 11:59:32.945048684 +0200
@@ -47,8 +47,11 @@
 
     virtual ~SingleValueImp() {}
 
+    mutable QMutex mutex;
+
     void load_data()
     {
+        QMutexLocker locker(&mutex);
         if (!m_ready)
         {
             fill_from_db();
@@ -271,8 +274,11 @@
                 .arg(m_value_name).arg(m_table_name).arg(m_id_name);
     }
 
+    mutable QMutex mutex;
+
     void load_data()
     {
+        QMutexLocker locker(&mutex);
         if (!m_ready)
         {
             fill_from_db();
@@ -730,8 +736,11 @@
         }
     }
 
+    mutable QMutex mutex;
+
     void load_data()
     {
+        QMutexLocker locker(&mutex);
         if (!m_ready)
         {
             fill_from_db();
