diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/mainserver.cpp	2014-05-05 11:59:31.641722783 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/mainserver.cpp	2014-05-05 11:59:48.834958091 +0200
@@ -443,8 +443,24 @@
 
 void MainServer::ProcessRequestWork(MythSocket *sock)
 {
+    sockListLock.lockForRead();
+    PlaybackSock *pbs = GetPlaybackBySock(sock);
+    if (pbs)
+        pbs->IncrRef();
+    sockListLock.unlock();
+
     QStringList listline;
-    if (!sock->ReadStringList(listline) || listline.empty())
+    if (pbs)
+    {
+        if (!pbs->ReadStringList(listline) || listline.empty())
+        {
+            pbs->DecrRef();
+            LOG(VB_GENERAL, LOG_INFO, "No data in ProcessRequestWork()");
+            return;
+        }
+        pbs->DecrRef();
+    }
+    else if (!sock->ReadStringList(listline) || listline.empty())
     {
         LOG(VB_GENERAL, LOG_INFO, "No data in ProcessRequestWork()");
         return;
@@ -475,7 +491,7 @@
     }
 
     sockListLock.lockForRead();
-    PlaybackSock *pbs = GetPlaybackBySock(sock);
+    pbs = GetPlaybackBySock(sock);
     if (!pbs)
     {
         sockListLock.unlock();
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/playbacksock.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/playbacksock.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/playbacksock.cpp	2014-05-05 11:59:31.641722783 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/playbacksock.cpp	2014-05-05 11:59:48.834958091 +0200
@@ -10,6 +10,7 @@
 #include "mythcorecontext.h"
 #include "mythdate.h"
 #include "inputinfo.h"
+#include "referencecounter.h"
 
 #define LOC QString("PlaybackSock: ")
 #define LOC_ERR QString("PlaybackSock, Error: ")
@@ -71,6 +72,20 @@
     return m_eventsMode;
 }
 
+bool PlaybackSock::ReadStringList(QStringList &list)
+{
+    sock->IncrRef();
+    ReferenceLocker rlocker(sock);
+    QMutexLocker locker(&sockLock);
+    if (!sock->IsDataAvailable())
+    {
+        LOG(VB_GENERAL, LOG_DEBUG,
+            "PlaybackSock::ReadStringList(): Data vanished !!!");
+        return false;
+    }
+    return sock->ReadStringList(list);
+}
+
 bool PlaybackSock::SendReceiveStringList(
     QStringList &strlist, uint min_reply_length)
 {
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/playbacksock.h mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/playbacksock.h
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/playbacksock.h	2014-05-05 11:59:31.641722783 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/playbacksock.h	2014-05-05 11:59:48.834958091 +0200
@@ -101,6 +101,8 @@
 
     QStringList ForwardRequest(const QStringList&);
 
+    bool ReadStringList(QStringList &list);
+
   private:
     bool SendReceiveStringList(QStringList &strlist, uint min_reply_length = 0);
 
