diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-05 11:59:43.301656299 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-05 11:59:56.021583803 +0200
@@ -3918,13 +3918,13 @@
     bool need_clear = false;
     while (decoderSeek >= 0)
     {
-        usleep(1000);
+        usleep(50 * 1000);
 
         // provide some on screen feedback if seeking is slow
         count++;
-        if (!(count % 150) && !hasFullPositionMap)
+        if (!(count % 3) && !hasFullPositionMap)
         {
-            int num = (count / 150) % 4;
+            int num = count % 3;
             SetOSDMessage(tr("Searching") + QString().fill('.', num),
                           kOSDTimeout_Short);
             DisplayPauseFrame();
