diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/ringbuffer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/ringbuffer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/ringbuffer.cpp	2014-05-05 12:02:42.773969789 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/ringbuffer.cpp	2014-05-05 12:03:03.033855069 +0200
@@ -1032,6 +1032,7 @@
             {
                 poslock.lockForWrite();
                 rbwlock.lockForWrite();
+
                 if (rbwposcopy == rbwpos)
                 {
                     internalreadpos += read_return;
@@ -1040,6 +1041,8 @@
                         LOC + QString("rbwpos += %1K requested %2K in read")
                         .arg(read_return/1024,3).arg(totfree/1024,3));
                 }
+                numfailures = 0;
+
                 rbwlock.unlock();
                 poslock.unlock();
 
