diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_play.cpp	2014-05-05 12:03:09.790483495 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_play.cpp	2014-05-05 12:03:21.180419050 +0200
@@ -2436,8 +2436,8 @@
             if (ctx->IsPlayerErrored())
             {
                 ShowNotificationError(ctx->player->GetError(),
-                                      _Location,
-                                      buffer->GetFilename());
+                                      TV::tr( "TV Player" ),
+                                      playbackURL);
                 // We're going to display this error as notification
                 // no need to display it later as popup
                 ctx->player->ResetErrored();
