diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/ringbuffer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/ringbuffer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/ringbuffer.cpp	2014-05-05 12:03:16.507112156 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/ringbuffer.cpp	2014-05-05 12:03:35.923669004 +0200
@@ -1400,15 +1400,6 @@
         return -1;
     }
 
-    if (commserror)
-    {
-        LOG(VB_GENERAL, LOG_ERR, LOC + loc_desc +
-            ": Attempt to read after commserror set");
-        errno = EIO;
-        rwlock.unlock();
-        return -1;
-    }
-
     if (!readInternalMode &&
         (request_pause || stopreads || !readaheadrunning || (ignorereadpos>=0)))
     {
