diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-05 12:03:35.430338461 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-05 12:03:49.166927478 +0200
@@ -2988,6 +2988,13 @@
         return;
     }
 
+    // Check if we got a communication error, and if so pause playback
+    if (player_ctx->buffer->GetCommsError())
+    {
+        Pause();
+        player_ctx->buffer->ResetCommsError();
+    }
+
     // Handle end of file
     EofState eof = GetEof();
     if (HasReachedEof())
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/ringbuffer.h mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/ringbuffer.h
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/ringbuffer.h	2014-05-05 12:03:35.427005146 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/ringbuffer.h	2014-05-05 12:03:49.170260792 +0200
@@ -67,6 +67,8 @@
     QString   GetFilename(void)      const;
     QString   GetSubtitleFilename(void) const;
     QString   GetLastError(void)     const;
+    bool      GetCommsError(void) const { return commserror; }
+    void      ResetCommsError(void) { commserror = 0; }
 
     /// Returns value of stopreads
     /// \sa StartReads(void), StopReads(void)
