diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/main.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythfrontend/main.cpp	2014-05-05 12:03:46.043611797 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythfrontend/main.cpp	2014-05-05 12:03:59.233537263 +0200
@@ -1485,6 +1485,11 @@
     // of the MythPushButton widgets, and they don't use the themed background.
     QApplication::setDesktopSettingsAware(false);
 #endif
+#ifdef Q_OS_LINUX
+    // This makes Xlib calls thread-safe which seems to be required for hardware
+    // accelerated Flash playback to work without causing mythfrontend to abort.
+    QApplication::setAttribute(Qt::AA_X11InitThreads);
+#endif
     new QApplication(argc, argv);
     QCoreApplication::setApplicationName(MYTH_APPNAME_MYTHFRONTEND);
 
