diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/tv_rec.cpp	2014-05-05 12:03:51.403581505 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/tv_rec.cpp	2014-05-05 12:04:11.756799863 +0200
@@ -3635,7 +3635,7 @@
         request.input   = input;
 
         if (TuningOnSameMultiplex(request))
-            LOG(VB_PLAYBACK, LOG_INFO, LOC + "On same multiplex");
+            LOG(VB_CHANNEL, LOG_INFO, LOC + "On same multiplex");
 
         TuningShutdowns(request);
 
