diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmyth/programinfo.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmyth/programinfo.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmyth/programinfo.cpp	2014-05-05 12:04:03.966843858 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmyth/programinfo.cpp	2014-05-05 12:04:21.056747354 +0200
@@ -175,7 +175,7 @@
     oldrecstatus(rsUnknown),
     rectype(kNotRecording),
     dupin(kDupsInAll),
-    dupmethod(kDupCheckSubDesc),
+    dupmethod(kDupCheckSubThenDesc),
 
     // everything below this line is not serialized
     availableStatus(asAvailable),
@@ -645,7 +645,7 @@
     oldrecstatus(rsUnknown),
     rectype(_rectype),
     dupin(kDupsInAll),
-    dupmethod(kDupCheckSubDesc),
+    dupmethod(kDupCheckSubThenDesc),
 
     // everything below this line is not serialized
     availableStatus(asAvailable),
@@ -785,7 +785,7 @@
     oldrecstatus(rsUnknown),
     rectype(kNotRecording),
     dupin(kDupsInAll),
-    dupmethod(kDupCheckSubDesc),
+    dupmethod(kDupCheckSubThenDesc),
 
     // everything below this line is not serialized
     availableStatus(asAvailable),
@@ -1129,7 +1129,7 @@
 
     rectype = kNotRecording;
     dupin = kDupsInAll;
-    dupmethod = kDupCheckSubDesc;
+    dupmethod = kDupCheckSubThenDesc;
 
     sourceid = 0;
     inputid = 0;
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/recordinginfo.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/recordinginfo.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/recordinginfo.cpp	2014-05-05 12:04:03.956843916 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/recordinginfo.cpp	2014-05-05 12:04:21.663410592 +0200
@@ -1234,7 +1234,7 @@
 {
     uint erecid = parentid ? parentid : recordid;
     uint din = dupin ? dupin : kDupsInAll;
-    uint dmeth = dupmethod ? dupmethod : kDupCheckSubDesc;
+    uint dmeth = dupmethod ? dupmethod : kDupCheckSubThenDesc;
 
     MSqlQuery result(MSqlQuery::InitCon());
 
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/recordingrule.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/recordingrule.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/recordingrule.cpp	2014-05-05 12:04:03.943510657 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/recordingrule.cpp	2014-05-05 12:04:21.686743802 +0200
@@ -45,7 +45,7 @@
     m_startOffset(gCoreContext->GetNumSetting("DefaultStartOffset", 0)),
     m_endOffset(gCoreContext->GetNumSetting("DefaultEndOffset", 0)),
     m_dupMethod(static_cast<RecordingDupMethodType>(
-                gCoreContext->GetNumSetting("prefDupMethod", kDupCheckSubDesc))),
+                gCoreContext->GetNumSetting("prefDupMethod", kDupCheckSubThenDesc))),
     m_dupIn(kDupsInAll),
     m_filter(GetDefaultFilter()),
     m_recProfile(tr("Default")),
