diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/external/FFmpeg/libavcodec/mpeg12.c mythtv-0.27-20140504-ge2acf0d-new/mythtv/external/FFmpeg/libavcodec/mpeg12.c
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/external/FFmpeg/libavcodec/mpeg12.c	2014-05-05 12:04:32.726681488 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/external/FFmpeg/libavcodec/mpeg12.c	2014-05-05 12:04:41.423299088 +0200
@@ -1202,6 +1202,7 @@
 #endif
 #if CONFIG_MPEG1_VDPAU_HWACCEL
     AV_PIX_FMT_VDPAU_MPEG1,
+    AV_PIX_FMT_VDPAU,
 #endif
     AV_PIX_FMT_YUV420P,
     AV_PIX_FMT_NONE
@@ -1214,6 +1215,7 @@
 #endif
 #if CONFIG_MPEG2_VDPAU_HWACCEL
     AV_PIX_FMT_VDPAU_MPEG2,
+    AV_PIX_FMT_VDPAU,
 #endif
 #if CONFIG_MPEG2_DXVA2_HWACCEL
     AV_PIX_FMT_DXVA2_VLD,
