diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/fileringbuffer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/fileringbuffer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/fileringbuffer.cpp	2014-05-05 12:04:36.476660328 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/fileringbuffer.cpp	2014-05-05 12:04:51.756574138 +0200
@@ -615,6 +615,10 @@
 {
     long long ret = -1;
 
+    // Ticket 12128
+    StopReads();
+    StartReads();
+
     if (writemode)
     {
         ret = WriterSeek(pos, whence, true);
