diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/avformatdecoder.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/avformatdecoder.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/avformatdecoder.cpp	2014-05-05 12:05:09.096476382 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/avformatdecoder.cpp	2014-05-05 12:05:20.943076294 +0200
@@ -222,6 +222,30 @@
     return 1;
 }
 
+static bool force_sw_decode(AVCodecContext *avctx)
+{
+    switch (avctx->codec_id)
+    {
+        case AV_CODEC_ID_H264:
+            switch (avctx->profile)
+            {
+                case FF_PROFILE_H264_HIGH_10:
+                case FF_PROFILE_H264_HIGH_10_INTRA:
+                case FF_PROFILE_H264_HIGH_422:
+                case FF_PROFILE_H264_HIGH_422_INTRA:
+                case FF_PROFILE_H264_HIGH_444_PREDICTIVE:
+                case FF_PROFILE_H264_HIGH_444_INTRA:
+                case FF_PROFILE_H264_CAVLC_444:
+                    return true;
+                default:
+                    break;
+            }
+        default:
+            break;
+    }
+    return false;
+}
+
 static void myth_av_log(void *ptr, int level, const char* fmt, va_list vl)
 {
     if (silence_ffmpeg_logging)
@@ -2273,6 +2297,18 @@
             if (version)
                 video_codec_id = (MythCodecID)(kCodec_MPEG1 + version - 1);
 
+                // Check it's a codec we can decode using GPU
+            if (force_sw_decode(enc))
+            {
+                dec = "ffmpeg";
+                if (FlagIsSet(kDecodeAllowGPU))
+                {
+                    LOG(VB_PLAYBACK, LOG_WARNING, LOC +
+                        QString("Unsupported Video Profile, "
+                                "forcing software decode"));
+                }
+            }
+
             if (version)
             {
 #if defined(USING_VDPAU)
