diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/external/FFmpeg/libavcodec/vdpau_vc1.c mythtv-0.27-20140504-ge2acf0d-new/mythtv/external/FFmpeg/libavcodec/vdpau_vc1.c
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/external/FFmpeg/libavcodec/vdpau_vc1.c	2014-05-05 12:05:17.316430061 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/external/FFmpeg/libavcodec/vdpau_vc1.c	2014-05-05 12:05:31.456350409 +0200
@@ -59,7 +59,7 @@
     else
         info->picture_type  = s->pict_type - 1 + s->pict_type / 3;
 
-    info->frame_coding_mode = v->fcm;
+    info->frame_coding_mode = v->fcm ? (v->fcm + 1) : 0;
     info->postprocflag      = v->postprocflag;
     info->pulldown          = v->broadcast;
     info->interlace         = v->interlace;
