diff -Naur mythtv-0.27-20140708-gf4825ca-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-0.27-20140708-gf4825ca-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-0.27-20140708-gf4825ca-old/mythtv/libs/libmythtv/mythplayer.cpp	2014-07-08 18:09:00.014380306 +0200
+++ mythtv-0.27-20140708-gf4825ca-new/mythtv/libs/libmythtv/mythplayer.cpp	2014-07-08 22:28:36.654961654 +0200
@@ -1024,8 +1024,20 @@
     if (!gCoreContext->IsDatabaseIgnored() &&
         player_ctx->playingInfo->QueryAutoExpire() == kLiveTVAutoExpire)
     {
-        gCoreContext->SaveSetting(
-            "DefaultChanid", player_ctx->playingInfo->GetChanID());
+        int livetv_always_same_chan = gCoreContext->GetNumSetting("NoRememberLastTVChan", 0);
+        if (livetv_always_same_chan == 1)
+        {
+            LOG(VB_GENERAL, LOG_INFO,
+                "Saving DefaultChanid disabled by NoRememberLastTVChan=1");
+        }
+        else
+        {
+            gCoreContext->SaveSetting(
+                "DefaultChanid", player_ctx->playingInfo->GetChanID());
+            LOG(VB_GENERAL, LOG_INFO, LOC +
+                QString("DefaultChanid saved as '%1'").arg(player_ctx->playingInfo->GetChanID()));
+        }
+
         if (player_ctx->recorder && player_ctx->recorder->IsValidRecorder())
         {
             int cardid = player_ctx->recorder->GetRecorderNumber();
