diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-22 11:31:54.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-22 13:44:40.000000000 +0200
@@ -1017,8 +1017,12 @@
     if (!gCoreContext->IsDatabaseIgnored() &&
         player_ctx->playingInfo->QueryAutoExpire() == kLiveTVAutoExpire)
     {
-        gCoreContext->SaveSetting(
-            "DefaultChanid", player_ctx->playingInfo->GetChanID());
+        int livetv_always_same_chan = gCoreContext->GetNumSetting("NoRememberLastTVChan", 0);
+        if (livetv_always_same_chan = 0)
+        {
+            gCoreContext->SaveSetting(
+                "DefaultChanid", player_ctx->playingInfo->GetChanID());
+        }
         if (player_ctx->recorder && player_ctx->recorder->IsValidRecorder())
         {
             int cardid = player_ctx->recorder->GetRecorderNumber();
