From 4a6d58407abaef6aa6bb08d5344878918e3dc838 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Fri, 30 May 2014 19:30:04 +1000
Subject: [PATCH 08/28] No point de-interleaving mono.

---
 mythtv/libs/libmyth/audio/audioconvert.cpp |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/mythtv/libs/libmyth/audio/audioconvert.cpp b/mythtv/libs/libmyth/audio/audioconvert.cpp
index b868a61..972e5a4 100644
--- a/mythtv/libs/libmyth/audio/audioconvert.cpp
+++ b/mythtv/libs/libmyth/audio/audioconvert.cpp
@@ -758,6 +758,13 @@ void AudioConvert::DeinterleaveSamples(AudioFormat format, int channels,
                                           uint8_t* output, const uint8_t* input,
                                           int data_size)
 {
+    if (channels == 1)
+    {
+        // If channel count is 1, planar and non-planar formats are the same
+        memcpy(output, input, data_size);
+        return;
+    }
+
     int bits = AudioOutputSettings::FormatToBits(format);
     if (bits == 8)
     {
-- 
1.7.10.2

