diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/programs/mythtranscode/mpeg2fix.cpp mythtv-0.27-20140519-gf4825ca-new/mythtv/programs/mythtranscode/mpeg2fix.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/programs/mythtranscode/mpeg2fix.cpp	2014-06-01 00:42:48.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/programs/mythtranscode/mpeg2fix.cpp	2014-06-01 00:48:16.741903455 +0200
@@ -206,6 +206,7 @@
                        const char *fmt, int norp, int fixPTS, int maxf,
                        bool showprog, int otype, void (*update_func)(float),
                        int (*check_func)())
+            : inputFC(NULL), picture(NULL)
 {
     displayFrame = 0;
 
@@ -283,6 +284,7 @@
 
     if (inputFC)
         avformat_close_input(&inputFC);
+    av_frame_free(&picture);
 
     MPEG2frame *tmpFrame;
 
@@ -1087,7 +1089,17 @@
         WriteYUV(tmpstr, info);
     }
 
-    picture = avcodec_alloc_frame();
+    if (!picture)
+    {
+        if (!(picture = av_frame_alloc()))
+        {
+            return true;
+        }
+    }
+    else
+    {
+        av_frame_unref(picture);
+    }
 
     pkt->data = (uint8_t *)av_malloc(outbuf_size);
 
@@ -1128,7 +1140,6 @@
 
     if (!out_codec)
     {
-        free(picture);
         LOG(VB_GENERAL, LOG_ERR, "Couldn't find MPEG2 encoder");
         return true;
     }
@@ -1171,7 +1182,6 @@
 
     if (avcodec_open2(c, out_codec, NULL) < 0)
     {
-        free(picture);
         LOG(VB_GENERAL, LOG_ERR, "could not open codec");
         return true;
     }
@@ -1189,7 +1199,6 @@
 
         if (ret < 0)
         {
-            free(picture);
             LOG(VB_GENERAL, LOG_ERR,
                 QString("avcodec_encode_video2 failed (%1)").arg(ret));
             return true;
@@ -1214,7 +1223,6 @@
 
     avcodec_close(c);
     av_freep(&c);
-    av_freep(&picture);
 
     return false;
 }
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/programs/mythtranscode/mpeg2fix.h mythtv-0.27-20140519-gf4825ca-new/mythtv/programs/mythtranscode/mpeg2fix.h
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/programs/mythtranscode/mpeg2fix.h	2014-06-01 00:42:48.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/programs/mythtranscode/mpeg2fix.h	2014-06-01 00:44:57.139297751 +0200
@@ -247,6 +247,8 @@
     pthread_t thread;
 
     AVFormatContext *inputFC;
+    AVFrame *picture;
+
     int vid_id;
     int ext_count;
     QMap <int, int> aud_map;
