From bce987c017cb1d89102cb9de484adf598b2a808d Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Thu, 5 Jun 2014 21:50:06 +1000
Subject: [PATCH 03/34] Fix compilation without VAAPI

---
 mythtv/libs/libmythtv/mythcodecid.h |   19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/mythtv/libs/libmythtv/mythcodecid.h b/mythtv/libs/libmythtv/mythcodecid.h
index b464133..91ce15f 100644
--- a/mythtv/libs/libmythtv/mythcodecid.h
+++ b/mythtv/libs/libmythtv/mythcodecid.h
@@ -103,12 +103,27 @@ int mpeg_version(int codec_id);
 #define CODEC_IS_H264(id)     (mpeg_version(id) == 5)
 #define CODEC_IS_MPEG(id)     (mpeg_version(id) && mpeg_version(id) <= 2)
 #define CODEC_IS_FFMPEG_MPEG(id) (CODEC_IS_MPEG(id))
+#ifdef USING_VDPAU
 #define CODEC_IS_VDPAU(codec) (codec &&\
                                codec->capabilities & CODEC_CAP_HWACCEL_VDPAU)
+#else
+#define CODEC_IS_VDPAU(codec) (0)
+#endif
+
+#ifdef USING_VAAPI
 #define CODEC_IS_VAAPI(codec, enc) (codec && IS_VAAPI_PIX_FMT(enc->pix_fmt))
+#else
+#define CODEC_IS_VAAPI(codec, enc) (0)
+#endif
+
+#ifdef USING_DXVA2
 #define CODEC_IS_DXVA2(codec, enc) (codec && (enc->pix_fmt == PIX_FMT_DXVA2_VLD))
-#define CODEC_IS_HWACCEL(codec, enc) (CODEC_IS_VDPAU(codec) ||\
-                                      CODEC_IS_VAAPI(codec, enc) ||\
+#else
+#define CODEC_IS_DXVA2(codec, enc) (0)
+#endif
+
+#define CODEC_IS_HWACCEL(codec, enc) (CODEC_IS_VDPAU(codec)      ||     \
+                                      CODEC_IS_VAAPI(codec, enc) ||     \
                                       CODEC_IS_DXVA2(codec, enc))
 
 #endif // _MYTH_CODEC_ID_H_
-- 
1.7.10.2

