diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/README.sync mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/README.sync
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/README.sync	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/README.sync	2014-06-06 16:15:47.623466104 +0200
@@ -15,7 +15,8 @@
 git://source.ffmpeg.org/ffmpeg.git at SHA1 875649bf on July 27th, 2013 (jya) (release/1.2 branch)
 git://source.ffmpeg.org/ffmpeg.git at SHA1 f9c87262 on August 12th, 2013 (jya) (release/1.2 branch)
 git://source.ffmpeg.org/ffmpeg.git at SHA1 3d79041f on May 4th, 2014 (jya) (release/1.2 branch, release 1.2.6))
-git://source.ffmpeg.org/ffmpeg.git at n2.2.2-30-gacafd18 on May 4th, 2014 (jya) (release/2.2 branch, release 2.2.2))
+git://source.ffmpeg.org/ffmpeg.git at n2.2.2-30-gacafd18 on May 30th, 2014 (jya) (release/2.2 branch, release 2.2.2))
+git://source.ffmpeg.org/ffmpeg.git at n2.2.3 on May 6th, 2014 (jya) (release/2.2 branch, release 2.2.3))
 
 List of files modified from original FFmpeg:
 Makefile
@@ -23,44 +24,45 @@
 common.mak
 configure
 libavcodec/Makefile
-libavcodec/avcodec.h
-libavcodec/old_codec_ids.h
-libavcodec/dsputil.h
-libavcodec/dvbsub_parser.c
-libavcodec/dvbsubdec.c
+libavcodec/avcodec.h            // add subtitle support
+libavcodec/codec_desc.c         // add subtitle codec description
+libavcodec/dvbsubdec.c          // DVB subtitles fixes
+libavcodec/dvbsub_parser.c      // Fix for #5978
 libavcodec/dvdsubdec.c
 libavcodec/get_bits.h
 libavcodec/golomb.h
 libavcodec/h264_refs.c
 libavcodec/libavcodec.v
-libavcodec/mpeg12dec.c
-libavcodec/mpeg4videodec.c
+libavcodec/mpeg12dec.c          // CEA-708/608 support
+libavcodec/mpeg4videodec.c      // fixes for #3001 + fix for h263 frame rate
 libavcodec/mpeg4videoenc.c
-libavcodec/mpegaudio_parser.c
+libavcodec/mpegaudio_parser.c   // add support for dual language
+libavcodec/mpegaudio_template.c // small change to reduce verbosity
 libavcodec/mpegaudiodec.c
-libavcodec/mpegaudiodecheader.c
-libavcodec/mpegvideo.c
-libavcodec/mpegvideo.h
+libavcodec/mpegaudiodecheader.c // Add support for dual language detection
+libavcodec/mpegaudiodecheader.h
+libavcodec/mpegvideo.c          // CEA-708/608 support
+libavcodec/mpegvideo.h          // rename member as it cause compilation with C++,  CEA-708/608 support
 libavcodec/utils-mythtv.c <- new file
-libavcodec/pcm.c
-libavcodec/pgssubdec.c
+libavcodec/pcm.c                // Fix to allow playing stream marked with 0 channels
+libavcodec/pgssubdec.c          // support for forced display of PGS subtitle
 libavformat/Makefile
-libavformat/allformats.c
-libavformat/avformat.h
-libavformat/avidec.c   <-- to revisit
-libavformat/mov.c
-libavformat/mpeg.c
-libavformat/mpeg.h
+libavformat/allformats.c        // add MythTV mpegts demuxer
+libavformat/avformat.h          // extra field to all callbacks for stream change
+libavformat/mov.c               // Fix http streaming of iPhone videos (Fix from xbmc)
+libavformat/mpeg.c              // support for AAC/LATM, mpeg2vbi stream and stream change callback
+libavformat/mpeg.h              // AAC/LATM stream type definition
 libavformat/mpegts-mythtv.c  <- new file
 libavformat/mpegts-mythtv.h  <- new file
-libavformat/mpegts.c
+libavformat/mpegts.c            // rename some functions so they do not conflict with mythtv mpegts demuxer
 libavformat/mpegts.h
 libavformat/utils-mythtv.c  <- new file
-libavformat/utils.c
+libavformat/utils.c             // fix to make Myth mpegts demux higher priority than FFmpeg's own, some minor diff to prevent corner cases crash
 libavformat/rtpdec_mpegts.c <- header change
 libavformat/rtsp.c <- header change
 libavformat/wtvdec.c <- header change
 libavutil/bswap.h
+libavutil/frame.c               // CEA-708/608 AVFrame members addition
 libpostproc/Makefile
 libpostproc/postprocess.c
-library.mak
+library.mak                     // change so it doesn't install pkgconfig info
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/RELEASE mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/RELEASE
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/RELEASE	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/RELEASE	2014-06-06 16:15:42.780135704 +0200
@@ -1 +1 @@
-2.2.2
+2.2.3
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/VERSION mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/VERSION
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/VERSION	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/VERSION	2014-06-06 16:15:42.780135704 +0200
@@ -1 +1 @@
-2.2.2
+2.2.3
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/compat/avisynth/avisynth_c.h mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/compat/avisynth/avisynth_c.h
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/compat/avisynth/avisynth_c.h	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/compat/avisynth/avisynth_c.h	2014-06-06 16:13:39.970032397 +0200
@@ -13,7 +13,8 @@
 //
 // You should have received a copy of the GNU General Public License
 // along with this program; if not, write to the Free Software
-// Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA, or visit
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+// MA 02110-1301 USA, or visit
 // http://www.gnu.org/copyleft/gpl.html .
 //
 // As a special exception, I give you permission to link to the
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/compat/avisynth/avxsynth_c.h mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/compat/avisynth/avxsynth_c.h
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/compat/avisynth/avxsynth_c.h	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/compat/avisynth/avxsynth_c.h	2014-06-06 16:13:39.970032397 +0200
@@ -13,7 +13,8 @@
 //
 // You should have received a copy of the GNU General Public License
 // along with this program; if not, write to the Free Software
-// Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA, or visit
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+// MA 02110-1301 USA, or visit
 // http://www.gnu.org/copyleft/gpl.html .
 //
 // As a special exception, I give you permission to link to the
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/configure mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/configure
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/configure	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/configure	2014-06-06 16:13:33.233371549 +0200
@@ -3386,7 +3386,7 @@
     tilegx|tile-gx)
         arch="tilegx"
     ;;
-    i[3-6]86|i86pc|BePC|x86pc|x86_64|x86_32|amd64)
+    i[3-6]86*|i86pc|BePC|x86pc|x86_64|x86_32|amd64)
         arch="x86"
     ;;
 esac
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/Doxyfile mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/Doxyfile
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/Doxyfile	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/Doxyfile	2014-06-06 16:15:42.780135704 +0200
@@ -31,7 +31,7 @@
 # This could be handy for archiving the generated documentation or
 # if some version control system is used.
 
-PROJECT_NUMBER         = 2.2.2
+PROJECT_NUMBER         = 2.2.3
 
 # With the PROJECT_LOGO tag one can specify a logo or icon that is included
 # in the documentation. The maximum height of the logo should not exceed 55
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/demuxers.texi mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/demuxers.texi
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/demuxers.texi	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/demuxers.texi	2014-06-06 16:13:46.493388313 +0200
@@ -296,7 +296,7 @@
 
 Raw video demuxer.
 
-This demuxer allows to read raw video data. Since there is no header
+This demuxer allows one to read raw video data. Since there is no header
 specifying the assumed video parameters, the user must specify them
 in order to be able to decode the data correctly.
 
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/encoders.texi mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/encoders.texi
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/encoders.texi	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/encoders.texi	2014-06-06 16:13:46.496721471 +0200
@@ -807,7 +807,7 @@
 
 @item reservoir
 Enable use of bit reservoir when set to 1. Default value is 1. LAME
-has this enabled by default, but can be overriden by use
+has this enabled by default, but can be overridden by use
 @option{--nores} option.
 
 @item joint_stereo (@emph{-m j})
@@ -1525,7 +1525,7 @@
 Many libx264 encoder options are mapped to FFmpeg global codec
 options, while unique encoder options are provided through private
 options. Additionally the @option{x264opts} and @option{x264-params}
-private options allows to pass a list of key=value tuples as accepted
+private options allows one to pass a list of key=value tuples as accepted
 by the libx264 @code{x264_param_parse} function.
 
 The x264 project website is at
@@ -1853,7 +1853,7 @@
 parameters.
 
 This option is functionally the same as the @option{x264opts}, but is
-duplicated for compability with the Libav fork.
+duplicated for compatibility with the Libav fork.
 
 For example to specify libx264 encoding options with @command{ffmpeg}:
 @example
@@ -2047,7 +2047,7 @@
 Apple ProRes encoder.
 
 FFmpeg contains 2 ProRes encoders, the prores-aw and prores-ks encoder.
-The used encoder can be choosen with the @code{-vcodec} option.
+The used encoder can be chosen with the @code{-vcodec} option.
 
 @subsection Private Options for prores-ks
 
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/ffmpeg-resampler.texi mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/ffmpeg-resampler.texi
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/ffmpeg-resampler.texi	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/ffmpeg-resampler.texi	2014-06-06 16:13:46.496721471 +0200
@@ -14,7 +14,7 @@
 
 The FFmpeg resampler provides a high-level interface to the
 libswresample library audio resampling utilities. In particular it
-allows to perform audio resampling, audio channel layout rematrixing,
+allows one to perform audio resampling, audio channel layout rematrixing,
 and convert audio format and packing layout.
 
 @c man end DESCRIPTION
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/ffmpeg-scaler.texi mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/ffmpeg-scaler.texi
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/ffmpeg-scaler.texi	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/ffmpeg-scaler.texi	2014-06-06 16:13:46.496721471 +0200
@@ -13,7 +13,7 @@
 @c man begin DESCRIPTION
 
 The FFmpeg rescaler provides a high-level interface to the libswscale
-library image conversion utilities. In particular it allows to perform
+library image conversion utilities. In particular it allows one to perform
 image rescaling and pixel format conversion.
 
 @c man end DESCRIPTION
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/filters.texi mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/filters.texi
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/filters.texi	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/filters.texi	2014-06-06 16:13:46.496721471 +0200
@@ -172,7 +172,7 @@
 
 The name and arguments of the filter are optionally preceded and
 followed by a list of link labels.
-A link label allows to name a link and associate it to a filter output
+A link label allows one to name a link and associate it to a filter output
 or input pad. The preceding labels @var{in_link_1}
 ... @var{in_link_N}, are associated to the filter input pads,
 the following labels @var{out_link_1} ... @var{out_link_M}, are
@@ -3921,7 +3921,7 @@
 connected through 8-connectivity with the "weak" edge pixels selected
 by the low threshold.
 
-@var{low} and @var{high} threshold values must be choosen in the range
+@var{low} and @var{high} threshold values must be chosen in the range
 [0,1], and @var{low} should be lesser or equal to @var{high}.
 
 Default value for @var{low} is @code{20/255}, and default value for @var{high}
@@ -5238,7 +5238,7 @@
 
 Deinterleave or interleave fields.
 
-This filter allows to process interlaced images fields without
+This filter allows one to process interlaced images fields without
 deinterlacing them. Deinterleaving splits the input frame into 2
 fields (so called half pictures). Odd lines are moved to the top
 half of the output image, even lines to the bottom half.
@@ -6860,7 +6860,7 @@
 @end example
 
 @item
-Rotate the video, output size is choosen so that the whole rotating
+Rotate the video, output size is chosen so that the whole rotating
 input video is always completely contained in the output:
 @example
 rotate='2*PI*t:ow=hypot(iw,ih):oh=ow'
@@ -6983,7 +6983,7 @@
 @item flags
 Set libswscale scaling flags. See
 @ref{sws_flags,,the ffmpeg-scaler manual,ffmpeg-scaler} for the
-complete list of values. If not explictly specified the filter applies
+complete list of values. If not explicitly specified the filter applies
 the default flags.
 
 @item size, s
@@ -7797,7 +7797,7 @@
 ffmpeg -skip_frame nokey -i file.avi -vf 'scale=128:72,tile=8x8' -an -vsync 0 keyframes%03d.png
 @end example
 The @option{-vsync 0} is necessary to prevent @command{ffmpeg} from
-duplicating each output frame to accomodate the originally detected frame
+duplicating each output frame to accommodate the originally detected frame
 rate.
 
 @item
@@ -8394,7 +8394,7 @@
 (enabled).
 
 @item aspect
-Set vignette aspect. This setting allows to adjust the shape of the vignette.
+Set vignette aspect. This setting allows one to adjust the shape of the vignette.
 Setting this value to the SAR of the input will make a rectangular vignetting
 following the dimensions of the video.
 
@@ -8903,7 +8903,7 @@
 
 At each interaction the grid evolves according to the adopted rule,
 which specifies the number of neighbor alive cells which will make a
-cell stay alive or born. The @option{rule} option allows to specify
+cell stay alive or born. The @option{rule} option allows one to specify
 the rule to adopt.
 
 This source accepts the following options:
@@ -10323,7 +10323,7 @@
 changed, so it will generate non monotonically increasing timestamps.
 @end table
 
-This filter allows to overlay a second video on top of main input of
+This filter allows one to overlay a second video on top of main input of
 a filtergraph as shown in this graph:
 @example
 input -----------> deltapts0 --> overlay --> output
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/git-howto.texi mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/git-howto.texi
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/git-howto.texi	2014-05-19 15:29:18.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/git-howto.texi	2014-06-06 16:13:46.496721471 +0200
@@ -299,7 +299,7 @@
 git commit --amend
 @end example
 
-allows to amend the last commit details quickly.
+allows one to amend the last commit details quickly.
 
 @example
 git rebase -i origin/master
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/indevs.texi mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/indevs.texi
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/indevs.texi	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/indevs.texi	2014-06-06 16:13:46.496721471 +0200
@@ -409,7 +409,7 @@
 See @url{http://developer.apple.com/technologies/mac/audio-and-video.html}
 @end table
 
-This device allows to capture from an audio input device handled
+This device allows one to capture from an audio input device handled
 through OpenAL.
 
 You need to specify the name of the device to capture in the provided
@@ -617,7 +617,7 @@
 
 @item input_format
 Set the preferred pixel format (for raw video) or a codec name.
-This option allows to select the input format, when several are
+This option allows one to select the input format, when several are
 available.
 
 @item framerate
@@ -678,7 +678,7 @@
 
 X11 video input device.
 
-This device allows to capture a region of an X11 display.
+This device allows one to capture a region of an X11 display.
 
 The filename passed as input has the syntax:
 @example
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/muxers.texi mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/muxers.texi
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/muxers.texi	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/muxers.texi	2014-06-06 16:13:46.500054602 +0200
@@ -758,7 +758,7 @@
 @table @option
 @item reference_stream @var{specifier}
 Set the reference stream, as specified by the string @var{specifier}.
-If @var{specifier} is set to @code{auto}, the reference is choosen
+If @var{specifier} is set to @code{auto}, the reference is chosen
 automatically. Otherwise it must be a stream specifier (see the ``Stream
 specifiers'' chapter in the ffmpeg manual) which specifies the
 reference stream. The default value is @code{auto}.
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/outdevs.texi mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/outdevs.texi
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/outdevs.texi	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/outdevs.texi	2014-06-06 16:13:46.500054602 +0200
@@ -42,7 +42,7 @@
 
 CACA output device.
 
-This output device allows to show a video stream in CACA window.
+This output device allows one to show a video stream in CACA window.
 Only one CACA window is allowed per application, so you can
 have only one instance of this output device in an application.
 
@@ -302,7 +302,7 @@
 
 SDL (Simple DirectMedia Layer) output device.
 
-This output device allows to show a video stream in an SDL
+This output device allows one to show a video stream in an SDL
 window. Only one SDL window is allowed per application, so you can
 have only one instance of this output device in an application.
 
@@ -361,7 +361,7 @@
 
 XV (XVideo) output device.
 
-This output device allows to show a video stream in a X Window System
+This output device allows one to show a video stream in a X Window System
 window.
 
 @subsection Options
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/protocols.texi mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/protocols.texi
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/protocols.texi	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/protocols.texi	2014-06-06 16:13:46.500054602 +0200
@@ -960,7 +960,7 @@
 
 @section tcp
 
-Trasmission Control Protocol.
+Transmission Control Protocol.
 
 The required syntax for a TCP url is:
 @example
@@ -1066,7 +1066,7 @@
 @var{options} contains a list of &-separated options of the form @var{key}=@var{val}.
 
 In case threading is enabled on the system, a circular buffer is used
-to store the incoming data, which allows to reduce loss of data due to
+to store the incoming data, which allows one to reduce loss of data due to
 UDP socket buffer overruns. The @var{fifo_size} and
 @var{overrun_nonfatal} options are related to this buffer.
 
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/utils.texi mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/utils.texi
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/doc/utils.texi	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/doc/utils.texi	2014-06-06 16:13:46.500054602 +0200
@@ -1056,7 +1056,7 @@
 @item device_idx
 Select the index of the device used to run OpenCL code.
 
-The specifed index must be one of the indexes in the device list which
+The specified index must be one of the indexes in the device list which
 can be obtained with @code{ffmpeg -opencl_bench} or @code{av_opencl_get_device_list()}.
 
 @end table
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/aac.h mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/aac.h
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/aac.h	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/aac.h	2014-06-06 16:14:29.286742040 +0200
@@ -81,7 +81,7 @@
     INTENSITY_BT   = 15,    ///< Scalefactor data are intensity stereo positions.
 };
 
-#define IS_CODEBOOK_UNSIGNED(x) ((x - 1) & 10)
+#define IS_CODEBOOK_UNSIGNED(x) (((x) - 1) & 10)
 
 enum ChannelPosition {
     AAC_CHANNEL_OFF   = 0,
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/dcadec.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/dcadec.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/dcadec.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/dcadec.c	2014-06-06 16:13:56.816733946 +0200
@@ -2426,7 +2426,7 @@
          * masks in some sense -- unfortunately some channels could overlap */
         if (av_popcount(channel_mask) != av_popcount(channel_layout)) {
             av_log(avctx, AV_LOG_DEBUG,
-                   "DTS-XXCH: Inconsistant avcodec/dts channel layouts\n");
+                   "DTS-XXCH: Inconsistent avcodec/dts channel layouts\n");
             return AVERROR_INVALIDDATA;
         }
 
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/diracdec.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/diracdec.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/diracdec.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/diracdec.c	2014-06-06 16:14:11.536737937 +0200
@@ -201,6 +201,7 @@
 
     uint16_t *mctmp;            /* buffer holding the MC data multipled by OBMC weights */
     uint8_t *mcscratch;
+    int buffer_stride;
 
     DECLARE_ALIGNED(16, uint8_t, obmc_weight)[3][MAX_BLOCKSIZE*MAX_BLOCKSIZE];
 
@@ -343,19 +344,41 @@
             return AVERROR(ENOMEM);
     }
 
-    w = s->source.width;
-    h = s->source.height;
-
     /* fixme: allocate using real stride here */
-    s->sbsplit  = av_malloc(sbwidth * sbheight);
-    s->blmotion = av_malloc(sbwidth * sbheight * 16 * sizeof(*s->blmotion));
-    s->edge_emu_buffer_base = av_malloc((w+64)*MAX_BLOCKSIZE);
+    s->sbsplit  = av_malloc_array(sbwidth, sbheight);
+    s->blmotion = av_malloc_array(sbwidth, sbheight * 16 * sizeof(*s->blmotion));
+
+    if (!s->sbsplit || !s->blmotion)
+        return AVERROR(ENOMEM);
+    return 0;
+}
+
+static int alloc_buffers(DiracContext *s, int stride)
+{
+    int w = s->source.width;
+    int h = s->source.height;
+
+    av_assert0(stride >= w);
+    stride += 64;
+
+    if (s->buffer_stride >= stride)
+        return 0;
+    s->buffer_stride = 0;
+
+    av_freep(&s->edge_emu_buffer_base);
+    memset(s->edge_emu_buffer, 0, sizeof(s->edge_emu_buffer));
+    av_freep(&s->mctmp);
+    av_freep(&s->mcscratch);
 
-    s->mctmp     = av_malloc((w+64+MAX_BLOCKSIZE) * (h+MAX_BLOCKSIZE) * sizeof(*s->mctmp));
-    s->mcscratch = av_malloc((w+64)*MAX_BLOCKSIZE);
+    s->edge_emu_buffer_base = av_malloc_array(stride, MAX_BLOCKSIZE);
 
-    if (!s->sbsplit || !s->blmotion || !s->mctmp || !s->mcscratch)
+    s->mctmp     = av_malloc_array((stride+MAX_BLOCKSIZE), (h+MAX_BLOCKSIZE) * sizeof(*s->mctmp));
+    s->mcscratch = av_malloc_array(stride, MAX_BLOCKSIZE);
+
+    if (!s->edge_emu_buffer_base || !s->mctmp || !s->mcscratch)
         return AVERROR(ENOMEM);
+
+    s->buffer_stride = stride;
     return 0;
 }
 
@@ -382,6 +405,7 @@
         av_freep(&s->plane[i].idwt_tmp);
     }
 
+    s->buffer_stride = 0;
     av_freep(&s->sbsplit);
     av_freep(&s->blmotion);
     av_freep(&s->edge_emu_buffer_base);
@@ -1854,6 +1878,9 @@
         s->plane[1].stride = pic->avframe->linesize[1];
         s->plane[2].stride = pic->avframe->linesize[2];
 
+        if (alloc_buffers(s, FFMAX3(FFABS(s->plane[0].stride), FFABS(s->plane[1].stride), FFABS(s->plane[2].stride))) < 0)
+            return AVERROR(ENOMEM);
+
         /* [DIRAC_STD] 11.1 Picture parse. picture_parse() */
         if (dirac_decode_picture_header(s))
             return -1;
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/golomb-test.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/golomb-test.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/golomb-test.c	2014-05-19 15:29:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/golomb-test.c	2014-06-06 16:14:34.703422211 +0200
@@ -58,7 +58,7 @@
         }
     }
 
-#define EXTEND(i) (i << 3 | i & 7)
+#define EXTEND(i) ((i) << 3 | (i) & 7)
     init_put_bits(&pb, temp, SIZE);
     for (i = 0; i < COUNT; i++)
         set_ue_golomb(&pb, EXTEND(i));
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/h264.h mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/h264.h
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/h264.h	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/h264.h	2014-06-06 16:14:43.933441541 +0200
@@ -61,10 +61,10 @@
 #define MAX_SLICES 16
 
 #ifdef ALLOW_INTERLACE
-#define MB_MBAFF(h)    h->mb_mbaff
-#define MB_FIELD(h)    h->mb_field_decoding_flag
-#define FRAME_MBAFF(h) h->mb_aff_frame
-#define FIELD_PICTURE(h) (h->picture_structure != PICT_FRAME)
+#define MB_MBAFF(h)    (h)->mb_mbaff
+#define MB_FIELD(h)    (h)->mb_field_decoding_flag
+#define FRAME_MBAFF(h) (h)->mb_aff_frame
+#define FIELD_PICTURE(h) ((h)->picture_structure != PICT_FRAME)
 #define LEFT_MBS 2
 #define LTOP     0
 #define LBOT     1
@@ -84,12 +84,12 @@
 #define FIELD_OR_MBAFF_PICTURE(h) (FRAME_MBAFF(h) || FIELD_PICTURE(h))
 
 #ifndef CABAC
-#define CABAC(h) h->pps.cabac
+#define CABAC(h) (h)->pps.cabac
 #endif
 
-#define CHROMA(h)    (h->sps.chroma_format_idc)
-#define CHROMA422(h) (h->sps.chroma_format_idc == 2)
-#define CHROMA444(h) (h->sps.chroma_format_idc == 3)
+#define CHROMA(h)    ((h)->sps.chroma_format_idc)
+#define CHROMA422(h) ((h)->sps.chroma_format_idc == 2)
+#define CHROMA444(h) ((h)->sps.chroma_format_idc == 3)
 
 #define EXTENDED_SAR       255
 
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/hevc.h mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/hevc.h
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/hevc.h	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/hevc.h	2014-06-06 16:14:49.286784266 +0200
@@ -81,10 +81,10 @@
 #define SAMPLE_CTB(tab, x, y) ((tab)[(y) * min_cb_width + (x)])
 #define SAMPLE_CBF(tab, x, y) ((tab)[((y) & ((1<<log2_trafo_size)-1)) * MAX_CU_SIZE + ((x) & ((1<<log2_trafo_size)-1))])
 
-#define IS_IDR(s) (s->nal_unit_type == NAL_IDR_W_RADL || s->nal_unit_type == NAL_IDR_N_LP)
-#define IS_BLA(s) (s->nal_unit_type == NAL_BLA_W_RADL || s->nal_unit_type == NAL_BLA_W_LP || \
-                   s->nal_unit_type == NAL_BLA_N_LP)
-#define IS_IRAP(s) (s->nal_unit_type >= 16 && s->nal_unit_type <= 23)
+#define IS_IDR(s) ((s)->nal_unit_type == NAL_IDR_W_RADL || (s)->nal_unit_type == NAL_IDR_N_LP)
+#define IS_BLA(s) ((s)->nal_unit_type == NAL_BLA_W_RADL || (s)->nal_unit_type == NAL_BLA_W_LP || \
+                   (s)->nal_unit_type == NAL_BLA_N_LP)
+#define IS_IRAP(s) ((s)->nal_unit_type >= 16 && (s)->nal_unit_type <= 23)
 
 /**
  * Table 7-3: NAL unit type codes
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/ivi_dsp.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/ivi_dsp.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/ivi_dsp.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/ivi_dsp.c	2014-06-06 16:14:53.143440953 +0200
@@ -235,15 +235,15 @@
 
 /** butterfly operation for the inverse Haar transform */
 #define IVI_HAAR_BFLY(s1, s2, o1, o2, t) \
-    t  = (s1 - s2) >> 1;\
-    o1 = (s1 + s2) >> 1;\
-    o2 = t;\
+    t  = ((s1) - (s2)) >> 1;\
+    o1 = ((s1) + (s2)) >> 1;\
+    o2 = (t);\
 
 /** inverse 8-point Haar transform */
 #define INV_HAAR8(s1, s5, s3, s7, s2, s4, s6, s8,\
                   d1, d2, d3, d4, d5, d6, d7, d8,\
                   t0, t1, t2, t3, t4, t5, t6, t7, t8) {\
-    t1 = s1 << 1; t5 = s5 << 1;\
+    t1 = (s1) << 1; t5 = (s5) << 1;\
     IVI_HAAR_BFLY(t1, t5, t1, t5, t0); IVI_HAAR_BFLY(t1, s3, t1, t3, t0);\
     IVI_HAAR_BFLY(t5, s7, t5, t7, t0); IVI_HAAR_BFLY(t1, s2, t1, t2, t0);\
     IVI_HAAR_BFLY(t3, s4, t3, t4, t0); IVI_HAAR_BFLY(t5, s6, t5, t6, t0);\
@@ -485,21 +485,21 @@
 
 /** butterfly operation for the inverse slant transform */
 #define IVI_SLANT_BFLY(s1, s2, o1, o2, t) \
-    t  = s1 - s2;\
-    o1 = s1 + s2;\
-    o2 = t;\
+    t  = (s1) - (s2);\
+    o1 = (s1) + (s2);\
+    o2 = (t);\
 
 /** This is a reflection a,b = 1/2, 5/4 for the inverse slant transform */
 #define IVI_IREFLECT(s1, s2, o1, o2, t) \
-    t  = ((s1 + s2*2 + 2) >> 2) + s1;\
-    o2 = ((s1*2 - s2 + 2) >> 2) - s2;\
-    o1 = t;\
+    t  = (((s1) + (s2)*2 + 2) >> 2) + (s1);\
+    o2 = (((s1)*2 - (s2) + 2) >> 2) - (s2);\
+    o1 = (t);\
 
 /** This is a reflection a,b = 1/2, 7/8 for the inverse slant transform */
 #define IVI_SLANT_PART4(s1, s2, o1, o2, t) \
-    t  = s2 + ((s1*4  - s2 + 4) >> 3);\
-    o2 = s1 + ((-s1 - s2*4 + 4) >> 3);\
-    o1 = t;\
+    t  = (s2) + (((s1)*4  - (s2) + 4) >> 3);\
+    o2 = (s1) + ((-(s1) - (s2)*4 + 4) >> 3);\
+    o1 = (t);\
 
 /** inverse slant8 transform */
 #define IVI_INV_SLANT8(s1, s4, s8, s5, s2, s6, s3, s7,\
@@ -557,7 +557,7 @@
     }
 #undef COMPENSATE
 
-#define COMPENSATE(x) ((x + 1)>>1)
+#define COMPENSATE(x) (((x) + 1)>>1)
     src = tmp;
     for (i = 0; i < 8; i++) {
         if (!src[0] && !src[1] && !src[2] && !src[3] && !src[4] && !src[5] && !src[6] && !src[7]) {
@@ -597,7 +597,7 @@
     }
 #undef COMPENSATE
 
-#define COMPENSATE(x) ((x + 1)>>1)
+#define COMPENSATE(x) (((x) + 1)>>1)
     src = tmp;
     for (i = 0; i < 4; i++) {
         if (!src[0] && !src[1] && !src[2] && !src[3]) {
@@ -631,7 +631,7 @@
     int     i;
     int     t0, t1, t2, t3, t4, t5, t6, t7, t8;
 
-#define COMPENSATE(x) ((x + 1)>>1)
+#define COMPENSATE(x) (((x) + 1)>>1)
     for (i = 0; i < 8; i++) {
         if (!in[0] && !in[1] && !in[2] && !in[3] && !in[4] && !in[5] && !in[6] && !in[7]) {
             memset(out, 0, 8*sizeof(out[0]));
@@ -673,7 +673,7 @@
     row4 = pitch << 2;
     row8 = pitch << 3;
 
-#define COMPENSATE(x) ((x + 1)>>1)
+#define COMPENSATE(x) (((x) + 1)>>1)
     for (i = 0; i < 8; i++) {
         if (flags[i]) {
             IVI_INV_SLANT8(in[0], in[8], in[16], in[24], in[32], in[40], in[48], in[56],
@@ -710,7 +710,7 @@
     int     i;
     int     t0, t1, t2, t3, t4;
 
-#define COMPENSATE(x) ((x + 1)>>1)
+#define COMPENSATE(x) (((x) + 1)>>1)
     for (i = 0; i < 4; i++) {
         if (!in[0] && !in[1] && !in[2] && !in[3]) {
             memset(out, 0, 4*sizeof(out[0]));
@@ -732,7 +732,7 @@
 
     row2 = pitch << 1;
 
-#define COMPENSATE(x) ((x + 1)>>1)
+#define COMPENSATE(x) (((x) + 1)>>1)
     for (i = 0; i < 4; i++) {
         if (flags[i]) {
             IVI_INV_SLANT4(in[0], in[4], in[8], in[12],
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/libvorbisenc.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/libvorbisenc.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/libvorbisenc.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/libvorbisenc.c	2014-06-06 16:14:24.216737557 +0200
@@ -349,7 +349,8 @@
             avctx->delay              = duration;
             av_assert0(!s->afq.remaining_delay);
             s->afq.frames->duration  += duration;
-            s->afq.frames->pts       -= duration;
+            if (s->afq.frames->pts != AV_NOPTS_VALUE)
+                s->afq.frames->pts       -= duration;
             s->afq.remaining_samples += duration;
         }
         ff_af_queue_remove(&s->afq, duration, &avpkt->pts, &avpkt->duration);
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/mlpdec.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/mlpdec.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/mlpdec.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/mlpdec.c	2014-06-06 16:14:58.240110413 +0200
@@ -872,7 +872,7 @@
     return 0;
 }
 
-#define MSB_MASK(bits)  (-1u << bits)
+#define MSB_MASK(bits)  (-1u << (bits))
 
 /** Generate PCM samples using the prediction filters and residual values
  *  read from the data stream, and update the filter state. */
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/mss34dsp.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/mss34dsp.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/mss34dsp.c	2014-05-19 15:29:15.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/mss34dsp.c	2014-06-06 16:15:03.360122581 +0200
@@ -84,8 +84,8 @@
     blk[6 * step] = (-(t3 + t7) + t8 + tA) >> shift;                \
     blk[7 * step] = (-(t1 + t6) + t9 + tB) >> shift;                \
 
-#define SOP_ROW(a) ((a) << 16) + 0x2000
-#define SOP_COL(a) ((a + 32) << 16)
+#define SOP_ROW(a) (((a) << 16) + 0x2000)
+#define SOP_COL(a) (((a) + 32) << 16)
 
 void ff_mss34_dct_put(uint8_t *dst, int stride, int *block)
 {
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/mss4.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/mss4.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/mss4.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/mss4.c	2014-06-06 16:15:07.886777110 +0200
@@ -364,7 +364,7 @@
     return prev[component];
 }
 
-#define MKVAL(vals)  (vals[0] | (vals[1] << 3) | (vals[2] << 6))
+#define MKVAL(vals)  ((vals)[0] | ((vals)[1] << 3) | ((vals)[2] << 6))
 
 /* Image mode - the hardest to comprehend MSS4 coding mode.
  *
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/pgssubdec.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/pgssubdec.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/pgssubdec.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/pgssubdec.c	2014-06-06 16:15:47.623466104 +0200
@@ -53,7 +53,7 @@
 typedef struct PGSSubPresentation {
     int                    id_number;
     int                    object_count;
-    int object_forced;
+    int                    object_forced;
     PGSSubPictureReference *objects;
     int64_t pts;
 } PGSSubPresentation;
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/ppc/asm.S mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/ppc/asm.S
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/ppc/asm.S	2014-05-19 15:29:15.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/ppc/asm.S	2014-06-06 16:13:26.850006173 +0200
@@ -36,12 +36,20 @@
 
 .macro extfunc name
     .global X(\name)
+#if _CALL_ELF == 2
+    .text
+X(\name):
+    addis %r2, %r12, .TOC.-X(\name)@ha
+    addi %r2, %r2, .TOC.-X(\name)@l
+    .localentry X(\name), .-X(\name)
+#else
     .section .opd, "aw"
 X(\name):
     .quad L(\name), .TOC.@tocbase, 0
     .previous
     .type X(\name), STT_FUNC
 L(\name):
+#endif
 .endm
 
 .macro movrel rd, sym, gp
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/svq3.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/svq3.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavcodec/svq3.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavcodec/svq3.c	2014-06-06 16:15:22.430123815 +0200
@@ -1187,7 +1187,7 @@
     if (h->pict_type != AV_PICTURE_TYPE_I) {
         if (!s->last_pic->f.data[0]) {
             av_log(avctx, AV_LOG_ERROR, "Missing reference frame.\n");
-            av_frame_unref(s->last_pic);
+            av_frame_unref(&s->last_pic->f);
             ret = get_buffer(avctx, s->last_pic);
             if (ret < 0)
                 return ret;
@@ -1200,7 +1200,7 @@
 
         if (h->pict_type == AV_PICTURE_TYPE_B && !s->next_pic->f.data[0]) {
             av_log(avctx, AV_LOG_ERROR, "Missing reference frame.\n");
-            av_frame_unref(s->next_pic);
+            av_frame_unref(&s->next_pic->f);
             ret = get_buffer(avctx, s->next_pic);
             if (ret < 0)
                 return ret;
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/flvenc.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/flvenc.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/flvenc.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/flvenc.c	2014-06-06 16:15:12.606780352 +0200
@@ -536,7 +536,7 @@
         sc->last_ts = ts;
 
     avio_wb24(pb, size + flags_size);
-    avio_wb24(pb, ts);
+    avio_wb24(pb, ts & 0xFFFFFF);
     avio_w8(pb, (ts >> 24) & 0x7F); // timestamps are 32 bits _signed_
     avio_wb24(pb, flv->reserved);
 
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/h263dec.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/h263dec.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/h263dec.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/h263dec.c	2014-06-06 16:15:38.783472209 +0200
@@ -31,24 +31,37 @@
     int res_change=0;
     int src_fmt, last_src_fmt=-1;
     int last_gn=0;
+    int tr, last_tr = -1;
 
     for(i=0; i<p->buf_size; i++){
         code = (code<<8) + p->buf[i];
-        if ((code & 0xfffffc0000) == 0x800000) {
-            src_fmt= (code>>2)&7;
+        if ((code & 0xfffffc000000) == 0x80000000) {
+            tr = (code >> 18) & 0xFF;
+            src_fmt= (code>>10)&7;
             if(   src_fmt != last_src_fmt
                && last_src_fmt>0 && last_src_fmt<6
                && src_fmt<6)
                 res_change++;
 
-            if((code&0x300)==0x200 && src_fmt){
+            if (tr == last_tr) {
+                invalid_psc++;
+                continue;
+            }
+
+            if (src_fmt != 7 && !(code&(1<<9)) && (code&(1<<5))) {
+                invalid_psc++;
+                continue;
+            }
+
+            if((code&0x30000)==0x20000 && src_fmt){
                 valid_psc++;
                 last_gn=0;
             }else
                 invalid_psc++;
             last_src_fmt= src_fmt;
-        } else if((code & 0xffff800000) == 0x800000) {
-            int gn= (code>>(23-5)) & 0x1F;
+            last_tr = tr;
+        } else if((code & 0xffff80000000) == 0x80000000) {
+            int gn= (code>>(31-5)) & 0x1F;
             if(gn<last_gn){
                 invalid_psc++;
             }else
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/isom.h mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/isom.h
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/isom.h	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/isom.h	2014-06-06 16:15:47.623466104 +0200
@@ -37,7 +37,7 @@
 int ff_mov_iso639_to_lang(const char lang[4], int mp4);
 int ff_mov_lang_to_iso639(unsigned code, char to[4]);
 
-/* the QuicTime file format is quite convoluted...
+/* the QuickTime file format is quite convoluted...
  * it has lots of index tables, each indexing something in another one...
  * Here we just use what is needed to read the chunks
  */
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/matroskadec.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/matroskadec.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/matroskadec.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/matroskadec.c	2014-06-06 16:15:47.623466104 +0200
@@ -2591,6 +2591,7 @@
     uint32_t *lace_size = NULL;
     int n, flags, laces = 0;
     uint64_t num;
+    int trust_default_duration = 1;
 
     if ((n = matroska_ebmlnum_uint(matroska, data, size, &num)) < 0) {
         av_log(matroska->ctx, AV_LOG_ERROR, "EBML block data error\n");
@@ -2645,7 +2646,15 @@
     if (res)
         goto end;
 
-    if (!block_duration)
+    if (track->audio.samplerate == 8000) {
+        // If this is needed for more codecs, then add them here
+        if (st->codec->codec_id == AV_CODEC_ID_AC3) {
+            if(track->audio.samplerate != st->codec->sample_rate || !st->codec->frame_size)
+                trust_default_duration = 0;
+        }
+    }
+
+    if (!block_duration && trust_default_duration)
         block_duration = track->default_duration * laces / matroska->time_scale;
 
     if (cluster_time != (uint64_t)-1 && (block_time >= 0 || cluster_time >= -block_time))
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/mov.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/mov.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/mov.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/mov.c	2014-06-06 16:14:03.526733113 +0200
@@ -2167,7 +2167,7 @@
                 if (sc->keyframe_absent
                     && !sc->stps_count
                     && !rap_group_present
-                    && st->codec->codec_type == AVMEDIA_TYPE_AUDIO)
+                    && (st->codec->codec_type == AVMEDIA_TYPE_AUDIO || (i==0 && j==0)))
                      keyframe = 1;
                 if (keyframe)
                     distance = 0;
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/utils.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/utils.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavformat/utils.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavformat/utils.c	2014-06-06 16:15:17.550114056 +0200
@@ -3324,7 +3324,8 @@
             }
         }
 #if FF_API_R_FRAME_RATE
-        ff_rfps_add_frame(ic, st, pkt->dts);
+        if (st->codec->codec_type == AVMEDIA_TYPE_VIDEO)
+            ff_rfps_add_frame(ic, st, pkt->dts);
 #endif
         if (st->parser && st->parser->parser->split && !st->codec->extradata) {
             int i = st->parser->parser->split(st->codec, pkt->data, pkt->size);
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavutil/aarch64/asm.S mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavutil/aarch64/asm.S
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavutil/aarch64/asm.S	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavutil/aarch64/asm.S	2014-06-06 16:13:19.473364603 +0200
@@ -58,8 +58,8 @@
 
 .macro  movrel rd, val
 #if CONFIG_PIC
-        adrp            \rd, #:pg_hi21:\val
-        add             \rd, \rd, #:lo12:\val
+        adrp            \rd, :pg_hi21:\val
+        add             \rd, \rd, :lo12:\val
 #else
         ldr             \rd, =\val
 #endif
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavutil/cpu.c mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavutil/cpu.c
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/external/FFmpeg/libavutil/cpu.c	2014-06-06 16:08:14.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/external/FFmpeg/libavutil/cpu.c	2014-06-06 16:14:18.896746215 +0200
@@ -47,6 +47,26 @@
 static int flags, checked;
 
 void av_force_cpu_flags(int arg){
+    if (   (arg & ( AV_CPU_FLAG_3DNOW    |
+                    AV_CPU_FLAG_3DNOWEXT |
+                    AV_CPU_FLAG_SSE      |
+                    AV_CPU_FLAG_SSE2     |
+                    AV_CPU_FLAG_SSE2SLOW |
+                    AV_CPU_FLAG_SSE3     |
+                    AV_CPU_FLAG_SSE3SLOW |
+                    AV_CPU_FLAG_SSSE3    |
+                    AV_CPU_FLAG_SSE4     |
+                    AV_CPU_FLAG_SSE42    |
+                    AV_CPU_FLAG_AVX      |
+                    AV_CPU_FLAG_XOP      |
+                    AV_CPU_FLAG_FMA3     |
+                    AV_CPU_FLAG_FMA4     |
+                    AV_CPU_FLAG_AVX2     ))
+        && !(arg & AV_CPU_FLAG_MMX)) {
+        av_log(NULL, AV_LOG_WARNING, "MMX implied by specified flags\n");
+        arg |= AV_CPU_FLAG_MMX;
+    }
+
     flags   = arg;
     checked = arg != -1;
 }
