From bc382fc990ab827237f20a24ff01e0c9b7242c25 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Sun, 29 Jun 2014 05:13:37 -0700
Subject: [PATCH] fix compilation due to missing header

---
 mythtv/libs/libmythtv/privatedecoder_crystalhd.cpp |    1 +
 1 file changed, 1 insertion(+)

diff --git a/mythtv/libs/libmythtv/privatedecoder_crystalhd.cpp b/mythtv/libs/libmythtv/privatedecoder_crystalhd.cpp
index 85b5cce..f3662c4 100644
--- a/mythtv/libs/libmythtv/privatedecoder_crystalhd.cpp
+++ b/mythtv/libs/libmythtv/privatedecoder_crystalhd.cpp
@@ -2,6 +2,7 @@
 #include "myth_imgconvert.h"
 #include "mythlogging.h"
 #include "mythavutil.h"
+#include "fourcc.h"
 
 #define LOC  QString("CrystalHD: ")
 #define ERR  QString("CrystalHD Err: ")
-- 
1.7.10.2

