From a4176cadf39804891e3f4dcf75cd6f04575f4494 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Mon, 30 Jun 2014 12:06:58 -0700
Subject: [PATCH 02/12] OSX: Allow PiP with VDA decoder

---
 mythtv/libs/libmythtv/videoout_null.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/mythtv/libs/libmythtv/videoout_null.cpp b/mythtv/libs/libmythtv/videoout_null.cpp
index eee155d..425a97d 100644
--- a/mythtv/libs/libmythtv/videoout_null.cpp
+++ b/mythtv/libs/libmythtv/videoout_null.cpp
@@ -24,6 +24,8 @@ void VideoOutputNull::GetRenderOptions(render_opts &opts,
         (*opts.safe_renderers)["ffmpeg"].append("null");
     if (opts.decoders->contains("crystalhd"))
         (*opts.safe_renderers)["crystalhd"].append("null");
+    if (opts.decoders->contains("vda"))
+        (*opts.safe_renderers)["vda"].append("null");
 
     opts.priorities->insert("null", 10);
 }
-- 
1.7.10.2

