From 89f2588eecd4b684d85779fa638a8a6e014cceb0 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Mon, 30 Jun 2014 12:20:02 -0700
Subject: [PATCH 05/12] Remove unused header

---
 mythtv/libs/libmythtv/DVD/mythdvdplayer.cpp |    1 -
 1 file changed, 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/DVD/mythdvdplayer.cpp b/mythtv/libs/libmythtv/DVD/mythdvdplayer.cpp
index 3df9b9e..6cff40d 100644
--- a/mythtv/libs/libmythtv/DVD/mythdvdplayer.cpp
+++ b/mythtv/libs/libmythtv/DVD/mythdvdplayer.cpp
@@ -1,7 +1,6 @@
 #include "dvdringbuffer.h"
 #include "DetectLetterbox.h"
 #include "audiooutput.h"
-#include "myth_imgconvert.h"
 #include "avformatdecoderdvd.h"
 #include "mythdvdplayer.h"
 
-- 
1.7.10.2

