From 8e79b0a61d898c46d3fe29a57379841fcc396f69 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Fri, 4 Jul 2014 08:01:45 +1000
Subject: [PATCH 18/19] Fix crashes when changing files

---
 mythplugins/mythmusic/mythmusic/avfdecoder.cpp |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/mythplugins/mythmusic/mythmusic/avfdecoder.cpp b/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
index 3af3289..8c8b69d 100644
--- a/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
+++ b/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
@@ -410,8 +410,11 @@ void avfDecoder::deinit()
     m_audioDec = NULL;
     m_inputFormat = NULL;
 
-    av_freep(&m_byteIOContext->buffer);
-    av_freep(&m_byteIOContext);
+    if (m_byteIOContext)
+    {
+        av_freep(&m_byteIOContext->buffer);
+        av_freep(&m_byteIOContext);
+    }
 }
 
 void avfDecoder::run()
-- 
1.7.10.2

