diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythplugins/mytharchive/mytharchivehelper/main.cpp mythtv-0.27-20140716-gf4825ca-new/mythplugins/mytharchive/mytharchivehelper/main.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythplugins/mytharchive/mytharchivehelper/main.cpp	2014-11-26 14:33:47.000000000 +0100
+++ mythtv-0.27-20140716-gf4825ca-new/mythplugins/mytharchive/mytharchivehelper/main.cpp	2014-11-26 14:36:33.709131689 +0100
@@ -1001,8 +1001,8 @@
         }
     }
 
-    QString destFile = gCoreContext->GenMythURL(gCoreContext->GetSetting("MasterServerIP"),
-                                                gCoreContext->GetSetting("MasterServerPort"),
+    QString destFile = gCoreContext->GenMythURL(gCoreContext->GetMasterServerIP(),
+                                                gCoreContext->GetMasterServerPort(),
                                                 basename , "Default");
 
     // copy file to recording directory
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netcommon.cpp mythtv-0.27-20140716-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netcommon.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netcommon.cpp	2014-07-16 11:43:18.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netcommon.cpp	2014-11-26 14:36:33.709131689 +0100
@@ -41,26 +41,8 @@
 
 QString GetMythXMLURL(void)
 {
-    // Get MasterServerIP
-    //   The data for MasterServerIP setting is the same as the data for
-    //   BackendServerIP setting
-    QString MasterIP = gCoreContext->GetSetting("MasterServerIP");
-
-    MSqlQuery query(MSqlQuery::InitCon());
-    // Get hostname of Master Server by comparing BackendServerIP to
-    // the just-retrieved MasterServerIP
-    query.prepare("SELECT hostname FROM settings "
-                  " WHERE value = 'BackendServerIP'"
-                  "   AND data = :IPADDRESS");
-    query.bindValue(":IPADDRESS", MasterIP);
-    if (!query.exec() || !query.next())
-        MythDB::DBError("Find Master Server Hostname", query);
-
-    QString MasterHost = query.value(0).toString();
-
-    // Use hostname to get BackendStatusPort
-    int MasterStatusPort = gCoreContext->GetNumSettingOnHost("BackendStatusPort",
-                                                          MasterHost);
+    QString MasterIP = gCoreContext->GetMasterServerIP();
+    int MasterStatusPort = gCoreContext->GetMasterServerStatusPort();
 
     return QString("http://%1:%2/InternetContent/").arg(MasterIP).arg(MasterStatusPort);
 }
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmyth/programinfo.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmyth/programinfo.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmyth/programinfo.cpp	2014-11-26 14:33:43.000000000 +0100
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmyth/programinfo.cpp	2014-11-26 14:36:33.709131689 +0100
@@ -2429,18 +2429,18 @@
         (gCoreContext->GetNumSetting("MasterBackendOverride", 0)) &&
         (RemoteCheckFile(this, false)))
     {
-        tmpURL = gCoreContext->GenMythURL(gCoreContext->GetSetting("MasterServerIP"),
-                                          gCoreContext->GetSetting("MasterServerPort").toInt(),
+        tmpURL = gCoreContext->GenMythURL(gCoreContext->GetMasterServerIP(),
+                                          gCoreContext->GetMasterServerPort(),
                                           basename);
 
         LOG(VB_FILE, LOG_INFO, LOC +
             QString("GetPlaybackURL: Found @ '%1'").arg(tmpURL));
         return tmpURL;
-        }
+    }
 
     // Fallback to streaming from the backend the recording was created on
     tmpURL = gCoreContext->GenMythURL(gCoreContext->GetBackendServerIP(hostname),
-                                      gCoreContext->GetSettingOnHost("BackendServerPort", hostname).toInt(),
+                                      gCoreContext->GetBackendServerPort(hostname),
                                       basename);
 
     LOG(VB_FILE, LOG_INFO, LOC +
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmyth/remoteutil.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmyth/remoteutil.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmyth/remoteutil.cpp	2014-07-16 11:42:35.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmyth/remoteutil.cpp	2014-11-26 14:36:33.709131689 +0100
@@ -458,8 +458,7 @@
         QString ann = QString("ANN Playback %1 0")
                         .arg(gCoreContext->GetHostName());
         QString addr = gCoreContext->GetBackendServerIP(host);
-        int port = gCoreContext->GetNumSettingOnHost("BackendServerPort",
-                                                     host, 6543);
+        int port = gCoreContext->GetBackendServerPort(host);
         bool mismatch = false;
 
         MythSocket *sock = gCoreContext->ConnectCommandSocket(
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythbase/mythcorecontext.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythbase/mythcorecontext.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythbase/mythcorecontext.cpp	2014-11-26 14:33:48.000000000 +0100
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythbase/mythcorecontext.cpp	2014-11-26 14:36:33.709131689 +0100
@@ -335,8 +335,8 @@
         return false;
     }
 
-    QString server = GetSetting("MasterServerIP", "localhost");
-    int     port   = GetNumSetting("MasterServerPort", 6543);
+    QString server = GetMasterServerIP();
+    int     port   = GetMasterServerPort();
     bool    proto_mismatch = false;
 
     if (d->m_serverSock && !d->m_serverSock->IsConnected())
@@ -605,7 +605,7 @@
 
 bool MythCoreContext::IsMasterHost(const QString &host)
 {
-    return IsThisHost(GetSetting("MasterServerIP"), host);
+    return IsThisHost(GetMasterServerIP(), host);
 }
 
 bool MythCoreContext::IsMasterBackend(void)
@@ -636,10 +636,13 @@
 
 bool MythCoreContext::IsThisHost(const QString &addr, const QString &host)
 {
-    QString thisip = GetSettingOnHost("BackendServerIP", host);
-    QString thisip6 = GetSettingOnHost("BackendServerIP6", host);
+    QHostAddress addrfix(addr);
+    addrfix.setScopeId(QString());
+    QString addrstr = addrfix.toString();
+    QString thisip  = GetBackendServerIP4(host);
+    QString thisip6 = GetBackendServerIP6(host);
 
-    return ((addr == thisip) || (addr == thisip6));
+    return ((addrstr == thisip) || (addrstr == thisip6));
 }
 
 bool MythCoreContext::IsFrontendOnly(void)
@@ -710,10 +713,10 @@
 QString MythCoreContext::GetMasterHostPrefix(const QString &storageGroup,
                                              const QString &path)
 {
-    return GenMythURL(GetSetting("MasterServerIP"),
-                        GetNumSetting("MasterServerPort", 6543),
-                        path,
-                        storageGroup);
+    return GenMythURL(GetMasterServerIP(),
+                      GetMasterServerPort(),
+                      path,
+                      storageGroup);
 }
 
 QString MythCoreContext::GetMasterHostName(void)
@@ -849,30 +852,64 @@
     return d->m_database->GetFloatSettingOnHost(key, host, defaultval);
 }
 
-#if 0
+/**
+ * \fn QString GetMasterServerIP(void)
+ * Returns the Master Backend IP address
+ * If the address is an IPv6 address, the scope Id is removed.
+ * If no master server address has been defined in the database, return localhost
+ */
 QString MythCoreContext::GetMasterServerIP(void)
 {
-    QString saddr = GetSetting("MasterServerIP");
-    QHostAddress addr(saddr);
+    QHostAddress addr(GetSetting("MasterServerIP", "localhost"));
+    addr.setScopeId(QString());
 
-    if (!d->m_IPv6.empty() &&
-            (addr.protocol() == QAbstractSocket::IPv6Protocol))
-        // we have IPv6 addresses, assume we can connect to them
-        return saddr;
-    else if (!d->m_IPv4.empty() &&
-            (addr.protocol() == QAbstractSocket::IPv4Protocol))
-        // we have IPv4 addresses, assume we can connect to them
-        return saddr;
-    else
-        return GetBackendServerIP(GetMasterHostName());
+    return addr.toString();
 }
-#endif
 
+/**
+ * \fn int GetMasterServerPort(void)
+ * Returns the Master Backend control port
+ * If no master server port has been defined in the database, return the default
+ * 6543
+ */
+int MythCoreContext::GetMasterServerPort(void)
+{
+    return GetNumSetting("MasterServerPort", 6543);
+}
+
+/**
+ * \fn int GetMasterServerStatusPort(void)
+ * Returns the Master Backend status port
+ * If no master server status port has been defined in the database,
+ * return the default 65434
+ */
+int MythCoreContext::GetMasterServerStatusPort(void)
+{
+    QString masterhost = GetMasterHostName();
+
+    return GetBackendStatusPort(masterhost);
+}
+
+/**
+ * \fn QString GetBackendServerIP(void)
+ * Returns the IP address of the locally defined backend IP.
+ * See GetBackendServerIP(host)
+ */
 QString MythCoreContext::GetBackendServerIP(void)
 {
     return GetBackendServerIP(d->m_localHostname);
 }
 
+/**
+ * \fn QString GetBackendServerIP(const QString &host)
+ * Returns the IP address of backend "host"
+ * If an IPv6 address has been defined, will use it whenever possible, unless
+ * the IPv6 address is a localhost address, in which case the IPv4 address will
+ * be used.
+ * Will return an empty string if neither an IPv4 nor an IPv6 address has been
+ * defined.
+ * The address returned is free of scope Id if IPv6
+ */
 QString MythCoreContext::GetBackendServerIP(const QString &host)
 {
     QString addr4, addr6;
@@ -880,22 +917,18 @@
     if (!ServerPool::DefaultListenIPv6().isEmpty())
     {
         // we have IPv6 addresses, assume we can connect to them
-        addr6 = GetSettingOnHost("BackendServerIP6", host, "");
-        QHostAddress addr(addr6);
-        // remove scope Id
-        addr.setScopeId(QString());
-        addr6 = addr.toString();
+        addr6 = GetBackendServerIP6(host);
     }
 #endif
     if (!ServerPool::DefaultListenIPv4().isEmpty())
-        addr4 = GetSettingOnHost("BackendServerIP", host, "");
+        addr4 = GetBackendServerIP4(host);
 
     if (addr6.isEmpty())
     {
         if (addr4.isEmpty())
         {
             LOG(VB_GENERAL, LOG_ERR, "No address defined for host: "+host);
-            return "";
+            return QString();
         }
 
         // IPv6 is empty, so return this regardless
@@ -911,6 +944,102 @@
 }
 
 /**
+ * \fn QStringGetBackendServerIP4(void)
+ * Returns the IPv4 address defined for the current host
+ * see GetBackendServerIP4(host)
+ */
+QString MythCoreContext::GetBackendServerIP4(void)
+{
+    return GetBackendServerIP4(d->m_localHostname);
+}
+
+/**
+ * \fn QString GetBackendServerIP4(const QString &host)
+ * Returns the IPv4 address defined for the backend "host".
+ * returns an empty string if the defined IP is invalid
+ */
+QString MythCoreContext::GetBackendServerIP4(const QString &host)
+{
+    QHostAddress addr(GetSettingOnHost("BackendServerIP", host, ""));
+
+    if (!addr.isNull())
+    {
+        return addr.toString();
+    }
+    else
+    {
+        return QString();
+    }
+}
+
+/**
+ * \fn QString GetBackendServerIP6(void)
+ * Returns the IPv6 address defined for the current host
+ * see GetBackendServerIP6(host)
+ */
+QString MythCoreContext::GetBackendServerIP6(void)
+{
+    return GetBackendServerIP4(d->m_localHostname);
+}
+
+/**
+ * \fn QString GetBackendServerIP6(const QString &host)
+ * Returns the IPv6 address defined for the backend "host".
+ * returns an empty string if the defined IP is invalid.
+ * The IP address returned doesn't contain a scope Id
+ */
+QString MythCoreContext::GetBackendServerIP6(const QString &host)
+{
+    QHostAddress addr(GetSettingOnHost("BackendServerIP6", host, ""));
+    if (!addr.isNull())
+    {
+        // remove scope Id
+        addr.setScopeId(QString());
+        return addr.toString();
+    }
+    else
+    {
+        return QString();
+    }
+}
+
+/**
+ * \fn int GetBackendServerPort(void)
+ * Returns the locally defined backend control port
+ */
+int MythCoreContext::GetBackendServerPort(void)
+{
+    return GetBackendServerPort(d->m_localHostname);
+}
+
+/**
+ * \fn vint GetBackendServerPort(const QString &host)
+ * Returns the backend "hosts"'s control port
+ */
+int MythCoreContext::GetBackendServerPort(const QString &host)
+{
+    return GetNumSettingOnHost("BackendServerPort", host, 6543);
+}
+
+/**
+ * \fn int GetBackendStatusPort(void)
+ * Returns the locally defined backend status port
+ */
+int MythCoreContext::GetBackendStatusPort(void)
+{
+    return GetBackendServerPort(d->m_localHostname);
+}
+
+/**
+ * \fn int GetBackendStatusPort(const QString &host)
+ * Returns the backend "hosts"'s status port
+ */
+int MythCoreContext::GetBackendStatusPort(const QString &host)
+{
+    return GetNumSettingOnHost("BackendStatusPort", host, 6544);
+}
+
+/**
  * \fn void GetScopeForAddress(QHostAddress &addr)
  * Return the cached scope Id for the given address.
  * If unknown returns false, else returns true and set scope Id to given address
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythbase/mythcorecontext.h mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythbase/mythcorecontext.h
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythbase/mythcorecontext.h	2014-11-26 14:33:48.000000000 +0100
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythbase/mythcorecontext.h	2014-11-26 14:36:33.712465775 +0100
@@ -155,6 +155,17 @@
 
     QString GetBackendServerIP(void);
     QString GetBackendServerIP(const QString &host);
+    QString GetBackendServerIP4(void);
+    QString GetBackendServerIP4(const QString &host);
+    QString GetBackendServerIP6(void);
+    QString GetBackendServerIP6(const QString &host);
+    QString GetMasterServerIP(void);
+    int GetMasterServerPort(void);
+    int GetMasterServerStatusPort(void);
+    int GetBackendServerPort(void);
+    int GetBackendServerPort(const QString &host);
+    int GetBackendStatusPort(void);
+    int GetBackendStatusPort(const QString &host);
 
     bool GetScopeForAddress(QHostAddress &addr) const;
     void SetScopeForAddress(const QHostAddress &addr);
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythbase/remotefile.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythbase/remotefile.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythbase/remotefile.cpp	2014-11-26 14:33:39.000000000 +0100
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythbase/remotefile.cpp	2014-11-26 14:36:33.712465775 +0100
@@ -121,11 +121,7 @@
 
     if (port <= 0)
     {
-        port = GetMythDB()->GetSettingOnHost("BackendServerPort", host).toInt();
-
-        // if we still have no port use the default
-        if (port <= 0)
-            port = 6543;
+        port = gCoreContext->GetBackendServerPort(host);
     }
 
     if (!lsock->ConnectToHost(host, port))
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythmetadata/metadataimagedownload.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythmetadata/metadataimagedownload.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythmetadata/metadataimagedownload.cpp	2014-07-16 11:42:34.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythmetadata/metadataimagedownload.cpp	2014-11-26 14:36:33.712465775 +0100
@@ -473,9 +473,8 @@
 QString getStorageGroupURL(VideoArtworkType type, QString host)
 {
     QString sgroup;
-    QString ip = gCoreContext->GetSettingOnHost("BackendServerIP", host);
-    uint port = gCoreContext->GetSettingOnHost("BackendServerPort",
-                                               host).toUInt();
+    QString ip = gCoreContext->GetBackendServerIP(host);
+    uint port = gCoreContext->GetBackendServerPort(host);
 
     if (type == kArtworkCoverart)
         sgroup = "Coverart";
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythmetadata/videoutils.h mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythmetadata/videoutils.h
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythmetadata/videoutils.h	2014-07-16 11:42:34.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythmetadata/videoutils.h	2014-11-26 14:36:33.712465775 +0100
@@ -66,8 +66,7 @@
     const QString &storage_group, const QString &host, const QString &path)
 {
     QString ip = gCoreContext->GetBackendServerIP(host);
-    uint port = gCoreContext->GetSettingOnHost("BackendServerPort",
-                                               host).toUInt();
+    uint port = gCoreContext->GetBackendServerPort(host);
 
     return gCoreContext->GenMythURL(ip,port,path,
                                     StorageGroup::GetGroupToUse(host, storage_group));
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp	2014-07-16 11:42:34.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythprotoserver/requesthandler/outboundhandler.cpp	2014-11-26 14:36:33.712465775 +0100
@@ -33,9 +33,9 @@
 
     m_socket = new MythSocket(-1, m_parent);
 
-    QString server   = gCoreContext->GetSetting("MasterServerIP", "localhost");
+    QString server   = gCoreContext->GetMasterServerIP();
     QString hostname = gCoreContext->GetMasterHostName();
-    int port         = gCoreContext->GetNumSetting("MasterServerPort", 6543);
+    int port         = gCoreContext->GetMasterServerPort();
 
     if (!m_socket->ConnectToHost(server, port))
     {
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/HLS/httplivestream.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/HLS/httplivestream.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/HLS/httplivestream.cpp	2014-11-26 14:33:29.000000000 +0100
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/HLS/httplivestream.cpp	2014-11-26 14:36:33.712465775 +0100
@@ -750,8 +750,8 @@
 
     m_httpPrefix = gCoreContext->GetSetting("HTTPLiveStreamPrefix", QString(
         "http://%1:%2/StorageGroup/Streaming/")
-        .arg(gCoreContext->GetSetting("MasterServerIP"))
-        .arg(gCoreContext->GetSetting("BackendStatusPort")));
+        .arg(gCoreContext->GetMasterServerIP())
+        .arg(gCoreContext->GetMasterServerStatusPort()));
 
     if (!m_httpPrefix.endsWith("/"))
         m_httpPrefix.append("/");
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/metadataimagehelper.h mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/metadataimagehelper.h
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/metadataimagehelper.h	2014-07-16 11:42:34.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/metadataimagehelper.h	2014-11-26 14:36:33.712465775 +0100
@@ -49,8 +49,7 @@
     const QString &storage_group, const QString &host, const QString &path)
 {
     QString ip = gCoreContext->GetBackendServerIP(host);
-    uint port = gCoreContext->GetSettingOnHost("BackendServerPort",
-                                               host).toUInt();
+    uint port = gCoreContext->GetBackendServerPort(host);
 
     return gCoreContext->GenMythURL(ip,port,path,
                                     StorageGroup::GetGroupToUse(host, storage_group));
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/tv_rec.cpp	2014-11-26 14:33:46.000000000 +0100
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/tv_rec.cpp	2014-11-26 14:36:33.712465775 +0100
@@ -2733,7 +2733,7 @@
 
     QString hostprefix = gCoreContext->GenMythURL(
                     gCoreContext->GetBackendServerIP(),
-                    gCoreContext->GetSetting("BackendServerPort").toInt());
+                    gCoreContext->GetBackendServerPort());
 
     tvchain->SetHostPrefix(hostprefix);
     tvchain->SetCardType(genOpt.cardtype);
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/httpstatus.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/httpstatus.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/httpstatus.cpp	2014-07-16 11:43:04.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/httpstatus.cpp	2014-11-26 14:36:33.712465775 +0100
@@ -330,8 +330,8 @@
     {
         numbes++;
         QString masterhost = gCoreContext->GetMasterHostName();
-        QString masterip   = gCoreContext->GetSetting("MasterServerIP");
-        QString masterport = gCoreContext->GetSettingOnHost("BackendStatusPort", masterhost, "6544");
+        QString masterip   = gCoreContext->GetMasterServerIP();
+        int masterport = gCoreContext->GetMasterServerStatusPort();
 
         QDomElement mbe = pDoc->createElement("Backend");
         backends.appendChild(mbe);
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/main_helpers.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/main_helpers.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/main_helpers.cpp	2014-11-26 14:33:26.000000000 +0100
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/main_helpers.cpp	2014-11-26 14:36:33.715799926 +0100
@@ -431,8 +431,8 @@
 {
     MythSocket *tempMonitorConnection = new MythSocket();
     if (tempMonitorConnection->ConnectToHost(
-            gCoreContext->GetSetting("MasterServerIP", "127.0.0.1"),
-            gCoreContext->GetNumSetting("MasterServerPort", 6543)))
+            gCoreContext->GetMasterServerIP(),
+            gCoreContext->GetMasterServerPort()))
     {
         if (!gCoreContext->CheckProtoVersion(tempMonitorConnection))
         {
@@ -566,9 +566,8 @@
             return ret;
     }
 
-    int     port = gCoreContext->GetNumSetting("BackendServerPort", 6543);
-    if (gCoreContext->GetSetting("BackendServerIP").isEmpty() &&
-        gCoreContext->GetSetting("BackendServerIP6").isEmpty())
+    int     port = gCoreContext->GetBackendServerPort();
+    if (gCoreContext->GetBackendServerIP().isEmpty())
     {
         cerr << "No setting found for this machine's BackendServerIP.\n"
              << "Please run setup on this machine and modify the first page\n"
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/mainserver.cpp	2014-11-26 14:33:47.000000000 +0100
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/mainserver.cpp	2014-11-26 14:44:46.839732930 +0100
@@ -37,6 +37,7 @@
 #include <QTimer>
 #include <QNetworkInterface>
 #include <QNetworkProxy>
+#include <QHostAddress>
 
 #include "previewgeneratorqueue.h"
 #include "mythmiscutil.h"
@@ -1787,7 +1788,7 @@
     QMap<QString, QString> backendIpMap;
     QMap<QString, QString> backendPortMap;
     QString ip   = gCoreContext->GetBackendServerIP();
-    QString port = gCoreContext->GetSetting("BackendServerPort");
+    int port = gCoreContext->GetBackendServerPort();
 
     ProgramList::iterator it = destination.begin();
     for (it = destination.begin(); it != destination.end(); ++it)
@@ -1856,18 +1857,18 @@
             }
             else
             {
-                ProgramInfo *p = proginfo;
-                if (!backendIpMap.contains(p->GetHostname()))
-                    backendIpMap[p->GetHostname()] =
-                        gCoreContext->GetSettingOnHost("BackendServerIp",
-                                                   p->GetHostname());
-                if (!backendPortMap.contains(p->GetHostname()))
-                    backendPortMap[p->GetHostname()] =
-                        gCoreContext->GetSettingOnHost("BackendServerPort",
-                                                   p->GetHostname());
-                p->SetPathname(gCoreContext->GenMythURL(backendIpMap[p->GetHostname()],
-                                                        backendPortMap[p->GetHostname()],
-                                                        p->GetBasename()));
+                ProgramInfo *p      = proginfo;
+                QString hostname    = p->GetHostname();
+
+                if (!backendIpMap.contains(hostname))
+                    backendIpMap[hostname] =
+                        gCoreContext->GetBackendServerIP(hostname);
+                if (!backendPortMap.contains(hostname))
+                    backendPortMap[hostname] =
+                        gCoreContext->GetBackendServerPort(hostname);
+                p->SetPathname(gCoreContext->GenMythURL(backendIpMap[hostname],
+                                                        backendPortMap[hostname],
+                                                        hostname));
             }
         }
 
@@ -1943,7 +1944,7 @@
     {
         QString lpath = GetPlaybackURL(&pginfo);
         QString ip    = gCoreContext->GetBackendServerIP();
-        QString port  = gCoreContext->GetSetting("BackendServerPort");
+        int port  = gCoreContext->GetBackendServerPort();
 
         if (playbackhost == gCoreContext->GetHostName())
             pginfo.SetPathname(lpath);
@@ -2144,7 +2145,7 @@
 
             QString url = gCoreContext->GenMythURL(
                                   gCoreContext->GetBackendServerIP(hostname),
-                                  gCoreContext->GetSettingOnHost("BackendServerPort", hostname),
+                                  gCoreContext->GetBackendServerPort(hostname),
                                   basename,
                                   storagegroup);
 
@@ -3354,6 +3355,7 @@
 
     QString host = gCoreContext->GetHostName();
     QString wantHost = sList.at(1);
+    QHostAddress wantHostaddr(wantHost);
     QString groupname = sList.at(2);
     QString path = sList.at(3);
     bool fileNamesOnly = false;
@@ -3368,8 +3370,8 @@
             .arg(groupname).arg(host).arg(path).arg(wantHost));
 
     if ((host.toLower() == wantHost.toLower()) ||
-        (gCoreContext->GetSetting("BackendServerIP") == wantHost) ||
-        (gCoreContext->GetSetting("BackendServerIP6") == wantHost))
+        (gCoreContext->GetBackendServerIP4() == wantHostaddr.toString()) ||
+        (gCoreContext->GetBackendServerIP6() == wantHostaddr.toString()))
     {
         StorageGroup sg(groupname, host);
         LOG(VB_FILE, LOG_INFO, "HandleSGGetFileList: Getting local info");
@@ -3425,6 +3427,7 @@
     }
 
     QString wantHost = sList.at(1);
+    QHostAddress wantHostaddr(wantHost);
     QString groupname = sList.at(2);
     QString filename = sList.at(3);
 
@@ -3434,8 +3437,8 @@
             .arg(gCoreContext->GenMythURL(wantHost, 0, filename, groupname)));
 
     if ((wantHost.toLower() == gCoreContext->GetHostName().toLower()) ||
-        (wantHost == gCoreContext->GetSetting("BackendServerIP")) ||
-        (wantHost == gCoreContext->GetSetting("BackendServerIP6")))
+        (wantHostaddr.toString() == gCoreContext->GetBackendServerIP4()) ||
+        (wantHostaddr.toString() == gCoreContext->GetBackendServerIP6()))
     {
         LOG(VB_FILE, LOG_INFO, "HandleSGFileQuery: Getting local info");
         StorageGroup sg(groupname, gCoreContext->GetHostName());
@@ -3638,13 +3641,12 @@
         if (encoder->IsLocal())
         {
             strlist << gCoreContext->GetBackendServerIP();
-            strlist << gCoreContext->GetSetting("BackendServerPort");
+            strlist << QString::number(gCoreContext->GetBackendServerPort());
         }
         else
         {
             strlist << gCoreContext->GetBackendServerIP(encoder->GetHostName());
-            strlist << gCoreContext->GetSettingOnHost("BackendServerPort",
-                                                  encoder->GetHostName(), "-1");
+            strlist << QString::number(gCoreContext->GetBackendServerPort(encoder->GetHostName()));
         }
     }
     else
@@ -3789,13 +3791,12 @@
         if (encoder->IsLocal())
         {
             strlist << gCoreContext->GetBackendServerIP();
-            strlist << gCoreContext->GetSetting("BackendServerPort");
+            strlist << QString::number(gCoreContext->GetBackendServerPort());
         }
         else
         {
             strlist << gCoreContext->GetBackendServerIP(encoder->GetHostName());
-            strlist << gCoreContext->GetSettingOnHost("BackendServerPort",
-                                                  encoder->GetHostName(), "-1");
+            strlist << QString::number(gCoreContext->GetBackendServerPort(encoder->GetHostName()));
         }
     }
     else
@@ -5264,13 +5265,12 @@
         if (encoder->IsLocal())
         {
             strlist << gCoreContext->GetBackendServerIP();
-            strlist << gCoreContext->GetSetting("BackendServerPort");
+            strlist << QString::number(gCoreContext->GetBackendServerPort());
         }
         else
         {
             strlist << gCoreContext->GetBackendServerIP(encoder->GetHostName());
-            strlist << gCoreContext->GetSettingOnHost("BackendServerPort",
-                                                  encoder->GetHostName(), "-1");
+            strlist << QString::number(gCoreContext->GetBackendServerPort(encoder->GetHostName()));
         }
     }
     else
@@ -5301,13 +5301,12 @@
         if (encoder->IsLocal())
         {
             strlist << gCoreContext->GetBackendServerIP();
-            strlist << gCoreContext->GetSetting("BackendServerPort");
+            strlist << QString::number(gCoreContext->GetBackendServerPort());
         }
         else
         {
             strlist << gCoreContext->GetBackendServerIP(encoder->GetHostName());
-            strlist << gCoreContext->GetSettingOnHost("BackendServerPort",
-                                                  encoder->GetHostName(), "-1");
+            strlist << QString::number(gCoreContext->GetBackendServerPort(encoder->GetHostName()));
         }
     }
     else
@@ -6245,8 +6244,8 @@
 {
     MythSocket *masterServerSock = new MythSocket(-1, this);
 
-    QString server = gCoreContext->GetSetting("MasterServerIP", "127.0.0.1");
-    int port = gCoreContext->GetNumSetting("MasterServerPort", 6543);
+    QString server = gCoreContext->GetMasterServerIP();
+    int port = gCoreContext->GetMasterServerPort();
 
     LOG(VB_GENERAL, LOG_NOTICE, QString("Connecting to master server: %1:%2")
                            .arg(server).arg(port));
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/upnpcdsmusic.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/upnpcdsmusic.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/upnpcdsmusic.cpp	2014-07-16 11:43:04.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/upnpcdsmusic.cpp	2014-11-26 14:36:33.715799926 +0100
@@ -298,8 +298,8 @@
         m_mapBackendPort[ sHostName ] = gCoreContext->GetSettingOnHost("BackendStatusPort", sHostName);
 #endif
 
-    QString sServerIp = gCoreContext->GetSetting( "BackendServerIp"   );
-    QString sPort     = gCoreContext->GetSetting( "BackendStatusPort" );
+    QString sServerIp   = gCoreContext->GetBackendServerIP4();
+    int sPort           = gCoreContext->GetBackendStatusPort();
 
     // ----------------------------------------------------------------------
     // Build Support Strings
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/upnpcdstv.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/upnpcdstv.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/upnpcdstv.cpp	2014-07-16 11:43:04.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/upnpcdstv.cpp	2014-11-26 14:36:33.715799926 +0100
@@ -307,10 +307,10 @@
     // ----------------------------------------------------------------------
 
     if (!m_mapBackendIp.contains( sHostName ))
-        m_mapBackendIp[ sHostName ] = gCoreContext->GetSettingOnHost( "BackendServerIp", sHostName);
+        m_mapBackendIp[ sHostName ] = gCoreContext->GetBackendServerIP4(sHostName);
 
     if (!m_mapBackendPort.contains( sHostName ))
-        m_mapBackendPort[ sHostName ] = gCoreContext->GetSettingOnHost("BackendStatusPort", sHostName);
+        m_mapBackendPort[ sHostName ] = gCoreContext->GetBackendStatusPort(sHostName);
 
     // ----------------------------------------------------------------------
     // Build Support Strings
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/upnpcdsvideo.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/upnpcdsvideo.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/upnpcdsvideo.cpp	2014-07-16 11:43:04.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/upnpcdsvideo.cpp	2014-11-26 14:36:33.719134057 +0100
@@ -269,12 +269,12 @@
         if (sHostName.isEmpty())
         {
             m_mapBackendIp[sHostName] = 
-                gCoreContext->GetSetting( "BackendServerIP" );
+                gCoreContext->GetBackendServerIP4();
         }
         else
         {
             m_mapBackendIp[sHostName] = 
-                gCoreContext->GetSettingOnHost( "BackendServerIp", sHostName);
+                gCoreContext->GetBackendServerIP4(sHostName);
         }
     }
 
@@ -283,12 +283,12 @@
         if (sHostName.isEmpty())
         {
             m_mapBackendPort[sHostName] = 
-                gCoreContext->GetSetting( "BackendStatusPort" );
+                gCoreContext->GetBackendStatusPort();
         }
         else
         {
             m_mapBackendPort[sHostName] = 
-                gCoreContext->GetSettingOnHost("BackendStatusPort", sHostName);
+                gCoreContext->GetBackendStatusPort(sHostName);
         }
     }
     
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythfrontend/mediarenderer.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythfrontend/mediarenderer.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythfrontend/mediarenderer.cpp	2014-07-16 11:43:03.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythfrontend/mediarenderer.cpp	2014-11-26 14:36:33.719134057 +0100
@@ -60,9 +60,8 @@
         QString hostname   = gCoreContext->GetHostName();
         QDateTime qdtNow   = MythDate::current();
         QString masterhost = gCoreContext->GetMasterHostName();
-        QString masterip   = gCoreContext->GetSetting("MasterServerIP");
-        QString masterport = gCoreContext->GetSettingOnHost("BackendStatusPort",
-                                                            masterhost, "6544");
+        QString masterip   = gCoreContext->GetMasterServerIP();
+        int masterport = gCoreContext->GetMasterServerStatusPort();
 
         QTextStream stream ( &pRequest->m_response );
         stream.setCodec("UTF-8");
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythfrontend/themechooser.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythfrontend/themechooser.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythfrontend/themechooser.cpp	2014-07-16 11:43:03.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythfrontend/themechooser.cpp	2014-11-26 14:36:33.719134057 +0100
@@ -928,8 +928,8 @@
         m_mythVersion.replace(QRegExp("\\.[0-9]{8,}.*"), "");
     }
 
-    m_infoPackage = gCoreContext->GenMythURL(gCoreContext->GetSetting("MasterServerIP"),
-                                             0,
+    m_infoPackage = gCoreContext->GenMythURL(gCoreContext->GetMasterServerIP(),
+                                             gCoreContext->GetMasterServerPort(),
                                              "remotethemes/themes.zip",
                                              "Temp");
 
@@ -960,8 +960,8 @@
     if (RemoteFile::Exists(m_infoPackage))
     {
         QString remoteThemeDir =
-            gCoreContext->GenMythURL(gCoreContext->GetSetting("MasterServerIP"),
-                                     0,
+            gCoreContext->GenMythURL(gCoreContext->GetMasterServerIP(),
+                                     gCoreContext->GetMasterServerPort(),
                                      QString("remotethemes/%1/%2")
                                              .arg(m_mythVersion)
                                              .arg(GetMythUI()->GetThemeName()),
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythfrontend/upnpscanner.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythfrontend/upnpscanner.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythfrontend/upnpscanner.cpp	2014-07-16 11:43:03.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythfrontend/upnpscanner.cpp	2014-11-26 14:36:33.719134057 +0100
@@ -441,9 +441,8 @@
     m_watchdogTimer->start(1000 * 10); // every 10s
 
     // avoid connecting to the master backend
-    m_masterHost = gCoreContext->GetSetting("MasterServerIP");
-    m_masterPort = gCoreContext->GetSettingOnHost("BackendStatusPort",
-                                             m_masterHost, "6544").toInt();
+    m_masterHost = gCoreContext->GetMasterServerIP();
+    m_masterPort = gCoreContext->GetMasterServerStatusPort();
 
     m_lock.unlock();
     LOG(VB_GENERAL, LOG_INFO, LOC + "Started");
@@ -925,9 +924,8 @@
     // sometimes initialisation is too early and m_masterHost is empty
     if (m_masterHost.isEmpty())
     {
-        m_masterHost = gCoreContext->GetSetting("MasterServerIP");
-        m_masterPort = gCoreContext->GetSettingOnHost("BackendStatusPort",
-                                                  m_masterHost, "6544").toInt();
+        m_masterHost = gCoreContext->GetMasterServerIP();
+        m_masterPort = gCoreContext->GetMasterServerStatusPort();
     }
 
     QUrl qurl(url);
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythmediaserver/main.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythmediaserver/main.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythmediaserver/main.cpp	2014-07-16 11:43:03.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythmediaserver/main.cpp	2014-11-26 14:36:33.719134057 +0100
@@ -141,9 +141,8 @@
         return GENERIC_EXIT_CONNECT_ERROR;
     }
 
-    int     port = gCoreContext->GetNumSetting("BackendServerPort", 6543);
-    if (gCoreContext->GetSetting("BackendServerIP").isEmpty() &&
-        gCoreContext->GetSetting("BackendServerIP6").isEmpty())
+    int     port = gCoreContext->GetBackendServerPort();
+    if (gCoreContext->GetBackendServerIP().isEmpty())
     {
         cerr << "No setting found for this machine's BackendServerIP.\n"
              << "Please run setup on this machine and modify the first page\n"
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythtranscode/main.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythtranscode/main.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythtranscode/main.cpp	2014-07-16 11:43:03.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythtranscode/main.cpp	2014-11-26 14:36:33.719134057 +0100
@@ -722,12 +722,13 @@
 
 static int transUnlink(QString filename, ProgramInfo *pginfo)
 {
+    QString hostname = pginfo->GetHostname();
+
     if (pginfo != NULL && !pginfo->GetStorageGroup().isEmpty() &&
-        !pginfo->GetHostname().isEmpty())
+        !hostname.isEmpty())
     {
-        QString ip = gCoreContext->GetBackendServerIP(pginfo->GetHostname());
-        QString port = gCoreContext->GetSettingOnHost("BackendServerPort",
-                                                      pginfo->GetHostname());
+        QString ip = gCoreContext->GetBackendServerIP(hostname);
+        int port = gCoreContext->GetBackendServerPort(hostname);
         QString basename = filename.section('/', -1);
         QString uri = gCoreContext->GenMythURL(ip, port, basename,
                                                pginfo->GetStorageGroup());
