From 373abe42fbd99960c176cdb0026fa61f0a51616f Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Mon, 9 Sep 2013 19:24:10 +0100
Subject: [PATCH] Re-enable Artist and Genre groupings for upnp

---
 mythtv/programs/mythbackend/upnpcdsmusic.cpp |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/mythtv/programs/mythbackend/upnpcdsmusic.cpp b/mythtv/programs/mythbackend/upnpcdsmusic.cpp
index 0d4396a..e2106af 100644
--- a/mythtv/programs/mythbackend/upnpcdsmusic.cpp
+++ b/mythtv/programs/mythbackend/upnpcdsmusic.cpp
@@ -80,10 +80,8 @@ UPnpCDSRootInfo UPnpCDSMusic::g_RootNodes[] =
             "GROUP BY a.album_name "
             "ORDER BY a.album_name",
         "WHERE song.album_id=:KEY", "album.album_name" },
-
-#if 0
     {   "By Artist",
-        "artist_id",
+        "song.artist_id",
         "SELECT a.artist_id as id, "
           "a.artist_name as name, "
           "count( distinct song.artist_id ) as children "
@@ -94,7 +92,7 @@ UPnpCDSRootInfo UPnpCDSMusic::g_RootNodes[] =
         "WHERE song.artist_id=:KEY", "" },
 
 {   "By Genre",
-        "genre_id",
+        "song.genre_id",
         "SELECT g.genre_id as id, "
           "genre as name, "
           "count( distinct song.genre_id ) as children "
@@ -103,7 +101,6 @@ UPnpCDSRootInfo UPnpCDSMusic::g_RootNodes[] =
             "GROUP BY g.genre_id "
             "ORDER BY g.genre",
         "WHERE song.genre_id=:KEY", "" },
-#endif
 
 };
 
-- 
1.7.10.2

