From f93822a8cadc7e0687ee97a207a16525776467fd Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Sat, 7 Jun 2014 22:42:59 +1000
Subject: [PATCH 083/945] Fix GetBackendServerIP6 from returning IPv4 address

---
 mythtv/libs/libmythbase/mythcorecontext.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythbase/mythcorecontext.cpp b/mythtv/libs/libmythbase/mythcorecontext.cpp
index ebd0b54..71876e3 100644
--- a/mythtv/libs/libmythbase/mythcorecontext.cpp
+++ b/mythtv/libs/libmythbase/mythcorecontext.cpp
@@ -979,7 +979,7 @@ QString MythCoreContext::GetBackendServerIP4(const QString &host)
  */
 QString MythCoreContext::GetBackendServerIP6(void)
 {
-    return GetBackendServerIP4(d->m_localHostname);
+    return GetBackendServerIP6(d->m_localHostname);
 }
 
 /**
-- 
1.7.10.2

