diff --git a/mythtv/libs/libmythtv/videoout_nullvdpau.cpp b/mythtv/libs/libmythtv/videoout_nullvdpau.cpp
index bcee8a4..bdad0ee 100644
--- a/mythtv/libs/libmythtv/videoout_nullvdpau.cpp
+++ b/mythtv/libs/libmythtv/videoout_nullvdpau.cpp
@@ -299,11 +299,11 @@ void VideoOutputNullVDPAU::DrawSlice(VideoFrame *frame, int x, int y, int w, int
                                                FMT_VDPAU))
                         {
                             created++;
-                            int size = (m_surfaceSize.width() *
-                                        m_surfaceSize.height() * 3) / 2;
+                            int size = buffersize(FMT_YV12, width, height);
+                            uint8_t* buffer = (uint8_t*)av_malloc(size);
                             m_shadowBuffers->AddBuffer(m_surfaceSize.width(),
                                                        m_surfaceSize.height(),
-                                                       new unsigned char[size],
+                                                       buffer,
                                                        FMT_YV12);
                         }
                     }
