diff -Naur master-old/mythtv/libs/libmythtv/mythframe.h master-new/mythtv/libs/libmythtv/mythframe.h
--- master-old/mythtv/libs/libmythtv/mythframe.h	2014-07-01 07:17:04.000000000 +0200
+++ master-new/mythtv/libs/libmythtv/mythframe.h	2014-07-05 15:15:52.986810409 +0200
@@ -95,7 +95,7 @@
                         const int *p = 0,
                         const int *o = 0,
                         float _aspect = -1.0f, double _rate = -1.0f,
-                        int _aligned = 64) MUNUSED;
+                        int _aligned = 0) MUNUSED;
 static inline void clear(VideoFrame *vf);
 static inline bool compatible(const VideoFrame *a,
                               const VideoFrame *b) MUNUSED;
diff -Naur master-old/mythtv/libs/libmythtv/videobuffers.h master-new/mythtv/libs/libmythtv/videobuffers.h
--- master-old/mythtv/libs/libmythtv/videobuffers.h	2014-07-01 07:17:04.000000000 +0200
+++ master-new/mythtv/libs/libmythtv/videobuffers.h	2014-07-05 15:16:41.503304950 +0200
@@ -47,7 +47,7 @@
 {
   public:
     YUVInfo(uint w, uint h, uint size, const int *p, const int *o,
-            int aligned = 64);
+            int aligned = 0);
 
   public:
     uint width;
