From d3bb51292227cbc1165a4de10d8d376128ffa181 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Mon, 6 Aug 2012 15:43:14 +0200
Subject: [PATCH 41/56] ringbuffer: Prevent a divide by 0

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
Signed-off-by: Jean-Yves Avenard <jyavenard@mythtv.org>
---
 mythtv/libs/libmythtv/ringbuffer.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/mythtv/libs/libmythtv/ringbuffer.cpp b/mythtv/libs/libmythtv/ringbuffer.cpp
index 4601bb2..fdb786d 100644
--- a/mythtv/libs/libmythtv/ringbuffer.cpp
+++ b/mythtv/libs/libmythtv/ringbuffer.cpp
@@ -452,6 +452,8 @@ bool RingBuffer::IsNearEnd(double fps, uint vvf) const
     // telecom kilobytes (i.e. 1000 per k not 1024)
     uint   tmp = (uint) max(abs(rawbitrate * playspeed), 0.5f * rawbitrate);
     uint   kbits_per_sec = min(rawbitrate * 3, tmp);
+    if (kbits_per_sec == 0)
+        return false;
 
     double readahead_time   = sz / (kbits_per_sec * (1000.0/8.0));
 
-- 
1.7.10.2

