From 27b8c68a9bb7a37ebf7beccb0c7dc335e2358c44 Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Mon, 14 Jul 2014 20:41:26 +0100
Subject: [PATCH 4/8] UPnP: Properly sort the video genre list alphabetically

---
 mythtv/programs/mythbackend/upnpcdsvideo.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/programs/mythbackend/upnpcdsvideo.cpp b/mythtv/programs/mythbackend/upnpcdsvideo.cpp
index 1987ce9..f612af2 100644
--- a/mythtv/programs/mythbackend/upnpcdsvideo.cpp
+++ b/mythtv/programs/mythbackend/upnpcdsvideo.cpp
@@ -111,7 +111,7 @@ UPnpCDSRootInfo UPnpCDSVideo::g_RootNodes[] =
             "LEFT JOIN videocategory ON videocategory.intid = videometadata.category "
           "%1 "
           "GROUP BY videometadata.category "
-          "ORDER BY videometadata.category",
+          "ORDER BY videocategory.category",
         "WHERE videometadata.category=:KEY",
         "title",
         "object.container",
-- 
1.7.10.2

