diff -Naur mythtv-0.27-20140708-gf4825ca-old/mythtv/programs/mythbackend/upnpcdstv.cpp mythtv-0.27-20140708-gf4825ca-new/mythtv/programs/mythbackend/upnpcdstv.cpp
--- mythtv-0.27-20140708-gf4825ca-old/mythtv/programs/mythbackend/upnpcdstv.cpp	2014-07-15 10:35:06.189201119 +0200
+++ mythtv-0.27-20140708-gf4825ca-new/mythtv/programs/mythbackend/upnpcdstv.cpp	2014-07-15 14:03:45.198542321 +0200
@@ -149,7 +149,6 @@
                   "r.basename, r.progstart, r.progend, "        \
                   "r.storagegroup, r.inetref, "                 \
                   "p.category_type, c.callsign, c.channum, "    \
-                  "p.episode, p.totalepisodes, p.season, "      \
                   "r.programid, r.seriesid, r.recordid, "       \
                   "c.default_authority, c.name "                \
            "FROM recorded r "                                   \
@@ -314,16 +313,16 @@
     QString        sCallsign    = query.value(16).toString();
     QString        sChanNum     = query.value(17).toString();
 
-    int            nEpisode      = query.value(18).toInt();
-    int            nEpisodeTotal = query.value(19).toInt();
-    int            nSeason       = query.value(20).toInt();
-
-    QString        sProgramId   = query.value(21).toString();
-    QString        sSeriesId    = query.value(22).toString();
-    int            nRecordId    = query.value(23).toInt();
+    int            nEpisode      = 0;
+    int            nEpisodeTotal = 0;
+    int            nSeason       = 0;
+
+    QString        sProgramId   = query.value(18).toString();
+    QString        sSeriesId    = query.value(19).toString();
+    int            nRecordId    = query.value(20).toInt();
 
-    QString        sDefaultAuthority = query.value(24).toString();
-    QString        sChanName    = query.value(25).toString();
+    QString        sDefaultAuthority = query.value(21).toString();
+    QString        sChanName    = query.value(22).toString();
 
     // ----------------------------------------------------------------------
     // Cache Host ip Address & Port
