diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythmetadata/metadataimagedownload.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythmetadata/metadataimagedownload.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythmetadata/metadataimagedownload.cpp	2014-07-17 16:35:08.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythmetadata/metadataimagedownload.cpp	2014-07-17 16:40:17.943647152 +0200
@@ -216,12 +216,12 @@
                 bool exists = false;
                 bool onMaster = false;
                 QString resolvedFN;
-                if ((lookup->GetHost().toLower() == gCoreContext->GetHostName().toLower()) ||
-                    (gCoreContext->IsThisHost(lookup->GetHost())))
+                if (gCoreContext->IsMasterBackend() &&
+                    gCoreContext->IsThisHost(lookup->GetHost()))
                 {
-                    StorageGroup sg;
+                    StorageGroup sg(getStorageGroupName(type), lookup->GetHost());
                     resolvedFN = sg.FindFile(filename);
-                    exists = QFile::exists(resolvedFN);
+                    exists = !resolvedFN.isEmpty() && QFile::exists(resolvedFN);
                     if (!exists)
                     {
                         resolvedFN = getLocalStorageGroupPath(type,
