From 823e8511070c122657ed22238a40ab41cec0c545 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Thu, 17 Jul 2014 20:36:42 +1000
Subject: [PATCH 07/16] metadata: properly update artwork images paths

---
 .../libs/libmythmetadata/metadataimagedownload.cpp   |   18 ++++++------------
 1 file changed, 6 insertions(+), 12 deletions(-)

diff --git a/mythtv/libs/libmythmetadata/metadataimagedownload.cpp b/mythtv/libs/libmythmetadata/metadataimagedownload.cpp
index 96800ce..6f86051 100644
--- a/mythtv/libs/libmythmetadata/metadataimagedownload.cpp
+++ b/mythtv/libs/libmythmetadata/metadataimagedownload.cpp
@@ -198,14 +198,9 @@ void MetadataImageDownload::run()
                                         "to file: %1").arg(finalfile));
                             errored = true;
                         }
-                        else
-                            downloaded.insert(type, info);
                     }
-
                     delete download;
                 }
-                else
-                    downloaded.insert(type, info);
             }
             else
             {
@@ -234,7 +229,6 @@ void MetadataImageDownload::run()
 
                 if (!exists || lookup->GetAllowOverwrites())
                 {
-
                     if (exists && !onMaster)
                     {
                         QFileInfo fi(finalfile);
@@ -295,8 +289,6 @@ void MetadataImageDownload::run()
                                             "to file: %1").arg(finalfile));
                                 errored = true;
                             }
-                            else
-                                downloaded.insert(type, info);
                         }
                     }
                     else
@@ -316,15 +308,17 @@ void MetadataImageDownload::run()
                                             "to file: %1").arg(finalfile));
                                 errored = true;
                             }
-                            else
-                                downloaded.insert(type, info);
                         }
                     }
 
                     delete download;
                 }
-                else
-                    downloaded.insert(type, info);
+            }
+            if (!errored)
+            {
+                // update future Artwork Map with what we've successfully
+                // retrieved (either downloaded or already existing
+                downloaded.insert(type, info);
             }
         }
         if (errored)
-- 
1.7.10.2

