From d90fd4e75f738a8d5f202d4b7d1290994ada556d Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Sat, 26 Jul 2014 08:34:32 +0100
Subject: [PATCH 04/30] UPnP: Fix copy/paste error spotted by Coverity
 (1228783).

---
 mythtv/programs/mythbackend/upnpcdstv.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/programs/mythbackend/upnpcdstv.cpp b/mythtv/programs/mythbackend/upnpcdstv.cpp
index e9fa047..07a8a45 100644
--- a/mythtv/programs/mythbackend/upnpcdstv.cpp
+++ b/mythtv/programs/mythbackend/upnpcdstv.cpp
@@ -419,7 +419,7 @@ void UPnpCDSTv::AddItem( const UPnpCDSRequest    *pRequest,
         // FIXME: This should be set correctly for EIT data to SI_SERIESID and
         //        for known sources such as TMS to the correct identifier
         QString sIdType = "mythtv.org_XMLTV";
-        if (sProgramId.contains(sDefaultAuthority))
+        if (sSeriesId.contains(sDefaultAuthority))
             sIdType = "mythtv.org_EIT";
 
         pItem->SetPropValue( "seriesID", sSeriesId, sIdType );
-- 
1.7.10.2

