diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythprotoserver/sockethandler/filetransfer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythprotoserver/sockethandler/filetransfer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythprotoserver/sockethandler/filetransfer.cpp	2014-05-05 12:00:11.198164015 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythprotoserver/sockethandler/filetransfer.cpp	2014-05-05 12:00:26.128079012 +0200
@@ -5,6 +5,7 @@
 #include "ringbuffer.h"
 #include "programinfo.h"
 #include "mythsocket.h"
+#include "mythlogging.h"
 
 FileTransfer::FileTransfer(QString &filename, MythSocket *remote,
                            MythSocketManager *parent,
@@ -61,11 +62,23 @@
     return false;
 }
 
+bool FileTransfer::ReOpen(QString newFilename)
+{
+    if (!writemode)
+        return false;
+
+    if (rbuffer)
+        return rbuffer->ReOpen(newFilename);
+
+    return false;
+}
+
 void FileTransfer::Stop(void)
 {
     if (readthreadlive)
     {
         readthreadlive = false;
+        LOG(VB_FILE, LOG_INFO, "calling StopReads()");
         rbuffer->StopReads();
         QMutexLocker locker(&lock);
         readsLocked = true;
@@ -80,6 +93,7 @@
 
 void FileTransfer::Pause(void)
 {
+    LOG(VB_FILE, LOG_INFO, "calling StopReads()");
     rbuffer->StopReads();
     QMutexLocker locker(&lock);
     readsLocked = true;
@@ -90,6 +104,7 @@
 
 void FileTransfer::Unpause(void)
 {
+    LOG(VB_FILE, LOG_INFO, "calling StartReads()");
     rbuffer->StartReads();
     {
         QMutexLocker locker(&lock);
@@ -153,19 +168,48 @@
 
     requestBuffer.resize(max((size_t)max(size,0) + 128, requestBuffer.size()));
     char *buf = &requestBuffer[0];
+    int attempts = 0;
+
     while (tot < size)
     {
         int request = size - tot;
+        int received;
 
-        if (GetSocket()->Read(buf, (uint)request, 25 /*ms*/) != request)
-            break;
+        received = GetSocket()->Read(buf, (uint)request, 50 /*ms */);
 
-        ret = rbuffer->Write(buf, request);
-        
+        if (received != request)
+        {
+            LOG(VB_FILE, LOG_DEBUG,
+                QString("WriteBlock(): Read failed. Requested %1 got %2")
+                .arg(request).arg(received));
+            if (received < 0)
+            {
+                // An error occurred, abort immediately
+                break;
+            }
+            if (received == 0)
+            {
+                attempts++;
+                if (attempts >= 3)
+                {
+                    LOG(VB_FILE, LOG_ERR,
+                        "WriteBlock(): Read tried too many times, aborting "
+                        "(client or network too slow?)");
+                    break;
+                }
+                continue;
+            }
+        }
+        ret = rbuffer->Write(buf, received);
         if (ret <= 0)
+        {
+            LOG(VB_FILE, LOG_DEBUG,
+                QString("WriteBlock(): Write failed. Requested %1 got %2")
+                .arg(received).arg(ret));
             break;
+        }
 
-        tot += request;
+        tot += received;
     }
 
     if (pginfo)
@@ -211,7 +255,7 @@
     if (pginfo)
         pginfo->UpdateInUseMark();
 
-    return QFileInfo(rbuffer->GetFilename()).size();
+    return rbuffer->GetRealFileSize();
 }
 
 void FileTransfer::SetTimeout(bool fast)
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythprotoserver/sockethandler/filetransfer.h mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythprotoserver/sockethandler/filetransfer.h
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythprotoserver/sockethandler/filetransfer.h	2014-05-05 12:00:11.198164015 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythprotoserver/sockethandler/filetransfer.h	2014-05-05 12:00:26.348077761 +0200
@@ -28,6 +28,7 @@
                  MythSocketManager *parent, bool write);
 
     bool isOpen(void);
+    bool ReOpen(QString newFilename = "");
 
     void Stop(void);
 
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/filetransfer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/filetransfer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/filetransfer.cpp	2014-05-05 12:00:13.391484857 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/filetransfer.cpp	2014-05-05 12:00:26.654742730 +0200
@@ -137,7 +137,7 @@
         int request = size - tot;
 
         ret = rbuffer->Read(buf, request);
-        
+
         if (rbuffer->GetStopReads() || ret <= 0)
             break;
 
@@ -257,7 +257,7 @@
     if (pginfo)
         pginfo->UpdateInUseMark();
 
-    return QFileInfo(rbuffer->GetFilename()).size();
+    return rbuffer->GetRealFileSize();
 }
 
 void FileTransfer::SetTimeout(bool fast)
