diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/filters/filter-avcodec.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/filters/filter-avcodec.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/filters/filter-avcodec.pro	2014-07-16 11:42:29.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/filters/filter-avcodec.pro	2014-08-10 11:35:35.426966926 +0200
@@ -1,10 +1,12 @@
 # A few filters use routines from libavcodec. Include this in their .pro file
 
-LIBS += -L../../external/FFmpeg/libavcodec -lmythavcodec
+LIBS += -L../../external/FFmpeg/libswresample -lmythswresample
 LIBS += -L../../external/FFmpeg/libavutil  -lmythavutil
+LIBS += -L../../external/FFmpeg/libavcodec -lmythavcodec
 LIBS += -L../../external/FFmpeg/libavformat  -lmythavformat
 
 # Rebuild (link) this filter if the lib changes
+POST_TARGETDEPS += ../../external/FFmpeg/libswresample/$$avLibName(swresample)
 POST_TARGETDEPS += ../../external/FFmpeg/libavutil/$$avLibName(avutil)
 POST_TARGETDEPS += ../../external/FFmpeg/libavcodec/$$avLibName(avcodec)
 POST_TARGETDEPS += ../../external/FFmpeg/libavformat/$$avLibName(avformat)
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmyth/libmyth.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmyth/libmyth.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmyth/libmyth.pro	2014-07-16 11:42:35.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmyth/libmyth.pro	2014-08-10 11:42:04.160306402 +0200
@@ -103,17 +103,17 @@
 LIBS += -L../libmythupnp         -lmythupnp-$${LIBVERSION}
 LIBS += -L../libmythfreesurround -lmythfreesurround-$${LIBVERSION}
 LIBS += -L../libmythservicecontracts -lmythservicecontracts-$${LIBVERSION}
-LIBS += -L../../external/FFmpeg/libavcodec -lmythavcodec
+LIBS += -L../../external/FFmpeg/libswresample -lmythswresample
 LIBS += -L../../external/FFmpeg/libavutil  -lmythavutil
+LIBS += -L../../external/FFmpeg/libavcodec -lmythavcodec
 LIBS += -L../../external/FFmpeg/libavformat  -lmythavformat
-LIBS += -L../../external/FFmpeg/libswresample -lmythswresample
 
 POST_TARGETDEPS += ../../external/libsamplerate/libmythsamplerate-$${MYTH_LIB_EXT}
 POST_TARGETDEPS += ../libmythsoundtouch/libmythsoundtouch-$${MYTH_LIB_EXT}
-POST_TARGETDEPS += ../libmythfreesurround/libmythfreesurround-$${MYTH_LIB_EXT}
-POST_TARGETDEPS += ../../external/FFmpeg/libavcodec/$$avLibName(avcodec)
-POST_TARGETDEPS += ../../external/FFmpeg/libavutil/$$avLibName(avutil)
 POST_TARGETDEPS += ../../external/FFmpeg/libswresample/$$avLibName(swresample)
+POST_TARGETDEPS += ../../external/FFmpeg/libavutil/$$avLibName(avutil)
+POST_TARGETDEPS += ../../external/FFmpeg/libavcodec/$$avLibName(avcodec)
+POST_TARGETDEPS += ../libmythfreesurround/libmythfreesurround-$${MYTH_LIB_EXT}
 
 # Install headers so that plugins can compile independently
 inc.path = $${PREFIX}/include/mythtv/
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmyth/test/test_audioconvert/test_audioconvert.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmyth/test/test_audioconvert/test_audioconvert.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmyth/test/test_audioconvert/test_audioconvert.pro	2014-07-16 11:42:35.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmyth/test/test_audioconvert/test_audioconvert.pro	2014-08-10 11:35:35.430300023 +0200
@@ -17,10 +17,10 @@
 LIBS += -L../../../libmythui -lmythui-$$LIBVERSION
 LIBS += -L../../../libmythupnp -lmythupnp-$$LIBVERSION
 LIBS += -L../../../libmythservicecontracts -lmythservicecontracts-$$LIBVERSION
+LIBS += -L../../../../external/FFmpeg/libswresample -lmythswresample
+LIBS += -L../../../../external/FFmpeg/libavutil -lmythavutil
 LIBS += -L../../../../external/FFmpeg/libavcodec -lmythavcodec
 LIBS += -L../../../../external/FFmpeg/libavformat -lmythavformat
-LIBS += -L../../../../external/FFmpeg/libavutil -lmythavutil
-LIBS += -L../../../../external/FFmpeg/libswresample -lmythswresample
 LIBS += -L../.. -lmyth-$$LIBVERSION
 
 contains(QMAKE_CXX, "g++") {
@@ -31,10 +31,10 @@
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/zeromq/src/.libs/
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/nzmqt/src/
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/qjson/lib/
+QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libswresample
+QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavutil
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavcodec
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavformat
-QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavutil
-QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libswresample
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../libmythbase
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../libmythui
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../libmythupnp
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmyth/test/test_audioutils/test_audioutils.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmyth/test/test_audioutils/test_audioutils.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmyth/test/test_audioutils/test_audioutils.pro	2014-07-16 11:42:35.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmyth/test/test_audioutils/test_audioutils.pro	2014-08-10 11:35:35.430300023 +0200
@@ -17,10 +17,10 @@
 LIBS += -L../../../libmythui -lmythui-$$LIBVERSION
 LIBS += -L../../../libmythupnp -lmythupnp-$$LIBVERSION
 LIBS += -L../../../libmythservicecontracts -lmythservicecontracts-$$LIBVERSION
+LIBS += -L../../../../external/FFmpeg/libswresample -lmythswresample
+LIBS += -L../../../../external/FFmpeg/libavutil -lmythavutil
 LIBS += -L../../../../external/FFmpeg/libavcodec -lmythavcodec
 LIBS += -L../../../../external/FFmpeg/libavformat -lmythavformat
-LIBS += -L../../../../external/FFmpeg/libavutil -lmythavutil
-LIBS += -L../../../../external/FFmpeg/libswresample -lmythswresample
 LIBS += -L../.. -lmyth-$$LIBVERSION
 
 contains(QMAKE_CXX, "g++") {
@@ -31,10 +31,10 @@
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/zeromq/src/.libs/
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/nzmqt/src/
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/qjson/lib/
+QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libswresample
+QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavutil
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavcodec
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavformat
-QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavutil
-QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libswresample
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../libmythbase
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../libmythui
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../libmythupnp
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythmetadata/libmythmetadata.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythmetadata/libmythmetadata.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythmetadata/libmythmetadata.pro	2014-08-10 10:41:03.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythmetadata/libmythmetadata.pro	2014-08-10 11:35:35.430300023 +0200
@@ -62,8 +62,9 @@
 LIBS += -L../libmythupnp         -lmythupnp-$${LIBVERSION}
 LIBS += -L../libmythservicecontracts -lmythservicecontracts-$${LIBVERSION}
 LIBS += -L../libmythfreesurround -lmythfreesurround-$${LIBVERSION}
+LIBS += -L../../external/FFmpeg/libswresample -lmythswresample
+LIBS += -L../../external/FFmpeg/libavutil -lmythavutil
 LIBS += -L../../external/FFmpeg/libavcodec -lmythavcodec
-LIBS += -L../../external/FFmpeg/libavutil  -lmythavutil
 LIBS += -L../../external/FFmpeg/libavformat -lmythavformat
 LIBS += -L../libmyth              -lmyth-$${LIBVERSION}
 LIBS += -L../libmythtv              -lmythtv-$${LIBVERSION}
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythprotoserver/libmythprotoserver.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythprotoserver/libmythprotoserver.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythprotoserver/libmythprotoserver.pro	2014-07-16 11:42:34.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythprotoserver/libmythprotoserver.pro	2014-08-10 11:35:35.430300023 +0200
@@ -38,6 +38,7 @@
 
 LIBS += -L../libmythbase -L../libmyth -L../libmythtv -L../libmythui
 LIBS += -L../libmythupnp -L../libmythservicecontracts
+LIBS += -L../../external/FFmpeg/libswresample -lmythswresample
 LIBS += -L../../external/FFmpeg/libavutil -L../../external/FFmpeg/libavcodec
 LIBS += -L../../external/FFmpeg/libavformat -L../../external/FFmpeg/libswscale
 LIBS += -lmythbase-$$LIBVERSION -lmyth-$$LIBVERSION -lmythui-$$LIBVERSION
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/libmythtv.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/libmythtv.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/libmythtv.pro	2014-08-10 10:40:54.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/libmythtv.pro	2014-08-10 11:44:57.383655151 +0200
@@ -799,6 +799,7 @@
 # Have them at the end in order to properly resolve on mingw platform
 # where the order is of significance
 LIBS += -L../libmyth
+LIBS += -L../../external/FFmpeg/libswresample -lmythswresample
 LIBS += -L../../external/FFmpeg/libavutil
 LIBS += -L../../external/FFmpeg/libavcodec
 LIBS += -L../../external/FFmpeg/libavformat
@@ -821,6 +822,7 @@
 LIBS += $$EXTRA_LIBS $$QMAKE_LIBS_DYNLOAD
 
 POST_TARGETDEPS += ../libmyth/libmyth-$${MYTH_SHLIB_EXT}
+POST_TARGETDEPS += ../../external/FFmpeg/libswresample/$$avLibName(swresample)
 POST_TARGETDEPS += ../../external/FFmpeg/libavutil/$$avLibName(avutil)
 POST_TARGETDEPS += ../../external/FFmpeg/libavcodec/$$avLibName(avcodec)
 POST_TARGETDEPS += ../../external/FFmpeg/libavformat/$$avLibName(avformat)
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/test/test_copyframes/test_copyframes.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/test/test_copyframes/test_copyframes.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/test/test_copyframes/test_copyframes.pro	2014-08-10 10:40:51.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/test/test_copyframes/test_copyframes.pro	2014-08-10 11:35:35.430300023 +0200
@@ -20,11 +20,11 @@
 LIBS += -L../../../libmythupnp -lmythupnp-$$LIBVERSION
 LIBS += -L../../../libmythservicecontracts -lmythservicecontracts-$$LIBVERSION
 LIBS += -L../../../libmyth -lmyth-$$LIBVERSION
+LIBS += -L../../../../external/FFmpeg/libswresample -lmythswresample
+LIBS += -L../../../../external/FFmpeg/libavutil -lmythavutil
 LIBS += -L../../../../external/FFmpeg/libavcodec -lmythavcodec
 LIBS += -L../../../../external/FFmpeg/libswscale -lmythswscale
 LIBS += -L../../../../external/FFmpeg/libavformat -lmythavformat
-LIBS += -L../../../../external/FFmpeg/libavutil -lmythavutil
-LIBS += -L../../../../external/FFmpeg/libswresample -lmythswresample
 LIBS += -L../../../../external/qjson/lib -lmythqjson
 using_mheg:LIBS += -L../../../libmythfreemheg -lmythfreemheg-$$LIBVERSION
 using_hdhomerun:LIBS += -L../../../../external/libhdhomerun -lmythhdhomerun-$$LIBVERSION
@@ -43,11 +43,11 @@
 }
 
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/qjson/lib/
+QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libswresample
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavutil
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libswscale
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavformat
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavcodec
-QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libswresample
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/libhdhomerun
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../libmythbase
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../libmyth
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/test/test_mpegtables/test_mpegtables.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/test/test_mpegtables/test_mpegtables.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/test/test_mpegtables/test_mpegtables.pro	2014-07-16 11:42:34.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/test/test_mpegtables/test_mpegtables.pro	2014-08-10 11:35:35.430300023 +0200
@@ -19,11 +19,11 @@
 LIBS += -L../../../libmythupnp -lmythupnp-$$LIBVERSION
 LIBS += -L../../../libmythservicecontracts -lmythservicecontracts-$$LIBVERSION
 LIBS += -L../../../libmyth -lmyth-$$LIBVERSION
+LIBS += -L../../../../external/FFmpeg/libswresample -lmythswresample
+LIBS += -L../../../../external/FFmpeg/libavutil -lmythavutil
 LIBS += -L../../../../external/FFmpeg/libavcodec -lmythavcodec
 LIBS += -L../../../../external/FFmpeg/libswscale -lmythswscale
 LIBS += -L../../../../external/FFmpeg/libavformat -lmythavformat
-LIBS += -L../../../../external/FFmpeg/libavutil -lmythavutil
-LIBS += -L../../../../external/FFmpeg/libswresample -lmythswresample
 LIBS += -L../../../../external/qjson/lib -lmythqjson
 using_mheg:LIBS += -L../../../libmythfreemheg -lmythfreemheg-$$LIBVERSION
 using_hdhomerun:LIBS += -L../../../../external/libhdhomerun -lmythhdhomerun-$$LIBVERSION
@@ -42,11 +42,11 @@
 }
 
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/qjson/lib/
+QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libswresample
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavutil
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libswscale
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavformat
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libavcodec
-QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/FFmpeg/libswresample
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../../external/libhdhomerun
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../libmythbase
 QMAKE_LFLAGS += -Wl,$$_RPATH_$(PWD)/../../../libmyth
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythtranscode/mythtranscode.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythtranscode/mythtranscode.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythtranscode/mythtranscode.pro	2014-08-10 10:40:49.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythtranscode/mythtranscode.pro	2014-08-10 11:50:48.466850119 +0200
@@ -30,9 +30,10 @@
            replex/pes.h     replex/ringbuffer.h replex/ts.h
 
 DEPENDPATH += replex
+DEPENDPATH += ../../libs/libswresample
+DEPENDPATH += ../../libs/libavutil
 DEPENDPATH += ../../libs/libavcodec
 DEPENDPATH += ../../libs/libavformat
-DEPENDPATH += ../../libs/libavutil
 DEPENDPATH += ../../libs/libmythtv/recorders
 
 !contains( CONFIG_LIBMPEG2EXTERNAL, yes) {
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythtranscode/replex/replex.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythtranscode/replex/replex.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythtranscode/replex/replex.pro	2014-07-16 11:43:03.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythtranscode/replex/replex.pro	2014-08-10 11:49:52.956883152 +0200
@@ -20,17 +20,14 @@
 
 QMAKE_CFLAGS += -w
 
-LIBS += -L../../../external/FFmpeg/libavutil
-LIBS += -L../../../external/FFmpeg/libavcodec
-LIBS += -L../../../external/FFmpeg/libavformat
+LIBS += -L../../../external/FFmpeg/libswresample -lmythswresample
+LIBS += -L../../../external/FFmpeg/libavutil -lmythavutil
+LIBS += -L../../../external/FFmpeg/libavcodec -lmythavcodec
+LIBS += -L../../../external/FFmpeg/libavformat -lmythavformat
 LIBS += -L../../../external/zeromq/src/.libs
 LIBS += -L../../../external/nzmqt/src
 LIBS += -L../../../external/qjson/lib
-LIBS += -L../../../libs/libmythbase
-LIBS += -lmythavformat
-LIBS += -lmythavcodec
-LIBS += -lmythavutil
-LIBS += -lmythbase-$$LIBVERSION
+LIBS += -L../../../libs/libmythbase -lmythbase-$$LIBVERSION
 LIBS += $$EXTRA_LIBS
 
 isEmpty(QMAKE_EXTENSION_SHLIB) {
@@ -40,6 +37,7 @@
 # llvm/xcode doesn't compile if -O3 is enabled (or any other -O for that matter)
 macx: QMAKE_CFLAGS -= -O3 -O2 -O1 -Os
 
+POST_TARGETDEPS += ../../../external/FFmpeg/libswresample/$$avLibName(swresample)
 POST_TARGETDEPS += ../../../external/FFmpeg/libavutil/$$avLibName(avutil)
 POST_TARGETDEPS += ../../../external/FFmpeg/libavcodec/$$avLibName(avcodec)
 POST_TARGETDEPS += ../../../external/FFmpeg/libavformat/$$avLibName(avformat)
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/programs-libs.pro mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/programs-libs.pro
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/programs-libs.pro	2014-07-16 11:43:03.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/programs-libs.pro	2014-08-10 11:52:28.226813207 +0200
@@ -10,11 +10,11 @@
 INCLUDEPATH += ../../libs/libmythprotoserver
 
 LIBS += -L../../libs/libmyth -L../../libs/libmythtv
+LIBS += -L../../external/FFmpeg/libswresample
 LIBS += -L../../external/FFmpeg/libavutil
 LIBS += -L../../external/FFmpeg/libavcodec
 LIBS += -L../../external/FFmpeg/libavformat
 LIBS += -L../../external/FFmpeg/libswscale
-LIBS += -L../../external/FFmpeg/libswresample
 LIBS += -L../../libs/libmythbase
 LIBS += -L../../libs/libmythui
 LIBS += -L../../libs/libmythupnp
@@ -24,9 +24,9 @@
 
 LIBS += -lmythswscale
 LIBS += -lmythavformat
-LIBS += -lmythavcodec
-LIBS += -lmythavutil
 LIBS += -lmythswresample
+LIBS += -lmythavutil
+LIBS += -lmythavcodec
 LIBS += -lmythtv-$$LIBVERSION
 LIBS += -lmythupnp-$$LIBVERSION
 LIBS += -lmythbase-$$LIBVERSION
@@ -47,6 +47,7 @@
 POST_TARGETDEPS += ../../libs/libmythui/libmythui-$${MYTH_SHLIB_EXT}
 POST_TARGETDEPS += ../../libs/libmyth/libmyth-$${MYTH_SHLIB_EXT}
 POST_TARGETDEPS += ../../libs/libmythtv/libmythtv-$${MYTH_SHLIB_EXT}
+POST_TARGETDEPS += ../../external/FFmpeg/libswresample/$$avLibName(swresample)
 POST_TARGETDEPS += ../../external/FFmpeg/libavutil/$$avLibName(avutil)
 POST_TARGETDEPS += ../../external/FFmpeg/libavcodec/$$avLibName(avcodec)
 POST_TARGETDEPS += ../../external/FFmpeg/libavformat/$$avLibName(avformat)
