diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/ringbuffer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/ringbuffer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/ringbuffer.cpp	2014-05-05 12:01:28.797722701 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/ringbuffer.cpp	2014-05-05 12:01:46.484288953 +0200
@@ -194,7 +194,7 @@
     request_pause(false),     paused(false),
     ateof(false),             readsallowed(false),
     setswitchtonext(false),
-    rawbitrate(8000),         playspeed(1.0f),
+    rawbitrate(800),          playspeed(1.0f),
     fill_threshold(65536),    fill_min(-1),
     readblocksize(CHUNK),     wanttoread(0),
     numfailures(0),           commserror(false),
@@ -373,7 +373,14 @@
                      (estbitrate >=  500) ? KB16  :
                      (estbitrate >   250) ? KB8   :
                      (estbitrate >   125) ? KB4   : KB2;
-    readblocksize = rbs;
+    if (rbs < CHUNK)
+    {
+        readblocksize = rbs;
+    }
+    else
+    {
+        readblocksize = max(rbs,readblocksize);
+    }
 
     // minumum seconds of buffering before allowing read
     float secs_min = 0.15;
