diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythbase/remotefile.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythbase/remotefile.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythbase/remotefile.cpp	2014-05-05 12:01:43.150974540 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythbase/remotefile.cpp	2014-05-05 12:01:57.057562287 +0200
@@ -33,6 +33,8 @@
 #include "mythmiscutil.h"
 #include "threadedfilewriter.h"
 
+#define MAX_FILE_CHECK 500  // in ms
+
 RemoteFile::RemoteFile(const QString &_path, bool write, bool useRA,
                        int _timeout_ms,
                        const QStringList *possibleAuxiliaryFiles) :
@@ -855,7 +857,7 @@
 
     QMutexLocker locker(&lock);
 
-    if (completed)
+    if (completed || lastSizeCheck.elapsed() < MAX_FILE_CHECK)
     {
         return filesize;
     }
@@ -886,15 +888,20 @@
             if (strlist.count() >= 2)
             {
                 completed = strlist[1].toInt();
-                filesize = size;
             }
-            return size;
+            filesize = size;
         }
-        struct stat fileinfo;
-        if (Exists(path, &fileinfo))
+        else
         {
-            return fileinfo.st_size;
+            struct stat fileinfo;
+
+            if (Exists(path, &fileinfo))
+            {
+                filesize = fileinfo.st_size;
+            }
         }
+        lastSizeCheck.restart();
+        return filesize;
     }
 
     return -1;
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythbase/remotefile.h mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythbase/remotefile.h
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythbase/remotefile.h	2014-05-05 12:01:43.147641225 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythbase/remotefile.h	2014-05-05 12:01:57.057562287 +0200
@@ -8,6 +8,7 @@
 #include <QMutex>
 
 #include "mythbaseexp.h"
+#include "mythtimer.h"
 
 class MythSocket;
 class QFile;
@@ -73,6 +74,7 @@
 
     bool            writemode;
     bool            completed;
+    MythTimer       lastSizeCheck;
 
     QStringList     possibleauxfiles;
     QStringList     auxfiles;
