From 410aebfdf26fb37a3bc0c0331d262197ef948966 Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Wed, 3 Sep 2014 19:42:14 +0100
Subject: [PATCH 04/11] Avoid calling QFile::exists() on an empty string.

---
 mythtv/libs/libmythupnp/httprequest.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mythtv/libs/libmythupnp/httprequest.cpp b/mythtv/libs/libmythupnp/httprequest.cpp
index 81437bd..dfa9892 100644
--- a/mythtv/libs/libmythupnp/httprequest.cpp
+++ b/mythtv/libs/libmythupnp/httprequest.cpp
@@ -863,7 +863,7 @@ void HTTPRequest::FormatFileResponse( const QString &sFileName )
 {
     m_sFileName = sFileName;
 
-    if (QFile::exists( m_sFileName ))
+    if (!m_sFileName.isEmpty() && QFile::exists( m_sFileName ))
     {
 
         if (m_eResponseType == ResponseTypeUnknown)
@@ -876,7 +876,7 @@ void HTTPRequest::FormatFileResponse( const QString &sFileName )
         m_eResponseType   = ResponseTypeHTML;
         m_nResponseStatus = 404;
         LOG(VB_UPNP, LOG_INFO,
-            QString("HTTPRequest::FormatFileResponse(%1) - cannot find file")
+            QString("HTTPRequest::FormatFileResponse('%1') - cannot find file")
                 .arg(sFileName));
     }
 }
-- 
1.7.10.2

