From ce8b5f8be1db62ab3cc8064a89a4893cdd0fce41 Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Sun, 19 Oct 2014 12:22:44 +0100
Subject: [PATCH] Coverity: Fix tiny leak if gif is not animated. ID 1248082

---
 mythtv/libs/libmythui/mythuiimage.cpp |    1 +
 1 file changed, 1 insertion(+)

diff --git a/mythtv/libs/libmythui/mythuiimage.cpp b/mythtv/libs/libmythui/mythuiimage.cpp
index 0ec2bec..a930a6e 100644
--- a/mythtv/libs/libmythui/mythuiimage.cpp
+++ b/mythtv/libs/libmythui/mythuiimage.cpp
@@ -516,6 +516,7 @@ class ImageLoadThread : public QRunnable
 
                 return;
              }
+             delete frames;
         }
 
         MythImage *image = ImageLoader::LoadImage(m_painter,
-- 
1.7.10.2

