diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/ringbuffer.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/ringbuffer.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/ringbuffer.cpp	2014-10-16 19:24:07.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/ringbuffer.cpp	2014-10-22 10:16:06.640307609 +0200
@@ -408,7 +408,7 @@
     }
 
     // minumum seconds of buffering before allowing read
-    float secs_min = 0.3;
+    float secs_min = 0.5;
     // set the minimum buffering before allowing ffmpeg read
     fill_min  = (uint) ((estbitrate * 1000 * secs_min) * 0.125f);
     // make this a multiple of ffmpeg block size..
