From 2f2261d6e3bb78f029dbb38e4746314d6ca0a2b6 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Mon, 10 Nov 2014 12:41:52 +0000
Subject: [PATCH] mythfilldatabase: remove the redundant assign to userid

Refs #12300.
---
 mythtv/libs/libmythtv/datadirect.cpp |    1 -
 1 file changed, 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/datadirect.cpp b/mythtv/libs/libmythtv/datadirect.cpp
index 4a11c0c8..7c62597 100644
--- a/mythtv/libs/libmythtv/datadirect.cpp
+++ b/mythtv/libs/libmythtv/datadirect.cpp
@@ -1178,7 +1178,6 @@ bool DataDirectProcessor::GrabData(const QDateTime &pstartDate,
 
     if (m_cacheData)
     {
-        QByteArray userid = GetUserID().toLatin1();
         cache_dd_data = m_tmpDir +
             QString("/mythtv_dd_cache_%1_UTC_%2_to_%3")
             .arg(GetListingsProvider())
-- 
1.7.10.2

