diff --git a/mythtv/libs/libmythtv/avformatdecoder.cpp b/mythtv/libs/libmythtv/avformatdecoder.cpp
index b6dc9f6..4385f4d 100644
--- a/mythtv/libs/libmythtv/avformatdecoder.cpp
+++ b/mythtv/libs/libmythtv/avformatdecoder.cpp
@@ -1271,7 +1271,7 @@ int AvFormatDecoder::OpenFile(RingBuffer *rbuffer, bool novideo,
         if ((m_playbackinfo) || livetv || watchingrecording)
         {
             recordingHasPositionMap |= SyncPositionMap();
-            if (recordingHasPositionMap && !livetv && !watchingrecording)
+            if (recordingHasPositionMap && !watchingrecording) // && !livetv)
             {
                 hasFullPositionMap = true;
                 gopset = true;
@@ -3118,12 +3118,12 @@ void AvFormatDecoder::HandleGopStart(
         int tempKeyFrameDist = framesRead - 1 - prevgoppos;
         bool reset_kfd = false;
 
-        if (!gopset || livetv) // gopset: we've seen 2 keyframes
+        if (!gopset) // || livetv) // gopset: we've seen 2 keyframes
         {
             LOG(VB_PLAYBACK, LOG_INFO, LOC +
                 "gopset not set, syncing positionMap");
             SyncPositionMap();
-            if (tempKeyFrameDist > 0 && !livetv)
+            if (tempKeyFrameDist > 0) // && !livetv)
             {
                 LOG(VB_PLAYBACK, LOG_INFO, LOC +
                     QString("Initial key frame distance: %1.")
@@ -3164,7 +3164,7 @@ void AvFormatDecoder::HandleGopStart(
     lastKey = prevgoppos = framesRead - 1;
 
     if (can_reliably_parse_keyframes &&
-        !hasFullPositionMap && !livetv && !watchingrecording)
+        !hasFullPositionMap && !watchingrecording) // && !livetv)
     {
         long long last_frame = 0;
         {

