diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythtv/tv_play.cpp	2014-11-29 19:37:44.176994910 +0100
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythtv/tv_play.cpp	2014-11-29 19:28:51.000000000 +0100
@@ -1260,6 +1260,16 @@
                 // big in fullscreen mode
                 if (fullscreen)
                 {
+                    // Hack for ticket12311
+                    if ((maxHeight >= 1072) && (maxHeight <= 1200))
+                    { 
+                        LOG(VB_GENERAL, LOG_DEBUG, LOC + "Hack for ticket12311: Re-Adjusted Screen size. Original: "
+                        + QString("%1x").arg(maxWidth)
+                        + QString("%1. New: ").arg(maxHeight)
+                        + QString("%1x1080").arg(maxWidth));
+                        maxHeight = 1080;
+                    }
+
                     player_bounds.setSize(QSize(maxWidth, maxHeight));
 
                     // resize possibly avoids a bug on some systems
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythui/mythuihelper.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythui/mythuihelper.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/libs/libmythui/mythuihelper.cpp	2014-07-16 11:42:35.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/libs/libmythui/mythuihelper.cpp	2014-11-28 15:35:33.000000000 +0100
@@ -860,8 +860,8 @@
 void MythUIHelper::GetScreenSettings(int &xbase, int &width, float &wmult,
                                      int &ybase, int &height, float &hmult)
 {
-    xbase  = d->m_screenxbase;
-    ybase  = d->m_screenybase;
+    xbase  = d->m_screenxbase - 1;
+    ybase  = d->m_screenybase - 1;
 
     height = d->m_screenheight;
     width = d->m_screenwidth;
