From 0f7d7ff0660a12022b76cdb8d6d6bceda032f540 Mon Sep 17 00:00:00 2001
From: Karl Dietz <dekarl@mythtv.org>
Date: Sun, 30 Nov 2014 23:41:58 +0100
Subject: [PATCH] fix braino in c12af87

---
 mythtv/programs/mythfilldatabase/xmltvparser.cpp |   11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/mythtv/programs/mythfilldatabase/xmltvparser.cpp b/mythtv/programs/mythfilldatabase/xmltvparser.cpp
index 618331c..02d5ae1 100644
--- a/mythtv/programs/mythfilldatabase/xmltvparser.cpp
+++ b/mythtv/programs/mythfilldatabase/xmltvparser.cpp
@@ -362,7 +362,8 @@ ProgInfo *XMLTVParser::parseProgram(QDomElement &element)
             {
                 QDomNodeList values = info.elementsByTagName("value");
                 QDomElement item;
-                QString stars, num, den;
+                QString stars;
+                float num, den;
                 float rating = 0.0;
 
                 // Use the first rating to appear in the xml, this should be
@@ -386,10 +387,10 @@ ProgInfo *XMLTVParser::parseProgram(QDomElement &element)
                 if (!item.isNull())
                 {
                     stars = getFirstText(item);
-                    num = stars.section('/', 0, 0) + 1;
-                    den = stars.section('/', 1, 1) + 1;
-                    if (0.0 < den.toFloat())
-                        rating = num.toFloat()/den.toFloat();
+                    num = stars.section('/', 0, 0).toFloat() + 1;
+                    den = stars.section('/', 1, 1).toFloat() + 1;
+                    if (0.0 < den)
+                        rating = num/den;
                 }
 
                 pginfo->stars = rating;
-- 
1.7.10.2

