From 2ab60e37c0c84243ca73fea6f0dd03c114633883 Mon Sep 17 00:00:00 2001
From: Karl Dietz <dekarl@mythtv.org>
Date: Mon, 1 Dec 2014 00:36:04 +0100
Subject: [PATCH] Add HEVC to the ComponentDescriptor

---
 mythtv/libs/libmythtv/mpeg/dvbdescriptors.h |   21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/mythtv/libs/libmythtv/mpeg/dvbdescriptors.h b/mythtv/libs/libmythtv/mpeg/dvbdescriptors.h
index 7fe40a0..278adfb 100644
--- a/mythtv/libs/libmythtv/mpeg/dvbdescriptors.h
+++ b/mythtv/libs/libmythtv/mpeg/dvbdescriptors.h
@@ -383,8 +383,9 @@ class ComponentDescriptor : public MPEGDescriptor
 
     bool IsVideo(void) const
     {
-        return 0x1 == StreamContent() ||
-               0x5 == StreamContent();
+        return 0x1 == StreamContent() || // MPEG-2
+               0x5 == StreamContent() || // H.264
+               0x9 == StreamContent();   // HEVC
     }
     bool IsAudio(void) const
     {
@@ -407,6 +408,8 @@ class ComponentDescriptor : public MPEGDescriptor
             return MPEG2Properties();
         if (0x5 == StreamContent())
             return VID_AVC | AVCProperties();
+        if (0x9 == StreamContent())
+            return /* VID_HEVC | */ HEVCProperties();
 
         return VID_UNKNOWN;
     }
@@ -447,6 +450,20 @@ class ComponentDescriptor : public MPEGDescriptor
         }
     }
 
+    unsigned char HEVCProperties(void) const
+    {
+        switch(ComponentType())
+        {
+            case 0x0: case 0x1:
+            case 0x2: case 0x3:
+                return VID_HDTV;
+            case 0x5:
+                return VID_HDTV; // | VID_UHDTV;
+            default:
+                return VID_UNKNOWN;
+        }
+    }
+
     unsigned char AudioProperties(void) const
     {
         switch (StreamContent())
-- 
1.7.10.2

