diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/services/guide.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/services/guide.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythbackend/services/guide.cpp	2014-07-16 11:43:04.000000000 +0200
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythbackend/services/guide.cpp	2014-12-12 16:12:31.440194144 +0100
@@ -95,16 +95,18 @@
                          "AND program.chanid <= :EndChanId "
                          "AND program.endtime >= :StartDate "
                          "AND program.starttime <= :EndDate "
+                         "AND program.starttime >= :StartDateLimit "
                          "GROUP BY program.starttime, channel.chanid "
                          "ORDER BY lpad(channel.channum, 10, 0), "
                          "         callsign,                     "
                          "         lpad(program.chanid, 10, 0),  "
                          "         program.starttime ";
 
-    bindings[":StartChanId"] = nStartChanId;
-    bindings[":EndChanId"  ] = nEndChanId;
-    bindings[":StartDate"  ] = dtStartTime;
-    bindings[":EndDate"    ] = dtEndTime;
+    bindings[":StartChanId"   ] = nStartChanId;
+    bindings[":EndChanId"     ] = nEndChanId;
+    bindings[":StartDate"     ] = dtStartTime;
+    bindings[":StartDateLimit"] = dtStartTime.addDays(-1);
+    bindings[":EndDate"       ] = dtEndTime;
 
     // ----------------------------------------------------------------------
     // Get all Pending Scheduled Programs
diff -Naur mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythfrontend/guidegrid.cpp mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythfrontend/guidegrid.cpp
--- mythtv-0.27-20140716-gf4825ca-old/mythtv/programs/mythfrontend/guidegrid.cpp	2014-12-12 15:32:48.000000000 +0100
+++ mythtv-0.27-20140716-gf4825ca-new/mythtv/programs/mythfrontend/guidegrid.cpp	2014-12-12 16:06:39.625294438 +0100
@@ -982,14 +982,15 @@
     ProgramList proglist;
     MSqlBindings bindings;
     QString querystr =
-        "WHERE program.chanid     = :CHANID  AND "
-        "      program.endtime   >= :STARTTS AND "
-        "      program.starttime <= :ENDTS   AND "
+        "WHERE program.chanid     = :CHANID       AND "
+        "      program.endtime   >= :STARTTS      AND "
+        "      program.starttime <= :ENDTS        AND "
+        "      program.starttime >= :STARTLIMITTS AND "
         "      program.manualid   = 0 ";
-    bindings[":STARTTS"] =
-        m_currentStartTime.addSecs(0 - m_currentStartTime.time().second());
-    bindings[":ENDTS"] =
-        m_currentEndTime.addSecs(0 - m_currentEndTime.time().second());
+    QDateTime starttime = m_currentStartTime.addSecs(0 - m_currentStartTime.time().second());
+    bindings[":STARTTS"] = starttime;
+    bindings[":STARTLIMITTS"] = starttime.addDays(-1);
+    bindings[":ENDTS"] = m_currentEndTime.addSecs(0 - m_currentEndTime.time().second());
     bindings[":CHANID"]  = chanid;
 
     ProgramList dummy;
@@ -1337,12 +1338,13 @@
         QString querystr = "WHERE program.chanid = :CHANID "
                            "  AND program.endtime >= :STARTTS "
                            "  AND program.starttime <= :ENDTS "
+                           "  AND program.starttime >= :STARTLIMITTS "
                            "  AND program.manualid = 0 ";
+        QDateTime starttime = m_currentStartTime.addSecs(0 - m_currentStartTime.time().second());
         bindings[":CHANID"]  = GetChannelInfo(chanNum)->chanid;
-        bindings[":STARTTS"] =
-            m_currentStartTime.addSecs(0 - m_currentStartTime.time().second());
-        bindings[":ENDTS"] =
-            m_currentEndTime.addSecs(0 - m_currentEndTime.time().second());
+        bindings[":STARTTS"] = starttime;
+        bindings[":STARTLIMITTS"] = starttime.addDays(-1);
+        bindings[":ENDTS"] = m_currentEndTime.addSecs(0 - m_currentEndTime.time().second());
 
         LoadFromProgram(*proglist, querystr, bindings, m_recList);
     }
