diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/fileringbuffer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/fileringbuffer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/fileringbuffer.cpp	2014-05-05 11:59:49.868285535 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/fileringbuffer.cpp	2014-05-05 12:00:05.191531560 +0200
@@ -590,9 +590,24 @@
     rwlock.lockForRead();
     long long ret = -1;
     if (remotefile)
+    {
         ret = remotefile->GetFileSize();
+    }
     else
+    {
+        if (fd2 >= 0)
+        {
+            struct stat sb;
+
+            ret = fstat(fd2, &sb);
+            if (ret == 0 && S_ISREG(sb.st_mode))
+            {
+                rwlock.unlock();
+                return sb.st_size;
+            }
+        }
         ret = QFileInfo(filename).size();
+    }
     rwlock.unlock();
     return ret;
 }
