diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-05 11:59:58.011572464 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/mythplayer.cpp	2014-05-05 12:00:15.814804393 +0200
@@ -2774,10 +2774,10 @@
     ChangeSpeed();
 
     // check that we aren't too close to the end of program.
-    // and if so set it to 15s from the end if completed recordings
+    // and if so set it to 10s from the end if completed recordings
     // or 3s if live
     long long duration = player_ctx->tvchain->GetLengthAtCurPos();
-    int maxpos = player_ctx->tvchain->HasNext() ? 15 : 3;
+    int maxpos = player_ctx->tvchain->HasNext() ? 10 : 3;
 
     if (nextpos > (duration - maxpos))
     {
diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/ringbuffer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/ringbuffer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/ringbuffer.cpp	2014-05-05 11:59:58.008239150 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/ringbuffer.cpp	2014-05-05 12:00:15.814804393 +0200
@@ -409,8 +409,13 @@
 bool RingBuffer::IsNearEnd(double fps, uint vvf) const
 {
     rwlock.lockForRead();
-    int    sz  = ReadBufAvail();
-    uint   rbs = readblocksize;
+    long long sz2 = ReadBufAvail();
+    poslock.lockForRead();
+    long long rp  = readpos;
+    poslock.unlock();
+    long long sz  = GetRealFileSize() - rp;
+    sz = max(sz, sz2);
+
     // telecom kilobytes (i.e. 1000 per k not 1024)
     uint   tmp = (uint) max(abs(rawbitrate * playspeed), 0.5f * rawbitrate);
     uint   kbits_per_sec = min(rawbitrate * 3, tmp);
@@ -422,11 +427,11 @@
     double bytes_per_frame = kbits_per_sec * (1000.0/8.0) / fps;
     double readahead_frames = sz / bytes_per_frame;
 
-    bool near_end = ((vvf + readahead_frames) < 20.0) || (sz < rbs*1.5);
+    bool near_end = (vvf + readahead_frames) < 20.0;
 
     LOG(VB_PLAYBACK, LOG_INFO, LOC + "IsReallyNearEnd()" +
             QString(" br(%1KB)").arg(kbits_per_sec/8) +
-            QString(" sz(%1KB)").arg(sz / 1000) +
+            QString(" sz(%1KB)").arg(sz / 1000LL) +
             QString(" vfl(%1)").arg(vvf) +
             QString(" frh(%1)").arg(((uint)readahead_frames)) +
             QString(" ne:%1").arg(near_end));
@@ -914,7 +919,7 @@
                 rbwlock.unlock();
                 poslock.unlock();
 
-                LOG(VB_FILE, LOG_DEBUG, LOC +
+                LOG(VB_FILE, LOG_INFO, LOC +
                     QString("total read so far: %1 bytes")
                     .arg(internalreadpos));
             }
