diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/autoexpire.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/autoexpire.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/programs/mythbackend/autoexpire.cpp	2014-05-05 12:01:19.684441131 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/programs/mythbackend/autoexpire.cpp	2014-05-05 12:01:33.801027621 +0200
@@ -941,8 +941,8 @@
         case emShortLiveTVPrograms:
             msg = "Adding Short LiveTV programs in starttime order";
             where = "recgroup = 'LiveTV' "
-                    "AND endtime < DATE_ADD(starttime, INTERVAL '2' MINUTE) "
-                    "AND endtime <= DATE_ADD(NOW(), INTERVAL '-1' MINUTE) ";
+                    "AND endtime < DATE_ADD(starttime, INTERVAL '30' SECOND) "
+                    "AND endtime <= DATE_ADD(NOW(), INTERVAL '-35' MINUTE) ";
             orderby = "starttime ASC";
             break;
         case emNormalLiveTVPrograms:
