diff -Naur mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/fileringbuffer.cpp mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/fileringbuffer.cpp
--- mythtv-0.27-20140504-ge2acf0d-old/mythtv/libs/libmythtv/fileringbuffer.cpp	2014-05-05 12:01:56.167567337 +0200
+++ mythtv-0.27-20140504-ge2acf0d-new/mythtv/libs/libmythtv/fileringbuffer.cpp	2014-05-05 12:02:06.300843188 +0200
@@ -379,7 +379,7 @@
     commserror = false;
     numfailures = 0;
 
-    rawbitrate = 8000;
+    rawbitrate = 800;
     CalcReadAheadThresh();
 
     bool ok = fd2 >= 0 || remotefile;
