diff -Naur mythtv-0.27-build-old/mythtv/libs/libmythbase/mythcorecontext.cpp mythtv-0.27-build-new/mythtv/libs/libmythbase/mythcorecontext.cpp
--- mythtv-0.27-build-old/mythtv/libs/libmythbase/mythcorecontext.cpp	2014-07-22 21:53:37.000000000 +0200
+++ mythtv-0.27-build-new/mythtv/libs/libmythbase/mythcorecontext.cpp	2014-07-22 21:54:44.667541494 +0200
@@ -630,6 +630,11 @@
     return (res == GENERIC_EXIT_OK);
 }
 
+bool MythCoreContext::IsThisBackend(const QString &addr)
+{
+    return IsBackend() && IsThisHost(addr);
+}
+
 bool MythCoreContext::IsThisHost(const QString &addr)
 {
     return IsThisHost(addr, GetHostName());
diff -Naur mythtv-0.27-build-old/mythtv/libs/libmythbase/mythcorecontext.h mythtv-0.27-build-new/mythtv/libs/libmythbase/mythcorecontext.h
--- mythtv-0.27-build-old/mythtv/libs/libmythbase/mythcorecontext.h	2014-07-22 21:53:36.000000000 +0200
+++ mythtv-0.27-build-new/mythtv/libs/libmythbase/mythcorecontext.h	2014-07-22 21:55:48.154020192 +0200
@@ -103,6 +103,7 @@
     bool IsMasterBackend(void);  ///< is this the actual MBE process
     bool BackendIsRunning(void); ///< a backend process is running on this host
 
+    bool IsThisBackend(const QString &addr);    /// it this address mapped to this backend host
     bool IsThisHost(const QString &addr); //< is this address mapped to this host
     bool IsThisHost(const QString &addr, const QString &host);
 
