diff -Naur mythtv-0.27-build-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-0.27-build-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-0.27-build-old/mythtv/programs/mythbackend/mainserver.cpp	2014-07-23 09:24:34.000000000 +0200
+++ mythtv-0.27-build-new/mythtv/programs/mythbackend/mainserver.cpp	2014-07-23 09:31:19.653853553 +0200
@@ -5511,7 +5511,7 @@
 
     QString hostname = slist[1];
 
-    if (ismaster && hostname != gCoreContext->GetHostName())
+    if (ismaster && !gCoreContext->IsThisHost(hostname))
     {
         // forward the request to the slave BE
         PlaybackSock *slave = GetMediaServerByHostname(hostname);
@@ -5573,7 +5573,7 @@
 
     QString hostname = slist[1];
 
-    if (ismaster && hostname != gCoreContext->GetHostName())
+    if (ismaster && !gCoreContext->IsThisHost(hostname))
     {
         // forward the request to the slave BE
         PlaybackSock *slave = GetMediaServerByHostname(hostname);
@@ -5633,7 +5633,7 @@
 
     QString hostname = slist[1];
 
-    if (ismaster && hostname != gCoreContext->GetHostName())
+    if (ismaster && !gCoreContext->IsThisHost(hostname))
     {
         // forward the request to the slave BE
         PlaybackSock *slave = GetMediaServerByHostname(hostname);
@@ -5719,7 +5719,7 @@
 
     QString hostname = slist[1];
 
-    if (ismaster && hostname != gCoreContext->GetHostName())
+    if (ismaster && !gCoreContext->IsThisHost(hostname))
     {
         // forward the request to the slave BE
         PlaybackSock *slave = GetMediaServerByHostname(hostname);
@@ -5875,7 +5875,7 @@
     QString songid = slist[2];
     QString imagetype = slist[3];
 
-    if (ismaster && hostname != gCoreContext->GetHostName())
+    if (ismaster && !gCoreContext->IsThisHost(hostname))
     {
 
         // forward the request to the slave BE
@@ -5930,7 +5930,7 @@
 
     QString hostname = slist[1];
 
-    if (ismaster && hostname != gCoreContext->GetHostName())
+    if (ismaster && !gCoreContext->IsThisHost(hostname))
     {
         // forward the request to the slave BE
         PlaybackSock *slave = GetMediaServerByHostname(hostname);
@@ -6103,7 +6103,7 @@
 
     QString hostname = slist[1];
 
-    if (ismaster && hostname != gCoreContext->GetHostName())
+    if (ismaster && !gCoreContext->IsThisHost(hostname))
     {
         // forward the request to the slave BE
         PlaybackSock *slave = GetMediaServerByHostname(hostname);
@@ -6262,7 +6262,7 @@
 
     QString hostname = slist[1];
 
-    if (ismaster && hostname != gCoreContext->GetHostName())
+    if (ismaster && !gCoreContext->IsThisHost(hostname))
     {
         // forward the request to the slave BE
         PlaybackSock *slave = GetMediaServerByHostname(hostname);
