From e4d1f7e56743bfa26f0fe1a2e5be62ab1f5c6667 Mon Sep 17 00:00:00 2001
From: Roger Siddons <dizygotheca@ntlworld.com>
Date: Thu, 30 Oct 2014 18:41:31 +0000
Subject: [PATCH 04/11] New Image Gallery: Fix segfault from using INFO on
 Slideshow

Signed-off-by: Paul Harrison <pharrison@mythtv.org>
---
 mythtv/programs/mythfrontend/gallerywidget.cpp |    2 --
 1 file changed, 2 deletions(-)

diff --git a/mythtv/programs/mythfrontend/gallerywidget.cpp b/mythtv/programs/mythfrontend/gallerywidget.cpp
index 58495e5..d22b29c4 100644
--- a/mythtv/programs/mythfrontend/gallerywidget.cpp
+++ b/mythtv/programs/mythfrontend/gallerywidget.cpp
@@ -499,8 +499,6 @@ void GalleryWidget::ShowFileDetails()
     // All widgets are visible, remember this
     m_infoVisible = true;
     SetFocusWidget(m_infoList);
-
-    delete im;
 }
 
 
-- 
1.7.10.2

