From 327f5fa9e83e59098822d96bbd92a0e6dc504715 Mon Sep 17 00:00:00 2001
From: Roger Siddons <dizygotheca@ntlworld.com>
Date: Thu, 30 Oct 2014 18:43:12 +0000
Subject: [PATCH 05/11] New Image Gallery: Restore failed transform correctly

Signed-off-by: Paul Harrison <pharrison@mythtv.org>
---
 mythtv/programs/mythfrontend/galleryviewhelper.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/programs/mythfrontend/galleryviewhelper.cpp b/mythtv/programs/mythfrontend/galleryviewhelper.cpp
index b12ada5..4ea28cd 100644
--- a/mythtv/programs/mythfrontend/galleryviewhelper.cpp
+++ b/mythtv/programs/mythfrontend/galleryviewhelper.cpp
@@ -401,7 +401,7 @@ void GalleryViewHelper::SetFileOrientation(int fileOrientation)
     }
     else
     {
-        im->SetOrientation(oldFileOrientation, false);
+        im->SetOrientation(oldFileOrientation, true);
 
         LOG(VB_GENERAL, LOG_ERR,
             QString("Could not write the angle %1 into the file %2. "
-- 
1.7.10.2

