diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythgallery/mythgallery/galleryutil.cpp mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythgallery/mythgallery/galleryutil.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythgallery/mythgallery/galleryutil.cpp	2014-05-19 15:29:33.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythgallery/mythgallery/galleryutil.cpp	2014-06-08 19:22:06.565374292 +0200
@@ -484,7 +484,7 @@
     // rename the file's thumbnail if it exists
     if (QFile::exists(currDir + "/.thumbcache/" + oldName))
     {
-        QDir d(currDir + "/.thumbcache/");
+        QDir d(currDir + "/cache/gallery-thumbcache/");
         d.rename(oldName, newName);
     }
 
@@ -656,7 +656,7 @@
     // rename the directory's thumbnail if it exists in the parent directory
     if (QFile::exists(currDir + "/.thumbcache/" + oldName))
     {
-        QDir d(currDir + "/.thumbcache/");
+        QDir d(currDir + "/cache/gallery-thumbcache/");
         d.rename(oldName, newName);
     }
 
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2014-06-08 19:20:28.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2014-06-08 19:22:06.565374292 +0200
@@ -276,7 +276,7 @@
     }
 
     MusicMetadata *titleTrack = NULL;
-    QString saveDir = GetConfDir() + "/MythMusic/RipTemp/";
+    QString saveDir = GetConfDir() + "/tmp/RipTemp/";
     QString outfile;
 
     std::auto_ptr<Encoder> encoder;
@@ -578,10 +578,10 @@
 
     // make sure the directory where we temporarily save the rips is present
     QDir dir;
-    dir.mkpath(GetConfDir() + "/MythMusic/RipTemp/");
+    dir.mkpath(GetConfDir() + "/tmp/RipTemp/");
 
     // remove any ripped tracks from the temp rip directory
-    QString command = "rm -f " + GetConfDir() + "/MythMusic/RipTemp/*";
+    QString command = "rm -f " + GetConfDir() + "/tmp/RipTemp/*";
     myth_system(command);
 
     // get last host and directory we ripped to
@@ -594,7 +594,7 @@
 Ripper::~Ripper(void)
 {
     // remove any ripped tracks from the temp rip directory
-    QString command = "rm -f " + GetConfDir() + "/MythMusic/RipTemp/*";
+    QString command = "rm -f " + GetConfDir() + "/tmp/RipTemp/*";
     myth_system(command);
 
     if (m_decoder)
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netcommon.cpp mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netcommon.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netcommon.cpp	2014-05-19 15:29:44.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netcommon.cpp	2014-06-08 19:22:06.565374292 +0200
@@ -18,13 +18,7 @@
     if (!dir.exists())
         dir.mkdir(fileprefix);
 
-    fileprefix += "/MythNetvision";
-
-    dir = QDir(fileprefix);
-    if (!dir.exists())
-        dir.mkdir(fileprefix);
-
-    fileprefix += "/thumbcache";
+    fileprefix += "/cache/netvision-thumbcache";
 
     dir = QDir(fileprefix);
     if (!dir.exists())
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netsearch.cpp mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netsearch.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netsearch.cpp	2014-06-08 19:20:24.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netsearch.cpp	2014-06-08 19:22:06.565374292 +0200
@@ -313,7 +313,7 @@
 
 void NetSearch::cleanCacheDir()
 {
-    QString cache = QString("%1/MythNetvision/thumbcache")
+    QString cache = QString("%1/cache/netvision-thumbcache")
                        .arg(GetConfDir());
     QDir cacheDir(cache);
     QStringList thumbs = cacheDir.entryList(QDir::Files);
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/nettree.cpp mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/nettree.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/nettree.cpp	2014-05-19 15:29:48.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/nettree.cpp	2014-06-08 19:22:06.565374292 +0200
@@ -196,7 +196,7 @@
 
 void NetTree::cleanCacheDir()
 {
-    QString cache = QString("%1/thumbcache")
+    QString cache = QString("%1/cache/netvision-thumbcache")
                        .arg(GetConfDir());
     QDir cacheDir(cache);
     QStringList thumbs = cacheDir.entryList(QDir::Files);
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythbase/mythdownloadmanager.cpp mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythbase/mythdownloadmanager.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythbase/mythdownloadmanager.cpp	2014-05-19 15:28:57.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythbase/mythdownloadmanager.cpp	2014-06-08 19:22:06.565374292 +0200
@@ -251,7 +251,7 @@
     m_manager = new QNetworkAccessManager(this);
     m_diskCache = new QNetworkDiskCache(this);
     m_proxy = new QNetworkProxy();
-    m_diskCache->setCacheDirectory(GetConfDir() + "/Cache-" +
+    m_diskCache->setCacheDirectory(GetConfDir() + "/cache/" +
                                    QCoreApplication::applicationName() + "-" +
                                    gCoreContext->GetHostName());
     m_manager->setCache(m_diskCache);
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythmetadata/metadataimagedownload.cpp mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythmetadata/metadataimagedownload.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythmetadata/metadataimagedownload.cpp	2014-05-19 15:28:54.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythmetadata/metadataimagedownload.cpp	2014-06-08 19:22:06.565374292 +0200
@@ -357,7 +357,7 @@
     if (!dir.exists())
         dir.mkdir(fileprefix);
 
-    fileprefix += "/thumbcache";
+    fileprefix += "/cache/metadata-thumbcache";
 
     dir = QDir(fileprefix);
     if (!dir.exists())
@@ -515,7 +515,7 @@
 
 void cleanThumbnailCacheDir()
 {
-    QString cache = QString("%1/thumbcache")
+    QString cache = QString("%1/cache/metadata-thumbcache")
                .arg(GetConfDir());
     QDir cacheDir(cache);
     QStringList thumbs = cacheDir.entryList(QDir::Files);
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythmetadata/mythuiimageresults.cpp mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythmetadata/mythuiimageresults.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythmetadata/mythuiimageresults.cpp	2014-05-19 15:28:54.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythmetadata/mythuiimageresults.cpp	2014-06-08 19:22:06.565374292 +0200
@@ -94,7 +94,7 @@
 
 void ImageSearchResultsDialog::cleanCacheDir()
 {
-    QString cache = QString("%1/thumbcache")
+    QString cache = QString("%1/cache/metadata-thumbcache")
                .arg(GetConfDir());
     QDir cacheDir(cache);
     QStringList thumbs = cacheDir.entryList(QDir::Files);
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythmetadata/mythuimetadataresults.cpp mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythmetadata/mythuimetadataresults.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythmetadata/mythuimetadataresults.cpp	2014-05-19 15:28:54.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythmetadata/mythuimetadataresults.cpp	2014-06-08 19:22:06.568708013 +0200
@@ -100,7 +100,7 @@
 
 void MetadataResultsDialog::cleanCacheDir()
 {
-    QString cache = QString("%1/thumbcache")
+    QString cache = QString("%1/cache/metadata-thumbcache")
                        .arg(GetConfDir());
     QDir cacheDir(cache);
     QStringList thumbs = cacheDir.entryList(QDir::Files);
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythtv/previewgenerator.cpp mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythtv/previewgenerator.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythtv/previewgenerator.cpp	2014-05-19 15:28:52.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythtv/previewgenerator.cpp	2014-06-08 19:22:06.568708013 +0200
@@ -479,7 +479,7 @@
     if (outFileName.isEmpty())
     {
         QString remotecachedirname =
-            QString("%1/remotecache").arg(GetConfDir());
+            QString("%1/cache/remotecache").arg(GetConfDir());
         QDir remotecachedir(remotecachedirname);
 
         if (!remotecachedir.exists())
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythtv/previewgeneratorqueue.cpp mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythtv/previewgeneratorqueue.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythtv/previewgeneratorqueue.cpp	2014-05-19 15:28:52.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythtv/previewgeneratorqueue.cpp	2014-06-08 19:22:06.568708013 +0200
@@ -289,7 +289,7 @@
 
         if (streaming)
         {
-            ret_file = QString("%1/remotecache/%2")
+            ret_file = QString("%1/cache/remotecache/%2")
                 .arg(GetConfDir()).arg(filename.section('/', -1));
 
             QFileInfo finfo(ret_file);
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythui/mythuihelper.cpp mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythui/mythuihelper.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/libs/libmythui/mythuihelper.cpp	2014-06-08 19:20:28.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/libs/libmythui/mythuihelper.cpp	2014-06-08 19:22:06.568708013 +0200
@@ -719,7 +719,7 @@
 
 QString MythUIHelper::GetThemeCacheDir(void)
 {
-    QString cachedirname = GetConfDir() + "/themecache/";
+    QString cachedirname = GetConfDir() + "/cache/themecache/";
 
     QString tmpcachedir = cachedirname +
                           GetMythDB()->GetSetting("Theme", DEFAULT_UI_THEME) +
@@ -731,7 +731,7 @@
 
 void MythUIHelper::ClearOldImageCache(void)
 {
-    QString cachedirname = GetConfDir() + "/themecache/";
+    QString cachedirname = GetConfDir() + "/cache/themecache/";
 
     d->themecachedir = GetThemeCacheDir();
 
@@ -795,7 +795,7 @@
 
 void MythUIHelper::RemoveCacheDir(const QString &dirname)
 {
-    QString cachedirname = GetConfDir() + "/themecache/";
+    QString cachedirname = GetConfDir() + "/cache/themecache/";
 
     if (!dirname.startsWith(cachedirname))
         return;
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/programs/mythtv-setup/importicons.cpp mythtv-0.27-20140519-gf4825ca-new/mythtv/programs/mythtv-setup/importicons.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/programs/mythtv-setup/importicons.cpp	2014-05-19 15:29:28.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/programs/mythtv-setup/importicons.cpp	2014-06-08 19:22:06.568708013 +0200
@@ -39,7 +39,7 @@
 
     m_popupStack = GetMythMainWindow()->GetStack("popup stack");
 
-    m_tmpDir = QDir(QString("%1/icontmp").arg(GetConfDir()));
+    m_tmpDir = QDir(QString("%1/tmp/icon").arg(GetConfDir()));
 
     if (!m_tmpDir.exists())
         m_tmpDir.mkpath(m_tmpDir.absolutePath());
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythtv/programs/scripts/metadata/Movie/tmdb3.py mythtv-0.27-20140519-gf4825ca-new/mythtv/programs/scripts/metadata/Movie/tmdb3.py
--- mythtv-0.27-20140519-gf4825ca-old/mythtv/programs/scripts/metadata/Movie/tmdb3.py	2014-05-19 15:29:28.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythtv/programs/scripts/metadata/Movie/tmdb3.py	2014-06-08 19:25:10.160256707 +0200
@@ -275,7 +275,7 @@
                 print "Unable to find MythTV directory for metadata cache."
                 sys.exit(1)
             confdir = os.path.join(confdir, '.mythtv')
-        confpath = os.path.join(confdir, 'pytmdb3.cache')
+        confpath = os.path.join(confdir, 'cache/pytmdb3.cache')
         set_cache(engine='file', filename=confpath)
 
     if opts.language:
