diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netbase.cpp mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netbase.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netbase.cpp	2014-06-08 21:23:56.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netbase.cpp	2014-06-08 21:25:38.438350980 +0200
@@ -15,12 +15,12 @@
 
 NetBase::NetBase(MythScreenStack *parent, const char *name)
     : MythScreenType(parent, name),
-    m_thumbImage(NULL),
-    m_downloadable(NULL),
-    m_busyPopup(NULL),
-    m_popupStack(GetMythMainWindow()->GetStack("popup stack")),
-    m_progressDialog(NULL),
-    m_imageDownload(new MetadataImageDownload(this))
+      m_thumbImage(NULL),
+      m_downloadable(NULL),
+      m_busyPopup(NULL),
+      m_popupStack(GetMythMainWindow()->GetStack("popup stack")),
+      m_progressDialog(NULL),
+      m_imageDownload(new MetadataImageDownload(this))
 {
     gCoreContext->addListener(this);
 }
@@ -58,9 +58,7 @@
         m_popupStack, "videodownloadprogressdialog");
 
     if (m_progressDialog->Create())
-    {
         m_popupStack->AddScreen(m_progressDialog, false);
-    }
     else
     {
         delete m_progressDialog;
@@ -117,8 +115,7 @@
         return;
     }
 
-    GetMythMainWindow()->HandleMedia(
-        "Internal", item->GetMediaURL(),
+    GetMythMainWindow()->HandleMedia("Internal", item->GetMediaURL(),
         item->GetDescription(), item->GetTitle(), item->GetSubtitle(),
         QString(), item->GetSeason(), item->GetEpisode(), QString(),
         item->GetTime().toInt(), item->GetDate().toString("yyyy"));
@@ -143,11 +140,13 @@
         }
         else
         {
-            args.replaceInStrings("%DIR%", QString(GetConfDir() + "/MythNetvision"));
+            args.replaceInStrings("%DIR%",
+                                  QString(GetConfDir() + "/MythNetvision"));
             args.replaceInStrings("%MEDIAURL%", item->GetMediaURL());
             args.replaceInStrings("%URL%", item->GetURL());
             args.replaceInStrings("%TITLE%", item->GetTitle());
         }
+
         QString playerCommand = cmd + " " + args.join(" ");
         RunCmdWithoutScreensaver(playerCommand);
     }
@@ -165,8 +164,8 @@
 
         if (browser.isEmpty())
         {
-            ShowOkPopup(tr("No browser command set! MythNetTree needs MythBrowser "
-                           "installed to display the video."));
+            ShowOkPopup(tr("No browser command set! MythNetTree needs "
+                           "MythBrowser installed to display the video."));
             return;
         }
 
@@ -260,8 +259,10 @@
             {
                 QString message = tr("Downloading Video...\n"
                                      "(%1 of %2 MB)")
-                    .arg(QString::number(args[2].toInt() / 1024.0 / 1024.0, 'f', 2))
-                    .arg(QString::number(args[3].toInt() / 1024.0 / 1024.0, 'f', 2));
+                    .arg(QString::number(args[2].toInt() / 1024.0 / 1024.0,
+                                         'f', 2))
+                    .arg(QString::number(args[3].toInt() / 1024.0 / 1024.0,
+                                         'f', 2));
                 m_progressDialog->SetMessage(message);
                 m_progressDialog->SetTotal(args[3].toInt());
                 m_progressDialog->SetProgress(args[2].toInt());
@@ -298,14 +299,14 @@
         return;
 
     QString baseFilename = GetDownloadFilename(item->GetTitle(),
-                                          item->GetMediaURL());
+                                               item->GetMediaURL());
 
     QString finalFilename = generate_file_url("Default",
-                              gCoreContext->GetMasterHostName(),
-                              baseFilename);
+                                              gCoreContext->GetMasterHostName(),
+                                              baseFilename);
 
     LOG(VB_GENERAL, LOG_INFO, QString("Downloading %1 to %2")
-        .arg(item->GetMediaURL()) .arg(finalFilename));
+        .arg(item->GetMediaURL()).arg(finalFilename));
 
     // Does the file already exist?
     bool exists = RemoteFile::Exists(finalFilename);
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netcommon.cpp mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netcommon.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netcommon.cpp	2014-06-08 21:23:56.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netcommon.cpp	2014-06-08 21:25:38.438350980 +0200
@@ -52,7 +52,8 @@
     int MasterStatusPort = gCoreContext->GetNumSettingOnHost("BackendStatusPort",
                                                           MasterHost);
 
-    return QString("http://%1:%2/InternetContent/").arg(MasterIP).arg(MasterStatusPort);
+    return QString("http://%1:%2/InternetContent/").arg(MasterIP)
+        .arg(MasterStatusPort);
 }
 
 QUrl GetMythXMLSearch(QString url, QString query, QString grabber,
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netsearch.cpp mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netsearch.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/netsearch.cpp	2014-06-08 21:23:56.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/netsearch.cpp	2014-06-08 21:25:38.438350980 +0200
@@ -36,7 +36,6 @@
       m_search(NULL),                m_pageText(NULL),
       m_noSites(NULL),               m_progress(NULL),
       m_okPopup(NULL),               m_netSearch(NULL),
-
       m_reply(NULL),
       m_currentSearch(QString()),    m_currentGrabber(0),
       m_currentCmd(QString()),       m_pagenum(0),
@@ -82,13 +81,13 @@
 
     // UI Hookups
     connect(m_siteList, SIGNAL(itemSelected(MythUIButtonListItem *)),
-                       SLOT(SlotItemChanged()));
+            SLOT(SlotItemChanged()));
     connect(m_siteList, SIGNAL(itemClicked(MythUIButtonListItem *)),
-                       SLOT(DoSearch(void)));
+            SLOT(DoSearch(void)));
     connect(m_searchResultList, SIGNAL(itemClicked(MythUIButtonListItem *)),
-                       SLOT(StreamWebVideo(void)));
+            SLOT(StreamWebVideo(void)));
     connect(m_searchResultList, SIGNAL(itemSelected(MythUIButtonListItem *)),
-                       SLOT(SlotItemChanged()));
+            SLOT(SlotItemChanged()));
 
     BuildFocusList();
 
@@ -176,7 +175,7 @@
     QString label = tr("Search Options");
 
     MythDialogBox *menuPopup = new MythDialogBox(label, m_popupStack,
-                                                    "mythnetvisionmenupopup");
+                                                 "mythnetvisionmenupopup");
 
     if (menuPopup->Create())
     {
@@ -190,11 +189,12 @@
             if (item)
             {
                 if (item->GetDownloadable())
-                    menuPopup->AddButton(tr("Stream Video"), SLOT(StreamWebVideo()));
+                    menuPopup->AddButton(tr("Stream Video"),
+                                         SLOT(StreamWebVideo()));
                 menuPopup->AddButton(tr("Open Web Link"), SLOT(ShowWebVideo()));
 
                 QString filename = GetDownloadFilename(item->GetTitle(),
-                    item->GetMediaURL());
+                                                       item->GetMediaURL());
                 bool exists = false;
 
                 if (filename.startsWith("myth://"))
@@ -206,9 +206,11 @@
                     GetFocusWidget() == m_searchResultList)
                 {
                     if (exists)
-                        menuPopup->AddButton(tr("Play"), SLOT(DoPlayVideo(filename)));
+                        menuPopup->AddButton(tr("Play"),
+                                             SLOT(DoPlayVideo(filename)));
                     else
-                        menuPopup->AddButton(tr("Save This Video"), SLOT(DoDownloadAndPlay()));
+                        menuPopup->AddButton(tr("Save This Video"),
+                                             SLOT(DoDownloadAndPlay()));
                 }
 
                 if (item->GetDownloadable() &&
@@ -220,17 +222,16 @@
             }
 
             if (m_pagenum > 1)
-                menuPopup->AddButton(tr("Previous Page"), SLOT(GetLastResults()));
+                menuPopup->AddButton(tr("Previous Page"),
+                                     SLOT(GetLastResults()));
             if (m_searchResultList->GetCount() > 0 && m_pagenum < m_maxpage)
                 menuPopup->AddButton(tr("Next Page"), SLOT(GetMoreResults()));
         }
-        menuPopup->AddButton(tr("Manage Search Scripts"), SLOT(RunSearchEditor()));
-
+        menuPopup->AddButton(tr("Manage Search Scripts"),
+                             SLOT(RunSearchEditor()));
     }
     else
-    {
         delete menuPopup;
-    }
 }
 
 void NetSearch::FillGrabberButtonList()
@@ -283,10 +284,10 @@
     if (!m_netSearch)
         m_netSearch = new QNetworkAccessManager(this);
     connect(m_netSearch, SIGNAL(finished(QNetworkReply*)),
-                       SLOT(SearchFinished(void)));
+            SLOT(SearchFinished(void)));
 
     QUrl init = GetMythXMLSearch(m_mythXML, m_currentSearch,
-                                m_currentCmd, m_pagenum);
+                                 m_currentCmd, m_pagenum);
     QUrl req(init.toEncoded(), QUrl::TolerantMode);
     LOG(VB_GENERAL, LOG_INFO,
         QString("Using Search URL %1").arg(req.toString()));
@@ -360,8 +361,7 @@
         else
             m_maxpage = (searchresults/returned);
     }
-    if (m_pageText && m_maxpage > 0 && m_pagenum > 0 &&
-        returned > 0)
+    if (m_pageText && m_maxpage > 0 && m_pagenum > 0 && returned > 0)
         m_pageText->SetText(QString("%1 / %2")
                         .arg(QString::number(m_pagenum))
                         .arg(QString::number(m_maxpage)));
@@ -383,8 +383,7 @@
 
     if (!m_okPopup)
     {
-        m_okPopup = new MythConfirmationDialog(m_popupStack,
-                                         message, false);
+        m_okPopup = new MythConfirmationDialog(m_popupStack, message, false);
 
         if (m_okPopup->Create())
             m_popupStack->AddScreen(m_okPopup);
@@ -444,15 +443,13 @@
 
     if (searchedit->Create())
     {
-        connect(searchedit, SIGNAL(ItemsChanged()), this,
-                       SLOT(DoListRefresh()));
+        connect(searchedit, SIGNAL(ItemsChanged()),
+                this, SLOT(DoListRefresh()));
 
         mainStack->AddScreen(searchedit);
     }
     else
-    {
         delete searchedit;
-    }
 }
 
 void NetSearch::DoListRefresh()
@@ -547,20 +544,13 @@
         uint pos = qVariantValue<uint>(data->data);
 
         if (file.isEmpty() || !((uint)m_searchResultList->GetCount() >= pos))
-        {
             return;
-        }
 
-        MythUIButtonListItem *item =
-                  m_searchResultList->GetItemAt(pos);
+        MythUIButtonListItem *item = m_searchResultList->GetItemAt(pos);
 
         if (item && item->GetText() == title)
-        {
             item->SetImage(file);
-        }
     }
     else
-    {
         NetBase::customEvent(event);
-    }
 }
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/nettree.cpp mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/nettree.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/nettree.cpp	2014-06-08 21:23:56.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/nettree.cpp	2014-06-08 21:25:38.438350980 +0200
@@ -145,11 +145,8 @@
 
 NetTree::~NetTree()
 {
-    if (m_siteGeneric)
-    {
-        delete m_siteGeneric;
-        m_siteGeneric = NULL;
-    }
+    delete m_siteGeneric;
+    m_siteGeneric = NULL;
 
     delete m_gdt;
     m_gdt = NULL;
@@ -226,7 +223,8 @@
 
     if (nodeInt == kSubFolder)
     {
-        item->SetText(QString("%1").arg(node->visibleChildCount()), "childcount");
+        item->SetText(QString("%1").arg(node->visibleChildCount()),
+                      "childcount");
         item->DisplayState("subfolder", "nodetype");
         item->SetText(node->GetText(), "title");
         item->SetText(node->GetText());
@@ -265,7 +263,7 @@
             dlfile.replace("%SHAREDIR%", GetShareDir());
         else
             dlfile = getDownloadFilename(video->GetTitle(),
-                         video->GetThumbnail());
+                                         video->GetThumbnail());
 
         if (QFile::exists(dlfile))
             item->SetImage(dlfile);
@@ -414,7 +412,8 @@
     }
     else
     {
-        MythGenericTree *node = GetNodePtrFromButton(m_siteButtonList->GetItemCurrent());
+        MythGenericTree *node =
+            GetNodePtrFromButton(m_siteButtonList->GetItemCurrent());
 
         if (node)
             item = qVariantValue<ResultItem *>(node->GetData());
@@ -433,7 +432,8 @@
     menu->AddItem(tr("Scan/Manage Subscriptions"), NULL, CreateShowManageMenu());
     menu->AddItem(tr("Change View"), NULL, CreateShowViewMenu());
 
-    MythDialogBox *menuPopup = new MythDialogBox(menu, m_popupStack, "mythnettreemenupopup");
+    MythDialogBox *menuPopup =
+        new MythDialogBox(menu, m_popupStack, "mythnettreemenupopup");
 
     if (menuPopup->Create())
         m_popupStack->AddScreen(menuPopup);
@@ -463,15 +463,16 @@
 
     MythMenu *menu = new MythMenu(label, this, "options");
 
-
     menu->AddItem(tr("Update Site Maps"), SLOT(UpdateTrees()));
     menu->AddItem(tr("Update RSS"), SLOT(UpdateRSS()));
     menu->AddItem(tr("Manage Site Subscriptions"), SLOT(RunTreeEditor()));
     menu->AddItem(tr("Manage RSS Subscriptions"), SLOT(RunRSSEditor()));
     if (!m_treeAutoUpdate)
-        menu->AddItem(tr("Enable Automatic Site Updates"), SLOT(ToggleTreeUpdates()));
+        menu->AddItem(tr("Enable Automatic Site Updates"),
+                      SLOT(ToggleTreeUpdates()));
     else
-        menu->AddItem(tr("Disable Automatic Site Updates"), SLOT(ToggleTreeUpdates()));
+        menu->AddItem(tr("Disable Automatic Site Updates"),
+                      SLOT(ToggleTreeUpdates()));
 //    if (!m_rssAutoUpdate)
 //        menu->AddItem(tr("Enable Automatic RSS Updates"), SLOT(ToggleRSSUpdates()));
 //    else
@@ -501,7 +502,7 @@
 void NetTree::SwitchView()
 {
     NetTree *nettree =
-            new NetTree(m_type, GetMythMainWindow()->GetMainStack(), "nettree");
+        new NetTree(m_type, GetMythMainWindow()->GetMainStack(), "nettree");
 
     if (nettree->Create())
     {
@@ -523,43 +524,36 @@
         MythGenericTree *rssGeneric =
             new MythGenericTree(RSSNode, kSubFolder, false);
 
-    // Add an upfolder
-    if (m_type != DLG_TREE)
-    {
-        rssGeneric->addNode(tr("Back"), kUpFolder, true, false);
-    }
-
-    rssGeneric->SetData(QString("%1/mythnetvision/icons/rss.png")
-         .arg(GetShareDir()));
-
-    RSSSite::rssList::iterator i = m_rssList.begin();
-    for (; i != m_rssList.end(); ++i)
-    {
-        ResultItem::resultList items =
-                   getRSSArticles((*i)->GetTitle(), VIDEO_PODCAST);
-        MythGenericTree *ret = new MythGenericTree(
-                   (*i)->GetTitle(), kSubFolder, false);
-        ret->SetData(qVariantFromValue(*i));
-        rssGeneric->addNode(ret);
-
         // Add an upfolder
         if (m_type != DLG_TREE)
-        {
-            ret->addNode(tr("Back"), kUpFolder, true, false);
-        }
+            rssGeneric->addNode(tr("Back"), kUpFolder, true, false);
 
-        ResultItem::resultList::iterator it = items.begin();
-        for (; it != items.end(); ++it)
+        rssGeneric->SetData(QString("%1/mythnetvision/icons/rss.png")
+                            .arg(GetShareDir()));
+
+        RSSSite::rssList::iterator i = m_rssList.begin();
+        for (; i != m_rssList.end(); ++i)
         {
-            AddFileNode(ret, *it);
+            ResultItem::resultList items = getRSSArticles((*i)->GetTitle(),
+                                                          VIDEO_PODCAST);
+            MythGenericTree *ret =
+                new MythGenericTree((*i)->GetTitle(), kSubFolder, false);
+            ret->SetData(qVariantFromValue(*i));
+            rssGeneric->addNode(ret);
+
+            // Add an upfolder
+            if (m_type != DLG_TREE)
+                ret->addNode(tr("Back"), kUpFolder, true, false);
+
+            ResultItem::resultList::iterator it = items.begin();
+            for (; it != items.end(); ++it)
+                AddFileNode(ret, *it);
         }
-    }
 
-    m_siteGeneric->addNode(rssGeneric);
+        m_siteGeneric->addNode(rssGeneric);
     }
 
     // Now let's add all the grabber trees
-
     for (GrabberScript::scriptList::iterator i = m_grabberList.begin();
             i != m_grabberList.end(); ++i)
     {
@@ -577,9 +571,7 @@
 
         // Add an upfolder
         if (m_type != DLG_TREE)
-        {
             ret->addNode(tr("Back"), kUpFolder, true, false);
-        }
 
         for (QList<QPair<QString, QString> >::iterator i = paths.begin();
                 i != paths.end(); ++i)
@@ -609,8 +601,7 @@
         if (tmp)
             dst = tmp;
         else
-            folder = new MythGenericTree(
-                     curPath, kSubFolder, false);
+            folder = new MythGenericTree(curPath, kSubFolder, false);
     }
 
     if (!folder)
@@ -621,9 +612,7 @@
 
     // Add an upfolder
     if (m_type != DLG_TREE)
-    {
-          folder->addNode(tr("Back"), kUpFolder, true, false);
-    }
+        folder->addNode(tr("Back"), kUpFolder, true, false);
 
     if (paths.size())
         BuildGenericTree(folder, paths, dirthumb, videos);
@@ -640,8 +629,7 @@
 {
     QString title = video->GetTitle();
     title.replace("&amp;", "&");
-    MythGenericTree *sub_node = where_to_add->
-                    addNode(title, 0, true);
+    MythGenericTree *sub_node = where_to_add->addNode(title, 0, true);
     sub_node->SetData(qVariantFromValue(video));
     m_videos.append(video);
     return 1;
@@ -652,7 +640,8 @@
     ResultItem *item = NULL;
 
     if (m_type == DLG_TREE)
-        item = qVariantValue<ResultItem *>(m_siteMap->GetCurrentNode()->GetData());
+        item =
+            qVariantValue<ResultItem *>(m_siteMap->GetCurrentNode()->GetData());
     else
     {
         MythGenericTree *node =
@@ -671,12 +660,14 @@
 
     if (m_type == DLG_TREE)
     {
-        item = qVariantValue<ResultItem *>(m_siteMap->GetCurrentNode()->GetData());
+        item =
+            qVariantValue<ResultItem *>(m_siteMap->GetCurrentNode()->GetData());
         site = qVariantValue<RSSSite *>(m_siteMap->GetCurrentNode()->GetData());
     }
     else
     {
-        MythGenericTree *node = GetNodePtrFromButton(m_siteButtonList->GetItemCurrent());
+        MythGenericTree *node =
+            GetNodePtrFromButton(m_siteButtonList->GetItemCurrent());
 
         if (!node)
             return;
@@ -703,7 +694,8 @@
             }
             else
             {
-                QString sFilename = getDownloadFilename(item->GetTitle(), item->GetThumbnail());
+                QString sFilename = getDownloadFilename(item->GetTitle(),
+                                                        item->GetThumbnail());
 
                 bool exists = QFile::exists(sFilename);
                 if (exists)
@@ -726,10 +718,12 @@
     }
     else if (site)
     {
-        ResultItem res = ResultItem(site->GetTitle(), QString(), site->GetDescription(),
-              site->GetURL(), site->GetImage(), QString(), site->GetAuthor(), QDateTime(),
-              0, 0, -1, QString(), QStringList(), QString(), QStringList(), 0, 0, QString(),
-              0, QStringList(), 0, 0, 0);
+        ResultItem res =
+            ResultItem(site->GetTitle(), QString(), site->GetDescription(),
+                       site->GetURL(), site->GetImage(), QString(),
+                       site->GetAuthor(), QDateTime(), 0, 0, -1, QString(),
+                       QStringList(), QString(), QStringList(), 0, 0, QString(),
+                       0, QStringList(), 0, 0, 0);
 
         InfoMap metadataMap;
         res.toMap(metadataMap);
@@ -744,9 +738,7 @@
             m_thumbImage->Reset();
 
         if (m_downloadable)
-        {
             m_downloadable->Reset();
-        }
     }
     else
     {
@@ -763,16 +755,18 @@
                         GetData().toString();
         else
         {
-            MythGenericTree *node = GetNodePtrFromButton(m_siteButtonList->GetItemCurrent());
+            MythGenericTree *node =
+                GetNodePtrFromButton(m_siteButtonList->GetItemCurrent());
 
             if (node)
                 thumb = node->GetData().toString();
         }
 
-        ResultItem res = ResultItem(title, QString(), QString(),
-              QString(), thumb, QString(), QString(), QDateTime(),
-              0, 0, -1, QString(), QStringList(), QString(), QStringList(), 0, 0, QString(),
-              0, QStringList(), 0, 0, 0);
+        ResultItem res =
+            ResultItem(title, QString(), QString(), QString(), thumb, QString(),
+                       QString(), QDateTime(), 0, 0, -1, QString(),
+                       QStringList(), QString(), QStringList(), 0, 0, QString(),
+                       0, QStringList(), 0, 0, 0);
 
         InfoMap metadataMap;
         res.toMap(metadataMap);
@@ -831,15 +825,12 @@
 
     if (treeedit->Create())
     {
-        connect(treeedit, SIGNAL(ItemsChanged()), this,
-                       SLOT(DoTreeRefresh()));
+        connect(treeedit, SIGNAL(ItemsChanged()), this, SLOT(DoTreeRefresh()));
 
         mainStack->AddScreen(treeedit);
     }
     else
-    {
         delete treeedit;
-    }
 }
 
 void NetTree::RunRSSEditor()
@@ -850,15 +841,12 @@
 
     if (rssedit->Create())
     {
-        connect(rssedit, SIGNAL(ItemsChanged()), this,
-                       SLOT(UpdateRSS()));
+        connect(rssedit, SIGNAL(ItemsChanged()), this, SLOT(UpdateRSS()));
 
         mainStack->AddScreen(rssedit);
     }
     else
-    {
         delete rssedit;
-    }
 }
 
 void NetTree::DoTreeRefresh()
@@ -895,8 +883,7 @@
     CreateBusyDialog(title);
 
     RSSManager *rssMan = new RSSManager();
-    connect(rssMan, SIGNAL(finished()), this,
-                   SLOT(DoTreeRefresh()));
+    connect(rssMan, SIGNAL(finished()), this, SLOT(DoTreeRefresh()));
     rssMan->startTimer();
     rssMan->doUpdate();
 }
@@ -915,14 +902,14 @@
 {
     m_rssAutoUpdate = !m_rssAutoUpdate;
     gCoreContext->SaveSetting("mythnetvision.rssBackgroundFetch",
-                         m_rssAutoUpdate);
+                              m_rssAutoUpdate);
 }
 
 void NetTree::ToggleTreeUpdates()
 {
     m_treeAutoUpdate = !m_treeAutoUpdate;
     gCoreContext->SaveSetting("mythnetvision.backgroundFetch",
-                         m_treeAutoUpdate);
+                              m_treeAutoUpdate);
 }
 
 void NetTree::customEvent(QEvent *event)
@@ -959,21 +946,14 @@
         else
         {
             if (!((uint)m_siteButtonList->GetCount() >= pos))
-            {
                 return;
-            }
 
-            MythUIButtonListItem *item =
-                      m_siteButtonList->GetItemAt(pos);
+            MythUIButtonListItem *item = m_siteButtonList->GetItemAt(pos);
 
             if (item && item->GetText() == title)
-            {
                 item->SetImage(file);
-            }
         }
     }
     else
-    {
-      NetBase::customEvent(event);
-    }
+        NetBase::customEvent(event);
 }
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/nettree.h mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/nettree.h
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/nettree.h	2014-06-08 21:23:56.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/nettree.h	2014-06-08 21:25:38.438350980 +0200
@@ -62,7 +62,7 @@
     void UpdateItem(MythUIButtonListItem *item);
 
     void BuildGenericTree(MythGenericTree* dst,
-                    QDomElement& domElem);
+                          QDomElement& domElem);
     void BuildGenericTree(MythGenericTree *dst,
                           QStringList paths,
                           QString dirthumb,
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/rsseditor.cpp mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/rsseditor.cpp
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/rsseditor.cpp	2014-06-08 21:23:56.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/rsseditor.cpp	2014-06-08 21:25:38.438350980 +0200
@@ -35,9 +35,7 @@
 
         QList<QByteArray> exts = QImageReader::supportedImageFormats();
         for (QList<QByteArray>::iterator p = exts.begin(); p != exts.end(); ++p)
-        {
             ret.append(QString("*.").append(*p));
-        }
 
         return ret;
     }
@@ -75,7 +73,8 @@
 bool RSSEditPopup::Create(void)
 {
     // Load the theme for this screen
-    bool foundtheme = LoadWindowFromXML("netvision-ui.xml", "rsseditpopup", this);
+    bool foundtheme =
+        LoadWindowFromXML("netvision-ui.xml", "rsseditpopup", this);
 
     if (!foundtheme)
         return false;
@@ -123,7 +122,7 @@
         }
 
         if (m_site->GetDownload() == 1)
-           m_download->SetCheckState(MythUIStateType::Full);
+            m_download->SetCheckState(MythUIStateType::Full);
     }
 
     BuildFocusList();
@@ -164,8 +163,9 @@
 
         removeFromDB(m_urlText, VIDEO_PODCAST);
 
-        if (insertInDB(new RSSSite(title, filename, VIDEO_PODCAST,
-                desc, link, author, download, MythDate::current())))
+        RSSSite site(title, filename, VIDEO_PODCAST,
+                     desc, link, author, download, MythDate::current());
+        if (insertInDB(&site))
             emit Saving();
         Close();
     }
@@ -193,13 +193,13 @@
 void RSSEditPopup::SlotCheckRedirect(QNetworkReply* reply)
 {
     QVariant possibleRedirectUrl =
-         reply->attribute(QNetworkRequest::RedirectionTargetAttribute);
+        reply->attribute(QNetworkRequest::RedirectionTargetAttribute);
 
     QUrl urlRedirectedTo;
     urlRedirectedTo = redirectUrl(
         possibleRedirectUrl.toUrl(), urlRedirectedTo);
 
-    if(!urlRedirectedTo.isEmpty())
+    if (!urlRedirectedTo.isEmpty())
     {
         m_urlEdit->SetText(urlRedirectedTo.toString());
         m_manager->get(QNetworkRequest(urlRedirectedTo));
@@ -228,7 +228,7 @@
 
     QDomElement root = document.documentElement();
     QDomElement channel = root.firstChildElement ("channel");
-    if (!channel.isNull ())
+    if (!channel.isNull())
     {
         Parse parser;
         if (title.isEmpty())
@@ -242,7 +242,8 @@
         if (author.isEmpty())
             author = channel.firstChildElement("webMaster").text();
 
-        QString thumbnailURL = channel.firstChildElement("image").attribute("url");
+        QString thumbnailURL =
+            channel.firstChildElement("image").attribute("url");
         if (thumbnailURL.isEmpty())
         {
             QDomElement thumbElem = channel.firstChildElement("image");
@@ -252,10 +253,12 @@
         if (thumbnailURL.isEmpty())
         {
             QDomNodeList nodes = channel.elementsByTagNameNS(
-                           "http://www.itunes.com/dtds/podcast-1.0.dtd", "image");
+                           "http://www.itunes.com/dtds/podcast-1.0.dtd",
+                           "image");
             if (nodes.size())
             {
-                thumbnailURL = nodes.at(0).toElement().attributeNode("href").value();
+                thumbnailURL =
+                    nodes.at(0).toElement().attributeNode("href").value();
                 if (thumbnailURL.isEmpty())
                     thumbnailURL = nodes.at(0).toElement().text();
             }
@@ -270,50 +273,16 @@
         QDateTime updated = MythDate::current();
         QString filename("");
 
-        if (file.isEmpty())
+        if (!file.isEmpty())
             filename = file;
+        else if (!thumbnailURL.isEmpty())
+            filename = thumbnailURL;
 
         QString link = m_urlEdit->GetText();
 
-        if (!thumbnailURL.isEmpty() && filename.isEmpty())
-        {
-            QString fileprefix = GetConfDir();
-
-            QDir dir(fileprefix);
-            if (!dir.exists())
-                    dir.mkdir(fileprefix);
-
-            fileprefix += "/MythNetvision";
-
-            dir = QDir(fileprefix);
-            if (!dir.exists())
-                dir.mkdir(fileprefix);
-
-            fileprefix += "/sitecovers";
-
-            dir = QDir(fileprefix);
-            if (!dir.exists())
-                dir.mkdir(fileprefix);
-
-            QFileInfo fi(thumbnailURL);
-            QString rawFilename = fi.fileName();
-
-            filename = QString("%1/%2").arg(fileprefix).arg(rawFilename);
-
-            if (!QFile::exists(filename))
-            {
-                if (!GetMythDownloadManager()->download(thumbnailURL, filename))
-                {
-                    LOG(VB_GENERAL, LOG_ERR,
-                        QString("RSSEditPopup: failed to download thumbnail from: %1")
-                                .arg(thumbnailURL));
-                    filename = "";
-                }
-            }
-        }
-
-        if (insertInDB(new RSSSite(title, filename, VIDEO_PODCAST, description, link,
-                author, download, MythDate::current())))
+        RSSSite site(title, filename, VIDEO_PODCAST, description, link,
+                     author, download, MythDate::current());
+        if (insertInDB(&site))
             emit Saving();
     }
 
@@ -322,24 +291,24 @@
 
 void RSSEditPopup::DoFileBrowser()
 {
-    SelectImagePopup(GetConfDir() + "/MythNetvision" + "/sitecovers", *this, CEID_NEWIMAGE);
+    SelectImagePopup(GetConfDir() + "/MythNetvision" + "/sitecovers", *this,
+                     CEID_NEWIMAGE);
 }
 
 void RSSEditPopup::SelectImagePopup(const QString &prefix,
                             QObject &inst, const QString &returnEvent)
 {
-        MythScreenStack *popupStack =
-                GetMythMainWindow()->GetStack("popup stack");
+    MythScreenStack *popupStack = GetMythMainWindow()->GetStack("popup stack");
 
-        MythUIFileBrowser *fb = new MythUIFileBrowser(popupStack, prefix);
-        fb->SetNameFilter(GetSupportedImageExtensionFilter());
-        if (fb->Create())
-        {
-            fb->SetReturnEvent(&inst, returnEvent);
-            popupStack->AddScreen(fb);
-        }
-        else
-            delete fb;
+    MythUIFileBrowser *fb = new MythUIFileBrowser(popupStack, prefix);
+    fb->SetNameFilter(GetSupportedImageExtensionFilter());
+    if (fb->Create())
+    {
+        fb->SetReturnEvent(&inst, returnEvent);
+        popupStack->AddScreen(fb);
+    }
+    else
+        delete fb;
 }
 
 void RSSEditPopup::customEvent(QEvent *levent)
@@ -449,7 +418,8 @@
         return true;
 
     QStringList actions;
-    bool handled = GetMythMainWindow()->TranslateKeyPress("Internet Video", event, actions);
+    bool handled = GetMythMainWindow()->TranslateKeyPress("Internet Video",
+                                                          event, actions);
 
     for (int i = 0; i < actions.size() && !handled; i++)
     {
@@ -497,7 +467,8 @@
 
 void RSSEditor::SlotItemChanged()
 {
-    RSSSite *site = qVariantValue<RSSSite *>(m_sites->GetItemCurrent()->GetData());
+    RSSSite *site =
+        qVariantValue<RSSSite *>(m_sites->GetItemCurrent()->GetData());
 
     if (site)
     {
@@ -509,7 +480,7 @@
 
             if (!thumb.isEmpty())
             {
-                m_image->SetFilename(site->GetImage());
+                m_image->SetFilename(thumb);
                 m_image->Load();
             }
         }
@@ -528,12 +499,14 @@
 {
     QMutexLocker locker(&m_lock);
 
-    QString message = tr("Are you sure you want to unsubscribe from this feed?");
+    QString message =
+        tr("Are you sure you want to unsubscribe from this feed?");
 
-    MythScreenStack *m_popupStack = GetMythMainWindow()->GetStack("popup stack");
+    MythScreenStack *m_popupStack =
+        GetMythMainWindow()->GetStack("popup stack");
 
     MythConfirmationDialog *confirmdialog =
-            new MythConfirmationDialog(m_popupStack,message);
+        new MythConfirmationDialog(m_popupStack,message);
 
     if (confirmdialog->Create())
     {
@@ -552,23 +525,22 @@
 
     MythScreenStack *mainStack = GetMythMainWindow()->GetMainStack();
 
-    RSSSite *site = qVariantValue<RSSSite *>(m_sites->GetItemCurrent()->GetData());
+    RSSSite *site =
+        qVariantValue<RSSSite *>(m_sites->GetItemCurrent()->GetData());
 
     if (site)
     {
-        RSSEditPopup *rsseditpopup = new RSSEditPopup(site->GetURL(), true, mainStack, "rsseditpopup");
+        RSSEditPopup *rsseditpopup =
+            new RSSEditPopup(site->GetURL(), true, mainStack, "rsseditpopup");
 
         if (rsseditpopup->Create())
         {
-            connect(rsseditpopup, SIGNAL(Saving()), this,
-                           SLOT(ListChanged()));
+            connect(rsseditpopup, SIGNAL(Saving()), this, SLOT(ListChanged()));
 
             mainStack->AddScreen(rsseditpopup);
         }
         else
-        {
             delete rsseditpopup;
-        }
     }
 }
 
@@ -578,19 +550,17 @@
 
     MythScreenStack *mainStack = GetMythMainWindow()->GetMainStack();
 
-    RSSEditPopup *rsseditpopup = new RSSEditPopup("", false, mainStack, "rsseditpopup");
+    RSSEditPopup *rsseditpopup =
+        new RSSEditPopup("", false, mainStack, "rsseditpopup");
 
     if (rsseditpopup->Create())
     {
-        connect(rsseditpopup, SIGNAL(Saving()), this,
-                       SLOT(ListChanged()));
+        connect(rsseditpopup, SIGNAL(Saving()), this, SLOT(ListChanged()));
 
         mainStack->AddScreen(rsseditpopup);
     }
     else
-    {
         delete rsseditpopup;
-    }
 }
 
 void RSSEditor::DoDeleteSite(bool remove)
@@ -600,7 +570,8 @@
     if (!remove)
         return;
 
-    RSSSite *site = qVariantValue<RSSSite *>(m_sites->GetItemCurrent()->GetData());
+    RSSSite *site =
+        qVariantValue<RSSSite *>(m_sites->GetItemCurrent()->GetData());
 
     if (removeFromDB(site))
         ListChanged();
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/rsseditor.h mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/rsseditor.h
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/rsseditor.h	2014-06-08 21:23:56.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/rsseditor.h	2014-06-08 21:25:38.438350980 +0200
@@ -76,8 +76,7 @@
     Q_OBJECT
 
   public:
-    RSSEditor(MythScreenStack *parent,
-              const QString &name = "RSSEditor");
+    RSSEditor(MythScreenStack *parent, const QString &name = "RSSEditor");
    ~RSSEditor();
 
     bool Create(void);
diff -Naur mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/treeeditor.h mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/treeeditor.h
--- mythtv-0.27-20140519-gf4825ca-old/mythplugins/mythnetvision/mythnetvision/treeeditor.h	2014-06-08 21:23:56.000000000 +0200
+++ mythtv-0.27-20140519-gf4825ca-new/mythplugins/mythnetvision/mythnetvision/treeeditor.h	2014-06-08 21:25:38.438350980 +0200
@@ -11,8 +11,7 @@
     Q_OBJECT
 
   public:
-    TreeEditor(MythScreenStack *parent,
-               const QString &name = "TreeEditor");
+    TreeEditor(MythScreenStack *parent, const QString &name = "TreeEditor");
 
   protected:
     bool InsertInDB(GrabberScript *script);
