From fabc4642f1cf855d08a3ba23ed8668f56661bed2 Mon Sep 17 00:00:00 2001
From: Jonatan Lindblad <jlindblad@mythtv.org>
Date: Sun, 8 Jun 2014 01:10:43 +0200
Subject: [PATCH 16/23] MythNetvision: Fix thumbnail loading in the list view

Fixes #11800
---
 mythplugins/mythnetvision/mythnetvision/nettree.cpp |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mythplugins/mythnetvision/mythnetvision/nettree.cpp b/mythplugins/mythnetvision/mythnetvision/nettree.cpp
index 58a58b2..33c9580 100644
--- a/mythplugins/mythnetvision/mythnetvision/nettree.cpp
+++ b/mythplugins/mythnetvision/mythnetvision/nettree.cpp
@@ -703,6 +703,12 @@ void NetTree::SlotItemChanged()
                     m_thumbImage->SetFilename(sFilename);
                     m_thumbImage->Load();
                 }
+                else if (item->GetThumbnail().startsWith("http"))
+                {
+                    m_imageDownload->addThumb(item->GetTitle(),
+                                              item->GetThumbnail(),
+                                              qVariantFromValue<uint>(0));
+                }
             }
         }
         else if (m_thumbImage)
-- 
1.7.10.2

