diff -Naur rtl8821CU-93b04bfcd293fdd7d98d5a7c964ae9416a40159c-old/Makefile rtl8821CU-93b04bfcd293fdd7d98d5a7c964ae9416a40159c-new/Makefile
--- rtl8821CU-93b04bfcd293fdd7d98d5a7c964ae9416a40159c-old/Makefile	2020-07-28 16:41:06.119999993 +0200
+++ rtl8821CU-93b04bfcd293fdd7d98d5a7c964ae9416a40159c-new/Makefile	2020-07-28 16:40:45.073333331 +0200
@@ -81,10 +81,10 @@
 CONFIG_RTW_REPEATER_SON = n
 CONFIG_RTW_WIFI_HAL = n
 ########################## Debug ###########################
-CONFIG_RTW_DEBUG = y
+CONFIG_RTW_DEBUG = n
 # default log level is _DRV_INFO_ = 4,
 # please refer to "How_to_set_driver_debug_log_level.doc" to set the available level.
-CONFIG_RTW_LOG_LEVEL = 3
+CONFIG_RTW_LOG_LEVEL = 2
 ######################## Wake On Lan ##########################
 CONFIG_WOWLAN = n
 CONFIG_WAKEUP_TYPE = 0x7 #bit2: deauth, bit1: unicast, bit0: magic pkt.
