From 657b7cfde9958e273febdeaeac579902e407f577 Mon Sep 17 00:00:00 2001
From: Rin Cat <me@rin.cat>
Date: Sun, 4 Apr 2021 14:21:45 -0400
Subject: [PATCH] Linux 5.12 compat: GRO_DROP

---
 os_dep/linux/recv_linux.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/os_dep/linux/recv_linux.c b/os_dep/linux/recv_linux.c
index 11c0d47..ece18f5 100644
--- a/os_dep/linux/recv_linux.c
+++ b/os_dep/linux/recv_linux.c
@@ -355,7 +355,11 @@ static int napi_recv(_adapter *padapter, int budget)
 
 #ifdef CONFIG_RTW_GRO
 		if (pregistrypriv->en_gro) {
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0))
+			if (rtw_napi_gro_receive(&padapter->napi, pskb) != GRO_MERGED_FREE)
+#else
 			if (rtw_napi_gro_receive(&padapter->napi, pskb) != GRO_DROP)
+#endif
 				rx_ok = _TRUE;
 			goto next;
 		}
diff -Naur rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-old/Makefile rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-new/Makefile
--- rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-old/Makefile	2021-03-20 17:21:54.623329445 +0100
+++ rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-new/Makefile	2021-03-20 17:11:49.866662782 +0100
@@ -1213,7 +1213,7 @@
 ifeq ($(CONFIG_PLATFORM_I386_PC), y)
 EXTRA_CFLAGS += -DCONFIG_LITTLE_ENDIAN
 EXTRA_CFLAGS += -DCONFIG_IOCTL_CFG80211 -DRTW_USE_CFG80211_STA_EVENT
-EXTRA_CFLAGS += -mhard-float
+# EXTRA_CFLAGS += -mhard-float
 SUBARCH := $(shell uname -m | sed -e s/i.86/i386/)
 ARCH ?= $(SUBARCH)
 CROSS_COMPILE ?=
