diff -Naur rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-old/Makefile rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-new/Makefile
--- rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-old/Makefile	2021-03-20 17:21:54.623329445 +0100
+++ rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-new/Makefile	2021-03-20 17:11:49.866662782 +0100
@@ -1213,7 +1213,7 @@
 ifeq ($(CONFIG_PLATFORM_I386_PC), y)
 EXTRA_CFLAGS += -DCONFIG_LITTLE_ENDIAN
 EXTRA_CFLAGS += -DCONFIG_IOCTL_CFG80211 -DRTW_USE_CFG80211_STA_EVENT
-EXTRA_CFLAGS += -mhard-float
+# EXTRA_CFLAGS += -mhard-float
 SUBARCH := $(shell uname -m | sed -e s/i.86/i386/)
 ARCH ?= $(SUBARCH)
 CROSS_COMPILE ?=
diff -Naur rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-old/os_dep/linux/recv_linux.c rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-new/os_dep/linux/recv_linux.c
--- rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-old/os_dep/linux/recv_linux.c	2021-01-13 20:39:20.000000000 +0100
+++ rtl8821CU-bd2ef2d6990d0e42d03042f1dd1e9eba82f1ccb0-new/os_dep/linux/recv_linux.c	2021-03-20 17:19:17.883329438 +0100
@@ -355,8 +355,12 @@
 
 #ifdef CONFIG_RTW_GRO
 		if (pregistrypriv->en_gro) {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 12, 0)
 			if (rtw_napi_gro_receive(&padapter->napi, pskb) != GRO_DROP)
 				rx_ok = _TRUE;
+#else
+			rx_ok = _TRUE;
+#endif
 			goto next;
 		}
 #endif /* CONFIG_RTW_GRO */
