diff -Naur lame-3.99.5-old/configure.in lame-3.99.5-new/configure.in
--- lame-3.99.5-old/configure.in	2012-07-16 17:44:16.000000000 -0700
+++ lame-3.99.5-new/configure.in	2012-07-16 17:44:56.000000000 -0700
@@ -390,7 +390,7 @@
 
 dnl configure use of features
 
-AM_PATH_GTK(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
+AM_PATH_GTK_2_0(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
 
 dnl ElectricFence malloc debugging
 AC_MSG_CHECKING(use of ElectricFence malloc debugging)
