diff -Naur lame-3.99.5-old/libmp3lame/vector/Makefile.am lame-3.99.5-new/libmp3lame/vector/Makefile.am
--- lame-3.99.5-old/libmp3lame/vector/Makefile.am	2007-01-09 02:15:53.000000000 -0800
+++ lame-3.99.5-new/libmp3lame/vector/Makefile.am	2015-06-25 15:34:55.438994241 -0700
@@ -20,6 +20,7 @@
 
 if WITH_XMM
 liblamevectorroutines_la_SOURCES = $(xmm_sources)
+liblamevectorroutines_la_CFLAGS = -msse
 endif
 
 noinst_HEADERS = lame_intrin.h
