diff -Naur lame-3.99.5-old/configure.in lame-3.99.5-new/configure.in
--- lame-3.99.5-old/configure.in	2012-02-28 10:50:27.000000000 -0800
+++ lame-3.99.5-new/configure.in	2012-07-17 04:14:12.000000000 -0700
@@ -77,9 +77,6 @@
 	AC_MSG_RESULT(${GCC_version})
 fi
 
-dnl more automake stuff
-AM_C_PROTOTYPES
-
 AC_CHECK_HEADER(dmalloc.h)
 if test "${ac_cv_header_dmalloc_h}" = "yes"; then
 	AM_WITH_DMALLOC
diff -Naur lame-3.99.5-old/doc/html/Makefile.am lame-3.99.5-new/doc/html/Makefile.am
--- lame-3.99.5-old/doc/html/Makefile.am	2010-09-30 13:58:40.000000000 -0700
+++ lame-3.99.5-new/doc/html/Makefile.am	2012-07-17 04:14:53.000000000 -0700
@@ -1,6 +1,6 @@
 ## $Id: Makefile.am,v 1.7 2010/09/30 20:58:40 jaz001 Exp $
 
-AUTOMAKE_OPTIONS = foreign ansi2knr
+AUTOMAKE_OPTIONS = foreign
 
 docdir = $(datadir)/doc
 pkgdocdir = $(docdir)/$(PACKAGE)
diff -Naur lame-3.99.5-old/doc/man/Makefile.am lame-3.99.5-new/doc/man/Makefile.am
--- lame-3.99.5-old/doc/man/Makefile.am	2000-10-22 04:39:44.000000000 -0700
+++ lame-3.99.5-new/doc/man/Makefile.am	2012-07-17 04:14:56.000000000 -0700
@@ -1,6 +1,6 @@
 ## $Id: Makefile.am,v 1.1 2000/10/22 11:39:44 aleidinger Exp $
 
-AUTOMAKE_OPTIONS = foreign ansi2knr
+AUTOMAKE_OPTIONS = foreign
 
 man_MANS = lame.1
 EXTRA_DIST = ${man_MANS}
diff -Naur lame-3.99.5-old/libmp3lame/i386/Makefile.am lame-3.99.5-new/libmp3lame/i386/Makefile.am
--- lame-3.99.5-old/libmp3lame/i386/Makefile.am	2011-04-04 02:42:34.000000000 -0700
+++ lame-3.99.5-new/libmp3lame/i386/Makefile.am	2012-07-17 04:15:09.000000000 -0700
@@ -1,6 +1,6 @@
 ## $Id: Makefile.am,v 1.26 2011/04/04 09:42:34 aleidinger Exp $
 
-AUTOMAKE_OPTIONS = foreign $(top_srcdir)/ansi2knr
+AUTOMAKE_OPTIONS = foreign
 
 DEFS = @DEFS@ @CONFIG_DEFS@
 
