diff -Naur linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
--- linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2020-04-21 20:43:06.070000000 +0200
+++ linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2020-04-21 20:53:37.476666630 +0200
@@ -4,6 +4,7 @@
 /dts-v1/;
 
 #include "sun50i-h6.dtsi"
+#include "sun50i-h6-cpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
@@ -70,6 +71,10 @@
 	};
 };
 
+&cpu0 {
+	cpu-supply = <&reg_dcdca>;
+};
+
 &de {
 	status = "okay";
 };
@@ -231,7 +236,8 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
+				regulator-ramp-delay = <2500>;
 				regulator-name = "vdd-cpu";
 			};
 
diff -Naur linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts
--- linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts	2020-04-21 20:43:06.070000000 +0200
+++ linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts	2020-04-21 20:47:24.450000001 +0200
@@ -6,6 +6,7 @@
 /dts-v1/;
 
 #include "sun50i-h6.dtsi"
+#include "sun50i-h6-cpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
diff -Naur linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts
--- linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2020-04-17 16:13:49.000000000 +0200
+++ linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2020-04-21 20:55:17.429999990 +0200
@@ -4,6 +4,7 @@
 /dts-v1/;
 
 #include "sun50i-h6.dtsi"
+#include "sun50i-h6-cpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
@@ -241,6 +242,7 @@
 				regulator-always-on;
 				regulator-min-microvolt = <800000>;
 				regulator-max-microvolt = <1160000>;
+				regulator-ramp-delay = <2500>;
 				regulator-name = "vdd-cpu";
 			};
 
diff -Naur linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts
--- linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts	2020-04-17 16:13:49.000000000 +0200
+++ linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts	2020-04-21 21:00:26.913333329 +0200
@@ -4,6 +4,7 @@
 /dts-v1/;
 
 #include "sun50i-h6.dtsi"
+#include "sun50i-h6-cpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
@@ -90,6 +91,10 @@
 	};
 };
 
+&cpu0 {
+	cpu-supply = <&reg_dcdca>;
+};
+
 &de {
 	status = "okay";
 };
@@ -230,7 +235,8 @@
 			reg_dcdca: dcdca {
 				regulator-always-on;
 				regulator-min-microvolt = <810000>;
-				regulator-max-microvolt = <1080000>;
+				regulator-max-microvolt = <1160000>;
+				regulator-ramp-delay = <2500>;
 				regulator-name = "vdd-cpu";
 			};
 
diff -Naur linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-qplus.dts linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-qplus.dts
--- linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-qplus.dts	2020-04-21 20:43:06.070000000 +0200
+++ linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-qplus.dts	2020-04-21 20:48:02.776666660 +0200
@@ -6,6 +6,7 @@
 /dts-v1/;
 
 #include "sun50i-h6.dtsi"
+#include "sun50i-h6-cpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
diff -Naur linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts
--- linux-5.6.5-old/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts	2020-04-21 20:43:06.070000000 +0200
+++ linux-5.6.5-new/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6.dts	2020-04-28 17:12:36.840000001 +0200
@@ -4,6 +4,7 @@
 /dts-v1/;
 
 #include "sun50i-h6.dtsi"
+#include "sun50i-h6-cpu-opp.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 
@@ -38,6 +39,14 @@
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
 	};
+
+	reg_vcccpu: vcccpu {
+		compatible = "regulator-fixed";
+		regulator-name = "vcccpu";
+		regulator-min-microvolt = <1160000>;
+		regulator-max-microvolt = <1160000>;
+		regulator-always-on;
+	};
 };
 
 &de {
