diff -Naur u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-old/arch/arm/dts/sun50i-h6-beelink-gs1.dts u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-new/arch/arm/dts/sun50i-h6-beelink-gs1.dts
--- u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-old/arch/arm/dts/sun50i-h6-beelink-gs1.dts	2019-07-16 13:43:15.000000000 +0200
+++ u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-new/arch/arm/dts/sun50i-h6-beelink-gs1.dts	2019-09-05 22:17:12.670000007 +0200
@@ -15,6 +15,7 @@
 
 	aliases {
 		serial0 = &uart0;
+		ethernet0 = &emac;
 	};
 
 	chosen {
diff -Naur u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-old/arch/arm/dts/sun50i-h6-eachlink-h6-mini.dts u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-new/arch/arm/dts/sun50i-h6-eachlink-h6-mini.dts
--- u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-old/arch/arm/dts/sun50i-h6-eachlink-h6-mini.dts	2019-09-05 22:20:23.826666674 +0200
+++ u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-new/arch/arm/dts/sun50i-h6-eachlink-h6-mini.dts	2019-09-05 22:17:39.343333340 +0200
@@ -17,6 +17,7 @@
 
 	aliases {
 		serial0 = &uart0;
+		ethernet0 = &emac;
 	};
 
 	chosen {
