diff -Naur u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-old/arch/arm/dts/sun50i-h6-beelink-gs1.dts u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-new/arch/arm/dts/sun50i-h6-beelink-gs1.dts
--- u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-old/arch/arm/dts/sun50i-h6-beelink-gs1.dts	2019-07-16 13:43:15.000000000 +0200
+++ u-boot-sunxi-f96238e253661b43168352010a2628957940bd4b-new/arch/arm/dts/sun50i-h6-beelink-gs1.dts	2019-09-04 10:41:38.599999966 +0200
@@ -15,12 +15,25 @@
 
 	aliases {
 		serial0 = &uart0;
+		ethernet0 = &emac;
 	};
 
 	chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
+	connector {
+		compatible = "hdmi-connector";
+		type = "a";
+		ddc-supply = <&reg_ddc>;
+
+		port {
+			hdmi_con_in: endpoint {
+				remote-endpoint = <&hdmi_out_con>;
+			};
+		};
+	};
+
 	leds {
 		compatible = "gpio-leds";
 
@@ -31,6 +44,15 @@
 		};
 	};
 
+	reg_ddc: ddc-io {
+		compatible = "regulator-fixed";
+		regulator-name = "ddc-io";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		gpio = <&pio 7 2 GPIO_ACTIVE_HIGH>; /* PH2 */
+	};
+
 	reg_vcc5v: vcc5v {
 		/* board wide 5V supply directly from the DC jack */
 		compatible = "regulator-fixed";
@@ -41,6 +63,17 @@
 	};
 };
 
+&hdmi {
+	status = "okay";
+};
+
+&hdmi_out {
+	hdmi_out_con: endpoint {
+		remote-endpoint = <&hdmi_con_in>;
+	};
+
+};
+
 &mmc0 {
 	vmmc-supply = <&reg_cldo1>;
 	cd-gpios = <&pio 5 6 GPIO_ACTIVE_LOW>;
