diff -Naur xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-old/Makefile xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-new/Makefile
--- xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-old/Makefile	2020-05-02 19:17:06.000000000 +0200
+++ xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-new/Makefile	2021-06-18 11:50:14.496644304 +0200
@@ -1,8 +1,8 @@
 # SPDX-License-Identifier: GPL-2.0-only
 
 # # Standalone Makefile - uncomment for out-of-tree compilation
-# CONFIG_WLAN_VENDOR_XRADIO := m
-# ccflags-y += -DCONFIG_XRADIO_USE_EXTENSIONS
+CONFIG_WLAN_VENDOR_XRADIO := m
+ccflags-y += -DCONFIG_XRADIO_USE_EXTENSIONS
 # ccflags-y += -DCONFIG_XRADIO_WAPI_SUPPORT
 
 # Kernel part
@@ -54,7 +54,7 @@
 # # Not functional at this point
 # #ccflags-y += -DROAM_OFFLOAD
 
-# #ccflags-y += -DDEBUG
+# ccflags-y += -DDEBUG
 
-ldflags-y += --strip-debug
+ldflags-y += --strip-debug
 
diff -Naur xradio-6bf0e2e21c80456e2a3d4ad1267caecde7165871-old/sdio.c xradio-6bf0e2e21c80456e2a3d4ad1267caecde7165871-new/sdio.c
--- xradio-6bf0e2e21c80456e2a3d4ad1267caecde7165871-old/sdio.c	2020-04-20 12:51:11.000000000 +0200
+++ xradio-6bf0e2e21c80456e2a3d4ad1267caecde7165871-new/sdio.c	2020-07-20 16:27:42.436666669 +0200
@@ -15,7 +15,7 @@
 #include <linux/mmc/sdio_func.h>
 #include <linux/mmc/card.h>
 #include <linux/mmc/sdio.h>
-#include <asm/mach-types.h>
+//#include <asm/mach-types.h>
 #include <linux/of.h>
 #include <linux/of_irq.h>
