diff -Naur xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-old/main.c xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-new/main.c
--- xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-old/main.c	2020-05-02 19:17:06.000000000 +0200
+++ xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-new/main.c	2021-06-30 13:50:39.963331816 +0200
@@ -498,6 +498,7 @@
 	int err = -ENOMEM;
 	u16 ctrl_reg;
 	int if_id;
+	int ret;
 	u8 b;		/* MRK 5.5a */
 	struct ieee80211_hw *dev;
 	struct xradio_common *hw_priv;
@@ -516,14 +517,17 @@
 	sdio_set_drvdata(func, hw_priv);
 
 	// fill in mac addresses
+	ret = 1;
 	if (hw_priv->pdev->of_node) {
-		addr = of_get_mac_address(hw_priv->pdev->of_node);
+		ret = of_get_mac_address(hw_priv->pdev->of_node, randomaddr);
 	}
-	if (!addr) {
+	if (ret) {
 		dev_warn(hw_priv->pdev, "no mac address provided, using random\n");
 		eth_random_addr(randomaddr);
-		addr = randomaddr;
+	} else {
+		dev_warn(hw_priv->pdev, "using mac address from device-tree\n");
 	}
+	addr = randomaddr;
 	for (b = 0; b < XRWL_MAX_VIFS; b++) {				/* MRK 5.5a */
 		memcpy(hw_priv->addresses[b].addr, addr, ETH_ALEN);
 		hw_priv->addresses[b].addr[5] += b;
