diff -Naur xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-old/main.c xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-new/main.c
--- xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-old/main.c	2020-05-02 19:17:06.000000000 +0200
+++ xradio-354e8c32e7948d46a63796d0ca266b1f702999b0-new/main.c	2021-06-18 11:57:06.633310994 +0200
@@ -517,7 +517,11 @@
 
 	// fill in mac addresses
 	if (hw_priv->pdev->of_node) {
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 13, 0))
+		addr = of_get_mac_address(hw_priv->pdev->of_node, 0);
+#else
 		addr = of_get_mac_address(hw_priv->pdev->of_node);
+#endif
 	}
 	if (!addr) {
 		dev_warn(hw_priv->pdev, "no mac address provided, using random\n");
