diff --speed-large-files --no-dereference --minimal -Naur xradio-3a1f77fb2db248b7d18d93b67b16e0d6c91db184/bh.c xradio-3a1f77fb2db248b7d18d93b67b16e0d6c91db184/bh.c
--- xradio-3a1f77fb2db248b7d18d93b67b16e0d6c91db184/bh.c	2023-12-25 02:12:36.000000000 +0100
+++ xradio-3a1f77fb2db248b7d18d93b67b16e0d6c91db184/bh.c	2024-03-22 13:38:52.064945539 +0100
@@ -776,7 +776,7 @@
 		} else if (!status) {
 			/* check if there is data waiting but we missed the interrupt*/
 			if (xradio_bh_rx_availlen(hw_priv) > 0) {
-				dev_warn(hw_priv->pdev, "missed interrupt\n");
+				//dev_warn(hw_priv->pdev, "missed interrupt\n");
 				if(xradio_bh_exchange(hw_priv) < 0){
 					break;
 				}
