diff -Naur xmacro-pre0.3-20000911-old/chartbl.h xmacro-pre0.3-20000911-new/chartbl.h
--- xmacro-pre0.3-20000911-old/chartbl.h	2000-09-05 13:21:23.000000000 -0700
+++ xmacro-pre0.3-20000911-new/chartbl.h	2012-03-24 07:14:39.000000000 -0700
@@ -22,7 +22,7 @@
 
 /* Version 0.1 (20000817) probably still incomplete... */
 
-char *chartbl_lat1[] =
+const char *chartbl_lat1[] =
 {
 	"",		//   0  0
 	"",		//   1  1
@@ -282,7 +282,7 @@
 	"ydiaeresis",	// 255 FF
 };
 
-char *chartbl_lat2[] =
+const char *chartbl_lat2[] =
 {
 	"",		//   0  0
 	"",		//   1  1
@@ -542,4 +542,4 @@
 	"abovedot",	// 255 FF
 };
 
-char **chartbl[]={ chartbl_lat1, chartbl_lat2 };
+const char **chartbl[]={ chartbl_lat1, chartbl_lat2 };
diff -Naur xmacro-pre0.3-20000911-old/xmacroplay.cpp xmacro-pre0.3-20000911-new/xmacroplay.cpp
--- xmacro-pre0.3-20000911-old/xmacroplay.cpp	2000-09-11 11:25:11.000000000 -0700
+++ xmacro-pre0.3-20000911-new/xmacroplay.cpp	2012-03-24 07:14:57.000000000 -0700
@@ -34,11 +34,14 @@
 /***************************************************************************** 
  * Includes
  ****************************************************************************/
-#include <stdio.h>		
+#include <iostream>
+#include <cstdio>
+#include <cstring>
+using namespace std;
+
 #include <stdlib.h>
 #include <unistd.h>
-#include <ctype.h>
-#include <X11/Xlibint.h>
+
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <X11/cursorfont.h>
@@ -47,19 +50,10 @@
 #include <X11/extensions/XTest.h>
 
 #include "chartbl.h"
-/***************************************************************************** 
- * What iostream do we have?
- ****************************************************************************/
-#ifdef HAVE_IOSTREAM
-#include <iostream>
-#include <iomanip>
-#else
-#include <iostream.h>
-#include <iomanip.h>
-#endif
 
 #define PROG "xmacroplay"
 
+
 /***************************************************************************** 
  * The delay in milliseconds when sending events to the remote display
  ****************************************************************************/
diff -Naur xmacro-pre0.3-20000911-old/xmacrorec2.cpp xmacro-pre0.3-20000911-new/xmacrorec2.cpp
--- xmacro-pre0.3-20000911-old/xmacrorec2.cpp	2000-09-05 16:46:15.000000000 -0700
+++ xmacro-pre0.3-20000911-new/xmacrorec2.cpp	2012-03-24 07:14:39.000000000 -0700
@@ -38,9 +38,13 @@
 /***************************************************************************** 
  * Includes
  ****************************************************************************/
-#include <stdio.h>		
+#include <iostream>
+#include <cstdio>
+#include <cstring>
+using namespace std;
+
 #include <stdlib.h>
-#include <X11/Xlibint.h>
+
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <X11/cursorfont.h>
@@ -48,17 +52,6 @@
 #include <X11/keysym.h>
 #include <X11/extensions/record.h>
 
-/***************************************************************************** 
- * What iostream do we have?
- ****************************************************************************/
-#ifdef HAVE_IOSTREAM
-#include <iostream>
-#include <iomanip>
-#else
-#include <iostream.h>
-#include <iomanip.h>
-#endif
-
 #define PROG "xmacrorec2"
 
 /***************************************************************************** 
diff -Naur xmacro-pre0.3-20000911-old/xmacrorec.cpp xmacro-pre0.3-20000911-new/xmacrorec.cpp
--- xmacro-pre0.3-20000911-old/xmacrorec.cpp	2000-09-05 13:21:23.000000000 -0700
+++ xmacro-pre0.3-20000911-new/xmacrorec.cpp	2012-03-24 07:14:39.000000000 -0700
@@ -36,9 +36,13 @@
 /***************************************************************************** 
  * Includes
  ****************************************************************************/
-#include <stdio.h>		
+#include <iostream>
+#include <cstdio>
+#include <cstring>
+using namespace std;
+
 #include <stdlib.h>
-#include <X11/Xlibint.h>
+
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <X11/cursorfont.h>
@@ -46,17 +50,6 @@
 #include <X11/keysym.h>
 #include <X11/extensions/XTest.h>
 
-/***************************************************************************** 
- * What iostream do we have?
- ****************************************************************************/
-#ifdef HAVE_IOSTREAM
-#include <iostream>
-#include <iomanip>
-#else
-#include <iostream.h>
-#include <iomanip.h>
-#endif
-
 #define PROG "xmacrorec"
 
 /***************************************************************************** 
