From a33da99cd385e66c1401842c4ea03d878db46915 Mon Sep 17 00:00:00 2001
From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Thu, 29 Aug 2019 17:11:08 +0200
Subject: [PATCH 08/13] h264: set frame_num in slice_params

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 src/h264.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/h264.c b/src/h264.c
index bd3b6ab..ecbb3d9 100644
--- a/src/h264.c
+++ b/src/h264.c
@@ -336,6 +336,7 @@ static void h264_va_slice_to_v4l2(struct request_data *driver_data,
 	slice->header_bit_size = VASlice->slice_data_bit_offset;
 	slice->first_mb_in_slice = VASlice->first_mb_in_slice;
 	slice->slice_type = VASlice->slice_type;
+	slice->frame_num = VAPicture->frame_num;
 	slice->cabac_init_idc = VASlice->cabac_init_idc;
 	slice->slice_qp_delta = VASlice->slice_qp_delta;
 	slice->disable_deblocking_filter_idc =
-- 
2.30.2

