diff -Naur BeautifulSoup-3.2.0-old/setup.py BeautifulSoup-3.2.0-new/setup.py
--- BeautifulSoup-3.2.0-old/setup.py	2010-11-21 05:15:26.000000000 -0800
+++ BeautifulSoup-3.2.0-new/setup.py	2012-02-10 10:50:01.000000000 -0800
@@ -1,4 +1,4 @@
-from distutils.core import setup
+from setuptools import setup
 import unittest
 import warnings
 warnings.filterwarnings("ignore", "Unknown distribution option")
