diff -Naur a52dec-0.7.4-old/configure.in a52dec-0.7.4-new/configure.in
--- a52dec-0.7.4-old/configure.in	2002-07-27 20:50:20.000000000 -0700
+++ a52dec-0.7.4-new/configure.in	2012-12-31 05:55:17.000000000 -0800
@@ -6,7 +6,7 @@
 AC_CONFIG_FILES([Makefile autotools/Makefile include/Makefile test/Makefile
     doc/Makefile src/Makefile liba52/Makefile libao/Makefile vc++/Makefile])
 AM_INIT_AUTOMAKE([a52dec],[0.7.4])
-AM_CONFIG_HEADER(include/config.h)
+AC_CONFIG_HEADERS(include/config.h)
 AM_MAINTAINER_MODE
 AC_CANONICAL_HOST
 
