diff --git a/drivers/staging/media/rkvdec/rkvdec.c b/drivers/staging/media/rkvdec/rkvdec.c
index 1c0e35f9d84f..f69e2dbf28a5 100644
--- a/drivers/staging/media/rkvdec/rkvdec.c
+++ b/drivers/staging/media/rkvdec/rkvdec.c
@@ -1343,11 +1343,9 @@ static int rkvdec_probe(struct platform_device *pdev)
 
 	rkvdec->capabilities = variant->capabilities;
 
-	/*
-	 * Bump ACLK to max. possible freq. (500 MHz) to improve performance
-	 * When 4k video playback.
-	 */
-	clk_set_rate(rkvdec->clocks[0].clk, 500 * 1000 * 1000);
+	clk_set_rate(rkvdec->clocks[0].clk, RKVDEC_AXI_CLK_RATE);
+	clk_set_rate(rkvdec->clocks[2].clk, RKVDEC_CABAC_CLK_RATE);
+	clk_set_rate(rkvdec->clocks[3].clk, RKVDEC_CORE_CLK_RATE);
 
 	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 	rkvdec->regs = devm_ioremap_resource(&pdev->dev, res);

diff --git a/drivers/staging/media/rkvdec/rkvdec.h b/drivers/staging/media/rkvdec/rkvdec.h
index cc505bc4a042..a8f7236fe880 100644
--- a/drivers/staging/media/rkvdec/rkvdec.h
+++ b/drivers/staging/media/rkvdec/rkvdec.h
@@ -33,6 +33,10 @@
 #define RKVDEC_CAPABILITY_HEVC	BIT(1)
 #define RKVDEC_CAPABILITY_VP9	BIT(2)
 
+#define RKVDEC_AXI_CLK_RATE	400 * 1000 * 1000
+#define RKVDEC_CABAC_CLK_RATE	200 * 1000 * 1000
+#define RKVDEC_CORE_CLK_RATE	200 * 1000 * 1000
+
 struct rkvdec_ctx;
 
 struct rkvdec_ctrl_desc {
