diff --speed-large-files --no-dereference --minimal -Naur pango-1.50.14/meson.build pango-1.50.14/meson.build
--- pango-1.50.14/meson.build	2023-03-02 14:19:55.000000000 +0100
+++ pango-1.50.14/meson.build	2023-04-29 17:55:54.986625336 +0200
@@ -89,7 +89,6 @@
     '-Wuninitialized',
     '-Wunused',
     '-Werror=address',
-    '-Werror=array-bounds',
     '-Werror=empty-body',
     '-Werror=implicit',
     '-Werror=implicit-fallthrough',
@@ -132,7 +131,6 @@
     '-Werror=sequence-point',
     '-Werror=return-type',
     '-Werror=trigraphs',
-    '-Werror=array-bounds',
     '-Werror=write-strings',
     '-Werror=address',
     '-Werror=int-to-pointer-cast',
