diff -Naur rxvt-2.6.4-old/autoconf/Make.common.in rxvt-2.6.4-new/autoconf/Make.common.in
--- rxvt-2.6.4-old/autoconf/Make.common.in	1999-03-27 22:15:42.000000000 -0800
+++ rxvt-2.6.4-new/autoconf/Make.common.in	2009-07-27 22:09:40.000000000 -0700
@@ -44,7 +44,7 @@
 CMP = @CMP@
 TBL = @TBL@
 INSTALL = @INSTALL@
-INSTALL_PROGRAM = @INSTALL@ -s -m 755
+INSTALL_PROGRAM = @INSTALL@ -m 755
 INSTALL_DATA = @INSTALL@ -m 644
 
 # Flags & libs
diff -Naur rxvt-2.6.4-old/src/Makefile.in rxvt-2.6.4-new/src/Makefile.in
--- rxvt-2.6.4-old/src/Makefile.in	1999-10-14 16:48:55.000000000 -0700
+++ rxvt-2.6.4-new/src/Makefile.in	2009-07-27 22:10:03.000000000 -0700
@@ -60,7 +60,7 @@
 .c.o:
 	$(CC) $(DEFS) $(DEBUG) -c $(CPPFLAGS) $(XINC)  -I. -I$(basedir) -I$(srcdir) $(DINCLUDE) $(CFLAGS) $<
 #-------------------------------------------------------------------------
-all: protos rxvt
+all: rxvt
 
 rxvt: version.h $(OBJS)
 	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LIBS) $(XLIB) $(DLIB)
