diff -Naur xf86-video-ati-7.1.0-old/src/radeon_dri2.c xf86-video-ati-7.1.0-new/src/radeon_dri2.c
--- xf86-video-ati-7.1.0-old/src/radeon_dri2.c	2013-01-30 07:52:53.000000000 -0800
+++ xf86-video-ati-7.1.0-new/src/radeon_dri2.c	2013-02-02 06:02:25.000000000 -0800
@@ -431,10 +431,15 @@
     }
 
     if (translate && drawable->type == DRAWABLE_WINDOW) {
+#ifdef COMPOSITE
 	PixmapPtr pPix = GetDrawablePixmap(drawable);
 
 	off_x = drawable->x - pPix->screen_x;
 	off_y = drawable->y - pPix->screen_y;
+#else
+		off_x = 0;
+		off_y = 0;
+#endif
     }
     gc = GetScratchGC(dst_drawable->depth, pScreen);
     copy_clip = REGION_CREATE(pScreen, NULL, 0);
