diff -Naur gcc-16.1.1/gcc/gcc.cc gcc-16.1.1.new/gcc/gcc.cc
--- gcc-16.1.1/gcc/gcc.cc	2026-06-13 16:43:14.000000000 +0200
+++ gcc-16.1.1.new/gcc/gcc.cc	2026-06-13 20:14:28.058758713 +0200
@@ -1850,7 +1850,7 @@
   char *buf;
 
 #if USE_LD_AS_NEEDED
-#if defined(USE_LD_AS_NEEDED_LDSCRIPT) && !defined(USE_LIBUNWIND_EXCEPTIONS)
+#if defined(_USE_LD_AS_NEEDED_LDSCRIPT) && !defined(USE_LIBUNWIND_EXCEPTIONS)
   buf = concat ("%{static|static-libgcc|static-pie:", static_name, " ", eh_name, "}"
 		"%{!static:%{!static-libgcc:%{!static-pie:"
 		"%{!shared-libgcc:",
