diff -Naur gcc-7.4.0-old/libstdc++-v3/configure gcc-7.4.0-new/libstdc++-v3/configure
--- gcc-7.4.0-old/libstdc++-v3/configure	2018-08-13 19:16:02.093382000 +0000
+++ gcc-7.4.0-new/libstdc++-v3/configure	2019-12-20 02:16:52.259323666 +0000
@@ -21762,7 +21762,7 @@
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
-      #include <sys/sdt.h>
+      #include <sys/sdtxx.h>
       int f() { STAP_PROBE(hi, bob); }
 
 _ACEOF
