diff -Naur PyXML-0.8.4-old/setup.py PyXML-0.8.4-new/setup.py
--- PyXML-0.8.4-old/setup.py	2004-11-29 06:01:15.000000000 -0800
+++ PyXML-0.8.4-new/setup.py	2010-06-26 16:56:59.000000000 -0700
@@ -6,7 +6,7 @@
 
 import sys, os, string
 
-from distutils.core import setup, Extension
+from setuptools import setup, Extension
 from setupext import Data_Files, install_Data_Files, wininst_request_delete
 from distutils.sysconfig import get_config_var, get_config_vars
 from distutils.sysconfig import parse_config_h, get_config_h_filename
