diff --speed-large-files --no-dereference --minimal -Naur xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/CMakeLists.txt xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/CMakeLists.txt
--- xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/CMakeLists.txt	2023-12-09 22:24:46.000000000 +0100
+++ xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/CMakeLists.txt	2023-12-18 15:53:45.693404488 +0100
@@ -566,7 +566,7 @@
 endif()
 
 # Documentation
-find_package(Doxygen)
+# find_package(Doxygen)
 if(DOXYGEN_FOUND)
   add_custom_target(doc
                     COMMAND ${DOXYGEN_EXECUTABLE} ${CMAKE_SOURCE_DIR}/docs/doxygen/Doxyfile.doxy
