diff --speed-large-files --no-dereference --minimal -Naur xbmc-68149f72ee5fc31e7f3dc32ef5733ddec3cc5b05/tools/depends/target/libdvdread/LIBDVDREAD-VERSION xbmc-68149f72ee5fc31e7f3dc32ef5733ddec3cc5b05/tools/depends/target/libdvdread/LIBDVDREAD-VERSION
--- xbmc-68149f72ee5fc31e7f3dc32ef5733ddec3cc5b05/tools/depends/target/libdvdread/LIBDVDREAD-VERSION	2026-08-08 10:34:34.000000000 +0200
+++ xbmc-68149f72ee5fc31e7f3dc32ef5733ddec3cc5b05/tools/depends/target/libdvdread/LIBDVDREAD-VERSION	2026-08-08 18:49:35.457500619 +0200
@@ -1,6 +1,6 @@
 LIBNAME=libdvdread
 VERSION=7.0.1
 ARCHIVE=$(LIBNAME)-$(VERSION).tar.bz2
-SHA512=b5390a1acb8fdf6e24188f9259199eb0fef2dd4b332447ba2bfa571f9fc0c3bfad00c1d27a8b8806180a853c04a818b7802d3075ff787142c1a175ad798c6a3d
+SHA512=03e7376327ba211fcc51ee4e4705b1cc7649c613a96e186eb62821b57a11a519c46720e33cf3351cd95fa02db200e40d3b1b9f7e4fe5a3003853fc40c5f511f0
 BYPRODUCT=libdvdread.a
 BYPRODUCT_WIN=dvdread.lib
