diff --speed-large-files --no-dereference --minimal -Naur xbmc-68149f72ee5fc31e7f3dc32ef5733ddec3cc5b05/tools/depends/target/libdvdnav/LIBDVDNAV-VERSION xbmc-68149f72ee5fc31e7f3dc32ef5733ddec3cc5b05/tools/depends/target/libdvdnav/LIBDVDNAV-VERSION
--- xbmc-68149f72ee5fc31e7f3dc32ef5733ddec3cc5b05/tools/depends/target/libdvdnav/LIBDVDNAV-VERSION	2026-08-08 10:34:34.000000000 +0200
+++ xbmc-68149f72ee5fc31e7f3dc32ef5733ddec3cc5b05/tools/depends/target/libdvdnav/LIBDVDNAV-VERSION	2026-08-08 18:51:44.101163626 +0200
@@ -1,6 +1,6 @@
 LIBNAME=libdvdnav
 VERSION=7.0.0
 ARCHIVE=$(LIBNAME)-$(VERSION).tar.bz2
-SHA512=8d12d476e352def9716ecbd7ba184a70dd9bae87dcde669b8647a0ef9ac1991f617c9918c043e159eb353302c4cc50fb71bc591ceaeba1d3b9d47af77fa72c71
+SHA512=4df6336050f3b79ca2edafac28dd0383b950272ba6d8ebe6e43a85eddc737ce740d72a3fc1fa03c26fc5f285125e3672a022555a5399259386b2036c467a47f3
 BYPRODUCT=libdvdnav.a
 BYPRODUCT_WIN=dvdnav.lib
