diff --speed-large-files --no-dereference --minimal -Naur xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/tools/depends/target/libdvdnav/LIBDVDNAV-VERSION xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/tools/depends/target/libdvdnav/LIBDVDNAV-VERSION
--- xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/tools/depends/target/libdvdnav/LIBDVDNAV-VERSION	2023-12-09 22:24:46.000000000 +0100
+++ xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/tools/depends/target/libdvdnav/LIBDVDNAV-VERSION	2023-12-18 19:45:37.673338743 +0100
@@ -3,6 +3,6 @@
 VERSION=6.1.1-Next-Nexus-Alpha2-2
 FULL_URL=$(BASE_URL)/archive/$(VERSION).tar.gz
 ARCHIVE=$(LIBNAME)-$(VERSION).tar.gz
-SHA512=51e6fc033121241354a5f0b3fc9a430577ae3ff6bb7f31445aa548ef4893037fb80eea3b2c6774c81e9ebaf9c45e9b490c98c2c65eb38f9f7daba84b236f7e1d
+SHA512=889bc987113189bfdfc1ebf603b256e5b228be37196455dcfdc7f66debe18e7298b82b807e86ff18a8925e491395ed4007c87cf18ba1ff1ccc213d86f5342a10
 BYPRODUCT=libdvdnav.a
 BYPRODUCT_WIN=libdvdnav.lib
