diff --speed-large-files --no-dereference --minimal -Naur xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/CMakeLists.txt xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/CMakeLists.txt
--- xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/CMakeLists.txt	2023-12-18 20:30:53.753347738 +0100
+++ xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/CMakeLists.txt	2023-12-18 19:54:07.590007112 +0100
@@ -241,7 +241,6 @@
                   SmbClient
                   Sndio
                   UDEV
-                  Udfread
                   XSLT
                   ${PLATFORM_OPTIONAL_DEPS})
 
