diff --speed-large-files --no-dereference --minimal -Naur xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/cmake/modules/buildtools/FindClangFormat.cmake xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/cmake/modules/buildtools/FindClangFormat.cmake
--- xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/cmake/modules/buildtools/FindClangFormat.cmake	2023-12-09 22:24:46.000000000 +0100
+++ xbmc-979dbf5fe5d0b42daf633434afc3986651c8646a/cmake/modules/buildtools/FindClangFormat.cmake	2023-12-19 17:18:49.040098863 +0100
@@ -3,7 +3,7 @@
 # ----------
 # Finds clang-format
 
-find_program(CLANG_FORMAT_EXECUTABLE clang-format)
+# find_program(CLANG_FORMAT_EXECUTABLE clang-format)
 
 if(CLANG_FORMAT_EXECUTABLE)
   execute_process(COMMAND "${CLANG_FORMAT_EXECUTABLE}" --version
