diff --speed-large-files --no-dereference --minimal -Naur xbmc-2dc676eb7519299ea005ad55a8d4888af0b45860/xbmc/windowing/gbm/drm/DRMUtils.cpp xbmc-2dc676eb7519299ea005ad55a8d4888af0b45860/xbmc/windowing/gbm/drm/DRMUtils.cpp
--- xbmc-2dc676eb7519299ea005ad55a8d4888af0b45860/xbmc/windowing/gbm/drm/DRMUtils.cpp	2025-08-22 13:36:02.000000000 +0200
+++ xbmc-2dc676eb7519299ea005ad55a8d4888af0b45860/xbmc/windowing/gbm/drm/DRMUtils.cpp	2025-08-22 17:49:29.127802341 +0200
@@ -190,8 +190,7 @@
           {
             return (plane->GetPlaneId() != videoPlaneId &&
                     (videoPlaneId == 0 || plane->SupportsFormat(DRM_FORMAT_ARGB8888)) &&
-                    (plane->SupportsFormat(DRM_FORMAT_XRGB2101010) ||
-                     plane->SupportsFormat(DRM_FORMAT_XRGB8888)));
+                     plane->SupportsFormat(DRM_FORMAT_XRGB8888));
           }
           return false;
         });
@@ -221,17 +220,9 @@
   if (m_video_plane)
     CLog::LogF(LOGDEBUG, "Using video plane {}", m_video_plane->GetPlaneId());
 
-  if (m_gui_plane->SupportsFormat(DRM_FORMAT_XRGB2101010))
-  {
-    m_gui_plane->SetFormat(DRM_FORMAT_XRGB2101010);
-    CLog::LogF(LOGDEBUG, "Using 10bit gui plane {}", m_gui_plane->GetPlaneId());
-  }
-  else
-  {
-    m_gui_plane->SetFormat(DRM_FORMAT_XRGB8888);
-    CLog::LogF(LOGDEBUG, "Using gui plane {}", m_gui_plane->GetPlaneId());
-  }
-
+  m_gui_plane->SetFormat(DRM_FORMAT_XRGB8888);
+  CLog::Log(LOGDEBUG, "CDRMUtils::{} - using gui plane {}", __FUNCTION__,
+            m_gui_plane->GetPlaneId());
   return true;
 }
 
