diff --speed-large-files --no-dereference --minimal -Naur xbmc-4211a62d86456e9bdd9a9be5169e5ae68e2872d0/xbmc/guilib/GUIWindowManager.cpp xbmc-4211a62d86456e9bdd9a9be5169e5ae68e2872d0/xbmc/guilib/GUIWindowManager.cpp
--- xbmc-4211a62d86456e9bdd9a9be5169e5ae68e2872d0/xbmc/guilib/GUIWindowManager.cpp	2025-02-17 20:56:21.000000000 +0100
+++ xbmc-4211a62d86456e9bdd9a9be5169e5ae68e2872d0/xbmc/guilib/GUIWindowManager.cpp	2025-02-18 10:10:07.464932263 +0100
@@ -43,6 +43,7 @@
 #include "settings/windows/GUIWindowSettingsCategory.h"
 #include "settings/windows/GUIWindowSettingsScreenCalibration.h"
 #include "threads/SingleLock.h"
+#include "utils/Geometry.h"
 #include "utils/StringUtils.h"
 #include "utils/URIUtils.h"
 #include "utils/Variant.h"
@@ -1395,6 +1396,9 @@
 
       if (!hasRendered)
         CServiceBroker::GetWinSystem()->SetDirtyRegions(dirtyRegions);
+
+      if (!hasRendered)
+        CServiceBroker::GetWinSystem()->SetDirtyRegions(dirtyRegions);
 
       CServiceBroker::GetWinSystem()->GetGfxContext().SetScissors(i);
       RenderPass();
