diff --speed-large-files --no-dereference --minimal -Naur u-boot-2a0fc326204da252a921750b9888f5f25b79ce09/arch/arm/dts/rk3588s-rock-5a.dts u-boot-2a0fc326204da252a921750b9888f5f25b79ce09/arch/arm/dts/rk3588s-rock-5a.dts
--- u-boot-2a0fc326204da252a921750b9888f5f25b79ce09/arch/arm/dts/rk3588s-rock-5a.dts	2023-10-07 13:56:06.000000000 +0200
+++ u-boot-2a0fc326204da252a921750b9888f5f25b79ce09/arch/arm/dts/rk3588s-rock-5a.dts	2023-10-08 18:24:20.836688996 +0200
@@ -13,6 +13,13 @@
 	model = "Radxa ROCK 5A";
 	compatible = "radxa,rock-5a", "rockchip,rk3588";
 
+	aliases {
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci;
+		ethernet0 = &gmac1;
+		serial2 = &uart2;
+	};
+
 	vcc12v_dcin: vcc12v-dcin {
 		u-boot,dm-pre-reloc;
 		compatible = "regulator-fixed";
