diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-6.11.0/qttools/src/qdoc/CMakeLists.txt qt-everywhere-src-6.11.0/qttools/src/qdoc/CMakeLists.txt
--- qt-everywhere-src-6.11.0/qttools/src/qdoc/CMakeLists.txt	2026-03-10 06:07:41.000000000 +0100
+++ qt-everywhere-src-6.11.0/qttools/src/qdoc/CMakeLists.txt	2026-03-31 16:13:31.166361102 +0200
@@ -5,6 +5,6 @@
 add_subdirectory(catch_conversions)
 add_subdirectory(catch_generators)
 
-if(QT_FEATURE_qdoc)
-    add_subdirectory(qdoc)
-endif()
+#if(QT_FEATURE_qdoc)
+#    add_subdirectory(qdoc)
+#endif()
