diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-6.5.0/qtwebengine/cmake/FindGn.cmake qt-everywhere-src-6.5.0/qtwebengine/cmake/FindGn.cmake
--- qt-everywhere-src-6.5.0/qtwebengine/cmake/FindGn.cmake	2023-03-28 21:45:02.000000000 +0200
+++ qt-everywhere-src-6.5.0/qtwebengine/cmake/FindGn.cmake	2023-05-18 19:06:18.106826898 +0200
@@ -38,6 +38,12 @@
         Gn_EXECUTABLE
 )
 
+message(Gn_VERSION="${Gn_VERSION}")
+message(Gn_EXECUTABLE="${Gn_EXECUTABLE}")
+message(Gn_FOUND="${Gn_FOUND}")
+
+set(Gn_FOUND "ON")
+
 if(Gn_FOUND AND NOT TARGET Gn::gn AND NOT CMAKE_SCRIPT_MODE_FILE)
     add_executable(Gn::gn IMPORTED)
     set_property(TARGET Gn::gn PROPERTY IMPORTED_LOCATION ${Gn_EXECUTABLE})
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-6.5.0/qtwebengine/src/3rdparty/chromium/build/config/linux/pkg-config.py qt-everywhere-src-6.5.0/qtwebengine/src/3rdparty/chromium/build/config/linux/pkg-config.py
--- qt-everywhere-src-6.5.0/qtwebengine/src/3rdparty/chromium/build/config/linux/pkg-config.py	2023-03-28 21:45:02.000000000 +0200
+++ qt-everywhere-src-6.5.0/qtwebengine/src/3rdparty/chromium/build/config/linux/pkg-config.py	2023-05-18 19:06:40.246826240 +0200
@@ -80,8 +80,6 @@
   # from pkg-config's |prefix| variable.
   prefix = subprocess.check_output([options.pkg_config,
       "--variable=prefix"] + args, env=os.environ).decode('utf-8')
-  if prefix[-4] == '/usr':
-    return prefix[4:]
   return prefix
 
 
diff --speed-large-files --no-dereference --minimal -Naur qt-everywhere-src-6.5.0/qtwebengine/src/core/configure/BUILD.root.gn.in qt-everywhere-src-6.5.0/qtwebengine/src/core/configure/BUILD.root.gn.in
--- qt-everywhere-src-6.5.0/qtwebengine/src/core/configure/BUILD.root.gn.in	2023-03-28 21:45:02.000000000 +0200
+++ qt-everywhere-src-6.5.0/qtwebengine/src/core/configure/BUILD.root.gn.in	2023-05-18 17:34:34.249298689 +0200
@@ -97,7 +97,6 @@
 }
 
 shared_library("QtWebEngineCore") {
-  rsp_types = [ "objects", "archives", "libs", "lflags"]
   configs += [
     ":cpp17_config",
     ":QtWebEngineCore_config",
@@ -691,7 +690,6 @@
 
 if (enable_spellcheck) {
   shared_library("convert_dict") {
-    rsp_types = [ "objects", "archives", "libs", "lflags" ]
     configs += [ "//build/config/compiler:wexit_time_destructors" ]
     deps = [
       "//chrome/tools/convert_dict:lib",
