diff -Naur kbdd-20060719Z-old/kbd.c kbdd-20060719Z-new/kbd.c
--- kbdd-20060719Z-old/kbd.c	2006-07-18 09:34:52.000000000 -0700
+++ kbdd-20060719Z-new/kbd.c	2007-02-12 07:26:44.000000000 -0800
@@ -1412,7 +1412,7 @@
 		compaq_microkbd();
 	else if (kbdtype == KBD_TYPE_LIRC)
 #ifdef USELIRC
-		lirc(basename(argv[0]);
+		lirc(basename(argv[0]));
 #else
 	{
 		fprintf(stderr, "keyboard type 'lirc' is not supported in this version\n");
