diff --speed-large-files --no-dereference --minimal -Naur sysvinit-2.99/src/init.c sysvinit-2.99/src/init.c
--- sysvinit-2.99/src/init.c	2021-02-21 19:02:25.000000000 +0100
+++ sysvinit-2.99/src/init.c	2022-12-19 11:10:20.106632157 +0100
@@ -2587,7 +2587,7 @@
 {
   CHILD		*ch;
   static int	newlevel = 0;
-  static int	warn = 1;
+  static int	warn = 0;
   int		loglevel;
   int		oldlevel;
 
@@ -2643,9 +2643,9 @@
 				ch->flags &= ~(FAILING|WAITING|XECUTED);
 			break;
 		default:
-			if (warn)
+/*			if (warn)
 			  initlog(L_VB,
-				"no more processes left in this runlevel");
+				"no more processes left in this runlevel"); */
 			warn = 0;
 			loglevel = -1;
 			if (got_signals == 0)
@@ -2653,7 +2653,7 @@
 			break;
 	}
 	if (loglevel > 0) {
-		initlog(L_VB, "Entering runlevel: %c", runlevel);
+/*		initlog(L_VB, "Entering runlevel: %c", runlevel); */
 		wrote_utmp_rlevel = 0;
 		wrote_wtmp_rlevel = 0;
 		write_utmp_wtmp("runlevel", "~~", runlevel + 256 * oldlevel, RUN_LVL, "~");
@@ -2885,7 +2885,7 @@
   	/*
 	 *	Say hello to the world
 	 */
-  	initlog(L_CO, bootmsg, "booting");
+  	/* initlog(L_CO, bootmsg, "booting"); */
 
   	/*
 	 *	See if we have to start an emergency shell.
