diff -Naur sysvinit-2.88dsf-old/src/Makefile sysvinit-2.88dsf-new/src/Makefile
--- sysvinit-2.88dsf-old/src/Makefile	2010-04-11 02:30:27.000000000 -0700
+++ sysvinit-2.88dsf-new/src/Makefile	2011-03-24 11:28:33.000000000 -0700
@@ -66,6 +66,12 @@
 endif
 INSTALL_DIR	= install -m 755 -d
 MANDIR		= /usr/share/man
+BINDIR          = /bin
+USRBINDIR       = /usr/bin
+SBINDIR         = /sbin
+USRLIBDIR       = /usr/lib
+CONFDIR         = /etc
+INCLUDEDIR      = /usr/include
 
 ifeq ($(WITH_SELINUX),yes)
   SELINUX_DEF	=  -DWITH_SELINUX
@@ -78,7 +84,7 @@
 endif
 
 # Additional libs for GNU libc.
-ifneq ($(wildcard /usr/lib*/libcrypt.a),)
+ifneq ($(wildcard $(ROOT)$(USRLIBDIR)/libcrypt.a),)
   SULOGINLIBS	+= -lcrypt
 endif
 
@@ -138,28 +144,28 @@
 distclean:	clobber
 
 install:
-		$(INSTALL_DIR) $(ROOT)/bin/ $(ROOT)/sbin/
-		$(INSTALL_DIR) $(ROOT)/usr/bin/
+		$(INSTALL_DIR) $(ROOT)/bin/ $(ROOT)/$(SBINDIR)/
+		$(INSTALL_DIR) $(ROOT)/$(USRBINDIR)/
 		for i in $(BIN); do \
-			$(INSTALL_EXEC) $$i $(ROOT)/bin/ ; \
+			$(INSTALL_EXEC) $$i $(ROOT)/$(BINDIR)/ ; \
 		done
 		for i in $(SBIN); do \
-			$(INSTALL_EXEC) $$i $(ROOT)/sbin/ ; \
+			$(INSTALL_EXEC) $$i $(ROOT)/$(SBINDIR)/ ; \
 		done
 		for i in $(USRBIN); do \
-			$(INSTALL_EXEC) $$i $(ROOT)/usr/bin/ ; \
+			$(INSTALL_EXEC) $$i $(ROOT)/$(USRBINDIR)/ ; \
 		done
 		# $(INSTALL_DIR) $(ROOT)/etc/
 		# $(INSTALL_EXEC) initscript.sample $(ROOT)/etc/
-		ln -sf halt $(ROOT)/sbin/reboot
-		ln -sf halt $(ROOT)/sbin/poweroff
-		ln -sf init $(ROOT)/sbin/telinit
-		ln -sf /sbin/killall5 $(ROOT)/bin/pidof
-		if [ ! -f $(ROOT)/usr/bin/lastb ]; then \
-			ln -sf last $(ROOT)/usr/bin/lastb; \
+		ln -sf halt $(ROOT)/$(SBINDIR)/reboot
+		ln -sf halt $(ROOT)/$(SBINDIR)/poweroff
+		ln -sf init $(ROOT)/$(SBINDIR)/telinit
+		ln -sf /sbin/killall5 $(ROOT)/$(BINDIR)/pidof
+		if [ ! -f $(ROOT)/$(USRBINDIR)/lastb ]; then \
+			ln -sf last $(ROOT)/$(USRBINDIR)/lastb; \
 		fi
-		$(INSTALL_DIR) $(ROOT)/usr/include/
-		$(INSTALL_DATA) initreq.h $(ROOT)/usr/include/
+		$(INSTALL_DIR) $(ROOT)/$(INCLUDEDIR)/
+		$(INSTALL_DATA) initreq.h $(ROOT)/$(INCLUDEDIR)/
 		$(INSTALL_DIR) $(ROOT)$(MANDIR)/man1/
 		$(INSTALL_DIR) $(ROOT)$(MANDIR)/man5/
 		$(INSTALL_DIR) $(ROOT)$(MANDIR)/man8/
