diff -Naur sysvinit-2.88dsf-old/src/init.c sysvinit-2.88dsf-new/src/init.c
--- sysvinit-2.88dsf-old/src/init.c	2010-04-13 02:48:54.000000000 -0700
+++ sysvinit-2.88dsf-new/src/init.c	2011-02-05 09:45:11.000000000 -0800
@@ -2369,7 +2369,7 @@
 			break;
 		default:
 			if (warn)
-			  initlog(L_VB,
+			  initlog(L_SY,
 				"no more processes left in this runlevel");
 			warn = 0;
 			loglevel = -1;
@@ -2378,7 +2378,7 @@
 			break;
 	}
 	if (loglevel > 0) {
-		initlog(L_VB, "Entering runlevel: %c", runlevel);
+		initlog(L_SY, "Entering runlevel: %c", runlevel);
 		wrote_utmp_rlevel = 0;
 		wrote_wtmp_rlevel = 0;
 		write_utmp_wtmp("runlevel", "~~", runlevel + 256 * oldlevel, RUN_LVL, "~");
@@ -2598,7 +2598,7 @@
   	/*
 	 *	Say hello to the world
 	 */
-  	initlog(L_CO, bootmsg, "booting");
+  	initlog(L_SY, bootmsg, "booting");
 
   	/*
 	 *	See if we have to start an emergency shell.
@@ -2624,7 +2624,7 @@
 	/*
 	 *	Restart: unblock signals and let the show go on
 	 */
-	initlog(L_CO, bootmsg, "reloading");
+	initlog(L_SY, bootmsg, "reloading");
 	sigfillset(&sgt);
 	sigprocmask(SIG_UNBLOCK, &sgt, NULL);
 
