diff -Naur sysvinit-2.88dsf-old/src/halt.c sysvinit-2.88dsf-new/src/halt.c
--- sysvinit-2.88dsf-old/src/halt.c	2011-02-05 09:45:51.000000000 -0800
+++ sysvinit-2.88dsf-new/src/halt.c	2011-02-05 09:56:07.000000000 -0800
@@ -246,8 +246,19 @@
 		 *	See if we are in runlevel 0 or 6.
 		 */
 		c = get_runlevel();
-		if (c != '0' && c != '6')
+		if (c != '0' && c != '6') {
+			char *file;
+
+			if (do_poweroff) {
+				file = strdup("/poweroff");
+			} else {
+				file = strdup("/halt");
+			}
+			close(open(file, O_CREAT|O_RDWR, 0644));
+			free(file);
+
 			do_shutdown(do_reboot ? "-r" : "-h", tm);
+		}
 	}
 
 	/*
