diff -Naur sysvinit-2.99-old/src/init.c sysvinit-2.99-new/src/init.c
--- sysvinit-2.99-old/src/init.c	2021-02-21 19:02:25.000000000 +0100
+++ sysvinit-2.99-new/src/init.c	2021-11-03 12:01:13.409831432 +0100
@@ -2587,7 +2587,7 @@
 {
   CHILD		*ch;
   static int	newlevel = 0;
-  static int	warn = 1;
+  static int	warn = 0;
   int		loglevel;
   int		oldlevel;
 
@@ -2653,7 +2653,7 @@
 			break;
 	}
 	if (loglevel > 0) {
-		initlog(L_VB, "Entering runlevel: %c", runlevel);
+/*		initlog(L_VB, "Entering runlevel: %c", runlevel); */
 		wrote_utmp_rlevel = 0;
 		wrote_wtmp_rlevel = 0;
 		write_utmp_wtmp("runlevel", "~~", runlevel + 256 * oldlevel, RUN_LVL, "~");
@@ -2885,7 +2885,7 @@
   	/*
 	 *	Say hello to the world
 	 */
-  	initlog(L_CO, bootmsg, "booting");
+  	/* initlog(L_CO, bootmsg, "booting"); */
 
   	/*
 	 *	See if we have to start an emergency shell.
