diff --speed-large-files --no-dereference --minimal -Naur sysvinit-2.99/src/sulogin.c sysvinit-2.99/src/sulogin.c
--- sysvinit-2.99/src/sulogin.c	2021-02-21 19:02:25.000000000 +0100
+++ sysvinit-2.99/src/sulogin.c	2022-08-05 19:39:12.176664049 +0200
@@ -51,7 +51,6 @@
 #ifdef __linux__
 #  include <sys/statfs.h>
 #  include <sys/mount.h>
-#  include <linux/fs.h>
 #  include <linux/magic.h>
 #  include <linux/major.h>
 #  ifndef TMPFS_MAGIC
