diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/cardutil.cpp mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/cardutil.cpp
--- mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/cardutil.cpp	2019-01-05 12:44:38.820000000 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/cardutil.cpp	2019-01-05 12:44:40.590000001 +0100
@@ -1948,7 +1948,7 @@
 {
     QString devlabel;
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("SELECT cardtype, videodevice "
+    query.prepare("SELECT cardtype, displayname "
                   "FROM capturecard WHERE cardid = :INPUTID ");
     query.bindValue(":INPUTID", inputid);
 
