diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-master-20190105-g3382e1a-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-master-20190105-g3382e1a-old/mythtv/programs/mythbackend/mainserver.cpp	2019-01-05 12:44:49.443333333 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/programs/mythbackend/mainserver.cpp	2019-01-05 12:44:50.750000000 +0100
@@ -2015,6 +2015,7 @@
             if (!controlSocketList.remove(socket))
                 return; // socket was disconnected
             ft = new FileTransfer(filename, socket, writemode);
+            fileTransferList.push_back(ft);
         }
         else
         {
@@ -2022,6 +2023,7 @@
             if (!controlSocketList.remove(socket))
                 return; // socket was disconnected
             ft = new FileTransfer(filename, socket, usereadahead, timeout_ms);
+            fileTransferList.push_back(ft);
         }
 
         if (!ft->isOpen())
@@ -2039,7 +2041,6 @@
             QString("adding: %1(%2) as a file transfer")
                                       .arg(commands[2])
                                       .arg(quintptr(socket),0,16));
-        fileTransferList.push_back(ft);
 
         retlist << QString::number(socket->GetSocketDescriptor());
         retlist << QString::number(ft->GetFileSize());
