diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythui/mythmainwindow.cpp mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythui/mythmainwindow.cpp
--- mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythui/mythmainwindow.cpp	2020-01-10 09:49:58.760000001 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythui/mythmainwindow.cpp	2020-01-10 09:49:59.846666668 +0100
@@ -2864,12 +2864,12 @@
 
 int MythMainWindow::NormX(const int x)
 {
-    return qRound(x * d->wmult);
+    return (int)(x * d->wmult);
 }
 
 int MythMainWindow::NormY(const int y)
 {
-    return qRound(y * d->hmult);
+    return (int)(y * d->hmult);
 }
 
 void MythMainWindow::SetScalingFactors(float wmult, float hmult)
