diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/programs/mythcommflag/ClassicLogoDetector.cpp mythtv-30-20200110-ga27754a-new/mythtv/programs/mythcommflag/ClassicLogoDetector.cpp
--- mythtv-30-20200110-ga27754a-old/mythtv/programs/mythcommflag/ClassicLogoDetector.cpp	2020-01-10 09:49:59.563333334 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/programs/mythcommflag/ClassicLogoDetector.cpp	2020-01-10 09:50:00.840000002 +0100
@@ -49,6 +49,12 @@
     commDetectLogoBadEdgeThreshold =
         gCoreContext->GetSetting("CommDetectLogoBadEdgeThreshold", "0.85")
         .toDouble();
+    commDetectLogoLocation =
+        gCoreContext->GetSetting("CommDetectLogoLocation", "");
+    commDetectLogoWidthRatio =
+        gCoreContext->GetNumSetting("CommDetectLogoWidthRatio", 4);
+    commDetectLogoHeightRatio =
+        gCoreContext->GetNumSetting("CommDetectLogoHeightRatio", 4);
 }
 
 unsigned int ClassicLogoDetector::getRequiredAvailableBufferForSearch()
@@ -239,8 +245,8 @@
         cerr << "Hit ENTER to continue" << endl;
         getchar();
 #endif
-        if (((logoMaxX - logoMinX) < (width / 4)) &&
-            ((logoMaxY - logoMinY) < (height / 4)) &&
+        if (((logoMaxX - logoMinX) < (width / commDetectLogoWidthRatio)) &&
+            ((logoMaxY - logoMinY) < (height / commDetectLogoHeightRatio)) &&
             (pixelsInMask > minPixelsInMask))
         {
             logoInfoAvailable = true;
@@ -461,14 +467,22 @@
 
     for (y = commDetectBorder + r; y < (height - commDetectBorder - r); y++)
     {
-        if ((y > (height/4)) && (y < (height * 3 / 4)))
+        if (
+            (commDetectLogoLocation.contains("N") && (y > (height/4))) ||
+            (commDetectLogoLocation.contains("S") && (y < (height * 3 / 4))) ||
+            ((y > (height/4)) && (y < (height * 3 / 4)))
+            )
             continue;
 
         for (x = commDetectBorder + r; x < (width - commDetectBorder - r); x++)
         {
             int edgeCount = 0;
 
-            if ((x > (width/4)) && (x < (width * 3 / 4)))
+            if (
+                (commDetectLogoLocation.contains("W") && (x > (width/4))) ||
+                (commDetectLogoLocation.contains("E") && (x < (width * 3 / 4))) ||
+                ((x > (width/4)) && (x < (width * 3 / 4)))
+                )
                 continue;
 
             pos = y * width + x;
diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/programs/mythcommflag/ClassicLogoDetector.h mythtv-30-20200110-ga27754a-new/mythtv/programs/mythcommflag/ClassicLogoDetector.h
--- mythtv-30-20200110-ga27754a-old/mythtv/programs/mythcommflag/ClassicLogoDetector.h	2020-01-10 09:49:59.560000002 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/programs/mythcommflag/ClassicLogoDetector.h	2020-01-10 09:50:00.840000002 +0100
@@ -35,8 +35,11 @@
     int commDetectLogoSamplesNeeded;
     int commDetectLogoSampleSpacing;
     int commDetectLogoSecondsNeeded;
+    int commDetectLogoWidthRatio;
+    int commDetectLogoHeightRatio;
     double commDetectLogoGoodEdgeThreshold;
     double commDetectLogoBadEdgeThreshold;
+    QString commDetectLogoLocation;
 
     EdgeMaskEntry *edgeMask;
 
