diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythupnp/upnpcds.h mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythupnp/upnpcds.h
--- mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythupnp/upnpcds.h	2020-01-10 09:50:00.776666668 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythupnp/upnpcds.h	2020-01-10 09:50:01.836666668 +0100
@@ -320,7 +320,7 @@
         // Implement UPnpServiceImpl methods that we can
 
         QString GetServiceType() override // UPnpServiceImpl
-            { return "urn:schemas-upnp-org:service:ContentDirectory:4"; }
+            { return "urn:schemas-upnp-org:service:ContentDirectory:1"; }
         QString GetServiceId() override // UPnpServiceImpl
             { return "urn:upnp-org:serviceId:ContentDirectory"; }
         QString GetServiceControlURL() override // UPnpServiceImpl
diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythupnp/upnpcmgr.h mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythupnp/upnpcmgr.h
--- mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythupnp/upnpcmgr.h	2020-01-10 09:50:00.776666668 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythupnp/upnpcmgr.h	2020-01-10 09:50:01.836666668 +0100
@@ -68,7 +68,7 @@
         // Implement UPnpServiceImpl methods that we can
 
         QString GetServiceType() override // UPnpServiceImpl
-            { return "urn:schemas-upnp-org:service:ConnectionManager:3"; }
+            { return "urn:schemas-upnp-org:service:ConnectionManager:1"; }
         QString GetServiceId() override // UPnpServiceImpl
             { return "urn:upnp-org:serviceId:ConnectionManager"; }
         QString GetServiceControlURL() override // UPnpServiceImpl
diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/programs/mythbackend/devicemaster.xml mythtv-30-20200110-ga27754a-new/mythtv/programs/mythbackend/devicemaster.xml
--- mythtv-30-20200110-ga27754a-old/mythtv/programs/mythbackend/devicemaster.xml	2020-01-10 09:50:00.540000001 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/programs/mythbackend/devicemaster.xml	2020-01-10 09:50:01.840000001 +0100
@@ -7,7 +7,7 @@
     <device>
         <!-- UDN auto generated -->
         <friendlyName>MythTV AV Media Server</friendlyName>
-        <deviceType>urn:schemas-upnp-org:device:MediaServer:4</deviceType>
+        <deviceType>urn:schemas-upnp-org:device:MediaServer:1</deviceType>
         <manufacturer>MythTV</manufacturer>
         <manufacturerURL>http://www.mythtv.org/</manufacturerURL>
         <modelName>MythTV AV Media Server</modelName>
