diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-01-10 09:50:04.733333334 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-01-10 09:50:05.796666668 +0100
@@ -508,6 +508,9 @@
         // FIXME assert if TableID == SDTo
     }
 
+    DBG_SM("HandleSDT()", QString("networkID = %1 orig_net_id = %2 transportID = %3 orig_transport_id = %4")
+               .arg(networkID).arg(sdt->OriginalNetworkID()).arg(transportID).arg(sdt->TSID()));
+
     if (sdt->OriginalNetworkID() != networkID || sdt->TSID() != transportID)
     {
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
