diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp
--- mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-01-05 12:45:02.280000000 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/recorders/dvbsignalmonitor.cpp	2019-01-05 12:45:03.526666667 +0100
@@ -298,7 +298,7 @@
     // Debug output
     if (wasLocked != isLocked)
     {
-        LOG(VB_CHANNEL, LOG_INFO, LOC + "UpdateValues -- Signal " +
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Signal " +
                 (isLocked ? "Locked" : "Lost"));
     }
 
