diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/tv_rec.cpp	2020-01-10 09:50:07.766666668 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/tv_rec.cpp	2020-01-10 09:50:08.983333335 +0100
@@ -3958,7 +3958,7 @@
     // Start signal monitoring for devices capable of monitoring
     if (use_sm)
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "Starting Signal Monitor");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Starting Signal Monitor");
         bool error = false;
         if (!SetupSignalMonitor(
                 !antadj, request.flags & kFlagEITScan, livetv | antadj))
@@ -4008,7 +4008,7 @@
                     signalMonitorDeadline =
                         curRecording->GetRecordingEndTime().addSecs(-10);
 
-                    LOG(VB_CHANNEL, LOG_DEBUG, LOC +
+                    LOG(VB_GENERAL, LOG_DEBUG, LOC +
                         QString("Pre-fail start deadline: %1 "
                                 "Start recording deadline: %2 "
                                 "Good signal deadline: %3")
@@ -4080,7 +4080,7 @@
 
     if (signalMonitor->IsAllGood())
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
         if (curRecording && (current_time > startRecordingDeadline))
         {
             newRecStatus = RecStatus::Failing;
@@ -4169,7 +4169,7 @@
             --signalMonitorCheckCnt;
         else
         {
-            LOG(VB_RECORD, LOG_INFO, LOC +
+            LOG(VB_GENERAL, LOG_INFO, LOC +
                 QString("TuningSignalCheck: Still waiting.  Will timeout @ %1")
                 .arg(signalMonitorDeadline.toLocalTime()
                      .toString("hh:mm:ss.zzz")));
