diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/avformatdecoder.h mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/avformatdecoder.h
--- mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/avformatdecoder.h	2020-01-10 09:50:08.906666668 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/avformatdecoder.h	2020-01-10 09:50:09.960000001 +0100
@@ -75,10 +75,11 @@
     }
     QString toString() const
     {
-        return QString("id(%1) %2Hz %3ch %4bps %5 (profile %6)")
+        return QString("id(%1) %2Hz %3/%7ch %4bps %5 (profile %6)")
             .arg(ff_codec_id_string(codec_id),4).arg(sample_rate,6)
             .arg(channels,2).arg(AudioOutputSettings::FormatToBits(format),2)
-            .arg((do_passthru) ? "pt":"",3).arg(codec_profile);
+            .arg((do_passthru) ? "pt":"",3).arg(codec_profile)
+            .arg(original_channels);
     }
 };
 
