diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp
--- mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-01-10 09:50:09.886666668 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/recorders/dtvsignalmonitor.cpp	2020-01-10 09:50:10.963333335 +0100
@@ -513,6 +513,11 @@
 
     if (sdt->OriginalNetworkID() != networkID || sdt->TSID() != transportID)
     {
+        LOG(VB_GENERAL, LOG_WARNING, LOC +
+            QString("Wrong SDT; onid %1 tsid %2, desired %3 %4\n"
+                    "\t\t\tRescan your transports.")
+                .arg(detectedNetworkID).arg(detectedTransportID)
+                .arg(networkID).arg(transportID) );
         GetDVBStreamData()->SetVersionSDT(sdt->TSID(), -1, 0);
     }
     else
