diff -Naur mythtv-30-20200110-ga27754a-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp mythtv-30-20200110-ga27754a-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp
--- mythtv-30-20200110-ga27754a-old/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2020-01-10 09:50:10.940000001 +0100
+++ mythtv-30-20200110-ga27754a-new/mythplugins/mythmusic/mythmusic/avfdecoder.cpp	2020-01-10 09:50:11.966666668 +0100
@@ -162,6 +162,12 @@
             if (m_meta_album_pos > 0)
                 result["album"] = rx.cap(m_meta_album_pos);
         }
+        else
+        {
+            LOG(VB_PLAYBACK, LOG_INFO, QString(
+                    "ShoutCast: Meta StreamTitle'%1' not matched by '%2'")
+                .arg(title).arg(m_meta_format) );
+        }
     }
 
     return result;
