diff -Naur mythtv-30-20200110-ga27754a-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-30-20200110-ga27754a-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-30-20200110-ga27754a-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2020-01-10 09:50:11.906666668 +0100
+++ mythtv-30-20200110-ga27754a-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2020-01-10 09:50:13.196666668 +0100
@@ -466,6 +466,11 @@
 
     // get the MMAP path
     m_mmapPath = getZMSetting("ZM_PATH_MAP");
+    if (m_mmapPath == "")
+    {
+        m_mmapPath = "/dev/shm";
+    }
+
     if (m_debug)
     {
         cout << "Memory path directory is: " << m_mmapPath << endl;
