diff -Naur mythtv-30-20200110-ga27754a-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp mythtv-30-20200110-ga27754a-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp
--- mythtv-30-20200110-ga27754a-old/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2020-01-10 09:50:14.646666668 +0100
+++ mythtv-30-20200110-ga27754a-new/mythplugins/mythzoneminder/mythzmserver/zmserver.cpp	2020-01-10 09:50:16.080000001 +0100
@@ -25,6 +25,7 @@
 #include <sys/stat.h>
 #include <sys/shm.h>
 #include <sys/mman.h>
+#include <algorithm>
 
 #ifdef linux
 #  include <sys/vfs.h>
@@ -1105,9 +1106,13 @@
     int frameNo = atoi(tokens[3].c_str());
     string eventTime(tokens[4]);
 
+    std::replace( eventTime.begin(), eventTime.end(), '/', '-');
+    std::string eventDate = eventTime.substr(0, 8);
+
     if (m_debug)
         cout << "Getting frame " << frameNo << " for event " << eventID
-             << " on monitor " << monitorID  << " event time is " << eventTime << endl;
+             << " on monitor " << monitorID  << " event time is " << eventTime
+             << " event date is " << eventDate << endl;
 
     string outStr("");
 
@@ -1119,7 +1124,7 @@
 
     if (m_useDeepStorage)
     {
-        filepath = g_webPath + "/events/" + monitorID + "/" + eventTime + "/";
+        filepath = g_webPath + "/events/" + monitorID + "/20" + eventDate + "/" + eventID + "/" ;
         sprintf(str, m_eventFileFormat.c_str(), frameNo);
         filepath += str;
     }
@@ -1172,9 +1177,13 @@
     int frameID;
     int frameCount = 0;
 
+    std::replace( eventTime.begin(), eventTime.end(), '/', '-');
+    std::string eventDate = eventTime.substr(0, 8);
+
     if (m_debug)
         cout << "Getting analysis frame " << frameNo << " for event " << eventID
-             << " on monitor " << monitorID << " event time is " << eventTime << endl;
+             << " on monitor " << monitorID << " event time is " << eventTime
+             << " event date is " << eventDate << endl;
 
     // get the 'alarm' frames from the Frames table for this event
     MYSQL_RES *res;
@@ -1252,7 +1261,7 @@
     string frameFile("");
 
     if (m_useDeepStorage)
-        filepath = g_webPath + "/events/" + monitorID + "/" + eventTime + "/";
+        filepath = g_webPath + "/events/" + monitorID + "/20" + eventDate + "/" + eventID + "/" ;
     else
         filepath = g_webPath + "/events/" + monitorID + "/" + eventID + "/";
 
