diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/libs/libmyth/mythmediamonitor.cpp mythtv-30-20200110-ga27754a-new/mythtv/libs/libmyth/mythmediamonitor.cpp
--- mythtv-30-20200110-ga27754a-old/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-01-10 09:50:24.146666668 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/libs/libmyth/mythmediamonitor.cpp	2020-01-10 09:50:25.386666668 +0100
@@ -770,6 +770,12 @@
                  "Ignoring device: " + device->getDevicePath());
         return true;
     }
+    // If the disk is the local MiniMyth disk/conf device, stop now:
+    else if (device->getMountPath() == "/minimyth")
+    {
+        LOG(VB_MEDIA, LOG_INFO, "Ignoring device: " + device->getDevicePath());
+        return true;
+    }
 #if 0
     else
     {
