diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/configure mythtv-30-20200110-ga27754a-new/mythtv/configure
--- mythtv-30-20200110-ga27754a-old/mythtv/configure	2020-01-10 09:50:29.403333335 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/configure	2020-01-10 09:50:30.900000001 +0100
@@ -5179,7 +5179,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.3"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
