diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/programs/mythfrontend/main.cpp mythtv-30-20200110-ga27754a-new/mythtv/programs/mythfrontend/main.cpp
--- mythtv-30-20200110-ga27754a-old/mythtv/programs/mythfrontend/main.cpp	2020-01-10 09:50:31.603333334 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/programs/mythfrontend/main.cpp	2020-01-10 09:50:33.193333334 +0100
@@ -833,8 +833,22 @@
     {
         case 0 : // Do nothing
             break;
-        case 1 : // Display menu (mythdvd)*/
-            break;
+        case 1 : // Display menu (mythdvd)
+        {
+            MythThemedMenuState *menu = dynamic_cast<MythThemedMenuState *>
+                (GetMythMainWindow()->GetMainStack()->GetTopScreen());
+            if (menu)
+            {
+                menu->m_buttonList->MoveToNamedPosition("Videos");
+                QKeyEvent key(QEvent::KeyPress, Qt::Key_Return, Qt::NoModifier);
+                menu->m_buttonList->keyPressEvent(&key);
+                menu = dynamic_cast<MythThemedMenuState *>
+                    (GetMythMainWindow()->GetMainStack()->GetTopScreen());
+                if (menu)
+                    menu->m_buttonList->MoveToNamedPosition("Play a DVD or Blu-ray");
+            }
+             break;
+        }
         case 2 : // play DVD or Blu-ray
             playDisc();
             break;
