diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/configure mythtv-30-20200110-ga27754a-new/mythtv/configure
--- mythtv-30-20200110-ga27754a-old/mythtv/configure	2020-01-10 09:50:32.640000001 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/configure	2020-01-10 09:50:34.723333334 +0100
@@ -144,6 +144,8 @@
   --disable-systemd_notify disable systemd notify support
   --disable-systemd_journal disable systemd journal support
 
+  --enable-opengles        enable OpenGL ES
+
   --enable-mac-bundle      produce standalone OS X apps (e.g. mythfrontend.app)
 
   --disable-libxml2        disable libxml2 support (disc metadata)
@@ -1973,6 +1975,7 @@
     lirc
     mheg
     opengl
+    opengles
     opengl_video
     opengl_themepainter
     openmax
@@ -2775,7 +2778,7 @@
 enable mheg
 enable mythtranscode
 enable opengl
-enable opengles
+disable opengles
 enable opengl_video
 enable opengl_themepainter
 enable symbol_visibility
@@ -6026,21 +6029,34 @@
                              { check_cpp_condition x264.h "X264_MPEG2" &&
                                enable libx262; }
 enabled libxml2           && require_pkg_config libxml2 libxml-2.0 libxml/xmlversion.h xmlCheckVersion
-enabled opengl            && { check_lib opengl GL/glx.h glXGetProcAddress "-lGL" ||
-                               check_lib opengl windows.h wglGetProcAddress "-lopengl32 -lgdi32" ||
-                               check_lib opengl OpenGL/gl3.h glGetError "-Wl,-framework,OpenGL" ||
-                               check_lib opengl ES2/gl.h glGetError "-isysroot=${sysroot} -Wl,-framework,OpenGLES" ||
-                               check_lib opengl GLES/gl.h glGetError "-Wl,-framework,OpenGLES" ||
-                               { check_pkg_config GLESv2 glesv2 GLES2/gl2.h glGetError &&
-                                 check_pkg_config EGL egl EGL/egl.h eglGetProcAddress &&
-                                 enable opengl
-                               } ||
-                               { check_lib GLESv2 GLES2/gl2.h glGetError "-lGLESv2" &&
-                                 check_lib EGL EGL/egl.h eglGetProcAddress "-lEGL" &&
-                                 enable opengl
-                               } ||
-                               die "ERROR: opengl not found."
-                             }
+
+if enabled opengl; then
+  check_lib opengl GL/glx.h glXGetProcAddress "-lGL" ||
+  check_lib opengl windows.h wglGetProcAddress "-lopengl32 -lgdi32" ||
+  check_lib opengl OpenGL/gl3.h glGetError "-Wl,-framework,OpenGL" ||
+  check_lib opengl ES2/gl.h glGetError "-isysroot=${sysroot} -Wl,-framework,OpenGLES" ||
+  check_lib opengl GLES/gl.h glGetError "-Wl,-framework,OpenGLES" ||
+  { check_lib GLESv2 GLES2/gl2.h glGetError "-lGLESv2" &&
+    check_lib EGL EGL/egl.h eglGetProcAddress "-lEGL"; } ||
+  die "ERROR: OpenGL not found!";
+  enable opengl
+  if enabled opengl_video; then
+    enable opengl_video
+  fi
+  if enabled opengl_themepainter; then
+    enable opengl_themepainter
+  fi
+fi
+
+if enabled opengles; then
+    check_pkg_config GLESv2 glesv2 GLES2/gl2.h glGetError &&
+    check_pkg_config EGL egl EGL/egl.h eglGetProcAddress ||
+    die "ERROR: OpenGL ES not found!";
+    check_lib GLESv2 GLES2/gl2.h glGetError "-lGLESv2" &&
+    check_lib EGL EGL/egl.h eglGetProcAddress "-lEGL" ||
+    die "ERROR: OpenGL ES not found!";
+fi
+
 if enabled libmpeg2external; then
     check_lib libmpeg2external mpeg2dec/mpeg2.h mpeg2_init -lmpeg2 || disable libmpeg2external
     if disabled libmpeg2external; then
@@ -6547,9 +6563,9 @@
 # Disable OpenGL ES support if Qt does not support it
 QTCONF=$(pkg-config --variable=qt_config Qt5Core || pkg-config --variable=qt_config QtCore)
 if echo "$QTCONF" | grep -qv opengles2 ; then
-    if test $target_os != android ; then
-        disable opengles
-    fi
+    true
+else
+    die "ERROR: Qt not provides OpenGL ES support"
 fi
 
 if enabled firewire; then
@@ -7397,8 +7413,9 @@
   fi
 fi
   echo "OpenGL support            ${opengl-no}"
-  echo "OpenGL video              ${opengl_video-no}"
-  echo "OpenGL ThemePainter       ${opengl_themepainter-no}"
+  echo "  OpenGL ES 2.0           ${opengles-no}"
+  echo "  OpenGL video            ${opengl_video-no}"
+  echo "  OpenGL ThemePainter     ${opengl_themepainter-no}"
   if test x"$target_os" = x"darwin" ; then
     echo "VDA support               ${vda-no}"
   fi
@@ -7420,7 +7437,6 @@
 fi
 echo "libdns_sd (Bonjour)       ${libdns_sd-no}"
 echo "libcrypto                 ${libcrypto-no}"
-echo "OpenGL ES 2.0             ${opengles-no}"
 if enabled libbluray_external; then
     echo "bluray support            yes (system)"
 else
