diff -Naur mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20190105-g3382e1a-old/mythtv/libs/libmythtv/tv_play.cpp	2019-01-05 12:45:31.766666667 +0100
+++ mythtv-master-20190105-g3382e1a-new/mythtv/libs/libmythtv/tv_play.cpp	2019-01-05 12:45:33.120000000 +0100
@@ -8393,7 +8393,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
