diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/tv_play.cpp	2020-01-10 09:50:37.100000002 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/tv_play.cpp	2020-01-10 09:50:38.383333334 +0100
@@ -8420,7 +8420,7 @@
     else if (tuned == 3)
         tuneCode = 'T';
     else
-        tuneCode = '_';
+        tuneCode = 'T';
 
     sigDesc = sigDesc + QString(" | (%1%2%3%4%5%6%7%8%9) %10")
               .arg(tuneCode).arg(slock).arg(pat).arg(pmt).arg(mgt).arg(vct)
