diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/osd.cpp mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/osd.cpp	2020-01-10 09:50:38.236666668 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/osd.cpp	2020-01-10 09:50:39.586666668 +0100
@@ -565,7 +565,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
