diff -Naur mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/mythplayer.cpp mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/mythplayer.cpp
--- mythtv-30-20200110-ga27754a-old/mythtv/libs/libmythtv/mythplayer.cpp	2020-01-10 09:50:43.056666668 +0100
+++ mythtv-30-20200110-ga27754a-new/mythtv/libs/libmythtv/mythplayer.cpp	2020-01-10 09:50:44.180000001 +0100
@@ -5130,13 +5130,13 @@
     if (decoder)
         infoMap["videocodecdesc"] = decoder->GetRawEncodingType();
     infoMap["videowidth"]     = QString::number(width);
-    infoMap["videoheight"]    = QString::number(height);
+    bool interlaced = is_interlaced(m_scan);
+    infoMap["videoheight"]    = QString::number(height) + (interlaced ? "i" : "");
     infoMap["videoframerate"] = QString::number(video_frame_rate, 'f', 2);
 
     if (width < 640)
         return;
 
-    bool interlaced = is_interlaced(m_scan);
     if (width == 1920 || height == 1080 || height == 1088)
         infoMap["videodescrip"] = interlaced ? "HD_1080_I" : "HD_1080_P";
     else if ((width == 1280 || height == 720) && !interlaced)
