diff -Naur mythtv-30-20200110-ga27754a-old/mythplugins/mythmusic/mythmusic/cdrip.cpp mythtv-30-20200110-ga27754a-new/mythplugins/mythmusic/mythmusic/cdrip.cpp
--- mythtv-30-20200110-ga27754a-old/mythplugins/mythmusic/mythmusic/cdrip.cpp	2020-01-10 09:50:56.283333334 +0100
+++ mythtv-30-20200110-ga27754a-new/mythplugins/mythmusic/mythmusic/cdrip.cpp	2020-01-10 09:50:57.330000001 +0100
@@ -592,6 +592,18 @@
     QString command = "rm -f " + GetConfDir() + "/tmp/RipTemp/*";
     myth_system(command);
 
+    if (m_scanThread)
+    {
+        m_scanThread->wait();
+        delete m_scanThread;
+    }
+
+    if (m_ejectThread)
+    {
+        m_ejectThread->wait();
+        delete m_ejectThread;
+    }
+
     if (m_decoder)
         delete m_decoder;
 
@@ -790,6 +802,7 @@
 
 void Ripper::ScanFinished()
 {
+    m_scanThread->wait();
     delete m_scanThread;
     m_scanThread = nullptr;
 
@@ -1196,6 +1209,7 @@
 
 void Ripper::EjectFinished()
 {
+    m_ejectThread->wait();
     delete m_ejectThread;
     m_ejectThread = nullptr;
 
