diff -Naur libaacs-0.6.0-old/configure.ac libaacs-0.6.0-new/configure.ac
--- libaacs-0.6.0-old/configure.ac	2013-09-27 21:10:25.000000000 +0200
+++ libaacs-0.6.0-new/configure.ac	2013-09-27 21:29:03.207513565 +0200
@@ -145,9 +145,6 @@
 [-Wredundant-decls -Wcast-qual -Wwrite-strings -Wtype-limits -Wundef ]dnl
 [-Wmissing-prototypes -Wshadow])
 
-CC_CHECK_CFLAGS_APPEND([-Werror=implicit-function-declaration ]dnl
-                       [-Werror-implicit-function-declaration],
-                       [break;])
 
 CC_CHECK_WERROR
 AS_IF([test "x$enable_werror" = "xyes"], [
