diff -Naur perl-5.12.3-old/ext/Errno/Errno_pm.PL perl-5.12.3-new/ext/Errno/Errno_pm.PL
--- perl-5.12.3-old/ext/Errno/Errno_pm.PL	2011-01-26 17:59:36.000000000 -0800
+++ perl-5.12.3-new/ext/Errno/Errno_pm.PL	2011-01-26 18:00:06.000000000 -0800
@@ -344,10 +344,6 @@
 use Config;
 use strict;
 
-"\$Config{'archname'}-\$Config{'osvers'}" eq
-"$Config{'archname'}-$Config{'osvers'}" or
-	die "Errno architecture ($Config{'archname'}-$Config{'osvers'}) does not match executable architecture (\$Config{'archname'}-\$Config{'osvers'})";
-
 \$VERSION = "$VERSION";
 \$VERSION = eval \$VERSION;
 \@ISA = qw(Exporter);
