diff -Naur makedepend-1.0.4-old/main.c makedepend-1.0.4-new/main.c
--- makedepend-1.0.4-old/main.c	2012-03-07 21:54:20.000000000 -0800
+++ makedepend-1.0.4-new/main.c	2012-03-08 07:22:56.000000000 -0800
@@ -422,10 +422,12 @@
 		}
 	    }
 #else /* !__UNIXOS2__, does not use INCLUDEDIR at all */
+#ifdef INCLUDEDIR
 	    if (incp >= includedirs + MAXDIRS)
 		fatalerr("Too many -I flags.\n");
 	    *incp++ = INCLUDEDIR;
 #endif
+#endif
 
 #ifdef EXTRAINCDIR
 	    if (incp >= includedirs + MAXDIRS)
diff -Naur makedepend-1.0.4-old/Makefile.am makedepend-1.0.4-new/Makefile.am
--- makedepend-1.0.4-old/Makefile.am	2012-03-07 21:54:20.000000000 -0800
+++ makedepend-1.0.4-new/Makefile.am	2012-03-08 07:22:56.000000000 -0800
@@ -13,7 +13,7 @@
                      def.h \
                      imakemdep.h
 
-AM_CPPFLAGS = -DOBJSUFFIX=\".${OBJEXT}\" -DINCLUDEDIR=\"/usr/include\"
+AM_CPPFLAGS = -DOBJSUFFIX=\".${OBJEXT}\"
 AM_CFLAGS = $(CWARNFLAGS)
 
 EXTRA_DIST = autogen.sh makedepend.man
diff -Naur makedepend-1.0.4-old/Makefile.in makedepend-1.0.4-new/Makefile.in
--- makedepend-1.0.4-old/Makefile.in	2012-03-07 21:54:26.000000000 -0800
+++ makedepend-1.0.4-new/Makefile.in	2012-03-08 07:22:56.000000000 -0800
@@ -250,7 +250,7 @@
                      def.h \
                      imakemdep.h
 
-AM_CPPFLAGS = -DOBJSUFFIX=\".${OBJEXT}\" -DINCLUDEDIR=\"/usr/include\"
+AM_CPPFLAGS = -DOBJSUFFIX=\".${OBJEXT}\"
 AM_CFLAGS = $(CWARNFLAGS)
 EXTRA_DIST = autogen.sh makedepend.man ChangeLog \
 	tests/28045/makedep.sh tests/28045/foo.cpp
