diff -Naur List-MoreUtils-0.25_02-old/Makefile.PL List-MoreUtils-0.25_02-new/Makefile.PL
--- List-MoreUtils-0.25_02-old/Makefile.PL	2009-05-31 14:27:24.000000000 -0700
+++ List-MoreUtils-0.25_02-new/Makefile.PL	2012-03-14 18:27:20.000000000 -0700
@@ -51,9 +51,9 @@
 
 exit if defined $no_xs;
 
-print "Testing if you have a C compiler\n";
-
-open F, ">test.c" or no_cc();
-print F "int main() { return 0; }\n";
-close F or no_cc();
-system ("$Config{make} test$Config{obj_ext}") and no_cc();
+#print "Testing if you have a C compiler\n";
+#
+#open F, ">test.c" or no_cc();
+#print F "int main() { return 0; }\n";
+#close F or no_cc();
+#system ("$Config{make} test$Config{obj_ext}") and no_cc();
