diff -Naur perl-5.14.0-old/dist/XSLoader/XSLoader_pm.PL perl-5.14.0-new/dist/XSLoader/XSLoader_pm.PL
--- perl-5.14.0-old/dist/XSLoader/XSLoader_pm.PL	2011-05-18 17:54:36.000000000 -0700
+++ perl-5.14.0-new/dist/XSLoader/XSLoader_pm.PL	2011-05-18 18:06:39.000000000 -0700
@@ -38,7 +38,7 @@
     my $boots = "$module\::bootstrap";
     goto &$boots if defined &$boots;
 
-    goto \&XSLoader::bootstrap_inherit unless $module and defined &dl_load_file;
+    goto \&XSLoader::bootstrap_inherit;
 
     my @modparts = split(/::/,$module);
     my $modfname = $modparts[-1];
