diff -Naur perl-5.14.0-old/ext/Errno/Errno_pm.PL perl-5.14.0-new/ext/Errno/Errno_pm.PL
--- perl-5.14.0-old/ext/Errno/Errno_pm.PL	2011-05-18 18:07:17.000000000 -0700
+++ perl-5.14.0-new/ext/Errno/Errno_pm.PL	2011-05-18 18:09:13.000000000 -0700
@@ -335,10 +335,6 @@
 use Config;
 use strict;
 
-"\$Config{'archname'}-\$Config{'osvers'}" eq
-"$Config{'archname'}-$Config{'osvers'}" or
-	die "Errno architecture ($Config{'archname'}-$Config{'osvers'}) does not match executable architecture (\$Config{'archname'}-\$Config{'osvers'})";
-
 our \$VERSION = "$VERSION";
 \$VERSION = eval \$VERSION;
 our \@ISA = 'Exporter';
