diff --speed-large-files --no-dereference --minimal -Naur node-20.2.0/configure.py node-20.2.0/configure.py
--- node-20.2.0/configure.py	2023-05-15 16:59:31.000000000 +0200
+++ node-20.2.0/configure.py	2023-05-18 19:44:23.543425577 +0200
@@ -1260,9 +1260,7 @@
 
   o['variables']['want_separate_host_toolset'] = int(cross_compiling)
 
-  # Enable branch protection for arm64
   if target_arch == 'arm64':
-    o['cflags']+=['-msign-return-address=all']
     o['variables']['arm_fpu'] = options.arm_fpu or 'neon'
 
   if options.node_snapshot_main is not None:
