diff -Naur pjproject-2.3-old/build/cc-auto.mak.in pjproject-2.3-new/build/cc-auto.mak.in
--- pjproject-2.3-old/build/cc-auto.mak.in	2013-10-21 09:52:35.000000000 +0200
+++ pjproject-2.3-new/build/cc-auto.mak.in	2015-03-23 12:42:32.510310695 +0100
@@ -2,7 +2,7 @@
 export CXX = @CXX@ -c
 export AR = @AR@
 export AR_FLAGS = @AR_FLAGS@
-export LD = @LD@
+export LD = @CXX@
 export LDOUT = -o 
 export RANLIB = @RANLIB@
 
