diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/a800 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/a800
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/a800	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/a800	2012-12-10 17:00:20.000000000 -0800
@@ -1,17 +1,17 @@
 # table a800, type: UNKNOWN
 0x0201 KEY_MODE
 0x0200 KEY_POWER2
-0x0205 KEY_1
-0x0206 KEY_2
-0x0207 KEY_3
-0x0209 KEY_4
-0x020a KEY_5
-0x020b KEY_6
-0x020d KEY_7
-0x020e KEY_8
-0x020f KEY_9
+0x0205 KEY_NUMERIC_1
+0x0206 KEY_NUMERIC_2
+0x0207 KEY_NUMERIC_3
+0x0209 KEY_NUMERIC_4
+0x020a KEY_NUMERIC_5
+0x020b KEY_NUMERIC_6
+0x020d KEY_NUMERIC_7
+0x020e KEY_NUMERIC_8
+0x020f KEY_NUMERIC_9
 0x0212 KEY_LEFT
-0x0211 KEY_0
+0x0211 KEY_NUMERIC_0
 0x0213 KEY_RIGHT
 0x0217 KEY_CAMERA
 0x0210 KEY_LAST
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/adstech_dvb_t_pci v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/adstech_dvb_t_pci
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/adstech_dvb_t_pci	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/adstech_dvb_t_pci	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table adstech_dvb_t_pci, type: UNKNOWN
-0x4d KEY_0
-0x57 KEY_1
-0x4f KEY_2
-0x53 KEY_3
-0x56 KEY_4
-0x4e KEY_5
-0x5e KEY_6
-0x54 KEY_7
-0x4c KEY_8
-0x5c KEY_9
+0x4d KEY_NUMERIC_0
+0x57 KEY_NUMERIC_1
+0x4f KEY_NUMERIC_2
+0x53 KEY_NUMERIC_3
+0x56 KEY_NUMERIC_4
+0x4e KEY_NUMERIC_5
+0x5e KEY_NUMERIC_6
+0x54 KEY_NUMERIC_7
+0x4c KEY_NUMERIC_8
+0x5c KEY_NUMERIC_9
 0x5b KEY_POWER
 0x5f KEY_MUTE
 0x55 KEY_GOTO
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/af9005 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/af9005
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/af9005	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/af9005	2012-12-10 17:00:20.000000000 -0800
@@ -5,17 +5,17 @@
 0x017f KEY_MUTE
 0x01bf KEY_VOLUMEDOWN
 0x013f KEY_CHANNELDOWN
-0x01df KEY_1
-0x015f KEY_2
-0x019f KEY_3
-0x011f KEY_4
-0x01ef KEY_5
-0x016f KEY_6
-0x01af KEY_7
-0x0127 KEY_8
-0x0107 KEY_9
+0x01df KEY_NUMERIC_1
+0x015f KEY_NUMERIC_2
+0x019f KEY_NUMERIC_3
+0x011f KEY_NUMERIC_4
+0x01ef KEY_NUMERIC_5
+0x016f KEY_NUMERIC_6
+0x01af KEY_NUMERIC_7
+0x0127 KEY_NUMERIC_8
+0x0107 KEY_NUMERIC_9
 0x01cf KEY_ZOOM
-0x014f KEY_0
+0x014f KEY_NUMERIC_0
 0x018f KEY_GOTO
 0x00bd KEY_POWER
 0x007d KEY_VOLUMEUP
@@ -23,15 +23,15 @@
 0x009d KEY_MUTE
 0x005d KEY_VOLUMEDOWN
 0x00dd KEY_CHANNELDOWN
-0x00ad KEY_1
-0x006d KEY_2
-0x00ed KEY_3
-0x008d KEY_4
-0x004d KEY_5
-0x00cd KEY_6
-0x00b5 KEY_7
-0x0075 KEY_8
-0x00f5 KEY_9
+0x00ad KEY_NUMERIC_1
+0x006d KEY_NUMERIC_2
+0x00ed KEY_NUMERIC_3
+0x008d KEY_NUMERIC_4
+0x004d KEY_NUMERIC_5
+0x00cd KEY_NUMERIC_6
+0x00b5 KEY_NUMERIC_7
+0x0075 KEY_NUMERIC_8
+0x00f5 KEY_NUMERIC_9
 0x0095 KEY_ZOOM
-0x0055 KEY_0
+0x0055 KEY_NUMERIC_0
 0x00d5 KEY_GOTO
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/alink_dtu_m v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/alink_dtu_m
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/alink_dtu_m	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/alink_dtu_m	2012-12-10 17:00:20.000000000 -0800
@@ -1,19 +1,19 @@
 # table alink_dtu_m, type: NEC
 0x0800 KEY_VOLUMEUP
-0x0801 KEY_1
-0x0802 KEY_3
-0x0803 KEY_7
-0x0804 KEY_9
+0x0801 KEY_NUMERIC_1
+0x0802 KEY_NUMERIC_3
+0x0803 KEY_NUMERIC_7
+0x0804 KEY_NUMERIC_9
 0x0805 KEY_NEW
-0x0806 KEY_0
+0x0806 KEY_NUMERIC_0
 0x0807 KEY_CHANNEL
-0x080d KEY_5
-0x080f KEY_2
+0x080d KEY_NUMERIC_5
+0x080f KEY_NUMERIC_2
 0x0812 KEY_POWER2
 0x0814 KEY_CHANNELUP
 0x0816 KEY_VOLUMEDOWN
-0x0818 KEY_6
+0x0818 KEY_NUMERIC_6
 0x081a KEY_MUTE
-0x081b KEY_8
-0x081c KEY_4
+0x081b KEY_NUMERIC_8
+0x081c KEY_NUMERIC_4
 0x081d KEY_CHANNELDOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/anysee v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/anysee
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/anysee	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/anysee	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table anysee, type: NEC
-0x0800 KEY_0
-0x0801 KEY_1
-0x0802 KEY_2
-0x0803 KEY_3
-0x0804 KEY_4
-0x0805 KEY_5
-0x0806 KEY_6
-0x0807 KEY_7
-0x0808 KEY_8
-0x0809 KEY_9
+0x0800 KEY_NUMERIC_0
+0x0801 KEY_NUMERIC_1
+0x0802 KEY_NUMERIC_2
+0x0803 KEY_NUMERIC_3
+0x0804 KEY_NUMERIC_4
+0x0805 KEY_NUMERIC_5
+0x0806 KEY_NUMERIC_6
+0x0807 KEY_NUMERIC_7
+0x0808 KEY_NUMERIC_8
+0x0809 KEY_NUMERIC_9
 0x080a KEY_POWER2
 0x080b KEY_VIDEO
 0x080c KEY_CHANNEL
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/apac_viewcomp v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/apac_viewcomp
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/apac_viewcomp	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/apac_viewcomp	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table apac_viewcomp, type: UNKNOWN
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
-0x00 KEY_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
+0x00 KEY_NUMERIC_0
 0x17 KEY_LAST
 0x0a KEY_LIST
 0x1c KEY_TUNER
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/asus_pc39 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/asus_pc39
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/asus_pc39	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/asus_pc39	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table asus_pc39, type: RC5
-0x082a KEY_0
-0x0816 KEY_1
-0x0812 KEY_2
-0x0814 KEY_3
-0x0836 KEY_4
-0x0832 KEY_5
-0x0834 KEY_6
-0x080e KEY_7
-0x080a KEY_8
-0x080c KEY_9
+0x082a KEY_NUMERIC_0
+0x0816 KEY_NUMERIC_1
+0x0812 KEY_NUMERIC_2
+0x0814 KEY_NUMERIC_3
+0x0836 KEY_NUMERIC_4
+0x0832 KEY_NUMERIC_5
+0x0834 KEY_NUMERIC_6
+0x080e KEY_NUMERIC_7
+0x080a KEY_NUMERIC_8
+0x080c KEY_NUMERIC_9
 0x0801 KEY_RADIO
 0x083c KEY_MENU
 0x0815 KEY_VOLUMEUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/asus_ps3_100 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/asus_ps3_100
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/asus_ps3_100	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/asus_ps3_100	2012-12-10 17:00:20.000000000 -0800
@@ -7,16 +7,16 @@
 0x080d KEY_YELLOW
 0x0806 KEY_BLUE
 0x0807 KEY_GREEN
-0x082a KEY_0
-0x0816 KEY_1
-0x0812 KEY_2
-0x0814 KEY_3
-0x0836 KEY_4
-0x0832 KEY_5
-0x0834 KEY_6
-0x080e KEY_7
-0x080a KEY_8
-0x080c KEY_9
+0x082a KEY_NUMERIC_0
+0x0816 KEY_NUMERIC_1
+0x0812 KEY_NUMERIC_2
+0x0814 KEY_NUMERIC_3
+0x0836 KEY_NUMERIC_4
+0x0832 KEY_NUMERIC_5
+0x0834 KEY_NUMERIC_6
+0x080e KEY_NUMERIC_7
+0x080a KEY_NUMERIC_8
+0x080c KEY_NUMERIC_9
 0x0815 KEY_VOLUMEUP
 0x0826 KEY_VOLUMEDOWN
 0x0835 KEY_CHANNELUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/ati_x10 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/ati_x10
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/ati_x10	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/ati_x10	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table ati_x10, type: OTHER
-0x0d KEY_1
-0x0e KEY_2
-0x0f KEY_3
-0x10 KEY_4
-0x11 KEY_5
-0x12 KEY_6
-0x13 KEY_7
-0x14 KEY_8
-0x15 KEY_9
-0x17 KEY_0
+0x0d KEY_NUMERIC_1
+0x0e KEY_NUMERIC_2
+0x0f KEY_NUMERIC_3
+0x10 KEY_NUMERIC_4
+0x11 KEY_NUMERIC_5
+0x12 KEY_NUMERIC_6
+0x13 KEY_NUMERIC_7
+0x14 KEY_NUMERIC_8
+0x15 KEY_NUMERIC_9
+0x17 KEY_NUMERIC_0
 0x00 KEY_A
 0x01 KEY_B
 0x19 KEY_C
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table avermedia, type: UNKNOWN
-0x28 KEY_1
-0x18 KEY_2
-0x38 KEY_3
-0x24 KEY_4
-0x14 KEY_5
-0x34 KEY_6
-0x2c KEY_7
-0x1c KEY_8
-0x3c KEY_9
-0x22 KEY_0
+0x28 KEY_NUMERIC_1
+0x18 KEY_NUMERIC_2
+0x38 KEY_NUMERIC_3
+0x24 KEY_NUMERIC_4
+0x14 KEY_NUMERIC_5
+0x34 KEY_NUMERIC_6
+0x2c KEY_NUMERIC_7
+0x1c KEY_NUMERIC_8
+0x3c KEY_NUMERIC_9
+0x22 KEY_NUMERIC_0
 0x20 KEY_TV
 0x10 KEY_CD
 0x30 KEY_TEXT
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_a16d v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_a16d
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_a16d	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_a16d	2012-12-10 17:00:20.000000000 -0800
@@ -1,17 +1,17 @@
 # table avermedia_a16d, type: UNKNOWN
 0x20 KEY_LIST
 0x00 KEY_POWER
-0x28 KEY_1
-0x18 KEY_2
-0x38 KEY_3
-0x24 KEY_4
-0x14 KEY_5
-0x34 KEY_6
-0x2c KEY_7
-0x1c KEY_8
-0x3c KEY_9
+0x28 KEY_NUMERIC_1
+0x18 KEY_NUMERIC_2
+0x38 KEY_NUMERIC_3
+0x24 KEY_NUMERIC_4
+0x14 KEY_NUMERIC_5
+0x34 KEY_NUMERIC_6
+0x2c KEY_NUMERIC_7
+0x1c KEY_NUMERIC_8
+0x3c KEY_NUMERIC_9
 0x12 KEY_SUBTITLE
-0x22 KEY_0
+0x22 KEY_NUMERIC_0
 0x32 KEY_REWIND
 0x3a KEY_SHUFFLE
 0x02 KEY_PRINT
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_cardbus v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_cardbus
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_cardbus	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_cardbus	2012-12-10 17:00:20.000000000 -0800
@@ -3,19 +3,19 @@
 0x01 KEY_TUNER
 0x03 KEY_TEXT
 0x04 KEY_EPG
-0x05 KEY_1
-0x06 KEY_2
-0x07 KEY_3
+0x05 KEY_NUMERIC_1
+0x06 KEY_NUMERIC_2
+0x07 KEY_NUMERIC_3
 0x08 KEY_AUDIO
-0x09 KEY_4
-0x0a KEY_5
-0x0b KEY_6
+0x09 KEY_NUMERIC_4
+0x0a KEY_NUMERIC_5
+0x0b KEY_NUMERIC_6
 0x0c KEY_ZOOM
-0x0d KEY_7
-0x0e KEY_8
-0x0f KEY_9
+0x0d KEY_NUMERIC_7
+0x0e KEY_NUMERIC_8
+0x0f KEY_NUMERIC_9
 0x10 KEY_PAGEUP
-0x11 KEY_0
+0x11 KEY_NUMERIC_0
 0x12 KEY_INFO
 0x13 KEY_AGAIN
 0x14 KEY_MUTE
@@ -35,7 +35,7 @@
 0x27 KEY_ANGLE
 0x28 KEY_SELECT
 0x29 KEY_BLUE
-0x2a KEY_BACKSPACE
+0x2a KEY_BACK
 0x2b KEY_VIDEO
 0x2c KEY_DOWN
 0x2e KEY_DOT
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_dvbt v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_dvbt
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_dvbt	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_dvbt	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table avermedia_dvbt, type: UNKNOWN
-0x28 KEY_0
-0x22 KEY_1
-0x12 KEY_2
-0x32 KEY_3
-0x24 KEY_4
-0x14 KEY_5
-0x34 KEY_6
-0x26 KEY_7
-0x16 KEY_8
-0x36 KEY_9
+0x28 KEY_NUMERIC_0
+0x22 KEY_NUMERIC_1
+0x12 KEY_NUMERIC_2
+0x32 KEY_NUMERIC_3
+0x24 KEY_NUMERIC_4
+0x14 KEY_NUMERIC_5
+0x34 KEY_NUMERIC_6
+0x26 KEY_NUMERIC_7
+0x16 KEY_NUMERIC_8
+0x36 KEY_NUMERIC_9
 0x20 KEY_VIDEO
 0x10 KEY_TEXT
 0x00 KEY_POWER
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_m135a v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_m135a
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_m135a	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_m135a	2012-12-10 17:00:20.000000000 -0800
@@ -2,16 +2,16 @@
 0x0200 KEY_POWER2
 0x022e KEY_DOT
 0x0201 KEY_MODE
-0x0205 KEY_1
-0x0206 KEY_2
-0x0207 KEY_3
-0x0209 KEY_4
-0x020a KEY_5
-0x020b KEY_6
-0x020d KEY_7
-0x020e KEY_8
-0x020f KEY_9
-0x0211 KEY_0
+0x0205 KEY_NUMERIC_1
+0x0206 KEY_NUMERIC_2
+0x0207 KEY_NUMERIC_3
+0x0209 KEY_NUMERIC_4
+0x020a KEY_NUMERIC_5
+0x020b KEY_NUMERIC_6
+0x020d KEY_NUMERIC_7
+0x020e KEY_NUMERIC_8
+0x020f KEY_NUMERIC_9
+0x0211 KEY_NUMERIC_0
 0x0213 KEY_RIGHT
 0x0212 KEY_LEFT
 0x0217 KEY_SLEEP
@@ -37,17 +37,17 @@
 0x0401 KEY_POWER2
 0x0406 KEY_MUTE
 0x0408 KEY_MODE
-0x0409 KEY_1
-0x040a KEY_2
-0x040b KEY_3
-0x040c KEY_4
-0x040d KEY_5
-0x040e KEY_6
-0x040f KEY_7
-0x0410 KEY_8
-0x0411 KEY_9
+0x0409 KEY_NUMERIC_1
+0x040a KEY_NUMERIC_2
+0x040b KEY_NUMERIC_3
+0x040c KEY_NUMERIC_4
+0x040d KEY_NUMERIC_5
+0x040e KEY_NUMERIC_6
+0x040f KEY_NUMERIC_7
+0x0410 KEY_NUMERIC_8
+0x0411 KEY_NUMERIC_9
 0x044c KEY_DOT
-0x0412 KEY_0
+0x0412 KEY_NUMERIC_0
 0x0407 KEY_REFRESH
 0x0413 KEY_AUDIO
 0x0440 KEY_SCREEN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_m733a_rm_k6 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_m733a_rm_k6
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_m733a_rm_k6	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_m733a_rm_k6	2012-12-10 17:00:20.000000000 -0800
@@ -2,17 +2,17 @@
 0x0401 KEY_POWER2
 0x0406 KEY_MUTE
 0x0408 KEY_MODE
-0x0409 KEY_1
-0x040a KEY_2
-0x040b KEY_3
-0x040c KEY_4
-0x040d KEY_5
-0x040e KEY_6
-0x040f KEY_7
-0x0410 KEY_8
-0x0411 KEY_9
+0x0409 KEY_NUMERIC_1
+0x040a KEY_NUMERIC_2
+0x040b KEY_NUMERIC_3
+0x040c KEY_NUMERIC_4
+0x040d KEY_NUMERIC_5
+0x040e KEY_NUMERIC_6
+0x040f KEY_NUMERIC_7
+0x0410 KEY_NUMERIC_8
+0x0411 KEY_NUMERIC_9
 0x044c KEY_DOT
-0x0412 KEY_0
+0x0412 KEY_NUMERIC_0
 0x0407 KEY_REFRESH
 0x0413 KEY_AUDIO
 0x0440 KEY_SCREEN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_rm_ks v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_rm_ks
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avermedia_rm_ks	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avermedia_rm_ks	2012-12-10 17:00:20.000000000 -0800
@@ -7,16 +7,16 @@
 0x0506 KEY_MUTE
 0x0507 KEY_RIGHT
 0x0508 KEY_RED
-0x0509 KEY_1
-0x050a KEY_2
-0x050b KEY_3
-0x050c KEY_4
-0x050d KEY_5
-0x050e KEY_6
-0x050f KEY_7
-0x0510 KEY_8
-0x0511 KEY_9
-0x0512 KEY_0
+0x0509 KEY_NUMERIC_1
+0x050a KEY_NUMERIC_2
+0x050b KEY_NUMERIC_3
+0x050c KEY_NUMERIC_4
+0x050d KEY_NUMERIC_5
+0x050e KEY_NUMERIC_6
+0x050f KEY_NUMERIC_7
+0x0510 KEY_NUMERIC_8
+0x0511 KEY_NUMERIC_9
+0x0512 KEY_NUMERIC_0
 0x0513 KEY_AUDIO
 0x0515 KEY_EPG
 0x0516 KEY_PLAY
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avertv_303 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avertv_303
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/avertv_303	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/avertv_303	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table avertv_303, type: UNKNOWN
 0x2a KEY_1
-0x32 KEY_2
-0x3a KEY_3
-0x4a KEY_4
-0x52 KEY_5
-0x5a KEY_6
-0x6a KEY_7
-0x72 KEY_8
-0x7a KEY_9
-0x0e KEY_0
+0x32 KEY_NUMERIC_2
+0x3a KEY_NUMERIC_3
+0x4a KEY_NUMERIC_4
+0x52 KEY_NUMERIC_5
+0x5a KEY_NUMERIC_6
+0x6a KEY_NUMERIC_7
+0x72 KEY_NUMERIC_8
+0x7a KEY_NUMERIC_9
+0x0e KEY_NUMERIC_0
 0x02 KEY_POWER
 0x22 KEY_VIDEO
 0x42 KEY_AUDIO
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/azurewave_ad_tu700 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/azurewave_ad_tu700
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/azurewave_ad_tu700	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/azurewave_ad_tu700	2012-12-10 17:00:20.000000000 -0800
@@ -1,17 +1,17 @@
 # table azurewave_ad_tu700, type: NEC
 0x0000 KEY_TAB
-0x0001 KEY_2
+0x0001 KEY_NUMERIC_2
 0x0002 KEY_CHANNELDOWN
-0x0003 KEY_1
+0x0003 KEY_NUMERIC_1
 0x0004 KEY_MENU
 0x0005 KEY_CHANNELUP
-0x0006 KEY_3
+0x0006 KEY_NUMERIC_3
 0x0007 KEY_SLEEP
 0x0008 KEY_VIDEO
-0x0009 KEY_4
+0x0009 KEY_NUMERIC_4
 0x000a KEY_VOLUMEDOWN
 0x000c KEY_CANCEL
-0x000d KEY_7
+0x000d KEY_NUMERIC_7
 0x000e KEY_AGAIN
 0x000f KEY_TEXT
 0x0010 KEY_MUTE
@@ -19,17 +19,17 @@
 0x0012 KEY_FASTFORWARD
 0x0013 KEY_BACK
 0x0014 KEY_PLAY
-0x0015 KEY_0
+0x0015 KEY_NUMERIC_0
 0x0016 KEY_POWER2
 0x0017 KEY_FAVORITES
 0x0018 KEY_RED
-0x0019 KEY_8
+0x0019 KEY_NUMERIC_8
 0x001a KEY_STOP
-0x001b KEY_9
+0x001b KEY_NUMERIC_9
 0x001c KEY_EPG
-0x001d KEY_5
+0x001d KEY_NUMERIC_5
 0x001e KEY_VOLUMEUP
-0x001f KEY_6
+0x001f KEY_NUMERIC_6
 0x0040 KEY_REWIND
 0x0041 KEY_PREVIOUS
 0x0042 KEY_NEXT
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/behold v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/behold
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/behold	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/behold	2012-12-10 17:00:20.000000000 -0800
@@ -1,17 +1,17 @@
 # table behold, type: NEC
 0x6b861c KEY_TUNER
 0x6b8612 KEY_POWER
-0x6b8601 KEY_1
-0x6b8602 KEY_2
-0x6b8603 KEY_3
-0x6b8604 KEY_4
-0x6b8605 KEY_5
-0x6b8606 KEY_6
-0x6b8607 KEY_7
-0x6b8608 KEY_8
-0x6b8609 KEY_9
+0x6b8601 KEY_NUMERIC_1
+0x6b8602 KEY_NUMERIC_2
+0x6b8603 KEY_NUMERIC_3
+0x6b8604 KEY_NUMERIC_4
+0x6b8605 KEY_NUMERIC_5
+0x6b8606 KEY_NUMERIC_6
+0x6b8607 KEY_NUMERIC_7
+0x6b8608 KEY_NUMERIC_8
+0x6b8609 KEY_NUMERIC_9
 0x6b860a KEY_AGAIN
-0x6b8600 KEY_0
+0x6b8600 KEY_NUMERIC_0
 0x6b8617 KEY_MODE
 0x6b8614 KEY_SCREEN
 0x6b8610 KEY_ZOOM
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/behold_columbus v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/behold_columbus
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/behold_columbus	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/behold_columbus	2012-12-10 17:00:20.000000000 -0800
@@ -3,20 +3,20 @@
 0x11 KEY_VIDEO
 0x1C KEY_TUNER
 0x12 KEY_POWER
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
 0x0D KEY_SETUP
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
 0x19 KEY_CAMERA
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
 0x10 KEY_ZOOM
 0x0A KEY_AGAIN
-0x00 KEY_0
+0x00 KEY_NUMERIC_0
 0x0B KEY_CHANNELUP
 0x0C KEY_VOLUMEUP
 0x1B KEY_TIME
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/budget_ci_old v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/budget_ci_old
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/budget_ci_old	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/budget_ci_old	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table budget_ci_old, type: UNKNOWN
-0x00 KEY_0
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
+0x00 KEY_NUMERIC_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
 0x0a KEY_ENTER
 0x0b KEY_RED
 0x0c KEY_POWER
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/cinergy v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/cinergy
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/cinergy	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/cinergy	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table cinergy, type: UNKNOWN
-0x00 KEY_0
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
+0x00 KEY_NUMERIC_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
 0x0a KEY_POWER
 0x0b KEY_MEDIA
 0x0c KEY_ZOOM
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/cinergy_1400 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/cinergy_1400
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/cinergy_1400	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/cinergy_1400	2012-12-10 17:00:20.000000000 -0800
@@ -1,15 +1,15 @@
 # table cinergy_1400, type: UNKNOWN
 0x01 KEY_POWER
-0x02 KEY_1
-0x03 KEY_2
-0x04 KEY_3
-0x05 KEY_4
-0x06 KEY_5
-0x07 KEY_6
-0x08 KEY_7
-0x09 KEY_8
-0x0a KEY_9
-0x0c KEY_0
+0x02 KEY_NUMERIC_1
+0x03 KEY_NUMERIC_2
+0x04 KEY_NUMERIC_3
+0x05 KEY_NUMERIC_4
+0x06 KEY_NUMERIC_5
+0x07 KEY_NUMERIC_6
+0x08 KEY_NUMERIC_7
+0x09 KEY_NUMERIC_8
+0x0a KEY_NUMERIC_9
+0x0c KEY_NUMERIC_0
 0x0b KEY_VIDEO
 0x0d KEY_REFRESH
 0x0e KEY_SELECT
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/cinergyt2 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/cinergyt2
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/cinergyt2	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/cinergyt2	2012-12-10 17:00:20.000000000 -0800
@@ -1,15 +1,15 @@
 # table cinergyt2, type: UNKNOWN
 0x0401 KEY_POWER
-0x0402 KEY_1
-0x0403 KEY_2
-0x0404 KEY_3
-0x0405 KEY_4
-0x0406 KEY_5
-0x0407 KEY_6
-0x0408 KEY_7
-0x0409 KEY_8
-0x040a KEY_9
-0x040c KEY_0
+0x0402 KEY_NUMERIC_1
+0x0403 KEY_NUMERIC_2
+0x0404 KEY_NUMERIC_3
+0x0405 KEY_NUMERIC_4
+0x0406 KEY_NUMERIC_5
+0x0407 KEY_NUMERIC_6
+0x0408 KEY_NUMERIC_7
+0x0409 KEY_NUMERIC_8
+0x040a KEY_NUMERIC_9
+0x040c KEY_NUMERIC_0
 0x040b KEY_VIDEO
 0x040d KEY_REFRESH
 0x040e KEY_SELECT
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/d680_dmb v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/d680_dmb
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/d680_dmb	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/d680_dmb	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table d680_dmb, type: UNKNOWN
 0x0038 KEY_UNKNOWN
 0x080c KEY_ZOOM
-0x0800 KEY_0
-0x0001 KEY_1
-0x0802 KEY_2
-0x0003 KEY_3
-0x0804 KEY_4
-0x0005 KEY_5
-0x0806 KEY_6
-0x0007 KEY_7
-0x0808 KEY_8
-0x0009 KEY_9
+0x0800 KEY_NUMERIC_0
+0x0001 KEY_NUMERIC_1
+0x0802 KEY_NUMERIC_2
+0x0003 KEY_NUMERIC_3
+0x0804 KEY_NUMERIC_4
+0x0005 KEY_NUMERIC_5
+0x0806 KEY_NUMERIC_6
+0x0007 KEY_NUMERIC_7
+0x0808 KEY_NUMERIC_8
+0x0009 KEY_NUMERIC_9
 0x000a KEY_MUTE
 0x0829 KEY_BACK
 0x0012 KEY_CHANNELUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dib0700_nec v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dib0700_nec
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dib0700_nec	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dib0700_nec	2012-12-10 17:00:20.000000000 -0800
@@ -1,23 +1,23 @@
 # table dib0700_nec, type: NEC
 0x866b13 KEY_MUTE
 0x866b12 KEY_POWER
-0x866b01 KEY_1
-0x866b02 KEY_2
-0x866b03 KEY_3
-0x866b04 KEY_4
-0x866b05 KEY_5
-0x866b06 KEY_6
-0x866b07 KEY_7
-0x866b08 KEY_8
-0x866b09 KEY_9
-0x866b00 KEY_0
+0x866b01 KEY_NUMERIC_1
+0x866b02 KEY_NUMERIC_2
+0x866b03 KEY_NUMERIC_3
+0x866b04 KEY_NUMERIC_4
+0x866b05 KEY_NUMERIC_5
+0x866b06 KEY_NUMERIC_6
+0x866b07 KEY_NUMERIC_7
+0x866b08 KEY_NUMERIC_8
+0x866b09 KEY_NUMERIC_9
+0x866b00 KEY_NUMERIC_0
 0x866b0d KEY_CHANNELUP
 0x866b19 KEY_CHANNELDOWN
 0x866b10 KEY_VOLUMEUP
 0x866b0c KEY_VOLUMEDOWN
 0x866b0a KEY_CAMERA
 0x866b0b KEY_ZOOM
-0x866b1b KEY_BACKSPACE
+0x866b1b KEY_BACK
 0x866b15 KEY_ENTER
 0x866b1d KEY_UP
 0x866b1e KEY_DOWN
@@ -37,17 +37,17 @@
 0x7a41 KEY_MUTE
 0x4501 KEY_POWER
 0x4502 KEY_MUTE
-0x4503 KEY_1
-0x4504 KEY_2
-0x4505 KEY_3
-0x4506 KEY_4
-0x4507 KEY_5
-0x4508 KEY_6
-0x4509 KEY_7
-0x450a KEY_8
-0x450b KEY_9
+0x4503 KEY_NUMERIC_1
+0x4504 KEY_NUMERIC_2
+0x4505 KEY_NUMERIC_3
+0x4506 KEY_NUMERIC_4
+0x4507 KEY_NUMERIC_5
+0x4508 KEY_NUMERIC_6
+0x4509 KEY_NUMERIC_7
+0x450a KEY_NUMERIC_8
+0x450b KEY_NUMERIC_9
 0x450c KEY_LAST
-0x450d KEY_0
+0x450d KEY_NUMERIC_0
 0x450e KEY_ENTER
 0x450f KEY_RED
 0x4510 KEY_CHANNELUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dib0700_rc5 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dib0700_rc5
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dib0700_rc5	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dib0700_rc5	2012-12-10 17:00:20.000000000 -0800
@@ -6,16 +6,16 @@
 0x0709 KEY_VOLUMEDOWN
 0x0706 KEY_CHANNELUP
 0x070c KEY_CHANNELDOWN
-0x070f KEY_1
-0x0715 KEY_2
-0x0710 KEY_3
-0x0718 KEY_4
-0x071b KEY_5
-0x071e KEY_6
-0x0711 KEY_7
-0x0721 KEY_8
-0x0712 KEY_9
-0x0727 KEY_0
+0x070f KEY_NUMERIC_1
+0x0715 KEY_NUMERIC_2
+0x0710 KEY_NUMERIC_3
+0x0718 KEY_NUMERIC_4
+0x071b KEY_NUMERIC_5
+0x071e KEY_NUMERIC_6
+0x0711 KEY_NUMERIC_7
+0x0721 KEY_NUMERIC_8
+0x0712 KEY_NUMERIC_9
+0x0727 KEY_NUMERIC_0
 0x0724 KEY_SCREEN
 0x072a KEY_TEXT
 0x072d KEY_REWIND
@@ -25,17 +25,17 @@
 0x073c KEY_STOP
 0x073f KEY_CANCEL
 0xeb01 KEY_POWER
-0xeb02 KEY_1
-0xeb03 KEY_2
-0xeb04 KEY_3
-0xeb05 KEY_4
-0xeb06 KEY_5
-0xeb07 KEY_6
-0xeb08 KEY_7
-0xeb09 KEY_8
-0xeb0a KEY_9
+0xeb02 KEY_NUMERIC_1
+0xeb03 KEY_NUMERIC_2
+0xeb04 KEY_NUMERIC_3
+0xeb05 KEY_NUMERIC_4
+0xeb06 KEY_NUMERIC_5
+0xeb07 KEY_NUMERIC_6
+0xeb08 KEY_NUMERIC_7
+0xeb09 KEY_NUMERIC_8
+0xeb0a KEY_NUMERIC_9
 0xeb0b KEY_VIDEO
-0xeb0c KEY_0
+0xeb0c KEY_NUMERIC_0
 0xeb0d KEY_REFRESH
 0xeb0f KEY_EPG
 0xeb10 KEY_UP
@@ -72,17 +72,17 @@
 0xeb54 KEY_PREVIOUS
 0xeb58 KEY_RECORD
 0xeb5c KEY_NEXT
-0x1e00 KEY_0
-0x1e01 KEY_1
-0x1e02 KEY_2
-0x1e03 KEY_3
-0x1e04 KEY_4
-0x1e05 KEY_5
-0x1e06 KEY_6
-0x1e07 KEY_7
-0x1e08 KEY_8
-0x1e09 KEY_9
-0x1e0a KEY_KPASTERISK
+0x1e00 KEY_NUMERIC_0
+0x1e01 KEY_NUMERIC_1
+0x1e02 KEY_NUMERIC_2
+0x1e03 KEY_NUMERIC_3
+0x1e04 KEY_NUMERIC_4
+0x1e05 KEY_NUMERIC_5
+0x1e06 KEY_NUMERIC_6
+0x1e07 KEY_NUMERIC_7
+0x1e08 KEY_NUMERIC_8
+0x1e09 KEY_NUMERIC_9
+0x1e0a KEY_NUMERIC_STAR
 0x1e0b KEY_RED
 0x1e0c KEY_RADIO
 0x1e0d KEY_MENU
@@ -122,16 +122,16 @@
 0x0f4e KEY_PRINT
 0x0840 KEY_SCREEN
 0x0f71 KEY_DOT
-0x0743 KEY_0
-0x0c41 KEY_1
-0x0443 KEY_2
-0x0b7f KEY_3
-0x0e41 KEY_4
-0x0643 KEY_5
-0x097f KEY_6
-0x0d7e KEY_7
-0x057c KEY_8
-0x0a40 KEY_9
+0x0743 KEY_NUMERIC_0
+0x0c41 KEY_NUMERIC_1
+0x0443 KEY_NUMERIC_2
+0x0b7f KEY_NUMERIC_3
+0x0e41 KEY_NUMERIC_4
+0x0643 KEY_NUMERIC_5
+0x097f KEY_NUMERIC_6
+0x0d7e KEY_NUMERIC_7
+0x057c KEY_NUMERIC_8
+0x0a40 KEY_NUMERIC_9
 0x0e4e KEY_CLEAR
 0x047c KEY_CHANNEL
 0x0f41 KEY_LAST
@@ -144,16 +144,16 @@
 0x017d KEY_VOLUMEDOWN
 0x0242 KEY_CHANNELUP
 0x007d KEY_CHANNELDOWN
-0x1d00 KEY_0
-0x1d01 KEY_1
-0x1d02 KEY_2
-0x1d03 KEY_3
-0x1d04 KEY_4
-0x1d05 KEY_5
-0x1d06 KEY_6
-0x1d07 KEY_7
-0x1d08 KEY_8
-0x1d09 KEY_9
+0x1d00 KEY_NUMERIC_0
+0x1d01 KEY_NUMERIC_1
+0x1d02 KEY_NUMERIC_2
+0x1d03 KEY_NUMERIC_3
+0x1d04 KEY_NUMERIC_4
+0x1d05 KEY_NUMERIC_5
+0x1d06 KEY_NUMERIC_6
+0x1d07 KEY_NUMERIC_7
+0x1d08 KEY_NUMERIC_8
+0x1d09 KEY_NUMERIC_9
 0x1d0a KEY_TEXT
 0x1d0d KEY_MENU
 0x1d0f KEY_MUTE
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dibusb v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dibusb
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dibusb	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dibusb	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table dibusb, type: UNKNOWN
 0x0016 KEY_POWER
 0x0010 KEY_MUTE
-0x0003 KEY_1
-0x0001 KEY_2
-0x0006 KEY_3
-0x0009 KEY_4
-0x001d KEY_5
-0x001f KEY_6
-0x000d KEY_7
-0x0019 KEY_8
-0x001b KEY_9
-0x0015 KEY_0
+0x0003 KEY_NUMERIC_1
+0x0001 KEY_NUMERIC_2
+0x0006 KEY_NUMERIC_3
+0x0009 KEY_NUMERIC_4
+0x001d KEY_NUMERIC_5
+0x001f KEY_NUMERIC_6
+0x000d KEY_NUMERIC_7
+0x0019 KEY_NUMERIC_8
+0x001b KEY_NUMERIC_9
+0x0015 KEY_NUMERIC_0
 0x0005 KEY_CHANNELUP
 0x0002 KEY_CHANNELDOWN
 0x001e KEY_VOLUMEUP
@@ -36,16 +36,16 @@
 0x860c KEY_UNKNOWN
 0x860b KEY_EPG
 0x8610 KEY_MUTE
-0x8601 KEY_1
-0x8602 KEY_2
-0x8603 KEY_3
-0x8604 KEY_4
-0x8605 KEY_5
-0x8606 KEY_6
-0x8607 KEY_7
-0x8608 KEY_8
-0x8609 KEY_9
-0x860a KEY_0
+0x8601 KEY_NUMERIC_1
+0x8602 KEY_NUMERIC_2
+0x8603 KEY_NUMERIC_3
+0x8604 KEY_NUMERIC_4
+0x8605 KEY_NUMERIC_5
+0x8606 KEY_NUMERIC_6
+0x8607 KEY_NUMERIC_7
+0x8608 KEY_NUMERIC_8
+0x8609 KEY_NUMERIC_9
+0x860a KEY_NUMERIC_0
 0x8618 KEY_ZOOM
 0x861c KEY_UNKNOWN
 0x8613 KEY_UNKNOWN
@@ -94,19 +94,19 @@
 0x801d KEY_REWIND
 0x801e KEY_FASTFORWARD
 0x801f KEY_NEXT
-0x8040 KEY_1
-0x8041 KEY_2
-0x8042 KEY_3
+0x8040 KEY_NUMERIC_1
+0x8041 KEY_NUMERIC_2
+0x8042 KEY_NUMERIC_3
 0x8043 KEY_CHANNELUP
-0x8044 KEY_4
-0x8045 KEY_5
-0x8046 KEY_6
+0x8044 KEY_NUMERIC_4
+0x8045 KEY_NUMERIC_5
+0x8046 KEY_NUMERIC_6
 0x8047 KEY_CHANNELDOWN
-0x8048 KEY_7
-0x8049 KEY_8
-0x804a KEY_9
+0x8048 KEY_NUMERIC_7
+0x8049 KEY_NUMERIC_8
+0x804a KEY_NUMERIC_9
 0x804b KEY_VOLUMEUP
 0x804c KEY_CLEAR
-0x804d KEY_0
+0x804d KEY_NUMERIC_0
 0x804e KEY_ENTER
 0x804f KEY_VOLUMEDOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/digitalnow_tinytwin v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/digitalnow_tinytwin
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/digitalnow_tinytwin	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/digitalnow_tinytwin	2012-12-10 17:00:20.000000000 -0800
@@ -2,14 +2,14 @@
 0x0000 KEY_MUTE
 0x0001 KEY_VOLUMEUP
 0x0002 KEY_POWER2
-0x0003 KEY_2
-0x0004 KEY_3
-0x0005 KEY_4
-0x0006 KEY_6
-0x0007 KEY_7
-0x0008 KEY_8
+0x0003 KEY_NUMERIC_2
+0x0004 KEY_NUMERIC_3
+0x0005 KEY_NUMERIC_4
+0x0006 KEY_NUMERIC_6
+0x0007 KEY_NUMERIC_7
+0x0008 KEY_NUMERIC_8
 0x0009 KEY_NUMERIC_STAR
-0x000a KEY_0
+0x000a KEY_NUMERIC_0
 0x000b KEY_NUMERIC_POUND
 0x000c KEY_RIGHT
 0x000d KEY_HOMEPAGE
@@ -26,10 +26,10 @@
 0x0019 KEY_BLUE
 0x001a KEY_REWIND
 0x001b KEY_PLAY
-0x001c KEY_5
-0x001d KEY_9
+0x001c KEY_NUMERIC_5
+0x001d KEY_NUMERIC_9
 0x001e KEY_VOLUMEDOWN
-0x001f KEY_1
+0x001f KEY_NUMERIC_1
 0x0040 KEY_STOP
 0x0042 KEY_PAUSE
 0x0043 KEY_SCREEN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/digittrade v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/digittrade
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/digittrade	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/digittrade	2012-12-10 17:00:20.000000000 -0800
@@ -1,9 +1,9 @@
 # table digittrade, type: NEC
-0x0000 KEY_9
+0x0000 KEY_NUMERIC_9
 0x0001 KEY_EPG
 0x0002 KEY_VOLUMEDOWN
 0x0003 KEY_TEXT
-0x0004 KEY_8
+0x0004 KEY_NUMERIC_8
 0x0005 KEY_MUTE
 0x0006 KEY_POWER2
 0x0009 KEY_ZOOM
@@ -11,19 +11,19 @@
 0x000d KEY_SUBTITLE
 0x000e KEY_STOP
 0x0010 KEY_OK
-0x0011 KEY_2
-0x0012 KEY_4
-0x0015 KEY_3
-0x0016 KEY_5
+0x0011 KEY_NUMERIC_2
+0x0012 KEY_NUMERIC_4
+0x0015 KEY_NUMERIC_3
+0x0016 KEY_NUMERIC_5
 0x0017 KEY_CHANNELDOWN
 0x0019 KEY_CHANNELUP
 0x001a KEY_PAUSE
-0x001b KEY_1
+0x001b KEY_NUMERIC_1
 0x001d KEY_AUDIO
 0x001e KEY_PLAY
 0x001f KEY_CAMERA
 0x0040 KEY_VOLUMEUP
-0x0048 KEY_7
-0x004c KEY_6
+0x0048 KEY_NUMERIC_7
+0x004c KEY_NUMERIC_6
 0x004d KEY_PLAYPAUSE
-0x0054 KEY_0
+0x0054 KEY_NUMERIC_0
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/digitv v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/digitv
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/digitv	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/digitv	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table digitv, type: UNKNOWN
-0x5f55 KEY_0
-0x6f55 KEY_1
-0x9f55 KEY_2
-0xaf55 KEY_3
-0x5f56 KEY_4
-0x6f56 KEY_5
-0x9f56 KEY_6
-0xaf56 KEY_7
-0x5f59 KEY_8
-0x6f59 KEY_9
+0x5f55 KEY_NUMERIC_0
+0x6f55 KEY_NUMERIC_1
+0x9f55 KEY_NUMERIC_2
+0xaf55 KEY_NUMERIC_3
+0x5f56 KEY_NUMERIC_4
+0x6f56 KEY_NUMERIC_5
+0x9f56 KEY_NUMERIC_6
+0xaf56 KEY_NUMERIC_7
+0x5f59 KEY_NUMERIC_8
+0x6f59 KEY_NUMERIC_9
 0x9f59 KEY_TV
 0xaf59 KEY_AUX
 0x5f5a KEY_DVD
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dm1105_nec v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dm1105_nec
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dm1105_nec	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dm1105_nec	2012-12-10 17:00:20.000000000 -0800
@@ -1,23 +1,23 @@
 # table dm1105_nec, type: UNKNOWN
 0x0a KEY_POWER2
 0x0c KEY_MUTE
-0x11 KEY_1
-0x12 KEY_2
-0x13 KEY_3
-0x14 KEY_4
-0x15 KEY_5
-0x16 KEY_6
-0x17 KEY_7
-0x18 KEY_8
-0x19 KEY_9
-0x10 KEY_0
+0x11 KEY_NUMERIC_1
+0x12 KEY_NUMERIC_2
+0x13 KEY_NUMERIC_3
+0x14 KEY_NUMERIC_4
+0x15 KEY_NUMERIC_5
+0x16 KEY_NUMERIC_6
+0x17 KEY_NUMERIC_7
+0x18 KEY_NUMERIC_8
+0x19 KEY_NUMERIC_9
+0x10 KEY_NUMERIC_0
 0x1c KEY_CHANNELUP
 0x0f KEY_CHANNELDOWN
 0x1a KEY_VOLUMEUP
 0x0e KEY_VOLUMEDOWN
 0x04 KEY_RECORD
 0x09 KEY_CHANNEL
-0x08 KEY_BACKSPACE
+0x08 KEY_BACK
 0x07 KEY_FASTFORWARD
 0x0b KEY_PAUSE
 0x02 KEY_ESC
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dntv_live_dvb_t v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dntv_live_dvb_t
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dntv_live_dvb_t	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dntv_live_dvb_t	2012-12-10 17:00:20.000000000 -0800
@@ -1,15 +1,15 @@
 # table dntv_live_dvb_t, type: UNKNOWN
 0x00 KEY_ESC
-0x0a KEY_0
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
+0x0a KEY_NUMERIC_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
 0x0b KEY_TUNER
 0x0c KEY_SEARCH
 0x0d KEY_STOP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dntv_live_dvbt_pro v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dntv_live_dvbt_pro
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dntv_live_dvbt_pro	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dntv_live_dvbt_pro	2012-12-10 17:00:20.000000000 -0800
@@ -5,17 +5,17 @@
 0x58 KEY_TUNER
 0x5a KEY_RADIO
 0x59 KEY_DVD
-0x03 KEY_1
-0x01 KEY_2
-0x06 KEY_3
-0x09 KEY_4
-0x1d KEY_5
-0x1f KEY_6
-0x0d KEY_7
-0x19 KEY_8
-0x1b KEY_9
+0x03 KEY_NUMERIC_1
+0x01 KEY_NUMERIC_2
+0x06 KEY_NUMERIC_3
+0x09 KEY_NUMERIC_4
+0x1d KEY_NUMERIC_5
+0x1f KEY_NUMERIC_6
+0x0d KEY_NUMERIC_7
+0x19 KEY_NUMERIC_8
+0x1b KEY_NUMERIC_9
 0x0c KEY_CANCEL
-0x15 KEY_0
+0x15 KEY_NUMERIC_0
 0x4a KEY_CLEAR
 0x13 KEY_BACK
 0x00 KEY_TAB
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dtt200u v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dtt200u
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dtt200u	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dtt200u	2012-12-10 17:00:20.000000000 -0800
@@ -2,18 +2,18 @@
 0x8001 KEY_MUTE
 0x8002 KEY_CHANNELDOWN
 0x8003 KEY_VOLUMEDOWN
-0x8004 KEY_1
-0x8005 KEY_2
-0x8006 KEY_3
-0x8007 KEY_4
-0x8008 KEY_5
-0x8009 KEY_6
-0x800a KEY_7
+0x8004 KEY_NUMERIC_1
+0x8005 KEY_NUMERIC_2
+0x8006 KEY_NUMERIC_3
+0x8007 KEY_NUMERIC_4
+0x8008 KEY_NUMERIC_5
+0x8009 KEY_NUMERIC_6
+0x800a KEY_NUMERIC_7
 0x800c KEY_ZOOM
-0x800d KEY_0
+0x800d KEY_NUMERIC_0
 0x800e KEY_SELECT
 0x8012 KEY_POWER
 0x801a KEY_CHANNELUP
-0x801b KEY_8
+0x801b KEY_NUMERIC_8
 0x801e KEY_VOLUMEUP
-0x801f KEY_9
+0x801f KEY_NUMERIC_9
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dvico_mce v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dvico_mce
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dvico_mce	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dvico_mce	2012-12-10 17:00:20.000000000 -0800
@@ -25,17 +25,17 @@
 0xfe52 KEY_CAMERA
 0xfe5a KEY_TUNER
 0xfe19 KEY_OPEN
-0xfe0b KEY_1
-0xfe17 KEY_2
-0xfe1b KEY_3
-0xfe07 KEY_4
-0xfe50 KEY_5
-0xfe54 KEY_6
-0xfe48 KEY_7
-0xfe4c KEY_8
-0xfe58 KEY_9
+0xfe0b KEY_NUMERIC_1
+0xfe17 KEY_NUMERIC_2
+0xfe1b KEY_NUMERIC_3
+0xfe07 KEY_NUMERIC_4
+0xfe50 KEY_NUMERIC_5
+0xfe54 KEY_NUMERIC_6
+0xfe48 KEY_NUMERIC_7
+0xfe4c KEY_NUMERIC_8
+0xfe58 KEY_NUMERIC_9
 0xfe13 KEY_ANGLE
-0xfe03 KEY_0
+0xfe03 KEY_NUMERIC_0
 0xfe1f KEY_ZOOM
 0xfe43 KEY_REWIND
 0xfe47 KEY_PLAYPAUSE
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dvico_portable v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dvico_portable
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dvico_portable	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dvico_portable	2012-12-10 17:00:20.000000000 -0800
@@ -14,17 +14,17 @@
 0xfc16 KEY_CAMERA
 0xfc40 KEY_TUNER
 0xfc45 KEY_OPEN
-0xfc19 KEY_1
-0xfc18 KEY_2
-0xfc1b KEY_3
-0xfc1a KEY_4
-0xfc58 KEY_5
-0xfc59 KEY_6
-0xfc15 KEY_7
-0xfc14 KEY_8
-0xfc17 KEY_9
+0xfc19 KEY_NUMERIC_1
+0xfc18 KEY_NUMERIC_2
+0xfc1b KEY_NUMERIC_3
+0xfc1a KEY_NUMERIC_4
+0xfc58 KEY_NUMERIC_5
+0xfc59 KEY_NUMERIC_6
+0xfc15 KEY_NUMERIC_7
+0xfc14 KEY_NUMERIC_8
+0xfc17 KEY_NUMERIC_9
 0xfc44 KEY_ANGLE
-0xfc55 KEY_0
+0xfc55 KEY_NUMERIC_0
 0xfc07 KEY_ZOOM
 0xfc0a KEY_REWIND
 0xfc08 KEY_PLAYPAUSE
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dw210x v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dw210x
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/dw210x	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/dw210x	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table dw210x, type: UNKNOWN
 0xf80a KEY_POWER2
 0xf80c KEY_MUTE
-0xf811 KEY_1
-0xf812 KEY_2
-0xf813 KEY_3
-0xf814 KEY_4
-0xf815 KEY_5
-0xf816 KEY_6
-0xf817 KEY_7
-0xf818 KEY_8
-0xf819 KEY_9
-0xf810 KEY_0
+0xf811 KEY_NUMERIC_1
+0xf812 KEY_NUMERIC_2
+0xf813 KEY_NUMERIC_3
+0xf814 KEY_NUMERIC_4
+0xf815 KEY_NUMERIC_5
+0xf816 KEY_NUMERIC_6
+0xf817 KEY_NUMERIC_7
+0xf818 KEY_NUMERIC_8
+0xf819 KEY_NUMERIC_9
+0xf810 KEY_NUMERIC_0
 0xf81c KEY_CHANNELUP
 0xf80f KEY_CHANNELDOWN
 0xf81a KEY_VOLUMEUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/em_terratec v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/em_terratec
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/em_terratec	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/em_terratec	2012-12-10 17:00:20.000000000 -0800
@@ -3,19 +3,19 @@
 0x02 KEY_SELECT
 0x03 KEY_MUTE
 0x04 KEY_POWER
-0x05 KEY_1
-0x06 KEY_2
-0x07 KEY_3
+0x05 KEY_NUMERIC_1
+0x06 KEY_NUMERIC_2
+0x07 KEY_NUMERIC_3
 0x08 KEY_CHANNELUP
-0x09 KEY_4
-0x0a KEY_5
-0x0b KEY_6
+0x09 KEY_NUMERIC_4
+0x0a KEY_NUMERIC_5
+0x0b KEY_NUMERIC_6
 0x0c KEY_CHANNELDOWN
-0x0d KEY_7
-0x0e KEY_8
-0x0f KEY_9
+0x0d KEY_NUMERIC_7
+0x0e KEY_NUMERIC_8
+0x0f KEY_NUMERIC_9
 0x10 KEY_VOLUMEUP
-0x11 KEY_0
+0x11 KEY_NUMERIC_0
 0x12 KEY_MENU
 0x13 KEY_PRINT
 0x14 KEY_VOLUMEDOWN
@@ -24,6 +24,6 @@
 0x19 KEY_REWIND
 0x1a KEY_PLAY
 0x1b KEY_FORWARD
-0x1c KEY_BACKSPACE
+0x1c KEY_BACK
 0x1e KEY_STOP
 0x40 KEY_ZOOM
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/encore_enltv v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/encore_enltv
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/encore_enltv	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/encore_enltv	2012-12-10 17:00:20.000000000 -0800
@@ -4,16 +4,16 @@
 0x00 KEY_VIDEO
 0x01 KEY_AUDIO
 0x02 KEY_CAMERA
-0x1f KEY_1
-0x03 KEY_2
-0x04 KEY_3
-0x05 KEY_4
-0x1c KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x1d KEY_9
-0x0a KEY_0
+0x1f KEY_NUMERIC_1
+0x03 KEY_NUMERIC_2
+0x04 KEY_NUMERIC_3
+0x05 KEY_NUMERIC_4
+0x1c KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x1d KEY_NUMERIC_9
+0x0a KEY_NUMERIC_0
 0x09 KEY_LIST
 0x0b KEY_LAST
 0x14 KEY_HOME
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/encore_enltv2 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/encore_enltv2
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/encore_enltv2	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/encore_enltv2	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table encore_enltv2, type: UNKNOWN
 0x4c KEY_POWER2
 0x4a KEY_TUNER
-0x40 KEY_1
-0x60 KEY_2
-0x50 KEY_3
-0x70 KEY_4
-0x48 KEY_5
-0x68 KEY_6
-0x58 KEY_7
-0x78 KEY_8
-0x44 KEY_9
-0x54 KEY_0
+0x40 KEY_NUMERIC_1
+0x60 KEY_NUMERIC_2
+0x50 KEY_NUMERIC_3
+0x70 KEY_NUMERIC_4
+0x48 KEY_NUMERIC_5
+0x68 KEY_NUMERIC_6
+0x58 KEY_NUMERIC_7
+0x78 KEY_NUMERIC_8
+0x44 KEY_NUMERIC_9
+0x54 KEY_NUMERIC_0
 0x64 KEY_LAST
 0x4e KEY_AGAIN
 0x6c KEY_VIDEO
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/encore_enltv_fm53 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/encore_enltv_fm53
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/encore_enltv_fm53	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/encore_enltv_fm53	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table encore_enltv_fm53, type: UNKNOWN
 0x10 KEY_POWER2
 0x06 KEY_MUTE
-0x09 KEY_1
-0x1d KEY_2
-0x1f KEY_3
-0x19 KEY_4
-0x1b KEY_5
-0x11 KEY_6
-0x17 KEY_7
-0x12 KEY_8
-0x16 KEY_9
-0x48 KEY_0
+0x09 KEY_NUMERIC_1
+0x1d KEY_NUMERIC_2
+0x1f KEY_NUMERIC_3
+0x19 KEY_NUMERIC_4
+0x1b KEY_NUMERIC_5
+0x11 KEY_NUMERIC_6
+0x17 KEY_NUMERIC_7
+0x12 KEY_NUMERIC_8
+0x16 KEY_NUMERIC_9
+0x48 KEY_NUMERIC_0
 0x04 KEY_LIST
 0x40 KEY_LAST
 0x02 KEY_MODE
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/eztv v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/eztv
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/eztv	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/eztv	2012-12-10 17:00:20.000000000 -0800
@@ -27,16 +27,16 @@
 0x2c KEY_STOP
 0x2d KEY_PLAY
 0x2e KEY_CAMERA
-0x00 KEY_0
-0x05 KEY_1
-0x06 KEY_2
-0x07 KEY_3
-0x09 KEY_4
-0x0a KEY_5
-0x0b KEY_6
-0x0d KEY_7
-0x0e KEY_8
-0x0f KEY_9
+0x00 KEY_NUMERIC_0
+0x05 KEY_NUMERIC_1
+0x06 KEY_NUMERIC_2
+0x07 KEY_NUMERIC_3
+0x09 KEY_NUMERIC_4
+0x0a KEY_NUMERIC_5
+0x0b KEY_NUMERIC_6
+0x0d KEY_NUMERIC_7
+0x0e KEY_NUMERIC_8
+0x0f KEY_NUMERIC_9
 0x2a KEY_VOLUMEUP
 0x11 KEY_VOLUMEDOWN
 0x18 KEY_CHANNELUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/flydvb v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/flydvb
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/flydvb	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/flydvb	2012-12-10 17:00:20.000000000 -0800
@@ -1,17 +1,17 @@
 # table flydvb, type: UNKNOWN
 0x01 KEY_ZOOM
 0x00 KEY_POWER
-0x03 KEY_1
-0x04 KEY_2
-0x05 KEY_3
-0x07 KEY_4
-0x08 KEY_5
-0x09 KEY_6
-0x0b KEY_7
-0x0c KEY_8
-0x0d KEY_9
+0x03 KEY_NUMERIC_1
+0x04 KEY_NUMERIC_2
+0x05 KEY_NUMERIC_3
+0x07 KEY_NUMERIC_4
+0x08 KEY_NUMERIC_5
+0x09 KEY_NUMERIC_6
+0x0b KEY_NUMERIC_7
+0x0c KEY_NUMERIC_8
+0x0d KEY_NUMERIC_9
 0x06 KEY_AGAIN
-0x0f KEY_0
+0x0f KEY_NUMERIC_0
 0x10 KEY_MUTE
 0x02 KEY_RADIO
 0x1b KEY_LANGUAGE
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/flyvideo v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/flyvideo
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/flyvideo	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/flyvideo	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table flyvideo, type: UNKNOWN
-0x0f KEY_0
-0x03 KEY_1
-0x04 KEY_2
-0x05 KEY_3
-0x07 KEY_4
-0x08 KEY_5
-0x09 KEY_6
-0x0b KEY_7
-0x0c KEY_8
-0x0d KEY_9
+0x0f KEY_NUMERIC_0
+0x03 KEY_NUMERIC_1
+0x04 KEY_NUMERIC_2
+0x05 KEY_NUMERIC_3
+0x07 KEY_NUMERIC_4
+0x08 KEY_NUMERIC_5
+0x09 KEY_NUMERIC_6
+0x0b KEY_NUMERIC_7
+0x0c KEY_NUMERIC_8
+0x0d KEY_NUMERIC_9
 0x0e KEY_MODE
 0x11 KEY_VIDEO
 0x15 KEY_AUDIO
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/fusionhdtv_mce v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/fusionhdtv_mce
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/fusionhdtv_mce	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/fusionhdtv_mce	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table fusionhdtv_mce, type: UNKNOWN
-0x0b KEY_1
-0x17 KEY_2
-0x1b KEY_3
-0x07 KEY_4
-0x50 KEY_5
-0x54 KEY_6
-0x48 KEY_7
-0x4c KEY_8
-0x58 KEY_9
-0x03 KEY_0
+0x0b KEY_NUMERIC_1
+0x17 KEY_NUMERIC_2
+0x1b KEY_NUMERIC_3
+0x07 KEY_NUMERIC_4
+0x50 KEY_NUMERIC_5
+0x54 KEY_NUMERIC_6
+0x48 KEY_NUMERIC_7
+0x4c KEY_NUMERIC_8
+0x58 KEY_NUMERIC_9
+0x03 KEY_NUMERIC_0
 0x5e KEY_OK
 0x51 KEY_UP
 0x53 KEY_DOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/gadmei_rm008z v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/gadmei_rm008z
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/gadmei_rm008z	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/gadmei_rm008z	2012-12-10 17:00:20.000000000 -0800
@@ -5,18 +5,18 @@
 0x0e KEY_VIDEO
 0x0b KEY_AUDIO
 0x0f KEY_RADIO
-0x00 KEY_1
-0x01 KEY_2
-0x02 KEY_3
-0x03 KEY_4
-0x04 KEY_5
-0x05 KEY_6
-0x06 KEY_7
-0x07 KEY_8
-0x08 KEY_9
-0x09 KEY_0
+0x00 KEY_NUMERIC_1
+0x01 KEY_NUMERIC_2
+0x02 KEY_NUMERIC_3
+0x03 KEY_NUMERIC_4
+0x04 KEY_NUMERIC_5
+0x05 KEY_NUMERIC_6
+0x06 KEY_NUMERIC_7
+0x07 KEY_NUMERIC_8
+0x08 KEY_NUMERIC_9
+0x09 KEY_NUMERIC_0
 0x0a KEY_INFO
-0x1c KEY_BACKSPACE
+0x1c KEY_BACK
 0x0d KEY_PLAY
 0x1e KEY_CAMERA
 0x1a KEY_RECORD
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/genius_tvgo_a11mce v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/genius_tvgo_a11mce
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/genius_tvgo_a11mce	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/genius_tvgo_a11mce	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table genius_tvgo_a11mce, type: UNKNOWN
-0x48 KEY_0
-0x09 KEY_1
-0x1d KEY_2
-0x1f KEY_3
-0x19 KEY_4
-0x1b KEY_5
-0x11 KEY_6
-0x17 KEY_7
-0x12 KEY_8
-0x16 KEY_9
+0x48 KEY_NUMERIC_0
+0x09 KEY_NUMERIC_1
+0x1d KEY_NUMERIC_2
+0x1f KEY_NUMERIC_3
+0x19 KEY_NUMERIC_4
+0x1b KEY_NUMERIC_5
+0x11 KEY_NUMERIC_6
+0x17 KEY_NUMERIC_7
+0x12 KEY_NUMERIC_8
+0x16 KEY_NUMERIC_9
 0x54 KEY_RECORD
 0x06 KEY_MUTE
 0x10 KEY_POWER
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/gotview7135 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/gotview7135
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/gotview7135	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/gotview7135	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table gotview7135, type: UNKNOWN
 0x11 KEY_POWER
 0x35 KEY_TV
-0x1b KEY_0
-0x29 KEY_1
-0x19 KEY_2
-0x39 KEY_3
-0x1f KEY_4
-0x2c KEY_5
-0x21 KEY_6
-0x24 KEY_7
-0x18 KEY_8
-0x2b KEY_9
+0x1b KEY_NUMERIC_0
+0x29 KEY_NUMERIC_1
+0x19 KEY_NUMERIC_2
+0x39 KEY_NUMERIC_3
+0x1f KEY_NUMERIC_4
+0x2c KEY_NUMERIC_5
+0x21 KEY_NUMERIC_6
+0x24 KEY_NUMERIC_7
+0x18 KEY_NUMERIC_8
+0x2b KEY_NUMERIC_9
 0x3b KEY_AGAIN
 0x06 KEY_AUDIO
 0x31 KEY_PRINT
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/haupp v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/haupp
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/haupp	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/haupp	2012-12-10 17:00:20.000000000 -0800
@@ -1,15 +1,15 @@
 # table haupp, type: UNKNOWN
-0x1e00 KEY_0
-0x1e01 KEY_1
-0x1e02 KEY_2
-0x1e03 KEY_3
-0x1e04 KEY_4
-0x1e05 KEY_5
-0x1e06 KEY_6
-0x1e07 KEY_7
-0x1e08 KEY_8
-0x1e09 KEY_9
-0x1e0a KEY_KPASTERISK
+0x1e00 KEY_NUMERIC_0
+0x1e01 KEY_NUMERIC_1
+0x1e02 KEY_NUMERIC_2
+0x1e03 KEY_NUMERIC_3
+0x1e04 KEY_NUMERIC_4
+0x1e05 KEY_NUMERIC_5
+0x1e06 KEY_NUMERIC_6
+0x1e07 KEY_NUMERIC_7
+0x1e08 KEY_NUMERIC_8
+0x1e09 KEY_NUMERIC_9
+0x1e0a KEY_NUMERIC_STAR
 0x1e0b KEY_RED
 0x1e0c KEY_RADIO
 0x1e0d KEY_MENU
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/hauppauge v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/hauppauge
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/hauppauge	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/hauppauge	2012-12-10 17:01:42.000000000 -0800
@@ -28,17 +28,17 @@
 0x1e24 KEY_PREVIOUSSONG
 0x1e30 KEY_PAUSE
 0x1e1e KEY_NEXTSONG
-0x1e01 KEY_1
-0x1e02 KEY_2
-0x1e03 KEY_3
-0x1e04 KEY_4
-0x1e05 KEY_5
-0x1e06 KEY_6
-0x1e07 KEY_7
-0x1e08 KEY_8
-0x1e09 KEY_9
+0x1e01 KEY_NUMERIC_1
+0x1e02 KEY_NUMERIC_2
+0x1e03 KEY_NUMERIC_3
+0x1e04 KEY_NUMERIC_4
+0x1e05 KEY_NUMERIC_5
+0x1e06 KEY_NUMERIC_6
+0x1e07 KEY_NUMERIC_7
+0x1e08 KEY_NUMERIC_8
+0x1e09 KEY_NUMERIC_9
 0x1e0a KEY_TEXT
-0x1e00 KEY_0
+0x1e00 KEY_NUMERIC_0
 0x1e0e KEY_SUBTITLE
 0x1e0b KEY_RED
 0x1e2e KEY_GREEN
@@ -46,16 +46,16 @@
 0x1e29 KEY_BLUE
 0x1f3d KEY_POWER2
 0x1f3b KEY_SELECT
-0x1f00 KEY_0
-0x1f01 KEY_1
-0x1f02 KEY_2
-0x1f03 KEY_3
-0x1f04 KEY_4
-0x1f05 KEY_5
-0x1f06 KEY_6
-0x1f07 KEY_7
-0x1f08 KEY_8
-0x1f09 KEY_9
+0x1f00 KEY_NUMERIC_0
+0x1f01 KEY_NUMERIC_1
+0x1f02 KEY_NUMERIC_2
+0x1f03 KEY_NUMERIC_3
+0x1f04 KEY_NUMERIC_4
+0x1f05 KEY_NUMERIC_5
+0x1f06 KEY_NUMERIC_6
+0x1f07 KEY_NUMERIC_7
+0x1f08 KEY_NUMERIC_8
+0x1f09 KEY_NUMERIC_9
 0x1f1f KEY_EXIT
 0x1f0d KEY_MENU
 0x1f10 KEY_VOLUMEUP
@@ -78,16 +78,16 @@
 0x1f30 KEY_PAUSE
 0x1f24 KEY_PREVIOUSSONG
 0x1f1e KEY_NEXTSONG
-0x1d00 KEY_0
-0x1d01 KEY_1
-0x1d02 KEY_2
-0x1d03 KEY_3
-0x1d04 KEY_4
-0x1d05 KEY_5
-0x1d06 KEY_6
-0x1d07 KEY_7
-0x1d08 KEY_8
-0x1d09 KEY_9
+0x1d00 KEY_NUMERIC_0
+0x1d01 KEY_NUMERIC_1
+0x1d02 KEY_NUMERIC_2
+0x1d03 KEY_NUMERIC_3
+0x1d04 KEY_NUMERIC_4
+0x1d05 KEY_NUMERIC_5
+0x1d06 KEY_NUMERIC_6
+0x1d07 KEY_NUMERIC_7
+0x1d08 KEY_NUMERIC_8
+0x1d09 KEY_NUMERIC_9
 0x1d0a KEY_TEXT
 0x1d0d KEY_MENU
 0x1d0f KEY_MUTE
@@ -121,16 +121,16 @@
 0x1c16 KEY_LEFT
 0x1c17 KEY_RIGHT
 0x1c25 KEY_OK
-0x1c00 KEY_0
-0x1c01 KEY_1
-0x1c02 KEY_2
-0x1c03 KEY_3
-0x1c04 KEY_4
-0x1c05 KEY_5
-0x1c06 KEY_6
-0x1c07 KEY_7
-0x1c08 KEY_8
-0x1c09 KEY_9
+0x1c00 KEY_NUMERIC_0
+0x1c01 KEY_NUMERIC_1
+0x1c02 KEY_NUMERIC_2
+0x1c03 KEY_NUMERIC_3
+0x1c04 KEY_NUMERIC_4
+0x1c05 KEY_NUMERIC_5
+0x1c06 KEY_NUMERIC_6
+0x1c07 KEY_NUMERIC_7
+0x1c08 KEY_NUMERIC_8
+0x1c09 KEY_NUMERIC_9
 0x1c1f KEY_EXIT
 0x1c0d KEY_MENU
 0x1c1c KEY_TV
@@ -159,15 +159,15 @@
 0x000d KEY_MUTE
 0x0021 KEY_CHANNELDOWN
 0x0022 KEY_VIDEO
-0x0001 KEY_1
-0x0002 KEY_2
-0x0003 KEY_3
-0x0004 KEY_4
-0x0005 KEY_5
-0x0006 KEY_6
-0x0007 KEY_7
-0x0008 KEY_8
-0x0009 KEY_9
+0x0001 KEY_NUMERIC_1
+0x0002 KEY_NUMERIC_2
+0x0003 KEY_NUMERIC_3
+0x0004 KEY_NUMERIC_4
+0x0005 KEY_NUMERIC_5
+0x0006 KEY_NUMERIC_6
+0x0007 KEY_NUMERIC_7
+0x0008 KEY_NUMERIC_8
+0x0009 KEY_NUMERIC_9
 0x001e KEY_RED
-0x0000 KEY_0
+0x0000 KEY_NUMERIC_0
 0x0026 KEY_SLEEP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/imon_pad v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/imon_pad
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/imon_pad	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/imon_pad	2012-12-10 17:00:20.000000000 -0800
@@ -25,8 +25,8 @@
 0x288195b7 KEY_EXIT
 0x02000029 KEY_ESC
 0x2bb715b7 KEY_ESC
-0x0200002a KEY_BACKSPACE
-0x28a115b7 KEY_BACKSPACE
+0x0200002a KEY_BACK
+0x28a115b7 KEY_BACK
 0x2b9595b7 KEY_MUTE
 0x28a395b7 KEY_VOLUMEUP
 0x28a595b7 KEY_VOLUMEDOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/iodata_bctv7e v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/iodata_bctv7e
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/iodata_bctv7e	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/iodata_bctv7e	2012-12-10 17:00:20.000000000 -0800
@@ -3,16 +3,16 @@
 0x20 KEY_RADIO
 0x60 KEY_EPG
 0x00 KEY_POWER
-0x44 KEY_0
-0x50 KEY_1
-0x30 KEY_2
-0x70 KEY_3
-0x48 KEY_4
-0x28 KEY_5
-0x68 KEY_6
-0x58 KEY_7
-0x38 KEY_8
-0x78 KEY_9
+0x44 KEY_NUMERIC_0
+0x50 KEY_NUMERIC_1
+0x30 KEY_NUMERIC_2
+0x70 KEY_NUMERIC_3
+0x48 KEY_NUMERIC_4
+0x28 KEY_NUMERIC_5
+0x68 KEY_NUMERIC_6
+0x58 KEY_NUMERIC_7
+0x38 KEY_NUMERIC_8
+0x78 KEY_NUMERIC_9
 0x10 KEY_L
 0x08 KEY_TIME
 0x18 KEY_PLAYPAUSE
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/it913x_v1 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/it913x_v1
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/it913x_v1	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/it913x_v1	2012-12-10 17:00:20.000000000 -0800
@@ -1,21 +1,21 @@
 # table it913x_v1, type: NEC
 0x61d601 KEY_VIDEO
-0x61d602 KEY_3
+0x61d602 KEY_NUMERIC_3
 0x61d603 KEY_POWER
-0x61d604 KEY_1
-0x61d605 KEY_5
-0x61d606 KEY_6
+0x61d604 KEY_NUMERIC_1
+0x61d605 KEY_NUMERIC_5
+0x61d606 KEY_NUMERIC_6
 0x61d607 KEY_CHANNELDOWN
-0x61d608 KEY_2
+0x61d608 KEY_NUMERIC_2
 0x61d609 KEY_CHANNELUP
-0x61d60a KEY_9
+0x61d60a KEY_NUMERIC_9
 0x61d60b KEY_ZOOM
-0x61d60c KEY_7
-0x61d60d KEY_8
+0x61d60c KEY_NUMERIC_7
+0x61d60d KEY_NUMERIC_8
 0x61d60e KEY_VOLUMEUP
-0x61d60f KEY_4
+0x61d60f KEY_NUMERIC_4
 0x61d610 KEY_ESC
-0x61d611 KEY_0
+0x61d611 KEY_NUMERIC_0
 0x61d612 KEY_OK
 0x61d613 KEY_VOLUMEDOWN
 0x61d614 KEY_RECORD
@@ -31,16 +31,16 @@
 0x61d61e KEY_YELLOW
 0x61d61f KEY_BLUE
 0x61d643 KEY_POWER2
-0x807f0d KEY_0
-0x807f04 KEY_1
-0x807f05 KEY_2
-0x807f06 KEY_3
-0x807f07 KEY_4
-0x807f08 KEY_5
-0x807f09 KEY_6
-0x807f0a KEY_7
-0x807f1b KEY_8
-0x807f1f KEY_9
+0x807f0d KEY_NUMERIC_0
+0x807f04 KEY_NUMERIC_1
+0x807f05 KEY_NUMERIC_2
+0x807f06 KEY_NUMERIC_3
+0x807f07 KEY_NUMERIC_4
+0x807f08 KEY_NUMERIC_5
+0x807f09 KEY_NUMERIC_6
+0x807f0a KEY_NUMERIC_7
+0x807f1b KEY_NUMERIC_8
+0x807f1f KEY_NUMERIC_9
 0x807f12 KEY_POWER
 0x807f01 KEY_MEDIA_REPEAT
 0x807f19 KEY_PAUSE
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/it913x_v2 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/it913x_v2
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/it913x_v2	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/it913x_v2	2012-12-10 17:00:20.000000000 -0800
@@ -8,28 +8,28 @@
 0x807f04 KEY_VOLUMEUP
 0x807f05 KEY_SCREEN
 0x807f06 KEY_VOLUMEDOWN
-0x807f07 KEY_0
+0x807f07 KEY_NUMERIC_0
 0x807f08 KEY_CHANNELDOWN
 0x807f09 KEY_PREVIOUS
-0x807f0a KEY_1
-0x807f1b KEY_2
-0x807f1f KEY_3
-0x807f0c KEY_4
-0x807f0d KEY_5
-0x807f0e KEY_6
-0x807f00 KEY_7
-0x807f0f KEY_8
-0x807f19 KEY_9
-0x866b00 KEY_0
-0x866b01 KEY_1
-0x866b02 KEY_2
-0x866b03 KEY_3
-0x866b04 KEY_4
-0x866b05 KEY_5
-0x866b06 KEY_6
-0x866b07 KEY_7
-0x866b08 KEY_8
-0x866b09 KEY_9
+0x807f0a KEY_NUMERIC_1
+0x807f1b KEY_NUMERIC_2
+0x807f1f KEY_NUMERIC_3
+0x807f0c KEY_NUMERIC_4
+0x807f0d KEY_NUMERIC_5
+0x807f0e KEY_NUMERIC_6
+0x807f00 KEY_NUMERIC_7
+0x807f0f KEY_NUMERIC_8
+0x807f19 KEY_NUMERIC_9
+0x866b00 KEY_NUMERIC_0
+0x866b01 KEY_NUMERIC_1
+0x866b02 KEY_NUMERIC_2
+0x866b03 KEY_NUMERIC_3
+0x866b04 KEY_NUMERIC_4
+0x866b05 KEY_NUMERIC_5
+0x866b06 KEY_NUMERIC_6
+0x866b07 KEY_NUMERIC_7
+0x866b08 KEY_NUMERIC_8
+0x866b09 KEY_NUMERIC_9
 0x866b12 KEY_POWER
 0x866b13 KEY_MUTE
 0x866b0a KEY_PREVIOUS
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/kaiomy v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/kaiomy
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/kaiomy	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/kaiomy	2012-12-10 17:00:20.000000000 -0800
@@ -3,16 +3,16 @@
 0x01 KEY_LIST
 0x0b KEY_ZOOM
 0x03 KEY_POWER
-0x04 KEY_1
-0x08 KEY_2
-0x02 KEY_3
-0x0f KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x0c KEY_7
-0x0d KEY_8
-0x0a KEY_9
-0x11 KEY_0
+0x04 KEY_NUMERIC_1
+0x08 KEY_NUMERIC_2
+0x02 KEY_NUMERIC_3
+0x0f KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x0c KEY_NUMERIC_7
+0x0d KEY_NUMERIC_8
+0x0a KEY_NUMERIC_9
+0x11 KEY_NUMERIC_0
 0x09 KEY_CHANNELUP
 0x07 KEY_CHANNELDOWN
 0x0e KEY_VOLUMEUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/kworld_315u v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/kworld_315u
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/kworld_315u	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/kworld_315u	2012-12-10 17:00:20.000000000 -0800
@@ -3,20 +3,20 @@
 0x6101 KEY_VIDEO
 0x610b KEY_ZOOM
 0x6103 KEY_POWER2
-0x6104 KEY_1
-0x6108 KEY_2
-0x6102 KEY_3
+0x6104 KEY_NUMERIC_1
+0x6108 KEY_NUMERIC_2
+0x6102 KEY_NUMERIC_3
 0x6109 KEY_CHANNELUP
-0x610f KEY_4
-0x6105 KEY_5
-0x6106 KEY_6
+0x610f KEY_NUMERIC_4
+0x6105 KEY_NUMERIC_5
+0x6106 KEY_NUMERIC_6
 0x6107 KEY_CHANNELDOWN
-0x610c KEY_7
-0x610d KEY_8
-0x610a KEY_9
+0x610c KEY_NUMERIC_7
+0x610d KEY_NUMERIC_8
+0x610a KEY_NUMERIC_9
 0x610e KEY_VOLUMEUP
 0x6110 KEY_LAST
-0x6111 KEY_0
+0x6111 KEY_NUMERIC_0
 0x6112 KEY_ENTER
 0x6113 KEY_VOLUMEDOWN
 0x6114 KEY_RECORD
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/kworld_pc150u v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/kworld_pc150u
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/kworld_pc150u	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/kworld_pc150u	2012-12-10 17:00:20.000000000 -0800
@@ -2,16 +2,16 @@
 0x0c KEY_MEDIA
 0x16 KEY_EJECTCLOSECD
 0x1d KEY_POWER2
-0x00 KEY_1
-0x01 KEY_2
-0x02 KEY_3
-0x03 KEY_4
-0x04 KEY_5
-0x05 KEY_6
-0x06 KEY_7
-0x07 KEY_8
-0x08 KEY_9
-0x0a KEY_0
+0x00 KEY_NUMERIC_1
+0x01 KEY_NUMERIC_2
+0x02 KEY_NUMERIC_3
+0x03 KEY_NUMERIC_4
+0x04 KEY_NUMERIC_5
+0x05 KEY_NUMERIC_6
+0x06 KEY_NUMERIC_7
+0x07 KEY_NUMERIC_8
+0x08 KEY_NUMERIC_9
+0x0a KEY_NUMERIC_0
 0x09 KEY_AGAIN
 0x14 KEY_MUTE
 0x1e KEY_LAST
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/kworld_plus_tv_analog v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/kworld_plus_tv_analog
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/kworld_plus_tv_analog	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/kworld_plus_tv_analog	2012-12-10 17:00:20.000000000 -0800
@@ -2,16 +2,16 @@
 0x0c KEY_MEDIA
 0x16 KEY_CLOSECD
 0x1d KEY_POWER2
-0x00 KEY_1
-0x01 KEY_2
-0x02 KEY_3
-0x03 KEY_4
-0x04 KEY_5
-0x05 KEY_6
-0x06 KEY_7
-0x07 KEY_8
-0x08 KEY_9
-0x0a KEY_0
+0x00 KEY_NUMERIC_1
+0x01 KEY_NUMERIC_2
+0x02 KEY_NUMERIC_3
+0x03 KEY_NUMERIC_4
+0x04 KEY_NUMERIC_5
+0x05 KEY_NUMERIC_6
+0x06 KEY_NUMERIC_7
+0x07 KEY_NUMERIC_8
+0x08 KEY_NUMERIC_9
+0x0a KEY_NUMERIC_0
 0x09 KEY_AGAIN
 0x14 KEY_MUTE
 0x20 KEY_UP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/leadtek_y04g0051 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/leadtek_y04g0051
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/leadtek_y04g0051	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/leadtek_y04g0051	2012-12-10 17:00:20.000000000 -0800
@@ -2,20 +2,20 @@
 0x0300 KEY_POWER2
 0x0303 KEY_SCREEN
 0x0304 KEY_RIGHT
-0x0305 KEY_1
-0x0306 KEY_2
-0x0307 KEY_3
+0x0305 KEY_NUMERIC_1
+0x0306 KEY_NUMERIC_2
+0x0307 KEY_NUMERIC_3
 0x0308 KEY_LEFT
-0x0309 KEY_4
-0x030a KEY_5
-0x030b KEY_6
+0x0309 KEY_NUMERIC_4
+0x030a KEY_NUMERIC_5
+0x030b KEY_NUMERIC_6
 0x030c KEY_UP
-0x030d KEY_7
-0x030e KEY_8
-0x030f KEY_9
+0x030d KEY_NUMERIC_7
+0x030e KEY_NUMERIC_8
+0x030f KEY_NUMERIC_9
 0x0310 KEY_DOWN
 0x0311 KEY_AGAIN
-0x0312 KEY_0
+0x0312 KEY_NUMERIC_0
 0x0313 KEY_OK
 0x0314 KEY_MUTE
 0x0316 KEY_OK
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/lme2510 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/lme2510
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/lme2510	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/lme2510	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table lme2510, type: NEC
-0x10ed45 KEY_0
-0x10ed5f KEY_1
-0x10ed50 KEY_2
-0x10ed5d KEY_3
-0x10ed41 KEY_4
-0x10ed0a KEY_5
-0x10ed42 KEY_6
-0x10ed47 KEY_7
-0x10ed49 KEY_8
-0x10ed05 KEY_9
+0x10ed45 KEY_NUMERIC_0
+0x10ed5f KEY_NUMERIC_1
+0x10ed50 KEY_NUMERIC_2
+0x10ed5d KEY_NUMERIC_3
+0x10ed41 KEY_NUMERIC_4
+0x10ed0a KEY_NUMERIC_5
+0x10ed42 KEY_NUMERIC_6
+0x10ed47 KEY_NUMERIC_7
+0x10ed49 KEY_NUMERIC_8
+0x10ed05 KEY_NUMERIC_9
 0x10ed43 KEY_POWER
 0x10ed46 KEY_SUBTITLE
 0x10ed06 KEY_PAUSE
@@ -25,16 +25,16 @@
 0x10ed18 KEY_RECORD
 0x10ed07 KEY_EPG
 0x10ed01 KEY_STOP
-0xbf15 KEY_0
-0xbf08 KEY_1
-0xbf09 KEY_2
-0xbf0a KEY_3
-0xbf0c KEY_4
-0xbf0d KEY_5
-0xbf0e KEY_6
-0xbf10 KEY_7
-0xbf11 KEY_8
-0xbf12 KEY_9
+0xbf15 KEY_NUMERIC_0
+0xbf08 KEY_NUMERIC_1
+0xbf09 KEY_NUMERIC_2
+0xbf0a KEY_NUMERIC_3
+0xbf0c KEY_NUMERIC_4
+0xbf0d KEY_NUMERIC_5
+0xbf0e KEY_NUMERIC_6
+0xbf10 KEY_NUMERIC_7
+0xbf11 KEY_NUMERIC_8
+0xbf12 KEY_NUMERIC_9
 0xbf00 KEY_POWER
 0xbf04 KEY_MEDIA_REPEAT
 0xbf1a KEY_PAUSE
@@ -45,16 +45,16 @@
 0xbf14 KEY_ZOOM
 0xbf18 KEY_RECORD
 0xbf16 KEY_STOP
-0x1c KEY_0
-0x07 KEY_1
-0x15 KEY_2
-0x09 KEY_3
-0x16 KEY_4
-0x19 KEY_5
-0x0d KEY_6
-0x0c KEY_7
-0x18 KEY_8
-0x5e KEY_9
+0x1c KEY_NUMERIC_0
+0x07 KEY_NUMERIC_1
+0x15 KEY_NUMERIC_2
+0x09 KEY_NUMERIC_3
+0x16 KEY_NUMERIC_4
+0x19 KEY_NUMERIC_5
+0x0d KEY_NUMERIC_6
+0x0c KEY_NUMERIC_7
+0x18 KEY_NUMERIC_8
+0x5e KEY_NUMERIC_9
 0x45 KEY_POWER
 0x44 KEY_MEDIA_REPEAT
 0x4a KEY_PAUSE
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/manli v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/manli
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/manli	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/manli	2012-12-10 17:00:20.000000000 -0800
@@ -1,17 +1,17 @@
 # table manli, type: UNKNOWN
 0x1c KEY_RADIO
 0x12 KEY_POWER
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
 0x0a KEY_AGAIN
-0x00 KEY_0
+0x00 KEY_NUMERIC_0
 0x17 KEY_DIGITS
 0x14 KEY_MENU
 0x10 KEY_INFO
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/medion_x10 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/medion_x10
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/medion_x10	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/medion_x10	2012-12-10 17:00:20.000000000 -0800
@@ -19,16 +19,16 @@
 0x34 KEY_YELLOW
 0x35 KEY_BLUE
 0x16 KEY_TEXT
-0x0d KEY_1
-0x0e KEY_2
-0x0f KEY_3
-0x10 KEY_4
-0x11 KEY_5
-0x12 KEY_6
-0x13 KEY_7
-0x14 KEY_8
-0x15 KEY_9
-0x17 KEY_0
+0x0d KEY_NUMERIC_1
+0x0e KEY_NUMERIC_2
+0x0f KEY_NUMERIC_3
+0x10 KEY_NUMERIC_4
+0x11 KEY_NUMERIC_5
+0x12 KEY_NUMERIC_6
+0x13 KEY_NUMERIC_7
+0x14 KEY_NUMERIC_8
+0x15 KEY_NUMERIC_9
+0x17 KEY_NUMERIC_0
 0x1c KEY_SEARCH
 0x20 KEY_DELETE
 0x36 KEY_KEYBOARD
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/medion_x10_digitainer v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/medion_x10_digitainer
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/medion_x10_digitainer	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/medion_x10_digitainer	2012-12-10 17:00:20.000000000 -0800
@@ -36,15 +36,15 @@
 0x24 KEY_REWIND
 0x27 KEY_RECORD
 0x26 KEY_FORWARD
-0x0d KEY_1
-0x0e KEY_2
-0x0f KEY_3
-0x10 KEY_4
-0x11 KEY_5
-0x12 KEY_6
-0x13 KEY_7
-0x14 KEY_8
-0x15 KEY_9
-0x17 KEY_0
+0x0d KEY_NUMERIC_1
+0x0e KEY_NUMERIC_2
+0x0f KEY_NUMERIC_3
+0x10 KEY_NUMERIC_4
+0x11 KEY_NUMERIC_5
+0x12 KEY_NUMERIC_6
+0x13 KEY_NUMERIC_7
+0x14 KEY_NUMERIC_8
+0x15 KEY_NUMERIC_9
+0x17 KEY_NUMERIC_0
 0x1a KEY_UP
 0x22 KEY_DOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/medion_x10_or2x v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/medion_x10_or2x
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/medion_x10_or2x	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/medion_x10_or2x	2012-12-10 17:00:20.000000000 -0800
@@ -30,16 +30,16 @@
 0x28 KEY_STOP
 0x29 KEY_PAUSE
 0x27 KEY_RECORD
-0x0d KEY_1
-0x0e KEY_2
-0x0f KEY_3
-0x10 KEY_4
-0x11 KEY_5
-0x12 KEY_6
-0x13 KEY_7
-0x14 KEY_8
-0x15 KEY_9
-0x17 KEY_0
+0x0d KEY_NUMERIC_1
+0x0e KEY_NUMERIC_2
+0x0f KEY_NUMERIC_3
+0x10 KEY_NUMERIC_4
+0x11 KEY_NUMERIC_5
+0x12 KEY_NUMERIC_6
+0x13 KEY_NUMERIC_7
+0x14 KEY_NUMERIC_8
+0x15 KEY_NUMERIC_9
+0x17 KEY_NUMERIC_0
 0x30 KEY_CLEAR
 0x36 KEY_ENTER
 0x37 KEY_NUMERIC_STAR
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/msi_digivox_ii v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/msi_digivox_ii
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/msi_digivox_ii	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/msi_digivox_ii	2012-12-10 17:00:20.000000000 -0800
@@ -1,19 +1,19 @@
 # table msi_digivox_ii, type: NEC
-0x0302 KEY_2
+0x0302 KEY_NUMERIC_2
 0x0303 KEY_UP
-0x0304 KEY_3
+0x0304 KEY_NUMERIC_3
 0x0305 KEY_CHANNELDOWN
-0x0308 KEY_5
-0x0309 KEY_0
-0x030b KEY_8
+0x0308 KEY_NUMERIC_5
+0x0309 KEY_NUMERIC_0
+0x030b KEY_NUMERIC_8
 0x030d KEY_DOWN
-0x0310 KEY_9
-0x0311 KEY_7
+0x0310 KEY_NUMERIC_9
+0x0311 KEY_NUMERIC_7
 0x0314 KEY_VOLUMEUP
 0x0315 KEY_CHANNELUP
 0x0316 KEY_OK
 0x0317 KEY_POWER2
-0x031a KEY_1
-0x031c KEY_4
-0x031d KEY_6
+0x031a KEY_NUMERIC_1
+0x031c KEY_NUMERIC_4
+0x031d KEY_NUMERIC_6
 0x031f KEY_VOLUMEDOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/msi_digivox_iii v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/msi_digivox_iii
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/msi_digivox_iii	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/msi_digivox_iii	2012-12-10 17:00:20.000000000 -0800
@@ -1,21 +1,21 @@
 # table msi_digivox_iii, type: NEC
 0x61d601 KEY_VIDEO
-0x61d602 KEY_3
+0x61d602 KEY_NUMERIC_3
 0x61d603 KEY_POWER
-0x61d604 KEY_1
-0x61d605 KEY_5
-0x61d606 KEY_6
+0x61d604 KEY_NUMERIC_1
+0x61d605 KEY_NUMERIC_5
+0x61d606 KEY_NUMERIC_6
 0x61d607 KEY_CHANNELDOWN
-0x61d608 KEY_2
+0x61d608 KEY_NUMERIC_2
 0x61d609 KEY_CHANNELUP
-0x61d60a KEY_9
+0x61d60a KEY_NUMERIC_9
 0x61d60b KEY_ZOOM
-0x61d60c KEY_7
-0x61d60d KEY_8
+0x61d60c KEY_NUMERIC_7
+0x61d60d KEY_NUMERIC_8
 0x61d60e KEY_VOLUMEUP
-0x61d60f KEY_4
+0x61d60f KEY_NUMERIC_4
 0x61d610 KEY_ESC
-0x61d611 KEY_0
+0x61d611 KEY_NUMERIC_0
 0x61d612 KEY_OK
 0x61d613 KEY_VOLUMEDOWN
 0x61d614 KEY_RECORD
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/msi_tvanywhere v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/msi_tvanywhere
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/msi_tvanywhere	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/msi_tvanywhere	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table msi_tvanywhere, type: UNKNOWN
-0x00 KEY_0
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
+0x00 KEY_NUMERIC_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
 0x0c KEY_MUTE
 0x0f KEY_SCREEN
 0x10 KEY_FN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/msi_tvanywhere_plus v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/msi_tvanywhere_plus
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/msi_tvanywhere_plus	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/msi_tvanywhere_plus	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table msi_tvanywhere_plus, type: UNKNOWN
-0x01 KEY_1
-0x0b KEY_2
-0x1b KEY_3
-0x05 KEY_4
-0x09 KEY_5
-0x15 KEY_6
-0x06 KEY_7
-0x0a KEY_8
-0x12 KEY_9
-0x02 KEY_0
+0x01 KEY_NUMERIC_1
+0x0b KEY_NUMERIC_2
+0x1b KEY_NUMERIC_3
+0x05 KEY_NUMERIC_4
+0x09 KEY_NUMERIC_5
+0x15 KEY_NUMERIC_6
+0x06 KEY_NUMERIC_7
+0x0a KEY_NUMERIC_8
+0x12 KEY_NUMERIC_9
+0x02 KEY_NUMERIC_0
 0x10 KEY_KPPLUS
 0x13 KEY_AGAIN
 0x1e KEY_POWER
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/nebula v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/nebula
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/nebula	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/nebula	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table nebula, type: UNKNOWN
-0x00 KEY_0
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
+0x00 KEY_NUMERIC_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
 0x0a KEY_TV
 0x0b KEY_AUX
 0x0c KEY_DVD
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/nec_terratec_cinergy_xs v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/nec_terratec_cinergy_xs
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/nec_terratec_cinergy_xs	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/nec_terratec_cinergy_xs	2012-12-10 17:00:20.000000000 -0800
@@ -5,16 +5,16 @@
 0x1443 KEY_SUBTITLE
 0x1444 KEY_TEXT
 0x1445 KEY_DELETE
-0x1402 KEY_1
-0x1403 KEY_2
-0x1404 KEY_3
-0x1405 KEY_4
-0x1406 KEY_5
-0x1407 KEY_6
-0x1408 KEY_7
-0x1409 KEY_8
-0x140a KEY_9
-0x140c KEY_0
+0x1402 KEY_NUMERIC_1
+0x1403 KEY_NUMERIC_2
+0x1404 KEY_NUMERIC_3
+0x1405 KEY_NUMERIC_4
+0x1406 KEY_NUMERIC_5
+0x1407 KEY_NUMERIC_6
+0x1408 KEY_NUMERIC_7
+0x1409 KEY_NUMERIC_8
+0x140a KEY_NUMERIC_9
+0x140c KEY_NUMERIC_0
 0x140b KEY_TUNER
 0x140d KEY_MODE
 0x1446 KEY_TV
@@ -29,7 +29,7 @@
 0x1414 KEY_DOWN
 0x140f KEY_EPG
 0x1416 KEY_INFO
-0x144d KEY_BACKSPACE
+0x144d KEY_BACK
 0x141c KEY_VOLUMEUP
 0x141e KEY_VOLUMEDOWN
 0x144c KEY_PLAY
@@ -48,16 +48,16 @@
 0x144f KEY_FASTFORWARD
 0x145c KEY_NEXT
 0x04eb01 KEY_POWER2
-0x04eb02 KEY_1
-0x04eb03 KEY_2
-0x04eb04 KEY_3
-0x04eb05 KEY_4
-0x04eb06 KEY_5
-0x04eb07 KEY_6
-0x04eb08 KEY_7
-0x04eb09 KEY_8
-0x04eb0a KEY_9
-0x04eb0c KEY_0
+0x04eb02 KEY_NUMERIC_1
+0x04eb03 KEY_NUMERIC_2
+0x04eb04 KEY_NUMERIC_3
+0x04eb05 KEY_NUMERIC_4
+0x04eb06 KEY_NUMERIC_5
+0x04eb07 KEY_NUMERIC_6
+0x04eb08 KEY_NUMERIC_7
+0x04eb09 KEY_NUMERIC_8
+0x04eb0a KEY_NUMERIC_9
+0x04eb0c KEY_NUMERIC_0
 0x04eb0b KEY_TEXT
 0x04eb0d KEY_REFRESH
 0x04eb0e KEY_HOME
@@ -67,7 +67,7 @@
 0x04eb12 KEY_OK
 0x04eb13 KEY_RIGHT
 0x04eb14 KEY_DOWN
-0x04eb15 KEY_BACKSPACE
+0x04eb15 KEY_BACK
 0x04eb16 KEY_INFO
 0x04eb17 KEY_RED
 0x04eb18 KEY_GREEN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/norwood v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/norwood
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/norwood	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/norwood	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table norwood, type: UNKNOWN
-0x20 KEY_0
-0x21 KEY_1
-0x22 KEY_2
-0x23 KEY_3
-0x24 KEY_4
-0x25 KEY_5
-0x26 KEY_6
-0x27 KEY_7
-0x28 KEY_8
-0x29 KEY_9
+0x20 KEY_NUMERIC_0
+0x21 KEY_NUMERIC_1
+0x22 KEY_NUMERIC_2
+0x23 KEY_NUMERIC_3
+0x24 KEY_NUMERIC_4
+0x25 KEY_NUMERIC_5
+0x26 KEY_NUMERIC_6
+0x27 KEY_NUMERIC_7
+0x28 KEY_NUMERIC_8
+0x29 KEY_NUMERIC_9
 0x78 KEY_VIDEO
 0x2c KEY_EXIT
 0x2a KEY_SELECT
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/npgtech v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/npgtech
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/npgtech	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/npgtech	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table npgtech, type: UNKNOWN
 0x1d KEY_SWITCHVIDEOMODE
 0x2a KEY_FRONT
-0x3e KEY_1
-0x02 KEY_2
-0x06 KEY_3
-0x0a KEY_4
-0x0e KEY_5
-0x12 KEY_6
-0x16 KEY_7
-0x1a KEY_8
-0x1e KEY_9
-0x3a KEY_0
+0x3e KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x06 KEY_NUMERIC_3
+0x0a KEY_NUMERIC_4
+0x0e KEY_NUMERIC_5
+0x12 KEY_NUMERIC_6
+0x16 KEY_NUMERIC_7
+0x1a KEY_NUMERIC_8
+0x1e KEY_NUMERIC_9
+0x3a KEY_NUMERIC_0
 0x22 KEY_NUMLOCK
 0x20 KEY_REFRESH
 0x03 KEY_BRIGHTNESSDOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/opera1 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/opera1
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/opera1	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/opera1	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table opera1, type: UNKNOWN
-0x5fa0 KEY_1
-0x51af KEY_2
-0x5da2 KEY_3
-0x41be KEY_4
-0x0bf5 KEY_5
-0x43bd KEY_6
-0x47b8 KEY_7
-0x49b6 KEY_8
-0x05fa KEY_9
-0x45ba KEY_0
+0x5fa0 KEY_NUMERIC_1
+0x51af KEY_NUMERIC_2
+0x5da2 KEY_NUMERIC_3
+0x41be KEY_NUMERIC_4
+0x0bf5 KEY_NUMERIC_5
+0x43bd KEY_NUMERIC_6
+0x47b8 KEY_NUMERIC_7
+0x49b6 KEY_NUMERIC_8
+0x05fa KEY_NUMERIC_9
+0x45ba KEY_NUMERIC_0
 0x09f6 KEY_CHANNELUP
 0x1be5 KEY_CHANNELDOWN
 0x5da3 KEY_VOLUMEDOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pctv_sedna v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pctv_sedna
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pctv_sedna	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pctv_sedna	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table pctv_sedna, type: UNKNOWN
-0x00 KEY_0
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
+0x00 KEY_NUMERIC_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
 0x0a KEY_AGAIN
 0x0b KEY_CHANNELUP
 0x0c KEY_VOLUMEUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pinnacle310e v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pinnacle310e
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pinnacle310e	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pinnacle310e	2012-12-10 17:00:20.000000000 -0800
@@ -5,16 +5,16 @@
 0x48 KEY_PROGRAM
 0x1c KEY_EPG
 0x04 KEY_LIST
-0x03 KEY_1
-0x01 KEY_2
-0x06 KEY_3
-0x09 KEY_4
-0x1d KEY_5
-0x1f KEY_6
-0x0d KEY_7
-0x19 KEY_8
-0x1b KEY_9
-0x15 KEY_0
+0x03 KEY_NUMERIC_1
+0x01 KEY_NUMERIC_2
+0x06 KEY_NUMERIC_3
+0x09 KEY_NUMERIC_4
+0x1d KEY_NUMERIC_5
+0x1f KEY_NUMERIC_6
+0x0d KEY_NUMERIC_7
+0x19 KEY_NUMERIC_8
+0x1b KEY_NUMERIC_9
+0x15 KEY_NUMERIC_0
 0x0c KEY_CANCEL
 0x4a KEY_CLEAR
 0x13 KEY_BACK
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pinnacle_color v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pinnacle_color
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pinnacle_color	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pinnacle_color	2012-12-10 17:00:20.000000000 -0800
@@ -29,15 +29,15 @@
 0x53 KEY_PREVIOUS
 0x4c KEY_STOP
 0x54 KEY_NEXT
-0x69 KEY_0
-0x6a KEY_1
-0x6b KEY_2
-0x6c KEY_3
-0x6d KEY_4
-0x6e KEY_5
-0x6f KEY_6
-0x70 KEY_7
-0x71 KEY_8
-0x72 KEY_9
+0x69 KEY_NUMERIC_0
+0x6a KEY_NUMERIC_1
+0x6b KEY_NUMERIC_2
+0x6c KEY_NUMERIC_3
+0x6d KEY_NUMERIC_4
+0x6e KEY_NUMERIC_5
+0x6f KEY_NUMERIC_6
+0x70 KEY_NUMERIC_7
+0x71 KEY_NUMERIC_8
+0x72 KEY_NUMERIC_9
 0x74 KEY_CHANNEL
-0x0a KEY_BACKSPACE
+0x0a KEY_BACK
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pinnacle_grey v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pinnacle_grey
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pinnacle_grey	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pinnacle_grey	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table pinnacle_grey, type: UNKNOWN
-0x3a KEY_0
-0x31 KEY_1
-0x32 KEY_2
-0x33 KEY_3
-0x34 KEY_4
-0x35 KEY_5
-0x36 KEY_6
-0x37 KEY_7
-0x38 KEY_8
-0x39 KEY_9
+0x3a KEY_NUMERIC_0
+0x31 KEY_NUMERIC_1
+0x32 KEY_NUMERIC_2
+0x33 KEY_NUMERIC_3
+0x34 KEY_NUMERIC_4
+0x35 KEY_NUMERIC_5
+0x36 KEY_NUMERIC_6
+0x37 KEY_NUMERIC_7
+0x38 KEY_NUMERIC_8
+0x39 KEY_NUMERIC_9
 0x2f KEY_POWER
 0x2e KEY_P
 0x1f KEY_L
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pinnacle_pctv_hd v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pinnacle_pctv_hd
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pinnacle_pctv_hd	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pinnacle_pctv_hd	2012-12-10 17:00:20.000000000 -0800
@@ -7,16 +7,16 @@
 0x0709 KEY_VOLUMEDOWN
 0x0706 KEY_CHANNELUP
 0x070c KEY_CHANNELDOWN
-0x070f KEY_1
-0x0715 KEY_2
-0x0710 KEY_3
-0x0718 KEY_4
-0x071b KEY_5
-0x071e KEY_6
-0x0711 KEY_7
-0x0721 KEY_8
-0x0712 KEY_9
-0x0727 KEY_0
+0x070f KEY_NUMERIC_1
+0x0715 KEY_NUMERIC_2
+0x0710 KEY_NUMERIC_3
+0x0718 KEY_NUMERIC_4
+0x071b KEY_NUMERIC_5
+0x071e KEY_NUMERIC_6
+0x0711 KEY_NUMERIC_7
+0x0721 KEY_NUMERIC_8
+0x0712 KEY_NUMERIC_9
+0x0727 KEY_NUMERIC_0
 0x0724 KEY_ZOOM
 0x072a KEY_SUBTITLE
 0x072d KEY_REWIND
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pixelview v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pixelview
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pixelview	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pixelview	2012-12-10 17:00:20.000000000 -0800
@@ -9,16 +9,16 @@
 0x1a KEY_PLAYPAUSE
 0x19 KEY_ZOOM
 0x0f KEY_TEXT
-0x01 KEY_1
-0x0b KEY_2
-0x1b KEY_3
-0x05 KEY_4
-0x09 KEY_5
-0x15 KEY_6
-0x06 KEY_7
-0x0a KEY_8
-0x12 KEY_9
-0x02 KEY_0
+0x01 KEY_NUMERIC_1
+0x0b KEY_NUMERIC_2
+0x1b KEY_NUMERIC_3
+0x05 KEY_NUMERIC_4
+0x09 KEY_NUMERIC_5
+0x15 KEY_NUMERIC_6
+0x06 KEY_NUMERIC_7
+0x0a KEY_NUMERIC_8
+0x12 KEY_NUMERIC_9
+0x02 KEY_NUMERIC_0
 0x10 KEY_LAST
 0x13 KEY_LIST
 0x1f KEY_CHANNELUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pixelview_002t v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pixelview_002t
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pixelview_002t	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pixelview_002t	2012-12-10 17:00:20.000000000 -0800
@@ -1,23 +1,23 @@
 # table pixelview_002t, type: NEC
 0x866b13 KEY_MUTE
 0x866b12 KEY_POWER2
-0x866b01 KEY_1
-0x866b02 KEY_2
-0x866b03 KEY_3
-0x866b04 KEY_4
-0x866b05 KEY_5
-0x866b06 KEY_6
-0x866b07 KEY_7
-0x866b08 KEY_8
-0x866b09 KEY_9
-0x866b00 KEY_0
+0x866b01 KEY_NUMERIC_1
+0x866b02 KEY_NUMERIC_2
+0x866b03 KEY_NUMERIC_3
+0x866b04 KEY_NUMERIC_4
+0x866b05 KEY_NUMERIC_5
+0x866b06 KEY_NUMERIC_6
+0x866b07 KEY_NUMERIC_7
+0x866b08 KEY_NUMERIC_8
+0x866b09 KEY_NUMERIC_9
+0x866b00 KEY_NUMERIC_0
 0x866b0d KEY_CHANNELUP
 0x866b19 KEY_CHANNELDOWN
 0x866b10 KEY_VOLUMEUP
 0x866b0c KEY_VOLUMEDOWN
 0x866b0a KEY_CAMERA
 0x866b0b KEY_ZOOM
-0x866b1b KEY_BACKSPACE
+0x866b1b KEY_BACK
 0x866b15 KEY_ENTER
 0x866b1d KEY_UP
 0x866b1e KEY_DOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pixelview_mk12 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pixelview_mk12
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pixelview_mk12	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pixelview_mk12	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table pixelview_mk12, type: NEC
 0x866b03 KEY_TUNER
 0x866b1e KEY_POWER2
-0x866b01 KEY_1
-0x866b0b KEY_2
-0x866b1b KEY_3
-0x866b05 KEY_4
-0x866b09 KEY_5
-0x866b15 KEY_6
-0x866b06 KEY_7
-0x866b0a KEY_8
-0x866b12 KEY_9
-0x866b02 KEY_0
+0x866b01 KEY_NUMERIC_1
+0x866b0b KEY_NUMERIC_2
+0x866b1b KEY_NUMERIC_3
+0x866b05 KEY_NUMERIC_4
+0x866b09 KEY_NUMERIC_5
+0x866b15 KEY_NUMERIC_6
+0x866b06 KEY_NUMERIC_7
+0x866b0a KEY_NUMERIC_8
+0x866b12 KEY_NUMERIC_9
+0x866b02 KEY_NUMERIC_0
 0x866b13 KEY_AGAIN
 0x866b10 KEY_DIGITS
 0x866b00 KEY_VIDEO
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pixelview_new v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pixelview_new
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pixelview_new	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pixelview_new	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table pixelview_new, type: UNKNOWN
 0x3c KEY_TIME
 0x12 KEY_POWER
-0x3d KEY_1
-0x38 KEY_2
-0x18 KEY_3
-0x35 KEY_4
-0x39 KEY_5
-0x15 KEY_6
-0x36 KEY_7
-0x3a KEY_8
-0x1e KEY_9
-0x3e KEY_0
+0x3d KEY_NUMERIC_1
+0x38 KEY_NUMERIC_2
+0x18 KEY_NUMERIC_3
+0x35 KEY_NUMERIC_4
+0x39 KEY_NUMERIC_5
+0x15 KEY_NUMERIC_6
+0x36 KEY_NUMERIC_7
+0x3a KEY_NUMERIC_8
+0x1e KEY_NUMERIC_9
+0x3e KEY_NUMERIC_0
 0x1c KEY_AGAIN
 0x3f KEY_VIDEO
 0x1f KEY_LAST
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/powercolor_real_angel v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/powercolor_real_angel
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/powercolor_real_angel	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/powercolor_real_angel	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table powercolor_real_angel, type: UNKNOWN
 0x38 KEY_SWITCHVIDEOMODE
 0x0c KEY_MEDIA
-0x00 KEY_0
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
+0x00 KEY_NUMERIC_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
 0x0a KEY_DIGITS
 0x29 KEY_PREVIOUS
 0x12 KEY_BRIGHTNESSUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/proteus_2309 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/proteus_2309
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/proteus_2309	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/proteus_2309	2012-12-10 17:00:20.000000000 -0800
@@ -1,17 +1,17 @@
 # table proteus_2309, type: UNKNOWN
-0x00 KEY_0
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
+0x00 KEY_NUMERIC_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
 0x5c KEY_POWER
 0x20 KEY_ZOOM
-0x0f KEY_BACKSPACE
+0x0f KEY_BACK
 0x1b KEY_ENTER
 0x41 KEY_RECORD
 0x43 KEY_STOP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/purpletv v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/purpletv
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/purpletv	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/purpletv	2012-12-10 17:00:20.000000000 -0800
@@ -1,17 +1,17 @@
 # table purpletv, type: UNKNOWN
 0x03 KEY_POWER
 0x6f KEY_MUTE
-0x10 KEY_BACKSPACE
-0x11 KEY_0
-0x04 KEY_1
-0x05 KEY_2
-0x06 KEY_3
-0x08 KEY_4
-0x09 KEY_5
-0x0a KEY_6
-0x0c KEY_7
-0x0d KEY_8
-0x0e KEY_9
+0x10 KEY_BACK
+0x11 KEY_NUMERIC_0
+0x04 KEY_NUMERIC_1
+0x05 KEY_NUMERIC_2
+0x06 KEY_NUMERIC_3
+0x08 KEY_NUMERIC_4
+0x09 KEY_NUMERIC_5
+0x0a KEY_NUMERIC_6
+0x0c KEY_NUMERIC_7
+0x0d KEY_NUMERIC_8
+0x0e KEY_NUMERIC_9
 0x12 KEY_DOT
 0x07 KEY_VOLUMEUP
 0x0b KEY_VOLUMEDOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pv951 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pv951
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/pv951	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/pv951	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table pv951, type: UNKNOWN
-0x00 KEY_0
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
+0x00 KEY_NUMERIC_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
 0x12 KEY_POWER
 0x10 KEY_MUTE
 0x1f KEY_VOLUMEDOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/rc6_mce v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/rc6_mce
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/rc6_mce	2012-12-09 04:32:42.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/rc6_mce	2012-12-10 17:00:20.000000000 -0800
@@ -12,7 +12,7 @@
 0x800f040a KEY_DELETE
 0x800f040b KEY_ENTER
 0x800f040c KEY_SLEEP
-0x800f040d KEY_MEDIA
+0x800f040d KEY_MENU                     # Windows menu button
 0x800f040e KEY_MUTE
 0x800f040f KEY_INFO
 0x800f0410 KEY_VOLUMEUP
@@ -27,8 +27,8 @@
 0x800f0419 KEY_STOP
 0x800f041a KEY_NEXT
 0x800f041b KEY_PREVIOUS
-0x800f041c KEY_NUMERIC_POUND
-0x800f041d KEY_NUMERIC_STAR
+0x800f041c KEY_TEXT                     # '#' button (KEY_NUMERIC_POUND)
+0x800f041d KEY_ZOOM                     # '*' button (KEY_NUMERIC_STAR)
 0x800f041e KEY_UP
 0x800f041f KEY_DOWN
 0x800f0420 KEY_LEFT
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/real_audio_220_32_keys v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/real_audio_220_32_keys
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/real_audio_220_32_keys	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/real_audio_220_32_keys	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table real_audio_220_32_keys, type: UNKNOWN
 0x1c KEY_RADIO
 0x12 KEY_POWER2
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
-0x00 KEY_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
+0x00 KEY_NUMERIC_0
 0x0c KEY_VOLUMEUP
 0x18 KEY_VOLUMEDOWN
 0x0b KEY_CHANNELUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/snapstream_firefly v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/snapstream_firefly
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/snapstream_firefly	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/snapstream_firefly	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table snapstream_firefly, type: OTHER
 0x2c KEY_ZOOM
 0x02 KEY_CLOSE
-0x0d KEY_1
-0x0e KEY_2
-0x0f KEY_3
-0x10 KEY_4
-0x11 KEY_5
-0x12 KEY_6
-0x13 KEY_7
-0x14 KEY_8
-0x15 KEY_9
-0x17 KEY_0
+0x0d KEY_NUMERIC_1
+0x0e KEY_NUMERIC_2
+0x0f KEY_NUMERIC_3
+0x10 KEY_NUMERIC_4
+0x11 KEY_NUMERIC_5
+0x12 KEY_NUMERIC_6
+0x13 KEY_NUMERIC_7
+0x14 KEY_NUMERIC_8
+0x15 KEY_NUMERIC_9
+0x17 KEY_NUMERIC_0
 0x16 KEY_BACK
 0x18 KEY_KPENTER
 0x09 KEY_VOLUMEUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/su3000 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/su3000
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/su3000	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/su3000	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table su3000, type: UNKNOWN
 0x25 KEY_POWER
 0x0a KEY_MUTE
-0x01 KEY_1
-0x02 KEY_2
-0x03 KEY_3
-0x04 KEY_4
-0x05 KEY_5
-0x06 KEY_6
-0x07 KEY_7
-0x08 KEY_8
-0x09 KEY_9
-0x00 KEY_0
+0x01 KEY_NUMERIC_1
+0x02 KEY_NUMERIC_2
+0x03 KEY_NUMERIC_3
+0x04 KEY_NUMERIC_4
+0x05 KEY_NUMERIC_5
+0x06 KEY_NUMERIC_6
+0x07 KEY_NUMERIC_7
+0x08 KEY_NUMERIC_8
+0x09 KEY_NUMERIC_9
+0x00 KEY_NUMERIC_0
 0x20 KEY_UP
 0x21 KEY_DOWN
 0x12 KEY_VOLUMEUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/tbs v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/tbs
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/tbs	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/tbs	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table tbs, type: UNKNOWN
 0xf884 KEY_POWER
 0xf894 KEY_MUTE
-0xf887 KEY_1
-0xf886 KEY_2
-0xf885 KEY_3
-0xf88b KEY_4
-0xf88a KEY_5
-0xf889 KEY_6
-0xf88f KEY_7
-0xf88e KEY_8
-0xf88d KEY_9
-0xf892 KEY_0
+0xf887 KEY_NUMERIC_1
+0xf886 KEY_NUMERIC_2
+0xf885 KEY_NUMERIC_3
+0xf88b KEY_NUMERIC_4
+0xf88a KEY_NUMERIC_5
+0xf889 KEY_NUMERIC_6
+0xf88f KEY_NUMERIC_7
+0xf88e KEY_NUMERIC_8
+0xf88d KEY_NUMERIC_9
+0xf892 KEY_NUMERIC_0
 0xf896 KEY_CHANNELUP
 0xf891 KEY_CHANNELDOWN
 0xf893 KEY_VOLUMEUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/tbs_nec v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/tbs_nec
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/tbs_nec	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/tbs_nec	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table tbs_nec, type: UNKNOWN
 0x84 KEY_POWER2
 0x94 KEY_MUTE
-0x87 KEY_1
-0x86 KEY_2
-0x85 KEY_3
-0x8b KEY_4
-0x8a KEY_5
-0x89 KEY_6
-0x8f KEY_7
-0x8e KEY_8
-0x8d KEY_9
-0x92 KEY_0
+0x87 KEY_NUMERIC_1
+0x86 KEY_NUMERIC_2
+0x85 KEY_NUMERIC_3
+0x8b KEY_NUMERIC_4
+0x8a KEY_NUMERIC_5
+0x89 KEY_NUMERIC_6
+0x8f KEY_NUMERIC_7
+0x8e KEY_NUMERIC_8
+0x8d KEY_NUMERIC_9
+0x92 KEY_NUMERIC_0
 0xc0 KEY_10CHANNELSUP
 0xd0 KEY_10CHANNELSDOWN
 0x96 KEY_CHANNELUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/technisat_usb2 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/technisat_usb2
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/technisat_usb2	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/technisat_usb2	2012-12-10 17:00:20.000000000 -0800
@@ -1,17 +1,17 @@
 # table technisat_usb2, type: RC5
 0x0a0c KEY_POWER
-0x0a01 KEY_1
-0x0a02 KEY_2
-0x0a03 KEY_3
+0x0a01 KEY_NUMERIC_1
+0x0a02 KEY_NUMERIC_2
+0x0a03 KEY_NUMERIC_3
 0x0a0d KEY_MUTE
-0x0a04 KEY_4
-0x0a05 KEY_5
-0x0a06 KEY_6
+0x0a04 KEY_NUMERIC_4
+0x0a05 KEY_NUMERIC_5
+0x0a06 KEY_NUMERIC_6
 0x0a38 KEY_VIDEO
-0x0a07 KEY_7
-0x0a08 KEY_8
-0x0a09 KEY_9
-0x0a00 KEY_0
+0x0a07 KEY_NUMERIC_7
+0x0a08 KEY_NUMERIC_8
+0x0a09 KEY_NUMERIC_9
+0x0a00 KEY_NUMERIC_0
 0x0a4f KEY_INFO
 0x0a20 KEY_CHANNELUP
 0x0a52 KEY_MENU
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/terratec_cinergy_xs v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/terratec_cinergy_xs
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/terratec_cinergy_xs	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/terratec_cinergy_xs	2012-12-10 17:00:20.000000000 -0800
@@ -2,20 +2,20 @@
 0x41 KEY_HOME
 0x01 KEY_POWER
 0x42 KEY_MENU
-0x02 KEY_1
-0x03 KEY_2
-0x04 KEY_3
+0x02 KEY_NUMERIC_1
+0x03 KEY_NUMERIC_2
+0x04 KEY_NUMERIC_3
 0x43 KEY_SUBTITLE
-0x05 KEY_4
-0x06 KEY_5
-0x07 KEY_6
+0x05 KEY_NUMERIC_4
+0x06 KEY_NUMERIC_5
+0x07 KEY_NUMERIC_6
 0x44 KEY_TEXT
-0x08 KEY_7
-0x09 KEY_8
-0x0a KEY_9
+0x08 KEY_NUMERIC_7
+0x09 KEY_NUMERIC_8
+0x0a KEY_NUMERIC_9
 0x45 KEY_DELETE
 0x0b KEY_TUNER
-0x0c KEY_0
+0x0c KEY_NUMERIC_0
 0x0d KEY_MODE
 0x46 KEY_TV
 0x47 KEY_DVD
@@ -28,7 +28,7 @@
 0x14 KEY_DOWN
 0x0f KEY_EPG
 0x16 KEY_INFO
-0x4d KEY_BACKSPACE
+0x4d KEY_BACK
 0x1c KEY_VOLUMEUP
 0x4c KEY_PLAY
 0x1b KEY_CHANNELUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/terratec_slim v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/terratec_slim
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/terratec_slim	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/terratec_slim	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table terratec_slim, type: NEC
-0x02bd00 KEY_1
-0x02bd01 KEY_2
-0x02bd02 KEY_3
-0x02bd03 KEY_4
-0x02bd04 KEY_5
-0x02bd05 KEY_6
-0x02bd06 KEY_7
-0x02bd07 KEY_8
-0x02bd08 KEY_9
-0x02bd09 KEY_0
+0x02bd00 KEY_NUMERIC_1
+0x02bd01 KEY_NUMERIC_2
+0x02bd02 KEY_NUMERIC_3
+0x02bd03 KEY_NUMERIC_4
+0x02bd04 KEY_NUMERIC_5
+0x02bd05 KEY_NUMERIC_6
+0x02bd06 KEY_NUMERIC_7
+0x02bd07 KEY_NUMERIC_8
+0x02bd08 KEY_NUMERIC_9
+0x02bd09 KEY_NUMERIC_0
 0x02bd0a KEY_MUTE
 0x02bd0b KEY_NEW
 0x02bd0e KEY_VOLUMEDOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/terratec_slim_2 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/terratec_slim_2
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/terratec_slim_2	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/terratec_slim_2	2012-12-10 17:00:20.000000000 -0800
@@ -2,18 +2,18 @@
 0x8001 KEY_MUTE
 0x8002 KEY_VOLUMEDOWN
 0x8003 KEY_CHANNELDOWN
-0x8004 KEY_1
-0x8005 KEY_2
-0x8006 KEY_3
-0x8007 KEY_4
-0x8008 KEY_5
-0x8009 KEY_6
-0x800a KEY_7
+0x8004 KEY_NUMERIC_1
+0x8005 KEY_NUMERIC_2
+0x8006 KEY_NUMERIC_3
+0x8007 KEY_NUMERIC_4
+0x8008 KEY_NUMERIC_5
+0x8009 KEY_NUMERIC_6
+0x800a KEY_NUMERIC_7
 0x800c KEY_ZOOM
-0x800d KEY_0
+0x800d KEY_NUMERIC_0
 0x800e KEY_AGAIN
 0x8012 KEY_POWER2
 0x801a KEY_VOLUMEUP
-0x801b KEY_8
+0x801b KEY_NUMERIC_8
 0x801e KEY_CHANNELUP
-0x801f KEY_9
+0x801f KEY_NUMERIC_9
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/tevii v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/tevii
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/tevii	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/tevii	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table tevii, type: UNKNOWN
 0xf80a KEY_POWER
 0xf80c KEY_MUTE
-0xf811 KEY_1
-0xf812 KEY_2
-0xf813 KEY_3
-0xf814 KEY_4
-0xf815 KEY_5
-0xf816 KEY_6
-0xf817 KEY_7
-0xf818 KEY_8
-0xf819 KEY_9
-0xf810 KEY_0
+0xf811 KEY_NUMERIC_1
+0xf812 KEY_NUMERIC_2
+0xf813 KEY_NUMERIC_3
+0xf814 KEY_NUMERIC_4
+0xf815 KEY_NUMERIC_5
+0xf816 KEY_NUMERIC_6
+0xf817 KEY_NUMERIC_7
+0xf818 KEY_NUMERIC_8
+0xf819 KEY_NUMERIC_9
+0xf810 KEY_NUMERIC_0
 0xf81c KEY_MENU
 0xf80f KEY_VOLUMEDOWN
 0xf81a KEY_LAST
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/tevii_nec v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/tevii_nec
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/tevii_nec	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/tevii_nec	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table tevii_nec, type: UNKNOWN
 0x0a KEY_POWER2
 0x0c KEY_MUTE
-0x11 KEY_1
-0x12 KEY_2
-0x13 KEY_3
-0x14 KEY_4
-0x15 KEY_5
-0x16 KEY_6
-0x17 KEY_7
-0x18 KEY_8
-0x19 KEY_9
-0x10 KEY_0
+0x11 KEY_NUMERIC_1
+0x12 KEY_NUMERIC_2
+0x13 KEY_NUMERIC_3
+0x14 KEY_NUMERIC_4
+0x15 KEY_NUMERIC_5
+0x16 KEY_NUMERIC_6
+0x17 KEY_NUMERIC_7
+0x18 KEY_NUMERIC_8
+0x19 KEY_NUMERIC_9
+0x10 KEY_NUMERIC_0
 0x1c KEY_MENU
 0x0f KEY_VOLUMEDOWN
 0x1a KEY_LAST
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/total_media_in_hand v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/total_media_in_hand
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/total_media_in_hand	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/total_media_in_hand	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table total_media_in_hand, type: NEC
-0x02bd00 KEY_1
-0x02bd01 KEY_2
-0x02bd02 KEY_3
-0x02bd03 KEY_4
-0x02bd04 KEY_5
-0x02bd05 KEY_6
-0x02bd06 KEY_7
-0x02bd07 KEY_8
-0x02bd08 KEY_9
-0x02bd09 KEY_0
+0x02bd00 KEY_NUMERIC_1
+0x02bd01 KEY_NUMERIC_2
+0x02bd02 KEY_NUMERIC_3
+0x02bd03 KEY_NUMERIC_4
+0x02bd04 KEY_NUMERIC_5
+0x02bd05 KEY_NUMERIC_6
+0x02bd06 KEY_NUMERIC_7
+0x02bd07 KEY_NUMERIC_8
+0x02bd08 KEY_NUMERIC_9
+0x02bd09 KEY_NUMERIC_0
 0x02bd0a KEY_MUTE
 0x02bd0b KEY_CYCLEWINDOWS
 0x02bd0c KEY_VIDEO
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/trekstor v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/trekstor
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/trekstor	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/trekstor	2012-12-10 17:00:20.000000000 -0800
@@ -1,5 +1,5 @@
 # table trekstor, type: NEC
-0x0084 KEY_0
+0x0084 KEY_NUMERIC_0
 0x0085 KEY_MUTE
 0x0086 KEY_HOMEPAGE
 0x0087 KEY_UP
@@ -11,17 +11,17 @@
 0x008d KEY_PLAY
 0x008e KEY_STOP
 0x008f KEY_EPG
-0x0090 KEY_7
-0x0091 KEY_4
-0x0092 KEY_1
+0x0090 KEY_NUMERIC_7
+0x0091 KEY_NUMERIC_4
+0x0092 KEY_NUMERIC_1
 0x0093 KEY_CHANNELDOWN
-0x0094 KEY_8
-0x0095 KEY_5
-0x0096 KEY_2
+0x0094 KEY_NUMERIC_8
+0x0095 KEY_NUMERIC_5
+0x0096 KEY_NUMERIC_2
 0x0097 KEY_CHANNELUP
-0x0098 KEY_9
-0x0099 KEY_6
-0x009a KEY_3
+0x0098 KEY_NUMERIC_9
+0x0099 KEY_NUMERIC_6
+0x009a KEY_NUMERIC_3
 0x009b KEY_VOLUMEDOWN
 0x009c KEY_TV
 0x009d KEY_RECORD
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/tt_1500 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/tt_1500
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/tt_1500	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/tt_1500	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table tt_1500, type: RC5
 0x1501 KEY_POWER
 0x1502 KEY_SHUFFLE
-0x1503 KEY_1
-0x1504 KEY_2
-0x1505 KEY_3
-0x1506 KEY_4
-0x1507 KEY_5
-0x1508 KEY_6
-0x1509 KEY_7
-0x150a KEY_8
-0x150b KEY_9
-0x150c KEY_0
+0x1503 KEY_NUMERIC_1
+0x1504 KEY_NUMERIC_2
+0x1505 KEY_NUMERIC_3
+0x1506 KEY_NUMERIC_4
+0x1507 KEY_NUMERIC_5
+0x1508 KEY_NUMERIC_6
+0x1509 KEY_NUMERIC_7
+0x150a KEY_NUMERIC_8
+0x150b KEY_NUMERIC_9
+0x150c KEY_NUMERIC_0
 0x150d KEY_UP
 0x150e KEY_LEFT
 0x150f KEY_OK
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/twinhan_vp1027_dvbs v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/twinhan_vp1027_dvbs
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/twinhan_vp1027_dvbs	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/twinhan_vp1027_dvbs	2012-12-10 17:00:20.000000000 -0800
@@ -5,19 +5,19 @@
 0x48 KEY_INFO
 0x1c KEY_EPG
 0x04 KEY_LIST
-0x03 KEY_1
-0x01 KEY_2
-0x06 KEY_3
-0x09 KEY_4
-0x1d KEY_5
-0x1f KEY_6
-0x0d KEY_7
-0x19 KEY_8
-0x1b KEY_9
-0x15 KEY_0
+0x03 KEY_NUMERIC_1
+0x01 KEY_NUMERIC_2
+0x06 KEY_NUMERIC_3
+0x09 KEY_NUMERIC_4
+0x1d KEY_NUMERIC_5
+0x1f KEY_NUMERIC_6
+0x0d KEY_NUMERIC_7
+0x19 KEY_NUMERIC_8
+0x1b KEY_NUMERIC_9
+0x15 KEY_NUMERIC_0
 0x0c KEY_CANCEL
 0x4a KEY_CLEAR
-0x13 KEY_BACKSPACE
+0x13 KEY_BACK
 0x00 KEY_TAB
 0x4b KEY_UP
 0x51 KEY_DOWN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/videomate_k100 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/videomate_k100
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/videomate_k100	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/videomate_k100	2012-12-10 17:00:20.000000000 -0800
@@ -14,7 +14,7 @@
 0x15 KEY_OK
 0x14 KEY_RIGHT
 0x12 KEY_DOWN
-0x16 KEY_BACKSPACE
+0x16 KEY_BACK
 0x02 KEY_ZOOM
 0x04 KEY_INFO
 0x05 KEY_VOLUMEUP
@@ -30,17 +30,17 @@
 0x10 KEY_PREVIOUS
 0x0d KEY_PAUSE
 0x0f KEY_NEXT
-0x1e KEY_1
-0x1f KEY_2
-0x20 KEY_3
-0x21 KEY_4
-0x22 KEY_5
-0x23 KEY_6
-0x24 KEY_7
-0x25 KEY_8
-0x26 KEY_9
+0x1e KEY_NUMERIC_1
+0x1f KEY_NUMERIC_2
+0x20 KEY_NUMERIC_3
+0x21 KEY_NUMERIC_4
+0x22 KEY_NUMERIC_5
+0x23 KEY_NUMERIC_6
+0x24 KEY_NUMERIC_7
+0x25 KEY_NUMERIC_8
+0x26 KEY_NUMERIC_9
 0x2a KEY_NUMERIC_STAR
-0x1d KEY_0
+0x1d KEY_NUMERIC_0
 0x29 KEY_SUBTITLE
 0x27 KEY_CLEAR
 0x34 KEY_SCREEN
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/videomate_s350 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/videomate_s350
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/videomate_s350	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/videomate_s350	2012-12-10 17:00:20.000000000 -0800
@@ -12,16 +12,16 @@
 0x13 KEY_CHANNELDOWN
 0x14 KEY_MUTE
 0x15 KEY_VOLUMEDOWN
-0x16 KEY_1
-0x17 KEY_2
-0x18 KEY_3
-0x19 KEY_4
-0x1a KEY_5
-0x1b KEY_6
-0x1c KEY_7
-0x1d KEY_8
-0x1e KEY_9
-0x1f KEY_0
+0x16 KEY_NUMERIC_1
+0x17 KEY_NUMERIC_2
+0x18 KEY_NUMERIC_3
+0x19 KEY_NUMERIC_4
+0x1a KEY_NUMERIC_5
+0x1b KEY_NUMERIC_6
+0x1c KEY_NUMERIC_7
+0x1d KEY_NUMERIC_8
+0x1e KEY_NUMERIC_9
+0x1f KEY_NUMERIC_0
 0x21 KEY_SLEEP
 0x24 KEY_ZOOM
 0x25 KEY_LAST
@@ -36,7 +36,7 @@
 0x06 KEY_NEXTSONG
 0x03 KEY_EPG
 0x09 KEY_SETUP
-0x22 KEY_BACKSPACE
+0x22 KEY_BACK
 0x0c KEY_UP
 0x0e KEY_DOWN
 0x0b KEY_LEFT
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/videomate_tv_pvr v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/videomate_tv_pvr
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/videomate_tv_pvr	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/videomate_tv_pvr	2012-12-10 17:00:20.000000000 -0800
@@ -24,15 +24,15 @@
 0x10 KEY_CHANNELUP
 0x13 KEY_CHANNELDOWN
 0x04 KEY_RECORD
-0x16 KEY_1
-0x17 KEY_2
-0x18 KEY_3
-0x19 KEY_4
-0x1a KEY_5
-0x1b KEY_6
-0x1c KEY_7
-0x1d KEY_8
-0x1e KEY_9
-0x1f KEY_0
+0x16 KEY_NUMERIC_1
+0x17 KEY_NUMERIC_2
+0x18 KEY_NUMERIC_3
+0x19 KEY_NUMERIC_4
+0x1a KEY_NUMERIC_5
+0x1b KEY_NUMERIC_6
+0x1c KEY_NUMERIC_7
+0x1d KEY_NUMERIC_8
+0x1e KEY_NUMERIC_9
+0x1f KEY_NUMERIC_0
 0x20 KEY_LANGUAGE
 0x21 KEY_SLEEP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/vp7045 v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/vp7045
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/vp7045	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/vp7045	2012-12-10 17:00:20.000000000 -0800
@@ -1,16 +1,16 @@
 # table vp7045, type: UNKNOWN
 0x0016 KEY_POWER
 0x0010 KEY_MUTE
-0x0003 KEY_1
-0x0001 KEY_2
-0x0006 KEY_3
-0x0009 KEY_4
-0x001d KEY_5
-0x001f KEY_6
-0x000d KEY_7
-0x0019 KEY_8
-0x001b KEY_9
-0x0015 KEY_0
+0x0003 KEY_NUMERIC_1
+0x0001 KEY_NUMERIC_2
+0x0006 KEY_NUMERIC_3
+0x0009 KEY_NUMERIC_4
+0x001d KEY_NUMERIC_5
+0x001f KEY_NUMERIC_6
+0x000d KEY_NUMERIC_7
+0x0019 KEY_NUMERIC_8
+0x001b KEY_NUMERIC_9
+0x0015 KEY_NUMERIC_0
 0x0005 KEY_CHANNELUP
 0x0002 KEY_CHANNELDOWN
 0x001e KEY_VOLUMEUP
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/winfast v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/winfast
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/winfast	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/winfast	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table winfast, type: UNKNOWN
-0x12 KEY_0
-0x05 KEY_1
-0x06 KEY_2
-0x07 KEY_3
-0x09 KEY_4
-0x0a KEY_5
-0x0b KEY_6
-0x0d KEY_7
-0x0e KEY_8
-0x0f KEY_9
+0x12 KEY_NUMERIC_0
+0x05 KEY_NUMERIC_1
+0x06 KEY_NUMERIC_2
+0x07 KEY_NUMERIC_3
+0x09 KEY_NUMERIC_4
+0x0a KEY_NUMERIC_5
+0x0b KEY_NUMERIC_6
+0x0d KEY_NUMERIC_7
+0x0e KEY_NUMERIC_8
+0x0f KEY_NUMERIC_9
 0x00 KEY_POWER2
 0x1b KEY_AUDIO
 0x02 KEY_TUNER
diff -Naur v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/winfast_usbii_deluxe v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/winfast_usbii_deluxe
--- v4l-utils-0.9.3-old/utils/keytable/rc_keymaps/winfast_usbii_deluxe	2012-12-01 10:24:31.000000000 -0800
+++ v4l-utils-0.9.3-new/utils/keytable/rc_keymaps/winfast_usbii_deluxe	2012-12-10 17:00:20.000000000 -0800
@@ -1,14 +1,14 @@
 # table winfast_usbii_deluxe, type: UNKNOWN
-0x62 KEY_0
-0x75 KEY_1
-0x76 KEY_2
-0x77 KEY_3
-0x79 KEY_4
-0x7a KEY_5
-0x7b KEY_6
-0x7d KEY_7
-0x7e KEY_8
-0x7f KEY_9
+0x62 KEY_NUMERIC_0
+0x75 KEY_NUMERIC_1
+0x76 KEY_NUMERIC_2
+0x77 KEY_NUMERIC_3
+0x79 KEY_NUMERIC_4
+0x7a KEY_NUMERIC_5
+0x7b KEY_NUMERIC_6
+0x7d KEY_NUMERIC_7
+0x7e KEY_NUMERIC_8
+0x7f KEY_NUMERIC_9
 0x38 KEY_CAMERA
 0x37 KEY_RECORD
 0x35 KEY_TIME
