diff -Naur Python-2.6.5-old/configure.in Python-2.6.5-new/configure.in
--- Python-2.6.5-old/configure.in	2010-03-09 22:47:28.000000000 +0100
+++ Python-2.6.5-new/configure.in	2017-08-27 17:55:04.000000000 +0200
@@ -3725,7 +3725,7 @@
 ],
 ac_cv_working_tzset=yes,
 ac_cv_working_tzset=no,
-ac_cv_working_tzset=no)])
+ac_cv_working_tzset=yes)])
 AC_MSG_RESULT($ac_cv_working_tzset)
 if test "$ac_cv_working_tzset" = yes
 then
