diff -Naur Python-2.7.16-old/Include/Python.h Python-2.7.16-new/Include/Python.h
--- Python-2.7.16-old/Include/Python.h	2019-03-02 19:17:42.000000000 +0100
+++ Python-2.7.16-new/Include/Python.h	2019-05-18 14:28:02.129999979 +0200
@@ -1,3 +1,6 @@
+/* Whether to use feature set F.  */
+#define __GLIBC_USE(F)  __GLIBC_USE_ ## F
+
 #ifndef Py_PYTHON_H
 #define Py_PYTHON_H
 /* Since this is a "meta-include" file, no #ifdef __cplusplus / extern "C" { */
