diff -Naur pmount-0.9.23-old/src/Makefile.am pmount-0.9.23-new/src/Makefile.am
--- pmount-0.9.23-old/src/Makefile.am	2010-05-04 13:22:07.000000000 -0700
+++ pmount-0.9.23-new/src/Makefile.am	2010-06-16 15:52:33.000000000 -0700
@@ -36,8 +36,8 @@
 
 install-data-hook:
 	test -z $(INSTALL_DIR) || mkdir -p -- $(INSTALL_DIR)
-	$(INSTALL_DATA) -o root -g root -m 4755 -D $(INSTALL_SRC)/pmount $(INSTALL_DIR)/pmount
-	$(INSTALL_DATA) -o root -g root -m 4755 -D $(INSTALL_SRC)/pumount $(INSTALL_DIR)/pumount
+	$(INSTALL_DATA) -m 755 -D $(INSTALL_SRC)/pmount $(INSTALL_DIR)/pmount
+	$(INSTALL_DATA) -m 755 -D $(INSTALL_SRC)/pumount $(INSTALL_DIR)/pumount
 
 uninstall-hook:
 	rm -f $(INSTALL_DIR)/pmount
diff -Naur pmount-0.9.23-old/src/Makefile.in pmount-0.9.23-new/src/Makefile.in
--- pmount-0.9.23-old/src/Makefile.in	2010-06-15 16:23:08.000000000 -0700
+++ pmount-0.9.23-new/src/Makefile.in	2010-06-16 15:52:33.000000000 -0700
@@ -605,8 +605,8 @@
 
 install-data-hook:
 	test -z $(INSTALL_DIR) || mkdir -p -- $(INSTALL_DIR)
-	$(INSTALL_DATA) -o root -g root -m 4755 -D $(INSTALL_SRC)/pmount $(INSTALL_DIR)/pmount
-	$(INSTALL_DATA) -o root -g root -m 4755 -D $(INSTALL_SRC)/pumount $(INSTALL_DIR)/pumount
+	$(INSTALL_DATA) -m 755 -D $(INSTALL_SRC)/pmount $(INSTALL_DIR)/pmount
+	$(INSTALL_DATA) -m 755 -D $(INSTALL_SRC)/pumount $(INSTALL_DIR)/pumount
 
 uninstall-hook:
 	rm -f $(INSTALL_DIR)/pmount
