diff --speed-large-files --no-dereference --minimal -Naur linux-5.17-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6-mini.dts linux-5.17-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6-mini.dts
--- linux-5.17-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6-mini.dts	2022-01-30 14:37:07.000000000 +0100
+++ linux-5.17-rc2/arch/arm64/boot/dts/allwinner/sun50i-h6-tanix-tx6-mini.dts	2022-02-01 13:36:51.993284703 +0100
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: (GPL-2.0+ OR MIT)
-// Copyright (c) 2021 Jernej Skrabec <jernej.skrabec@gmail.com>
+// Copyright (c) 2021 Jernej Skrabec <jernej.skrabec@gmail.com>, piotr.oniszczuk@gmail.com
 
 /dts-v1/;
 
@@ -8,8 +8,40 @@
 / {
 	model = "Tanix TX6 mini";
 	compatible = "oranth,tanix-tx6-mini", "allwinner,sun50i-h6";
+
+	aliases {
+		ethernet0 = &emac;
+	};
+
 };
 
 &r_ir {
 	linux,rc-map-name = "rc-tanix-tx3mini";
 };
+
+&ac200_pwm_clk {
+	status = "okay";
+};
+
+&emac {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ext_rmii_pins>;
+	phy-mode = "rmii";
+	phy-handle = <&ext_rmii_phy>;
+	status = "okay";
+};
+
+&i2c3 {
+	status = "okay";
+};
+
+&mdio {
+	ext_rmii_phy: ethernet-phy@1 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <1>;
+	};
+};
+
+&pwm {
+	status = "okay";
+};
