diff --speed-large-files --no-dereference --minimal -Naur linux-6.2.10/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts linux-6.2.10/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts
--- linux-6.2.10/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts	2023-04-12 09:53:08.913236440 +0200
+++ linux-6.2.10/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts	2023-04-12 09:58:36.399892251 +0200
@@ -17,6 +17,7 @@
 
 	aliases {
 		ethernet0 = &emac0;
+		ethernet1 = &wlan;
 		serial0 = &uart0;
 		mmc0 = &mmc0;  /* mmc0/mmcblk0 for SD */
 		mmc1 = &mmc1;  /* mmc1/mmcblk1 for SDIO */
@@ -73,6 +74,36 @@
 		enable-active-high;
 		gpio = <&pio 2 16 GPIO_ACTIVE_HIGH>; /* PC16 */
 	};
+
+	reg_vcc33_wifi: vcc33-wifi {
+		/* Always on 3.3V regulator for WiFi and BT */
+		compatible = "regulator-fixed";
+		regulator-name = "vcc33-wifi";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+		vin-supply = <&reg_vcc5v>;
+	};
+
+	reg_vcc_wifi_io: vcc-wifi-io {
+		/* Always on 1.8V/300mA regulator for WiFi and BT IO */
+		compatible = "regulator-fixed";
+		regulator-name = "vcc-wifi-io";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-always-on;
+		vin-supply = <&reg_vcc33_wifi>;
+	};
+
+	wifi_pwrseq: wifi-pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		clocks = <&rtc CLK_OSC32K_FANOUT>;
+		clock-names = "ext_clock";
+		pinctrl-names = "default";
+		pinctrl-0 = <&clk_losc>; /* PG10 with MUX3 set */
+		reset-gpios = <&pio 6 18 GPIO_ACTIVE_LOW>; /* PG18 */
+		post-power-on-delay-ms = <200>;
+	};
 };
 
 &cpu0 {
@@ -111,6 +142,20 @@
 	status = "okay";
 };
 
+&mmc1 {
+	vmmc-supply = <&reg_vcc33_wifi>;
+	vqmmc-supply = <&reg_vcc_wifi_io>;
+	mmc-pwrseq = <&wifi_pwrseq>;
+	bus-width = <4>;
+	non-removable;
+	mmc-ddr-1_8v;
+	status = "okay";
+
+	wlan: wifi@1 {
+		reg = <1>;
+	};
+};
+
 &ohci1 {
 	status = "okay";
 };
