diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
--- linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts	2022-12-10 15:15:50.109791985 +0100
+++ linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts	2022-12-10 15:05:54.136469387 +0100
@@ -6,6 +6,7 @@
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/soc/rockchip,vop2.h>
 #include "rk3566.dtsi"
+#include <dt-bindings/leds/common.h>
 
 / {
 	model = "Pine64 RK3566 Quartz64-A Board";
@@ -72,7 +73,8 @@
 
 		led-work {
 			label = "work-led";
-			default-state = "off";
+			default-state = "on";
+			linux,default-trigger = "heartbeat";
 			gpios = <&gpio0 RK_PD3 GPIO_ACTIVE_HIGH>;
 			pinctrl-names = "default";
 			pinctrl-0 = <&work_led_enable_h>;
@@ -81,9 +83,9 @@
 
 		led-diy {
 			label = "diy-led";
-			default-state = "on";
 			gpios = <&gpio0 RK_PD4 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 			pinctrl-names = "default";
 			pinctrl-0 = <&diy_led_enable_h>;
 			retain-state-suspended;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts
--- linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2022-12-10 15:15:50.333125314 +0100
+++ linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-quartz64-b.dts	2022-12-10 15:05:59.916469285 +0100
@@ -6,6 +6,7 @@
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/soc/rockchip,vop2.h>
 #include "rk3566.dtsi"
+#include <dt-bindings/leds/common.h>
 
 / {
 	model = "Pine64 RK3566 Quartz64-B Board";
@@ -45,9 +46,9 @@
 
 		led-user {
 			label = "user-led";
-			default-state = "on";
 			gpios = <&gpio0 RK_PA0 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 			pinctrl-names = "default";
 			pinctrl-0 = <&user_led_enable_h>;
 			retain-state-suspended;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-rock-3c.dts linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-rock-3c.dts
--- linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-rock-3c.dts	2022-12-10 15:15:50.366458647 +0100
+++ linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-rock-3c.dts	2022-12-10 15:06:44.003135164 +0100
@@ -12,6 +12,7 @@
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/soc/rockchip,vop2.h>
 #include "rk3566.dtsi"
+#include <dt-bindings/leds/common.h>
 
 / {
 	model = "Radxa ROCK3 Model C";
@@ -51,9 +52,9 @@
 
 		led-user {
 			label = "user-led";
-			default-state = "on";
 			gpios = <&gpio0 RK_PA0 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 			pinctrl-names = "default";
 			pinctrl-0 = <&user_led_enable_h>;
 			retain-state-suspended;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-roc-pc.dts linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-roc-pc.dts
--- linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-roc-pc.dts	2022-12-04 23:48:12.000000000 +0100
+++ linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-roc-pc.dts	2022-12-10 15:06:13.299802373 +0100
@@ -6,6 +6,7 @@
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/soc/rockchip,vop2.h>
 #include "rk3566.dtsi"
+#include <dt-bindings/leds/common.h>
 
 / {
 	model = "Firefly Station M2";
@@ -46,7 +47,8 @@
 			label = "user-led";
 			default-state = "on";
 			gpios = <&gpio0 RK_PD3 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 			pinctrl-names = "default";
 			pinctrl-0 = <&user_led_enable_h>;
 			retain-state-suspended;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-soquartz.dtsi linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-soquartz.dtsi
--- linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-soquartz.dtsi	2022-12-04 23:48:12.000000000 +0100
+++ linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3566-soquartz.dtsi	2022-12-10 15:07:55.396467210 +0100
@@ -5,6 +5,7 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/pinctrl/rockchip.h>
 #include "rk3566.dtsi"
+#include <dt-bindings/leds/common.h>
 
 / {
 	model = "Pine64 RK3566 SoQuartz SOM";
@@ -44,8 +45,9 @@
 
 		led_work: led-work {
 			label = "work-led";
-			default-state = "off";
 			gpios = <&gpio0 RK_PC0 GPIO_ACTIVE_LOW>;
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 			pinctrl-names = "default";
 			pinctrl-0 = <&work_led_enable_h>;
 			retain-state-suspended;
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts
--- linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts	2022-12-04 23:48:12.000000000 +0100
+++ linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-evb1-v10.dts	2022-12-10 14:19:31.763186238 +0100
@@ -51,9 +51,9 @@
 
 		led_work: led-0 {
 			gpios = <&gpio0 RK_PC0 GPIO_ACTIVE_HIGH>;
-			function = LED_FUNCTION_HEARTBEAT;
 			color = <LED_COLOR_ID_BLUE>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 			pinctrl-names = "default";
 			pinctrl-0 = <&led_work_en>;
 		};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts
--- linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts	2022-12-10 15:15:50.296458648 +0100
+++ linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-rock-3a.dts	2022-12-10 14:19:31.763186238 +0100
@@ -37,9 +37,9 @@
 
 		led_user: led-0 {
 			gpios = <&gpio0 RK_PB7 GPIO_ACTIVE_HIGH>;
-			function = LED_FUNCTION_HEARTBEAT;
 			color = <LED_COLOR_ID_BLUE>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 			pinctrl-names = "default";
 			pinctrl-0 = <&led_user_en>;
 		};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-rock-3b.dts linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-rock-3b.dts
--- linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-rock-3b.dts	2022-12-10 15:15:50.186458650 +0100
+++ linux-6.1-rc8/arch/arm64/boot/dts/rockchip/rk3568-rock-3b.dts	2022-12-10 14:19:31.763186238 +0100
@@ -54,9 +54,9 @@
 
 		led_user: led-0 {
 			gpios = <&gpio0 RK_PB7 GPIO_ACTIVE_HIGH>;
-			function = LED_FUNCTION_HEARTBEAT;
 			color = <LED_COLOR_ID_BLUE>;
-			linux,default-trigger = "heartbeat";
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc0";
 			pinctrl-names = "default";
 			pinctrl-0 = <&led_user_en>;
 		};
