diff --speed-large-files --no-dereference --minimal -Naur linux-6.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts linux-6.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
--- linux-6.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2023-04-25 17:40:55.619866592 +0200
+++ linux-6.3/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts	2023-04-25 17:39:48.265047696 +0200
@@ -4,6 +4,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/pinctrl/rockchip.h>
+#include <dt-bindings/leds/common.h>
 #include "rk3588.dtsi"
 #include "rk3588-rk806-single.dtsi"
 
@@ -13,6 +14,7 @@
 
 	aliases {
 		mmc0 = &sdhci;
+		mmc1 = &sdmmc;
 		serial2 = &uart2;
 	};
 
@@ -48,6 +50,19 @@
 		regulator-max-microvolt = <1100000>;
 		vin-supply = <&vcc5v0_sys>;
  	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_user: led-0 {
+			gpio = <&gpio0 RK_PB7 GPIO_ACTIVE_HIGH>;
+			color = <LED_COLOR_ID_BLUE>;
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc1";
+			pinctrl-names = "default";
+			pinctrl-0 = <&led_user_en>;
+		};
+	};
 };
 
 &i2c0 {
@@ -145,3 +160,11 @@
 	pinctrl-0 = <&sdmmc_bus4 &sdmmc_clk &sdmmc_cmd &sdmmc_det>;
 	status = "okay";
 };
+
+&pinctrl {
+	leds {
+		led_user_en: led_user_en {
+			rockchip,pins = <3 RK_PD5 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+};
diff --speed-large-files --no-dereference --minimal -Naur linux-6.3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts linux-6.3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
--- linux-6.3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2023-04-25 17:40:55.619866592 +0200
+++ linux-6.3/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2023-04-25 17:37:05.663291167 +0200
@@ -4,6 +4,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/pinctrl/rockchip.h>
+#include <dt-bindings/leds/common.h>
 #include "rk3588s.dtsi"
 #include "rk3588-rk806-single.dtsi"
 
@@ -13,6 +14,7 @@
 
 	aliases {
 		mmc0 = &sdhci;
+		mmc1 = &sdmmc;
 		serial2 = &uart2;
 	};
 
@@ -48,6 +50,19 @@
 		regulator-max-microvolt = <1100000>;
 		vin-supply = <&vcc5v0_sys>;
 	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_user: led-0 {
+			gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
+			color = <LED_COLOR_ID_BLUE>;
+			function = LED_FUNCTION_DISK_ACTIVITY;
+			linux,default-trigger = "mmc1";
+			pinctrl-names = "default";
+			pinctrl-0 = <&led_user_en>;
+		};
+	};
 };
 
 &i2c0 {
@@ -148,6 +163,12 @@
 			rockchip,pins = <3 RK_PB7 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	leds {
+		led_user_en: led_user_en {
+			rockchip,pins = <3 RK_PD5 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
 };
 
 &sdhci {
