diff -Naur bzip2-1.0.6-old/Makefile-libbz2_so bzip2-1.0.6-new/Makefile-libbz2_so
--- bzip2-1.0.6-old/Makefile-libbz2_so	2010-09-10 16:07:52.000000000 -0700
+++ bzip2-1.0.6-new/Makefile-libbz2_so	2010-09-20 09:28:01.000000000 -0700
@@ -1,6 +1,6 @@
 
 # This Makefile builds a shared version of the library, 
-# libbz2.so.1.0.6, with soname libbz2.so.1.0,
+# libbz2.so.1.0.6, with soname libbz2.so.1,
 # at least on x86-Linux (RedHat 7.2), 
 # with gcc-2.96 20000731 (Red Hat Linux 7.1 2.96-98).  
 # Please see the README file for some important info 
@@ -37,11 +37,12 @@
 all: $(OBJS)
 	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 $(OBJS)
 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.6
-	rm -f libbz2.so.1.0
+	rm -f libbz2.so.1.0 libbz2.so.1
 	ln -s libbz2.so.1.0.6 libbz2.so.1.0
+	ln -s libbz2.so.1.0 libbz2.so.1
 
 clean: 
-	rm -f $(OBJS) bzip2.o libbz2.so.1.0.6 libbz2.so.1.0 bzip2-shared
+	rm -f $(OBJS) bzip2.o libbz2.so.1.0.6 libbz2.so.1.0 libbz2.so.1 bzip2-shared
 
 blocksort.o: blocksort.c
 	$(CC) $(CFLAGS) -c blocksort.c
