diff -Naur exiv2-0.24-old/src/Makefile exiv2-0.24-new/src/Makefile
--- exiv2-0.24-old/src/Makefile	2013-12-01 13:13:42.000000000 +0100
+++ exiv2-0.24-new/src/Makefile	2015-03-23 12:02:19.000000000 +0100
@@ -231,7 +231,7 @@
 
 $(EXIV2BIN): lib $(EXIV2OBJ) $(EXIV2COBJ)
 	-mkdir ../bin >/dev/null 2>&1
-	@$(LIBTOOL) --mode=link $(LINK.cc) -o $@ $(LIBRARY) $(EXIV2OBJ) $(EXIV2COBJ) -rpath $(libdir)
+	@$(LIBTOOL) --mode=link $(LINK.cc) -o $@ $(LIBRARY) $(EXIV2OBJ) $(EXIV2COBJ) -rpath $(DESTDIR)$(libdir)
 
 install-header:
 	$(INSTALL_DIRS) $(DESTDIR)$(incdir)
