diff -Naur exiv2.old/exiv2Targets.cmake exiv2.new/exiv2Targets.cmake
--- exiv2.old/exiv2Targets.cmake	2025-06-25 14:12:20.000000000 +0200
+++ exiv2.new/exiv2Targets.cmake	2025-06-25 14:23:45.218064584 +0200
@@ -47,7 +47,10 @@
 
 
 # The installation prefix configured by this project.
-set(_IMPORT_PREFIX "/usr")
+get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
 
 # Create imported target Exiv2::exiv2lib
 add_library(Exiv2::exiv2lib SHARED IMPORTED)
diff -Naur exiv2.old/exiv2Targets-release.cmake exiv2.new/exiv2Targets-release.cmake
--- exiv2.old/exiv2Targets-release.cmake	2025-06-25 14:12:20.000000000 +0200
+++ exiv2.new/exiv2Targets-release.cmake	2025-06-25 14:16:49.594740000 +0200
@@ -8,12 +8,12 @@
 # Import target "Exiv2::exiv2lib" for configuration "Release"
 set_property(TARGET Exiv2::exiv2lib APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
 set_target_properties(Exiv2::exiv2lib PROPERTIES
-  IMPORTED_LOCATION_RELEASE "/usr/lib/libexiv2.so.0.28.5"
+  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/libexiv2.so.0.28.5"
   IMPORTED_SONAME_RELEASE "libexiv2.so.28"
   )
 
 list(APPEND _cmake_import_check_targets Exiv2::exiv2lib )
-list(APPEND _cmake_import_check_files_for_Exiv2::exiv2lib "/usr/lib/libexiv2.so.0.28.5" )
+list(APPEND _cmake_import_check_files_for_Exiv2::exiv2lib "${_IMPORT_PREFIX}/lib/libexiv2.so.0.28.5" )
 
 # Commands beyond this point should not need to know the version.
 set(CMAKE_IMPORT_FILE_VERSION)
