diff -Naur xbmc-19.2-Matrix-old/xbmc/interfaces/swig/CMakeLists.txt xbmc-19.2-Matrix-new/xbmc/interfaces/swig/CMakeLists.txt
--- xbmc-19.2-Matrix-old/xbmc/interfaces/swig/CMakeLists.txt	2021-10-06 10:49:25.000000000 +0200
+++ xbmc-19.2-Matrix-new/xbmc/interfaces/swig/CMakeLists.txt	2021-10-19 22:10:33.872544192 +0200
@@ -29,7 +29,7 @@
   set(SOURCES ${SOURCES} "${CPP_FILE}" PARENT_SCOPE)
 endfunction()
 
-find_package(Java COMPONENTS Runtime REQUIRED)
+#find_package(Java COMPONENTS Runtime REQUIRED)
 find_package(SWIG REQUIRED)
 
 # The generated bindings
