diff -Naur mythtv-master-20160114-ga6fec7d-old/mythtv/programs/mythbackend/mainserver.cpp mythtv-master-20160114-ga6fec7d-new/mythtv/programs/mythbackend/mainserver.cpp
--- mythtv-master-20160114-ga6fec7d-old/mythtv/programs/mythbackend/mainserver.cpp	2016-01-14 16:32:23.000000000 +0100
+++ mythtv-master-20160114-ga6fec7d-new/mythtv/programs/mythbackend/mainserver.cpp	2016-01-16 16:14:47.786665376 +0100
@@ -1694,9 +1694,9 @@
         playbackList.push_back(pbs);
         lock.unlock();
 
-        LOG(VB_GENERAL, LOG_INFO, LOC + QString("MainServer::ANN %1")
+        LOG(VB_NETWORK, LOG_INFO, LOC + QString("MainServer::ANN %1")
                                       .arg(commands[1]));
-        LOG(VB_GENERAL, LOG_INFO, LOC +
+        LOG(VB_NETWORK, LOG_INFO, LOC +
             QString("adding: %1(%2) as a client (events: %3)")
                                       .arg(commands[2])
                                       .arg(quintptr(socket),0,16)
@@ -7620,7 +7620,7 @@
                 }
             }
 
-            LOG(VB_GENERAL, LOG_INFO, QString("%1 sock(%2) '%3' disconnected")
+            LOG(VB_NETWORK, LOG_INFO, QString("%1 sock(%2) '%3' disconnected")
                 .arg(pbs->getBlockShutdown() ? "Playback" : "Monitor")
                 .arg(quintptr(socket),0,16)
                 .arg(pbs->getHostname()) );
@@ -7650,7 +7650,7 @@
         MythSocket *sock = (*ft)->getSocket();
         if (sock == socket)
         {
-            LOG(VB_GENERAL, LOG_INFO, QString("FileTransfer sock(%1) disconnected")
+            LOG(VB_NETWORK, LOG_INFO, QString("FileTransfer sock(%1) disconnected")
                 .arg(quintptr(socket),0,16) );
             (*ft)->DecrRef();
             fileTransferList.erase(ft);
@@ -7662,7 +7662,7 @@
     QSet<MythSocket*>::iterator cs = controlSocketList.find(socket);
     if (cs != controlSocketList.end())
     {
-        LOG(VB_GENERAL, LOG_INFO, QString("Control sock(%1) disconnected")
+        LOG(VB_NETWORK, LOG_INFO, QString("Control sock(%1) disconnected")
             .arg(quintptr(socket),0,16) );
         (*cs)->DecrRef();
         controlSocketList.erase(cs);
