diff -Naur mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/tv_rec.cpp mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/tv_rec.cpp
--- mythtv-master-20150429-g6523186-old/mythtv/libs/libmythtv/tv_rec.cpp	2015-04-30 10:53:06.864892666 +0200
+++ mythtv-master-20150429-g6523186-new/mythtv/libs/libmythtv/tv_rec.cpp	2015-04-30 10:53:57.234688837 +0200
@@ -3961,7 +3961,7 @@
     // Start signal monitoring for devices capable of monitoring
     if (use_sm)
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "Starting Signal Monitor");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "Starting Signal Monitor");
         bool error = false;
         if (!SetupSignalMonitor(
                 !antadj, request.flags & kFlagEITScan, livetv | antadj))
@@ -4009,7 +4009,7 @@
                     signalMonitorDeadline =
                         curRecording->GetRecordingEndTime();
 
-                    LOG(VB_RECORD, LOG_DEBUG, LOC +
+                    LOG(VB_GENERAL, LOG_DEBUG, LOC +
                         QString("Start recording deadline: %1 "
                                 "Good signal deadline: %2")
                         .arg(startRecordingDeadline.toLocalTime()
@@ -4065,7 +4065,7 @@
 
     if (signalMonitor->IsAllGood())
     {
-        LOG(VB_RECORD, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
+        LOG(VB_GENERAL, LOG_INFO, LOC + "TuningSignalCheck: Good signal");
         if (curRecording && (MythDate::current() > startRecordingDeadline))
         {
             newRecStatus = RecStatus::Failing;
@@ -4151,7 +4151,7 @@
             --signalMonitorCheckCnt;
         else
         {
-            LOG(VB_RECORD, LOG_INFO, LOC +
+            LOG(VB_GENERAL, LOG_INFO, LOC +
                 QString("TuningSignalCheck: Still waiting.  Will timeout @ %1")
                 .arg(signalMonitorDeadline.toLocalTime()
                      .toString("hh:mm:ss.zzz")));
