diff -Naur mythtv-master-20160505-ged7473f-old/mythtv/libs/libmythtv/tv_play.cpp mythtv-master-20160505-ged7473f-new/mythtv/libs/libmythtv/tv_play.cpp
--- mythtv-master-20160505-ged7473f-old/mythtv/libs/libmythtv/tv_play.cpp	2016-05-05 10:00:14.000000000 +0200
+++ mythtv-master-20160505-ged7473f-new/mythtv/libs/libmythtv/tv_play.cpp	2016-05-11 16:45:59.573333323 +0200
@@ -1274,6 +1274,11 @@
         {
             int xbase, width, ybase, height;
             GetMythUI()->GetScreenBounds(xbase, ybase, width, height);
+
+            // Hack for ticket12311
+            xbase = xbase - 1;
+            ybase = ybase - 1;
+
             player_bounds = QRect(xbase, ybase, width, height);
         }
 
@@ -1294,6 +1299,16 @@
                 // big in fullscreen mode
                 if (fullscreen)
                 {
+                    // Hack for ticket12311
+                    if ((maxHeight >= 1072) && (maxHeight <= 1200))
+                    { 
+                        LOG(VB_GENERAL, LOG_DEBUG, LOC + "Hack for ticket12311: Re-Adjusted Screen size. Original: "
+                        + QString("%1x").arg(maxWidth)
+                        + QString("%1. New: ").arg(maxHeight)
+                        + QString("%1x1080").arg(maxWidth));
+                        maxHeight = 1080;
+                    }
+
                     player_bounds.setSize(QSize(maxWidth, maxHeight));
 
                     // resize possibly avoids a bug on some systems
diff -Naur mythtv-master-20160505-ged7473f-old/mythtv/libs/libmythui/mythuihelper.cpp mythtv-master-20160505-ged7473f-new/mythtv/libs/libmythui/mythuihelper.cpp
--- mythtv-master-20160505-ged7473f-old/mythtv/libs/libmythui/mythuihelper.cpp	2016-05-05 10:00:14.000000000 +0200
+++ mythtv-master-20160505-ged7473f-new/mythtv/libs/libmythui/mythuihelper.cpp	2016-05-11 16:46:43.770000001 +0200
@@ -318,8 +318,8 @@
         GetMythDB()->OverrideSettingForSession("GuiHeight", QString::number(h_override));
     }
 
-    m_screenxbase  = GetMythDB()->GetNumSetting("GuiOffsetX");
-    m_screenybase  = GetMythDB()->GetNumSetting("GuiOffsetY");
+    m_screenxbase  = GetMythDB()->GetNumSetting("GuiOffsetX") - 1;
+    m_screenybase  = GetMythDB()->GetNumSetting("GuiOffsetY") - 1;
 
     m_screenwidth = m_screenheight = 0;
     GetMythDB()->GetResolutionSetting("Gui", m_screenwidth, m_screenheight);
