diff -Naur mythtv-master-20151221-ga0ea915-old/mythtv/programs/mythfrontend/proglist.cpp mythtv-master-20151221-ga0ea915-new/mythtv/programs/mythfrontend/proglist.cpp
--- mythtv-master-20151221-ga0ea915-old/mythtv/programs/mythfrontend/proglist.cpp	2015-12-21 11:41:54.000000000 +0100
+++ mythtv-master-20151221-ga0ea915-new/mythtv/programs/mythfrontend/proglist.cpp	2015-12-21 19:15:26.909527824 +0100
@@ -1568,7 +1568,7 @@
     ProgramList::const_iterator it = m_itemList.begin();
     for (; it != m_itemList.end(); ++it)
         new MythUIButtonListItem(m_progList, "", qVariantFromValue(*it));
-    m_progList->LoadInBackground();
+    //m_progList->LoadInBackground();
 
     if (m_positionText)
     {
