diff -Naur libva-vdpau-driver-0.7.4-old/src/Makefile.am libva-vdpau-driver-0.7.4-new/src/Makefile.am
--- libva-vdpau-driver-0.7.4-old/src/Makefile.am	2011-12-06 02:07:06.000000000 -0800
+++ libva-vdpau-driver-0.7.4-new/src/Makefile.am	2012-10-06 07:08:29.000000000 -0700
@@ -75,6 +75,7 @@
 install-data-hook:
 	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
 	for drv in $(DRIVERS); do				\
+	    rm -f $${drv}_drv_video.so;                         \
 	    ln -s vdpau_drv_video.so $${drv}_drv_video.so;	\
 	done
 
diff -Naur libva-vdpau-driver-0.7.4-old/src/Makefile.in libva-vdpau-driver-0.7.4-new/src/Makefile.in
--- libva-vdpau-driver-0.7.4-old/src/Makefile.in	2012-10-05 08:03:17.000000000 -0700
+++ libva-vdpau-driver-0.7.4-new/src/Makefile.in	2012-10-06 07:08:29.000000000 -0700
@@ -687,6 +687,7 @@
 install-data-hook:
 	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
 	for drv in $(DRIVERS); do				\
+	    rm -f $${drv}_drv_video.so;                         \
 	    ln -s vdpau_drv_video.so $${drv}_drv_video.so;	\
 	done
 
