diff -Naur wlroots-0.12.0-old/meson.build wlroots-0.12.0-new/meson.build
--- wlroots-0.12.0-old/meson.build	2020-11-08 15:01:44.000000000 +0100
+++ wlroots-0.12.0-new/meson.build	2021-01-28 20:54:07.996666123 +0100
@@ -26,7 +26,6 @@
 add_project_arguments(cc.get_supported_arguments([
 	'-Wundef',
 	'-Wlogical-op',
-	'-Wmissing-include-dirs',
 	'-Wold-style-definition',
 	'-Wpointer-arith',
 	'-Winit-self',
