diff --speed-large-files --no-dereference --minimal -Naur waylandpp-1.0.0/include/wayland-client.hpp waylandpp-1.0.0/include/wayland-client.hpp
--- waylandpp-1.0.0/include/wayland-client.hpp	2022-04-25 18:04:33.000000000 +0200
+++ waylandpp-1.0.0/include/wayland-client.hpp	2023-05-10 15:29:16.909887177 +0200
@@ -29,6 +29,7 @@
 /** \file */
 
 #include <atomic>
+#include <cstdint>
 #include <functional>
 #include <memory>
 #include <string>
diff --speed-large-files --no-dereference --minimal -Naur waylandpp-1.0.0/scanner/scanner.cpp waylandpp-1.0.0/scanner/scanner.cpp
--- waylandpp-1.0.0/scanner/scanner.cpp	2022-04-25 18:04:33.000000000 +0200
+++ waylandpp-1.0.0/scanner/scanner.cpp	2023-05-10 15:30:52.236551010 +0200
@@ -15,6 +15,7 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <cstdint>
 #include <fstream>
 #include <iostream>
 #include <list>
@@ -1106,6 +1107,7 @@
   wayland_hpp << "#pragma once" << std::endl
               << std::endl
               << "#include <array>" << std::endl
+              << "#include <cstdint>" << std::endl
               << "#include <functional>" << std::endl
               << "#include <memory>" << std::endl
               << "#include <string>" << std::endl
@@ -1125,6 +1127,7 @@
     wayland_server_hpp << "#pragma once" << std::endl
                        << std::endl
                        << "#include <array>" << std::endl
+                       << "#include <cstdint>" << std::endl
                        << "#include <functional>" << std::endl
                        << "#include <memory>" << std::endl
                        << "#include <string>" << std::endl
