diff --speed-large-files --no-dereference --minimal -Naur waylandpp-1.0.0/CMakeLists.txt waylandpp-1.0.0/CMakeLists.txt
--- waylandpp-1.0.0/CMakeLists.txt	2022-04-25 18:04:33.000000000 +0200
+++ waylandpp-1.0.0/CMakeLists.txt	2023-12-18 15:23:15.823398388 +0100
@@ -29,7 +29,7 @@
 include(GNUInstallDirs)
 include(CMakeDependentOption)
 include(CMakePackageConfigHelpers)
-find_package(Doxygen)
+# find_package(Doxygen)
 
 # version information
 configure_file(include/wayland-version.hpp.in wayland-version.hpp @ONLY)
