diff -Naur mythtv-old/mythplugins/mythweb/modules/stream/stream_flv.pl mythtv-new/mythplugins/mythweb/modules/stream/stream_flv.pl
--- mythtv-old/mythplugins/mythweb/modules/stream/stream_flv.pl	2018-02-27 12:29:33.000000000 +0100
+++ mythtv-new/mythplugins/mythweb/modules/stream/stream_flv.pl	2018-02-27 14:10:34.533333302 +0100
@@ -133,7 +133,7 @@
                         .' -ar 44100'
                         .' -ab '.shell_escape("${abitrate}k")
                         .' -b '.shell_escape("${vbitrate}k")
-                        .' /dev/stdout 2>/dev/null |';
+                        .' /dev/stdout 2>/var/log/mythweb-streaming-flv.log |';
 
 # Print the movie
     $ffmpeg_pid = open(DATA, $ffmpeg_command);
