diff -Naur mythtv-master-20200311-g625fc71-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c mythtv-master-20200311-g625fc71-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c
--- mythtv-master-20200311-g625fc71-old/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-03-14 13:30:22.146666668 +0100
+++ mythtv-master-20200311-g625fc71-new/mythtv/external/FFmpeg/libavutil/hwcontext_drm.c	2020-03-14 13:30:23.853333334 +0100
@@ -43,6 +43,11 @@
     AVDRMDeviceContext *hwctx = hwdev->hwctx;
     drmVersionPtr version;
 
+    if (device == NULL) {
+      hwctx->fd = -1;
+      return 0;
+    }
+
     hwctx->fd = open(device, O_RDWR);
     if (hwctx->fd < 0)
         return AVERROR(errno);
