diff -Naur mythtv-master-20200311-g625fc71-old/mythtv/configure mythtv-master-20200311-g625fc71-new/mythtv/configure
--- mythtv-master-20200311-g625fc71-old/mythtv/configure	2020-03-14 13:30:58.756666667 +0100
+++ mythtv-master-20200311-g625fc71-new/mythtv/configure	2020-03-14 13:31:01.823333334 +0100
@@ -5189,7 +5189,7 @@
 }
 
 # Minimum supported Qt version
-QT_MIN_VERSION_STR="5.7"
+QT_MIN_VERSION_STR="4.8"
 QT_MIN_VERSION=$(version2num $QT_MIN_VERSION_STR)
 QT_MIN_VERSION_HEX=$(printf "0x%06x" $QT_MIN_VERSION)
 
