diff -Naur mythtv-master-20200207-g2869463-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20200207-g2869463-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20200207-g2869463-old/mythtv/libs/libmythtv/osd.cpp	2020-02-08 14:19:38.073333335 +0100
+++ mythtv-master-20200207-g2869463-new/mythtv/libs/libmythtv/osd.cpp	2020-02-08 14:19:39.616666668 +0100
@@ -528,7 +528,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
