diff -Naur mythtv-master-20200311-g625fc71-old/mythtv/libs/libmythtv/osd.cpp mythtv-master-20200311-g625fc71-new/mythtv/libs/libmythtv/osd.cpp
--- mythtv-master-20200311-g625fc71-old/mythtv/libs/libmythtv/osd.cpp	2020-03-14 13:31:05.793333334 +0100
+++ mythtv-master-20200311-g625fc71-new/mythtv/libs/libmythtv/osd.cpp	2020-03-14 13:31:08.496666667 +0100
@@ -528,7 +528,7 @@
             else
                 bar->SetUsed(0);
         }
-        bar->SetVisible(startts > 0);
+        bar->SetVisible(startts > 0 && endts >= nowts && startts <= nowts);
         bar->SetStart(0);
         bar->SetTotal(1000);
     }
