diff --speed-large-files --no-dereference --minimal -Naur nss-3.89.1/nss/coreconf/Linux.mk nss-3.89.1/nss/coreconf/Linux.mk
--- nss-3.89.1/nss/coreconf/Linux.mk	2023-05-05 19:06:27.000000000 +0200
+++ nss-3.89.1/nss/coreconf/Linux.mk	2023-05-17 14:25:51.846281063 +0200
@@ -3,10 +3,10 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-CC     ?= gcc
-CCC    ?= g++
-RANLIB ?= ranlib
-LD     ?= ld
+CC     ?= $(CC)
+CCC    ?= $(CXX)
+RANLIB ?= $(RANLIB)
+LD     ?= ld
 
 include $(CORE_DEPTH)/coreconf/UNIX.mk
 
