diff -Naur nss-3.38-old/nss/coreconf/Linux.mk nss-3.38-new/nss/coreconf/Linux.mk
--- nss-3.38-old/nss/coreconf/Linux.mk	2018-06-21 11:24:45.000000000 +0200
+++ nss-3.38-new/nss/coreconf/Linux.mk	2018-06-28 12:20:03.000000000 +0200
@@ -3,9 +3,9 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-CC     ?= gcc
-CCC    ?= g++
-RANLIB ?= ranlib
+CC     ?= $(CC)
+CCC    ?= $(CXX)
+RANLIB ?= $(RANLIB)
 
 include $(CORE_DEPTH)/coreconf/UNIX.mk
 
