diff --speed-large-files --no-dereference --minimal -Naur nss-3.122/nss/cmd/shlibsign/sign.sh nss-3.122/nss/cmd/shlibsign/sign.sh
--- nss-3.122/nss/cmd/shlibsign/sign.sh	2026-03-19 17:58:13.000000000 +0100
+++ nss-3.122/nss/cmd/shlibsign/sign.sh	2026-04-16 09:17:00.500952686 +0200
@@ -33,19 +33,7 @@
     "${2}"/shlibsign -v -C -i "${5}"
     ;;
 *)
-    LIBPATH=`(cd "${1}"/lib; pwd)`:`(cd "${4}"; pwd)`:$LIBPATH
-    export LIBPATH
-    SHLIB_PATH=${1}/lib:${4}:$SHLIB_PATH
-    export SHLIB_PATH
-    LD_LIBRARY_PATH=${1}/lib:${4}:$LD_LIBRARY_PATH
-    export LD_LIBRARY_PATH
-    DYLD_LIBRARY_PATH=${1}/lib:${4}:$DYLD_LIBRARY_PATH
-    export DYLD_LIBRARY_PATH
-    LIBRARY_PATH=${1}/lib:${4}:$LIBRARY_PATH
-    export LIBRARY_PATH
-    ADDON_PATH=${1}/lib:${4}:$ADDON_PATH
-    export ADDON_PATH
-    echo "${2}"/shlibsign -C -v -i "${5}"
-    "${2}"/shlibsign -C -v -i "${5}"
+    echo nss-shlibsign -v -i "${5}"
+    nss-shlibsign -v -i "${5}"
     ;;
 esac
