diff -Naur nss-3.14.1-old/mozilla/security/nss/lib/Makefile nss-3.14.1-new/mozilla/security/nss/lib/Makefile
--- nss-3.14.1-old/mozilla/security/nss/lib/Makefile	2012-04-25 07:49:24.000000000 -0700
+++ nss-3.14.1-new/mozilla/security/nss/lib/Makefile	2013-01-04 08:52:28.000000000 -0800
@@ -42,6 +42,22 @@
 endif
 endif
 
+ifeq ($(OS_TARGET),Linux)
+# If FREEBL_NO_DEPEND has not yet been defined,
+# we'll default to FREEBL_NO_DEPEND=1.
+# This can be overridden by defining FREEBL_NO_DEPEND=0
+ifndef FREEBL_NO_DEPEND
+FREEBL_NO_DEPEND := 1
+export FREEBL_NO_DEPEND
+else
+ifeq ($(FREEBL_NO_DEPEND), 0)
+then
+FREEBL_NO_DEPEND := 
+unexport FREEBL_NO_DEPEND
+endif
+endif
+endif
+
 #######################################################################
 # (5) Execute "global" rules. (OPTIONAL)                              #
 #######################################################################
