diff --speed-large-files --no-dereference --minimal -Naur splashutils-1.5.4.4/src/libfbsplash.c splashutils-1.5.4.4/src/libfbsplash.c
--- splashutils-1.5.4.4/src/libfbsplash.c	2011-06-12 23:43:39.000000000 +0200
+++ splashutils-1.5.4.4/src/libfbsplash.c	2022-08-05 22:16:55.616664499 +0200
@@ -23,8 +23,6 @@
 #include <sys/ioctl.h>
 #include <sys/mount.h>
 
-#include <linux/fs.h>
-
 #if !defined(MNT_DETACH)
 	#define MNT_DETACH 2
 #endif
