diff -Naur splashutils-1.5.4.4-old/configure.ac splashutils-1.5.4.4-new/configure.ac
--- splashutils-1.5.4.4-old/configure.ac	2011-06-12 14:43:39.000000000 -0700
+++ splashutils-1.5.4.4-new/configure.ac	2011-11-03 06:56:07.000000000 -0700
@@ -267,7 +267,7 @@
   AS_HELP_STRING([--with-freetype2-src=DIR], [freetype2 source to use when linking with klibc]),
   [
 	AS_IF(
-	  [test "`echo ${withval} | sed -e 's,^/.*,,'`" = "${withval}"],
+	  [test "x`AS_ECHO([${withval}]) | $SED -e 's,^/.*,,'`" = "x${withval}"],
 	  [LIBFREETYPE2_SOURCE="\$(abs_top_srcdir)/${withval}"],
 	  [LIBFREETYPE2_SOURCE="${withval}"]
 	)
@@ -280,7 +280,7 @@
   AS_HELP_STRING([--with-jpeg-src=DIR], [jpeglib source to use when linking with klibc]),
   [
 	AS_IF(
-	  [test "`echo ${withval} | sed -e 's,^/.*,,'`" = "${withval}"],
+	  [test "x`AS_ECHO([${withval}]) | $SED -e 's,^/.*,,'`" = "x${withval}"],
 	  [LIBJPEG_SOURCE="\$(abs_top_srcdir)/${withval}"],
 	  [LIBJPEG_SOURCE="${withval}"]
 	)
@@ -293,7 +293,7 @@
   AS_HELP_STRING([--with-lpng-src=DIR], [libpng source to use when linking with klibc]),
   [
 	AS_IF(
-	  [test "`echo ${withval} | sed -e 's,^/.*,,'`" = "${withval}"],
+	  [test "x`AS_ECHO([${withval}]) | $SED -e 's,^/.*,,'`" = "x${withval}"],
 	  [LIBPNG_SOURCE="\$(abs_top_srcdir)/${withval}"],
 	  [LIBPNG_SOURCE="${withval}"]
 	)
@@ -306,7 +306,7 @@
   AS_HELP_STRING([--with-zlib-src=DIR], [zlib source to use when linking with klibc]),
   [
 	AS_IF(
-	  [test "`echo ${withval} | sed -e 's,^/.*,,'`" = "${withval}"],
+	  [test "x`AS_ECHO([${withval}]) | $SED -e 's,^/.*,,'`" = "x${withval}"],
 	  [LIBZ_SOURCE="\$(abs_top_srcdir)/${withval}"],
 	  [LIBZ_SOURCE="${withval}"]
 	)
@@ -317,7 +317,7 @@
 
 AC_ARG_WITH([essential-prefix],
   AS_HELP_STRING([--with-essential-prefix=ESPREFIX], [install essential files in ESPREFIX @<:@/@:>@]),
-  [essential_prefix="$(echo ${withval} | sed -e 's#/$##')"],
+  [essential_prefix=`AS_ECHO([${withval}]) | $SED -e 's#/$##')`],
   [essential_prefix="/"]
 )
 AC_SUBST([essential_prefix])
@@ -448,7 +448,7 @@
   ]
 )
 AS_IF(
-  [test "x${JPEG_CFLAGS}" = "no"],
+  [test "x${JPEG_CFLAGS}" = "xno"],
   [JPEG_CFLAGS=""]
 )
 AS_IF(
@@ -461,7 +461,7 @@
   ]
 )
 AS_IF(
-  [test "x${JPEG_LIBS}" = "no"],
+  [test "x${JPEG_LIBS}" = "xno"],
   [JPEG_LIBS=""]
 )
 AC_SUBST([JPEG_CFLAGS])
@@ -484,7 +484,7 @@
       ]
     )
     AS_IF(
-      [test "x${GPM_CFLAGS}" = "no"],
+      [test "x${GPM_CFLAGS}" = "xno"],
       [GPM_CFLAGS=""]
     )
     AS_IF(
@@ -498,7 +498,7 @@
       ]
     )
     AS_IF(
-      [test "x${GPM_LIBS}" = "no"],
+      [test "x${GPM_LIBS}" = "xno"],
       [GPM_LIBS=""]
     )
   ]
