diff -Naur splashutils-1.5.4.4-old/src/common.h splashutils-1.5.4.4-new/src/common.h
--- splashutils-1.5.4.4-old/src/common.h	2011-06-12 23:43:39.000000000 +0200
+++ splashutils-1.5.4.4-new/src/common.h	2020-04-01 08:59:04.317707014 +0200
@@ -6,6 +6,7 @@
 #include <stdio.h>
 #include <stdbool.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <linux/kd.h>
 #include <linux/tty.h>
 #include <linux/input.h>
