diff -Naur parted-3.2-old/doc/pt_BR/Makefile.in parted-3.2-new/doc/pt_BR/Makefile.in
--- parted-3.2-old/doc/pt_BR/Makefile.in	2014-07-29 05:02:56.000000000 +0200
+++ parted-3.2-new/doc/pt_BR/Makefile.in	2016-02-28 20:50:29.000000000 +0100
@@ -1043,7 +1043,7 @@
 lispdir = @lispdir@
 localedir = @localedir@
 localstatedir = @localstatedir@
-mandir = $(mandir)/$(lang)
+mandir = @mandir@/$(lang)
 mkdir_p = @mkdir_p@
 oldincludedir = @oldincludedir@
 pdfdir = @pdfdir@
